From d8dba9141d985b65fd947ad1407f0adbb97c10b4 Mon Sep 17 00:00:00 2001
From: miwu04 <me@alanlin.icu>
Date: Thu, 6 Aug 2026 22:55:45 +0800
Subject: [PATCH 4/5] chore(pnpm): relax supportedArchitectures.cpu to current

---
 pnpm-workspace.yaml | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/pnpm-workspace.yaml b/pnpm-workspace.yaml
index 9eff6408..87bb5df8 100644
--- a/pnpm-workspace.yaml
+++ b/pnpm-workspace.yaml
@@ -17,5 +17,4 @@ supportedArchitectures:
   os:
     - current
   cpu:
-    - x64
-    - arm64
+    - current
-- 
2.55.0

