From 86f62eb349d62df21f827f9a17c5150297e8e8ac Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 6 May 2024 23:41:59 -0700
Subject: [PATCH 05/55] Fedora qt prefer_adwaita_on_gnome

---
 src/gui/kernel/qguiplatformplugin.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/gui/kernel/qguiplatformplugin.cpp b/src/gui/kernel/qguiplatformplugin.cpp
index 1b1a863e..439cd5ed 100644
--- a/src/gui/kernel/qguiplatformplugin.cpp
+++ b/src/gui/kernel/qguiplatformplugin.cpp
@@ -165,8 +165,13 @@ QString QGuiPlatformPlugin::styleName()
     case DE_GNOME: {
         QStringList availableStyles = QStyleFactory::keys();
         // Set QGtkStyle for GNOME if available
+        QString adwaitaStyleKey = QString::fromLatin1("adwaita");
         QString gtkStyleKey = QString::fromLatin1("GTK+");
-        if (availableStyles.contains(gtkStyleKey)) {
+        if (availableStyles.contains(adwaitaStyleKey)) {
+            stylename = adwaitaStyleKey;
+            break;
+        }
+        else if (availableStyles.contains(gtkStyleKey)) {
             stylename = gtkStyleKey;
             break;
         }
-- 
2.45.0

