From 2e19d2d5d56dfe501b0376a95664b661f4495d52 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 6 May 2024 23:42:03 -0700
Subject: [PATCH 19/55] Fedora qt everywhere opensource src 4.8.7 qmake_LFLAGS

---
 qmake/Makefile.unix | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qmake/Makefile.unix b/qmake/Makefile.unix
index ee9b4bd7..be0ea5f8 100644
--- a/qmake/Makefile.unix
+++ b/qmake/Makefile.unix
@@ -3,7 +3,7 @@ BUILD_PATH = @BUILD_PATH@
 QTOBJS = @QMAKE_QTOBJS@
 QTSRCS = @QMAKE_QTSRCS@
 QMAKESPEC = @QMAKESPEC@
-LFLAGS = @QMAKE_LFLAGS@
+LFLAGS = @QMAKE_LFLAGS@ $(QMAKE_LFLAGS_RELEASE)
 
 #qmake code
 OBJS=project.o property.o main.o makefile.o unixmake2.o unixmake.o  \
-- 
2.45.0

