From dfe09b1e34a866a69ed61195d949288d2e08573a Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 6 May 2024 23:42:04 -0700
Subject: [PATCH 24/55] Fedora qt everywhere opensource src 4.8.0 QTBUG 22037

---
 src/corelib/tools/qlist.h | 30 +++++++++++-------------------
 1 file changed, 11 insertions(+), 19 deletions(-)

diff --git a/src/corelib/tools/qlist.h b/src/corelib/tools/qlist.h
index fab6a8cf..c6c3a6a0 100644
--- a/src/corelib/tools/qlist.h
+++ b/src/corelib/tools/qlist.h
@@ -769,26 +769,18 @@ Q_OUTOFLINE_TEMPLATE void QList<T>::clear()
 template <typename T>
 Q_OUTOFLINE_TEMPLATE int QList<T>::removeAll(const T &_t)
 {
-    int index = indexOf(_t);
-    if (index == -1)
-        return 0;
-
+    detachShared();
     const T t = _t;
-    detach();
-
-    Node *i = reinterpret_cast<Node *>(p.at(index));
-    Node *e = reinterpret_cast<Node *>(p.end());
-    Node *n = i;
-    node_destruct(i);
-    while (++i != e) {
-        if (i->t() == t)
-            node_destruct(i);
-        else
-            *n++ = *i;
-    }
-
-    int removedCount = e - n;
-    d->end -= removedCount;
+    int removedCount=0, i=0;
+    Node *n;
+    while (i < p.size())
+        if ((n = reinterpret_cast<Node *>(p.at(i)))->t() == t) {
+            node_destruct(n);
+            p.remove(i);
+            ++removedCount;
+        } else {
+            ++i;
+        }
     return removedCount;
 }
 
-- 
2.45.0

