From ac98e18018f21893c14f144fb80e443a0e580dfb Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 6 May 2024 23:42:05 -0700
Subject: [PATCH 28/55] Fedora qt everywhere opensource src 4.8.3 icu_no_debug

---
 src/corelib/tools/qlocale_icu.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/corelib/tools/qlocale_icu.cpp b/src/corelib/tools/qlocale_icu.cpp
index 0b2ac2c8..7dd75ea7 100644
--- a/src/corelib/tools/qlocale_icu.cpp
+++ b/src/corelib/tools/qlocale_icu.cpp
@@ -88,7 +88,9 @@ bool qt_initIcu(const QString &localeString)
         QLibrary lib(QLatin1String("icui18n"), QLatin1String(U_ICU_VERSION_SHORT));
         lib.setLoadHints(QLibrary::ImprovedSearchHeuristics);
         if (!lib.load()) {
+#ifndef QT_NO_DEBUG
             qWarning() << "Unable to load library icui18n" << lib.errorString();
+#endif
             status = ErrorLoading;
             return false;
         }
@@ -118,7 +120,9 @@ bool qt_initIcu(const QString &localeString)
         QLibrary ucLib(QLatin1String("icuuc"), QLatin1String(U_ICU_VERSION_SHORT));
         ucLib.setLoadHints(QLibrary::ImprovedSearchHeuristics);
         if (!ucLib.load()) {
+#ifndef QT_NO_DEBUG
             qWarning() << "Unable to load library icuuc" << ucLib.errorString();
+#endif
             status = ErrorLoading;
             return false;
         }
-- 
2.45.0

