From 357190c828b72da08c5712758618c0da71847ae4 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 6 May 2024 23:42:06 -0700
Subject: [PATCH 32/55] Fedora qt everywhere opensource src 4.8.5 QTBUG 35459

---
 src/xml/sax/qxml.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/xml/sax/qxml.cpp b/src/xml/sax/qxml.cpp
index ff717766..eb098c63 100644
--- a/src/xml/sax/qxml.cpp
+++ b/src/xml/sax/qxml.cpp
@@ -433,7 +433,7 @@ private:
     // for the DTD currently being parsed.
     static const int dtdRecursionLimit = 2;
     // The maximum amount of characters an entity value may contain, after expansion.
-    static const int entityCharacterLimit = 1024;
+    static const int entityCharacterLimit = 4096;
 
     const QString &string();
     void stringClear();
-- 
2.45.0

