From a904b80a33b413758391778d5a6060a39c1fc4c8 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 6 May 2024 23:42:07 -0700
Subject: [PATCH 37/55] Fedora qt everywhere opensource src 4.8.7 mips64

---
 configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure b/configure
index 9162af63..f8f49c25 100755
--- a/configure
+++ b/configure
@@ -3331,6 +3331,9 @@ arm*)
     CFG_ARCH=arm
     COMPAT_ARCH=armv6
     ;;
+mips*)
+    CFG_ARCH=mips
+    ;;
 esac
 
 if [ -d "$relpath/src/corelib/arch/$CFG_ARCH" ]; then
-- 
2.45.0

