From d2695ce55f28417aa6d787402f74a629bac51472 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 6 May 2024 23:42:08 -0700
Subject: [PATCH 41/55] Fedora qt everywhere opensource src 4.8.7 icu59

---
 src/corelib/tools/tools.pri | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/corelib/tools/tools.pri b/src/corelib/tools/tools.pri
index 1a7153ad..8d372b4a 100644
--- a/src/corelib/tools/tools.pri
+++ b/src/corelib/tools/tools.pri
@@ -102,7 +102,15 @@ contains(QT_CONFIG, zlib):include($$PWD/../../3rdparty/zlib.pri)
 else:include($$PWD/../../3rdparty/zlib_dependency.pri)
 
 contains(QT_CONFIG,icu) {
-    SOURCES += tools/qlocale_icu.cpp
+    cpp11.name = cpp11
+    cpp11.input = SOURCES_CPP11
+    cpp11.dependency_type = TYPE_C
+    cpp11.variable_out = OBJECTS
+    cpp11.output = ${QMAKE_VAR_OBJECTS_DIR}${QMAKE_FILE_IN_BASE}$${first(QMAKE_EXT_OBJ)}
+    cpp11.commands = $${QMAKE_CXX} $(CXXFLAGS) -std=c++11 $(INCPATH) -c ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
+    QMAKE_EXTRA_COMPILERS += cpp11
+
+    SOURCES_CPP11 += tools/qlocale_icu.cpp
     DEFINES += QT_USE_ICU
 }
 
-- 
2.45.0

