From 8075c14bd9985cdf91acfc93bb49f4a27c1a8a46 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 6 May 2024 23:42:08 -0700
Subject: [PATCH 42/55] Fedora qt everywhere opensource src 4.8.7 gcc8_qtscript

---
 src/script/script.pro | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/script/script.pro b/src/script/script.pro
index d9472764..fb8cabf1 100644
--- a/src/script/script.pro
+++ b/src/script/script.pro
@@ -91,6 +91,9 @@ symbian {
     TARGET.UID3=0x2001B2E1
 }
 
+# hack around gcc8 optimization bug with -O2
+QMAKE_CXXFLAGS_RELEASE += -O1
+
 symbian {
     symbian-abld|symbian-sbsv2 {
         MMP_RULES += ALWAYS_BUILD_AS_ARM
-- 
2.45.0

