From b534da2819c6f72ad66e07b91d4b63395eeb359d Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 6 May 2024 23:42:09 -0700
Subject: [PATCH 44/55] Fedora qt everywhere opensource src 4.8.7 hardcode
 buildkey

---
 configure | 26 +-------------------------
 1 file changed, 1 insertion(+), 25 deletions(-)

diff --git a/configure b/configure
index f8f49c25..ecd3ef28 100755
--- a/configure
+++ b/configure
@@ -7742,18 +7742,6 @@ case "$XPLATFORM" in
 	canBuildQtXmlPatterns="no"
 	;;
     *-g++*)
-	# Check gcc's version
-	case "$(${QMAKE_CONF_COMPILER} -dumpversion)" in
-	    5*|4*|3.4*)
-		;;
-            3.3*)
-                canBuildWebKit="no"
-                ;;
-	    *)
-		canBuildWebKit="no"
-		canBuildQtXmlPatterns="no"
-		;;
-	esac
 	;;
     solaris-cc*)
         # Check the compiler version
@@ -8053,19 +8041,7 @@ g++*)
         ;;
     esac
 
-    case "$COMPILER_VERSION" in
-    2.95.*)
-        COMPILER_VERSION="2.95.*"
-        ;;
-    3.*)
-        COMPILER_VERSION="3.*"
-        ;;
-    5*|4.*)
-        COMPILER_VERSION="4"
-        ;;
-    *)
-        ;;
-    esac
+    COMPILER_VERSION="4"
     [ '!' -z "$COMPILER_VERSION" ] && COMPILER="g++-${COMPILER_VERSION}"
     ;;
 icc*)
-- 
2.45.0

