From e55c0345a603c1ad1b476da4b0de2dde71b740ec Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 6 May 2024 23:42:09 -0700
Subject: [PATCH 47/55] Fedora qt everywhere opensource src 4.8.6 QTBUG 22829

---
 src/tools/moc/main.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/tools/moc/main.cpp b/src/tools/moc/main.cpp
index 147e3f5a..3139983f 100644
--- a/src/tools/moc/main.cpp
+++ b/src/tools/moc/main.cpp
@@ -188,8 +188,12 @@ int runMoc(int _argc, char **_argv)
     pp.macros["Q_MOC_RUN"];
     pp.macros["__cplusplus"];
 
-    // Workaround a bug while parsing the boost/type_traits/has_operator.hpp header. See QTBUG-22829
+    // Workaround a bugs while parsing some boost headers. See QTBUG-22829 
     pp.macros["BOOST_TT_HAS_OPERATOR_HPP_INCLUDED"];
+    pp.macros["BOOST_LEXICAL_CAST_INCLUDED"];
+    pp.macros["BOOST_NEXT_PRIOR_HPP_INCLUDED"];
+    pp.macros["BOOST_TYPE_TRAITS_HPP"];
+    pp.macros["_SYS_SYSMACROS_H_OUTER"];
 
     QByteArray filename;
     QByteArray output;
-- 
2.45.0

