From 3f88c059c7b57800bfd75c0f21a7c337c5148820 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 6 May 2024 23:42:11 -0700
Subject: [PATCH 54/55] Debian riscv64 support

---
 .../webkit/Source/JavaScriptCore/wtf/Platform.h       | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h b/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h
index 635df39e..ea7251e8 100644
--- a/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h
+++ b/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h
@@ -377,8 +377,15 @@
 #endif
 #endif
 
+/* CPU(RISCV64) - RISC-V 64 */
+#if defined(__riscv) && (__riscv_xlen == 64)
+#define WTF_CPU_RISCV64 1
+#define ENABLE_JIT 0
+#define ENABLE_YARR_JIT 0
+#endif
+
 /* Not sure about this one, qt5-qtwebkit doesn't include it -- rex */
-#if CPU(ARM) || CPU(MIPS) || CPU(SH4) || CPU(AARCH64)
+#if CPU(ARM) || CPU(MIPS) || CPU(SH4) || CPU(AARCH64) || CPU(RISCV64)
 #define WTF_CPU_NEEDS_ALIGNED_ACCESS 1
 #endif
 
@@ -1012,7 +1019,7 @@
     || CPU(SPARC64) \
     || CPU(S390X) \
     || CPU(PPC64) \
-    || CPU(MIPS64) || CPU(AARCH64)
+    || CPU(MIPS64) || CPU(AARCH64) || CPU(RISCV64)
 #define WTF_USE_JSVALUE64 1
 #else
 #define WTF_USE_JSVALUE32_64 1
-- 
2.45.0

