diff -Naur qt-everywhere-opensource-src-4.8.7.riscv64/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h qt-everywhere-opensource-src-4.8.7/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h
--- qt-everywhere-opensource-src-4.8.7.riscv64/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h	2023-01-06 11:37:08.025075249 -0800
+++ qt-everywhere-opensource-src-4.8.7/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h	2023-01-06 11:39:09.778098264 -0800
@@ -377,8 +377,15 @@
 #endif
 #endif
 
+/* CPU(RISCV64) - RISC-V 64 */
+#if defined(__riscv) && (__riscv_xlen == 64)
+#define WTF_CPU_RISCV64 1
+#define ENABLE_JIT 0
+#define ENABLE_YARR_JIT 0
+#endif
+
 /* Not sure about this one, qt5-qtwebkit doesn't include it -- rex */
-#if CPU(ARM) || CPU(MIPS) || CPU(SH4) || CPU(AARCH64)
+#if CPU(ARM) || CPU(MIPS) || CPU(SH4) || CPU(AARCH64) || CPU(RISCV64)
 #define WTF_CPU_NEEDS_ALIGNED_ACCESS 1
 #endif
 
@@ -1012,7 +1019,7 @@
     || CPU(SPARC64) \
     || CPU(S390X) \
     || CPU(PPC64) \
-    || CPU(MIPS64) || CPU(AARCH64)
+    || CPU(MIPS64) || CPU(AARCH64) || CPU(RISCV64)
 #define WTF_USE_JSVALUE64 1
 #else
 #define WTF_USE_JSVALUE32_64 1
