From d8b304c565d69d44d28c52185c500af4bd4a63f1 Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Tue, 6 Jan 2026 09:49:22 +0800
Subject: [PATCH 1/2] dbus fix invalid conversion from 'const void*' to 'void*'

---
 src/dbus/qdbusintegrator.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/dbus/qdbusintegrator.cpp b/src/dbus/qdbusintegrator.cpp
index eaf3e291..10b246ca 100644
--- a/src/dbus/qdbusintegrator.cpp
+++ b/src/dbus/qdbusintegrator.cpp
@@ -73,7 +73,7 @@ QT_BEGIN_NAMESPACE
 static bool isDebugging;
 #define qDBusDebug              if (!::isDebugging); else qDebug
 
-Q_GLOBAL_STATIC_WITH_ARGS(const QString, orgFreedesktopDBusString, (QLatin1String(DBUS_SERVICE_DBUS)))
+Q_GLOBAL_STATIC_WITH_ARGS(QString, orgFreedesktopDBusString, (QLatin1String(DBUS_SERVICE_DBUS)))
 
 static inline QString dbusServiceString()
 { return *orgFreedesktopDBusString(); }
-- 
2.52.0

