From 48d574ea01731eab9f3ff5435faae3902e41934c Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sat, 14 Nov 2020 23:35:32 -0600
Subject: [PATCH] Fix other arch build

---
 cmake/scripts/linux/ArchSetup.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/scripts/linux/ArchSetup.cmake b/cmake/scripts/linux/ArchSetup.cmake
index 624edf6..f8a2a00 100644
--- a/cmake/scripts/linux/ArchSetup.cmake
+++ b/cmake/scripts/linux/ArchSetup.cmake
@@ -33,7 +33,7 @@ else()
     set(ARCH aarch64)
     set(NEON True)
   else()
-    message(SEND_ERROR "Unknown CPU: ${CPU}")
+    set(NEON False)
   endif()
 endif()
 
-- 
2.27.0

