From f9900876ac7b087987ed27a18105e6dc98d1bbc9 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 7 Feb 2024 19:32:03 +0800
Subject: [PATCH 3/5] Fix build for other architectures

---
 cmake/scripts/linux/ArchSetup.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/scripts/linux/ArchSetup.cmake b/cmake/scripts/linux/ArchSetup.cmake
index 4083483173..7ff62cab05 100644
--- a/cmake/scripts/linux/ArchSetup.cmake
+++ b/cmake/scripts/linux/ArchSetup.cmake
@@ -47,7 +47,7 @@ else()
     set(ARCH loongarch64)
     set(NEON False)
   else()
-    message(SEND_ERROR "Unknown CPU: ${CPU}")
+    set(NEON False)
   endif()
 endif()
 
-- 
2.43.0

