From 3397b1f9cee047d7496ac92af90e4092cde20f26 Mon Sep 17 00:00:00 2001
From: xunpod <xunpod@hotmail.com>
Date: Tue, 25 Jun 2024 12:32:04 +0800
Subject: [PATCH 2/3] feat: disable UnlockNeteaseMusic by default

---
 src/store/initLocalStorage.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/store/initLocalStorage.js b/src/store/initLocalStorage.js
index ce0e802..cf4fb2b 100644
--- a/src/store/initLocalStorage.js
+++ b/src/store/initLocalStorage.js
@@ -16,7 +16,7 @@ let localStorage = {
     lyricFontSize: 28,
     outputDevice: 'default',
     showPlaylistsByAppleMusic: true,
-    enableUnblockNeteaseMusic: true,
+    enableUnblockNeteaseMusic: false,
     automaticallyCacheSongs: true,
     cacheLimit: 8192,
     enableReversedMode: false,
-- 
2.34.1

