From 987cec7a8fb5a7684a005cb7e7c54befd33cfb7d Mon Sep 17 00:00:00 2001
From: xunpod <xunpod@hotmail.com>
Date: Tue, 25 Jun 2024 12:32:37 +0800
Subject: [PATCH 3/3] fix(tray): set the app icon as a tray icon

---
 src/electron/tray.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/electron/tray.js b/src/electron/tray.js
index ed314f2..a061764 100644
--- a/src/electron/tray.js
+++ b/src/electron/tray.js
@@ -201,7 +201,7 @@ export function createTray(win, eventEmitter) {
   let iconTheme = nativeTheme.shouldUseDarkColors ? 'light' : 'dark';
 
   let icon = nativeImage
-    .createFromPath(path.join(__static, `img/icons/menu-${iconTheme}@88.png`))
+    .createFromPath(path.join(__static, `img/icons/64x64.png`))
     .resize({
       height: 20,
       width: 20,
-- 
2.34.1

