diff --git a/src/utils/Player.js b/src/utils/Player.js
index a0073ec45b..11af73aaef 100644
--- a/src/utils/Player.js
+++ b/src/utils/Player.js
@@ -3,7 +3,7 @@ import { getArtist } from '@/api/artist';
 import { trackScrobble, trackUpdateNowPlaying } from '@/api/lastfm';
 import { fmTrash, personalFM } from '@/api/others';
 import { getPlaylistDetail, intelligencePlaylist } from '@/api/playlist';
-import { getLyric, getMP3, getTrackDetail, scrobble } from '@/api/track';
+import { getLyric, getMP3, getTrackDetail } from '@/api/track';
 import store from '@/store';
 import { isAccountLoggedIn } from '@/utils/auth';
 import { cacheTrackSource, getTrackSource } from '@/utils/db';
@@ -309,11 +309,6 @@ export default class {
     );
     const trackDuration = ~~(track.dt / 1000);
     time = completed ? trackDuration : ~~time;
-    scrobble({
-      id: track.id,
-      sourceid: this.playlistSource.id,
-      time,
-    });
     if (
       store.state.lastfm.key !== undefined &&
       (time >= trackDuration / 2 || time >= 240)
diff --git a/src/api/track.js b/src/api/track.js
index 490c91c3ad..58e3066898 100644
--- a/src/api/track.js
+++ b/src/api/track.js
@@ -126,23 +126,3 @@ export function likeATrack(params) {
     params,
   });
 }
-
-/**
- * 听歌打卡
- * 说明 : 调用此接口 , 传入音乐 id, 来源 id，歌曲时间 time，更新听歌排行数据
- * - id - 歌曲 id
- * - sourceid - 歌单或专辑 id
- * - time - 歌曲播放时间,单位为秒
- * @param {Object} params
- * @param {number} params.id
- * @param {number} params.sourceid
- * @param {number=} params.time
- */
-export function scrobble(params) {
-  params.timestamp = new Date().getTime();
-  return request({
-    url: '/scrobble',
-    method: 'get',
-    params,
-  });
-}
diff --git a/src/api/user.js b/src/api/user.js
index ec020144c7..553323ea85 100644
--- a/src/api/user.js
+++ b/src/api/user.js
@@ -84,23 +84,6 @@ export function userLikedSongsIDs(uid) {
   });
 }
 
-/**
- * 每日签到
- * 说明 : 调用此接口可签到获取积分
- * -  type: 签到类型 , 默认 0, 其中 0 为安卓端签到 ,1 为 web/PC 签到
- * @param {number} type
- */
-export function dailySignin(type = 0) {
-  return request({
-    url: '/daily_signin',
-    method: 'post',
-    params: {
-      type,
-      timestamp: new Date().getTime(),
-    },
-  });
-}
-
 /**
  * 获取收藏的专辑（需要登录）
  * 说明 : 调用此接口可获取到用户收藏的专辑
diff --git a/src/utils/common.js b/src/utils/common.js
index a89d500d82..8aa95baf9e 100644
--- a/src/utils/common.js
+++ b/src/utils/common.js
@@ -1,6 +1,5 @@
 import { isAccountLoggedIn } from './auth';
 import { refreshCookie } from '@/api/auth';
-import { dailySignin } from '@/api/user';
 import dayjs from 'dayjs';
 import store from '@/store';
 
@@ -110,12 +109,6 @@ export function dailyTask() {
         value: dayjs().date(),
       });
     });
-    dailySignin(0).catch(() => {
-      console.debug('[debug][common.js] 手机端重复签到');
-    });
-    dailySignin(1).catch(() => {
-      console.debug('[debug][common.js] PC端重复签到');
-    });
   }
 }
