Author: Cyril Brulebois <kibi@debian.org>
Description: Fix GNU/kFreeBSD build.
Bug-Debian: https://bugs.debian.org/417865
--- wdm.orig/src/wdm/daemon.c
+++ wdm/src/wdm/daemon.c
@@ -41,7 +41,7 @@
 #include <sys/ioctl.h>
 #endif
 #if defined(__osf__) || defined(linux) || defined(__GNU__) || defined(__CYGWIN__) \
-	|| (defined(IRIX) && !defined(_IRIX4))
+	|| (defined(IRIX) && !defined(_IRIX4)) || defined(__GLIBC__)
 #define setpgrp setpgid
 #endif
 #ifdef hpux
