Author: Agustin Martin Domingo <agmartin@debian.org>
Description: Fixed to work with composite extension
Bug-Debian: https://bugs.debian.org/488715, https://bugs.debian.org/547251
--- wdm.orig/src/wdmLogin/Login.c
+++ wdm/src/wdmLogin/Login.c
@@ -1190,6 +1190,9 @@
 	rattr.render_mode = render_mode;
 	rattr.colors_per_channel = cpc;
 	/* use of scr->screen is temporary hack */
+	rattr.flags |= RC_VisualID;
+        rattr.visualid = DefaultVisual(WMScreenDisplay(scr), scr->screen)->visualid;
+
 	rcontext = RCreateContext(WMScreenDisplay(scr), scr->screen, &rattr);
 	if(rcontext == NULL)
 	{
