Author: Matthias Klose <doko@ubuntu.com>
Description: Fix FTBFS with ld --as-needed.
Bug-Debian: https://bugs.debian.org/556673
--- wdm.orig/Makefile.in
+++ wdm/Makefile.in
@@ -14,7 +14,7 @@
 XLIBS = @XLIBS@
 WCFLAGS = @WCFLAGS@
 WLDFLAGS = @WLDFLAGS@
-WLIBS = @WLIBS@
+WLIBS = @WLIBS@ -lfontconfig
 SELINUX_CFLAGS = @SELINUX_CFLAGS@
 SELINUX_LIBS = @SELINUX_LIBS@
 PASSWDLIB = @PASSWDLIB@
@@ -35,23 +35,23 @@
 
 wdmLogin_cflags = $(cflags) $(XCFLAGS) $(WCFLAGS)
 wdmLogin_ldflags = $(ldflags) $(XLDFLAGS) $(WLDFLAGS)
-wdmLogin_libs = $(libs) $(XLIBS) $(WLIBS)
+wdmLogin_libs = $(WLIBS) $(libs) $(XLIBS)
 
 wdmPrefs_cflags = $(cflags) $(XCFLAGS) $(WCFLAGS)
 wdmPrefs_ldflags = $(ldflags) $(XLDFLAGS) $(WLDFLAGS)
-wdmPrefs_libs = $(libs) $(XLIBS) $(WLIBS)
+wdmPrefs_libs = $(WLIBS) $(libs) $(XLIBS)
 
 wdmChooser_cflags = $(cflags) $(XCFLAGS) $(WCFLAGS)
 wdmChooser_ldflags = $(ldflags) $(XLDFLAGS) $(WLDFLAGS)
-wdmChooser_libs = $(libs) $(XLIBS) $(WLIBS)
+wdmChooser_libs = $(WLIBS) $(libs) $(XLIBS)
 
 TestLogin_cflags = $(cflags) $(XCFLAGS) $(WCFLAGS)
 TestLogin_ldflags = $(ldflags) $(XLDFLAGS) $(WLDFLAGS)
-TestLogin_libs = $(libs) $(XLIBS) $(WLIBS)
+TestLogin_libs = $(WLIBS) $(libs) $(XLIBS)
 
 wdm_cflags = $(cflags) $(XCFLAGS) $(WCFLAGS) $(SELINUX_CFLAGS)
 wdm_ldflags = $(ldflags) $(XLDFLAGS) $(WLDFLAGS)
-wdm_libs = $(libs) $(XLIBS) $(WLIBS) $(PASSWDLIB) $(SELINUX_LIBS) \
+wdm_libs = $(WLIBS) $(libs) $(XLIBS) $(PASSWDLIB) $(SELINUX_LIBS) \
 	-lXmu -lXau -lXdmcp
 
 libwdm_cflags = $(cflags) $(XCFLAGS) $(WCFLAGS)
--- wdm.orig/etc/Howto.conf.in
+++ wdm/etc/Howto.conf.in
@@ -14,7 +14,7 @@
 XLIBS = @XLIBS@;
 WCFLAGS = @WCFLAGS@;
 WLDFLAGS = @WLDFLAGS@;
-WLIBS = @WLIBS@;
+WLIBS = @WLIBS@ -lfontconfig;
 SELINUX_CFLAGS = @SELINUX_CFLAGS@;
 SELINUX_LIBS = @SELINUX_LIBS@;
 PASSWDLIB = @PASSWDLIB@;
@@ -35,23 +35,23 @@
 
 wdmLogin_cflags = [cflags] [XCFLAGS] [WCFLAGS];
 wdmLogin_ldflags = [ldflags] [XLDFLAGS] [WLDFLAGS];
-wdmLogin_libs = [libs] [XLIBS] [WLIBS];
+wdmLogin_libs = [WLIBS] [libs] [XLIBS];
 
 wdmPrefs_cflags = [cflags] [XCFLAGS] [WCFLAGS];
 wdmPrefs_ldflags = [ldflags] [XLDFLAGS] [WLDFLAGS];
-wdmPrefs_libs = [libs] [XLIBS] [WLIBS];
+wdmPrefs_libs = [WLIBS] [libs] [XLIBS];
 
 wdmChooser_cflags = [cflags] [XCFLAGS] [WCFLAGS];
 wdmChooser_ldflags = [ldflags] [XLDFLAGS] [WLDFLAGS];
-wdmChooser_libs = [libs] [XLIBS] [WLIBS];
+wdmChooser_libs = [WLIBS] [libs] [XLIBS] -lfontconfig;
 
 TestLogin_cflags = [cflags] [XCFLAGS] [WCFLAGS];
 TestLogin_ldflags = [ldflags] [XLDFLAGS] [WLDFLAGS];
-TestLogin_libs = [libs] [XLIBS] [WLIBS];
+TestLogin_libs = [WLIBS] [libs] [XLIBS];
 
 wdm_cflags = [cflags] [XCFLAGS] [WCFLAGS] [SELINUX_CFLAGS];
 wdm_ldflags = [ldflags] [XLDFLAGS] [WLDFLAGS];
-wdm_libs = [libs] [XLIBS] [WLIBS] [PASSWDLIB] [SELINUX_LIBS] \
+wdm_libs = [WLIBS] [libs] [XLIBS] [PASSWDLIB] [SELINUX_LIBS] \
 	-lXmu -lXau -lXdmcp;
 
 libwdm_cflags = [cflags] [XCFLAGS] [WCFLAGS];
