From: Michael Gilbert <mgilbert@debian.org>
Date: Tue, 4 Oct 2016 11:43:45 +0200
Subject: Don't build icc2ps but use psicc from liblcms2-utils instead

---
 Makefile | 22 +++++-----------------
 1 file changed, 5 insertions(+), 17 deletions(-)

diff --git a/Makefile b/Makefile
index ea8b2a4..ddd03e4 100644
--- a/Makefile
+++ b/Makefile
@@ -196,13 +196,6 @@ FILES	=	\
 		foomatic-db/*/*.xml \
 		foomatic-test \
 		getweb.in \
-		icc2ps/*.[ch] \
-		icc2ps/*.1in \
-		icc2ps/Makefile \
-		icc2ps/AUTHORS \
-		icc2ps/COPYING \
-		icc2ps/README \
-		icc2ps/README.foo2zjs \
 		osx-hotplug/Makefile \
 		osx-hotplug/*.m \
 		osx-hotplug/*.1in \
@@ -394,7 +387,7 @@ JBGOPTS=-m 16 -d 0 -p 92	# Equivalent options for pbmtojbg
 # The usual build rules
 #
 all:	all-test $(PROGS) $(BINPROGS) $(SHELLS) getweb \
-	all-icc2ps all-osx-hotplug man doc \
+	all-osx-hotplug man doc \
 	all-done
 
 MACOSX_stdio=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h
@@ -631,7 +624,7 @@ command2foo2lava-pjl.o: command2foo2lava-pjl.c
 #
 # Installation rules
 #
-install: all install-test install-prog install-icc2ps install-osx-hotplug \
+install: all install-test install-prog install-osx-hotplug \
 	    install-extra install-crd install-foo install-ppd \
 	    install-gui install-desktop install-filter \
 	    install-man install-doc install-aa
@@ -1194,7 +1187,6 @@ uninstall: uninstall-aa
 	-rm -f $(MANDIR)/man1/foo2ddst*.1 $(MANDIR)/man1/ddstdecode.1
 	-rm -f $(MANDIR)/man1/gipddecode.1
 	-rm -f $(MANDIR)/man1/arm2hpdl.1 $(MANDIR)/man1/usb_printerid.1
-	-rm -f $(MANDIR)/man1/foo2zjs-icc2ps.1
 	-rm -rf /usr/share/foo2zjs/
 	-rm -rf /usr/share/foo2hp/
 	-rm -rf /usr/share/foo2oak/
@@ -1221,7 +1213,6 @@ uninstall: uninstall-aa
 	-rm -f /usr/bin/gipddecode
 	-rm -f /usr/bin/opldecode
 	-rm -f /usr/bin/rodecode
-	-rm -f /usr/bin/foo2zjs-icc2ps
 	-rm -f /usr/bin/foo2zjs-pstops
 	-rm -f /usr/bin/command2foo2lava-pjl
 	-rm -f /usr/lib/cups/filter/command2foo2lava-pjl
@@ -1266,7 +1257,6 @@ clean:
 	-rm -f *.zjs *.zm *.zc *.zc? *.zc?? *.oak *.pbm *.pksm *.cmyk
 	-rm -f pksm2bitcmyk
 	-rm -f *.icm.*.ps
-	cd icc2ps; $(MAKE) $@
 	cd osx-hotplug; $(MAKE) $@
 
 #
@@ -1420,7 +1410,7 @@ pprtest-3.oak: FRC
 #
 #	icc2ps regression tests
 #
-ICC2PS=./icc2ps/foo2zjs-icc2ps
+ICC2PS=/usr/bin/psicc
 icctest:
 	for g in *.icm; do \
 	    for i in 0 1 2 3; do \
@@ -1530,7 +1520,7 @@ oldppd:
 # Manpage generation.  No, I am not interested in "info" files or
 # HTML documentation.
 #
-man: $(MANPAGES) man-icc2ps man-osx-hotplug
+man: $(MANPAGES) man-osx-hotplug
 
 $(MANPAGES): macros.man includer-man
 
@@ -1601,7 +1591,6 @@ install-man: man
 	$(INSTALL) -c -m 644 arm2hpdl.1 $(MANDIR)/man1/
 	$(INSTALL) -c -m 644 usb_printerid.1 $(MANDIR)/man1/
 	$(INSTALL) -c -m 644 printer-profile.1 $(MANDIR)/man1/
-	cd icc2ps; $(MAKE) install-man
 ifeq ($(UNAME),Darwin)
 	cd osx-hotplug; $(MAKE) install-man
 endif
@@ -1622,10 +1611,9 @@ install-doc: doc
 
 GROFF=/usr/local/test/bin/groff
 GROFF=groff
-manual.pdf: $(MANPAGES) icc2ps/foo2zjs-icc2ps.1 osx-hotplug/osx-hplj-hotplug.1
+manual.pdf: $(MANPAGES) osx-hotplug/osx-hplj-hotplug.1
 	-$(GROFF) -t -man \
 	    `ls $(MANPAGES) \
-		icc2ps/foo2zjs-icc2ps.1 \
 		osx-hotplug/osx-hplj-hotplug.1 \
 		| sort` \
 	    | ps2pdf - $@
