From: Didier Raboud <odyx@debian.org>
Date: Tue, 4 Oct 2016 11:43:47 +0200
Subject: Use the Debian-provided libjbig.

---
 Makefile | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index ddd03e4..ec3c3b6 100644
--- a/Makefile
+++ b/Makefile
@@ -126,10 +126,6 @@ FILES	=	\
 		Makefile \
 		foo2zjs.c \
 		foo2zjs.1in \
-		jbig.c \
-		jbig.h \
-		jbig_ar.c \
-		jbig_ar.h \
 		zjsdecode.c \
 		zjsdecode.1in \
 		zjs.h \
@@ -287,7 +283,7 @@ MANPAGES+=	foo2ddst-wrapper.1 foo2ddst.1 ddstdecode.1
 MANPAGES+=	gipddecode.1
 MANPAGES+=	foo2zjs-pstops.1 arm2hpdl.1 usb_printerid.1
 MANPAGES+=	printer-profile.1
-LIBJBG	=	jbig.o jbig_ar.o
+LIBJBG =       -ljbig
 BINPROGS=
 
 ifeq ($(UNAME),Linux)
