From: Didier Raboud <odyx@debian.org>
Date: Tue, 4 Oct 2016 11:43:50 +0200
Subject: Remove 'Encoding' from .desktop file

---
 hplj1020.desktop | 1 -
 1 file changed, 1 deletion(-)

diff --git a/hplj1020.desktop b/hplj1020.desktop
index 121ab2b..49f9705 100644
--- a/hplj1020.desktop
+++ b/hplj1020.desktop
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=Reactivate HP LaserJet 1018/1020 after reloading paper
 Comment=Click the button in this program when your printer ran out of paper and you have refilled the paper
 # zExec=usb_printerid /dev/usb/lp0
