From: Didier Raboud <odyx@debian.org>
Date: Tue, 4 Oct 2016 11:43:53 +0200
Subject: Correct the path for the usb_printerid program

---
 hplj1000 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hplj1000 b/hplj1000
index 323069c..bb87a79 100755
--- a/hplj1000
+++ b/hplj1000
@@ -60,7 +60,7 @@ FWDIR=/usr/share/foo2zjs/firmware
 #    [snip]
 #    # Mount all other filesystems (except for NFS and /proc, which is already
 #
-PRINTERID=/bin/usb_printerid
+PRINTERID=`which usb_printerid`
 
 #
 #	Figure out how to log our messages
