From: Didier Raboud <odyx@debian.org>
Date: Tue, 4 Oct 2016 11:44:00 +0200
Subject: Delay firmware transfer for UDEV automatic printer setup

---
 hplj1000 | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/hplj1000 b/hplj1000
index 21ca570..6b74601 100755
--- a/hplj1000
+++ b/hplj1000
@@ -32,6 +32,15 @@ esac
 
 PROGNAME="$0"
 
+#
+# Wait some time before installing the firmware so that probing for
+# automatic printer setup which happens immediately after turning on
+# the printer does not interfere with the firmware transfer to the
+# printer
+# https://bugs.launchpad.net/ubuntu/+source/foo2zjs/+bug/543177
+#
+sleep 3
+
 #
 # Set $DEV to, e.g. /dev/usb/lp0, to force the device you want
 # Else, leave it null to automatically detect the device
