From: Didier Raboud <odyx@debian.org>
Date: Tue, 8 May 2018 09:40:27 +0200
Subject: Don't try to install-hotplug-osx on all UNAME's

---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 6d0177e..04085c2 100644
--- a/Makefile
+++ b/Makefile
@@ -1063,7 +1063,9 @@ install-hotplug-prog:
 	$(USBDIR)/hplj1000 install-usblp
 
 install-hotplug-osx:
+ifeq ($(UNAME),Darwin)
 	cd osx-hotplug; $(MAKE) PREFIX=$(PREFIX) install-hotplug
+endif
 
 install-filter:
 	if [ "$(CUPS_SERVERBIN)" != "" ]; then \
@@ -1526,7 +1528,9 @@ man-icc2ps:
 	cd icc2ps; $(MAKE) man
 
 man-osx-hotplug:
+ifeq ($(UNAME),Darwin)
 	cd osx-hotplug; $(MAKE) man
+endif
 
 .1in.1: 
 	-rm -f $*.1
