From: Helmut Grohne <helmut@subdivi.de>
Date: Sat, 18 Jan 2020 09:50:26 +0100
Subject: Drop -lsomelib make dependencies to progress towards
 cross-buildability

Closes: #949207
---
 Makefile | 46 +++++++++++++++++++++++-----------------------
 1 file changed, 23 insertions(+), 23 deletions(-)

diff --git a/Makefile b/Makefile
index 33ec176..55d7ced 100644
--- a/Makefile
+++ b/Makefile
@@ -450,34 +450,34 @@ all-done:
 	@echo "yourself."
 
 
-foo2ddst: foo2ddst.o $(LIBJBG)
+foo2ddst: foo2ddst.o
 	$(CC) $(CFLAGS) -o $@ foo2ddst.o $(LIBJBG) $(LDFLAGS)
 
-foo2hbpl2: foo2hbpl2.o $(LIBJBG)
+foo2hbpl2: foo2hbpl2.o
 	$(CC) $(CFLAGS) -o $@ foo2hbpl2.o $(LIBJBG) $(LDFLAGS)
 
-foo2hp: foo2hp.o $(LIBJBG)
+foo2hp: foo2hp.o
 	$(CC) $(CFLAGS) -o $@ foo2hp.o $(LIBJBG) $(LDFLAGS)
 
-foo2hiperc: foo2hiperc.o $(LIBJBG)
+foo2hiperc: foo2hiperc.o
 	$(CC) $(CFLAGS) -o $@ foo2hiperc.o $(LIBJBG) $(LDFLAGS)
 
-foo2lava: foo2lava.o $(LIBJBG)
+foo2lava: foo2lava.o
 	$(CC) $(CFLAGS) -o $@ foo2lava.o $(LIBJBG) $(LDFLAGS)
 
-foo2oak: foo2oak.o $(LIBJBG)
+foo2oak: foo2oak.o
 	$(CC) $(CFLAGS) -o $@ foo2oak.o $(LIBJBG) $(LDFLAGS)
 
-foo2qpdl: foo2qpdl.o $(LIBJBG)
+foo2qpdl: foo2qpdl.o
 	$(CC) $(CFLAGS) -o $@ foo2qpdl.o $(LIBJBG) $(LDFLAGS)
 
-foo2slx: foo2slx.o $(LIBJBG)
+foo2slx: foo2slx.o
 	$(CC) $(CFLAGS) -o $@ foo2slx.o $(LIBJBG) $(LDFLAGS)
 
-foo2xqx: foo2xqx.o $(LIBJBG)
+foo2xqx: foo2xqx.o
 	$(CC) $(CFLAGS) -o $@ foo2xqx.o $(LIBJBG) $(LDFLAGS)
 
-foo2zjs: foo2zjs.o $(LIBJBG)
+foo2zjs: foo2zjs.o
 	$(CC) $(CFLAGS) -o $@ foo2zjs.o $(LIBJBG) $(LDFLAGS)
 
 
@@ -563,43 +563,43 @@ ifeq ($(UNAME),Darwin)
 	cd osx-hotplug; $(MAKE) all
 endif
 
-ok: ok.o $(LIBJBG)
+ok: ok.o
 	$(CC) $(CFLAGS) ok.o $(LIBJBG) -o $@ $(LDFLAGS)
 
-ddstdecode: ddstdecode.o $(LIBJBG)
+ddstdecode: ddstdecode.o
 	$(CC) $(CFLAGS) ddstdecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
-gipddecode: gipddecode.o $(LIBJBG)
+gipddecode: gipddecode.o
 	$(CC) $(CFLAGS) gipddecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
-hbpldecode: hbpldecode.o $(LIBJBG)
+hbpldecode: hbpldecode.o
 	$(CC) $(CFLAGS) hbpldecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
-hipercdecode: hipercdecode.o $(LIBJBG)
+hipercdecode: hipercdecode.o
 	$(CC) $(CFLAGS) hipercdecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
-lavadecode: lavadecode.o $(LIBJBG)
+lavadecode: lavadecode.o
 	$(CC) $(CFLAGS) lavadecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
-oakdecode: oakdecode.o $(LIBJBG)
+oakdecode: oakdecode.o
 	$(CC) $(CFLAGS) -g oakdecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
-opldecode: opldecode.o $(LIBJBG)
+opldecode: opldecode.o
 	$(CC) $(CFLAGS) -g opldecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
-qpdldecode: qpdldecode.o $(LIBJBG)
+qpdldecode: qpdldecode.o
 	$(CC) $(CFLAGS) qpdldecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
-splcdecode: splcdecode.o $(LIBJBG)
+splcdecode: splcdecode.o
 	$(CC) $(CFLAGS) splcdecode.o $(LIBJBG) -lz -o $@ $(LDFLAGS)
 
-slxdecode: slxdecode.o $(LIBJBG)
+slxdecode: slxdecode.o
 	$(CC) $(CFLAGS) slxdecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
-xqxdecode: xqxdecode.o $(LIBJBG)
+xqxdecode: xqxdecode.o
 	$(CC) $(CFLAGS) xqxdecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
-zjsdecode: zjsdecode.o $(LIBJBG)
+zjsdecode: zjsdecode.o
 	$(CC) $(CFLAGS) zjsdecode.o $(LIBJBG) -o $@ $(LDFLAGS)
 
 command2foo2lava-pjl: command2foo2lava-pjl.o
