# Ref: https://github.com/xushengfeng/eSearch/pull/441

diff --git a/esearch/lib/ipc.ts b/esearch/lib/ipc.ts
index 2017b72f..bea872cf 100644
--- a/esearch/lib/ipc.ts
+++ b/esearch/lib/ipc.ts
@@ -119,6 +119,7 @@ type Message = {
     setAutoStart: (value: boolean) => void;
     theme: (t: NativeTheme["themeSource"]) => void;
     dingIgnore: (ignore: boolean) => void;
+    recorderReady: ()=> void,
     recordInit: (
         id: string,
         r: [number, number, number, number],
diff --git a/esearch/src/main/main.ts b/esearch/src/main/main.ts
index b29a5e0c..f165568c 100644
--- a/esearch/src/main/main.ts
+++ b/esearch/src/main/main.ts
@@ -1278,7 +1278,7 @@ function createRecorderWindow(
         }
     });

-    recorder.webContents.on("did-finish-load", () => {
+    mainOn("recorderReady", () => {
         desktopCapturer.getSources({ types: ["screen"] }).then((sources) => {
             let dId = sources.find((s) => s.display_id === screenx.id)?.id;
             if (!dId) dId = sources[0].id;
diff --git a/esearch/src/renderer/recorder/recorder.ts b/esearch/src/renderer/recorder/recorder.ts
index 92071e3a..96cd8254 100644
--- a/esearch/src/renderer/recorder/recorder.ts
+++ b/esearch/src/renderer/recorder/recorder.ts
@@ -1155,3 +1155,5 @@ renderOn("recordState", ([s]) => {
         }
     }
 });
+
+renderSend("recorderReady", []);
--
2.52.0
