From 14de61e0dadd762ea37d32a0f1c186357519a863 Mon Sep 17 00:00:00 2001
From: "Lain \"Fearyncess\" Yang" <fsf@live.com>
Date: Fri, 30 Aug 2024 17:27:37 +0800
Subject: [PATCH 2/5] r600: disable buggy vc-1 hardware decoding

---
 src/gallium/drivers/r600/radeon_video.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gallium/drivers/r600/radeon_video.c b/src/gallium/drivers/r600/radeon_video.c
index 670e0aafb9e..6e5abd8d1a1 100644
--- a/src/gallium/drivers/r600/radeon_video.c
+++ b/src/gallium/drivers/r600/radeon_video.c
@@ -237,7 +237,7 @@ int rvid_get_video_param(struct pipe_screen *screen,
 		case PIPE_VIDEO_FORMAT_MPEG4_AVC:
 			return true;
 		case PIPE_VIDEO_FORMAT_VC1:
-			return true;
+			return false;
 		case PIPE_VIDEO_FORMAT_HEVC:
 			return false;
 		case PIPE_VIDEO_FORMAT_JPEG:
-- 
2.46.0.windows.1

