From 51c1fbbe7a098c93b6a07f1ed4daa64b1800f044 Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Wed, 21 May 2025 19:49:12 +0800
Subject: [PATCH] cpp system process add mips64 cpu

---
 cmake_modules/SetupCxxFlags.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake_modules/SetupCxxFlags.cmake b/cmake_modules/SetupCxxFlags.cmake
index fdb28b540..dad97f88d 100644
--- a/cmake_modules/SetupCxxFlags.cmake
+++ b/cmake_modules/SetupCxxFlags.cmake
@@ -40,6 +40,8 @@ if(NOT DEFINED ARROW_CPU_FLAG)
     set(ARROW_CPU_FLAG "riscv64")
   elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "loongarch64")
     set(ARROW_CPU_FLAG "loongarch64")
+  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "mips64")
+    set(ARROW_CPU_FLAG "mips64")
   else()
     message(FATAL_ERROR "Unknown system processor")
   endif()
-- 
2.49.0

