diff -aur qownnotes-22.1.0/dialogs/settingsdialog.cpp qownnotes-22.1.0.aosc/dialogs/settingsdialog.cpp
--- qownnotes-22.1.0/dialogs/settingsdialog.cpp	2022-01-01 16:35:41.000000000 +0800
+++ qownnotes-22.1.0.aosc/dialogs/settingsdialog.cpp	2022-01-01 22:55:44.924298337 +0800
@@ -1245,10 +1245,11 @@
     const QSignalBlocker blocker(ui->appMetricsCheckBox);
     Q_UNUSED(blocker)
     ui->appMetricsCheckBox->setChecked(
-        settings.value(QStringLiteral("appMetrics/disableTracking")).toBool());
+        settings.value(QStringLiteral("appMetrics/disableTracking"), true)
+            .toBool());
 
     ui->appHeartbeatCheckBox->setChecked(
-        settings.value(QStringLiteral("appMetrics/disableAppHeartbeat"))
+        settings.value(QStringLiteral("appMetrics/disableAppHeartbeat"), true)
             .toBool());
 
     ui->darkModeColorsCheckBox->setChecked(
