diff -Naur tdemultimedia-trinity-14.0.13/akode_artsplugin/akodePlayObject_impl.h tdemultimedia-trinity-14.0.13.arts/akode_artsplugin/akodePlayObject_impl.h
--- tdemultimedia-trinity-14.0.13/akode_artsplugin/akodePlayObject_impl.h	2022-10-13 16:31:52.000000000 -0700
+++ tdemultimedia-trinity-14.0.13.arts/akode_artsplugin/akodePlayObject_impl.h	2022-12-04 08:50:47.666837739 -0800
@@ -26,8 +26,8 @@
 #include <akode/decoder.h>
 #include <akode/resampler.h>
 
-#include <stdsynthmodule.h>
-#include <kmedia2.h>
+#include <arts/stdsynthmodule.h>
+#include <arts/kmedia2.h>
 #include <queue>
 #include "akodearts.h"
 
