From 67ebdc6edc14b14c1c64fec1af0ba740eedfefca Mon Sep 17 00:00:00 2001
From: Zane van Iperen <zane@zanevaniperen.com>
Date: Mon, 15 Mar 2021 17:23:23 +1000
Subject: [PATCH 01/10] lib/moviedecoder: remove unused variable

---
 libffmpegthumbnailer/moviedecoder.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/libffmpegthumbnailer/moviedecoder.cpp b/libffmpegthumbnailer/moviedecoder.cpp
index 290e212..aa44adf 100644
--- a/libffmpegthumbnailer/moviedecoder.cpp
+++ b/libffmpegthumbnailer/moviedecoder.cpp
@@ -570,8 +570,6 @@ bool MovieDecoder::getVideoPacket()
     bool framesAvailable = true;
     bool frameDecoded = false;
 
-    int attempts = 0;
-
     if (m_pPacket)
     {
         av_packet_unref(m_pPacket);
-- 
2.46.0

