From 65ea730dd3f99e289cebe6d85af904a9e66ce8f2 Mon Sep 17 00:00:00 2001
From: Paul Jonkins <paul.ionkin@gmail.com>
Date: Wed, 23 Jun 2021 17:53:55 +0200
Subject: [PATCH 10/10] Constify AVCodec

---
 libffmpegthumbnailer/moviedecoder.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libffmpegthumbnailer/moviedecoder.h b/libffmpegthumbnailer/moviedecoder.h
index 3ef5f12..fb6add2 100644
--- a/libffmpegthumbnailer/moviedecoder.h
+++ b/libffmpegthumbnailer/moviedecoder.h
@@ -78,7 +78,7 @@ private:
     int                     m_VideoStream;
     AVFormatContext*        m_pFormatContext;
     AVCodecContext*         m_pVideoCodecContext;
-    AVCodec*                m_pVideoCodec;
+    const AVCodec*          m_pVideoCodec;
     AVFilterGraph*          m_pFilterGraph;
     AVFilterContext*        m_pFilterSource;
     AVFilterContext*        m_pFilterSink;
-- 
2.46.0

