From 16cf4fd4c6b16a08a341014fc2196b57bd7859d0 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Wed, 1 Apr 2020 20:21:26 -0500
Subject: [PATCH 1/2] neofetch/aosc: split AOSC OS and AOSC OS/Retro

---
 neofetch | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/neofetch b/neofetch
index 5359f81..e4cd3c8 100755
--- a/neofetch
+++ b/neofetch
@@ -753,8 +753,8 @@ image_source="auto"
 # Default: 'auto'
 # Values:  'auto', 'distro_name'
 # Flag:    --ascii_distro
-# NOTE: AIX, Alpine, Anarchy, Android, Antergos, antiX, AOSC,
-#       Apricity, ArcoLinux, ArchBox, ARCHlabs, ArchStrike,
+# NOTE: AIX, Alpine, Anarchy, Android, Antergos, antiX, "AOSC OS",
+#       "AOSC OS/Retro", Apricity, ArcoLinux, ArchBox, ARCHlabs, ArchStrike,
 #       XFerience, ArchMerge, Arch, Artix, Arya, Bedrock, Bitrig,
 #       BlackArch, BLAG, BlankOn, BlueLight, bonsai, BSD,
 #       BunsenLabs, Calculate, Carbs, CentOS, Chakra, ChaletOS,
@@ -4817,8 +4817,8 @@ ASCII:
     --ascii_colors x x x x x x  Colors to print the ascii art
     --ascii_distro distro       Which Distro's ascii art to print
 
-                                NOTE: AIX, Alpine, Anarchy, Android, Antergos, antiX, AOSC,
-                                Apricity, ArcoLinux, ArchBox, ARCHlabs, ArchStrike,
+                                NOTE: AIX, Alpine, Anarchy, Android, Antergos, antiX, "AOSC OS",
+                                "AOSC OS/Retro", Apricity, ArcoLinux, ArchBox, ARCHlabs, ArchStrike,
                                 XFerience, ArchMerge, Arch, Artix, Arya, Bedrock, Bitrig,
                                 BlackArch, BLAG, BlankOn, BlueLight, bonsai, BSD,
                                 BunsenLabs, Calculate, Carbs, CentOS, Chakra, ChaletOS,
-- 
2.24.1

