From 2c392729e23db9a89307c2fa032d4427d9471c27 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Fri, 21 Oct 2022 18:16:03 -0700
Subject: [PATCH] neofetch: (AOSC OS) add quotation marks around architecture
 name

---
 neofetch | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/neofetch b/neofetch
index 341aca51..abe510e3 100755
--- a/neofetch
+++ b/neofetch
@@ -1226,7 +1226,7 @@ get_distro() {
     esac
 
     [[ $os_arch == on ]] && \
-        distro+=" $machine_arch"
+        distro+=" ($machine_arch)"
 
     [[ ${ascii_distro:-auto} == auto ]] && \
         ascii_distro=$(trim "$distro")
-- 
2.37.1
