diff -Naur OpenBLAS-0.3.10/Makefile.system OpenBLAS-0.3.10.loongson3/Makefile.system
--- OpenBLAS-0.3.10/Makefile.system	2020-06-14 13:03:04.000000000 -0700
+++ OpenBLAS-0.3.10.loongson3/Makefile.system	2020-12-02 13:05:41.276652024 -0800
@@ -704,8 +704,8 @@
 endif
 
 ifeq ($(CORE), LOONGSON3A)
-CCOMMON_OPT += -march=mips64
-FCOMMON_OPT += -march=mips64
+CCOMMON_OPT += -march=gs464
+FCOMMON_OPT += -march=gs464
 endif
 
 ifeq ($(CORE), LOONGSON3B)
@@ -969,7 +969,7 @@
 FCOMMON_OPT += -n64
 endif
 ifeq ($(CORE), LOONGSON3A)
-FCOMMON_OPT += -loongson3 -static
+FCOMMON_OPT += -march=gs464 -static
 endif
 
 ifeq ($(CORE), LOONGSON3B)
@@ -999,7 +999,7 @@
 CCOMMON_OPT += -n64
 endif
 ifeq ($(CORE), LOONGSON3A)
-CCOMMON_OPT += -loongson3 -static
+CCOMMON_OPT += -march=gs464 -static
 endif
 
 ifeq ($(CORE), LOONGSON3B)
