From 480a38d4ad2f6fa2137e81e9f318dda445858e9c Mon Sep 17 00:00:00 2001
From: Sylvain Henry <sylvain@haskus.fr>
Date: Wed, 2 Dec 2020 12:25:41 +0100
Subject: [PATCH] rts: don't use siginterrupt (#19019)

Signed-off-by: Camber Huang <camber@poi.science>
---
 rts/posix/Signals.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/rts/posix/Signals.c b/rts/posix/Signals.c
index 5ad688bc2f..57b1873c77 100644
--- a/rts/posix/Signals.c
+++ b/rts/posix/Signals.c
@@ -680,15 +680,11 @@ initDefaultHandlers(void)
     // install the SIGINT handler
     action.sa_handler = shutdown_handler;
     sigemptyset(&action.sa_mask);
-    action.sa_flags = 0;
+    action.sa_flags = 0; // disable SA_RESTART
     if (sigaction(SIGINT, &action, &oact) != 0) {
         sysErrorBelch("warning: failed to install SIGINT handler");
     }
 
-#if defined(HAVE_SIGINTERRUPT)
-    siginterrupt(SIGINT, 1);    // isn't this the default? --SDM
-#endif
-
     // install the SIGFPE handler
 
     // In addition to handling SIGINT, also handle SIGFPE by ignoring it.
-- 
2.37.1

