From 642b25e46cf9860e988cb439760b530ce96b8476 Mon Sep 17 00:00:00 2001
From: Camber Huang <camber@poi.science>
Date: Fri, 29 Jul 2022 18:35:44 +0800
Subject: [PATCH] Fix doc generate error

Signed-off-by: Camber Huang <camber@poi.science>
---
 docs/users_guide/conf.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/docs/users_guide/conf.py b/docs/users_guide/conf.py
index 83e0215640..bdcf328b4f 100644
--- a/docs/users_guide/conf.py
+++ b/docs/users_guide/conf.py
@@ -100,13 +100,13 @@ htmlhelp_basename = 'GHCUsersGuide'
 latex_elements = {
     'inputenc': '',
     'utf8extra': '',
-    'preamble': '''
+    'preamble': r'''
 \usepackage{fontspec}
 \usepackage{makeidx}
 \setsansfont{DejaVu Sans}
 \setromanfont{DejaVu Serif}
 \setmonofont{DejaVu Sans Mono}
-\setlength{\\tymin}{45pt}
+\setlength{\tymin}{45pt}
 ''',
 }
 
-- 
2.37.1

