--- shadow-4.2.1-orig/configure.in	2014-05-09 11:53:10.000000000 +0000
+++ id32bit/configure.in	2015-05-25 09:28:37.478198111 +0000
@@ -335,14 +335,7 @@
 	dnl
 	dnl FIXME: check if 32 bit UIDs/GIDs are supported by libc
 	dnl
-	AC_RUN_IFELSE([AC_LANG_SOURCE([
-#include <sys/types.h>
-int main(void) {
-	uid_t u;
-	gid_t g;
-	return (sizeof u < 4) || (sizeof g < 4);
-}
-	])], [id32bit="yes"], [id32bit="no"])
+	id32bit="yes"
 
 	if test "x$id32bit" = "xyes"; then
 		AC_DEFINE(ENABLE_SUBIDS, 1, [Define to support the subordinate IDs.])
