From 00621208b411011d9f858923fa9d2ccc024767b5 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Thu, 30 Jun 2016 11:32:03 -0400
Subject: [PATCH] allow deprecated declarations for now to let readdir_r
 work...

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 gcc.specs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc.specs b/gcc.specs
index 0d4bbda..d25ac5e 100644
--- a/gcc.specs
+++ b/gcc.specs
@@ -2,7 +2,7 @@
 + -D_GNU_SOURCE
 
 *efivar_cpp_options:
- -Werror -Wall -std=gnu11 -Wextra
+ -Werror -Wall -std=gnu11 -Wextra -Wno-error=deprecated-declarations
 
 *cpp_options:
 + %(efivar_cpp_options)
-- 
2.7.4


