From 6d9763483d86c135b9fcc05b021f566d27b80e94 Mon Sep 17 00:00:00 2001
From: m3tav3rse <n6maa10816@tuta.io>
Date: Thu, 19 Jul 2018 16:16:46 +0200
Subject: [PATCH] linux-pci: fix acpi uid parsing based on 2050c87

---
 src/linux-pci.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/linux-pci.c b/src/linux-pci.c
index 87878c3..f764235 100644
--- a/src/linux-pci.c
+++ b/src/linux-pci.c
@@ -104,9 +104,11 @@ parse_pci(struct device *dev, const char *current)
         rc = read_sysfs_file(&fbuf,
                              "devices/pci%04hx:%02hhx/firmware_node/uid",
                              root_domain, root_bus);
-        if ((rc <= 0 && errno != ENOENT) || fbuf == NULL)
+        if (rc <= 0 && errno != ENOENT)
                 return -1;
-        if (rc > 0) {
+        else if (rc > 0) {
+                if (fbuf == NULL)
+                    return -1;
                 rc = sscanf((char *)fbuf, "%"PRIu64"\n", &acpi_uid_int);
                 if (rc == 1) {
                         dev->pci_root.pci_root_acpi_uid = acpi_uid_int;
