From fe927ddb39f68b51a26f76d92df9dda29a05d65a Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Thu, 28 Oct 2021 23:43:51 -0500
Subject: [PATCH] cmake: link against embree avx512

---
 build_files/cmake/Modules/FindEmbree.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build_files/cmake/Modules/FindEmbree.cmake b/build_files/cmake/Modules/FindEmbree.cmake
index 7f7f2ae0fb3..8851d2cf607 100644
--- a/build_files/cmake/Modules/FindEmbree.cmake
+++ b/build_files/cmake/Modules/FindEmbree.cmake
@@ -39,6 +39,7 @@ IF(NOT (APPLE AND ("${CMAKE_OSX_ARCHITECTURES}" STREQUAL "arm64")))
     embree_sse42
     embree_avx
     embree_avx2
+    embree_avx512
   )
 ENDIF()
 
-- 
2.30.2

