diff -Naur blender-2.79.old/source/creator/CMakeLists.txt blender-2.79/source/creator/CMakeLists.txt
--- blender-2.79.old/source/creator/CMakeLists.txt	2017-09-12 14:17:03.361182908 +0200
+++ blender-2.79/source/creator/CMakeLists.txt	2017-09-12 14:17:22.264536022 +0200
@@ -342,8 +342,8 @@
 	endif()
 
 	install(
-		DIRECTORY ${CMAKE_SOURCE_DIR}/release/scripts
-		DESTINATION ${TARGETDIR_VER}
+		DIRECTORY ${CMAKE_SOURCE_DIR}/release/scripts/
+		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/blender/scripts/
 		PATTERN ".git" EXCLUDE
 		PATTERN ".gitignore" EXCLUDE
 		PATTERN ".arcconfig" EXCLUDE
