diff -Naur blender-2.79.old/source/blender/blenfont/intern/blf_font_i18n.c blender-2.79/source/blender/blenfont/intern/blf_font_i18n.c
--- blender-2.79.old/source/blender/blenfont/intern/blf_font_i18n.c	2017-09-11 06:34:59.000000000 +0200
+++ blender-2.79/source/blender/blenfont/intern/blf_font_i18n.c	2017-09-12 14:12:51.239473283 +0200
@@ -57,7 +57,7 @@
 {
 #ifdef WITH_INTERNATIONAL
 	if (unifont_ttf == NULL) {
-		const char * const fontpath = BKE_appdir_folder_id(BLENDER_DATAFILES, "fonts");
+		const char * const fontpath = "/usr/share/fonts/blender";
 		if (fontpath) {
 			char unifont_path[1024];
 
@@ -92,7 +92,7 @@
 {
 #ifdef WITH_INTERNATIONAL
 	if (unifont_mono_ttf == NULL) {
-		const char *fontpath = BKE_appdir_folder_id(BLENDER_DATAFILES, "fonts");
+		const char * const fontpath = "/usr/share/fonts/blender";
 		if (fontpath) {
 			char unifont_path[1024];
 
diff -Naur blender-2.79.old/source/creator/CMakeLists.txt blender-2.79/source/creator/CMakeLists.txt
--- blender-2.79.old/source/creator/CMakeLists.txt	2017-09-12 05:44:18.000000000 +0200
+++ blender-2.79/source/creator/CMakeLists.txt	2017-09-12 14:12:51.239473283 +0200
@@ -359,9 +359,8 @@
 # localization
 if(WITH_INTERNATIONAL)
 	install(
-		DIRECTORY
-			${CMAKE_SOURCE_DIR}/release/datafiles/fonts
-		DESTINATION ${TARGETDIR_VER}/datafiles
+		DIRECTORY ${CMAKE_SOURCE_DIR}/release/datafiles/fonts/
+		DESTINATION ${CMAKE_INSTALL_PREFIX}/share/fonts/blender
 	)
 
 	set(_locale_dir "${CMAKE_SOURCE_DIR}/release/datafiles/locale")
