diff -Naur blender-2.79.old/intern/ghost/intern/GHOST_SystemPathsUnix.cpp blender-2.79/intern/ghost/intern/GHOST_SystemPathsUnix.cpp
--- blender-2.79.old/intern/ghost/intern/GHOST_SystemPathsUnix.cpp	2017-09-12 14:49:10.055232962 +0200
+++ blender-2.79/intern/ghost/intern/GHOST_SystemPathsUnix.cpp	2017-09-12 14:50:46.149035501 +0200
@@ -61,6 +61,9 @@
 
 const GHOST_TUns8 *GHOST_SystemPathsUnix::getSystemDir(int, const char *versionstr) const
 {
+	/* ignore versionstr when building the system path */
+	versionstr = "";
+
 	/* no prefix assumes a portable build which only uses bundled scripts */
 	if (static_path) {
 		static string system_path = string(static_path) + "/blender/" + versionstr;
diff -Naur blender-2.79.old/source/creator/CMakeLists.txt blender-2.79/source/creator/CMakeLists.txt
--- blender-2.79.old/source/creator/CMakeLists.txt	2017-09-12 14:49:10.243236700 +0200
+++ blender-2.79/source/creator/CMakeLists.txt	2017-09-12 14:50:46.149035501 +0200
@@ -295,7 +295,7 @@
 		if(WITH_INSTALL_PORTABLE)
 			set(TARGETDIR_VER ${BLENDER_VERSION})
 		else()
-			set(TARGETDIR_VER share/blender/${BLENDER_VERSION})
+			set(TARGETDIR_VER share/blender)
 		endif()
 	endif()
 
