diff -Naur blender-2.79.old/blender.thumbnailer blender-2.79/blender.thumbnailer
--- blender-2.79.old/blender.thumbnailer	1970-01-01 01:00:00.000000000 +0100
+++ blender-2.79/blender.thumbnailer	2017-09-12 14:14:51.034711054 +0200
@@ -0,0 +1,4 @@
+[Thumbnailer Entry]
+TryExec=/usr/share/blender/scripts/blender-thumbnailer.py
+Exec=/usr/share/blender/scripts/blender-thumbnailer.py %i %o
+MimeType=application/x-blender;
diff -Naur blender-2.79.old/source/creator/CMakeLists.txt blender-2.79/source/creator/CMakeLists.txt
--- blender-2.79.old/source/creator/CMakeLists.txt	2017-09-12 14:15:10.335071584 +0200
+++ blender-2.79/source/creator/CMakeLists.txt	2017-09-12 14:14:51.034711054 +0200
@@ -514,7 +514,7 @@
 		)
 		install(
 			PROGRAMS ${CMAKE_SOURCE_DIR}/release/bin/blender-thumbnailer.py
-			DESTINATION bin
+			DESTINATION ${CMAKE_INSTALL_PREFIX}/share/blender/scripts
 		)
 		set(BLENDER_TEXT_FILES_DESTINATION share/doc/blender)
 	endif()
