From b2a8a7d600aa7124e02f65b406d14f0624e8677b Mon Sep 17 00:00:00 2001
From: Dima Pasechnik <dima@pasechnik.info>
Date: Fri, 8 Aug 2025 22:25:52 -0500
Subject: [PATCH] allow external flint 3.3.*

---
 build/pkgs/flint/spkg-configure.m4 | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/build/pkgs/flint/spkg-configure.m4 b/build/pkgs/flint/spkg-configure.m4
index 50a57f8f9f2..d6d21b80fae 100644
--- a/build/pkgs/flint/spkg-configure.m4
+++ b/build/pkgs/flint/spkg-configure.m4
@@ -3,12 +3,12 @@ SAGE_SPKG_CONFIGURE([flint], [
         AC_CHECK_HEADERS([flint/flint.h flint/padic.h], [dnl
           dnl gr_get_fexpr appears in Flint 3.0
           AC_SEARCH_LIBS([gr_get_fexpr], [flint], [dnl
-            dnl Assume Flint 3.3 is too new
-            AC_MSG_CHECKING([whether FLINT version is >= 3.3.0])
+            dnl Assume Flint 3.4 is too new
+            AC_MSG_CHECKING([whether FLINT version is >= 3.4.0])
             AC_COMPILE_IFELSE([dnl
               AC_LANG_PROGRAM([[#include <flint/flint.h>
-                                #if __FLINT_RELEASE >= 30300
-                                # error "FLINT 3.3 is too new"
+                                #if __FLINT_RELEASE > 30399
+                                # error "FLINT 3.4 is too new"
                                 #endif
                               ]])
             ], [dnl
-- 
2.52.0

