From 6ac1bf1ca87fb147c2bc1c4811716982d2151904 Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Fri, 28 Nov 2025 06:36:21 +0800
Subject: [PATCH 1/2] flint build without LTO on Loongson3

---
 build/pkgs/flint/spkg-build.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/build/pkgs/flint/spkg-build.in b/build/pkgs/flint/spkg-build.in
index ffb822c8a97..972be9e57a5 100644
--- a/build/pkgs/flint/spkg-build.in
+++ b/build/pkgs/flint/spkg-build.in
@@ -1,3 +1,8 @@
 cd src
+
+if [ `uname -m` = "mips64" ]; then
+    export CFLAGS="$CFLAGS -fno-lto"
+fi
+
 sdh_configure
 sdh_make
-- 
2.52.0

