From cb5284f0b73c28b10d3014aa57bf4ac8e0c0c85a Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Fri, 28 Nov 2025 06:40:04 +0800
Subject: [PATCH 2/2] giac build without LTO on Loongson3

---
 build/pkgs/giac/spkg-install.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/build/pkgs/giac/spkg-install.in b/build/pkgs/giac/spkg-install.in
index ceed3899be6..b59fd4f0a5a 100644
--- a/build/pkgs/giac/spkg-install.in
+++ b/build/pkgs/giac/spkg-install.in
@@ -52,6 +52,13 @@ if [ "$UNAME" = "Darwin" ]; then
     DISABLENLS="--disable-nls"
 fi
 
+if [ `uname -m` = "mips64" ]; then
+    echo "Loongson3 Building without LTO ..."
+    CFLAGS="$CFLAGS -fno-lto"
+    CXXFLAGS="$CXXFLAGS -fno-lto"
+    CPPFLAGS="$CPPFLAGS -fno-lto"
+fi
+
 sdh_configure --disable-gui --disable-ao "$DISABLENLS" --enable-png=no --disable-samplerate --disable-static --disable-micropy --disable-quickjs
 
 #############################################################
-- 
2.52.0

