diff -up pciutils-3.9.0/lib/internal.h.extvisible pciutils-3.9.0/lib/internal.h
--- pciutils-3.9.0/lib/internal.h.extvisible	2022-12-16 19:15:16.914850590 +0100
+++ pciutils-3.9.0/lib/internal.h	2022-12-16 19:15:38.248989198 +0100
@@ -20,7 +20,7 @@
 #define SYMBOL_VERSION(_int, _ext)
 #else
 #define STATIC_ALIAS(_decl, _for)
-#define DEFINE_ALIAS(_decl, _for) extern _decl __attribute__((alias(#_for)))
+#define DEFINE_ALIAS(_decl, _for) extern _decl __attribute__((alias(#_for),visibility("default"),externally_visible))
 #ifdef _WIN32
 /* GCC does not support asm .symver directive for Windows targets, so define new external global function symbol as alias to internal symbol */
 #define SYMBOL_VERSION(_int, _ext) asm(".globl\t" PCI_STRINGIFY(__MINGW_USYMBOL(_ext)) "\n\t" \
