diff --git a/CMakeLists.txt b/CMakeLists.txt
index f731916d8..16cd04a0a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -367,7 +367,7 @@ if(TRITON_BUILD_PYTHON_MODULE)
 
     # LLVM
     LLVMPasses
-    LLVMNVPTXCodeGen
+    # LLVMNVPTXCodeGen
     # LLVMNVPTXAsmPrinter
     LLVMAMDGPUCodeGen
     LLVMAMDGPUAsmParser
@@ -393,6 +393,11 @@ if(TRITON_BUILD_PYTHON_MODULE)
         LLVMPowerPCAsmParser
         LLVMPowerPCCodeGen
       )
+  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "loongarch64")
+      list(APPEND TRITON_LIBRARIES
+        LLVMLoongArchAsmParser
+        LLVMLoongArchCodeGen
+      )
   else()
     message(FATAL_ERROR "LLVM codegen/ASM parser libs: This HW architecture (${CMAKE_SYSTEM_PROCESSOR}) is not configured in cmake lib dependencies.")
   endif()
diff --git a/third_party/proton/csrc/lib/Driver/CMakeLists.txt b/third_party/proton/csrc/lib/Driver/CMakeLists.txt
index 438f24f49..eb6b320cc 100644
--- a/third_party/proton/csrc/lib/Driver/CMakeLists.txt
+++ b/third_party/proton/csrc/lib/Driver/CMakeLists.txt
@@ -1,9 +1,9 @@
 add_proton_library(ProtonDriver
   Device.cpp
-  GPU/CudaApi.cpp
-  GPU/CuptiApi.cpp
+#  GPU/CudaApi.cpp
+#  GPU/CuptiApi.cpp
   GPU/HipApi.cpp
   GPU/HsaApi.cpp
-  GPU/RoctracerApi.cpp
-  GPU/NvtxApi.cpp
+#  GPU/RoctracerApi.cpp
+#  GPU/NvtxApi.cpp
 )
diff --git a/third_party/proton/csrc/lib/Driver/Device.cpp b/third_party/proton/csrc/lib/Driver/Device.cpp
index 24f4d1612..13f46c8e4 100644
--- a/third_party/proton/csrc/lib/Driver/Device.cpp
+++ b/third_party/proton/csrc/lib/Driver/Device.cpp
@@ -1,5 +1,5 @@
 #include "Device.h"
-#include "Driver/GPU/CudaApi.h"
+//#include "Driver/GPU/CudaApi.h"
 #include "Driver/GPU/HipApi.h"
 
 #include "Utility/Errors.h"
@@ -7,9 +7,9 @@
 namespace proton {
 
 Device getDevice(DeviceType type, uint64_t index) {
-  if (type == DeviceType::CUDA) {
-    return cuda::getDevice(index);
-  }
+//  if (type == DeviceType::CUDA) {
+//    return cuda::getDevice(index);
+//  }
   if (type == DeviceType::HIP) {
     return hip::getDevice(index);
   }
@@ -17,9 +17,10 @@ Device getDevice(DeviceType type, uint64_t index) {
 }
 
 const std::string getDeviceTypeString(DeviceType type) {
-  if (type == DeviceType::CUDA) {
-    return DeviceTraits<DeviceType::CUDA>::name;
-  } else if (type == DeviceType::HIP) {
+//  if (type == DeviceType::CUDA) {
+//    return DeviceTraits<DeviceType::CUDA>::name;
+//  } else
+  if (type == DeviceType::HIP) {
     return DeviceTraits<DeviceType::HIP>::name;
   }
   throw std::runtime_error("DeviceType not supported");
diff --git a/third_party/proton/csrc/lib/Profiler/CMakeLists.txt b/third_party/proton/csrc/lib/Profiler/CMakeLists.txt
index 00dcaef97..f766965ca 100644
--- a/third_party/proton/csrc/lib/Profiler/CMakeLists.txt
+++ b/third_party/proton/csrc/lib/Profiler/CMakeLists.txt
@@ -2,8 +2,8 @@ add_proton_library(ProtonProfiler
   Profiler.cpp
   GPUProfiler.cpp
   Graph.cpp
-  Cupti/CuptiPCSampling.cpp
-  Cupti/CuptiProfiler.cpp
+#  Cupti/CuptiPCSampling.cpp
+#  Cupti/CuptiProfiler.cpp
   RocTracer/RoctracerProfiler.cpp
   Instrumentation/InstrumentationProfiler.cpp
   Instrumentation/Metadata.cpp
diff --git a/third_party/proton/csrc/lib/Runtime/CMakeLists.txt b/third_party/proton/csrc/lib/Runtime/CMakeLists.txt
index 8c8e93e35..7cc52e719 100644
--- a/third_party/proton/csrc/lib/Runtime/CMakeLists.txt
+++ b/third_party/proton/csrc/lib/Runtime/CMakeLists.txt
@@ -1,4 +1,4 @@
 add_proton_library(ProtonRuntime
-  CudaRuntime.cpp
+#  CudaRuntime.cpp
   HipRuntime.cpp
 )
