--- a/llvm/lib/Analysis/ConstantFolding.cpp     2026-01-05 03:01:19.301122220 +0000
+++ b/llvm/lib/Analysis/ConstantFolding.cpp     2026-01-05 02:13:54.290355620 +0000
@@ -1982,7 +1982,7 @@
 }
 
 #if defined(HAS_IEE754_FLOAT128) && defined(HAS_LOGF128)
-Constant *ConstantFoldFP128(float128 (*NativeFP)(float128), const APFloat &V,
+Constant *ConstantFoldFP128(_Float128 (*NativeFP)(_Float128), const APFloat &V,
                             Type *Ty) {
   llvm_fenv_clearexcept();
   float128 Result = NativeFP(V.convertToQuad());
