From 885879ae6d91a261a4f74d43b2c1d64b73e5bd7b Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 13 Jul 2026 12:32:16 +0800
Subject: [PATCH 1/9] AOSCOS: doc/siegerc.in: set default URLs file path

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 doc/siegerc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/siegerc.in b/doc/siegerc.in
index 07e9346..26d49c8 100644
--- a/doc/siegerc.in
+++ b/doc/siegerc.in
@@ -317,7 +317,7 @@ concurrent = 25
 # NOTE: The command line -f FILE / --file=FILE takes precedence over 
 # this directive
 #
-# file =
+file = /etc/urls.txt
 
 #
 # Default URL, this is a single URL that you want to test. This is 
-- 
2.52.0

