From ebbead794ce7d1272c70cb2bcd330f293f86496a Mon Sep 17 00:00:00 2001
From: Alexandr Nedvedicky <sashan@openssl.org>
Date: Wed, 11 Feb 2026 12:48:53 +0100
Subject: [PATCH 2/9] UPSTREAM: - add missing detection for strcasecmp()

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index a187f6d..9dd1bcc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -325,7 +325,7 @@ dnl Checks for library functions.
 dnl
 AC_FUNC_ALLOCA
 AC_CHECK_FUNCS(strchr memcpy strncpy strstr strlen)
-AC_CHECK_FUNCS(strncasecmp strncmp)
+AC_CHECK_FUNCS(strcasecmp strncasecmp strncmp)
 AC_CHECK_FUNCS(socket, , AC_CHECK_LIB(socket, socket))
 AC_CHECK_FUNCS(gethostbyname, , AC_CHECK_LIB(nsl, gethostbyname))
 AC_CHECK_FUNCS(snprintf)
-- 
2.52.0

