From 47130956b1e5d3ee44424f08f860a214c5720594 Mon Sep 17 00:00:00 2001
From: Denis Fateyev <denis@fateyev.com>
Date: Fri, 12 Jan 2024 04:24:30 +0600
Subject: [PATCH 4/9] UPSTREAM: Fix bindir value passing in script

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 configure.ac | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9dd1bcc..77b6900 100644
--- a/configure.ac
+++ b/configure.ac
@@ -481,7 +481,7 @@ AC_OUTPUT_COMMANDS([
   echo "exit"     >> $outfile
   echo ""         >> $outfile
  
-], [ bindir=$exec_prefix$bindir sh=$SHELL ]) 
+], [ bindir=$bindir sh=$SHELL ])
 
 dnl
 dnl create bombardment utility
@@ -494,7 +494,7 @@ AC_OUTPUT_COMMANDS([
       -e "s|%_SHELL%|$sh|" \
       < $infile > $outfile
  
-], [ bindir=$exec_prefix$bindir sh=$SHELL ]) 
+], [ bindir=$bindir sh=$SHELL ])
 
 dnl
 dnl create siege2csv utility
@@ -507,7 +507,7 @@ AC_OUTPUT_COMMANDS([
       -e "s|%_PERL%|$LREP|" \
       < $infile > $outfile
  
-], [ bindir=$exec_prefix$bindir LREP=$PERL ]) 
+], [ bindir=$bindir LREP=$PERL ])
 
 dnl
 dnl Write platform to file for support reporting
-- 
2.52.0

