From a86f132a603c9a47310ae7d03759997580697c99 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bj=C3=B6rn=20Esser?= <besser82@fedoraproject.org>
Date: Thu, 13 Feb 2025 14:17:25 +0100
Subject: [PATCH 9/9] UPSTREAM: Update bootstrap script
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Björn Esser <besser82@fedoraproject.org>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 utils/bootstrap | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/utils/bootstrap b/utils/bootstrap
index de49f63..023d8df 100755
--- a/utils/bootstrap
+++ b/utils/bootstrap
@@ -6,10 +6,7 @@
 # part of siege distribution
 # automates the autotools
 set -x
-aclocal 
-autoheader
-automake --foreign --copy
-autoconf
+autoreconf --force --install --verbose --warnings=no-syntax
 utils/manifier doc/siege.pod        doc/siege.1.in        'Siege Load Tester' 1
 utils/manifier doc/siege.config.pod doc/siege.config.1.in 'siege.config utility' 1
 utils/manifier doc/bombardment.pod  doc/bombardment.1.in  'bombardment' 1
-- 
2.52.0

