diff -Naur ProxyMan.orig/bash.sh ProxyMan/bash.sh
--- ProxyMan.orig/bash.sh	2018-05-27 18:14:48.940151601 +0800
+++ ProxyMan/bash.sh	2018-05-27 18:33:43.360201321 +0800
@@ -85,31 +85,31 @@
 		var="$5:$6@"
 	fi
 
-	echo -n "" > bash_config.tmp
+	echo -n "" > /tmp/bash_config.tmp
 	if [ "$3" = "y" ]; then
 		newvar="://$var$1:$2"
-		echo "http_proxy=\"http$newvar\"" >> bash_config.tmp
-		echo "https_proxy=\"http$newvar\"" >> bash_config.tmp
-		echo "ftp_proxy=\"ftp$newvar\"" >> bash_config.tmp
-		echo "HTTP_PROXY=\"http$newvar\"" >> bash_config.tmp
-		echo "HTTPS_PROXY=\"http$newvar\"" >> bash_config.tmp
-		echo "FTP_PROXY=\"ftp$newvar\"" >> bash_config.tmp
+		echo "http_proxy=\"http$newvar\"" >> /tmp/bash_config.tmp
+		echo "https_proxy=\"http$newvar\"" >> /tmp/bash_config.tmp
+		echo "ftp_proxy=\"ftp$newvar\"" >> /tmp/bash_config.tmp
+		echo "HTTP_PROXY=\"http$newvar\"" >> /tmp/bash_config.tmp
+		echo "HTTPS_PROXY=\"http$newvar\"" >> /tmp/bash_config.tmp
+		echo "FTP_PROXY=\"ftp$newvar\"" >> /tmp/bash_config.tmp
 
     fix_new_line $HOME/.bashrc
-		cat bash_config.tmp | tee -a $HOME/.bashrc > /dev/null
-		rm bash_config.tmp
+		cat /tmp/bash_config.tmp | tee -a $HOME/.bashrc > /dev/null
+		rm /tmp/bash_config.tmp
 		return
 
 	elif [ "$3" = "n" ]; then
-		echo "http_proxy=\"http://$var$1:$2\"" >> bash_config.tmp
-		echo "http_proxy=\"http://$var$7:$8\"" >> bash_config.tmp
-		echo "ftp_proxy=\"ftp://$var$9:$10\"" >> bash_config.tmp
-		echo "HTTP_PROXY=\"http://$var$1:$2\"" >> bash_config.tmp
-		echo "HTTPS_PROXY=\"http://$var$7:$8\"" >> bash_config.tmp
-		echo "FTP_PROXY=\"ftp://$var$9:$10\"" >> bash_config.tmp
+		echo "http_proxy=\"http://$var$1:$2\"" >> /tmp/bash_config.tmp
+		echo "http_proxy=\"http://$var$7:$8\"" >> /tmp/bash_config.tmp
+		echo "ftp_proxy=\"ftp://$var$9:$10\"" >> /tmp/bash_config.tmp
+		echo "HTTP_PROXY=\"http://$var$1:$2\"" >> /tmp/bash_config.tmp
+		echo "HTTPS_PROXY=\"http://$var$7:$8\"" >> /tmp/bash_config.tmp
+		echo "FTP_PROXY=\"ftp://$var$9:$10\"" >> /tmp/bash_config.tmp
 
-		cat bash_config.tmp | tee -a $HOME/.bashrc > /dev/null
-		rm bash_config.tmp
+		cat /tmp/bash_config.tmp | tee -a $HOME/.bashrc > /dev/null
+		rm /tmp/bash_config.tmp
 		return
 	fi
 }
diff -Naur ProxyMan.orig/main.sh ProxyMan/main.sh
--- ProxyMan.orig/main.sh	2018-05-27 18:19:44.353497882 +0800
+++ ProxyMan/main.sh	2018-05-27 18:34:15.850202746 +0800
@@ -39,6 +39,12 @@
 
 # here your code starts
 
+# Create profile dir
+
+if [[ ! -d ~/.proxyman/profiles/ ]]; then
+	mkdir -pv ~/.proxyman/profiles/
+fi
+
 # This is the main script which calls other respective scripts.
 # In case of doubts regarding how to use, refer the README.md file.
 clear
@@ -158,7 +164,7 @@
 
         if [[ "$1" != "load"  && ( "$save_for_reuse" = "y" || "$save_for_reuse" = "Y" ) ]]; then
             config_file="http_host=$http_host%s\nhttp_port=$http_port%s\nuse_same=$use_same\nuse_auth=$use_auth\nusername=$username\npassword=$password\nhttps_host=$https_host\nhttps_port=$https_port\nftp_host=$ftp_host\nftp_port=$ftp_port"
-            printf $config_file > "profiles/$profile_name".txt
+            printf $config_file > ~/.proxyman/profiles/"$profile_name".txt
         fi
 
         if [[ $choice == "load" || $1 == "load" ]]; then
@@ -178,22 +184,22 @@
                 config_name=$2
             fi
     
-            if [ ! -e profiles/"$config_name".txt ]; then
+            if [ ! -e ~/.proxyman/profiles/"$config_name".txt ]; then
                 echo -ne "\e[1m \e[31mFile does not exist! \e[0m"
                 echo
                 exit
             fi
 
-            http_host=`grep http_host -i profiles/$config_name.txt  | cut -d= -f2`
-            http_port=`grep http_port -i profiles/$config_name.txt  | cut -d= -f2`
-            use_same=`grep use_same -i profiles/$config_name.txt  | cut -d= -f2`
-            use_auth=`grep use_auth -i profiles/$config_name.txt  | cut -d= -f2`
-            username=`grep username -i profiles/$config_name.txt  | cut -d= -f2`
-            password=`grep password -i profiles/$config_name.txt  | cut -d= -f2`
-            https_host=`grep https_host -i profiles/$config_name.txt  | cut -d= -f2`
-            https_port=`grep https_port -i profiles/$config_name.txt  | cut -d= -f2`
-            ftp_host=`grep ftp_host -i profiles/$config_name.txt  | cut -d= -f2`
-            ftp_port=`grep ftp_port -i profiles/$config_name.txt  | cut -d= -f2`
+            http_host=`grep http_host -i ~/.proxyman/profiles/$config_name.txt  | cut -d= -f2`
+            http_port=`grep http_port -i ~/.proxyman/profiles/$config_name.txt  | cut -d= -f2`
+            use_same=`grep use_same -i ~/.proxyman/profiles/$config_name.txt  | cut -d= -f2`
+            use_auth=`grep use_auth -i ~/.proxyman/profiles/$config_name.txt  | cut -d= -f2`
+            username=`grep username -i ~/.proxyman/profiles/$config_name.txt  | cut -d= -f2`
+            password=`grep password -i ~/.proxyman/profiles/$config_name.txt  | cut -d= -f2`
+            https_host=`grep https_host -i ~/.proxyman/profiles/$config_name.txt  | cut -d= -f2`
+            https_port=`grep https_port -i ~/.proxyman/profiles/$config_name.txt  | cut -d= -f2`
+            ftp_host=`grep ftp_host -i ~/.proxyman/profiles/$config_name.txt  | cut -d= -f2`
+            ftp_port=`grep ftp_port -i ~/.proxyman/profiles/$config_name.txt  | cut -d= -f2`
 
             echo -e " Config \033[0;36m$config_name\033[0m successfully loaded"
         fi
