diff -Naur fontconfig/conf.d/10-autohint.conf fontconfig.rev/conf.d/10-autohint.conf
--- fontconfig/conf.d/10-autohint.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-autohint.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Enable autohinter</description>
 <!--  Use the Autohinter --> 
   <match target="pattern">
     <!--
diff -Naur fontconfig/conf.d/10-hinting-full.conf fontconfig.rev/conf.d/10-hinting-full.conf
--- fontconfig/conf.d/10-hinting-full.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-hinting-full.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,12 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Set hintfull to hintstyle</description>
-
   <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
diff -Naur fontconfig/conf.d/10-hinting-medium.conf fontconfig.rev/conf.d/10-hinting-medium.conf
--- fontconfig/conf.d/10-hinting-medium.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-hinting-medium.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,12 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Set hintmedium to hintstyle</description>
-
   <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
diff -Naur fontconfig/conf.d/10-hinting-none.conf fontconfig.rev/conf.d/10-hinting-none.conf
--- fontconfig/conf.d/10-hinting-none.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-hinting-none.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,12 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Set hintnone to hintstyle</description>
-
   <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
diff -Naur fontconfig/conf.d/10-hinting-slight.conf fontconfig.rev/conf.d/10-hinting-slight.conf
--- fontconfig/conf.d/10-hinting-slight.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-hinting-slight.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,12 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Set hintslight to hintstyle</description>
-
   <match target="pattern">
     <!--
       This configuration is available on the major desktop environments.
diff -Naur fontconfig/conf.d/10-no-sub-pixel.conf fontconfig.rev/conf.d/10-no-sub-pixel.conf
--- fontconfig/conf.d/10-no-sub-pixel.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-no-sub-pixel.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Disable sub-pixel rendering</description>
 <!--  Disable sub-pixel rendering --> 
   <match target="pattern">
     <!--
diff -Naur fontconfig/conf.d/10-scale-bitmap-fonts.conf fontconfig.rev/conf.d/10-scale-bitmap-fonts.conf
--- fontconfig/conf.d/10-scale-bitmap-fonts.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-scale-bitmap-fonts.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,11 +1,7 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
 
-  <description>Bitmap scaling</description>
 <!--
   If font is bitmap, calculate scale factor.
   Note that color bitmap fonts have scalable=true, while
diff -Naur fontconfig/conf.d/10-sub-pixel-bgr.conf fontconfig.rev/conf.d/10-sub-pixel-bgr.conf
--- fontconfig/conf.d/10-sub-pixel-bgr.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-sub-pixel-bgr.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Enable sub-pixel rendering with the BGR stripes layout</description>
 <!--  Enable sub-pixel rendering --> 
   <match target="pattern">
     <!--
diff -Naur fontconfig/conf.d/10-sub-pixel-rgb.conf fontconfig.rev/conf.d/10-sub-pixel-rgb.conf
--- fontconfig/conf.d/10-sub-pixel-rgb.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-sub-pixel-rgb.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Enable sub-pixel rendering with the RGB stripes layout</description>
 <!--  Enable sub-pixel rendering --> 
   <match target="pattern">
     <!--
diff -Naur fontconfig/conf.d/10-sub-pixel-vbgr.conf fontconfig.rev/conf.d/10-sub-pixel-vbgr.conf
--- fontconfig/conf.d/10-sub-pixel-vbgr.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-sub-pixel-vbgr.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Enable sub-pixel rendering with the vertical BGR stripes layout</description>
 <!--  Enable sub-pixel rendering --> 
   <match target="pattern">
     <!--
diff -Naur fontconfig/conf.d/10-sub-pixel-vrgb.conf fontconfig.rev/conf.d/10-sub-pixel-vrgb.conf
--- fontconfig/conf.d/10-sub-pixel-vrgb.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-sub-pixel-vrgb.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Enable sub-pixel rendering with the vertical RGB stripes layout</description>
 <!--  Enable sub-pixel rendering --> 
   <match target="pattern">
     <!--
diff -Naur fontconfig/conf.d/10-unhinted.conf fontconfig.rev/conf.d/10-unhinted.conf
--- fontconfig/conf.d/10-unhinted.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/10-unhinted.conf	2018-04-09 21:30:12.331222902 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Disable hinting</description>
 <!--  Disable hinting --> 
   <match target="pattern">
     <!--
diff -Naur fontconfig/conf.d/11-lcdfilter-default.conf fontconfig.rev/conf.d/11-lcdfilter-default.conf
--- fontconfig/conf.d/11-lcdfilter-default.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/11-lcdfilter-default.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Use lcddefault as default for LCD filter</description>
 <!--  Use lcddefault as default for LCD filter -->
   <match target="pattern">
     <!--
diff -Naur fontconfig/conf.d/11-lcdfilter-legacy.conf fontconfig.rev/conf.d/11-lcdfilter-legacy.conf
--- fontconfig/conf.d/11-lcdfilter-legacy.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/11-lcdfilter-legacy.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Use lcdlegacy as default for LCD filter</description>
 <!--  Use lcdlegacy as default for LCD filter -->
   <match target="pattern">
     <!--
diff -Naur fontconfig/conf.d/11-lcdfilter-light.conf fontconfig.rev/conf.d/11-lcdfilter-light.conf
--- fontconfig/conf.d/11-lcdfilter-light.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/11-lcdfilter-light.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Use lcdlight as default for LCD filter</description>
 <!--  Use lcdlight as default for LCD filter -->
   <match target="pattern">
     <!--
diff -Naur fontconfig/conf.d/20-unhint-small-vera.conf fontconfig.rev/conf.d/20-unhint-small-vera.conf
--- fontconfig/conf.d/20-unhint-small-vera.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/20-unhint-small-vera.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Disable hinting for Bitstream Vera fonts when the size is less than 8ppem</description>
 <!-- 
 	The Bitstream Vera fonts have GASP entries suggesting that hinting be
 	disabled below 8 ppem, but FreeType ignores those, preferring to use
diff -Naur fontconfig/conf.d/25-unhint-nonlatin.conf fontconfig.rev/conf.d/25-unhint-nonlatin.conf
--- fontconfig/conf.d/25-unhint-nonlatin.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/25-unhint-nonlatin.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,7 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
 
-  <description>Disable hinting for CJK fonts</description>
 <!-- We can't hint CJK fonts well, so turn off hinting for CJK fonts. -->
 
 	<match target="font">
diff -Naur fontconfig/conf.d/30-metric-aliases.conf fontconfig.rev/conf.d/30-metric-aliases.conf
--- fontconfig/conf.d/30-metric-aliases.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/30-metric-aliases.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Set substitutions for similar/metric-compatible families</description>
 
 <!--
 
diff -Naur fontconfig/conf.d/40-nonlatin.conf fontconfig.rev/conf.d/40-nonlatin.conf
--- fontconfig/conf.d/40-nonlatin.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/40-nonlatin.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Set substitutions for non-Latin fonts</description>
 <!--
   Mark common families with their generics so we'll get
   something reasonable
diff -Naur fontconfig/conf.d/45-generic.conf fontconfig.rev/conf.d/45-generic.conf
--- fontconfig/conf.d/45-generic.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/45-generic.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,12 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Set substitutions for emoji/math fonts</description>
-
 <!-- Keep in sync with 60-generic.conf -->
 
 <!-- Emoji -->
diff -Naur fontconfig/conf.d/45-latin.conf fontconfig.rev/conf.d/45-latin.conf
--- fontconfig/conf.d/45-latin.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/45-latin.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Set substitutions for Latin fonts</description>
 <!--
   Mark common families with their generics so we'll get
   something reasonable
diff -Naur fontconfig/conf.d/49-sansserif.conf fontconfig.rev/conf.d/49-sansserif.conf
--- fontconfig/conf.d/49-sansserif.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/49-sansserif.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Add sans-serif to the family when no generic name</description>
 <!--
   If the font still has no generic name, add sans-serif
  -->
diff -Naur fontconfig/conf.d/50-user.conf fontconfig.rev/conf.d/50-user.conf
--- fontconfig/conf.d/50-user.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/50-user.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Load per-user customization files</description>
 	<!--
 	    Load per-user customization files where stored on XDG Base Directory
 	    specification compliant places. it should be usually:
diff -Naur fontconfig/conf.d/51-local.conf fontconfig.rev/conf.d/51-local.conf
--- fontconfig/conf.d/51-local.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/51-local.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Load local customization file</description>
 	<!-- Load local system customization file -->
 	<include ignore_missing="yes">local.conf</include>
 </fontconfig>
diff -Naur fontconfig/conf.d/60-generic.conf fontconfig.rev/conf.d/60-generic.conf
--- fontconfig/conf.d/60-generic.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/60-generic.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Set preferable fonts for emoji/math fonts</description>
 <!-- Keep in sync with 45-generic.conf -->
 
 <!-- Emoji -->
diff -Naur fontconfig/conf.d/60-latin.conf fontconfig.rev/conf.d/60-latin.conf
--- fontconfig/conf.d/60-latin.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/60-latin.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Set preferable fonts for Latin</description>
 	<alias>
 		<family>serif</family>
 		<prefer>
diff -Naur fontconfig/conf.d/65-fonts-persian.conf fontconfig.rev/conf.d/65-fonts-persian.conf
--- fontconfig/conf.d/65-fonts-persian.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/65-fonts-persian.conf	2018-04-09 21:30:12.334556235 +0000
@@ -31,10 +31,6 @@
     2008	Behdad Esfahbod:  Cleanup.  Add fantasy and cursive.
   -->
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
 
 
 <!-- Deprecated fonts are discouraged -->
diff -Naur fontconfig/conf.d/65-khmer.conf fontconfig.rev/conf.d/65-khmer.conf
--- fontconfig/conf.d/65-khmer.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/65-khmer.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,10 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
 	<alias>
 		<family>serif</family>
 		<prefer>
diff -Naur fontconfig/conf.d/65-nonlatin.conf fontconfig.rev/conf.d/65-nonlatin.conf
--- fontconfig/conf.d/65-nonlatin.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/65-nonlatin.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Set preferable fonts for non-Latin</description>
 	<alias>
 		<family>serif</family>
 		<prefer>
diff -Naur fontconfig/conf.d/69-unifont.conf fontconfig.rev/conf.d/69-unifont.conf
--- fontconfig/conf.d/69-unifont.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/69-unifont.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,10 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
 	<alias>
 		<family>serif</family>
 		<prefer>
diff -Naur fontconfig/conf.d/70-no-bitmaps.conf fontconfig.rev/conf.d/70-no-bitmaps.conf
--- fontconfig/conf.d/70-no-bitmaps.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/70-no-bitmaps.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Reject bitmap fonts</description>
 <!-- Reject bitmap fonts -->
  <selectfont>
   <rejectfont>
diff -Naur fontconfig/conf.d/70-yes-bitmaps.conf fontconfig.rev/conf.d/70-yes-bitmaps.conf
--- fontconfig/conf.d/70-yes-bitmaps.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/70-yes-bitmaps.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,11 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
-  <description>Accept bitmap fonts</description>
 <!-- Accept bitmap fonts -->
  <selectfont>
   <acceptfont>
diff -Naur fontconfig/conf.d/80-delicious.conf fontconfig.rev/conf.d/80-delicious.conf
--- fontconfig/conf.d/80-delicious.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/80-delicious.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,10 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
 <!-- Fix-ups for Delicious family -->
 
 <!-- Delicious 'heavy' variant says its Medium weight -->
diff -Naur fontconfig/conf.d/90-synthetic.conf fontconfig.rev/conf.d/90-synthetic.conf
--- fontconfig/conf.d/90-synthetic.conf	2018-04-09 21:29:12.397889275 +0000
+++ fontconfig.rev/conf.d/90-synthetic.conf	2018-04-09 21:30:12.334556235 +0000
@@ -1,10 +1,6 @@
 <?xml version="1.0"?>
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <fontconfig>
-  <its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-    <its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-  </its:rules>
-
 <!-- 
  Artificial oblique for fonts without an italic or oblique version
  -->
diff -Naur fontconfig/fonts.conf.in fontconfig.rev/fonts.conf.in
--- fontconfig/fonts.conf.in	2018-04-09 21:29:12.404555942 +0000
+++ fontconfig.rev/fonts.conf.in	2018-04-09 21:30:21.171222937 +0000
@@ -2,11 +2,6 @@
 <!DOCTYPE fontconfig SYSTEM "fonts.dtd">
 <!-- /etc/fonts/fonts.conf file to configure system font access -->
 <fontconfig>
-	<its:rules xmlns:its="http://www.w3.org/2005/11/its" version="1.0">
-		<its:translateRule translate="no" selector="/fontconfig/*[not(self::description)]"/>
-	</its:rules>
-
-	<description>Default configuration file</description>
 
 <!--
 	DO NOT EDIT THIS FILE.
