Index: fontconfig-2.11.0/src/fclang.c
===================================================================
--- fontconfig-2.11.0.orig/src/fclang.c	2013-11-23 19:51:18.830649663 +0100
+++ fontconfig-2.11.0/src/fclang.c	2013-11-23 19:51:37.697966141 +0100
@@ -189,6 +189,8 @@ FcLangNormalize (const FcChar8 *lang)
 	return NULL;
 
     if (FcStrCmpIgnoreCase (lang, (const FcChar8 *)"C") == 0 ||
+	FcStrCmpIgnoreCase (lang, (const FcChar8 *)"C.UTF-8") == 0 ||
+	FcStrCmpIgnoreCase (lang, (const FcChar8 *)"C.utf8") == 0 ||
 	FcStrCmpIgnoreCase (lang, (const FcChar8 *)"POSIX") == 0)
     {
 	result = FcStrCopy ((const FcChar8 *)"en");
