From 456d30bad5efd7b1daf33e503966773ab18bcc60 Mon Sep 17 00:00:00 2001
From: Cyan <cyan@cyano.uk>
Date: Wed, 20 Sep 2023 18:46:38 -0700
Subject: [PATCH 1/4] plugins/amdgpu/Makefile: add missing defines

---
 plugins/amdgpu/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/amdgpu/Makefile b/plugins/amdgpu/Makefile
index a20d1d163..20bb0e347 100644
--- a/plugins/amdgpu/Makefile
+++ b/plugins/amdgpu/Makefile
@@ -28,7 +28,7 @@ criu-amdgpu.pb-c.c: criu-amdgpu.proto
 		protoc-c --proto_path=. --c_out=. criu-amdgpu.proto
 
 amdgpu_plugin.so: amdgpu_plugin.c amdgpu_plugin_drm.c amdgpu_plugin_topology.c amdgpu_plugin_util.c criu-amdgpu.pb-c.c
-	$(CC) $(PLUGIN_CFLAGS) $(shell $(COMPEL) includes) $^ -o $@ $(PLUGIN_INCLUDE) $(PLUGIN_LDFLAGS) $(LIBDRM_INC)
+	$(CC) $(DEFINES) $(PLUGIN_CFLAGS) $(shell $(COMPEL) includes) $^ -o $@ $(PLUGIN_INCLUDE) $(PLUGIN_LDFLAGS) $(LIBDRM_INC)
 
 amdgpu_plugin_clean:
 	$(call msg-clean, $@)
-- 
2.51.0

