From fbe9855d513deaa65528fb7a962001aeba8cfce6 Mon Sep 17 00:00:00 2001
From: Cyan <cyan@cyano.uk>
Date: Wed, 20 Sep 2023 18:46:38 -0700
Subject: [PATCH] plugins/amdgpu/Makefile: add missing defines

---
 plugins/amdgpu/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/amdgpu/Makefile b/plugins/amdgpu/Makefile
index 64a923d..d5c2652 100644
--- a/plugins/amdgpu/Makefile
+++ b/plugins/amdgpu/Makefile
@@ -29,7 +29,7 @@ criu-amdgpu.pb-c.c: criu-amdgpu.proto
 		protoc-c --proto_path=. --c_out=. criu-amdgpu.proto
 
 amdgpu_plugin.so: amdgpu_plugin.c amdgpu_plugin_topology.c criu-amdgpu.pb-c.c
-	$(CC) $(PLUGIN_CFLAGS) $(shell $(COMPEL) includes) $^ -o $@ $(PLUGIN_INCLUDE) $(PLUGIN_LDFLAGS) $(LIBDRM_INC)
+	$(CC) $(DEFINES) $(PLUGIN_CFLAGS) $(shell $(COMPEL) includes) $^ -o $@ $(PLUGIN_INCLUDE) $(PLUGIN_LDFLAGS) $(LIBDRM_INC)
 
 amdgpu_plugin_clean:
 	$(call msg-clean, $@)
-- 
2.39.1

