From 02014738d46869241a5e518ebcf14b70a66fbeb0 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Thu, 11 Sep 2025 21:32:26 +0800
Subject: [PATCH 2/4] use longlong on mips

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5d8e89ac1..397a0f62e 100644
--- a/Makefile
+++ b/Makefile
@@ -77,7 +77,7 @@ ifeq ($(ARCH),x86)
 endif
 
 ifeq ($(ARCH),mips)
-        DEFINES		:= -DCONFIG_MIPS
+        DEFINES		:= -DCONFIG_MIPS -D__SANE_USERSPACE_TYPES__
 endif
 
 ifeq ($(ARCH),loongarch64)
-- 
2.51.0

