From 36f89186d62aba394afd392ef1d89ab0034cabb6 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sat, 13 Sep 2025 16:07:27 +0800
Subject: [PATCH 3/4] drop versioned protobuf requirement

---
 lib/pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/pyproject.toml b/lib/pyproject.toml
index 8eb4b7084..2e0e3960f 100644
--- a/lib/pyproject.toml
+++ b/lib/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools", "protobuf<4.0"]
+requires = ["setuptools", "protobuf"]
 build-backend = "setuptools.build_meta"
 
 [project]
-- 
2.51.0

