--- bsd-finger-0.17/finger/Makefile.strip	Wed Jun 19 15:22:39 2002
+++ bsd-finger-0.17/finger/Makefile	Wed Jun 19 15:25:11 2002
@@ -11,7 +11,7 @@
 
 install: finger
 	install -d $(INSTALLROOT)$(BINDIR)
-	install -s -m$(BINMODE) finger $(INSTALLROOT)$(BINDIR)
+	install -m$(BINMODE) finger $(INSTALLROOT)$(BINDIR)
 	install -d $(INSTALLROOT)$(MANDIR)/man1
 	install -m$(MANMODE) finger.1 $(INSTALLROOT)$(MANDIR)/man1
 
--- bsd-finger-0.17/fingerd/Makefile.strip	Wed Jun 19 15:22:39 2002
+++ bsd-finger-0.17/fingerd/Makefile	Wed Jun 19 15:25:27 2002
@@ -10,7 +10,7 @@
 
 install: fingerd
 	install -d $(INSTALLROOT)$(SBINDIR)
-	install -s -m$(DAEMONMODE) fingerd $(INSTALLROOT)$(SBINDIR)/in.fingerd
+	install -m$(DAEMONMODE) fingerd $(INSTALLROOT)$(SBINDIR)/in.fingerd
 	install -d $(INSTALLROOT)$(MANDIR)/man8
 	install -m$(MANMODE) fingerd.8 $(INSTALLROOT)$(MANDIR)/man8/in.fingerd.8
 	ln -sf in.fingerd.8 $(INSTALLROOT)$(MANDIR)/man8/fingerd.8
