diff --git a/Makefile.in b/Makefile.in
index e822b04..a0574bf 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -41,7 +41,7 @@ install-conf:
 	done; \
 	for j in $(top_srcdir)/docs/conf $(top_builddir)/docs/conf ; do \
 	    cd $$j ; \
-	    for i in httpd.conf extra/*.conf; do \
+	    for i in httpd.conf modules.conf extra/*.conf; do \
 	    	if [ -f $$i ] ; then \
 	    	( \
 	    		n_lm=`awk 'BEGIN {n=0} /@@LoadModule@@/ {n+=1} END {print n}' < $$i`; \
diff --git a/configure.in b/configure.in
index 62ed7b0..209b4ef 100644
--- a/configure.in
+++ b/configure.in
@@ -937,7 +937,7 @@ AC_SUBST(ap_make_delimiter)
 dnl Ensure that docs/conf is created.
 test -d docs/conf||$mkdir_p docs/conf
 
-AC_CONFIG_FILES(docs/conf/httpd.conf docs/conf/extra/httpd-autoindex.conf docs/conf/extra/httpd-dav.conf docs/conf/extra/httpd-default.conf docs/conf/extra/httpd-info.conf docs/conf/extra/httpd-languages.conf docs/conf/extra/httpd-manual.conf docs/conf/extra/httpd-mpm.conf docs/conf/extra/httpd-multilang-errordoc.conf docs/conf/extra/httpd-ssl.conf docs/conf/extra/httpd-userdir.conf docs/conf/extra/httpd-vhosts.conf docs/conf/extra/proxy-html.conf include/ap_config_layout.h support/apxs support/apachectl support/dbmmanage support/envvars-std support/log_server_status support/logresolve.pl support/phf_abuse_log.cgi support/split-logfile build/rules.mk build/pkg/pkginfo build/config_vars.sh)
+AC_CONFIG_FILES(docs/conf/httpd.conf docs/conf/modules.conf docs/conf/extra/httpd-autoindex.conf docs/conf/extra/httpd-dav.conf docs/conf/extra/httpd-default.conf docs/conf/extra/httpd-info.conf docs/conf/extra/httpd-languages.conf docs/conf/extra/httpd-manual.conf docs/conf/extra/httpd-mpm.conf docs/conf/extra/httpd-multilang-errordoc.conf docs/conf/extra/httpd-ssl.conf docs/conf/extra/httpd-userdir.conf docs/conf/extra/httpd-vhosts.conf docs/conf/extra/proxy-html.conf include/ap_config_layout.h support/apxs support/apachectl support/dbmmanage support/envvars-std support/log_server_status support/logresolve.pl support/phf_abuse_log.cgi support/split-logfile build/rules.mk build/pkg/pkginfo build/config_vars.sh)
 AC_CONFIG_COMMANDS([default], [true], [APACHE_GEN_MAKEFILES])
 AC_OUTPUT
 AC_MSG_NOTICE([summary of build options:
diff --git a/docs/conf/httpd.conf.in b/docs/conf/httpd.conf.in
index 37d7c0b..489ea87 100644
--- a/docs/conf/httpd.conf.in
+++ b/docs/conf/httpd.conf.in
@@ -51,19 +51,10 @@ ServerRoot "@@ServerRoot@@"
 #Listen 12.34.56.78:80
 Listen @@Port@@
 
-#
-# Dynamic Shared Object (DSO) Support
-#
-# To be able to use the functionality of a module which was built as a DSO you
-# have to place corresponding `LoadModule' lines at this location so the
-# directives contained in it are actually available _before_ they are used.
-# Statically compiled modules (those listed by `httpd -l') do not need
-# to be loaded here.
-#
-# Example:
-# LoadModule foo_module modules/mod_foo.so
-#
-@@LoadModule@@
+# AOSC: We split the loadable modules section to a different config file.
+# To configure loadable modules, please head to `/etc/httpd/conf/modules.conf'
+# instead.
+Include @rel_sysconfdir@/modules.conf
 
 <IfModule unixd_module>
 #
@@ -74,8 +65,8 @@ Listen @@Port@@
 # It is usually good practice to create a dedicated user and group for
 # running httpd, as with most system services.
 #
-User daemon
-Group daemon
+User httpd
+Group httpd
 
 </IfModule>
 
@@ -416,3 +407,7 @@ SSLRandomSeed startup builtin
 SSLRandomSeed connect builtin
 </IfModule>
 
+# AOSC: In practice you should configure your websites as virtual hosts.
+# Please add your VirtualHost.conf to /etc/httpd/sites-available and link
+# it to /etc/httpd/sites-enabled.
+IncludeOptional /etc/httpd/sites-enabled/*.conf
