diff -Naur httpd-2.4.46.orig/support/Makefile.in httpd-2.4.46/support/Makefile.in
--- httpd-2.4.46.orig/support/Makefile.in	2018-02-09 10:17:30.000000000 +0000
+++ httpd-2.4.46/support/Makefile.in	2021-08-26 04:23:04.161224664 +0000
@@ -30,9 +30,9 @@
 	    fi ; \
 	done
 	@if test -f "$(builddir)/envvars-std"; then \
-	    cp -p envvars-std $(DESTDIR)$(sbindir); \
-	    if test ! -f $(DESTDIR)$(sbindir)/envvars; then \
-	        cp -p envvars-std $(DESTDIR)$(sbindir)/envvars ; \
+	    cp -p envvars-std $(DESTDIR)$(libexecdir); \
+	    if test ! -f $(DESTDIR)$(libexecdir)/envvars; then \
+	        cp -p envvars-std $(DESTDIR)$(libexecdir)/envvars ; \
 	    fi ; \
 	fi
 
diff -Naur httpd-2.4.46.orig/support/apachectl.in httpd-2.4.46/support/apachectl.in
--- httpd-2.4.46.orig/support/apachectl.in	2012-02-01 03:47:28.000000000 +0000
+++ httpd-2.4.46/support/apachectl.in	2021-08-26 04:23:56.299424715 +0000
@@ -45,8 +45,8 @@
 HTTPD='@exp_sbindir@/@progname@'
 #
 # pick up any necessary environment variables
-if test -f @exp_sbindir@/envvars; then
-  . @exp_sbindir@/envvars
+if test -f @exp_libexecdir@/envvars; then
+  . @exp_libexecdir@/envvars
 fi
 #
 # a command that outputs a formatted text version of the HTML at the
