diff -Naur arch-install-scripts/pacstrap.in arch-install-scripts.pacstrap/pacstrap.in
--- arch-install-scripts/pacstrap.in	2023-11-13 08:01:08.375239616 +0000
+++ arch-install-scripts.pacstrap/pacstrap.in	2023-11-13 08:01:00.810213447 +0000
@@ -52,6 +52,10 @@
   exit $(( $# ? 0 : 1 ))
 fi
 
+# We ship `pacstrap' with an extra configuration file that complies with Arch Linux,
+# as the standard package is meant for *any* Pacman distributions/repositories.
+pacman_config="/etc/pacman.conf.pacstrap"
+
 while getopts ':C:cDGiKMNPU' flag; do
   case $flag in
     C)
