diff -Naur arch-install-scripts/pacstrap.in arch-install-scripts.mirrorlist/pacstrap.in
--- arch-install-scripts/pacstrap.in	2023-11-13 08:01:08.375239616 +0000
+++ arch-install-scripts.mirrorlist/pacstrap.in	2023-11-13 08:02:27.949515487 +0000
@@ -137,7 +137,7 @@
 
   if (( copymirrorlist )); then
     # install the host's mirrorlist onto the new root
-    cp -a /etc/pacman.d/mirrorlist "$newroot/etc/pacman.d/"
+    cp -a /etc/pacman.d/mirrorlist.pacstrap "$newroot/etc/pacman.d/mirrorlist"
   fi
 
   if (( copyconf )); then
