From 25e3d287418547d1caff5c269f4ac6eaaf509a94 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Mon, 13 Nov 2023 08:12:32 +0000
Subject: [PATCH] common: add fuse.portal

---
 common | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common b/common
index 5768dd2..e3c4fe6 100644
--- a/common
+++ b/common
@@ -26,6 +26,7 @@ declare -A pseudofs_types=([anon_inodefs]=1
                            [fuse.gvfs-fuse-daemon]=1
                            [fuse.gvfsd-fuse]=1
                            [fuse.lxcfs]=1
+                           [fuse.portal]=1
                            [fuse.rofiles-fuse]=1
                            [fuse.vmware-vmblock]=1
                            [fuse.xwmfs]=1
-- 
2.39.1

