From: Stefano Zacchiroli <zack@debian.org>
Date: Tue, 20 Apr 2010 14:54:09 +0200
Subject: [PATCH] Enable installing to a target directory passed from debian/rules

Signed-off-by: Stefano Zacchiroli <zack@debian.org>
Signed-off-by: Stephane Glondu <steph@glondu.net>
---
 Makefile |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index c398c15..4f05503 100644
--- a/Makefile
+++ b/Makefile
@@ -14,8 +14,9 @@ MODS=ulexing utf16 utf8
 install: all
 	cd _build && $(MAKE) -f ../Makefile realinstall
 
+DESTDIR =
 realinstall:
-	ocamlfind install ulex ../META $(wildcard $(MODS:=.mli) $(MODS:=.cmi) $(MODS:=.cmx) pa_ulex.cma ulexing.a ulexing.cma ulexing.cmxa)
+	ocamlfind install -destdir $(DESTDIR) ulex ../META $(wildcard $(MODS:=.mli) $(MODS:=.cmi) $(MODS:=.cmx) pa_ulex.cma ulexing.a ulexing.cma ulexing.cmxa)
 
 uninstall:
 	ocamlfind remove ulex
-- 
