From 49fc84573347c4439d2af5d2e1ef717760ec4344 Mon Sep 17 00:00:00 2001
From: Errant404 <erigidissimus@gmail.com>
Date: Mon, 19 Aug 2024 22:10:09 +0800
Subject: [PATCH 1/4] replace `linux-version` script with plain shell commands

---
 u-boot-update | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/u-boot-update b/u-boot-update
index 6f39993..9b73c2c 100755
--- a/u-boot-update
+++ b/u-boot-update
@@ -113,7 +113,7 @@ timeout ${U_BOOT_TIMEOUT}
 "
 
 # Find linux versions
-_KERNELS=$(linux-version list --paths | linux-version sort --reverse | sed -e 's,.*/boot/,,g')
+_KERNELS=$(ls /boot/vmlinu* | sed -e 's,.*/boot/,,g' | sort --reverse)
 
 for _KERNEL in ${_KERNELS}
 do
-- 
2.52.0

