From 2852bbda71a64862262ce8c4abe8dbe122284572 Mon Sep 17 00:00:00 2001
From: Errant404 <erigidissimus@gmail.com>
Date: Wed, 21 Aug 2024 17:29:00 +0800
Subject: [PATCH] customize configuration files

---
 etc/default/u-boot | 3 ++-
 read-config        | 4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/etc/default/u-boot b/etc/default/u-boot
index e8ec95c..6ccc7b8 100644
--- a/etc/default/u-boot
+++ b/etc/default/u-boot
@@ -6,7 +6,7 @@
 #U_BOOT_DEFAULT="l0"
 #U_BOOT_PROMPT="1"
 #U_BOOT_ENTRIES="all"
-#U_BOOT_MENU_LABEL="Debian GNU/Linux"
+#U_BOOT_MENU_LABEL="AOSC OS"
 #U_BOOT_PARAMETERS="ro quiet"
 #U_BOOT_ROOT=""
 #U_BOOT_TIMEOUT="50"
@@ -14,4 +14,5 @@
 #U_BOOT_FDT_DIR="/usr/lib/aosc-os-${ARCH}-boot/linux-kernel-"
 #U_BOOT_FDT_OVERLAYS=""
 #U_BOOT_FDT_OVERLAYS_DIR="/boot/dtbo/"
+#U_BOOT_INITRD="initramfs"
 #U_BOOT_SYNC_DTBS="false"
diff --git a/read-config b/read-config
index 2bdcf0e..15e5018 100644
--- a/read-config
+++ b/read-config
@@ -65,11 +65,11 @@ U_BOOT_DEFAULT="${U_BOOT_DEFAULT:-l0}"
 U_BOOT_PROMPT="${U_BOOT_PROMPT:-1}"
 U_BOOT_ENTRIES="${U_BOOT_ENTRIES:-all}"
 U_BOOT_TIMEOUT="${U_BOOT_TIMEOUT:-50}"
-U_BOOT_MENU_LABEL="${U_BOOT_MENU_LABEL:-${PRETTY_NAME:-Debian GNU/Linux kernel}}"
+U_BOOT_MENU_LABEL="${U_BOOT_MENU_LABEL:-${PRETTY_NAME:-AOSC OS kernel}}"
 U_BOOT_FDT_DIR="${U_BOOT_FDT_DIR:-${_FDT_DIR}}"
 U_BOOT_FDT_OVERLAYS="${U_BOOT_FDT_OVERLAYS:-}"
 U_BOOT_FDT_OVERLAYS_DIR="${U_BOOT_FDT_OVERLAYS_DIR:-/boot/dtbo}"
-U_BOOT_INITRD="${U_BOOT_INITRD:-initrd.img}"
+U_BOOT_INITRD="${U_BOOT_INITRD:-initramfs}"
 
 if [ -z "${U_BOOT_PARAMETERS}" ] && [ -f /etc/kernel/cmdline ]
 then
-- 
2.46.0

