From 4ae23eda57d5ffbecaead5fa4f01d784466d011c Mon Sep 17 00:00:00 2001
From: Errant404 <erigidissimus@gmail.com>
Date: Wed, 21 Aug 2024 20:09:15 +0800
Subject: [PATCH] make the initrd matching pattern suitable for AOSC style.

---
 u-boot-update | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/u-boot-update b/u-boot-update
index ab6eb68..b7778c4 100755
--- a/u-boot-update
+++ b/u-boot-update
@@ -127,6 +127,9 @@ do
 	if [ -e "/boot/${U_BOOT_INITRD}-${_VERSION}" ]
 	then
 		_INITRD="initrd ${_BOOT_DIRECTORY}/${U_BOOT_INITRD}-${_VERSION}"
+	elif [ -e "/boot/${U_BOOT_INITRD}-${_VERSION}.img" ]
+	then
+		_INITRD="initrd ${_BOOT_DIRECTORY}/${U_BOOT_INITRD}-${_VERSION}.img"
 	elif [ -e "/boot/${U_BOOT_INITRD}" ]
 	then
 		_INITRD="initrd ${_BOOT_DIRECTORY}/${U_BOOT_INITRD}"
-- 
2.46.0

