--- a/Freedesktop/Aria.desktop	2017-01-02 14:42:08.000000000 -0600
+++ b/Freedesktop/Aria.desktop	2017-01-31 20:07:10.799887684 -0600
@@ -2,12 +2,12 @@
 [Desktop Entry]
 Encoding=UTF-8
-Version=1.4.1
+Version=1.4.13
 Type=Application
 Terminal=false
-Exec=Aria %f
+Exec=ariamaestosa %f
 Comment[en_US]=Aria Maestosa music editor
 Name=AriaMaestosa
 Comment=Aria Maestosa music editor
-Icon=/usr/local/share/Aria/aria64.png
+Icon=ariamaestosa
 MimeType=application/x-ariamaestosa
 Categories=AudioVideo;Audio
