diff --git a/dub.sdl b/dub.sdl
index 5bac5f9..80c79c1 100644
--- a/dub.sdl
+++ b/dub.sdl
@@ -8,12 +8,10 @@ dependency "taggedalgebraic" version="~>0.11.22"
 
 configuration "lua51" {
 	targetType "library"
-	dflags "-L$PACKAGE_DIR/deps/linux64/lua51.a" platform="linux-x86_64"
-	dflags "-L$PACKAGE_DIR/deps/win64/lua51.lib" platform="windows-x86_64"
-	dflags "-L$PACKAGE_DIR/deps/macamd/lua51.a" platform="osx-aarch64"
-	dflags "-L$PACKAGE_DIR/deps/macx64/lua51.a" platform="osx-x86_64"
+	dflags "-L-llua5.1" platform="linux"
 	versions "BindLua_Static" "LUA_51"
     subConfiguration "bindbc-lua" "static"
+        libs "lua5.1" platform="linux"
 }
 
 configuration "lua51-dynamic" {
@@ -26,5 +24,7 @@ configuration "lua51-dynamic" {
 
 configuration "lua52-dynamic" {
 	targetType "library"
+        dflags "-L-llua5.2" platform="linux"
 	versions "LUA_52"
+        libs "lua5.2" platform="linux"
 }
