From a742e817918bf13cd1ab04c14c7cc726a9afb782 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Thu, 15 Feb 2024 19:59:18 -0700
Subject: [PATCH 2/4] fix broken ccache detection

---
 common/autoconf/generated-configure.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/autoconf/generated-configure.sh b/common/autoconf/generated-configure.sh
index b3c5819161..b6b18c7293 100644
--- a/common/autoconf/generated-configure.sh
+++ b/common/autoconf/generated-configure.sh
@@ -55734,7 +55734,7 @@ $as_echo "$as_me: WARNING: --with-ccache-dir has no meaning when ccache is not e
     # precompiled headers.
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking if ccache supports precompiled headers" >&5
 $as_echo_n "checking if ccache supports precompiled headers... " >&6; }
-    HAS_GOOD_CCACHE=`($CCACHE --version | head -n 1 | grep -E 3.1.[456789]) 2> /dev/null`
+    HAS_GOOD_CCACHE=`($CCACHE --version | head -n 1 | grep -E 3.[23456789].[123456789]) 2> /dev/null`
     if test "x$HAS_GOOD_CCACHE" = x; then
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: no, disabling ccache" >&5
 $as_echo "no, disabling ccache" >&6; }
-- 
2.43.2

