Description: fix wrong return value
Bug: http://bugzilla.gnome.org/show_bug.cgi?id=582865
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=532119

--- a/gconf/gconf-client.c
+++ b/gconf/gconf-client.c
@@ -2373,7 +2373,7 @@ gconf_client_lookup (GConfClient *client
       {
         g_free (dir);
         trace ("Negative cache hit on %s", key);
-        return TRUE;
+        return FALSE;
       }
     else 
       {
@@ -2389,7 +2389,7 @@ gconf_client_lookup (GConfClient *client
               {
                 g_free (dir);
                 trace ("Non-existing dir for %s", key);
-                return TRUE;
+                return FALSE;
               }
             not_cached = TRUE;
           }
