--- a/backends/markup-tree.c
+++ b/backends/markup-tree.c
@@ -71,6 +71,11 @@
 }
 #endif
 
+/* PATH_MAX is not defined in limits.h on some platforms */
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 typedef struct
 {
   char       *locale;
--- a/backends/xml-dir.c
+++ b/backends/xml-dir.c
@@ -36,6 +36,11 @@
 
 #include "gconf/gconf-internals.h"
 
+/* PATH_MAX is not defined in limits.h on some platforms */
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 /* This makes hash table safer when debugging */
 #ifndef GCONF_ENABLE_DEBUG
 #define safe_g_hash_table_insert g_hash_table_insert
