From 931a4f7945e06cd80f97f3fda4529e0885887b08 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <origincode@aosc.io>
Date: Thu, 7 May 2026 16:51:58 +0800
Subject: [PATCH 2/2] Do not build deb

Signed-off-by: Kaiyang Wu <origincode@aosc.io>
---
 apps/desktop/electron-builder.ts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apps/desktop/electron-builder.ts b/apps/desktop/electron-builder.ts
index 9bcd771a73..692ac08849 100644
--- a/apps/desktop/electron-builder.ts
+++ b/apps/desktop/electron-builder.ts
@@ -129,7 +129,7 @@ const config: Omit<Writable<Configuration>, "electronFuses"> & {
         electron_protocol: variant.protocols[0],
     },
     linux: {
-        target: ["tar.gz", "deb"],
+        target: ["tar.gz"],
         category: "Network;InstantMessaging;Chat",
         icon: "icon.png",
         executableName: variant.name, // element-desktop or element-desktop-nightly
-- 
2.52.0

