From be8a93f6357b926cc40229b38d0cea94dbd96f13 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <origincode@aosc.io>
Date: Thu, 7 May 2026 16:51:58 +0800
Subject: [PATCH 2/2] Do not build deb

Signed-off-by: Kaiyang Wu <origincode@aosc.io>
---
 apps/desktop/electron-builder.ts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apps/desktop/electron-builder.ts b/apps/desktop/electron-builder.ts
index 88961e6860..76b5d9e90e 100644
--- a/apps/desktop/electron-builder.ts
+++ b/apps/desktop/electron-builder.ts
@@ -138,7 +138,7 @@ const config: Omit<Writable<Configuration>, "electronFuses"> & {
         electron_protocol: variant.protocols[0],
     },
     linux: {
-        target: ["tar.gz", "deb"],
+        target: ["tar.gz"],
         category: "Network;InstantMessaging;Chat",
         icon: "icon.png",
         executableName: variant.name, // element-desktop or element-desktop-nightly
-- 
2.55.0

