Description: Link libdb_sql-5.3 with the right libraries.
Author: Iain Lane <iain.lane@canonical.com>
Bug-Upstream: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=712262

--- db5.3.orig/dist/Makefile.in
+++ db5.3/dist/Makefile.in
@@ -897,7 +897,7 @@ $(libsql_version): $(SQL_OBJS)
 
 # Shared SQL library.
 $(libsqlso_target): $(SQL_OBJS)
-	$(SOLINK) $(SOFLAGS) $(LDFLAGS) -o $@ $(SQL_OBJS) $(LIBCSO_LIBS)
+	$(SOLINK) $(SOFLAGS) $(LDFLAGS) -o $@ $(SQL_OBJS) $(LIBCSO_LIBS) @SQL_LIBS@
 	$(RM) $(libsql)
 	$(LN) -s .libs/$(libsql_version) $(libsql)
 
