Index: db5.3-5.3.28/dist/s_config
===================================================================
--- db5.3-5.3.28.orig/dist/s_config
+++ db5.3-5.3.28/dist/s_config
@@ -35,3 +35,11 @@ rm -rf autom4te.cache
 chmod 755 configure
 
 chmod 755 config.guess config.sub install-sh
+
+echo "autoconf: running autoconf to build configure (lang/sql/sqlite/)"
+cp aclocal.m4 ../lang/sql/sqlite/
+cd ../lang/sql/sqlite/
+rm -f configure
+autoconf
+cd ../../../dist/
+
