From 0c50b686a9e10097440fff028bb8a519799377cb Mon Sep 17 00:00:00 2001
From: Kay Lin <kaymw@aosc.io>
Date: Mon, 30 May 2022 01:52:21 -0700
Subject: [PATCH 1/2] RenderTexture: include missing string header

---
 RenderTexture/TestRenderTexture.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/RenderTexture/TestRenderTexture.cpp b/RenderTexture/TestRenderTexture.cpp
index d3356e9..c61532c 100644
--- a/RenderTexture/TestRenderTexture.cpp
+++ b/RenderTexture/TestRenderTexture.cpp
@@ -8,6 +8,7 @@
 
 #include <assert.h>
 #include <stdio.h>
+#include <string.h>
 
 void Reshape(int w, int h);
 
-- 
2.36.0

