--- vid.stab-release-0.98b/CMakeLists.txt	2014-03-12 12:42:49.000000000 -0700
+++ vid.stab-release-0.98b.nosse2/CMakeLists.txt	2023-04-12 01:19:47.983378057 -0700
@@ -26,12 +26,6 @@
 add_definitions( -DDISABLE_ORC )
 endif()
 
-# here we should check for SSE2
-# our  -DUSE_SSE2_ASM code does not work with fpic
-if(SSE2_FOUND)
-add_definitions( -DUSE_SSE2 -msse2 -ffast-math )
-endif()
-
 set(SOURCES src/frameinfo.c src/transformtype.c src/libvidstab.c
   src/transform.c src/transformfixedpoint.c src/motiondetect.c
   src/motiondetect_opt.c src/serialize.c src/localmotion2transform.c
