From 133017a396fb001b03bd375df0ad6f9e8b19a0ee Mon Sep 17 00:00:00 2001
From: Ian Kent <raven@themaw.net>
Date: Wed, 17 Sep 2025 14:52:33 +0800
Subject: [PATCH 06/19] BACKPORT: autofs-5.1.9 - fix changelog

Add missed changelog entries.

Signed-off-by: Ian Kent <raven@themaw.net>
Signed-off-by: ilikara <3435193369@qq.com>
---
 CHANGELOG | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CHANGELOG b/CHANGELOG
index 3fe6b63..505993f 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,4 +1,7 @@
 
+- Update configure script.
+- fix ldap_parse_page_control() check.
+- fix crash in make_options_string().
 - Fix incompatible function pointer types in cyrus-sasl module.
 - fix always recreate credential cache.
 
-- 
2.51.0

