--- pyside-setup-opensource-src-5.15.2/sources/shiboken2/shibokenmodule/files.dir/shibokensupport/signature/parser.py	2020-11-11 04:51:30.000000000 -0800
+++ pyside-setup-opensource-src-5.15.2-new/sources/shiboken2/shibokenmodule/files.dir/shibokensupport/signature/parser.py	2021-11-02 01:58:40.272782292 -0700
@@ -222,7 +222,7 @@
 def to_string(thing):
     if isinstance(thing, str):
         return thing
-    if hasattr(thing, "__name__"):
+    if hasattr(thing, "__name__") and thing.__module__ != 'typing':
         dot = "." in str(thing)
         name = get_name(thing)
         return thing.__module__ + "." + name if dot else name
