diff -Naur a/ybin/ofpath b/ybin/ofpath
--- a/ybin/ofpath	2016-10-17 04:56:43.861671876 +0000
+++ b/ybin/ofpath	2016-10-17 04:56:29.473707713 +0000
@@ -213,11 +213,11 @@
 
     OF_PATH=;find_of_path "$SYS_DEVICE"
 
-    SCSI_DRIVER=`cat /sys/class/scsi_host/host$HOST_ID/proc_name 2>/dev/null`
+    SCSI_DRIVER=`cat /sys/class/scsi_host/host$HOST_ID/proc_name 2>/dev/null | tr -d '\000'`
 
-    SCSI_NAME=`cat /proc/device-tree$OF_PATH/name 2>/dev/null`
+    SCSI_NAME=`cat /proc/device-tree$OF_PATH/name 2>/dev/null | tr -d '\000'`
 
-    IEEE1394_ID=`cat /sys/block/$DEVNODE/device/ieee1394_id 2>/dev/null`
+    IEEE1394_ID=`cat /sys/block/$DEVNODE/device/ieee1394_id 2>/dev/null | tr -d '\000'`
     IEEE1394_ID=${IEEE1394_ID%%:*}
 
     PLUG_ID=$(ls -dv $SYS_PATH/*/host* 2>/dev/null | grep -n "/host$HOST_ID$")
