--- a/Makefile
+++ b/Makefile
@@ -39,7 +39,7 @@
 YBCFLAGS += -DTEXTADDR=$(TEXTADDR) -DDEBUG=$(DEBUG)
 YBCFLAGS += -DMALLOCADDR=$(MALLOCADDR) -DMALLOCSIZE=$(MALLOCSIZE)
 YBCFLAGS += -DKERNELADDR=$(KERNELADDR)
-YBCFLAGS += -Werror -fdiagnostics-show-option
+YBCFLAGS += -fgnu89-inline -fno-builtin-malloc
 YBCFLAGS += -I ./include
 YBCFLAGS += -fno-strict-aliasing
 
@@ -69,11 +69,15 @@
 
 # Libraries
 #
-LLIBS = -lext2fs
+LLIBS = -lext2fs -lcom_err \
+        `$(CC) -m32 -print-file-name=libpthread.a`
 
 # For compiling userland utils
 #
-UCFLAGS = -Os $(CFLAGS) -Wall -I/usr/include
+UCFLAGS = -Os -g $(CFLAGS) -Wall -I/usr/include
+UCFLAGS += -fstack-protector-strong
+UCFLAGS += -D_FORTIFY_SOURCE=2
+UCFLAGS += -Wl,-z,relro
 UCFLAGS += -Werror -fdiagnostics-show-option
 
 # For compiling build-tools that run on the host.
@@ -181,7 +185,7 @@
 	strip util/addnote
 	strip --remove-section=.comment --remove-section=.note util/addnote
 
-install: all strip
+install: all 
 	install -d -o root -g root -m 0755 ${ROOT}/etc/
 	install -d -o root -g root -m 0755 ${ROOT}/${PREFIX}/sbin/
 	install -d -o root -g root -m 0755 ${ROOT}/${PREFIX}/lib
