diff -Naur a/etc/yaboot.conf b/etc/yaboot.conf
--- a/etc/yaboot.conf	2011-10-18 04:11:10.000000000 +0000
+++ b/etc/yaboot.conf	2016-10-17 04:59:59.469184861 +0000
@@ -35,13 +35,14 @@
 ## section in this config file or the value of default=).
 
 timeout=40
-install=/usr/local/lib/yaboot/yaboot
-magicboot=/usr/local/lib/yaboot/ofboot
+install=/usr/lib/yaboot/yaboot
+magicboot=/usr/lib/yaboot/ofboot
+default=aoscuni
 
 ## Change the default colors, fgcolor is the text color, bgcolor is
 ## the screen background color. (default: fgcolor=white, bgcolor=black)
-#fgcolor=black
-#bgcolor=green
+fgcolor=black
+bgcolor=white
 
 ## Password supplied in plaintext, required for yaboot to boot, unless
 ## restricted is also present (see below). Be sure to
@@ -70,12 +71,10 @@
 ## image= to a symbolic link so long as the symlink does not cross
 ## partition boundries.
 
-image=/vmlinux
-	label=Linux
-	root=/dev/hda3
-	read-only
-
-image=/vmlinux.old
-	label=Linux.old
-	root=/dev/hda3
-	read-only
+image=/boot/vmlinux-uni
+	label=aoscuni
+	append="root=/dev/sda5 rw quiet radeon.agpmode=-1"
+
+image=/boot/vmlinux-smp
+	label=aoscsmp
+	append="root=/dev/sda5 rw quiet radeon.agpmode=-1"
