diff -Naur okular-22.04.3/shell/org.kde.okular.desktop okular-22.04.3.desktop/shell/org.kde.okular.desktop
--- okular-22.04.3/shell/org.kde.okular.desktop	2022-07-03 23:05:31.000000000 +0000
+++ okular-22.04.3.desktop/shell/org.kde.okular.desktop	2022-09-14 01:16:41.783944507 +0000
@@ -177,7 +177,7 @@
 Type=Application
 X-DocPath=okular/index.html
 InitialPreference=7
-Categories=Qt;KDE;Graphics;Office;Viewer;
+Categories=Qt;KDE;Office;Viewer;
 MimeType=application/vnd.kde.okular-archive;
 X-DBUS-ServiceName=org.kde.okular
 StartupWMClass=okular
