From 95eb7c98118244c235bcca5a0b9ff73cd08c1816 Mon Sep 17 00:00:00 2001
From: Christoph Cullmann <cullmann@kde.org>
Date: Thu, 3 Oct 2019 23:39:10 +0200
Subject: [PATCH] enable hidpi auto scaling

CCBUG: 373232
---
 src/main.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/main.cpp b/src/main.cpp
index 726ad802..06f3ce49 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -85,6 +85,10 @@ public:
 // ***
 extern "C" int Q_DECL_EXPORT kdemain(int argc, char *argv[])
 {
+    // enable high dpi support
+    QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps, true);
+    QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling, true);
+
     // Check if any of the arguments makes it impossible to re-use an existing process.
     // We need to do this manually and before creating a QApplication, because
     // QApplication takes/removes the Qt specific arguments that are incompatible.
@@ -116,9 +120,6 @@ extern "C" int Q_DECL_EXPORT kdemain(int argc, char *argv[])
     }
 #endif
 
-    // enable high dpi support
-    app->setAttribute(Qt::AA_UseHighDpiPixmaps, true);
-
 #if defined(Q_OS_MACOS)
     // this ensures that Ctrl and Meta are not swapped, so CTRL-C and friends
     // will work correctly in the terminal
-- 
2.17.1

