Index: b/avahi-daemon/avahi-daemon.conf
===================================================================
--- a/avahi-daemon/avahi-daemon.conf	2010-06-29 02:19:57.000000000 +0200
+++ b/avahi-daemon/avahi-daemon.conf	2010-07-07 20:03:01.987655481 +0200
@@ -21,9 +21,9 @@
 [server]
 #host-name=foo
 #domain-name=local
-browse-domains=0pointer.de, zeroconf.org
+#browse-domains=0pointer.de, zeroconf.org
 use-ipv4=yes
-use-ipv6=no
+use-ipv6=yes
 #allow-interfaces=eth0
 #deny-interfaces=eth1
 #check-response-ttl=no
