diff -Naur libtracefs-1.6.4/Documentation/Makefile libtracefs-1.6.4.mod/Documentation/Makefile
--- libtracefs-1.6.4/Documentation/Makefile	2023-01-09 20:14:30.000000000 -0500
+++ libtracefs-1.6.4.mod/Documentation/Makefile	2023-04-25 21:44:00.902801797 -0400
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: LGPL-2.1
 
-include $(src)/scripts/utils.mk
+include ../scripts/utils.mk
 
 
 # This Makefile and manpage XSL files were taken from libtraceevent
diff -Naur libtracefs-1.6.4/Makefile libtracefs-1.6.4.mod/Makefile
--- libtracefs-1.6.4/Makefile	2023-04-25 21:47:12.720952552 -0400
+++ libtracefs-1.6.4.mod/Makefile	2023-04-25 21:47:19.808881520 -0400
@@ -167,7 +167,7 @@
 # Make sure 32 bit stat() works on large file systems
 override CFLAGS += -D_FILE_OFFSET_BITS=64
 
-all: all_cmd
+all: all_cmd doc
 
 LIB_TARGET  = libtracefs.a libtracefs.so.$(TRACEFS_VERSION)
 LIB_INSTALL = libtracefs.a libtracefs.so*
@@ -280,7 +280,7 @@
 	$(Q)$(call do_install,$(src)/include/tracefs.h,$(includedir_SQ),644)
 	$(Q)$(call install_ld_config)
 
-install: install_libs
+install: install_libs install_doc
 
 install_pkgconfig: $(PKG_CONFIG_FILE)
 	$(Q)$(call , $(PKG_CONFIG_FILE)) \
diff -Naur libtracefs-1.6.4/scripts/utils.mk libtracefs-1.6.4.mod/scripts/utils.mk
--- libtracefs-1.6.4/scripts/utils.mk	2023-01-09 20:14:30.000000000 -0500
+++ libtracefs-1.6.4.mod/scripts/utils.mk	2023-04-25 21:44:58.765260220 -0400
@@ -101,11 +101,11 @@
 
 do_sample_build =							\
 	$(Q)($(print_sample_build)					\
-	$(CC) -o $1 $2 $(CFLAGS) $(LIBTRACEFS_STATIC) $(LIBTRACEEVENT_LIBS) -lpthread)
+	$(CC) -o $1 $2 $(CFLAGS) $(LIBTRACEFS_STATIC) $(LIBTRACEEVENT_LIBS) -lpthread $(LDFLAGS))
 
 do_sample_obj =									\
 	$(Q)($(print_sample_obj)						\
-	$(CC) -g -Wall -c $(CFLAGS) -o $1 $2 -I../include/ $(LIBTRACEEVENT_INCLUDES))
+	$(CC) -g -Wall -c $(CFLAGS) -o $1 $2 -I../include/ $(LIBTRACEEVENT_INCLUDES) $(LDFLAGS))
 
 ifneq ($(findstring $(MAKEFLAGS), w),w)
 PRINT_DIR = --no-print-directory
