diff -Naur Remmina-v1.4.31-7407cc3a27eb25ba4ddddeab565f67fd688324ed/data/desktop/org.remmina.Remmina.desktop.in Remmina-v1.4.31-7407cc3a27eb25ba4ddddeab565f67fd688324ed.nokiosk/data/desktop/org.remmina.Remmina.desktop.in
--- Remmina-v1.4.31-7407cc3a27eb25ba4ddddeab565f67fd688324ed/data/desktop/org.remmina.Remmina.desktop.in	2023-06-07 21:13:28.000000000 +0800
+++ Remmina-v1.4.31-7407cc3a27eb25ba4ddddeab565f67fd688324ed.nokiosk/data/desktop/org.remmina.Remmina.desktop.in	2023-08-21 12:22:11.927448436 +0800
@@ -79,7 +79,7 @@
 Keywords=remote desktop;rdp;vnc;ssh;spice;
 StartupWMClass=@REMMINA_APP_ID@
 MimeType=x-scheme-handler/rdp;x-scheme-handler/spice;x-scheme-handler/vnc;x-scheme-handler/ssh;x-scheme-handler/remmina;application/x-remmina;
-Actions=Kiosk;Profile;Tray;Quit;
+Actions=Profile;Tray;Quit;
 
 [Desktop Action Profile]
 Name=Create a New Connection Profile
@@ -106,16 +106,6 @@
 Name[zh_CN]=新建连接配置
 Exec=@REMMINA_BINARY_PATH@ --new
 
-[Desktop Action Kiosk]
-# Start Remmina with a minimal interface for kiosk/thin client mode
-Name=Start Remmina in Kiosk mode
-Name[da]=Start Remmina i kiosk-tilstand
-Name[fr]=Démarrer Remmina en mode Kiosque
-Name[hr]=Pokreni Remminu u Kisok načinu
-Name[it]=Avvia Remmina in modo Chiosco
-Name[uk]=Запустити Remmina у режимі кіоску
-Exec=@REMMINA_BINARY_PATH@ --kiosk
-
 [Desktop Action Tray]
 Name=Start Remmina Minimized
 Name[ca]=Inicia el Remmina minimitzat
