From 7942c66487f6f66e7553afda718370d3fb3bd665 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Tue, 23 Nov 2010 12:20:49 +0100
Subject: [PATCH] Do not use bundled LTDL.

---
 Makefile.am  |   15 +--------------
 configure.ac |   10 ++--------
 2 files changed, 3 insertions(+), 22 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 45a34d8..6bed17d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -26,23 +26,10 @@ MAINTAINERCLEANFILES = \
 	config.status \
 	.doc_stamp \
 	.DS_Store \
-	include/etl_profile.h.in \
-	libltdl/COPYING.LIB \
-	libltdl/Makefile.in \
-	libltdl/acinclude.m4 \
-	libltdl/config-h.in \
-	libltdl/configure.in \
-	libltdl/ltdl.h \
-	libltdl/Makefile.am \
-	libltdl/README \
-	libltdl/aclocal.m4 \
-	libltdl/configure \
-	libltdl/ltdl.c \
-	libltdl/stamp-h.in
+	include/etl_profile.h.in
 
 
 SUBDIRS = \
-	libltdl \
 	build_tools \
 	src \
 	examples \
diff --git a/configure.ac b/configure.ac
index 7833a08..98d81fd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -85,8 +85,8 @@ AC_LIBTOOL_DLOPEN
 AC_DISABLE_STATIC
 AC_ENABLE_SHARED
 AC_PROG_LIBTOOL
-AC_SUBST(LIBTOOL_DEPS)
-AC_LIBTOOL_PATCH
+#AC_SUBST(LIBTOOL_DEPS)
+#AC_LIBTOOL_PATCH
 
 #if [[ "$LIBTOOL_PATCH_SED""x" != "x" ]] ; then {
 #	printf "Patching libtool... "
@@ -622,12 +622,6 @@ AC_SUBST(CXXFLAGS)
 AC_SUBST(CPPFLAGS)
 AC_SUBST(LDFLAGS)
 
-# AC_CONFIG_SUBDIRS(libltdl) is required for libtool 1.5.26 but must not be present for
-# libtool 2.2.4. Problem seems to be that AC_LIB_LTDL (above) now calls AC_CONFIG_SUBDIRS
-# itself, through LTDL_INIT, _LTDL_SETUP, _LTDL_MODE_DISPATCH. We need to check the
-# condition that libltdl has already been registered with AC_CONFIG_SUBDIRS:
-m4_ifdef([_AC_SEEN_TAG(libltdl)], [], [AC_CONFIG_SUBDIRS(libltdl)])
-
 CONFIG_LIBS="-lsynfig"
 CONFIG_CFLAGS="$CONFIG_CFLAGS"
 AC_SUBST(CONFIG_LIBS)
-- 
1.7.3.1

