--- a/Makefile.am~	2015-07-09 02:33:03.000000000 -0500
+++ b/Makefile.am	2016-06-24 08:56:43.429236736 -0500
@@ -27,19 +27,6 @@
 	.doc_stamp \
 	.DS_Store \
 	include/etl_profile.h.in \
-	libltdl/COPYING.LIB \
-	libltdl/Makefile.in \
-	libltdl/acinclude.m4 \
-	libltdl/config-h.in \
-	libltdl/configure.in \
-	libltdl/ltdl.h \
-	libltdl/Makefile.am \
-	libltdl/README \
-	libltdl/aclocal.m4 \
-	libltdl/configure \
-	libltdl/ltdl.c \
-	libltdl/stamp-h.in
-
 
 SUBDIRS = \
 	libltdl \
--- a/configure.ac~	2015-10-06 04:13:36.000000000 -0500
+++ b/configure.ac	2016-06-24 08:58:04.641800077 -0500
@@ -654,12 +654,6 @@
 AC_SUBST(CPPFLAGS)
 AC_SUBST(LDFLAGS)
 
-# AC_CONFIG_SUBDIRS(libltdl) is required for libtool 1.5.26 but must not be present for
-# libtool 2.2.4. Problem seems to be that AC_LIB_LTDL (above) now calls AC_CONFIG_SUBDIRS
-# itself, through LTDL_INIT, _LTDL_SETUP, _LTDL_MODE_DISPATCH. We need to check the
-# condition that libltdl has already been registered with AC_CONFIG_SUBDIRS:
-m4_ifdef([_AC_SEEN_TAG(libltdl)], [], [AC_CONFIG_SUBDIRS(libltdl)])
-
 CONFIG_LIBS="-lsynfig"
 CONFIG_CFLAGS="$CONFIG_CFLAGS"
 AC_SUBST(CONFIG_LIBS)
--- a/Makefile.am~	2016-06-24 09:00:28.000000000 -0500
+++ b/Makefile.am	2016-06-24 09:01:12.531789841 -0500
@@ -26,7 +26,7 @@
 	config.status \
 	.doc_stamp \
 	.DS_Store \
-	include/etl_profile.h.in \
+	include/etl_profile.h.in
 
 SUBDIRS = \
 	libltdl \
--- a/Makefile.am~	2016-06-24 09:02:14.000000000 -0500
+++ b/Makefile.am	2016-06-24 09:02:38.773326142 -0500
@@ -29,7 +29,6 @@
 	include/etl_profile.h.in
 
 SUBDIRS = \
-	libltdl \
 	build_tools \
 	src \
 	examples \
