--- a/configure.ori
+++ b/configure
@@ -21614,7 +21614,7 @@ $as_echo_n "checking for boostlib >= $bo
                         libsubdirs="lib"
     ax_arch=`uname -m`
     case $ax_arch in
-      x86_64|ppc64|s390x|sparc64|aarch64)
+      x86_64|ppc64|ppc64le|s390x|sparc64|aarch64)
         libsubdirs="lib64 lib lib64"
         ;;
     esac
--- a/m4/ax_boost_base.m4.ori
+++ b/m4/ax_boost_base.m4
@@ -92,7 +92,7 @@ if test "x$want_boost" = "xyes"; then
     libsubdirs="lib"
     ax_arch=`uname -m`
     case $ax_arch in
-      x86_64|ppc64|s390x|sparc64|aarch64)
+      x86_64|ppc64|ppc64le|s390x|sparc64|aarch64)
         libsubdirs="lib64 lib lib64"
         ;;
     esac
