diff -Naur snappy-1.1.5/CMakeLists.txt snappy-1.1.5.notest/CMakeLists.txt
--- snappy-1.1.5/CMakeLists.txt	2017-06-29 09:37:30.000000000 +0800
+++ snappy-1.1.5.notest/CMakeLists.txt	2017-07-12 11:55:45.768763938 +0800
@@ -125,8 +125,6 @@
         ${Snappy_BINARY_DIR}/SnappyConfigVersion.cmake
         DESTINATION lib/cmake)
 
-ENABLE_TESTING()
-
 IF (HAVE_LIBZ)
     LIST(APPEND COMPRESSION_LIBS z)
 ENDIF (HAVE_LIBZ)
@@ -142,14 +140,3 @@
 IF (HAVE_LIBQUICKLZ)
     LIST(APPEND COMPRESSION_LIBS quicklz)
 ENDIF (HAVE_LIBQUICKLZ)
-
-ADD_EXECUTABLE(snappy-unittest snappy_unittest.cc snappy-test.cc)
-TARGET_COMPILE_DEFINITIONS(snappy-unittest PRIVATE -DHAVE_CONFIG_H)
-TARGET_LINK_LIBRARIES(snappy-unittest snappy ${COMPRESSION_LIBS}
-                      ${GFLAGS_LIBRARIES})
-TARGET_INCLUDE_DIRECTORIES(snappy-unittest BEFORE PRIVATE ${Snappy_SOURCE_DIR}
-                           ${GTEST_INCLUDE_DIRS} ${GFLAGS_INCLUDE_DIRS})
-
-ADD_TEST(NAME snappy-unittest
-         WORKING_DIRECTORY ${Snappy_SOURCE_DIR}
-         COMMAND ${Snappy_BINARY_DIR}/snappy-unittest)
