diff -Naur lib/lib/converters.h lib.qt514/lib/converters.h
--- lib/lib/converters.h	2020-03-27 04:11:07.330031252 +0000
+++ lib.qt514/lib/converters.h	2020-03-27 04:12:49.992331017 +0000
@@ -37,6 +37,7 @@
 using optional = std::experimental::optional<T>;
 #endif
 
+#if QT_VERSION < QT_VERSION_CHECK(5,14,0)
 // Enable std::unordered_map<QString, T>
 namespace std
 {
@@ -44,14 +45,12 @@
     {
         size_t operator()(const QString& s) const Q_DECL_NOEXCEPT
         {
-            return qHash(s
-#if (QT_VERSION >= QT_VERSION_CHECK(5, 6, 0))
-                         , uint(qGlobalQHashSeed())
-#endif
+            return qHash(s, uint(qGlobalQHashSeed())
                          );
         }
     };
 }
+#endif
 
 class QVariant;
 
