diff --git a/src/alsa_backend.c b/src/alsa_backend.c
index 4635231..d9d63a5 100644
--- a/src/alsa_backend.c
+++ b/src/alsa_backend.c
@@ -22,10 +22,8 @@
 //##############################################################################
 
 #include <alsa/asoundlib.h>
-#include <glib/gstring.h>
-#include <glib/gstdio.h>
-#include <glib/glist.h>
-#include <glib/giochannel.h>
+
+#include <glib.h>
 
 #include "alsa_backend.h"
 
diff --git a/src/volumeicon.c b/src/volumeicon.c
index 29d787b..5b7f6cd 100644
--- a/src/volumeicon.c
+++ b/src/volumeicon.c
@@ -761,7 +761,7 @@ static void volume_icon_load_icons()
 		gchar * icon_path = g_strdup_printf(ICONS_DIR"/%s/%d.png",
 			config_get_theme(), i+1);
 		if(icons_loaded && (NULL != m_icons[i]))
-			gdk_pixbuf_unref(m_icons[i]);
+			g_object_unref(m_icons[i]);
 		m_icons[i] = gdk_pixbuf_new_from_file(icon_path, NULL);
 		if(NULL == m_icons[i])
 			g_message("Failed to load '%s'", icon_path);
