From 77df268dabee033ccb3dc9cc95a909073ee07030 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 30 Nov 2020 17:15:32 +0800
Subject: [PATCH] load 32-bit library /opt/32/lib for aosc os

---
 bin/mangohud.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/mangohud.in b/bin/mangohud.in
index eadce2d..bb266b2 100755
--- a/bin/mangohud.in
+++ b/bin/mangohud.in
@@ -23,6 +23,6 @@ fi
 # figure out whether the 32 or 64 bit version should be used, and will search
 # for it in the correct directory
 LD_PRELOAD="${LD_PRELOAD}:${MANGOHUD_LIB_NAME}"
-LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:@ld_libdir_mangohud@"
+LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:@ld_libdir_mangohud@:/opt/32/lib"
 
 exec env MANGOHUD=1 LD_LIBRARY_PATH="${LD_LIBRARY_PATH}" LD_PRELOAD="${LD_PRELOAD}" "$@"
-- 
2.28.0

