diff --git a/src/framework/audio/qml/MuseScore/Audio/DevTools/AudioEngineTests.qml b/src/framework/audio/qml/MuseScore/Audio/DevTools/AudioEngineTests.qml
index 5e5eb93e..874c6112 100644
--- a/src/framework/audio/qml/MuseScore/Audio/DevTools/AudioEngineTests.qml
+++ b/src/framework/audio/qml/MuseScore/Audio/DevTools/AudioEngineTests.qml
@@ -179,12 +179,12 @@ Rectangle {
 
                 Switch {
                     text: qsTr("eq")
-                    onPositionChanged: devtools.enableNoiseEq(position > 0.5);
+                    onClicked: devtools.enableNoiseEq(checked);
                 }
 
                 Switch {
                     text: qsTr("mute")
-                    onPositionChanged: devtools.setMuteNoise(position > 0.5);
+                    onClicked: devtools.setMuteNoise(checked);
                 }
 
                 Slider {
@@ -236,7 +236,7 @@ Rectangle {
 
                 Switch {
                     text: qsTr("mute")
-                    onPositionChanged: devtools.setMuteSine(position > 0.5);
+                    onClicked: devtools.setMuteSine(checked);
                 }
 
                 Slider {
