From 666e4695c7cb7cb244456e58c5318d37d0f5bd71 Mon Sep 17 00:00:00 2001
From: WhiredPlanck <whiredplanck@outlook.com>
Date: Tue, 12 Oct 2021 22:24:11 +0800
Subject: [PATCH] qmake: fix bad include and libs path prefix

---
 LANDrop.pro | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/LANDrop.pro b/LANDrop.pro
index 8474775..ea2baa9 100644
--- a/LANDrop.pro
+++ b/LANDrop.pro
@@ -51,14 +51,14 @@ RC_ICONS = icons/app.ico
 ICON = icons/app.icns
 
 unix {
-    INCLUDEPATH += /usr/local/include
-    LIBS += -L/usr/local/lib -lsodium
-
     PREFIX = $$(PREFIX)
     isEmpty(PREFIX) {
-        PREFIX = /usr/local
+        PREFIX = /usr
     }
 
+    INCLUDEPATH += $$PREFIX/include
+    LIBS += -lsodium
+
     binary.path = $$PREFIX/bin
     binary.files = $$OUT_PWD/landrop
     binary.extra = cp "$$OUT_PWD/LANDrop" "$$OUT_PWD/landrop"
-- 
2.30.2

