diff -Naur yubioath-desktop-yubioath-desktop-4.0.1/main.cpp yubioath-desktop-yubioath-desktop-4.0.1.qtsingle/main.cpp
--- yubioath-desktop-yubioath-desktop-4.0.1/main.cpp	2017-03-27 08:06:12.000000000 -0500
+++ yubioath-desktop-yubioath-desktop-4.0.1.qtsingle/main.cpp	2017-04-03 12:00:18.318176977 -0500
@@ -5,7 +5,7 @@
 #include <QtGlobal>
 #include <QtWidgets>
 #ifndef Q_OS_DARWIN
-#include <QtSingleApplication>
+#include <QtSolutions/QtSingleApplication>
 #endif
 #include "screenshot.h"
 
diff -Naur yubioath-desktop-yubioath-desktop-4.0.1/yubioath-desktop.pro yubioath-desktop-yubioath-desktop-4.0.1.qtsingle/yubioath-desktop.pro
--- yubioath-desktop-yubioath-desktop-4.0.1/yubioath-desktop.pro	2017-03-27 08:06:12.000000000 -0500
+++ yubioath-desktop-yubioath-desktop-4.0.1.qtsingle/yubioath-desktop.pro	2017-04-03 12:00:14.008024153 -0500
@@ -66,3 +66,5 @@
     py/* \
     py/qr/* \
     qml/*
+
+QMAKE_LFLAGS += -lQt5Solutions_SingleApplication-head
