diff --git a/qt3d/src/3rdparty/assimp/src/code/AssetLib/FBX/FBXBinaryTokenizer.cpp b/qt3d/src/3rdparty/assimp/src/code/AssetLib/FBX/FBXBinaryTokenizer.cpp
index 3488120..79b838d 100644
--- a/qt3d/src/3rdparty/assimp/src/code/AssetLib/FBX/FBXBinaryTokenizer.cpp
+++ b/qt3d/src/3rdparty/assimp/src/code/AssetLib/FBX/FBXBinaryTokenizer.cpp
@@ -56,6 +56,8 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <assimp/DefaultLogger.hpp>
 #include <assimp/StringUtils.h>
 
+#include <cstdint>
+
 namespace Assimp {
 namespace FBX {
 
diff --git a/qt3d/src/3rdparty/assimp/src/code/Common/Importer.h b/qt3d/src/3rdparty/assimp/src/code/Common/Importer.h
index e7da7f4..c41dd53 100644
--- a/qt3d/src/3rdparty/assimp/src/code/Common/Importer.h
+++ b/qt3d/src/3rdparty/assimp/src/code/Common/Importer.h
@@ -50,6 +50,9 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <string>
 #include <assimp/matrix4x4.h>
 
+#include <exception>
+#include <stdexcept>
+
 struct aiScene;
 
 namespace Assimp    {
diff --git a/qt3d/src/3rdparty/assimp/src/include/assimp/Importer.hpp b/qt3d/src/3rdparty/assimp/src/include/assimp/Importer.hpp
index 09b5b68..327c71f 100644
--- a/qt3d/src/3rdparty/assimp/src/include/assimp/Importer.hpp
+++ b/qt3d/src/3rdparty/assimp/src/include/assimp/Importer.hpp
@@ -61,6 +61,9 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <exception>
 
+#include <exception>
+#include <stdexcept>
+
 namespace Assimp {
 // =======================================================================
 // Public interface to Assimp
