diff -Naur a/policy/org.freedesktop.packagekit.policy.in b/policy/org.freedesktop.packagekit.policy.in
--- a/policy/org.freedesktop.packagekit.policy.in	2015-10-28 11:23:18.000000000 +0000
+++ b/policy/org.freedesktop.packagekit.policy.in	2016-02-07 03:53:03.329440450 +0000
@@ -131,7 +131,7 @@
     <defaults>
       <allow_any>auth_admin</allow_any>
       <allow_inactive>auth_admin</allow_inactive>
-      <allow_active>yes</allow_active>
+      <allow_active>auth_admin_keep</allow_active>
     </defaults>
   </action>
 
@@ -170,7 +170,7 @@
     <defaults>
       <allow_any>auth_admin</allow_any>
       <allow_inactive>auth_admin</allow_inactive>
-      <allow_active>yes</allow_active>
+      <allow_active>auth_admin_keep</allow_active>
     </defaults>
   </action>
 
@@ -273,7 +273,7 @@
 
   <action id="org.freedesktop.packagekit.trigger-offline-update">
     <!-- SECURITY:
-          - Normal users are able to ask updates to be installed at
+          - Administrators are able to ask updates to be installed at
             early boot time without a password.
      -->
     <_description>Trigger offline updates</_description>
@@ -282,7 +282,7 @@
     <defaults>
       <allow_any>auth_admin</allow_any>
       <allow_inactive>auth_admin</allow_inactive>
-      <allow_active>yes</allow_active>
+      <allow_active>auth_admin_keep</allow_active>
     </defaults>
   </action>
 
diff -Naur a/policy/org.freedesktop.packagekit.rules b/policy/org.freedesktop.packagekit.rules
--- a/policy/org.freedesktop.packagekit.rules	2015-09-29 17:05:46.000000000 +0000
+++ b/policy/org.freedesktop.packagekit.rules	2016-02-07 03:53:17.542535896 +0000
@@ -1,5 +1,6 @@
 polkit.addRule(function(action, subject) {
-    if (action.id == "org.freedesktop.packagekit.package-install" &&
+    if ((action.id == "org.freedesktop.packagekit.upgrade-system" ||
+         action.id == "org.freedesktop.packagekit.trigger-offline-update") &&
         subject.active == true && subject.local == true &&
         subject.isInGroup("wheel")) {
             return polkit.Result.YES;
