diff -Naur PackageKit-PACKAGEKIT_1_2_1/backends/aptcc/meson.build PackageKit-PACKAGEKIT_1_2_1.apt21/backends/aptcc/meson.build
--- PackageKit-PACKAGEKIT_1_2_1/backends/aptcc/meson.build	2020-09-07 04:13:27.000000000 -0500
+++ PackageKit-PACKAGEKIT_1_2_1.apt21/backends/aptcc/meson.build	2020-09-24 23:38:17.441382935 -0500
@@ -8,7 +8,6 @@
 appstream_dep = dependency('appstream', version: '>=0.11')
 
 apt_pkg_dep = cpp_compiler.find_library('apt-pkg')
-apt_inst_dep = cpp_compiler.find_library('apt-inst')
 
 if not cpp_compiler.compiles(
   '''
@@ -19,8 +18,7 @@
     }
   ''',
   dependencies: [
-    apt_pkg_dep,
-    apt_inst_dep
+    apt_pkg_dep
   ]
 )
   error ('libapt-pkg 1.1 or later is required')
@@ -35,8 +33,7 @@
     }
   ''',
   dependencies: [
-    apt_pkg_dep,
-    apt_inst_dep
+    apt_pkg_dep
   ]
 )
   error ('libapt-pkg 1.7 or later is required - or backported frontend locking')
@@ -53,8 +50,7 @@
     }
   ''',
   dependencies: [
-    apt_pkg_dep,
-    apt_inst_dep
+    apt_pkg_dep
   ]
 )
   ddtp_flag = ['-DHAVE_DDTP']
@@ -86,7 +82,6 @@
     packagekit_glib2_dep,
     gmodule_dep,
     apt_pkg_dep,
-    apt_inst_dep,
     gstreamer_dep,
     gstreamer_base_dep,
     gstreamer_plugins_base_dep,
