diff --git a/src/player/out_alsa.c b/src/player/out_alsa.c
index 2498701..ed49147 100644
--- a/src/player/out_alsa.c
+++ b/src/player/out_alsa.c
@@ -45,7 +45,7 @@ static int open_alsa_output(const char *pcmname, unsigned int *rate) {
     int err;
 
     if (!pcmname || !*pcmname) {
-        pcmname = "default";
+        pcmname = "pulse";
     }
 
     if ((err = snd_pcm_open(&pcm, pcmname, SND_PCM_STREAM_PLAYBACK, 0)) < 0) {
