diff -Naur cwiid/wmdemo/Makefile.in cwiid.bluetooth/wmdemo/Makefile.in
--- cwiid/wmdemo/Makefile.in	2020-07-06 00:18:40.439804176 -0500
+++ cwiid.bluetooth/wmdemo/Makefile.in	2020-07-06 00:19:01.104889190 -0500
@@ -8,7 +8,7 @@
 
 CFLAGS += -I@top_builddir@/libcwiid
 LDFLAGS += -L@top_builddir@/libcwiid
-LDLIBS += -lcwiid
+LDLIBS += -lcwiid -lbluetooth
 INST_DIR = @bindir@
 
 include $(COMMON)/include/app.mak
