diff --git a/rust-launcher/src/utils.rs b/rust-launcher/src/utils.rs
index 89bf7966..0e1b2db8 100644
--- a/rust-launcher/src/utils.rs
+++ b/rust-launcher/src/utils.rs
@@ -25,8 +25,6 @@ pub fn find_jre(os: &os_access::Os) -> std::path::PathBuf {
                     os.log("itw-rust-debug: found and using");
                     let java_home = std::path::PathBuf::from(war);
                     let mut jre_dir = java_home.clone();
-                    jre_dir.push("jre");
-                    jre_dir.push("");
                     if jre_dir.exists() {
                         return jre_dir; 
                 }
