From 33de11135b5b123e7a2a5675151e548b747fbdd0 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sun, 12 Oct 2025 12:26:47 +0800
Subject: [PATCH 1/2] AOSCOS :launcher use aosc style path

---
 rust-launcher/src/utils.rs | 2 --
 1 file changed, 2 deletions(-)

diff --git a/rust-launcher/src/utils.rs b/rust-launcher/src/utils.rs
index 89bf796..0e1b2db 100644
--- a/rust-launcher/src/utils.rs
+++ b/rust-launcher/src/utils.rs
@@ -25,8 +25,6 @@ pub fn find_jre(os: &os_access::Os) -> std::path::PathBuf {
                     os.log("itw-rust-debug: found and using");
                     let java_home = std::path::PathBuf::from(war);
                     let mut jre_dir = java_home.clone();
-                    jre_dir.push("jre");
-                    jre_dir.push("");
                     if jre_dir.exists() {
                         return jre_dir; 
                 }
-- 
2.51.0

