diff -Naur a/mcs/class/System.Drawing/System.Drawing/SystemFonts.cs b/mcs/class/System.Drawing/System.Drawing/SystemFonts.cs
--- a/mcs/class/System.Drawing/System.Drawing/SystemFonts.cs	2016-02-22 08:08:45.000000000 -0700
+++ b/mcs/class/System.Drawing/System.Drawing/SystemFonts.cs	2016-05-05 21:36:33.232564512 -0600
@@ -68,35 +68,35 @@
 		}
 
 		public static Font CaptionFont { 
-			get { return new Font ("Microsoft Sans Serif", 11, "CaptionFont"); }
+			get { return new Font ("Sans", 11, "CaptionFont"); }
 		}
 
 		public static Font DefaultFont  { 
-			get { return new Font ("Microsoft Sans Serif", 8.25f, "DefaultFont"); }
+			get { return new Font ("Sans", 8.25f, "DefaultFont"); }
 		}
 
 		public static Font DialogFont  { 
-			get { return new Font ("Tahoma", 8, "DialogFont"); }
+			get { return new Font ("Sans", 8, "DialogFont"); }
 		}
 
 		public static Font IconTitleFont  { 
-			get { return new Font ("Microsoft Sans Serif", 11, "IconTitleFont"); }
+			get { return new Font ("Sans", 11, "IconTitleFont"); }
 		}
 
 		public static Font MenuFont  { 
-			get { return new Font ("Microsoft Sans Serif", 11, "MenuFont"); }
+			get { return new Font ("Sans", 11, "MenuFont"); }
 		}
 
 		public static Font MessageBoxFont  { 
-			get { return new Font ("Microsoft Sans Serif", 11, "MessageBoxFont"); }
+			get { return new Font ("Sans", 11, "MessageBoxFont"); }
 		}
 
 		public static Font SmallCaptionFont  { 
-			get { return new Font ("Microsoft Sans Serif", 11, "SmallCaptionFont"); }
+			get { return new Font ("Sans", 11, "SmallCaptionFont"); }
 		}
 
 		public static Font StatusFont  { 
-			get { return new Font ("Microsoft Sans Serif", 11, "StatusFont"); }
+			get { return new Font ("Sans", 11, "StatusFont"); }
 		}	      
 	}
 }
