--- a/mono/mini/exceptions-arm64.c	2025-04-12 05:43:09.420996006 +0000
+++ b/mono/mini/exceptions-arm64.c	2025-04-12 05:43:24.547107545 +0000
@@ -549,7 +549,7 @@
 	UCONTEXT_REG_R0 (sigctx) = (gsize)obj;
 
 	gpointer addr = (gpointer)handle_signal_exception;
-	UCONTEXT_REG_SET_PC (sigctx, addr);
+	UCONTEXT_REG_SET_PC (sigctx, (size_t)addr);
 	host_mgreg_t sp = UCONTEXT_REG_SP (sigctx) - MONO_ARCH_REDZONE_SIZE;
 	UCONTEXT_REG_SET_SP (sigctx, sp);
 #endif
