--- renderdoc-1.16/renderdoc/os/posix/linux/linux_process.cpp	2021-09-29 12:54:32.000000000 +0000
+++ renderdoc-1.16.ppc64el/renderdoc/os/posix/linux/linux_process.cpp	2021-11-17 01:00:35.524817303 +0000
@@ -218,6 +218,17 @@
 // the instruction pointer at all.
 #define BREAK_INST_INST_PTR_ADJUST 0
 
+#elif defined(__powerpc64__)
+
+#define user_regs_struct struct pt_regs
+
+#define INST_PTR_REG nip
+
+#define BREAK_INST 0x0800e07fULL
+#define BREAK_INST_BYTES_SIZE 4
+
+#define BREAK_INST_INST_PTR_ADJUST 4
+
 #else
 
 #define BREAK_INST 0xccULL
