diff --git a/src/framework/update/internal/updatescenario.cpp b/src/framework/update/internal/updatescenario.cpp
index 23b54be0c1..b3e67ac053 100644
--- a/src/framework/update/internal/updatescenario.cpp
+++ b/src/framework/update/internal/updatescenario.cpp
@@ -69,7 +69,7 @@ muse::async::Promise<Ret> UpdateScenario::checkForUpdate(bool manual)
                 (void)resolve(ret);
             };
 
-            const bool noUpdate = res.ret.code() == static_cast<int>(Err::NoUpdate);
+            const bool noUpdate = true;
             if (!noUpdate && !res.ret) {
                 LOGE() << "Unable to check for app update, error: " << res.ret.toString();
                 ret = muse::make_ret(Ret::Code::UnknownError);
@@ -221,8 +221,8 @@ void UpdateScenario::downloadRelease()
 
 void UpdateScenario::askToCloseAndCompleteInstall(const muse::io::path_t& installerPath)
 {
-    const std::string info = muse::trc("update", "MuseScore Studio needs to close to complete the installation. "
-                                                 "If you have any unsaved changes, you will be prompted to save them before MuseScore Studio closes.");
+    const std::string info = muse::trc("update", "Update is disabled."
+                                                 "Please update MuseScore with oma.");
     const int closeBtn = int(IInteractive::Button::CustomButton) + 1;
     const IInteractive::Result res = interactive()->infoSync("", info,
                                                              { interactive()->buttonData(IInteractive::Button::Cancel),
