From 1c6ceca69f81b08cd5c0954338e0ceddc8578cb6 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Mon, 13 May 2024 10:38:32 +0800
Subject: [PATCH] Disable self update

---
 src/update/internal/updatescenario.cpp | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/src/update/internal/updatescenario.cpp b/src/update/internal/updatescenario.cpp
index 666b581f44..fc6942b925 100644
--- a/src/update/internal/updatescenario.cpp
+++ b/src/update/internal/updatescenario.cpp
@@ -41,20 +41,17 @@ using namespace mu::framework;
 
 void UpdateScenario::delayedInit()
 {
-    if (configuration()->needCheckForUpdate() && multiInstancesProvider()->instances().size() == 1) {
-        QTimer::singleShot(AUTO_CHECK_UPDATE_INTERVAL, [this]() {
-            doCheckForUpdate(false);
-        });
-    }
 }
 
 void UpdateScenario::checkForUpdate()
 {
-    if (isCheckStarted()) {
-        return;
-    }
+    QString str = qtrc("update", "Self-update is disabled by AOSC OS. Please update via oma/apt.");
 
-    doCheckForUpdate(true);
+    IInteractive::Text text(str.toStdString(), IInteractive::TextFormat::RichText);
+    IInteractive::ButtonData okBtn = interactive()->buttonData(IInteractive::Button::Ok);
+
+    interactive()->info(trc("update", "Self-update disabled"), text, { okBtn }, okBtn.btn,
+                        IInteractive::Option::WithIcon);
 }
 
 bool UpdateScenario::isCheckStarted() const
-- 
2.45.0

