From 289cf4ceaf120ab17524bc0921d6b82d8560d7ac Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Tue, 28 Apr 2026 10:07:44 +0800
Subject: [PATCH 1/2] coreconf/Linux: add LFS support on glibc for 32-bit
 targets

---
 coreconf/Linux.mk | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/coreconf/Linux.mk b/coreconf/Linux.mk
index dcb432415..582705ff3 100644
--- a/coreconf/Linux.mk
+++ b/coreconf/Linux.mk
@@ -218,3 +218,13 @@ OS_CFLAGS += --coverage
 LDFLAGS += --coverage
 DSO_LDOPTS += --coverage
 endif
+
+# Check for glibc
+# _TIME_BITS is available since 47f24c21ee38701ae275aa9e451f70fa3e77478c (glibc 2.34).
+ifneq ($(strip $(shell echo -en "#include<features.h>\n#ifndef __GLIBC__\n#else\n#if (__GLIBC__>=2)&&(__GLIBC_MINOR__>=34)\n#define HAS_LFS 1\n#endif\n#endif\nHAS_LFS" | gcc -P -E -xc -)),1)
+# If the platform is using glibc and is a 32-bit architecture ...
+ifneq ($(USE_64), 1)
+# Enable largefile and time64 support
+DEFINES += -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64
+endif
+endif
-- 
2.52.0

