From fe63edd2d65a417e13593230c9de4df5b2fdbcee Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Tue, 12 May 2026 10:20:06 +0800
Subject: [PATCH 2/2] rules: allow LDFLAGS to be used during shlib linkage

---
 coreconf/rules.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/coreconf/rules.mk b/coreconf/rules.mk
index afd1b75b4..d7b130cb3 100644
--- a/coreconf/rules.mk
+++ b/coreconf/rules.mk
@@ -217,7 +217,7 @@ ifdef MT
 endif	# MSVC with manifest tool
 endif
 else
-	$(MKSHLIB) -o $@ $(OBJS) $(SUB_SHLOBJS) $(LD_LIBS) $(EXTRA_LIBS) $(EXTRA_SHARED_LIBS) $(OS_LIBS)
+	$(MKSHLIB) $(LDFLAGS) -o $@ $(OBJS) $(SUB_SHLOBJS) $(LD_LIBS) $(EXTRA_LIBS) $(EXTRA_SHARED_LIBS) $(OS_LIBS)
 	chmod +x $@
 endif
 endif
-- 
2.52.0

