diff -Naur electron/toolchain.gypi electron.gcc/toolchain.gypi
--- electron/toolchain.gypi	2016-11-28 17:35:34.300491057 -0600
+++ electron.gcc/toolchain.gypi	2016-11-28 17:35:45.903519242 -0600
@@ -3,7 +3,7 @@
     # Clang stuff.
     'make_clang_dir%': 'vendor/llvm-build/Release+Asserts',
     # Set this to true when building with Clang.
-    'clang%': 1,
+    'clang%': 0,
 
     'variables': {
       # The minimum macOS SDK version to use.
