--- a/peach.pas	2021-01-15 15:04:15.869899477 +0800
+++ b/peach.pas	2021-01-15 15:04:26.957043007 +0800
@@ -4818,7 +4818,7 @@
   {$ENDIF}
   {$IFDEF LINUX}
   EXEEXT        = '';
-  UNRARNAME     = 'unrar-nonfree';
+  UNRARNAME     = 'unrar';
   APPLICATION2  = 'szcnick/p7zip 17.02 (LGPL, fork of Myspace p7zip, based on Igor Pavlov''s 7z);';
   APPLICATION3  = 'PAQ8F/JD/L/O, LPAQ1/5/8, ZPAQ 7.05 [Matt Mahoney et al. (GPL)];';
   APPLICATION4  = 'Strip (GPL, GNU binutils), UPX 3.96 (GPL, Markus F.X.J. Oberhumer, Laszlo Molnar and John F. Reiser);';
