From 5af6b2852125d5d3e4b8a017ea9028e5d46842f8 Mon Sep 17 00:00:00 2001
From: Adrian Bunk <bunk@debian.org>
Date: Tue, 23 Sep 2025 12:34:53 +0300
Subject: [PATCH] Use #include <unistd.h> for obtaining the prototype of
 write()

---
 getopt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/getopt.c b/getopt.c
index 3f37c55..aa95877 100644
--- a/getopt.c
+++ b/getopt.c
@@ -43,9 +43,9 @@
 #include <stdio.h>
 /* We're ANSI now; we're guaranteed to have strchr(). */
 #include <string.h>
+#include <unistd.h>
 
 #define ERR(s, c)	if(x_opterr){\
-	extern int write();\
 	char errbuf[2];\
 	errbuf[0] = c; errbuf[1] = '\n';\
 	(void) write(2, argv[0], (unsigned)strlen(argv[0]));\
-- 
2.55.0

