From: Bartosz Fenski <fenio@debian.org>
Subject: Correct absolute path, use plain background etc.

--- a/ipcalc.cgi.orig	2006-09-22 12:33:45.000000000 +0200
+++ b/ipcalc.cgi	2006-09-22 12:35:58.000000000 +0200
@@ -40,7 +40,7 @@
 # 0.26   27.07.2006 Replaced REQUEST_URI with SCRIPT_URL to prevent cross-site-scripting attacks
 
 $|=1;
-$ipcalc = "/usr/local/bin/ipcalc";
+$ipcalc = "/usr/bin/ipcalc";
 $MAIL_ADDRESS="ipcalc-200502&#64;jodies.de";
 # history:
 # 200404 
@@ -137,7 +137,6 @@
 <!--
 body {
    background-color: white;
-   background-image: url("bg.gif");
    color: black;
    font-family: "Trebuchet MS", Verdana, Geneva, Helvetica, sans-serif;
    l//ine-height: 110%;
@@ -205,7 +205,7 @@
 <table border=0 width="100%">
 <tr>
 <td valign="top">&nbsp;</td>
-<td><a href="http://jodies.de/ipcalc"><img src="ipcal03.gif" align=right width=100 height=95 border=0 alt=""></a></td></tr>
+<td><a href="http://jodies.de/ipcalc"><img src="http://localhost/images/ipcalc/ipcal03.gif" align=right width=100 height=95 border=0 alt=""></a></td></tr>
 </table>
 EOF
 if ($help) {
@@ -247,7 +247,7 @@
 
 
 <p>You can have all this fun at your shell prompt. Originally ipcalc was not intended for 
-creating HTML and still works happily in /usr/local/bin/ :-)</p>
+creating HTML and still works happily in /usr/bin/ :-)</p>
 
 <p>Questions? Comments? Drop me a <a href="mailto:$MAIL_ADDRESS">mail</a>...</p>
 
@@ -333,17 +333,15 @@
   
   <td nowrap valign=top>
    <a href="http://jodies.de/ipcalc">
-   <img src="ipcalculator.png" alt="Thanks to http://www.netzwerkinfo.de/daemons/ for this ip calculator icon :-)" border=0></a><br>
+   <img src="http://localhost/images/ipcalc/ipcalculator.png" alt="Thanks to http://www.netzwerkinfo.de/daemons/ for this ip calculator icon :-)" border=0></a><br>
    <tt><span style="font-size: 8pt;">Version $version</span></tt></td>
  </tr>
 </table>
 <hr>
 <a href="http://jodies.de/ipcalc-archive/">Download</a><br>
-<a href="ipcalc.png">Screenshot</a> (ipcalc works also at the prompt)<br>
 <a href="http://jodies.de/ipcalc_cgi">CGI wrapper that produced this page</a>.<br>
 <a href="ipcalc-archive">Archive</a><br>
 Have a look in the archives for the <b>new version 0.38</b>, with the capability to deaggregate network ranges<br>
-<a href="ipcalc-faq/win32.html">How to run this under windows</a><br>
 Debian users can apt-get install ipcalc<br>
 2000-2004 <a href="mailto:$MAIL_ADDRESS">Krischan Jodies</a>
 
