From 154962ff4d2beb0b01f6f159339b887bae1b415a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=93=BA=E7=9B=96=E5=B4=BD?= <i@pugai.life>
Date: Tue, 8 Apr 2025 09:42:08 +0800
Subject: [PATCH 1/2] fix-exec-path

---
 resources/xwayland-satellite.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/resources/xwayland-satellite.service b/resources/xwayland-satellite.service
index 060ab6b..f68cefd 100644
--- a/resources/xwayland-satellite.service
+++ b/resources/xwayland-satellite.service
@@ -8,7 +8,7 @@ Requisite=graphical-session.target
 [Service]
 Type=notify
 NotifyAccess=all
-ExecStart=/usr/local/bin/xwayland-satellite
+ExecStart=/usr/bin/xwayland-satellite
 StandardOutput=journal
 
 [Install]
-- 
2.52.0

