diff -Naur sonic-visualiser-3.3/main/MainWindow.cpp sonic-visualiser-3.3.noupdate/main/MainWindow.cpp
--- sonic-visualiser-3.3/main/MainWindow.cpp	2019-05-21 17:33:11.000000000 +0000
+++ sonic-visualiser-3.3.noupdate/main/MainWindow.cpp	2019-10-05 06:10:00.050185322 +0000
@@ -89,7 +89,6 @@
 
 #include "Surveyer.h"
 #include "NetworkPermissionTester.h"
-#include "framework/VersionTester.h"
 
 // For version information
 #include <vamp/vamp.h>
@@ -343,15 +342,8 @@
         SVDEBUG << "MainWindow: Creating surveyer" << endl;
         m_surveyer = new Surveyer
             ("sonicvisualiser.org", "survey23-present.txt", "survey23.php");
-
-        SVDEBUG << "MainWindow: Creating version tester" << endl;
-        m_versionTester = new VersionTester
-            ("sonicvisualiser.org", "latest-version.txt", SV_VERSION);
-        connect(m_versionTester, SIGNAL(newerVersionAvailable(QString)),
-                this, SLOT(newerVersionAvailable(QString)));
     } else {
         m_surveyer = nullptr;
-        m_versionTester = nullptr;
     }
 
 /*
@@ -374,7 +366,6 @@
     delete m_unitConverter;
     delete m_preferencesDialog;
     delete m_layerTreeDialog;
-    delete m_versionTester;
     delete m_surveyer;
     Profiles::getInstance()->dump();
 //    SVDEBUG << "MainWindow::~MainWindow finishing" << endl;
@@ -5080,10 +5071,6 @@
     QTextEdit *textEdit = new QTextEdit;
     layout->addWidget(textEdit, row++, 1, 1, 2);
 
-    if (m_newerVersionIs != "") {
-        layout->addWidget(new QLabel(tr("<b>Note:</b> A newer version of Sonic Visualiser is available.<br>(Version %1 is available; you are using version %2)").arg(m_newerVersionIs).arg(SV_VERSION)), row++, 1, 1, 2);
-    }
-    
     QDialogButtonBox *bb = new QDialogButtonBox(QDialogButtonBox::Ok);
     layout->addWidget(bb, row++, 0, 1, 3);
     connect(bb, SIGNAL(accepted()), d, SLOT(accept()));
@@ -5314,22 +5301,3 @@
 {
     m_keyReference->show();
 }
-
-void
-MainWindow::newerVersionAvailable(QString version)
-{
-    m_newerVersionIs = version;
-    
-    QSettings settings;
-    settings.beginGroup("NewerVersionWarning");
-    QString tag = QString("version-%1-available-show").arg(version);
-    if (settings.value(tag, true).toBool()) {
-        QString title(tr("Newer version available"));
-        QString text(tr("<h3>Newer version available</h3><p>You are using version %1 of Sonic Visualiser, but version %2 is now available.</p><p>Please see the <a href=\"http://sonicvisualiser.org/\">Sonic Visualiser website</a> for more information.</p>").arg(SV_VERSION).arg(version));
-        QMessageBox::information(this, title, text);
-        settings.setValue(tag, false);
-    }
-    settings.endGroup();
-}
-
-
diff -Naur sonic-visualiser-3.3/main/MainWindow.h sonic-visualiser-3.3.noupdate/main/MainWindow.h
--- sonic-visualiser-3.3/main/MainWindow.h	2019-05-21 17:33:10.000000000 +0000
+++ sonic-visualiser-3.3.noupdate/main/MainWindow.h	2019-10-05 06:13:22.002541205 +0000
@@ -18,7 +18,6 @@
 
 #include "framework/MainWindowBase.h"
 
-class VersionTester;
 class Surveyer;
 class LayerTreeDialog;
 class ActivityLog;
@@ -167,7 +166,6 @@
     virtual void about();
     virtual void whatsNew();
     virtual void keyReference();
-    void newerVersionAvailable(QString) override;
 
 protected:
     Overview                *m_overview;
@@ -231,8 +229,6 @@
     QFileSystemWatcher      *m_templateWatcher;
 
     Surveyer                *m_surveyer;
-    VersionTester           *m_versionTester;
-    QString                  m_newerVersionIs;
 
     struct LayerConfiguration {
         LayerConfiguration(LayerFactory::LayerType _layer
diff -Naur sonic-visualiser-3.3/svapp/framework/MainWindowBase.h sonic-visualiser-3.3.noupdate/svapp/framework/MainWindowBase.h
--- sonic-visualiser-3.3/svapp/framework/MainWindowBase.h	2019-05-21 17:33:31.000000000 +0000
+++ sonic-visualiser-3.3.noupdate/svapp/framework/MainWindowBase.h	2019-10-05 06:10:58.606051680 +0000
@@ -332,8 +332,6 @@
 
     virtual void emitHideSplash();
 
-    virtual void newerVersionAvailable(QString) { }
-
     virtual void menuActionMapperInvoked(QObject *);
 
 protected:
