diff -Naur sonic-visualiser-3.0.3/piper-cpp/vamp-capnp/piper.capnp.h sonic-visualiser-3.0.3.ease/piper-cpp/vamp-capnp/piper.capnp.h
--- sonic-visualiser-3.0.3/piper-cpp/vamp-capnp/piper.capnp.h	2017-05-25 08:26:10.000000000 +0000
+++ sonic-visualiser-3.0.3.ease/piper-cpp/vamp-capnp/piper.capnp.h	2017-11-07 05:27:20.131091134 +0000
@@ -6,11 +6,6 @@
 
 #include <capnp/generated-header-support.h>
 
-#if CAPNP_VERSION != 6000
-#error "Version mismatch between generated code and library headers.  You must use the same version of the Cap'n Proto compiler and library."
-#endif
-
-
 namespace capnp {
 namespace schemas {
 

