diff -Naur sonic-visualiser-3.0.3/sonic-visualiser.pro sonic-visualiser-3.0.3.notest/sonic-visualiser.pro
--- sonic-visualiser-3.0.3/sonic-visualiser.pro	2017-05-25 08:26:10.000000000 +0000
+++ sonic-visualiser-3.0.3.notest/sonic-visualiser.pro	2017-11-07 05:32:54.228218738 +0000
@@ -1,25 +1,12 @@
 
 TEMPLATE = subdirs
 
-# We build the tests on every platform, though at the time of
-# writing they are only automatically run on non-Windows platforms
-# (because of the difficulty of getting them running nicely in the
-# IDE without causing great confusion if a test fails).
-SUBDIRS += \
-        sub_test_svcore_base \
-        sub_test_svcore_data_fileio \
-        sub_test_svcore_data_model
-
 SUBDIRS += \
 	checker \
 	sub_server \
         sub_convert \
 	sub_sv
 
-sub_test_svcore_base.file = test-svcore-base.pro
-sub_test_svcore_data_fileio.file = test-svcore-data-fileio.pro
-sub_test_svcore_data_model.file = test-svcore-data-model.pro
-
 sub_server.file = server.pro
 sub_convert.file = convert.pro
 sub_sv.file = sv.pro

