From ed081d7b2a068932de38888aa729b87f6f42b6cf Mon Sep 17 00:00:00 2001
From: huhubun <nongzc@outlook.com>
Date: Wed, 24 Dec 2025 00:25:22 +0800
Subject: [PATCH] Update Exec and Icon to zen-browser

---
 build/AppDir/zen.desktop | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/build/AppDir/zen.desktop b/build/AppDir/zen.desktop
index 9480304..5215ceb 100644
--- a/build/AppDir/zen.desktop
+++ b/build/AppDir/zen.desktop
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Name=Zen Browser
 Comment=Experience tranquillity while browsing the web without people tracking you!
-Exec=zen %u
-Icon=zen
+Exec=zen-browser %u
+Icon=zen-browser
 Type=Application
 MimeType=text/html;text/xml;application/xhtml+xml;x-scheme-handler/http;x-scheme-handler/https;application/x-xpinstall;application/pdf;application/json;
 StartupWMClass=zen
@@ -15,12 +15,12 @@ Actions=new-window;new-private-window;profilemanager;
 
 [Desktop Action new-window]
 Name=Open a New Window
-Exec=zen %u
+Exec=zen-browser %u
 
 [Desktop Action new-private-window]
 Name=Open a New Private Window
-Exec=zen --private-window %u
+Exec=zen-browser --private-window %u
 
 [Desktop Action profilemanager]
 Name=Open the Profile Manager
-Exec=zen --ProfileManager %u
+Exec=zen-browser --ProfileManager %u
-- 
2.52.0

