From 9c50edb9c3dc802aa0a48e0dfefb1057668ccdc1 Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <the@salted.fish>
Date: Mon, 20 Jul 2026 22:14:10 -0400
Subject: [PATCH 3/8] AOSCOS: Remove Google and Mozilla API key file path

---
 configs/common/mozconfig | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/configs/common/mozconfig b/configs/common/mozconfig
index 0ae698e3b90e5..0d14f732a5010 100644
--- a/configs/common/mozconfig
+++ b/configs/common/mozconfig
@@ -34,19 +34,6 @@ if test "$SCCACHE_GHA_ENABLED" = "true"; then
   mk_add_options 'export CCACHE_CPP2=yes'
 fi
 
-# add API keys if it exists on a file
-if test -f "$HOME/.zen-keys/safebrowsing.dat"; then
-  ac_add_options --with-google-safebrowsing-api-keyfile="$HOME/.zen-keys/safebrowsing.dat"
-fi
-
-if test -f "$HOME/.zen-keys/mozilla.dat"; then
-  ac_add_options --with-mozilla-api-keyfile="$HOME/.zen-keys/mozilla.dat"
-fi
-
-if test -f "$HOME/.zen-keys/google_location_service.dat"; then
-  ac_add_options --with-google-location-service-api-keyfile="$HOME/.zen-keys/google_location_service.dat"
-fi
-
 if test "$ZEN_RELEASE"; then
 
   ac_add_options --enable-clang-plugin
-- 
2.52.0

