From 13185058374a686ba9c8faa4a4e7d1804f48a63f Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Mon, 2 Feb 2026 23:32:26 -0800
Subject: [PATCH 5/7] AOSCOS: Update Google API keyfile path

---
 configs/common/mozconfig | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/configs/common/mozconfig b/configs/common/mozconfig
index 22b8e95c5..673f54d70 100644
--- a/configs/common/mozconfig
+++ b/configs/common/mozconfig
@@ -38,8 +38,9 @@ if test "$SCCACHE_GHA_ENABLED" = "true"; then
 fi
 
 # add API keys if it exists on a file
-if test -f "$HOME/.zen-keys/safebrowsing.dat"; then
-  ac_add_options --with-google-safebrowsing-api-keyfile="$HOME/.zen-keys/safebrowsing.dat"
+if test -f "$SRCDIR/autobuild/google-api-key"; then
+  ac_add_options --with-google-location-service-api-keyfile="$SRCDIR/autobuild/google-api-key"
+  ac_add_options --with-google-safebrowsing-api-keyfile="$SRCDIR/autobuild/google-api-key"
 fi
 
 if test -f "$HOME/.zen-keys/mozilla.dat"; then
@@ -93,7 +94,7 @@ if test "$ZEN_RELEASE"; then
   ac_add_options --disable-crashreporter
 
   # Experimental flag, enabled only on nightly for Firefox.
-  # Should bring in some nice performance improvements, 
+  # Should bring in some nice performance improvements,
   # but may cause stability issues.
   ac_add_options --enable-replace-malloc
 fi
-- 
2.52.0

