From 4257a5b4313c9c892b2986021e3bc22c4e556273 Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Mon, 2 Feb 2026 23:32:26 -0800
Subject: [PATCH 6/8] AOSCOS: Update Google API keyfile path

---
 configs/common/mozconfig | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/configs/common/mozconfig b/configs/common/mozconfig
index 3d41bc83aff94..297b559a1089a 100644
--- a/configs/common/mozconfig
+++ b/configs/common/mozconfig
@@ -36,8 +36,9 @@ if ! test "$SCCACHE_GHA_ENABLED" = "false"; then
 fi
 
 # add safe browsing key if it exists on a file
-if test -f "$HOME/.zen-keys/safebrowsing.dat"; then
-  ac_add_options --with-google-safebrowsing-api-keyfile="$HOME/.zen-keys/safebrowsing.dat"
+if test -f "$SRCDIR/autobuild/google-api-key"; then
+  ac_add_options --with-google-location-service-api-keyfile="$SRCDIR/autobuild/google-api-key"
+  ac_add_options --with-google-safebrowsing-api-keyfile="$SRCDIR/autobuild/google-api-key"
 fi
 
 if test "$ZEN_RELEASE"; then
-- 
2.52.0

