diff -Naur konsole-17.12.3/desktop/org.kde.konsole.desktop konsole-17.12.3.keyword/desktop/org.kde.konsole.desktop
--- konsole-17.12.3/desktop/org.kde.konsole.desktop	2018-03-01 22:54:01.000000000 +0000
+++ konsole-17.12.3.keyword/desktop/org.kde.konsole.desktop	2018-04-14 03:15:40.437768889 +0000
@@ -11,6 +11,8 @@
 StartupNotify=true
 X-KDE-AuthorizeAction=shell_access
 
+Keywords=terminal;
+
 Name=Konsole
 Name[af]=Konsole
 Name[ar]=كونسول
