diff -Naur rt-8u172-b00.old/buildSrc/linux.gradle rt-8u172-b00/buildSrc/linux.gradle
--- rt-8u172-b00.old/buildSrc/linux.gradle      2017-11-05 01:15:32.720575732 -0500
+++ rt-8u172-b00/buildSrc/linux.gradle  2017-11-05 01:17:06.316970448 -0500
@@ -46,6 +46,10 @@
         "-fstack-protector",
         "-Wextra", "-Wall", "-Wformat-security", "-Wno-unused", "-Wno-parentheses", "-Werror=implicit-function-declaration"] // warning flags

+if (System.env.CFLAGS) {
+    commonFlags = System.env.CFLAGS.tokenize(' ')
+}
+
 if (!IS_64) {
     commonFlags += "-m32"
 }

