diff -Naur rt-8u172-b00.old/buildSrc/linux.gradle rt-8u172-b00/buildSrc/linux.gradle
--- rt-8u172-b00.old/buildSrc/linux.gradle      2017-11-05 01:15:32.720575732 -0500
+++ rt-8u172-b00/buildSrc/linux.gradle  2017-11-05 01:17:06.316970448 -0500
@@ -45,6 +45,10 @@
         "-fno-strict-aliasing", "-fPIC", "-fno-omit-frame-pointer", // optimization flags
         "-W", "-Wall", "-Wno-unused", "-Wno-parentheses", "-Werror=implicit-function-declaration"] // warning flags
 
+if (System.env.CFLAGS) {
+    commonFlags = System.env.CFLAGS.tokenize(' ')
+}
+
 if (!IS_64) {
     commonFlags += "-m32"
 }

--- a/build.gradle     2018-02-01 19:40:35.616271756 -0500
+++ b/build.gradle     2018-02-01 19:46:40.152153968 -0500
@@ -2564,7 +2564,9 @@
                 } else if (t.name == "linux") {
                     cmakeArgs = "-DCMAKE_SYSTEM_NAME=Linux"
                     cmakeArgs = "-DCMAKE_SYSTEM_NAME=Linux"
                     if (IS_64) {
-                        cmakeArgs = "$cmakeArgs -DCMAKE_SYSTEM_PROCESSOR=x86_64"
+                        def ccCmakeFlags = System.env.CFLAGS.trim()
+                        def cxxCmakeFlags = System.env.CXXFLAGS.trim()
+                        cmakeArgs = "-DCMAKE_C_FLAGS=\"${ccCmakeFlags}\" -DCMAKE_CXX_FLAGS=\"${cxxCmakeFlags}\" -DCMAKE_SYSTEM_PROCESSOR=x86_64"
                     } else {
                         cmakeArgs = "$cmakeArgs -DCMAKE_SYSTEM_PROCESSOR=i586 -DCMAKE_C_FLAGS=-m32 -DCMAKE_CXX_FLAGS=-m32"
