--- a/build.gradle	2019-01-26 22:30:22.784116983 -0700
+++ b/build.gradle	2019-01-26 22:30:49.827445418 -0700
@@ -713,7 +713,7 @@
     throw new Exception("Unknown and unsupported build architecture: $OS_ARCH")
 } else if (IS_MAC && OS_ARCH != "x86_64") {
     throw new Exception("Unknown and unsupported build architecture: $OS_ARCH")
-} else if (IS_LINUX && OS_ARCH != "i386" && OS_ARCH != "amd64") {
+} else if (IS_LINUX && false) {
     throw new Exception("Unknown and unsupported build architecture: $OS_ARCH")
 }
 
