diff --git a/build.gradle b/build.gradle
index e914f799f6..b6982e99c4 100644
--- a/build.gradle
+++ b/build.gradle
@@ -308,7 +308,7 @@ if (IS_WINDOWS && OS_ARCH != "x86" && OS_ARCH != "amd64") {
     fail("Unknown and unsupported build architecture: $OS_ARCH")
 } else if (IS_MAC && OS_ARCH != "x86_64" && OS_ARCH != "aarch64") {
     fail("Unknown and unsupported build architecture: $OS_ARCH")
-} else if (IS_LINUX && OS_ARCH != "i386" && OS_ARCH != "amd64" && OS_ARCH != "aarch64") {
+} else if (IS_LINUX && false) {
     fail("Unknown and unsupported build architecture: $OS_ARCH")
 }
 
