diff -Naur rt-88287370703c.old/modules/media/src/main/native/gstreamer/3rd_party/glib/glib-2.56.1/glib/gstrfuncs.c rt-88287370703c/modules/media/src/main/native/gstreamer/3rd_party/glib/glib-2.56.1/glib/gstrfuncs.c
--- rt-88287370703c.old/modules/media/src/main/native/gstreamer/3rd_party/glib/glib-2.56.1/glib/gstrfuncs.c	2018-12-28 23:57:57.162860775 -0700
+++ rt-88287370703c/modules/media/src/main/native/gstreamer/3rd_party/glib/glib-2.56.1/glib/gstrfuncs.c	2018-12-28 23:58:49.799229502 -0700
@@ -39,7 +39,7 @@
 
 #ifdef HAVE_XLOCALE_H
 /* Needed on BSD/OS X for e.g. strtod_l */
-#include <xlocale.h>
+#include <locale.h>
 #endif
 
 #ifdef G_OS_WIN32
diff -Naur rt-88287370703c.old/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h rt-88287370703c/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h
--- rt-88287370703c.old/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h	2018-12-28 23:57:57.236193692 -0700
+++ rt-88287370703c/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h	2018-12-28 23:58:21.166058058 -0700
@@ -26,7 +26,7 @@
 #include <locale.h>
 #endif
 #ifdef HAVE_XLOCALE_H
-#include <xlocale.h>
+#include <locale.h>
 #endif
 
 typedef locale_t xsltLocale;
