--- rt-81e1980f9d54/modules/jdk.packager/src/main/native/library/common/PosixPlatform.cpp	2021-02-15 00:11:13.000000000 -0800
+++ rt-81e1980f9d54-new/modules/jdk.packager/src/main/native/library/common/PosixPlatform.cpp	2021-05-10 02:28:42.581351966 -0700
@@ -43,7 +43,11 @@
 #include <stdbool.h>
 #include <sys/types.h>
 #include <unistd.h>
+#ifdef linux
+#include <linux/sysctl.h>
+#else
 #include <sys/sysctl.h>
+#endif
 #include <sys/file.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
