From 4857db542e91cdc8c5d8c8a39aae43db4f97fdff Mon Sep 17 00:00:00 2001
From: Christian Franke <christian.franke@t-online.de>
Date: Fri, 6 Jun 2025 15:23:13 +0200
Subject: [PATCH 1/2] UPSTREAM: Add automake 1.18 to the list of tested
 versions

Signed-off-by: Christian Franke <christian.franke@t-online.de>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 smartmontools/autogen.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/autogen.sh b/autogen.sh
index ad80e708..405276e9 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -23,7 +23,7 @@ if [ -n "$AUTOMAKE" ]; then
   ver=$("$AUTOMAKE" --version) || exit 1
 else
   maxver=
-  for v in 1.17 1.16 1.15 1.14 1.13; do
+  for v in 1.18 1.17 1.16 1.15 1.14 1.13; do
     minver=$v; test -n "$maxver" || maxver=$v
     ver=$(automake-$v --version 2>/dev/null) || continue
     AUTOMAKE="automake-$v"
@@ -59,7 +59,7 @@ case "$ver" in
     # OK
     ;;
 
-  1.14|1.14.1|1.15|1.15.1|1.16|1.16.[1-5]|1.17)
+  1.14|1.14.1|1.15|1.15.1|1.16|1.16.[1-5]|1.17|1.18)
     # TODO: Enable 'subdir-objects' in configure.ac
     # For now, suppress 'subdir-objects' forward-incompatibility warning
     test -n "$warnings" || amwarnings="--warnings=no-unsupported"
-- 
2.51.0

