From 08b44eb291a93fe3f734c25be83b2062324180b0 Mon Sep 17 00:00:00 2001
From: Christian Franke <christian.franke@t-online.de>
Date: Sat, 12 Jul 2025 11:57:10 +0200
Subject: [PATCH 2/2] BACKPORT: UPSTREAM: Add automake 1.18.1 to the list of
 tested versions

Signed-off-by: Christian Franke <christian.franke@t-online.de>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 smartmontools/autogen.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/autogen.sh b/autogen.sh
index 405276e9..761b1a2a 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -55,7 +55,7 @@ case "$ver" in
     echo "GNU Automake $ver is not supported."; exit 1
     ;;
 
-  1.13.[34])
+  1.13.[34]|1.14|1.14.1|1.15|1.15.1|1.16|1.16.[1-5]|1.17|1.18|1.18.1)
     # OK
     ;;
 
-- 
2.51.0

