From 321e98a09b84a0b404428760c856532d7bf692ce Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 24 Aug 2026 23:06:22 +0800
Subject: [PATCH] AOSCOS: pyproject.toml: lift flit_core version limit

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 2e979bf..1a24ca2 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["flit_core>=3.12,<4"]
+requires = ["flit_core>=3.12"]
 build-backend = "flit_core.buildapi"
 
 [project]
-- 
2.55.0

