From 2318925199780bb8530b7ec1cabc8a9fd2b931f3 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Mon, 24 Feb 2020 23:19:11 +0800
Subject: [PATCH] add environment variables for PlaMo apps

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 data/phosh.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/data/phosh.in b/data/phosh.in
index b4854f5..ab1bce6 100755
--- a/data/phosh.in
+++ b/data/phosh.in
@@ -19,6 +19,12 @@ if [ -f "/etc/phosh/rootston.ini" ]; then
   ROOTSTON_INI=/etc/phosh/rootston.ini
 fi
 
+export EGL_PLATFORM=wayland
+export QT_QPA_PLATFORM=wayland
+export QT_WAYLAND_DISABLE_WINDOWDECORATION=1
+export PLASMA_PLATFORM=phone:handset
+export QT_QUICK_CONTROLS_MOBILE=true
+
 # Run gnome-session through a login shell so it picks
 # variables from /etc/profile.d (XDG_*)
 exec "${COMPOSITOR}" -C "${ROOTSTON_INI}" -E "bash -lc 'gnome-session --disable-acceleration-check --session=phosh'"
-- 
2.24.1

