--- webkitgtk-2.40.0/Source/WebCore/platform/graphics/SourceBrush.cpp	2023-02-20 09:22:18.621742500 +0000
+++ webkitgtk-2.40.0.gcc13/Source/WebCore/platform/graphics/SourceBrush.cpp	2023-04-19 07:20:22.960398765 +0000
@@ -65,12 +65,12 @@
 
 void SourceBrush::setGradient(Ref<Gradient>&& gradient, const AffineTransform& spaceTransform)
 {
-    m_brush = { Brush::LogicalGradient { WTFMove(gradient), spaceTransform } };
+    m_brush = Brush { Brush::LogicalGradient { WTFMove(gradient), spaceTransform } };
 }
 
 void SourceBrush::setPattern(Ref<Pattern>&& pattern)
 {
-    m_brush = { WTFMove(pattern) };
+    m_brush = Brush { WTFMove(pattern) };
 }
 
 WTF::TextStream& operator<<(TextStream& ts, const SourceBrush& brush)
