From 862714ced43f51fc3c5b3ccbf472fc79e392d269 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <origincode@aosc.io>
Date: Fri, 28 Mar 2025 05:11:12 -0700
Subject: [PATCH 7/7] Supply -mlasx to enable LASX intrinsics on loongarch64

Signed-off-by: Kaiyang Wu <origincode@aosc.io>
---
 Source/ThirdParty/skia/CMakeLists.txt | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/Source/ThirdParty/skia/CMakeLists.txt b/Source/ThirdParty/skia/CMakeLists.txt
index 92ed41cf..44ba62f6 100644
--- a/Source/ThirdParty/skia/CMakeLists.txt
+++ b/Source/ThirdParty/skia/CMakeLists.txt
@@ -1067,6 +1067,7 @@ endif ()
 #
 set(Skia_SkCMS_HSW_OPTS FALSE)
 set(Skia_SkCMS_SKX_OPTS FALSE)
+set(Skia_SkCMS_LASX_OPTS FALSE)
 
 set(Skia_SkCMS_HSW_FLAGS
     -ffp-contract=off
@@ -1081,6 +1082,9 @@ set(Skia_SkCMS_SKX_FLAGS
     -mavx512bw
     -mavx512vl
 )
+set(Skia_SkCMS_LASX_FLAGS
+    -mlasx
+)
 
 if (WTF_CPU_X86 OR WTF_CPU_X86_64)
     target_compile_definitions(Skia PRIVATE SK_ENABLE_AVX512_OPTS)
@@ -1092,6 +1096,12 @@ if (WTF_CPU_X86 OR WTF_CPU_X86_64)
     WEBKIT_ADD_COMPILER_FLAGS(CXX SOURCE src/opts/SkOpts_skx.cpp "-march=skylake-avx512")
 endif ()
 
+if (WTF_CPU_LOONGARCH64)
+    WEBKIT_CHECK_COMPILER_FLAGS(
+        CXX Skia_SkCMS_LASX_OPTS
+        ${SKIA_SkCMS_LASX_FLAGS})
+endif ()
+
 if (Skia_SkCMS_HSW_OPTS)
     WEBKIT_ADD_COMPILER_FLAGS(CXX SOURCE modules/skcms/src/skcms_TransformHsw.cc ${Skia_SkCMS_HSW_FLAGS})
 else ()
@@ -1104,4 +1114,8 @@ else ()
     target_compile_definitions(Skia PRIVATE SKCMS_DISABLE_SKX)
 endif ()
 
+if (Skia_SkCMS_LASX_OPTS)
+    WEBKIT_APPEND_GLOBAL_COMPILER_FLAGS(${Skia_SkCMS_LASX_FLAGS})
+endif ()
+
 add_library(Skia::Skia ALIAS Skia)
-- 
2.49.0

