From 6e7d65a45e1b5a79ef19632d4b67f0d6465739c7 Mon Sep 17 00:00:00 2001
From: ilikara <3435193369@qq.com>
Date: Tue, 5 Aug 2025 08:57:39 +0800
Subject: [PATCH 3/4] AOSCOS: src/netlib.c: include <sys/sendfile.h> when
 HAVE_SENDFILE is defined

Signed-off-by: ilikara <3435193369@qq.com>
---
 src/netlib.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/netlib.c b/src/netlib.c
index fe0e5e5..d0b07d1 100644
--- a/src/netlib.c
+++ b/src/netlib.c
@@ -148,6 +148,10 @@ char    netlib_id[]="\
 #include <sys/mpctl.h>
 #endif
 
+#ifdef HAVE_SENDFILE
+#include <sys/sendfile.h>
+#endif
+
 #if !defined(HAVE_GETADDRINFO) || !defined(HAVE_GETNAMEINFO)
 # include "missing/getaddrinfo.h"
 #endif
-- 
2.50.1

