From 126706fa9ae04517db57bc4cb9bb978c1177df19 Mon Sep 17 00:00:00 2001
From: ilikara <3435193369@qq.com>
Date: Tue, 5 Aug 2025 09:03:02 +0800
Subject: [PATCH 4/4] AOSCOS: src/net_uuid.c: include <unistd.h>

Signed-off-by: ilikara <3435193369@qq.com>
---
 src/net_uuid.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/net_uuid.c b/src/net_uuid.c
index 8701120..ecb5746 100644
--- a/src/net_uuid.c
+++ b/src/net_uuid.c
@@ -43,6 +43,8 @@
 #define snprintf _snprintf
 #else
 
+#include <unistd.h>
+
 #if HAVE_SYS_TYPES_H
 #include <sys/types.h>
 #else
-- 
2.50.1

