From ed47fa80997a221129326d59dbbe8174342ae12b Mon Sep 17 00:00:00 2001
From: Rong Bao <rong.bao@csmantle.top>
Date: Tue, 17 Feb 2026 18:25:57 +0800
Subject: [PATCH] Fix intltool path for non-Debian systems

Signed-off-by: Rong Bao <rong.bao@csmantle.top>
---
 Makefile         | 4 ++--
 debconf-updatepo | 2 +-
 po2debconf       | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index a2fab21..3d93fd6 100644
--- a/Makefile
+++ b/Makefile
@@ -9,14 +9,14 @@ fix: debconf-gettextize debconf-updatepo po2debconf
 
 debconf-updatepo po2debconf: FORCE
 	@sed \
-   -e 's@\(: \$${PODEBCONF_LIB=\)[^}]*}@\1$(prefix)/share/intltool-debian}@' \
+   -e 's@\(: \$${PODEBCONF_LIB=\)[^}]*}@\1$(prefix)/bin}@' \
    -e 's@\(: \$${PODEBCONF_ENCODINGS=\)[^}]*}@\1$(prefix)/share/po-debconf/encodings}@' \
 		$@ > $@.tmp && mv $@.tmp $@
 	@chmod a+x $@
 
 debconf-gettextize: FORCE
 	@sed \
-   -e 's@\(\$$ENV{PODEBCONF_LIB} ||= \)[^;]*;@\1"$(prefix)/share/intltool-debian";@' \
+   -e 's@\(\$$ENV{PODEBCONF_LIB} ||= \)[^;]*;@\1"$(prefix)/bin";@' \
    -e 's@\(\$$ENV{PODEBCONF_ENCODINGS} ||= \)[^;]*;@\1"$(prefix)/share/po-debconf/encodings";@' \
    -e 's@\(\$$ENV{PODEBCONF_HEADER} ||= \)[^;]*;@\1"$(prefix)/share/po-debconf/pot-header";@' \
 		$@ > $@.tmp && mv $@.tmp $@
diff --git a/debconf-updatepo b/debconf-updatepo
index 194c137..7d6316e 100755
--- a/debconf-updatepo
+++ b/debconf-updatepo
@@ -20,7 +20,7 @@
 #
 #   This script is part of po-debconf
 
-: ${PODEBCONF_LIB=/usr/share/intltool-debian}
+: ${PODEBCONF_LIB=/usr/bin}
 INTLTOOL_EXTRACT=${PODEBCONF_LIB}/intltool-extract
 export INTLTOOL_EXTRACT
 
diff --git a/po2debconf b/po2debconf
index 56c5a5b..2922037 100755
--- a/po2debconf
+++ b/po2debconf
@@ -20,7 +20,7 @@
 #
 #   This script is part of po-debconf
 
-: ${PODEBCONF_LIB=/usr/share/intltool-debian}
+: ${PODEBCONF_LIB=/usr/bin}
 : ${PODEBCONF_ENCODINGS=/usr/share/po-debconf/encodings}
 
 #   Prevent automatic conversion to UTF-8 by Perl
-- 
2.51.0

