Author: Osamu Aoki <osamu@debian.org>
--- a/configure
+++ b/configure
@@ -101,7 +101,7 @@
 	echo "uninstall: uninstall-i18n" >> Makefile.settings
 fi
 
-AXEL_CFLAGS="-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Os ${CFLAGS}"
+AXEL_CFLAGS="-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 ${CFLAGS}"
 if [ "$debug" = "1" ]; then
 	AXEL_CFLAGS="${AXEL_CFLAGS} -g"
 	echo 'DEBUG=1' >> Makefile.settings
@@ -232,7 +232,9 @@
 esac
 
 echo "CFLAGS=${AXEL_CFLAGS}" >> Makefile.settings
+echo "CFLAGS += `dpkg-buildflags --get CPPFLAGS`" >> Makefile.settings
 echo "LFLAGS=${AXEL_LFLAGS}" >> Makefile.settings
+echo "LFLAGS += `dpkg-buildflags --get LDFLAGS`" >> Makefile.settings
 
 echo 'Configuration done:'
 if [ "$i18n" = "1" ]; then
