From 8d21389323d2a7f0587e4f6efe6aa99457ec933a Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Mon, 24 Nov 2025 05:09:43 +0800
Subject: [PATCH 1/2] be liberal in what version to accept

---
 src/proto.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/proto.c b/src/proto.c
index 8d82dc1..15eb0de 100644
--- a/src/proto.c
+++ b/src/proto.c
@@ -54,6 +54,12 @@ gcry_error_t otrl_init(unsigned int ver_major, unsigned int ver_minor,
 {
     unsigned int api_version;
 
+/* See:
+ * https://bugzilla.redhat.com/show_bug.cgi?id=1634321
+ * https://github.com/psi-im/plugins/issues/7#issuecomment-423048948
+ */
+#if 0
+
     /* The major versions have to match, and you can't be using a newer
      * minor version than we expect. */
     if (ver_major != OTRL_VERSION_MAJOR || ver_minor > OTRL_VERSION_MINOR) {
@@ -63,6 +69,7 @@ gcry_error_t otrl_init(unsigned int ver_major, unsigned int ver_minor,
 		OTRL_VERSION_MAJOR, OTRL_VERSION_MINOR, OTRL_VERSION_SUB);
 	return gcry_error(GPG_ERR_INV_VALUE);
     }
+#endif
 
     /* Set the API version.  If we get called multiple times for some
      * reason, take the smallest value. */
-- 
2.51.1

