From bf588e0df5cbcdad0965a043cbcefe1b98e5d20b Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Mon, 24 Nov 2025 05:13:09 +0800
Subject: [PATCH 2/2] fix implicit function declaration

---
 tests/regression/client/client.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/regression/client/client.c b/tests/regression/client/client.c
index e72b661..908b875 100644
--- a/tests/regression/client/client.c
+++ b/tests/regression/client/client.c
@@ -27,6 +27,7 @@
 #include <syscall.h>
 #include <sys/epoll.h>
 #include <sys/types.h>
+#include <sys/socket.h>
 #include <sys/un.h>
 #include <unistd.h>
 
-- 
2.51.1

