diff --git a/externals/dynarmic/src/dynarmic/backend/x64/exception_handler_posix.cpp b/externals/dynarmic/src/dynarmic/backend/x64/exception_handler_posix.cpp
index ff01ab606..bdaac0c3e 100644
--- a/externals/dynarmic/src/dynarmic/backend/x64/exception_handler_posix.cpp
+++ b/externals/dynarmic/src/dynarmic/backend/x64/exception_handler_posix.cpp
@@ -64,7 +64,7 @@ class SigHandler {
 SigHandler sig_handler;
 
 SigHandler::SigHandler() {
-    constexpr size_t signal_stack_size = std::max(SIGSTKSZ, 2 * 1024 * 1024);
+    const size_t signal_stack_size = std::max<size_t>(SIGSTKSZ, 2 * 1024 * 1024);
 
     signal_stack_memory = std::malloc(signal_stack_size);
 
