From 66cc42b894d26d05bfaa147e30ba5c10b5d16390 Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint@balintreczey.hu>
Date: Sat, 26 Mar 2016 01:55:49 +0100
Subject: [PATCH 2/3] [core] added arm64 support to GetKernelCpuFamily()

---
 xbmc/utils/SystemInfo.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/utils/SystemInfo.cpp b/xbmc/utils/SystemInfo.cpp
index eb85abb..fcbfec3 100644
--- a/xbmc/utils/SystemInfo.cpp
+++ b/xbmc/utils/SystemInfo.cpp
@@ -991,7 +991,7 @@ const std::string& CSysInfo::GetKernelCpuFamily(void)
     if (uname(&un) == 0)
     {
       std::string machine(un.machine);
-      if (machine.compare(0, 3, "arm", 3) == 0)
+      if (machine.compare(0, 3, "arm", 3) == 0 || machine.compare(0, 7, "aarch64", 7) == 0)
         kernelCpuFamily = "ARM";
       else if (machine.compare(0, 4, "mips", 4) == 0)
         kernelCpuFamily = "MIPS";
-- 
2.1.4


