--- a/src/CMakeLists.txt	2021-09-02 00:53:18.839910999 -0500
+++ b/src/CMakeLists.txt	2021-09-02 00:11:42.318643959 -0500
@@ -388,6 +388,9 @@
     elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
       message("Host processor is 64bit ARM")
       sm_add_compile_definition("${SM_EXE_NAME}" CPU_AARCH64)
+    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc")
+      message("Host processor is PowerPC")
+      sm_add_compile_definition("${SM_EXE_NAME}" CPU_PPC)
     else()
       message("Unrecognized host processor type")
     endif()
