From 8a5850358d3bbd2329ae19b78406b5b296838df2 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 9 Feb 2024 15:06:37 +0800
Subject: [PATCH 2/2] Fix up .desktop entry

---
 stepmania.desktop | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/stepmania.desktop b/stepmania.desktop
index 4e3fb8ee74..84c9b67857 100644
--- a/stepmania.desktop
+++ b/stepmania.desktop
@@ -1,12 +1,9 @@
-
 [Desktop Entry]
-Encoding=UTF-8
 Name=StepMania
 GenericName=Rhythm and dance game
-TryExec=stepmania
-Exec=stepmania
+Exec=/usr/lib/stepmania/stepmania
 Terminal=false
 Icon=stepmania-ssc
 Type=Application
-Categories=Application;Game;ArcadeGame
+Categories=Game;ArcadeGame;
 Comment=A cross-platform rhythm video game.
-- 
2.43.0

