From 7efb6298139e68610d4cec8361c791b160fc1bb4 Mon Sep 17 00:00:00 2001
From: Cherrling <cherrling2021@outlook.com>
Date: Thu, 31 Jul 2025 12:27:23 +0800
Subject: [PATCH] FROM AOSC: disable updater

---
 component/updater/update_core.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/component/updater/update_core.go b/component/updater/update_core.go
index fedf6518..627e640e 100644
--- a/component/updater/update_core.go
+++ b/component/updater/update_core.go
@@ -73,6 +73,10 @@ func (u *CoreUpdater) Update(currentExePath string) (err error) {
 	u.mu.Lock()
 	defer u.mu.Unlock()
 
+ 	if true {
+ 		return fmt.Errorf("Please update using package manager")
+        }
+	
 	_, err = os.Stat(currentExePath)
 	if err != nil {
 		return fmt.Errorf("check currentExePath %q: %w", currentExePath, err)
-- 
2.50.1

