From f44894ea5a0c7753c5ecd0556b0a99ef888fcc5f Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Fri, 4 Oct 2024 10:03:53 +0800
Subject: [PATCH] FROM AOSC: disable updater

Signed-off-by: xtex <xtexchooser@duck.com>
---
 component/updater/update_core.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/component/updater/update_core.go b/component/updater/update_core.go
index 2aab7833ea2c..9687c53f4beb 100644
--- a/component/updater/update_core.go
+++ b/component/updater/update_core.go
@@ -71,6 +71,11 @@ func UpdateCore(execPath string) (err error) {
 	mu.Lock()
 	defer mu.Unlock()
 
+	if true {
+		err = &updateError{Message: "please update using package manager"}
+		return err
+	}
+
 	latestVersion, err = getLatestVersion()
 	if err != nil {
 		return err

base-commit: 4a16d22398116a88a230071278be536491a8cdce
-- 
2.46.2

