Index: xamarin-pkg-msbuild/mono/build/install.proj
===================================================================
--- xamarin-pkg-msbuild.orig/mono/build/install.proj
+++ xamarin-pkg-msbuild/mono/build/install.proj
@@ -100,7 +100,7 @@
 
         <Exec
             WorkingDirectory="$(XBuildDir)"
-            Command="ln -sfh Current 15.0" />
+            Command="ln -sfn Current 15.0" />
         <Exec
             WorkingDirectory="$(MonoInstallPrefix)\lib\mono\msbuild"
             Command="ln -s Current 15.0" />
