diff -Naur a/scripts/xdg-screensaver.in b/scripts/xdg-screensaver.in
--- a/scripts/xdg-screensaver.in	2015-10-05 14:59:19.000000000 -0400
+++ b/scripts/xdg-screensaver.in	2017-04-25 17:22:58.932024964 -0400
@@ -102,6 +102,11 @@
       screensaver_mate_screensaver "$1"
       ;;
 
+    slock)
+      slock
+      exit 0
+      ;;
+
     xscreensaver)
       screensaver_xscreensaver "$1"
       ;;
@@ -788,6 +793,8 @@
 dbus-send --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.GetNameOwner string:org.gnome.ScreenSaver > /dev/null 2>&1 && DE="gnome_screensaver"
 # Consider "mate-screensaver" a separate DE
 dbus-send --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.GetNameOwner string:org.mate.ScreenSaver > /dev/null 2>&1 && DE="mate_screensaver"
+# Consider "slock" a separate DE
+slock -v > /dev/null 2>&1 && DE="slock"
 # Consider "xautolock" a separate DE
 xautolock -enable > /dev/null 2>&1 && DE="xautolock_screensaver"
 
