diff -Naur portmidi.old/CMakeLists.txt portmidi/CMakeLists.txt
--- portmidi.old/CMakeLists.txt	2017-07-08 22:53:40.168463954 +0800
+++ portmidi/CMakeLists.txt	2017-07-08 23:28:41.243647825 +0800
@@ -2,7 +2,7 @@
 # Roger B. Dannenberg
 # 20 Sep 2009
 
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.8)
 
 if(UNIX)
   # allow user to set Release or Debug
diff -Naur portmidi.old/pm_java/CMakeLists.txt portmidi/pm_java/CMakeLists.txt
--- portmidi.old/pm_java/CMakeLists.txt	2017-07-08 22:56:45.352858917 +0800
+++ portmidi/pm_java/CMakeLists.txt	2017-07-09 00:03:22.759235786 +0800
@@ -5,43 +5,22 @@
     # java not dealt with in CMake -- see pm_mac/pm_mac.xcodeproj
   else(APPLE)
     # linux
+    find_package(Java REQUIRED)
+    include(UseJava)
     set(JPORTMIDICLASS JPortMidi.class JPortMidiException.class
     		       JPortMidiApi.class)
     set(PMDEFAULTSCLASS PmDefaultsFrame.class PmDefaults.class)
     prepend_path(JPORTMIDICLASS2 jportmidi/ ${JPORTMIDICLASS})
     prepend_path(PMDEFAULTSCLASS2 pmdefaults/ ${PMDEFAULTSCLASS})
     set(PMDEFAULTS_ALL_CLASSES ${JPORTMIDICLASS2} ${PMDEFAULTSCLASS2})
-    # message(STATUS "PMDEFAULTS_ALL_CLASSES is " ${PMDEFAULTS_ALL_CLASSES})
-    add_custom_command(OUTPUT pmdefaults/PmDefaultsFrame.class
-        COMMAND javac -classpath . pmdefaults/PmDefaultsFrame.java
-	MAIN_DEPENDENCY pmdefaults/PmDefaultsFrame.java
-	DEPENDS pmdefaults/PmDefaults.java
-	WORKING_DIRECTORY pm_java)
-    add_custom_command(OUTPUT pmdefaults/PmDefaults.class
-        COMMAND javac -classpath . pmdefaults/PmDefaults.java
-	MAIN_DEPENDENCY pmdefaults/PmDefaults.java
-	DEPENDS pmdefaults/PmDefaultsFrame.java
-	WORKING_DIRECTORY pm_java)
-    add_custom_command(OUTPUT ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pmdefaults.jar
-        COMMAND	cp pmdefaults/portmusic_logo.png .
-        COMMAND	jar cmf pmdefaults/manifest.txt pmdefaults.jar
-		pmdefaults/*.class portmusic_logo.png jportmidi/*.class
-  	COMMAND chmod +x pmdefaults/pmdefaults
-	COMMAND cp pmdefaults/pmdefaults ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
-	COMMAND mv pmdefaults.jar ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
-	COMMAND rm portmusic_logo.png
-	MAIN_DEPENDENCY pmdefaults/PmDefaults.class
-	DEPENDS ${PMDEFAULTS_ALL_CLASSES}
-	WORKING_DIRECTORY pm_java)
-    add_custom_target(pmdefaults_target ALL 
-        DEPENDS ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pmdefaults.jar)
-    # message(STATUS "add_custom_target: pmdefaults.jar")
-
+    file(COPY pmdefaults/portmusic_logo.png DESTINATION ${CMAKE_CURRENT_SOURCE_DIR})
+    file(COPY pmdefaults/pmdefaults DESTINATION ${CMAKE_BINARY_DIR}/pm_java/)
+    add_jar(pmdefaults MANIFEST pmdefaults/manifest.txt pmdefaults/PmDefaultsFrame.java pmdefaults/PmDefaults.java portmusic_logo.png)
     # install the libraries (Linux only)
-    INSTALL(FILES ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pmdefaults.jar
+    INSTALL(FILES ${CMAKE_BINARY_DIR}/pm_java/pmdefaults.jar
       DESTINATION /usr/share/java)
-    INSTALL(PROGRAMS ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/pmdefaults
-      DESTINATION /usr/local/bin)
+    INSTALL(PROGRAMS ${CMAKE_BINARY_DIR}/pm_java/pmdefaults
+      DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
   endif(APPLE)
 endif(UNIX)
 # In windows, use pm_java/make.bat
diff -Naur portmidi.old/pm_linux/finddefault.c portmidi/pm_linux/finddefault.c
--- portmidi.old/pm_linux/finddefault.c	2017-07-08 22:56:45.356192398 +0800
+++ portmidi/pm_linux/finddefault.c	2017-07-08 23:25:06.653275039 +0800
@@ -5,6 +5,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include <ctype.h>
 #include "portmidi.h"
 
 #define STRING_MAX 256
