Description: Fix FTBFS with GCC 7.
Author: Alessandro Furieri <a.furieri@lqt.it>
Origin: https://www.gaia-gis.it/fossil/spatialite_gui/fdiff?v1=78060ac75797c6bb&v2=d7a4a13c171c06f5&sbs=1
Bug-Debian: https://bugs.debian.org/853667

--- a/ResultSetView.cpp
+++ b/ResultSetView.cpp
@@ -494,7 +494,7 @@ const char *MyResultSetView::CleanSqlTai
 {
 // strips any leading white-space
   const char *p = dirty;
-  while (p != '\0')
+  while (*p != '\0')
     {
       if (*p == ' ')
         {
