Description: Add support for OpenJPEG 2.3.
Author: Bas Couwennberg <sebastic@debian.org>

--- a/config.h.in
+++ b/config.h.in
@@ -103,6 +103,9 @@
 /* Define to 1 if you have the <openjpeg-2.2/openjpeg.h> header file. */
 #undef HAVE_OPENJPEG_2_2_OPENJPEG_H
 
+/* Define to 1 if you have the <openjpeg-2.3/openjpeg.h> header file. */
+#undef HAVE_OPENJPEG_2_3_OPENJPEG_H
+
 /* Define to 1 if you have the <sqlite3ext.h> header file. */
 #undef HAVE_SQLITE3EXT_H
 
--- a/configure.ac
+++ b/configure.ac
@@ -200,13 +200,16 @@ AC_ARG_ENABLE(openjpeg, [AS_HELP_STRING(
     # they could be either on -/include/openjpeg-2.0
     #                   or on -/include/openjpeg-2.1
     #                   or on -/include/openjpeg-2.2
+    #                   or on -/include/openjpeg-2.3
     #
     AC_CHECK_HEADERS(openjpeg-2.0/openjpeg.h)
     AC_CHECK_HEADERS(openjpeg-2.1/openjpeg.h)
     AC_CHECK_HEADERS(openjpeg-2.2/openjpeg.h)
+    AC_CHECK_HEADERS(openjpeg-2.3/openjpeg.h)
     if test x"$ac_cv_header_openjpeg_2_0_openjpeg_h" != x"yes" &&
         test x"$ac_cv_header_openjpeg_2_1_openjpeg_h" != x"yes" &&
-        test x"$ac_cv_header_openjpeg_2_2_openjpeg_h" != x"yes";
+        test x"$ac_cv_header_openjpeg_2_2_openjpeg_h" != x"yes" &&
+        test x"$ac_cv_header_openjpeg_2_3_openjpeg_h" != x"yes";
     then
         AC_MSG_ERROR(['OpenJpeg-2' is required but the header (openjpeg.h) doesn't seem to be installed on this system])
     fi 
