From 130ba3812160ea576c05dd887e6801b8419f8e0c Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 24 Aug 2026 20:32:20 +0800
Subject: [PATCH 2/3] AOSCOS: multipathd: allow disabling Valgrind

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 multipathd/main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/multipathd/main.c b/multipathd/main.c
index d11a8576..46fc3042 100644
--- a/multipathd/main.c
+++ b/multipathd/main.c
@@ -83,7 +83,11 @@
 #include "dmevents.h"
 #include "io_err_stat.h"
 #include "foreign.h"
+#ifndef NO_VALGRIND
 #include "../third-party/valgrind/drd.h"
+#else
+#define ANNOTATE_BENIGN_RACE_SIZED(...)
+#endif
 #include "init_unwinder.h"
 
 #define CMDSIZE 160
-- 
2.55.0

