From 0533e5f3660934cee842caa623fb24211ea62005 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 24 Aug 2026 20:36:22 +0800
Subject: [PATCH 3/3] AOSCOS: Makefile.inc: install executables to /usr/bin

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 Makefile.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.inc b/Makefile.inc
index 9e3dc466..9b6af18d 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -62,7 +62,7 @@ tmpfilesdir	:= $(systemd_prefix)lib/tmpfiles.d
 modulesloaddir	:= $(systemd_prefix)lib/modules-load.d
 libudevdir	:= $(systemd_prefix)lib/udev
 udevrulesdir	:= $(libudevdir)/rules.d
-bindir		:= $(exec_prefix)sbin
+bindir		:= $(exec_prefix)bin
 mandir		:= $(usr_prefix)share/man
 LIB		:= $(if $(shell test -d /lib64 && echo 1),lib64,lib)
 syslibdir	:= $(prefix)$(LIB)
-- 
2.55.0

