From c645e67d9bdbcc469a9fe0ca21f8defc886df8c1 Mon Sep 17 00:00:00 2001
From: Carl Smedstad <carl.smedstad@protonmail.com>
Date: Sun, 17 Nov 2024 11:11:37 +0100
Subject: [PATCH 6/6] Fix Python 3.12 compatibility

---
 src/ieee1284module.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ieee1284module.c b/src/ieee1284module.c
index c6d6928..23a4426 100644
--- a/src/ieee1284module.c
+++ b/src/ieee1284module.c
@@ -579,7 +579,7 @@ WRITE_METHOD(ecp_write_addr)
 /***/
 
 static PyTypeObject ParportType = {
-	PyObject_HEAD_INIT(NULL)
+	PyVarObject_HEAD_INIT(NULL, 0)
 	"ieee1284.Parport",	/* const char *tp_name; */
 	sizeof (ParportObject),	/* Py_ssize_t tp_basicsize */
 	0,		/* Py_ssize_t tp_itemsize; */
-- 
2.49.0

