From 8e8f8c1238bfba759fa9dd16f1242d062b0e57f6 Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan@linux.intel.com>
Date: Sat, 2 Mar 2024 16:21:37 +0000
Subject: [PATCH 4/7] "on" is also good, as reported by Paul Menzel

---
 src/tuning/runtime.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/tuning/runtime.cpp b/src/tuning/runtime.cpp
index cb62533..e62fc6c 100644
--- a/src/tuning/runtime.cpp
+++ b/src/tuning/runtime.cpp
@@ -95,6 +95,8 @@ int runtime_tunable::good_bad(void)
 
 	if (strcmp(content.c_str(), "auto") == 0)
 		return TUNE_GOOD;
+	if (strcmp(content.c_str(), "on") == 0)
+		return TUNE_GOOD;
 
 	return TUNE_BAD;
 }
-- 
2.49.0

