diff --git a/hipamd/src/hip_graph_internal.cpp b/hipamd/src/hip_graph_internal.cpp
index 8ccab6a53e..f563724f3a 100644
--- a/hipamd/src/hip_graph_internal.cpp
+++ b/hipamd/src/hip_graph_internal.cpp
@@ -21,6 +21,17 @@
 #include "hip_graph_internal.hpp"
 #include <queue>

+#if defined(__aarch64__)
+#define _mm_sfence(ptr) asm volatile("DMB ishst":::"memory")
+#define _mm_mfence(ptr) asm volatile("DMB ish":::"memory")
+#elif defined(__loongarch_lp64)
+#define _mm_sfence(ptr) asm volatile ("dbar 0":::"memory")
+#define _mm_mfence(ptr) asm volatile ("dbar 0":::"memory")
+#elif defined(__riscv) && __riscv_xlen == 64
+#define _mm_sfence(ptr) asm volatile("fence w,w":::"memory")
+#define _mm_mfence(ptr) asm volatile("fence rw,rw":::"memory")
+#endif
+
 #define CASE_STRING(X, C)                                                                          \
   case X:                                                                                          \
     case_string = #C;                                                                              \
diff --git a/rocclr/device/rocm/rocvirtual.cpp b/rocclr/device/rocm/rocvirtual.cpp
index 5a0392292a..a3298a2c48 100644
--- a/rocclr/device/rocm/rocvirtual.cpp
+++ b/rocclr/device/rocm/rocvirtual.cpp
@@ -50,6 +50,15 @@
 #else
 #include <immintrin.h>
 #endif
+#elif defined(__aarch64__)
+#define _mm_sfence(ptr) asm volatile("DMB ishst":::"memory")
+#define _mm_mfence(ptr) asm volatile("DMB ish":::"memory")
+#elif defined(__loongarch_lp64)
+#define _mm_sfence(ptr) asm volatile ("dbar 0":::"memory")
+#define _mm_mfence(ptr) asm volatile ("dbar 0":::"memory")
+#elif defined(__riscv) && __riscv_xlen == 64
+#define _mm_sfence(ptr) asm volatile("fence w,w":::"memory")
+#define _mm_mfence(ptr) asm volatile("fence rw,rw":::"memory")
 #endif

 /**
diff --git a/rocclr/include/top.hpp b/rocclr/include/top.hpp
index f67d6c72f4..6b3f20acaf 100644
--- a/rocclr/include/top.hpp
+++ b/rocclr/include/top.hpp
@@ -28,6 +28,8 @@
     defined(__x86__) || defined(__x86_64) || defined(__x86_64__) || defined(__amd64) ||            \
     defined(__amd64__) || defined(_M_X64) || defined(_M_AMD64)
 #define ATI_ARCH_X86
+#elif defined(__riscv)
+#define ATI_ARCH_RISCV
 #endif

 #if defined(ATI_ARCH_ARM)
diff --git a/rocclr/os/os_posix.cpp b/rocclr/os/os_posix.cpp
index ddc060b0c7..5ea8a451bb 100644
--- a/rocclr/os/os_posix.cpp
+++ b/rocclr/os/os_posix.cpp
@@ -679,6 +679,12 @@ address Os::currentStackPtr() {
 #elif defined(ATI_ARCH_ARM)
       "mov %0,sp"
       : "=r"(value)
+#elif defined(__loongarch_lp64)
+      "addi.d %0,$sp,0"
+      : "=r"(value)
+#elif defined(ATI_ARCH_RISCV)
+      "mv %0,sp"
+      : "=r"(value)
 #else
       ""
 #endif
