diff --git a/rocclr/os/os_posix.cpp b/rocclr/os/os_posix.cpp
--- a/rocclr/os/os_posix.cpp	(revision dea3af7584625fdb5005ad5372d3a2e6a492706d)
+++ b/rocclr/os/os_posix.cpp	(date 1774860741997)
@@ -712,6 +712,9 @@
 #elif defined(ATI_ARCH_ARM)
       "mov %0,sp"
       : "=r"(value)
+#elif defined(ATI_ARCH_LOONGARCH)
+      "addi.d %0,$sp,0"
+      : "=r"(value)
 #else
       ""
 #endif
diff --git a/rocclr/include/top.hpp b/rocclr/include/top.hpp
--- a/rocclr/include/top.hpp	(revision dea3af7584625fdb5005ad5372d3a2e6a492706d)
+++ b/rocclr/include/top.hpp	(date 1774860866489)
@@ -14,6 +14,8 @@
     defined(__x86__) || defined(__x86_64) || defined(__x86_64__) || defined(__amd64) ||            \
     defined(__amd64__) || defined(_M_X64) || defined(_M_AMD64)
 #define ATI_ARCH_X86
+#elif defined(__loongarch_lp64)
+#define ATI_ARCH_LOONGARCH
 #endif
 
 #if defined(ATI_ARCH_ARM)

