From 292ff32cfbea9dcac24d69be845ec6f9eb9340d0 Mon Sep 17 00:00:00 2001
From: Qing Yun <kf@aosc.io>
Date: Tue, 12 May 2026 10:33:12 +0800
Subject: [PATCH] AOSCOS: remove platform limit

---
 umu/umu_runtime.py | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/umu/umu_runtime.py b/umu/umu_runtime.py
index 8494e00..a0152d1 100644
--- a/umu/umu_runtime.py
+++ b/umu/umu_runtime.py
@@ -507,15 +507,6 @@ RUNTIME_VERSIONS.update({
 })
 # fmt: on
 
-if platform.machine() == "x86_64":  # noqa: SIM114
-    pass
-elif platform.machine() == "aarch64":
-    pass
-else:
-    err: str = f"Unsupported platform {platform.machine()}"
-    raise RuntimeError(err)
-
-
 RUNTIME_NAMES = {RUNTIME_VERSIONS[key].name: key for key in RUNTIME_VERSIONS}
 
 
-- 
2.52.0

