From a2eaf2d29ae8b8f73ea3b640221a0664226bb9ad Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Wed, 13 Sep 2017 15:24:56 +0100
Subject: [PATCH] fwupd: Do not use deprecated API

---
 plugins/fwupd/gs-plugin-fwupd.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/plugins/fwupd/gs-plugin-fwupd.c b/plugins/fwupd/gs-plugin-fwupd.c
index 4a12bea2..112efbd7 100644
--- a/plugins/fwupd/gs-plugin-fwupd.c
+++ b/plugins/fwupd/gs-plugin-fwupd.c
@@ -351,11 +351,16 @@ gs_plugin_fwupd_new_app_from_results (GsPlugin *plugin, FwupdResult *res)
 	FwupdDevice *dev = fwupd_result_get_device (res);
 	FwupdRelease *rel = fwupd_result_get_release (res);
 	GsApp *app;
-	const gchar *id;
+	g_autofree gchar *id = NULL;
 	g_autoptr(AsIcon) icon = NULL;
 
 	/* get from cache */
-	id = fwupd_result_get_unique_id (res);
+	id = as_utils_unique_id_build (AS_APP_SCOPE_SYSTEM,
+				       AS_BUNDLE_KIND_UNKNOWN,
+				       NULL, /* origin */
+				       AS_APP_KIND_FIRMWARE,
+				       fwupd_release_get_appstream_id (rel),
+				       NULL);
 	app = gs_plugin_cache_lookup (plugin, id);
 	if (app == NULL) {
 		app = gs_plugin_app_new (plugin, id);
