Description: Remove not working items in Help menu
Origin: other, https://build.opensuse.org/package/view_file?expand=1&file=remove-non-working-help.patch&package=xsane&project=devel%3AopenSUSE%3AFactory
Forwarded: no

Index: xsane-0.998/src/xsane.c
===================================================================
--- xsane-0.998.orig/src/xsane.c	2013-01-27 01:44:39.000000000 +0100
+++ xsane-0.998/src/xsane.c	2013-01-29 01:14:28.983590459 +0100
@@ -4260,27 +4260,6 @@
   gtk_widget_show(item);
 
 
-  /* Backend doc -> html viewer */
-
-  if (xsane.backend)
-  {
-    item = gtk_menu_item_new_with_label(MENU_ITEM_BACKEND_DOC);
-    gtk_menu_append(GTK_MENU(menu), item);
-    g_signal_connect(GTK_OBJECT(item), "activate", (GtkSignalFunc) xsane_show_doc, (void *) xsane.backend);
-    gtk_widget_add_accelerator(item, "activate", xsane.accelerator_group, GDK_F2, 0, GTK_ACCEL_VISIBLE | DEF_GTK_ACCEL_LOCKED);
-    gtk_widget_show(item);
-  }
-
-
-  /* available backends -> html viewer */
-
-  item = gtk_menu_item_new_with_label(MENU_ITEM_AVAILABLE_BACKENDS);
-  gtk_menu_append(GTK_MENU(menu), item);
-  g_signal_connect(GTK_OBJECT(item), "activate", (GtkSignalFunc) xsane_show_doc, (void *) "sane-backends");
-  gtk_widget_add_accelerator(item, "activate", xsane.accelerator_group, GDK_F3, 0, GTK_ACCEL_VISIBLE | DEF_GTK_ACCEL_LOCKED);
-  gtk_widget_show(item);
-
-  
   /* problems -> html viewer */
 
   item = gtk_menu_item_new_with_label(MENU_ITEM_PROBLEMS);
