From 2024894dfdff3c2467478aaaa62db471d08421b0 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Tue, 5 Mar 2024 14:41:35 -0800
Subject: [PATCH] fix libutf8proc header file path

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 fcft.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fcft.c b/fcft.c
index 09728ab..8685ed0 100644
--- a/fcft.c
+++ b/fcft.c
@@ -25,7 +25,7 @@
  #include <harfbuzz/hb-ft.h>
 #endif
 #if defined(FCFT_HAVE_UTF8PROC)
- #include <utf8proc.h>
+ #include <libutf8proc/utf8proc.h>
 #endif
 
 #include <tllist.h>
-- 
2.44.0

