From d3357cbf36655dbf1b48158fcfbb2ad0a4a48c97 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Fri, 10 Apr 2026 12:12:16 +0800
Subject: [PATCH] AOSCOS: Fix build with ffmpeg >= 8.0

---
 src/pipewirerecord.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/pipewirerecord.cpp b/src/pipewirerecord.cpp
index ac8132e..11b5a86 100644
--- a/src/pipewirerecord.cpp
+++ b/src/pipewirerecord.cpp
@@ -242,7 +242,9 @@ void PipeWireRecordProduce::stateChanged(pw_stream_state state)
     qCDebug(PIPEWIRERECORD_LOGGING) << "finished";
     if (m_avCodecContext) {
         avio_closep(&m_avFormatContext->pb);
+#if LIBAVCODEC_VERSION_MAJOR < 61
         avcodec_close(m_avCodecContext);
+#endif
         av_free(m_avCodecContext);
         avformat_free_context(m_avFormatContext);
     }
@@ -555,7 +557,7 @@ void PipeWireRecordWrite::addFrame(const QImage &image, std::optional<int> seque
 
     // Let's add a key frame every 100 frames and also the first frame
     if (sequential && (*sequential == 0 || (*sequential - m_lastKeyFrame) > 100)) {
-        avFrame.m_avFrame->key_frame = 1;
+        avFrame.m_avFrame->flags |= AV_FRAME_FLAG_KEY;
         m_lastKeyFrame = *sequential;
     }
 
-- 
2.53.0

