From 2974e568e98d50a72fa2e72b6083dbdb314baa2f Mon Sep 17 00:00:00 2001
From: Frank Liberato <liberato@chromium.org>
Date: Wed, 7 Jul 2021 19:01:22 -0700
Subject: [PATCH 19/21] Add av_stream_get_first_dts for Chromium

---
 libavformat/avformat.h | 4 ++++
 libavformat/utils.c    | 7 +++++++
 2 files changed, 11 insertions(+)

diff --git a/libavformat/avformat.h b/libavformat/avformat.h
index 56c1c8028996..75221d9d1a1e 100644
--- a/libavformat/avformat.h
+++ b/libavformat/avformat.h
@@ -1202,6 +1202,10 @@ typedef struct AVStreamGroup {
 
 struct AVCodecParserContext *av_stream_get_parser(const AVStream *s);
 
+// Chromium: We use the internal field first_dts vvv
+int64_t    av_stream_get_first_dts(const AVStream *st);
+// Chromium: We use the internal field first_dts ^^^
+
 #define AV_PROGRAM_RUNNING 1
 
 /**
diff --git a/libavformat/utils.c b/libavformat/utils.c
index e9ded627ad67..399c777f2a24 100644
--- a/libavformat/utils.c
+++ b/libavformat/utils.c
@@ -44,6 +44,13 @@
  * various utility functions for use within FFmpeg
  */
 
+// Chromium: We use the internal field first_dts vvv
+int64_t av_stream_get_first_dts(const AVStream *st)
+{
+  return cffstream(st)->first_dts;
+}
+// Chromium: We use the internal field first_dts ^^^
+
 /* an arbitrarily chosen "sane" max packet size -- 50M */
 #define SANE_CHUNK_SIZE (50000000)
 
-- 
2.48.1.windows.1

