diff --git a/libavcodec/cbs_av1_syntax_template.c b/libavcodec/cbs_av1_syntax_template.c
index 5518544a4d..1b0f96e37f 100644
--- a/libavcodec/cbs_av1_syntax_template.c
+++ b/libavcodec/cbs_av1_syntax_template.c
@@ -936,10 +936,20 @@ static int FUNC(loop_filter_params)(CodedBitstreamContext *ctx, RWContext *rw,
                 infer(loop_filter_mode_deltas[i], ref_loop_filter_mode_deltas[i]);
         }
     } else {
+        const int8_t *ref_loop_filter_ref_deltas, *ref_loop_filter_mode_deltas;
+        if (current->primary_ref_frame == AV1_PRIMARY_REF_NONE) {
+            ref_loop_filter_ref_deltas = default_loop_filter_ref_deltas;
+            ref_loop_filter_mode_deltas = default_loop_filter_mode_deltas;
+        } else {
+            ref_loop_filter_ref_deltas =
+                priv->ref[current->ref_frame_idx[current->primary_ref_frame]].loop_filter_ref_deltas;
+            ref_loop_filter_mode_deltas =
+                priv->ref[current->ref_frame_idx[current->primary_ref_frame]].loop_filter_mode_deltas;
+        }
         for (i = 0; i < AV1_TOTAL_REFS_PER_FRAME; i++)
-            infer(loop_filter_ref_deltas[i], default_loop_filter_ref_deltas[i]);
+            infer(loop_filter_ref_deltas[i], ref_loop_filter_ref_deltas[i]);
         for (i = 0; i < 2; i++)
-            infer(loop_filter_mode_deltas[i], default_loop_filter_mode_deltas[i]);
+            infer(loop_filter_mode_deltas[i], ref_loop_filter_mode_deltas[i]);
     }
 
     return 0;
diff --git a/libavcodec/vaapi_av1.c b/libavcodec/vaapi_av1.c
index b8f5472c14..d5245edfbe 100644
--- a/libavcodec/vaapi_av1.c
+++ b/libavcodec/vaapi_av1.c
@@ -236,6 +236,7 @@ static int vaapi_av1_start_frame(AVCodecContext *avctx,
             .disable_frame_end_update_cdf = frame_header->disable_frame_end_update_cdf,
             .uniform_tile_spacing_flag    = frame_header->uniform_tile_spacing_flag,
             .allow_warped_motion          = frame_header->allow_warped_motion,
+            .reserved                     = frame_header->refresh_frame_flags,
         },
         .loop_filter_info_fields.bits = {
             .sharpness_level        = frame_header->loop_filter_sharpness,
diff --git a/libavcodec/vaapi_encode_h264.c b/libavcodec/vaapi_encode_h264.c
index 27f6551719..73ada8cdc6 100644
--- a/libavcodec/vaapi_encode_h264.c
+++ b/libavcodec/vaapi_encode_h264.c
@@ -951,6 +951,7 @@ static const VAAPIEncodeProfile vaapi_encode_h264_profiles[] = {
 #endif
     { AV_PROFILE_H264_HIGH, 8, 3, 1, 1, VAProfileH264High },
     { AV_PROFILE_H264_MAIN, 8, 3, 1, 1, VAProfileH264Main },
+    { AV_PROFILE_H264_MAIN, 8, 3, 1, 0, VAProfileH264Main },
     { AV_PROFILE_H264_CONSTRAINED_BASELINE,
                             8, 3, 1, 1, VAProfileH264ConstrainedBaseline },
     { AV_PROFILE_UNKNOWN }
diff --git a/libavcodec/vaapi_h264.c b/libavcodec/vaapi_h264.c
index 7f00da09fc..d2a7225d50 100644
--- a/libavcodec/vaapi_h264.c
+++ b/libavcodec/vaapi_h264.c
@@ -144,7 +144,7 @@ static int fill_vaapi_ReferenceFrames(VAPictureParameterBufferH264 *pic_param,
 
     for (i = 0; i < h->short_ref_count; i++) {
         const H264Picture *pic = h->short_ref[i];
-        if (pic && pic->reference && dpb_add(&dpb, pic) < 0)
+        if (pic && pic->reference && pic->field_poc[0] != INT_MAX && pic->field_poc[1] != INT_MAX && dpb_add(&dpb, pic) < 0)
             return -1;
     }
 
@@ -265,8 +265,8 @@ static int vaapi_h264_start_frame(AVCodecContext          *avctx,
             .log2_max_pic_order_cnt_lsb_minus4      = sps->log2_max_poc_lsb - 4,
             .delta_pic_order_always_zero_flag       = sps->delta_pic_order_always_zero_flag,
         },
-        .pic_init_qp_minus26                        = pps->init_qp - 26,
-        .pic_init_qs_minus26                        = pps->init_qs - 26,
+        .pic_init_qp_minus26                        = pps->init_qp - 26 - 6 *(sps->bit_depth_luma - 8),
+        .pic_init_qs_minus26                        = pps->init_qs - 26 - 6 *(sps->bit_depth_luma - 8),       
         .chroma_qp_index_offset                     = pps->chroma_qp_index_offset[0],
         .second_chroma_qp_index_offset              = pps->chroma_qp_index_offset[1],
         .pic_fields.bits = {
