From 6fd143b375e894a711187c79a0bf16a38d954d9f Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu@aosc.io>
Date: Fri, 9 Jan 2026 12:00:47 +0800
Subject: [PATCH] Hantro VA-API struct ABI match

---
 libavcodec/vaapi_av1.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libavcodec/vaapi_av1.c b/libavcodec/vaapi_av1.c
index 5451b6055b..327c329cf0 100644
--- a/libavcodec/vaapi_av1.c
+++ b/libavcodec/vaapi_av1.c
@@ -235,6 +235,7 @@ static int vaapi_av1_start_frame(AVCodecContext *avctx,
             .disable_frame_end_update_cdf = frame_header->disable_frame_end_update_cdf,
             .uniform_tile_spacing_flag    = frame_header->uniform_tile_spacing_flag,
             .allow_warped_motion          = frame_header->allow_warped_motion,
+            .reserved                     = frame_header->refresh_frame_flags,
         },
         .loop_filter_info_fields.bits = {
             .sharpness_level        = frame_header->loop_filter_sharpness,
-- 
2.52.0

