From 662e890ea591ab7a90c08211e708435aabfdf5ed Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Mon, 5 Sep 2016 02:12:59 +0000
Subject: [PATCH] configure: remove hard-coded stack-protector

---
 configure.in | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/configure.in b/configure.in
index 29b63c2..c0f0aba 100644
--- a/configure.in
+++ b/configure.in
@@ -871,23 +871,6 @@ if test "$GCC" = yes; then
       RUBY_TRY_CFLAGS(-D_FORTIFY_SOURCE=2, [RUBY_APPEND_OPTION(XCFLAGS, -D_FORTIFY_SOURCE=2)])
    ])
 
-    # -fstack-protector
-    AS_CASE(["$target_os"],
-    [mingw*|nacl], [
-	stack_protector=no
-    ])
-    if test -z "${stack_protector+set}"; then
-	RUBY_TRY_CFLAGS(-fstack-protector, [stack_protector=yes], [stack_protector=no])
-	if test "x$stack_protector" = xyes; then
-	    RUBY_TRY_LDFLAGS(-fstack-protector, [], [stack_protector=broken])
-	fi
-    fi
-    if test "x$stack_protector" = xyes; then
-	RUBY_APPEND_OPTION(XCFLAGS, -fstack-protector)
-	RUBY_APPEND_OPTION(XLDFLAGS, -fstack-protector)
-	RUBY_APPEND_OPTION(LDFLAGS, -fstack-protector)
-    fi
-
     AS_CASE(["$target_os"],[mingw*], [
       # On  Windows  platforms,   system  provided  headers  are  VC++
       # optimized.  That  is, C++  habits are often  contaminated into
-- 
2.7.3

