From 1175915a7d1da6cb3cb39e0e305d0ddb7de1af95 Mon Sep 17 00:00:00 2001
From: Kay Lin <i@v2bv.net>
Date: Fri, 12 Mar 2021 02:53:27 -0800
Subject: [PATCH 5/5] [autotools] Accept V=1 for controlling verbosity

Signed-off-by: Kay Lin <i@v2bv.net>
---
 Makefile.in | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 3e32149..99625f1 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -16,7 +16,12 @@
 # with thin-provisioning-tools.  If not, see
 # <http://www.gnu.org/licenses/>.
 
-V=@
+V?=0
+ifeq ($(V), 0)
+    Q=@
+else
+    Q=
+endif
 
 PROGRAMS=\
 	bin/pdata_tools
@@ -220,35 +225,35 @@ endif
 %.o: %.cc
 	@echo "    [CXX] $<"
 	@mkdir -p $(dir $@)
-	$(V) $(CXX) -c $(INCLUDES) $(CXXFLAGS) -o $@ $<
-	$(V) $(CXX) -MM -MT $(subst .cc,.o,$<) $(INCLUDES) $(TEST_INCLUDES) $(CXXFLAGS) $< > $*.$$$$; \
+	$(Q) $(CXX) -c $(INCLUDES) $(CXXFLAGS) -o $@ $<
+	$(Q) $(CXX) -MM -MT $(subst .cc,.o,$<) $(INCLUDES) $(TEST_INCLUDES) $(CXXFLAGS) $< > $*.$$$$; \
 	sed 's,\([^ :]*\)\.o[ :]*,\1.o \1.gmo $* : Makefile ,g' < $*.$$$$ > $*.d; \
 	$(RM) $*.$$$$
 
 %.o: %.c
 	@echo "    [CC] $<"
 	@mkdir -p $(dir $@)
-	$(V) $(CC) -c $(INCLUDES) $(CFLAGS) -o $@ $<
-	$(V) $(CC) -MM -MT $(subst .cc,.o,$<) $(INCLUDES) $(TEST_INCLUDES) $(CFLAGS) $< > $*.$$$$; \
+	$(Q) $(CC) -c $(INCLUDES) $(CFLAGS) -o $@ $<
+	$(Q) $(CC) -MM -MT $(subst .cc,.o,$<) $(INCLUDES) $(TEST_INCLUDES) $(CFLAGS) $< > $*.$$$$; \
 	sed 's,\([^ :]*\)\.o[ :]*,\1.o \1.gmo $* : Makefile ,g' < $*.$$$$ > $*.d; \
 	$(RM) $*.$$$$
 
 %.8: %.txt bin/txt2man
 	@echo "    [txt2man] $<"
 	@mkdir -p $(dir $@)
-	$(V) bin/txt2man -p -t $(basename $(notdir $<)) $< > $@
+	$(Q) bin/txt2man -p -t $(basename $(notdir $<)) $< > $@
 
 #----------------------------------------------------------------
 
 lib/libpdata.a: $(OBJECTS) $(EMITTERS)
 	@echo "    [AR]  $<"
 	@mkdir -p $(dir $@)
-	$(V)$(AR) -rv $@ $(OBJECTS) $(EMITTERS) > /dev/null 2>&1
+	$(Q)$(AR) -rv $@ $(OBJECTS) $(EMITTERS) > /dev/null 2>&1
 
 bin/pdata_tools: $(OBJECTS) $(EMITTERS)
 	@echo "    [LD]  $@"
 	@mkdir -p $(dir $@)
-	$(V) $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@ $+ $(LIBS) $(CXXLIB)
+	$(Q) $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@ $+ $(LIBS) $(CXXLIB)
 
 #----------------------------------------------------------------
 
@@ -368,7 +373,7 @@ LIBFT_OBJECTS=$(subst .c,.o,$(LIBFT_SOURCE))
 
 lib/libft.so: $(LIBFT_OBJECTS)
 	@echo "    [LD]" $@
-	$(V) gcc -shared -o $@ $+ -laio
+	$(Q) gcc -shared -o $@ $+ -laio
 
 .PHONEY: functional-test unit-test
 
-- 
2.30.2

