From 2e8c8ccd27675a89d9b1f88e7761335a1d2d9528 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Tue, 13 Apr 2021 23:35:07 -0700
Subject: [PATCH] Makefile: drop update-icon-caches

---
 Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 6dd53bd..5bbeabb 100644
--- a/Makefile
+++ b/Makefile
@@ -61,7 +61,6 @@ install-desktop:
 		${PREFIX}/share/icons/hicolor/scalable/apps/easy-ebook-viewer.svg
 	install -m 644 misc/easy-ebook-viewer.desktop \
 		${PREFIX}/share/applications/easy-ebook-viewer.desktop
-	update-icon-caches ${PREFIX}/share/icons/hicolor
 
 clean:
 	rm -f easy-ebook-viewer
@@ -76,4 +75,4 @@ uninstall-desktop:
 	rm -f ${PREFIX}/share/applications/easy-ebook-viewer.desktop
 	rm -f ${PREFIX}/share/icons/hicolor/*/apps/easy-ebook-viewer.png
 
-.PHONY: all install install-bin install-desktop
\ No newline at end of file
+.PHONY: all install install-bin install-desktop
-- 
2.30.2

