From 4043b3d0973103163dd9865dd9f5db1f0cee30ad Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Tue, 24 Jun 2025 02:07:23 -0700
Subject: [PATCH] chore: bump original version

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index e356f3ee..6f16b1ed 100644
--- a/meson.build
+++ b/meson.build
@@ -11,7 +11,7 @@ project(
 	],
 )
 
-original_version = '1.10.1'
+original_version = '1.11'
 
 add_project_arguments(
 	[
-- 
2.50.0

