From c6c753797f6669a1a6dc4b9839c625911aaf2cd5 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Mon, 7 Aug 2023 15:39:45 +0800
Subject: [PATCH] backport to tbb 2020

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 src/libslic3r/GCode.cpp | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/src/libslic3r/GCode.cpp b/src/libslic3r/GCode.cpp
index fe1b5972f..ffb416b1c 100644
--- a/src/libslic3r/GCode.cpp
+++ b/src/libslic3r/GCode.cpp
@@ -55,9 +55,11 @@
 #if TBB_VERSION_MAJOR >= 2021
     #include <tbb/parallel_pipeline.h>
     using slic3r_tbb_filtermode = tbb::filter_mode;
+    #define slic3r_tbb_filter tbb::filter
 #else
     #include <tbb/pipeline.h>
     using slic3r_tbb_filtermode = tbb::filter;
+    #define slic3r_tbb_filter tbb::interface6::filter_t
 #endif
 
 #include <Shiny/Shiny.h>
@@ -2198,15 +2200,15 @@ void GCode::process_layers(
 
     // The pipeline elements are joined using const references, thus no copying is performed.
     output_stream.find_replace_supress();
-    tbb::filter<void, LayerResult> pipeline_to_layerresult = generator;
+    slic3r_tbb_filter<void, LayerResult> pipeline_to_layerresult = generator;
     if (m_spiral_vase)
         pipeline_to_layerresult = pipeline_to_layerresult & spiral_vase;
     if (m_pressure_equalizer)
         pipeline_to_layerresult = pipeline_to_layerresult & pressure_equalizer;
-    tbb::filter<void, std::string> pipeline_to_string = pipeline_to_layerresult & cooling & fan_mover;
+    slic3r_tbb_filter<void, std::string> pipeline_to_string = pipeline_to_layerresult & cooling & fan_mover;
     if (m_find_replace)
         pipeline_to_string = pipeline_to_string & find_replace;
-    tbb::filter<void, void> full_pipeline = pipeline_to_string & output;
+    slic3r_tbb_filter<void, void> full_pipeline = pipeline_to_string & output;
     tbb::parallel_pipeline(12, full_pipeline);
     output_stream.find_replace_enable();
 }
@@ -2293,15 +2295,15 @@ void GCode::process_layers(
 
     // The pipeline elements are joined using const references, thus no copying is performed.
     output_stream.find_replace_supress();
-    tbb::filter<void, LayerResult> pipeline_to_layerresult = generator;
+    slic3r_tbb_filter<void, LayerResult> pipeline_to_layerresult = generator;
     if (m_spiral_vase)
         pipeline_to_layerresult = pipeline_to_layerresult & spiral_vase;
     if (m_pressure_equalizer)
         pipeline_to_layerresult = pipeline_to_layerresult & pressure_equalizer;
-    tbb::filter<void, std::string> pipeline_to_string = pipeline_to_layerresult & cooling & fan_mover;
+    slic3r_tbb_filter<void, std::string> pipeline_to_string = pipeline_to_layerresult & cooling & fan_mover;
     if (m_find_replace)
         pipeline_to_string = pipeline_to_string & find_replace;
-    tbb::filter<void, void> full_pipeline = pipeline_to_string & output;
+    slic3r_tbb_filter<void, void> full_pipeline = pipeline_to_string & output;
     tbb::parallel_pipeline(12, full_pipeline);
     output_stream.find_replace_enable();
 }
-- 
2.39.1

