--- a/src/hints/HintsToPot.cpp	2023-10-14 00:25:54.878963595 -0700
+++ b/src/hints/HintsToPot.cpp	2023-10-14 00:25:59.514112596 -0700
@@ -2,6 +2,7 @@
 #include <vector>
 #include <string>
 #include <boost/filesystem.hpp>
+#include <boost/filesystem/fstream.hpp>
 #include <boost/dll.hpp>
 #include <boost/property_tree/ini_parser.hpp>
 #include <boost/nowide/fstream.hpp>
--- a/src/slic3r/GUI/ScriptExecutor.cpp	2023-10-14 00:41:49.017788985 -0700
+++ b/src/slic3r/GUI/ScriptExecutor.cpp	2023-10-14 00:42:02.011231082 -0700
@@ -13,7 +13,7 @@
 #include <angelscript/add_on/scriptbuilder/scriptbuilder.h>
 #include <angelscript/add_on/scriptstdstring/scriptstdstring.h>
 #include <angelscript/add_on/scriptmath/scriptmath.h>
-
+#include <boost/filesystem/string_file.hpp>
 using namespace gw;
 
 namespace Slic3r {  namespace GUI {
--- a/src/occt_wrapper/CMakeLists.txt	2023-10-14 00:45:14.781801593 -0700
+++ b/src/occt_wrapper/CMakeLists.txt	2023-10-14 00:45:25.059152430 -0700
@@ -19,7 +19,7 @@
 
 generate_export_header(OCCTWrapper)
 
-find_package(OpenCASCADE 7.6.2 REQUIRED)
+find_package(OpenCASCADE 7.6.3 REQUIRED)
 
 set(OCCT_LIBS
     TKXDESTEP
