From 9d2ea3620e7d14e60fbd4607e7b2c7402569c706 Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Mon, 8 Jan 2024 04:20:47 -0800
Subject: [PATCH 2/3] fix compilation with wxgtk 3.2.4

---
 src/slic3r/GUI/PhysicalPrinterDialog.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/slic3r/GUI/PhysicalPrinterDialog.cpp b/src/slic3r/GUI/PhysicalPrinterDialog.cpp
index 7ca307ff9..299b3d418 100644
--- a/src/slic3r/GUI/PhysicalPrinterDialog.cpp
+++ b/src/slic3r/GUI/PhysicalPrinterDialog.cpp
@@ -564,7 +564,8 @@ void PhysicalPrinterDialog::build_printhost_settings(ConfigOptionsGroup* m_optgr
     // Always fill in the "printhost_port" combo box from the config and select it.
     {
         Choice* choice = dynamic_cast<Choice*>(m_optgroup->get_field("printhost_port"));
-        choice->set_values({ m_config->opt_string("printhost_port") });
+        const wxArrayString &val = { m_config->opt_string("printhost_port") };
+        choice->set_values(val);
         choice->set_selection();
     }
 
-- 
2.44.0

