From fe900cfc5ca66692a9d493a922b5407cfbf04920 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Sat, 9 Mar 2024 15:16:29 +0800
Subject: [PATCH] fix build with cereal 1.3.0

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 src/libslic3r/ObjectID.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libslic3r/ObjectID.hpp b/src/libslic3r/ObjectID.hpp
index 782c58741..eb0fe5182 100644
--- a/src/libslic3r/ObjectID.hpp
+++ b/src/libslic3r/ObjectID.hpp
@@ -1,7 +1,7 @@
 #ifndef slic3r_ObjectID_hpp_
 #define slic3r_ObjectID_hpp_
 
-#include <cereal/access.hpp>
+#include <cereal/cereal.hpp>
 
 namespace Slic3r {
 
-- 
2.43.0

