From 87584ebc4c6c0db63159d772eec7c9389de5131b Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Sat, 23 Aug 2025 23:01:43 -0400
Subject: [PATCH 4/6] AOSCOS: Disable version_check

---
 src/libslic3r/AppConfig.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libslic3r/AppConfig.cpp b/src/libslic3r/AppConfig.cpp
index 1c18ebb85f..7a6ec1df28 100644
--- a/src/libslic3r/AppConfig.cpp
+++ b/src/libslic3r/AppConfig.cpp
@@ -221,7 +221,7 @@ void AppConfig::set_defaults()
             set("color_light", key2color["Gui_color_light"]);
 
         if (get("version_check").empty())
-            set("version_check", "1");
+            set("version_check", "0");
 
         if (get("preset_update").empty())
             set("preset_update", "0");
-- 
2.50.1

