From 3ae30536ff1116f999a7dbb326e2418e3cd557ae Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Sat, 9 Mar 2024 15:25:06 +0800
Subject: [PATCH] fix test-utils build on Linux

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 src/test-utils/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/test-utils/CMakeLists.txt b/src/test-utils/CMakeLists.txt
index 713593a9a..c2cf0cf99 100644
--- a/src/test-utils/CMakeLists.txt
+++ b/src/test-utils/CMakeLists.txt
@@ -5,6 +5,10 @@ add_executable(stl_to_cpp stl_to_cpp.cpp)
 
 target_link_libraries(stl_to_cpp libslic3r)
 
+if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    target_link_libraries(stl_to_cpp -lxcb)
+endif()
+
 if (WIN32)
     prusaslicer_copy_dlls(stl_to_cpp)
 endif()
-- 
2.43.0

