From 043c92ce6a9216d47e19028bf8260fb2f329337e Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Sun, 24 Aug 2025 00:19:32 -0400
Subject: [PATCH 6/6] AOSCOS: update version number

---
 version.inc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/version.inc b/version.inc
index d127e2d899..95386c39d1 100644
--- a/version.inc
+++ b/version.inc
@@ -10,9 +10,9 @@ set(SLIC3R_APP_CMD "superslicer")
 # id for windows registry
 set(SLIC3R_APP_PROG_ID "Slic3r.1")
 # versions
-set(SLIC3R_VERSION "2.7-beta+UNKNOWN")
+set(SLIC3R_VERSION "2.7.61.8+AOSC")
 set(SLIC3R_VERSION_FULL "2.7.61.8")
-set(SLIC3R_BUILD_ID "${SLIC3R_APP_KEY}_${SLIC3R_VERSION_FULL}+UNKNOWN")
+set(SLIC3R_BUILD_ID "${SLIC3R_APP_KEY}_${SLIC3R_VERSION_FULL}+AOSC")
 set(SLIC3R_RC_VERSION "2,7,61,8")
 set(SLIC3R_RC_VERSION_DOTS "${SLIC3R_VERSION_FULL}")
 
@@ -28,4 +28,4 @@ set(SLIC3R_BASED_ON " based on PrusaSlicer")
 set(SLIC3R_GITHUB "supermerill/SuperSlicer")
 # download url
 set(SLIC3R_DOWNLOAD "https://github.com/supermerill/SuperSlicer/releases")
-set(SLIC3R_DOC_URL "https://manual.slic3r.org/")
\ No newline at end of file
+set(SLIC3R_DOC_URL "https://manual.slic3r.org/")
-- 
2.50.1

