diff --git a/src/libslic3r/CMakeLists.txt b/src/libslic3r/CMakeLists.txt
index 84be9c12e..b1eca701a 100644
--- a/src/libslic3r/CMakeLists.txt
+++ b/src/libslic3r/CMakeLists.txt
@@ -439,7 +439,7 @@ target_link_libraries(libslic3r
     qoi
     )
 
-if (APPLE)
+if (TRUE)
     # TODO: we need to fix notarization with the separate shared library
     target_link_libraries(libslic3r OCCTWrapper)
 endif ()
diff --git a/src/libslic3r/Format/STEP.cpp b/src/libslic3r/Format/STEP.cpp
index 5165bb701..7587335ab 100644
--- a/src/libslic3r/Format/STEP.cpp
+++ b/src/libslic3r/Format/STEP.cpp
@@ -22,7 +22,7 @@
 
 namespace Slic3r {
 
-#if __APPLE__
+#if 1
 extern "C" bool load_step_internal(const char *path, OCCTResult* res);
 #endif
 
@@ -30,7 +30,7 @@ LoadStepFn get_load_step_fn()
 {
     static LoadStepFn load_step_fn = nullptr;
 
-#ifndef __APPLE__
+#if 0
     constexpr const char* fn_name = "load_step_internal";
 #endif
 
@@ -54,7 +54,7 @@ LoadStepFn get_load_step_fn()
             FreeLibrary(module);
             throw;
         }
-#elif __APPLE__
+#elif 1
         load_step_fn = &load_step_internal;
 #else
         libpath /= "OCCTWrapper.so";
diff --git a/src/occt_wrapper/CMakeLists.txt b/src/occt_wrapper/CMakeLists.txt
index ed75531a9..d2c92e1f6 100644
--- a/src/occt_wrapper/CMakeLists.txt
+++ b/src/occt_wrapper/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.13)
 project(OCCTWrapper)
 
-if (APPLE)
+if (TRUE)
     # TODO: we need to fix notarization with the separate shared library
     add_library(OCCTWrapper STATIC OCCTWrapper.cpp)
 else ()
