diff --git a/src/libslic3r/CMakeLists.txt b/src/libslic3r/CMakeLists.txt
index 570a36f637..622edbd1d6 100644
--- a/src/libslic3r/CMakeLists.txt
+++ b/src/libslic3r/CMakeLists.txt
@@ -619,7 +619,7 @@ target_link_libraries(libslic3r
     LibBGCode::bgcode_convert
     )
 
-if (APPLE)
+if (TRUE)
     # TODO: we need to fix notarization with the separate shared library
     target_link_libraries(libslic3r OCCTWrapper)
 endif ()
diff --git a/src/libslic3r/Format/STEP.cpp b/src/libslic3r/Format/STEP.cpp
index 2dbb5ee41a..d2f5229e63 100644
--- a/src/libslic3r/Format/STEP.cpp
+++ b/src/libslic3r/Format/STEP.cpp
@@ -26,7 +26,7 @@
 
 namespace Slic3r {
 
-#if __APPLE__
+#if 1
 extern "C" bool load_step_internal(const char *path, OCCTResult* res);
 #endif
 
@@ -34,7 +34,7 @@ LoadStepFn get_load_step_fn()
 {
     static LoadStepFn load_step_fn = nullptr;
 
-#ifndef __APPLE__
+#if 0
     constexpr const char* fn_name = "load_step_internal";
 #endif
 
@@ -58,7 +58,7 @@ LoadStepFn get_load_step_fn()
             FreeLibrary(module);
             throw;
         }
-#elif __APPLE__
+#elif 1
         load_step_fn = &load_step_internal;
 #else
         libpath /= "OCCTWrapper.so";
diff --git a/src/occt_wrapper/CMakeLists.txt b/src/occt_wrapper/CMakeLists.txt
index f3e3bc7768..5247e74379 100644
--- a/src/occt_wrapper/CMakeLists.txt
+++ b/src/occt_wrapper/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.13)
 project(OCCTWrapper)
 
-if (APPLE)
+if (TRUE)
     # TODO: we need to fix notarization with the separate shared library
     add_library(OCCTWrapper STATIC OCCTWrapper.cpp)
 else ()
