diff --git a/src/libslic3r/AppConfig.cpp b/src/libslic3r/AppConfig.cpp
index 4850b9a91..eadf41b0e 100644
--- a/src/libslic3r/AppConfig.cpp
+++ b/src/libslic3r/AppConfig.cpp
@@ -340,7 +340,7 @@ void AppConfig::set_defaults()
             set("color_light", key2color["Gui_color_light"]);
 
         if (get("version_check").empty())
-            set("version_check", "1");
+            set("version_check", "0");
 
         if (get("preset_update").empty())
             set("preset_update", "0");
