Index: subtitleripper-0.3.4/vobsub.c
===================================================================
--- subtitleripper-0.3.4.orig/vobsub.c	2010-02-11 12:10:06.000000000 +0100
+++ subtitleripper-0.3.4/vobsub.c	2010-02-11 12:10:34.000000000 +0100
@@ -18,7 +18,7 @@
  * Some code freely inspired from VobSub <URL:http://vobsub.edensrising.com>,
  * with kind permission from Gabest <gabest@freemail.hu>
  */
-/* #define HAVE_GETLINE */
+#define HAVE_GETLINE
 #include <ctype.h>
 #include <errno.h>
 #include <stdio.h>
