diff -Nur midori-0.5.12.orig/midori/main.c midori-0.5.12/midori/main.c
--- midori-0.5.12.orig/midori/main.c	2016-09-05 14:45:28.000000000 -0600
+++ midori-0.5.12/midori/main.c	2016-12-27 11:25:00.527569564 -0700
@@ -58,6 +58,9 @@
     gchar** uris;
     gchar* block_uris;
     gint inactivity_reset;
+
+g_setenv("G_TLS_GNUTLS_PRIORITY", "@SYSTEM", FALSE);
+
     GOptionEntry entries[] =
     {
        { "app", 'a', 0, G_OPTION_ARG_STRING, &webapp,
