diff -up SDL2-2.0.4/src/video/wayland/SDL_waylanddyn.h.fix SDL2-2.0.4/src/video/wayland/SDL_waylanddyn.h
--- SDL2-2.0.4/src/video/wayland/SDL_waylanddyn.h.fix	2016-02-05 12:18:29.514348406 -0500
+++ SDL2-2.0.4/src/video/wayland/SDL_waylanddyn.h	2016-02-05 12:18:45.898990489 -0500
@@ -79,6 +79,7 @@ void SDL_WAYLAND_UnloadSymbols(void);
 #define wl_proxy_get_user_data (*WAYLAND_wl_proxy_get_user_data)
 #define wl_proxy_add_listener (*WAYLAND_wl_proxy_add_listener)
 #define wl_proxy_marshal_constructor (*WAYLAND_wl_proxy_marshal_constructor)
+#define wl_proxy_marshal_constructor_versioned (*WAYLAND_wl_proxy_marshal_constructor_versioned)
 
 #define wl_seat_interface (*WAYLAND_wl_seat_interface)
 #define wl_surface_interface (*WAYLAND_wl_surface_interface)
