From 1b10d4c888d406c6e4b56b487cb30aaffde409f7 Mon Sep 17 00:00:00 2001
From: falkTX <falktx@falktx.com>
Date: Tue, 3 Dec 2019 12:22:02 +0000
Subject: [PATCH 2/9] Always use -Wl,--no-undefined (unless macOS)

Signed-off-by: falkTX <falktx@falktx.com>
---
 common/wscript |  4 ++--
 wscript        | 18 ++++++++----------
 2 files changed, 10 insertions(+), 12 deletions(-)

diff --git a/common/wscript b/common/wscript
index 70503788..f81a285c 100644
--- a/common/wscript
+++ b/common/wscript
@@ -388,12 +388,12 @@ def build(bld):
     if bld.env['BUILD_ADAPTER'] and bld.env['IS_LINUX'] and bld.env['BUILD_DRIVER_ALSA']:
          audio_adapter_sources += ['../linux/alsa/JackAlsaAdapter.cpp']
          process = create_jack_process_obj(bld, 'audioadapter', audio_adapter_sources, serverlib)
-         process.use = ['ALSA', 'SAMPLERATE']
+         process.use += ['ALSA', 'SAMPLERATE']
 
     if bld.env['BUILD_ADAPTER'] and bld.env['IS_SUN']:
          audio_adapter_sources += ['../solaris/oss/JackOSSAdapter.cpp', 'memops.c']
          process = create_jack_process_obj(bld, 'audioadapter', audio_adapter_sources, serverlib)
-         process.use = 'SAMPLERATE'
+         process.use += 'SAMPLERATE'
 
     if bld.env['BUILD_ADAPTER'] and bld.env['IS_WINDOWS']:
          audio_adapter_sources += ['../windows/portaudio/JackPortAudioAdapter.cpp', '../windows/portaudio/JackPortAudioDevices.cpp']
diff --git a/wscript b/wscript
index 0621ae71..dbae6cac 100644
--- a/wscript
+++ b/wscript
@@ -212,11 +212,13 @@ def configure(conf):
         conf.env.append_unique('CCDEFINES', '_POSIX')
         conf.env.append_unique('CXXDEFINES', '_POSIX')
 
+    conf.env.append_unique('CFLAGS', '-Wall')
     conf.env.append_unique('CXXFLAGS', '-Wall')
     conf.env.append_unique('CXXFLAGS', '-std=gnu++11')
-    conf.env.append_unique('CFLAGS', '-Wall')
 
-    if conf.env['IS_MACOSX']:
+    if not conf.env['IS_MACOSX']:
+        conf.env.append_unique('LDFLAGS', '-Wl,--no-undefined')
+    else:
         conf.check(lib='aften', uselib='AFTEN', define_name='AFTEN')
         conf.check_cxx(
             fragment=''
@@ -524,14 +526,10 @@ def build_jackd(bld):
 
 # FIXME: Is SERVER_SIDE needed?
 def create_driver_obj(bld, **kw):
-    if bld.env['IS_MACOSX'] or bld.env['IS_WINDOWS']:
-        # On MacOSX this is necessary.
-        # I do not know if this is necessary on Windows.
-        # Note added on 2015-12-13 by karllinden.
-        if 'use' in kw:
-            kw['use'] += ['serverlib']
-        else:
-            kw['use'] = ['serverlib']
+    if 'use' in kw:
+        kw['use'] += ['serverlib']
+    else:
+        kw['use'] = ['serverlib']
 
     driver = bld(
         features = ['c', 'cxx', 'cshlib', 'cxxshlib'],
-- 
2.24.1

