From a39d2df15e13bc406f780ae6e91c7c1a61b68f76 Mon Sep 17 00:00:00 2001
From: falkTX <falktx@falktx.com>
Date: Tue, 3 Dec 2019 12:31:06 +0000
Subject: [PATCH 3/9] Force fix for libdb mixed mode build

Signed-off-by: falkTX <falktx@falktx.com>
---
 common/JackMetadata.h | 6 ++++++
 wscript               | 3 ++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/common/JackMetadata.h b/common/JackMetadata.h
index 7469737a..a6cf18f9 100644
--- a/common/JackMetadata.h
+++ b/common/JackMetadata.h
@@ -24,6 +24,12 @@
 #include "config.h"
 #endif
 
+// libdb does not work in 32bit mixed mode
+#ifdef BUILD_WITH_32_64
+#undef HAVE_DB
+#define HAVE_DB 0
+#endif
+
 #include <stdint.h>
 
 #if HAVE_DB
diff --git a/wscript b/wscript
index dbae6cac..3a187d8f 100644
--- a/wscript
+++ b/wscript
@@ -392,8 +392,9 @@ def configure(conf):
 
     if Options.options.mixed:
         conf.setenv(lib32, env=conf.env.derive())
-        conf.env.append_unique('CXXFLAGS', '-m32')
         conf.env.append_unique('CFLAGS', '-m32')
+        conf.env.append_unique('CXXFLAGS', '-m32')
+        conf.env.append_unique('CXXFLAGS', '-DBUILD_WITH_32_64')
         conf.env.append_unique('LINKFLAGS', '-m32')
         if Options.options.libdir32:
             conf.env['LIBDIR'] = Options.options.libdir32
-- 
2.24.1

