From 608cfbe74cceaedea58c746f19117e568c5f8728 Mon Sep 17 00:00:00 2001
From: falkTX <falktx@falktx.com>
Date: Tue, 10 Dec 2019 18:44:06 +0000
Subject: [PATCH 5/9] Fix freedesktop specification link

Signed-off-by: falkTX <falktx@falktx.com>
---
 common/jack/metadata.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/jack/metadata.h b/common/jack/metadata.h
index 9e026161..891dbd62 100644
--- a/common/jack/metadata.h
+++ b/common/jack/metadata.h
@@ -265,7 +265,7 @@ extern const char* JACK_METADATA_ICON_LARGE;
  * themes.  Icons should be searched for according to the freedesktop Icon
  *
  * Theme Specification:
- * http://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html
+ * https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html
  */
 extern const char* JACK_METADATA_ICON_NAME;
 
-- 
2.24.1

