--- a/camlibs/ax203/jpeg_memsrcdest.h	2012-12-13 08:49:31.000000000 +0000
+++ b/camlibs/ax203/jpeg_memsrcdest.h	2016-01-23 03:21:16.000000000 +0000
@@ -1,7 +1,7 @@
 #include <jpeglib.h>
 
 void
-jpeg_mem_src (j_decompress_ptr cinfo, unsigned char * buffer,
+jpeg_mem_src (j_decompress_ptr cinfo, const unsigned char * buffer,
 	unsigned long bufsize);
 
 void
--- a/camlibs/jl2005c/jpeg_memsrcdest.h	2012-12-13 08:49:29.000000000 +0000
+++ b/camlibs/jl2005c/jpeg_memsrcdest.h	2016-01-23 03:22:40.000000000 +0000
@@ -1,7 +1,7 @@
 #include <jpeglib.h>
 
 void
-jpeg_mem_src (j_decompress_ptr cinfo, unsigned char * buffer,
+jpeg_mem_src (j_decompress_ptr cinfo, const unsigned char * buffer,
 	unsigned long bufsize);
 
 void
