diff --git a/dnscrypt-proxy/example-dnscrypt-proxy.toml b/dnscrypt-proxy/example-dnscrypt-proxy.toml
index 0f973adb..a3222bf0 100644
--- a/dnscrypt-proxy/example-dnscrypt-proxy.toml
+++ b/dnscrypt-proxy/example-dnscrypt-proxy.toml
@@ -170,7 +170,7 @@ keepalive = 30
 ## This file is different from other log files, and will not be
 ## automatically rotated by the application.
 
-# log_file = 'dnscrypt-proxy.log'
+log_file = '/var/log/dnscrypt-proxy/dnscrypt-proxy.log'
 
 
 ## When using a log file, only keep logs from the most recent launch.
@@ -180,7 +180,7 @@ keepalive = 30
 
 ## Use the system logger (syslog on Unix, Event Log on Windows)
 
-# use_syslog = true
+use_syslog = true
 
 
 ## Delay, in minutes, after which certificates are reloaded
@@ -372,7 +372,7 @@ reject_ttl = 10
 
 ## See the `example-forwarding-rules.txt` file for an example
 
-# forwarding_rules = 'forwarding-rules.txt'
+forwarding_rules = '/etc/dnscrypt-proxy/forwarding-rules.txt'
 
 
 
@@ -388,7 +388,7 @@ reject_ttl = 10
 ##
 ## See the `example-cloaking-rules.txt` file for an example
 
-# cloaking_rules = 'cloaking-rules.txt'
+# cloaking_rules = '/etc/dnscrypt-proxy/cloaking-rules.txt'
 
 ## TTL used when serving entries in cloaking-rules.txt
 
@@ -442,7 +442,7 @@ cache_neg_max_ttl = 600
 ## check for connectivity and captive portals, along with hard-coded
 ## IP addresses to return.
 
-# map_file = 'example-captive-portals.txt'
+# map_file = '/etc/dnscrypt-proxy/example-captive-portals.txt'
 
 
 
@@ -474,8 +474,8 @@ cache_neg_max_ttl = 600
 ## openssl req -x509 -nodes -newkey rsa:2048 -days 5000 -sha256 -keyout localhost.pem -out localhost.pem
 ## See the documentation (wiki) for more information.
 
-# cert_file = 'localhost.pem'
-# cert_key_file = 'localhost.pem'
+# cert_file = "/var/lib/dnscrypt-proxy/localhost.pem"
+# cert_key_file = "/var/lib/dnscrypt-proxy/localhost.pem"
 
 
 
@@ -546,12 +546,12 @@ format = 'tsv'
 
 ## Path to the file of blocking rules (absolute, or relative to the same directory as the config file)
 
-# blocked_names_file = 'blocked-names.txt'
+# blocked_names_file = '/etc/dnscrypt-proxy/blocked-names.txt'
 
 
 ## Optional path to a file logging blocked queries
 
-# log_file = 'blocked-names.log'
+# log_file = '/var/log/dnscrypt-proxy/blocked-names.log'
 
 
 ## Optional log format: tsv or ltsv (default: tsv)
@@ -574,12 +574,12 @@ format = 'tsv'
 
 ## Path to the file of blocking rules (absolute, or relative to the same directory as the config file)
 
-# blocked_ips_file = 'blocked-ips.txt'
+# blocked_ips_file = '/etc/dnscrypt-proxy/blocked-ips.txt'
 
 
 ## Optional path to a file logging blocked queries
 
-# log_file = 'blocked-ips.log'
+# log_file = '/var/log/dnscrypt-proxy/blocked-ips.log'
 
 
 ## Optional log format: tsv or ltsv (default: tsv)
@@ -602,12 +602,12 @@ format = 'tsv'
 
 ## Path to the file of allow list rules (absolute, or relative to the same directory as the config file)
 
-# allowed_names_file = 'allowed-names.txt'
+# allowed_names_file = '/etc/dnscrypt-proxy/allowed-names.txt'
 
 
 ## Optional path to a file logging allowed queries
 
-# log_file = 'allowed-names.log'
+# log_file = '/var/log/dnscrypt-proxy/allowed-names.log'
 
 
 ## Optional log format: tsv or ltsv (default: tsv)
@@ -630,12 +630,12 @@ format = 'tsv'
 
 ## Path to the file of allowed ip rules (absolute, or relative to the same directory as the config file)
 
-# allowed_ips_file = 'allowed-ips.txt'
+# allowed_ips_file = '/etc/dnscrypt-proxy/allowed-ips.txt'
 
 
 ## Optional path to a file logging allowed queries
 
-# log_file = 'allowed-ips.log'
+# log_file = '/var/log/dnscrypt-proxy/allowed-ips.log'
 
 ## Optional log format: tsv or ltsv (default: tsv)
 
@@ -740,7 +740,7 @@ format = 'tsv'
   # [sources.quad9-resolvers]
   #   urls = ['https://www.quad9.net/quad9-resolvers.md']
   #   minisign_key = 'RWQBphd2+f6eiAqBsvDZEBXBGHQBJfeG6G+wJPPKxCZMoEQYpmoysKUN'
-  #   cache_file = 'quad9-resolvers.md'
+  #   cache_file = '/var/cache/dnscrypt-proxy/quad9-resolvers.md'
   #   prefix = 'quad9-'
 
   ### Another example source, with resolvers censoring some websites not appropriate for children
