diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0e853fee..32a40d8e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,7 @@ find_package(folly CONFIG REQUIRED)
 find_package(fmt CONFIG REQUIRED)
 
 find_package(OpenSSL REQUIRED)
-find_package(Glog REQUIRED)
+find_package(glog CONFIG QUIET)
 find_package(Threads REQUIRED)
 find_package(Zstd REQUIRED)
 if (UNIX AND NOT APPLE)
@@ -204,7 +204,6 @@ target_include_directories(
     ${sodium_INCLUDE_DIR}
     ${ZSTD_INCLUDE_DIR}
   PRIVATE
-    ${GLOG_INCLUDE_DIRS}
     ${FIZZ_INCLUDE_DIRECTORIES}
 )
 
@@ -218,7 +217,7 @@ target_link_libraries(fizz
     ZLIB::ZLIB
     ${ZSTD_LIBRARY}
   PRIVATE
-    ${GLOG_LIBRARIES}
+    glog::glog
     ${GFLAGS_LIBRARIES}
     ${FIZZ_LINK_LIBRARIES}
     ${CMAKE_DL_LIBS}
