From 15e93fe87097ef7a0c70412cbd489a5cd9f8f748 Mon Sep 17 00:00:00 2001
From: stydxm <stydxm@outlook.com>
Date: Tue, 2 Sep 2025 12:11:05 +0800
Subject: [PATCH 1/2] Update glog import logic for glog 0.7.1

---
 fizz/CMakeLists.txt | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/fizz/CMakeLists.txt b/fizz/CMakeLists.txt
index 995d2076..ad753ec7 100644
--- a/fizz/CMakeLists.txt
+++ b/fizz/CMakeLists.txt
@@ -60,7 +60,7 @@ endif()
 find_package(fmt CONFIG REQUIRED)
 
 find_package(OpenSSL REQUIRED)
-find_package(Glog REQUIRED)
+find_package(glog CONFIG QUIET)
 add_compile_definitions(GLOG_USE_GLOG_EXPORT)
 find_package(Threads REQUIRED)
 find_package(Zstd REQUIRED)
@@ -214,7 +214,6 @@ target_include_directories(
     ${sodium_INCLUDE_DIR}
     ${ZSTD_INCLUDE_DIR}
   PRIVATE
-    ${GLOG_INCLUDE_DIRS}
     ${FIZZ_INCLUDE_DIRECTORIES}
 )
 
@@ -266,7 +265,7 @@ target_link_libraries(fizz
     ZLIB::ZLIB
     ${ZSTD_LIBRARY}
   PRIVATE
-    ${GLOG_LIBRARIES}
+    glog::glog
     ${GFLAGS_LIBRARIES}
     ${FIZZ_LINK_LIBRARIES}
     ${CMAKE_DL_LIBS}
-- 
2.52.0

