From 137426a26f46cd6b45e2499f5e6eb7c55211d7ef Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 16 Mar 2026 13:42:29 +0800
Subject: [PATCH 2/2] Fix links to follybenchmark

---
 fizz/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fizz/CMakeLists.txt b/fizz/CMakeLists.txt
index ad753ec7..2e5554bf 100644
--- a/fizz/CMakeLists.txt
+++ b/fizz/CMakeLists.txt
@@ -494,7 +494,7 @@ if(BUILD_EXAMPLES)
     DESTINATION ${BIN_INSTALL_DIR}
   )
   add_executable(EncryptedRecordBench record/test/EncryptedRecordBench.cpp)
-  target_link_libraries(EncryptedRecordBench fizz Folly::folly_benchmark)
+  target_link_libraries(EncryptedRecordBench fizz Folly::follybenchmark)
   set_target_properties(
     EncryptedRecordBench PROPERTIES OUTPUT_NAME fizz-encryptedrecordbench
   )
-- 
2.52.0

