From 08876a8e1f7281b70762b171bb5382c714907117 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Thu, 4 Jul 2024 11:32:43 +0800
Subject: [PATCH] Set binary name as mihomo

---
 src/service/web.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/service/web.rs b/src/service/web.rs
index b461ee2..95e6add 100644
--- a/src/service/web.rs
+++ b/src/service/web.rs
@@ -79,7 +79,7 @@ pub fn stop_clash() -> Result<()> {
 
     let mut system = System::new();
     system.refresh_all();
-    let procs = system.processes_by_name("verge-mihomo");
+    let procs = system.processes_by_name("mihomo");
     for proc in procs {
         proc.kill();
     }
-- 
2.45.2

