From 83f01520f21f2e6bfecaf0cdb7d6f16f0deb28c3 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sun, 27 Apr 2025 00:29:37 +0800
Subject: [PATCH 2/4] fix: fix get root access security issue

---
 src/service/core.rs | 20 +++++++++++---------
 1 file changed, 11 insertions(+), 9 deletions(-)

diff --git a/src/service/core.rs b/src/service/core.rs
index 4817674..8f28cbd 100644
--- a/src/service/core.rs
+++ b/src/service/core.rs
@@ -137,15 +137,14 @@ impl CoreManager {
                 .clone();
             let config = config.ok_or(anyhow!("Runtime config is not set"))?;
 
-            let bin_path = config.bin_path.as_str();
             let config_dir = config.config_dir.as_str();
             let config_file = config.config_file.as_str();
             let log_file = config.log_file.as_str();
             let args = vec!["-d", config_dir, "-f", config_file];
 
             println!(
-                "Starting mihomo with bin_path: {}, config_dir: {}, config_file: {}, log_file: {}",
-                bin_path, config_dir, config_file, log_file
+                "Starting mihomo with config_dir: {}, config_file: {}, log_file: {}",
+                config_dir, config_file, log_file
             );
 
             // Open log file
@@ -153,7 +152,7 @@ impl CoreManager {
                 .with_context(|| format!("Failed to open log file: {}", log_file))?;
 
             // Spawn process
-            let pid = process::spawn_process(bin_path, &args, log)?;
+            let pid = process::spawn_process("/usr/bin/mihomo", &args, log)?;
             println!("Mihomo started with PID: {}", pid);
 
             // Update mihomo status
@@ -227,12 +226,15 @@ impl CoreManager {
             .unwrap()
             .running_pid
             .load(Ordering::Relaxed) as u32;
-        
+
         match process::find_processes("mihomo") {
             Ok(pids) => {
                 // 直接在迭代过程中过滤和终止
-                let kill_count = pids.into_iter()
-                    .filter(|&pid| pid != current_pid && (tracked_mihomo_pid <= 0 || pid != tracked_mihomo_pid))
+                let kill_count = pids
+                    .into_iter()
+                    .filter(|&pid| {
+                        pid != current_pid && (tracked_mihomo_pid <= 0 || pid != tracked_mihomo_pid)
+                    })
                     .map(|pid| {
                         println!("Found other mihomo process with PID: {}, stopping it", pid);
                         match process::kill_process(pid) {
@@ -248,14 +250,14 @@ impl CoreManager {
                     })
                     .filter(|&success| success)
                     .count();
-                    
+
                 println!("Successfully stopped {} mihomo processes", kill_count);
             }
             Err(e) => {
                 eprintln!("Error finding mihomo processes: {}", e);
             }
         }
-        
+
         Ok(())
     }
 
-- 
2.49.0

