diff --git a/aiter/jit/utils/cpp_extension.py b/aiter/jit/utils/cpp_extension.py
index 444a5669f..02986a201 100644
--- a/aiter/jit/utils/cpp_extension.py
+++ b/aiter/jit/utils/cpp_extension.py
@@ -17,6 +17,7 @@ import subprocess
 import sys
 import sysconfig
 import warnings
+import platform
 from typing import Dict, List, Optional, Tuple, Union

 import setuptools
@@ -34,7 +35,9 @@ EXEC_EXT = ""
 CLIB_PREFIX = "lib"
 CLIB_EXT = ".so"
 SHARED_FLAG = "-shared"
-
+CMODEL_LARGE = "large"
+if platform.machine() == 'loongarch64':
+    CMODEL_LARGE = 'extreme'
 SUBPROCESS_DECODE_ARGS = ()
 MINIMUM_GCC_VERSION = (5, 0, 0)
 MINIMUM_MSVC_VERSION = (19, 0, 24215)
@@ -303,7 +306,7 @@ COMMON_HIPCC_FLAGS = [
     "-DCUDA_HAS_FP16=1",
     "-D__HIP_NO_HALF_OPERATORS__=1",
     "-D__HIP_NO_HALF_CONVERSIONS__=1",
-    "-mcmodel=large",
+    "-mcmodel=" + CMODEL_LARGE,
     "-fno-unique-section-names",
     "-ffunction-sections",
     "-fdata-sections",
@@ -1475,7 +1478,7 @@ def verify_ninja_availability():


 def _prepare_ldflags(extra_ldflags, with_cuda, verbose, is_standalone, torch_exclude):
-    extra_ldflags.append("-mcmodel=large")
+    extra_ldflags.append("-mcmodel=" + CMODEL_LARGE)
     extra_ldflags.append("-ffunction-sections")
     extra_ldflags.append("-fdata-sections ")
     extra_ldflags.append("-Wl,--gc-sections")
