--- aMule-2.3.2/configure.in	2016-09-16 07:55:06.000000000 +0000
+++ aMule-2.3.2.nobfd/configure.in	2020-01-28 21:07:19.257744273 +0000
@@ -335,12 +335,6 @@
 AC_LANG_PUSH([C++])
 
 
-# Check for bfd.h (to have proper backtraces without using wxExecute)
-dnl This must be *after* MULE_CHECK_NLS
-MULE_CHECK_BFD
-MULE_CHECK_EXECINFO
-
-
 # aMule plasma widget
 MULE_IF_ENABLED([plasmamule], [PLASMAMULE_CHECKS])
 
