From 12eca130e743932c17343a1fe82f88a31bd1453a Mon Sep 17 00:00:00 2001
From: poplite <poplite.xyz@gmail.com>
Date: Mon, 6 May 2019 21:41:25 +0800
Subject: [PATCH] amulegui: Initialize wx sockets

---
 src/amule-remote-gui.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/amule-remote-gui.cpp b/src/amule-remote-gui.cpp
index 1af561aa8..724d7438f 100644
--- a/src/amule-remote-gui.cpp
+++ b/src/amule-remote-gui.cpp
@@ -27,6 +27,7 @@
 #include <wx/cmdline.h>			// Needed for wxCmdLineParser
 #include <wx/config.h>			// Do_not_auto_remove (win32)
 #include <wx/fileconf.h>		// Needed for wxFileConfig
+#include <wx/socket.h>			// Needed for wxSocketBase
 
 
 #include <common/Format.h>
@@ -138,6 +139,8 @@ int CamuleRemoteGuiApp::OnExit()
 {
 	StopTickTimer();
 
+	wxSocketBase::Shutdown();	// needed because we also called Initialize() manually
+
 	return wxApp::OnExit();
 }
 
@@ -253,6 +256,9 @@ bool CamuleRemoteGuiApp::OnInit()
 		return false;
 	}
 
+	// Initialize wx sockets (needed for http download in background with Asio sockets)
+	wxSocketBase::Initialize();
+
 	// Create the polling timer
 	poll_timer = new wxTimer(this,ID_CORE_TIMER_EVENT);
 	if (!poll_timer) {
