Last-Update: 2013-04-16
Forwarded: no
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=705498
Description: provides MIME ed2k:// protocol handler registration

--- a/amule.desktop
+++ b/amule.desktop
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=aMule
-Exec=amule
+Exec=amule %u
 Icon=amule
 Terminal=false
 Type=Application
@@ -8,3 +8,4 @@ Categories=Network;P2P;
 Comment=A client for the eD2k network
 Comment[fr]=Un client pour le réseau eD2k
 Comment[tr]=eD2k ağı için istemci
+MimeType=x-scheme-handler/ed2k;
