diff -Naur ncbi-vdb-2.9.2-1/libs/krypto/Makefile ncbi-vdb-2.9.2-1.generic64/libs/krypto/Makefile
--- ncbi-vdb-2.9.2-1/libs/krypto/Makefile	2018-08-01 14:42:17.000000000 -0500
+++ ncbi-vdb-2.9.2-1.generic64/libs/krypto/Makefile	2020-05-01 22:08:08.813179001 -0500
@@ -135,8 +135,8 @@
 # though other compilers could also be supported
 ifeq ($(COMP),gcc)
 CC_LISTING = -Wa,-ahlms=$(<D)/$(@F).list
-_CC_AES_NI  = -funsafe-math-optimizations -mmmx -msse -msse2 -msse3 -mssse3 -msse4.1 -maes -Wa,-march=generic64+aes $(CC_LISTING)
-_CC_VECREG  = -funsafe-math-optimizations -mmmx -msse -msse2 -msse3 -mssse3 -msse4.1 -Wa,-march=generic64+sse4 $(CC_LISTING)
+_CC_AES_NI  = -funsafe-math-optimizations -mmmx -msse -msse2 -msse3 -mssse3 -msse4.1 -maes $(CC_LISTING)
+_CC_VECREG  = -funsafe-math-optimizations -mmmx -msse -msse2 -msse3 -mssse3 -msse4.1 $(CC_LISTING)
 _CC_VEC     = $(CC_LISTING)
 else
 _CC_LISTING =
