diff -Naur ncbi-vdb-2.8.0.fix-install/build/Makefile.install ncbi-vdb-2.8.0/build/Makefile.install
--- ncbi-vdb-2.8.0.fix-install/build/Makefile.install	2016-12-08 19:09:59.181830996 -0600
+++ ncbi-vdb-2.8.0/build/Makefile.install	2016-12-08 19:10:25.382039092 -0600
@@ -72,7 +72,6 @@
 ifeq (true, $(LINUX_ROOT))
 	@ # install symlink to includes 
 	@ echo "Updating $(INCLUDE_SYMLINK)"
-	@ rm -f $(INCLUDE_SYMLINK)
 	@ ln -s $(INST_INCDIR)/ncbi-vdb $(INCLUDE_SYMLINK)
 	@ #
 	@ # install configuration file(s)
