diff -Naur ncbi-vdb-2.9.2-1/libs/kproc/linux/sysmgr.c ncbi-vdb-2.9.2-1.generic64/libs/kproc/linux/sysmgr.c
--- ncbi-vdb-2.9.2-1/libs/kproc/linux/sysmgr.c	2018-08-01 14:42:17.000000000 -0500
+++ ncbi-vdb-2.9.2-1.generic64/libs/kproc/linux/sysmgr.c	2020-05-01 22:13:10.199092326 -0500
@@ -33,7 +33,7 @@
 static __thread bool have_tid, on_main_thread;
 
 static
-pid_t gettid ( void )
+pid_t sys_gettid ( void )
 {
     return syscall ( SYS_gettid );
 }
@@ -45,7 +45,7 @@
 {
     if ( ! have_tid )
     {
-        on_main_thread = gettid () == getpid ();
+        on_main_thread = sys_gettid () == getpid ();
         have_tid = true;
     }
     return on_main_thread;
