diff -Naur ncbi-vdb-2.9.2-1/setup/konfigure.perl ncbi-vdb-2.9.2-1.aarch64/setup/konfigure.perl
--- ncbi-vdb-2.9.2-1/setup/konfigure.perl	2018-08-01 14:42:17.000000000 -0500
+++ ncbi-vdb-2.9.2-1.aarch64/setup/konfigure.perl	2020-05-30 12:44:26.688237680 -0500
@@ -217,7 +217,7 @@
 
 print "checking machine architecture... " unless ($AUTORUN);
 println $MARCH unless ($AUTORUN);
-unless ($MARCH =~ /x86_64/i || $MARCH =~ /i?86/i) {
+unless ($MARCH =~ /x86_64/i || $MARCH =~ /i?86/i || $MARCH =~ /aarch64/i) {
     println "configure: error: unsupported architecture '$OSTYPE'";
     exit 1;
 }
@@ -308,6 +308,8 @@
     $BITS = '32_64';
 } elsif ($MARCH =~ /i?86/i) {
     $BITS = 32;
+} elsif ($MARCH =~ /aarch64/i) {
+    $BITS = 64;
 } else {
     die "unrecognized Architecture '$ARCH'";
 }
