--- a/build/env.cmake
+++ b/build/env.cmake
@@ -26,6 +26,8 @@
 # Calculation of the global settings for the CMake build.
 #
 
+include(CheckTypeSize)
+
 # allow implicit source file extensions
 if ( ${CMAKE_VERSION} VERSION_EQUAL "3.20" OR
      ${CMAKE_VERSION} VERSION_GREATER "3.20")
@@ -135,6 +137,9 @@ elseif ("x86_64" STREQUAL ${ARCH} )
     endif()
 endif()
 
+check_type_size("void *" SIZEOF_VOID_P)
+math(EXPR BITS "8 * ${SIZEOF_VOID_P}")
+
 # now any unique combinations of OS and ARCH
 if     ( "mac-x86_84" STREQUAL ${OS}-${ARCH})
 elseif ( "linux-x86_64" STREQUAL ${OS}-${ARCH})
