Author: Andreas Tille <tille@debian.org>
Last-Update: Wed, 31 Aug 2016 07:55:11 +0200
Bug-Debian: https://bugs.debian.org/836128
Description: Add -msse2 to libs/search

--- a/libs/search/Makefile
+++ b/libs/search/Makefile
@@ -39,6 +39,8 @@ OBJFILES = \
 
 include $(TOP)/build/Makefile.env
 
+CFLAGS += -msse2
+
 #-------------------------------------------------------------------------------
 # outer targets
 #
