Author: Andreas Tille <tille@debian.org>
Last-Update: Tue, 12 Jul 2016 16:15:32 +0200
Description: Enable finding ngs-sdk

--- a/setup/konfigure.perl
+++ b/setup/konfigure.perl
@@ -1310,7 +1310,9 @@ sub find_in_dir {
         print "\tlibraries... " unless ($AUTORUN);
         if ($lib) {
             my $builddir = File::Spec->catdir($dir, $OS, $TOOLS, $ARCH, $BUILD);
-            my $libdir  = File::Spec->catdir($builddir, 'lib');
+            my $buildarch = `dpkg-architecture -qDEB_BUILD_MULTIARCH`;
+            chomp $buildarch;
+            my $libdir  = File::Spec->catdir($dir, 'lib', $buildarch);
             my $ilibdir = File::Spec->catdir($builddir, 'ilib');
             my $f = File::Spec->catdir($libdir, $lib);
             print "\n\t\tchecking $f\n\t" if ($OPT{'debug'});
@@ -1391,8 +1393,8 @@ sub find_in_dir {
         }
     }
     if ($jar) {
-        print "\tjar... " unless ($AUTORUN);
-        my $try = "$dir/jar/$jar";
+        print "\tjar... (seeking for $jar in dir $dir)" unless ($AUTORUN);
+        my $try = "$dir/$jar";
         if (-e "$try") {
             println $try unless ($AUTORUN);
             $found_lib = $try;
--- a/setup/package.prl
+++ b/setup/package.prl
@@ -30,8 +30,8 @@ sub REQ { ( { name    => 'ngs-sdk',
               option  => 'with-ngs-java-prefix',
               origin  => 'I',
               type    => 'JO',
-              pkgpath => '/usr/local/ngs/ngs-java',
-              usrpath =>      '$HOME/ngs/ngs-java',
+              pkgpath => '/usr/share/java/jar', # grmpf, why is this ignored after it is changed :-(((
+              usrpath =>      '/usr/share/java/jar', # try again with same value
               bldpath => '$HOME/ncbi-outdir/ngs-java',
               locbldpath=>'$Bin/../../OUTDIR/ngs-java',
               lib     => 'ngs-java.jar',
