Author: Andreas Tille <tille@debian.org>
Last-Update: Wed, 31 Aug 2016 07:55:11 +0200
Bug-Debian: https://bugs.debian.org/836129
Description: Enable build on x32 by removing explicit -m32/-m64

--- a/build/Makefile.cc
+++ b/build/Makefile.cc
@@ -46,12 +46,6 @@ LP = @ $(TOP)/build/ld.sh $(OS) $(ARCH)
 # tool options
 WARN =
 
-ifeq (64,$(BITS))
-	CARCH = -m64
-else
-	CARCH = -m32
-endif
-
 ifeq (prof, $(BUILD))
 	PROF = -xpg
 endif
