diff --git a/pyproject.toml b/pyproject.toml
index a0c7f8e..d1a36b1 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -16,7 +16,7 @@
 requires = [
     "setuptools>=77.0.1",
     "wheel",
-    "cython==3.1.1"
+    "cython"
 ]
 build-backend = "setuptools.build_meta"
 
diff --git a/tools/distrib/python/grpcio_tools/pyproject.toml b/tools/distrib/python/grpcio_tools/pyproject.toml
index bb25192..2a8d03c 100644
--- a/tools/distrib/python/grpcio_tools/pyproject.toml
+++ b/tools/distrib/python/grpcio_tools/pyproject.toml
@@ -16,7 +16,7 @@
 requires = [
     "setuptools>=77.0.1",
     "wheel",
-    "Cython==3.1.1",
+    "Cython",
 ]
 build-backend = "setuptools.build_meta"
 
