--- a/comm/mailnews/compose/src/nsMsgCompUtils.cpp	2020-02-18 02:36:19.000000000 +0300
+++ b/comm/mailnews/compose/src/nsMsgCompUtils.cpp-OK	2021-01-27 02:20:24.665901361 +0300
@@ -357,18 +357,28 @@
 
     finalHeaders->SetRawHeader(HEADER_X_MOZILLA_DRAFT_INFO, draftInfo, nullptr);
   }
 
   nsCOMPtr<nsIHttpProtocolHandler> pHTTPHandler = do_GetService(NS_NETWORK_PROTOCOL_CONTRACTID_PREFIX "http", &rv);
   if (NS_SUCCEEDED(rv) && pHTTPHandler)
   {
     nsAutoCString userAgentString;
-    // Ignore error since we're testing the return value.
-    mozilla::Unused << pHTTPHandler->GetUserAgent(userAgentString);
+
+    // allow a user to override the default UA
+    // an empty string value drops UA header
+
+    nsCString userAgentOverride;
+    rv = prefs->GetCharPref("mailnews.useragent.override", getter_Copies(userAgentOverride));
+
+    if (NS_SUCCEEDED(rv))
+      userAgentString = userAgentOverride;
+    else
+      // Ignore error since we're testing the return value.
+      mozilla::Unused << pHTTPHandler->GetUserAgent(userAgentString);
 
     if (!userAgentString.IsEmpty())
       finalHeaders->SetUnstructuredHeader("User-Agent",
         NS_ConvertUTF8toUTF16(userAgentString));
   }
 
   finalHeaders->SetUnstructuredHeader("MIME-Version", NS_LITERAL_STRING("1.0"));
 
