diff -Nrup mozilla-OLD/layout/base/PresShell.cpp mozilla/layout/base/PresShell.cpp
--- mozilla-OLD/layout/base/PresShell.cpp	2021-08-08 16:01:59.000000000 +0300
+++ mozilla/layout/base/PresShell.cpp	2022-02-02 10:28:07.295905935 +0300
@@ -4200,6 +4200,9 @@ PresShell::DoFlushPendingNotifications(m
     if (!mIsDestroying) {
       nsAutoScriptBlocker scriptBlocker;
       mPresContext->RestyleManager()->ProcessPendingRestyles();
+      // Clear mNeedStyleFlush here agagin to make this flag work properly for
+      // optimization since the flag might have set in ProcessPendingRestyles().
+      mNeedStyleFlush = false;
     }
 
     didStyleFlush = true;
