--- firefox-52.1.0esr/layout/style/nsCSSPseudoElements.h.old	2017-04-11 04:13:10.000000000 +0200
+++ firefox-52.1.0esr/layout/style/nsCSSPseudoElements.h	2017-05-15 23:27:57.470258443 +0200
@@ -109,7 +109,8 @@
 
   // Work around https://gcc.gnu.org/bugzilla/show_bug.cgi?id=64037 ,
   // which is a general gcc bug that we seem to have hit only on Android/x86.
-#if defined(ANDROID) && defined(__i386__) && defined(__GNUC__) && \
+  // ...and CentOS 6/x86
+#if defined(__i386__) && defined(__GNUC__) && \
     !defined(__clang__)
 #if (MOZ_GCC_VERSION_AT_LEAST(4,8,0) && MOZ_GCC_VERSION_AT_MOST(4,8,4)) || \
     (MOZ_GCC_VERSION_AT_LEAST(4,9,0) && MOZ_GCC_VERSION_AT_MOST(4,9,2))
