--- a/comm/suite/installer/Makefile.in	2020-12-22 19:18:42.000000000 +0300
+++ b/comm/suite/installer/Makefile.in-OK	2021-01-23 00:34:17.416407249 +0300
@@ -140,16 +140,20 @@
 	defaults/profile/panels.rdf \
 	defaults/profile/mimeTypes.rdf \
 	defaults/profile/chrome/userChrome-example.css \
 	defaults/profile/chrome/userContent-example.css \
 	$(NULL)
 
 include $(topsrcdir)/toolkit/mozapps/installer/packager.mk
 
+install::
+	$(NSINSTALL) -D $(DESTDIR)$(mandir)/man1
+	$(SYSINSTALL) $(IFLAGS1) $(DIST)/man/man1/$(MOZ_APP_NAME).1 $(DESTDIR)$(mandir)/man1
+
 ifeq (Darwin, $(OS_ARCH))
 BINPATH = $(_BINPATH)
 DEFINES += -DAPPNAME=$(_APPNAME)
 else
 BINPATH = bin
 endif
 DEFINES += -DBINPATH=$(BINPATH)
 
--- a/build/moz.configure/old.configure	2022-02-02 15:44:48.908904555 +0300
+++ b/build/moz.configure/old.configure-OK	2022-02-02 15:48:09.807512967 +0300
@@ -220,16 +220,17 @@
     '--enable-universalchardet',
     '--enable-unsigned-overflow-sanitizer',
     '--enable-updater',
     '--enable-verify-mar',
     '--enable-xul',
     '--enable-zipwriter',
     '--includedir',
     '--libdir',
+    '--mandir',
     '--no-create',
     '--prefix',
     '--with-android-distribution-directory',
     '--with-android-max-sdk',
     '--with-android-min-sdk',
     '--with-android-sdk',
     '--with-app-basename',
     '--with-app-name',
