--- mozilla/toolkit/mozapps/extensions/internal/XPIInstall.jsm.orig	2022-06-08 22:10:29.000000000 +0300
+++ mozilla/toolkit/mozapps/extensions/internal/XPIInstall.jsm	2022-06-15 21:14:29.984281233 +0300
@@ -691,17 +691,17 @@ async function loadManifestFromRDF(aUri,
 
   // A theme's userDisabled value is true if the theme is not the selected skin
   // or if there is an active lightweight theme. We ignore whether softblocking
   // is in effect since it would change the active theme.
   if (isTheme(addon.type)) {
     addon.userDisabled = !!LightweightThemeManager.currentTheme ||
                          addon.internalName != XPIProvider.selectedSkin;
   } else {
-    addon.userDisabled = false;
+    addon.userDisabled = getRDFProperty(ds, root, "userDisabled") == "true";
   }
 
   addon.softDisabled = addon.blocklistState == nsIBlocklistService.STATE_SOFTBLOCKED;
   addon.applyBackgroundUpdates = AddonManager.AUTOUPDATE_DEFAULT;
 
   // icons will be filled by the calling function
   addon.icons = {};
   addon.userPermissions = null;
