--- a/comm/calendar/lightning/install.rdf.orig	2020-06-06 23:01:30.000000000 +0300
+++ b/comm/calendar/lightning/install.rdf	2021-03-31 02:40:43.366666893 +0300
@@ -45,5 +45,6 @@
     <em:optionsURL>chrome://communicator/content/pref/preferences.xul</em:optionsURL>
 #endif
     <em:strictCompatibility>true</em:strictCompatibility>
+    <em:userDisabled>true</em:userDisabled>
   </Description>
 </RDF>
--- a/comm/suite/extensions/inspector/install.rdf.orig	2020-02-18 02:40:17.000000000 +0300
+++ b/comm/suite/extensions/inspector/install.rdf	2021-03-31 02:44:18.618860758 +0300
@@ -103,6 +103,7 @@
     <em:description>Inspects the structure and properties of a window and its contents.</em:description>
     <em:creator>mozilla.org</em:creator>
     <em:homepageURL>http://www.mozilla.org/projects/inspector/</em:homepageURL>
+    <em:userDisabled>true</em:userDisabled>
 
     <em:localized>
       <Description>
