--- a/comm/mailnews/base/util/nsMsgDBFolder.cpp.orig	2021-05-25 21:36:20.000000000 +0300
+++ b/comm/mailnews/base/util/nsMsgDBFolder.cpp	2021-07-09 21:03:53.266796854 +0300
@@ -3900,15 +3900,17 @@ nsresult nsMsgDBFolder::CreateDirectoryF
 
   if(!pathIsDirectory)
   {
-    //If the current path isn't a directory, add directory separator
-    //and test it out.
-    rv = AddDirectorySeparator(path);
-    if(NS_FAILED(rv))
-      return rv;
-
-    //If that doesn't exist, then we have to create this directory
-    pathIsDirectory = false;
-    path->IsDirectory(&pathIsDirectory);
+    if(!isServer) {
+      //If the current path isn't a directory, add directory separator
+      //and test it out.
+      rv = AddDirectorySeparator(path);
+      if(NS_FAILED(rv))
+        return rv;
+  
+      //If that doesn't exist, then we have to create this directory
+      pathIsDirectory = false;
+      path->IsDirectory(&pathIsDirectory);
+    }
     if(!pathIsDirectory)
     {
       bool pathExists;
