diff -Nrup -U 8 mozilla/startupcache/StartupCache.cpp mozilla/startupcache/StartupCache.cpp
--- mozilla/startupcache/StartupCache.cpp	2019-12-29 20:38:43.000000000 +0300
+++ mozilla/startupcache/StartupCache.cpp	2020-02-24 01:24:52.935332070 +0300
@@ -76,21 +76,22 @@ StartupCache::CollectReports(nsIHandleRe
 
 StartupCache*
 StartupCache::GetSingleton()
 {
   if (!gStartupCache) {
     if (!XRE_IsParentProcess()) {
       return nullptr;
     }
-#ifdef MOZ_DISABLE_STARTUPCACHE
-    return nullptr;
-#else
-    StartupCache::InitSingleton();
-#endif
+
+    if (PR_GetEnv("MOZ_DISABLE_STARTUPCACHE")) {
+      return nullptr;
+    } else {
+      StartupCache::InitSingleton();
+    }
   }
 
   return StartupCache::gStartupCache;
 }
 
 void
 StartupCache::DeleteSingleton()
 {
