--- mozilla/dom/events/EventStateManager.cpp.orig	2021-08-17 20:56:14.484483225 +0300
+++ mozilla/dom/events/EventStateManager.cpp	2021-08-17 20:56:23.604425650 +0300
@@ -657,16 +657,19 @@ EventStateManager::PreHandleEvent(nsPres
   }
   case eMouseUp: {
     switch (mouseEvent->button) {
       case WidgetMouseEvent::eLeftButton:
         if (Prefs::ClickHoldContextMenu()) {
           KillClickHoldTimer();
         }
         StopTrackingDragGesture();
+        if (sActiveESM && sActiveESM != this) {
+          sActiveESM->StopTrackingDragGesture();
+        }
         sNormalLMouseEventInProcess = false;
         // then fall through...
         MOZ_FALLTHROUGH;
       case WidgetMouseEvent::eRightButton:
       case WidgetMouseEvent::eMiddleButton:
         SetClickCount(mouseEvent, aStatus);
         break;
     }
