--- a/netwerk/protocol/http/nsHttpHandler.cpp.orig	2021-11-23 02:47:09.005462286 +0300
+++ b/netwerk/protocol/http/nsHttpHandler.cpp	2021-11-23 02:48:27.445894730 +0300
@@ -452,9 +452,9 @@
 
     nsHttpChannelAuthProvider::InitializePrefs();
 
-    mMisc.AssignLiteral("rv:68.0");
+    mMisc.AssignLiteral("rv:" MOZILLA_UAVERSION);
 
-    mCompatFirefox.AssignLiteral("Firefox/68.0");
+    mCompatFirefox.AssignLiteral("Firefox/" MOZILLA_UAVERSION);
 
     nsCOMPtr<nsIXULAppInfo> appInfo =
         do_GetService("@mozilla.org/xre/app-info;1");
