--- mozilla/netwerk/protocol/http/nsHttpHandler.cpp.orig	2022-02-15 16:26:14.697023318 +0300
+++ mozilla/netwerk/protocol/http/nsHttpHandler.cpp	2022-02-15 16:26:21.180976404 +0300
@@ -370,11 +370,18 @@ nsHttpHandler::EnsureUAOverridesInit()
     MOZ_ASSERT(XRE_IsParentProcess());
     MOZ_ASSERT(NS_IsMainThread());
 
+    static bool initDone = false;
+
+    if (initDone)
+        return;
+
     nsresult rv;
     nsCOMPtr<nsISupports> bootstrapper
         = do_GetService("@mozilla.org/network/ua-overrides-bootstrapper;1", &rv);
     MOZ_ASSERT(bootstrapper);
     MOZ_ASSERT(NS_SUCCEEDED(rv));
+
+    initDone = true;
 }
 
 nsHttpHandler::~nsHttpHandler()
