--- a/toolkit/crashreporter/google-breakpad/src/third_party/curl/curlbuild.h	2021-05-20 02:07:53.799731821 -0500
+++ b/toolkit/crashreporter/google-breakpad/src/third_party/curl/curlbuild.h	2021-05-20 02:10:17.127497350 -0500
@@ -155,7 +155,7 @@
 
 /* The size of `long', as computed by sizeof. */
 #if defined(_M_X64) || (defined(__x86_64__) && !defined(__ILP32__)) ||      \
-    defined(__aarch64__) || (defined(__mips__) && _MIPS_SIM == _ABI64)
+    defined(__aarch64__) || (defined(__mips__) && _MIPS_SIM == _ABI64) || (defined(__powerpc64__))
 #define CURL_SIZEOF_LONG 8
 #else
 #define CURL_SIZEOF_LONG 4
