--- a/js/src/jit/arm64/MacroAssembler-arm64.h	2020-02-17 23:37:53.000000000 +0000
+++ b/js/src/jit/arm64/MacroAssembler-arm64.h	2022-03-27 05:56:00.542861646 +0000
@@ -253,6 +253,11 @@
         if (!GetStackPointer64().Is(vixl::sp))
             Mov(vixl::sp, GetStackPointer64());
     }
+    void initPseudoStackPtr() {
+      if (!GetStackPointer64().Is(vixl::sp)) {
+        Mov(GetStackPointer64(), vixl::sp);
+      }
+    }
     void initStackPtr() {
         if (!GetStackPointer64().Is(vixl::sp))
             Mov(GetStackPointer64(), vixl::sp);
