--- a/comm/mailnews/compose/src/nsMsgCompUtils.cpp.orig	2021-05-18 17:31:14.226249883 -0600
+++ b/comm/mailnews/compose/src/nsMsgCompUtils.cpp	2021-05-18 17:33:40.068526633 -0600
@@ -371,9 +371,12 @@
 
     if (NS_SUCCEEDED(rv))
       userAgentString = userAgentOverride;
-    else
+    else {
       // Ignore error since we're testing the return value.
       mozilla::Unused << pHTTPHandler->GetUserAgent(userAgentString);
+      // Properly advertise compatibility (if any).
+      userAgentString.ReplaceSubstring("Firefox", "Thunderbird");
+    }
 
     if (!userAgentString.IsEmpty())
       finalHeaders->SetUnstructuredHeader("User-Agent",
