--- mozilla/toolkit/mozapps/extensions/internal/XPIInstall.jsm.orig	2021-03-01 21:17:58.000000000 +0300
+++ mozilla/toolkit/mozapps/extensions/internal/XPIInstall.jsm	2021-03-31 02:33:44.392181653 +0300
@@ -698,17 +698,17 @@ async function loadManifestFromRDF(aUri,
   if (isTheme(addon.type)) {
     addon.userDisabled = !!LightweightThemeManager.currentTheme ||
                          addon.internalName != XPIProvider.selectedSkin;
   } else if (addon.type == "experiment") {
     // Experiments are disabled by default. It is up to the Experiments Manager
     // to enable them (it drives installation).
     addon.userDisabled = true;
   } else {
-    addon.userDisabled = false;
+    addon.userDisabled = getRDFProperty(ds, root, "userDisabled") == "true";
   }
 
   addon.softDisabled = addon.blocklistState == nsIBlocklistService.STATE_SOFTBLOCKED;
   addon.applyBackgroundUpdates = AddonManager.AUTOUPDATE_DEFAULT;
 
   // Experiments are managed and updated through an external "experiments
   // manager." So disable some built-in mechanisms.
   if (addon.type == "experiment") {
