--- a/comm/suite/components/places/PlacesUIUtils.jsm.orig	2021-04-08 19:35:52.322459790 +0300
+++ b/comm/suite/components/places/PlacesUIUtils.jsm	2021-04-08 19:35:56.256427521 +0300
@@ -921,17 +921,18 @@ var PlacesUIUtils = {
                   createInstance(Ci.nsIMutableArray);
       args.appendElement(uriList);
       browserWindow = Services.ww.openWindow(aWindow,
                                              "chrome://navigator/content/navigator.xul",
                                              null, "chrome,dialog=no,all", args);
       return;
     }
 
-    var loadInBackground = where == "tabshifted";
+    var loadInBackground = Services.prefs.getBoolPref("browser.tabs.loadBookmarksInBackground") ?
+                           (where == "current" || where == "tab") : (where == "tabshifted");
     // For consistency, we want all the bookmarks to open in new tabs, instead
     // of having one of them replace the currently focused tab.  Hence we call
     // loadTabs with aReplace set to false.
     browserWindow.gBrowser.loadTabs(urls, {
       inBackground: loadInBackground,
       replace: false,
       triggeringPrincipal: Services.scriptSecurityManager.getSystemPrincipal(),
     });
