diff -up firefox-51.0/dom/u2f/U2F.cpp.prbool firefox-51.0/dom/u2f/U2F.cpp
--- firefox-51.0/dom/u2f/U2F.cpp.prbool	2016-07-26 12:27:56.903381382 +0200
+++ firefox-51.0/dom/u2f/U2F.cpp	2016-07-26 12:28:08.924399413 +0200
@@ -4,6 +4,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "prtypes.h"
 #include "hasht.h"
 #include "mozilla/dom/ContentChild.h"
 #include "mozilla/dom/CryptoBuffer.h"
diff -up firefox-51.0/security/certverifier/OCSPCache.h.prbool firefox-51.0/security/certverifier/OCSPCache.h
--- firefox-51.0/security/certverifier/OCSPCache.h.prbool	2016-05-12 19:13:32.000000000 +0200
+++ firefox-51.0/security/certverifier/OCSPCache.h	2016-07-26 12:26:56.366290580 +0200
@@ -25,6 +25,7 @@
 #ifndef mozilla_psm_OCSPCache_h
 #define mozilla_psm_OCSPCache_h
 
+#include "prtypes.h"
 #include "hasht.h"
 #include "mozilla/Mutex.h"
 #include "mozilla/Vector.h"
diff -up firefox-51.0/security/certverifier/CTLogVerifier.cpp.old firefox-51.0/security/certverifier/CTLogVerifier.cpp
--- firefox-51.0/security/certverifier/CTLogVerifier.cpp.old	2017-01-18 13:03:34.758386735 +0100
+++ firefox-51.0/security/certverifier/CTLogVerifier.cpp	2017-01-18 13:03:38.753367776 +0100
@@ -7,6 +7,7 @@
 #include "CTLogVerifier.h"
 
 #include "CTSerialization.h"
+#include "prtypes.h"
 #include "hasht.h"
 #include "mozilla/ArrayUtils.h"
 #include "mozilla/Assertions.h"
diff -up firefox-51.0/security/certverifier/CTObjectsExtractor.cpp.old firefox-51.0/security/certverifier/CTObjectsExtractor.cpp
--- firefox-51.0/security/certverifier/CTObjectsExtractor.cpp.old	2017-01-18 13:01:57.629847680 +0100
+++ firefox-51.0/security/certverifier/CTObjectsExtractor.cpp	2017-01-18 13:02:01.892827449 +0100
@@ -6,6 +6,7 @@
 
 #include "CTObjectsExtractor.h"
 
+#include "prtypes.h"
 #include "hasht.h"
 #include "mozilla/Assertions.h"
 #include "mozilla/Casting.h"
