diff -Nrbu mozilla/netwerk/base/nsILoadGroup.idl mozilla-OK/netwerk/base/nsILoadGroup.idl
--- mozilla/netwerk/base/nsILoadGroup.idl	2016-12-14 05:08:20.000000000 +0300
+++ mozilla-OK/netwerk/base/nsILoadGroup.idl	2016-12-31 04:10:40.511379017 +0300
@@ -97,10 +97,4 @@
      * the docShell has created the default request.)
      */
     attribute nsLoadFlags defaultLoadFlags;
-
-    /**
-     * The cached user agent override created by UserAgentOverrides.jsm. Used
-     * for all sub-resource requests in the loadgroup.
-     */
-    attribute ACString userAgentOverrideCache;
 };
diff -Nrbu mozilla/netwerk/base/nsLoadGroup.cpp mozilla-OK/netwerk/base/nsLoadGroup.cpp
--- mozilla/netwerk/base/nsLoadGroup.cpp	2016-12-14 05:09:59.000000000 +0300
+++ mozilla-OK/netwerk/base/nsLoadGroup.cpp	2016-12-31 04:10:40.537378405 +0300
@@ -809,20 +809,6 @@
     return NS_OK;
 }
 
-NS_IMETHODIMP
-nsLoadGroup::GetUserAgentOverrideCache(nsACString & aUserAgentOverrideCache)
-{
-  aUserAgentOverrideCache = mUserAgentOverrideCache;
-  return NS_OK;
-}
-
-NS_IMETHODIMP
-nsLoadGroup::SetUserAgentOverrideCache(const nsACString & aUserAgentOverrideCache)
-{
-  mUserAgentOverrideCache = aUserAgentOverrideCache;
-  return NS_OK;
-}
-
 
 ////////////////////////////////////////////////////////////////////////////////
 
diff -Nrbu mozilla/netwerk/base/nsLoadGroup.h mozilla-OK/netwerk/base/nsLoadGroup.h
--- mozilla/netwerk/base/nsLoadGroup.h	2016-12-14 05:08:20.000000000 +0300
+++ mozilla-OK/netwerk/base/nsLoadGroup.h	2016-12-31 04:12:09.525381401 +0300
@@ -95,8 +95,6 @@
 
     /* For nsPILoadGroupInternal */
     uint32_t                        mTimedNonCachedRequestsUntilOnEndPageLoad;
-
-    nsCString                       mUserAgentOverrideCache;
 };
 
 } // namespace net
diff -Nrbu mozilla/netwerk/protocol/http/UserAgentOverrides.jsm mozilla-OK/netwerk/protocol/http/UserAgentOverrides.jsm
--- mozilla/netwerk/protocol/http/UserAgentOverrides.jsm	2016-12-14 05:08:21.000000000 +0300
+++ mozilla-OK/netwerk/protocol/http/UserAgentOverrides.jsm	2016-12-31 04:10:46.388378377 +0300
@@ -46,9 +46,9 @@
     Services.prefs.addObserver(PREF_OVERRIDES_ENABLED, buildOverrides, false);
 
     try {
-      Services.obs.addObserver(HTTP_on_useragent_request, "http-on-useragent-request", false);
+      Services.obs.addObserver(HTTP_on_modify_request, "http-on-modify-request", false);
     } catch (x) {
-      // The http-on-useragent-request notification is disallowed in content processes.
+      // The http-on-modify-request notification is disallowed in content processes.
     }
 
     UserAgentUpdates.init(function(overrides) {
@@ -118,7 +118,7 @@
 
     Services.prefs.removeObserver(PREF_OVERRIDES_ENABLED, buildOverrides);
 
-    Services.obs.removeObserver(HTTP_on_useragent_request, "http-on-useragent-request");
+    Services.obs.removeObserver(HTTP_on_modify_request, "http-on-modify-request");
   },
 
   receiveMessage: function(aMessage) {
@@ -169,7 +169,7 @@
   }
 }
 
-function HTTP_on_useragent_request(aSubject, aTopic, aData) {
+function HTTP_on_modify_request(aSubject, aTopic, aData) {
   let channel = aSubject.QueryInterface(Ci.nsIHttpChannel);
 
   for (let callback of gOverrideFunctions) {
diff -Nrbu mozilla/netwerk/protocol/http/nsHttpChannel.cpp mozilla-OK/netwerk/protocol/http/nsHttpChannel.cpp
--- mozilla/netwerk/protocol/http/nsHttpChannel.cpp	2016-12-14 05:10:00.000000000 +0300
+++ mozilla-OK/netwerk/protocol/http/nsHttpChannel.cpp	2016-12-31 04:16:48.141377976 +0300
@@ -5512,8 +5512,6 @@
     // notify "http-on-modify-request" observers
     CallOnModifyRequestObservers();
 
-    SetLoadGroupUserAgentOverride();
-
     // Check to see if we should redirect this channel elsewhere by
     // nsIHttpChannel.redirectTo API request
     if (mAPIRedirectToURI) {
@@ -7592,50 +7590,6 @@
 }
 
 void
-nsHttpChannel::SetLoadGroupUserAgentOverride()
-{
-    nsCOMPtr<nsIURI> uri;
-    GetURI(getter_AddRefs(uri));
-    nsAutoCString uriScheme;
-    if (uri) {
-        uri->GetScheme(uriScheme);
-    }
-
-    // We don't need a UA for file: protocols.
-    if (uriScheme.EqualsLiteral("file")) {
-        gHttpHandler->OnUserAgentRequest(this);
-        return;
-    }
-
-    nsIRequestContextService* rcsvc = gHttpHandler->GetRequestContextService();
-    nsCOMPtr<nsIRequestContext> rc;
-    if (rcsvc) {
-        rcsvc->GetRequestContext(mRequestContextID,
-                                    getter_AddRefs(rc));
-    }
-
-    nsAutoCString ua;
-    if (nsContentUtils::IsNonSubresourceRequest(this)) {
-        gHttpHandler->OnUserAgentRequest(this);
-        if (rc) {
-            GetRequestHeader(NS_LITERAL_CSTRING("User-Agent"), ua);
-            rc->SetUserAgentOverride(ua);
-        }
-    } else {
-        GetRequestHeader(NS_LITERAL_CSTRING("User-Agent"), ua);
-        // Don't overwrite the UA if it is already set (eg by an XHR with explicit UA).
-        if (ua.IsEmpty()) {
-            if (rc) {
-                rc->GetUserAgentOverride(ua);
-                SetRequestHeader(NS_LITERAL_CSTRING("User-Agent"), ua, false);
-            } else {
-                gHttpHandler->OnUserAgentRequest(this);
-            }
-        }
-    }
-}
-
-void
 nsHttpChannel::SetDoNotTrack()
 {
   /**
diff -Nrbu mozilla/netwerk/protocol/http/nsHttpChannel.h mozilla-OK/netwerk/protocol/http/nsHttpChannel.h
--- mozilla/netwerk/protocol/http/nsHttpChannel.h	2016-12-14 05:10:00.000000000 +0300
+++ mozilla-OK/netwerk/protocol/http/nsHttpChannel.h	2016-12-31 04:15:24.541376728 +0300
@@ -434,8 +434,6 @@
 
     void MaybeWarnAboutAppCache();
 
-    void SetLoadGroupUserAgentOverride();
-
     void SetDoNotTrack();
 
 private:
diff -Nrbu mozilla/netwerk/protocol/http/nsHttpHandler.h mozilla-OK/netwerk/protocol/http/nsHttpHandler.h
--- mozilla/netwerk/protocol/http/nsHttpHandler.h	2016-12-14 05:08:22.000000000 +0300
+++ mozilla-OK/netwerk/protocol/http/nsHttpHandler.h	2016-12-31 04:10:46.392378377 +0300
@@ -278,12 +278,6 @@
         NotifyObservers(chan, NS_HTTP_ON_MODIFY_REQUEST_TOPIC);
     }
 
-    // Called by the channel and cached in the loadGroup
-    void OnUserAgentRequest(nsIHttpChannel *chan)
-    {
-      NotifyObservers(chan, NS_HTTP_ON_USERAGENT_REQUEST_TOPIC);
-    }
-
     // Called by the channel once headers are available
     void OnExamineResponse(nsIHttpChannel *chan)
     {
diff -Nrbu mozilla/netwerk/protocol/http/nsIHttpProtocolHandler.idl mozilla-OK/netwerk/protocol/http/nsIHttpProtocolHandler.idl
--- mozilla/netwerk/protocol/http/nsIHttpProtocolHandler.idl	2016-12-14 05:08:22.000000000 +0300
+++ mozilla-OK/netwerk/protocol/http/nsIHttpProtocolHandler.idl	2016-12-31 04:10:46.392378377 +0300
@@ -113,14 +113,5 @@
  */
 #define NS_HTTP_ON_EXAMINE_CACHED_RESPONSE_TOPIC "http-on-examine-cached-response"
 
-/**
- * Before an HTTP request corresponding to a channel with the LOAD_DOCUMENT_URI
- * flag is sent to the server, this observer topic is notified. The observer of
- * this topic can then choose to modify the user agent for this request before
- * the request is actually sent to the server. Additionally, the modified user
- * agent will be propagated to sub-resource requests from the same load group.
- */
-#define NS_HTTP_ON_USERAGENT_REQUEST_TOPIC "http-on-useragent-request"
-
 
 %}
