diff -Nrup -U 8 seamonkey-2.53-comm-2_53_beta_01/mozilla/startupcache/StartupCache.cpp seamonkey-2.53-comm-2_53_beta_01-OK/mozilla/startupcache/StartupCache.cpp
--- seamonkey-2.53-comm-2_53_beta_01/mozilla/startupcache/StartupCache.cpp	2019-12-29 20:38:43.000000000 +0300
+++ seamonkey-2.53-comm-2_53_beta_01-OK/mozilla/startupcache/StartupCache.cpp	2020-02-24 01:24:52.935332070 +0300
@@ -76,21 +76,22 @@ StartupCache::CollectReports(nsIHandleRe
 
 StartupCache*
 StartupCache::GetSingleton()
 {
   if (!gStartupCache) {
     if (!XRE_IsParentProcess()) {
       return nullptr;
     }
-#ifdef MOZ_DISABLE_STARTUPCACHE
-    return nullptr;
-#else
-    StartupCache::InitSingleton();
-#endif
+
+    if (PR_GetEnv("MOZ_DISABLE_STARTUPCACHE")) {
+      return nullptr;
+    } else {
+      StartupCache::InitSingleton();
+    }
   }
 
   return StartupCache::gStartupCache;
 }
 
 void
 StartupCache::DeleteSingleton()
 {
diff -Nrup -U 8 seamonkey-2.53-comm-2_53_beta_01/suite/components/pref/content/pref-cache.js seamonkey-2.53-comm-2_53_beta_01-OK/suite/components/pref/content/pref-cache.js
--- seamonkey-2.53-comm-2_53_beta_01/suite/components/pref/content/pref-cache.js	2019-12-31 03:14:14.000000000 +0300
+++ seamonkey-2.53-comm-2_53_beta_01-OK/suite/components/pref/content/pref-cache.js	2020-02-24 01:24:52.955331955 +0300
@@ -106,8 +106,15 @@ function ReadSmartSizeEnabled()
 }
 
 function updateActualCacheSize()
 {
   Cc["@mozilla.org/netwerk/cache-storage-service;1"]
     .getService(Ci.nsICacheStorageService)
     .asyncGetDiskConsumption(CacheObserver);
 }
+
+function prefClearStartupCache(aButton)
+{
+  Services.appinfo.invalidateCachesOnRestart();
+  aButton.disabled = true;
+}
+
diff -Nrup -U 8 seamonkey-2.53-comm-2_53_beta_01/suite/components/pref/content/pref-cache.xul seamonkey-2.53-comm-2_53_beta_01-OK/suite/components/pref/content/pref-cache.xul
--- seamonkey-2.53-comm-2_53_beta_01/suite/components/pref/content/pref-cache.xul	2019-12-31 03:14:14.000000000 +0300
+++ seamonkey-2.53-comm-2_53_beta_01-OK/suite/components/pref/content/pref-cache.xul	2020-02-24 02:54:04.763717723 +0300
@@ -103,16 +103,28 @@
             <menuitem value="0" label="&checkOncePerSession.label;"/>
             <menuitem value="2" label="&checkNever.label;"/>
           </menupopup>
         </menulist>
       </hbox>
 
     </groupbox>
 
+    <groupbox id="startupCache">
+      <caption id="startupCacheLabel" label="&startupCacheTitle.label;"/>
+      <description>&startupCachePara;</description>
+      <hbox id="startupCacheBox" align="center">
+        <button id="clearStartupCacheButton"
+                class="indent"
+                label="&clearStartupCache.label;"
+                accesskey="&clearStartupCache.accesskey;"
+                oncommand="prefClearStartupCache(this);"/>
+      </hbox>
+    </groupbox>
+
     <groupbox id="prefetch">
       <caption id="prefetchLabel" label="&prefetchTitle.label;"/>
       <vbox id="prefetchBox" align="start">
         <checkbox id="enablePrefetch"
                   label="&enablePrefetch.label;"
                   accesskey="&enablePrefetch.accesskey;"
                   preference="network.prefetch-next"/>
       </vbox>
diff -Nrup -U 8 seamonkey-2.53-comm-2_53_beta_01/suite/locales/en-US/chrome/common/pref/pref-cache.dtd seamonkey-2.53-comm-2_53_beta_01-OK/suite/locales/en-US/chrome/common/pref/pref-cache.dtd
--- seamonkey-2.53-comm-2_53_beta_01/suite/locales/en-US/chrome/common/pref/pref-cache.dtd	2019-12-31 03:14:14.000000000 +0300
+++ seamonkey-2.53-comm-2_53_beta_01-OK/suite/locales/en-US/chrome/common/pref/pref-cache.dtd	2020-02-24 02:50:26.000000000 +0300
@@ -21,11 +21,16 @@
 <!ENTITY diskCacheFolderExplanation      "Cache files will be stored in a subfolder named &quot;Cache&quot; of the directory you specify. Restart &brandShortName; for changes to take effect.">
 <!ENTITY docCache.label                  "Compare the page in the cache to the page on the network:">
 <!ENTITY docCache.accesskey              "m">
 <!ENTITY checkOncePerSession.label       "Once per session">
 <!ENTITY checkEveryTime.label            "Every time I view the page">
 <!ENTITY checkNever.label                "Never">
 <!ENTITY checkAutomatically.label        "When the page is out of date">
 
+<!ENTITY startupCacheTitle.label         "Startup Cache">
+<!ENTITY startupCachePara                "Startup cache is used to speed up SeaMonkey loading. (Updates or changes of some addons and extensions might require to flush the cache afterwards.)">
+<!ENTITY clearStartupCache.label         "Clear startupCache">
+<!ENTITY clearStartupCache.accesskey     "s">
+
 <!ENTITY prefetchTitle.label             "Link Prefetching">
 <!ENTITY enablePrefetch.label            "Prefetch web pages when idle, so that links in web pages designed for prefetching can load more quickly">
 <!ENTITY enablePrefetch.accesskey        "P">
