--- seamonkey-2.53.1/mozilla/widget/gtk/nsFilePicker.cpp.orig	2020-01-03 01:35:33.934711196 +0300
+++ seamonkey-2.53.1/mozilla/widget/gtk/nsFilePicker.cpp	2020-01-03 01:39:14.120136390 +0300
@@ -396,9 +396,7 @@
 
   void *file_chooser = GtkFileChooserNew(title.get(), parent_widget, action, accept_button);
 
-  if (mAllowURLs) {
-    gtk_file_chooser_set_local_only(GTK_FILE_CHOOSER(file_chooser), FALSE);
-  }
+  gtk_file_chooser_set_local_only(GTK_FILE_CHOOSER(file_chooser), FALSE);
 
   if (action == GTK_FILE_CHOOSER_ACTION_OPEN || action == GTK_FILE_CHOOSER_ACTION_SAVE) {
     GtkWidget *img_preview = gtk_image_new();
