diff -Nrup -U 8 a/mozilla/tools/profiler/core/platform.h b/mozilla/tools/profiler/core/platform.h
--- a/mozilla/tools/profiler/core/platform.h	2020-02-18 02:39:44.000000000 +0300
+++ b/mozilla/tools/profiler/core/platform.h	2020-03-01 06:04:13.875681471 +0300
@@ -38,32 +38,26 @@
 #include "mozilla/StaticMutex.h"
 #include "mozilla/TimeStamp.h"
 #include "mozilla/UniquePtr.h"
 #include "mozilla/Unused.h"
 #include "PlatformMacros.h"
 #include <vector>
 #include "StackTop.h"
 
-// We need a definition of gettid(), but glibc doesn't provide a
+// We need a definition of gettid(), but old glibc versions don't provide a
 // wrapper for it.
 #if defined(__GLIBC__)
 #include <unistd.h>
 #include <sys/syscall.h>
-static inline pid_t gettid()
-{
-  return (pid_t) syscall(SYS_gettid);
-}
+#define gettid() static_cast<pid_t>(syscall(SYS_gettid))
 #elif defined(GP_OS_darwin)
 #include <unistd.h>
 #include <sys/syscall.h>
-static inline pid_t gettid()
-{
-  return (pid_t) syscall(SYS_thread_selfid);
-}
+#define gettid() static_cast<pid_t>(syscall(SYS_thread_selfid))
 #elif defined(GP_OS_android)
 #include <unistd.h>
 #elif defined(GP_OS_windows)
 #include <windows.h>
 #include <process.h>
 #ifndef getpid
 #define getpid _getpid
 #endif
