diff -Nrbu seamonkey-2.53.1/mozilla/security/certverifier/NSSCertDBTrustDomain.cpp seamonkey-2.53.1-OK/mozilla/security/certverifier/NSSCertDBTrustDomain.cpp
--- seamonkey-2.53.1/mozilla/security/certverifier/NSSCertDBTrustDomain.cpp	2019-12-29 20:38:43.000000000 +0300
+++ seamonkey-2.53.1-OK/mozilla/security/certverifier/NSSCertDBTrustDomain.cpp	2020-01-03 01:20:05.309371162 +0300
@@ -1146,12 +1146,10 @@
   if (!loadPKCS11Modules) {
     flags |= NSS_INIT_NOMODDB;
   }
-  nsAutoCString dbTypeAndDirectory("sql:");
-  dbTypeAndDirectory.Append(dir);
   MOZ_LOG(gCertVerifierLog, LogLevel::Debug,
-          ("InitializeNSS(%s, %d, %d)", dbTypeAndDirectory.get(), readOnly,
+          ("InitializeNSS(%s, %d, %d)", PromiseFlatCString(dir).get(), readOnly,
            loadPKCS11Modules));
-  SECStatus srv = NSS_Initialize(dbTypeAndDirectory.get(), "", "",
+  SECStatus srv = NSS_Initialize(PromiseFlatCString(dir).get(), "", "",
                                  SECMOD_DB, flags);
   if (srv != SECSuccess) {
     return srv;
