diff -Nrbu seamonkey-2.49.5/mozilla/security/manager/ssl/moz.build seamonkey-2.49.5/mozilla/security/manager/ssl/moz.build
--- seamonkey-2.49.5/mozilla/security/manager/ssl/moz.build	2019-05-06 20:30:52.000000000 +0300
+++ seamonkey-2.49.5/mozilla/security/manager/ssl/moz.build	2019-08-12 01:40:45.891454909 +0300
@@ -184,10 +184,6 @@
 
 DEFINES['CERT_AddTempCertToPerm'] = '__CERT_AddTempCertToPerm'
 
-USE_LIBS += [
-    'crmf',
-]
-
 include('/ipc/chromium/chromium-config.mozbuild')
 
 if CONFIG['GNU_CXX']:
