diff -Nrbu seamonkey-2.49.5/suite/app/Makefile.in seamonkey-2.49.5/suite/app/Makefile.in
--- seamonkey-2.49.5/suite/app/Makefile.in	2019-07-14 00:01:29.000000000 +0300
+++ seamonkey-2.49.5/suite/app/Makefile.in	2019-08-19 22:45:02.346141684 +0300
@@ -97,6 +97,7 @@
 	$(foreach dir,$(filter-out $(DONOTPACK),$(subst $(STAGEDIST)/,,$(wildcard $(STAGEDIST)/*))),$(_PACKAGE_EXTENSIONS))
 
 tools::
+	$(NSINSTALL) -D $(DIST)/bin/distribution/extensions
 	@$(MAKE) pack-ext STAGEDIST="$(DIST)/bin/extensions"
 	@$(MAKE) pack-ext STAGEDIST="$(DIST)/bin/distribution/extensions"
 
diff -Nrbu seamonkey-2.49.5/suite/locales/Makefile.in seamonkey-2.49.5/suite/locales/Makefile.in
--- seamonkey-2.49.5/suite/locales/Makefile.in	2019-07-14 00:01:29.000000000 +0300
+++ seamonkey-2.49.5/suite/locales/Makefile.in	2019-08-19 22:45:21.358643583 +0300
@@ -123,7 +123,7 @@
 	@$(MAKE) -C $(DEPTH)/toolkit/locales libs-$*
 	@$(MAKE) -C $(DEPTH)/services/sync/locales AB_CD=$* XPI_NAME=locale-$*
 	@$(MAKE) -C ../../editor/ui/locales AB_CD=$* XPI_NAME=locale-$*
-	@$(MAKE) -C $(DEPTH)/extensions/irc/locales libs-$*
+	@$(MAKE) -C $(DEPTH)/extensions/irc/locales libs-$* || :
 	@$(MAKE) -C $(DEPTH)/extensions/spellcheck/locales AB_CD=$* XPI_NAME=locale-$*
 	@$(MAKE) -C $(DEPTH)/intl/locales AB_CD=$* XPI_NAME=locale-$*
 	@$(MAKE) -C $(DEPTH)/devtools/client/locales AB_CD=$* XPI_NAME=locale-$*
