--- seamonkey-2.53.1/mozilla/old-configure.in.orig	2019-12-29 20:38:43.000000000 +0300
+++ seamonkey-2.53.1/mozilla/old-configure.in	2020-01-06 03:19:40.952857698 +0300
@@ -48,7 +48,7 @@
 MOZJPEG=62
 MOZPNG=10635
 NSPR_VERSION=4
-NSPR_MINVER=4.25
+NSPR_MINVER=4.21
 NSS_VERSION=3
 
 dnl Set the minimum version of toolkit libs used by mozilla
@@ -1808,7 +1808,7 @@
     _USE_SYSTEM_NSS=1 )
 
 if test -n "$_USE_SYSTEM_NSS"; then
-    AM_PATH_NSS(3.50, [MOZ_SYSTEM_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
+    AM_PATH_NSS(3.44.0, [MOZ_SYSTEM_NSS=1], [AC_MSG_ERROR([you don't have NSS installed or your version is too old])])
 fi
 
 NSS_CFLAGS="$NSS_CFLAGS -I${DIST}/include/nss"
