diff -Nrbu seamonkey-2.46/seamonkey-2.46/mozilla/config/baseconfig.mk seamonkey-2.46-OK/seamonkey-2.46/mozilla/config/baseconfig.mk
--- seamonkey-2.46/seamonkey-2.46/mozilla/config/baseconfig.mk	2016-10-05 10:31:33.000000000 +0300
+++ seamonkey-2.46-OK/seamonkey-2.46/mozilla/config/baseconfig.mk	2016-10-14 00:21:15.964029335 +0300
@@ -2,10 +2,10 @@
 # directly in python/mozbuild/mozbuild/base.py for gmake validation.
 # We thus use INCLUDED_AUTOCONF_MK to enable/disable some parts depending
 # whether a normal build is happening or whether the check is running.
-includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
+includedir := $(includedir)/$(MOZ_APP_NAME)
+idldir = $(datadir)/idl/$(MOZ_APP_NAME)
+installdir = $(libdir)/$(MOZ_APP_NAME)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel
 ifeq (.,$(DEPTH))
 DIST = dist
 else
