From efc94a5b15ffa544c9dbeb92bca2b43ebbc9f0bc Mon Sep 17 00:00:00 2001
From: Leslie Zhai <zhaixiang@loongson.cn>
Date: Sat, 13 Apr 2024 15:39:48 +0800
Subject: [PATCH 2/2] 485432: Fix libavcodec version major typo

---
 plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp b/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
index dd51afee3..e8eeb4ba8 100644
--- a/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
+++ b/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
@@ -139,7 +139,7 @@ bool K3bFFMpegFile::open()
     }
 
     d->sampleFormat = d->codecContext->sample_fmt;
-#if LIBAVCODEC_VERSION_MAJOR < 70
+#if LIBAVCODEC_VERSION_MAJOR < 61
     d->isSpacious = ::av_sample_fmt_is_planar(d->sampleFormat) &&
         d->codecContext->channels > 1;
 #else
@@ -190,7 +190,7 @@ int K3bFFMpegFile::sampleRate() const
 
 int K3bFFMpegFile::channels() const
 {
-#if LIBAVCODEC_VERSION_MAJOR < 70
+#if LIBAVCODEC_VERSION_MAJOR < 61
     return d->codecContext->channels;
 #else
 #pragma Unimplemented
-- 
2.46.0

