diff --git a/kdev-pg/CMakeLists.txt b/kdev-pg/CMakeLists.txt
index 5490835..19dc3a4 100644
--- a/kdev-pg/CMakeLists.txt
+++ b/kdev-pg/CMakeLists.txt
@@ -76,7 +76,7 @@ IF(FLEX_EXECUTABLE)
         DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/kdev-pg-lexer.ll"
                  ${OPTIONAL_PARSER_HEADER_DEPENDENCY}
         COMMAND ${FLEX_EXECUTABLE}
-        ARGS    --nounistd -o"${CMAKE_CURRENT_BINARY_DIR}/kdev-pg-lexer.cc"
+        ARGS    --yymore --nounistd -o"${CMAKE_CURRENT_BINARY_DIR}/kdev-pg-lexer.cc"
                 "${CMAKE_CURRENT_SOURCE_DIR}/kdev-pg-lexer.ll"
         )
 
