diff -Naur kdeconnect-kde-22.04.3/data/org.kde.kdeconnect_open.desktop kdeconnect-kde-22.04.3.nourlhandler/data/org.kde.kdeconnect_open.desktop
--- kdeconnect-kde-22.04.3/data/org.kde.kdeconnect_open.desktop	2022-09-13 23:42:14.238526915 +0000
+++ kdeconnect-kde-22.04.3.nourlhandler/data/org.kde.kdeconnect_open.desktop	2022-09-13 23:41:14.132477561 +0000
@@ -126,3 +126,4 @@
 Terminal=false
 Categories=Qt;KDE;Utility;
 X-KDE-FormFactor=desktop;tablet;handset;
+NoDisplay=true
