From 7ab2f9da443704ba006565b13bf456c9c825daf1 Mon Sep 17 00:00:00 2001
From: SkyBird233 <52884766+SkyBird233@users.noreply.github.com>
Date: Thu, 29 Jan 2026 14:32:03 +0800
Subject: [PATCH] fix: f-string syntax error on Python < 3.12

---
 legendary/core.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/legendary/core.py b/legendary/core.py
index 7e47003..6620059 100644
--- a/legendary/core.py
+++ b/legendary/core.py
@@ -1098,7 +1098,7 @@ class LegendaryCore:
             self.log.info(f'Downloading {len(chunkLinks)} chunks...')
 
             def log_download(r, *args, **kwargs):
-                self.log.debug(f'Downloaded chunk {'/'.join(urlparse(r.url).path.split('/')[5:])} successfully.')
+                self.log.debug(f'Downloaded chunk {"/".join(urlparse(r.url).path.split("/")[5:])} successfully.')
 
             # map chunkLinks to self.egs.unauth_future_session.get
             futures = [self.egs.unauth_future_session.get(link, hooks={'response': log_download}) for link in chunkLinks]
-- 
2.52.0

