diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9965d21b..4235fd08 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,7 +69,7 @@ find_package(folly CONFIG REQUIRED)
 find_package(fizz CONFIG REQUIRED)
 find_package(fmt CONFIG REQUIRED)
 find_package(OpenSSL REQUIRED)
-find_package(Glog REQUIRED)
+find_package(glog CONFIG QUIET)
 find_package(gflags CONFIG QUIET)
 if (gflags_FOUND)
   message(STATUS "Found gflags from package config")
@@ -165,7 +165,6 @@ target_include_directories(
     ${FOLLY_INCLUDE_DIR}
     ${Boost_INCLUDE_DIRS}
     ${OPENSSL_INCLUDE_DIR}
-    ${GLOG_INCLUDE_DIRS}
     ${GFLAGS_INCLUDE_DIRS}
     ${LIBEVENT_INCLUDE_DIR}
     ${DOUBLE_CONVERSION_INCLUDE_DIR}
@@ -175,7 +174,7 @@ target_link_libraries(wangle PUBLIC
   ${FIZZ_LIBRARIES}
   ${Boost_LIBRARIES}
   ${OPENSSL_LIBRARIES}
-  ${GLOG_LIBRARIES}
+  glog::glog
   ${GFLAGS_LIBRARIES}
   ${LIBEVENT_LIB}
   ${DOUBLE_CONVERSION_LIBRARY}
