From a7957ae0a731d3508728b32b99a9986c3cbe99fb Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Wed, 4 Mar 2026 20:16:58 +0800
Subject: [PATCH 1/3] AOSCOS: use local patched rusty_v8

---
 Cargo.lock | 2 --
 Cargo.toml | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index cdfd1d25b..b1a9ad35b 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -11172,8 +11172,6 @@ dependencies = [
 [[package]]
 name = "v8"
 version = "146.8.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "67519d234f7bc5a54a71f7fbc52258b0f3a7e9c11f365213d49016a59207ec10"
 dependencies = [
  "bindgen 0.72.1",
  "bitflags 2.9.3",
diff --git a/Cargo.toml b/Cargo.toml
index e57819f8e..7f296e004 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -601,3 +601,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.twox-hash]
 opt-level = 3
+
+[patch.crates-io]
+v8 = { path = '../rusty_v8' }
-- 
2.53.0

