From 4b633c15b4bf42ad2a480aa74f57e0b90730aa9f Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Fri, 24 Jul 2026 20:23:18 +0800
Subject: [PATCH 1/3] AOSCOS: use local patched rusty_v8

---
 Cargo.lock | 2 --
 Cargo.toml | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 4726162123..0dce0dd827 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -11230,8 +11230,6 @@ dependencies = [
 [[package]]
 name = "v8"
 version = "150.4.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "42a978ff11f15b24e5c05a7123cf2b68f41e763546699781a924ef4e2cf43a49"
 dependencies = [
  "bindgen 0.72.1",
  "bitflags 2.9.3",
diff --git a/Cargo.toml b/Cargo.toml
index bef2a7d583..4b9c00a1bc 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -637,3 +637,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.twox-hash]
 opt-level = 3
+
+[patch.crates-io]
+v8 = { path = "../rusty_v8" }
-- 
2.55.0

