From 74fe87db73d0950fb0e63a73629170a3b01a1b52 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Thu, 11 Dec 2025 10:52:06 +0800
Subject: [PATCH 1/2] use local patched rusty_v8

---
 Cargo.lock | 2 --
 Cargo.toml | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 3a89641b..6722cc6f 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -10025,8 +10025,6 @@ dependencies = [
 [[package]]
 name = "v8"
 version = "142.2.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "f566072bd76b2631d0dca1d90a766c943863b1fd6b01312281dc919816de976d"
 dependencies = [
  "bindgen 0.72.1",
  "bitflags 2.9.3",
diff --git a/Cargo.toml b/Cargo.toml
index d3c49bf2..36ed4d4f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -542,3 +542,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.deno_npm_cache]
 opt-level = 3
+
+[patch.crates-io]
+v8 = { path = '../rusty_v8' }
-- 
2.52.0

