From 8ecf71079aaef11bb4f4dcb5908eb9cbf74fc380 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sat, 23 May 2026 00:33:21 +0800
Subject: [PATCH 1/2] AOSCOS: use local patched rusty_v8

---
 Cargo.lock | 2 --
 Cargo.toml | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 4651851f3..c28575193 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -11336,8 +11336,6 @@ dependencies = [
 [[package]]
 name = "v8"
 version = "149.0.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "55897294cf631f99d5ff57c6a0611648861645ed7c8f251579161e0768812840"
 dependencies = [
  "bindgen 0.72.1",
  "bitflags 2.9.3",
diff --git a/Cargo.toml b/Cargo.toml
index 2be6c83cc..982a10cdd 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -612,3 +612,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.twox-hash]
 opt-level = 3
+
+[patch.crates-io]
+v8 = { path = '../rusty_v8' }
-- 
2.54.0

