From 1a17cfe538213353d56dd66a00c1a511e7edd34f Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Thu, 29 Jan 2026 23:16:48 +0800
Subject: [PATCH 1/3] AOSCOS: use local patched rusty_v8

---
 Cargo.lock | 2 --
 Cargo.toml | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 3e89f78ae..8fdafe06c 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -10153,8 +10153,6 @@ dependencies = [
 [[package]]
 name = "v8"
 version = "145.0.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "61d9a107e16bae35a0be2bb0096ac1d2318aac352c82edd796ab2b9cac66d8f0"
 dependencies = [
  "bindgen 0.72.1",
  "bitflags 2.9.3",
diff --git a/Cargo.toml b/Cargo.toml
index b9c04b529..4c5c49929 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -547,3 +547,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.deno_npm_cache]
 opt-level = 3
+
+[patch.crates-io]
+v8 = { path = '../rusty_v8' }
-- 
2.52.0

