From 5655e5ae6348f57709bb87b59061c7a56325acb8 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Wed, 4 Mar 2026 20:16:58 +0800
Subject: [PATCH 1/2] AOSCOS: use local patched rusty_v8

---
 Cargo.lock | 2 --
 Cargo.toml | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 77a5e6474..06911ea73 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -11266,8 +11266,6 @@ dependencies = [
 [[package]]
 name = "v8"
 version = "147.4.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "2df8fffd507fb18ed000673a83d937f58e60fb07f3306b2274284125b15137cd"
 dependencies = [
  "bindgen 0.72.1",
  "bitflags 2.9.3",
diff --git a/Cargo.toml b/Cargo.toml
index 401fc3c0a..f4167b1ef 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -610,3 +610,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.twox-hash]
 opt-level = 3
+
+[patch.crates-io]
+v8 = { path = '../rusty_v8' }
-- 
2.54.0

