From a06fec63128b2a23c471481dab4d271a8c7b9ef6 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Wed, 4 Mar 2026 20:16:58 +0800
Subject: [PATCH 1/2] AOSCOS: use local patched rusty_v8

---
 Cargo.lock | 2 --
 Cargo.toml | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 3b7c5bccc..d16844c88 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -11195,8 +11195,6 @@ dependencies = [
 [[package]]
 name = "v8"
 version = "147.0.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "a98a7a54a2cbb941924658340efeff052840ab2dcb925c34260142ba85310023"
 dependencies = [
  "bindgen 0.72.1",
  "bitflags 2.9.3",
diff --git a/Cargo.toml b/Cargo.toml
index d1878f148..2865d7de0 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -604,3 +604,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.twox-hash]
 opt-level = 3
+
+[patch.crates-io]
+v8 = { path = '../rusty_v8' }
-- 
2.53.0

