From a41d3b0ee475cd45169edf9fa0cf14c46ed863a6 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Fri, 24 Jul 2026 20:23:18 +0800
Subject: [PATCH 1/2] AOSCOS: use local patched rusty_v8

---
 Cargo.lock | 2 --
 Cargo.toml | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 8739cfa1c2..f812300546 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -11273,8 +11273,6 @@ dependencies = [
 [[package]]
 name = "v8"
 version = "150.4.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "42a978ff11f15b24e5c05a7123cf2b68f41e763546699781a924ef4e2cf43a49"
 dependencies = [
  "bindgen 0.72.1",
  "bitflags 2.9.3",
diff --git a/Cargo.toml b/Cargo.toml
index 2b660a0bfd..9ff6f4b84c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -640,3 +640,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.twox-hash]
 opt-level = 3
+
+[patch.crates-io]
+v8 = { path = "../rusty_v8" }
-- 
2.55.0

