--- a/Cargo.toml	2024-06-11 23:19:13.929010256 -0600
+++ b/Cargo.toml	2024-06-11 23:19:21.258108450 -0600
@@ -207,9 +207,9 @@
 # NB: the `bench` and `release` profiles must remain EXACTLY the same.
 [profile.release]
 codegen-units = 1
-incremental = true
+incremental = false
 lto = true
-opt-level = 'z' # Optimize for size
+opt-level = 3
 
 # Build release with debug symbols: cargo build --profile=release-with-debug
 [profile.release-with-debug]
