From 90578710abd5b90fee06d9f7fd0b3bb6209a6004 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Fri, 24 Jul 2026 20:23:18 +0800
Subject: [PATCH] AOSCOS: use local patched rusty_v8

---
 Cargo.lock | 2 --
 Cargo.toml | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 640923eb0c..c66237bb4e 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -11174,8 +11174,6 @@ dependencies = [
 [[package]]
 name = "v8"
 version = "150.2.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "c7f4e905df70d6c00b95e69c5f0831fd5eb5889b0116ae2b30293578c19cd1bc"
 dependencies = [
  "bindgen 0.72.1",
  "bitflags 2.9.3",
diff --git a/Cargo.toml b/Cargo.toml
index ade71b26c4..4a3b71d99b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -635,3 +635,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.twox-hash]
 opt-level = 3
+
+[patch.crates-io]
+v8 = { path = "../rusty_v8" }
-- 
2.55.0

