From ac1651b777831646a65fc002f883017178429162 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Thu, 25 Jun 2026 23:16:45 +0800
Subject: [PATCH] AOSCOS: use local patched rusty_v8

---
 Cargo.lock | 2 --
 Cargo.toml | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index f17e7b4588..0380247f3e 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -11173,8 +11173,6 @@ dependencies = [
 [[package]]
 name = "v8"
 version = "149.4.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "ebfd8d6919bfb4b627ca778a67f85ca45e5fb442d352947ba093798bdf9b07e0"
 dependencies = [
  "bindgen 0.72.1",
  "bitflags 2.9.3",
diff --git a/Cargo.toml b/Cargo.toml
index 5226def2af..31da04774f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -629,3 +629,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.twox-hash]
 opt-level = 3
+
+[patch.crates-io]
+v8 = { path = "../rusty_v8" }
-- 
2.54.0

