From 084161a50b4c171faf1add4e60f487c2fbda79f3 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Wed, 4 Mar 2026 20:16:58 +0800
Subject: [PATCH 1/2] AOSCOS: use local patched rusty_v8

---
 Cargo.lock | 2 --
 Cargo.toml | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index c7e80d3b9..9baed9a62 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -11173,8 +11173,6 @@ dependencies = [
 [[package]]
 name = "v8"
 version = "146.8.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "67519d234f7bc5a54a71f7fbc52258b0f3a7e9c11f365213d49016a59207ec10"
 dependencies = [
  "bindgen 0.72.1",
  "bitflags 2.9.3",
diff --git a/Cargo.toml b/Cargo.toml
index 4017762d8..894d8376b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -602,3 +602,6 @@ opt-level = 3
 opt-level = 3
 [profile.release.package.twox-hash]
 opt-level = 3
+
+[patch.crates-io]
+v8 = { path = '../rusty_v8' }
-- 
2.53.0

