From 40b97a769118e7071e5ac5349b61bac63e73bef6 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sat, 31 Jan 2026 16:45:26 +0800
Subject: [PATCH 3/3] AOSCOS: disable self-update feature

---
 cli/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cli/Cargo.toml b/cli/Cargo.toml
index db2173ee2..2dd742614 100644
--- a/cli/Cargo.toml
+++ b/cli/Cargo.toml
@@ -32,7 +32,7 @@ harness = false
 path = "./bench/lsp_bench_standalone.rs"
 
 [features]
-default = ["upgrade"]
+default = []
 # A feature that enables heap profiling with dhat on Linux.
 # 1. Compile with `cargo build --profile=release-with-debug --features=dhat-heap`
 # 2. Run the executable. It will output a dhat-heap.json file.
-- 
2.52.0

