From 12c97d4a8466b9bcdabac35e9d7f44744a4f72e0 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Wed, 18 Feb 2026 11:44:43 +0800
Subject: [PATCH 3/3] AOSCOS: disable self-update feature

---
 cli/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cli/Cargo.toml b/cli/Cargo.toml
index 98d816f2b..64561535a 100644
--- a/cli/Cargo.toml
+++ b/cli/Cargo.toml
@@ -24,7 +24,7 @@ path = "integration_tests_runner.rs"
 harness = false
 
 [features]
-default = ["upgrade"]
+default = []
 # A feature that enables heap profiling with dhat on Linux.
 # 1. Compile with `cargo build --profile=release-with-debug --features=dhat-heap`
 # 2. Run the executable. It will output a dhat-heap.json file.
-- 
2.53.0

