From c31c7f20686e6ae967356f0791f2e65a2a5a7de2 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Wed, 10 Feb 2021 03:10:23 -0700
Subject: [PATCH] fsys: resolve symlinks on *nix platforms

---
 gutils/fsys.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gutils/fsys.c b/gutils/fsys.c
index 22eb873f8..13fd9fcd7 100644
--- a/gutils/fsys.c
+++ b/gutils/fsys.c
@@ -533,6 +533,7 @@ char *_GFile_find_program_dir(char *prog) {
 	free(tmppath);
     }
 #else
+    char* buf = NULL;
     if ( (pt = strrchr(prog,'/'))!=NULL )
 	program_dir = copyn(prog,pt-prog);
     else if ( (path = getenv("PATH"))!=NULL ) {
@@ -552,6 +553,10 @@ char *_GFile_find_program_dir(char *prog) {
 		program_dir = copy(path);
 	}
     }
+	if ( (buf = realpath(program_dir, NULL)) ) {
+		if ( program_dir!=NULL ) free(program_dir);
+		program_dir = buf;
+	}
 #endif
 
     if ( program_dir==NULL )
-- 
2.30.0

