Description: Make program compile and work on Linux.
Author: Michael Meskes <meskes@debian.org>
--- netdiag-1.0.orig/tcpblast-20011111/tcpblast.c
+++ netdiag-1.0/tcpblast-20011111/tcpblast.c
@@ -434,7 +434,7 @@ parse_rates(char *spec) {
 
 void
 sndbuf(int f, int setbufsize, char *name) {
-	int	size=sizeof(int); 	/* getsockopt() should know how much space we have */
+	socklen_t size=sizeof(int); 	/* getsockopt() should know how much space we have */
 
 	/* get/setsockopt doesn't return any error really for SO_SNDBUF,
 	   at least on Linux; it limits the buffer to [2048..65536]
@@ -453,7 +453,8 @@ sndbuf(int f, int setbufsize, char *name
 
 void
 setmaxseg(int f, int maxseg, char *name) {
-	int	currseg, size=sizeof(int); 	/* getsockopt() should know how much space we have */
+	int	currseg;
+	socklen_t size=sizeof(int); 	/* getsockopt() should know how much space we have */
 
 	if (getsockopt(f, IPPROTO_TCP, TCP_MAXSEG, (void *)&currseg, &size)==-1)
 		printf("%s getsockopt: %s", name, strerror(errno));
--- netdiag-1.0.orig/tcpblast-20011111/Makefile.in
+++ netdiag-1.0/tcpblast-20011111/Makefile.in
@@ -13,10 +13,10 @@
 CCOPT = $(OPTOPT)
 INCLS = -I.
 DEFS = @DEFS@ -DVERSION=\"@VERSION@\"
-CFLAGS = @CFLAGS@ $(CCOPT) $(DEFS) $(INCLS)
+CFLAGS += @CFLAGS@ $(CCOPT) $(DEFS) $(INCLS)
 # CFLAGS = -g -Wall $(DEFS) $(INCLS)
 STATIC = 
-LDFLAGS = $(OPTOPT) @LDFLAGS@ $(STATIC)
+LDFLAGS += $(OPTOPT) @LDFLAGS@ $(STATIC)
 LIBS =  @LIBS@
 SHTOOL = ./shtool
 INSTALL = $(SHTOOL) install 
@@ -23,7 +23,6 @@
 MAKEDEPEND = @MAKEDEPEND@
 RPMTDIR=$(HOME)/rpmdir
 RH = /usr/src/redhat
-HOME=root@6bone-gw.6bone.pl:~ftp/pub/blast
 
 TARGET = tcpblast
 TARGET2 = discard
@@ -62,7 +61,7 @@
 
 install: all
 	$(SHTOOL) mkdir -f -p -m 755 $(sbindir) $(mandir)/man8
-	$(INSTALL) -s -c -m 755 $(TARGET) $(sbindir)/$(TARGET)
+	$(INSTALL) -c -m 755 $(TARGET) $(sbindir)/$(TARGET)
 	$(INSTALL) -c -m 644 docs/$(TARGET).8 $(mandir)/man8/$(TARGET).8
 	echo Ignore error which may follow.
 	cd $(sbindir); ln -s tcpblast udpblast
--- netdiag-1.0.orig/tcpblast-20011111/docs/tcpblast.8
+++ netdiag-1.0/tcpblast-20011111/docs/tcpblast.8
@@ -1,8 +1,4 @@
 .rn '' }`
-''' $RCSfile$$Revision$$Date$
-'''
-''' $Log$
-'''
 .de Sh
 .br
 .if t .Sp
@@ -31,12 +27,6 @@
 
 .fi
 ..
-'''
-'''
-'''     Set up \*(-- to give an unbreakable dash;
-'''     string Tr holds user defined translation string.
-'''     Bell System Logo is used as a dummy character.
-'''
 .tr \(*W-|\(bv\*(Tr
 .ie n \{\
 .ds -- \(*W-
@@ -45,10 +35,6 @@
 .if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\" diablo 12 pitch
 .ds L" ""
 .ds R" ""
-'''   \*(M", \*(S", \*(N" and \*(T" are the equivalent of
-'''   \*(L" and \*(R", except that they are used on ".xx" lines,
-'''   such as .IP and .SH, which do another additional levels of
-'''   double-quote interpretation
 .ds M" """
 .ds S" """
 .ds N" """""
