From 32d52128340162afb07e0d7b5c14dec5ca079584 Mon Sep 17 00:00:00 2001
From: salieri <maliya355@outlook.com>
Date: Tue, 13 Aug 2024 12:45:26 +0800
Subject: [PATCH] tools/logger/convert: work around implicit sign conversion 
 for mips64

---
 logger/convert.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/logger/convert.c b/logger/convert.c
index 4e0a6cdc5..bbf69411f 100644
--- a/logger/convert.c
+++ b/logger/convert.c
@@ -354,7 +354,7 @@ static inline void print_table_header(void)
 	if (global_config->time_precision >= 0) {
 		const uint8_t ts_width = timestamp_width(global_config->time_precision);
 
-		fprintf(out_fd, "%*s(us)%*s  ", -ts_width, " TIMESTAMP", ts_width, "DELTA");
+		fprintf(out_fd, "%*s(us)%*s  ", -(int)ts_width, " TIMESTAMP", ts_width, "DELTA");
 	}
 
 	fprintf(out_fd, "%2s %-18s ", "C#", "COMPONENT");
-- 
2.34.1

