From 47667a444403a85cebbabf8aeacd905bd949b6e7 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Fri, 19 Jul 2024 12:34:18 +0800
Subject: [PATCH] tools/logger/convert: work around implicit sign conversion
 for mips64

---
 logger/convert.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/logger/convert.c b/logger/convert.c
index e393f828e..5e39ad35c 100644
--- a/logger/convert.c
+++ b/logger/convert.c
@@ -354,7 +354,7 @@ static inline void print_table_header(void)
 	if (global_config->time_precision >= 0) {
 		const unsigned int ts_width = timestamp_width(global_config->time_precision);
 
-		fprintf(out_fd, "%*s(us)%*s  ", -ts_width, " TIMESTAMP", ts_width, "DELTA");
+		fprintf(out_fd, "%*s(us)%*s  ", -(int)ts_width, " TIMESTAMP", ts_width, "DELTA");
 	}
 
 	fprintf(out_fd, "%2s %-18s ", "C#", "COMPONENT");
-- 
2.45.2

