From 6239d9a0d4634d179594199e0e15a686f9108d22 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Thu, 2 Apr 2026 14:26:20 +0000
Subject: [PATCH] tools/ctl: use correct integer type for defined macros

---
 tools/ctl/ctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/ctl/ctl.c b/tools/ctl/ctl.c
index c669349..a06a192 100644
--- a/ctl/ctl.c
+++ b/ctl/ctl.c
@@ -503,7 +503,7 @@ static int ctl_set_get(struct ctl_data *ctl_data)
 		ref_size = ctl_data->buffer[BUFFER_SIZE_OFFSET] + BUFFER_TLV_HEADER_BYTES;
 		if (read_size != ref_size) {
 			fprintf(stderr,
-				"Error: Blob TLV header size %u (plus %lu) does not match with read bytes count %zu.\n",
+				"Error: Blob TLV header size %u (plus %zu) does not match with read bytes count %zu.\n",
 				ctl_data->buffer[BUFFER_SIZE_OFFSET], BUFFER_TLV_HEADER_BYTES,
 				read_size);
 			return -EINVAL;
-- 
2.52.0

