From c12b7c3a6f5643696fe82e0a2dd9d435346463f0 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Thu, 4 Jun 2026 07:54:52 +0000
Subject: [PATCH] logger: make format specifiers platform-agnostic for time
 related vars

---
 logger/convert.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/logger/convert.c b/logger/convert.c
index 4e0a6cd..f38931e 100644
--- a/logger/convert.c
+++ b/logger/convert.c
@@ -10,6 +10,7 @@
 #include <stdbool.h>
 #include <stdlib.h>
 #include <stdint.h>
+#include <inttypes.h>
 #include <string.h>
 #include <errno.h>
 #include <unistd.h>
@@ -366,7 +367,12 @@ static inline void print_table_header(void)
 		struct tm *ltime = localtime(&epoc_secs);
 
 		/* e.g.: ktime=4263.487s @ 2021-04-27 14:21:13 -0700 PDT */
-		fprintf(out_fd, "\tktime=%lu.%03lus",
+		fprintf(out_fd,
+#if defined(_TIME_BITS) && _TIME_BITS == 64
+				"\tktime=%" PRIu64 ".%03lus",
+#else
+				"\tktime=%" PRIuPTR ".%03lus",
+#endif
 			ktime.tv_sec, ktime.tv_nsec / 1000000);
 
 		if (ltime && strftime(date_string, sizeof(date_string),
-- 
2.52.0

