From: Xu Colin <colinxu2020@gmail.com>
Date: Thu, 20 Aug 2026 23:21:36 +0800
Subject: [PATCH] Mark AsyncSafeSignalManager.__current as optional because mypy 2.x does not permits assigning None to non-Optional variable.

---

diff --git a/launch/launch/utilities/signal_management.py b/launch/launch/utilities/signal_management.py
index a078c69..5566cb5 100644
--- a/launch/utilities/signal_management.py
+++ b/launch/utilities/signal_management.py
@@ -60,7 +60,7 @@ class AsyncSafeSignalManager:
         is not broken by third-party code or by asyncio itself in some platforms.
     """

-    __current = None  # type: AsyncSafeSignalManager
+    __current = None  # type: Optional[AsyncSafeSignalManager]

     __set_wakeup_fd = signal.set_wakeup_fd  # type: Callable[[int], int]
