diff -up cdrdao-1.2.3/dao/cdrdao.man.helpmansync cdrdao-1.2.3/dao/cdrdao.man
--- cdrdao-1.2.3/dao/cdrdao.man.helpmansync	2012-10-08 15:54:01.598543254 +0200
+++ cdrdao-1.2.3/dao/cdrdao.man	2012-10-08 16:01:11.439957988 +0200
@@ -3,7 +3,7 @@
 cdrdao \- reads and writes CDs in disc-at-once mode
 .SH SYNOPSIS
 .B cdrdao
-.RB { show-toc|read-toc|read-cd|read-cddb|show-data|read-test|disk-info|msinfo|unlock|simulate|write|copy|blank }
+.RB { show-toc|toc-info|toc-size|read-toc|read-cd|read-cddb|show-data|read-test|disk-info|discid|msinfo|drive-info|unlock|scanbus|simulate|write|copy|blank }
 .RB [ --device
 .IR device ]
 .RB [ --source-device
@@ -77,6 +77,12 @@ The first argument must be one of the fo
 .BI show-toc
 Print out a summary about what will be written to the CD-R.  
 .TP
+.BI toc-info
+Prints out short toc-file summary.
+.TP
+.BI toc-size
+Prints total number of blocks for toc.
+.TP
 .BI read-toc
 Analyze each track of the inserted CD and create a
 .I toc-file
@@ -122,10 +128,16 @@ Shows information about the inserted CD-
 it will also print the start of the last and current session which is
 used by mkisofs to create an image for a second or higher session.
 .TP
+.BI discid
+Prints out CDDB information.
+.TP
 .BI msinfo
 Shows information required for creating multi session disks with
 mkisofs. The output is meant for processing by scripts.
 .TP
+.BI drive-info
+Shows drive information.
+.TP
 .BI unlock
 Tries to unlock the recorder device after a failed write or simulation
 run. If you cannot eject the CD after a cdrdao run try this command.
@@ -139,6 +151,9 @@ blanking operation. Use option
 .BI --speed
 to select another blanking speed.
 .TP
+.BI scanbus
+Scan for devices.
+.TP
 .BI simulate
 Like
 .BI write
