From 39a89ffd68dee692bd1cd58e972ebd3ec099e486 Mon Sep 17 00:00:00 2001
From: Martin Rodriguez Reboredo <yakoyoku@gmail.com>
Date: Sun, 19 May 2024 23:23:45 -0300
Subject: [PATCH 6/7] FROM UPSTREAM: Gui: Fix NaviCube for Qt6

---
 src/Gui/NaviCube.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Gui/NaviCube.cpp b/src/Gui/NaviCube.cpp
index 454c8b55f185..a9abe94409e1 100644
--- a/src/Gui/NaviCube.cpp
+++ b/src/Gui/NaviCube.cpp
@@ -907,6 +907,7 @@ NaviCubeImplementation::PickId NaviCubeImplementation::pickFace(short x, short y
     GLubyte pixels[4] = {0};
     if (m_PickingFramebuffer && std::abs(x) <= m_CubeWidgetSize / 2 &&
         std::abs(y) <= m_CubeWidgetSize / 2) {
+        static_cast<QtGLWidget*>(m_View3DInventorViewer->viewport())->makeCurrent();
         m_PickingFramebuffer->bind();
 
         glViewport(0, 0, m_CubeWidgetSize * 2, m_CubeWidgetSize * 2);
@@ -917,6 +918,7 @@ NaviCubeImplementation::PickId NaviCubeImplementation::pickFace(short x, short y
         glReadPixels(2 * x + m_CubeWidgetSize, 2 * y + m_CubeWidgetSize, 1, 1,
                      GL_RGBA, GL_UNSIGNED_BYTE, &pixels);
         m_PickingFramebuffer->release();
+        static_cast<QtGLWidget*>(m_View3DInventorViewer->viewport())->doneCurrent();
     }
     return pixels[3] == 255 ? static_cast<PickId>(pixels[0]) : PickId::None;
 }
-- 
2.46.0

