diff -Naur rpm-4.17.0/configure.ac rpm-4.17.0.lua53/configure.ac
--- rpm-4.17.0/configure.ac	2021-09-03 09:42:16.790237970 +0000
+++ rpm-4.17.0.lua53/configure.ac	2021-12-30 10:00:54.250969534 +0000
@@ -746,7 +746,7 @@
 AM_CONDITIONAL(ACL,[test "$with_acl" = yes])
 
 PKG_CHECK_MODULES([LUA],
-    [lua >= 5.3],
+    [lua5.3 >= 5.3],
     [],
     [AC_MSG_ERROR([lua not present or too old)])])
 AC_SUBST(LUA_CFLAGS)
