diff -Naur rpm-4.16.0/configure.ac rpm-4.16.0.lua53/configure.ac
--- rpm-4.16.0/configure.ac	2020-09-30 00:48:01.214567716 -0700
+++ rpm-4.16.0.lua53/configure.ac	2020-12-08 23:14:12.902323058 -0800
@@ -830,7 +830,7 @@
 
 AS_IF([test "$with_lua" != no],[
   PKG_CHECK_MODULES([LUA],
-    [lua >= 5.2],
+    [lua5.3 >= 5.2],
     [AC_DEFINE(WITH_LUA, 1, [Build with lua support?])],
     [AC_MSG_ERROR([lua not present (--without-lua to disable)])])
   AC_SUBST(LUA_CFLAGS)
