--- a/configure.ac	2024-04-23 20:34:08.039900414 +0800
+++ b/configure.ac	2024-04-23 20:34:16.276052006 +0800
@@ -45,10 +45,10 @@
 # Default CFLAGS, CXXFLAGS for GCC in case they were not set by the user.
 if test "$GCC" = "yes"; then
   if test "$ac_test_CFLAGS" != "set"; then
-    CFLAGS="-W -Wall -g -O3 -fomit-frame-pointer -D_FORTIFY_SOURCE=2 $NUFRAW_MARCH"
+    CFLAGS="-W -Wall -g -O3 -fomit-frame-pointer -D_FORTIFY_SOURCE=2 $NUFRAW_MARCH $CFLAGS"
   fi
   if test "$ac_test_CXXFLAGS" != "set"; then
-    CXXFLAGS="-W -Wall -g -O3 -fomit-frame-pointer -D_FORTIFY_SOURCE=2 $NUFRAW_MARCH"
+    CXXFLAGS="-W -Wall -g -O3 -fomit-frame-pointer -D_FORTIFY_SOURCE=2 $NUFRAW_MARCH $CXXFLAGS"
   fi
 fi
 
