From cc8b8db3b7493324d6a436a31c7db78bb0f99ad3 Mon Sep 17 00:00:00 2001
From: LaserEyess <lasereyess@lasereyess.net>
Date: Sun, 11 Nov 2018 14:37:13 -0500
Subject: [PATCH] Parse headers case insensitively

---
 src/trg-client.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/trg-client.c b/src/trg-client.c
index 6b47d7a..9e92368 100644
--- a/src/trg-client.c
+++ b/src/trg-client.c
@@ -478,7 +478,8 @@ header_callback(void *ptr, size_t size, size_t nmemb, void *data)
     TrgClient *tc = TRG_CLIENT(data);
     gchar *session_id;
 
-    if (g_str_has_prefix(header, X_TRANSMISSION_SESSION_ID_HEADER_PREFIX)) {
+    if (g_ascii_strncasecmp(header, X_TRANSMISSION_SESSION_ID_HEADER_PREFIX,
+            strlen(X_TRANSMISSION_SESSION_ID_HEADER_PREFIX)) == 0) {
         char *nl;
 
         session_id = g_strdup(header);
