--- a/org.kde.skanpage.desktop.orig	2024-03-11 19:31:01.258446227 -0700
+++ b/org.kde.skanpage.desktop	2024-03-11 19:31:11.414780998 -0700
@@ -2,7 +2,7 @@
 #
 # SPDX-License-Identifier: CC0-1.0
 [Desktop Entry]
-Categories=Qt;KDE;Graphics;Scanning;
+Categories=Qt;KDE;Office;Scanning;
 Comment=Scan and save images
 Comment[ar]=امسح الصّور ضوئيًّا واحفظها
 Comment[bg]=Сканиране и запазване на изображения
