diff --git a/build/env.cmake b/build/env.cmake
index 5c5b6e4b..beff5e7f 100644
--- a/build/env.cmake
+++ b/build/env.cmake
@@ -780,7 +780,7 @@ if ( SINGLE_CONFIG )
                 ${VDB_INCDIR}/kfg/ncbi              \
                 '${VDB_COPY_DIR}'                   \
                 ${CMAKE_INSTALL_PREFIX}/bin/ncbi    \
-                /etc/ncbi                           \
+		${CMAKE_CURRENT_SOURCE_DIR}/etc/ncbi\
                 ${CMAKE_INSTALL_BINDIR}             \
                 ${CMAKE_INSTALL_LIBDIR}             \
                 ${CMAKE_SOURCE_DIR}/shared/kfgsums  \
diff --git a/build/install.sh b/build/install.sh
index 39947280..fe9f0391 100755
--- a/build/install.sh
+++ b/build/install.sh
@@ -49,7 +49,7 @@ if [ "$EUID" -eq 0 ]; then
         $SCRIPT_DIR/install-kfg.sh vdb-copy.kfg $SRC_DIR_COPY $KONFIG_DIR_ROOT $KFGSUMS_FILE
     fi
 
-PROFILE_FILE=/etc/profile.d/sra-tools
+PROFILE_FILE=${SRD_DIR}/etc/profile.d/sra-tools
 
     echo "Updating ${PROFILE_FILE}.sh"
     printf \
