Author: Aaron M. Ucko
Last-Update: 2023-01-20 17:48:36
Description: Fix linking target

Index: b/ngs/ngs-sdk/CMakeLists.txt
===================================================================
--- a/ngs/ngs-sdk/CMakeLists.txt
+++ b/ngs/ngs-sdk/CMakeLists.txt
@@ -52,6 +52,7 @@ endif()
 
 # Merging NGS_SDK_TARGETS with NCBI_NGS_TARGETS for ngs-sdk lib to contain all necessary functions for python and java to import (VDB-4713)
 set(NGS_SDK_TARGETS ${NGS_SDK_TARGETS} ${NCBI_NGS_TARGETS})
+set(NGS_CPP_TARGETS ${NGS_CPP_TARGETS} ${NCBI_NGS_TARGETS})
 
 if( SINGLE_CONFIG OR WIN32 )
     include( CMakeLists.posix.txt )
