diff -Naur encfs-1.9.5/CMakeLists.txt encfs-1.9.5.staticinternal/CMakeLists.txt
--- encfs-1.9.5/CMakeLists.txt	2018-04-27 03:52:22.000000000 -0500
+++ encfs-1.9.5.staticinternal/CMakeLists.txt	2019-03-12 12:00:47.696474130 -0500
@@ -226,10 +226,6 @@
   message("-- Using vendored TinyXML2")
   set(TINYXML_DIR vendor/github.com/leethomason/tinyxml2)
   if (BUILD_SHARED_LIBS)
-    set(BUILD_STATIC_LIBS OFF CACHE BOOL "build static libs")
-    set(BUILD_SHARED_LIBS ON CACHE BOOL "build shared libs")
-    set(TINYXML_LIBRARIES tinyxml2)
-  else ()
     set(BUILD_STATIC_LIBS ON CACHE BOOL "build static libs")
     set(BUILD_SHARED_LIBS OFF CACHE BOOL "build shared libs")
     set(TINYXML_LIBRARIES tinyxml2_static)
