diff -Naur tegaki-wagomu-0.3.1-old/tegakiwagomu.py tegaki-wagomu-0.3.1/tegakiwagomu.py
--- tegaki-wagomu-0.3.1-old/tegakiwagomu.py	2018-05-11 14:04:24.162188549 +0800
+++ tegaki-wagomu-0.3.1/tegakiwagomu.py	2018-05-11 14:08:37.790764311 +0800
@@ -25,7 +25,7 @@
 import struct
 from array import array
 
-from tegaki.recognizer import Results, Recognizer, RecognizerError
+from tegaki.recognizer import Recognizer, RecognizerError
 from tegaki.trainer import Trainer, TrainerError
 from tegaki.arrayutils import array_flatten
 from tegaki.dictutils import SortedDict
@@ -231,7 +231,7 @@
             if not ret: 
                 raise RecognizerError, self._recognizer.get_error_message()
 
-        def _recognize(self, writing, n=10):
+        def recognize(self, writing, n=10):
             n_strokes = writing.get_n_strokes()
             feat = self.get_features(writing)
             nfeat = len(feat) 
@@ -248,7 +248,7 @@
                 utf8 = unichr(res.get_unicode(i)).encode("utf8")
                 candidates.append((utf8, res.get_distance(i)))
 
-            return Results(candidates)
+            return candidates
 
     RECOGNIZER_CLASS = WagomuRecognizer
 
