From 68370600530b6514877793193c1b0ab9a6f46acc Mon Sep 17 00:00:00 2001
From: Mike FABIAN <mfabian@redhat.com>
Date: Mon, 18 May 2015 07:03:13 +0200
Subject: [PATCH 1/5] Remove duplicate transliterations for U+0152 and U+0153
 from C-translit.h.in

---
 locale/C-translit.h.in | 2 --
 1 file changed, 2 deletions(-)

diff --git a/locale/C-translit.h.in b/locale/C-translit.h.in
index 71bef64..7aed945 100644
--- a/locale/C-translit.h.in
+++ b/locale/C-translit.h.in
@@ -38,8 +38,6 @@
 "\x0133"	"ij"	/* <U0133> LATIN SMALL LIGATURE IJ */
 "\x0149"	"'n"	/* <U0149> LATIN SMALL LETTER N PRECEDED BY APOSTROPHE */
 "\x0152"	"OE"	/* <U0152> LATIN CAPITAL LIGATURE OE */
-"\x0152"	"OE"	/* <U0152> LATIN CAPITAL LIGATURE OE */
-"\x0153"	"oe"	/* <U0153> LATIN SMALL LIGATURE OE */
 "\x0153"	"oe"	/* <U0153> LATIN SMALL LIGATURE OE */
 "\x017f"	"s"	/* <U017F> LATIN SMALL LETTER LONG S */
 "\x01c7"	"LJ"	/* <U01C7> LATIN CAPITAL LETTER LJ */
-- 
2.4.3


