From 5fe3b68742a63022a80a69e1a72f4697cd752ee6 Mon Sep 17 00:00:00 2001
From: stydxm <stydxm@outlook.com>
Date: Tue, 2 Sep 2025 12:11:05 +0800
Subject: [PATCH 2/2] Update glog import logic for glog 0.7.1

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 02f0710ba..b57a1483e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,7 +56,7 @@ set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake"
 
 find_package(Gflags REQUIRED)
 
-find_package(Glog MODULE REQUIRED)
+find_package(glog CONFIG QUIET)
 
 find_package(folly CONFIG REQUIRED)
 
-- 
2.52.0

