diff -Naur flightgear-2017.3.1/3rdparty/sqlite3/CMakeLists.txt flightgear-2017.3.1.pic/3rdparty/sqlite3/CMakeLists.txt
--- flightgear-2017.3.1/3rdparty/sqlite3/CMakeLists.txt 2017-09-20 07:51:30.000000000 +0000
+++ flightgear-2017.3.1.pic/3rdparty/sqlite3/CMakeLists.txt     2018-03-02 16:59:05.651186835 +0000
@@ -2,4 +2,4 @@
 add_definitions(-DSQLITE_OMIT_LOAD_EXTENSION)
 add_definitions(-DNDEBUG)
 add_library(fgsqlite3 sqlite3.c)
-set_target_properties(fgsqlite3 PROPERTIES COMPILE_FLAGS "-fpic -fno-fast-math")
+set_target_properties(fgsqlite3 PROPERTIES COMPILE_FLAGS "-fPIC -fno-fast-math")
