diff -Naur mozjs17.0.0-old/mfbt/double-conversion/utils.h mozjs17.0.0/mfbt/double-conversion/utils.h
--- mozjs17.0.0-old/mfbt/double-conversion/utils.h	2018-02-27 13:19:26.000000000 +0800
+++ mozjs17.0.0/mfbt/double-conversion/utils.h	2018-02-27 13:25:30.000000000 +0800
@@ -58,7 +58,7 @@
     defined(__mips__) || defined(__powerpc__) || \
     defined(__sparc__) || defined(__sparc) || defined(__s390__) || \
     defined(__SH4__) || defined(__alpha__) || \
-    defined(__aarch64__) || \
+    defined(__aarch64__) || defined(__riscv) || \
     defined(_MIPS_ARCH_MIPS32R2)
 #define DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS 1
 #elif defined(_M_IX86) || defined(__i386__) || defined(__i386)
