--- a/js/src/jsval.h.old        2015-06-19 14:38:47.000000000 -0400
+++ b/js/src/jsval.h    2015-06-19 17:28:00.653643399 -0400
@@ -188,8 +188,14 @@ typedef uint64_t JSValueShiftedTag;

 #elif JS_BITS_PER_WORD == 64

+#if defined(__aarch64__)
+#define JSVAL_PAYLOAD_MASK           0x0000FFFFFFFFFFFFLL
+#define JSVAL_TAG_MASK               0xFFFF000000000000LL
+#else
 #define JSVAL_PAYLOAD_MASK           0x00007FFFFFFFFFFFLL
 #define JSVAL_TAG_MASK               0xFFFF800000000000LL
+#endif
+
 #define JSVAL_TYPE_TO_TAG(type)
((JSValueTag)(JSVAL_TAG_MAX_DOUBLE | (type)))
 #define JSVAL_TYPE_TO_SHIFTED_TAG(type)
(((uint64_t)JSVAL_TYPE_TO_TAG(type)) << JSVAL_TAG_SHIFT)