From: Hideki Yamane (Debian-JP) <henrich@debian.or.jp>
Subject: Enable Japanese translations

--- a/configure.ac	2014-09-06 21:57:31.815538838 +0200
+++ b/configure.ac	2014-09-06 21:57:44.635393843 +0200
@@ -44,7 +44,7 @@
 PKG_CHECK_MODULES(GKSU_PROPERTIES, [gtk+-2.0 >= 2.12, gconf-2.0])
 
 # Checks for library functions.
-ALL_LINGUAS="ca cs da de es eu fr hu it ko lt pl pt_BR ro ru sk sv nb nl th zh_CN"
+ALL_LINGUAS="ca cs da de es eu fr hu it ja ko lt pl pt_BR ro ru sk sv nb nl th zh_CN"
 
 AC_DEFINE(GETTEXT_PACKAGE, AC_PACKAGE_NAME, [gettext package])
 GETTEXT_PACKAGE=AC_PACKAGE_NAME
