From: Helmut Grohne <helmut@subdivi.de>
Subject: fix cross compilation by using the cross tools discovered by configure

Index: libgksu-2.0.13~pre1/libgksuui/Makefile.am
===================================================================
--- libgksu-2.0.13~pre1.orig/libgksuui/Makefile.am
+++ libgksu-2.0.13~pre1/libgksuui/Makefile.am
@@ -1,10 +1,10 @@
 AM_CFLAGS = -g -O2 -Wall
-INCLUDES = `pkg-config --cflags gtk+-2.0 gconf-2.0`
+INCLUDES = `$(PKG_CONFIG) --cflags gtk+-2.0 gconf-2.0`
 AM_CPPFLAGS = -DLOCALEDIR=\"$(datadir)/locale\" -DDATA_DIR=\"$(datadir)\" -DPREFIX=\"$(prefix)\"
 
 noinst_LTLIBRARIES = libgksuui1.0.la
 libgksuui1_0_la_SOURCES = gksuui-dialog.c
-libgksuui1_0_la_LDFLAGS = -Wl,-O1 `pkg-config --libs gtk+-2.0 gconf-2.0 x11`
+libgksuui1_0_la_LDFLAGS = -Wl,-O1 `$(PKG_CONFIG) --libs gtk+-2.0 gconf-2.0 x11`
 
 noinst_HEADERS = defines.h gksuui.h gksuui-dialog.h
 includedir = ${prefix}/include/$(PACKAGE)
@@ -12,4 +12,4 @@
 noinst_PROGRAMS = test-gksuui
 test_gksuui_SOURCES = test-gksuui.c
 test_gksuui_LDADD = libgksuui1.0.la
-test_gksuui_LDFLAGS = `pkg-config --libs glib-2.0 x11`
+test_gksuui_LDFLAGS = `$(PKG_CONFIG) --libs glib-2.0 x11`
Index: libgksu-2.0.13~pre1/libgksu/Makefile.am
===================================================================
--- libgksu-2.0.13~pre1.orig/libgksu/Makefile.am
+++ libgksu-2.0.13~pre1/libgksu/Makefile.am
@@ -28,12 +28,12 @@
 
 pkglibdir = ${libdir}/${PACKAGE}
 pkglibexec_PROGRAMS = gksu-run-helper
-gksu_run_helper_LDFLAGS = `pkg-config --libs glib-2.0`
+gksu_run_helper_LDFLAGS = `$(PKG_CONFIG) --libs glib-2.0`
 gksu_run_helper_SOURCES = gksu-run-helper.c
 
 noinst_PROGRAMS = test-gksu
 test_gksu_SOURCES = test-gksu.c
 test_gksu_LDADD = libgksu2.la
-test_gksu_LDFLAGS = `pkg-config --libs glib-2.0 gtk+-2.0`
+test_gksu_LDFLAGS = `$(PKG_CONFIG) --libs glib-2.0 gtk+-2.0`
 
 EXTRA_DIST = libgksu.ver
