From f4a1d6be9351559bb31d42fb56cff2fe929ff86e Mon Sep 17 00:00:00 2001
From: Mag Mell <sakiiily@aosc.io>
Date: Mon, 14 Sep 2020 06:25:08 +0800
Subject: [PATCH] Fix QT 5.14 build

---
 src/migrator/lib/migratorfactory.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/migrator/lib/migratorfactory.cpp b/src/migrator/lib/migratorfactory.cpp
index 5686320..d7f08d3 100644
--- a/src/migrator/lib/migratorfactory.cpp
+++ b/src/migrator/lib/migratorfactory.cpp
@@ -11,6 +11,8 @@
 #include <QDir>
 #include <QFileInfo>
 #include <QPluginLoader>
+#include <QVariant>
+#include <QSet>
 #include <fcitx-utils/standardpath.h>
 
 namespace fcitx {
-- 
2.27.0

