--- a/CMakeLists.txt	2017-06-01 12:06:24.000000000 -0600
+++ b/CMakeLists.txt	2017-12-26 19:06:51.042314377 -0700
@@ -222,6 +222,8 @@
 target_link_libraries           (libtesseract ${LIB_Ws2_32} ${LIB_pthread})
 set_target_properties           (libtesseract PROPERTIES VERSION ${VERSION_MAJOR}.${VERSION_MINOR_0}.${VERSION_MINOR_1})
 set_target_properties           (libtesseract PROPERTIES SOVERSION ${VERSION_MAJOR}.${VERSION_MINOR_0}.${VERSION_MINOR_1})
+set_target_properties           (libtesseract PROPERTIES OUTPUT_NAME tesseract.${VERSION_MAJOR}.${VERSION_MINOR})
+
 if (WIN32)
 set_target_properties           (libtesseract PROPERTIES OUTPUT_NAME tesseract${VERSION_MAJOR}${VERSION_MINOR})
 set_target_properties           (libtesseract PROPERTIES DEBUG_OUTPUT_NAME tesseract${VERSION_MAJOR}${VERSION_MINOR}d)
