From 5608a60b7dc14d8d0a63c10dd0952ae15072c396 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Tue, 21 May 2024 10:49:06 +0800
Subject: [PATCH 01/16] [Fedora] Fix build on NEON-enabled ARM targets

---
 gfx/ycbcr/moz.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gfx/ycbcr/moz.build b/gfx/ycbcr/moz.build
index 3c1a96409c..21b0880c1e 100644
--- a/gfx/ycbcr/moz.build
+++ b/gfx/ycbcr/moz.build
@@ -53,7 +53,7 @@ else:
         'yuv_row_other.cpp',
     ]
 
-if CONFIG['TARGET_CPU'] == 'arm' and CONFIG['HAVE_ARM_NEON']:
+if CONFIG['HAVE_ARM_NEON']:
     SOURCES += [
         'yuv_row_arm.s',
     ]
-- 
2.46.0.rc1.windows.1

