From 75e9441c58f4d26d4fa77fbeff6abadf9632d0fe Mon Sep 17 00:00:00 2001
From: Han Gao <rabenda.cn@gmail.com>
Date: Wed, 1 Oct 2025 21:48:27 +0800
Subject: [PATCH 2/4] Enable GLES on riscv64.

---
 toolkit/xre/glxtest/glxtest.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolkit/xre/glxtest/glxtest.cpp b/toolkit/xre/glxtest/glxtest.cpp
index 4adbcc888b3bc1e075b3e950345a8da54dd5d955..10b7c63a7af602246b622749ba89f8896257e713 100644
--- a/toolkit/xre/glxtest/glxtest.cpp
+++ b/toolkit/xre/glxtest/glxtest.cpp
@@ -433,7 +433,7 @@ static bool get_egl_gl_status(EGLDisplay dpy,
   PFNGLGETSTRING glGetString =
       cast<PFNGLGETSTRING>(eglGetProcAddress("glGetString"));
 
-#if defined(__arm__) || defined(__aarch64__)
+#if defined(__arm__) || defined(__aarch64__) || (defined(__riscv) && __riscv_xlen == 64)
   bool useGles = true;
 #else
   bool useGles = false;
-- 
2.54.0

