From aba4f6b082ead80e8ce3e9c1a0524a671fbd91e0 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Tue, 21 May 2024 10:54:01 +0800
Subject: [PATCH 04/16] Mozilla 526293: show remote shares in file selector

---
 widget/gtk/nsFilePicker.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/widget/gtk/nsFilePicker.cpp b/widget/gtk/nsFilePicker.cpp
index 751da3dc6b..e7b5f6dbe5 100644
--- a/widget/gtk/nsFilePicker.cpp
+++ b/widget/gtk/nsFilePicker.cpp
@@ -442,9 +442,7 @@ nsFilePicker::Open(nsIFilePickerShownCallback* aCallback) {
   // If we have --enable-proxy-bypass-protection, then don't allow
   // remote URLs to be used.
 #ifndef MOZ_PROXY_BYPASS_PROTECTION
-  if (mAllowURLs) {
-    gtk_file_chooser_set_local_only(GTK_FILE_CHOOSER(file_chooser), FALSE);
-  }
+  gtk_file_chooser_set_local_only(GTK_FILE_CHOOSER(file_chooser), FALSE);
 #endif
 
   if (action == GTK_FILE_CHOOSER_ACTION_OPEN ||
-- 
2.46.0.rc1.windows.1

