diff -up firefox-97.0/toolkit/moz.configure.disable-elfhack firefox-97.0/toolkit/moz.configure
--- firefox-97.0/toolkit/moz.configure.disable-elfhack	2022-02-08 09:58:47.518047952 +0100
+++ firefox-97.0/toolkit/moz.configure	2022-02-08 10:17:49.552945956 +0100
@@ -1273,7 +1273,7 @@ with only_when("--enable-compile-environ
             help="{Enable|Disable} elf hacks",
         )
 
-        set_config("USE_ELF_HACK", depends_if("--enable-elf-hack")(lambda _: True))
+        set_config("USE_ELF_HACK", depends_if("--enable-elf-hack")(lambda _: False))
 
 
 @depends(build_environment)
