--- a/third_party/libwebrtc/moz.build   2023-04-17 16:31:46.251131174 -0700
+++ b/third_party/libwebrtc/moz.build   2023-04-17 16:33:03.683124514 -0700
@@ -519,6 +519,9 @@
     DIRS += [
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_decoder_isac_float_gn",
         "/third_party/libwebrtc/api/audio_codecs/isac/audio_encoder_isac_float_gn",
+        "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_generic_gn",
+        "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn",
+        "/third_party/libwebrtc/modules/desktop_capture/primitives_gn",
         "/third_party/libwebrtc/modules/audio_coding/isac_c_gn",
         "/third_party/libwebrtc/modules/audio_coding/isac_gn"
     ]
