From 1f13fb1a05bb96547e80fdf428c4a95ddf43bd38 Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Sat, 27 Sep 2025 18:50:55 +0800
Subject: [PATCH 1/4] fix-submodules

---
 .gitmodules | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/.gitmodules b/.gitmodules
index cfcef0a..5b2f6db 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,10 +1,8 @@
 [submodule "core/Clash.Meta"]
 	path = core/Clash.Meta
-	url = git@github.com:chen08209/Clash.Meta.git
+	url = https://github.com/chen08209/Clash.Meta.git
 	branch = FlClash
 [submodule "plugins/flutter_distributor"]
 	path = plugins/flutter_distributor
-	url = git@github.com:chen08209/flutter_distributor.git
+	url = https://github.com/chen08209/flutter_distributor.git
 	branch = FlClash
-
-
-- 
2.51.0

