From 2c4415bd298fdfa49de0a8aa5faa422366877290 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Tue, 19 Sep 2023 18:27:36 -0700
Subject: [PATCH] load 32-bit library /opt/32/lib for AOSC OS

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 bin/mangohud.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/bin/mangohud.in b/bin/mangohud.in
index f975224..8b957a5 100755
--- a/bin/mangohud.in
+++ b/bin/mangohud.in
@@ -9,6 +9,7 @@ if [ "$#" -eq 0 ]; then
 fi
 
 MANGOHUD_LIB_NAME="@ld_libdir_mangohud@libMangoHud.so"
+MANGOHUD_32_LIB_NAME="/opt/32/lib/libMangoHud.so"
 
 if [ "$1" = "--dlsym" ]; then
 	MANGOHUD_DLSYM=1
@@ -18,6 +19,6 @@ fi
 
 # Preload using the plain filenames of the libs, the dynamic linker will
 # figure out whether the 32 or 64 bit version should be used
-LD_PRELOAD="${LD_PRELOAD:+$LD_PRELOAD:}${MANGOHUD_LIB_NAME}"
+LD_PRELOAD="${LD_PRELOAD:+$LD_PRELOAD:}${MANGOHUD_LIB_NAME}:${MANGOHUD_32_LIB_NAME}"
 
 exec env MANGOHUD=1 LD_PRELOAD="${LD_PRELOAD}" "$@"
-- 
2.39.1

