From d47f7fcc07877ed54b03ff0d951826359de10edd Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <origincode@aosc.io>
Date: Thu, 10 Nov 2022 13:30:41 -0800
Subject: [PATCH] load 32-bit library /opt/32/lib for AOSC OS

Signed-off-by: Kaiyang Wu <origincode@aosc.io>
---
 bin/mangohud.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/mangohud.in b/bin/mangohud.in
index e13da99..560b3a2 100755
--- a/bin/mangohud.in
+++ b/bin/mangohud.in
@@ -23,6 +23,6 @@ fi
 # figure out whether the 32 or 64 bit version should be used, and will search
 # for it in the correct directory
 LD_PRELOAD="${LD_PRELOAD}${LD_PRELOAD:+:}${MANGOHUD_LIB_NAME}"
-LD_LIBRARY_PATH="${LD_LIBRARY_PATH}${LD_LIBRARY_PATH:+:}@ld_libdir_mangohud@"
+LD_LIBRARY_PATH="${LD_LIBRARY_PATH}${LD_LIBRARY_PATH:+:}/usr/lib/mangohud:/opt/32/lib/mangohud"
 
 exec env MANGOHUD=1 LD_LIBRARY_PATH="${LD_LIBRARY_PATH}" LD_PRELOAD="${LD_PRELOAD}" "$@"
-- 
2.37.1

