From 3f8f036ee8773ae1af23dd0848b6ab487b5ac7de Mon Sep 17 00:00:00 2001
From: FlightlessMango <flightlessmangoyt@gmail.com>
Date: Sun, 7 May 2023 23:01:10 +0200
Subject: [PATCH] params: add cstdint header

---
 src/overlay_params.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/overlay_params.h b/src/overlay_params.h
index 8fa2485..0878dca 100644
--- a/src/overlay_params.h
+++ b/src/overlay_params.h
@@ -5,6 +5,7 @@
 #include <string>
 #include <vector>
 #include <unordered_map>
+#include <cstdint>
 
 #ifdef __cplusplus
 extern "C" {
-- 
2.39.1

