From 4c81c318e3e751b0c27fedd16576a24f08fd8e37 Mon Sep 17 00:00:00 2001
From: Cyan <cyan@cyano.uk>
Date: Fri, 16 May 2025 10:33:37 +0800
Subject: [PATCH] acinclude: Prepend '-L' to the libdir returned by pg_config

---
 acinclude.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/acinclude.m4 b/acinclude.m4
index 8e89af8..119d377 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -191,7 +191,7 @@ if test "$ac_pgsql" = "yes"; then
 		PGSQL_INCLUDE=-I$ac_pgsql_incdir
 	fi
 	if test "$ac_pgsql_libdir" = "no"; then
-		PGSQL_LDFLAGS=`pg_config --libdir`
+		PGSQL_LDFLAGS="-L"`pg_config --libdir`
 	else
 		PGSQL_LDFLAGS=-L$ac_pgsql_libdir
 	fi
-- 
2.49.0

