From 26ad070f061ed0e2ebb015382c8048b346c29b56 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Mon, 4 May 2026 23:56:18 +0800
Subject: [PATCH] patching config to apply best practices

---
 ssh_config  | 3 ++-
 sshd_config | 6 ++++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/ssh_config b/ssh_config
index 238a0c5e3..24dfd0f56 100644
--- a/ssh_config
+++ b/ssh_config
@@ -17,7 +17,7 @@
 # list of available options, their meanings and defaults, please see the
 # ssh_config(5) man page.
 
-# Host *
+Host *
 #   ForwardAgent no
 #   ForwardX11 no
 #   PasswordAuthentication yes
@@ -43,3 +43,4 @@
 #   ProxyCommand ssh -q -W %h:%p gateway.example.com
 #   RekeyLimit 1G 1h
 #   UserKnownHostsFile ~/.ssh/known_hosts.d/%k
+Include /etc/ssh/ssh_config.d/*
diff --git a/sshd_config b/sshd_config
index 0f4a3a724..4d8cba58a 100644
--- a/sshd_config
+++ b/sshd_config
@@ -81,7 +81,7 @@ AuthorizedKeysFile	.ssh/authorized_keys
 # If you just want the PAM account and session checks to run without
 # PAM authentication, then enable this but set PasswordAuthentication
 # and KbdInteractiveAuthentication to 'no'.
-#UsePAM no
+UsePAM yes
 
 #AllowAgentForwarding yes
 #AllowTcpForwarding yes
@@ -90,7 +90,7 @@ AuthorizedKeysFile	.ssh/authorized_keys
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PermitTTY yes
-#PrintMotd yes
+PrintMotd no # PAM does that
 #PrintLastLog yes
 #TCPKeepAlive yes
 #PermitUserEnvironment no
@@ -116,3 +116,5 @@ Subsystem	sftp	/usr/libexec/sftp-server
 #	AllowTcpForwarding no
 #	PermitTTY no
 #	ForceCommand cvs server
+
+Include /etc/ssh/sshd_config.d/*
-- 
2.52.0

