diff --git a/configure.ac b/configure.ac
index 379cd746b..c0f7aeb69 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1013,6 +1013,9 @@ int main(void) { if (NSVersionOfRunTimeLibrary("System") >= (60 << 16))
 	riscv64-*)
 		seccomp_audit_arch=AUDIT_ARCH_RISCV64
 		;;
+	loongarch64-*)
+		seccomp_audit_arch=AUDIT_ARCH_LOONGARCH64
+		;;
 	esac
 	if test "x$seccomp_audit_arch" != "x" ; then
 		AC_MSG_RESULT(["$seccomp_audit_arch"])
