--- a/python/mozbuild/mozbuild/action/check_binary.py	2021-12-10 10:04:00.432682212 +0000
+++ b/python/mozbuild/mozbuild/action/check_binary.py	2021-12-10 10:04:18.154828362 +0000
@@ -27,7 +27,7 @@
 
 TARGET = {
     "platform": buildconfig.substs["OS_TARGET"],
-    "readelf": "{}readelf".format(buildconfig.substs.get("TOOLCHAIN_PREFIX", "")),
+    "readelf": "{}readelf".format(buildconfig.substs.get("TOOLCHAIN_PREFIX", "llvm")),
 }
 
 ADDR_RE = re.compile(r"[0-9a-f]{8,16}")
