diff -up mozilla/intl/icu/source/i18n/digitlst.cpp.xlocale mozilla/intl/icu/source/i18n/digitlst.cpp
--- mozilla/intl/icu/source/i18n/digitlst.cpp.xlocale	2017-08-24 14:42:48.634084293 +0200
+++ mozilla/intl/icu/source/i18n/digitlst.cpp	2017-08-24 14:42:50.534084676 +0200
@@ -64,7 +64,7 @@
 # if U_PLATFORM_USES_ONLY_WIN32_API || U_PLATFORM == U_PF_CYGWIN
 #   include <locale.h>
 # else
-#   include <xlocale.h>
+#   include <locale.h>
 # endif
 #endif
 
