--- mozilla-release/configure.in	2015-09-17 22:13:29.000000000 +0000
+++ mozilla-release-changed/configure.in	2015-09-23 03:43:55.004658906 +0000
@@ -2183,8 +2183,8 @@
         # while; Intel recommends against using it.
         MOZ_OPTIMIZE_FLAGS="-O2"
     elif test "$GNU_CC" -o "$GNU_CXX"; then
-        MOZ_PGO_OPTIMIZE_FLAGS="-O3"
-        MOZ_OPTIMIZE_FLAGS="-Os"
+        MOZ_PGO_OPTIMIZE_FLAGS=""
+        MOZ_OPTIMIZE_FLAGS=""
         if test -z "$CLANG_CC"; then
            MOZ_OPTIMIZE_FLAGS="-freorder-blocks $MOZ_OPTIMIZE_FLAGS"
         fi
--- mozilla-release/configure	2015-09-17 22:14:09.000000000 +0000
+++ mozilla-release-changed/configure	2015-09-23 03:44:24.317991931 +0000
@@ -9535,8 +9535,8 @@
         # while; Intel recommends against using it.
         MOZ_OPTIMIZE_FLAGS="-O2"
     elif test "$GNU_CC" -o "$GNU_CXX"; then
-        MOZ_PGO_OPTIMIZE_FLAGS="-O3"
-        MOZ_OPTIMIZE_FLAGS="-Os"
+        MOZ_PGO_OPTIMIZE_FLAGS=""
+        MOZ_OPTIMIZE_FLAGS=""
         if test -z "$CLANG_CC"; then
            MOZ_OPTIMIZE_FLAGS="-freorder-blocks $MOZ_OPTIMIZE_FLAGS"
         fi
