diff -up firefox-53.0/dom/u2f/U2F.cpp.prbool firefox-53.0/dom/u2f/U2F.cpp
--- firefox-53.0/dom/u2f/U2F.cpp.prbool	2017-04-11 06:15:14.000000000 +0200
+++ firefox-53.0/dom/u2f/U2F.cpp	2017-04-18 09:19:20.524285200 +0200
@@ -4,6 +4,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "prtypes.h"
 #include "hasht.h"
 #include "mozilla/dom/CallbackFunction.h"
 #include "mozilla/dom/ContentChild.h"
diff -up firefox-53.0/security/certverifier/CTLogVerifier.cpp.prbool firefox-53.0/security/certverifier/CTLogVerifier.cpp
--- firefox-53.0/security/certverifier/CTLogVerifier.cpp.prbool	2017-04-18 09:19:20.524285200 +0200
+++ firefox-53.0/security/certverifier/CTLogVerifier.cpp	2017-04-18 09:52:34.189402811 +0200
@@ -9,6 +9,7 @@
 #include <stdint.h>
 
 #include "CTSerialization.h"
+#include "prtypes.h"
 #include "hasht.h"
 #include "mozilla/ArrayUtils.h"
 #include "mozilla/Assertions.h"
diff -up firefox-53.0/security/certverifier/CTObjectsExtractor.cpp.prbool firefox-53.0/security/certverifier/CTObjectsExtractor.cpp
--- firefox-53.0/security/certverifier/CTObjectsExtractor.cpp.prbool	2017-01-16 17:16:51.000000000 +0100
+++ firefox-53.0/security/certverifier/CTObjectsExtractor.cpp	2017-04-18 09:19:20.525285196 +0200
@@ -6,6 +6,7 @@
 
 #include "CTObjectsExtractor.h"
 
+#include "prtypes.h"
 #include "hasht.h"
 #include "mozilla/Assertions.h"
 #include "mozilla/Casting.h"
diff -up firefox-53.0/security/certverifier/OCSPCache.h.prbool firefox-53.0/security/certverifier/OCSPCache.h
--- firefox-53.0/security/certverifier/OCSPCache.h.prbool	2017-04-11 06:15:19.000000000 +0200
+++ firefox-53.0/security/certverifier/OCSPCache.h	2017-04-18 09:19:20.525285196 +0200
@@ -25,6 +25,7 @@
 #ifndef mozilla_psm_OCSPCache_h
 #define mozilla_psm_OCSPCache_h
 
+#include "prtypes.h"
 #include "hasht.h"
 #include "mozilla/Mutex.h"
 #include "mozilla/Vector.h"
