--- a/layout/generic/nsIFrame.h.old	2016-07-11 13:41:39.688276559 +0200
+++ b/layout/generic/nsIFrame.h	2016-07-11 13:42:12.791406976 +0200
@@ -13,7 +13,7 @@
 #error This header/class should only be used within Mozilla code. It should not be used by extensions.
 #endif
 
-#define MAX_REFLOW_DEPTH 200
+#define MAX_REFLOW_DEPTH 1000
 
 /* nsIFrame is in the process of being deCOMtaminated, i.e., this file is eventually
    going to be eliminated, and all callers will use nsFrame instead.  At the moment
