diff -up firefox-65.0/layout/generic/nsIFrame.h.256180 firefox-65.0/layout/generic/nsIFrame.h
--- firefox-65.0/layout/generic/nsIFrame.h.256180	2019-01-28 14:42:58.245798815 +0100
+++ firefox-65.0/layout/generic/nsIFrame.h	2019-01-28 14:44:24.844449119 +0100
@@ -13,7 +13,7 @@
 #error This header/class should only be used within Mozilla code. It should not be used by extensions.
 #endif
 
-#define MAX_REFLOW_DEPTH 200
+#define MAX_REFLOW_DEPTH 1000
 
 /* nsIFrame is in the process of being deCOMtaminated, i.e., this file is
    eventually going to be eliminated, and all callers will use nsFrame instead.
