diff -up firefox-53.0/js/src/jit/AtomicOperations.h.1253216 firefox-53.0/js/src/jit/AtomicOperations.h
--- firefox-53.0/js/src/jit/AtomicOperations.h.1253216	2017-04-18 10:07:25.939282630 +0200
+++ firefox-53.0/js/src/jit/AtomicOperations.h	2017-04-18 10:15:50.965436343 +0200
@@ -348,7 +348,7 @@ AtomicOperations::isLockfree(int32_t siz
 #elif defined(__sh__)
 #  include "jit/none/AtomicOperations-ppc.h"
 # else
-#  include "jit/none/AtomicOperations-none.h" // These MOZ_CRASH() always
+#  include "jit/none/AtomicOperations-ppc.h"
 # endif
 #elif defined(JS_CODEGEN_X86) || defined(JS_CODEGEN_X64)
 # include "jit/x86-shared/AtomicOperations-x86-shared.h"
