diff -up gv-3.7.4/src/zoom.c.bug1071238 gv-3.7.4/src/zoom.c
--- gv-3.7.4/src/zoom.c.bug1071238	2012-03-29 10:03:50.000000000 -0600
+++ gv-3.7.4/src/zoom.c	2014-03-03 09:13:09.724941964 -0700
@@ -104,12 +104,12 @@ zoom_createZoom(Widget w _GL_UNUSED, XtP
 
     BEGINMESSAGE(zoom_createZoom)
 
-    if (!(gv_psfile != NULL || (gv_gs_arguments && *gv_gs_arguments))) {
+    if (!gv_psfile) {
        INFMESSAGE(no file) ENDMESSAGE(zoom_createZoom)
        return;
     }
 
-    if (strcmp(gv_filename, "-")) {
+    if (gv_filename != NULL && strcmp(gv_filename, "-")) {
        stat(gv_filename, &sbuf);
        if (mtime != sbuf.st_mtime) {
 	  INFMESSAGE1(file has changed) ENDMESSAGE1(zoom_createZoom)return;
