diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/types.h b/Ghidra/Features/Decompiler/src/decompile/cpp/types.h
index d10906637..3c374556a 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/types.h
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/types.h
@@ -101,6 +101,21 @@ typedef char int1;
 typedef uint8 uintp;
 #endif
 
+#if defined (__linux__) && defined (__aarch64__) && defined(__LP64__)
+#define HOST_ENDIAN 0
+typedef unsigned int uintm;
+typedef int intm;
+typedef unsigned long uint8;
+typedef long int8;
+typedef unsigned int uint4;
+typedef int int4;
+typedef unsigned short uint2;
+typedef short int2;
+typedef unsigned char uint1;
+typedef char int1;
+typedef uint8 uintp;
+#endif
+
 #if defined(_WINDOWS)
 
 #if defined(_WIN64)
diff --git a/build.gradle b/build.gradle
index 0df62fc36..2251b6041 100644
--- a/build.gradle
+++ b/build.gradle
@@ -295,6 +295,9 @@ String getCurrentPlatformName() {
 		if (isX86_64) {
 			return 'linux64'
 		}
+		if (isAARCH_64) {
+			return 'linux64'
+		}
 	}
 	else if (osName.startsWith("Mac OS X")) {
 		if (isX86_64) {
diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/Makefile b/Ghidra/Features/Decompiler/src/decompile/cpp/Makefile
index a19dfe4a9..b4496f998 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/Makefile
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/Makefile
@@ -18,7 +18,7 @@ ifndef ARCH
   ARCH=$(CPU)
 endif
 ifeq ($(ARCH),x86_64)
-  ARCH_TYPE=-m64
+  ARCH_TYPE=
   OSDIR=linux64
 else
   ARCH_TYPE=-m32
diff --git a/Ghidra/Debug/Debugger-gadp/build.gradle b/Ghidra/Debug/Debugger-gadp/build.gradle
index 86185f5b4..c45da9f02 100644
--- a/Ghidra/Debug/Debugger-gadp/build.gradle
+++ b/Ghidra/Debug/Debugger-gadp/build.gradle
@@ -33,7 +33,11 @@ configurations {
 def os = org.gradle.internal.os.OperatingSystem.current()
 
 dependencies {
+	String archName = System.getProperty("os.arch")
+	boolean isAARCH_64 = archName.equals("aarch64")
+
 	allProtocArtifacts 'com.google.protobuf:protoc:3.11.1:windows-x86_64@exe'
+    allProtocArtifacts 'com.google.protobuf:protoc:3.11.1:linux-aarch_64@exe'
     allProtocArtifacts 'com.google.protobuf:protoc:3.11.1:linux-x86_64@exe'
     allProtocArtifacts 'com.google.protobuf:protoc:3.11.1:osx-x86_64@exe'
 
@@ -41,7 +45,11 @@ dependencies {
     	protocArtifact 'com.google.protobuf:protoc:3.11.1:windows-x86_64@exe'
 	}
 	if (os.isLinux()) {
-    	protocArtifact 'com.google.protobuf:protoc:3.11.1:linux-x86_64@exe'
+		if (isAARCH_64) {
+		    	protocArtifact 'com.google.protobuf:protoc:3.11.1:linux-aarch_64@exe'
+		} else {
+		    	protocArtifact 'com.google.protobuf:protoc:3.11.1:linux-x86_64@exe'
+		}
 	}
 	if (os.isMacOsX()) {
     	protocArtifact 'com.google.protobuf:protoc:3.11.1:osx-x86_64@exe'
diff --git a/Ghidra/Debug/Framework-Debugging/build.gradle b/Ghidra/Debug/Framework-Debugging/build.gradle
index 841b955a2..413073c22 100644
--- a/Ghidra/Debug/Framework-Debugging/build.gradle
+++ b/Ghidra/Debug/Framework-Debugging/build.gradle
@@ -134,6 +134,11 @@ model {
 				linker.args("-lpthread")
 				linker.args("-lutil")
 			}
+			if (toolChain in Clang) {
+				cCompiler.args("-std=c99")
+				linker.args("-lpthread")
+				linker.args("-lutil")
+			}
 			if (toolChain in VisualCpp) {
 				cppCompiler.define("VS_PROJECT")
 				// NB. No /SUBSYSTEM:CONSOLE
