--- a/build.gradle	2020-02-11 14:52:10.000000000 -0600
+++ b/build.gradle	2020-10-17 20:55:12.870240241 -0500
@@ -263,7 +263,7 @@
 	String archName = System.getProperty("os.arch")
 		
 	boolean isX86_32 = archName.equals("x86") || archName.equals("i386");
-	boolean isX86_64 = archName.equals("x86_64") || archName.equals("amd64");
+	boolean isX86_64 = archName.equals("x86_64") || archName.equals("amd64") || archName.equals("aarch64");
 
 	if (osName.startsWith("Windows")) {
 		if (isX86_32) {
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/types.h	2020-02-11 14:52:10.000000000 -0600
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/types.h	2020-10-17 21:27:15.310873475 -0500
@@ -86,7 +86,7 @@
 typedef uint4 uintp;
 #endif
 
-#if defined (__linux__) && defined (__x86_64__)
+#if defined (__linux__) && (defined (__x86_64__) || defined (__aarch64__))
 #define HOST_ENDIAN 0
 typedef unsigned int uintm;
 typedef int intm;
