diff -Naur libstoragemgmt-1.6.2/NEWS libstoragemgmt-1.6.2.run/NEWS
--- libstoragemgmt-1.6.2/NEWS	2017-09-30 13:28:40.000000000 +0000
+++ libstoragemgmt-1.6.2.run/NEWS	2018-11-30 07:23:24.264513335 +0000
@@ -317,7 +317,7 @@
     - Simulator updates (python & C)
     - Removed tog-pegasus dependency (SMI-S is python plug-in)
     - Removed lsmVolumeStatus as it was implemented and redundant
-    - initscript, check for /var/run and create if missing
+    - initscript, check for /run and create if missing
 
 0.0.10: July 20 2012
     - Simulator plug-in written in C, simc_lsmplugin is available
diff -Naur libstoragemgmt-1.6.2/c_binding/lsm_datatypes.cpp libstoragemgmt-1.6.2.run/c_binding/lsm_datatypes.cpp
--- libstoragemgmt-1.6.2/c_binding/lsm_datatypes.cpp	2017-03-27 12:59:21.000000000 +0000
+++ libstoragemgmt-1.6.2.run/c_binding/lsm_datatypes.cpp	2018-11-30 07:23:24.214511021 +0000
@@ -53,7 +53,7 @@
 #ifdef  __cplusplus
 extern "C" {
 #endif
-#define LSM_DEFAULT_PLUGIN_DIR "/var/run/lsm/ipc"
+#define LSM_DEFAULT_PLUGIN_DIR "/run/lsm/ipc"
 
 int lsm_string_list_append(lsm_string_list * sl, const char *value)
 {
diff -Naur libstoragemgmt-1.6.2/daemon/lsm_daemon.c libstoragemgmt-1.6.2.run/daemon/lsm_daemon.c
--- libstoragemgmt-1.6.2/daemon/lsm_daemon.c	2018-05-14 12:40:58.000000000 +0000
+++ libstoragemgmt-1.6.2.run/daemon/lsm_daemon.c	2018-11-30 07:23:24.284514261 +0000
@@ -44,7 +44,7 @@
 #include <limits.h>
 #include <libconfig.h>
 
-#define BASE_DIR  "/var/run/lsm"
+#define BASE_DIR  "/run/lsm"
 #define SOCKET_DIR BASE_DIR"/ipc"
 #define PLUGIN_DIR "/usr/bin"
 #define LSM_USER "libstoragemgmt"
diff -Naur libstoragemgmt-1.6.2/packaging/daemon/libstoragemgmt.conf libstoragemgmt-1.6.2.run/packaging/daemon/libstoragemgmt.conf
--- libstoragemgmt-1.6.2/packaging/daemon/libstoragemgmt.conf	2018-05-14 12:40:58.000000000 +0000
+++ libstoragemgmt-1.6.2.run/packaging/daemon/libstoragemgmt.conf	2018-11-30 07:23:24.357850988 +0000
@@ -1,2 +1,2 @@
-D /var/run/lsm 0775 root libstoragemgmt -
-D /var/run/lsm/ipc 0775 root libstoragemgmt -
+D /run/lsm 0775 root libstoragemgmt -
+D /run/lsm/ipc 0775 root libstoragemgmt -
diff -Naur libstoragemgmt-1.6.2/packaging/daemon/libstoragemgmtd libstoragemgmt-1.6.2.run/packaging/daemon/libstoragemgmtd
--- libstoragemgmt-1.6.2/packaging/daemon/libstoragemgmtd	2016-09-14 08:13:21.000000000 +0000
+++ libstoragemgmt-1.6.2.run/packaging/daemon/libstoragemgmtd	2018-11-30 07:23:24.357850988 +0000
@@ -15,7 +15,7 @@
 
 NAME=lsmd
 PROG=/usr/bin/$NAME
-PID=/var/run/lsm/lsmd.pid
+PID=/run/lsm/lsmd.pid
 LOCK=/var/lock/subsys/libstoragemgmtd
 
 RETVAL=0
@@ -27,8 +27,8 @@
 fi
 
 check_dirs() {
-	test -d /var/run/lsm || mkdir -p /var/run/lsm/ipc
-	test -d /var/run/lsm/ipc || mkdir -p /var/run/lsm/ipc
+	test -d /run/lsm || mkdir -p /run/lsm/ipc
+	test -d /run/lsm/ipc || mkdir -p /run/lsm/ipc
 }
 
 status_lsm() {
diff -Naur libstoragemgmt-1.6.2/packaging/libstoragemgmt.spec libstoragemgmt-1.6.2.run/packaging/libstoragemgmt.spec
--- libstoragemgmt-1.6.2/packaging/libstoragemgmt.spec	2018-05-18 08:12:00.000000000 +0000
+++ libstoragemgmt-1.6.2.run/packaging/libstoragemgmt.spec	2018-11-30 07:23:24.357850988 +0000
@@ -311,7 +311,7 @@
 %pre
 getent group libstoragemgmt >/dev/null || groupadd -r libstoragemgmt
 getent passwd libstoragemgmt >/dev/null || \
-    useradd -r -g libstoragemgmt -d /var/run/lsm -s /sbin/nologin \
+    useradd -r -g libstoragemgmt -d /run/lsm -s /sbin/nologin \
     -c "daemon account for libstoragemgmt" libstoragemgmt
 
 %post
diff -Naur libstoragemgmt-1.6.2/packaging/libstoragemgmt.spec.in libstoragemgmt-1.6.2.run/packaging/libstoragemgmt.spec.in
--- libstoragemgmt-1.6.2/packaging/libstoragemgmt.spec.in	2018-05-18 08:10:52.000000000 +0000
+++ libstoragemgmt-1.6.2.run/packaging/libstoragemgmt.spec.in	2018-11-30 07:23:24.357850988 +0000
@@ -311,7 +311,7 @@
 %pre
 getent group libstoragemgmt >/dev/null || groupadd -r libstoragemgmt
 getent passwd libstoragemgmt >/dev/null || \
-    useradd -r -g libstoragemgmt -d /var/run/lsm -s /sbin/nologin \
+    useradd -r -g libstoragemgmt -d /run/lsm -s /sbin/nologin \
     -c "daemon account for libstoragemgmt" libstoragemgmt
 
 %post
diff -Naur libstoragemgmt-1.6.2/python_binding/lsm/_common.py libstoragemgmt-1.6.2.run/python_binding/lsm/_common.py
--- libstoragemgmt-1.6.2/python_binding/lsm/_common.py	2017-11-03 13:14:55.000000000 +0000
+++ libstoragemgmt-1.6.2.run/python_binding/lsm/_common.py	2018-11-30 07:23:24.237845434 +0000
@@ -147,7 +147,7 @@
         return getattr(self.proxied_obj, _proxy_method_name)(*args, **kwargs)
 
 # variable in client and specified on the command line for the daemon
-UDS_PATH = '/var/run/lsm/ipc'
+UDS_PATH = '/run/lsm/ipc'
 
 # Set to True for verbose logging
 LOG_VERBOSE = True
