--- qemu-5.1.0/configure	2020-08-11 14:17:15.000000000 -0500
+++ qemu-5.1.0.kvmpr/configure	2020-10-15 13:02:23.636254047 -0500
@@ -204,7 +204,8 @@
         i386:i386 | i386:x86_64 | i386:x32 | \
         x86_64:i386 | x86_64:x86_64 | x86_64:x32 | \
         mips:mips | mipsel:mips | mips64:mips | mips64el:mips | \
-        ppc:ppc | ppc64:ppc | ppc:ppc64 | ppc64:ppc64 | ppc64:ppc64le | \
+        ppc:ppc | ppc64:ppc | ppc:ppc64 | ppc64:ppc64 | \
+        ppc:ppc64le | ppc64:ppc64le | \
         s390x:s390x)
             return 0
         ;;
