diff --git a/html2ps b/html2ps
index aea859e..ba6bf3b 100755
--- a/html2ps
+++ b/html2ps
@@ -1608,7 +1608,6 @@ $defs$fontdef$reenc
  Fi cH 1 eq and{fill}if} {Bz 0 RL 0 Bz RL Bz neg 0 RL 0 Bz neg RL
  Fi cH 1 eq and{fill}if} {0 -5 R Bz 0 RL 0 21 RL Bz neg 0 RL 0 -21 RL}]} D
 /MS {/Sm E D WB}D
-/O {BN()Sm BX} D
 /O {BN()0 Sm BX} D
 /BX {/Bt E D Bt 2 lt{/Ch E D CS 0.8 mul}{11 mul}ie W XO sub MR sub
  2 copy gt{E}if pop /HZ E D Bt 2 eq{Fi not{pop()}if ( )E join /Ft E D TT
