From b0a06d52e39438fae2afbf98bafe6c794d13b83e Mon Sep 17 00:00:00 2001
From: Christoph Willing <chris.willing@linux.com>
Date: Sat, 14 Jul 2018 21:34:03 +1000
Subject: [PATCH] Include opencv2/imgproc.hpp for CV_RGB()

Signed-off-by: Christoph Willing <chris.willing@linux.com>
---
 src/filter/facebl0r/facebl0r.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/filter/facebl0r/facebl0r.cpp b/src/filter/facebl0r/facebl0r.cpp
index 6d6e8f2..823462a 100644
--- a/src/filter/facebl0r/facebl0r.cpp
+++ b/src/filter/facebl0r/facebl0r.cpp
@@ -19,6 +19,7 @@
 #include <stdlib.h>
 #include <opencv/cv.h>
 #include <opencv/highgui.h>
+#include <opencv2/imgproc.hpp>
 
 #include <frei0r.hpp>
 #include <frei0r_math.h>

