From 003d64e41f73524ec4aa474bc0ac90a737647a86 Mon Sep 17 00:00:00 2001
From: Christoph Willing <chris.willing@linux.com>
Date: Sun, 15 Jul 2018 10:30:23 +1000
Subject: [PATCH] Check opencv version

Signed-off-by: Christoph Willing <chris.willing@linux.com>
---
 src/filter/facebl0r/facebl0r.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/filter/facebl0r/facebl0r.cpp b/src/filter/facebl0r/facebl0r.cpp
index 823462a..10fcdb7 100644
--- a/src/filter/facebl0r/facebl0r.cpp
+++ b/src/filter/facebl0r/facebl0r.cpp
@@ -15,11 +15,17 @@
  */
 
 
+#include <opencv2/core/version.hpp>
+#define CV_VERSION_NUM (CV_VERSION_MAJOR * 10000 \
+                      + CV_VERSION_MINOR * 100 \
+                      + CV_VERSION_REVISION)
 #include <stdio.h>
 #include <stdlib.h>
 #include <opencv/cv.h>
 #include <opencv/highgui.h>
+#if CV_VERSION_NUM > 30401
 #include <opencv2/imgproc.hpp>
+#endif
 
 #include <frei0r.hpp>
 #include <frei0r_math.h>

