diff -Naur gnome-multi-writer-3.26.0/man/meson.build gnome-multi-writer-3.26.0.manfix/man/meson.build
--- gnome-multi-writer-3.26.0/man/meson.build	2017-09-11 10:43:11.000000000 +0000
+++ gnome-multi-writer-3.26.0.manfix/man/meson.build	2017-10-02 20:40:53.914129100 +0000
@@ -3,7 +3,7 @@
   custom_target('gnome-multi-writer-man',
     output : 'gnome-multi-writer.1',
     input : 'gnome-multi-writer.sgml',
-    command : [docbook2man, '@INPUT@', '--output', 'man'],
+    command : [docbook2man, '--sgml', '@INPUT@', '--output', 'man'],
     install : true,
     install_dir : 'share/man/man1'
   )
