diff -ruN xrdp-0.9.5-v/xrdp/xrdp.ini xrdp-0.9.5/xrdp/xrdp.ini
--- xrdp-0.9.5-v/xrdp/xrdp.ini	2017-12-28 01:30:25.000000000 +1100
+++ xrdp-0.9.5/xrdp/xrdp.ini	2017-12-30 18:14:09.862621310 +1100
@@ -153,72 +153,72 @@
 ; Some session types such as Xorg, X11rdp and Xvnc start a display server.
 ; Startup command-line parameters for the display server are configured
 ; in sesman.ini. See and configure also sesman.ini.
-[Xorg]
-name=Xorg
-lib=libxup.so
+[Xvnc]
+name=Xvnc
+lib=libvnc.so
 username=ask
 password=ask
 ip=127.0.0.1
 port=-1
-code=20
+#xserverbpp=24
+#delay_ms=2000
 
-[X11rdp]
-name=X11rdp
+[Xorg]
+name=Xorg
 lib=libxup.so
 username=ask
 password=ask
 ip=127.0.0.1
 port=-1
-xserverbpp=24
-code=10
+code=20
 
-[Xvnc]
-name=Xvnc
-lib=libvnc.so
-username=ask
-password=ask
-ip=127.0.0.1
-port=-1
+#[X11rdp]
+#name=X11rdp
+#lib=libxup.so
+#username=ask
+#password=ask
+#ip=127.0.0.1
+#port=-1
 #xserverbpp=24
-#delay_ms=2000
+#code=10
 
-[console]
-name=console
-lib=libvnc.so
-ip=127.0.0.1
-port=5900
-username=na
-password=ask
+#[console]
+#name=console
+#lib=libvnc.so
+#ip=127.0.0.1
+#port=5900
+#username=na
+#password=ask
 #delay_ms=2000
 
-[vnc-any]
-name=vnc-any
-lib=libvnc.so
-ip=ask
-port=ask5900
-username=na
-password=ask
+#[vnc-any]
+#name=vnc-any
+#lib=libvnc.so
+#ip=ask
+#port=ask5900
+#username=na
+#password=ask
 #pamusername=asksame
 #pampassword=asksame
 #pamsessionmng=127.0.0.1
 #delay_ms=2000
 
-[sesman-any]
-name=sesman-any
-lib=libvnc.so
-ip=ask
-port=-1
-username=ask
-password=ask
+#[sesman-any]
+#name=sesman-any
+#lib=libvnc.so
+#ip=ask
+#port=-1
+#username=ask
+#password=ask
 #delay_ms=2000
 
-[neutrinordp-any]
-name=neutrinordp-any
-lib=libxrdpneutrinordp.so
-ip=ask
-port=ask3389
-username=ask
-password=ask
+#[neutrinordp-any]
+#name=neutrinordp-any
+#lib=libxrdpneutrinordp.so
+#ip=ask
+#port=ask3389
+#username=ask
+#password=ask
 
 ; You can override the common channel settings for each session type
 #channel.rdpdr=true
