--- xrdp-0.9.2-vanilla/sesman/sesman.ini	2017-02-20 18:11:37.000000000 +1100
+++ xrdp-0.9.2/sesman/sesman.ini	2017-04-07 11:03:29.929069805 +1000
@@ -3,7 +3,7 @@
 ListenPort=3350
 EnableUserWindowManager=true
 UserWindowManager=startwm.sh
-DefaultWindowManager=startwm.sh
+DefaultWindowManager=startwm-bash.sh
 
 [Security]
 AllowRootLogin=true
@@ -54,12 +54,12 @@
 EnableSyslog=1
 SyslogLevel=DEBUG
 
-[X11rdp]
-param=X11rdp
-param=-bs
-param=-nolisten
-param=tcp
-param=-uds
+#[X11rdp]
+#param=X11rdp
+#param=-bs
+#param=-nolisten
+#param=tcp
+#param=-uds
 
 [Xvnc]
 param=Xvnc
