diff -ruN xrdp-0.9.0-vanilla/instfiles/xrdp.service xrdp-0.9.0/instfiles/xrdp.service
--- xrdp-0.9.0-vanilla/instfiles/xrdp.service	2015-07-15 07:17:49.000000000 +1000
+++ xrdp-0.9.0/instfiles/xrdp.service	2015-07-17 10:19:19.000000000 +1000
@@ -1,14 +1,11 @@
 [Unit]
 Description=xrdp daemon
 Requires=xrdp-sesman.service
-After=syslog.target network.target xrdp-sesman.service
+After=xrdp-sesman.service
 
 [Service]
-Type=forking
-PIDFile=/var/run/xrdp.pid
 EnvironmentFile=/etc/sysconfig/xrdp
-ExecStart=/usr/sbin/xrdp $XRDP_OPTIONS
-ExecStop=/usr/sbin/xrdp $XRDP_OPTIONS --kill
+ExecStart=/usr/sbin/xrdp $XRDP_OPTIONS --nodaemon
 
 [Install]
 WantedBy=multi-user.target
diff -ruN xrdp-0.9.0-vanilla/instfiles/xrdp-sesman.service xrdp-0.9.0/instfiles/xrdp-sesman.service
--- xrdp-0.9.0-vanilla/instfiles/xrdp-sesman.service	2015-07-15 07:17:49.000000000 +1000
+++ xrdp-0.9.0/instfiles/xrdp-sesman.service	2015-07-17 10:19:46.000000000 +1000
@@ -1,14 +1,13 @@
 [Unit]
 Description=xrdp session manager
-After=syslog.target network.target
+BindsTo=xrdp.service
+Wants=network-online.target nss-lookup.target nss-user-lookup.target
+After=network-online.target nss-lookup.target nss-user-lookup.target
 StopWhenUnneeded=true
 
 [Service]
-Type=forking
-PIDFile=/var/run/xrdp-sesman.pid
 EnvironmentFile=/etc/sysconfig/xrdp
-ExecStart=/usr/sbin/xrdp-sesman $SESMAN_OPTIONS
-ExecStop=/usr/sbin/xrdp-sesman $SESMAN_OPTIONS --kill
+ExecStart=/usr/sbin/xrdp-sesman $SESMAN_OPTIONS --nodaemon
 
 [Install]
 WantedBy=multi-user.target
