diff -ruN xrdp-0.9.0-vanilla/sesman/sesman.ini xrdp-0.9.0/sesman/sesman.ini
--- xrdp-0.9.0-vanilla/sesman/sesman.ini	2015-07-15 07:17:49.000000000 +1000
+++ xrdp-0.9.0/sesman/sesman.ini	2015-07-17 10:25:13.449024694 +1000
@@ -3,7 +3,7 @@
 ListenPort=3350
 EnableUserWindowManager=1
 UserWindowManager=startwm.sh
-DefaultWindowManager=startwm.sh
+DefaultWindowManager=startwm-bash.sh
 
 [Security]
 AllowRootLogin=1
@@ -64,19 +64,17 @@
 
 [X11rdp]
 param1=-bs
-param2=-ac
-param3=-nolisten
-param4=tcp
-param5=-uds
+param2=-nolisten
+param3=tcp
+param4=-uds
 
 [Xvnc]
 param1=-bs
-param2=-ac
-param3=-nolisten
-param4=tcp
-param5=-localhost
-param6=-dpi
-param7=96
+param2=-nolisten
+param3=tcp
+param4=-localhost
+param5=-dpi
+param6=96
 
 [Xorg]
 param1=-config
@@ -84,9 +82,8 @@
 param3=-logfile
 param4=/dev/null
 param5=-noreset
-param6=-ac
-param7=-nolisten
-param8=tcp
+param6=-nolisten
+param7=tcp
 
 [Chansrv]
 # drive redirection, defaults to xrdp_client if not set
