diff -ruN xrdp-0.9.5-v/sesman/sesman.ini xrdp-0.9.5/sesman/sesman.ini
--- xrdp-0.9.5-v/sesman/sesman.ini	2017-12-28 01:30:25.000000000 +1100
+++ xrdp-0.9.5/sesman/sesman.ini	2017-12-30 18:08:03.726887379 +1100
@@ -3,7 +3,7 @@
 ListenPort=3350
 EnableUserWindowManager=true
 UserWindowManager=startwm.sh
-DefaultWindowManager=startwm.sh
+DefaultWindowManager=startwm-bash.sh
 
 [Security]
 AllowRootLogin=true
@@ -78,12 +78,12 @@
 param=-logfile
 param=.xorgxrdp.%s.log
 
-[X11rdp]
-param=X11rdp
-param=-bs
-param=-nolisten
-param=tcp
-param=-uds
+#[X11rdp]
+#param=X11rdp
+#param=-bs
+#param=-nolisten
+#param=tcp
+#param=-uds
 
 [Xvnc]
 param=Xvnc
