From 984e24cf9ecaf61c109b36556db8411ac1aa9b30 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Thu, 1 Feb 2024 04:27:11 -0800
Subject: [PATCH 2/2] [Gentoo] make the build system respect system flags

---
 configure.ac | 1 -
 1 file changed, 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index a8934381..5c9097eb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -22,7 +22,6 @@ AC_PROG_CC
 AC_PROG_CXX
 AX_PROG_CC_FOR_BUILD
 
-AX_CC_MAXOPT
 AX_GCC_WARN_UNUSED_RESULT
 AX_C___ATTRIBUTE__
 
-- 
2.43.0

