diff -Naur a/src/calamares/CalamaresWindow.cpp b/src/calamares/CalamaresWindow.cpp
--- a/src/calamares/CalamaresWindow.cpp	2015-07-08 14:31:01.000000000 +0000
+++ b/src/calamares/CalamaresWindow.cpp	2016-01-06 18:54:28.645784669 +0000
@@ -65,28 +65,6 @@
     sideBox->setFixedWidth( qMax( 190, CalamaresUtils::defaultFontHeight() * 12 ) );
     sideBox->setSizePolicy( QSizePolicy::Expanding, QSizePolicy::Expanding );
 
-    QHBoxLayout* logoLayout = new QHBoxLayout;
-    sideLayout->addLayout( logoLayout );
-    logoLayout->addStretch();
-    QLabel* logoLabel = new QLabel( sideBox );
-    {
-        QPalette plt = sideBox->palette();
-        sideBox->setAutoFillBackground( true );
-        plt.setColor( sideBox->backgroundRole(), Calamares::Branding::instance()->
-                      styleString( Calamares::Branding::SidebarBackground ) );
-        plt.setColor( sideBox->foregroundRole(), Calamares::Branding::instance()->
-                      styleString( Calamares::Branding::SidebarText ) );
-        sideBox->setPalette( plt );
-        logoLabel->setPalette( plt );
-    }
-    logoLabel->setAlignment( Qt::AlignCenter );
-    logoLabel->setFixedSize( 80, 80 );
-    logoLabel->setPixmap( Calamares::Branding::instance()->
-                          image( Calamares::Branding::ProductIcon,
-                                 logoLabel->size() ) );
-    logoLayout->addWidget( logoLabel );
-    logoLayout->addStretch();
-
     ProgressTreeView* tv = new ProgressTreeView( sideBox );
     sideLayout->addWidget( tv );
     tv->setFocusPolicy( Qt::NoFocus );
diff -Naur a/src/calamares/CalamaresWindow.ui b/src/calamares/CalamaresWindow.ui
--- a/src/calamares/CalamaresWindow.ui	2015-04-10 11:29:20.000000000 +0000
+++ b/src/calamares/CalamaresWindow.ui	2016-01-06 18:47:06.821911124 +0000
@@ -57,70 +57,6 @@
        <number>0</number>
       </property>
       <item>
-       <layout class="QHBoxLayout" name="horizontalLayout_3">
-        <property name="topMargin">
-         <number>15</number>
-        </property>
-        <property name="rightMargin">
-         <number>0</number>
-        </property>
-        <property name="bottomMargin">
-         <number>10</number>
-        </property>
-        <item>
-         <spacer name="horizontalSpacer">
-          <property name="orientation">
-           <enum>Qt::Horizontal</enum>
-          </property>
-          <property name="sizeHint" stdset="0">
-           <size>
-            <width>40</width>
-            <height>20</height>
-           </size>
-          </property>
-         </spacer>
-        </item>
-        <item>
-         <widget class="QLabel" name="topIcon">
-          <property name="minimumSize">
-           <size>
-            <width>80</width>
-            <height>80</height>
-           </size>
-          </property>
-          <property name="maximumSize">
-           <size>
-            <width>80</width>
-            <height>80</height>
-           </size>
-          </property>
-          <property name="text">
-           <string/>
-          </property>
-          <property name="pixmap">
-           <pixmap>:/Images/images/install.png</pixmap>
-          </property>
-          <property name="scaledContents">
-           <bool>true</bool>
-          </property>
-         </widget>
-        </item>
-        <item>
-         <spacer name="horizontalSpacer_2">
-          <property name="orientation">
-           <enum>Qt::Horizontal</enum>
-          </property>
-          <property name="sizeHint" stdset="0">
-           <size>
-            <width>40</width>
-            <height>20</height>
-           </size>
-          </property>
-         </spacer>
-        </item>
-       </layout>
-      </item>
-      <item>
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
@@ -143,26 +79,7 @@
           <property name="leftMargin">
            <number>2</number>
           </property>
-          <item row="0" column="0">
-           <widget class="QLabel" name="welcomeIcon">
-            <property name="minimumSize">
-             <size>
-              <width>18</width>
-              <height>18</height>
-             </size>
-            </property>
-            <property name="maximumSize">
-             <size>
-              <width>18</width>
-              <height>18</height>
-             </size>
-            </property>
-            <property name="text">
-             <string/>
-            </property>
-           </widget>
-          </item>
-          <item row="0" column="1" colspan="2">
+          <item row="0" column="0" colspan="2">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
@@ -173,7 +90,7 @@
             </property>
            </widget>
           </item>
-          <item row="1" column="1">
+          <item row="1" column="0">
            <widget class="QLabel" name="releaseIcon">
             <property name="minimumSize">
              <size>
@@ -192,14 +109,14 @@
             </property>
            </widget>
           </item>
-          <item row="1" column="2">
+          <item row="1" column="1">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Location</string>
             </property>
            </widget>
           </item>
-          <item row="2" column="1">
+          <item row="2" column="0">
            <widget class="QLabel" name="licenseIcon">
             <property name="minimumSize">
              <size>
@@ -218,7 +135,7 @@
             </property>
            </widget>
           </item>
-          <item row="2" column="2">
+          <item row="2" column="1">
            <widget class="QLabel" name="licenseLabel">
             <property name="text">
              <string>License Approval</string>
