diff -Naur a/src/modules/welcome/WelcomePage.cpp b/src/modules/welcome/WelcomePage.cpp
--- a/src/modules/welcome/WelcomePage.cpp	2015-08-06 09:58:32.000000000 +0000
+++ b/src/modules/welcome/WelcomePage.cpp	2016-01-07 01:41:06.888131993 +0000
@@ -45,7 +45,7 @@
 {
     ui->setupUi( this );
 
-    ui->verticalLayout->insertSpacing( 1, CalamaresUtils::defaultFontHeight() * 4 );
+    ui->verticalLayout->insertSpacing( 1, CalamaresUtils::defaultFontHeight() * 2 );
     initLanguages();
 
     ui->mainText->setAlignment( Qt::AlignCenter );
diff -Naur a/src/modules/welcome/WelcomePage.ui b/src/modules/welcome/WelcomePage.ui
--- a/src/modules/welcome/WelcomePage.ui	2015-08-06 09:58:32.000000000 +0000
+++ b/src/modules/welcome/WelcomePage.ui	2016-01-07 01:44:10.491636951 +0000
@@ -27,7 +27,7 @@
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
-         <height>40</height>
+         <height>10</height>
         </size>
        </property>
       </spacer>
