diff -Naur a/src/calamares/progresstree/ProgressTreeDelegate.cpp b/src/calamares/progresstree/ProgressTreeDelegate.cpp
--- a/src/calamares/progresstree/ProgressTreeDelegate.cpp	2015-06-19 20:47:39.000000000 +0000
+++ b/src/calamares/progresstree/ProgressTreeDelegate.cpp	2016-01-07 02:50:23.178802450 +0000
@@ -104,7 +104,8 @@
     {
         painter->setPen( Calamares::Branding::instance()->
                          styleString( Calamares::Branding::SidebarTextSelect ) );
-        painter->setBrush( APP->mainWindow()->palette().background() );
+        painter->setBrush( QColor( Calamares::Branding::instance()->
+                           styleString( Calamares::Branding::SidebarTextHighlight ) ) );
     }
 
     painter->fillRect( option.rect, painter->brush().color() );
diff -Naur a/src/libcalamaresui/Branding.cpp b/src/libcalamaresui/Branding.cpp
--- a/src/libcalamaresui/Branding.cpp	2015-06-19 21:33:51.000000000 +0000
+++ b/src/libcalamaresui/Branding.cpp	2016-01-07 02:44:27.541809495 +0000
@@ -71,7 +71,8 @@
 {
     "sidebarBackground",
     "sidebarText",
-    "sidebarTextSelect"
+    "sidebarTextSelect",
+    "sidebarTextHighlight"
 };
 
 
diff -Naur a/src/libcalamaresui/Branding.h b/src/libcalamaresui/Branding.h
--- a/src/libcalamaresui/Branding.h	2015-06-19 21:33:37.000000000 +0000
+++ b/src/libcalamaresui/Branding.h	2016-01-07 02:45:10.258514987 +0000
@@ -62,7 +62,8 @@
     {
         SidebarBackground,
         SidebarText,
-        SidebarTextSelect
+        SidebarTextSelect,
+        SidebarTextHighlight
     };
 
     static Branding* instance();
