From e2732f187667baebdc2a8ec3102cc6851f66e576 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Wed, 6 Mar 2024 14:34:07 -0500
Subject: [PATCH] fix document install

---
 documents/CMakeLists.txt | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/documents/CMakeLists.txt b/documents/CMakeLists.txt
index d435b674..e3991426 100644
--- a/documents/CMakeLists.txt
+++ b/documents/CMakeLists.txt
@@ -20,15 +20,16 @@ find_package(Doxygen)
 
 if(DOXYGEN_FOUND)
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile.in ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile)
+    configure_file(DoxygenAwesome/doxygen-awesome.css doxygen-awesome.css COPYONLY)
+    file(GLOB IMAGES "Images/*")
+    foreach(IMAGE ${IMAGES})
+        get_filename_component(IMGAE_FILE ${IMAGE} NAME)
+        configure_file(${IMAGE} ${IMGAE_FILE} COPYONLY)
+    endforeach()
     add_custom_target(MaterialXDocs ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile
                       WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+                      DEPENDS doxygen-awesome.css MaterialXLogo_200x155.png
                       COMMENT "Generating HTML documentation: ${DOXYGEN_HTML_OUTPUT_DIR}/index.html")
-    add_custom_command(TARGET MaterialXDocs PRE_BUILD
-                       COMMAND ${CMAKE_COMMAND} -E copy_directory
-                       ${CMAKE_SOURCE_DIR}/documents/DoxygenAwesome ${CMAKE_CURRENT_BINARY_DIR})
-    add_custom_command(TARGET MaterialXDocs PRE_BUILD
-                       COMMAND ${CMAKE_COMMAND} -E copy_directory
-                       ${CMAKE_SOURCE_DIR}/documents/Images ${CMAKE_CURRENT_BINARY_DIR})
     install(DIRECTORY ${DOXYGEN_HTML_OUTPUT_DIR}
 	    DESTINATION "/share/doc/materialx/" MESSAGE_NEVER)
 endif(DOXYGEN_FOUND)
-- 
2.44.0

