diff -Naur xdg-user-dirs-gtk-0.10/user-dirs-update-gtk.desktop.in xdg-user-dirs-gtk-0.10.morede/user-dirs-update-gtk.desktop.in
--- xdg-user-dirs-gtk-0.10/user-dirs-update-gtk.desktop.in	2013-01-21 13:18:32.000000000 +0000
+++ xdg-user-dirs-gtk-0.10.morede/user-dirs-update-gtk.desktop.in	2018-11-30 06:34:42.252428539 +0000
@@ -4,7 +4,7 @@
 _Name=User folders update
 _Comment=Update common folders names to match current locale
 Terminal=false
-OnlyShowIn=GNOME;LXDE;Unity;
+OnlyShowIn=GNOME;LXDE;Unity;MATE;Cinnamon;
 Type=Application
 StartupNotify=false
 X-KDE-autostart-after=panel
