--- a/src/utils/cas/configfile.c
+++ b/src/utils/cas/configfile.c
@@ -52,7 +52,7 @@ int writeconfig(void)
 		"# font_size - size the font\n",
 		"# source_image - image where the text will be written\n",
 		"# *_line - x,y,[1/0] enabled or disabled\n\n",
-		"font /usr/share/fonts/corefonts/times.ttf\n",
+		"font /usr/share/fonts/truetype/ttf-dejavu/DejaVuSerif.ttf\n",
 		"font_size 10.5\n",
 		"source_image /usr/share/cas/stat.png\n",
 		"first_line 23,17,1\n",
