diff -uNrp b/tools/udisks-bash-completion.sh a/tools/udisks-bash-completion.sh
--- b/tools/udisks-bash-completion.sh	2011-08-25 11:27:33.000000000 -0700
+++ a/tools/udisks-bash-completion.sh	2013-02-24 19:39:41.968418728 -0800
@@ -1,6 +1,6 @@
 
 # Check for bash
-[ -z "$BASH_VERSION" ] && return
+[ -z "$BASH_VERSION" -o -z "$PS1" ] && return
 
 ####################################################################################################
 
