--- zyn-fusion-ui-src-3.0.5/Makefile	2019-04-09 17:35:34.000000000 -0600
+++ zyn-fusion-ui-src-3.0.5.new/Makefile	2019-04-18 10:57:29.781164861 -0600
@@ -5,18 +5,18 @@
 
 all:
 	ruby ./rebuild-fcache.rb
-	cd deps/nanovg/src   && $(CC) nanovg.c -c -fPIC
+	cd deps/nanovg/src   && $(CC) nanovg.c -c -fPIC $(CFLAGS)
 	$(AR) rc deps/libnanovg.a deps/nanovg/src/*.o
-#	cd deps/pugl         && python2 ./waf configure --no-cairo --static
-	cd deps/pugl         && python2 ./waf configure --no-cairo --static --debug
+	cd deps/pugl         && python2 ./waf configure --no-cairo --static
+#	cd deps/pugl         && python2 ./waf configure --no-cairo --static --debug
 	cd deps/pugl         && python2 ./waf
-	cd src/osc-bridge    && CFLAGS="-I ../../deps/$(UV_DIR)/include " make lib
+	cd src/osc-bridge    && make lib
 	cd mruby             && MRUBY_CONFIG=../build_config.rb rake
-	$(CC) -shared -o libzest.so `find mruby/build/host -type f | grep -e "\.o$$" | grep -v bin` ./deps/libnanovg.a \
+	$(CC) -shared -o libzest.so -Wl,--whole-archive ./mruby/build/host/lib/libmruby.a -Wl,--no-whole-archive  ./deps/libnanovg.a \
 		./deps/libnanovg.a \
 		src/osc-bridge/libosc-bridge.a \
-		./deps/$(UV_DIR)/.libs/libuv.a  -lm -lX11 -lGL -lpthread
-	$(CC) test-libversion.c deps/pugl/build/libpugl-0.a -ldl -o zest -lX11 -lGL -lpthread -I deps/pugl -std=gnu99
+		-luv -lm -lX11 -lGL -lpthread $(LDFLAGS)
+	$(CC) test-libversion.c deps/pugl/build/libpugl-0.a $(CFLAGS) -ldl -o zest -lX11 -lGL -lpthread -I deps/pugl -std=gnu99
 
 osx:
 	ruby ./rebuild-fcache.rb
--- zyn-fusion-ui-src-3.0.5/build_config.rb	2019-04-14 22:15:52.000000000 -0600
+++ zyn-fusion-ui-src-3.0.5.new/build_config.rb	2019-04-18 10:58:35.227391177 -0600
@@ -92,7 +92,7 @@
   conf.cc do |cc|
       cc.include_paths << "#{`pwd`.strip}/../deps/nanovg/src"
       cc.include_paths << "#{`pwd`.strip}/../deps/pugl/"
-      cc.include_paths << "#{`pwd`.strip}/../deps/libuv-v1.9.1/include/"
+      cc.include_paths << "#{`pwd`.strip}/../deps/libuv-v1.9.1/include/" if windows
       cc.include_paths << "/usr/share/mingw-w64/include/" if windows
       cc.include_paths << "/usr/x86_64-w64-mingw32/include/" if windows
       cc.flags << "-DLDBL_EPSILON=1e-6" if windows
@@ -113,7 +113,7 @@
       linker.flags_after_libraries  << "#{`pwd`.strip}/../deps/pugl/build/libpugl-0.a"
       linker.flags_after_libraries  << "#{`pwd`.strip}/../deps/libnanovg.a"
       if(!windows)
-        linker.flags_after_libraries  << "#{`pwd`.strip}/../deps/libuv.a"
+        linker.flags_after_libraries  << "-luv"
         if(ENV['OS'] != "Mac")
           linker.libraries << 'GL'
           linker.libraries << 'X11'
