From 9a14038287ff48dd361904e9bee1a191cb2e18ec Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 15 Jul 2024 15:17:29 +0800
Subject: [PATCH 2/2] fix(mruby-widget-lib): fix asset search path

---
 src/mruby-widget-lib/src/api.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/mruby-widget-lib/src/api.c b/src/mruby-widget-lib/src/api.c
index 6949413..0ea74a6 100644
--- a/src/mruby-widget-lib/src/api.c
+++ b/src/mruby-widget-lib/src/api.c
@@ -25,6 +25,15 @@ static void
 check_error(mrb_state *mrb);
 
 char *get_search_path(void) {
+    const char* assets_path = "/usr/share/zyn-fusion/";
+    char buf[256];
+    snprintf(buf, sizeof(buf), "%s/qml/MainWindow.qml", assets_path);
+    FILE *f = fopen(buf, "r");
+    if (f) {
+        fclose(f);
+        return strdup(assets_path);
+    }
+
     char buffer[1024];
     GetModuleFileName(GetModuleHandle("libzest.dll"), buffer, sizeof(buffer));
     //printf("get_search_path() => <%s>\n", buffer);
-- 
2.45.2

