diff --git a/src/mruby-widget-lib/src/api.c b/src/mruby-widget-lib/src/api.c
index b443da1..20af3f4 100644
--- a/src/mruby-widget-lib/src/api.c
+++ b/src/mruby-widget-lib/src/api.c
@@ -12,7 +12,15 @@ static void
 check_error(mrb_state *mrb);
 
 char *get_search_path(void) {
+    const char* assets_path = "/usr/share/zyn-fusion/";
+    char buf[256];
+    snprintf(buf, sizeof(buf), "%s/qml/MainWindow.qml", assets_path);
     Dl_info dl_info;
+    FILE *f = fopen(buf, "r");
+    if (f) {
+        fclose(f);
+        return strdup(assets_path);
+    }
     dladdr((void*)check_error, &dl_info);
     return strdup(dl_info.dli_fname);
 }
