diff -Naur PyQwt-5.2.0/configure/configure.py PyQwt-5.2.0.qt4/configure/configure.py
--- PyQwt-5.2.0/configure/configure.py	2009-08-01 02:28:33.000000000 -0500
+++ PyQwt-5.2.0.qt4/configure/configure.py	2017-07-10 01:46:14.929847855 -0500
@@ -183,7 +183,7 @@
 ''' % ' '.join(extra_include_dirs))
     f.close()
 
-    qmake = os.path.join(configuration.qt_dir, 'bin', 'qmake')
+    qmake = os.path.join('/usr/lib/qt4', 'bin', 'qmake')
     os.system('%s -spec %s -o %s %s' % (
         qmake, configuration.platform, make_file, pro_file))
 
