--- extra-devel/extra-cmake-modules/spec	2019-07-24 14:43:26 +0000
+++ extra-devel/extra-cmake-modules/spec	2019-07-24 14:43:32 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/extra-cmake-modules-$VER.tar.xz"
-CHKSUM="sha256::1a630b242242ac32d20d14e7ae5645aa28a47676eae0e62ed76af8bc1f56c438"
+CHKSUM="sha256::2bd9da815de98d5908d3371815df963d305c828f90ba1a076f38543876690248"
--- extra-kde/syntax-highlighting/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/syntax-highlighting/spec	2019-07-24 14:43:41 +0000
@@ -1,4 +1,4 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/syntax-highlighting-$VER.tar.xz"
-CHKSUM="sha256::8ed700bb070638a5e7ad33f45ba29bc88f9420ee1ece1230296e3da12b5e91f0"
+CHKSUM="sha256::45ada4dca20641654a2d076ecb45b9ef78a4aa8032afe54ade33f140cdca504d"
 
--- extra-kde/kapidox/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kapidox/spec	2019-07-24 14:43:46 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kapidox-$VER.tar.xz"
-CHKSUM="sha256::4445a89fa9ca34fdec46c296997fedaaf7e76b3e80bc1526cd948e167d341c01"
+CHKSUM="sha256::88cb8b8637e1c6e93d908d3384253d8d99efaa28ef7ba9c7a3089544e1f114dc"
--- extra-kde/kwayland/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kwayland/spec	2019-07-24 14:43:51 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kwayland-$VER.tar.xz"
-CHKSUM="sha256::39ae3b71b452e555ef2715a0b1305d44e52b1913882981f8e2e036f481f3e580"
+CHKSUM="sha256::a0645594c2dc7c121b11eecf1c1d31448d110d13dab200caadddb7e1d836d905"
--- extra-kde/kcoreaddons/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kcoreaddons/spec	2019-07-24 14:43:57 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kcoreaddons-$VER.tar.xz"
-CHKSUM="sha256::9df3668696b2e722196ce93fff86bae8ea343c13c008bf2ad6f8d3cff15b5aca"
+CHKSUM="sha256::c78e41596e65fec08b1f7b212dd584cb61b5a9fe89c1bb312e06238fac3e137d"
--- extra-kde/ki18n/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/ki18n/spec	2019-07-24 14:44:04 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/ki18n-$VER.tar.xz"
-CHKSUM="sha256::76d3306da302250c0b78441d1ddeb8cbae803958b75f30514f181566d0efce2c"
+CHKSUM="sha256::b94598e6cb40019c9db403cb392b647d798692899467883f2f9419e1de5782c3"
--- extra-kde/kdoctools/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kdoctools/spec	2019-07-24 14:44:10 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kdoctools-$VER.tar.xz"
-CHKSUM="sha256::ec732210181d42a7dd61d66954f302afd4444e436081212a85e0e35c405c8e09"
+CHKSUM="sha256::b231ea9857c5c1335fbf1a63907311a136651d93af4a4143e0b4caad833ae834"
--- extra-kde/kauth/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kauth/spec	2019-07-24 14:44:15 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kauth-$VER.tar.xz"
-CHKSUM="sha256::e7d9cbbc108df1c2d41038785967ed9a9489987462ba5f1f1a9d270ae9e1093c"
+CHKSUM="sha256::50a2a2eb90d7529bb6fc4ddd77e37ef5b25c612b18487cc63d8d086d5ec28916"
--- extra-kde/kcodecs/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kcodecs/spec	2019-07-24 14:44:21 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kcodecs-$VER.tar.xz"
-CHKSUM="sha256::768875bba3612a47419ac600107e045467feb55a4bf6ceb1cf51085c85c7a13a"
+CHKSUM="sha256::cf8d5e5d47cbee31ea45faa1ca4d6043d3f8d9cbff2e5d40ce749a0b2cec50df"
--- extra-kde/kconfig/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kconfig/spec	2019-07-24 14:44:26 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kconfig-$VER.tar.xz"
-CHKSUM="sha256::28aac471e6960636afbcbd525b88b4f186180e23a76c140f48d256a7a9e1254a"
+CHKSUM="sha256::541acceead9ca516e3562109e7e94351ce378c234bea968c8dfde78f00559233"
--- extra-kde/kwidgetsaddons/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kwidgetsaddons/spec	2019-07-24 14:44:37 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kwidgetsaddons-$VER.tar.xz"
-CHKSUM="sha256::69e22fd0a685386a0c00dd4b3865471affbd8a947bba5b2e70bd8eb42a82f3bb"
+CHKSUM="sha256::2e74d3897290c43871836a55ce0dbf4e6f3563f97016827d7ffccf0e5342a39d"
--- extra-kde/kcompletion/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kcompletion/spec	2019-07-24 14:44:42 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kcompletion-$VER.tar.xz"
-CHKSUM="sha256::23d758648c44090b34d5852f45323a5efd7592a99dd260c4d59b97a809bdc0e5"
+CHKSUM="sha256::d5afb075661dfe2da5e300bd73dee89f92736f39859be711f186310bed245f7c"
--- extra-kde/kwindowsystem/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kwindowsystem/spec	2019-07-24 14:44:53 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kwindowsystem-$VER.tar.xz"
-CHKSUM="sha256::ecb39fcbe48e24c0a3b9c70b9b08c3c95ade446759b850bf5079df7669f56936"
+CHKSUM="sha256::3bdce23b75b4b1d1eea69fb2035ff5be94ea2c1b7bb3d4e529a9b1740b106315"
--- extra-kde/kcrash/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kcrash/spec	2019-07-24 14:44:57 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kcrash-$VER.tar.xz"
-CHKSUM="sha256::581e8215212a584886e8d4b6d5c1aac277590bcb343697d8b3653e1c22f6b1e0"
+CHKSUM="sha256::4f45a5ac0620a0fbf4b983ac3d3e68053a3a0f94bae86076f3bc3ec82ecac299"
--- extra-kde/kdbusaddons/spec	2019-07-24 14:43:26 +0000
+++ extra-kde/kdbusaddons/spec	2019-07-24 14:45:02 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kdbusaddons-$VER.tar.xz"
-CHKSUM="sha256::b930918ccece1d80aaf0eb770d4e17e5926d7da3a97d1f8506f28829d267b972"
+CHKSUM="sha256::996a9c41d6290e9520dfaa88a97a476f29b2992a135024ab0ad2bc707b0df881"
