--- extra-kde/kguiaddons/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kguiaddons/spec	2019-07-24 13:52:21 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kguiaddons-$VER.tar.xz"
-CHKSUM="sha256::b190b601679bb5f6994f85d452b0a182283dbb872b688447449aefe09388e23e"
+CHKSUM="sha256::228dc1715a35979072381658ca244795e989ea941e703559de398fdea793c548"
--- extra-kde/kconfigwidgets/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kconfigwidgets/spec	2019-07-24 13:52:26 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kconfigwidgets-$VER.tar.xz"
-CHKSUM="sha256::c5cfcd8ba356563d83c11a2ebe78d163336df2ef3f97faed0f487ddc5f67a503"
+CHKSUM="sha256::74ad4cf5b2858955e7966bb27f7b7ec38c6ddc7c95aa10ba356--- extra-kde/attica5/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/attica5/spec	2019-07-24 13:52:57 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/attica-$VER.tar.xz"
-CHKSUM="sha256::abbaeed8ec027735d6504d31387e4115400457e14b8983a780debd769e4766c5"
+CHKSUM="sha256::6658a886950ab87d779991b4c39beeff250f5aff64c71ee98b5c472a219ac6de"
--- extra-kde/kglobalaccel/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kglobalaccel/spec	2019-07-24 13:54:26 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kglobalaccel-$VER.tar.xz"
-CHKSUM="sha256::c3a7c694f257d9331c8cb6f398532e89e9cb962f2224524fe69937c17edcaf5e"
+CHKSUM="sha256::660f2b930fd2ea4ad935f37d4e26253a7537ff30cf665ac57a41087dd4c2e805"
--- extra-kde/kitemviews/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kitemviews/spec	2019-07-24 13:54:47 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kitemviews-$VER.tar.xz"
-CHKSUM="sha256::a84036e6dd2974ddcfcc4d8c0f5a1eab59dafdb20f60fbd8df3d5f7fa54824ab"
+CHKSUM="sha256::e43c1479bd5a8c90ba8e396a34299e0a96143ad6cdfd1edef0be5839fb95437f"
--- extra-kde/kbookmarks/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kbookmarks/spec	2019-07-24 14:08:30 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kbookmarks-$VER.tar.xz"
-CHKSUM="sha256::02011a370959e9fe91db499ba8b12284f2274471760b1bd466587aa9c0c1953b"
+CHKSUM="sha256::a2b77f0b084211badce2eb90c19e00caf078ab59512c326ed1c90f4a58de0d9f"
19ac6de"
--- extra-kde/kservice/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kservice/spec	2019-07-24 13:52:50 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kservice-$VER.tar.xz"
-CHKSUM="sha256::dbc67e2b0cdc2616e39842a1a9a7ab0ab99a96a083de98b6bf9abff8d4454cc3"
+CHKSUM="sha256::2d891e2781da5a562515ae09f5fb9ebd85ce397631cb217de11cb998e79516f8"
--- extra-kde/kglobalaccel/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kglobalaccel/spec	2019-07-24 13:52:55 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kglobalaccel-$VER.tar.xz"
-CHKSUM="sha256::c3a7c694f257d9331c8cb6f398532e89e9cb962f2224524fe69937c17edcaf5e"
+CHKSUM="sha256::660f2b930fd2ea4ad935f37d4e26253a7537ff30cf665ac57a41087dd4c2e805"
--- extra-kde/kitemviews/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kitemviews/spec	2019-07-24 13:53:00 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kitemviews-$VER.tar.xz"
-CHKSUM="sha256::a84036e6dd2974ddcfcc4d8c0f5a1eab59dafdb20f60fbd8df3d5f7fa54824ab"
+CHKSUM="sha256::e43c1479bd5a8c90ba8e396a34299e0a96143ad6cdfd1edef0be5839fb95437f"
--- extra-kde/kiconthemes/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kiconthemes/spec	2019-07-24 13:53:05 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kiconthemes-$VER.tar.xz"
-CHKSUM="sha256::0ae1525aafc9083218a1b0540510b3c65f5843a5f1bd4adf88988ba5f1dab765"
+CHKSUM="sha256::054bfdb9129ae8610cad742198e2ad047f3960f810b8153dc28ead76687d9e79"
--- extra-kde/sonnet/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/sonnet/spec	2019-07-24 13:53:10 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/sonnet-$VER.tar.xz"
-CHKSUM="sha256::3c6539eb6e50a01ec0a823dd21ba8a49e1b9fb709cf91c25d95be6a48ed30b65"
+CHKSUM="sha256::f6734ac4a8dcef69f159ce0810a66ebbc2bdb8f67aac58376e8af35eb7c3dfc3"
--- extra-kde/ktextwidgets/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/ktextwidgets/spec	2019-07-24 13:53:16 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/ktextwidgets-$VER.tar.xz"
-CHKSUM="sha256::17af96f8fcc4dc67245d2d3c4d1c997dab8619a35599fb516c49f873ca4fa1ce"
+CHKSUM="sha256::8149a999ca6d2dda523d6047956ca62fd283d71356b84dfd7db6fd1582b5d43e"
--- extra-kde/kxmlgui/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kxmlgui/spec	2019-07-24 13:53:23 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kxmlgui-$VER.tar.xz"
-CHKSUM="sha256::34af5e4b23c13a92d7103fd0c6464da886ed816367ede78d1bfa19382d06eac6"
+CHKSUM="sha256::c81c2794610add13fd8019031fb40b0d002085d40ba358f6406c54f79581c149"
--- extra-kde/kbookmarks/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kbookmarks/spec	2019-07-24 13:53:28 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kbookmarks-$VER.tar.xz"
-CHKSUM="sha256::02011a370959e9fe91db499ba8b12284f2274471760b1bd466587aa9c0c1953b"
+CHKSUM="sha256::a2b77f0b084211badce2eb90c19e00caf078ab59512c326ed1c90f4a58de0d9f"
--- extra-kde/solid/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/solid/spec	2019-07-24 13:53:33 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/solid-$VER.tar.xz"
-CHKSUM="sha256::841d90346cdc51214076cf26357701781d8d534dd209d92768f306e281e46e9e"
+CHKSUM="sha256::bf3dbf22dc63946cf83cd69882f0d5fb8bf938c90252b85c849ef644caebac8a"
--- extra-kde/kjobwidgets/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kjobwidgets/spec	2019-07-24 13:53:38 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kjobwidgets-$VER.tar.xz"
-CHKSUM="sha256::3ea6001724c82e2158f6ee3719f7b4974f271b056661fac037b39ce2338d04b6"
+CHKSUM="sha256::b9d2a044a17eff6ced2cf6c4bd06661a0d64cbeea2b18248cdac4f969ea69353"
--- extra-kde/knotifications/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/knotifications/spec	2019-07-24 13:53:43 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/knotifications-$VER.tar.xz"
-CHKSUM="sha256::44ac6a710a0d89947c4f1412a8b324d89db18b9f1179ff98e22c61070e12ce16"
+CHKSUM="sha256::fbed417c689b8abf08e1f29df5094cab55b193ce03483658dfb8cdbea3f2d928"
--- extra-kde/kwallet/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kwallet/spec	2019-07-24 13:53:49 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kwallet-$VER.tar.xz"
-CHKSUM="sha256::d2957ca5e0539a4f7689373ded3bbe642cb48bb37f76b6e5f2dac499f0ec260a"
+CHKSUM="sha256::2ac2186dd7d004b6740cb14e7651ddcb7d74390acb382c9d55874f3a03a8111d"
--- extra-kde/kio/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kio/spec	2019-07-24 13:54:03 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kio-$VER.tar.xz"
-CHKSUM="sha256::d6fb0bdae9454cd67de19806e338fdcb72e8678a27e95fad3626491d8b1a4cd4"
+CHKSUM="sha256::be433fcf665679671a2e320556c39b708ed0789271e76ebb6142df058260238d"
--- extra-kde/kinit/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kinit/spec	2019-07-24 13:54:08 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kinit-$VER.tar.xz"
-CHKSUM="sha256::473f6ba85f99d0d907fdb8105935854ec9b687191e0391a3e66348e5ad5519c6"
+CHKSUM="sha256::c8c109e220e8eb812a09d3474439b0f6f6af4d25861322a9494fc206da51afdc"
--- extra-kde/kpackage/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kpackage/spec	2019-07-24 13:54:12 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kpackage-$VER.tar.xz"
-CHKSUM="sha256::59fe9c9ec38ec9afb0381921ad72593772d28273ad85469df20083731b05f90c"
+CHKSUM="sha256::18d951c2ae16cfe2765dfec3ff42ce88f4b716b242294690efe8ff727395406d"
--- extra-kde/kdeclarative/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kdeclarative/spec	2019-07-24 13:54:18 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kdeclarative-$VER.tar.xz"
-CHKSUM="sha256::799c21af97dfbd25d844ff4839461a7ce29bd7e8ec04e00edf43c8bc3eee906a"
+CHKSUM="sha256::bbfd1438752866352e4fc7cb69b616c7413eb0b765f73777aa52d5063e2eefb4"
--- extra-kde/kparts/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kparts/spec	2019-07-24 13:54:23 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kparts-$VER.tar.xz"
-CHKSUM="sha256::1ff0d7ffc327f29c250d9c13c782a772b630d6146de68fe5fd1c38575a5645ee"
+CHKSUM="sha256::6df0c2bf4481716e68d57b0e2040e9c2f0ee66e81309ba594d868026ea9f5ac6"
--- extra-kde/kplotting/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kplotting/spec	2019-07-24 13:54:29 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kplotting-$VER.tar.xz"
-CHKSUM="sha256::39e44138af405f8e07eb91d660447a770394bf8fae166634a386e873761c33c3"
+CHKSUM="sha256::75e8c21c2417a65cb423d11e4abd28a93a9cf9e3b2f01695cdedf9ee4b2fa7a0"
--- extra-kde/kdesignerplugin/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kdesignerplugin/spec	2019-07-24 13:54:34 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kdesignerplugin-$VER.tar.xz"
-CHKSUM="sha256::9bf43498f631dbf1c2bc15af7f6e6494d979223cf2a34ec9f95f58a7be57d816"
+CHKSUM="sha256::9472332c5a8f7c27d6c581a4f0f4f837d82e2780c1f9b25f2f034afd1d250607"
--- extra-kde/kpty/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kpty/spec	2019-07-24 13:54:39 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kpty-$VER.tar.xz"
-CHKSUM="sha256::f310c89854a10000dabc3e67b227bf6b28b1511c87968a7cb81a9a9f8d22f7b7"
+CHKSUM="sha256::757daec3b77395cdf390b35419972641b2ca18c986b39b98ccf8fb3384547338"
--- extra-kde/kunitconversion/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kunitconversion/spec	2019-07-24 13:54:45 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kunitconversion-$VER.tar.xz"
-CHKSUM="sha256::848472340fb24fdb57c0ea60bc4bafcb2b3a26d0eede21a2e8b39e730d6bc5cd"
+CHKSUM="sha256::0e377f67b6977787861aba971167aedfaa0e4c023609d505ea2d57357c082a79"
--- extra-kde/threadweaver/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/threadweaver/spec	2019-07-24 13:54:54 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/threadweaver-$VER.tar.xz"
-CHKSUM="sha256::bd300ce0dc7d8b796fdcc257b9d27eb3f35cdcdef18611a146d3c11f3690dbbf"
+CHKSUM="sha256::8bf6520e006dadacc82735242500c73810ff3ed4e3053c8352724c9d45f4a122"
--- extra-kde/kcmutils/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kcmutils/spec	2019-07-24 13:55:00 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kcmutils-$VER.tar.xz"
-CHKSUM="sha256::7cf6a81818745852c38558160e3888bdd2a9bcce135aee53c3df99e939633ab8"
+CHKSUM="sha256::22226ca10caedc6021b966a26a45096f30837ca9eedcc97479e30f950abc7b8c"
--- extra-kde/kactivities/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kactivities/spec	2019-07-24 13:55:05 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kactivities-$VER.tar.xz"
-CHKSUM="sha256::e76a4d44ca6e7ce2273d8c3b133fdd12e32367aeffd66098ec46e7b28c491750"
+CHKSUM="sha256::7703b894c25a576a87c201d41899e793c3c2f89eaa57c1ab71266ae950091883"
--- extra-kde/kactivities-stats/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kactivities-stats/spec	2019-07-24 13:55:11 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kactivities-stats-$VER.tar.xz"
-CHKSUM="sha256::6ed0f91dd1dee3db2b0ac8e9a344eb32a881cf243dbf34c0d95888955b7277ec"
+CHKSUM="sha256::f7374d1d2fe94bae935796193a62e46ee1963a39e11b183cc0a40baedb973788"
--- extra-kde/kirigami2/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kirigami2/spec	2019-07-24 13:55:16 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kirigami2-$VER.tar.xz"
-CHKSUM="sha256::fea01ea5d277a1d415a3b2f0fe8f7659bf345933c4fd1c78c5f92fab40c97e5d"
+CHKSUM="sha256::f016481d393041513dda11345e9ee84723587fa4be9f17391ed1a5868477e153"
--- extra-kde/plasma-framework/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/plasma-framework/spec	2019-07-24 13:55:29 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/plasma-framework-$VER.tar.xz"
-CHKSUM="sha256::0accc21e8dd47531821d9a4a339fb790d3695fe12868d8c92e1b00564e028943"
+CHKSUM="sha256::96c93ea54d1ce29f930e707ab4b0eafc9e31d022f9939d2bd42b2d884e450141"
--- extra-kde/kjs/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kjs/spec	2019-07-24 13:55:34 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/portingAids/kjs-$VER.tar.xz"
-CHKSUM="sha256::963e452ff3aa18ddc858fbfdcf234bdc97576b9496b871f742d90177a33974ac"
+CHKSUM="sha256::9f4beaf9d773afbd73de5a9adf1f481c9b95fb02cc7cfe8ec050625c13811355"
--- extra-kde/kfilemetadata/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kfilemetadata/spec	2019-07-24 13:55:40 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kfilemetadata-$VER.tar.xz"
-CHKSUM="sha256::269db442adef2409ccf6b2078e3b644eb504b36f84c33c2809b1ff739c68f512"
+CHKSUM="sha256::6d2ddfe0b16d73a139573dd02766825b21b958af16f51682c1f4cb816b3929f8"
--- extra-kde/bluez-qt/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/bluez-qt/spec	2019-07-24 13:55:45 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/bluez-qt-$VER.tar.xz"
-CHKSUM="sha256::d592b84dd52aeaa0a5453949aced2fa7b160830d7325382f1616e92a338017de"
+CHKSUM="sha256::328aba853ec0c034e777261d9e3ac3202ad941ffde844392e196afaed04be0d4"
--- extra-kde/knewstuff/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/knewstuff/spec	2019-07-24 13:55:53 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/knewstuff-$VER.tar.xz"
-CHKSUM="sha256::3249464773b90c03178fb4be3c874644629f611cef72c3ea89d9ed200a501164"
+CHKSUM="sha256::64744c6775c839d9d8216c4a81161df5544e113570b4cee8533b5a9b4c7b36f2"
--- extra-kde/frameworkintegration/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/frameworkintegration/spec	2019-07-24 13:56:03 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/frameworkintegration-$VER.tar.xz"
-CHKSUM="sha256::ee8d46b629a95e49fda80e50ba8863504a6c463fec887f83370b7e7977db80f5"
+CHKSUM="sha256::c82600be42b1db398acf4059bde92b3449bb6e52fd0180e66b41060e185e4872"
--- extra-kde/kded/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kded/spec	2019-07-24 13:56:07 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kded-$VER.tar.xz"
-CHKSUM="sha256::ff2a105edaec825bc9d964df8e181c0f4910432bbfbf2238d61acb17b3bebf39"
+CHKSUM="sha256::cd472a852dcb3206b219a46660558ee24b09356d7342ce39b89ab54ea160c388"
--- extra-kde/kdesu/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kdesu/spec	2019-07-24 13:56:12 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kdesu-$VER.tar.xz"
-CHKSUM="sha256::949a23837de945131042e795daaa8a1abc4d9e03cf149cc9cea78636573e8739"
+CHKSUM="sha256::3a040ae9825738d9b4953086d6c262009dcff877a4ed7e6925e3a34086007687"
--- extra-kde/kdnssd/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kdnssd/spec	2019-07-24 13:56:16 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kdnssd-$VER.tar.xz"
-CHKSUM="sha256::7e1383580f97248d8bfb6522bd230657bd002d3eb0e74a178bccc8a511e353bf"
+CHKSUM="sha256::8313b1a418559032a110f07b8816afd676770e0897d72d1d8cc353349a281964"
--- extra-kde/kemoticons/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kemoticons/spec	2019-07-24 13:56:26 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kemoticons-$VER.tar.xz"
-CHKSUM="sha256::82eeb425a0b4eebd60c743433e09bb6aadcb512dce27c5dbd7ce73fb12680b02"
+CHKSUM="sha256::60e2f227ed7094a0d4d973f7968d4d70b7c9d176dadb45c540541a3d448a5125"
--- extra-kde/kidletime/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kidletime/spec	2019-07-24 13:56:30 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kidletime-$VER.tar.xz"
-CHKSUM="sha256::252965db39e1c9f927896176b8b9683c9d7ca25cd1373a3924036f87e2222824"
+CHKSUM="sha256::e7c4ca5ca3e847c37add4c81b3d430f4c4bf0214433a234a32d766c9199adb27"
--- extra-kde/kimageformats/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kimageformats/spec	2019-07-24 13:56:35 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kimageformats-$VER.tar.xz"
-CHKSUM="sha256::a58ddeb83295d1c0305897788aa303aba6ec8d3f9663f3d175cb8337a518f0b1"
+CHKSUM="sha256::4e2fd48a2c5b7ee3af0b82bce9aaa0847e840b881bbb307c049bb8c881302635"
--- extra-kde/kitemmodels/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kitemmodels/spec	2019-07-24 13:56:41 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kitemmodels-$VER.tar.xz"
-CHKSUM="sha256::2fcf1c08766096c9e62fd40087013e7fd69900d8e5d306d392817949e15ea18e"
+CHKSUM="sha256::364799922efa153d5bd6c326a8c3311d7170356807be1e67eb4d5135ec62f5ff"
--- extra-kde/knotifyconfig/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/knotifyconfig/spec	2019-07-24 13:56:46 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/knotifyconfig-$VER.tar.xz"
-CHKSUM="sha256::b4efe3e636a542531b28841e92c732d6c778670e5f98277bd7029174ce05987f"
+CHKSUM="sha256::77e78f15226334ffaaa0469c06052cca60a9c18063140f38b9cd6dbf8de8f099"
--- extra-kde/kpeople/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kpeople/spec	2019-07-24 13:56:51 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kpeople-$VER.tar.xz"
-CHKSUM="sha256::79e6d54a1b5c3d41717cb94898ab52a5e0c6187a069cb1e35cd4930ee5348540"
+CHKSUM="sha256::ec72c0a0b8e5ee9541eb3d0dedfffc78aee39d54fec91d78adb33243a8b3bd1d"
--- extra-kde/krunner/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/krunner/spec	2019-07-24 13:56:56 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/krunner-$VER.tar.xz"
-CHKSUM="sha256::ea2176751010e19a0f4df76cfefb178441fd51d2447e0418a21dae8f4e7640bf"
+CHKSUM="sha256::e0fce95407c52d1769a38a2f95f80637bbedb039b0df702b71c5b191795fdce3"
--- extra-kde/ktexteditor/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/ktexteditor/spec	2019-07-24 13:57:07 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/ktexteditor-$VER.tar.xz"
-CHKSUM="sha256::4c88d0033775e0715ad115370a36680ede95d29c7c4b840c0e06f23ef64959a8"
+CHKSUM="sha256::7484dfaa4df868c8193e15cc760b3ce7e8e0948daaa4a1b3ca6b96550041385d"
--- extra-kde/kxmlrpcclient/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kxmlrpcclient/spec	2019-07-24 13:57:12 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/kxmlrpcclient-$VER.tar.xz"
-CHKSUM="sha256::14a300a68ac6ade06206ebfc296c5ae2aa55b6f95629f87e19fb055108bd56ae"
+CHKSUM="sha256::e5f3bef626027b3cd84873362a5c6ee26692ee99a394ef9d944741670b1ca322"
--- extra-kde/modemmanager-qt/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/modemmanager-qt/spec	2019-07-24 13:57:16 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/modemmanager-qt-$VER.tar.xz"
-CHKSUM="sha256::6cd1490efcfeddd64cc536889b9e675077d76cbce6fcebb0a216c9b8bc379df7"
+CHKSUM="sha256::461b99099376d1210dad7867627c67b2a64af908a4fca0aeed75fc36059cad4e"
--- extra-kde/networkmanager-qt/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/networkmanager-qt/spec	2019-07-24 13:57:22 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/networkmanager-qt-$VER.tar.xz"
-CHKSUM="sha256::61d212ef98c5ce6406bfa8f6e84e4bea43c0e4b7ddb4abbff1df077d701e0b5c"
+CHKSUM="sha256::fae1d2e36e0c79095cfa3d907c86c47272226a6952ab93139b750f29e9215a65"
--- extra-kde/prison5/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/prison5/spec	2019-07-24 13:57:27 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/prison-$VER.tar.xz"
-CHKSUM="sha256::1dbe7531d1ebe6adcef6b56553812ae6557d32c3a3da65be7cd7ea02be331e16"
+CHKSUM="sha256::1b8149b8965c287f7fddfd91fb1d9cf95f150dda5b70b19480452c2e855945b5"
--- extra-kde/qqc2-desktop-style/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/qqc2-desktop-style/spec	2019-07-24 13:57:31 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/qqc2-desktop-style-$VER.tar.xz"
-CHKSUM="sha256::0b00eb10646593a57aff1b1ba79ddb34f5e2a3fb5f955df1bf9b02af5ff04e21"
+CHKSUM="sha256::75e2cf793f1ba6af6592edf910412214a640366e08fea8487f20251a3e535140"
--- extra-kde/syndication/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/syndication/spec	2019-07-24 13:57:36 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/syndication-$VER.tar.xz"
-CHKSUM="sha256::09874ad643da79c538592800d307ab8f2e8785032c0d885fac8f9355ae83d972"
+CHKSUM="sha256::67bf74432d362af7e7d21984f6915b047b596a8a0ac789cbf724ea7c4f8943d9"
--- extra-kde/kdelibs4support/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kdelibs4support/spec	2019-07-24 13:57:51 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/portingAids/kdelibs4support-$VER.tar.xz"
-CHKSUM="sha256::edcc14715699f247c57303d7418014202bd0ed3b06cf9752ddd156fefeb60efa"
+CHKSUM="sha256::91b207b3d46c6bc5e97b9b89717db1ef7c4359c65bc4595794143ba78ce92a60"
--- extra-kde/khtml/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/khtml/spec	2019-07-24 13:58:02 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/portingAids/khtml-$VER.tar.xz"
-CHKSUM="sha256::7d381c2a7d15710b4ba404111135c1aadbd3fc8027ad77f153b81a5208c3acf2"
+CHKSUM="sha256::c18a5fa51cbf9f9a0a0f104e2fc0581abdb99a5bd36f8426ef129ea848414037"
--- extra-kde/kjsembed/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kjsembed/spec	2019-07-24 13:58:07 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/portingAids/kjsembed-$VER.tar.xz"
-CHKSUM="sha256::7bde7a742c05c5ba0b8faef4681dcec0a0b91ad0a3c7cc3cdd3cfa94c0c16e52"
+CHKSUM="sha256::91c8767a533ee414cd0b4789270881495a8e06482d68bda988cc5fdeb987c421"
--- extra-kde/kmediaplayer/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kmediaplayer/spec	2019-07-24 13:58:12 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/portingAids/kmediaplayer-$VER.tar.xz"
-CHKSUM="sha256::508251507d8432ad29d6cde3d5da258b9d8c14519014901c46f3060a485c982e"
+CHKSUM="sha256::c27ab1227276a65b1c98796b5bae5e62f0e5c45a904760d94f5624bd3a46949e"
--- extra-kde/kross/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kross/spec	2019-07-24 13:58:17 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/portingAids/kross-$VER.tar.xz"
-CHKSUM="sha256::bbfafe4abd2c2166f31eb36daa1849f4dc4a829c06fe85402e1fad0ce875c667"
+CHKSUM="sha256::e83f3388ca8e89057e272b3b98233f294528ed0dcb62a6edb6cf37b7a1891f45"
--- extra-kde/baloo/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/baloo/spec	2019-07-24 13:58:24 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER:0:4}/baloo-$VER.tar.xz"
-CHKSUM="sha256::9ed5ba80305815cde3ebe1fe707c35661cedba39539645e815d408fe28a41212"
+CHKSUM="sha256::d96a19ff94caf3e3b10e8d5f165e276829a42a1c70fb8451d70cb3764f06671e"
--- extra-kde/kholidays/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/kholidays/spec	2019-07-24 13:58:31 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="http://download.kde.org/stable/frameworks/${VER:0:4}/kholidays-$VER.tar.xz"
-CHKSUM="sha256::28bc4a987adac5313dcc03b50eb34d910b93f76f54c0aaa0d30b97a086a6a252"
+CHKSUM="sha256::af1954758e946b62e387e2197f128e7ae53b6054717f35bcabc674bec19b2d14"
--- extra-kde/purpose/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/purpose/spec	2019-07-24 13:58:38 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="http://download.kde.org/stable/frameworks/${VER:0:4}/purpose-$VER.tar.xz"
-CHKSUM="sha256::882aa64c55e134d1aff06d533f8c3c6f8e5886b5b782925e1aad2e0ca7e37e67"
+CHKSUM="sha256::830669315e15a3edbf91130711b604967d1dae2cabb1fa79e0d72866f05b9f0a"
--- extra-kde/oxygen-icons/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/oxygen-icons/spec	2019-07-24 14:10:07 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER%.*}/oxygen-icons5-$VER.tar.xz"
-CHKSUM="sha256::0b07c6092d46ee6359bcf101a85a55342214c11a1a4c262c82dc6fc4aa63929d"
+CHKSUM="sha256::dca378431ec83106859736938dcc81156435b71ead564ff7ddfda85311b94e5f"
--- extra-kde/breeze-icons/spec	2019-07-24 13:50:45 +0000
+++ extra-kde/breeze-icons/spec	2019-07-24 14:10:19 +0000
@@ -1,3 +1,3 @@
 VER=5.60.0
 SRCTBL="https://download.kde.org/stable/frameworks/${VER%.*}/breeze-icons-$VER.tar.xz"
-CHKSUM="sha256::7d98d6be0795cc5dd7468e3d8093541726dcc10586a91477c02879f5eb8fcf58"
+CHKSUM="sha256::99d0cbda6a38766079166633a06493929222c6443e8718c1a2db4aeb5b5f20b7"
