From 21bb51566cfa6b01c0330ce1bc1b0366caf9241f Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Sun, 7 Nov 2021 16:20:10 -0600
Subject: [PATCH 1/2] Explicit cast to int when calling Qt sizing functions

---
 src/canvaspreviewframe.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/canvaspreviewframe.py b/src/canvaspreviewframe.py
index e04611e..5e4cc6b 100644
--- a/src/canvaspreviewframe.py
+++ b/src/canvaspreviewframe.py
@@ -73,8 +73,8 @@ class CanvasPreviewFrame(QFrame):
         self.fFakeWidth  = float(realWidth) / 15
         self.fFakeHeight = float(realHeight) / 15
 
-        self.setMinimumSize(self.fFakeWidth+padding,   self.fFakeHeight+padding)
-        self.setMaximumSize(self.fFakeWidth*4+padding, self.fFakeHeight+padding)
+        self.setMinimumSize(int(self.fFakeWidth+padding),   int(self.fFakeHeight+padding))
+        self.setMaximumSize(int(self.fFakeWidth*4+padding), int(self.fFakeHeight+padding))
 
         self.fRenderTarget.setWidth(realWidth)
         self.fRenderTarget.setHeight(realHeight)
@@ -198,7 +198,7 @@ class CanvasPreviewFrame(QFrame):
 
         painter.setBrush(self.fViewBrush)
         painter.setPen(self.fViewPen)
-        painter.drawRect(self.fViewRect[iX], self.fViewRect[iY], maxWidth, maxHeight)
+        painter.drawRect(int(self.fViewRect[iX]), int(self.fViewRect[iY]), int(maxWidth), int(maxHeight))
 
         if self.fUseCustomPaint:
             event.accept()
-- 
2.30.2

