From 2a8b33da1f1f45dacdcdd2a34b18703819d13a31 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Sun, 7 Nov 2021 16:35:38 -0600
Subject: [PATCH 2/2] Pixmap Dial: include QPainterPath

---
 c++/widgets/pixmapdial.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/c++/widgets/pixmapdial.cpp b/c++/widgets/pixmapdial.cpp
index 37dcffd..7638c57 100644
--- a/c++/widgets/pixmapdial.cpp
+++ b/c++/widgets/pixmapdial.cpp
@@ -20,6 +20,7 @@
 #include <cmath>
 
 #include <QtCore/QTimer>
+#include <QtGui/QPainterPath>
 #include <QtGui/QPainter>
 #include <QtGui/QPaintEvent>
 
-- 
2.30.2

