--- libmcrypt-2.5.8.orig/Makefile.am
+++ libmcrypt-2.5.8/Makefile.am
@@ -2,7 +2,7 @@
 
 INCLUDES = $(INCLTDL)
 EXTRA_DIST = KNOWN-BUGS THANKS NEWS libmcrypt.spec.in libmcrypt.spec COPYING.LIB
-DIST_SUBDIRS = libltdl modules include lib src doc
+DIST_SUBDIRS = $(LIBLTDL_DIR) modules include lib src doc
 SUBDIRS = $(LIBLTDL_DIR) modules include lib src doc
 
 ChangeLog:
--- libmcrypt-2.5.8.orig/debian/changelog
+++ libmcrypt-2.5.8/debian/changelog
@@ -0,0 +1,198 @@
+libmcrypt (2.5.8-3.3) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Really fix #754132 
+
+ -- Wookey <wookey@debian.org>  Mon, 15 Sep 2014 08:01:17 +0000
+
+libmcrypt (2.5.8-3.2) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Fix FTBFS by using dh --with autoreconf. Closes: #754132
+
+ -- Andreas Barth <aba@ayous.org>  Mon, 25 Aug 2014 16:08:33 +0000
+
+libmcrypt (2.5.8-3.1) unstable; urgency=high
+
+  * Non-maintainer upload.
+  * High-urgency upload for RC bugfix.
+  * Add missing Replaces: from libmcrypt-dev to libmcrypt4 (<< 2.5.8-1).
+    Closes: #523310.
+
+ -- Steve Langasek <vorlon@debian.org>  Fri, 14 Aug 2009 14:50:23 -0700
+
+libmcrypt (2.5.8-3) unstable; urgency=low
+
+  * Applied the following patches from sourceforge.net's bugtracker:
+    - libmcrypt-2.5.8-prototypes.diff
+      http://sourceforge.net/tracker/?func=detail&aid=1872799&group_id=87941&atid=584895
+    - libmcrypt-rotate-mask.patch (this should fix build on 64bit archs)
+      http://sourceforge.net/tracker/?func=detail&aid=2424445&group_id=87941&atid=584895
+
+ -- RISKO Gergely <risko@debian.org>  Mon, 30 Mar 2009 01:59:02 +0300
+
+libmcrypt (2.5.8-2) unstable; urgency=low
+
+  * debian/rules: fix the clean target, which wasn't runnable after
+    extracting the source package
+  * remove unneeded libltdl-dev build dependency
+
+ -- RISKO Gergely <risko@debian.org>  Sun, 29 Mar 2009 16:53:12 +0300
+
+libmcrypt (2.5.8-1) unstable; urgency=low
+
+  * Imported Upstream version 2.5.8 (closes: #427437)
+    - fixes lib/libmcrypt.m4 quoting bug (closes: #305288)
+  * debian/rules: simplified packaging by using dh
+  * new control fields (homepage, vcs-*), watch file, debian/compat = 7
+  * include the mcrypt.3 manpage as libmcrypt.3 in the dev package
+  * enable static library in the dev package
+  * put .so and .la into the dev instead of the lib package
+    (closes: #501680)
+  * debian/copyright cleanups
+  * ${Source-Versions} -> ${binary:Version} in debian/control
+  * Depends: ${misc:Depends} in debian/control
+  * lintian source package overrides: debian/source.lintian-overrides
+  * config.sub, config.guess patch to use the versions from
+    /usr/share/misc
+  * support 40 and 80 bits long keys in cast5
+    patch sent to upstream
+    (closes: #299509)
+
+ -- RISKO Gergely <risko@debian.org>  Sat, 28 Mar 2009 22:25:40 +0200
+
+libmcrypt (2.5.7-5.1) unstable; urgency=low
+
+  * NMU upload.
+  * Fix spurious aclocal warning thanks to a patch from Alban
+    Browaeys. Closes: #305288.
+
+ -- Vincent Bernat <bernat@debian.org>  Sat, 13 Dec 2008 15:10:43 +0100
+
+libmcrypt (2.5.7-5) unstable; urgency=low
+
+  * updated the description about the new algorithms
+  * new sourceforge project url added to the copyright file
+
+ -- RISKO Gergely <risko@debian.org>  Mon, 21 Mar 2005 00:59:11 +0100
+
+libmcrypt (2.5.7-4) unstable; urgency=low
+
+  * relibtoolize for GNU/k*BSD (closes: #266323)
+
+ -- RISKO Gergely <risko@debian.org>  Sun, 26 Sep 2004 17:40:08 +0200
+
+libmcrypt (2.5.7-3) unstable; urgency=low
+
+  * adding Replaces: libmcrypt4 (<< 2.5.7-2) to libmcrypt-dev, thanks
+    to Ray (closes: #265586)
+
+ -- RISKO Gergely <risko@debian.org>  Sat, 14 Aug 2004 01:05:56 +0200
+
+libmcrypt (2.5.7-2) unstable; urgency=low
+
+  * copyright file fixes
+  * section of libmcrypt-dev is libdevel from now
+  * added code to debian/rules to automatically update config.sub,
+    config.guess
+  * added a manpage for libmcrypt-config, so can remove lintian file
+  * mcrypt.3 is a symlink to libmcrypt.3 from now (closes: #208877)
+
+ -- RISKO Gergely <risko@debian.org>  Fri, 13 Aug 2004 17:22:52 +0200
+
+libmcrypt (2.5.7-1.1) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * New upstream version.
+
+ -- Andrés Roldán <aroldan@debian.org>  Sun,  5 Oct 2003 01:03:32 +0000
+
+libmcrypt (2.5.5-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- RISKO Gergely <risko@debian.org>  Fri,  3 Jan 2003 20:07:14 +0100
+
+libmcrypt (2.5.3-2) unstable; urgency=low
+
+  * upload to main
+
+ -- RISKO Gergely <risko@debian.org>  Sun, 10 Nov 2002 18:47:36 +0100
+
+libmcrypt (2.5.3-1) unstable; urgency=low
+
+  * new upstream version
+
+ -- RISKO Gergely <risko@debian.org>  Sat, 14 Sep 2002 15:37:21 +0200
+
+libmcrypt (2.5.2-1) unstable; urgency=low
+
+  * new upstream version
+
+ -- RISKO Gergely <risko@debian.org>  Tue,  2 Jul 2002 11:50:01 +0200
+
+libmcrypt (2.5.0-1) unstable; urgency=low
+
+  * new upstream version (closes: Bug#133260)
+  * only suggests mcrypt (closes: Bug#137064)
+
+ -- RISKO Gergely <risko@debian.org>  Wed, 13 Mar 2002 19:52:30 +0100
+
+libmcrypt (2.4.22-2) unstable; urgency=low
+
+  * mcrypt_check_version added to includes for ia64
+  * .la files is now in run-time package (closes: Bug#135418)
+
+ -- RISKO Gergely <risko@debian.org>  Sun, 24 Feb 2002 23:12:24 +0100
+
+libmcrypt (2.4.22-1) unstable; urgency=low
+
+  * New upstream. (closes: Bug#131074)
+
+ -- RISKO Gergely <risko@debian.org>  Sun,  3 Feb 2002 20:58:10 +0100
+
+libmcrypt (2.4.18-1) unstable; urgency=low
+
+  * New-maintainer. (closes: Bug#123488)
+  * New upstream version. (closes: Bug#64322, Bug#98156, Bug#81919)
+
+ -- RISKO Gergely <risko@debian.org>  Sun, 16 Dec 2001 16:32:37 +0100
+
+libmcrypt (2.4.11-0.3) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * debian/control: add libltdl3-dev to libmcrypt-dev's dependencies,
+    closes: #100846.
+  * Adrian's last NMU fixed the section, closes: #64322.
+  * Updated config.{guess,sub}, closes: #98156.
+
+ -- James Troup <james@nocrew.org>  Thu, 14 Jun 2001 19:38:04 +0100
+
+libmcrypt (2.4.11-0.2) unstable; urgency=low
+
+  * NMU with permission of Raphael Bossek.
+  * Corrected the section in debian/control to non-US.
+
+ -- Adrian Bunk <bunk@fs.tum.de>  Wed, 16 May 2001 21:41:07 +0200
+
+libmcrypt (2.4.11-0.1) unstable; urgency=low
+
+  * NMU with permission of Raphael Bossek.
+  * New upstream release.
+
+ -- Adrian Bunk <bunk@fs.tum.de>  Tue, 15 May 2001 00:44:16 +0200
+
+libmcrypt (2.4.3-1.1) unstable; urgency=low
+
+  * NMU
+  * Added build dependancy on debhelper, Closes: #80556
+  * Fixed man page installation, Closes: #81919
+
+ -- Joey Hess <joeyh@debian.org>  Sat, 14 Apr 2001 21:27:50 -0700
+
+libmcrypt (2.4.3-1) unstable; urgency=low
+
+  * New upstream version.
+
+ -- Raphael Bossek <bossekr@debian.org>  Sat, 15 Apr 2000 10:45:13 +0200
+
--- libmcrypt-2.5.8.orig/debian/compat
+++ libmcrypt-2.5.8/debian/compat
@@ -0,0 +1 @@
+7
--- libmcrypt-2.5.8.orig/debian/control
+++ libmcrypt-2.5.8/debian/control
@@ -0,0 +1,48 @@
+Source: libmcrypt
+Section: libs
+Priority: optional
+Maintainer: RISKO Gergely <risko@debian.org>
+Build-Depends: debhelper (>= 7.0.50), dh-autoreconf, libltdl-dev
+Vcs-Browser: http://git.debian.org/?p=collab-maint/libmcrypt.git;a=summary
+Vcs-Git: git://git.debian.org/collab-maint/libmcrypt.git
+Homepage: http://mcrypt.sourceforge.net/
+Standards-Version: 3.8.1
+
+Package: libmcrypt4
+Section: libs
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Suggests: libmcrypt-dev, mcrypt
+Description: De-/Encryption Library
+ libmcrypt is the library which implements all the algorithms and
+ modes found in mcrypt.
+ .
+ libmcrypt supports the algorithms: BLOWFISH, TWOFISH, DES, TripleDES,
+ 3-WAY, SAFER-sk64, SAFER-sk128, SAFER+, LOKI97, GOST, RC2, RC6, MARS,
+ IDEA, RIJNDAEL-128, RIJNDAEL-192, SERPENT, RIJNDAEL-256, CAST-128
+ (known as CAST5), CAST-256, ARCFOUR, ENIGMA, PANAMA, XTEA and WAKE.
+ Block algorithms can be used in: CBC, ECB, CFB and OFB (8 bit and n
+ bit, where n is the size of the algorithm's block length).
+ .
+ More information can be found at the libmcrypt homepage
+ http://mcrypt.sourceforge.net/ .
+
+Package: libmcrypt-dev
+Section: libdevel
+Architecture: any
+Depends: libmcrypt4 (= ${binary:Version}), ${misc:Depends}
+Replaces: libmcrypt4 (<< 2.5.8-1)
+Description: De-/Encryption Library development files
+ libmcrypt is the library which implements all the algorithms and
+ modes found in mcrypt.
+ .
+ libmcrypt supports the algorithms: BLOWFISH, TWOFISH, DES, TripleDES,
+ 3-WAY, SAFER-sk64, SAFER-sk128, SAFER+, LOKI97, GOST, RC2, RC6, MARS,
+ IDEA, RIJNDAEL-128, RIJNDAEL-192, SERPENT, RIJNDAEL-256, CAST-128
+ (known as CAST5), CAST-256, ARCFOUR, ENIGMA, PANAMA, XTEA and WAKE.
+ Block algorithms can be used in: CBC, ECB, CFB and OFB (8 bit and n
+ bit, where n is the size of the algorithm's block length).
+ .
+ More information can be found at the libmcrypt homepage
+ http://mcrypt.sourceforge.net/ .
+
--- libmcrypt-2.5.8.orig/debian/copyright
+++ libmcrypt-2.5.8/debian/copyright
@@ -0,0 +1,40 @@
+Author: Nikos Mavroyanopoulos <nmav@hellug.gr>.
+
+The author says in the README file:
+Look at COPYING for license information and the individual files for more
+information. COPYING applies only for the mcrypt program and not the
+algorithms which most of them are public domain.
+
+And the attached COPYING.LIB is the GNU Library General Public License, so
+this software is released under the terms of the GNU Library General Public
+License.
+
+In the headers of the source files he says:
+/*
+ * Copyright (C) 1998,1999,2001 Nikos Mavroyanopoulos
+ *
+ * This library is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU Library General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any
+ * later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ */
+
+On Debian systems, the complete text of the GNU Library General Public License
+can be found in /usr/share/common-licenses/LGPL-2.
+
+Debian packaging:
+It was put together by Raphael Bossek <bossekr@debian.org>
+from the following sources:
+
+  ftp://mcrypt.hellug.gr/pub/mcrypt/libmcrypt
+
+Changes:
+  * Added Debian GNU/Linux packaging files.
+
+The actual project page for the upstream is http://mcrypt.sourceforge.net/.
+The actual maintainer of the package is RISKO Gergely <risko@debian.org>.
--- libmcrypt-2.5.8.orig/debian/libmcrypt-config.1
+++ libmcrypt-2.5.8/debian/libmcrypt-config.1
@@ -0,0 +1,53 @@
+.\" This manpage has been automatically generated by docbook2man 
+.\" from a DocBook document.  This tool can be found at:
+.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
+.\" Please send any bug reports, improvements, comments, patches, 
+.\" etc. to Steve Cheng <steve@ggi-project.org>.
+.TH "LIBMCRYPT-CONFIG" "1" "13 August 2004" "" ""
+.SH NAME
+libmcrypt-config \- script to get information about the installed version of      LIBMCRYPT.
+.SH SYNOPSIS
+
+\fBlibmcrypt-config\fR [ \fB --prefix\fI[=DIR]\fB \fR ] [ \fB --exec-prefix\fI[=DIR]\fB \fR ] [ \fB --libs \fR ] [ \fB --cflags \fR ] [ \fB --version \fR ]
+
+.SH "DESCRIPTION"
+.PP
+\fBlibmcrypt-config\fR is a tool that is used to determine
+the compiler and linker flags that should be used to compile and
+link programs that use \fILIBMCRYPT\fR library.
+.SH "OPTIONS"
+.TP
+\fB--prefix=PREFIX\fR
+If specified, use \fIPREFIX\fR instead
+of the installation prefix that
+\fILIBMCRYPT\fR was built with when computing
+the output for the \fB--cflags\fR and
+\fB--libs\fR options. This option is also used for
+the exec prefix if \fB--exec-prefix\fR was
+not specified. This option must be specified before any
+\fB--libs\fR or \fB--cflags\fR options.
+.TP
+\fB--exec-prefix=PREFIX\fR
+If specified, use \fIPREFIX\fR instead
+of the installation exec prefix that
+\fILIBMCRYPT\fR was built with when computing
+the output for the \fB--cflags\fR and
+\fB--libs\fR options. This option must be specified
+before any \fB--libs\fR or \fB--cflags\fR options.
+.TP
+\fB--libs\fR
+Print the linker flags that are necessary to link a
+\fILIBMCRYPT\fR program.
+.TP
+\fB--cflags\fR
+Print the compiler flags that are necessary to compile a
+\fILIBMCRYPT\fR program.
+.TP
+\fB--version\fR
+Print the currently installed version of
+\fILIBMCRYPT\fR on the standard output.
+.SH "AUTHOR"
+.PP
+This manual page was written by Christian Marillat <marillat@debian.org> for
+the Debian GNU/Linux system (but may be used by others).  Modified by RISKO
+Gergely <risko@debian.org> for libmcrpyt.
--- libmcrypt-2.5.8.orig/debian/libmcrypt-dev.docs
+++ libmcrypt-2.5.8/debian/libmcrypt-dev.docs
@@ -0,0 +1,4 @@
+README
+TODO
+doc/README.key
+doc/README.xtea
--- libmcrypt-2.5.8.orig/debian/libmcrypt-dev.examples
+++ libmcrypt-2.5.8/debian/libmcrypt-dev.examples
@@ -0,0 +1 @@
+doc/example.c
--- libmcrypt-2.5.8.orig/debian/libmcrypt-dev.install
+++ libmcrypt-2.5.8/debian/libmcrypt-dev.install
@@ -0,0 +1,6 @@
+usr/lib/libmcrypt.la
+usr/lib/libmcrypt.a
+usr/lib/libmcrypt.so
+usr/include
+usr/bin/libmcrypt-config
+usr/share/aclocal/libmcrypt.m4
--- libmcrypt-2.5.8.orig/debian/libmcrypt-dev.manpages
+++ libmcrypt-2.5.8/debian/libmcrypt-dev.manpages
@@ -0,0 +1,2 @@
+debian/tmp/usr/share/man/man3/mcrypt.3
+debian/libmcrypt-config.1
--- libmcrypt-2.5.8.orig/debian/libmcrypt4.install
+++ libmcrypt-2.5.8/debian/libmcrypt4.install
@@ -0,0 +1 @@
+usr/lib/libmcrypt.so.4*
--- libmcrypt-2.5.8.orig/debian/rules
+++ libmcrypt-2.5.8/debian/rules
@@ -0,0 +1,11 @@
+#!/usr/bin/make -f
+
+%:
+	dh "$@" --with autoreconf
+
+override_dh_installman:
+	dh_installman
+	mv debian/libmcrypt-dev/usr/share/man/man3/mcrypt.3 debian/libmcrypt-dev/usr/share/man/man3/libmcrypt.3
+
+override_dh_auto_configure:
+	dh_auto_configure -- --enable-static
--- libmcrypt-2.5.8.orig/debian/source.lintian-overrides
+++ libmcrypt-2.5.8/debian/source.lintian-overrides
@@ -0,0 +1,8 @@
+# upstream source contains these files, but in debian/rules clean they
+# got deleted, so this is OK
+libmcrypt source: configure-generated-file-in-source config.status
+libmcrypt source: configure-generated-file-in-source config.log
+
+# I want to run uscan for myself too, so I can't depend on the qa
+# redirector.  This solution works without any kind of redirectors.
+libmcrypt source: debian-watch-file-should-use-sf-redirector line 2
--- libmcrypt-2.5.8.orig/debian/watch
+++ libmcrypt-2.5.8/debian/watch
@@ -0,0 +1,2 @@
+version=3
+http://sourceforge.net/project/showfiles.php?group_id=87941&package_id=91774 .+/libmcrypt-([\d\.]+)\.tar\.gz.+
--- libmcrypt-2.5.8.orig/lib/mcrypt_modules.h
+++ libmcrypt-2.5.8/lib/mcrypt_modules.h
@@ -1,7 +1,7 @@
-#define rotl32(x,n)   (((x) << ((word32)(n))) | ((x) >> (32 - (word32)(n))))
-#define rotr32(x,n)   (((x) >> ((word32)(n))) | ((x) << (32 - (word32)(n))))
-#define rotl16(x,n)   (((x) << ((word16)(n))) | ((x) >> (16 - (word16)(n))))
-#define rotr16(x,n)   (((x) >> ((word16)(n))) | ((x) << (16 - (word16)(n))))
+#define rotl32(x,n)   (((x) << ((word32)(n & 31))) | ((x) >> (32 - (word32)(n & 31))))
+#define rotr32(x,n)   (((x) >> ((word32)(n & 31))) | ((x) << (32 - (word32)(n & 31))))
+#define rotl16(x,n)   (((x) << ((word16)(n & 15))) | ((x) >> (16 - (word16)(n & 15))))
+#define rotr16(x,n)   (((x) >> ((word16)(n & 15))) | ((x) << (16 - (word16)(n & 15))))
 
 /* Use hardware rotations.. when available */
 #ifdef swap32
--- libmcrypt-2.5.8.orig/modules/algorithms/cast-128.c
+++ libmcrypt-2.5.8/modules/algorithms/cast-128.c
@@ -317,7 +317,7 @@
 	return 16;
 }
 
-static const int key_sizes[] = { 16 };
+static const int key_sizes[] = { 5, 10, 16 };
 WIN32DLL_DEFINE const int *_mcrypt_get_supported_key_sizes(int *len)
 {
 	*len = sizeof(key_sizes)/sizeof(int);
--- libmcrypt-2.5.8.orig/modules/algorithms/des.c
+++ libmcrypt-2.5.8/modules/algorithms/des.c
@@ -35,9 +35,12 @@
 
 /* #define	NULL	0 */
 
-static void permute_ip(), permute_fp(), perminit_ip(), spinit(),
-perminit_fp();
-static word32 f();
+static void permute_ip(char *, DES_KEY *, char *);
+static void permute_fp(char *, DES_KEY *, char *);
+static void perminit_ip(DES_KEY *);
+static void spinit(DES_KEY *);
+static void perminit_fp(DES_KEY *);
+static word32 f(DES_KEY *, register word32, register char *);
 
 
 /* Tables defined in the Data Encryption Standard documents */
--- libmcrypt-2.5.8.orig/modules/algorithms/tripledes.c
+++ libmcrypt-2.5.8/modules/algorithms/tripledes.c
@@ -36,8 +36,10 @@
 
 /* #define	NULL	0 */
 
-static void permute(), perminit(), spinit();
-static word32 f();
+static void permute(char *, char[16][16][8], char *);
+static void perminit(char[16][16][8], char[64]);
+static void spinit(TRIPLEDES_KEY *, int);
+static word32 f(TRIPLEDES_KEY *, int, register word32, register char *);
 
 
 /* Tables defined in the Data Encryption Standard documents */
