diff -Naur qucs-0.0.19/Makefile.am qucs-0.0.19.nodoc/Makefile.am
--- qucs-0.0.19/Makefile.am	2017-01-22 07:13:23.000000000 -0600
+++ qucs-0.0.19.nodoc/Makefile.am	2017-08-06 05:11:02.001174104 -0500
@@ -22,16 +22,7 @@
 # Boston, MA 02110-1301, USA.
 #
 
-# will not work anytime soon.
-AM_DISTCHECK_CONFIGURE_FLAGS=--disable-doc
-
-if ENABLE_DOC
-DOC_SUBDIR = qucs-doc
-else
-DOC_SUBDIR =
-endif
-
-SUBDIRS = qucs qucs-core $(DOC_SUBDIR)
+SUBDIRS = qucs qucs-core
 
 EXTRA_DIST = bootstrap VERSION README.md NEWS.md CONTRIBUTING.md
 
@@ -104,23 +95,3 @@
 		echo removing qucs-test; \
 		rm -rf qucs-test; \
 	fi
-
-#if DO_DOXYGEN
-#-  # Publish HTML into Qucs/qucs-doxygen repo
-#-  # See: http://awestruct.org/auto-deploy-to-github-pages/
-#-  # Clone repo, put the html pages on its gh_pages branch
-#-  # Credentials taken from the secure environment variables GIT_NAME, GIT_EMAIL and GH_TOKEN
-#-  - if [[ $LINUX && $CLANG && $MASTER ]]; then
-#-      git config --global user.name ${GIT_NAME} ;
-#-      git config --global user.email ${GIT_EMAIL} ;
-#-      git clone https://github.com/Qucs/qucs-doxygen.git qucs-doxygen ;
-#-      cd qucs-doxygen ;
-#-      git checkout gh-pages ;
-#-      git rm -r * ;
-#-      mv ../qucs/contrib/doxygen/html/ qucs/ ;
-#-      mv ../qucs-doc/doxygen/html/ qucs-core/ ;
-#-      git add --all . ;
-#-      git commit -m "Auto-updating Doxygen developer documentation" ;
-#-      git push https://${GH_TOKEN}@github.com/Qucs/qucs-doxygen gh-pages ;
-#-    fi
-#endif
diff -Naur qucs-0.0.19/configure.ac qucs-0.0.19.nodoc/configure.ac
--- qucs-0.0.19/configure.ac	2017-01-22 07:13:23.000000000 -0600
+++ qucs-0.0.19.nodoc/configure.ac	2017-08-06 05:11:17.001173636 -0500
@@ -19,11 +19,6 @@
 
 # Options
 
-# hmm maybe the default should be --disable-doc
-AC_ARG_ENABLE([doc],
-  AS_HELP_STRING([--disable-doc], [Disable doc subpackage]))
-AM_CONDITIONAL([ENABLE_DOC], [test "$enable_doc" != no])
-
 AM_CONDITIONAL([QUCS_TEST], [test -f "$srcdir/qucs-test/run.py"])
 
 dnl Check for MacOSX.
@@ -46,8 +41,5 @@
 AC_CONFIG_SUBDIRS([qucs
                    qucs-core])
 
-AS_IF([test -d "$srcdir/qucs-doc"],
-	AC_CONFIG_SUBDIRS([qucs-doc])
-)
 AC_CONFIG_FILES([Makefile])
 AC_OUTPUT
