--- a/jdk/make/lib/Awt2dLibraries.gmk	2016-04-21 20:51:22.855199482 -0600
+++ b/jdk/make/lib/Awt2dLibraries.gmk	2016-04-21 20:41:28.557107120 -0600
@@ -1205,7 +1205,7 @@
     LIBSPLASHSCREEN_DIRS += $(JDK_TOPDIR)/src/macosx/native/sun/awt/splashscreen
   endif
 
-  LIBSPLASHSCREEN_CFLAGS := -DSPLASHSCREEN -DPNG_NO_MMX_CODE \
+  LIBSPLASHSCREEN_CFLAGS := -DSPLASHSCREEN -DPNG_NO_MMX_CODE -DPNG_ARM_NEON_OPT=0 \
       $(foreach dir, $(LIBSPLASHSCREEN_DIRS), -I$(dir))
 
   ifeq ($(OPENJDK_TARGET_OS), macosx)
