From 84a6a8f0d69c939143a4c242806024092c4b4dee Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 25 Nov 2024 15:09:30 +0800
Subject: [PATCH 3/8] AOSCOS: libiberty: also generate a PIC version of
 libiberty.a

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 libiberty/configure | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/libiberty/configure b/libiberty/configure
index 02fbaabe89e..21ba89fcde3 100755
--- a/libiberty/configure
+++ b/libiberty/configure
@@ -5409,10 +5409,6 @@ if [ "${enable_host_shared}" = "yes" ] || [ "${enable_host_pie}" = "yes" ]; then
   shared=yes
 fi
 
-if [ "${shared}" != "yes" ]; then
-  PICFLAG=
-fi
-
 
 NOASANFLAG=
 case " ${CFLAGS} " in
-- 
2.54.0

