From b1edcc7cbb2b90cc1cfebce7dcca0b92cf66c164 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 5 Sep 2022 16:57:54 +0200
Subject: [PATCH 14/21] linux: Test whether BT mice get de-duplicated

---
 src/linux/integration-test.py | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/linux/integration-test.py b/src/linux/integration-test.py
index 2702a29..1e48fa2 100755
--- a/src/linux/integration-test.py
+++ b/src/linux/integration-test.py
@@ -1422,6 +1422,21 @@ class Tests(dbusmock.DBusTestCase):
         self.assertEqual(self.get_dbus_display_property('WarningLevel'), UP_DEVICE_LEVEL_NONE)
         self.stop_daemon()
 
+    def test_dup_bluetooth_mouse(self):
+        '''BT mouse that also supports HID++'''
+
+        # power_supply interface
+        self._add_bt_mouse()
+
+        # BlueZ BATT service
+        alias = 'Arc Touch Mouse SE'
+        battery_level = 99
+        device_properties = {
+            'Appearance': dbus.UInt16(0x03c2, variant_level=1)
+        }
+        devs = self._add_bluez_battery_device(alias, device_properties, battery_level)
+        self.assertEqual(len(devs), 1)
+
     def test_bluetooth_mouse_reconnect(self):
         '''bluetooth mouse powerdown/reconnect'''
 
-- 
2.39.1

