From 4051aa8f80c2d2710b028d69c6d9e49b4b665c26 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 16 Nov 2022 16:14:40 +0100
Subject: [PATCH 20/21] idevice: Simplify setting the model

---
 src/linux/up-device-idevice.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/linux/up-device-idevice.c b/src/linux/up-device-idevice.c
index dad245c..cc65c44 100644
--- a/src/linux/up-device-idevice.c
+++ b/src/linux/up-device-idevice.c
@@ -193,7 +193,7 @@ up_device_idevice_coldplug (UpDevice *device)
 		      "type", kind,
 		      "serial", uuid,
 		      "vendor", g_udev_device_get_property (native, "ID_VENDOR"),
-		      "model", g_udev_device_get_property (native, "ID_MODEL"),
+		      "model", model,
 		      "power-supply", FALSE,
 		      "is-present", FALSE,
 		      "is-rechargeable", TRUE,
-- 
2.39.1

