From 73952df5f5d554797cdd963b312fd9fa987faa7b Mon Sep 17 00:00:00 2001
From: Xu Colin <colinxu2020@gmail.com>
Date: Mon, 24 Aug 2026 21:07:18 +0800
Subject: [PATCH 1/2] AOSCOS: relax SIP version requirement

Signed-off-by: Xu Colin <colinxu2020@gmail.com>
---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 5125d701..91488100 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -74,7 +74,7 @@ requires = [
     "setuptools>=70.1",
     "cython >= 3.0.10",
     "requests >= 2.26.0",
-    "sip == 6.12.0",
+    "sip >= 6.12.0, < 7",
     "typing-extensions; python_version < '3.11'",
 ]
 build-backend = "setuptools.build_meta"
-- 
2.53.0.windows.1


