From ecccd2fb822c2743ec04a949d396475688997db1 Mon Sep 17 00:00:00 2001
From: Xu Colin <colinxu2020@gmail.com>
Date: Mon, 24 Aug 2026 21:08:15 +0800
Subject: [PATCH 2/2] AOSCOS: build wxPython against system wxWidgets

Signed-off-by: Xu Colin <colinxu2020@gmail.com>
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 9bcef61c..c767de49 100644
--- a/setup.py
+++ b/setup.py
@@ -102,7 +102,7 @@ class wx_build(orig_build):
                 'message and the wxWidgets and Phoenix build steps in the future.\n')
 
             # Use the same Python that is running this script.
-            cmd = ['"{}"'.format(sys.executable), '-u', 'build.py', 'build']
+            cmd = ['"{}"'.format(sys.executable), '-u', 'build.py', 'build', '--use_syswx']
             cmd = ' '.join(cmd)
             runcmd(cmd)
 
-- 
2.53.0.windows.1


