diff --git a/gui/mistral/family.cmake b/gui/mistral/family.cmake
index c78ff6c1..072fa5cd 100644
--- a/gui/mistral/family.cmake
+++ b/gui/mistral/family.cmake
@@ -1 +1 @@
-target_include_directories(gui_mistral PRIVATE ${MISTRAL_ROOT}/libmistral ${CMAKE_BINARY_DIR}/libmistral)
+target_include_directories(gui_mistral PRIVATE ${MISTRAL_ROOT}/libmistral ${CMAKE_BINARY_DIR}/libmistral ${CMAKE_BINARY_DIR}/tools)
