From 9ab217341b8c4ceb2afbccbaa9e102f7bf42ac94 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Tue, 3 Jun 2025 05:47:00 -0700
Subject: [PATCH 1/2] mpv-settings: use /usr/bin/mpv as default path

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 .../components/playback/mpv-settings.tsx      | 20 ++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/src/renderer/features/settings/components/playback/mpv-settings.tsx b/src/renderer/features/settings/components/playback/mpv-settings.tsx
index c54e17be..d0881cfc 100644
--- a/src/renderer/features/settings/components/playback/mpv-settings.tsx
+++ b/src/renderer/features/settings/components/playback/mpv-settings.tsx
@@ -39,19 +39,21 @@ export const MpvSettings = memo(() => {
     const [mpvPath, setMpvPath] = useState(
         (localSettings?.get('mpv_path') as string | undefined) || '',
     );
+    
+    const defaultMpvPath = '/usr/bin/mpv';
 
     const handleSetMpvPath = async (clear?: boolean) => {
         if (clear) {
-            localSettings?.set('mpv_path', undefined);
-            setMpvPath('');
+            localSettings?.set('mpv_path', defaultMpvPath);
+            setMpvPath(defaultMpvPath);
             return;
         }
 
         const result = await localSettings?.openFileSelector();
 
         if (result === null) {
-            localSettings?.set('mpv_path', undefined);
-            setMpvPath('');
+            localSettings?.set('mpv_path', defaultMpvPath);
+            setMpvPath(defaultMpvPath);
             return;
         }
 
@@ -61,9 +63,13 @@ export const MpvSettings = memo(() => {
 
     useEffect(() => {
         const getMpvPath = async () => {
-            if (!localSettings) return setMpvPath('');
-            const mpvPath = (await localSettings.get('mpv_path')) as string;
-            return setMpvPath(mpvPath);
+            if (!localSettings) return setMpvPath(defaultMpvPath);
+            const mpvPath = await localSettings.get('mpv_path');
+            if (mpvPath === undefined) {
+                return setMpvPath(defaultMpvPath);
+            } else {
+                return setMpvPath(mpvPath as string);
+            }
         };
 
         getMpvPath();
-- 
2.52.0

