From 754c18f2e6361818417158acd072be6fd67a23ec Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Wed, 6 Mar 2024 21:24:14 -0800
Subject: [PATCH] mpv-settings: use /usr/bin/mpv as default path

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 .../components/playback/mpv-settings.tsx         | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/src/renderer/features/settings/components/playback/mpv-settings.tsx b/src/renderer/features/settings/components/playback/mpv-settings.tsx
index 9004a465..b2443b3f 100644
--- a/src/renderer/features/settings/components/playback/mpv-settings.tsx
+++ b/src/renderer/features/settings/components/playback/mpv-settings.tsx
@@ -80,10 +80,12 @@ export const MpvSettings = () => {
 
     const [mpvPath, setMpvPath] = useState('');
 
+    const defaultMpvPath = '/usr/bin/mpv';
+
     const handleSetMpvPath = (e: File | null) => {
         if (e === null) {
-            localSettings?.set('mpv_path', undefined);
-            setMpvPath('');
+            localSettings?.set('mpv_path', defaultMpvPath);
+            setMpvPath(defaultMpvPath);
             return;
         }
 
@@ -93,9 +95,13 @@ export const MpvSettings = () => {
 
     useEffect(() => {
         const getMpvPath = async () => {
-            if (!localSettings) return setMpvPath('');
-            const mpvPath = (await localSettings.get('mpv_path')) as string;
-            return setMpvPath(mpvPath);
+            if (!localSettings) return setMpvPath(defaultMpvPath);
+            const mpvPath = await localSettings.get('mpv_path');
+            if (mpvPath === undefined) {
+              return setMpvPath(defaultMpvPath);
+            } else {
+              return setMpvPath(mpvPath as string);
+            }
         };
 
         getMpvPath();
-- 
2.44.0

