From cc170ac2ef5e37a3bb214bca8ce7f41ec24a46df Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 12 Jul 2026 17:24:03 +0800
Subject: [PATCH] AOSCOS: common.mk: correct shared data installation path

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 common.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/common.mk b/common.mk
index 749488c2..d9bc2590 100644
--- a/common.mk
+++ b/common.mk
@@ -482,7 +482,7 @@ endif
 
 PKGMANSUBDIRS = man1 man3 man5 web
 
-PKGSUBDIRS = bin include include/netpbm lib sharedlink staticlink misc \
+PKGSUBDIRS = bin include include/netpbm lib sharedlink staticlink share/netpbm \
   $(PKGMANSUBDIRS:%=$(PKGMANDIR)/%)
 
 $(PKGSUBDIRS:%=$(PKGDIR)/%):
@@ -516,7 +516,7 @@ $(SCRIPTS:%=%_installscript): $(PKGDIR)/bin
 install.data: $(DATAFILES:%=%_installdata) $(SUBDIRS:%=%/install.data)
 
 .PHONY: $(DATAFILES:%=%_installdata) 
-$(DATAFILES:%=%_installdata): $(PKGDIR)/misc
+$(DATAFILES:%=%_installdata): $(PKGDIR)/share/netpbm
 	$(INSTALL) -c -m $(INSTALL_PERM_DATA) \
 	  $(SRCDIR)/$(SUBDIR)/$(@:%_installdata=%) $<
 
-- 
2.52.0

