diff -up chromium-99.0.4844.51/chrome/browser/component_updater/registration.cc.widevine-no-download chromium-99.0.4844.51/chrome/browser/component_updater/registration.cc
--- chromium-99.0.4844.51/chrome/browser/component_updater/registration.cc.widevine-no-download	2022-03-04 14:57:16.459599123 +0000
+++ chromium-99.0.4844.51/chrome/browser/component_updater/registration.cc	2022-03-04 14:58:39.560763939 +0000
@@ -105,10 +105,6 @@ void RegisterComponentsForUpdate() {
   RegisterMediaFoundationWidevineCdmComponent(cus);
 #endif
 
-#if BUILDFLAG(ENABLE_WIDEVINE_CDM_COMPONENT)
-  RegisterWidevineCdmComponent(cus);
-#endif  // BUILDFLAG(ENABLE_WIDEVINE_CDM_COMPONENT)
-
 #if BUILDFLAG(ENABLE_NACL) && !BUILDFLAG(IS_ANDROID)
 #if BUILDFLAG(IS_CHROMEOS_ASH)
   // PNaCl on Chrome OS is on rootfs and there is no need to download it. But
