diff -up chromium-107.0.5304.110/remoting/host/mojom/remoting_mojom_traits.h.remoting-extra-qualification chromium-107.0.5304.110/remoting/host/mojom/remoting_mojom_traits.h
--- chromium-107.0.5304.110/remoting/host/mojom/remoting_mojom_traits.h.remoting-extra-qualification	2022-11-20 19:56:00.897740616 +0100
+++ chromium-107.0.5304.110/remoting/host/mojom/remoting_mojom_traits.h	2022-11-21 09:21:47.967180137 +0100
@@ -42,7 +42,7 @@
 namespace mojo {
 
 template <>
-class mojo::StructTraits<remoting::mojom::BoolDataView, bool> {
+class StructTraits<remoting::mojom::BoolDataView, bool> {
  public:
   static bool value(bool value) { return value; }
 
@@ -53,7 +53,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::FloatDataView, float> {
+class StructTraits<remoting::mojom::FloatDataView, float> {
  public:
   static float value(float value) { return value; }
 
@@ -64,7 +64,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::Int32DataView, int32_t> {
+class StructTraits<remoting::mojom::Int32DataView, int32_t> {
  public:
   static int32_t value(int32_t value) { return value; }
 
@@ -76,7 +76,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::UInt32DataView, uint32_t> {
+class StructTraits<remoting::mojom::UInt32DataView, uint32_t> {
  public:
   static uint32_t value(uint32_t value) { return value; }
 
@@ -88,7 +88,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::DesktopCaptureOptionsDataView,
+class StructTraits<remoting::mojom::DesktopCaptureOptionsDataView,
                          ::webrtc::DesktopCaptureOptions> {
  public:
   static bool use_update_notifications(
@@ -113,7 +113,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::DesktopEnvironmentOptionsDataView,
+class StructTraits<remoting::mojom::DesktopEnvironmentOptionsDataView,
                          ::remoting::DesktopEnvironmentOptions> {
  public:
   static bool enable_curtaining(
@@ -210,7 +210,7 @@ struct EnumTraits<remoting::mojom::Deskt
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::DesktopRectDataView,
+class StructTraits<remoting::mojom::DesktopRectDataView,
                          ::webrtc::DesktopRect> {
  public:
   static int32_t left(const ::webrtc::DesktopRect& rect) { return rect.left(); }
@@ -230,7 +230,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::DesktopSizeDataView,
+class StructTraits<remoting::mojom::DesktopSizeDataView,
                          ::webrtc::DesktopSize> {
  public:
   static int32_t width(const ::webrtc::DesktopSize& size) {
@@ -246,7 +246,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::DesktopVectorDataView,
+class StructTraits<remoting::mojom::DesktopVectorDataView,
                          ::webrtc::DesktopVector> {
  public:
   static int32_t x(const ::webrtc::DesktopVector& vector) { return vector.x(); }
@@ -258,7 +258,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::MouseCursorDataView,
+class StructTraits<remoting::mojom::MouseCursorDataView,
                          ::webrtc::MouseCursor> {
  public:
   static const webrtc::DesktopSize& image_size(
@@ -511,7 +511,7 @@ struct EnumTraits<remoting::mojom::Audio
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::AudioPacketDataView,
+class StructTraits<remoting::mojom::AudioPacketDataView,
                          ::std::unique_ptr<::remoting::AudioPacket>> {
  public:
   static int32_t timestamp(
@@ -549,7 +549,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::ClipboardEventDataView,
+class StructTraits<remoting::mojom::ClipboardEventDataView,
                          ::remoting::protocol::ClipboardEvent> {
  public:
   static const std::string& mime_type(
@@ -567,7 +567,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::UnionTraits<
+class UnionTraits<
     remoting::mojom::ReadChunkResultDataView,
     ::remoting::Result<std::vector<uint8_t>,
                        ::remoting::protocol::FileTransfer_Error>> {
@@ -606,7 +606,7 @@ class mojo::UnionTraits<
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::FileTransferErrorDataView,
+class StructTraits<remoting::mojom::FileTransferErrorDataView,
                          ::remoting::protocol::FileTransfer_Error> {
  public:
   static ::remoting::protocol::FileTransfer_Error_Type type(
@@ -745,7 +745,7 @@ class mojo::UnionTraits<
 #endif  // BUILDFLAG(IS_WIN)
 
 template <>
-class mojo::StructTraits<remoting::mojom::KeyboardLayoutDataView,
+class StructTraits<remoting::mojom::KeyboardLayoutDataView,
                          ::remoting::protocol::KeyboardLayout> {
  public:
   static const ::google::protobuf::
@@ -759,7 +759,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::UnionTraits<remoting::mojom::KeyActionDataView,
+class UnionTraits<remoting::mojom::KeyActionDataView,
                         ::remoting::protocol::KeyboardLayout_KeyAction> {
  public:
   static remoting::mojom::KeyActionDataView::Tag GetTag(
@@ -792,7 +792,7 @@ class mojo::UnionTraits<remoting::mojom:
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::KeyBehaviorDataView,
+class StructTraits<remoting::mojom::KeyBehaviorDataView,
                          ::remoting::protocol::KeyboardLayout_KeyBehavior> {
  public:
   static const ::google::protobuf::Map<
@@ -1150,7 +1150,7 @@ struct EnumTraits<remoting::mojom::Layou
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::KeyEventDataView,
+class StructTraits<remoting::mojom::KeyEventDataView,
                          ::remoting::protocol::KeyEvent> {
  public:
   static bool pressed(const ::remoting::protocol::KeyEvent& event) {
@@ -1186,7 +1186,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::MouseEventDataView,
+class StructTraits<remoting::mojom::MouseEventDataView,
                          ::remoting::protocol::MouseEvent> {
  public:
   static absl::optional<int32_t> x(
@@ -1275,7 +1275,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::ScreenResolutionDataView,
+class StructTraits<remoting::mojom::ScreenResolutionDataView,
                          ::remoting::ScreenResolution> {
  public:
   static const ::webrtc::DesktopSize& dimensions(
@@ -1293,7 +1293,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::TextEventDataView,
+class StructTraits<remoting::mojom::TextEventDataView,
                          ::remoting::protocol::TextEvent> {
  public:
   static const std::string& text(const ::remoting::protocol::TextEvent& event) {
@@ -1305,7 +1305,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::TouchEventPointDataView,
+class StructTraits<remoting::mojom::TouchEventPointDataView,
                          ::remoting::protocol::TouchEventPoint> {
  public:
   static uint32_t id(const ::remoting::protocol::TouchEventPoint& event) {
@@ -1382,7 +1382,7 @@ struct EnumTraits<remoting::mojom::Touch
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::TouchEventDataView,
+class StructTraits<remoting::mojom::TouchEventDataView,
                          ::remoting::protocol::TouchEvent> {
  public:
   static ::remoting::protocol::TouchEvent::TouchEventType event_type(
@@ -1442,7 +1442,7 @@ struct EnumTraits<remoting::mojom::Trans
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::TransportRouteDataView,
+class StructTraits<remoting::mojom::TransportRouteDataView,
                          ::remoting::protocol::TransportRoute> {
  public:
   static ::remoting::protocol::TransportRoute::RouteType type(
@@ -1589,7 +1589,7 @@ struct EnumTraits<remoting::mojom::Proto
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::VideoLayoutDataView,
+class StructTraits<remoting::mojom::VideoLayoutDataView,
                          ::remoting::protocol::VideoLayout> {
  public:
   static const ::google::protobuf::RepeatedPtrField<
@@ -1608,7 +1608,7 @@ class mojo::StructTraits<remoting::mojom
 };
 
 template <>
-class mojo::StructTraits<remoting::mojom::VideoTrackLayoutDataView,
+class StructTraits<remoting::mojom::VideoTrackLayoutDataView,
                          ::remoting::protocol::VideoTrackLayout> {
  public:
   static int64_t screen_id(
