diff -up chromium-108.0.5359.124/ui/linux/linux_ui_factory.cc.e chromium-108.0.5359.124/ui/linux/linux_ui_factory.cc
--- chromium-108.0.5359.124/ui/linux/linux_ui_factory.cc.e	2023-01-10 11:40:46.836056080 +0100
+++ chromium-108.0.5359.124/ui/linux/linux_ui_factory.cc	2023-01-10 11:41:06.355411862 +0100
@@ -93,7 +93,7 @@ LinuxUiAndTheme* GetDefaultLinuxUiAndThe
 
 }  // namespace
 
-BASE_FEATURE(kAllowQt, "AllowQt", base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kAllowQt, "AllowQt", base::FEATURE_ENABLED_BY_DEFAULT);
 
 LinuxUi* GetDefaultLinuxUi() {
   return GetDefaultLinuxUiAndTheme();
