diff -up chromium-104.0.5112.101/chrome/browser/about_flags.cc.accel-mjpeg chromium-104.0.5112.101/chrome/browser/about_flags.cc
--- chromium-104.0.5112.101/chrome/browser/about_flags.cc.accel-mjpeg	2022-08-16 23:29:08.000000000 +0000
+++ chromium-104.0.5112.101/chrome/browser/about_flags.cc	2022-08-18 14:21:36.762266571 +0000
@@ -4485,12 +4485,12 @@ const FeatureEntry kFeatureEntries[] = {
      flag_descriptions::kWebXrForceRuntimeDescription, kOsDesktop,
      MULTI_VALUE_TYPE(kWebXrForceRuntimeChoices)},
 #endif  // ENABLE_VR
-#if BUILDFLAG(IS_CHROMEOS_ASH)
+#if BUILDFLAG(IS_CHROMEOS_ASH) || BUILDFLAG(IS_LINUX)
     {"disable-accelerated-mjpeg-decode",
      flag_descriptions::kAcceleratedMjpegDecodeName,
-     flag_descriptions::kAcceleratedMjpegDecodeDescription, kOsCrOS,
+     flag_descriptions::kAcceleratedMjpegDecodeDescription, kOsCrOS | kOsLinux,
      SINGLE_DISABLE_VALUE_TYPE(switches::kDisableAcceleratedMjpegDecode)},
-#endif  // BUILDFLAG(IS_CHROMEOS_ASH)
+#endif  // BUILDFLAG(IS_CHROMEOS_ASH) || BUILDFLAG(IS_LINUX)
     {"system-keyboard-lock", flag_descriptions::kSystemKeyboardLockName,
      flag_descriptions::kSystemKeyboardLockDescription, kOsDesktop,
      FEATURE_VALUE_TYPE(features::kSystemKeyboardLock)},
diff -up chromium-104.0.5112.101/chrome/browser/flag_descriptions.cc.accel-mjpeg chromium-104.0.5112.101/chrome/browser/flag_descriptions.cc
--- chromium-104.0.5112.101/chrome/browser/flag_descriptions.cc.accel-mjpeg	2022-08-16 23:29:09.000000000 +0000
+++ chromium-104.0.5112.101/chrome/browser/flag_descriptions.cc	2022-08-18 14:25:23.609572649 +0000
@@ -4170,7 +4170,7 @@ const char kUseAngleGL[] = "OpenGL";
 
 // ChromeOS -------------------------------------------------------------------
 
-#if BUILDFLAG(IS_CHROMEOS_ASH)
+#if BUILDFLAG(IS_CHROMEOS_ASH) || BUILDFLAG(IS_LINUX)
 
 const char kAcceleratedMjpegDecodeName[] =
     "Hardware-accelerated mjpeg decode for captured frame";
@@ -4178,6 +4178,10 @@ const char kAcceleratedMjpegDecodeDescri
     "Enable hardware-accelerated mjpeg decode for captured frame where "
     "available.";
 
+#endif
+
+#if BUILDFLAG(IS_CHROMEOS_ASH)
+
 const char kAdaptiveChargingForTestingName[] =
     "Show adaptive charging notifications for testing";
 const char kAdaptiveChargingForTestingDescription[] =
diff -up chromium-104.0.5112.101/chrome/browser/flag_descriptions.h.accel-mjpeg chromium-104.0.5112.101/chrome/browser/flag_descriptions.h
--- chromium-104.0.5112.101/chrome/browser/flag_descriptions.h.accel-mjpeg	2022-08-18 14:21:36.765266469 +0000
+++ chromium-104.0.5112.101/chrome/browser/flag_descriptions.h	2022-08-18 14:26:10.469983301 +0000
@@ -2390,11 +2390,15 @@ extern const char kUseAngleGL[];
 
 // ChromeOS -------------------------------------------------------------------
 
-#if BUILDFLAG(IS_CHROMEOS_ASH)
+#if BUILDFLAG(IS_CHROMEOS_ASH) || BUILDFLAG(IS_LINUX)
 
 extern const char kAcceleratedMjpegDecodeName[];
 extern const char kAcceleratedMjpegDecodeDescription[];
 
+#endif // BUILDFLAG(IS_CHROMEOS_ASH) || BUILDFLAG(IS_LINUX)
+
+#if BUILDFLAG(IS_CHROMEOS_ASH)
+
 extern const char kAdaptiveChargingName[];
 extern const char kAdaptiveChargingDescription[];
 
