--- chromium-121.0.6167.75.orig/mojo/public/cpp/bindings/type_converter.h
+++ chromium-121.0.6167.75/mojo/public/cpp/bindings/type_converter.h
@@ -95,16 +95,19 @@
   return TypeConverter<T, std::remove_cv_t<U>*>::Convert(obj);
 }
 
+template <class T> constexpr T *clang_to_address(T *P) { return P; }
+template <class T> auto clang_to_address(const T &P) -> decltype(P.operator->()) { return P.operator->(); }
+
 template <typename T, typename U>
   requires(!std::is_pointer_v<U>)
 inline T ConvertTo(const U& obj) {
   if constexpr (requires {
                   {
-                    mojo::ConvertTo<T>(std::to_address(obj))
+                    mojo::ConvertTo<T>(clang_to_address(obj))
 
                   } -> std::same_as<T>;
                 }) {
-    return mojo::ConvertTo<T>(std::to_address(obj));
+    return mojo::ConvertTo<T>(clang_to_address(obj));
   } else {
     return TypeConverter<T, U>::Convert(obj);
   }
