From 42d0584bbcc2b567d1a64986a88f147df55bb794 Mon Sep 17 00:00:00 2001
From: meta-legend <110937181+meta-legend@users.noreply.github.com>
Date: Sun, 5 Jul 2026 00:16:37 -0500
Subject: [PATCH 2/3] Fix floor1 inverse_db_table indexing

---
 stb_vorbis.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/stb_vorbis.c b/stb_vorbis.c
index 3e5c250..077a8d3 100644
--- a/stb_vorbis.c
+++ b/stb_vorbis.c
@@ -23,7 +23,7 @@
 //    Dougall Johnson (sample-exact seeking)
 //
 // Bugfix/warning contributors:
-//    Terje Mathisen     Niklas Frykholm     Andy Hill
+//    Terje Mathisen     Niklas Frykholm     Andy Hill 
 //    Casey Muratori     John Bolton         Gargaj
 //    Laurent Gomila     Marc LeBlanc        Ronny Chevalier
 //    Bernhard Wodo      Evan Balster        github:alxprd
@@ -34,6 +34,7 @@
 //    github:audinowho   Dougall Johnson     David Reid
 //    github:Clownacy    Pedro J. Estebanez  Remi Verschelde
 //    AnthoFoxo          github:morlat       Gabriel Ravier
+//    github:meta-legend  
 //
 // Partial history:
 //    1.22    - 2021-07-11 - various small fixes
@@ -3100,7 +3101,7 @@ static int do_floor(vorb *f, Mapping *map, int i, int n, float *target, YTYPE *f
       if (lx < n2) {
          // optimization of: draw_line(target, lx,ly, n,ly, n2);
          for (j=lx; j < n2; ++j)
-            LINE_OP(target[j], inverse_db_table[ly]);
+            LINE_OP(target[j], inverse_db_table[ly&255]);
          CHECK(f);
       }
    }
-- 
2.55.0

