From 1de0bf9890f61dc205954954b1e46eb66f99288b Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Mon, 17 Nov 2025 09:42:27 +0800
Subject: [PATCH 3/3] use 64 bit stbsp__uintptr on loongarch64

---
 stb_sprintf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/stb_sprintf.h b/stb_sprintf.h
index ca432a6..0086ad7 100644
--- a/stb_sprintf.h
+++ b/stb_sprintf.h
@@ -230,7 +230,7 @@ STBSP__PUBLICDEC void STB_SPRINTF_DECORATE(set_separators)(char comma, char peri
 #define stbsp__uint16 unsigned short
 
 #ifndef stbsp__uintptr
-#if defined(__ppc64__) || defined(__powerpc64__) || defined(__aarch64__) || defined(_M_X64) || defined(__x86_64__) || defined(__x86_64) || defined(__s390x__)
+#if defined(__ppc64__) || defined(__powerpc64__) || defined(__aarch64__) || defined(_M_X64) || defined(__x86_64__) || defined(__x86_64) || defined(__s390x__) || defined(__loongarch_lp64)
 #define stbsp__uintptr stbsp__uint64
 #else
 #define stbsp__uintptr stbsp__uint32
-- 
2.55.0

