diff -Naur recommonmark/recommonmark/parser.py recommonmark.commonmark/recommonmark/parser.py
--- recommonmark/recommonmark/parser.py	2018-09-29 02:58:42.185349287 +0000
+++ recommonmark.commonmark/recommonmark/parser.py	2018-09-29 02:59:49.888249227 +0000
@@ -6,7 +6,7 @@
 from docutils import parsers, nodes
 from sphinx import addnodes
 
-from CommonMark import Parser
+from commonmark import Parser
 
 from warnings import warn
 
