From 28139816d62b8444ca61a000a87c71e59fef104d Mon Sep 17 00:00:00 2001
From: Akira TAGOH <akira@tagoh.org>
Date: Mon, 5 Jun 2017 21:00:36 +0900
Subject: Fix the build failure when srcdir != builddir and have gperf 3.1 or
 later installed


diff --git a/src/Makefile.am b/src/Makefile.am
index bbbfeda..bd5ac1f 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -104,7 +104,11 @@ noinst_PROGRAMS = fcarch
 ../fc-lang/fclang.h:
 	cd ../fc-lang && $(MAKE) $(AM_MAKEFLAGS) fclang.h
 
-fcobjshash.gperf: fcobjshash.gperf.h fcobjs.h
+fcobjshash.gperf: Makefile stamp-fcobjshash.gperf
+	-@$(RM) stamp-fcobjshash.gperf
+	@$(MAKE) stamp-fcobjshash.gperf
+	@touch -r stamp-fcobjshash.gperf $@
+stamp-fcobjshash.gperf: fcobjshash.gperf.h fcobjs.h
 	$(AM_V_GEN) $(CPP) -I$(top_srcdir) $< | \
 	$(SED) 's/^ *//;s/ *, */,/' | \
 	$(GREP) '^[^#]' | \
@@ -113,7 +117,7 @@ fcobjshash.gperf: fcobjshash.gperf.h fcobjs.h
 		/CUT_OUT_END/ { no_write=0; next; }; \
 		{ if (!no_write) print; next; }; \
 	' - > $@.tmp && \
-	mv -f $@.tmp $@ || ( $(RM) $@.tmp && false )
+	mv -f $@.tmp fcobjshash.gperf && touch $@ || ( $(RM) $@.tmp && false )
 
 fcobjshash.h: Makefile fcobjshash.gperf
 	$(AM_V_GEN) $(GPERF) -m 100 fcobjshash.gperf > $@.tmp && \
@@ -198,7 +202,7 @@ stamp-fcstdint: $(top_builddir)/config.status
 	@touch $@
 
 CLEANFILES = $(ALIAS_FILES) fontconfig.def
-DISTCLEANFILES = stamp-fcstdint fcstdint.h
+DISTCLEANFILES = stamp-fcstdint fcstdint.h stamp-fcobjshash.gperf
 
 fontconfig.def: $(PUBLIC_FILES) $(PUBLIC_FT_FILES)
 	echo Generating $@
-- 
cgit v0.10.2

