diff -Naur a/conf.d/60-latin.conf b/conf.d/60-latin.conf
--- a/conf.d/60-latin.conf	2014-11-05 21:13:44.000000000 -0700
+++ b/conf.d/60-latin.conf	2016-03-11 22:56:32.202010128 -0700
@@ -4,6 +4,7 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
+			<family>Noto Serif</family>
 			<family>Bitstream Vera Serif</family>
 			<family>DejaVu Serif</family>
 			<family>Times New Roman</family>
@@ -17,6 +18,7 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
+			<family>Noto Sans</family>
 			<family>Bitstream Vera Sans</family>
 			<family>DejaVu Sans</family>
 			<family>Verdana</family>
@@ -34,6 +36,7 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
+			<family>Noto Mono</family>
 			<family>Bitstream Vera Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
 			<family>Inconsolata</family>
@@ -49,7 +52,7 @@
 <!--
   Fantasy faces
  -->
- 	<alias>
+	<alias>
 		<family>fantasy</family>
 		<prefer>
 			<family>Impact</family>
