diff -u -p -r xf86-video-fbturbo-0.4.0.orig/src/backing_store_tuner.c xf86-video-fbturbo-0.4.0/src/backing_store_tuner.c
--- xf86-video-fbturbo-0.4.0.orig/src/backing_store_tuner.c	2021-11-12 13:30:19.847975437 +0000
+++ xf86-video-fbturbo-0.4.0/src/backing_store_tuner.c	2021-11-12 13:33:23.871445322 +0000
@@ -109,7 +109,7 @@ xPostValidateTree(WindowPtr pWin, Window
     private->PostValidateTreeNestingLevel++;

     /* Disable backing store for the focus window */
-    if (!private->ForceBackingStore && focusWin->backStorage) {
+    if (!private->ForceBackingStore) {
         DebugMsg("Disable backing store for the focus window 0x%x\n",
                  (unsigned int)focusWin->drawable.id);
         pScreen->backingStoreSupport = Always;
@@ -125,8 +125,7 @@ xPostValidateTree(WindowPtr pWin, Window
     /* And enable backing store for all the other children of root */
     curWin = pScreen->root->firstChild;
     while (curWin) {
-        if (!curWin->backStorage && (private->ForceBackingStore ||
-                                     curWin != focusWin)) {
+        if (private->ForceBackingStore || curWin != focusWin) {
             DebugMsg("Enable backing store for window 0x%x\n",
                      (unsigned int)curWin->drawable.id);
             pScreen->backingStoreSupport = Always;
@@ -158,7 +157,7 @@ xReparentWindow(WindowPtr pWin, WindowPt
     }

     /* We only want backing store set for direct children of root */
-    if (pPriorParent == pScreen->root && pWin->backStorage) {
+    if (pPriorParent == pScreen->root) {
         DebugMsg("Reparent window 0x%x from root, disabling backing store\n",
                  (unsigned int)pWin->drawable.id);
         pScreen->backingStoreSupport = Always;

