diff -Naur fuse-3.10.1.orig/util/install_helper.sh fuse-3.10.1.mod/util/install_helper.sh
--- fuse-3.10.1.orig/util/install_helper.sh	2020-12-13 15:42:07.149454208 -0600
+++ fuse-3.10.1.mod/util/install_helper.sh	2020-12-13 15:41:07.686333784 -0600
@@ -30,10 +30,12 @@
     chown root:root "${DESTDIR}${bindir}/fusermount3"
     chmod u+s "${DESTDIR}${bindir}/fusermount3"
 
+    if false; then
     if test ! -e "${DESTDIR}/dev/fuse"; then
         mkdir -p "${DESTDIR}/dev"
         mknod "${DESTDIR}/dev/fuse" -m 0666 c 10 229
     fi
+    fi
 fi
 
 install -D -m 644 "${MESON_SOURCE_ROOT}/util/udev.rules" \
