diff --git a/libs/schema/schema-ast.y b/libs/schema/schema-ast.y
index a53dafc13..4ff9b12eb 100644
--- a/libs/schema/schema-ast.y
+++ b/libs/schema/schema-ast.y
@@ -40,7 +40,7 @@
         return p_sb . NextToken ( p_token -> tok );
     }
 
-    void AST_error ( ctx_t ctx, void * p_parser, ASTBuilder & p_builder, ParseTreeScanner & p_sb, const char * p_msg )
+    void schema_error ( ctx_t ctx, void * p_parser, ASTBuilder & p_builder, ParseTreeScanner & p_sb, const char * p_msg )
     {
         INTERNAL_ERROR ( xcUnexpected, "%s: %s", p_sb . GetSourceFileName (), p_msg );
     }
@@ -53,6 +53,11 @@
 %parse-param { ASTBuilder& p_builder }
 %param { ParseTreeScanner& p_sb }
 
+%code {
+#undef yyerror
+#define yyerror schema_error
+}
+
  /*%define api.value.type { const Token* }*/
 
 %union {

