Author: Mathieu Malaterre
Last-Update: 2022-02-21 09:28:30 +0100
Origin: https://lists.debian.org/debian-med/2022/02/msg00094.html
Description: Set linker flags properly

--- a/libs/CMakeLists.txt
+++ b/libs/CMakeLists.txt
@@ -30,7 +30,6 @@ add_subdirectory(align)
 add_subdirectory(axf) # sra tools dependency
 add_subdirectory(blast)
 add_subdirectory(cloud)
-add_subdirectory(ext)
 add_subdirectory(kapp)
 add_subdirectory(kdb)
 add_subdirectory(kfc)
@@ -54,7 +53,7 @@ if ( SINGLE_CONFIG OR _NCBIVDB_CFG_PACKA
     #
     # Install header files
     set( HEADERS align  cloud  insdc  kdb  kfg  klib  kproc   ktst  os     search  sra   vfs
-    cc     csra2  ext  kapp   kfc  kfs  kns   krypto  ncbi  vdb  )
+    cc     csra2   kapp   kfc  kfs  kns   krypto  ncbi  vdb  )
 
     foreach( h IN ITEMS ${HEADERS} )
         install( DIRECTORY ${CMAKE_SOURCE_DIR}/interfaces/${h} DESTINATION ${INST_INCDIR} )
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -38,13 +38,12 @@ if( WIN32 )
     set( COMMON_LIBS_WRITE ${COMMON_LINK_LIBRARIES} ncbi-wvdb Ws2_32 Crypt32 ${MBEDTLS_LIBS} )
 else()
     link_libraries( ${COMMON_LINK_LIBRARIES} )
-    set( COMMON_LIBS_READ  ncbi-vdb  dl pthread ${MBEDTLS_LIBS} )
-    set( COMMON_LIBS_WRITE ncbi-wvdb dl pthread ${MBEDTLS_LIBS} )
+    set( COMMON_LIBS_READ  ncbi-vdb  dl pthread z bz2 zstd ${MBEDTLS_LIBS} )
+    set( COMMON_LIBS_WRITE ncbi-wvdb dl pthread z bz2 zstd ${MBEDTLS_LIBS} )
 endif()
 
 add_subdirectory(ktst)
 add_subdirectory(cc)
-add_subdirectory(ext)
 add_subdirectory(judy_test)
 add_subdirectory(klib)
 add_subdirectory(kfs)
--- a/libs/ncbi-vdb/CMakeLists.posix.txt
+++ b/libs/ncbi-vdb/CMakeLists.posix.txt
@@ -50,13 +50,13 @@ else()
 
 add_library( ncbi-vdb STATIC "${VDB_LIB_RD}" )
 ExportStatic( ncbi-vdb true )
-ExportShared( ncbi-vdb true "m;dl;pthread;${MBEDTLS_LIBS}")
+ExportShared( ncbi-vdb true "m;dl;pthread;z;bz2;zstd;${MBEDTLS_LIBS}")
 
 ################################
 
 add_library( ncbi-wvdb STATIC "${VDB_LIB_RDWR}" )
 ExportStatic( ncbi-wvdb true )
-ExportShared( ncbi-wvdb true "m;dl;pthread;${MBEDTLS_LIBS}")
+ExportShared( ncbi-wvdb true "m;dl;pthread;z;bz2;zstd;${MBEDTLS_LIBS}")
 
 if( RUN_SANITIZER_TESTS )
     set( asan_defs "-fsanitize=address" )
--- a/libs/ncbi-vdb/CMakeLists.txt
+++ b/libs/ncbi-vdb/CMakeLists.txt
@@ -47,9 +47,9 @@ else()
     list( APPEND VDB_LIB_CMN_TARGETS mbedx509 mbedtls mbedcrypto )
 endif()
 
-if(NOT _NCBIVDB_CFG_PACKAGING)
-    list( APPEND VDB_LIB_CMN_TARGETS bz2 z zstd )
-endif()
+# if(NOT _NCBIVDB_CFG_PACKAGING)
+#     list( APPEND VDB_LIB_CMN_TARGETS bz2 z zstd )
+# endif()
 
 set( VDB_LIB_RD_TARGETS
     ${VDB_LIB_CMN_TARGETS}
