Description: Use cmake mechanism for SONAME
Author: Mathieu Malaterre <malat@debian.org>
Last-Update: 2022-02-24

--- a/build/common.cmake
+++ b/build/common.cmake
@@ -137,20 +137,21 @@ function(MakeLinksShared target name ins
             set( LIBSUFFIX ".${SHLX}.${VERSION}" )
             set( MAJLIBSUFFIX ".${SHLX}.${MAJVERS}" )
         endif()
-        add_custom_command(TARGET ${target}
-            POST_BUILD
-            COMMAND rm -f lib${name}${LIBSUFFIX}
-            COMMAND mv lib${name}.${SHLX} lib${name}${LIBSUFFIX}
-            COMMAND ln -f -s lib${name}${LIBSUFFIX} lib${name}${MAJLIBSUFFIX}
-            COMMAND ln -f -s lib${name}${MAJLIBSUFFIX} lib${name}.${SHLX}
-            WORKING_DIRECTORY ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
-        )
+        set_target_properties(${target} PROPERTIES VERSION ${VERSION} SOVERSION ${MAJVERS})
+        #add_custom_command(TARGET ${target}
+        #    POST_BUILD
+        #    COMMAND rm -f lib${name}${LIBSUFFIX}
+        #    COMMAND mv lib${name}.${SHLX} lib${name}${LIBSUFFIX}
+        #    COMMAND ln -f -s lib${name}${LIBSUFFIX} lib${name}${MAJLIBSUFFIX}
+        #    COMMAND ln -f -s lib${name}${MAJLIBSUFFIX} lib${name}.${SHLX}
+        #    WORKING_DIRECTORY ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
+        #)
 
-        set_property(
-            TARGET    ${target}
-            APPEND
-            PROPERTY ADDITIONAL_CLEAN_FILES "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/lib${name}${LIBSUFFIX};${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/lib${name}${MAJLIBSUFFIX};${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/lib${name}.${SHLX}"
-        )
+        #set_property(
+        #    TARGET    ${target}
+        #    APPEND
+        #    PROPERTY ADDITIONAL_CLEAN_FILES "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/lib${name}${LIBSUFFIX};${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/lib${name}${MAJLIBSUFFIX};${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/lib${name}.${SHLX}"
+        #)
 
         if ( ${install} )
             install( PROGRAMS  ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/lib${name}${LIBSUFFIX}
