--- a/test/vdb/CMakeLists.txt
+++ b/test/vdb/CMakeLists.txt
@@ -66,7 +66,9 @@ AddExecutableTest( Test_VDB_META_CMP_COP
 # Slow tests
 AddExecutableTest( SlowTest_VDB_tablecursor "test-tablecursor" "${COMMON_LIBS_READ}" )
 AddExecutableTest( SlowTest_VDB_vdb-slow "test-vdb-slow" "${COMMON_LIBS_READ}" )
+if (${BITS} GREATER 32)
 AddExecutableTest( SlowTest_VDB_wvdb-slow "test-wvdb-slow" "${COMMON_LIBS_WRITE};${ADDITIONAL_LIBS}" )
+endif()
 
 # TODO: make sure runs are not cached in the user repository when running tests
 
