From 7ec4a42459b090b23a772d6a568f910f380e44db Mon Sep 17 00:00:00 2001
From: YOSHIOKA Takuma <nop_thread@nops.red>
Date: Mon, 20 May 2024 06:56:28 +0900
Subject: [PATCH] Fix build failure with gcc-14

---
 libskk/file-dict.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libskk/file-dict.vala b/libskk/file-dict.vala
index ebaa74f..0f5fae0 100644
--- a/libskk/file-dict.vala
+++ b/libskk/file-dict.vala
@@ -70,7 +70,7 @@ namespace Skk {
         // Skip until the first occurrence of line.  This moves offset
         // at the beginning of the next line.
         bool read_until (ref long offset, string line) {
-            return_val_if_fail (offset < mmap.length, null);
+            return_val_if_fail (offset < mmap.length, false);
             while (offset + line.length < mmap.length) {
                 char *p = ((char *)mmap.memory + offset);
                 if (*p == '\n' &&
-- 
2.47.0

From 97b8bea049bbacf4fee90d86388718089490d687 Mon Sep 17 00:00:00 2001
From: Daiki Ueno <dueno@redhat.com>
Date: Fri, 8 Nov 2024 09:41:35 +0900
Subject: [PATCH] fep: Avoid Vala generics to work around build issues with GCC
 14
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Otherwise, there will be -Wint-conversion warnings:

 fep.c:232:91: error: initialization of ‘void *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
   232 | static const ClientEntry CLIENT_input_mode_labels[5] = {{SKK_INPUT_MODE_HIRAGANA, "あ"}, {SKK_INPUT_MODE_KATAKANA, "ア"}, ...};
       |                                                                                           ^~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Daiki Ueno <dueno@redhat.com>
---
 tools/fep.vala | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/tools/fep.vala b/tools/fep.vala
index 92fb984..5782e5c 100644
--- a/tools/fep.vala
+++ b/tools/fep.vala
@@ -268,12 +268,14 @@ class Client : Fep.GClient {
         return retval;
     }
 
-    struct Entry<K,V> {
-        K key;
-        V value;
+    // We can't use Entry<InputMode,*> here because of Vala bug:
+    // https://bugzilla.gnome.org/show_bug.cgi?id=684262
+    struct Entry {
+        Skk.InputMode key;
+        string value;
     }
 
-    static const Entry<Skk.InputMode,string>[] input_mode_labels = {
+    static const Entry[] input_mode_labels = {
         { Skk.InputMode.HIRAGANA, "あ" },
         { Skk.InputMode.KATAKANA, "ア" },
         { Skk.InputMode.HANKAKU_KATAKANA, "_ｱ" },
-- 
2.47.0

