diff -Naur bumblebee-3.2.1/configure.ac bumblebee-3.2.1-gid-change/configure.ac
--- bumblebee-3.2.1/configure.ac	2013-04-26 15:49:03.022174178 +0000
+++ bumblebee-3.2.1-gid-change/configure.ac	2015-12-25 03:15:04.655453092 +0000
@@ -26,7 +26,7 @@
 # Don't be stupid by using values like "path/$(nuke_machine)", it gets expanded!
 AC_DEFINE_SUBST(CONF_XDISP, ":8", [X screen number])
 AC_DEFINE_SUBST(CONF_SOCKPATH, "/var/run/bumblebee.socket", [socket path])
-AC_DEFINE_SUBST(CONF_GID, "bumblebee", [group to use when setting GID])
+AC_DEFINE_SUBST(CONF_GID, "video", [group to use when setting GID])
 AC_DEFINE_SUBST(CONF_KEEPONEXIT, "false", [stop secondary X on exit of last optirun executable])
 AC_DEFINE_SUBST(CONF_FALLBACKSTART, "false", [make optirun start applications normally if secondary is unavailable])
 AC_DEFINE_SUBST(CONF_VGLCOMPRESS, "proxy", [vglclient transport method])
