From 2875344cdf2562376199552afe9e98edd1dbb681 Mon Sep 17 00:00:00 2001
From: ilikara <3435193369@qq.com>
Date: Wed, 24 Sep 2025 17:41:53 +0800
Subject: [PATCH 6/6] AOSCOS: fix build with GCC>=14

Signed-off-by: ilikara <3435193369@qq.com>
---
 src/util/pathname.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/util/pathname.hpp b/src/util/pathname.hpp
index c30df0a83..c7d1c7ca0 100644
--- a/src/util/pathname.hpp
+++ b/src/util/pathname.hpp
@@ -19,6 +19,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 /** Simple class to allow a distinction of paths that refer to the
  *  filesystem and paths that refer to the datadir, it also hides
-- 
2.51.0

