diff -Naur ceph-15.2.8/src/test/CMakeLists.txt ceph-15.2.8-new/src/test/CMakeLists.txt
--- ceph-15.2.8/src/test/CMakeLists.txt        2020-12-16 09:29:50.000000000 -0800
+++ ceph-15.2.8-new/src/test/CMakeLists.txt    2021-02-05 20:00:48.012811334 -0800
@@ -3,6 +3,10 @@
 set(UNITTEST_LIBS GMock::Main GMock::GMock GTest::GTest ${CMAKE_THREAD_LIBS_INIT} 
     ${GSSAPI_LIBRARIES} ${OPENLDAP_LIBRARIES} ${CMAKE_DL_LIBS})
 
+if(Boost_VERSION VERSION_GREATER_EQUAL 1.74)
+  add_definitions(-DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT)
+endif()
+
 add_library(unit-main OBJECT unit.cc)
 target_include_directories(unit-main PRIVATE
   $<TARGET_PROPERTY:GTest::GTest,INTERFACE_INCLUDE_DIRECTORIES>)
