diff -Naur a/common/autoconf/generated-configure.sh b/common/autoconf/generated-configure.sh
--- a/common/autoconf/generated-configure.sh	2016-04-11 15:54:34.000000000 -0600
+++ b/common/autoconf/generated-configure.sh	2016-04-20 23:36:14.951354165 -0600
@@ -16163,17 +16163,8 @@
 
   # Maximum amount of heap memory.
   # Maximum stack size.
-  if test "x$BOOT_JDK_BITS" = x32; then
-    JVM_MAX_HEAP=1100M
-    STACK_SIZE=768
-  else
-    # Running Javac on a JVM on a 64-bit machine, takes more space since 64-bit
-    # pointers are used. Apparently, we need to increase the heap and stack
-    # space for the jvm. More specifically, when running javac to build huge
-    # jdk batch
-    JVM_MAX_HEAP=1600M
-    STACK_SIZE=1536
-  fi
+  JVM_MAX_HEAP=1600M
+  STACK_SIZE=1536
 
   $ECHO "Check if jvm arg is ok: -Xmx$JVM_MAX_HEAP" >&5
   $ECHO "Command: $JAVA -Xmx$JVM_MAX_HEAP -version" >&5
