diff --git a/onnxruntime/core/flatbuffers/flatbuffers_utils.cc b/onnxruntime/core/flatbuffers/flatbuffers_utils.cc
index 42dff12..ee22264 100644
--- a/onnxruntime/core/flatbuffers/flatbuffers_utils.cc
+++ b/onnxruntime/core/flatbuffers/flatbuffers_utils.cc
@@ -200,7 +200,6 @@ static Status LoadTensorShapeOrtFormat(const fbs::Shape& fbs_shape, TensorShapeP
     dims->Reserve(fbs_dims->size());
     for (const auto fbs_dim : *fbs_dims) {
       ORT_RETURN_IF(nullptr == fbs_dim, "Null entry in dimensions. Invalid ORT format model.");
-      TensorShapeProto_Dimension dim;
       ORT_RETURN_IF_ERROR(LoadTensorDimensionOrtFormat(*fbs_dim, *dims->Add()));
     }
   }
diff --git a/onnxruntime/core/framework/graph_partitioner.cc b/onnxruntime/core/framework/graph_partitioner.cc
index 2bebab9..b1d7ca7 100644
--- a/onnxruntime/core/framework/graph_partitioner.cc
+++ b/onnxruntime/core/framework/graph_partitioner.cc
@@ -1121,7 +1121,7 @@ static Status CreateEpContextModel(const ExecutionProviders& execution_providers
 
     AllocatorPtr allocator = output_buffer_holder->buffer_allocator;
     IAllocatorUniquePtr<void> buffer = IAllocator::MakeUniquePtr<void>(allocator, buffer_size);
-    model_proto.SerializeToArray(buffer.get(), static_cast<int>(buffer_size));
+    static_cast<void>(model_proto.SerializeToArray(buffer.get(), static_cast<int>(buffer_size)));
 
     *output_buffer_holder->buffer_size_ptr = buffer_size;
     *output_buffer_holder->buffer_ptr = buffer.release();
@@ -1134,7 +1134,7 @@ static Status CreateEpContextModel(const ExecutionProviders& execution_providers
     auto out_stream_buf = std::make_unique<epctx::OutStreamBuf>(*output_write_func_holder);
     std::ostream out_stream(out_stream_buf.get());
 
-    model_proto.SerializeToOstream(&out_stream);
+    static_cast<void>(model_proto.SerializeToOstream(&out_stream));
     out_stream.flush();
     ORT_RETURN_IF_ERROR(out_stream_buf->GetStatus());
   } else {
diff --git a/onnxruntime/core/framework/tensorprotoutils.cc b/onnxruntime/core/framework/tensorprotoutils.cc
index 3e928af..4ef0058 100644
--- a/onnxruntime/core/framework/tensorprotoutils.cc
+++ b/onnxruntime/core/framework/tensorprotoutils.cc
@@ -2093,14 +2093,18 @@ static Status CopySparseData(const std::string& name,
                           indices_elements * sizeof(int32_t));
         ORT_RETURN_IF_ERROR(UnpackInitializerData(indices, model_path, unpack_buffer));
         auto int32_span = ReinterpretAsSpan<const int32_t>(gsl::make_span(unpack_buffer));
-        indices_values.insert(indices_values.cend(), int32_span.begin(), int32_span.end());
+        for (const auto index : int32_span) {
+          indices_values.push_back(index);
+        }
         unpack_buffer.clear();
         unpack_buffer.shrink_to_fit();
       } else {
         ORT_RETURN_IF_NOT(indices.int32_data_size() == indices_elements,
                           "Sparse tensor: ", name, " indices int32 data size does not match expected: ",
                           indices_elements);
-        indices_values.insert(indices_values.cend(), indices.int32_data().cbegin(), indices.int32_data().cend());
+        for (const auto index : indices.int32_data()) {
+          indices_values.push_back(index);
+        }
       }
       indices_data = gsl::make_span(indices_values);
       break;
@@ -2112,14 +2116,18 @@ static Status CopySparseData(const std::string& name,
                           indices_elements * sizeof(int16_t));
         ORT_RETURN_IF_ERROR(UnpackInitializerData(indices, model_path, unpack_buffer));
         auto int16_span = ReinterpretAsSpan<const int16_t>(gsl::make_span(unpack_buffer));
-        indices_values.insert(indices_values.cend(), int16_span.begin(), int16_span.end());
+        for (const auto index : int16_span) {
+          indices_values.push_back(index);
+        }
         unpack_buffer.clear();
         unpack_buffer.shrink_to_fit();
       } else {
         ORT_RETURN_IF_NOT(indices.int32_data_size() == indices_elements,
                           "Sparse tensor: ", name, " indices int16 data size does not match expected: ",
                           indices_elements);
-        indices_values.insert(indices_values.cend(), indices.int32_data().cbegin(), indices.int32_data().cend());
+        for (const auto index : indices.int32_data()) {
+          indices_values.push_back(index);
+        }
       }
       indices_data = gsl::make_span(indices_values);
       break;
@@ -2131,14 +2139,18 @@ static Status CopySparseData(const std::string& name,
                           indices_elements * sizeof(int8_t));
         ORT_RETURN_IF_ERROR(UnpackInitializerData(indices, model_path, unpack_buffer));
         auto int8_span = ReinterpretAsSpan<const int8_t>(gsl::make_span(unpack_buffer));
-        indices_values.insert(indices_values.cend(), int8_span.begin(), int8_span.end());
+        for (const auto index : int8_span) {
+          indices_values.push_back(index);
+        }
         unpack_buffer.clear();
         unpack_buffer.shrink_to_fit();
       } else {
         ORT_RETURN_IF_NOT(indices.int32_data_size() == indices_elements,
                           "Sparse tensor: ", name, " indices int8 data size does not match expected: ",
                           indices_elements);
-        indices_values.insert(indices_values.cend(), indices.int32_data().cbegin(), indices.int32_data().cend());
+        for (const auto index : indices.int32_data()) {
+          indices_values.push_back(index);
+        }
       }
       indices_data = gsl::make_span(indices_values);
       break;
diff --git a/onnxruntime/core/graph/contrib_ops/bert_defs.cc b/onnxruntime/core/graph/contrib_ops/bert_defs.cc
index 1209446..ed6a3fd 100644
--- a/onnxruntime/core/graph/contrib_ops/bert_defs.cc
+++ b/onnxruntime/core/graph/contrib_ops/bert_defs.cc
@@ -1405,7 +1405,6 @@ void PagedAttentionTypeAndShapeInference(ONNX_NAMESPACE::InferenceContext& ctx)
     }
 
     if (ctx.hasInput(2)) {
-      ONNX_NAMESPACE::TensorShapeProto output_shape;
       propagateShapeFromInputToOutput(ctx, 0, 0);
     } else {  // packed QKV
       ONNX_NAMESPACE::TensorShapeProto output_shape;
diff --git a/onnxruntime/core/optimizer/skip_layer_norm_fusion.cc b/onnxruntime/core/optimizer/skip_layer_norm_fusion.cc
index 3727ac0..1d6095c 100644
--- a/onnxruntime/core/optimizer/skip_layer_norm_fusion.cc
+++ b/onnxruntime/core/optimizer/skip_layer_norm_fusion.cc
@@ -199,9 +199,9 @@ Status SkipLayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_le
     Format matched_format = Format::None;
 
     // Format 1
-    std::vector<graph_utils::EdgeEndToMatch> format1_parent_path{
+    std::array<graph_utils::EdgeEndToMatch, 2> format1_parent_path{{
         {0, 0, "Add", {7, 13, 14}, kOnnxDomain},
-        {0, 0, "Add", {7, 13, 14}, kOnnxDomain}};
+        {0, 0, "Add", {7, 13, 14}, kOnnxDomain}}};
 
     std::vector<const Node::EdgeEnd*> edges;
     if (graph_utils::FindPath(ln_node, true, format1_parent_path, edges, logger)) {
@@ -218,9 +218,9 @@ Status SkipLayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_le
 
     if (matched_format == Format::None) {
       // Format 2
-      std::vector<graph_utils::EdgeEndToMatch> format2_parent_path{
+      std::array<graph_utils::EdgeEndToMatch, 2> format2_parent_path{{
           {0, 0, "Add", {7, 13, 14}, kOnnxDomain},
-          {0, 1, "Add", {7, 13, 14}, kOnnxDomain}};
+          {0, 1, "Add", {7, 13, 14}, kOnnxDomain}}};
 
       if (graph_utils::FindPath(ln_node, true, format2_parent_path, edges, logger)) {
         p_add1 = const_cast<Node*>(&edges[0]->GetNode());
@@ -237,8 +237,8 @@ Status SkipLayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_le
 
     if (matched_format == Format::None) {
       // Format 3
-      std::vector<graph_utils::EdgeEndToMatch> format3_parent_path{
-          {0, 0, "Add", {7, 13, 14}, kOnnxDomain}};
+      std::array<graph_utils::EdgeEndToMatch, 1> format3_parent_path{{
+          {0, 0, "Add", {7, 13, 14}, kOnnxDomain}}};
 
       if (graph_utils::FindPath(ln_node, true, format3_parent_path, edges, logger)) {
         p_add1 = const_cast<Node*>(&edges[0]->GetNode());
diff --git a/onnxruntime/core/session/onnxruntime_c_api.cc b/onnxruntime/core/session/onnxruntime_c_api.cc
index 3f28529..428fa01 100644
--- a/onnxruntime/core/session/onnxruntime_c_api.cc
+++ b/onnxruntime/core/session/onnxruntime_c_api.cc
@@ -3549,7 +3549,6 @@ ORT_API_STATUS_IMPL(OrtApis::OpAttr_GetTensorAttributeAsOrtValue, _In_ const Ort
 
 ORT_API_STATUS_IMPL(OrtApis::OpAttr_GetType, _In_ const OrtOpAttr* attribute, _Out_ OrtOpAttrType* type) {
   API_IMPL_BEGIN
-  const auto attr = attribute->attr_proto;
   auto onnx_attr_type = attribute->attr_proto.type();
   switch (onnx_attr_type) {
     case ONNX_NAMESPACE::AttributeProto_AttributeType::AttributeProto_AttributeType_UNDEFINED: {
