diff -ruN libewf-20140816.orig/common/byte_stream.h libewf-20140816/common/byte_stream.h
--- libewf-20140816.orig/common/byte_stream.h	2023-12-03 17:03:25.000000000 +0800
+++ libewf-20140816/common/byte_stream.h	2026-06-22 10:38:38.158532314 +0800
@@ -1,7 +1,7 @@
 /*
  * Byte stream functions
  *
- * Copyright (C) 2006-2023, Joachim Metz <joachim.metz@gmail.com>
+ * Copyright (C) 2006-2026, Joachim Metz <joachim.metz@gmail.com>
  *
  * Refer to AUTHORS for acknowledgements.
  *
@@ -29,13 +29,72 @@
 extern "C" {
 #endif
 
-#define _BYTE_STREAM_HOST_IS_ENDIAN_BIG		( *((uint32_t *) "\x01\x02\x03\x04" ) == 0x01020304 )
-#define _BYTE_STREAM_HOST_IS_ENDIAN_LITTLE	( *((uint32_t *) "\x01\x02\x03\x04" ) == 0x04030201 )
-#define _BYTE_STREAM_HOST_IS_ENDIAN_MIDDLE	( *((uint32_t *) "\x01\x02\x03\x04" ) == 0x02010403 )
-
-#define _BYTE_STREAM_ENDIAN_BIG			(uint8_t) 'b'
-#define _BYTE_STREAM_ENDIAN_LITTLE		(uint8_t) 'l'
-#define _BYTE_STREAM_ENDIAN_MIDDLE		(uint8_t) 'm'
+#define _BYTE_STREAM_ENDIAN_BIG		'b'
+#define _BYTE_STREAM_ENDIAN_LITTLE	'l'
+#define _BYTE_STREAM_ENDIAN_MIDDLE	'm'
+
+#undef _BYTE_STREAM_HOST_BYTE_ORDER
+
+#if defined( __BYTE_ORDER__ ) && defined( __ORDER_LITTLE_ENDIAN__ ) && defined( __ORDER_BIG_ENDIAN__ ) && defined( __ORDER_PDP_ENDIAN__ )
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_BIG
+
+#elif __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_LITTLE
+
+#elif __BYTE_ORDER__ == __ORDER_PDP_ENDIAN__
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_MIDDLE
+#endif
+
+#elif defined( __linux__ ) || defined( __GLIBC__ )
+#include <endian.h>
+
+#if defined( __BYTE_ORDER ) && defined( __LITTLE_ENDIAN ) && defined( __BIG_ENDIAN ) && defined( __PDP_ENDIAN )
+#if __BYTE_ORDER == __BIG_ENDIAN
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_BIG
+
+#elif __BYTE_ORDER == __LITTLE_ENDIAN
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_LITTLE
+
+#elif __BYTE_ORDER == __PDP_ENDIAN
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_MIDDLE
+
+#endif
+#endif
+
+#elif defined( __APPLE__ ) || defined( __FreeBSD__ ) || defined( __OpenBSD__ ) || defined( __NetBSD__ )
+#include <machine/endian.h>
+
+#if defined( BYTE_ORDER ) && defined( LITTLE_ENDIAN ) && defined( BIG_ENDIAN )
+#if BYTE_ORDER == BIG_ENDIAN
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_BIG
+
+#elif BYTE_ORDER == LITTLE_ENDIAN
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_LITTLE
+
+#endif
+#endif
+
+#elif defined( _MSC_VER ) || defined( _WIN32 )
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_LITTLE
+
+#elif defined( __hppa__ ) || defined( __m68k__ ) || defined( __mips__ ) || defined( __powerpc__ ) || defined( __sparc__ )
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_BIG
+
+#elif defined( __i386__ ) || defined( __x86_64__ ) || defined( __arm__ ) || defined( __aarch64__ )
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_LITTLE
+
+#elif defined( __pdp11__ )
+#define _BYTE_STREAM_HOST_BYTE_ORDER _BYTE_STREAM_ENDIAN_MIDDLE
+#endif
+
+#if !defined( _BYTE_STREAM_HOST_BYTE_ORDER )
+#error "Unable to determine host byte-order"
+#endif
+
+#define _BYTE_STREAM_HOST_IS_ENDIAN_BIG		( _BYTE_STREAM_HOST_BYTE_ORDER == _BYTE_STREAM_ENDIAN_BIG )
+#define _BYTE_STREAM_HOST_IS_ENDIAN_LITTLE	( _BYTE_STREAM_HOST_BYTE_ORDER == _BYTE_STREAM_ENDIAN_LITTLE )
+#define _BYTE_STREAM_HOST_IS_ENDIAN_MIDDLE	( _BYTE_STREAM_HOST_BYTE_ORDER == _BYTE_STREAM_ENDIAN_MIDDLE )
 
 typedef union byte_stream_float32
 {
