From b529fb089fd400b00c5a98fbd6c10b85dd8983ff Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Fri, 23 Aug 2024 17:27:30 +0800
Subject: [PATCH] FROM AOSC: fix python libdir

---
 SConscript | 2 --
 1 file changed, 2 deletions(-)

diff --git a/SConscript b/SConscript
index c7eb1b6f7c4f..b11bfdf5eeb1 100644
--- a/SConscript
+++ b/SConscript
@@ -1611,8 +1611,6 @@ if not cleaning and not helping and config.env['python']:
             # follow FHS, put in /usr/local/libXX, not /usr/libXX
             # may be lib, lib32 or lib64
             python_libdir = polystr(python_libdir)
-            python_libdir = python_libdir.replace("/usr/lib",
-                                                  "/usr/local/lib")
         python_module_dir = str(python_libdir) + os.sep
         # Many systems can have a problem with the Python path
         if 'PYTHONPATH' in os.environ:

base-commit: 8501f01b257b226b9ad34190628e23838babadff
-- 
2.46.0

