From 5a523f3bd824f55e0ffa51acedf3cfedb9c944b3 Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Thu, 2 Oct 2025 08:43:28 +0800
Subject: [PATCH 1/4] Allow CMAKE_INSTALL_PREFIX to be set externally

---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ea7f41e..9b8e0ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,8 +31,9 @@ include(GNUInstallDirs)
 
 # FHS compliant paths for Linux installation
 if(NOT WIN32 AND CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
-#  set(CMAKE_INSTALL_PREFIX "/opt/${PROJECT_NAME}")
-  set(CMAKE_INSTALL_PREFIX "/usr")
+	if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  		set(CMAKE_INSTALL_PREFIX "/usr" CACHE PATH "Install path" FORCE)
+	endif()
 endif()
 
 # Pull in the rest of the pieces
-- 
2.51.0

