From 1802c3235d9d2e36522eb44ddfbd2809ecfb5372 Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Thu, 2 Oct 2025 09:37:02 +0800
Subject: [PATCH 3/4] lib_srecord build as shared library

---
 srecord/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/srecord/CMakeLists.txt b/srecord/CMakeLists.txt
index 280a4f2..765dbd4 100644
--- a/srecord/CMakeLists.txt
+++ b/srecord/CMakeLists.txt
@@ -21,7 +21,8 @@ file(GLOB_RECURSE LIB_SRECORD_SRC "*.cc")
 file(GLOB_RECURSE LIB_SRECORD_HDR "*.h")
 
 message(STATUS "gcrypt location ${LIB_GCRYPT}")
-add_library(lib_srecord STATIC ${LIB_SRECORD_SRC} ${LIB_SRECORD_HDR} ${LIB_GCRYPT})
+add_library(lib_srecord SHARED ${LIB_SRECORD_SRC} ${LIB_SRECORD_HDR} ${LIB_GCRYPT})
+target_link_libraries(lib_srecord PRIVATE gcrypt)
 
 # Install the library
 install(TARGETS lib_srecord
-- 
2.51.0

