From 98f2188d7b38995fec1b74dbba498a7bf3686fdb Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Thu, 2 Oct 2025 10:21:46 +0800
Subject: [PATCH 4/4] Allow disable install runtime dependancies

---
 etc/packaging.cmake | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/etc/packaging.cmake b/etc/packaging.cmake
index ee2a32c..6b88a43 100644
--- a/etc/packaging.cmake
+++ b/etc/packaging.cmake
@@ -32,6 +32,7 @@ FUNCTION(INSTALL_SRECORD_EXECUTABLE_TARGET target)
     )
   message(STATUS "CMAKE_INSTALL_SYSTEM_RUNTIME_DESTINATION ${CMAKE_INSTALL_SYSTEM_RUNTIME_DESTINATION}")
   message(STATUS "CMAKE_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX}")
+  if(WITH_INSTALL_RUNTIME_DEPENDANCIES)
   # Find standard library DLL's (if needed, this covers mingw clang and gcc on windows)
   cmake_path(REMOVE_FILENAME CMAKE_C_COMPILER OUTPUT_VARIABLE COMPILER_C_DIR)
   cmake_path(REMOVE_FILENAME CMAKE_CXX_COMPILER OUTPUT_VARIABLE COMPILER_CXX_DIR)
@@ -49,8 +50,8 @@ FUNCTION(INSTALL_SRECORD_EXECUTABLE_TARGET target)
     PRE_EXCLUDE_REGEXES "api-ms-" "ext-ms-"
     POST_EXCLUDE_REGEXES ".*system32/.*\\.dll"
     DIRECTORIES ${DLL_SEARCH_DIRS}
-    COMPONENT ${target}
-    )
+    COMPONENT ${target} )
+  endif()
 ENDFUNCTION()
 
 # Packaging
-- 
2.51.0

