diff -up app_main.cpp.orig app_main.cpp
--- a/app_main.cpp.orig	2012-07-05 12:10:18.377378026 -0400
+++ b/app_main.cpp	2012-07-05 12:10:24.414378093 -0400
@@ -76,14 +76,6 @@ public:
 
   int main()
   {
-    if(name1[0]==0)
-      {
-        debug<<"This is the Gfan program for computing Groebner fans and tropical varieties.\n"
-        "Use the command \"gfan list\" to view all subcommands.\n"
-        "The command \"gfan\" is deprecate for computing all Groebner bases of an ideal.\n"
-        "Please use subcommand \"gfan _bases\" instead. Awaiting input. <Ctrl>-D to end.\n";
-      }
-
     LexicographicTermOrder myOrder;
 
     PolynomialSet g=FileParser(Stdin).parsePolynomialSetWithRing();
