diff --git a/scripts/devInstall.js b/scripts/devInstall.js
index 3203e99..a24db38 100644
--- a/scripts/devInstall.js
+++ b/scripts/devInstall.js
@@ -25,9 +25,9 @@ let result;
 spawnProcess("npm", "rm", "electron")
 spawnProcess("npm", "i")
 if (process.platform === "win32"){
-    result = spawnProcess("npm", "i", "--save-dev", "--arch=ia32", "electron@9.3.1")
+    result = spawnProcess("npm", "i", "--save-dev", "--arch=ia32", "electron@9.3.5")
 } else {
-    result = spawnProcess("npm", "i", "--save-dev", "--arch=x64", "electron@9.3.1")
+    result = spawnProcess("npm", "i", "--save-dev", "--arch=x64", "electron@9.3.5")
 }
 
 let runString = "`npm test`"
@@ -35,4 +35,4 @@ let runString = "`npm test`"
 spawnProcess("node", join(PROJECT_DIRNAME,"scripts", "installSubModules.js"))
 
 
-console.log("Everything is installed. You should be able to do "+runString+" to compile everything and launch.")
\ No newline at end of file
+console.log("Everything is installed. You should be able to do "+runString+" to compile everything and launch.")
