diff --git a/Lightcord.desktop b/Lightcord.desktop
index 433aa70..d4f8fc6 100644
--- a/Lightcord.desktop
+++ b/Lightcord.desktop
@@ -1,8 +1,9 @@
 [Desktop Entry]
 Name=Lightcord
 Comment[fr_FR]=Un client Discord simple et personalisable
+Comment[zh_CN]=可灵活定制的 Discord 客户端
 Comment=A simple - customizable - Discord Client
-Exec=/opt/Lightcord/Lightcord
+Exec=/usr/bin/lightcord
 Icon=lightcord
 Terminal=false
 Type=Application
