--- a/Cargo.toml	2022-03-30 23:14:28.870879012 +0800
+++ b/Cargo.toml	2022-03-30 23:15:37.317310573 +0800
@@ -35,7 +35,7 @@
 percent-encoding = "2.1.0"
 rand = "0.8.4"
 region = "3.0.0"
-reqwest = { version = "0.11.9", default-features = false, features = ["blocking", "json", "rustls-tls-native-roots"] }
+reqwest = { version = "0.11.9", default-features = false, features = ["blocking", "json", "native-tls"] }
 rsa = "0.5.0"
 serde = { version = "1.0.136", features = ["derive"] }
 serde_json = "1.0.78"
