From ab82fd6d5e979ef29dc88358bc4071507d6ac93e Mon Sep 17 00:00:00 2001
From: Shyliuli <1658347525@qq.com>
Date: Wed, 30 Apr 2025 21:57:42 +0800
Subject: [PATCH 2/7] add install for CmakeLists.txt

---
 QtScrcpy/CMakeLists.txt | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/QtScrcpy/CMakeLists.txt b/QtScrcpy/CMakeLists.txt
index 2a151e7..74887ce 100755
--- a/QtScrcpy/CMakeLists.txt
+++ b/QtScrcpy/CMakeLists.txt
@@ -393,3 +393,14 @@ target_link_libraries(${PROJECT_NAME} PRIVATE
     ${LINK_LIBS}
     QtScrcpyCore
 )
+
+install(TARGETS QtScrcpy)
+install(FILES
+    "${CMAKE_CURRENT_SOURCE_DIR}/sndcpy/sndcpy.sh"
+    "${CMAKE_CURRENT_SOURCE_DIR}/sndcpy/sndcpy.apk"
+    DESTINATION share/QtScrcpy
+)
+install(PROGRAMS
+    "${CMAKE_CURRENT_SOURCE_DIR}/QtScrcpyCore/src/third_party/scrcpy-server"
+    DESTINATION share/QtScrcpy
+)
-- 
2.53.0

