From d010ec25b35762d84bdbabc5ae82f617531afba8 Mon Sep 17 00:00:00 2001
From: Shyliuli <1658347525@qq.com>
Date: Fri, 2 May 2025 16:01:08 +0800
Subject: [PATCH 4/7] modify the adb location in sndcpy.sh

---
 QtScrcpy/sndcpy/sndcpy.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/QtScrcpy/sndcpy/sndcpy.sh b/QtScrcpy/sndcpy/sndcpy.sh
index 5906b78..47b885c 100755
--- a/QtScrcpy/sndcpy/sndcpy.sh
+++ b/QtScrcpy/sndcpy/sndcpy.sh
@@ -3,7 +3,7 @@
 echo Begin Runing...
 SNDCPY_PORT=28200
 SNDCPY_APK=sndcpy.apk
-ADB=./adb
+ADB=adb
 
 serial=
 if [[ $# -ge 2 ]]
@@ -43,4 +43,4 @@ do
     fi
 done
 
-echo Ready playing...
\ No newline at end of file
+echo Ready playing...
-- 
2.53.0

