From 03b013c5c24edc01692613c1942ec8875616dc52 Mon Sep 17 00:00:00 2001
From: Shyliuli <1658347525@qq.com>
Date: Fri, 2 May 2025 16:13:11 +0800
Subject: [PATCH 5/7] modify the sndcpy.apk location in sndcpy.sh

---
 QtScrcpy/sndcpy/sndcpy.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/QtScrcpy/sndcpy/sndcpy.sh b/QtScrcpy/sndcpy/sndcpy.sh
index 47b885c..b0f8db0 100755
--- a/QtScrcpy/sndcpy/sndcpy.sh
+++ b/QtScrcpy/sndcpy/sndcpy.sh
@@ -2,7 +2,7 @@
 
 echo Begin Runing...
 SNDCPY_PORT=28200
-SNDCPY_APK=sndcpy.apk
+SNDCPY_APK=/usr/share/QtScrcpy/sndcpy.apk
 ADB=adb
 
 serial=
-- 
2.53.0

