From 879a0fa2c9a639582fe9c6174c6fdcde6c2724c2 Mon Sep 17 00:00:00 2001
From: Ruikai Liu <rickliu2000@outlook.com>
Date: Tue, 11 Nov 2025 09:56:02 +0800
Subject: [PATCH 6/7] Fix build with Qt 6.10

---
 QtScrcpy/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/QtScrcpy/CMakeLists.txt b/QtScrcpy/CMakeLists.txt
index 74887ce..ebef2d8 100755
--- a/QtScrcpy/CMakeLists.txt
+++ b/QtScrcpy/CMakeLists.txt
@@ -122,6 +122,10 @@ endif()
 
 find_package(Qt${QT_DESIRED_VERSION} REQUIRED COMPONENTS ${qt_required_components})
 
+if(Qt6_VERSION VERSION_GREATER_EQUAL 6.10)
+  find_package(Qt6 COMPONENTS CorePrivate GuiPrivate REQUIRED)
+endif()
+
 set(LINK_LIBS
     Qt${QT_DESIRED_VERSION}::Widgets
     Qt${QT_DESIRED_VERSION}::Network
-- 
2.53.0

