From c75e1d128474b9f3b8bfb5d928fc214702a2b762 Mon Sep 17 00:00:00 2001
From: Yanhao Mo <yanhaocs@gmail.com>
Date: Wed, 8 Aug 2018 15:34:54 +0800
Subject: [PATCH] bugfix about test cases

---
 tests/tst_GSettings.qml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tests/tst_GSettings.qml
+++ b/tests/tst_GSettings.qml
@@ -12,7 +12,7 @@
     id: settings
     schema.id: "com.ubports.gsettings.Test"
     // has to be "valueChanged" signal, not "changed"; the latter doesn't work reliably with the in-memory gsettings backend
-    onValueChanged: changes.push([key, value]);
+    onValueChanged: testCase.changes.push([key, value]);
   }
 
   SignalSpy {
