From 1eb5833dbc17fc66a558056e6b7d5b3bc381cfa1 Mon Sep 17 00:00:00 2001
From: RyogiMutsuki <me@mutsuki.ink>
Date: Sun, 31 Aug 2025 18:15:34 +0800
Subject: [PATCH] skip broken build requirements check

---
 wrappers/python/pyproject.toml | 2 --
 1 file changed, 2 deletions(-)

diff --git a/wrappers/python/pyproject.toml b/wrappers/python/pyproject.toml
index 16484fe3..4aae9747 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -3,7 +3,5 @@ requires = [
     "setuptools>=42",
     "setuptools_scm",
     "wheel",
-    "cmake>=3.15",
-    "pybind11[global]",
 ]
 build-backend = "setuptools.build_meta"
-- 
2.51.0

