--- gtk+-1.2.10/gtk/gtkentry.c.kpenter	Sat Jul 21 20:27:08 2001
+++ gtk+-1.2.10/gtk/gtkentry.c	Sat Jul 21 20:27:16 2001
@@ -1184,6 +1184,7 @@
 	}
       break;
     case GDK_Return:
+    case GDK_KP_Enter:
       return_val = TRUE;
       gtk_widget_activate (widget);
       break;
--- gtk+-1.2.10/gtk/gtktext.c.kpenter	Sat Jul 21 20:28:56 2001
+++ gtk+-1.2.10/gtk/gtktext.c	Sat Jul 21 20:29:17 2001
@@ -2055,6 +2055,7 @@
 	case GDK_Up:        scroll_int (text, -KEY_SCROLL_PIXELS); break;
 	case GDK_Down:      scroll_int (text, +KEY_SCROLL_PIXELS); break;
 	case GDK_Return:
+	case GDK_KP_Enter:
 	  if (event->state & GDK_CONTROL_MASK)
 	    gtk_signal_emit_by_name (GTK_OBJECT (text), "activate");
 	  else
@@ -2161,6 +2162,7 @@
 	  gtk_editable_insert_text (editable, "\t", 1, &position);
 	  break;
 	case GDK_Return:
+	case GDK_KP_Enter:
 	  if (event->state & GDK_CONTROL_MASK)
 	    gtk_signal_emit_by_name (GTK_OBJECT (text), "activate");
 	  else
--- gtk+-1.2.10/gtk/gtkmenushell.c.kpenter	Sat Jul 21 20:27:59 2001
+++ gtk+-1.2.10/gtk/gtkmenushell.c	Sat Jul 21 20:28:05 2001
@@ -258,6 +258,11 @@
 				GTK_TYPE_BOOL,
 				TRUE);
   gtk_binding_entry_add_signal (binding_set,
+				GDK_KP_Enter, 0,
+				"activate_current", 1,
+				GTK_TYPE_BOOL,
+				TRUE);
+  gtk_binding_entry_add_signal (binding_set,
 				GDK_space, 0,
 				"activate_current", 1,
 				GTK_TYPE_BOOL,
--- gtk+-1.2.10/gtk/gtknotebook.c.kpenter	Sat Jul 21 20:28:30 2001
+++ gtk+-1.2.10/gtk/gtknotebook.c	Sat Jul 21 20:28:38 2001
@@ -1369,6 +1369,7 @@
 	gtk_notebook_switch_focus_tab (notebook, list);
       return TRUE;
     case GDK_Return:
+    case GDK_KP_Enter:
     case GDK_space:
       gtk_notebook_page_select (GTK_NOTEBOOK (widget));
       return TRUE;
