Index: gtk/gtkrange.c
===================================================================
RCS file: /cvs/gnome/gtk+/gtk/gtkrange.c,v
retrieving revision 1.26.2.3
diff -u -p -r1.26.2.3 gtkrange.c
--- a/gtk/gtkrange.c	22 Feb 2001 20:38:14 -0000	1.26.2.3
+++ b/gtk/gtkrange.c	19 Jul 2002 15:41:58 -0000
@@ -829,7 +828,10 @@ gtk_range_expose (GtkWidget      *widget
 	    (event->area.x + event->area.width <= 
 	     widget->allocation.width - trough_border) &&
 	    (event->area.y + event->area.height <= 
-	     widget->allocation.height - trough_border)))
+	     widget->allocation.height - trough_border)) ||
+	  gtk_style_get_prop_experimental (widget->style,
+					   "GtkRange::always_draw_trough",
+					   0))
 	gtk_range_draw_trough (range);
     }
   else if (event->window == widget->window)
