--- gtk+-1.2.8/gtk/gtklabel.c.alnum	Wed Jan 24 05:56:00 2001
+++ gtk+-1.2.8/gtk/gtklabel.c	Wed Jan 24 07:44:22 2001
@@ -507,8 +507,11 @@
 static gboolean
 is_ideogram (GdkWChar wc)
 {
-  return !(gdk_iswalnum (wc) || gdk_iswspace (wc) ||
-	   gdk_iswpunct (wc) || gdk_iswcntrl (wc));
+    if (gdk_iswalpha (wc) && (!gdk_iswupper (wc) && !gdk_iswlower (wc)))
+	return TRUE;
+    
+    return !(gdk_iswspace (wc) || gdk_iswalnum (wc) ||
+	     gdk_iswpunct (wc) || gdk_iswcntrl (wc));
 }
 
 /* this needs to handle white space better. */
