https://github.com/libimobiledevice/libimobiledevice/issues/1666

diff --git a/cython/debugserver.pxi b/cython/debugserver.pxi
index a3b7d1e..0ec864c 100644
--- a/cython/debugserver.pxi
+++ b/cython/debugserver.pxi
@@ -45,11 +45,7 @@ cdef class DebugServerError(BaseError):

 # from http://stackoverflow.com/a/17511714
 # https://github.com/libimobiledevice/libimobiledevice/pull/198
-from cpython cimport PY_MAJOR_VERSION
-if PY_MAJOR_VERSION <= 2:
-    from cpython.string cimport PyString_AsString
-else:
-    from cpython.bytes cimport PyBytes_AsString as PyString_AsString
+from cpython.bytes cimport PyBytes_AsString as PyString_AsString
 cdef char ** to_cstring_array(list_str):
     if not list_str:
         return NULL
