diff -rupN qt-creator-opensource-src-3.3.1/src/rpath.pri qt-creator-opensource-src-3.3.1-new/src/rpath.pri
--- qt-creator-opensource-src-3.3.1/src/rpath.pri	2015-02-19 11:16:46.000000000 +0100
+++ qt-creator-opensource-src-3.3.1-new/src/rpath.pri	2015-02-24 19:06:30.603703530 +0100
@@ -3,9 +3,8 @@ macx {
 } else:linux-* {
     #do the rpath by hand since it's not possible to use ORIGIN in QMAKE_RPATHDIR
     # this expands to $ORIGIN (after qmake and make), it does NOT read a qmake var
-    QMAKE_RPATHDIR += \$\$ORIGIN
-    QMAKE_RPATHDIR += \$\$ORIGIN/..
-    QMAKE_RPATHDIR += \$\$ORIGIN/../$$IDE_LIBRARY_BASENAME/qtcreator
+    QMAKE_RPATHDIR += $${QTC_PREFIX}/$${IDE_LIBRARY_BASENAME}/qtcreator
+    QMAKE_RPATHDIR += $${QTC_PREFIX}/$${IDE_LIBRARY_BASENAME}/qtcreator/plugins
     IDE_PLUGIN_RPATH = $$join(QMAKE_RPATHDIR, ":")
 
     QMAKE_LFLAGS += -Wl,-z,origin \'-Wl,-rpath,$${IDE_PLUGIN_RPATH}\'
