Index: gtk+-3.14.1/gtk/gtksettings.c
===================================================================
--- gtk+-3.14.1.orig/gtk/gtksettings.c	2014-10-02 01:53:10.669684152 +0200
+++ gtk+-3.14.1/gtk/gtksettings.c	2014-10-02 01:53:10.665684097 +0200
@@ -456,7 +456,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
                                                                   P_("Fallback Icon Theme Name"),
                                                                   P_("Name of a icon theme to fall back to"),
-                                                                  NULL,
+                                                                  "gnome",
                                                                   GTK_PARAM_READWRITE | G_PARAM_DEPRECATED),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
