Description: Don't let offscreen widget do grabbing
Author: Cody Russell <bratsche@gnome.org>
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=607668
Bug-Ubuntu: https://bugs.launchpad.net/bugs/512427

Index: gtk+3.0-3.14.3/gtk/gtkmain.c
===================================================================
--- gtk+3.0-3.14.3.orig/gtk/gtkmain.c
+++ gtk+3.0-3.14.3/gtk/gtkmain.c
@@ -2060,9 +2060,14 @@ gtk_grab_add (GtkWidget *widget)
 {
   GtkWindowGroup *group;
   GtkWidget *old_grab_widget;
+  GtkWidget *toplevel;
 
   g_return_if_fail (widget != NULL);
 
+  toplevel = gtk_widget_get_toplevel (widget);
+  if (toplevel && gdk_window_get_window_type (gtk_widget_get_window (toplevel)) == GDK_WINDOW_OFFSCREEN)
+    return;
+
   if (!gtk_widget_has_grab (widget) && gtk_widget_is_sensitive (widget))
     {
       _gtk_widget_set_has_grab (widget, TRUE);
