Bug: https://bugzilla.gnome.org/show_bug.cgi?id=702371

Index: gtk+-3.14.0/testsuite/gtk/treeview.c
===================================================================
--- gtk+-3.14.0.orig/testsuite/gtk/treeview.c	2014-09-23 19:07:32.556451645 +0200
+++ gtk+-3.14.0/testsuite/gtk/treeview.c	2014-09-23 19:07:32.556451645 +0200
@@ -300,8 +300,10 @@
   g_test_add_func ("/TreeView/cursor/bug-539377", test_bug_539377);
   g_test_add_func ("/TreeView/cursor/select-collapsed_row",
                    test_select_collapsed_row);
+  /* See http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=711107
+   * and https://bugzilla.gnome.org/show_bug.cgi?id=702371
   g_test_add_func ("/TreeView/sizing/row-separator-height",
-                   test_row_separator_height);
+                   test_row_separator_height);*/
   g_test_add_func ("/TreeView/selection/count", test_selection_count);
   g_test_add_func ("/TreeView/selection/empty", test_selection_empty);
 

