Origin: vendor, Debian

Index: gtk+3.0-3.3.4/gtk/gtktreeview.c
===================================================================
--- gtk+3.0-3.3.4.orig/gtk/gtktreeview.c	2011-11-21 14:30:11.000000000 +0100
+++ gtk+3.0-3.3.4/gtk/gtktreeview.c	2011-12-07 18:33:01.585621058 +0100
@@ -10113,7 +10113,8 @@
 
       if (cursor_path)
 	{
-	  if (gtk_tree_selection_get_mode (tree_view->priv->selection) == GTK_SELECTION_MULTIPLE)
+	  if (gtk_tree_selection_get_mode (tree_view->priv->selection) == GTK_SELECTION_MULTIPLE ||
+	      gtk_tree_selection_get_mode (tree_view->priv->selection) == GTK_SELECTION_SINGLE)
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, FALSE, FALSE);
 	  else
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, TRUE, FALSE);
