Index: gtk+3.0-3.12.2/gtk/updateiconcache.c
===================================================================
--- gtk+3.0-3.12.2.orig/gtk/updateiconcache.c	2014-06-05 13:04:31.402060363 +0200
+++ gtk+3.0-3.12.2/gtk/updateiconcache.c	2014-06-05 13:04:31.398060363 +0200
@@ -676,7 +676,7 @@
 		      directories = g_list_append (directories, g_strdup (subdir));
 		    }
 		  else
-		    dir_index = 0xffff;
+		    continue;
 		}
 
 	      image = g_new0 (Image, 1);
