Index: gtk+3.0-3.14.3/gtk/gtkeventbox.c
===================================================================
--- gtk+3.0-3.14.3.orig/gtk/gtkeventbox.c
+++ gtk+3.0-3.14.3/gtk/gtkeventbox.c
@@ -411,7 +411,8 @@ gtk_event_box_realize (GtkWidget *widget
                         | GDK_BUTTON_RELEASE_MASK
                         | GDK_EXPOSURE_MASK
                         | GDK_ENTER_NOTIFY_MASK
-                        | GDK_LEAVE_NOTIFY_MASK;
+                        | GDK_LEAVE_NOTIFY_MASK
+                        | GDK_SCROLL_MASK;
 
   priv = GTK_EVENT_BOX (widget)->priv;
 
