--- a/testsuite/reftests/gtk-reftest.c
+++ b/testsuite/reftests/gtk-reftest.c
@@ -230,6 +230,20 @@
   g_free (filename);
 }
 
+static gboolean
+known_fail(const char *test_name)
+{
+  char *filename = get_test_file (test_name, ".ui.known_fail", TRUE);
+
+  if (filename)
+    {
+      g_free (filename);
+      return TRUE;
+    }
+
+  return FALSE;
+}
+
 static void
 test_ui_file (GFile *file)
 {
@@ -261,7 +275,13 @@
   if (diff_image)
     {
       save_image (diff_image, ui_file, ".diff.png");
-      g_test_fail ();
+      if (known_fail(ui_file))
+        {
+          printf("KNOWN FAILURE - %s\n", ui_file);
+          g_test_message ("KNOWN FAIL: %s", ui_file);
+        }
+      else
+        g_test_fail ();
     }
 
   remove_extra_css (provider);
