diff -Naur stackless-3.7.5-slp/Include/internal/slp_switch_ppc_unix.h stackless-3.7.5-slp-patch/Include/internal/slp_switch_ppc_unix.h
--- stackless-3.7.5-slp/Include/internal/slp_switch_ppc_unix.h	2019-10-20 02:12:19.000000000 -0500
+++ stackless-3.7.5-slp-patch/Include/internal/slp_switch_ppc_unix.h	2020-11-10 18:39:28.724538418 -0600
@@ -34,9 +34,15 @@
 
 #define SLP_STACK_MAGIC 3
 
+#ifdef __powerpc64__
 #define REGS_TO_SAVE "r13", "r14", "r15", "r16", "r17", "r18", "r19", "r20", \
-       "r21", "r22", "r23", "r24", "r25", "r26", "r27", "r28", "r29", "r31", \
+        "r21", "r22", "r23", "r24", "r25", "r26", "r27", "r28", "r29", \
+        "cr2", "cr3", "cr4"
+#else
+#define REGS_TO_SAVE "r13", "r14", "r15", "r16", "r17", "r18", "r19", "r20", \
+       "r21", "r22", "r23", "r24", "r25", "r26", "r27", "r28", "r29", "r31" \
        "cr2", "cr3", "cr4"
+#endif
 static int
 slp_switch(void)
 {
