--- valgrind/glibc-2.34567-NPTL-helgrind.supp.jj	2009-08-19 15:37:48.000000000 +0200
+++ valgrind/glibc-2.34567-NPTL-helgrind.supp	2009-10-21 16:46:31.000000000 +0200
@@ -88,6 +88,12 @@
    obj:*/lib*/libpthread-2.*so*
 }
 {
+   helgrind-glibc2X-102a
+   Helgrind:Race
+   fun:mythread_wrapper
+   obj:*vgpreload_helgrind*.so
+}
+{
    helgrind-glibc2X-103
    Helgrind:Race
    fun:pthread_cond_*@@GLIBC_2.*
