From f2f1f74c643f5af48090f6dbf10a245620cf94ce Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Sun, 4 Feb 2024 22:19:17 +0800
Subject: [PATCH 2/4] helgrind race supp

---
 glibc-2.X-helgrind.supp.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/glibc-2.X-helgrind.supp.in b/glibc-2.X-helgrind.supp.in
index 62492c9aa..f39f60a88 100644
--- a/glibc-2.X-helgrind.supp.in
+++ b/glibc-2.X-helgrind.supp.in
@@ -109,6 +109,12 @@
    fun:mythread_wrapper
    obj:@GLIBC_LIBPTHREAD_PATH@
 }
+{
+   helgrind-glibc2X-102a
+   Helgrind:Race
+   fun:mythread_wrapper
+   obj:*vgpreload_helgrind*.so
+}
 {
    helgrind-glibc2X-103
    Helgrind:Race
-- 
2.44.0

