From 65bcbcd8bc1dafa0a6e8331489d7f68f5e56b696 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Sun, 4 Feb 2024 22:19:17 +0800
Subject: [PATCH 2/5] AOSCOS: helgrind race supp

Signed-off-by: Jiajie Chen <c@jia.je>
---
 glibc-2.X-helgrind.supp.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/glibc-2.X-helgrind.supp.in b/glibc-2.X-helgrind.supp.in
index 61d4e1d72..e932f4a4c 100644
--- a/glibc-2.X-helgrind.supp.in
+++ b/glibc-2.X-helgrind.supp.in
@@ -118,6 +118,12 @@
    fun:mythread_wrapper
    obj:@GLIBC_LIBPTHREAD_PATH@
 }
+{
+   helgrind-glibc2X-102a
+   Helgrind:Race
+   fun:mythread_wrapper
+   obj:*vgpreload_helgrind*.so
+}
 {
    helgrind-glibc2X-103
    Helgrind:Race
-- 
2.52.0

