From b0f9b43a3a9f377eb79d87291ad636b26b74dba1 Mon Sep 17 00:00:00 2001
From: Feiyang Chen <chenfeiyang@loongson.cn>
Date: Tue, 22 Mar 2022 10:00:53 +0800
Subject: [PATCH v4 054/123] loongarch64: Add isel support for Ist_MBE

---
 VEX/priv/host_loongarch64_isel.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/VEX/priv/host_loongarch64_isel.c b/VEX/priv/host_loongarch64_isel.c
index ef86c38bc..ecff42e65 100644
--- a/VEX/priv/host_loongarch64_isel.c
+++ b/VEX/priv/host_loongarch64_isel.c
@@ -1044,6 +1044,22 @@ static void iselStmtCas ( ISelEnv* env, IRStmt* stmt )
    }
 }
 
+static void iselStmtMBE ( ISelEnv* env, IRStmt* stmt )
+{
+   switch (stmt->Ist.MBE.event) {
+      case Imbe_Fence:
+      case Imbe_CancelReservation:
+         addInstr(env, LOONGARCH64Instr_Bar(LAbar_DBAR, 0));
+         break;
+      case Imbe_InsnFence:
+         addInstr(env, LOONGARCH64Instr_Bar(LAbar_IBAR, 0));
+         break;
+      default:
+         vpanic("iselStmt(loongarch64): Ist_MBE");
+         break;
+   }
+}
+
 static void iselStmtExit ( ISelEnv* env, IRStmt* stmt )
 {
    if (stmt->Ist.Exit.dst->tag != Ico_U64)
@@ -1141,6 +1157,11 @@ static void iselStmt(ISelEnv* env, IRStmt* stmt)
          iselStmtCas(env, stmt);
          break;
 
+      /* --------- MEM FENCE --------- */
+      case Ist_MBE:
+         iselStmtMBE(env, stmt);
+         break;
+
       /* --------- INSTR MARK --------- */
       /* Doesn't generate any executable code ... */
       case Ist_IMark:
-- 
2.39.1

