From 070175b34b15963130230ca58869b2826453b811 Mon Sep 17 00:00:00 2001
From: Feiyang Chen <chenfeiyang@loongson.cn>
Date: Wed, 23 Mar 2022 16:28:54 +0800
Subject: [PATCH v4 080/123] loongarch64: Add support for cachegrind

---
 cachegrind/cg_arch.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/cachegrind/cg_arch.c b/cachegrind/cg_arch.c
index c1a2e960d..5442982bd 100644
--- a/cachegrind/cg_arch.c
+++ b/cachegrind/cg_arch.c
@@ -477,7 +477,10 @@ configure_caches(cache_t *I1c, cache_t *D1c, cache_t *LLc,
 
 #elif defined(VGA_loongarch64)
 
-   /* TODO */
+   // Set caches to default (for LOONGARCH64 - 3A5000)
+   *I1c = (cache_t) {    65536,  4, 64 };
+   *D1c = (cache_t) {    65536,  4, 64 };
+   *LLc = (cache_t) {   262144, 16, 64 };
 
 #else
 
-- 
2.39.1

