From e8f66d23fdbdf57e42ad17999e8e74a58256edc8 Mon Sep 17 00:00:00 2001
From: Feiyang Chen <chenfeiyang@loongson.cn>
Date: Wed, 23 Mar 2022 16:34:04 +0800
Subject: [PATCH v4 089/123] loongarch64: Add support for debuglog

---
 coregrind/m_debuglog.c | 29 +++++++++++++++++++++++++----
 1 file changed, 25 insertions(+), 4 deletions(-)

diff --git a/coregrind/m_debuglog.c b/coregrind/m_debuglog.c
index 671fc57fd..1d4f08984 100644
--- a/coregrind/m_debuglog.c
+++ b/coregrind/m_debuglog.c
@@ -605,14 +605,35 @@ static UInt local_sys_getpid ( void )
 
 static UInt local_sys_write_stderr ( const HChar* buf, Int n )
 {
-   /* TODO */
-   return 0;
+   ULong ret;
+   __asm__ volatile (
+      "li.w    $a0, 2  \n\t" // stderr
+      "move    $a1, %1 \n\t"
+      "move    $a2, %2 \n\t"
+      "li.w    $a7, " VG_STRINGIFY(__NR_write) " \n\t"
+      "syscall 0       \n\t"
+      "move    %0, $a0 \n\t"
+      : "=r" (ret)
+      : "r" (buf), "r" (n)
+      : "memory", "$a0", "$a1", "$a2", "$a3", "$a4", "$a5", "$a6", "$a7",
+        "$t0", "$t1", "$t2", "$t3", "$t4", "$t5", "$t6", "$t7", "$t8"
+   );
+   return ret >= 0 ? (UInt)ret : -1;
 }
 
 static UInt local_sys_getpid ( void )
 {
-   /* TODO */
-   return 0;
+   ULong ret;
+   __asm__ volatile (
+      "li.w    $a7, " VG_STRINGIFY(__NR_getpid) " \n\t"
+      "syscall 0       \n\t"
+      "move    %0, $a0 \n\t"
+      : "=r" (ret)
+      :
+      : "memory", "$a0", "$a1", "$a2", "$a3", "$a4", "$a5", "$a6", "$a7",
+        "$t0", "$t1", "$t2", "$t3", "$t4", "$t5", "$t6", "$t7", "$t8"
+   );
+   return (UInt)ret;
 }
 
 #elif defined(VGP_x86_solaris)
-- 
2.39.1

