From 76a45535f10a36b179b9a95ab95f956cd6f2b75a Mon Sep 17 00:00:00 2001
From: Feiyang Chen <chenfeiyang@loongson.cn>
Date: Wed, 23 Mar 2022 16:37:18 +0800
Subject: [PATCH v4 092/123] loongarch64: Add support for trampoline

---
 coregrind/m_trampoline.S | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/coregrind/m_trampoline.S b/coregrind/m_trampoline.S
index 5679e504b..022bfaddb 100644
--- a/coregrind/m_trampoline.S
+++ b/coregrind/m_trampoline.S
@@ -1541,13 +1541,27 @@ VG_(loongarch64_linux_SUBST_FOR_rt_sigreturn):
 .global VG_(loongarch64_linux_REDIR_FOR_index)
 .type   VG_(loongarch64_linux_REDIR_FOR_index), @function
 VG_(loongarch64_linux_REDIR_FOR_index):
-   /* TODO */
+   index_loop:
+      ld.bu  $t0, $a0, 0
+      beq    $t0, $a1, index_end
+      addi.d $a0, $a0, 1
+      bne    $t0, $zero, index_loop
+      move   $a0, $zero
+   index_end:
+      jr     $ra
 .size VG_(loongarch64_linux_REDIR_FOR_index), .-VG_(loongarch64_linux_REDIR_FOR_index)
 
 .global VG_(loongarch64_linux_REDIR_FOR_strlen)
 .type   VG_(loongarch64_linux_REDIR_FOR_strlen), @function
 VG_(loongarch64_linux_REDIR_FOR_strlen):
-   /* TODO */
+      move   $t0, $a0
+   strlen_loop:
+      ld.bu  $t1, $a0, 0
+      addi.d $a0, $a0, 1
+      bne    $t1, $zero, strlen_loop
+      sub.d  $a0, $a0, $t0
+      addi.d $a0, $a0, -1
+      jr     $ra
 .size VG_(loongarch64_linux_REDIR_FOR_strlen), .-VG_(loongarch64_linux_REDIR_FOR_strlen)
 
 .global VG_(trampoline_stuff_end)
-- 
2.39.1

