From 3dc4eebec467da71376d22d822e78d6c53b4d583 Mon Sep 17 00:00:00 2001
From: Feiyang Chen <chenfeiyang@loongson.cn>
Date: Wed, 23 Mar 2022 16:37:39 +0800
Subject: [PATCH v4 093/123] loongarch64: Add support for libcassert

---
 coregrind/m_libcassert.c | 21 ++++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/coregrind/m_libcassert.c b/coregrind/m_libcassert.c
index 56a3e2000..0ad514129 100644
--- a/coregrind/m_libcassert.c
+++ b/coregrind/m_libcassert.c
@@ -266,9 +266,24 @@
       }
 #elif defined(VGP_loongarch64_linux)
 #  define GET_STARTREGS(srP)                              \
-   do {                                                   \
-      /* TODO */                                          \
-   } while (0)
+   {                                                      \
+      ULong pc, sp, fp, ra;                               \
+      __asm__ __volatile__(                               \
+         "pcaddi %0, 0   \n\t"                            \
+         "move   %1, $sp \n\t"                            \
+         "move   %2, $fp \n\t"                            \
+         "move   %3, $ra \n\t"                            \
+         : "=r" (pc),                                     \
+           "=r" (sp),                                     \
+           "=r" (fp),                                     \
+           "=r" (ra)                                      \
+         : /* reads none */                               \
+         : /* no trashed */ );                            \
+      (srP)->r_pc = (ULong)pc;                            \
+      (srP)->r_sp = (ULong)sp;                            \
+      (srP)->misc.LOONGARCH64.r_fp = (ULong)fp;           \
+      (srP)->misc.LOONGARCH64.r_ra = (ULong)ra;           \
+   }
 #else
 #  error Unknown platform
 #endif
-- 
2.39.1

