From 5dd85a6b8415e928da9f2372b3e10581cd66a94c Mon Sep 17 00:00:00 2001
From: Feiyang Chen <chenfeiyang@loongson.cn>
Date: Mon, 4 Jul 2022 14:19:12 +0800
Subject: [PATCH v4 117/123] loongarch64: Add CLEAR_CALLER_SAVED_REGS macro for
 memcheck leak tests

---
 memcheck/tests/leak.h | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/memcheck/tests/leak.h b/memcheck/tests/leak.h
index 79e3cd6ac..df9d2e759 100644
--- a/memcheck/tests/leak.h
+++ b/memcheck/tests/leak.h
@@ -148,6 +148,27 @@
    do {                                                                      \
       __asm__ __volatile__ ("movl $0, %ecx\n\t"); \
    } while (0)
+#elif defined (VGA_loongarch64)
+#define CLEAR_CALLER_SAVED_REGS                                              \
+   do {                                                                      \
+      __asm__ __volatile__ ("move $a0, $zero \n\t"                           \
+                            "move $a1, $zero \n\t"                           \
+                            "move $a2, $zero \n\t"                           \
+                            "move $a3, $zero \n\t"                           \
+                            "move $a4, $zero \n\t"                           \
+                            "move $a5, $zero \n\t"                           \
+                            "move $a6, $zero \n\t"                           \
+                            "move $a7, $zero \n\t"                           \
+                            "move $t0, $zero \n\t"                           \
+                            "move $t1, $zero \n\t"                           \
+                            "move $t2, $zero \n\t"                           \
+                            "move $t3, $zero \n\t"                           \
+                            "move $t4, $zero \n\t"                           \
+                            "move $t5, $zero \n\t"                           \
+                            "move $t6, $zero \n\t"                           \
+                            "move $t7, $zero \n\t"                           \
+                            "move $t8, $zero \n\t");                         \
+   } while (0)
 #else
 #define CLEAR_CALLER_SAVED_REGS  /*nothing*/
 #endif
-- 
2.39.1

