diff --git a/Makefile b/Makefile
index fe3baa4..e6de6e5 100644
--- a/Makefile
+++ b/Makefile
@@ -9,12 +9,12 @@ all:
 	$(AR) rc deps/libnanovg.a deps/nanovg/src/*.o
 	cd deps/pugl         && ./waf configure --no-cairo --static
 	cd deps/pugl         && ./waf
-	cd src/osc-bridge    && CFLAGS="-I ../../deps/libuv-v1.9.1/include " make lib
+	cd src/osc-bridge    && make lib
 	cd mruby             && MRUBY_CONFIG=../build_config.rb rake
-	$(CC) -shared -o libzest.so `find mruby/build/host -type f | grep -e "\.o$$" | grep -v bin` ./deps/libnanovg.a \
+	$(CC) -shared -o libzest.so -Wl,--whole-archive ./mruby/build/host/lib/libmruby.a -Wl,--no-whole-archive  ./deps/libnanovg.a \
 		./deps/libnanovg.a \
 		src/osc-bridge/libosc-bridge.a \
-		./deps/libuv-v1.9.1/.libs/libuv.a  -lm -lX11 -lGL -lpthread
+		-luv  -lm -lX11 -lGL -lpthread
 	$(CC) test-libversion.c deps/pugl/build/libpugl-0.a -ldl -o zest -lX11 -lGL -lpthread -I deps/pugl -std=gnu99
 
 windows:
diff --git a/build_config.rb b/build_config.rb
index b30b14a..2999a11 100644
--- a/build_config.rb
+++ b/build_config.rb
@@ -110,7 +110,7 @@ build_type.new(build_name) do |conf|
       #linker.library_paths  << "#{`pwd`.strip}/../deps/rtosc/build/"
       linker.library_paths  << "#{`pwd`.strip}/../src/osc-bridge/"
       linker.libraries << 'osc-bridge'
-      linker.flags_after_libraries  << "#{`pwd`.strip}/../deps/libuv.a"
+      linker.flags_after_libraries  << "-luv"
       linker.flags_after_libraries  << "#{`pwd`.strip}/../deps/pugl/build/libpugl-0.a"
       linker.flags_after_libraries  << "#{`pwd`.strip}/../deps/libnanovg.a"
       if(!windows)
