diff --git a/src/conditions/BUILD b/src/conditions/BUILD
index 30b896e..48223e3 100755
--- a/src/conditions/BUILD
+++ b/src/conditions/BUILD
@@ -93,6 +93,15 @@ config_setting(
     visibility = ["//visibility:public"],
 )

+config_setting(
+    name = "linux_loongarch64",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:loongarch64",
+    ],
+    visibility = ["//visibility:public"],
+)
+
 config_setting(
     name = "darwin",
     constraint_values = ["@platforms//os:macos"],
diff --git a/src/conditions/BUILD.tools b/src/conditions/BUILD.tools
index 8c21fa5..8ee859f 100755
--- a/src/conditions/BUILD.tools
+++ b/src/conditions/BUILD.tools
@@ -76,6 +76,15 @@ config_setting(
     visibility = ["//visibility:public"],
 )

+config_setting(
+    name = "linux_loongarch64",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:loongarch64",
+    ],
+    visibility = ["//visibility:public"],
+)
+
 config_setting(
     name = "linux_aarch64",
     constraint_values = [
diff --git a/src/main/java/com/google/devtools/build/lib/analysis/config/AutoCpuConverter.java b/src/main/java/com/google/devtools/build/lib/analysis/config/AutoCpuConverter.java
index 6581fad..3cef313 100755
--- a/src/main/java/com/google/devtools/build/lib/analysis/config/AutoCpuConverter.java
+++ b/src/main/java/com/google/devtools/build/lib/analysis/config/AutoCpuConverter.java
@@ -56,6 +56,7 @@ public class AutoCpuConverter extends Converter.Contextless<String> {
               case S390X -> "s390x";
               case MIPS64 -> "mips64";
               case RISCV64 -> "riscv64";
+              case LOONGARCH64 -> "loongarch64";
               default -> "unknown";
             };
         default -> "unknown";
diff --git a/src/main/java/com/google/devtools/build/lib/util/CPU.java b/src/main/java/com/google/devtools/build/lib/util/CPU.java
index 88ae4a3..15759cb 100755
--- a/src/main/java/com/google/devtools/build/lib/util/CPU.java
+++ b/src/main/java/com/google/devtools/build/lib/util/CPU.java
@@ -29,6 +29,7 @@ public enum CPU {
   S390X("s390x", ImmutableSet.of("s390x", "s390")),
   MIPS64("mips64", ImmutableSet.of("mips64el", "mips64")),
   RISCV64("riscv64", ImmutableSet.of("riscv64")),
+  LOONGARCH64("loongarch64", ImmutableSet.of("loongarch64","loong64")),
   UNKNOWN("unknown", ImmutableSet.<String>of());

   private final String canonicalName;
diff --git a/third_party/BUILD b/third_party/BUILD
index 023a760..21151e4 100755
--- a/third_party/BUILD
+++ b/third_party/BUILD
@@ -525,6 +525,7 @@ UNNECESSARY_DYNAMIC_LIBRARIES = select({
     # The .so file is an x86/s390x one, so we can just remove it if the CPU is not x86/s390x
     "//src/conditions:arm": "*.so *.jnilib *.dll",
     "//src/conditions:linux_aarch64": "*.so *.jnilib *.dll",
+    "//src/conditions:linux_loongarch64": "*.so *.jnilib *.dll",
     "//src/conditions:linux_ppc": "*.so *.jnilib *.dll",
     "//src/conditions:freebsd": "*.so *.jnilib *.dll",
     "//src/conditions:openbsd": "*.so *.jnilib *.dll",
@@ -761,10 +762,15 @@ config_setting(
     values = {"host_cpu": "ppc"},
 )

+config_setting(
+    name = "loongarch64",
+    values = {"host_cpu": "loongarch64"},
+)
+
 test_suite(
     name = "all_windows_tests",
     tests = [
         "//third_party/def_parser:windows_tests",
     ],
     visibility = ["//src:__pkg__"],
-)
\ No newline at end of file
+)
