diff --git a/go/private/platforms.bzl b/go/private/platforms.bzl
index c3009b2..62bb4fd 100644
--- a/go/private/platforms.bzl
+++ b/go/private/platforms.bzl
@@ -35,6 +35,8 @@ BAZEL_GOARCH_CONSTRAINTS = {
     "ppc64le": "@platforms//cpu:ppc64le",
     "s390x": "@platforms//cpu:s390x",
     "riscv64": "@platforms//cpu:riscv64",
+    "loong64": "@platforms//cpu:loongarch64",
+    "mips64": "@platforms//cpu:mips64",
 }
 
 GOOS_GOARCH = (
@@ -96,6 +98,7 @@ GOOS_GOARCH = (
     ("windows", "amd64"),
     ("windows", "arm"),
     ("windows", "arm64"),
+    ("linux", "loongarch64"),
 )
 
 RACE_GOOS_GOARCH = {
@@ -149,6 +152,7 @@ CGO_GOOS_GOARCH = {
     ("windows", "386"): None,
     ("windows", "amd64"): None,
     ("windows", "arm64"): None,
+    ("linux", "loong64"): None,
 }
 
 def _generate_constraints(names, bazel_constraints):
