--- a/go/private/platforms.bzl  2024-05-21 17:54:21.000000000 +0000
+++ b/go/private/platforms.bzl  2025-12-22 05:53:04.511106240 +0000
@@ -33,6 +33,8 @@
     "ppc64": "@platforms//cpu:ppc",
     "ppc64le": "@platforms//cpu:ppc",
     "s390x": "@platforms//cpu:s390x",
+    "loong64": "@platforms//cpu:loongarch64",
+    "mips64": "@platforms//cpu:mips64",
 }

 GOOS_GOARCH = (
@@ -66,6 +68,7 @@
     ("linux", "ppc64le"),
     ("linux", "riscv64"),
     ("linux", "s390x"),
+    ("linux", "loongarch64"),
     ("nacl", "386"),
     ("nacl", "amd64p32"),
     ("nacl", "arm"),
@@ -124,6 +127,7 @@
     ("linux", "mipsle"): None,
     ("linux", "ppc64le"): None,
     ("linux", "riscv64"): None,
+    ("linux", "loong64"): None,
     ("linux", "s390x"): None,
     ("linux", "sparc64"): None,
     ("netbsd", "386"): None,
