Index: gnome-session-3.15.92/gnome-session/gsm-fail-whale.c
===================================================================
--- gnome-session-3.15.92.orig/gnome-session/gsm-fail-whale.c
+++ gnome-session-3.15.92/gnome-session/gsm-fail-whale.c
@@ -21,6 +21,8 @@
 #include <signal.h>
 #include <stdlib.h>
 
+#include <stdlib.h>
+
 #include <glib/gi18n.h>
 #include <glib/gstdio.h>
 
@@ -37,25 +39,6 @@ gsm_fail_whale_dialog_we_failed  (gboole
                                   gboolean            allow_logout,
                                   GsmShellExtensions *extensions)
 {
-        gint i;
-        gchar *argv[5];
-        GPid  pid;
-
-        i = 0;
-        argv[i++] = LIBEXECDIR "/gnome-session-failed";
-        if (debug_mode)
-                argv[i++] = "--debug";
-        if (allow_logout)
-                argv[i++] = "--allow-logout";
-        if (extensions != NULL && gsm_shell_extensions_n_extensions (extensions) > 0)
-                argv[i++] = "--extensions";
-        argv[i++] = NULL;
-
-        if (!g_spawn_async (NULL, argv, NULL, G_SPAWN_DO_NOT_REAP_CHILD, NULL, NULL, &pid, NULL)) {
-                exit (1);
-        }
-
-        g_child_watch_add (pid,
-                           (GChildWatchFunc)on_fail_whale_failed,
-                           NULL);
+        g_critical ("We failed, but the fail whale is dead. Sorry....");
+        exit (1);
 }
