diff --git a/source/lib/rocprofiler-sdk/platform/gnulinux/agent.cpp b/source/lib/rocprofiler-sdk/platform/gnulinux/agent.cpp
--- a/source/lib/rocprofiler-sdk/platform/gnulinux/agent.cpp
+++ b/source/lib/rocprofiler-sdk/platform/gnulinux/agent.cpp
@@ -138,6 +138,8 @@

     for(const auto& bitr : processor_blocks)
     {
+        if(bitr.size() != 0 && bitr[0].find("system type") == 0)
+            continue;
         auto info_v = cpu_info{};
         for(const auto& itr : bitr)
         {
@@ -179,7 +181,7 @@

                 if(itr.find("vendor_id") == 0)
                     info_v.vendor_id = value;
-                else if(itr.find("model name") == 0)
+                else if(itr.find("model name") == 0 || itr.find("Model Name") == 0)
                 {
                     info_v.model_name = value;
                     // Remove leading and trailing whitespaces
@@ -192,11 +194,11 @@
                     info_v.family = get_stol(value);
                 else if(itr.find("model") == 0 && itr.find("model name") != 0)
                     info_v.model = get_stol(value);
-                else if(itr.find("physical id") == 0)
+                else if(itr.find("physical id") == 0 || itr.find("package") == 0)
                     info_v.physical_id = get_stol(value);
-                else if(itr.find("core id") == 0)
+                else if(itr.find("core id") == 0 || itr.find("core") == 0)
                     info_v.core_id = get_stol(value);
-                else if(itr.find("apicid") == 0)
+                else if(itr.find("apicid") == 0 || itr.find("global_id") == 0)
                     info_v.apicid = get_stol(value);
             }
             else
@@ -211,6 +213,14 @@
                     << fmt::format("Encountered unexpected /proc/cpuinfo line format: '{}'", itr);
             }
         }
+#ifdef __loongarch_lp64
+// FIXME: Irtysh?
+        info_v.vendor_id = "Loongson";
+// FIXME: May not make sense
+        info_v.family = 1;
+// FIXME: May not make sense
+        info_v.model = 1;
+#endif

         if(info_v.is_valid())
             processor_info.emplace_back(info_v);
