From ded368bb1f1c5be0b9ee6bc3fbedf440f637cae9 Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 31 May 2025 22:00:02 +0800
Subject: [PATCH] AOSCOS; Fix man generation

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 199cd305f848..93357ae0efec 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = src python man html
+SUBDIRS = src python
 htmldir = ${datadir}/doc/@PACKAGE@-@VERSION@
 html_DATA = AUTHORS COPYING ChangeLog NEWS README
 CLEANFILES = xml/libpst xml/Makefile

base-commit: cc600ee98c4ed23b8ab0bc2cf6b6c6e9cb587e89
-- 
2.49.0

