--- a/wheel_artifacts/pyproject.toml.base	2025-03-31 18:17:07.000000000 +0800
+++ b/wheel_artifacts/pyproject.toml.base	2025-05-31 14:08:26.926488859 +0800
@@ -38,6 +38,7 @@ classifiers = [
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
     "Programming Language :: Python :: 3.13",
+    "Programming Language :: Python :: 3.14",
     "Topic :: Database",
     "Topic :: Software Development",
     "Topic :: Software Development :: Code Generators",
