From 8845d9d45ac754400e19cc2c706dba70a16ef0d5 Mon Sep 17 00:00:00 2001
From: fractal <99148867+fractal-l@users.noreply.github.com>
Date: Sat, 8 Aug 2026 14:41:46 +0300
Subject: [PATCH] fix: lang key subsets scanning for ayu_ prefixed keys

---
 codegen/lang/subsets.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/codegen/lang/subsets.cpp b/codegen/lang/subsets.cpp
index c1a3373..dcb38a9 100644
--- a/codegen/lang/subsets.cpp
+++ b/codegen/lang/subsets.cpp
@@ -193,10 +193,11 @@ void Scan(const QByteArray &content, Scanned &result) {
 	const auto data = content.constData();
 	const auto size = content.size();
 	for (auto i = qsizetype(0); i + 4 < size; ++i) {
-		if (data[i] != 'l'
-			|| data[i + 1] != 'n'
-			|| data[i + 2] != 'g'
-			|| data[i + 3] != '_'
+		const auto isKeyPrefix = (data[i] == 'l' && data[i + 1] == 'n'
+			&& data[i + 2] == 'g' && data[i + 3] == '_')
+			|| (data[i] == 'a' && data[i + 1] == 'y'
+				&& data[i + 2] == 'u' && data[i + 3] == '_');
+		if (!isKeyPrefix
 			|| (i > 0 && IsIdentifierChar(data[i - 1]))) {
 			continue;
 		}
