From 668ecc1ea70d747861e717144a867e4e176c11af Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Jun 2025 22:42:40 +0800
Subject: [PATCH 2/3] AOSCOS: fix(Completion): fix paths in built-in
 completions

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 Completion/Unix/Command/_loadkeys | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Completion/Unix/Command/_loadkeys b/Completion/Unix/Command/_loadkeys
index d0a87b6f6..d57767764 100644
--- a/Completion/Unix/Command/_loadkeys
+++ b/Completion/Unix/Command/_loadkeys
@@ -9,7 +9,7 @@ case $OSTYPE in
       '(--mktable -m)'{--mktable,-m}'[output a "defkeymap.c" to stdout]' \
       '(--clearstrings)'{--clearstrings,-s}'[clear kernel string table]' \
       '*'{--verbose,-v}'[print more details]' \
-      ':keymap:_files -W /usr/share/keymaps'
+      ':keymap:_files -W /usr/share/kbd/keymaps'
   ;;
   solaris*)
     _files -W /usr/share/lib/keytables
-- 
2.49.0

