--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -291,14 +291,14 @@
         qt5_add_dbus_interfaces(CANTATA_SRCS dbus/org.freedesktop.PowerManagement.Inhibit.xml)
     else (ENABLE_QT5)
         if ( (QT_VERSION_MINOR LESS 8) OR (QT_VERSION_MINOR EQUAL 8 AND QT_VERSION_PATCH LESS 2) )
-            qt4_add_dbus_adaptor(CANTATA_SRCS dbus/org.mpris.MediaPlayer2.Player.OLD.xml dbus/mpris.h Mpris)
+            qt4_add_dbus_adaptor(CANTATA_SRCS dbus/org.mpris.MediaPlayer2.Player.xml dbus/mpris.h Mpris)
         else ( (QT_VERSION_MINOR LESS 8) OR (QT_VERSION_MINOR EQUAL 8 AND QT_VERSION_PATCH LESS 2) )
             qt4_add_dbus_adaptor(CANTATA_SRCS dbus/org.mpris.MediaPlayer2.Player.xml dbus/mpris.h Mpris)
         endif ( (QT_VERSION_MINOR LESS 8) OR (QT_VERSION_MINOR EQUAL 8 AND QT_VERSION_PATCH LESS 2) )
         qt4_add_dbus_adaptor(CANTATA_SRCS dbus/org.mpris.MediaPlayer2.root.xml dbus/mpris.h Mpris)
         qt4_add_dbus_adaptor(CANTATA_SRCS dbus/com.googlecode.cantata.xml gui/mainwindow.h MainWindow)
         if ( (QT_VERSION_MINOR LESS 8) OR (QT_VERSION_MINOR EQUAL 8 AND QT_VERSION_PATCH LESS 2) )
-            qt4_add_dbus_interfaces(CANTATA_SRCS dbus/org.freedesktop.Notifications.OLD.xml)
+            qt4_add_dbus_interfaces(CANTATA_SRCS dbus/org.freedesktop.Notifications.xml)
         else ( (QT_VERSION_MINOR LESS 8) OR (QT_VERSION_MINOR EQUAL 8 AND QT_VERSION_PATCH LESS 2) )
             qt4_add_dbus_interfaces(CANTATA_SRCS dbus/org.freedesktop.Notifications.xml)
         endif ( (QT_VERSION_MINOR LESS 8) OR (QT_VERSION_MINOR EQUAL 8 AND QT_VERSION_PATCH LESS 2) )
--- a/dbus/org.mpris.MediaPlayer2.Player.xml
+++ b/dbus/org.mpris.MediaPlayer2.Player.xml
@@ -28,7 +28,7 @@
 		<property name='Rate' type='d' access='readwrite'/>
 		<property name='Shuffle' type='b' access='readwrite'/>
 		<property name='Metadata' type='a{sv}' access='read'>
-			<annotation name="org.qtproject.QtDBus.QtTypeName" value="QVariantMap"/>
+			<annotation name="com.trolltech.QtDBus.QtTypeName" value="QVariantMap"/>
 		</property>
 		<property name='Volume' type='d' access='readwrite'/>
 		<property name='Position' type='x' access='read'/>
--- a/dbus/org.freedesktop.Notifications.xml
+++ b/dbus/org.freedesktop.Notifications.xml
@@ -22,7 +22,7 @@
       <arg name="body" type="s" direction="in"/>
       <arg name="actions" type="as" direction="in"/>
       <arg name="hints" type="a{sv}" direction="in"/>
-      <annotation name="org.qtproject.QtDBus.QtTypeName.In6" value="QVariantMap"/>
+      <annotation name="com.trolltech.QtDBus.QtTypeName.In6" value="QVariantMap"/>
       <arg name="timeout" type="i" direction="in"/>
       <arg name="return_id" type="u" direction="out"/>
     </method>
