diff -Naur MoarVM-2017.04/build/setup.pm MoarVM-2017.04.nowerror/build/setup.pm
--- MoarVM-2017.04/build/setup.pm	2017-04-01 12:12:37.000000000 +0000
+++ MoarVM-2017.04.nowerror/build/setup.pm	2017-06-30 13:59:12.380585178 +0000
@@ -265,7 +265,7 @@
         cc => 'gcc',
         ld => undef,
 
-        ccmiscflags  => '-Wdeclaration-after-statement -Werror=declaration-after-statement',
+        ccmiscflags  => '',
         ccwarnflags  => '',
         ccoptiflags  => '-O%s -DNDEBUG',
         ccdebugflags => '-g%s',
