diff -up utfcpp-3.1/CMakeLists.txt.noarch utfcpp-3.1/CMakeLists.txt
--- utfcpp-3.1/CMakeLists.txt.noarch	2019-10-22 09:52:44.918701226 +0200
+++ utfcpp-3.1/CMakeLists.txt	2019-10-22 09:51:52.374148428 +0200
@@ -17,7 +17,7 @@ if(UTF8_INSTALL)
         set(DEF_INSTALL_CMAKE_DIR CMake)
     else()
         include(GNUInstallDirs) # define CMAKE_INSTALL_*
-        set(DEF_INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/utf8cpp)
+        set(DEF_INSTALL_CMAKE_DIR ${CMAKE_INSTALL_DATADIR}/cmake/utf8cpp)
     endif()
 
     install(DIRECTORY source/ DESTINATION include/utf8cpp)
