From 22b88311609d2a6e29104695446278ce5d758aa3 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 12 Aug 2024 16:22:31 +0800
Subject: [PATCH 1/2] Fix recursive FFLAGS expansion in Makefiles

---
 ctest/Makefile | 2 +-
 test/Makefile  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ctest/Makefile b/ctest/Makefile
index c02e04e1a..877a190c1 100644
--- a/ctest/Makefile
+++ b/ctest/Makefile
@@ -26,7 +26,7 @@ endif
 override CFLAGS += -DADD$(BU) -DCBLAS
 ifeq ($(F_COMPILER),GFORTRAN)
 ifneq (, $(filter $(CORE),LOONGSON3R3 LOONGSON3R4))
-	override FFLAGS = $(filter_out(-O2 -O3,$(FFLAGS))) -O0
+	override FFLAGS := $(filter_out(-O2 -O3,$(FFLAGS))) -O0
 endif
 	override FFLAGS += -fno-tree-vectorize
 endif
diff --git a/test/Makefile b/test/Makefile
index cfb2d41f5..65576d3dd 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -2,7 +2,7 @@ TOPDIR	= ..
 include ../Makefile.system
 ifeq ($(F_COMPILER),GFORTRAN)
 ifneq (, $(filter $(CORE),LOONGSON3R3 LOONGSON3R4))
-	override FFLAGS = $(filter_out(-O2 -O3,$(FFLAGS))) -O0
+	override FFLAGS := $(filter_out(-O2 -O3,$(FFLAGS))) -O0
 endif
         override FFLAGS += -fno-tree-vectorize
 endif
-- 
2.46.0

