diff -Naur gnome-power-manager-3.22.1/configure.ac gnome-power-manager-3.22.1.noman/configure.ac
--- gnome-power-manager-3.22.1/configure.ac	2016-10-12 04:31:14.000000000 -0500
+++ gnome-power-manager-3.22.1.noman/configure.ac	2016-11-24 20:37:27.518123935 -0600
@@ -117,21 +117,11 @@
 AM_CONDITIONAL([HAVE_TESTS], [test $have_tests = yes])
 
 dnl ---------------------------------------------------------------------------
-dnl - Is docbook2man available?
-dnl ---------------------------------------------------------------------------
-AC_PATH_PROG(DOCBOOK2MAN, docbook2man, no)
-if test "$DOCBOOK2MAN" = "no" ; then
-	AC_MSG_WARN([docbook2man not found, will not be able to build man documentation])
-fi
-AM_CONDITIONAL(HAVE_DOCBOOK2MAN, [test "$DOCBOOK2MAN" != "no"])
-
-dnl ---------------------------------------------------------------------------
 dnl - Makefiles, etc.
 dnl ---------------------------------------------------------------------------
 AC_CONFIG_FILES([
 Makefile
 src/Makefile
-man/Makefile
 po/Makefile.in
 data/Makefile
 data/org.gnome.PowerStats.desktop.in
diff -Naur gnome-power-manager-3.22.1/Makefile.am gnome-power-manager-3.22.1.noman/Makefile.am
--- gnome-power-manager-3.22.1/Makefile.am	2016-10-12 04:31:14.000000000 -0500
+++ gnome-power-manager-3.22.1.noman/Makefile.am	2016-11-24 20:37:35.918922740 -0600
@@ -4,7 +4,6 @@
 SUBDIRS = 						\
 	src						\
 	po						\
-	man						\
 	data
 
 EXTRA_DIST =						\
