From 77efdba84e8e245022886a90b5bc0e0e14b1667f Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Mon, 25 Mar 2019 07:12:46 +0000
Subject: [PATCH] src/ui: drop qtchooser

---
 src/ui/ui.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ui/ui.pro b/src/ui/ui.pro
index 90ee1f1..3e96ee4 100644
--- a/src/ui/ui.pro
+++ b/src/ui/ui.pro
@@ -43,7 +43,7 @@ isEmpty(DESTDIR) {
 }
 
 isEmpty(LRELEASE) {
-    !macx:!haiku: LRELEASE = qtchooser -run-tool=lrelease -qt=5
+    !macx:!haiku: LRELEASE = lrelease-qt5
     haiku: LRELEASE = lrelease
     macx: LRELEASE = lrelease
 }
-- 
2.21.0

