From d70eae71ab1ae52fddc9b3fe79f3e07d061c7d5e Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sun, 18 Aug 2019 19:52:27 +0800
Subject: [PATCH] fix usrmerge path not found issue

---
 frescobaldi_app/util.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/frescobaldi_app/util.py b/frescobaldi_app/util.py
index 5a25435e..af906728 100644
--- a/frescobaldi_app/util.py
+++ b/frescobaldi_app/util.py
@@ -55,8 +55,9 @@ def findexe(cmd, path=None):
             if path:
                 paths = itertools.chain(paths, path)
             for p in paths:
-                if os.access(os.path.join(p, cmd), os.X_OK):
-                    return os.path.join(p, cmd)
+                rp = os.path.realpath(p)
+                if os.access(os.path.join(rp, cmd), os.X_OK):
+                    return os.path.join(rp, cmd)
 
 
 def iswritable(path):
-- 
2.22.0

