diff -up nut-2.7.4/tools/nut-scanner/scan_avahi.c.dlfix nut-2.7.4/tools/nut-scanner/scan_avahi.c
diff -up nut-2.7.4/tools/nut-scanner/scan_ipmi.c.dlfix nut-2.7.4/tools/nut-scanner/scan_ipmi.c
diff -up nut-2.7.4/tools/nut-scanner/scan_nut.c.dlfix nut-2.7.4/tools/nut-scanner/scan_nut.c
diff -up nut-2.7.4/tools/nut-scanner/scan_snmp.c.dlfix nut-2.7.4/tools/nut-scanner/scan_snmp.c
diff -up nut-2.7.4/tools/nut-scanner/scan_usb.c.dlfix nut-2.7.4/tools/nut-scanner/scan_usb.c
diff -up nut-2.7.4/tools/nut-scanner/scan_xml_http.c.dlfix nut-2.7.4/tools/nut-scanner/scan_xml_http.c
--- nut-2.7.4/tools/nut-scanner/scan_xml_http.c.dlfix	2016-03-08 13:01:11.000000000 +0100
+++ nut-2.7.4/tools/nut-scanner/scan_xml_http.c	2016-03-12 00:23:02.554826665 +0100
@@ -37,7 +37,6 @@
 #include <ltdl.h>
 
 /* dynamic link library stuff */
-static char * libname = "libneon";
 static lt_dlhandle dl_handle = NULL;
 static const char *dl_error = NULL;
 
@@ -102,7 +101,7 @@ int nutscan_load_neon_library(const char
 
 	return 1;
 err:
-	fprintf(stderr, "Cannot load XML library (%s) : %s. XML search disabled.\n", libname, dl_error);
+	fprintf(stderr, "Cannot load XML library (%s) : %s. XML search disabled.\n", libname_path, dl_error);
 	dl_handle = (void *)1;
 	lt_dlexit();
 	return 0;
