diff -Naur ffmpeg-5.1/libavformat/avformat.h ffmpeg-5.1.dts/libavformat/avformat.h
--- ffmpeg-5.1/libavformat/avformat.h	2022-07-22 17:58:39.000000000 +0000
+++ ffmpeg-5.1.dts/libavformat/avformat.h	2022-07-27 07:02:27.260143833 +0000
@@ -1128,6 +1128,10 @@
  */
 int64_t    av_stream_get_end_pts(const AVStream *st);
 
+// Chromium: We use the internal field first_dts vvv
+int64_t    av_stream_get_first_dts(const AVStream *st);
+// Chromium: We use the internal field first_dts ^^^
+
 #define AV_PROGRAM_RUNNING 1
 
 /**
diff -Naur ffmpeg-5.1/libavformat/mux_utils.c ffmpeg-5.1.dts/libavformat/mux_utils.c
--- ffmpeg-5.1/libavformat/mux_utils.c	2022-07-22 17:58:39.000000000 +0000
+++ ffmpeg-5.1.dts/libavformat/mux_utils.c	2022-07-27 07:05:11.163475455 +0000
@@ -37,6 +37,13 @@
         return AV_NOPTS_VALUE;
 }
 
+// Chromium: We use the internal field first_dts vvv
+int64_t av_stream_get_first_dts(const AVStream *st)
+{
+  return cffstream(st)->first_dts;
+}
+// Chromium: We use the internal field first_dts ^^^
+
 int avformat_query_codec(const AVOutputFormat *ofmt, enum AVCodecID codec_id,
                          int std_compliance)
 {
