--- a/data/configs/org.blueman.policy.in	2018-09-09 13:31:26.752003945 +0800
+++ b/data/configs/org.blueman.policy.in	2018-09-09 13:31:55.545488315 +0800
@@ -13,7 +13,7 @@
 
     <defaults>
       <allow_inactive>no</allow_inactive>
-      <allow_active>auth_admin_keep</allow_active>
+      <allow_active>yes</allow_active>
 
     </defaults>
   </action>
@@ -23,7 +23,7 @@
     <_message>Launching DHCP client requires privileges</_message>
     <defaults>
       <allow_inactive>no</allow_inactive>
-      <allow_active>auth_admin_keep</allow_active>
+      <allow_active>yes</allow_active>
     </defaults>
   </action>
 
@@ -32,7 +32,7 @@
     <_message>Launching PPP daemon requires privileges</_message>
     <defaults>
       <allow_inactive>no</allow_inactive>
-      <allow_active>auth_admin_keep</allow_active>
+      <allow_active>yes</allow_active>
     </defaults>
   </action>
 
@@ -41,7 +41,7 @@
     <_message>Setting RfKill State requires privileges</_message>
     <defaults>
       <allow_inactive>no</allow_inactive>
-      <allow_active>auth_admin_keep</allow_active>
+      <allow_active>yes</allow_active>
     </defaults>
   </action>
 
