diff -Naur gdbm-1.18.1.old/src/parseopt.c gdbm-1.18.1/src/parseopt.c
--- gdbm-1.18.1.old/src/parseopt.c	2018-05-30 17:39:15.000000000 +0800
+++ gdbm-1.18.1/src/parseopt.c	2020-11-10 23:35:38.279144667 +0800
@@ -255,8 +255,6 @@
 }
 
 char *parseopt_program_name;
-char *parseopt_program_doc;
-char *parseopt_program_args;
 const char *program_bug_address = "<" PACKAGE_BUGREPORT ">";
 void (*parseopt_help_hook) (FILE *stream);
 
