From 91310ab025fadd40fa24beba5c9243d682b7ce6d Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 11 Mar 2024 17:02:53 +0800
Subject: [PATCH] HACK(CMakeLists.txt): drop broken QtFeedBack detection

---
 CMakeLists.txt | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fef526d..4b25d07 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,7 +56,6 @@ ecm_setup_version(${PROJECT_VERSION}
 ################# Find dependencies #################
 
 find_package(Qt${QT_MAJOR_VERSION} ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Core Quick Test Gui Svg QuickControls2 Sql Widgets)
-find_package(Qt${QT_MAJOR_VERSION} ${QT_MIN_VERSION} OPTIONAL_COMPONENTS Feedback)
 find_package(KF${QT_MAJOR_VERSION} ${KF_MIN_VERSION} REQUIRED COMPONENTS Kirigami2 Purpose I18n Config CoreAddons DBusAddons WindowSystem Notifications OPTIONAL_COMPONENTS QQC2DesktopStyle)
 find_package(KF${QT_MAJOR_VERSION}KirigamiAddons 0.6 REQUIRED)
 set_package_properties(KF${QT_MAJOR_VERSION}QQC2DesktopStyle PROPERTIES TYPE RUNTIME)
@@ -87,11 +86,6 @@ add_definitions(-DQT_NO_FOREACH -DQT_NO_URL_CAST_FROM_STRING)
 kde_enable_exceptions()
 qcoro_enable_coroutines()
 
-if (NOT Qt${QT_MAJOR_VERSION}Feedback_FOUND)
-    message(WARNING "Qt Feedback dependency is not found.\n"
-        "While it is optional since unmaintained, haptics feedback functionality is not supported without it.")
-endif()
-
 ################# build and install #################
 add_subdirectory(lib)
 add_subdirectory(src)
-- 
2.44.0

