From eef3de6bab1c5899345ffa5e475a56798ccddbf8 Mon Sep 17 00:00:00 2001
From: ouni666 <fangwenbin@vip.qq.com>
Date: Mon, 7 Jul 2025 14:55:29 +0800
Subject: [PATCH] fix loongarch64 electron

---
 electron-builder.yml | 36 +-----------------------------------
 package.json         |  8 +++++---
 2 files changed, 6 insertions(+), 38 deletions(-)

diff --git a/electron-builder.yml b/electron-builder.yml
index af9871d..fa02ad0 100644
--- a/electron-builder.yml
+++ b/electron-builder.yml
@@ -14,41 +14,7 @@ files:
 publish:
 - github
 
-
-# Mac OS configuration
-mac:
-  category: public.app-category.games
-  # icon: "src/app/assets/images/icon.icns"
-  target: 
-  - target: dmg
-    arch: 
-    - universal
-
-# Config for OSX dmg
-dmg:
-  contents:
-    - x: 130
-      y: 220
-    - x: 410
-      y: 220
-      type: "link"
-      path: "/Applications"
-
-# Windows configuration
-win:
-  # icon: "src/app/assets/images/icon.ico"
-  target:
-    - "zip"
-    - "nsis"
-
-# Config for the windows installer
-nsis:
-  oneClick: true
-  perMachine: true
-
 # Linux configuration
 linux:
   category: Game
-  target:
-    - "AppImage"
-    - "deb"
\ No newline at end of file
+  target: "dir"
diff --git a/package.json b/package.json
index 51335bf..7efdb84 100644
--- a/package.json
+++ b/package.json
@@ -7,7 +7,7 @@
   "main": "app/background.js",
   "scripts": {
     "dev": "DEBUG=xstreaming:authenticationV2 nextron",
-    "build": "nextron build",
+    "build": "nextron build --no-pack && electron-builder --dir",
     "lint": "eslint . --ext .ts && eslint renderer/ --ext .tsx",
     "lint:fix": "eslint . --ext .ts --fix && eslint renderer/ --ext .tsx --fix",
     "postinstall": "electron-builder install-app-deps",
@@ -52,8 +52,10 @@
     "@typescript-eslint/parser": "^8.6.0",
     "autoprefixer": "^10.4.20",
     "babel-loader": "^9.1.3",
-    "electron": "33",
-    "electron-builder": "^24.13.3",
+    "electron": "35.4.0",
+    "@darkyzhou-powered/electron-builder": "^25.1.8",
+    "@darkyzhou-powered/app-builder-lib": "^25.1.8",
+    "@darkyzhou-powered/builder-util": "^25.1.7",
     "eslint": "^8.57.0",
     "eslint-plugin-react-hooks": "^4.6.2",
     "eslint-plugin-react-refresh": "^0.4.12",
-- 
2.50.0

