diff -rupN openjpeg-2.1.0/CMakeLists.txt openjpeg-2.1.0-new/CMakeLists.txt
--- openjpeg-2.1.0/CMakeLists.txt	2014-04-29 09:15:05.000000000 +0200
+++ openjpeg-2.1.0-new/CMakeLists.txt	2014-05-23 13:12:19.513936024 +0200
@@ -249,7 +249,6 @@ if(BUILD_CODEC OR BUILD_MJ2)
   # OFF: It will only build 3rd party libs if they are not found on the system
   # ON: 3rd party libs will ALWAYS be build, and used
   option(BUILD_THIRDPARTY "Build the thirdparty executables if it is needed" OFF)
-  add_subdirectory(thirdparty)
   add_subdirectory(src/bin)
 endif ()
 add_subdirectory(wrapping)
