From 685add8c5611e324d78d731233c26c58e398f8b0 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Wed, 12 Nov 2025 18:42:34 +0800
Subject: [PATCH 7/9] chore: import .gitignore from loonggpu-kernel-dkms

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 .gitignore | 8 ++++++++
 1 file changed, 8 insertions(+)
 create mode 100644 .gitignore

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..223542a
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,8 @@
+*.o
+*.ko
+*.mod
+*.mod.c
+*.cmd
+*.symvers
+*.order
+conftest
-- 
2.52.0

