From 2fe3ca264a615d739b819ff55368c142a59104b1 Mon Sep 17 00:00:00 2001
From: gnunn1 <gerald.b.nunn@gmail.com>
Date: Fri, 6 Jan 2017 09:39:19 -0500
Subject: [PATCH] Update license, see #669

---
 source/gx/terminix/terminal/regex.d | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/source/gx/terminix/terminal/regex.d b/source/gx/terminix/terminal/regex.d
index c2a2da8..cf5dade 100644
--- a/source/gx/terminix/terminal/regex.d
+++ b/source/gx/terminix/terminal/regex.d
@@ -1,18 +1,8 @@
 /*
  * Copyright © 2015 Egmont Koblinger
  *
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not
+ * distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
 /*
