diff -Naur a/sfd/Makefile b/sfd/Makefile
--- a/sfd/Makefile	2016-10-19 00:10:43.613222321 -0500
+++ b/sfd/Makefile	2016-10-19 00:10:21.549807050 -0500
@@ -7,10 +7,7 @@
 TTFS=$(SFDS:.sfd=.ttf)
 OTFS=$(SFDS:.sfd=.otf)
 WOFFS=$(SFDS:.sfd=.woff)
-FFLIBS=/usr/local/lib/fontforge_22_02_2011
-FFBIN=/usr/local/bin/fontforge_22_02_2011
-#FIXME newer versions need flag --quiet for pretty output
-FF=LD_LIBRARY_PATH=${FFLIBS} ${FFBIN}
+FF=/usr/bin/fontforge
 FFFF=$(FF) -lang=ff -script 
 FFPY=$(FF) -lang=py -script 
 
