diff -Naur lightdm-settings-1.2.8/usr/share/polkit-1/actions/org.x.lightdm-settings.policy lightdm-settings-1.2.8.x/usr/share/polkit-1/actions/org.x.lightdm-settings.policy
--- lightdm-settings-1.2.8/usr/share/polkit-1/actions/org.x.lightdm-settings.policy	2019-07-28 13:27:04.000000000 +0000
+++ lightdm-settings-1.2.8.x/usr/share/polkit-1/actions/org.x.lightdm-settings.policy	2019-09-18 03:42:47.303467942 +0000
@@ -8,13 +8,14 @@
   <vendor_url>https://linuxmint.com/</vendor_url>
 
   <action id="org.x.lightdm-settings">
+    <message>Authentication is required to run LightDM Settings</message>
     <icon_name>preferences-system-login</icon_name>
     <defaults>
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
       <allow_active>auth_admin_keep</allow_active>
     </defaults>
-    <annotate key="org.freedesktop.policykit.exec.path">/usr/bin/lightdm-settings</annotate>
+    <annotate key="org.freedesktop.policykit.exec.path">/usr/bin/lightdm-settings-bin</annotate>
     <annotate key="org.freedesktop.policykit.exec.allow_gui">true</annotate>
   </action>
 
