From 51602d7a5f5e63b21d7e036880ee57911da2c653 Mon Sep 17 00:00:00 2001
From: Arseniy Obolenskiy <arseniy.obolenskiy@intel.com>
Date: Tue, 14 Jul 2020 10:24:54 +0300
Subject: [PATCH] [tracer] Fix build with libc++

Fix #2209
---
 tools/tracer/dumps/dump.h | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/tools/tracer/dumps/dump.h b/tools/tracer/dumps/dump.h
index 71a4806a53..0473e0e9ae 100644
--- a/tools/tracer/dumps/dump.h
+++ b/tools/tracer/dumps/dump.h
@@ -67,18 +67,19 @@ bool _IsBadReadPtr(void *ptr, size_t size);
 #define DUMP_FIELD_RESERVED(_field) \
     str += structName + "." #_field "[]=" + DUMP_RESERVED_ARRAY(_struct._field) + "\n";
 
-    #define ToString( x )  dynamic_cast< std::ostringstream & >( \
+    #define ToString( x )  static_cast< std::ostringstream const & >( \
         ( std::ostringstream() << std::dec << x ) ).str()
 
-    #define TimeToString( x ) dynamic_cast< std::ostringstream & >( \
+    #define TimeToString( x ) static_cast< std::ostringstream const & >( \
             ( std::ostringstream() << std::left << std::setw(4) << std::dec << x <<" msec") ).str()
 
     /*
-    #define ToHexFormatString( x ) dynamic_cast< std::ostringstream & >( \
+    #define ToHexFormatString( x ) static_cast< std::ostringstream const & >( \
             ( std::ostringstream() << std::hex << x ) ).str()
     */
 
-    #define ToHexFormatString( x ) (dynamic_cast< std::ostringstream & >( ( std::ostringstream() << std::hex << pVoidToHexString((void*)x) ) ).str() )
+    #define ToHexFormatString( x ) (static_cast< std::ostringstream const & >( \
+            ( std::ostringstream() << std::hex << pVoidToHexString((void*)x) ) ).str() )
 /*
 #define DEFINE_GET_TYPE(type) \
     template<> \
@@ -118,7 +119,8 @@ class DumpContext
 
     template<typename T>
     inline std::string toString( T x, eDumpFormat format = DUMP_DEC){
-        return dynamic_cast< std::ostringstream & >(( std::ostringstream() << ((format == DUMP_DEC) ? std::dec : std::hex) << x )).str();
+        return static_cast< std::ostringstream const & >(
+            ( std::ostringstream() << ((format == DUMP_DEC) ? std::dec : std::hex) << x )).str();
     }
 
     const char* get_bufferid_str(mfxU32 bufferid);
--- a/_studio/mfx_lib/mctf_package/mctf/src/mctf_common.cpp	2022-04-12 04:20:55.328283909 +0000
+++ b/_studio/mfx_lib/mctf_package/mctf/src/mctf_common.cpp	2022-04-12 04:21:07.470455354 +0000
@@ -43,6 +43,7 @@
 
 #include <algorithm>
 #include <climits>
+#include <limits>
 #include <cmath>
 #include "cmrt_cross_platform.h"
 
