From: Aron Xu <aron@debian.org>
Date: Wed, 3 Oct 2012 00:50:32 +0800
Subject: fix autoconf automake

---
 configure.in |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/configure.in b/configure.in
index fc8d5a8..767e980 100644
--- a/configure.in
+++ b/configure.in
@@ -84,7 +84,7 @@ VERSION=${LIBXSLT_VERSION}
 
 AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
 
-# AM_MAINTAINER_MODE
+AM_MAINTAINER_MODE
 
 # Support silent build rules, requires at least automake-1.11. Disable
 # by either passing --disable-silent-rules to configure or passing V=1
@@ -460,6 +460,7 @@ else
     WITH_DEBUGGER=1
     AC_DEFINE([WITH_DEBUGGER],[], [Define if debugging support is enabled])
 fi
+AM_CONDITIONAL(WITH_DEBUGGER, test "${WITH_DEBUGGER}" = "1")
 AC_SUBST(WITH_DEBUGGER)
 
 dnl
@@ -649,14 +650,14 @@ AC_SUBST(PYTHON_SITE_PACKAGES)
 
 XSLT_LIBDIR='-L${libdir}'
 XSLT_INCLUDEDIR='-I${includedir}'
-XSLT_LIBS="-lxslt $LIBXML_LIBS $M_LIBS"
+XSLT_LIBS="-lxslt $LIBXML_LIBS"
 AC_SUBST(XSLT_LIBDIR)
 AC_SUBST(XSLT_INCLUDEDIR)
 AC_SUBST(XSLT_LIBS)
 
 EXSLT_LIBDIR='-L${libdir}'
 EXSLT_INCLUDEDIR='-I${includedir}'
-EXSLT_LIBS="-lexslt $XSLT_LIBS $LIBGCRYPT_LIBS"
+EXSLT_LIBS="-lexslt $XSLT_LIBS"
 AC_SUBST(EXSLT_LIBDIR)
 AC_SUBST(EXSLT_INCLUDEDIR)
 AC_SUBST(EXSLT_LIBS)
@@ -669,7 +670,7 @@ dnl for the spec file
 RELDATE=`date +'%a %b %e %Y'`
 AC_SUBST(RELDATE)
 
-rm -f COPYING.LIB COPYING 2>/dev/null && $LN_S $srcdir/Copyright COPYING
+#rm -f COPYING.LIB COPYING 2>/dev/null && $LN_S $srcdir/Copyright COPYING
 
 
 AC_CONFIG_FILES([
