--- tdeaddons-trinity-14.0.10/ConfigureChecks.cmake	2021-02-27 16:36:23.000000000 -0800
+++ tdeaddons-trinity-14.0.10.xmms/ConfigureChecks.cmake	2021-06-26 13:24:14.037393347 -0700
@@ -35,14 +35,11 @@
   check_include_file( "fcntl.h" HAVE_FCNTL_H )
 
   if( WITH_XMMS )
-    check_library_exists( xmms xmms_remote_is_running "" HAVE_XMMS_LIB )
     tde_save_and_set( CMAKE_REQUIRED_LIBRARIES  "xmms" )
     check_symbol_exists( xmms_remote_is_running "xmms/xmmsctrl.h" HAVE_XMMS )
     tde_restore( CMAKE_REQUIRED_LIBRARIES )
-    if( HAVE_XMMS_LIB AND HAVE_XMMS )
+    if( HAVE_XMMS )
       set( XMMS_LIBRARIES xmms )
-    else( HAVE_XMMS )
-      tde_message_fatal( "xmms is required, but was not found on your system" )
     endif( HAVE_XMMS )
   endif( WITH_XMMS )
 endif( BUILD_KICKER_APPLETS )
