diff -Naur mesa-17.2.2/configure.ac mesa-17.2.2.gnu99/configure.ac
--- mesa-17.2.2/configure.ac  2017-10-02 15:49:02.000000000 +0000
+++ mesa-17.2.2.gnu99/configure.ac  2017-10-08 16:23:16.920188156 +0000
@@ -285,12 +285,7 @@
 dnl Add flags for gcc and g++
 if test "x$GCC" = xyes; then
     CFLAGS="$CFLAGS -Wall"
-
-    if test "x$USE_GNU99" = xyes; then
-        CFLAGS="$CFLAGS -std=gnu99"
-    else
-        CFLAGS="$CFLAGS -std=c99"
-    fi
+    CFLAGS="$CFLAGS -std=gnu99"
 
     # Enable -Werror=implicit-function-declaration and
     # -Werror=missing-prototypes, if available, or otherwise, just
