--- mesa-12.0.1/src/gallium/state_trackers/clover/llvm/invocation.cpp.orig	2016-08-30 08:27:41.566447916 +0200
+++ mesa-12.0.1/src/gallium/state_trackers/clover/llvm/invocation.cpp	2016-08-30 08:27:54.154379349 +0200
@@ -207,7 +207,7 @@
       c.getDiagnosticOpts().ShowCarets = false;
       c.getInvocation().setLangDefaults(c.getLangOpts(), clang::IK_OpenCL,
 #if HAVE_LLVM >= 0x0309
-                                        llvm::Triple(triple),
+                                        llvm::Triple(triple), c.getPreprocessorOpts(),
 #endif
                                         clang::LangStandard::lang_opencl11);
       c.createDiagnostics(
