From: Robert Luberda <robert@debian.org>
Date: Fri, 6 Jul 2007 08:04:24 +0100
Subject: 00 Makefile.

Fix CFLAGS.
---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index 5375f81..054ed1b 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -36,7 +36,7 @@ ifeq ($(USE_GNUC),1)
 ifeq ($(DEBUG),1)
 CXXFLAGS += -O0 -g
 else
-CXXFLAGS += -O2
+##CXXFLAGS += -O2
 endif
 CXXFLAGS_WERROR = -Werror
 CXXFLAGS += -Wall -W -Wcast-align -Wcast-qual -Wpointer-arith -Wshadow -Wwrite-strings
