From 54f4525f75fafd19ae20763a074879c079093fdc Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Mon, 8 Jan 2024 05:23:11 -0800
Subject: [PATCH] fix compilation with wxwidgets 3.2.4

---
 pcsx2/gui/i18n.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pcsx2/gui/i18n.cpp b/pcsx2/gui/i18n.cpp
index edcc5965b..2d059f9b3 100644
--- a/pcsx2/gui/i18n.cpp
+++ b/pcsx2/gui/i18n.cpp
@@ -36,7 +36,6 @@ static wxString i18n_GetBetterLanguageName( const wxLanguageInfo* info )
 	switch (info->Language)
 	{
 		case wxLANGUAGE_CHINESE:				return L"Chinese (Traditional)";
-		case wxLANGUAGE_CHINESE_TRADITIONAL:	return L"Chinese (Traditional)";
 		case wxLANGUAGE_CHINESE_TAIWAN:			return L"Chinese (Traditional)";
 		case wxLANGUAGE_CHINESE_HONGKONG:		return L"Chinese (Traditional, Hong Kong)";
 		case wxLANGUAGE_CHINESE_MACAU:			return L"Chinese (Traditional, Macau)";
-- 
2.39.1

