diff -Naur urfkill.orig/configure.ac urfkill/configure.ac
--- urfkill.orig/configure.ac	2024-07-11 09:56:46.185554582 +0800
+++ urfkill/configure.ac	2024-07-11 09:57:16.382909782 +0800
@@ -184,7 +184,7 @@
 AS_IF([test "$with_session_tracking" = "none"], with_session_tracking=no)
 # check value
 AS_IF([! (echo "$with_session_tracking" | grep -q -E "^(systemd|consolekit|no)$")],
-        AC_MSG_ERROR([--with-session-tracking must be systemd/consolekit/no, not $with_session_tracking]))
+        AC_MSG_ERROR([--with-session-tracking must be systemd/consolekit/no instead of $with_session_tracking]))
 # add conditionals and subtitution
 AM_CONDITIONAL(SESSION_TRACKING_CK, test "$with_session_tracking" = "consolekit")
 AM_CONDITIONAL(SESSION_TRACKING_SYSTEMD, test "$with_session_tracking" = "systemd")
