From 930b09bfd58b8c9588151ee2ecc22f06109bed08 Mon Sep 17 00:00:00 2001
From: Alessio Treglia <alessio@debian.org>
Date: Sun, 7 Jun 2026 01:10:33 +0800
Subject: [PATCH 01/10] DEBIAN: Zero errno before readdir call

Link: http://bugs.debian.org/375110
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 recurse.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/recurse.c b/recurse.c
index ad3622c..259a90e 100644
--- a/recurse.c
+++ b/recurse.c
@@ -408,6 +408,7 @@ static DIRECTORY *open_dir(const char *full_path)
  */
 static int read_dir(DIRECTORY *directory)
 {
+    errno = 0;
     directory->entry = readdir(directory->dir);
 
     if (directory->entry != NULL)
@@ -622,6 +623,7 @@ int process_argument(const char* path, SETTINGS* settings)
     /* Check for wildcards */
     settings->pattern = last_path(my_path);
 
+#if 0 /* Disable for Debian, as it suprises people. --liw */
     if (contains_pattern(settings->pattern))
     {
         /* Strip last part of path */
@@ -637,6 +639,7 @@ int process_argument(const char* path, SETTINGS* settings)
         }
     }
     else
+#endif
     {
         settings->pattern = NULL;
     }
-- 
2.52.0

