From 23fb0ceb3e8706ba8df69dcc7a60fa6fab97282e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Mart=C3=AD?= <mvdan@mvdan.cc>
Date: Sun, 7 Jun 2026 01:26:25 +0800
Subject: [PATCH 08/10] DEBIAN: Allow build with dh compat level 9 (debian
 build flags)

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 misc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/misc.c b/misc.c
index 5d3ee92..492995a 100644
--- a/misc.c
+++ b/misc.c
@@ -56,7 +56,7 @@ void file_error(const char* message, ...)
     vfprintf(stderr, message, args);
     va_end(args);
 
-    fprintf(stderr, strerror(err_num));
+    fputs(strerror(err_num), stderr);
     fprintf(stderr, "\n");
 }
 
-- 
2.52.0

