From 1557f0321a45ed2cdfb9cee531f54004f71fc949 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?H=C3=A5vard=20F=2E=20Aasen?= <havard.f.aasen@pfft.no>
Date: Sun, 7 Jun 2026 01:28:02 +0800
Subject: [PATCH 10/10] DEBIAN: Add missing header declaration for isatty()

Link: https://bugs.debian.org/1066500
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 misc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/misc.c b/misc.c
index 492995a..cf257f9 100644
--- a/misc.c
+++ b/misc.c
@@ -14,6 +14,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 #include <vorbis/codec.h>
 #include "i18n.h"
 #include "misc.h"
-- 
2.52.0

