--- a/configure.ac	2016-05-23 04:23:14.000000000 -0600
+++ b/configure.ac	2016-06-05 23:56:49.039658374 -0600
@@ -301,13 +301,6 @@
 if test "x${ac_cv_prog_cc_c99}" = "xno" ; then
    AC_MSG_ERROR([efl requires a c99-capable compiler])
 fi
-# We should be using ${CXX} here, but there is a bug in
-# autotools macro and CXX is always set to g++ even if
-# it's not found. So we are using an internal variable
-# that does the work for now, may get broken in the future.
-if test "x${ac_ct_CXX}" = "x" -a "x${CXX}" = "xg++"; then
-   AC_MSG_ERROR([efl requires a C++ compiler got ${ac_ct_CXX} and ${CXX}.])
-fi
 
 AC_SYS_LARGEFILE
 
