From 77365fa17263ed222f1e18c852711d63c9bf9709 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Sat, 16 Jul 2022 11:41:21 -0400
Subject: [PATCH 2/2] DIRTY HACK: workaround segv deref 0xaaaaa during tests

---
 tools/gegl-tester.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/gegl-tester.c b/tools/gegl-tester.c
index 8d10dab5e..963cb8e78 100644
--- a/tools/gegl-tester.c
+++ b/tools/gegl-tester.c
@@ -144,7 +144,7 @@ standard_output (const gchar *op_name)
   g_free (input_path);
   g_free (aux_path);
   g_free (output_path);
-  g_object_unref (composition);
+//g_object_unref (composition);
 }
 
 static gchar *
-- 
2.46.0

