Description: Build fix for powerpc / ppc64
Author: Alastair McKinstry <mckinstry@debian.org>
Bug-Origin: https://github.com/open-mpi/ompi/issues/2055
Last-Updated: 2016-09-05
Forwarded: yes

Index: openmpi-2.0.1/opal/include/opal/sys/powerpc/atomic.h
===================================================================
--- openmpi-2.0.1.orig/opal/include/opal/sys/powerpc/atomic.h
+++ openmpi-2.0.1/opal/include/opal/sys/powerpc/atomic.h
@@ -27,6 +27,13 @@
  * On powerpc ...
  */
 
+/* Hack on Debian. See: https://github.com/open-mpi/ompi/issues/2055
+ *   -- amck, 2016-09-05
+ */
+#undef OPAL_GCC_INLINE_ASSEMBLY
+#define OPAL_GCC_INLINE_ASSEMBLY 1
+
+
 #define MB()  __asm__ __volatile__ ("sync" : : : "memory")
 #define RMB() __asm__ __volatile__ ("lwsync" : : : "memory")
 #define WMB() __asm__ __volatile__ ("eieio" : : : "memory")

