From a1f3514d13303573794432ee8ee60bb94af82623 Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 31 May 2025 13:41:43 +0800
Subject: [PATCH 1/2] AOSCOS: Support Cython 3.1.0

---
 src/pygame_sdl2/event.pyx | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/pygame_sdl2/event.pyx b/src/pygame_sdl2/event.pyx
index 7b790eb53e47..4becb7370f4d 100644
--- a/src/pygame_sdl2/event.pyx
+++ b/src/pygame_sdl2/event.pyx
@@ -332,7 +332,7 @@ cdef object get_events(kinds):
     The lock must be held when calling this function.
     """
 
-    if isinstance(kinds, (int, long)):
+    if isinstance(kinds, int):
         kinds = [ kinds ]
 
     global event_queue
@@ -459,7 +459,7 @@ def set_blocked(t=None):
     if t == None:
         for et in event_names.keys():
             SDL_EventState(et, SDL_ENABLE)
-    elif isinstance(t, (int, long)):
+    elif isinstance(t, int):
         SDL_EventState(t, SDL_IGNORE)
     else:
         for et in t:
@@ -469,7 +469,7 @@ def set_allowed(t=None):
     if t == None:
         for et in event_names.keys():
             SDL_EventState(et, SDL_IGNORE)
-    elif isinstance(t, (int, long)):
+    elif isinstance(t, int):
         SDL_EventState(t, SDL_ENABLE)
     else:
         for et in t:

base-commit: 7e61ebdbf1fbc2cbff4994dd5c9bdd23411d8d37
-- 
2.49.0

