From 1cd966ef14b35010ae4f1ea3fbc1d7e3dd09c818 Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 31 May 2025 13:44:48 +0800
Subject: [PATCH 2/2] AOSCOS: Add missing import

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index 63bfaef9562f..abff6b0942a7 100755
--- a/setup.py
+++ b/setup.py
@@ -30,6 +30,7 @@ import os
 import platform
 import shutil
 import sys
+import sysconfig
 
 
 def setup_env(name):
-- 
2.49.0

