From dd1025c2c3b951821a8d122d6ed463ee24218a10 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Thu, 26 Oct 2017 17:17:21 -0600
Subject: [PATCH] pam: process limits entry

---
 src/login/systemd-user.m4 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/login/systemd-user.m4 b/src/login/systemd-user.m4
index 4f85b4b..6535844 100644
--- a/src/login/systemd-user.m4
+++ b/src/login/systemd-user.m4
@@ -7,6 +7,7 @@ m4_ifdef(`HAVE_SELINUX',
 session required pam_selinux.so close
 session required pam_selinux.so nottys open
 )m4_dnl
+session required pam_limits.so
 session required pam_loginuid.so
 session optional pam_keyinit.so force revoke
 session optional pam_systemd.so
-- 
2.14.2

