From 23f9c5afed974bf6d052fca4711017e704370462 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Sun, 28 Oct 2018 18:42:49 -0400
Subject: [PATCH 25/25] Disable test_lz4_decompress_partial

---
 src/journal/test-compress.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/journal/test-compress.c b/src/journal/test-compress.c
index 791c6fdffb..578a9c0f30 100644
--- a/src/journal/test-compress.c
+++ b/src/journal/test-compress.c
@@ -298,7 +298,6 @@ int main(int argc, char *argv[]) {
         test_compress_stream(OBJECT_COMPRESSED_LZ4, "lz4cat",
                              compress_stream_lz4, decompress_stream_lz4, srcfile);
 
-        test_lz4_decompress_partial();
 #else
         log_info("/* LZ4 test skipped */");
 #endif
-- 
2.19.1

