diff -Naur a/tmpfiles.d/tmp.conf b/tmpfiles.d/tmp.conf
--- a/tmpfiles.d/tmp.conf	2016-02-11 09:28:00.000000000 -0700
+++ b/tmpfiles.d/tmp.conf	2016-03-02 16:30:17.502036586 -0700
@@ -9,7 +9,7 @@
 
 # Clear tmp directories separately, to make them easier to override
 q /tmp 1777 root root 10d
-q /var/tmp 1777 root root 30d
+L /var/tmp - - - -
 
 # Exclude namespace mountpoints created with PrivateTmp=yes
 x /tmp/systemd-private-%b-*
