--- a/configure.ac	2016-02-18 11:58:52.511542205 -0700
+++ b/configure.ac	2016-02-18 12:00:41.967763337 -0700
@@ -1057,10 +1057,7 @@
         AS_HELP_STRING([--with-ntp-servers=NTPSERVERS],
                 [Space-separated list of default NTP servers]),
         [NTP_SERVERS="$withval"],
-        [NTP_SERVERS="time1.google.com time2.google.com time3.google.com time4.google.com"
-        AC_MSG_WARN([*** Using Google NTP servers.
-                        Do not ship OSes or devices with these default settings.
-                        See DISTRO_PORTING for details!])])
+        [NTP_SERVERS="0.pool.ntp.org 1.pool.ntp.org 2.pool.ntp.org 3.pool.ntp.org"])
 
 AC_DEFINE_UNQUOTED(NTP_SERVERS, ["$NTP_SERVERS"], [Default NTP Servers])
 AC_SUBST(NTP_SERVERS)
