Index: libgnomekbd-2.29.92/libgnomekbd.pc.in
===================================================================
--- libgnomekbd-2.29.92.orig/libgnomekbd.pc.in	2010-02-22 22:09:00.000000000 +0100
+++ libgnomekbd-2.29.92/libgnomekbd.pc.in	2010-03-28 11:13:14.730507782 +0200
@@ -6,6 +6,7 @@ includedir=@includedir@
 Name: libgnomekbd
 Description: GNOME keyboard shared library
 Version: @VERSION@
-Requires: gmodule-export-2.0 gdk-3.0 >= @GDK_REQUIRED@ libxklavier >= @LIBXKLAVIER_REQUIRED@
+Requires: gmodule-export-2.0 
+Requires.private: gdk-3.0 >= @GDK_REQUIRED@ libxklavier >= @LIBXKLAVIER_REQUIRED@
 Libs: -L${libdir} -lgnomekbd
 Cflags: -I${includedir}
