From 66d1668d79b768a2ef1d9e7374aad749ea934d9b Mon Sep 17 00:00:00 2001
From: salieri <maliya355@outlook.com>
Date: Wed, 31 Jan 2024 10:14:39 +0800
Subject: [PATCH] use system zlib

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 1532609f3..6b165253a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1403,7 +1403,7 @@ AC_MSG_CHECKING([for local zlib source])
 dnl zlib is needed for language level 3, and libpng
 # we must define ZLIBDIR regardless because png.mak does a -I$(ZLIBDIR)
 # this seems a harmless default
-ZLIBDIR=src
+ZLIBDIR=$includedir
 AUX_SHARED_ZLIB=
 ZLIBCFLAGS=""
 
-- 
2.34.1

