diff --git a/lld/ELF/InputSection.h b/lld/ELF/InputSection.h
index 26b6b4af0..5f9c164fe 100644
--- a/lld/ELF/InputSection.h
+++ b/lld/ELF/InputSection.h
@@ -137,7 +137,7 @@ public:
   // Used by --optimize-bb-jumps and RISC-V linker relaxation temporarily to
   // indicate the number of bytes which is not counted in the size. This should
   // be reset to zero after uses.
-  uint16_t bytesDropped = 0;
+  unsigned bytesDropped = 0;
 
   mutable bool compressed = false;
 
