From 9f4efcd4226269bed2affdaea1fc73d3e24c8051 Mon Sep 17 00:00:00 2001
From: Nikita Popov <npopov@redhat.com>
Date: Thu, 10 Feb 2022 09:59:11 +0100
Subject: [PATCH 1008/1013] [rust] Work around ICE in GCC 5.4

---
 lld/ELF/SyntheticSections.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lld/ELF/SyntheticSections.cpp b/lld/ELF/SyntheticSections.cpp
index 7778ae5f78e6..0a2d11218e70 100644
--- a/lld/ELF/SyntheticSections.cpp
+++ b/lld/ELF/SyntheticSections.cpp
@@ -1652,8 +1652,8 @@ void RelocationBaseSection::partitionRels() {
   if (!combreloc)
     return;
   const RelType relativeRel = target->relativeRel;
-  numRelativeRelocs =
-      llvm::partition(relocs, [=](auto &r) { return r.type == relativeRel; }) -
+  numRelativeRelocs = llvm::partition(
+      relocs, [=](const DynamicReloc &r) { return r.type == relativeRel; }) -
       relocs.begin();
 }
 
-- 
2.35.1

