From fbb6a3af950885934db5befd0a99c4bee1960c75 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 21 Oct 2025 10:49:23 +0800
Subject: [PATCH 1/2] AOSCOS: fix: use DocBook stylesheet paths found in AOSC
 OS

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 po4a/Makefile    | 4 ++--
 scripts/Makefile | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/po4a/Makefile b/po4a/Makefile
index 7d520c84..a36f8036 100644
--- a/po4a/Makefile
+++ b/po4a/Makefile
@@ -36,7 +36,7 @@ clean: ../doc/devscripts.7
 %.1:: %.dbk translate
 	mkdir -p $(LANGS)
 	xsltproc --nonet -o $@ \
-	  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl $<
+	  /usr/share/xml/docbook/xsl-stylesheets-1.79.2/manpages/docbook.xsl $<
 %.5:: %.pl translate
 	-podchecker $<
 	pod2man --utf8 --center=" " --release="$(DESC_$(dir $@))" $< > $@
@@ -46,4 +46,4 @@ clean: ../doc/devscripts.7
 %.5:: %.dbk translate
 	mkdir -p $(LANGS)
 	xsltproc --nonet -o $@ \
-	  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl $<
+	  /usr/share/xml/docbook/xsl-stylesheets-1.79.2/manpages/docbook.xsl $<
diff --git a/scripts/Makefile b/scripts/Makefile
index 0b8f1073..cf663995 100644
--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -99,7 +99,7 @@ endif
 	pod2man --utf8 --center=" " --release="Debian Utilities" --section=5 $< > $@
 %.1: %.dbk
 	xsltproc --nonet -o $@ \
-	  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl $<
+	  /usr/share/xml/docbook/xsl-stylesheets-1.79.2/manpages/docbook.xsl $<
 
 # Syntax checker
 test_sh: $(SH_CHECKS)
-- 
2.51.0

