diff --git a/po4a/Makefile b/po4a/Makefile
index d00ce013..6d320ef4 100644
--- a/po4a/Makefile
+++ b/po4a/Makefile
@@ -34,7 +34,7 @@ clean: ../doc/devscripts.1
 	pod2man --utf8 --center=" " --release="$(DESC_$(dir $@))" $< > $@
 %.1:: %.dbk translate
 	xsltproc --nonet \
-	  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl $<
+	  /usr/share/xml/docbook/xsl-stylesheets-1.79.2/manpages/docbook.xsl $<
 	# /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/other.xsl
 	# (which is imported by the above stylesheet) insists in writing the output
 	# to where it wants to.  we can only move the file ourselves.
diff --git a/scripts/Makefile b/scripts/Makefile
index 0e3f4ada..c247a82d 100644
--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -87,7 +87,7 @@ endif
 	pod2man --utf8 --center=" " --release="Debian Utilities" $< > $@
 %.1: %.dbk
 	xsltproc --nonet -o $@ \
-	  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl $<
+	  /usr/share/xml/docbook/xsl-stylesheets-1.79.2/manpages/docbook.xsl $<
 
 # Syntax checker
 test_sh: $(SH_CHECKS)
@@ -150,7 +150,7 @@ clean:
 test: test_pl test_sh test_py
 
 install: all
-	python3 setup.py install --root="$(DESTDIR)" --no-compile --install-layout=deb
+	python3 setup.py install --root="$(DESTDIR)" --no-compile
 	cp $(SCRIPTS) $(DESTDIR)$(BINDIR)
 	ln -sf edit-patch $(DESTDIR)$(BINDIR)/add-patch
 	install -d $(DESTDIR)$(COMPL_DIR)
