From e3d5bac8ffa389614ff81098999f2edc58f610db Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 21 Oct 2025 10:49:51 +0800
Subject: [PATCH 2/2] AOSCOS: fix(scripts/Makefile): drop --install-layout=deb
 option

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 scripts/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/Makefile b/scripts/Makefile
index cf663995..d4531e87 100644
--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -174,7 +174,7 @@ clean:
 test: test_pl test_sh test_py
 
 install: all
-	python3 setup.py install --root="$(DESTDIR)" --no-compile --install-layout=deb
+	python3 setup.py install --root="$(DESTDIR)" --no-compile
 	cp $(SCRIPTS) $(DESTDIR)$(BINDIR)
 	ln -sf edit-patch $(DESTDIR)$(BINDIR)/add-patch
 	install -d $(DESTDIR)$(COMPL_DIR)
-- 
2.51.0

