From 5e71cd4d958f2a4d739f978f46eb7a7e811de8e0 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Wed, 4 Jun 2025 10:45:50 +0800
Subject: [PATCH] disable anonymous data collection by default

---
 conf/configuration.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conf/configuration.go b/conf/configuration.go
index e2aca6c1..046bc5a6 100644
--- a/conf/configuration.go
+++ b/conf/configuration.go
@@ -612,7 +612,7 @@ func setViperDefaults() {
 	viper.SetDefault("defaultshareexpiration", 8760*time.Hour)
 	viper.SetDefault("defaultdownloadableshare", false)
 	viper.SetDefault("gatrackingid", "")
-	viper.SetDefault("enableinsightscollector", true)
+	viper.SetDefault("enableinsightscollector", false)
 	viper.SetDefault("enablelogredacting", true)
 	viper.SetDefault("authrequestlimit", 5)
 	viper.SetDefault("authwindowlength", 20*time.Second)
-- 
2.53.0

