--- a/doc/tm/Makefile
+++ b/doc/tm/Makefile
@@ -3,7 +3,7 @@
 # To create the SGML files, you will need to install the tmexpand
 # package.  See http://www.jedsoft.org/ for more information.
 #
-TM2SGML = tmexpand
+TM2SGML = tmexpand -I/usr/share/tmexpand/macros/
 SGML2HTML = sgml2html
 DOCBOOK2MAN = docbook2man
 DOCBOOK2TXT = docbook2txt
@@ -22,6 +22,6 @@
 clean:
 	-rm -f *~ *-error manpage.refs manpage.links
 distclean: clean
-	-rm -f most.1 most.sgml
+	-rm -f most.1 most.sgml most.txt
 install: most.1 most.txt
 	mv most.1 most.txt ..
--- a/doc/tm/most.tm
+++ b/doc/tm/most.tm
@@ -178,9 +178,15 @@
 
 \refsect1{COMMAND USAGE}
 \p
-   The commands take effect immediately; it is not necessary to type a
-   carriage return.  In the following commands, \i is a numerical
-   argument (1 by default).
+   Some commands have an optional numeric argument id (\i) entered before the
+   command.  If not given, the default value of \i is 1.
+\p
+   Commands take effect immediately; typing a carriage return isn't
+   necessary or correct.  (Besides, the carriage return itself is a
+   command.)
+\p
+   Example: to go down one line, press the DOWN_ARROW key. To go down 20
+   lines, press 2 then 0 then the DOWN_ARROW key.
 \desc-list
 \desc-item{SPACE, CTRL-D, NEXT_SCREEN}{
    Display another windowful, or jump \i windowfuls if \i is specified.}
@@ -339,8 +345,8 @@
 
 \desc-item{MOST_INITFILE}{Set this variable to specify the
 initialization file to load during startup.  The default action is to
-load the system configuration file and then a personal configuration
-file called .mostrc on Unix, and most.rc on other systems.}
+load the system configuration file from /etc/most.conf and then a
+personal configuration file located at $HOME/.mostrc}
 \desc-list-end
 \refsect1-end
 
