Description: add support for XV compressed data
Author: Alberto Garcia <berto@igalia.com>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=846465

--- a/src/file.c
+++ b/src/file.c
@@ -120,6 +120,13 @@
 	  {
 	     cmd = MOST_LZMA_POPEN_FORMAT;
 	  }
+	else if ((buf[0] == (char) 0xfd)
+	    && (buf[1] == '7') && (buf[2] == 'z')
+	    && (buf[3] == 'X') && (buf[4] == 'Z')
+	    && (buf[5] == (char) 0x00))
+	  {
+	     cmd = MOST_XZ_POPEN_FORMAT;
+	  }
 
 	if (cmd != NULL)
 	  {
--- a/src/file.h
+++ b/src/file.h
@@ -23,6 +23,7 @@
 #define MOST_GUNZIP_POPEN_FORMAT "gzip -dc \"%s\""
 #define MOST_BZIP2_POPEN_FORMAT "bzip2 -dc \"%s\""
 #define MOST_LZMA_POPEN_FORMAT "lzma -dc \"%s\""
+#define MOST_XZ_POPEN_FORMAT "xz -dc \"%s\""
 
 extern void most_reread_file (void);
 extern void most_read_to_line (MOST_INT);
