diff -Naur libilbc-3.0.4.orig/rtc_base/system/arch.h libilbc-3.0.4/rtc_base/system/arch.h
--- libilbc-3.0.4.orig/rtc_base/system/arch.h	2023-09-22 02:34:22.960691579 +0000
+++ libilbc-3.0.4/rtc_base/system/arch.h	2023-09-22 02:35:32.291912640 +0000
@@ -73,6 +73,9 @@
 #elif defined(__riscv) && __riscv_xlen == 32
 #define WEBRTC_ARCH_32_BITS
 #define WEBRTC_ARCH_LITTLE_ENDIAN
+#elif defined(__loongarch64)
+#define WEBRTC_ARCH_64_BITS
+#define WEBRTC_ARCH_LITTLE_ENDIAN
 #elif defined(__pnacl__)
 #define WEBRTC_ARCH_32_BITS
 #define WEBRTC_ARCH_LITTLE_ENDIAN
