From 743e59702c970c9eb54734378efd8609c32e4178 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <origincode@aosc.io>
Date: Wed, 27 Jul 2022 01:17:42 -0700
Subject: [PATCH] refactor: use sane jar installation paths

Signed-off-by: Kaiyang Wu <origincode@aosc.io>
---
 CMakeLists.txt           | 2 +-
 launcher/Application.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33c53b82..edfbf020 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -250,7 +250,7 @@ if(UNIX AND APPLE)
 elseif(UNIX)
     set(BINARY_DEST_DIR "bin")
     set(LIBRARY_DEST_DIR "lib${LIB_SUFFIX}")
-    set(JARS_DEST_DIR "share/jars")
+    set(JARS_DEST_DIR "lib/polymc")
     set(LAUNCHER_DESKTOP_DEST_DIR "share/applications" CACHE STRING "Path to the desktop file directory")
     set(LAUNCHER_METAINFO_DEST_DIR "share/metainfo" CACHE STRING "Path to the metainfo directory")
     set(LAUNCHER_ICON_DEST_DIR "share/icons/hicolor/scalable/apps" CACHE STRING "Path to the scalable icon directory")
diff --git a/launcher/Application.cpp b/launcher/Application.cpp
index 2bd91fd7..aa180b05 100644
--- a/launcher/Application.cpp
+++ b/launcher/Application.cpp
@@ -1578,7 +1578,7 @@ QString Application::getJarPath(QString jarFile)
 {
     QStringList potentialPaths = {
 #if defined(Q_OS_LINUX) || defined(Q_OS_FREEBSD) || defined(Q_OS_OPENBSD)
-        FS::PathCombine(m_rootPath, "share/jars"),
+        FS::PathCombine(m_rootPath, "lib/polymc"),
 #endif
         FS::PathCombine(m_rootPath, "jars"),
         FS::PathCombine(applicationDirPath(), "jars")
-- 
2.37.1

