From 370b0d7a591688a18f199ab8ea88eaa907c1630d Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <the@salted.fish>
Date: Sat, 18 Apr 2026 19:57:51 -0400
Subject: [PATCH 2/5] FEDORA: Do not export TLS libs

---
 config-scripts/cups-tls.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config-scripts/cups-tls.m4 b/config-scripts/cups-tls.m4
index 97c6fa293..d6b954d7b 100644
--- a/config-scripts/cups-tls.m4
+++ b/config-scripts/cups-tls.m4
@@ -150,5 +150,5 @@ AC_SUBST([IPPALIASES])
 AC_SUBST([TLSFLAGS])
 AC_SUBST([TLSLIBS])
 
-EXPORT_TLSLIBS="$TLSLIBS"
+EXPORT_TLSLIBS=""
 AC_SUBST([EXPORT_TLSLIBS])
-- 
2.52.0

