From c1cb1aefb5e7700710aed8f7924ba29599c849e1 Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <the@salted.fish>
Date: Sat, 18 Apr 2026 20:04:24 -0400
Subject: [PATCH 3/5] FEDORA: Use IP_FREEBIND when binding sockets

---
 cups/http-addr.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cups/http-addr.c b/cups/http-addr.c
index f93f00fa5..56fa30549 100644
--- a/cups/http-addr.c
+++ b/cups/http-addr.c
@@ -188,6 +188,10 @@ httpAddrListen(http_addr_t *addr,	/* I - Address to bind to */
   val = 1;
   setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, CUPS_SOCAST &val, sizeof(val));
 
+#ifdef __linux
+  setsockopt(fd, IPPROTO_IP, IP_FREEBIND, CUPS_SOCAST &val, sizeof(val));
+#endif /* __linux */
+
 #ifdef IPV6_V6ONLY
   if (addr->addr.sa_family == AF_INET6)
     setsockopt(fd, IPPROTO_IPV6, IPV6_V6ONLY, CUPS_SOCAST &val, sizeof(val));
-- 
2.52.0

