From b9986fd6d46e438c51c9be21e2d23f2d4488aee6 Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <the@salted.fish>
Date: Sat, 18 Apr 2026 20:05:22 -0400
Subject: [PATCH 4/5] FEDORA: Fix for jobs with multiple files and multiple
 formats

---
 backend/ipp.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/backend/ipp.c b/backend/ipp.c
index 4d1f308bd..ea7f76b65 100644
--- a/backend/ipp.c
+++ b/backend/ipp.c
@@ -1867,7 +1867,10 @@ main(int  argc,				/* I - Number of command-line args */
 	ippAddBoolean(request, IPP_TAG_OPERATION, "last-document",
         	      (i + 1) >= num_files);
 
-	if (document_format)
+	if (num_files > 1)
+	  ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_MIMETYPE,
+		       "document-format", NULL, "application/octet-stream");
+	else if (document_format)
 	  ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_MIMETYPE,
 		       "document-format", NULL, document_format);
 
-- 
2.52.0

