From b13527b81a4872772669fbcd73135121ce5fc40f Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <the@salted.fish>
Date: Sat, 18 Apr 2026 20:07:24 -0400
Subject: [PATCH 5/5] FEDORA: Add IEEE 1284 Device ID for a Dymo label printer

---
 ppdc/sample.drv | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ppdc/sample.drv b/ppdc/sample.drv
index 1c84aa3bf..97cecbf12 100644
--- a/ppdc/sample.drv
+++ b/ppdc/sample.drv
@@ -138,6 +138,7 @@ Version "2.4"
 {
   Manufacturer "DYMO"
   ModelName "Label Printer"
+  Attribute "1284DeviceID" "" "MFG:DYMO;MDL:LabelWriter 400;"
   Attribute NickName "" "DYMO Label Printer"
   PCFileName "dymo.ppd"
   DriverType label
-- 
2.52.0

