diff -Naur firefox-78.15.0/python/mozbuild/mozbuild/backend/configenvironment.py firefox-78.15.0-new/python/mozbuild/mozbuild/backend/configenvironment.py
--- firefox-78.15.0/python/mozbuild/mozbuild/backend/configenvironment.py	2021-09-28 10:04:57.000000000 +0000
+++ firefox-78.15.0-new/python/mozbuild/mozbuild/backend/configenvironment.py	2021-12-08 07:47:27.141438378 +0000
@@ -9,7 +9,8 @@
 import sys
 import json
 
-from collections import Iterable, OrderedDict
+from collections import OrderedDict
+from collections.abc import Iterable
 from types import ModuleType
 
 import mozpack.path as mozpath
diff -Naur firefox-78.15.0/python/mozbuild/mozbuild/makeutil.py firefox-78.15.0-new/python/mozbuild/mozbuild/makeutil.py
--- firefox-78.15.0/python/mozbuild/mozbuild/makeutil.py	2021-09-28 10:04:57.000000000 +0000
+++ firefox-78.15.0-new/python/mozbuild/mozbuild/makeutil.py	2021-12-08 07:48:40.069840520 +0000
@@ -7,7 +7,7 @@
 import os
 import re
 import six
-from collections import Iterable
+from collections.abc import Iterable
 
 
 class Makefile(object):
diff -Naur firefox-78.15.0/python/mozbuild/mozbuild/util.py firefox-78.15.0-new/python/mozbuild/mozbuild/util.py
--- firefox-78.15.0/python/mozbuild/mozbuild/util.py	2021-09-28 10:04:57.000000000 +0000
+++ firefox-78.15.0-new/python/mozbuild/mozbuild/util.py	2021-12-08 07:46:06.437856711 +0000
@@ -782,7 +782,7 @@
         self._strings = StrictOrderingOnAppendList()
         self._children = {}
 
-    class StringListAdaptor(collections.Sequence):
+    class StringListAdaptor(collections.abc.Sequence):
         def __init__(self, hsl):
             self._hsl = hsl
 
