diff -Naur calibre-3.10.0/src/calibre/gui2/ui.py calibre-3.10.0.noupdate/src/calibre/gui2/ui.py
--- calibre-3.10.0/src/calibre/gui2/ui.py	2017-10-20 03:44:45.000000000 +0000
+++ calibre-3.10.0.noupdate/src/calibre/gui2/ui.py	2017-10-20 13:38:39.718866955 +0000
@@ -342,9 +342,6 @@
         self.location_manager.update_device_metadata.connect(self.update_metadata_on_device)
         self.eject_action.triggered.connect(self.device_manager.umount_device)
 
-        # ################### Update notification ###################
-        UpdateMixin.init_update_mixin(self, opts)
-
         # ###################### Search boxes ########################
         SearchRestrictionMixin.init_search_restriction_mixin(self)
         SavedSearchBoxMixin.init_saved_seach_box_mixin(self)

