From 8bed72c59814c6df0e8e217a896fa93916fd8c40 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Wed, 2 Oct 2024 21:23:31 +0000
Subject: [PATCH 1/2] skip update checker

---
 src/calibre/gui2/ui.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/calibre/gui2/ui.py b/src/calibre/gui2/ui.py
index ca0fc26927..2c452509be 100644
--- a/src/calibre/gui2/ui.py
+++ b/src/calibre/gui2/ui.py
@@ -339,9 +339,6 @@ def initialize(self, library_path, db, actions, show_gui=True):
         self.location_manager.update_device_metadata.connect(self.update_metadata_on_device)
         self.eject_action.triggered.connect(self.device_manager.umount_device)
 
-        # ################### Update notification ###################
-        UpdateMixin.init_update_mixin(self, opts)
-
         # ###################### Search boxes ########################
         SearchRestrictionMixin.init_search_restriction_mixin(self)
         SavedSearchBoxMixin.init_saved_seach_box_mixin(self)
-- 
2.46.2

