From 46b94b1009c2b8da5c398f5fd7d64d6f2c48c80a Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Mon, 17 Nov 2025 05:54:37 +0800
Subject: [PATCH 3/3] Support 64 bit LoongArch architecture

---
 src/calibre/utils/stb_sprintf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/calibre/utils/stb_sprintf.h b/src/calibre/utils/stb_sprintf.h
index 0635360..4c0c195 100644
--- a/src/calibre/utils/stb_sprintf.h
+++ b/src/calibre/utils/stb_sprintf.h
@@ -215,7 +215,7 @@ STBSP__PUBLICDEF void STB_SPRINTF_DECORATE(set_separators)(char comma, char peri
 #define stbsp__uint16 unsigned short
 
 #ifndef stbsp__uintptr
-#if defined(__ppc64__) || defined(__powerpc64__) || defined(__aarch64__) || defined(_M_X64) || defined(__x86_64__) || defined(__x86_64)
+#if defined(__ppc64__) || defined(__powerpc64__) || defined(__aarch64__) || defined(_M_X64) || defined(__x86_64__) || defined(__x86_64) || defined(__loongarch_lp64)
 #define stbsp__uintptr stbsp__uint64
 #else
 #define stbsp__uintptr stbsp__uint32
-- 
2.51.1

