--- a/hupnp/src/devicehosting/devicehost/hservermodel_creator_p.cpp	2011-03-28 00:42:32.000000000 -0500
+++ b/hupnp/src/devicehosting/devicehost/hservermodel_creator_p.cpp	2016-02-29 16:57:50.879832371 -0600
@@ -493,7 +493,7 @@ HServerDevice* HServerModelCreator::pars
             QString("Failed to initialize device [%1]").arg(
                 deviceInfo.udn().toString());
 
-        return false;
+        return 0;
     }
 
     device->h_ptr->m_deviceDescription =
@@ -560,7 +560,7 @@ HServerDevice* HServerModelCreator::pars
                     "[%2]").arg(device->info().deviceType().toString(),
                                 dt.toString());
 
-                return false;
+                return 0;
             }
         }
     }
