diff --git a/NAPS2.App.Gtk/NAPS2.App.Gtk.csproj b/NAPS2.App.Gtk/NAPS2.App.Gtk.csproj
index 784a6a27d..1e1282838 100644
--- a/NAPS2.App.Gtk/NAPS2.App.Gtk.csproj
+++ b/NAPS2.App.Gtk/NAPS2.App.Gtk.csproj
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">

     <PropertyGroup>
-        <TargetFramework>net9</TargetFramework>
+        <TargetFramework>net10.0</TargetFramework>
         <OutputType>Exe</OutputType>
         <RootNamespace>NAPS2</RootNamespace>
         <AssemblyName>naps2</AssemblyName>
diff --git a/NAPS2.Lib.Gtk/NAPS2.Lib.Gtk.csproj b/NAPS2.Lib.Gtk/NAPS2.Lib.Gtk.csproj
index a2435d42f..8bcac2bf8 100644
--- a/NAPS2.Lib.Gtk/NAPS2.Lib.Gtk.csproj
+++ b/NAPS2.Lib.Gtk/NAPS2.Lib.Gtk.csproj
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">

     <PropertyGroup>
-        <TargetFrameworks>net9</TargetFrameworks>
+        <TargetFrameworks>net10.0</TargetFrameworks>
         <Nullable>enable</Nullable>
         <GenerateResourceUsePreserializedResources>true</GenerateResourceUsePreserializedResources>
         <RootNamespace>NAPS2</RootNamespace>
diff --git a/NAPS2.Lib/NAPS2.Lib.csproj b/NAPS2.Lib/NAPS2.Lib.csproj
index 88eb1d343..ed82d6643 100644
--- a/NAPS2.Lib/NAPS2.Lib.csproj
+++ b/NAPS2.Lib/NAPS2.Lib.csproj
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">

   <PropertyGroup>
-    <TargetFrameworks>net9</TargetFrameworks>
+    <TargetFrameworks>net10.0</TargetFrameworks>
     <TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('OSX'))">
       $(TargetFrameworks);net9-macos
     </TargetFrameworks>
