diff -up a/libraries/source/fcollada/src/Makefile.orig b/libraries/source/fcollada/src/Makefile
--- a/libraries/source/fcollada/src/Makefile.orig	2021-02-22 11:16:57.680708209 -0300
+++ b/libraries/source/fcollada/src/Makefile	2021-02-22 11:23:49.577033354 -0300
@@ -9,13 +9,10 @@ endif
 
 CXX ?= g++
 CXXFLAGS += -fvisibility=hidden -W -Wall -Wno-unused-parameter -Wno-unused-function $(OS_DEFINE) $(PIC_FLAGS) $(CPPFLAGS)
-CXXFLAGS_DEBUG := -O0 -g -D_DEBUG -DRETAIL
 CXXFLAGS_RELEASE := -O2 -DNDEBUG -DRETAIL
-CXXFLAGS_TEST := -O0 -g -D_DEBUG
 LDFLAGS_TEST= -ldl
 LIBS += `pkg-config libxml-2.0 --libs`
 INCLUDES += -IFCollada `pkg-config libxml-2.0 --cflags`
-INCLUDES_TEST := -IFCollada/FColladaTest $(INCLUDES)
 
 # FCollada is not aliasing-safe, so disallow dangerous optimisations
 # (TODO: It'd be nice to fix FCollada, but that looks hard)
@@ -181,57 +178,16 @@ SOURCE = \
 	FColladaPlugins/FArchiveXML/FAXSceneExport.cpp \
 	FColladaPlugins/FArchiveXML/FAXSceneImport.cpp \
 
-TEST_SOURCE = \
-	FCollada/FMath/FMArrayTest.cpp \
-	FCollada/FMath/FMQuaternionTest.cpp \
-	FCollada/FMath/FMTreeTest.cpp \
-	FCollada/FUtils/FUBoundingTest.cpp \
-	FCollada/FUtils/FUCrc32Test.cpp \
-	FCollada/FUtils/FUEventTest.cpp \
-	FCollada/FUtils/FUFileManagerTest.cpp \
-	FCollada/FUtils/FUFunctorTest.cpp \
-	FCollada/FUtils/FUObjectTest.cpp \
-	FCollada/FUtils/FUStringBuilderTest.cpp \
-	FCollada/FUtils/FUStringConversionTest.cpp \
-	FCollada/FUtils/FUStringTest.cpp \
-	FCollada/FUtils/FUTestBed.cpp \
-	FCollada/FUtils/FUUniqueStringMapTest.cpp \
-	FCollada/FColladaTest/FCTest.cpp \
-	FCollada/FColladaTest/FCTestAnimation.cpp \
-	FCollada/FColladaTest/FCTestArchiving.cpp \
-	FCollada/FColladaTest/FCTestController.cpp \
-	FCollada/FColladaTest/FCTestGeometryPolygonsTools.cpp \
-	FCollada/FColladaTest/FCTestParameters.cpp \
-	FCollada/FColladaTest/FCTestSceneGraph.cpp \
-	FCollada/FColladaTest/FCTestAssetManagement/FCTAMCrossCloning.cpp \
-	FCollada/FColladaTest/FCTestAssetManagement/FCTAssetManagement.cpp \
-	FCollada/FColladaTest/FCTestExportImport/FCTEIAnimation.cpp \
-	FCollada/FColladaTest/FCTestExportImport/FCTEICamera.cpp \
-	FCollada/FColladaTest/FCTestExportImport/FCTEIEmitter.cpp \
-	FCollada/FColladaTest/FCTestExportImport/FCTEIExtra.cpp \
-	FCollada/FColladaTest/FCTestExportImport/FCTEIGeometry.cpp \
-	FCollada/FColladaTest/FCTestExportImport/FCTEILight.cpp \
-	FCollada/FColladaTest/FCTestExportImport/FCTEIMaterial.cpp \
-	FCollada/FColladaTest/FCTestExportImport/FCTEIPhysics.cpp \
-	FCollada/FColladaTest/FCTestExportImport/FCTEIVisualScene.cpp \
-	FCollada/FColladaTest/FCTestExportImport/FCTestExportImport.cpp \
-	FCollada/FColladaTest/FCTestXRef/FCTestXRef.cpp \
-	FCollada/FColladaTest/FCTestXRef/FCTestXRefAcyclic.cpp \
-	FCollada/FColladaTest/FCTestXRef/FCTestXRefSimple.cpp \
-	FCollada/FColladaTest/FCTestXRef/FCTestXRefTree.cpp \
-
-OBJECTS_DEBUG = $(addprefix output/debug/,$(SOURCE:.cpp=.o))
 OBJECTS_RELEASE = $(addprefix output/release/,$(SOURCE:.cpp=.o))
-OBJECTS_TEST = $(addprefix output/test/,$(SOURCE:.cpp=.o) $(TEST_SOURCE:.cpp=.o))
-OBJECTS_ALL = $(OBJECTS_DEBUG) $(OBJECTS_RELEASE) $(OBJECTS_TEST)
+OBJECTS_ALL = $(OBJECTS_RELEASE)
 
-all: output/libFColladaSD.a output/libFColladaSR.a install
+all: output/libFColladaSR.a install
 
 output_dirs:
-	@for x in debug release test; \
+	@for x in release; \
 	do \
 		mkdir -p output/$$x/FColladaPlugins/FArchiveXML; \
-		for y in FCDocument FMath FUtils FColladaTest; \
+		for y in FCDocument FMath FUtils; \
 		do \
 			for z in FCTestAssetManagement FCTestExportImport FCTestXRef; \
 			do \
@@ -240,23 +196,11 @@ output_dirs:
 		done \
 	done
 
-test: FCollada/FColladaTest/ output/FColladaTest
-	(cd FCollada/FColladaTest/ ; ../../output/FColladaTest )
-	(cp -r FCollada/FColladaTest/Samples/ output/Samples && cd output/ && ./FColladaTest && cat FColladaTestLog.txt)
-
-output/libFColladaSD.a: $(OBJECTS_DEBUG) | output_dirs
-	@echo "$@"
-	@ar -cr $@ $(OBJECTS_DEBUG); ranlib $@
-
 output/libFColladaSR.a: $(OBJECTS_RELEASE) | output_dirs
 	@echo "$@"
 	@ar -cr $@ $(OBJECTS_RELEASE); ranlib $@
 
-output/FColladaTest: $(OBJECTS_TEST) | output_dirs
-	$(CXX) -o $@  $(OBJECTS_TEST) $(LIBS) $(LDFLAGS_TEST)
-
-install: output/libFColladaSD.a output/libFColladaSR.a
-	cp output/libFColladaSD.a ../lib/libFColladaSD.a
+install: output/libFColladaSR.a
 	cp output/libFColladaSR.a ../lib/libFColladaSR.a
 
 dfile = $(@:.o=.d)
@@ -266,18 +210,10 @@ gendep = \
 		sed -e 's/\#.*//' -e 's/^[^:]*: *//' -e 's/ *\\$$//' \
 		    -e '/^$$/ d' -e 's/$$/ :/' < $(dfile) >> $(Pfile) && \
 		rm -f $(dfile)
-output/debug/%.o: %.cpp | output_dirs
-	@echo "$<"
-	$(CXX) $(CXXFLAGS) $(CXXFLAGS_DEBUG) $(INCLUDES) -MD -MF $(dfile) -c $< -o $@
-	$(gendep)
 output/release/%.o: %.cpp | output_dirs
 	@echo "$<"
 	$(CXX) $(CXXFLAGS) $(CXXFLAGS_RELEASE) $(INCLUDES) -MD -MF $(dfile) -c $< -o $@
 	$(gendep)
-output/test/%.o: %.cpp | output_dirs
-	@echo "$<"
-	$(CXX) $(CXXFLAGS) $(CXXFLAGS_TEST) $(INCLUDES_TEST) -MD -MF $(dfile) -c $< -o $@
-	$(gendep)
 
 clean:
 	rm -rf output
