diff -up a/build/premake/extern_libs5.lua.orig b/build/premake/extern_libs5.lua
--- a/build/premake/extern_libs5.lua.orig	2021-02-22 11:26:03.499823098 -0300
+++ b/build/premake/extern_libs5.lua	2021-02-22 11:26:16.945902393 -0300
@@ -623,7 +623,7 @@ extern_lib_defs = {
 	},
 	valgrind = {
 		compile_settings = function()
-			add_source_include_paths("valgrind")
+			pkgconfig.add_includes("valgrind")
 		end,
 	},
 	vorbis = {
