diff -Naur libraries/source/nvtt-bak/src/src/nvmath/SimdVector_VE.h libraries/source/nvtt/src/src/nvmath/SimdVector_VE.h
--- a/libraries/source/nvtt/src/src/nvmath/SimdVector_VE.h	2021-02-05 18:28:50.000000000 -0600
+++ b/libraries/source/nvtt/src/src/nvmath/SimdVector_VE.h	2021-05-19 16:59:00.873805972 -0500
@@ -27,7 +27,7 @@
 #ifndef NV_SIMD_VECTOR_VE_H
 #define NV_SIMD_VECTOR_VE_H
 
-#ifndef __APPLE_ALTIVEC__
+#if !defined(__APPLE_ALTIVEC__) || !defined(__APPLE__)
 #include <altivec.h>
 #undef bool
 #endif
diff -Naur libraries/source/nvtt-bak/src/src/nvtt/squish/simd_ve.h libraries/source/nvtt/src/src/nvtt/squish/simd_ve.h
--- a/libraries/source/nvtt/src/src/nvtt/squish/simd_ve.h	2021-02-05 18:28:59.000000000 -0600
+++ b/libraries/source/nvtt/src/src/nvtt/squish/simd_ve.h	2021-05-19 16:57:40.488085146 -0500
@@ -27,7 +27,7 @@
 #ifndef SQUISH_SIMD_VE_H
 #define SQUISH_SIMD_VE_H
 
-#ifndef __APPLE_ALTIVEC__
+#if !defined(__APPLE_ALTIVEC__) || !defined(__APPLE__)
 #include <altivec.h>
 #undef bool
 #endif
