diff -uNr 0ad-0.0.21-alpha.orig/build/premake/premake4.lua 0ad-0.0.21-alpha/build/premake/premake4.lua
--- 0ad-0.0.21-alpha.orig/build/premake/premake4.lua	2016-09-18 10:54:27.000000000 +0200
+++ 0ad-0.0.21-alpha/build/premake/premake4.lua	2016-11-09 06:47:02.330099676 +0100
@@ -368,26 +368,6 @@
 			defines { "INSTALLED_LIBDIR=" .. _OPTIONS["libdir"] }
 		end
 
-		if os.is("linux") or os.is("bsd") then
-			-- To use our local shared libraries, they need to be found in the
-			-- runtime dynamic linker path. Add their path to -rpath.
-			if _OPTIONS["libdir"] then
-				linkoptions {"-Wl,-rpath," .. _OPTIONS["libdir"] }
-			else
-				-- On FreeBSD we need to allow use of $ORIGIN
-				if os.is("bsd") then
-					linkoptions { "-Wl,-z,origin" }
-				end
-
-				-- Adding the executable path and taking care of correct escaping
-				if _ACTION == "gmake" then
-					linkoptions { "-Wl,-rpath,'$$ORIGIN'" }
-				elseif _ACTION == "codeblocks" then
-					linkoptions { "-Wl,-R\\\\$$$ORIGIN" }
-				end
-			end
-		end
-
 	end
 end
 
