diff -up build/premake/premake4.lua.orig build/premake/premake4.lua
--- a/build/premake/premake4.lua.orig	2015-03-14 13:29:39.458872061 -0300
+++ b/build/premake/premake4.lua	2015-03-14 13:30:04.556873022 -0300
@@ -365,27 +365,6 @@ function project_set_build_flags()
 		if _OPTIONS["libdir"] then
 			defines { "INSTALLED_LIBDIR=" .. _OPTIONS["libdir"] }
 		end
-
-		if os.is("linux") or os.is("bsd") then
-			-- To use our local shared libraries, they need to be found in the
-			-- runtime dynamic linker path. Add their path to -rpath.
-			if _OPTIONS["libdir"] then
-				linkoptions {"-Wl,-rpath," .. _OPTIONS["libdir"] }
-			else
-				-- On FreeBSD we need to allow use of $ORIGIN
-				if os.is("bsd") then
-					linkoptions { "-Wl,-z,origin" }
-				end
-
-				-- Adding the executable path and taking care of correct escaping
-				if _ACTION == "gmake" then
-					linkoptions { "-Wl,-rpath,'$$ORIGIN'" } 
-				elseif _ACTION == "codeblocks" then
-					linkoptions { "-Wl,-R\\\\$$$ORIGIN" }
-				end
-			end
-		end
-
 	end
 end
 

