Index: trunk/source/lib/sysdep/arch/x86_x64/cache.cpp
===================================================================
--- trunk/source/lib/sysdep/arch/x86_x64/cache.cpp	(revision 23261)
+++ trunk/source/lib/sysdep/arch/x86_x64/cache.cpp	(revision 23262)
@@ -90,5 +90,6 @@
 {
 	0, 1, 2, 0, 4, 0, 8, 0,
-	16, 0, 32, 48, 64, 96, 128, x86_x64::Cache::fullyAssociative
+	// TODO: The second '16' does not obey to the specifications and is only a workaround. For a correct implementation please look here: https://community.amd.com/thread/244207
+	16, 16, 32, 48, 64, 96, 128, x86_x64::Cache::fullyAssociative
 };
 
