--- edi-0.8.0/meson.build      2020-04-29 13:40:58.000000000 +0000
+++ edi-0.8.0-new/meson.build  2021-12-08 00:15:30.736938772 +0000
@@ -85,7 +85,7 @@
   clang_include_dir_command = run_command(find_program('scripts/clang_include_dir.sh'))
   clang_include_dir = clang_include_dir_command.stdout().strip()
 
-  config_h.set_quoted('CLANG_INCLUDES', clang_include_dir)
+  config_h.set_quoted('CLANG_INCLUDES', clang_include_dir.replace('\n', ' '))
   config_h.set('HAVE_LIBCLANG', '1')
 endif
 
