From 7fa1749d954e242db0eeaeab9830b4c10217ed96 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 16 Dec 2024 16:37:55 +0800
Subject: [PATCH 1/5] Fix impossible constraint error for mips and riscv

---
 folly/lang/Hint-inl.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/folly/lang/Hint-inl.h b/folly/lang/Hint-inl.h
index 85a3adc70..d877948bf 100644
--- a/folly/lang/Hint-inl.h
+++ b/folly/lang/Hint-inl.h
@@ -101,7 +101,11 @@ FOLLY_ALWAYS_INLINE void compiler_must_not_elide(T const& t, std::false_type) {
   //
   // avoided for pointers to avoid fallout in calling code which mistakenly
   // applies the hint to the address of a value but not to the value itself
+#if defined (__mips__) || defined (__riscv)
+  asm volatile("" : : "rm"(t) : "memory");
+#else
   asm volatile("" : : "r"(t));
+#endif
 }
 
 template <typename T>
-- 
2.52.0

