From 74b428ab2e31ba8144bee10caada0f5dd391fab8 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 16 Dec 2024 16:37:55 +0800
Subject: [PATCH 1/2] Fix impossible constraint error for mips and riscv

---
 folly/lang/Hint-inl.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/folly/lang/Hint-inl.h b/folly/lang/Hint-inl.h
index 2c207a551..63126226d 100644
--- a/folly/lang/Hint-inl.h
+++ b/folly/lang/Hint-inl.h
@@ -100,7 +100,11 @@ FOLLY_ALWAYS_INLINE void compiler_must_not_elide(T const& t, std::false_type) {
   //
   // avoided for pointers to avoid fallout in calling code which mistakenly
   // applies the hint to the address of a value but not to the value itself
+#if defined (__mips__) || defined (__riscv)
+  asm volatile("" : : "rm"(t) : "memory");
+#else
   asm volatile("" : : "r"(t));
+#endif
 }
 
 template <typename T>
-- 
2.49.0

