From a35b49cf5b48db65025f7e749b53feeb874fc56e Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Fri, 28 Mar 2025 16:23:27 +0800
Subject: [PATCH 2/2] Use 64-byte cacheline alignment on RISC-V

---
 folly/lang/Align.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/folly/lang/Align.h b/folly/lang/Align.h
index b541156e2..522c30f65 100644
--- a/folly/lang/Align.h
+++ b/folly/lang/Align.h
@@ -145,7 +145,7 @@ using max_align_v_ = max_align_t_<
 constexpr std::size_t max_align_v = detail::max_align_v_::value();
 struct alignas(max_align_v) max_align_t {};
 
-#if defined(__cpp_lib_hardware_interference_size)
+#if defined(__cpp_lib_hardware_interference_size) && !defined(__riscv)
 
 //  GCC unconditionally warns about uses of the std's interference-size
 //  constants, on the basis that their uses in public ABIs is likely broken:
-- 
2.49.0

