From 476ccde4f1f5e223e5c3fd8c640ae8d0b30448c3 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Fri, 28 Mar 2025 16:23:27 +0800
Subject: [PATCH 2/5] Use 64-byte cacheline alignment on riscv64 and loongson3

---
 folly/lang/Align.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/folly/lang/Align.h b/folly/lang/Align.h
index 4642880c0..a779a46b8 100644
--- a/folly/lang/Align.h
+++ b/folly/lang/Align.h
@@ -145,7 +145,7 @@ using max_align_v_ = max_align_t_<
 constexpr std::size_t max_align_v = detail::max_align_v_::value();
 struct alignas(max_align_v) max_align_t {};
 
-#if defined(__cpp_lib_hardware_interference_size)
+#if defined(__cpp_lib_hardware_interference_size) && !defined(__riscv) && !defined(__mips64)
 
 //  GCC unconditionally warns about uses of the std's interference-size
 //  constants, on the basis that their uses in public ABIs is likely broken:
-- 
2.52.0

