diff --git a/CMake/folly-deps.cmake b/CMake/folly-deps.cmake
index 7dafece7d..ff56dc9bc 100644
--- a/CMake/folly-deps.cmake
+++ b/CMake/folly-deps.cmake
@@ -64,10 +64,18 @@ if(LIBGFLAGS_FOUND)
   set(FOLLY_LIBGFLAGS_INCLUDE ${LIBGFLAGS_INCLUDE_DIR})
 endif()
 
-find_package(Glog MODULE)
-set(FOLLY_HAVE_LIBGLOG ${GLOG_FOUND})
-list(APPEND FOLLY_LINK_LIBRARIES ${GLOG_LIBRARY})
-list(APPEND FOLLY_INCLUDE_DIRECTORIES ${GLOG_INCLUDE_DIR})
+# Detect glog with CONFIG first
+find_package(glog CONFIG QUIET)
+if(glog_FOUND)
+  set(FOLLY_HAVE_LIBGLOG True)
+  list(APPEND FOLLY_LINK_LIBRARIES glog::glog)
+else()
+  # Fallback to module mode
+  find_package(Glog MODULE REQUIRED)
+  set(FOLLY_HAVE_LIBGLOG ${GLOG_FOUND})
+  list(APPEND FOLLY_LINK_LIBRARIES ${GLOG_LIBRARY})
+  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${GLOG_INCLUDE_DIR})
+endif()
 
 find_package(LibEvent MODULE REQUIRED)
 list(APPEND FOLLY_LINK_LIBRARIES ${LIBEVENT_LIB})
