From 3bf15951362c6b9c4b852cfb940612cc05bdfa7d Mon Sep 17 00:00:00 2001
From: Claude <noreply@anthropic.com>
Date: Fri, 6 Mar 2026 11:05:07 -0800
Subject: [PATCH 4/5] UPSTREAM: Fix aarch64 duplicate symbol errors in libfolly

---
 folly/external/aor/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/folly/external/aor/CMakeLists.txt b/folly/external/aor/CMakeLists.txt
index e07e58745..c7b6e9338 100644
--- a/folly/external/aor/CMakeLists.txt
+++ b/folly/external/aor/CMakeLists.txt
@@ -34,6 +34,7 @@ folly_add_library(
 
 folly_add_library(
   NAME memcpy_aarch64-use
+  EXCLUDE_FROM_MONOLITH
   SRCS
     memcpy-advsimd.S
     memcpy-armv8.S
@@ -58,6 +59,7 @@ folly_add_library(
 
 folly_add_library(
   NAME memset_aarch64-use
+  EXCLUDE_FROM_MONOLITH
   SRCS
     memset-advsimd.S
     memset-mops.S
-- 
2.52.0

