From 9d79a0105d12833f3ee58b2f2f972b767e99175a Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Tue, 30 Jun 2026 17:19:52 +0800
Subject: [PATCH 5/5] Fix _FORTIFY_SOURCES + clang + glibc >= 2.40 death combo

---
 folly/Subprocess.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/folly/Subprocess.cpp b/folly/Subprocess.cpp
index 43cb04206..ebe1f5edc 100644
--- a/folly/Subprocess.cpp
+++ b/folly/Subprocess.cpp
@@ -148,6 +148,11 @@ static Ret subprocess_libc_load(
 #define FOLLY_DETAIL_SUBPROCESS_LIBC_X_OPEN(X) \
   X(open, __open_real)                         \
   X(openat, __openat_real)
+#elif defined(_FORTIFY_SOURCE) && defined(__clang__) && (__GLIBC__ > 2 || __GLIBC__ == 2 && __GLIBC_MINOR__ >= 40)
+// _FORTIFY_SOURCE overloads openat() on glibc >=2.40, so pick our overload explicitly.
+#define FOLLY_DETAIL_SUBPROCESS_LIBC_X_OPEN(X) \
+  X(open, open)                                \
+  X(openat, __openat_alias)
 #else
 #define FOLLY_DETAIL_SUBPROCESS_LIBC_X_OPEN(X) \
   X(open, open)                                \
-- 
2.52.0

