From f6f5417e5ae0e8b5a1e34d99d74123069613ae2c Mon Sep 17 00:00:00 2001
From: RawDiamondMC <RawDiamondMC@aosc.io>
Date: Tue, 28 Jul 2026 16:46:23 +0800
Subject: [PATCH 4/8] Use asmc64 on amd64

Signed-off-by: RawDiamondMC <RawDiamondMC@aosc.io>
---
 CPP/7zip/7zip_gcc.mak | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CPP/7zip/7zip_gcc.mak b/CPP/7zip/7zip_gcc.mak
index c766082..a96b930 100644
--- a/CPP/7zip/7zip_gcc.mak
+++ b/CPP/7zip/7zip_gcc.mak
@@ -1264,6 +1264,7 @@ ifdef USE_ASM
 ifdef IS_X64
 USE_X86_ASM=1
 USE_X64_ASM=1
+MY_ASM=asmc64
 else
 ifdef IS_X86
 USE_X86_ASM=1
-- 
2.52.0

