From da4b85a2cb7f4babdca30bb87127c37c12bdfdfd Mon Sep 17 00:00:00 2001
From: RawDiamondMC <RawDiamondMC@aosc.io>
Date: Tue, 28 Jul 2026 16:48:34 +0800
Subject: [PATCH 5/8] Compile for 80486 by default

Signed-off-by: RawDiamondMC <RawDiamondMC@aosc.io>
---
 CPP/7zip/7zip_gcc.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CPP/7zip/7zip_gcc.mak b/CPP/7zip/7zip_gcc.mak
index a96b930..3aa4dea 100644
--- a/CPP/7zip/7zip_gcc.mak
+++ b/CPP/7zip/7zip_gcc.mak
@@ -186,7 +186,7 @@ else  # IS_MINGW
 ifdef IS_X64
 AFLAGS_ABI = -elf64 -DABI_LINUX
 else
-AFLAGS_ABI = -elf -DABI_LINUX -DABI_CDECL
+AFLAGS_ABI = -4 -elf -DABI_LINUX -DABI_CDECL
 # -DABI_CDECL
 # -DABI_LINUX
 # -DABI_CDECL
-- 
2.52.0

