From ff52cf8899724e4869e451cdd4fb4592b69ce45c Mon Sep 17 00:00:00 2001
From: RawDiamondMC <RawDiamondMC@aosc.io>
Date: Tue, 28 Jul 2026 18:42:06 +0800
Subject: [PATCH 6/8] Disable use_vaes_256 in AesOpt as it's not supported on
 i486

Signed-off-by: RawDiamondMC <RawDiamondMC@aosc.io>
---
 Asm/x86/AesOpt.asm | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Asm/x86/AesOpt.asm b/Asm/x86/AesOpt.asm
index 84bf897..ec34998 100644
--- a/Asm/x86/AesOpt.asm
+++ b/Asm/x86/AesOpt.asm
@@ -4,14 +4,12 @@
 include 7zAsm.asm
 
 ifdef __ASMC__
-  use_vaes_256 equ 1
 else
 ifdef ymm0
   use_vaes_256 equ 1
 endif
 endif
 
-
 ifdef use_vaes_256
   ECHO "++ VAES 256"
 else
-- 
2.52.0

