diff -Naur mupen64plus-bundle-src-2.5.9.old/source/mupen64plus-video-rice/projects/unix/Makefile mupen64plus-bundle-src-2.5.9/source/mupen64plus-video-rice/projects/unix/Makefile
--- mupen64plus-bundle-src-2.5.9.old/source/mupen64plus-video-rice/projects/unix/Makefile	2020-02-22 02:46:19.434807182 +0800
+++ mupen64plus-bundle-src-2.5.9/source/mupen64plus-video-rice/projects/unix/Makefile	2020-02-22 02:48:03.845474253 +0800
@@ -113,6 +113,14 @@
     $(warning Architecture "$(HOST_CPU)" not officially supported.')
   endif
 endif
+ifneq ("$(filter aarch64,$(HOST_CPU))","")
+    CPU := AARCH
+    ARCH_DETECTED := 64BITS
+    PIC ?= 1
+    NO_ASM := 1
+    $(warning Architecture "$(HOST_CPU)" not officially supported.')
+endif
+
 ifeq ("$(CPU)","NONE")
   $(error CPU type "$(HOST_CPU)" not supported.  Please file bug report at 'https://github.com/mupen64plus/mupen64plus-core/issues')
 endif
