--- kbiff-trinity-14.0.10/ConfigureChecks.cmake	2021-02-27 16:42:51.000000000 -0800
+++ kbiff-trinity-14.0.10.mled/ConfigureChecks.cmake	2021-06-26 22:01:54.816370259 -0700
@@ -40,14 +40,3 @@
   # Therefore, there is no need for a direct dependence on the SSL library.
   set( USE_SSL 1 )
 endif( WITH_SSL )
-
-
-#### check for ASUS Mail LED
-
-if( WITH_MLED )
-  if( EXISTS "/proc/acpi/asus/mled" )
-    set( HAVE_MLED 1 )
-  else( )
-    tde_mesage_fatal( "ASUS Mail LED support was requested but not found in ACPI" )
-  endif( )
-endif( WITH_MLED )
