diff -ru liborcus-0.17.2.orig/include/orcus/base64.hpp liborcus-0.17.2/include/orcus/base64.hpp
--- liborcus-0.17.2.orig/include/orcus/base64.hpp	2023-01-23 20:34:32.641993292 +0000
+++ liborcus-0.17.2/include/orcus/base64.hpp	2023-01-23 20:43:45.653157969 +0000
@@ -9,8 +9,9 @@
 #define __ORCUS_BASE64_HPP__
 
 #include "env.hpp"
-#include <vector>
+#include <cstdint>
 #include <string>
+#include <vector>
 
 namespace orcus {
 
diff -ru liborcus-0.17.2.orig/include/orcus/types.hpp liborcus-0.17.2/include/orcus/types.hpp
--- liborcus-0.17.2.orig/include/orcus/types.hpp	2023-01-23 20:34:32.641993292 +0000
+++ liborcus-0.17.2/include/orcus/types.hpp	2023-01-23 20:35:00.060249357 +0000
@@ -8,6 +8,7 @@
 #ifndef INCLUDED_ORCUS_TYPES_HPP
 #define INCLUDED_ORCUS_TYPES_HPP
 
+#include <cstdint>
 #include <cstdlib>
 #include <vector>
 #include <string>
