diff -Naur osu-2021.820.0/osu.Android.props osu-2021.820.0.resources/osu.Android.props
--- osu-2021.820.0/osu.Android.props	2021-08-20 07:47:35.000000000 -0700
+++ osu-2021.820.0.resources/osu.Android.props	2021-08-26 23:41:31.999449812 -0700
@@ -51,7 +51,7 @@
     <Reference Include="Java.Interop" />
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="ppy.osu.Game.Resources" Version="2021.820.0" />
+    <PackageReference Include="ppy.osu.Game.Resources" Version="2021.827.0" />
     <PackageReference Include="ppy.osu.Framework.Android" Version="2021.819.0" />
   </ItemGroup>
   <ItemGroup Label="Transitive Dependencies">
diff -Naur osu-2021.820.0/osu.Game/osu.Game.csproj osu-2021.820.0.resources/osu.Game/osu.Game.csproj
--- osu-2021.820.0/osu.Game/osu.Game.csproj	2021-08-20 07:47:35.000000000 -0700
+++ osu-2021.820.0.resources/osu.Game/osu.Game.csproj	2021-08-26 23:41:47.926683725 -0700
@@ -1,46 +1,46 @@
-<Project Sdk="Microsoft.NET.Sdk">
-  <PropertyGroup Label="Project">
-    <TargetFramework>netstandard2.1</TargetFramework>
-    <OutputType>Library</OutputType>
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-  </PropertyGroup>
-  <PropertyGroup Label="Nuget">
-    <Title>osu!</Title>
-    <PackageId>ppy.osu.Game</PackageId>
-    <Version>0.0.0</Version>
-    <PackageIcon>icon.png</PackageIcon>
-    <IsPackable>true</IsPackable>
-  </PropertyGroup>
-  <ItemGroup Label="Icon">
-    <None Include="..\assets\lazer-nuget.png">
-      <Pack>True</Pack>
-      <PackagePath>icon.png</PackagePath>
-    </None>
-  </ItemGroup>
-  <ItemGroup Label="Package References">
-    <PackageReference Include="AutoMapper" Version="10.1.1" />
-    <PackageReference Include="DiffPlex" Version="1.7.0" />
-    <PackageReference Include="HtmlAgilityPack" Version="1.11.34" />
-    <PackageReference Include="Humanizer" Version="2.11.10" />
-    <PackageReference Include="MessagePack" Version="2.3.75" />
-    <PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="5.0.8" />
-    <PackageReference Include="Microsoft.AspNetCore.SignalR.Protocols.MessagePack" Version="5.0.8" />
-    <PackageReference Include="Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson" Version="5.0.8" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="2.2.6" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite.Core" Version="2.2.6" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="5.0.0" />
-    <PackageReference Include="Microsoft.NETCore.Targets" Version="3.1.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
-    <PackageReference Include="ppy.LocalisationAnalyser" Version="2021.725.0">
-      <PrivateAssets>all</PrivateAssets>
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-    </PackageReference>
-    <PackageReference Include="Realm" Version="10.3.0" />
-    <PackageReference Include="ppy.osu.Framework" Version="2021.819.0" />
-    <PackageReference Include="ppy.osu.Game.Resources" Version="2021.820.0" />
-    <PackageReference Include="Sentry" Version="3.8.3" />
-    <PackageReference Include="SharpCompress" Version="0.28.3" />
-    <PackageReference Include="NUnit" Version="3.13.2" />
-    <PackageReference Include="System.ComponentModel.Annotations" Version="5.0.0" />
-  </ItemGroup>
-</Project>
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup Label="Project">
+    <TargetFramework>netstandard2.1</TargetFramework>
+    <OutputType>Library</OutputType>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  </PropertyGroup>
+  <PropertyGroup Label="Nuget">
+    <Title>osu!</Title>
+    <PackageId>ppy.osu.Game</PackageId>
+    <Version>0.0.0</Version>
+    <PackageIcon>icon.png</PackageIcon>
+    <IsPackable>true</IsPackable>
+  </PropertyGroup>
+  <ItemGroup Label="Icon">
+    <None Include="..\assets\lazer-nuget.png">
+      <Pack>True</Pack>
+      <PackagePath>icon.png</PackagePath>
+    </None>
+  </ItemGroup>
+  <ItemGroup Label="Package References">
+    <PackageReference Include="AutoMapper" Version="10.1.1" />
+    <PackageReference Include="DiffPlex" Version="1.7.0" />
+    <PackageReference Include="HtmlAgilityPack" Version="1.11.34" />
+    <PackageReference Include="Humanizer" Version="2.11.10" />
+    <PackageReference Include="MessagePack" Version="2.3.75" />
+    <PackageReference Include="Microsoft.AspNetCore.SignalR.Client" Version="5.0.8" />
+    <PackageReference Include="Microsoft.AspNetCore.SignalR.Protocols.MessagePack" Version="5.0.8" />
+    <PackageReference Include="Microsoft.AspNetCore.SignalR.Protocols.NewtonsoftJson" Version="5.0.8" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="2.2.6" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite.Core" Version="2.2.6" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="5.0.0" />
+    <PackageReference Include="Microsoft.NETCore.Targets" Version="3.1.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
+    <PackageReference Include="ppy.LocalisationAnalyser" Version="2021.725.0">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+    <PackageReference Include="Realm" Version="10.3.0" />
+    <PackageReference Include="ppy.osu.Framework" Version="2021.819.0" />
+    <PackageReference Include="ppy.osu.Game.Resources" Version="2021.827.0" />
+    <PackageReference Include="Sentry" Version="3.8.3" />
+    <PackageReference Include="SharpCompress" Version="0.28.3" />
+    <PackageReference Include="NUnit" Version="3.13.2" />
+    <PackageReference Include="System.ComponentModel.Annotations" Version="5.0.0" />
+  </ItemGroup>
+</Project>
diff -Naur osu-2021.820.0/osu.iOS.props osu-2021.820.0.resources/osu.iOS.props
--- osu-2021.820.0/osu.iOS.props	2021-08-20 07:47:35.000000000 -0700
+++ osu-2021.820.0.resources/osu.iOS.props	2021-08-26 23:42:01.031877562 -0700
@@ -71,7 +71,7 @@
   </ItemGroup>
   <ItemGroup Label="Package References">
     <PackageReference Include="ppy.osu.Framework.iOS" Version="2021.819.0" />
-    <PackageReference Include="ppy.osu.Game.Resources" Version="2021.820.0" />
+    <PackageReference Include="ppy.osu.Game.Resources" Version="2021.827.0" />
   </ItemGroup>
   <!-- See https://github.com/dotnet/runtime/issues/35988 (can be removed after Xamarin uses net5.0 / net6.0) -->
   <PropertyGroup>
