From 6af77925dc17438dc27fa7e56c23798d3ed555c7 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Fri, 8 Mar 2024 18:47:56 +0800
Subject: [PATCH] Support llvm rc version

---
 cmake/modules/FindClang.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/modules/FindClang.cmake b/cmake/modules/FindClang.cmake
index 89f4c9face..2ccc3c8ec4 100644
--- a/cmake/modules/FindClang.cmake
+++ b/cmake/modules/FindClang.cmake
@@ -96,6 +96,8 @@ if(CLANG_FOUND)
   string(REPLACE "svn" "" CLANG_VERSION_CLEAN "${CLANG_VERSION}")
   # dito for git
   string(REPLACE "git" "" CLANG_VERSION_CLEAN "${CLANG_VERSION}")
+  # ditto for rc version
+  string(REPLACE "rc" "" CLANG_VERSION_CLEAN "${CLANG_VERSION}")
 
   find_path(CLANG_BUILTIN_DIR
             # cpuid.h because it is defined in ClangSupport constructor as valid clang builtin dir indicator
-- 
2.44.0

