From 47312552d1d49588a94397e0682b417613cc54db Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Sat, 30 Dec 2017 21:50:38 -0700
Subject: [PATCH] fix cmake file

---
 CMakeLists.txt | 97 +++++++++-------------------------------------------------
 1 file changed, 14 insertions(+), 83 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1382b76..d8d384d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,91 +9,22 @@ find_package(ALSA REQUIRED)
 find_package(Boost 1.47)
 
 # Find Airspy library.
-pkg_check_modules(PKG_AIRSPY libairspy)
-find_path(LIBAIRSPY_INCLUDE_DIR airspy.h
-          HINT ${PKG_LIBAIRSPY_INCLUDE_DIRS})
-find_library(LIBAIRSPY_LIBRARIES libairspy.a
-          HINT ${PKG_LIBAIRSPY_LIBRARIES_DIRS})
+pkg_check_modules(PKG_AIRSPY REQUIRED libairspy)
 
 # Find BladeRF library.
-pkg_check_modules(PKG_BLADERF libbladerf)
-find_path(LIBBLADERF_INCLUDE_DIR libbladeRF.h
-          HINT ${PKG_LIBBLADERF_INCLUDE_DIR})
-find_library(LIBBLADERF_LIBRARIES libbladeRF.so
-          HINT ${PKG_LIBBLADERF_LIBRARIES_DIRS})
+pkg_check_modules(PKG_BLADERF REQUIRED libbladeRF)
 
 # Find HackRF library.
-pkg_check_modules(PKG_HACKRF libhackrf)
-find_path(LIBHACKRF_INCLUDE_DIR hackrf.h
-          HINT ${PKG_LIBHACKRF_INCLUDE_DIRS})
-find_library(LIBHACKRF_LIBRARIES libhackrf.a
-          HINT ${PKG_LIBHACKRF_LIBRARIES_DIRS})
+pkg_check_modules(PKG_HACKRF REQUIRED libhackrf)
 
 # Find RTL-SDR library.
-pkg_check_modules(PKG_RTLSDR librtlsdr)
-find_path(LIBRTLSDR_INCLUDE_DIR rtl-sdr.h
-          HINT ${PKG_LIBRTLSDR_INCLUDE_DIRS})
-find_library(LIBRTLSDR_LIBRARIES librtlsdr.a
-          HINT ${PKG_LIBRTLSDR_LIBRARIES_DIRS})
+pkg_check_modules(PKG_RTLSDR REQUIRED librtlsdr)
 
 # Find libusb
-pkg_check_modules(PKG_LIBUSB libusb-1.0)
-find_path(LIBUSB_INCLUDE_DIR libusb.h
-          HINT ${PKG_LIBUSB_INCLUDE_DIRS}
-          PATH_SUFFIXES libusb-1.0)
-find_library(LIBUSB_LIBRARY usb-1.0
-             HINT ${PKG_LIBUSB_LIBRARY_DIRS})
-
-if(LIBRTLSDR_INCLUDE_DIR AND LIBRTLSDR_LIBRARIES)
-    message(STATUS "Found librtlsdr: ${LIBRTLSDR_INCLUDE_DIR}, ${LIBRTLSDR_LIBRARIES}")
-else()
-    message(WARNING "Can not find Osmocom RTL-SDR library")
-    message("Try again with environment variable PKG_CONFIG_PATH")
-    message("or with -DLIBRTLSDR_INCLUDE_DIR=/path/rtlsdr/include")
-    message("        -DLIBRTLSDR_LIBRARIES=/path/rtlsdr/lib/librtlsdr.so")
-endif()
-
-if(LIBAIRSPY_INCLUDE_DIR AND LIBAIRSPY_LIBRARIES)
-    message(STATUS "Found libairspy: ${LIBAIRSPY_INCLUDE_DIR}, ${LIBAIRSPY_LIBRARIES}")
-else()
-    message(WARNING "Can not find libairspy library")
-    message("Try again with environment variable PKG_CONFIG_PATH")
-    message("or with -DLIBAIRSPY_INCLUDE_DIR=/path/libairspy/include")
-    message("        -DLIBAIRSPY_LIBRARIES=/path/libairspy/lib/libairspy.so")
-endif()
-
-if(LIBBLADERF_INCLUDE_DIR AND LIBBLADERF_LIBRARIES)
-    message(STATUS "Found libbladeRF: ${LIBBLADERF_INCLUDE_DIR}, ${LIBBLADERF_LIBRARIES}")
-else()
-    message(WARNING "Can not find libbladeRF library")
-    message("Try again with environment variable PKG_CONFIG_PATH")
-    message("or with -DLIBBLADERF_INCLUDE_DIR=/path/libbladeRF/include")
-    message("        -DLIBBLADERF_LIBRARIES=/path/libbladeRF/lib/libbladeRF.so")
-endif()
-
-if(LIBHACKRF_INCLUDE_DIR AND LIBHACKRF_LIBRARIES)
-    message(STATUS "Found libhackrf: ${LIBHACKRF_INCLUDE_DIR}, ${LIBHACKRF_LIBRARIES}")
-else()
-    message(WARNING "Can not find libhackrf library")
-    message("Try again with environment variable PKG_CONFIG_PATH")
-    message("or with -DLIBHACKRF_INCLUDE_DIR=/path/libhackrf/include")
-    message("        -DLIBHACKRF_LIBRARIES=/path/libhackrf/lib/libbladeRF.so")
-endif()
-
-set(LIBAIRSPY_INCLUDE_DIRS ${LIBAIRSPY_INCLUDE_DIR} ${LIBUSB_INCLUDE_DIR})
-set(LIBAIRSPY_LIBRARIES    ${LIBAIRSPY_LIBRARIES} ${LIBUSB_LIBRARY})
-
-set(LIBBLADERF_INCLUDE_DIRS ${LIBBLADERF_INCLUDE_DIR} ${LIBUSB_INCLUDE_DIR})
-set(LIBBLADERF_LIBRARIES    ${LIBBLADERF_LIBRARIES} ${LIBUSB_LIBRARY})
-
-set(LIBHACKRF_INCLUDE_DIRS ${LIBHACKRF_INCLUDE_DIR} ${LIBUSB_INCLUDE_DIR})
-set(LIBHACKRF_LIBRARIES    ${LIBHACKRF_LIBRARIES} ${LIBUSB_LIBRARY})
-
-set(LIBRTLSDR_INCLUDE_DIRS ${LIBRTLSDR_INCLUDE_DIR} ${LIBUSB_INCLUDE_DIR})
-set(LIBRTLSDR_LIBRARIES    ${LIBRTLSDR_LIBRARIES} ${LIBUSB_LIBRARY} ${ZEROMQ_LIBRARIES})
+pkg_check_modules(PKG_LIBUSB REQUIRED libusb-1.0)
 
 # Compiler flags.
-set(CMAKE_CXX_FLAGS "-Wall -std=c++11 -O2 -ffast-math -ftree-vectorize ${EXTRA_FLAGS}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=c++11 -O2 -ffast-math -ftree-vectorize ${EXTRA_FLAGS}")
 
 set(sfmbase_SOURCES
     sfmbase/Filter.cpp
@@ -225,35 +156,35 @@ target_link_libraries(softfm
 )
 
 target_include_directories(sfmrtlsdr PUBLIC
-    ${LIBRTLSDR_INCLUDE_DIRS}
+    ${PKG_RTLSDR_INCLUDE_DIRS}
 )
 
 target_include_directories(sfmhackrf PUBLIC
-    ${LIBHACKRF_INCLUDE_DIRS}
+    ${PKG_HACKRF_INCLUDE_DIRS}
 )
 
 target_include_directories(sfmairspy PUBLIC
-    ${LIBAIRSPY_INCLUDE_DIRS}
+    ${PKG_AIRSPY_INCLUDE_DIRS}
 )
 
 target_include_directories(sfmbladerf PUBLIC
-    ${LIBBLADERF_INCLUDE_DIRS}
+    ${PKG_BLADERF_INCLUDE_DIRS}
 )
 
 target_link_libraries(sfmrtlsdr
-    ${LIBRTLSDR_LIBRARIES}
+    ${PKG_RTLSDR_LIBRARIES}
 )
 
 target_link_libraries(sfmhackrf
-    ${LIBHACKRF_LIBRARIES}
+    ${PKG_HACKRF_LIBRARIES}
 )
 
 target_link_libraries(sfmairspy
-    ${LIBAIRSPY_LIBRARIES}
+    ${PKG_AIRSPY_LIBRARIES}
 )
 
 target_link_libraries(sfmbladerf
-    ${LIBBLADERF_LIBRARIES}
+    ${PKG_BLADERF_LIBRARIES}
 )
 
 install(TARGETS softfm DESTINATION bin)
-- 
2.15.1

