diff -Naur a/conf.d/30-metric-aliases.conf b/conf.d/30-metric-aliases.conf
--- a/conf.d/30-metric-aliases.conf	2016-04-05 21:22:55.000000000 -0600
+++ b/conf.d/30-metric-aliases.conf	2016-05-02 19:05:02.119138347 -0600
@@ -7,8 +7,8 @@
 Alias similar/metric-compatible families from various sources:
 
 PostScript fonts:       URW fonts:              GUST fonts:        Windows fonts:
-======================  ======================  =================  ==================
-Helvetica               Nimbus Sans             TeX Gyre Heros
+======================  ======================  =================  ===================================
+Helvetica               Nimbus Sans             TeX Gyre Heros     Microsoft Sans Serif, MS Sans Serif
 Helvetica Condensed     Nimbus Sans Narrow      TeX Gyre Heros Cn
 Times                   Nimbus Roman            TeX Gyre Termes
 Courier                 Nimbus Mono             TeX Gyre Cursor
@@ -19,7 +19,7 @@
 New Century Schoolbook  Century SchoolBook URW  TeX Gyre Schola    Century Schoolbook
 
 Microsoft fonts:  Liberation fonts:       Google CrOS core fonts:  StarOffice fonts:  AMT fonts:
-================  ======================  =======================  =================  ==============
+================  ======================  =======================  =================  ===============
 Arial             Liberation Sans         Arimo                    Albany             Albany AMT
 Arial Narrow      Liberation Sans Narrow
 Times New Roman   Liberation Serif        Tinos                    Thorndale          Thorndale AMT
@@ -70,6 +70,27 @@
 	  </default>
 	</alias>
 
+        <alias binding="same">
+          <family>Microsoft Sans Serif</family>
+          <default>
+          <family>Helvetica</family>
+          </default>
+        </alias>
+
+        <alias binding="same">
+          <family>MS Sans Serif</family>
+          <default>
+          <family>Helvetica</family>
+          </default>
+        </alias>
+
+        <alias binding="same">
+          <family>Helv</family>
+          <default>
+          <family>Helvetica</family>
+          </default>
+        </alias>
+
 	<alias binding="same">
 	  <family>Nimbus Sans Narrow</family>
 	  <default>
