diff -Naur swift-4.0.2/Swiften/Network/PlatformNATTraversalWorker.cpp swift-4.0.2-new/Swiften/Network/PlatformNATTraversalWorker.cpp
--- swift-4.0.2/Swiften/Network/PlatformNATTraversalWorker.cpp	2018-04-06 03:06:46.000000000 -0700
+++ swift-4.0.2-new/Swiften/Network/PlatformNATTraversalWorker.cpp	2021-02-01 22:29:55.044648465 -0800
@@ -157,7 +157,7 @@
         miniUPnPInterface = new MiniUPnPInterface();
         miniUPnPSupported = miniUPnPInterface->isAvailable();
     }
-    SWIFT_LOG(debug) << "UPnP NAT traversal supported: " << miniUPnPSupported << std::endl;
+    SWIFT_LOG(debug) << "UPnP NAT traversal supported: " << miniUPnPSupported.value << std::endl;
     if (miniUPnPSupported) {
         return miniUPnPInterface;
     }
@@ -168,7 +168,7 @@
         natPMPInterface = new NATPMPInterface();
         natPMPSupported = natPMPInterface->isAvailable();
     }
-    SWIFT_LOG(debug) << "NAT-PMP NAT traversal supported: " << natPMPSupported << std::endl;
+    SWIFT_LOG(debug) << "NAT-PMP NAT traversal supported: " << natPMPSupported.value << std::endl;
     if (natPMPSupported) {
         return natPMPInterface;
     }
diff -Naur swift-4.0.2/Swiften/Base/Platform.h swift-4.0.2-new/Swiften/Base/Platform.h
--- swift-4.0.2/Swiften/Base/Platform.h	2018-04-06 03:06:46.000000000 -0700
+++ swift-4.0.2-new/Swiften/Base/Platform.h	2021-02-01 22:22:24.974507083 -0800
@@ -43,7 +43,7 @@
 #endif
 
 // Endianness
-#include <boost/detail/endian.hpp>
+#include <boost/endian.hpp>
 #if defined(BOOST_LITTLE_ENDIAN)
 #define SWIFTEN_LITTLE_ENDIAN
 #elif defined(BOOST_BIG_ENDIAN)
