From 34f413ee0be5b97a8fb8ce859001b4d57a8b89c5 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Tue, 10 Apr 2018 21:59:32 +0200
Subject: [PATCH] Fix build with Qt-5.11

---
 avogadro/qtplugins/gamessinput/gamessinput.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/avogadro/qtplugins/gamessinput/gamessinput.cpp b/avogadro/qtplugins/gamessinput/gamessinput.cpp
index 779d2d4f..ace0ad16 100644
--- a/avogadro/qtplugins/gamessinput/gamessinput.cpp
+++ b/avogadro/qtplugins/gamessinput/gamessinput.cpp
@@ -24,8 +24,8 @@
 
 #include <molequeue/client/jobobject.h>
 
+#include <QAction>
 #include <QtCore/QDebug>
-
 #include <QtWidgets/QMessageBox>
 
 namespace Avogadro {
