diff --git a/src/libresrc/libresrc.cpp b/src/libresrc/libresrc.cpp
index 79dc0f16c..9cde64c8b 100644
--- a/src/libresrc/libresrc.cpp
+++ b/src/libresrc/libresrc.cpp
@@ -23,8 +23,8 @@
 wxBitmap libresrc_getimage(const unsigned char *buff, size_t size, double scale, int dir) {
 	wxMemoryInputStream mem(buff, size);
 	if (dir != wxLayout_RightToLeft)
-		return wxBitmap(wxImage(mem), -1, scale);
-	return wxBitmap(wxImage(mem).Mirror(), -1, scale);
+		return wxBitmap(wxImage(mem), -1);
+	return wxBitmap(wxImage(mem).Mirror(), -1);
 }
 
 wxIcon libresrc_geticon(const unsigned char *buff, size_t size) {
