--- a/src/dialog_attachments.cpp	2023-10-13 18:33:52.649611127 -0700
+++ b/src/dialog_attachments.cpp	2023-10-13 18:34:25.793740795 -0700
@@ -161,7 +161,7 @@
 
 	// Multiple or single?
 	if (listView->GetNextSelected(i) != -1)
-		path = wxDirSelector(_("Select the path to save the files to:"), to_wx(OPT_GET("Path/Fonts Collector Destination")->GetString())).c_str();
+		path = wxDirSelector(_("Select the path to save the files to:"), to_wx(OPT_GET("Path/Fonts Collector Destination")->GetString())).ToStdString();
 	else {
 		path = SaveFileSelector(
 			_("Select the path to save the file to:"),
--- a/src/command/video.cpp	2023-10-13 18:33:01.306861209 -0700
+++ b/src/command/video.cpp	2023-10-13 18:33:09.711147652 -0700
@@ -475,7 +475,7 @@
 		// If where ever that is isn't defined, we can't save there
 		if ((basepath == "\\") || (basepath == "/")) {
 			// So save to the current user's home dir instead
-			basepath = wxGetHomeDir().c_str();
+			basepath = wxGetHomeDir().ToStdString();
 		}
 	}
 	// Actual fixed (possibly relative) path, decode it
