From d4f6ccb4a02eb586781aae35a9b21c26d384ba5c Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Thu, 15 Feb 2024 19:27:02 +0800
Subject: [PATCH] Set lto to true

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 9cf17c7f..c04010ff 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -181,7 +181,7 @@ opt-level = 1
 [profile.release]
 debug = true
 incremental = true
-lto = "thin"
+lto = true
 
 # windows-gnu should use it until rust-lang/rust#98302 is not fixed
 [profile.release-no-lto]
-- 
2.43.0

