From d185e6e24817aebf41d96e8d7691e67907ca4038 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Wed, 8 Apr 2026 07:03:13 +0000
Subject: [PATCH] cmake: disbale LTO for ia64

---
 CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a185657..6ef27dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,8 +47,10 @@ set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_STANDARD_REQUIRED   YES)
 set(CMAKE_CXX_STANDARD_REQUIRED YES)
 
-if (NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
-  set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+# NOTE: C++ LTO breaks programs on ia64.
+if (NOT CMAKE_BUILD_TYPE STREQUAL "Debug" AND
+    NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "ia64")
+    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
 endif()
 
 set(COMMON_SRC
-- 
2.52.0

