From 2ebde00827934975539d06039b4f523bfaa20ffc Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Tue, 14 Sep 2021 05:41:27 +0100
Subject: [PATCH 6/6] Unsquashfs: Add makefile entry for unsquash-12.o

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 39bbe87..95473d8 100755
--- a/Makefile
+++ b/Makefile
@@ -393,7 +393,7 @@ unsquash-34.o: unsquashfs.h unsquash-34.c unsquashfs_error.h
 
 unsquash-1234.o: unsquash-1234.c unsquashfs_error.h
 
-unsquash-1234.o: unsquash-12.c
+unsquash-12.o: unsquash-12.c unsquashfs.h
 
 unsquashfs_xattr.o: unsquashfs_xattr.c unsquashfs.h squashfs_fs.h xattr.h unsquashfs_error.h
 
-- 
2.33.0.windows.2

