--- efivar-38/src/include/defaults.mk	2022-01-14 11:29:35.000000000 -0800
+++ efivar-38.nomarch/src/include/defaults.mk	2022-12-14 22:24:34.252284498 -0800
@@ -72,11 +72,6 @@
 		   $(call family,SOFLAGS)
 
 HOST_ARCH=$(shell uname -m)
-ifneq ($(HOST_ARCH),ia64)
-	HOST_MARCH=-march=native
-else
-	HOST_MARCH=
-endif
 HOST_CPPFLAGS ?= $(CPPFLAGS)
 override _HOST_CPPFLAGS := $(HOST_CPPFLAGS)
 override HOST_CPPFLAGS = $(_HOST_CPPFLAGS) \
