diff -Naur ZeroBraneStudio-1.80/zbstudio/zbstudio.in ZeroBraneStudio-1.80.nohardcode/zbstudio/zbstudio.in
--- ZeroBraneStudio-1.80/zbstudio/zbstudio.in	2018-10-07 21:12:12.000000000 -0500
+++ ZeroBraneStudio-1.80.nohardcode/zbstudio/zbstudio.in	2018-10-25 20:38:44.502078153 -0500
@@ -3,4 +3,4 @@
 if [[ "$(uname -m)" == "x86_64" ]]; then ARCH="x64"; else ARCH="x86"; fi
 CWD="$PWD" # save the current directory, as it's going to change
 
-(cd "@IDE_DATADIR@"; bin/linux/$ARCH/lua src/main.lua zbstudio -cwd "$CWD" "$@") &
+(cd "@IDE_DATADIR@"; @LUA_EXECUTABLE@ src/main.lua zbstudio -cwd "$CWD" "$@") &
