From 31da05ce9cee026bda54d3227252d9eb986ee32a Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 27 Dec 2024 14:08:50 +0800
Subject: [PATCH 3/5] ARCHLINUX: install sensord

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index a5c038a3..2f04d898 100644
--- a/Makefile
+++ b/Makefile
@@ -86,7 +86,7 @@ MANDIR := $(PREFIX)/man
 ARCH := $(firstword $(subst -, ,$(shell $(CC) -dumpmachine)))
 
 # Extra non-default programs to build; e.g., sensord
-#PROG_EXTRA := sensord
+PROG_EXTRA := sensord
 
 # Build and install static library
 BUILD_STATIC_LIB := 1
-- 
2.53.0

