From 14af87f8e4e1537ab4c95a4247f4c0ce01a5f818 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 27 Dec 2024 14:24:08 +0800
Subject: [PATCH 4/5] FEDORA: sensors-detect: Add support for AMD CPU Family
 19h

---
 prog/detect/sensors-detect | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/prog/detect/sensors-detect b/prog/detect/sensors-detect
index c936dae7..20ad2885 100755
--- a/prog/detect/sensors-detect
+++ b/prog/detect/sensors-detect
@@ -2803,6 +2803,10 @@ use vars qw(@cpu_ids);
 		name => "AMD Family 17h thermal sensors",
 		driver => "k10temp",
 		detect => sub { amd_pci_detect('1463', '15d0', '1493', '1443') },
+	}, {
+		name => "AMD Family 19h thermal sensors",
+		driver => "k10temp",
+		detect => sub { amd_pci_detect('1653') },
 	}, {
 		name => "AMD Family 15h power sensors",
 		driver => "fam15h_power",
-- 
2.53.0

