From 9a0da43955ef1c843eded86f13e6ebc36713d9f3 Mon Sep 17 00:00:00 2001
From: Simon <h45h74x@gmail.com>
Date: Fri, 3 Jul 2020 00:15:44 +0200
Subject: [PATCH] Fixed typo that caused an error

---
 src/compare.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/compare.py b/src/compare.py
index 658f94f..5966b8d 100644
--- a/src/compare.py
+++ b/src/compare.py
@@ -291,5 +291,5 @@ while True:
 		# are captured and even after a delay it does not
 		# always work. Setting exposure at every frame is
 		# reliable though.
-		video_capture.intenal.set(cv2.CAP_PROP_AUTO_EXPOSURE, 1.0)  # 1 = Manual
-		video_capture.intenal.set(cv2.CAP_PROP_EXPOSURE, float(exposure))
+		video_capture.internal.set(cv2.CAP_PROP_AUTO_EXPOSURE, 1.0)  # 1 = Manual
+		video_capture.internal.set(cv2.CAP_PROP_EXPOSURE, float(exposure))
-- 
2.27.0

