From 76fe97a9bed7bd050fa2c395b66e776b20981aa1 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Wed, 16 Apr 2025 23:13:31 +0800
Subject: [PATCH 2/3] cmake: fix architecture detection on non-x86 and non-arm
 systems

---
 CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 765ed5f6..901a7545 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,8 +25,7 @@ set(GGML_CUDA_PEER_MAX_BATCH_SIZE 128)
 set(GGML_CUDA_GRAPHS ON)
 set(GGML_CUDA_FA ON)
 
-if((CMAKE_OSX_ARCHITECTURES AND NOT CMAKE_OSX_ARCHITECTURES MATCHES "arm64")
-    OR (NOT CMAKE_OSX_ARCHITECTURES AND NOT CMAKE_SYSTEM_PROCESSOR MATCHES "arm|aarch64|ARM64|ARMv[0-9]+"))
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|amd64|x86")
     set(GGML_CPU_ALL_VARIANTS ON)
 endif()
 
-- 
2.49.0

