From 36a76f9220a40618f6cce9ab620d7a33c58dfe48 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Fri, 10 Apr 2026 13:00:26 +0800
Subject: [PATCH] AOSCOS: Change the manual file's command name to
 moonlight-embedded

---
 docs/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/docs/CMakeLists.txt b/docs/CMakeLists.txt
index 3e328a3..b83d693 100644
--- a/docs/CMakeLists.txt
+++ b/docs/CMakeLists.txt
@@ -1,4 +1,4 @@
-add_custom_command(OUTPUT moonlight.1 COMMAND pod2man --section=1 --center="Moonlight Embedded Manual" --name="MOONLIGHT" --release="moonlight 2.5.0" ${CMAKE_CURRENT_SOURCE_DIR}/README.pod > moonlight.1)
-add_custom_target(docs ALL DEPENDS moonlight.1)
+add_custom_command(OUTPUT moonlight-embedded.1 COMMAND pod2man --section=1 --center="Moonlight Embedded Manual" --name="MOONLIGHT" --release="moonlight 2.5.0" ${CMAKE_CURRENT_SOURCE_DIR}/README.pod > moonlight-embedded.1)
+add_custom_target(docs ALL DEPENDS moonlight-embedded.1)
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/moonlight.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/moonlight-embedded.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
-- 
2.53.0

