diff -Naur stackless-2.7.16-slp/Stackless/platf/switch_ppc_unix.h stackless-2.7.16-slp-patch/Stackless/platf/switch_ppc_unix.h
--- stackless-2.7.16-slp/Stackless/platf/switch_ppc_unix.h	2019-03-09 05:47:11.000000000 -0600
+++ stackless-2.7.16-slp-patch/Stackless/platf/switch_ppc_unix.h	2020-11-10 18:15:32.274462422 -0600
@@ -34,9 +34,15 @@
 
 #define STACK_MAGIC 3
 
+#ifdef __powerpc64__
 #define REGS_TO_SAVE "r13", "r14", "r15", "r16", "r17", "r18", "r19", "r20", \
-       "r21", "r22", "r23", "r24", "r25", "r26", "r27", "r28", "r29", "r31", \
+       "r21", "r22", "r23", "r24", "r25", "r26", "r27", "r28", "r29", \
        "cr2", "cr3", "cr4"
+#else
+#define REGS_TO_SAVE "r13", "r14", "r15", "r16", "r17", "r18", "r19", "r20", \
+       "r21", "r22", "r23", "r24", "r25", "r26", "r27", "r28", "r29", "r31" \
+       "cr2", "cr3", "cr4"
+#endif
 /*
  * You may want to make the function static enable optimizations.
  * However, the ABI SPEC does not apply to static functions. Therefore
