diff -up wrk/genisoimage/rock.c.wrk wrk/genisoimage/rock.c
--- wrk/genisoimage/rock.c.wrk	2017-01-04 14:25:54.879670532 +0100
+++ wrk/genisoimage/rock.c	2017-01-04 15:11:38.061228649 +0100
@@ -39,6 +39,11 @@
 #include <device.h>
 #include <schily.h>
 
+/* "major" and "minor" macros will be moved to sysmacros.h
+ * silence the warning
+ * added by cdrkit-1.1.11-sysmacros.patch */
+#include <sys/sysmacros.h>
+
 #define	SU_VERSION 1
 
 #define	SL_ROOT    8
diff -up wrk/genisoimage/tree.c.wrk wrk/genisoimage/tree.c
--- wrk/genisoimage/tree.c.wrk	2017-01-04 14:21:47.472189363 +0100
+++ wrk/genisoimage/tree.c	2017-01-04 15:13:32.989466729 +0100
@@ -50,6 +50,11 @@
 #include <schily.h>
 #include <libgen.h>
 
+/* "major" and "minor" macros will be moved to sysmacros.h
+ * silence the warning
+ * added by cdrkit-1.1.11-sysmacros.patch */
+#include <sys/sysmacros.h>
+
 extern int allow_limited_size;
 
 #ifdef VMS
diff -up wrk/icedax/interface.c.wrk wrk/icedax/interface.c
--- wrk/icedax/interface.c.wrk	2017-01-04 14:46:54.298203958 +0100
+++ wrk/icedax/interface.c	2017-01-04 15:14:06.333535804 +0100
@@ -55,6 +55,11 @@
 #include <sys/ioctl.h>
 #include <statdefs.h>
 
+/* "major" and "minor" macros will be moved to sysmacros.h
+ * silence the warning
+ * added by cdrkit-1.1.11-sysmacros.patch */
+#include <sys/sysmacros.h>
+
 
 #include "mycdrom.h"
 #include "lowlevel.h"
diff -up wrk/icedax/ioctl.c.wrk wrk/icedax/ioctl.c
--- wrk/icedax/ioctl.c.wrk	2017-01-04 14:49:45.834583930 +0100
+++ wrk/icedax/ioctl.c	2017-01-04 15:15:10.268668248 +0100
@@ -36,6 +36,11 @@
 #include <schily.h>
 #include <device.h>
 
+/* "major" and "minor" macros will be moved to sysmacros.h
+ * silence the warning
+ * added by cdrkit-1.1.11-sysmacros.patch */
+#include <sys/sysmacros.h>
+
 #include <usal/scsitransp.h>
 
 #include "mycdrom.h"
