diff -rupN a/buildlib/buildcorelib.pro.old b/buildlib/buildcorelib.pro
--- a/buildlib/buildcorelib.pro.old	1969-12-31 19:00:00.000000000 -0500
+++ b/buildlib/buildcorelib.pro	2010-07-21 22:09:10.000000000 -0400
@@ -0,0 +1,13 @@
+TEMPLATE=lib
+CONFIG += qt dll qtsinglecoreapplication-buildlib
+mac:CONFIG += absolute_library_soname
+win32|mac:!wince*:!win32-msvc:!macx-xcode:CONFIG += debug_and_release build_all
+include(../src/qtsinglecoreapplication.pri)
+TARGET = $$QTSINGLECOREAPPLICATION_LIBNAME
+DESTDIR = $$QTSINGLEAPPLICATION_LIBDIR
+win32 {
+    DLLDESTDIR = $$[QT_INSTALL_BINS]
+    QMAKE_DISTCLEAN += $$[QT_INSTALL_BINS]\\$${QTSINGLECOREAPPLICATION_LIBNAME}.dll
+}
+target.path = $$DESTDIR
+INSTALLS += target
diff -rupN a/common.pri.old b/common.pri
--- a/common.pri.old	2015-09-13 18:29:19.955577604 +0200
+++ b/common.pri	2015-09-13 18:32:21.054587906 +0200
@@ -5,10 +5,12 @@ greaterThan(QT_MAJOR_VERSION, 5)|\
   if(equals(QT_MAJOR_VERSION, 5):greaterThan(QT_MINOR_VERSION, 4))|\
   if(equals(QT_MAJOR_VERSION, 5):equals(QT_MINOR_VERSION, 4):greaterThan(QT_PATCH_VERSION, 1)) {
     QTSINGLEAPPLICATION_LIBNAME = $$qt5LibraryTarget(QtSolutions_SingleApplication-head)
+    QTSINGLECOREAPPLICATION_LIBNAME = $$qt5LibraryTarget(QtSolutions_SingleCoreApplication-head)
 } else {
     QTSINGLEAPPLICATION_LIBNAME = $$qtLibraryTarget(QtSolutions_SingleApplication-head)
+    QTSINGLECOREAPPLICATION_LIBNAME = $$qtLibraryTarget(QtSolutions_SingleCoreApplication-head)
 }
 TEMPLATE -= fakelib
 
 QTSINGLEAPPLICATION_LIBDIR = $$PWD/lib
-unix:qtsingleapplication-uselib:!qtsingleapplication-buildlib:QMAKE_RPATHDIR += $$QTSINGLEAPPLICATION_LIBDIR
+QTSINGLECOREAPPLICATION_LIBDIR = $$PWD/lib
diff -rupN a/qtsingleapplication.pro.old b/qtsingleapplication.pro
--- a/qtsingleapplication.pro.old	2009-12-16 05:43:33.000000000 -0500
+++ b/qtsingleapplication.pro	2010-07-21 22:09:10.000000000 -0400
@@ -1,5 +1,4 @@
 TEMPLATE=subdirs
 CONFIG += ordered
 include(common.pri)
-qtsingleapplication-uselib:SUBDIRS=buildlib
-SUBDIRS+=examples
+qtsingleapplication-uselib:SUBDIRS=buildlib/buildlib.pro buildlib/buildcorelib.pro
diff -rupN a/src/QtSingleCoreApplication.old b/src/QtSingleCoreApplication
--- a/src/QtSingleCoreApplication.old	1969-12-31 19:00:00.000000000 -0500
+++ b/src/QtSingleCoreApplication	2010-07-21 22:09:10.000000000 -0400
@@ -0,0 +1 @@
+#include "qtsinglecoreapplication.h"
--- a/src/qtsinglecoreapplication.pri.old	2014-02-12 22:46:28.000000000 +0100
+++ b/src/qtsinglecoreapplication.pri	2014-08-15 15:28:16.604289958 +0200
@@ -1,10 +1,18 @@
-INCLUDEPATH	+= $$PWD
-DEPENDPATH      += $$PWD
-HEADERS		+= $$PWD/qtsinglecoreapplication.h $$PWD/qtlocalpeer.h
-SOURCES		+= $$PWD/qtsinglecoreapplication.cpp $$PWD/qtlocalpeer.cpp
+include(../common.pri)
+INCLUDEPATH += $$PWD
+DEPENDPATH += $$PWD
 
 QT *= network
+QT -= gui
 
-win32:contains(TEMPLATE, lib):contains(CONFIG, shared) {
-    DEFINES += QT_QTSINGLECOREAPPLICATION_EXPORT=__declspec(dllexport)
+qtsinglecoreapplication-uselib:!qtsinglecoreapplication-buildlib {
+    LIBS += -L$$QTSINGLECOREAPPLICATION_LIBDIR -l$$QTSINGLECOREAPPLICATION_LIBNAME
+} else {
+    SOURCES += $$PWD/qtsinglecoreapplication.cpp $$PWD/qtlocalpeer.cpp
+    HEADERS += $$PWD/qtsinglecoreapplication.h $$PWD/qtlocalpeer.h
+}
+
+win32 {
+    contains(TEMPLATE, lib):contains(CONFIG, shared):DEFINES += QT_QTSINGLECOREAPPLICATION_EXPORT
+    else:qtsinglecoreapplication-uselib:DEFINES += QT_QTSINGLECOREAPPLICATION_IMPORT
 }
