diff --git a/common.pri b/common.pri
index 8ca6ff1..32c3f79 100644
--- a/common.pri
+++ b/common.pri
@@ -12,5 +12,10 @@ greaterThan(QT_MAJOR_VERSION, 5)|\
 }
 TEMPLATE -= fakelib
 
-QTSINGLEAPPLICATION_LIBDIR = $$PWD/lib
-QTSINGLECOREAPPLICATION_LIBDIR = $$PWD/lib
+PREFIX = $$PREFIX
+isEmpty(PREFIX) {
+    PREFIX = /usr
+}
+
+QTSINGLEAPPLICATION_LIBDIR = $$PREFIX/lib
+QTSINGLECOREAPPLICATION_LIBDIR = $$PREFIX/lib
-- 
2.30.2

