From 1bc4b2ad138be75ebf58e8b7825ff2052a16ca06 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 1 Apr 2019 15:52:56 +0800
Subject: [PATCH] disable version check

---
 winetricks-zh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/winetricks-zh b/winetricks-zh
index 2135ef0..8844488 100755
--- a/winetricks-zh
+++ b/winetricks-zh
@@ -8,6 +8,9 @@
 # (This doesn't change often, use the sha256sum of the file when reporting problems)
 WINETRICKS_VERSION=20181203-next
 
+# Disable version check
+WINETRICKS_LATEST_VERSION_CHECK='disabled'
+
 # This is a UTF-8 file
 # You should see an o with two dots over it here [ö]
 # You should see a micro (u with a tail) here [µ]
-- 
2.21.0

