diff --git a/pppd/plugins/pppoe/Makefile.linux b/pppd/plugins/pppoe/Makefile.linux
index 2df887b..6cb8397 100644
--- a/pppd/plugins/pppoe/Makefile.linux
+++ b/pppd/plugins/pppoe/Makefile.linux
@@ -43,12 +43,12 @@ pppoe.so: plugin.o discovery.o if.o common.o
 
 install: all
 	$(INSTALL) -d -m 755 $(LIBDIR)
-	$(INSTALL) -c -m 4550 pppoe.so $(LIBDIR)
+	$(INSTALL) -c -m 755 pppoe.so $(LIBDIR)
 	# Symlink for backward compatibility
 	$(LN_S) pppoe.so $(LIBDIR)/rp-pppoe.so
 	$(INSTALL) -d -m 755 $(BINDIR)
-	$(INSTALL) -c -m 555 pppoe-discovery $(BINDIR)
-	$(INSTALL) -c -m 444 pppoe-discovery.8 $(MANDIR)
+	$(INSTALL) -c -m 755 pppoe-discovery $(BINDIR)
+	$(INSTALL) -c -m 644 pppoe-discovery.8 $(MANDIR)
 
 clean:
 	rm -f *.o *.so pppoe-discovery
