From: Didier Raboud <odyx@debian.org>
Date: Tue, 4 Oct 2016 11:43:48 +0200
Subject: Link against system liblcms1

---
 icc2ps/Makefile | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/icc2ps/Makefile b/icc2ps/Makefile
index 0aa49ba..87f57b5 100644
--- a/icc2ps/Makefile
+++ b/icc2ps/Makefile
@@ -6,9 +6,7 @@ BIN=	$(PREFIX)/bin
 MANDIR=$(PREFIX)/share/man
 
 SRC=	icc2ps.c xgetopt.c
-LIB=	cmscam97.c cmscnvrt.c cmserr.c cmsgamma.c cmsgmt.c cmsintrp.c cmsio1.c \
-	cmslut.c cmsmatsh.c cmsmtrx.c cmsnamed.c cmspack.c cmspcs.c cmsps2.c \
-	cmssamp.c cmswtpnt.c cmsxform.c cmsio0.c cmsvirt.c
+LIB=	-llcms
 CFLAGS+=	-O3
 
 # _XOPEN_SOURCE is required for swab() but undefines strcasecmp(), so add _GNU_SOURCE as well:
@@ -55,7 +53,7 @@ install-man: man
 clean:
 	rm -f *.o foo2zjs-icc2ps *.1
 
-foo2zjs-icc2ps: icc34.h lcms.h
+foo2zjs-icc2ps: icc34.h
 
 regress: all
 	@./foo2zjs-icc2ps a b 2>&1 | grep "little cms P"
