From: Didier Raboud <odyx@debian.org>
Date: Tue, 4 Oct 2016 11:43:51 +0200
Subject: Do not display the icon by default

See #579154
---
 hplj1020.desktop | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hplj1020.desktop b/hplj1020.desktop
index 49f9705..76adb31 100644
--- a/hplj1020.desktop
+++ b/hplj1020.desktop
@@ -9,3 +9,4 @@ Icon=/usr/share/pixmaps/hplj1020_icon.png
 StartupNotify=true
 Categories=System;
 X-Desktop-File-Install-Version=0.10
+NoDisplay=true
