From: Didier Raboud <odyx@debian.org>
Date: Mon, 25 May 2020 22:14:49 +0200
Subject: Specify FOOMATICDB when calling foomatic-ppfile

---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 55d7ced..79e1984 100644
--- a/Makefile
+++ b/Makefile
@@ -1490,6 +1490,7 @@ ppd:
 	    esac; \
 	    echo $$driver - $$printer; \
 	    if true; then \
+		FOOMATICDB=foomatic-db \
 		foomatic-ppdfile -d $$driver -p $$printer > PPD/$$printer.ppd; \
 	    else \
 		# 09/06/18: Use the older foomatic??? \
