From 449841003b045e9a135ba94dffa77dfc5bd61c0d Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Mon, 14 Jul 2025 15:40:51 +0800
Subject: [PATCH] Split google fonts initialization list

---
 src/core/textrenderer/qgsfontmanager.cpp | 34 +++++++++++++++++-------
 1 file changed, 24 insertions(+), 10 deletions(-)

diff --git a/src/core/textrenderer/qgsfontmanager.cpp b/src/core/textrenderer/qgsfontmanager.cpp
index 9d59560dec..159854428c 100644
--- a/src/core/textrenderer/qgsfontmanager.cpp
+++ b/src/core/textrenderer/qgsfontmanager.cpp
@@ -245,9 +245,9 @@ QgsFontDownloadDetails GoogleFontDetails( const QString &family, const QStringLi
 
 QgsFontDownloadDetails QgsFontManager::detailsForFontDownload( const QString &family, QString &matchedFamily ) const
 {
-  // this list is built using scripts/process_google_fonts.py
-  const thread_local std::vector< QgsFontDownloadDetails > sGoogleFonts
-  {
+  // NOTE: Init list is too large for Loongson3 and RISC-V, It will cause gcc crash or assembler
+  // messages: 'error: branch out of range'. So split initialization function.
+  const thread_local std::vector<QgsFontDownloadDetails> sGoogleFonts0{
     GoogleFontDetails( QStringLiteral( "ABeeZee" ), { QStringLiteral( "ofl/abeezee/ABeeZee-Regular.ttf" ), QStringLiteral( "ofl/abeezee/ABeeZee-Italic.ttf" ) }, QStringLiteral( "ofl/abeezee/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "ADLaM Display" ), { QStringLiteral( "ofl/adlamdisplay/ADLaMDisplay-Regular.ttf" ) }, QStringLiteral( "ofl/adlamdisplay/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Abel" ), { QStringLiteral( "ofl/abel/Abel-Regular.ttf" ) }, QStringLiteral( "ofl/abel/OFL.txt" ) ),
@@ -647,6 +647,10 @@ QgsFontDownloadDetails QgsFontManager::detailsForFontDownload( const QString &fa
     GoogleFontDetails( QStringLiteral( "Erica One" ), { QStringLiteral( "ofl/ericaone/EricaOne-Regular.ttf" ) }, QStringLiteral( "ofl/ericaone/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Esteban" ), { QStringLiteral( "ofl/esteban/Esteban-Regular.ttf" ) }, QStringLiteral( "ofl/esteban/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Estonia" ), { QStringLiteral( "ofl/estonia/Estonia-Regular.ttf" ) }, QStringLiteral( "ofl/estonia/OFL.txt" ) ),
+  };
+  //============================================================================
+
+  const thread_local std::vector<QgsFontDownloadDetails> sGoogleFonts1{
     GoogleFontDetails( QStringLiteral( "Euphoria Script" ), { QStringLiteral( "ofl/euphoriascript/EuphoriaScript-Regular.ttf" ) }, QStringLiteral( "ofl/euphoriascript/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Ewert" ), { QStringLiteral( "ofl/ewert/Ewert-Regular.ttf" ) }, QStringLiteral( "ofl/ewert/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Exo" ), { QStringLiteral( "ofl/exo/Exo%5Bwght%5D.ttf" ), QStringLiteral( "ofl/exo/Exo-Italic%5Bwght%5D.ttf" ) }, QStringLiteral( "ofl/exo/OFL.txt" ) ),
@@ -1047,6 +1051,10 @@ QgsFontDownloadDetails QgsFontManager::detailsForFontDownload( const QString &fa
     GoogleFontDetails( QStringLiteral( "Mochiy Pop P One" ), { QStringLiteral( "ofl/mochiypoppone/MochiyPopPOne-Regular.ttf" ) }, QStringLiteral( "ofl/mochiypoppone/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Modak" ), { QStringLiteral( "ofl/modak/Modak-Regular.ttf" ) }, QStringLiteral( "ofl/modak/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Modern Antiqua" ), { QStringLiteral( "ofl/modernantiqua/ModernAntiqua-Regular.ttf" ) }, QStringLiteral( "ofl/modernantiqua/OFL.txt" ) ),
+  };
+  //================================================================================
+
+  const thread_local std::vector<QgsFontDownloadDetails> sGoogleFonts2{
     GoogleFontDetails( QStringLiteral( "Mohave" ), { QStringLiteral( "ofl/mohave/Mohave%5Bwght%5D.ttf" ), QStringLiteral( "ofl/mohave/Mohave-Italic%5Bwght%5D.ttf" ) }, QStringLiteral( "ofl/mohave/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Moirai One" ), { QStringLiteral( "ofl/moiraione/MoiraiOne-Regular.ttf" ) }, QStringLiteral( "ofl/moiraione/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Molengo" ), { QStringLiteral( "ofl/molengo/Molengo-Regular.ttf" ) }, QStringLiteral( "ofl/molengo/OFL.txt" ) ),
@@ -1367,6 +1375,10 @@ QgsFontDownloadDetails QgsFontManager::detailsForFontDownload( const QString &fa
     GoogleFontDetails( QStringLiteral( "Palanquin" ), { QStringLiteral( "ofl/palanquin/Palanquin-Thin.ttf" ), QStringLiteral( "ofl/palanquin/Palanquin-ExtraLight.ttf" ), QStringLiteral( "ofl/palanquin/Palanquin-Light.ttf" ), QStringLiteral( "ofl/palanquin/Palanquin-Regular.ttf" ), QStringLiteral( "ofl/palanquin/Palanquin-Medium.ttf" ), QStringLiteral( "ofl/palanquin/Palanquin-SemiBold.ttf" ), QStringLiteral( "ofl/palanquin/Palanquin-Bold.ttf" ) }, QStringLiteral( "ofl/palanquin/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Palanquin Dark" ), { QStringLiteral( "ofl/palanquindark/PalanquinDark-Regular.ttf" ), QStringLiteral( "ofl/palanquindark/PalanquinDark-Medium.ttf" ), QStringLiteral( "ofl/palanquindark/PalanquinDark-SemiBold.ttf" ), QStringLiteral( "ofl/palanquindark/PalanquinDark-Bold.ttf" ) }, QStringLiteral( "ofl/palanquindark/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Palette Mosaic" ), { QStringLiteral( "ofl/palettemosaic/PaletteMosaic-Regular.ttf" ) }, QStringLiteral( "ofl/palettemosaic/OFL.txt" ) ),
+  };
+  //===========================================================================================
+
+  const thread_local std::vector<QgsFontDownloadDetails> sGoogleFonts3{
     GoogleFontDetails( QStringLiteral( "Pangolin" ), { QStringLiteral( "ofl/pangolin/Pangolin-Regular.ttf" ) }, QStringLiteral( "ofl/pangolin/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Paprika" ), { QStringLiteral( "ofl/paprika/Paprika-Regular.ttf" ) }, QStringLiteral( "ofl/paprika/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Parisienne" ), { QStringLiteral( "ofl/parisienne/Parisienne-Regular.ttf" ) }, QStringLiteral( "ofl/parisienne/OFL.txt" ) ),
@@ -1797,19 +1809,21 @@ QgsFontDownloadDetails QgsFontManager::detailsForFontDownload( const QString &fa
     GoogleFontDetails( QStringLiteral( "Zilla Slab" ), { QStringLiteral( "ofl/zillaslab/ZillaSlab-Light.ttf" ), QStringLiteral( "ofl/zillaslab/ZillaSlab-LightItalic.ttf" ), QStringLiteral( "ofl/zillaslab/ZillaSlab-Regular.ttf" ), QStringLiteral( "ofl/zillaslab/ZillaSlab-Italic.ttf" ), QStringLiteral( "ofl/zillaslab/ZillaSlab-Medium.ttf" ), QStringLiteral( "ofl/zillaslab/ZillaSlab-MediumItalic.ttf" ), QStringLiteral( "ofl/zillaslab/ZillaSlab-SemiBold.ttf" ), QStringLiteral( "ofl/zillaslab/ZillaSlab-SemiBoldItalic.ttf" ), QStringLiteral( "ofl/zillaslab/ZillaSlab-Bold.ttf" ), QStringLiteral( "ofl/zillaslab/ZillaSlab-BoldItalic.ttf" ) }, QStringLiteral( "ofl/zillaslab/OFL.txt" ) ),
     GoogleFontDetails( QStringLiteral( "Zilla Slab Highlight" ), { QStringLiteral( "ofl/zillaslabhighlight/ZillaSlabHighlight-Regular.ttf" ), QStringLiteral( "ofl/zillaslabhighlight/ZillaSlabHighlight-Bold.ttf" ) }, QStringLiteral( "ofl/zillaslabhighlight/OFL.txt" ) ),
   };
+  const std::vector<QgsFontDownloadDetails>* sGoogleFonts[4]{
+    &sGoogleFonts0, &sGoogleFonts1, &sGoogleFonts2, &sGoogleFonts3
+  };
 
   matchedFamily.clear();
   const QString cleanedFamily = QgsFontDownloadDetails::standardizeFamily( family );
 
-  for ( const QgsFontDownloadDetails &candidate : sGoogleFonts )
-  {
-    if ( candidate.standardizedFamily() == cleanedFamily )
-    {
-      matchedFamily = candidate.family();
-      return candidate;
+  for(int ii = 0; ii < 4; ii++){
+    for ( const QgsFontDownloadDetails &candidate : *sGoogleFonts[ii] ){
+      if ( candidate.standardizedFamily() == cleanedFamily ){
+        matchedFamily = candidate.family();
+        return candidate;
+      }
     }
   }
-
   return QgsFontDownloadDetails();
 }
 
-- 
2.50.1

