From 7d12314e86b7a38cfc45293734874a27301ea8c0 Mon Sep 17 00:00:00 2001
From: zhaixiaojuan <zhaixiaojuan@loongson.cn>
Date: Mon, 7 Jul 2025 07:40:19 +0000
Subject: [PATCH] Add loongarch64 support

---
 include/backward_ros/backward.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/backward_ros/backward.hpp b/include/backward_ros/backward.hpp
index 93037d5..3dab933 100644
--- a/include/backward_ros/backward.hpp
+++ b/include/backward_ros/backward.hpp
@@ -4220,6 +4220,8 @@ class SignalHandling {
     error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.__gregs[REG_PC]);
 #elif defined(__s390x__)
     error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.psw.addr);
+#elif defined(__loongarch64)
+    error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.__pc);
 #elif defined(__APPLE__) && defined(__x86_64__)
     error_addr = reinterpret_cast<void *>(uctx->uc_mcontext->__ss.__rip);
 #elif defined(__APPLE__)

