From 4e224b9f089f57011627184986dace1e9032ce22 Mon Sep 17 00:00:00 2001
From: "Lain \"Fearyncess\" Yang" <i@lain.vg>
Date: Wed, 3 Jun 2026 22:05:21 +0800
Subject: [PATCH 2/2] allow use boost 1.91.0

---
 cmake/dependencies/Boost_Sunshine.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/dependencies/Boost_Sunshine.cmake b/cmake/dependencies/Boost_Sunshine.cmake
index f0ff9a78..63b1f809 100644
--- a/cmake/dependencies/Boost_Sunshine.cmake
+++ b/cmake/dependencies/Boost_Sunshine.cmake
@@ -3,7 +3,7 @@
 #
 include_guard(GLOBAL)
 
-set(BOOST_VERSION "1.89.0")
+set(BOOST_VERSION "1.91.0")
 set(BOOST_COMPONENTS
         filesystem
         log
-- 
2.52.0

