From a4f977c44c38eb7b32d941f4cbddcc9feaf3867d Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Mon, 3 Mar 2025 22:24:02 +0800
Subject: [PATCH 2/3] Fix libSvtAv1Enc.a path

---
 cmake/dependencies/common.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/dependencies/common.cmake b/cmake/dependencies/common.cmake
index e38bf10a..27ddf9d3 100644
--- a/cmake/dependencies/common.cmake
+++ b/cmake/dependencies/common.cmake
@@ -51,7 +51,7 @@ if(NOT DEFINED FFMPEG_PREPARED_BINARIES)
             "${FFMPEG_PREPARED_BINARIES}/lib/libavcodec.a"
             "${FFMPEG_PREPARED_BINARIES}/lib/libavutil.a"
             "${FFMPEG_PREPARED_BINARIES}/lib/libcbs.a"
-            "${FFMPEG_PREPARED_BINARIES}/lib/libSvtAv1Enc.a"
+            "${FFMPEG_PREPARED_BINARIES}/lib64/libSvtAv1Enc.a"
             "${FFMPEG_PREPARED_BINARIES}/lib/libswscale.a"
             "${FFMPEG_PREPARED_BINARIES}/lib/libx264.a"
             "${FFMPEG_PREPARED_BINARIES}/lib/libx265.a"
-- 
2.48.1

