From 3d2ecea12364e4892d4646e788871a4dc6eec85c Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Tue, 4 Mar 2025 15:22:51 +0800
Subject: [PATCH 3/3] Specify SUNSHINE_ASSETS_DIR path manually

---
 cmake/compile_definitions/unix.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/compile_definitions/unix.cmake b/cmake/compile_definitions/unix.cmake
index 68660cd1..38422651 100644
--- a/cmake/compile_definitions/unix.cmake
+++ b/cmake/compile_definitions/unix.cmake
@@ -6,5 +6,5 @@ list(APPEND SUNSHINE_EXTERNAL_LIBRARIES
 
 # add install prefix to assets path if not already there
 if(NOT SUNSHINE_ASSETS_DIR MATCHES "^${CMAKE_INSTALL_PREFIX}")
-    set(SUNSHINE_ASSETS_DIR "${CMAKE_INSTALL_PREFIX}/${SUNSHINE_ASSETS_DIR}")
+    set(SUNSHINE_ASSETS_DIR "/usr/share/sunshine")
 endif()
-- 
2.48.1

