From a352dbcd88b2b83e02e2a0a055b2eb93c816fcc6 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Thu, 28 Oct 2021 23:43:51 -0500
Subject: [PATCH] cmake: link against embree avx512

---
 build_files/cmake/Modules/FindEmbree.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build_files/cmake/Modules/FindEmbree.cmake b/build_files/cmake/Modules/FindEmbree.cmake
index c7f9bc5b220..10f3bbf37cd 100644
--- a/build_files/cmake/Modules/FindEmbree.cmake
+++ b/build_files/cmake/Modules/FindEmbree.cmake
@@ -58,6 +58,7 @@ IF(EMBREE_STATIC_LIB)
       embree_sse42
       embree_avx
       embree_avx2
+    embree_avx512
     )
   ENDIF()
 
-- 
2.39.1

