--- a/intern/cycles/kernel/CMakeLists.txt	2023-10-14 20:20:57.272663506 -0700
+++ b/intern/cycles/kernel/CMakeLists.txt	2023-10-14 20:21:32.901902258 -0700
@@ -467,6 +467,8 @@
       -m ${CUDA_BITS}
       -I ${CMAKE_CURRENT_SOURCE_DIR}/..
       -I ${CMAKE_CURRENT_SOURCE_DIR}/device/cuda
+      -D__GNUC_MINOR__=10 -D__GNUC__=12
+      -allow-unsupported-compiler
       --use_fast_math
       -o ${CMAKE_CURRENT_BINARY_DIR}/${cuda_file}
       -Wno-deprecated-gpu-targets)
