diff --git a/Source/CursesDialog/cmCursesMainForm.cxx b/Source/CursesDialog/cmCursesMainForm.cxx
index dbd024da6..4a9dc473c 100644
--- a/Source/CursesDialog/cmCursesMainForm.cxx
+++ b/Source/CursesDialog/cmCursesMainForm.cxx
@@ -475,7 +475,7 @@ void cmCursesMainForm::UpdateStatusBar(const char* message)
       strncpy(bar + curFieldLen + 2, help, width - curFieldLen - 2);
       if (curFieldLen + helpLen + 2 < width) {
         memset(bar + curFieldLen + helpLen + 2, ' ',
-               width - curFieldLen + helpLen + 2);
+               width - (curFieldLen + helpLen + 2));
       }
     }
   }
