--- a/modules/berkeley-db/configure.orig	2017-01-28 11:49:37.845965965 -0700
+++ b/modules/berkeley-db/configure	2017-01-28 11:54:31.794501872 -0700
@@ -5214,7 +5214,7 @@ if ${ac_cv_dbe_set_errcall_accept_dbe+:}
 else
 
 CFLAGS_save="$CFLAGS"
-CFLAGS="$CFLAGS -Werror"
+CFLAGS="$CFLAGS -Wno-uninitialized -Werror"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <db.h>
--- a/modules/berkeley-db/configure.in.orig	2017-01-28 11:49:37.847965949 -0700
+++ b/modules/berkeley-db/configure.in	2017-01-28 11:54:31.792501889 -0700
@@ -48,7 +48,7 @@ dnl <http://www.sleepycat.com/docs/ref/u
 AC_CACHE_CHECK([whether DB_ENV->set_errcall() accepts DBE],
 ac_cv_dbe_set_errcall_accept_dbe,[
 CFLAGS_save="$CFLAGS"
-CFLAGS="$CFLAGS -Werror"
+CFLAGS="$CFLAGS -Wno-uninitialized -Werror"
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([#include <db.h>
 void my_callback (const DB_ENV* dbe, const char *errpfx, const char *msg) {}],
 [[DB_ENV dbe; dbe.set_errcall(&dbe,&my_callback);]])],
