--- a/modules/libsvm/libsvm.lisp.orig	2016-11-11 08:56:50.341694018 -0700
+++ b/modules/libsvm/libsvm.lisp	2016-11-11 09:05:19.341553889 -0700
@@ -31,7 +31,7 @@
 (c-lines "
 #if !defined(HAVE_SVM_DESTROY_MODEL)
 # if defined(HAVE_SVM_FREE_AND_DESTROY_MODEL)
-void svm_destroy_model (svm_model *model){ svm_free_and_destroy_model(&model); }
+void svm_destroy_model (struct svm_model *model){ svm_free_and_destroy_model(&model); }
 # else
 #  error No svm_destroy_model and no svm_free_and_destroy_model!
 # endif
