--- a/modules/bindings/glibc/linux.lisp.orig	2016-11-11 08:56:50.548677286 -0700
+++ b/modules/bindings/glibc/linux.lisp	2016-11-11 09:08:51.751465279 -0700
@@ -294,6 +294,8 @@
 ;; for robust mutexes
 (def-c-const EOWNERDEAD (:documentation "Owner died")) ; 130
 (def-c-const ENOTRECOVERABLE (:documentation "State not recoverable")) ; 131
+(def-c-const ERFKILL (:documentation "Operation not possible due to RF-kill")) ; 132
+(def-c-const EHWPOISON (:documentation "Memory page has hardware error")) ; 133
 
 ; -------------------------- <bits/errno.h> -----------------------------------
 
