From bda5d919b9a5e3e2555ef486614b1177bc92d5c9 Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Wed, 8 Nov 2023 23:12:53 -0800
Subject: [PATCH] Make VulkanLayerSettings a shared library

---
 src/layer/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/layer/CMakeLists.txt b/src/layer/CMakeLists.txt
index 5afff80..4cdc9f2 100644
--- a/src/layer/CMakeLists.txt
+++ b/src/layer/CMakeLists.txt
@@ -5,7 +5,7 @@
 # SPDX-License-Identifier: Apache-2.0
 set(CMAKE_FOLDER "${CMAKE_FOLDER}/VulkanLayerSettings")
 
-add_library(VulkanLayerSettings STATIC)
+add_library(VulkanLayerSettings SHARED)
 add_library(Vulkan::LayerSettings ALIAS VulkanLayerSettings)
 
 target_sources(VulkanLayerSettings PRIVATE
-- 
2.39.1

