diff -Naur a/etc/Vendor.conf b/etc/Vendor.conf
--- a/etc/Vendor.conf	2015-12-11 00:54:37.000000000 +0800
+++ b/etc/Vendor.conf	2016-07-07 21:41:41.341094303 +0800
@@ -12,7 +12,7 @@
 # If the value is set to 'none' then no link is shown.
 #
 # default=http://www.packagekit.org/pk-package-not-found.html
-DefaultUrl=http://www.packagekit.org/pk-package-not-found.html
+DefaultUrl=https://github.com/AOSC-Dev/aosc-os-abbs/blob/staging/CONTRIBUTING.md#hey-i-need-a-new-package
 
 # The URL which is shown to the user when a codec could not be found.
 # It should explain why certain codecs cannot be used, and perhaps show
@@ -21,7 +21,7 @@
 # If the value is set to 'none' then the value of DefaultUrl is used.
 #
 # default=none
-CodecUrl=none
+CodecUrl=https://github.com/AOSC-Dev/aosc-os-abbs/blob/staging/CONTRIBUTING.md#hey-i-need-a-new-package
 
 # The URL which is shown to the user when hardware drivers could not be found.
 # It should explain why some hardware is not supported, and links to futher
@@ -30,7 +30,7 @@
 # If the value is set to 'none' then the value of DefaultUrl is used.
 #
 # default=none
-HardwareUrl=none
+HardwareUrl=https://github.com/AOSC-Dev/aosc-os-abbs/blob/staging/CONTRIBUTING.md#hey-i-need-a-new-package
 
 # The URL which is shown to the user when fonts could not be found.
 # Alternatives should probably be suggested where possible.
@@ -38,7 +38,7 @@
 # If the value is set to 'none' then the value of DefaultUrl is used.
 #
 # default=none
-FontUrl=none
+FontUrl=https://github.com/AOSC-Dev/aosc-os-abbs/blob/staging/CONTRIBUTING.md#hey-i-need-a-new-package
 
 # The URL which is shown to the user when programs handing a mime tpye could not
 # be found. It should probably explain how to use wine if the program is a
@@ -47,4 +47,4 @@
 # If the value is set to 'none' then the value of DefaultUrl is used.
 #
 # default=none
-MimeUrl=none
+MimeUrl=https://github.com/AOSC-Dev/aosc-os-abbs/blob/staging/CONTRIBUTING.md#hey-i-need-a-new-package
