From c2fca81152d54249e47278c8755994f485904ed8 Mon Sep 17 00:00:00 2001
From: Linjie Fu <linjie.fu@intel.com>
Date: Wed, 11 Dec 2019 16:48:03 +0800
Subject: [PATCH 08/26] lavc/mips: simplify the switch code

Signed-off-by: Linjie Fu <linjie.fu@intel.com>
Signed-off-by: Michael Niedermayer <michael@niedermayer.cc>
---
 libavcodec/mips/h264pred_init_mips.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/libavcodec/mips/h264pred_init_mips.c b/libavcodec/mips/h264pred_init_mips.c
index 63637b8732..e537ad8bd4 100644
--- a/libavcodec/mips/h264pred_init_mips.c
+++ b/libavcodec/mips/h264pred_init_mips.c
@@ -73,10 +73,7 @@ static av_cold void h264_pred_init_msa(H264PredContext *h, int codec_id,
 
         switch (codec_id) {
         case AV_CODEC_ID_SVQ3:
-            ;
-            break;
         case AV_CODEC_ID_RV40:
-            ;
             break;
         case AV_CODEC_ID_VP7:
         case AV_CODEC_ID_VP8:
-- 
2.30.1

