From 9c18b08622a6ef56cbc69ff59a9e78ba8eebfcd5 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Tue, 20 Feb 2024 06:57:42 -0800
Subject: [PATCH 08/11] Add mips64r6 generic options

---
 configure | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure b/configure
index 2602bf584a..8218bbfa23 100755
--- a/configure
+++ b/configure
@@ -5096,6 +5096,10 @@ elif enabled mips; then
                 enable mips64r6
                 check_cflags "-mtune=p6600 -mabi=64" && check_cflags "-msched-weight -mload-store-pairs -funroll-loops" && check_ldflags "-mabi=64"
             ;;
+            mips64r6)
+                enable mips64r6
+                check_cflags "-mtune=mips64r6 -mabi=64" && check_cflags "-msched-weight -mload-store-pairs -funroll-loops" && check_ldflags "-mabi=64"
+            ;;
             # Cores from Loongson
             loongson2e|loongson2f|loongson3*)
                 enable local_aligned
-- 
2.45.0

