From 21da5df668bd3ca605f66e5886556e3518899781 Mon Sep 17 00:00:00 2001
From: James Almer <jamrial@gmail.com>
Date: Fri, 2 Jun 2023 14:07:20 -0300
Subject: [PATCH 16/17] avcodec/av1dec: reset the fragment on extradata reading
 failure

Signed-off-by: James Almer <jamrial@gmail.com>
---
 libavcodec/av1dec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/av1dec.c b/libavcodec/av1dec.c
index a3301f454f..1bc37d732a 100644
--- a/libavcodec/av1dec.c
+++ b/libavcodec/av1dec.c
@@ -760,7 +760,7 @@ static av_cold int av1_decode_init(AVCodecContext *avctx)
                                                avctx);
         if (ret < 0) {
             av_log(avctx, AV_LOG_WARNING, "Failed to read extradata.\n");
-            return ret;
+            goto end;
         }
 
         seq = ((CodedBitstreamAV1Context *)(s->cbc->priv_data))->sequence_header;
-- 
2.46.0

