From 87a85a0ffa1a0cfd6ee9ab1e6b310e083f6b234f Mon Sep 17 00:00:00 2001
From: MutsukiC <mutsuki@aosc.io>
Date: Sat, 9 May 2026 13:45:59 +0800
Subject: [PATCH] Skip broken build requirements check for python wrapper

---
 wrappers/python/pyproject.toml | 2 --
 1 file changed, 2 deletions(-)

diff --git a/wrappers/python/pyproject.toml b/wrappers/python/pyproject.toml
index 7c3aa927..866f6d10 100644
--- a/wrappers/python/pyproject.toml
+++ b/wrappers/python/pyproject.toml
@@ -3,8 +3,6 @@ requires = [
     "setuptools>=77",
 #    "setuptools-scm>=8",
     "wheel",
-    "cmake>=3.18",
-    "pybind11[global]",
 ]
 build-backend = "setuptools.build_meta"
 
-- 
2.52.0

