From 8f843c65acc771b172920fe16c5245d5b7cf340d Mon Sep 17 00:00:00 2001
From: MutsukiC <mutsuki@aosc.io>
Date: Thu, 23 Apr 2026 09:28:14 +0800
Subject: [PATCH] Skip broken build requirements check for python wrapper

---
 wrappers/python/pyproject.toml | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/wrappers/python/pyproject.toml b/wrappers/python/pyproject.toml
index 16484fe3..f6c16894 100644
--- a/wrappers/python/pyproject.toml
+++ b/wrappers/python/pyproject.toml
@@ -1,9 +1,6 @@
 [build-system]
 requires = [
     "setuptools>=42",
-    "setuptools_scm",
     "wheel",
-    "cmake>=3.15",
-    "pybind11[global]",
 ]
 build-backend = "setuptools.build_meta"
-- 
2.52.0

