From 56d01b6f654a7138486d5b29053898dc06947910 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 6 Mar 2024 15:36:55 +0800
Subject: [PATCH] feat(gimprc): enable color management

---
 etc/gimprc.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/etc/gimprc.in b/etc/gimprc.in
index 426cb0575f..9cf71ab655 100644
--- a/etc/gimprc.in
+++ b/etc/gimprc.in
@@ -317,9 +317,9 @@
 
 # Defines the color management behavior.  This is a parameter list.
 # 
-# (color-management
+(color-management
 #     (mode display)
-#     (display-profile-from-gdk no)
+      (display-profile-from-gdk no)
 #     (display-rendering-intent relative-colorimetric)
 #     (display-use-black-point-compensation yes)
 #     (display-optimize yes)
@@ -330,7 +330,7 @@
 #     (out-of-gamut-color (color-rgb 1 0 1))
 #     (show-rgb-u8 no)
 #     (show-hsv no)
-#     (display-module "CdisplayLcms"))
+      (display-module "CdisplayLcms"))
 
 # Keep a permanent record of all opened and saved files in the Recent
 # Documents list.  Possible values are yes and no.
-- 
2.44.0

