diff -up gimp-2.10.14/etc/gimprc.in.cm-system-monitor-profile-by-default gimp-2.10.14/etc/gimprc.in
--- gimp-2.10.14/etc/gimprc.in.cm-system-monitor-profile-by-default	2019-10-26 21:46:48.000000000 +0200
+++ gimp-2.10.14/etc/gimprc.in	2019-11-04 13:17:23.322750636 +0100
@@ -317,9 +317,9 @@
 
 # Defines the color management behavior.  This is a parameter list.
 # 
-# (color-management
+(color-management
 #     (mode display)
-#     (display-profile-from-gdk no)
+      (display-profile-from-gdk no)
 #     (display-rendering-intent relative-colorimetric)
 #     (display-use-black-point-compensation yes)
 #     (display-optimize yes)
@@ -328,7 +328,8 @@
 #     (simulation-optimize yes)
 #     (simulation-gamut-check no)
 #     (out-of-gamut-color (color-rgb 1 0 1))
-#     (display-module "CdisplayLcms"))
+#     (display-module "CdisplayLcms")
+)
 
 # Keep a permanent record of all opened and saved files in the Recent
 # Documents list.  Possible values are yes and no.
