Description: Fix FTBFS with GCC 15
Author: Adrian Bunk <bunk@debian.org>
Bug-Debian: https://bugs.debian.org/1097594

--- pktstat-1.8.5.orig/resize.c
+++ pktstat-1.8.5/resize.c
@@ -35,17 +35,12 @@
 
 #include "compat.h"
 
-#ifdef SIGWINCH
-static RETSIGTYPE sigwinch();
-#endif
-
 static volatile int sigwinch_seen;
 
 #ifdef SIGWINCH
 /* Set the flag when the window size changes */
 static RETSIGTYPE
-sigwinch(sig)
-	int sig;
+sigwinch(int sig)
 {
 	sigwinch_seen = 1;
 }
