--- a/mono/mini/mini-arm.c	2016-08-24 15:25:42.253619745 +0200
+++ b/mono/mini/mini-arm.c	2016-08-24 15:26:00.124173662 +0200
@@ -34,7 +34,7 @@
 #if (defined(HAVE_KW_THREAD) && defined(__linux__) && defined(__ARM_EABI__)) \
 	|| defined(TARGET_ANDROID) \
 	|| (defined(TARGET_IOS) && !defined(TARGET_WATCHOS))
-#define HAVE_FAST_TLS
+//#define HAVE_FAST_TLS
 #endif
 
 /* Sanity check: This makes no sense */
