diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2014-05-14 02:53:02.000000000 +0800
+++ b/configure.ac	2016-07-30 10:24:21.886455383 +0800
@@ -95,7 +95,7 @@
 	DO_CHECK=$enableval, DO_CHECK=yes)
 
 if test	"$DO_CHECK" = "yes"; then
-	accepted_versions="1.2 1.4 2.0"
+	accepted_versions="1.2 1.4 2.0 2.1"
 	AC_PATH_PROGS(GNUPG, [gpg gpg2], no)
 	AC_DEFINE_UNQUOTED(GNUPG, "$GNUPG", [Path to gpg executable.])
 	ok="no"
