commit 183d7619c29482a5d4ae8019692899f87f198171
commit b05e301d1b264a5d8f07cb96e5edc243d99bff79

diff -up libcryptui-3.12.2/configure.ac.gpg2 libcryptui-3.12.2/configure.ac
--- libcryptui-3.12.2/configure.ac.gpg2	2018-01-29 13:04:53.324022912 -0600
+++ libcryptui-3.12.2/configure.ac	2018-01-29 13:05:14.293824226 -0600
@@ -95,7 +95,7 @@ AC_ARG_ENABLE(gpg-check,
 	DO_CHECK=$enableval, DO_CHECK=yes)
 
 if test	"$DO_CHECK" = "yes"; then
-	accepted_versions="1.2 1.4 2.0"
+	accepted_versions="1.2 1.4 2.0 2.1 2.2"
 	AC_PATH_PROGS(GNUPG, [gpg gpg2], no)
 	AC_DEFINE_UNQUOTED(GNUPG, "$GNUPG", [Path to gpg executable.])
 	ok="no"
diff -up libcryptui-3.12.2/configure.gpg2 libcryptui-3.12.2/configure
--- libcryptui-3.12.2/configure.gpg2	2018-01-29 13:04:57.487983459 -0600
+++ libcryptui-3.12.2/configure	2018-01-29 13:05:27.721696999 -0600
@@ -17379,7 +17379,7 @@ fi
 
 
 if test	"$DO_CHECK" = "yes"; then
-	accepted_versions="1.2 1.4 2.0"
+	accepted_versions="1.2 1.4 2.0 2.1 2.2"
 	for ac_prog in gpg gpg2
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
