From a1c9ffb174d9588aab6f9f68909eb31c7867d076 Mon Sep 17 00:00:00 2001
From: "kernel-team@fedoraproject.org" <kernel-team@fedoraproject.org>
Date: Thu, 29 Jul 2010 16:46:31 -0700
Subject: [PATCH 02/22] input: kill stupid messages

Bugzilla: N/A
Upstream-status: Fedora mustard
---
 drivers/input/keyboard/atkbd.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/input/keyboard/atkbd.c b/drivers/input/keyboard/atkbd.c
index c229bd6b3f7f..07a014fc3d49 100644
--- a/drivers/input/keyboard/atkbd.c
+++ b/drivers/input/keyboard/atkbd.c
@@ -486,11 +486,15 @@ static void atkbd_receive_byte(struct ps2dev *ps2dev, u8 data)
 		return;
 	case ATKBD_RET_ACK:
 	case ATKBD_RET_NAK:
+# if 0
+		/* Quite a few key switchers and other tools trigger this
+		 * and it confuses people who can do nothing about it */
 		if (printk_ratelimit())
 			dev_warn(&serio->dev,
 				 "Spurious %s on %s. "
 				 "Some program might be trying to access hardware directly.\n",
 				 data == ATKBD_RET_ACK ? "ACK" : "NAK", serio->phys);
+#endif
 		return;
 	case ATKBD_RET_ERR:
 		atkbd->err_count++;
-- 
2.45.1

