diff -Naur cups-2.3.3/install-sh cups-2.3.3.mod/install-sh
--- cups-2.3.3/install-sh	2020-04-27 13:04:29.000000000 -0500
+++ cups-2.3.3.mod/install-sh	2021-08-01 14:13:53.459906353 -0500
@@ -219,7 +219,6 @@
 	# Update permissions and strip as needed, then move to the final name.
 	# If the chmod, strip, rm, or mv commands fail, remove the installed
 	# file...
-	if [ x"$stripcmd" != x ]; then $doit $stripcmd $stripopt "$dsttmp" || echo "warning: Unable to strip $dst!"; fi
 	if [ x"$chowncmd" != x ]; then $doit $chowncmd "$dsttmp" || echo "warning: Unable to change owner of $dst!"; fi
 	if [ x"$chgrpcmd" != x ]; then $doit $chgrpcmd "$dsttmp" || echo "warning: Unable to change group of $dst!"; fi
 
