diff -up cups-2.2.0/backend/org.cups.usb-quirks.libusb-quirks cups-2.2.0/backend/org.cups.usb-quirks
--- cups-2.2.0/backend/org.cups.usb-quirks.libusb-quirks	2016-09-14 01:39:47.000000000 +0200
+++ cups-2.2.0/backend/org.cups.usb-quirks	2016-11-07 18:46:32.913920903 +0100
@@ -76,6 +76,9 @@
 # Canon, Inc. MP510 Printer (https://bugs.launchpad.net/bugs/1050009)
 0x04a9 0x1717 unidir
 
+# Canon, Inc. MP540 Printer, https://bugzilla.redhat.com/967873
+0x04a9 0x1730 unidir
+
 # Canon, Inc. MP550 Printer (Issue #4155)
 0x04a9 0x173d unidir
 
@@ -127,6 +130,9 @@
 # All Samsung devices (https://bugs.launchpad.net/bugs/1032456)
 0x04e8 soft-reset
 
+# Samsung ML-2160 Series (https://bugzilla.redhat.com/show_bug.cgi?id=873123)
+0x04e8 0x330f unidir
+
 # All Zebra devices (https://bugs.launchpad.net/bugs/1001028)
 0x0a5f unidir
 
