diff --git a/scheduler/log.c b/scheduler/log.c
index 74790dc35..b6226201f 100644
--- a/scheduler/log.c
+++ b/scheduler/log.c
@@ -1046,7 +1047,7 @@ cupsdLogRequest(cupsd_client_t *con,	/* I - Request to log */
   * Filter requests as needed...
   */
 
-  if (AccessLogLevel == CUPSD_ACCESSLOG_NONE)
+  if (AccessLogLevel == CUPSD_ACCESSLOG_NONE || !AccessLog)
     return (1);
   else if (AccessLogLevel < CUPSD_ACCESSLOG_ALL)
   {
