diff -up cups-1.6.3/cups/usersys.c.use-ipp1.1 cups-1.6.3/cups/usersys.c
--- cups-1.6.3/cups/usersys.c.use-ipp1.1	2013-07-12 11:41:45.368837618 +0200
+++ cups-1.6.3/cups/usersys.c	2013-07-12 11:41:45.391837299 +0200
@@ -366,7 +366,7 @@ cupsSetServer(const char *server)	/* I -
         cg->server_version = 22;
     }
     else
-      cg->server_version = 20;
+      cg->server_version = 11;
 
     if (cg->server[0] != '/' && (port = strrchr(cg->server, ':')) != NULL &&
         !strchr(port, ']') && isdigit(port[1] & 255))
