diff -up cups-1.6.2/ppdc/sample.drv.dymo-deviceid cups-1.6.2/ppdc/sample.drv
--- cups-1.6.2/ppdc/sample.drv.dymo-deviceid	2013-06-18 16:57:02.110662953 +0100
+++ cups-1.6.2/ppdc/sample.drv	2013-06-18 16:58:56.513989117 +0100
@@ -125,6 +125,7 @@ Version "1.5"
 {
   Manufacturer "Dymo"
   ModelName "Label Printer"
+  Attribute "1284DeviceID" "" "MFG:DYMO;MDL:LabelWriter 400;"
   Attribute NickName "" "Dymo Label Printer"
   PCFileName "dymo.ppd"
   DriverType label
