diff -up cups-2.0rc1/cups/ppd.c.strict-ppd-line-length cups-2.0rc1/cups/ppd.c
--- cups-2.0rc1/cups/ppd.c.strict-ppd-line-length	2014-02-06 19:33:34.000000000 +0100
+++ cups-2.0rc1/cups/ppd.c	2014-09-12 18:07:44.227773710 +0200
@@ -2872,7 +2872,7 @@ ppd_read(cups_file_t    *fp,		/* I - Fil
 	*lineptr++ = (char)ch;
 	col ++;
 
-	if (col > (PPD_MAX_LINE - 1))
+	if (col > (PPD_MAX_LINE - 1) && pg->ppd_conform == PPD_CONFORM_STRICT)
 	{
 	 /*
           * Line is too long...
@@ -2933,7 +2933,7 @@ ppd_read(cups_file_t    *fp,		/* I - Fil
 	{
 	  col ++;
 
-	  if (col > (PPD_MAX_LINE - 1))
+	  if (col > (PPD_MAX_LINE - 1) && pg->ppd_conform == PPD_CONFORM_STRICT)
 	  {
 	   /*
             * Line is too long...
@@ -2992,7 +2992,7 @@ ppd_read(cups_file_t    *fp,		/* I - Fil
 	{
 	  col ++;
 
-	  if (col > (PPD_MAX_LINE - 1))
+	  if (col > (PPD_MAX_LINE - 1) && pg->ppd_conform == PPD_CONFORM_STRICT)
 	  {
 	   /*
             * Line is too long...
