Author: Nelson A. de Oliveira <naoliv@gmail.com>
Description: Cosmetic patch to manpage.
Bug-Debian: https://bugs.debian.org/319574
Reviewed-By: Amaya Rodrigo Sastre <amaya@debian.org>
--- wdm.orig/doc/wdm.man.in
+++ wdm/doc/wdm.man.in
@@ -7,7 +7,7 @@
 \fBwdm\fP is an X display manager based on the original X11 
 X Display Manager (xdm).
 It features numerous functionality improvements, including the ability
- to reboot or halt the machine (once
+to reboot or halt the machine (once
 the proper password has been supplied) and to select from a list of available
 X sessions to start. The look of the login panel is
 somewhat configurable, too.
@@ -42,11 +42,11 @@
 resource. This is factility to use with all messages if log is
 redirected to syslog.
 .IP "\fB\-usesyslog\fP"
-Specifies ``true'' as the value for the \fBDisplayManager.useSyslog\dP
+Specifies ``true'' as the value for the \fBDisplayManager.useSyslog\fP
 resource. This will force all messages go to syslog with facility
 specified by \fBDisplayManager.syslogFacility\fP resource.
 .IP "\fB\-useerrfile\fP"
-Specifies ``false'' as the value for the \fBDisplayManager.useSyslog\dP
+Specifies ``false'' as the value for the \fBDisplayManager.useSyslog\fP
 resource. This will force all messages go to log file
 specified by \fBDisplayManager.errorLogFile\fP resource.
 .IP "\fB\-resources\fP \fIresource_file\fP"
