Description: Miscellaneous spelling fixes reported by lintian
Author: Axel Beckert <abe@debian.org> 

Index: wdm/doc/wdm.man.in
===================================================================
--- wdm.orig/doc/wdm.man.in	2014-07-24 15:59:58.000000000 +0200
+++ wdm/doc/wdm.man.in	2014-07-24 16:01:31.000000000 +0200
@@ -639,14 +639,14 @@
 
 .fi
 .SH BACKGROUND IMAGE SPECIFICATION
-There are several possible ways of specifing a background image. The
+There are several possible ways of specifying a background image. The
 generic format is \fItype\fP:\fIimage\fP. \fItype\fP can be any of:
 .TP
 .B none
 The backgound is \fBnot\fP set.
 .TP
 .B solid
-it renders a solid backgroud, and \fIimage\fP is a color name
+it renders a solid background, and \fIimage\fP is a color name
 .TP
 \fBhgradient\fP, \fBvgradient\fP, \fBdgradient\fP
 a gradient (either horizontal, vertical or diagonal) will be
Index: wdm/doc/wdmLogin.man
===================================================================
--- wdm.orig/doc/wdmLogin.man	2014-07-24 15:22:12.000000000 +0200
+++ wdm/doc/wdmLogin.man	2014-07-24 16:02:54.000000000 +0200
@@ -2,7 +2,7 @@
 .SH NAME
 wdmLogin \- wdm login panel
 .SH SYNOPSIS
-\fBwdmLogin\fP [-d \fIdisplay\fP] [-l \fIlogo\fP] [-w \fI<window managers>\fP] [-b \fIbackground\fP] [-h \fIhelpfile\fP] [-u \fIusername\fP] [-p \fIpassword\fP]
+\fBwdmLogin\fP [\-d \fIdisplay\fP] [\-l \fIlogo\fP] [\-w \fI<window managers>\fP] [\-b \fIbackground\fP] [\-h \fIhelpfile\fP] [\-u \fIusername\fP] [\-p \fIpassword\fP]
 .SH DESCRIPTION
 \fBwdmLogin\fP is greeter designed to work with \fBwdm\fP(1x). It has
 several features not found on xdm's greeter. wdmLogin should not be
@@ -38,7 +38,7 @@
 uses the specified background image (see the next section).  If this is
 not specified, then the background is \fBNOT\fP set.
 .SH BACKGROUND IMAGE SPECIFICATION
-There are several possible ways of specifing a background image. The
+There are several possible ways of specifying a background image. The
 generic format is \-b\fItype\fP:\fIimage\fP. \fItype\fP can be any of:
 .TP
 .B none
@@ -46,7 +46,7 @@
 via other means.
 .TP
 .B solid
-it renders a solid backgroud, and \fIimage\fP is a color name
+it renders a solid background, and \fIimage\fP is a color name
 .TP
 \fBhgradient\fP, \fBvgradient\fP, \fBdgradient\fP
 a gradient (either horizontal, vertical or diagonal) will be
Index: wdm/src/wdm/choose.c
===================================================================
--- wdm.orig/src/wdm/choose.c	2014-07-24 15:22:12.000000000 +0200
+++ wdm/src/wdm/choose.c	2014-07-24 16:05:32.000000000 +0200
@@ -407,7 +407,7 @@
 	if (XdmcpReadARRAY8 (&buffer, &clientAddress)) {
 	    if (XdmcpReadCARD16 (&buffer, &connectionType)) {
 		if (XdmcpReadARRAY8 (&buffer, &choice)) {
-		    WDMDebug("Read from chooser succesfully\n");
+		    WDMDebug("Read from chooser successfully\n");
 		    RegisterIndirectChoice (&clientAddress, connectionType, &choice);
 		    XdmcpDisposeARRAY8 (&choice);
 		} else {
