From c8edd93d88b88d43748c7d3124a0292b6dbe500e Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Tue, 4 Mar 2025 13:49:45 +0800
Subject: [PATCH 2/2] Disable cuda and ffnvcodec on specific platforms

---
 cmake/ffmpeg/ffmpeg.cmake | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/cmake/ffmpeg/ffmpeg.cmake b/cmake/ffmpeg/ffmpeg.cmake
index e86b004..1065039 100644
--- a/cmake/ffmpeg/ffmpeg.cmake
+++ b/cmake/ffmpeg/ffmpeg.cmake
@@ -75,14 +75,9 @@ elseif(APPLE)
 elseif(UNIX)
     list(APPEND FFMPEG_EXTRA_CONFIGURE
             --enable-amf
-            --enable-cuda
-            --enable-cuda_llvm
             --enable-encoder=h264_amf,hevc_amf,av1_amf
-            --enable-encoder=h264_nvenc,hevc_nvenc,av1_nvenc
             --enable-encoder=h264_vaapi,hevc_vaapi,av1_vaapi
             --enable-encoder=h264_v4l2m2m
-            --enable-ffnvcodec
-            --enable-nvenc
             --enable-v4l2_m2m
             --enable-vaapi
     )
-- 
2.48.1

