diff --git a/.gitmodules b/.gitmodules
index a7cab832c4..c3ce3ed2d0 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +1,23 @@
 [submodule "meson"]
 	path = meson
 	url = https://gitlab.com/qemu-project/meson.git
+
+[submodule "slirp"]
+        path = slirp
+        url = https://git.qemu.org/libslirp.git
+
+[submodule "dtc"]
+        path = dtc
+        url = https://git.qemu.org/dtc.git
+
+[submodule "tests/fp/berkeley-testfloat-3"]
+        path = tests/fp/berkeley-testfloat-3
+        url = https://github.com/cota/berkeley-testfloat-3
+
+[submodule "tests/fp/berkeley-softfloat-3"]
+        path = tests/fp/berkeley-softfloat-3
+        url = https://github.com/cota/berkeley-softfloat-3
+
+[submodule "ui/keycodemapdb"]
+        path = ui/keycodemapdb
+        url = https://git.qemu.org/keycodemapdb.git
