From 9db022f85f7869e54ca71921ccfc01498720903e Mon Sep 17 00:00:00 2001
From: Yang Liu <numbksco@gmail.com>
Date: Thu, 27 Mar 2025 21:17:31 +0800
Subject: [PATCH 1006/1022] feat: optimize several vnor+vand to vandn

---
 target/i386/latx/translator/tr-simd.c | 32 ++++++++++-----------------
 1 file changed, 12 insertions(+), 20 deletions(-)

diff --git a/target/i386/latx/translator/tr-simd.c b/target/i386/latx/translator/tr-simd.c
index 9bf2a4aaec..22e36a2e92 100644
--- a/target/i386/latx/translator/tr-simd.c
+++ b/target/i386/latx/translator/tr-simd.c
@@ -807,11 +807,9 @@ bool translate_maxpd(IR1_INST *pir1)
     la_vfcmp_cond_d(mask, src, dest, 0x3);
     /* 2. when dest[i] > src[i] is true, dest[i] = dest[i] */
     la_vand_v(dest, dest, mask);
-    /* 3. mask = ~mask */
-    la_vnor_v(mask, mask, mask);
-    /* 4. when dest[i] > src[i] is false, dest[i] = src[i] */
-    la_vand_v(mask, src, mask);
-    /* 5. get final result */
+    /* 3. when dest[i] > src[i] is false, dest[i] = src[i] */
+    la_vandn_v(mask, mask, src);
+    /* 4. get final result */
     la_vor_v(dest, dest, mask);
     return true;
 }
@@ -836,11 +834,9 @@ bool translate_maxps(IR1_INST *pir1)
     la_vfcmp_cond_s(mask, src, dest, 0x3);
     /* 2. when dest[i] > src[i] is true, dest[i] = dest[i] */
     la_vand_v(dest, dest, mask);
-    /* 3. mask = ~mask */
-    la_vnor_v(mask, mask, mask);
-    /* 4. when dest[i] > src[i] is false, dest[i] = src[i] */
-    la_vand_v(mask, src, mask);
-    /* 5. get final result */
+    /* 3. when dest[i] > src[i] is false, dest[i] = src[i] */
+    la_vandn_v(mask, mask, src);
+    /* 4. get final result */
     la_vor_v(dest, dest, mask);
     return true;
 }
@@ -923,11 +919,9 @@ bool translate_minpd(IR1_INST *pir1)
     la_vfcmp_cond_d(mask, dest, src, 0x3);
     /* 2. when dest[i] > src[i] is true, dest[i] = dest[i] */
     la_vand_v(dest, dest, mask);
-    /* 3. mask = ~mask */
-    la_vnor_v(mask, mask, mask);
-    /* 4. when dest[i] > src[i] is false, dest[i] = src[i] */
-    la_vand_v(mask, src, mask);
-    /* 5. get final result */
+    /* 3. when dest[i] > src[i] is false, dest[i] = src[i] */
+    la_vandn_v(mask, mask, src);
+    /* 4. get final result */
     la_vor_v(dest, dest, mask);
     return true;
 }
@@ -952,11 +946,9 @@ bool translate_minps(IR1_INST *pir1)
     la_vfcmp_cond_s(mask, dest, src, 0x3);
     /* 2. when dest[i] > src[i] is true, dest[i] = dest[i] */
     la_vand_v(dest, dest, mask);
-    /* 3. mask = ~mask */
-    la_vnor_v(mask, mask, mask);
-    /* 4. when dest[i] > src[i] is false, dest[i] = src[i] */
-    la_vand_v(mask, src, mask);
-    /* 5. get final result */
+    /* 3. when dest[i] > src[i] is false, dest[i] = src[i] */
+    la_vandn_v(mask, mask, src);
+    /* 4. get final result */
     la_vor_v(dest, dest, mask);
     return true;
 }
-- 
2.49.0

