From f3735d80abb17fc3bebc33d45034cf8282587c50 Mon Sep 17 00:00:00 2001
From: Hanlu Li <heuleehanlu@gmail.com>
Date: Fri, 28 Mar 2025 14:06:35 +0800
Subject: [PATCH 1011/1022] LATX, fix: Fix encode_search() for TU

Signed-off-by: Hanlu Li <heuleehanlu@gmail.com>
---
 accel/tcg/translate-all.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/accel/tcg/translate-all.c b/accel/tcg/translate-all.c
index 9c9494a1a6..3ebee966d9 100644
--- a/accel/tcg/translate-all.c
+++ b/accel/tcg/translate-all.c
@@ -323,9 +323,7 @@ static target_long decode_sleb128(const uint8_t **pp)
 
 int encode_search(TranslationBlock *tb, uint8_t *block)
 {
-#ifndef CONFIG_LATX_TU
     uint8_t *highwater = tcg_ctx->code_gen_highwater;
-#endif
     uint8_t *p = block;
     int i, j, n;
 
@@ -347,11 +345,9 @@ int encode_search(TranslationBlock *tb, uint8_t *block)
            one row beginning below the high water mark cannot overrun
            the buffer completely.  Thus we can test for overflow after
            encoding a row without having to check during encoding.  */
-#ifndef CONFIG_LATX_TU
-        if (unlikely(p > highwater)) {
+        if (!in_pre_translate && unlikely(p > highwater)) {
             return -1;
         }
-#endif
     }
 
     return p - block;
-- 
2.49.0

