From c8e8c90f90b1ce0dc8e5e0516101f9e2bb509350 Mon Sep 17 00:00:00 2001
From: Xiaotian Wu <yetist@gmail.com>
Date: Mon, 31 Mar 2025 11:24:57 +0800
Subject: [PATCH 1016/1022] improve build scripts

---
 latxbuild/build-release.sh | 231 ++++++++++++++++---------------------
 1 file changed, 98 insertions(+), 133 deletions(-)

diff --git a/latxbuild/build-release.sh b/latxbuild/build-release.sh
index e4da894677..13d3bc54a0 100755
--- a/latxbuild/build-release.sh
+++ b/latxbuild/build-release.sh
@@ -1,140 +1,105 @@
 #!/usr/bin/bash
-
-# apt-get install aptitude
-# aptitude install git ninja-build libssl-dev libc6 gcc g++ pkg-config libglib2.0-dev libdrm-dev
-
-export CFLAGS="-Wno-error=unused-but-set-variable -Wno-error=unused-function -Wformat -Werror=format-y2k"
-
-source_dir=$(realpath "$(dirname "$0")/../")
-bin64=$source_dir/build64-release/latx-x86_64
-bin32=$source_dir/build32-release/latx-i386
-os=$(lsb_release -si)
-install_prefix=lat-$os
-install_dir=$source_dir
-tag=$1
-name=""
-
-help() {
-    echo "Usage:"
-    echo "  -n              name"
-    echo "  -h              help"
+set -e
+
+pkgname=lat
+pkgver=$(cat VERSION)
+pkgdate=$(date +%Y%m%d)
+srcdir=$(realpath "$(dirname "$0")/../")
+pkgdir=$srcdir/pkg
+tarballs=$pkgname-$pkgver-$pkgdate.tar.xz
+
+parpare() {
+    [ -d $srcdir/build32 ] || mkdir -p $srcdir/build32
+    [ -d $srcdir/build64 ] || mkdir -p $srcdir/build64
+    [ -d $pkgdir ] && rm -rf $pkgdir
+    mkdir -p $pkgdir
 }
 
-parseArgs() {
-    while getopts "n:h" opt; do
-        case ${opt} in
-        n)
-            name=-"$OPTARG"
-            ;;
-        h)
-            help
-            exit
-            ;;
-        # 若选项需要参数但未收到，则走冒号分支
-        :)
-            help
-            exit
-            ;;
-        # 若遇到未指定的选项，会走问号分支
-        ?)
-            help
-            exit
-            ;;
-        esac
-    done
+build() {
+    unset CXXFLAGS
+    unset CFLAGS
+    unset LDFLAGS
+    export CFLAGS="-Wno-error=unused-but-set-variable -Wno-error=unused-function -Wformat -Werror=format-y2k"
+
+    local _configure32_flags=(
+        --target-list=i386-linux-user
+        --disable-capstone
+        --disable-docs
+        --disable-gcrypt
+        --disable-glusterfs
+        --disable-gnutls
+        --disable-gtk
+        --disable-kvm
+        --disable-libiscsi
+        --disable-libnfs
+        --disable-libssh
+        --disable-linux-io-uring
+        --disable-nettle
+        --disable-opengl
+        --disable-qom-cast-debug
+        --disable-sdl
+        --disable-tools
+        --disable-tpm
+        --disable-vde
+        --disable-vhost-crypto
+        --disable-vhost-kernel
+        --disable-vhost-net
+        --disable-vhost-user
+        --disable-vnc
+        --enable-latx
+        --enable-guest-base-zero
+        --disable-debug-info
+        --optimize-O1
+        --extra-ldflags=-ldl
+        --disable-blobs
+        --disable-docs
+        --disable-werror
+        --disable-pie
+        --static
+        --disable-linux-io-uring
+    )
+
+    local _configure64_flags=(
+        --target-list=x86_64-linux-user
+        --disable-blobs
+        --disable-debug-info
+        --disable-docs
+        --disable-linux-io-uring
+        --disable-werror
+        --enable-kzt
+        --enable-latx
+        --extra-ldflags=-ldl
+        --optimize-O1
+    )
+
+    pushd $srcdir/build32 >/dev/null
+    ../configure "${_configure32_flags[@]}"
+    ninja -j$(nproc)
+    popd >/dev/null
+
+    pushd $srcdir/build64 >/dev/null
+    ../configure "${_configure64_flags[@]}"
+    ninja -j$(nproc)
+    popd >/dev/null
 }
 
-make_cmd() {
-    install_prefix=$install_prefix$name
-    install_dir=$install_dir/$install_prefix
-    cd $source_dir
-    rm -rf $install_prefix
-    rm -f $install_prefix.tar
-
-    # build latx-x86_64
-    rm -rf build64-release
-    mkdir -p build64-release
-    cd build64-release
-
-    ../configure --target-list=x86_64-linux-user --enable-latx \
-                --disable-debug-info --optimize-O1 --extra-ldflags=-ldl --enable-kzt \
-                --prefix=$install_dir --disable-blobs --disable-docs \
-                --disable-werror --disable-linux-io-uring
-
-    if [ ! -f "/usr/bin/ninja" ]; then
-        make -j $(nproc)
-    else
-        ninja
-        ninja install
-    fi
-
-    if [ ! -e $bin64 ]; then
-        echo "latx-x86_64 ERROR"
-        exit 1
-    fi
-
-    message64="latx-x86_64:\n"
-    message64="$message64 \t before strip: $(ls -lh $bin64| awk '{print $5}')"
-    strip latx-x86_64
-    message64="$message64 \t after strip:  $(ls -lh $bin64| awk '{print $5}')\n"
-    message64="$message64 \t MD5: $(md5sum $bin64)\n"
-    message64="$message64 \t VERSION: $($bin64 --version)\n"
-
-    # build latx-i386
-    cd $source_dir
-    rm -rf build32-release
-    mkdir -p build32-release
-    cd build32-release
-
-    ../configure --target-list=i386-linux-user --enable-latx \
-                --enable-guest-base-zero --disable-debug-info --optimize-O1 --static \
-                --extra-ldflags=-ldl --prefix=$install_dir --disable-blobs --disable-docs  \
-                --disable-werror --disable-pie --disable-linux-io-uring
-
-    if [ ! -f "/usr/bin/ninja" ]; then
-        make -j $(nproc)
-    else
-        ninja
-        ninja install
-    fi
-
-    if [ ! -e $bin32 ]; then
-        echo "latx-i386 ERROR"
-        exit 1
-    fi
-
-    message32="latx-i386:\n"
-    message32="$message32 \t before strip: $(ls -lh $bin32| awk '{print $5}')"
-    strip latx-i386
-    message32="$message32 \t after strip:  $(ls -lh $bin32| awk '{print $5}')\n"
-    message32="$message32 \t MD5: $(md5sum $bin32)\n"
-    message32="$message32 \t VERSION: $($bin32 --version)\n"
-    cd $source_dir
-
-    if [ -n "$tag" ]; then
-        commit_message=$(git show "$tag" --no-patch --format=%B)
-        printf "%s\n" "$commit_message" >> $install_dir/README
-    fi
-
-    cpu_info=$(lscpu)
-    model_name=$(echo "$cpu_info" | grep "Model name" | awk -F ':' '{print $2}' | xargs)
-    echo -e "\n\n\n////////////////////// $os on $model_name //////////////////////\n" >> $install_dir/README
-    echo -e "uname -a" >> $install_dir/README
-    uname -a  >> $install_dir/README
-    echo -e >> $install_dir/README
-    echo -e "/etc/os-release:" >> $install_dir/README
-    cat /etc/os-release >> $install_dir/README
-    echo -e >> $install_dir/README
-    echo -e "gcc version:" >> $install_dir/README
-    gcc --version | head -n 1  >> $install_dir/README
-    echo -e >> $install_dir/README
-    echo -e $message64 >> $install_dir/README
-    echo -e >> $install_dir/README
-    echo -e $message32 >> $install_dir/README
-    cat $install_dir/README
-
-    tar -czvf $install_prefix.tar.gz $install_prefix
+package() {
+    mkdir -p $pkgdir/$pkgname-$pkgver/usr/{bin,lib/binfmt.d,lib/sysctl.d}
+    install -Dm755 -s $srcdir/build32/latx-i386 $pkgdir/$pkgname-$pkgver/usr/bin/latx-i386
+    install -Dm755 -s $srcdir/build64/latx-x86_64 $pkgdir/$pkgname-$pkgver/usr/bin/latx-x86_64
+    cat >$pkgdir/$pkgname-$pkgver/usr/lib/binfmt.d/latx-i386.conf <<EOF
+:latx-i386:M::\x7fELF\x01\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x03\x00:\xff\xff\xff\xff\xff\xfe\xfe\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff:/usr/bin/latx-i386:
+EOF
+    cat >${pkgdir}/${pkgname}-${pkgver}/usr/lib/binfmt.d/latx-x86_64.conf <<EOF
+:latx-x86_64:M::\x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x3e\x00:\xff\xff\xff\xff\xff\xfe\xfe\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff:/usr/bin/latx-x86_64:
+EOF
+    echo "vm.mmap_min_addr = 65536" >${pkgdir}/$pkgname-$pkgver/usr/lib/sysctl.d/mmap_min_addr.conf
+    (
+        cd $pkgdir
+        tar Jcf $srcdir/$tarballs $pkgname-$pkgver
+    )
 }
 
-parseArgs "$@"
-make_cmd
+parpare
+build
+package
-- 
2.49.0

