From f81c45b3035d0d70da9e645a8a030182dc725f92 Mon Sep 17 00:00:00 2001
From: Hanlu Li <heuleehanlu@gmail.com>
Date: Wed, 2 Apr 2025 14:28:26 +0800
Subject: [PATCH 1025/1058] LATX, fix: disable KZT by default

Signed-off-by: Hanlu Li <heuleehanlu@gmail.com>
---
 target/i386/latx/latx-options.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/target/i386/latx/latx-options.c b/target/i386/latx/latx-options.c
index caf75d37ba..4d59aaba17 100644
--- a/target/i386/latx/latx-options.c
+++ b/target/i386/latx/latx-options.c
@@ -5,7 +5,7 @@
 #include "latx-debug.h"
 
 #if defined(CONFIG_LATX_KZT)
-int option_kzt = 1;
+int option_kzt = 0;
 #endif
 
 #ifdef CONFIG_LATX_FLAG_REDUCTION
-- 
2.49.0

