From f54ef63f86a3fde47d9ad77f2dac719999f053df Mon Sep 17 00:00:00 2001
From: Wenqiang Wei <weiwenqiang@mail.ustc.edu.cn>
Date: Tue, 8 Apr 2025 15:13:13 +0800
Subject: [PATCH 1033/1058] LATX, fix: SHBR src maybe not XMM.

---
 target/i386/latx/optimization/hbr.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/target/i386/latx/optimization/hbr.c b/target/i386/latx/optimization/hbr.c
index 2f34476a72..7eb57a7faa 100644
--- a/target/i386/latx/optimization/hbr.c
+++ b/target/i386/latx/optimization/hbr.c
@@ -858,8 +858,7 @@ static void over_tb_shbr_opt(TranslationBlock **tb_list, int tb_num_in_tu,
                         if ((ir1->xmm_def & SHBR_UPDATE_DES) && !(ir1->xmm_use & SHBR_NEED_DES)) {
                             no_opt_xmm &= ~(1 << dest_num);
                         }
-                        if (ir1->xmm_use & SHBR_NEED_SRC) {
-                            assert(ir1_opnd_is_xmm(opnd1));
+                        if ((ir1->xmm_use & SHBR_NEED_SRC) && ir1_opnd_is_xmm(opnd1)) {
                             src_num = ir1_opnd_base_reg_num(opnd1);
                             no_opt_xmm |= (1 << src_num);
                         }
-- 
2.49.0

