From b0f9d467b4a1accd10a068034dd8ea6f427d4b4d Mon Sep 17 00:00:00 2001
From: Wenqiang Wei <weiwenqiang@mail.ustc.edu.cn>
Date: Tue, 8 Apr 2025 11:32:20 +0800
Subject: [PATCH 1035/1058] LATX, fix: The warning of AOT only appears in debug
 mode.

---
 accel/tcg/translate-all.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/accel/tcg/translate-all.c b/accel/tcg/translate-all.c
index e67b97e4f6..445050b625 100644
--- a/accel/tcg/translate-all.c
+++ b/accel/tcg/translate-all.c
@@ -2404,7 +2404,9 @@ void aot_tb_register(TranslationBlock *tb)
     existing_tb = tb_link_page(tb, phys_pc, phys_page2);
     /* if the TB already exists, discard what we just translated */
     if (existing_tb != tb) {
+#ifdef CONFIG_LATX_DEBUG
         fprintf(stderr, "aot_register_tb failed due to tb_link page\n");
+#endif
     }
     tcg_tb_insert(tb);
 }
-- 
2.49.0

