From 7b5ba6187fe16d2ceae773ea8e5d0b4139ffb630 Mon Sep 17 00:00:00 2001
From: Hanlu Li <heuleehanlu@gmail.com>
Date: Sat, 19 Apr 2025 16:23:28 +0800
Subject: [PATCH 1053/1058] LATX, fix: Fix invalid NULL usage in epoll_wait
 call

Signed-off-by: Hanlu Li <heuleehanlu@gmail.com>
---
 target/i386/latx/context/wrappedlibc.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/target/i386/latx/context/wrappedlibc.c b/target/i386/latx/context/wrappedlibc.c
index fa5b70c644..f104feeb80 100644
--- a/target/i386/latx/context/wrappedlibc.c
+++ b/target/i386/latx/context/wrappedlibc.c
@@ -1877,8 +1877,13 @@ EXPORT int32_t my_epoll_ctl(int32_t epfd, int32_t op, int32_t fd, void* event)
 EXPORT int32_t my_epoll_wait(int32_t epfd, void* events, int32_t maxevents, int32_t timeout)
 {
     struct epoll_event _events[maxevents];
+    int32_t ret = 0;
     //AlignEpollEvent(_events, events, maxevents);
-    int32_t ret = epoll_wait(epfd, events?_events:NULL, maxevents, timeout);
+    if (events) {
+        ret = epoll_wait(epfd, _events, maxevents, timeout);
+    } else {
+        ret = -1;
+    }
     if(ret>0)
         UnalignEpollEvent(events, _events, ret);
     return ret;
@@ -1886,8 +1891,13 @@ EXPORT int32_t my_epoll_wait(int32_t epfd, void* events, int32_t maxevents, int3
 EXPORT int32_t my_epoll_pwait(int32_t epfd, void* events, int32_t maxevents, int32_t timeout, const sigset_t *sigmask)
 {
     struct epoll_event _events[maxevents];
+    int32_t ret = 0;
     //AlignEpollEvent(_events, events, maxevents);
-    int32_t ret = epoll_pwait(epfd, events?_events:NULL, maxevents, timeout, sigmask);
+    if (events) {
+        ret = epoll_pwait(epfd, _events, maxevents, timeout, sigmask);
+    } else {
+        ret = -1;
+    }
     if(ret>0)
         UnalignEpollEvent(events, _events, ret);
     return ret;
-- 
2.49.0

