diff -Naur libreoffice-7.0.0.3.old/external/firebird/ExternalProject_firebird.mk libreoffice-7.0.0.3/external/firebird/ExternalProject_firebird.mk
--- libreoffice-7.0.0.3.old/external/firebird/ExternalProject_firebird.mk	2020-07-30 03:29:17.000000000 +0800
+++ libreoffice-7.0.0.3/external/firebird/ExternalProject_firebird.mk	2020-10-31 06:40:33.683138836 +0800
@@ -57,6 +57,7 @@
 				-I$(call gb_UnpackedTarball_get_dir,icu)/source/common \
 			) \
 			$(if $(filter GCC-INTEL,$(COM)-$(CPUNAME)),-Di386=1) \
+			$(if $(ENABLE_LTO),$(gb_LTOFLAGS)) \
 			" \
 		&& export CXXFLAGS=" \
 			$(if $(filter MSC,$(COM)),$(if $(MSVC_USE_DEBUG_RUNTIME),-DMSVC_USE_DEBUG_RUNTIME)) \
@@ -76,11 +77,13 @@
 			$(CXXFLAGS_CXX11) \
 			$(firebird_NO_CXX11_NARROWING) \
 			$(if $(call gb_Module__symbols_enabled,firebird),$(gb_DEBUGINFO_FLAGS)) \
+			$(if $(ENABLE_LTO),$(gb_LTOFLAGS)) \
 		" \
 		&& export LDFLAGS=" \
 			$(if $(SYSTEM_ICU),$(ICU_LIBS), \
 				-L$(call gb_UnpackedTarball_get_dir,icu)/source/lib \
 			) \
+			$(if $(ENABLE_LTO),$(gb_LTOFLAGS)) \
 		" \
 		&& export LIBREOFFICE_ICU_LIB="$(call gb_UnpackedTarball_get_dir,icu)/source/lib" \
 		&& MAKE=$(MAKE) ./configure \
