--- libreoffice-7.1.0.3/external/firebird/ExternalProject_firebird.mk	2021-01-27 11:33:49.000000000 -0800
+++ libreoffice-7.1.0.3-new/external/firebird/ExternalProject_firebird.mk	2021-02-25 03:06:42.559331819 -0800
@@ -51,6 +51,7 @@
 			$(CXXFLAGS_CXX11) \
 			$(if $(filter TRUE,$(COM_IS_CLANG)), -Wno-c++11-narrowing) \
 			$(if $(call gb_Module__symbols_enabled,firebird),$(gb_DEBUGINFO_FLAGS)) \
+                        $(if $(ENABLE_LTO),$(gb_LTOFLAGS)) \
 		" \
 		&& export LDFLAGS=" \
 			$(if $(SYSTEM_LIBATOMIC_OPS),$(LIBATOMIC_OPS_LIBS), \
@@ -62,6 +63,7 @@
 			$(if $(SYSTEM_ICU),$(ICU_LIBS), \
 				-L$(call gb_UnpackedTarball_get_dir,icu)/source/lib \
 			) \
+                        $(if $(ENABLE_LTO),$(gb_LTOFLAGS)) \
 		" \
 		&& export LIBREOFFICE_ICU_LIB="$(call gb_UnpackedTarball_get_dir,icu)/source/lib" \
 		&& export MSVC_USE_INDIVIDUAL_PDBS=TRUE \
