diff -Naur libreoffice-5.4.2.2/Makefile.in libreoffice-5.4.2.2.root/Makefile.in
--- libreoffice-5.4.2.2/Makefile.in	2017-09-27 18:03:26.000000000 +0000
+++ libreoffice-5.4.2.2.root/Makefile.in	2017-10-09 05:43:30.892722495 +0000
@@ -57,14 +57,6 @@
 
 all: build
 
-check-if-root:
-	@if test `id -u` = 0 && ! grep -q 'lxc\|docker' /proc/self/cgroup; then \
-		echo; \
-		echo 'No. You make ME a sandwich.'; \
-		echo; \
-		exit 1; \
-	fi
-
 gb_Side ?= host
 
 include $(BUILDDIR)/config_$(gb_Side).mk
