diff -Naur libreoffice-6.1.2.1/Makefile.in libreoffice-6.1.2.1.root/Makefile.in
--- libreoffice-6.1.2.1/Makefile.in	2018-09-20 09:44:38.000000000 +0000
+++ libreoffice-6.1.2.1.root/Makefile.in	2018-10-06 02:49:43.091068224 +0000
@@ -59,14 +59,6 @@
 
 all: build
 
-check-if-root:
-	@if test ! `uname` = 'Haiku' -a `id -u` = 0 && ! grep -q 'lxc\|docker' /proc/self/cgroup; then \
-		echo; \
-		echo 'No. You make ME a sandwich.'; \
-		echo; \
-		exit 1; \
-	fi
-
 gb_Side ?= host
 
 include $(BUILDDIR)/config_$(gb_Side).mk
