diff -Naur libreoffice-5.3.0.3/Makefile.in libreoffice-5.3.0.3.rootbuild/Makefile.in
--- libreoffice-5.3.0.3/Makefile.in	2017-01-26 16:22:46.000000000 -0600
+++ libreoffice-5.3.0.3.rootbuild/Makefile.in	2017-02-03 00:46:03.465414621 -0600
@@ -54,14 +54,6 @@
 
 all: build
 
-check-if-root:
-	@if test `id -u` = 0; then \
-		echo; \
-		echo 'No. You make ME a sandwich.'; \
-		echo; \
-		exit 1; \
-	fi
-
 gb_Side ?= host
 
 include $(BUILDDIR)/config_$(gb_Side).mk
