--- a/build/depends.py.bak        2020-10-06 00:31:20.376856226 -0500
+++ b/build/depends.py    2020-10-06 00:33:05.061558502 -0500
@@ -615,7 +615,7 @@
             env.Append(CPPPATH=[self.SOUNDTOUCH_INTERNAL_PATH])
 
             # Prevents circular import.
-            from features import Optimize
+            from .features import Optimize
 
             # If we do not want optimizations then disable them.
             optimize = (build.flags['optimize'] if 'optimize' in build.flags
