diff -urN netbsd-sh/cd.c ash-0.3.7.orig/cd.c
--- netbsd-sh/cd.c	Fri Jul  9 13:02:05 1999
+++ ash-0.3.7.orig/cd.c	Mon Apr 23 22:16:46 2001
@@ -319,7 +319,7 @@
 	 * c implementation of getcwd, that does not open a pipe to
 	 * /bin/pwd.
 	 */
-#if defined(__NetBSD__) || defined(__SVR4)
+#if defined(__NetBSD__) || defined(__SVR4) || defined(__GLIBC__)
 		
 	if (getcwd(buf, sizeof(buf)) == NULL) {
 		char *pwd = getenv("PWD");

