diff -urN netbsd-sh/debian/README.debian ash-0.3.7.orig/debian/README.debian
--- netbsd-sh/debian/README.debian	Thu Jan  1 01:00:00 1970
+++ ash-0.3.7.orig/debian/README.debian	Mon Apr 23 22:16:46 2001
@@ -0,0 +1,14 @@
+ash for DEBIAN
+----------------------
+
+This is a very simple port of ash taken from NetBSD-current on 1997/06/18.  The
+file setmode.c was taken from src/libc/gen and is needed since the Linux libc
+does provide getmode(3) or setmode(3).  History editing is disabled to avoid
+code bloat.  This also means that building the package is possible without the
+BSD libedit.
+
+This port is preriodically revised to keep up to date with NetBSD's current
+release.
+
+Herbert Xu <herbert@debian.org>
+$Id: README.debian,v 1.6 2000/08/04 11:33:48 herbert Exp $
diff -urN netbsd-sh/debian/ash-medium.README.Debian ash-0.3.7.orig/debian/ash-medium.README.Debian
--- netbsd-sh/debian/ash-medium.README.Debian	Thu Jan  1 01:00:00 1970
+++ ash-0.3.7.orig/debian/ash-medium.README.Debian	Mon Apr 23 22:16:46 2001
@@ -0,0 +1,14 @@
+ash for DEBIAN
+----------------------
+
+This is a very simple port of ash taken from NetBSD-current on 1997/06/18.  The
+file setmode.c was taken from src/libc/gen and is needed since the Linux libc
+does provide getmode(3) or setmode(3).  History editing is disabled to avoid
+code bloat.  This also means that building the package is possible without the
+BSD libedit.
+
+This port is preriodically revised to keep up to date with NetBSD's current
+release.
+
+Herbert Xu <herbert@debian.org>
+$Id: ash-medium.README.Debian,v 1.1 2000/08/04 11:49:01 herbert Exp $
diff -urN netbsd-sh/debian/ash-medium.dirs ash-0.3.7.orig/debian/ash-medium.dirs
--- netbsd-sh/debian/ash-medium.dirs	Thu Jan  1 01:00:00 1970
+++ ash-0.3.7.orig/debian/ash-medium.dirs	Mon Apr 23 22:16:46 2001
@@ -0,0 +1,2 @@
+usr/bin
+usr/share/man/man1
diff -urN netbsd-sh/debian/ash-udeb.dirs ash-0.3.7.orig/debian/ash-udeb.dirs
--- netbsd-sh/debian/ash-udeb.dirs	Thu Jan  1 01:00:00 1970
+++ ash-0.3.7.orig/debian/ash-udeb.dirs	Mon Apr 23 22:16:46 2001
@@ -0,0 +1 @@
+bin
diff -urN netbsd-sh/debian/bsdyacc ash-0.3.7.orig/debian/bsdyacc
--- netbsd-sh/debian/bsdyacc	Thu Jan  1 01:00:00 1970
+++ ash-0.3.7.orig/debian/bsdyacc	Mon Apr 23 22:16:46 2001
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+
+if echo "$@" | grep -q -- -o; then
+	OUTPUT=$(echo "$@" |
+		 sed 's/.*-o[[:blank:]]\+\([^[:blank:]]\+\)\.c.*/\1/')
+	OPTIONS=$(echo "$@" |
+		 sed 's/\(.*\)-o[[:blank:]]\+[^[:blank:]]\+\(.*\)/\1\2/')
+	NEW=1
+else
+	OUTPUT=$(echo "$@" |
+		 sed -e 's/.*[[:blank:]]\+\([^[:blank:]]\+\)\.y.*/\1/')
+	OPTIONS="$@"
+	NEW=0
+fi
+
+byacc $OPTIONS
+if [ $NEW = 1 ]; then
+	mv y.tab.c $OUTPUT.c
+fi
+mv y.tab.h $OUTPUT.h
diff -urN netbsd-sh/debian/changelog ash-0.3.7.orig/debian/changelog
--- netbsd-sh/debian/changelog	Thu Jan  1 01:00:00 1970
+++ ash-0.3.7.orig/debian/changelog	Mon Apr 23 22:16:46 2001
@@ -0,0 +1,429 @@
+ash (0.3.7-14) unstable; urgency=low
+
+  * Removed predependency from udeb (closes: #81995).
+  * Added /bin/sh symlink to udeb (closes: #81967).
+
+ -- Herbert Xu <herbert@debian.org>  Sat, 13 Jan 2001 15:23:21 +1100
+
+ash (0.3.7-13) unstable; urgency=low
+
+  * Renamed the udeb to ash-udeb.
+
+ -- Herbert Xu <herbert@debian.org>  Wed, 20 Dec 2000 19:32:34 +1100
+
+ash (0.3.7-12) unstable; urgency=low
+
+  * Added support for udebs (Randolph Chung, closes: #79237).
+
+ -- Herbert Xu <herbert@debian.org>  Sat, 16 Dec 2000 13:53:28 +1100
+
+ash (0.3.7-11) unstable; urgency=low
+
+  * Preserve the previous exit status upon entering a function
+    (closes: #78374).
+
+ -- Herbert Xu <herbert@debian.org>  Sun,  3 Dec 2000 13:34:27 +1100
+
+ash (0.3.7-10) unstable; urgency=low
+
+  * Merged changes for GNU from Igor Khavkine.
+  * Minimise the number of sigactions.
+
+ -- Herbert Xu <herbert@debian.org>  Fri,  3 Nov 2000 20:31:52 +1100
+
+ash (0.3.7-9) unstable; urgency=low
+
+  * Predepend on the libraries.
+  * Always save fd 2 when it is redirected (closes: #75302).
+
+ -- Herbert Xu <herbert@debian.org>  Sun, 22 Oct 2000 08:40:40 +1100
+
+ash (0.3.7-8) unstable; urgency=high
+
+  * More redirection fixes (closes: #73613).
+
+ -- Herbert Xu <herbert@debian.org>  Thu,  5 Oct 2000 18:22:17 +1100
+
+ash (0.3.7-7) unstable; urgency=high
+
+  * Added missing break in redirection code (closes: #72956).
+
+ -- Herbert Xu <herbert@debian.org>  Tue,  3 Oct 2000 07:58:04 +1100
+
+ash (0.3.7-6) unstable; urgency=low
+
+  * command -[vV] no longer displays an error message on stdout.
+  * Redirecting to /proc/self/fd/* now works (closes: #72852).
+
+ -- Herbert Xu <herbert@debian.org>  Sun,  1 Oct 2000 12:56:39 +1100
+
+ash (0.3.7-5) unstable; urgency=low
+
+  * Implemented set -a.
+
+ -- Herbert Xu <herbert@debian.org>  Sat, 30 Sep 2000 16:00:33 +1100
+
+ash (0.3.7-4) unstable; urgency=low
+
+  * Added build-time dependency on debhelper (closes: #69920).
+  * Extended maximum length of arithmetic expansions to match 32-bit integers.
+
+ -- Herbert Xu <herbert@debian.org>  Wed, 20 Sep 2000 14:28:16 +1100
+
+ash (0.3.7-3) unstable; urgency=low
+
+  * Switch to the old globbing code since glob(3) is hopelessly broken
+    (closes: #69455).
+
+ -- Herbert Xu <herbert@debian.org>  Mon, 21 Aug 2000 20:37:15 +1000
+
+ash (0.3.7-2) unstable; urgency=low
+
+  * Call glob(3) with GLOB_NOMAGIC (ouch).
+
+ -- Herbert Xu <herbert@debian.org>  Sun,  6 Aug 2000 17:47:08 +1000
+
+ash (0.3.7-1) unstable; urgency=low
+
+  * NetBSD-current version as of 20000729.
+  * Use fnmatch(3) and glob(3).
+  * Fixed the use of backslashes in the pattern in parameter substitutions,
+    hopefully for the last time.
+  * Applied HETIO patch and built ash.medium (closes: #50788).  Will do ash.big
+    when readline is fixed so that it doesn't leak anymore.
+
+ -- Herbert Xu <herbert@debian.org>  Fri,  4 Aug 2000 21:36:44 +1000
+
+ash (0.3.6-5) unstable; urgency=low
+
+  * Fixed manpage entry for read with patch from Kevin Ryde (closes: #62500).
+  * Fixed a file descriptor leak for pipelines.
+
+ -- Herbert Xu <herbert@debian.org>  Wed, 19 Apr 2000 18:56:20 +1000
+
+ash (0.3.6-4) unstable; urgency=low
+
+  * Fixed the case of an empty command with redirections.
+
+ -- Herbert Xu <herbert@debian.org>  Fri,  7 Apr 2000 12:07:18 +1000
+
+ash (0.3.6-3) unstable; urgency=low
+
+  * ! is now recognised correctly.
+  * Ash is now more strict on the syntax, e.g., a lone ! is no longer accepted
+    as an alternative to ! true.
+
+ -- Herbert Xu <herbert@debian.org>  Fri,  7 Apr 2000 10:46:06 +1000
+
+ash (0.3.6-2) unstable; urgency=low
+
+  * Fixed a problem with fmtstr() which broke getopts.
+
+ -- Herbert Xu <herbert@debian.org>  Sun,  2 Apr 2000 10:49:26 +1000
+
+ash (0.3.6-1) unstable; urgency=low
+
+  * NetBSD-current version as of 20000326.
+  * Added a Build-Depends on groff (closes: #61041).
+  * Implemented noclobber (closes: #59028).
+  * Rewrote output.c to use stream IO.
+
+ -- Herbert Xu <herbert@debian.org>  Sat,  1 Apr 2000 19:24:31 +1000
+
+ash (0.3.5-10) frozen unstable; urgency=low
+
+  * Don't stat mail boxes in non-interactive mode (closes: #59213).
+  * Added an fflush(stdout) to the times builtin (closes: #59027).
+  * Documented the times builtin.
+  * Added source depends.
+
+ -- Herbert Xu <herbert@debian.org>  Sat, 18 Mar 2000 18:58:44 +1100
+
+ash (0.3.5-9) unstable; urgency=low
+
+  * Double quotes inside paramater substitutions inside double quotes are now
+    ignored as in bash (the originial behaviour was POSIX compliant too but
+    IMHO this one makes a little bit more sense).
+    This one broke mwm (but it was actually mwm's fault).
+  * Corrected backslash/CTLESC treatment for patterns in parameter
+    substitutions.
+
+ -- Herbert Xu <herbert@debian.org>  Sat,  6 Nov 1999 18:13:19 +1100
+
+ash (0.3.5-8) unstable; urgency=low
+
+  * Replaced use of echo -n in manual page with escape codes.
+  * Made FHS compliant (closes: #47978).
+  * Restored echo's option processing ability.
+
+ -- Herbert Xu <herbert@debian.org>  Fri, 22 Oct 1999 10:20:58 +1000
+
+ash (0.3.5-7) unstable; urgency=low
+
+  * echo no longer supports options.
+  * Don't quote patterns inside parameter substitutions enclosed by double
+    quotes (closes: #47842).
+
+ -- Herbert Xu <herbert@debian.org>  Wed, 20 Oct 1999 20:28:14 +1000
+
+ash (0.3.5-6) unstable; urgency=low
+
+  * Use getcwd() instead of /bin/pwd -- Zack Weinberg (closes: #46981).
+
+ -- Herbert Xu <herbert@debian.org>  Sun, 10 Oct 1999 16:31:49 +1000
+
+ash (0.3.5-5) unstable; urgency=low
+
+  * Only test for -e on simple commands (fixes #44559).
+
+ -- Herbert Xu <herbert@debian.org>  Wed,  8 Sep 1999 22:18:27 +1000
+
+ash (0.3.5-4) unstable; urgency=low
+
+  * Don't wait for stopped children if job control is disabled (fixes #42814).
+  * Allow an option '(' in a case statement (fixes #42364).
+
+ -- Herbert Xu <herbert@debian.org>  Thu, 12 Aug 1999 23:30:30 +1000
+
+ash (0.3.5-3) unstable; urgency=low
+
+  * OK, the fix to the esoteric problem in 0.3.5-1 actually breaks VSASSIGN
+    and VSQUESTION, they should work properly now (fixes #41327).
+
+ -- Herbert Xu <herbert@debian.org>  Thu, 15 Jul 1999 22:47:13 +1000
+
+ash (0.3.5-2) unstable; urgency=low
+
+  * PATH search and execution is now correct.
+  * hash no longer shows builtins.
+  * Added kill builtin.
+  * New description from James R. van Zandt reformatted by Josip Rodin.
+
+ -- Herbert Xu <herbert@debian.org>  Mon, 12 Jul 1999 18:51:42 +1000
+
+ash (0.3.5-1) unstable; urgency=low
+
+  * New upstream release.
+  * Adapted to new pmake (fixes #38737).
+  * Fixed behvaiour of backslashes preceding a closing brace for a parameter
+    substituion inside double quotes (even bash messes this one up :).
+  * Fixed command (fixes #34639).
+  * Fixed a pipe bug where stdin may be wrongly closed (fixes #35452).
+  * Revamped getopts (fixes #39694).
+
+ -- Herbert Xu <herbert@debian.org>  Sun,  4 Jul 1999 12:19:01 +1000
+
+ash (0.3.4-7) unstable; urgency=low
+
+  * Fixed a glibc 2.1 compatitibility problem.
+  * Fixed a PWD inconsistency that stuffed up the kernel compilation.
+
+ -- Herbert Xu <herbert@debian.org>  Mon, 17 May 1999 23:14:57 +1000
+
+ash (0.3.4-6) unstable; urgency=low
+
+  * Fixed incorrect -e test due to the last bug fix (fixes #26509).
+
+ -- Herbert Xu <herbert@debian.org>  Tue,  8 Sep 1998 10:02:46 +1000
+
+ash (0.3.4-5) unstable; urgency=low
+
+  * Use test_eaccess from bash instead of access(2) (fixes #26110).
+
+ -- Herbert Xu <herbert@debian.org>  Wed, 26 Aug 1998 21:22:49 +1000
+
+ash (0.3.4-4) unstable; urgency=low
+
+  * Only upload to unstable.
+
+ -- Herbert Xu <herbert@debian.org>  Tue,  5 May 1998 18:01:02 +1000
+
+ash (0.3.4-3) frozen unstable; urgency=low
+
+  * Applied sparc patch (fixes #21562).
+
+ -- Herbert Xu <herbert@debian.org>  Fri,  1 May 1998 19:48:13 +1000
+
+ash (0.3.4-2) frozen unstable; urgency=low
+
+  * Fixed the incorrect trap fixes (fixes #20363).
+
+ -- Herbert Xu <herbert@debian.org>  Thu, 16 Apr 1998 21:07:10 +1000
+
+ash (0.3.4-1) unstable; urgency=low
+
+  * New upstream release.
+  * Reverted word splitting change in 0.3.2-1 since the fix was broken and
+    major work (the quote removal is done too quickly at the moment) is needed
+    to fix it properly.
+  * Fixed more trap noncompliance.
+
+ -- Herbert Xu <herbert@debian.org>  Thu, 19 Mar 1998 22:59:12 +1100
+
+ash (0.3.2-5) unstable; urgency=low
+
+  * Fixed a bug when doing pattern matching in parameter expansions.
+
+ -- Herbert Xu <herbert@debian.org>  Tue, 10 Mar 1998 21:25:40 +1100
+
+ash (0.3.2-4) unstable; urgency=low
+
+  * Allow ] to be quoted in bracket expressions (fixes #17533).
+  * Move dh_fixperms to second last spot (fixes #18267).
+  * Don't do field splitting in evalfor.
+
+ -- Herbert Xu <herbert@debian.org>  Tue, 17 Feb 1998 13:32:09 +1100
+
+ash (0.3.2-3) unstable; urgency=low
+
+  * Fixed stupid core dump.
+
+ -- Herbert Xu <herbert@debian.org>  Wed, 11 Feb 1998 21:33:55 +1100
+
+ash (0.3.2-2) unstable; urgency=low
+
+  * Hack for special builtins (fixes #18055).
+  * Hack for command.
+
+ -- Herbert Xu <herbert@debian.org>  Wed, 11 Feb 1998 21:19:46 +1100
+
+ash (0.3.2-1) unstable; urgency=low
+
+  * NetBSD-current version as of 19980209.
+  * Fixed a word splitting problem after parameter expansion thanks to Alexey
+    Marinichev.
+  * Converted to debhelper (fixes #14612, #15005).
+
+ -- Herbert Xu <herbert@debian.org>  Mon,  9 Feb 1998 16:53:48 +1100
+
+ash (0.3.1-20) unstable; urgency=low
+
+  * Fixed -e problem with eval.
+
+ -- Herbert Xu <herbert@debian.org>  Sun,  7 Dec 1997 20:19:00 +1100
+
+ash (0.3.1-19) unstable; urgency=low
+
+  * Fixed -e problem with command substitution.
+
+ -- Herbert Xu <herbert@debian.org>  Sun,  7 Dec 1997 19:44:49 +1100
+
+ash (0.3.1-18) unstable; urgency=low
+
+  * Do not link with ncurses (#15485).
+
+ -- Herbert Xu <herbert@debian.org>  Sun, 30 Nov 1997 12:00:11 +1100
+
+ash (0.3.1-17) unstable; urgency=low
+
+  * Set PATH like bash (#15238).
+
+ -- Herbert Xu <herbert@debian.org>  Wed, 26 Nov 1997 16:17:27 +1100
+
+ash (0.3.1-16) unstable; urgency=low
+
+  * Fixed incorrect assignment builtin code.
+
+ -- Herbert Xu <herbert@debian.org>  Mon, 24 Nov 1997 16:19:10 +1100
+
+ash (0.3.1-15) unstable; urgency=low
+
+  * hash now returns error codes (needed by the Linux kernel).
+
+ -- Herbert Xu <herbert@debian.org>  Sun, 23 Nov 1997 21:37:08 +1100
+
+ash (0.3.1-14) unstable; urgency=low
+
+  * Disabled word-splitting for assignment builtins.
+
+ -- Herbert Xu <herbert@debian.org>  Sun, 23 Nov 1997 12:45:15 +1100
+
+ash (0.3.1-13) unstable; urgency=low
+
+  * ! is now recognised even after &&/||.
+
+ -- Herbert Xu <herbert@debian.org>  Fri, 21 Nov 1997 22:09:05 +1100
+
+ash (0.3.1-12) unstable; urgency=low
+
+  * More fixes to the handling of SIGINT when forking.
+
+ -- Herbert Xu <herbert@debian.org>  Fri, 14 Nov 1997 15:14:32 +1100
+
+ash (0.3.1-11) unstable; urgency=low
+
+  * Ignore SIGINT when forking non-interactively.
+
+ -- Herbert Xu <herbert@debian.org>  Mon,  3 Nov 1997 12:00:02 +1100
+
+ash (0.3.1-10) unstable; urgency=low
+
+  * echo now handles options correctly.
+  * echo nolonger returns 0 if erorrs occured while writing to stdout.
+  * New code from GNU echo merged.
+  * Error messages from test now work.
+
+ -- Herbert Xu <herbert@debian.org>  Wed,  8 Oct 1997 21:47:13 +1000
+
+ash (0.3.1-9) unstable; urgency=low
+
+  * ! is recognised at pipeline level like bash.
+
+ -- Herbert Xu <herbert@debian.org>  Mon, 15 Sep 1997 23:13:45 +1000
+
+ash (0.3.1-8) unstable; urgency=medium
+
+  * Old patch regarding SIGCHLD in again.
+
+ -- Herbert Xu <herbert@debian.org>  Sun, 31 Aug 1997 11:20:27 +1000
+
+ash (0.3.1-7) unstable; urgency=low
+
+  * /bin/sh -e is behaving even better now (for loops within conditionals).
+
+ -- Herbert Xu <herbert@debian.org>  Sat, 23 Aug 1997 22:08:19 +1000
+
+ash (0.3.1-6) unstable; urgency=low
+
+  * /bin/sh -e is behaving better now.
+
+ -- Herbert Xu <herbert@debian.org>  Sat, 23 Aug 1997 13:16:26 +1000
+
+ash (0.3.1-5) unstable; urgency=low
+
+  * hash -v /dir/command doesn't coredump anymore.
+  * type /dir/command now works correctly.
+
+ -- Herbert Xu <herbert@debian.org>  Fri,  1 Aug 1997 20:48:19 +1000
+
+ash (0.3.1-4) unstable; urgency=low
+
+  * trap now understands symbolic signal names.
+
+ -- Herbert Xu <herbert@debian.org>  Sat, 26 Jul 1997 14:04:46 +1000
+
+ash (0.3.1-3) unstable; urgency=low
+
+  * Added the builtin test command.
+
+ -- Herbert Xu <herbert@debian.org>  Sun, 20 Jul 1997 15:00:14 +1000
+
+ash (0.3.1-2) unstable; urgency=medium
+
+  * Fixed a coredump involving $*.
+
+ -- Herbert Xu <herbert@debian.org>  Sat, 19 Jul 1997 12:03:02 +1000
+
+ash (0.3.1-1) unstable; urgency=medium
+
+  * NetBSD-current version as of 19970715.
+  * Fixed a "use after free" bug (#11294).
+
+ -- Herbert Xu <herbert@debian.org>  Fri, 18 Jul 1997 13:48:09 +1000
+
+ash (0.3-1) unstable; urgency=low
+
+  * Initial Release.
+
+ -- Herbert Xu <herbert@debian.org>  Thu, 19 Jun 1997 19:29:16 +1000
+
diff -urN netbsd-sh/debian/control ash-0.3.7.orig/debian/control
--- netbsd-sh/debian/control	Thu Jan  1 01:00:00 1970
+++ ash-0.3.7.orig/debian/control	Mon Apr 23 22:16:46 2001
@@ -0,0 +1,69 @@
+Source: ash
+Section: shells
+Priority: optional
+Maintainer: Herbert Xu <herbert@debian.org>
+Build-Depends: byacc, debhelper, flex, pmake, groff
+Standards-Version: 3.2.1
+
+Package: ash
+Architecture: any
+Pre-Depends: ${shlibs:Depends}
+Description: NetBSD /bin/sh
+ "ash" is a POSIX compliant shell that is much smaller than "bash".
+ We take advantage of that by making it the shell on the installation
+ root floppy, where space is at a premium.
+ .
+ It can be usefully installed as /bin/sh (because it executes scripts
+ somewhat faster than "bash"), or as the default shell either of root
+ or of a second user with a userid of 0 (because it depends on fewer
+ libraries, and is therefore less likely to be affected by an upgrade
+ problem or a disk failure).  It is also useful for checking that a
+ script uses only POSIX syntax.
+ .
+ "bash" is a better shell for most users, since it has some nice
+ features absent from "ash", and is a required part of the system.
+
+Package: ash-medium
+Architecture: any
+Pre-Depends: ${shlibs:Depends}
+Priority: extra
+Description: NetBSD /bin/sh with HETIO
+ This is a slightly bigger version of the standard ash package with a
+ hack that provides primitive history support.  It may be useful on
+ boot floppies where space is at a premium, yet users still need the
+ ability to access previous commands in the same session.  If you're
+ not a boot floppies maintainer, you should probably go for ash,
+ ash.big (not yet available), or bash.
+ .
+ "ash" is a POSIX compliant shell that is much smaller than "bash".
+ We take advantage of that by making it the shell on the installation
+ root floppy, where space is at a premium.
+ .
+ It can be usefully installed as /bin/sh (because it executes scripts
+ somewhat faster than "bash"), or as the default shell either of root
+ or of a second user with a userid of 0 (because it depends on fewer
+ libraries, and is therefore less likely to be affected by an upgrade
+ problem or a disk failure).  It is also useful for checking that a
+ script uses only POSIX syntax.
+ .
+ "bash" is a better shell for most users, since it has some nice
+ features absent from "ash", and is a required part of the system.
+
+Package: ash-udeb
+Architecture: any
+Section: debian-installer
+Description: NetBSD /bin/sh for boot floppies
+ "ash" is a POSIX compliant shell that is much smaller than "bash".
+ We take advantage of that by making it the shell on the installation
+ root floppy, where space is at a premium.
+ .
+ It can be usefully installed as /bin/sh (because it executes scripts
+ somewhat faster than "bash"), or as the default shell either of root
+ or of a second user with a userid of 0 (because it depends on fewer
+ libraries, and is therefore less likely to be affected by an upgrade
+ problem or a disk failure).  It is also useful for checking that a
+ script uses only POSIX syntax.
+ .
+ "bash" is a better shell for most users, since it has some nice
+ features absent from "ash", and is a required part of the system.
+
diff -urN netbsd-sh/debian/copyright ash-0.3.7.orig/debian/copyright
--- netbsd-sh/debian/copyright	Thu Jan  1 01:00:00 1970
+++ ash-0.3.7.orig/debian/copyright	Mon Apr 23 22:16:46 2001
@@ -0,0 +1,18 @@
+This package was debianized by Mark W. Eichin eichin@kitten.gen.ma.us on
+Mon, 24 Feb 1997 16:00:16 -0500.
+
+This package was re-ported from NetBSD and debianized by
+Herbert Xu herbert@debian.org on Thu, 19 Jun 1997 19:29:16 +1000.
+
+It was downloaded from ftp.netbsd.org.
+
+Copyright:
+
+Copyright (c) 1991, 1993
+	The Regents of the University of California.  All rights reserved.
+
+This code is derived from software contributed to Berkeley by Kenneth Almquist.
+
+Please refer to /usr/share/common-licenses/BSD for details.
+
+$Id: copyright,v 1.3 2000/08/04 11:48:49 herbert Exp $
diff -urN netbsd-sh/debian/dirs ash-0.3.7.orig/debian/dirs
--- netbsd-sh/debian/dirs	Thu Jan  1 01:00:00 1970
+++ ash-0.3.7.orig/debian/dirs	Mon Apr 23 22:16:46 2001
@@ -0,0 +1,2 @@
+bin
+usr/share/man/man1
diff -urN netbsd-sh/debian/rules ash-0.3.7.orig/debian/rules
--- netbsd-sh/debian/rules	Thu Jan  1 01:00:00 1970
+++ ash-0.3.7.orig/debian/rules	Mon Apr 23 22:16:46 2001
@@ -0,0 +1,92 @@
+#!/usr/bin/make -f
+# $Id: rules,v 1.22 2001/01/13 04:23:49 herbert Exp $
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+CDEF = -g -O2 -Wall -DBSD=1 -DSMALL -D_GNU_SOURCE \
+       -DGLOB_BROKEN \
+       -D__COPYRIGHT\(x\)= -D__RCSID\(x\)= -D_DIAGASSERT\(x\)=
+CDEFSM = -Os -fomit-frame-pointer -Wall -DBSD=1 -DSMALL -D_GNU_SOURCE \
+       -DGLOB_BROKEN \
+       -D__COPYRIGHT\(x\)= -D__RCSID\(x\)= -D_DIAGASSERT\(x\)=
+
+sh.hetio:
+	chmod u+x debian/bsdyacc
+	rm -f *.o
+	pmake CFLAGS='$(CDEF)' HETIO= YACC=`pwd`/debian/bsdyacc
+	mv sh sh.hetio
+
+sh:
+	chmod u+x debian/bsdyacc
+	rm -f *.o
+	pmake CFLAGS='$(CDEF)' YACC=`pwd`/debian/bsdyacc
+
+sh.udeb:
+	chmod u+x debian/bsdyacc
+	rm -f *.o
+	pmake CFLAGS='$(CDEFSM)' YACC=`pwd`/debian/bsdyacc
+	mv sh sh.udeb
+
+build: build-stamp
+build-stamp: sh.hetio sh.udeb sh
+	dh_testdir
+
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp
+
+	pmake clean HETIO=
+	rm -f sh.cat1 mksignames signames.h sh.hetio sh.udeb
+
+	dh_clean
+
+# Build architecture-independent files here.
+binary-indep: build
+# We have nothing to do by default.
+
+# Build architecture-dependent files here.
+binary-arch: build
+#	dh_testversion
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+
+	install sh debian/tmp/bin/ash
+	install -m 644 sh.1 debian/tmp/usr/share/man/man1/ash.1
+	install sh.hetio debian/ash-medium/usr/bin/ash.medium
+	install -m 644 sh.1 debian/ash-medium/usr/share/man/man1/ash.medium.1
+	install sh.udeb debian/ash-udeb/bin/ash
+	ln -s ash debian/ash-udeb/bin/sh
+
+	dh_installdocs -Nash-udeb
+	dh_installexamples
+	dh_installmenu
+#	dh_installinit
+	dh_installcron
+#	dh_installmanpages
+#	dh_undocumented
+	dh_installchangelogs -Nash-udeb
+	dh_strip
+	dh_compress
+	dh_fixperms
+	dh_suidregister
+	dh_installdeb -Nash-udeb
+	dh_shlibdeps
+	dh_gencontrol
+#	dh_makeshlibs
+	dh_md5sums
+	dh_builddeb
+	for i in ../ash-udeb_*.deb; do mv $$i $${i%deb}udeb; done
+	sed '/^[^ ]*\.udeb/d; s/^\(ash-udeb_[^ ]*\.\)deb/\1udeb/' \
+		debian/files > debian/files.new
+	mv debian/files.new debian/files
+
+source diff:                                                                  
+	@echo >&2 'source and diff are obsolete - use dpkg-source -b'; false
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary

