From fa35f9d066587d7b8d6304099ecf09f7148e1257 Mon Sep 17 00:00:00 2001
From: Qing Yun <kf@aosc.io>
Date: Wed, 11 Feb 2026 22:47:58 +0800
Subject: [PATCH 3/3] Arch Linux: fltk 1.3

---
 CMakeLists.txt | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a5981938..1f011d22 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -267,12 +267,12 @@ if(BUILD_VIEWER)
   set(FLTK_SKIP_OPENGL TRUE)
   set(FLTK_SKIP_FORMS TRUE)
   if(BUILD_VIEWER STREQUAL "AUTO")
-    find_package(FLTK)
+    find_package(FLTK1.3)
   else()
-    find_package(FLTK REQUIRED)
+    find_package(FLTK1.3 REQUIRED)
   endif()
 
-  if(NOT FLTK_FOUND)
+  if(NOT FLTK1.3_FOUND)
     message(WARNING "FLTK NOT found. TigerVNC viewer disabled.")
     set(BUILD_VIEWER 0)
   endif()
@@ -304,11 +304,6 @@ if(BUILD_VIEWER)
     set(CMAKE_REQUIRED_INCLUDES ${FLTK_INCLUDE_DIR})
     set(CMAKE_REQUIRED_LIBRARIES ${FLTK_LIBRARIES})
 
-    check_cxx_source_compiles("#include <FL/Fl.H>\n#if FL_MAJOR_VERSION != 1 || FL_MINOR_VERSION != 3\n#error Wrong FLTK version\n#endif\nint main(int, char**) { return 0; }" OK_FLTK_VERSION)
-    if(NOT OK_FLTK_VERSION)
-      message(FATAL_ERROR "Incompatible version of FLTK")
-    endif()
-
     set(CMAKE_REQUIRED_FLAGS)
     set(CMAKE_REQUIRED_INCLUDES)
     set(CMAKE_REQUIRED_LIBRARIES)
-- 
2.52.0

