diff -Naur cinnamon-desktop-3.2.0/schemas/org.cinnamon.desktop.background.gschema.xml.in.in cinnamon-desktop-3.2.0.aosc/schemas/org.cinnamon.desktop.background.gschema.xml.in.in
--- cinnamon-desktop-3.2.0/schemas/org.cinnamon.desktop.background.gschema.xml.in.in	2016-11-07 04:33:02.000000000 -0600
+++ cinnamon-desktop-3.2.0.aosc/schemas/org.cinnamon.desktop.background.gschema.xml.in.in	2016-11-09 13:04:11.082341218 -0600
@@ -10,7 +10,7 @@
       </_description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/themes/Adwaita/backgrounds/adwaita-timed.xml'</default>
+      <default>'file://@datadir@//backgrounds/campanula-alt/campanula-alt.xml'</default>
       <_summary>Picture URI</_summary>
       <_description>
         URI to use for the background image. Not that the backend only supports
