diff -Naur a/bcloud/auth.py b/bcloud/auth.py
--- a/bcloud/auth.py	2015-12-25 22:35:16.000000000 +0800
+++ b/bcloud/auth.py	2016-08-02 14:49:58.511394388 +0800
@@ -289,6 +289,7 @@
     '''
     url = const.PAN_REFERER
     req = net.urlopen(url, headers={'Cookie': cookie.header_output()})
+    cookie.load_list(req.headers.get_all('Set-Cookie'))
     if req:
         return parse_bdstoken(req.data.decode())
     else:
diff -Naur a/bcloud/pcs.py b/bcloud/pcs.py
--- a/bcloud/pcs.py	2015-12-25 22:35:16.000000000 +0800
+++ b/bcloud/pcs.py	2016-08-02 14:52:45.215316213 +0800
@@ -542,7 +542,7 @@
     ])
     req = net.urlopen(url, headers={
         'Content-type': const.CONTENT_FORM_UTF8,
-        'Cookie': cookie.sub_output('BAIDUID', 'BDUSS', 'PANWEB', 'cflag'),
+        'Cookie': cookie.sub_output('BAIDUID', 'BDUSS', 'PANWEB', 'cflag', 'SCRC', 'STOKEN'),
     })
     if req:
         content = req.data
@@ -717,7 +717,7 @@
     dlink = metas['info'][0]['dlink']
     url = '{0}&cflg={1}'.format(dlink, cookie.get('cflag').value)
     req = net.urlopen_without_redirect(url, headers={
-        'Cookie': cookie.sub_output('BAIDUID', 'BDUSS', 'cflag'),
+        'Cookie': cookie.sub_output('BAIDUID', 'BDUSS', 'cflag', 'SCRC', 'STOKEN'),
         'Accept': const.ACCEPT_HTML,
     })
     if not req:
@@ -799,7 +799,7 @@
         '&ondup=', ondup,
         '&dir=', encoder.encode_uri_component(dir_name),
         '&filename=', encoder.encode_uri_component(file_name),
-        '&', cookie.sub_output('BDUSS'),
+        '&', cookie.sub_output('BDUSS', 'SCRC', 'STOKEN'),
     ])
     with open(source_path, 'rb') as fh:
         data = fh.read()
@@ -830,7 +830,7 @@
         '&content-md5=', content_md5,
         '&slice-md5=', slice_md5,
         '&path=', encoder.encode_uri_component(path),
-        '&', cookie.sub_output('BDUSS'),
+        '&', cookie.sub_output('BDUSS', 'SCRC', 'STOKEN'),
         '&bdstoken=', tokens['bdstoken'],
     ])
     req = net.urlopen(url, headers={'Cookie': cookie.header_output()})
@@ -850,7 +850,7 @@
     url = ''.join([
         const.PCS_URL_C,
         'file?method=upload&type=tmpfile&app_id=250528',
-        '&', cookie.sub_output('BDUSS'),
+        '&', cookie.sub_output('BDUSS', 'SCRC', 'STOKEN'),
     ])
     fields = []
     files = [('file', ' ', data)]
@@ -872,7 +872,7 @@
         const.PCS_URL_C,
         'file?method=createsuperfile&app_id=250528',
         '&path=', encoder.encode_uri_component(path),
-        '&', cookie.sub_output('BDUSS'),
+        '&', cookie.sub_output('BDUSS', 'SCRC', 'STOKEN'),
     ])
     param = {'block_list': block_list}
     data = 'param=' + json.dumps(param)
@@ -907,7 +907,7 @@
         data = ('dlink=0&target=' +
                 encoder.encode_uri_component(json.dumps(filelist)))
     req = net.urlopen(url, headers={
-        'Cookie': cookie.sub_output('BDUSS'),
+        'Cookie': cookie.sub_output('BDUSS', 'SCRC', 'STOKEN'),
         'Content-type': const.CONTENT_FORM,
         }, data=data.encode())
     if req:
