--- a/Makefile.am	2014-08-24 23:44:07.000000000 +0800
+++ b/Makefile.am	2016-08-24 20:10:50.724216634 +0800
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = lib objects plug-ins shapes app bindings samples po sheets data doc tests installer
+SUBDIRS = lib objects plug-ins shapes app bindings samples po sheets data tests installer
 
 gnomedatadir = $(datadir)
 
