Description: Add description to POD name section
Author: Ardo van Rangelrooij <ardo@debian.org>
Forwarded: not-needed
Last-Update: 2014-04-10
diff --git a/Refs.pm b/Refs.pm
index 3e3f563..fbc9e71 100644
--- a/Refs.pm
+++ b/Refs.pm
@@ -2,11 +2,11 @@ package SGMLS::Refs;
 
 use Carp;
 
-$version = '$Id: Refs.pm,v 1.5 1995/12/03 21:28:36 david Exp $';
+$version = '$Id: Refs.pm,v 1.1 1999/10/26 19:30:45 ardo Exp $';
 
 =head1 NAME
 
-SGMLS::Refs
+SGMLS::Refs - Forward reference handling
 
 =head1 SYNOPSIS
 
