Description: Fix errors with POD
Author: Daniel Lintott <daniel@serverb.co.uk>
Forwarded: no
Last-Update: 2014-04-11
--- a/SGMLS.pm
+++ b/SGMLS.pm
@@ -114,6 +114,8 @@
 collected and stored automatically and invisibly to the user.  The
 following list contains all of the methods for the C<SGMLS> class:
 
+=over
+
 =item C<next_event()>: Return an C<SGMLS_Event> object containing the
 next major event from the SGML parse.
 
@@ -144,6 +146,7 @@
 =item C<ext()>: Return a reference to an associative array for
 user-defined extensions.
 
+=back
 
 =head2 The C<SGMLS_Event> class
 
@@ -151,6 +154,8 @@
 C<next_event> method in the C<SGMLS> class.  It uses the following
 methods:
 
+=over
+
 =item C<type()>: Return a string describing the type of event:
 "start_element", "end_element", "cdata", "sdata", "re", "pi",
 "entity", "start_subdoc", "end_subdoc", and "conforming".  See
@@ -187,6 +192,7 @@
 =item C<ext()>: Return a reference to an associative array for
 user-defined extensions.
 
+=back
 
 =head2 The C<SGMLS_Element> class
 
@@ -194,6 +200,8 @@
 information (such as the element's attributes).  It recognises the
 following methods:
 
+=over
+
 =item C<name()>: Return a string containing the name, or Generic
 Identifier, of the element, in upper case.
 
@@ -228,12 +236,15 @@
 =item C<ext()>: Return a reference to an associative array for
 user-defined extensions.
 
+=back
 
 =head2 The C<SGMLS_Attribute> class
 
 Each instance of an attribute for each C<SGMLS_Element> is an object
 belonging to this class, which recognises the following methods:
 
+=over
+
 =item C<name()>: Return a string containing the name of the current
 attribute, all in upper case.
 
@@ -266,12 +277,15 @@
 =item C<ext()>: Return a reference to an associative array available
 for user-defined extensions.
 
+=back
 
 =head2 The C<SGMLS_Notation> class
 
 All declared notations appear as objects belonging to this class,
 which recognises the following methods:
 
+=over
+
 =item C<name()>: Return a string containing the name of the notation.
 
 =item C<sysid()>: Return a string containing the system identifier of
@@ -283,12 +297,15 @@
 =item C<ext()>: Return a reference to an associative array available
 for user-defined extensions.
 
+=back
 
 =head2 The C<SGMLS_Entity> class
 
 All declared entities appear as objects belonging to this class, which
 recognises the following methods:
 
+=over
+
 =item C<name()>: Return a string containing the name of the entity, in
 mixed case.
 
@@ -329,6 +346,7 @@
 =item C<ext()>: Return a reference to an associative array for
 user-defined extensions.
 
+=back
 
 =head1 AUTHOR AND COPYRIGHT
 
