Description: Don't clean documentation
Author: Ardo van Rangelrooij <ardo@debian.org>
Bug-Debian: https://bugs.debian.org/240645
Forwarded: not-needed
Last-Update: 2014-04-10
diff --git a/Makefile b/Makefile
index db522e1..49092a7 100644
--- a/Makefile
+++ b/Makefile
@@ -94,5 +94,5 @@ ${DIST}: clean docs
 	mv /tmp/${DIST} SGMLSpm
 
 clean:
-	cd DOC; make clean
+	#cd DOC; make clean
 	rm -f *~ core *.tar *.tar.gz
