From aca93e3364b9b837eb98eeb62322cf19d9819147 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 16 Jun 2015 15:08:23 +0200
Subject: [PATCH] uefi-lib: remove bogus comment

---
 modules.d/99uefi-lib/module-setup.sh | 1 -
 1 file changed, 1 deletion(-)

diff --git a/modules.d/99uefi-lib/module-setup.sh b/modules.d/99uefi-lib/module-setup.sh
index 4a856ea..3dcff22 100755
--- a/modules.d/99uefi-lib/module-setup.sh
+++ b/modules.d/99uefi-lib/module-setup.sh
@@ -1,5 +1,4 @@
 #!/bin/bash
-# module-setup for img-lib
 
 # called by dracut
 check() {
