From 47ff68e78b3bbd01367948b5b9fe1cbaeb45f262 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 23 Jun 2015 15:21:37 +0200
Subject: [PATCH] dracut.spec: Recommends grubby

---
 dracut.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dracut.spec b/dracut.spec
index 58510db..7a1a9c2 100644
--- a/dracut.spec
+++ b/dracut.spec
@@ -113,6 +113,10 @@ Requires: util-linux-ng >= 2.21
 
 Conflicts: mdadm < 3.2.6-14
 
+%if 0%{?fedora} > 22
+Recommends: grubby
+%endif
+
 %description
 dracut contains tools to create a bootable initramfs for 2.6 Linux kernels.
 Unlike existing implementations, dracut does hard-code as little as possible
