From c42064a684f3413dfce249c5bd6ebe0cb0a16908 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 23 Jun 2015 15:29:00 +0200
Subject: [PATCH] dracut.spec: turn requires in recommends

---
 dracut.spec | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/dracut.spec b/dracut.spec
index 7a1a9c2..4775210 100644
--- a/dracut.spec
+++ b/dracut.spec
@@ -92,11 +92,21 @@ Requires: cpio
 Requires: filesystem >= 2.1.0
 Requires: findutils
 Requires: grep
-Requires: hardlink
-Requires: gzip xz
 Requires: kmod
 Requires: sed
+
+%if 0%{?fedora} > 22
+Recommends: grubby
+Recommends: hardlink
+Recommends: gzip
+Recommends: xz
+Recommends: kpartx
+%else
+Requires: hardlink
+Requires: gzip
+Requires: xz
 Requires: kpartx
+%endif
 
 %if 0%{?fedora} || 0%{?rhel} > 6
 Requires: util-linux >= 2.21
@@ -113,10 +123,6 @@ Requires: util-linux-ng >= 2.21
 
 Conflicts: mdadm < 3.2.6-14
 
-%if 0%{?fedora} > 22
-Recommends: grubby
-%endif
-
 %description
 dracut contains tools to create a bootable initramfs for 2.6 Linux kernels.
 Unlike existing implementations, dracut does hard-code as little as possible
