From 3639fe543c06c79e8c7a4872844c95d7d9b61299 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 29 Jun 2015 16:14:58 +0200
Subject: [PATCH] lower the log level for dfatal

---
 dracut-logger.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dracut-logger.sh b/dracut-logger.sh
index 41e04f4..2a6019c 100755
--- a/dracut-logger.sh
+++ b/dracut-logger.sh
@@ -264,9 +264,9 @@ _lvl2syspri() {
 # Conversion is done as follows:
 #
 # <tt>
-#   FATAL(1) -> LOG_EMERG (0)
+#   none     -> LOG_EMERG (0)
 #   none     -> LOG_ALERT (1)
-#   none     -> LOG_CRIT (2)
+#   FATAL(1) -> LOG_CRIT (2)
 #   ERROR(2) -> LOG_ERR (3)
 #   WARN(3)  -> LOG_WARNING (4)
 #   none     -> LOG_NOTICE (5)
@@ -280,7 +280,7 @@ _dlvl2syslvl() {
     local lvl
 
     case "$1" in
-        1) lvl=0;;
+        1) lvl=2;;
         2) lvl=3;;
         3) lvl=4;;
         4) lvl=6;;
