From 8c7dc032f816d13d482bccebf2f7e56a81c207fd Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 13 Jul 2015 10:38:16 +0200
Subject: [PATCH] TODO: update

---
 TODO | 1 -
 1 file changed, 1 deletion(-)

diff --git a/TODO b/TODO
index 4a7d959..55fe13b 100644
--- a/TODO
+++ b/TODO
@@ -7,7 +7,6 @@ Items are ordered in priority.
 INITRAMFS TODO
 
 - search domain string
-- peerdns=no
 - allow dual stack configuration (IPv4, IPv6) for the same interface
 - "bind-mount" kernel drivers in real root for the rescue image,
   if the real root does not have any kernel modules for this kernel
