From cce34acace5876d06946dfb1fe70004a0b5176b0 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 28 Jul 2015 11:55:04 +0200
Subject: [PATCH] qemu: add spapr-vscsi kernel module

---
 modules.d/90qemu/module-setup.sh | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules.d/90qemu/module-setup.sh b/modules.d/90qemu/module-setup.sh
index 37972e4..66ef884 100755
--- a/modules.d/90qemu/module-setup.sh
+++ b/modules.d/90qemu/module-setup.sh
@@ -25,5 +25,6 @@ installkernel() {
         hostonly='' instmods \
             ata_piix ata_generic pata_acpi cdrom sr_mod ahci \
             virtio_blk virtio virtio_ring virtio_pci \
-            virtio_scsi virtio_console virtio_rng
+            virtio_scsi virtio_console virtio_rng \
+            spapr-vscsi
 }
