diff -Naur a/modules.d/50plymouth/module-setup.sh b/modules.d/50plymouth/module-setup.sh
--- a/modules.d/50plymouth/module-setup.sh	2015-06-15 18:27:21.000000000 +0800
+++ b/modules.d/50plymouth/module-setup.sh	2016-07-21 20:13:33.758877973 +0800
@@ -14,9 +14,6 @@
 # called by dracut
 install() {
     PKGLIBDIR="/usr/lib/plymouth"
-    if type -P dpkg-architecture &>/dev/null; then
-        PKGLIBDIR="/usr/lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH)/plymouth"
-    fi
     [ -x /usr/libexec/plymouth/plymouth-populate-initrd ] && PKGLIBDIR="/usr/libexec/plymouth"
 
     if grep -q nash ${PKGLIBDIR}/plymouth-populate-initrd \
