diff -Naur a/dracut.sh b/dracut.sh
--- a/dracut.sh	2016-09-18 08:25:51.647384475 -0500
+++ b/dracut.sh	2016-09-18 08:29:58.874772241 -0500
@@ -1638,7 +1638,7 @@
     # strip kernel modules, but do not touch signed modules
     find "$initdir" -type f -path '*/lib/modules/*.ko' -print0 \
         | while read -r -d $'\0' f || [ -n "$f" ]; do
-        SIG=$(tail -c 28 "$f")
+        SIG=$(tail -c 28 "$f" | tr -d '\000')
         [[ $SIG == '~Module signature appended~' ]] || { printf "%s\000" "$f"; }
     done | xargs -r -0 strip -g
 
diff -Naur a/modules.d/50drm/module-setup.sh b/modules.d/50drm/module-setup.sh
--- a/modules.d/50drm/module-setup.sh	2016-09-18 08:25:51.647384475 -0500
+++ b/modules.d/50drm/module-setup.sh	2016-09-18 08:29:58.874772241 -0500
@@ -24,9 +24,9 @@
             local _fname _fcont
             while read _fname || [ -n "$_fname" ]; do
                 case "$_fname" in
-                    *.ko)    _fcont="$(<        $_fname)" ;;
-                    *.ko.gz) _fcont="$(gzip -dc $_fname)" ;;
-                    *.ko.xz) _fcont="$(xz -dc   $_fname)" ;;
+                    *.ko)    _fcont="$(<        "$_fname" | tr -d '\000')" ;;
+                    *.ko.gz) _fcont="$(gzip -dc "$_fname" | tr -d '\000')" ;;
+                    *.ko.xz) _fcont="$(xz -dc   "$_fname" | tr -d '\000')" ;;
                 esac
                 [[   $_fcont =~ $_drm_drivers
                 && ! $_fcont =~ iw_handler_get_spy ]] \
diff -Naur a/modules.d/90kernel-modules/module-setup.sh b/modules.d/90kernel-modules/module-setup.sh
--- a/modules.d/90kernel-modules/module-setup.sh	2016-09-18 08:25:51.647384475 -0500
+++ b/modules.d/90kernel-modules/module-setup.sh	2016-09-18 08:29:58.874772241 -0500
@@ -10,9 +10,9 @@
             function bmf1() {
                 local _f
                 while read _f || [ -n "$_f" ]; do case "$_f" in
-                    *.ko)    [[ $(<         $_f) =~ $_blockfuncs ]] && echo "$_f" ;;
-                    *.ko.gz) [[ $(gzip -dc <$_f) =~ $_blockfuncs ]] && echo "$_f" ;;
-                    *.ko.xz) [[ $(xz -dc   <$_f) =~ $_blockfuncs ]] && echo "$_f" ;;
+                    *.ko)    [[ $(<         "$_f" | tr -d '\000') =~ $_blockfuncs ]] && echo "$_f" ;;
+                    *.ko.gz) [[ $(gzip -dc <"$_f" | tr -d '\000') =~ $_blockfuncs ]] && echo "$_f" ;;
+                    *.ko.xz) [[ $(xz -dc   <"$_f" | tr -d '\000') =~ $_blockfuncs ]] && echo "$_f" ;;
                     esac
                 done
                 return 0
diff -Naur a/modules.d/90kernel-network-modules/module-setup.sh b/modules.d/90kernel-network-modules/module-setup.sh
--- a/modules.d/90kernel-network-modules/module-setup.sh	2016-09-18 08:25:51.647384475 -0500
+++ b/modules.d/90kernel-network-modules/module-setup.sh	2016-09-18 08:29:26.008009706 -0500
@@ -26,9 +26,9 @@
             while read _fname; do
                 [[ $_fname =~ $_unwanted_drivers ]] && continue
                 case "$_fname" in
-                    *.ko)    _fcont="$(<        $_fname)" ;;
-                    *.ko.gz) _fcont="$(gzip -dc $_fname)" ;;
-                    *.ko.xz) _fcont="$(xz -dc   $_fname)" ;;
+                    *.ko)    _fcont="$(<        $_fname | tr -d '\000')" ;;
+                    *.ko.gz) _fcont="$(gzip -dc $_fname | tr -d '\000')" ;;
+                    *.ko.xz) _fcont="$(xz -dc   $_fname | tr -d '\000')" ;;
                 esac
                 [[   $_fcont =~ $_net_drivers
                 && ! $_fcont =~ iw_handler_get_spy ]] \
diff -Naur a/modules.d/90multipath/module-setup.sh b/modules.d/90multipath/module-setup.sh
--- a/modules.d/90multipath/module-setup.sh	2016-09-18 08:25:51.650717819 -0500
+++ b/modules.d/90multipath/module-setup.sh	2016-09-18 08:29:01.004603441 -0500
@@ -48,9 +48,9 @@
             local _f
             while read _f || [ -n "$_f" ]; do
                 case "$_f" in
-                    *.ko)    [[ $(<         $_f) =~ $_funcs ]] && echo "$_f" ;;
-                    *.ko.gz) [[ $(gzip -dc <$_f) =~ $_funcs ]] && echo "$_f" ;;
-                    *.ko.xz) [[ $(xz -dc   <$_f) =~ $_funcs ]] && echo "$_f" ;;
+                    *.ko)    [[ $(<         $_f | tr -d '\000') =~ $_funcs ]] && echo "$_f" ;;
+                    *.ko.gz) [[ $(gzip -dc <$_f | tr -d '\000') =~ $_funcs ]] && echo "$_f" ;;
+                    *.ko.xz) [[ $(xz -dc   <$_f | tr -d '\000') =~ $_funcs ]] && echo "$_f" ;;
                 esac
             done
             return 0
diff -Naur a/modules.d/95iscsi/module-setup.sh b/modules.d/95iscsi/module-setup.sh
--- a/modules.d/95iscsi/module-setup.sh	2016-09-18 08:25:51.650717819 -0500
+++ b/modules.d/95iscsi/module-setup.sh	2016-09-18 08:28:36.837866284 -0500
@@ -168,9 +168,9 @@
             local _f
             while read _f || [ -n "$_f" ]; do
                 case "$_f" in
-                    *.ko)    [[ $(<         $_f) =~ $_funcs ]] && echo "$_f" ;;
-                    *.ko.gz) [[ $(gzip -dc <$_f) =~ $_funcs ]] && echo "$_f" ;;
-                    *.ko.xz) [[ $(xz -dc   <$_f) =~ $_funcs ]] && echo "$_f" ;;
+                    *.ko)    [[ $(<         $_f | tr -d '\000') =~ $_funcs ]] && echo "$_f" ;;
+                    *.ko.gz) [[ $(gzip -dc <$_f | tr -d '\000') =~ $_funcs ]] && echo "$_f" ;;
+                    *.ko.xz) [[ $(xz -dc   <$_f | tr -d '\000') =~ $_funcs ]] && echo "$_f" ;;
                 esac
             done
             return 0
