--- a/configure-ng.py	2020-03-27 21:40:05.729683312 +0000
+++ b/configure-ng.py	2020-03-27 21:40:23.232569724 +0000
@@ -2219,10 +2219,6 @@
     # Build the SIP command line.
     argv = [quote(target_config.sip), '-w', '-n', 'PyQt4.sip', '-f', sip_flags]
 
-    # Make sure any unknown Qt version gets treated as the latest Qt v4.
-    argv.append('-B')
-    argv.append('Qt_5_0_0')
-
     if no_timestamp:
         argv.append('-T')
 
