From fcf341da9141e334945362396f31ebc72429eab7 Mon Sep 17 00:00:00 2001
From: Runhua He <hua@aosc.io>
Date: Thu, 18 Jul 2024 16:51:57 +0800
Subject: [PATCH] Cargo.toml: enable debug info for release profile

---
 bcrypt_src/src/_bcrypt/Cargo.toml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bcrypt_src/src/_bcrypt/Cargo.toml b/bcrypt_src/src/_bcrypt/Cargo.toml
index 58f6c13..df99d78 100644
--- a/src/_bcrypt/Cargo.toml
+++ b/src/_bcrypt/Cargo.toml
@@ -24,3 +24,5 @@ crate-type = ["cdylib"]
 [profile.release]
 lto = "thin"
 overflow-checks = true
+debug = 1
+strip = false
-- 
2.45.2

