diff -Naur tachyon/unix/Make-arch tachyon.nonx86/unix/Make-arch
--- tachyon/unix/Make-arch	2010-01-18 13:13:41.000000000 -0600
+++ tachyon.nonx86/unix/Make-arch	2020-10-24 23:42:01.167408283 -0500
@@ -1044,7 +1044,7 @@
 	$(MAKE) all \
 	"ARCH = linux-64" \
 	"CC = gcc" \
-	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux -DLP64 $(MISCFLAGS)" \
+	"CFLAGS = -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux -DLP64 $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = strip" \
@@ -1056,7 +1056,7 @@
 	$(MAKE) all \
 	"ARCH = linux-64-debug" \
 	"CC = gcc" \
-	"CFLAGS = -m64 -Wall -O -g -DLinux -DLP64 $(MISCFLAGS)" \
+	"CFLAGS = -Wall -O -g -DLinux -DLP64 $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = touch" \
@@ -1069,7 +1069,7 @@
 	$(MAKE) all \
 	"ARCH = linux-64-thr" \
 	"CC = gcc" \
-	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux -DLP64 -DTHR -D_REENTRANT $(MISCFLAGS)" \
+	"CFLAGS = -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux -DLP64 -DTHR -D_REENTRANT $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = strip" \
@@ -1227,7 +1227,7 @@
 	$(MAKE) all \
 	"ARCH = linux-lam-64" \
 	"CC = hcc" \
-	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -I$(LAMHOME)/h -DLinux -DMPI -DLP64 $(MISCFLAGS)" \
+	"CFLAGS = -Wall -O3 -fomit-frame-pointer -ffast-math -I$(LAMHOME)/h -DLinux -DMPI -DLP64 $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = strip" \
@@ -1274,7 +1274,7 @@
 	$(MAKE) all \
 	"ARCH = linux-lam-64-ogl" \
 	"CC = hcc" \
-	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -I$(LAMHOME)/h -DLinux -DMPI -DLP64 -DUSEOPENGL $(LINUX_GLX_INCS) $(MISCFLAGS)" \
+	"CFLAGS = -Wall -O3 -fomit-frame-pointer -ffast-math -I$(LAMHOME)/h -DLinux -DMPI -DLP64 -DUSEOPENGL $(LINUX_GLX_INCS) $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = strip" \
