diff --git a/src/term.c b/src/term.c
index 876e15d95..fefcccbd2 100644
--- a/src/term.c
+++ b/src/term.c
@@ -5317,6 +5317,13 @@ handle_version_response(int first, int *arg, int argc, char_u *tp)
 							// sequences
 	}
 
+	// KMSCON sends 1;1;0 and it cannot handle cursor style and blink
+	if (arg[0] == 1 && arg[1] == 1 && arg[2] == 0)
+	{
+	    term_props[TPR_CURSOR_STYLE].tpr_status = TPR_NO;
+	    term_props[TPR_CURSOR_BLINK].tpr_status = TPR_NO;
+	}
+
 	// Xterm first responded to this request at patch level
 	// 95, so assume anything below 95 is not xterm and hopefully supports
 	// the underline RGB color sequence.
