From 893f643a47087e2ed4bde09dbb205e2205625a5b Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Sat, 31 Oct 2020 18:59:34 +0800
Subject: [PATCH] Fix for kmscon

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 src/term.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/term.c b/src/term.c
index 7e39b037c..ad1d02c36 100644
--- a/src/term.c
+++ b/src/term.c
@@ -5198,6 +5198,13 @@ handle_version_response(int first, int *arg, int argc, char_u *tp)
 	    term_props[TPR_CURSOR_BLINK].tpr_status = TPR_NO;
 	}
 
+	// KMSCON sends 1;1;0 and it cannot handle cursor style and blink
+	if (arg[0] == 1 && arg[1] == 1 && arg[2] == 0)
+	{
+	    term_props[TPR_CURSOR_STYLE].tpr_status = TPR_NO;
+	    term_props[TPR_CURSOR_BLINK].tpr_status = TPR_NO;
+	}
+
 	// Xterm first responded to this request at patch level
 	// 95, so assume anything below 95 is not xterm and hopefully supports
 	// the underline RGB color sequence.
-- 
2.46.0

