diff -ur vim-8.2.1725-a/src/auto/configure vim-8.2.1725-b/src/auto/configure
--- vim-8.2.1725-a/src/auto/configure	2020-09-23 01:15:31.000000000 +0800
+++ vim-8.2.1725-b/src/auto/configure	2020-11-02 16:16:45.122859421 +0800
@@ -6136,9 +6136,9 @@
 		-e 's/-D_FORTIFY_SOURCE=.//g'`
             perllibs=`cd $srcdir; $vi_cv_path_perl -MExtUtils::Embed -e 'ldopts' | \
 		sed -e '/Warning/d' -e '/Note (probably harmless)/d' \
-			-e 's/-bE:perl.exp//' -e 's/-lc //'`
+			-e 's/-bE:perl.exp//' -e 's/-lc //' -e 's/-fno-lto//'`
                   perlldflags=`cd $srcdir; $vi_cv_path_perl -MExtUtils::Embed \
-		-e 'ccdlflags' | sed -e 's/-bE:perl.exp//'`
+		-e 'ccdlflags' | sed -e 's/-bE:perl.exp//' | sed -e 's/-fno-lto//'`
 
                   { $as_echo "$as_me:${as_lineno-$LINENO}: checking if compile and link flags for Perl are sane" >&5
 $as_echo_n "checking if compile and link flags for Perl are sane... " >&6; }
diff -ur vim-8.2.1725-a/src/configure.ac vim-8.2.1725-b/src/configure.ac
--- vim-8.2.1725-a/src/configure.ac	2020-09-23 01:15:31.000000000 +0800
+++ vim-8.2.1725-b/src/configure.ac	2020-11-02 13:50:44.110086408 +0800
@@ -1090,7 +1090,7 @@
       dnl Don't add perl lib to $LIBS: if it's not in LD_LIBRARY_PATH
       dnl a test in configure may fail because of that.
       perlldflags=`cd $srcdir; $vi_cv_path_perl -MExtUtils::Embed \
-		-e 'ccdlflags' | sed -e 's/-bE:perl.exp//'`
+		-e 'ccdlflags' | sed -e 's/-bE:perl.exp//' | sed -e 's/-fno-lto//'`
 
       dnl check that compiling a simple program still works with the flags
       dnl added for Perl.
