Index: zinnia-0.06/feature.cpp
===================================================================
--- zinnia-0.06.orig/feature.cpp
+++ zinnia-0.06/feature.cpp
@@ -42,7 +42,7 @@ float minimum_distance(const Node *first
 
   const float a = last->x - first->x;
   const float b = last->y - first->y;
-  const float c = last->y * first->x - last->x * first->y;
+  const float c = (double)last->y * first->x - (double)last->x * first->y;
 
   float max = -1.0;
   for (const Node *n = first; n != last; ++n) {
