diff -p -up zinnia-0.05/perl/Makefile.PL.bindings zinnia-0.05/perl/Makefile.PL
--- zinnia-0.05/perl/Makefile.PL.bindings	2008-09-13 18:59:36.000000000 +0200
+++ zinnia-0.05/perl/Makefile.PL	2009-11-19 14:17:39.000000000 +0100
@@ -3,8 +3,8 @@ WriteMakefile(
     'NAME'		=> 'zinnia',
     'CC'                => 'c++',
     'LD'                => 'c++',
-    'INC'               => '',
-    'LIBS'              => '-lzinnia',
+    'INC'               => '-I../',
+    'LIBS'              => '-L../.libs -lzinnia',
 #    'VERSION'	        => '0.1',
     'OBJECT'            => 'zinnia_wrap.o' 
 );
diff -p -up zinnia-0.05/perl/zinnia_wrap.cxx.bindings zinnia-0.05/perl/zinnia_wrap.cxx
--- zinnia-0.05/perl/zinnia_wrap.cxx.bindings	2009-05-31 06:39:39.000000000 +0200
+++ zinnia-0.05/perl/zinnia_wrap.cxx	2009-11-19 14:18:46.000000000 +0100
@@ -11,6 +11,7 @@
 #define SWIGPERL
 #define SWIG_CASTRANK_MODE
 
+#include <cstring>
 
 #ifdef __cplusplus
 /* SwigValueWrapper is described in swig.swg */
diff -p -up zinnia-0.05/python/zinnia_wrap.cxx.bindings zinnia-0.05/python/zinnia_wrap.cxx
--- zinnia-0.05/python/zinnia_wrap.cxx.bindings	2009-05-31 06:39:41.000000000 +0200
+++ zinnia-0.05/python/zinnia_wrap.cxx	2009-11-19 14:19:10.000000000 +0100
@@ -11,6 +11,7 @@
 #define SWIGPYTHON
 #define SWIG_PYTHON_DIRECTOR_NO_VTABLE
 
+#include <cstring>
 
 #ifdef __cplusplus
 /* SwigValueWrapper is described in swig.swg */
