From 2ea9abac327a620b5763cd3130aa2d2c1bc1ae2d Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Sat, 18 Oct 2025 10:09:16 +0800
Subject: [PATCH 03/10] loongvpu: fix error handling for vgem creation

---
 hantro/hantro_drm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hantro/hantro_drm.c b/hantro/hantro_drm.c
index 37de501..f6e89a9 100644
--- a/hantro/hantro_drm.c
+++ b/hantro/hantro_drm.c
@@ -154,6 +154,7 @@ static int hantro_gem_dumb_create_internal(struct drm_file *file_priv,
 	if (ret) {
 		FreeMem(cma_obj);
 		kfree(cma_obj);
+		goto out;
 	}
 	init_hantro_resv(&cma_obj->kresv, cma_obj);
 	cma_obj->handle = args->handle;
-- 
2.55.0

