From 54bdabab880cd71fd4620f01fb83ebbca23cf644 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Mon, 13 Oct 2025 11:26:11 +0800
Subject: [PATCH 05/13] loongvpu: remove driver date field for Linux 6.14+

---
 hantro/hantro_drm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/hantro/hantro_drm.c b/hantro/hantro_drm.c
index 96f9a28..50f2aae 100644
--- a/hantro/hantro_drm.c
+++ b/hantro/hantro_drm.c
@@ -1878,7 +1878,9 @@ struct drm_driver hantro_drm_driver = {
 	.fops = &hantro_fops,
 	.name = DRIVER_NAME,
 	.desc = DRIVER_DESC,
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 14, 0))
 	.date = DRIVER_DATE,
+#endif
 	.major = DRIVER_MAJOR,
 	.minor = DRIVER_MINOR,
 };
-- 
2.52.0

