From 666995e2267163439835f6838018d799aa3b2501 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 30 Dec 2025 22:09:04 +0800
Subject: [PATCH 07/10] fix(dkms.conf): drop deprecated DKMS options

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 dkms.conf | 2 --
 1 file changed, 2 deletions(-)

diff --git a/dkms.conf b/dkms.conf
index 384772e..7898d08 100644
--- a/dkms.conf
+++ b/dkms.conf
@@ -8,8 +8,6 @@ BUILT_MODULE_NAME[0]="ls2k3000_vpu"
 DEST_MODULE_NAME[0]="$PACKAGE_NAME"
 DEST_MODULE_LOCATION[0]="/updates/dkms"
 AUTOINSTALL=yes
-REMAKE_INITRD=yes
 
 MAKE[0]="unset ARCH; env LG_VERBOSE=1 \
     make ${parallel_jobs+-j$parallel_jobs} modules KERNEL_UNAME=${kernelver}"
-CLEAN="make KERNEL_UNAME=${kernelver} clean"
-- 
2.55.0

