From f2485aeb0445dbb5b672e44c8a7b3b9c7a557364 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Fri, 6 Mar 2026 13:27:30 +0800
Subject: [PATCH 08/10] loonggpu: add cflag '-fms-extensions' to conftest

---
 conftest.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conftest.sh b/conftest.sh
index 2e92b0e..c499031 100644
--- a/conftest.sh
+++ b/conftest.sh
@@ -89,7 +89,7 @@ build_cflags() {
     BASE_CFLAGS="-O2 -D__KERNEL__ \
 -DKBUILD_BASENAME=\"#conftest$$\" -DKBUILD_MODNAME=\"#conftest$$\" \
 -nostdinc -isystem $ISYSTEM \
--Wno-implicit-function-declaration -Wno-strict-prototypes -std=gnu11"
+-Wno-implicit-function-declaration -Wno-strict-prototypes -std=gnu11 -fms-extensions"
 
     if [ "$OUTPUT" != "$SOURCES" ]; then
         OUTPUT_CFLAGS="-I$OUTPUT/include2 -I$OUTPUT/include"
-- 
2.55.0

