From 7ceb6e6ef07c551aebd35ec8567844ee2cb2ea2b Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Thu, 23 Jul 2026 18:58:59 +0800
Subject: [PATCH 10/10] loongvpu: bump version to 1.0.0

---
 dkms.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dkms.conf b/dkms.conf
index 7898d08..90c3fe3 100644
--- a/dkms.conf
+++ b/dkms.conf
@@ -2,7 +2,7 @@
 
 # The version is replaced at build time by dh_dkms invoked in debian/rules.
 PACKAGE_NAME="ls2k3000_vpu"
-PACKAGE_VERSION="0.1"
+PACKAGE_VERSION="1.0"
 
 BUILT_MODULE_NAME[0]="ls2k3000_vpu"
 DEST_MODULE_NAME[0]="$PACKAGE_NAME"
-- 
2.55.0

