From 49fdae268ca0a0607bbbd0a7a1acf14e5d894d89 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 1 Sep 2025 03:28:12 +0800
Subject: [PATCH 4/5] AOSCOS: avoid creating empty GIT_COMMIT macro

---
 Main/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Main/CMakeLists.txt b/Main/CMakeLists.txt
index 15d3783..b2defb5 100644
--- a/Main/CMakeLists.txt
+++ b/Main/CMakeLists.txt
@@ -62,7 +62,9 @@ target_compile_definitions(usc-game PRIVATE VERSION_MAJOR=${PROJECT_VERSION_MAJO
 target_compile_definitions(usc-game PRIVATE VERSION_PATCH=${PROJECT_VERSION_PATCH})
 find_package (Git)
 if(GIT_FOUND)
-    target_compile_definitions(usc-game PRIVATE GIT_COMMIT=${GIT_DATE_HASH})
+    if(GIT_DATE_HASH)
+      target_compile_definitions(usc-game PRIVATE GIT_COMMIT=${GIT_DATE_HASH})
+    endif()
     execute_process(
       COMMAND git rev-parse --abbrev-ref HEAD
       WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-- 
2.52.0

