From 8ac9f9bd6c11210b58ad735209dcaa70cb9eb7ea Mon Sep 17 00:00:00 2001
From: Ruikai Liu <rickliu2000@outlook.com>
Date: Thu, 16 Apr 2026 15:26:43 +0800
Subject: [PATCH] fix systemd service file install path

---
 src/services/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/services/Makefile b/src/services/Makefile
index 818c106..79c832f 100644
--- a/src/services/Makefile
+++ b/src/services/Makefile
@@ -1,4 +1,4 @@
-SERVICEDIR=$(DESTDIR)/lib/systemd/system
+SERVICEDIR=$(DESTDIR)/usr/lib/systemd/system
 
 all:
 	@true
-- 
2.52.0

