diff -Naur boinc-client_release-7.8-7.8.3/clientgui/AsyncRPC.cpp boinc-client_release-7.8-7.8.3.glibc226/clientgui/AsyncRPC.cpp
--- boinc-client_release-7.8-7.8.3/clientgui/AsyncRPC.cpp	2017-10-03 18:38:01.000000000 +0000
+++ boinc-client_release-7.8-7.8.3.glibc226/clientgui/AsyncRPC.cpp	2017-10-20 14:20:01.358760945 +0000
@@ -19,10 +19,6 @@
 #pragma implementation "AsyncRPC.h"
 #endif
 
-#if !(defined(_WIN32) || (defined(__WXMAC__) && (MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_4)))
-#include <xlocale.h>
-#endif
-
 #include "stdwx.h"
 #include "BOINCGUIApp.h"
 #include "MainDocument.h"
diff -Naur boinc-client_release-7.8-7.8.3/lib/gui_rpc_client.h boinc-client_release-7.8-7.8.3.glibc226/lib/gui_rpc_client.h
--- boinc-client_release-7.8-7.8.3/lib/gui_rpc_client.h	2017-10-03 18:38:01.000000000 +0000
+++ boinc-client_release-7.8-7.8.3.glibc226/lib/gui_rpc_client.h	2017-10-20 14:21:48.446117193 +0000
@@ -807,7 +807,6 @@
 
 #elif defined(__APPLE__) && (MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_4)
 // uselocale() is not available in OS 10.3.9 so use weak linking
-#include <xlocale.h>
 extern int		freelocale(locale_t) __attribute__((weak_import));
 extern locale_t	newlocale(int, __const char *, locale_t) __attribute__((weak_import));
 extern locale_t	uselocale(locale_t) __attribute__((weak_import));

