diff -Naur boinc-client_release-7.14-7.14.2/clientgui/AdvancedFrame.cpp boinc-client_release-7.14-7.14.2.noupdate/clientgui/AdvancedFrame.cpp
--- boinc-client_release-7.14-7.14.2/clientgui/AdvancedFrame.cpp	2018-10-11 08:43:27.000000000 +0800
+++ boinc-client_release-7.14-7.14.2.noupdate/clientgui/AdvancedFrame.cpp	2019-06-28 12:03:51.756564456 +0800
@@ -198,7 +198,6 @@
     EVT_MENU(ID_HELPBOINCMANAGER, CAdvancedFrame::OnHelpBOINC)
     EVT_MENU(ID_HELPBOINCWEBSITE, CAdvancedFrame::OnHelpBOINC)
     EVT_MENU(wxID_ABOUT, CAdvancedFrame::OnHelpAbout)
-    EVT_MENU(ID_CHECK_VERSION, CAdvancedFrame::OnCheckVersion)
     EVT_HELP(wxID_ANY, CAdvancedFrame::OnHelp)
     // Custom Events & Timers
     EVT_FRAME_CONNECT(CAdvancedFrame::OnConnect)
@@ -688,21 +687,6 @@
         strMenuDescription
     );
     menuHelp->AppendSeparator();
-
-    strMenuName.Printf(
-        _("Check for new %s version"),
-        pSkinAdvanced->GetApplicationShortName().c_str()
-    );
-    strMenuDescription.Printf(
-        _("Check for new %s version"),
-        pSkinAdvanced->GetApplicationShortName().c_str()
-    );
-    menuHelp->Append(
-        ID_CHECK_VERSION,
-        strMenuName,
-        strMenuDescription
-    );
-    menuHelp->AppendSeparator();
 
     strMenuName.Printf(
         _("&About %s..."), 
diff -Naur boinc-client_release-7.14-7.14.2/clientgui/Events.h boinc-client_release-7.14-7.14.2.noupdate/clientgui/Events.h
--- boinc-client_release-7.14-7.14.2/clientgui/Events.h	2018-10-11 08:43:27.000000000 +0800
+++ boinc-client_release-7.14-7.14.2.noupdate/clientgui/Events.h	2019-06-28 12:03:36.116564753 +0800
@@ -95,7 +95,6 @@
 #define ID_HELPBOINC                            6035  // Locked: Used by manager_links.php
 #define ID_HELPBOINCWEBSITE                     6024  // Locked: Used by manager_links.php
 #define ID_HELPBOINCMANAGER                     6025  // Locked: Used by manager_links.php
-#define ID_CHECK_VERSION                        6026
 //#define wxID_ABOUT
 
 // Views
diff -Naur boinc-client_release-7.14-7.14.2/clientgui/sg_BoincSimpleFrame.cpp boinc-client_release-7.14-7.14.2.noupdate/clientgui/sg_BoincSimpleFrame.cpp
--- boinc-client_release-7.14-7.14.2/clientgui/sg_BoincSimpleFrame.cpp	2018-10-11 08:43:27.000000000 +0800
+++ boinc-client_release-7.14-7.14.2.noupdate/clientgui/sg_BoincSimpleFrame.cpp	2019-06-28 12:03:22.423231680 +0800
@@ -85,8 +85,6 @@
     EVT_MENU(ID_HELPBOINCMANAGER, CSimpleFrame::OnHelpBOINC)
     EVT_MENU(ID_HELPBOINCWEBSITE, CSimpleFrame::OnHelpBOINC)
     EVT_MENU(wxID_ABOUT, CSimpleFrame::OnHelpAbout)
-    EVT_MENU(ID_CHECK_VERSION, CSimpleFrame::OnCheckVersion)
-	EVT_MENU(ID_EVENTLOG, CSimpleFrame::OnEventLog)
     EVT_MOVE(CSimpleFrame::OnMove)
 #ifdef __WXMAC__
 	EVT_MENU(wxID_PREFERENCES, CSimpleFrame::OnPreferences)
@@ -259,21 +257,6 @@
         strMenuDescription
     );
     menuHelp->AppendSeparator();
-
-    strMenuName.Printf(
-        _("Check for new %s version"),
-        pSkinAdvanced->GetApplicationShortName().c_str()
-    );
-    strMenuDescription.Printf(
-        _("Check for new %s version"),
-        pSkinAdvanced->GetApplicationShortName().c_str()
-    );
-    menuHelp->Append(
-        ID_CHECK_VERSION,
-        strMenuName,
-        strMenuDescription
-    );
-    menuHelp->AppendSeparator();
 
     strMenuName.Printf(
         _("&About %s..."), 
