diff -Naur a/meson.build b/meson.build
--- a/meson.build	2016-08-09 00:01:45.000000000 +0800
+++ b/meson.build	2016-08-15 11:44:19.084340213 +0800
@@ -5,7 +5,7 @@
 itstool = find_program('itstool')
 msgfmt = find_program('msgfmt')
 
-python_dep = dependency('python3', version : '>= 3.3')
+python_dep = dependency('python-3.4', version : '>= 3.3')
 gst_dep = dependency('gstreamer-1.0', version : '>= 1.8.2')
 cairo_dep = dependency('cairo')
 pycairo_dep = dependency('py3cairo')
