--- pitivi-0.98.1/meson.build	2017-06-30 16:20:30.000000000 -0500
+++ pitivi-0.98.1.newmeson/meson.build	2017-08-05 04:43:34.292361252 -0500
@@ -78,9 +78,9 @@
 if archiver.found()
     prefix = 'pitivi-' + meson.project_version()
     archive = 'pitivi-' + meson.project_version() + '.tar.gz'
-    run_target('dist', command : ['git-archive-all', '--prefix=' + prefix, archive])
+    run_target('pitivi-dist', command : ['git-archive-all', '--prefix=' + prefix, archive])
 else
-    message('git-archive-all not found, you will not be able to run `ninja dist`')
+    message('git-archive-all not found, you will not be able to run `ninja pitivi-dist`')
     message('Install git-archive-all:')
     message('  - with pip:')
     message('    $ pip3 install git-archive-all')
--- pitivi-0.98.1/configure	2017-06-29 15:58:12.000000000 -0500
+++ pitivi-0.98.1.newmeson/configure	2017-08-05 04:44:21.135695212 -0500
@@ -22,7 +22,7 @@
 \tcd %(build_dir)s && %(ninja)s validate
 
 dist:
-\tcd %(build_dir)s && %(ninja)s dist
+\tcd %(build_dir)s && %(ninja)s pitivi-dist
 
 clean:
 \trm -Rf %(build_dir)s
