--- pitivi-0.999/meson.build~	2018-08-29 09:05:21.000000000 -0500
+++ pitivi-0.999/meson.build	2019-06-11 10:04:02.636744751 -0500
@@ -61,7 +61,8 @@
     dependencies: [gst_dep, python_dep, pycairo_dep, cairo_dep],
     name_prefix : '',
     install: true,
-    install_dir: get_option('libdir') + '/pitivi/python/pitivi/timeline')
+    install_dir: get_option('libdir') + '/pitivi/python/pitivi/timeline',
+    link_args : '-lpython3.8')
 
 pitivi_install_dir = get_option('libdir') + '/pitivi/python/'
 
