--- a/pitivi/utils/loggable.py~	2021-05-30 04:37:52.000000000 -0500
+++ b/pitivi/utils/loggable.py	2021-08-19 08:31:07.601086922 -0500
@@ -14,7 +14,7 @@
 #
 # You should have received a copy of the GNU Lesser General Public
 # License along with this program; if not, see <http://www.gnu.org/licenses/>.
-import collections
+import collections.abc as collections
 import errno
 import fnmatch
 import os
