diff -up lensfun-0.3.1/docs/CMakeLists.txt.INSTALL_HELPER_SCRIPTS lensfun-0.3.1/docs/CMakeLists.txt
--- lensfun-0.3.1/docs/CMakeLists.txt.INSTALL_HELPER_SCRIPTS	2015-05-10 14:04:53.000000000 +0200
+++ lensfun-0.3.1/docs/CMakeLists.txt	2015-05-13 15:28:13.569986646 +0200
@@ -22,6 +22,7 @@ ADD_CUSTOM_TARGET(doc
 # install documentation  
 INSTALL(DIRECTORY ${CMAKE_DOC_OUT}/ DESTINATION ${DOCDIR})
 
+IF(INSTALL_HELPER_SCRIPTS)
 # create and install man pages
 FIND_PROGRAM(RST2MAN_EXECUTABLE NAMES rst2man rst2man.py rst2man2 rst2man2.py)
 ADD_CUSTOM_TARGET(man ALL)
@@ -39,3 +40,4 @@ INSTALL(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/lensfun-add-adapter.1 
     DESTINATION ${CMAKE_INSTALL_PREFIX}/man/man1
 )
+ENDIF(INSTALL_HELPER_SCRIPTS)

