diff -up lensfun-0.3.1/CMakeLists.txt.mandir lensfun-0.3.1/CMakeLists.txt
--- lensfun-0.3.1/CMakeLists.txt.mandir	2015-05-13 15:30:51.240533510 +0200
+++ lensfun-0.3.1/CMakeLists.txt	2015-05-13 15:33:37.381948578 +0200
@@ -48,6 +48,7 @@ SET(BINDIR bin CACHE PATH "Main binary d
 SET(LIBDIR lib${LIB_SUFFIX} CACHE PATH "Directory for library itself")
 SET(INCLUDEDIR include CACHE PATH "Directory for library header files")
 SET(DATADIR share/lensfun CACHE PATH "Directory for library data files")
+SET(MANDIR share/man CACHE PATH "Root directory for man page files")
 
 IF(BUILD_FOR_SSE)
   SET(VECTORIZATION_SSE 1)
diff -up lensfun-0.3.1/docs/CMakeLists.txt.mandir lensfun-0.3.1/docs/CMakeLists.txt
--- lensfun-0.3.1/docs/CMakeLists.txt.mandir	2015-05-13 15:30:51.238533541 +0200
+++ lensfun-0.3.1/docs/CMakeLists.txt	2015-05-13 15:30:51.240533510 +0200
@@ -38,6 +38,6 @@ INSTALL(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/g-lensfun-update-data.1 
     ${CMAKE_CURRENT_BINARY_DIR}/lensfun-update-data.1 
     ${CMAKE_CURRENT_BINARY_DIR}/lensfun-add-adapter.1 
-    DESTINATION ${CMAKE_INSTALL_PREFIX}/man/man1
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/${MANDIR}/man1
 )
 ENDIF(INSTALL_HELPER_SCRIPTS)

