--- lensfun-0.3.1/apps/lensfun-update-data.orig	2015-07-14 11:33:08.643910978 +0200
+++ lensfun-0.3.1/apps/lensfun-update-data	2015-07-14 11:47:31.511063269 +0200
@@ -45,7 +45,7 @@
 import urllib.request, shutil, sys, os, getpass, time, calendar, tarfile, json
 
 lensfun_updates_dir = "/var/lib/lensfun-updates"
-if sys.platform != "win32" and getpass.getuser() != "root":
+if sys.platform != "win32" and os.geteuid() != 0:
     lensfun_updates_dir = os.path.join(os.path.expanduser("~"),
                                        ".local", "share", "lensfun", "updates")
     print("info: root privileges needed for updating the system database.")

