From 0abafe5841ae150827febc8bdda052ce3bfb72dd Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Mon, 29 Jun 2026 15:10:48 +0800
Subject: [PATCH] AOSCOS: spa/audiomixer: fix test_inc in meson.build

---
 spa/plugins/audiomixer/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/spa/plugins/audiomixer/meson.build b/spa/plugins/audiomixer/meson.build
index fa44c625c..b9b88f956 100644
--- a/spa/plugins/audiomixer/meson.build
+++ b/spa/plugins/audiomixer/meson.build
@@ -66,6 +66,7 @@ spa_audiomixer_lib = shared_library('spa-audiomixer',
 )
 spa_audiomixer_dep = declare_dependency(link_with: spa_audiomixer_lib)
 
+test_inc = include_directories('../test')
 test_apps = [
   'test-mix-ops',
   ]
-- 
2.52.0

