From 5e6c21cf9aaad08115c8025ae18ec5644b3d10f3 Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Sun, 28 Sep 2025 02:50:07 +0800
Subject: [PATCH] allow-make-install-without-build

---
 Makefile | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index e788df4..a3c932a 100644
--- a/Makefile
+++ b/Makefile
@@ -28,13 +28,12 @@ sbctl:
 	go build -ldflags="-X github.com/foxboron/sbctl.Version=$(VERSION)" -o $@ ./cmd/$@
 
 .PHONY: completions
-completions: sbctl
-	./sbctl completion bash | install -D /dev/stdin contrib/completions/bash-completion/completions/sbctl
-	./sbctl completion zsh | install -D /dev/stdin contrib/completions/zsh/site-functions/_sbctl
-	./sbctl completion fish | install -D /dev/stdin contrib/completions/fish/vendor_completions.d/sbctl.fish
+completions:
+	$(PKGDIR)/usr/bin/sbctl completion bash | install -Dv /dev/stdin contrib/completions/bash-completion/completions/sbctl
+	$(PKGDIR)/usr/bin/sbctl completion zsh | install -Dv /dev/stdin contrib/completions/zsh/site-functions/_sbctl
+	$(PKGDIR)/usr/bin/sbctl completion fish | install -Dv /dev/stdin contrib/completions/fish/vendor_completions.d/sbctl.fish
 
-install: sbctl completions man
-	install -Dm755 sbctl -t '$(DESTDIR)$(BINDIR)'
+install: completions man
 	for manfile in $(MANS); do \
 		install -Dm644 "$$manfile" -t '$(DESTDIR)$(MANDIR)/man'"$${manfile##*.}"; \
 	done;
-- 
2.51.0

