From 6d0770e78c7f83dc2fe7b29498d791be8c1e4fe2 Mon Sep 17 00:00:00 2001
From: Lunzima <lunzima@lunzima.net>
Date: Sun, 27 Apr 2025 22:05:33 +0800
Subject: [PATCH] feat: modify 00-auto.plan behavior; on battery, use balanced
 plan

---
 res/plans/00-auto.plan | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/res/plans/00-auto.plan b/res/plans/00-auto.plan
index 1e8f3b3..97e5126 100644
--- a/res/plans/00-auto.plan
+++ b/res/plans/00-auto.plan
@@ -16,7 +16,7 @@
 #   with this program; if not, write to the Free Software Foundation, Inc.,
 #   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-# On battery, executes powersave plan
 # On AC executes balanced plan
+# On battery, executes balanced plan
 PLAN_AUTO_AC=balanced
-PLAN_AUTO_BAT=powersave
+PLAN_AUTO_BAT=balanced
