diff -Naur sonnet-5.28.0/cmake/FindHUNSPELL.cmake sonnet-5.28.0.hunspell1.5/cmake/FindHUNSPELL.cmake
--- sonnet-5.28.0/cmake/FindHUNSPELL.cmake	2016-11-05 10:02:30.000000000 -0500
+++ sonnet-5.28.0.hunspell1.5/cmake/FindHUNSPELL.cmake	2016-11-28 21:53:35.060729203 -0600
@@ -14,7 +14,7 @@
 
 FIND_PATH(HUNSPELL_INCLUDE_DIR hunspell/hunspell.hxx )
 
-FIND_LIBRARY(HUNSPELL_LIBRARIES NAMES hunspell-1.4 hunspell-1.3 hunspell-1.2)
+FIND_LIBRARY(HUNSPELL_LIBRARIES NAMES hunspell-1.5 hunspell-1.4 hunspell-1.3 hunspell-1.2)
 
 # handle the QUIETLY and REQUIRED arguments and set HUNSPELL_FOUND to TRUE if 
 # all listed variables are TRUE
