From 2bdeccd26af6935e994747b92bc0e5270f9ae25b Mon Sep 17 00:00:00 2001
From: RawDiamondMC <RawDiamondMC@aosc.io>
Date: Sun, 5 Apr 2026 19:57:00 +0800
Subject: [PATCH] AOSCOS: generate C bindings

Signed-off-by: RawDiamondMC <RawDiamondMC@aosc.io>
---
 tree-sitter.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tree-sitter.json b/tree-sitter.json
index 7da9147..c68dc98 100644
--- a/tree-sitter.json
+++ b/tree-sitter.json
@@ -27,7 +27,7 @@
     }
   },
   "bindings": {
-    "c": false,
+    "c": true,
     "go": false,
     "java": false,
     "node": false,
-- 
2.52.0

