From c10d8cadbf00b308c907a989b2b93333a25ec625 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Mon, 3 Aug 2020 14:31:00 -0400
Subject: [PATCH] makefile: do not strip man page

---
 makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile b/makefile
index edf93aa..6e863a9 100644
--- a/makefile
+++ b/makefile
@@ -96,7 +96,7 @@ install: newflasher newflasher.1.gz
 	$(INSTALL) -o root -g root -d $(DESTDIR)/usr/bin
 	$(INSTALL) -o root -g root -m 755 -s newflasher $(DESTDIR)/usr/bin/
 	$(INSTALL) -o root -g root -d $(DESTDIR)/usr/share/man/man1
-	$(INSTALL) -o root -g root -m 644 -s newflasher.1.gz $(DESTDIR)/usr/share/man/man1
+	$(INSTALL) -o root -g root -m 644 newflasher.1.gz $(DESTDIR)/usr/share/man/man1
 
 .PHONY: clean
 clean:
-- 
2.27.0

