diff -up hplip-3.17.11/Makefile.am.udev-rules hplip-3.17.11/Makefile.am
--- hplip-3.17.11/Makefile.am.udev-rules	2017-12-07 13:44:04.472414453 +0100
+++ hplip-3.17.11/Makefile.am	2017-12-07 15:20:18.544755557 +0100
@@ -132,7 +132,7 @@ dist_rulessystem_DATA =data/rules/hplip-
 
 # hpmud.rules
 if !HPLIP_CLASS_DRIVER
-rulesdir = /etc/udev/rules.d
+rulesdir = /usr/lib/udev/rules.d
 if UDEV_SYSFS_RULES
 dist_rules_DATA = data/rules/56-hpmud_sysfs.rules
 else
