commit 4c913b68d8f8a79e11e247a8f7086f779f2685d4
Author: Tomas Korbar <tkorbar@redhat.com>
Date:   Tue Sep 18 16:17:06 2018 +0200

    Fix TypeError
    
    - 1564770 - TypeError: 'Error' object does not support indexing

diff --git a/clean.py b/clean.py
index 3d41f04..3a7248e 100755
--- a/clean.py
+++ b/clean.py
@@ -182,7 +182,7 @@ try:
                         log.error("Cleaning not needed or supported on this device.")
 
                 except Error as e:
-                    log.error("An error occured: %s" % e[0])
+                    log.error("An error occured: %s" % e.msg)
 
             else:
                 log.error("Device is busy or in an error state. Please check device and try again.")
diff --git a/ui/scrollunload.py b/ui/scrollunload.py
index fdb48e8..345ed82 100644
--- a/ui/scrollunload.py
+++ b/ui/scrollunload.py
@@ -120,7 +120,7 @@ class ScrollUnloadView(ScrollView):
                 self.pc = photocard.PhotoCard(None, self.cur_device.device_uri, self.cur_printer)
             except Error as e:
                 QApplication.restoreOverrideCursor()
-                self.form.FailureUI(self.__tr("An error occured: %s" % e[0]))
+                self.form.FailureUI(self.__tr("An error occured: %s" % e.msg))
                 self.cleanup(EVENT_PCARD_UNABLE_TO_MOUNT)
                 return False
 
