diff --git a/fax/backend/hpfax.py b/fax/backend/hpfax.py
index a63051d..3e187c9 100755
--- a/fax/backend/hpfax.py
+++ b/fax/backend/hpfax.py
@@ -89,7 +89,7 @@ try:
     from base import utils
     #from prnt import cups
 except ImportError as e:
-    bug("Error importing HPLIP modules: %s\n" % (pid, e))
+    bug("Error importing HPLIP modules: %s\n" % (e))
     sys.exit(1)
 
 def handle_sigpipe():
