diff -up hplip-3.23.3/ui5/devmgr5.py.qmsgbox-typos-fix hplip-3.23.3/ui5/devmgr5.py
--- hplip-3.23.3/ui5/devmgr5.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui5/devmgr5.py	2023-05-29 13:48:48.429731297 +0200
@@ -144,8 +144,7 @@ class PluginInstall(QObject):
             install_plugin = QMessageBox.warning(self.parent,
                                 self.parent.windowTitle(),
                                 self.__tr("<b>The HPLIP plugin is already installed.</b><p>Do you want to continue and re-install it?"),
-                                QMessageBox.Yes,
-                                QMessageBox.No,
+                                QMessageBox.Yes | QMessageBox.No,
                                 QMessageBox.NoButton) == QMessageBox.Yes
 
         if install_plugin:
@@ -154,8 +153,7 @@ class PluginInstall(QObject):
                 QMessageBox.critical(self.parent,
                     self.parent.windowTitle(),
                     self.__tr("<b>Unable to find an appropriate su/sudo utility to run hp-plugin.</b><p>Install kdesu, gnomesu, or gksu.</p>"),
-                    QMessageBox.Ok,
-                    QMessageBox.NoButton,
+                    QMessageBox.Ok | QMessageBox.NoButton,
                     QMessageBox.NoButton)
 
 
diff -up hplip-3.23.3/ui5/fabwindow.py.qmsgbox-typos-fix hplip-3.23.3/ui5/fabwindow.py
--- hplip-3.23.3/ui5/fabwindow.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui5/fabwindow.py	2023-05-29 13:48:48.429731297 +0200
@@ -344,7 +344,7 @@ class FABWindow(QMainWindow,  Ui_MainWin
             new_name = to_unicode(self.NameLineEdit.text())
             if new_name != self.name:
                 if QMessageBox.question(self, self.__tr("Rename?"), "Rename '%s' to '%s'?"%(self.name,new_name), \
-                                        QMessageBox.Yes | QMessageBox.No) == QMessageBox.Yes:
+                                        QMessageBox.Yes | QMessageBox.No, QMessageBox.No) == QMessageBox.Yes:
 
                     self.db.rename(self.name, new_name)
                     log.debug("Rename %s to %s" % (self.name, new_name))
diff -up hplip-3.23.3/ui5/nodevicesdialog.py.qmsgbox-typos-fix hplip-3.23.3/ui5/nodevicesdialog.py
--- hplip-3.23.3/ui5/nodevicesdialog.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui5/nodevicesdialog.py	2023-05-29 13:48:48.429731297 +0200
@@ -64,8 +64,7 @@ class NoDevicesDialog(QDialog, Ui_NoDevi
             QMessageBox.critical(self,
                                     self.windowTitle(),
                                     self.__tr("<b>An error occurred.</b><p>Please re-start the Device Manager and try again."),
-                                    QMessageBox.Ok,
-                                    QMessageBox.NoButton,
+                                    QMessageBox.Ok | QMessageBox.NoButton,
                                     QMessageBox.NoButton)
 
 
diff -up hplip-3.23.3/ui5/plugindialog.py.qmsgbox-typos-fix hplip-3.23.3/ui5/plugindialog.py
--- hplip-3.23.3/ui5/plugindialog.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui5/plugindialog.py	2023-05-29 13:48:48.429731297 +0200
@@ -252,7 +252,7 @@ class PluginDialog(QDialog, Ui_Dialog):
 
                 if QMessageBox.question(self, " ",
                         self.__tr("<b>%s</b><p>Without this, it is not possible to authenticate and validate the plug-in prior to installation.</p>Do you still want to install the plug-in?" %error_str),
-                        QMessageBox.Yes | QMessageBox.No) != QMessageBox.Yes:
+                                        QMessageBox.Yes | QMessageBox.No, QMessageBox.No) != QMessageBox.Yes:
 
                     self.pluginObj.deleteInstallationFiles(download_plugin_file)
                     self.close()
diff -up hplip-3.23.3/ui5/queuesconf.py.qmsgbox-typos-fix hplip-3.23.3/ui5/queuesconf.py
--- hplip-3.23.3/ui5/queuesconf.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui5/queuesconf.py	2023-05-29 13:48:48.430731291 +0200
@@ -245,7 +245,7 @@ class QueuesDiagnose(QDialog, Ui_Dialog)
                 
                     if QMessageBox.question(self, " ",
                         self.__tr("<b>%s</b><p>Without this, it is not possible to authenticate and validate this tool prior to installation.</p>Do you still want to run Smart Install disabler?" %error_str),
-                        QMessageBox.Yes | QMessageBox.No) == QMessageBox.Yes:
+                                            QMessageBox.Yes | QMessageBox.No, QMessageBox.No) == QMessageBox.Yes:
                         # Disabling without verification.
                         sts, out = utils.run("sh %s"%smart_install_run)
 
diff -up hplip-3.23.3/ui5/setupdialog.py.qmsgbox-typos-fix hplip-3.23.3/ui5/setupdialog.py
--- hplip-3.23.3/ui5/setupdialog.py.qmsgbox-typos-fix	2023-05-29 13:48:48.430731291 +0200
+++ hplip-3.23.3/ui5/setupdialog.py	2023-05-29 13:54:03.236851925 +0200
@@ -886,7 +886,7 @@ class SetupDialog(QDialog, Ui_Dialog):
                                     self.windowTitle(),
                                     warn_text,
                                     QMessageBox.Yes |
-                                    QMessageBox.No |
+                                    QMessageBox.No,
                                     QMessageBox.NoButton) == QMessageBox.Yes):
                 i = 2
                 while True:
@@ -1171,7 +1171,7 @@ class SetupDialog(QDialog, Ui_Dialog):
                     if QMessageBox.critical(self,
                                             self.windowTitle(),
                                             error_text,
-                                            QMessageBox.Retry | QMessageBox.Default,
+                                            QMessageBox.Retry | QMessageBox.Default |\
                                             QMessageBox.Cancel | QMessageBox.Escape,
                                             QMessageBox.NoButton) == QMessageBox.Cancel:
                         break
@@ -1223,7 +1223,7 @@ class SetupDialog(QDialog, Ui_Dialog):
                                 if QMessageBox.critical(self,
                                                         self.windowTitle(),
                                                         error_text,
-                                                        QMessageBox.Retry | QMessageBox.Default,
+                                                        QMessageBox.Retry | QMessageBox.Default |\
                                                         QMessageBox.Cancel | QMessageBox.Escape,
                                                         QMessageBox.NoButton) == QMessageBox.Cancel:
                                     break
diff -up hplip-3.23.3/ui/devmgr4.py.qmsgbox-typos-fix hplip-3.23.3/ui/devmgr4.py
--- hplip-3.23.3/ui/devmgr4.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/devmgr4.py	2023-05-29 13:48:48.424731328 +0200
@@ -1422,7 +1422,7 @@ class DevMgr4(DevMgr4_base):
             QMessageBox.critical(self,
                 self.caption(),
                 self.__tr("<b>Unable to find an appropriate su/sudo utility to run hp-plugin.</b><p>Install kdesu, gnomesu, or gksu.</p>"),
-                QMessageBox.Ok,
+                QMessageBox.Ok |\
                 QMessageBox.NoButton,
                 QMessageBox.NoButton)
         else:
@@ -2854,7 +2854,7 @@ class DevMgr4(DevMgr4_base):
             x = QMessageBox.critical(self,
            self.caption(),
            self.__tr("<b>Annoying Confirmation: Are you sure you want to remove this device?</b>"),
-            QMessageBox.Yes,
+            QMessageBox.Yes |\
             QMessageBox.No | QMessageBox.Default,
             QMessageBox.NoButton)
             if x == QMessageBox.Yes:
@@ -2949,7 +2949,7 @@ class DevMgr4(DevMgr4_base):
         QMessageBox.critical(self,
             self.caption(),
             error_text,
-            QMessageBox.Ok,
+            QMessageBox.Ok |\
             QMessageBox.NoButton,
             QMessageBox.NoButton)
 
@@ -2958,7 +2958,7 @@ class DevMgr4(DevMgr4_base):
         QMessageBox.warning(self,
             self.caption(),
             msg,
-            QMessageBox.Ok,
+            QMessageBox.Ok |\
             QMessageBox.NoButton,
             QMessageBox.NoButton)
 
@@ -3115,7 +3115,7 @@ class ScrollTestpageView(ScrollView):
             QMessageBox.information(self,
                 self.caption(),
                 self.__tr("<p><b>A test page should be printing on your printer.</b><p>If the page fails to print, please visit http://hplip.sourceforge.net for troubleshooting and support."),
-                QMessageBox.Ok,
+                QMessageBox.Ok |\
                 QMessageBox.NoButton,
                 QMessageBox.NoButton)
 
@@ -3130,7 +3130,7 @@ class ScrollTestpageView(ScrollView):
         QMessageBox.critical(self,
             self.caption(),
             error_text,
-            QMessageBox.Ok,
+            QMessageBox.Ok |\
             QMessageBox.NoButton,
             QMessageBox.NoButton)
 
@@ -3272,7 +3272,7 @@ class ScrollColorCalView(ScrollView):
             QMessageBox.information(self,
                 self.caption(),
                 self.__tr("<p><b>A test page should be printing on your printer.</b><p>If the page fails to print, please visit http://hplip.sourceforge.net for troubleshooting and support."),
-                QMessageBox.Ok,
+                QMessageBox.Ok |\
                 QMessageBox.NoButton,
                 QMessageBox.NoButton)
 
@@ -3287,7 +3287,7 @@ class ScrollColorCalView(ScrollView):
         QMessageBox.critical(self,
             self.caption(),
             error_text,
-            QMessageBox.Ok,
+            QMessageBox.Ok |\
             QMessageBox.NoButton,
             QMessageBox.NoButton)
 
diff -up hplip-3.23.3/ui/faxaddrbookform.py.qmsgbox-typos-fix hplip-3.23.3/ui/faxaddrbookform.py
--- hplip-3.23.3/ui/faxaddrbookform.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/faxaddrbookform.py	2023-05-29 13:48:48.424731328 +0200
@@ -228,7 +228,7 @@ class FaxAddrBookGroupsForm(FaxAddrBookG
         x = QMessageBox.critical(self,
                                  self.caption(),
                                  self.__tr("<b>Annoying Confirmation: Are you sure you want to delete this group?</b>"),
-                                  QMessageBox.Yes,
+                                  QMessageBox.Yes |\
                                   QMessageBox.No | QMessageBox.Default,
                                   QMessageBox.NoButton)
         if x == QMessageBox.Yes:
@@ -421,7 +421,7 @@ class FaxAddrBookForm(FaxAddrBookForm_ba
         if QMessageBox.critical(self,
              self.caption(),
              self.__tr("<b>Annoying Confirmation: Are you sure you want to delete this address book entry?</b>"),
-              QMessageBox.Yes,
+              QMessageBox.Yes |\
               QMessageBox.No | QMessageBox.Default,
               QMessageBox.NoButton) == QMessageBox.Yes:
             db.delete(self.current.entry['name'])
@@ -451,7 +451,7 @@ class FaxAddrBookForm(FaxAddrBookForm_ba
         QMessageBox.critical(self,
                              self.caption(),
                              QString(error_text),
-                              QMessageBox.Ok,
+                              QMessageBox.Ok |\
                               QMessageBox.NoButton,
                               QMessageBox.NoButton)
 
diff -up hplip-3.23.3/ui/faxsendjobform.py.qmsgbox-typos-fix hplip-3.23.3/ui/faxsendjobform.py
--- hplip-3.23.3/ui/faxsendjobform.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/faxsendjobform.py	2023-05-29 13:48:48.424731328 +0200
@@ -210,7 +210,7 @@ class FaxSendJobForm(QMainWindow):
         QMessageBox.information(self,
                              self.caption(),
                              self.__tr("<p><b>Fax send completed successfully.</b>"),
-                              QMessageBox.Ok,
+                              QMessageBox.Ok |\
                               QMessageBox.NoButton,
                               QMessageBox.NoButton)
 
@@ -219,7 +219,7 @@ class FaxSendJobForm(QMainWindow):
         QMessageBox.critical(self,
                              self.caption(),
                              error_text,
-                              QMessageBox.Ok,
+                              QMessageBox.Ok |\
                               QMessageBox.NoButton,
                               QMessageBox.NoButton)
 
@@ -228,7 +228,7 @@ class FaxSendJobForm(QMainWindow):
         QMessageBox.warning(self,
                              self.caption(),
                              error_text,
-                             QMessageBox.Ok,
+                             QMessageBox.Ok |\
                              QMessageBox.NoButton,
                              QMessageBox.NoButton)
 
diff -up hplip-3.23.3/ui/firmwaredialog.py.qmsgbox-typos-fix hplip-3.23.3/ui/firmwaredialog.py
--- hplip-3.23.3/ui/firmwaredialog.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/firmwaredialog.py	2023-05-29 13:48:48.424731328 +0200
@@ -109,7 +109,7 @@ class FirmwareDialog(QDialog, FirmwareDi
         QMessageBox.critical(self,
                 self.caption(),
                 error_text,
-                QMessageBox.Ok,
+                QMessageBox.Ok |\
                 QMessageBox.NoButton,
                 QMessageBox.NoButton)
 
diff -up hplip-3.23.3/ui/makecopiesform.py.qmsgbox-typos-fix hplip-3.23.3/ui/makecopiesform.py
--- hplip-3.23.3/ui/makecopiesform.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/makecopiesform.py	2023-05-29 13:48:48.424731328 +0200
@@ -156,7 +156,7 @@ class MakeCopiesForm(QMainWindow):
         QMessageBox.critical(self,
                              self.caption(),
                              error_text,
-                              QMessageBox.Ok,
+                              QMessageBox.Ok |\
                               QMessageBox.NoButton,
                               QMessageBox.NoButton)
 
@@ -164,7 +164,7 @@ class MakeCopiesForm(QMainWindow):
         QMessageBox.warning(self,
                              self.caption(),
                              msg,
-                              QMessageBox.Ok,
+                              QMessageBox.Ok |\
                               QMessageBox.NoButton,
                               QMessageBox.NoButton)
 
diff -up hplip-3.23.3/ui/nodevicesform.py.qmsgbox-typos-fix hplip-3.23.3/ui/nodevicesform.py
--- hplip-3.23.3/ui/nodevicesform.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/nodevicesform.py	2023-05-29 13:48:48.424731328 +0200
@@ -67,7 +67,7 @@ class NoDevicesForm(NoDevicesForm_base):
             QMessageBox.critical(self,
                                     self.caption(),
                                     self.__tr("<b>An error occurred.</b><p>Please re-start the Device Manager and try again."),
-                                    QMessageBox.Ok,
+                                    QMessageBox.Ok |\
                                     QMessageBox.NoButton,
                                     QMessageBox.NoButton)
 
diff -up hplip-3.23.3/ui/pluginform2.py.qmsgbox-typos-fix hplip-3.23.3/ui/pluginform2.py
--- hplip-3.23.3/ui/pluginform2.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/pluginform2.py	2023-05-29 13:48:48.424731328 +0200
@@ -173,7 +173,7 @@ class PluginForm2(PluginForm2_base):
         QMessageBox.critical(self,
             self.caption(),
             error_text,
-            QMessageBox.Ok,
+            QMessageBox.Ok |\
             QMessageBox.NoButton,
             QMessageBox.NoButton)
 
@@ -181,7 +181,7 @@ class PluginForm2(PluginForm2_base):
         QMessageBox.information(self,
                              self.caption(),
                              text,
-                              QMessageBox.Ok,
+                              QMessageBox.Ok |\
                               QMessageBox.NoButton,
                               QMessageBox.NoButton)
 
diff -up hplip-3.23.3/ui/printerform.py.qmsgbox-typos-fix hplip-3.23.3/ui/printerform.py
--- hplip-3.23.3/ui/printerform.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/printerform.py	2023-05-29 13:48:48.425731321 +0200
@@ -154,7 +154,7 @@ class PrinterForm(QMainWindow):
         QMessageBox.critical(self,
                              self.caption(),
                              error_text,
-                              QMessageBox.Ok,
+                              QMessageBox.Ok |\
                               QMessageBox.NoButton,
                               QMessageBox.NoButton)
 
diff -up hplip-3.23.3/ui/settingsdialog.py.qmsgbox-typos-fix hplip-3.23.3/ui/settingsdialog.py
--- hplip-3.23.3/ui/settingsdialog.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/settingsdialog.py	2023-05-29 13:48:48.425731321 +0200
@@ -146,7 +146,7 @@ class SettingsDialog(SettingsDialog_base
 ##            QMessageBox.warning(self,
 ##                                 self.caption(),
 ##                                 self.__tr("<b>One or more email addresses are missing.</b><p>Please enter this information and try again."),
-##                                  QMessageBox.Ok,
+##                                  QMessageBox.Ok |\
 ##                                  QMessageBox.NoButton,
 ##                                  QMessageBox.NoButton)
 ##            return
@@ -163,7 +163,7 @@ class SettingsDialog(SettingsDialog_base
 ##        QMessageBox.information(self,
 ##                     self.caption(),
 ##                     self.__tr("<p><b>Please check your email for a test message.</b><p>If the message doesn't arrive, please check your settings and try again."),
-##                      QMessageBox.Ok,
+##                      QMessageBox.Ok |\
 ##                      QMessageBox.NoButton,
 ##                      QMessageBox.NoButton)
 
diff -up hplip-3.23.3/ui/setupform.py.qmsgbox-typos-fix hplip-3.23.3/ui/setupform.py
--- hplip-3.23.3/ui/setupform.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/setupform.py	2023-05-29 13:48:48.425731321 +0200
@@ -602,7 +602,7 @@ class SetupForm(SetupForm_base):
             if ( QMessageBox.warning(self,
                                 self.caption(),
                                 warn_text,
-                                QMessageBox.Yes,
+                                QMessageBox.Yes |\
                                 QMessageBox.No,
                                 QMessageBox.NoButton) == QMessageBox.Yes ):
 
@@ -804,7 +804,7 @@ class SetupForm(SetupForm_base):
                     if QMessageBox.critical(self,
                                            self.caption(),
                                            error_text,
-                                           QMessageBox.Retry | QMessageBox.Default,
+                                           QMessageBox.Retry | QMessageBox.Default |\
                                            QMessageBox.Cancel | QMessageBox.Escape,
                                            QMessageBox.NoButton) == QMessageBox.Cancel:
                         break
@@ -832,7 +832,7 @@ class SetupForm(SetupForm_base):
                                 if QMessageBox.critical(self,
                                                        self.caption(),
                                                        error_text,
-                                                       QMessageBox.Retry | QMessageBox.Default,
+                                                       QMessageBox.Retry | QMessageBox.Default |\
                                                        QMessageBox.Cancel | QMessageBox.Escape,
                                                        QMessageBox.NoButton) == QMessageBox.Cancel:
                                     break
@@ -1003,7 +1003,7 @@ class SetupForm(SetupForm_base):
         QMessageBox.critical(self,
                              self.caption(),
                              error_text,
-                              QMessageBox.Ok,
+                              QMessageBox.Ok |\
                               QMessageBox.NoButton,
                               QMessageBox.NoButton)
 
@@ -1011,7 +1011,7 @@ class SetupForm(SetupForm_base):
         QMessageBox.warning(self,
                              self.caption(),
                              error_text,
-                              QMessageBox.Ok,
+                              QMessageBox.Ok |\
                               QMessageBox.NoButton,
                               QMessageBox.NoButton)
 
diff -up hplip-3.23.3/ui/unloadform.py.qmsgbox-typos-fix hplip-3.23.3/ui/unloadform.py
--- hplip-3.23.3/ui/unloadform.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/unloadform.py	2023-05-29 13:48:48.425731321 +0200
@@ -135,7 +135,7 @@ class UnloadForm(QMainWindow):
         QMessageBox.critical(self,
                              self.caption(),
                              error_text,
-                              QMessageBox.Ok,
+                              QMessageBox.Ok |\
                               QMessageBox.NoButton,
                               QMessageBox.NoButton)
 
diff -up hplip-3.23.3/ui/upgradeform.py.qmsgbox-typos-fix hplip-3.23.3/ui/upgradeform.py
--- hplip-3.23.3/ui/upgradeform.py.qmsgbox-typos-fix	2023-03-28 09:26:11.000000000 +0200
+++ hplip-3.23.3/ui/upgradeform.py	2023-05-29 13:48:48.425731321 +0200
@@ -118,7 +118,7 @@ class UpgradeForm(UpgradeForm_base):
         QMessageBox.critical(self,
             self.caption(),
             error_text,
-            QMessageBox.Ok,
+            QMessageBox.Ok |\
             QMessageBox.NoButton,
             QMessageBox.NoButton)
 
@@ -126,7 +126,7 @@ class UpgradeForm(UpgradeForm_base):
         QMessageBox.information(self,
                              self.caption(),
                              text,
-                              QMessageBox.Ok,
+                              QMessageBox.Ok |\
                               QMessageBox.NoButton,
                               QMessageBox.NoButton)
 
