diff --git a/base/sixext.py b/base/sixext.py
index 0bf4fc4..311bf72 100644
--- a/base/sixext.py
+++ b/base/sixext.py
@@ -110,11 +110,11 @@ if PY3:
 
 
     def to_string_utf8(s):
-        return s.decode("utf-8")
+        return s.decode("utf-8", 'ignore')
 
 
     def to_string_latin(s):
-        return s.decode("latin-1")
+        return s.decode("latin-1", 'ignore')
 
 
     def to_unicode(s, enc=None):
