diff --git a/scan/sane/escl.c b/scan/sane/escl.c
index 9fbbcaf..bc9e0b5 100644
--- a/scan/sane/escl.c
+++ b/scan/sane/escl.c
@@ -37,18 +37,12 @@ static int bb_load(struct escl_session *ps, const char *so)
    int stat=1;
    
    /* Load hpmud manually with symbols exported. Otherwise the plugin will not find it. */ 
-   if ((ps->hpmud_handle = load_library("libhpmud.so.0")) == NULL)
-   {
-	   if ((ps->hpmud_handle = load_library("libhpmud.so.0")) == NULL)
-           goto bugout;
-   }
+	 if ((ps->hpmud_handle = load_library("libhpmud.so.0")) == NULL)
+         goto bugout;
 
    /* Load math library manually with symbols exported (Ubuntu 8.04). Otherwise the plugin will not find it. */ 
-   if ((ps->math_handle = load_library("libm.so")) == NULL)
-   {
-      if ((ps->math_handle = load_library("libm.so.6")) == NULL)
-         goto bugout;
-   } 
+   if ((ps->math_handle = load_library("libm.so.6")) == NULL)
+      goto bugout;
    if ((ps->bb_handle = load_plugin_library(UTILS_SCAN_PLUGIN_LIBRARY, so)) == NULL)
    {
       SendScanEvent(ps->uri, EVENT_PLUGIN_FAIL);
diff --git a/scan/sane/marvell.c b/scan/sane/marvell.c
index 236f7c9..793104c 100644
--- a/scan/sane/marvell.c
+++ b/scan/sane/marvell.c
@@ -60,18 +60,12 @@ static int bb_load(struct marvell_session *ps, const char *so)
    int stat=1;
 
    /* Load hpmud manually with symbols exported. Otherwise the plugin will not find it. */ 
-   if ((ps->hpmud_handle = load_library("libhpmud.so")) == NULL)
-   {
-	   if ((ps->hpmud_handle = load_library("libhpmud.so.0")) == NULL)
-           goto bugout;
-   }
+	 if ((ps->hpmud_handle = load_library("libhpmud.so.0")) == NULL)
+         goto bugout;
 
    /* Load math library manually with symbols exported (Ubuntu 8.04). Otherwise the plugin will not find it. */ 
-   if ((ps->math_handle = load_library("libm.so")) == NULL)
-   {
-      if ((ps->math_handle = load_library("libm.so.6")) == NULL)
-         goto bugout;
-   } 
+   if ((ps->math_handle = load_library("libm.so.6")) == NULL)
+      goto bugout;
 
    if ((ps->bb_handle = load_plugin_library(UTILS_SCAN_PLUGIN_LIBRARY, so)) == NULL)
    {
diff --git a/scan/sane/soap.c b/scan/sane/soap.c
index 0d22e52..dcd8c7c 100644
--- a/scan/sane/soap.c
+++ b/scan/sane/soap.c
@@ -68,18 +68,12 @@ static int bb_load(struct soap_session *ps, const char *so)
    int stat=1;
 
    /* Load hpmud manually with symbols exported. Otherwise the plugin will not find it. */ 
-   if ((ps->hpmud_handle = load_library("libhpmud.so")) == NULL)
-   {
-	   if ((ps->hpmud_handle = load_library("libhpmud.so.0")) == NULL)
-           goto bugout;
-   }
+	 if ((ps->hpmud_handle = load_library("libhpmud.so.0")) == NULL)
+         goto bugout;
 
    /* Load math library manually with symbols exported (Ubuntu 8.04). Otherwise the plugin will not find it. */ 
-   if ((ps->math_handle = load_library("libm.so")) == NULL)
-   {
-      if ((ps->math_handle = load_library("libm.so.6")) == NULL)
-         goto bugout;
-   } 
+   if ((ps->math_handle = load_library("libm.so.6")) == NULL)
+      goto bugout;
 
    if ((ps->bb_handle = load_plugin_library(UTILS_SCAN_PLUGIN_LIBRARY, so)) == NULL)
    {
diff --git a/scan/sane/soapht.c b/scan/sane/soapht.c
index 139100d..74e8391 100644
--- a/scan/sane/soapht.c
+++ b/scan/sane/soapht.c
@@ -62,18 +62,12 @@ static int bb_load(struct soap_session *ps, const char *so)
    int stat=1;
 
    /* Load hpmud manually with symbols exported. Otherwise the plugin will not find it. */ 
-   if ((ps->hpmud_handle = load_library("libhpmud.so.0")) == NULL)
-   {
-	   if ((ps->hpmud_handle = load_library("libhpmud.so.0")) == NULL)
-           goto bugout;
-   }
+	 if ((ps->hpmud_handle = load_library("libhpmud.so.0")) == NULL)
+         goto bugout;
 
    /* Load math library manually with symbols exported (Ubuntu 8.04). Otherwise the plugin will not find it. */ 
-   if ((ps->math_handle = load_library("libm.so")) == NULL)
-   {
-      if ((ps->math_handle = load_library("libm.so.6")) == NULL)
-         goto bugout;
-   } 
+   if ((ps->math_handle = load_library("libm.so.6")) == NULL)
+      goto bugout;
 
    if ((ps->bb_handle = load_plugin_library(UTILS_SCAN_PLUGIN_LIBRARY, so)) == NULL)
    {
