diff --git a/base/validation.py b/base/validation.py
index 96e65be..35f22eb 100644
--- a/base/validation.py
+++ b/base/validation.py
@@ -58,7 +58,7 @@ class GPG_Verification(DigiSign_Verification):
 
         if not os.path.exists(self.__gpg_dir):
             try:
-                os.mkdir(self.__gpg_dir, 0o755)
+                os.mkdir(self.__gpg_dir, 0o700)
             except OSError:
                 log.error("Failed to create %s" % self.__gpg_dir)
         self.__change_owner()
