diff --git a/base/queues.py b/base/queues.py
index 2f56c8a..0818574 100755
--- a/base/queues.py
+++ b/base/queues.py
@@ -101,7 +101,7 @@ def parseQueues(mode):
             try:
                 match = LPSTAT_PATTERN.search(p)
                 printer_name = match.group(1)
-                device_uri = match.group(2)
+                device_uri = match.group(3)
                 cups_printers.append((printer_name, device_uri))
             except AttributeError:
                 pass
