From 839a216f17ce17a5667eecd54de2a70b1383df2c Mon Sep 17 00:00:00 2001
From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Wed, 5 Jan 2022 21:57:06 -0500
Subject: [PATCH 1/2] Info dlg: Use QDateTime.toString()

---
 ui5/infodialog.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ui5/infodialog.py b/ui5/infodialog.py
index 0de62fb..eaa846d 100644
--- a/ui5/infodialog.py
+++ b/ui5/infodialog.py
@@ -186,7 +186,7 @@ class InfoDialog(QDialog, Ui_Dialog):
         for row, h in enumerate(history):
             dt = QDateTime()
             dt.setTime_t(int(h.timedate))
-            dt = value_str(dt)
+            dt = value_str(dt.toString())
 
             ess = device.queryString(h.event_code, 0)
 
-- 
2.33.1

From f6e7ab1542afe7fbcd2d29b6cb15b97484e3779c Mon Sep 17 00:00:00 2001
From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Wed, 5 Jan 2022 21:56:39 -0500
Subject: [PATCH 2/2] value_str: Handle QDateTime correctly

---
 ui5/ui_utils.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/ui5/ui_utils.py b/ui5/ui_utils.py
index caba339..225ca1c 100644
--- a/ui5/ui_utils.py
+++ b/ui5/ui_utils.py
@@ -105,7 +105,10 @@ def value_str(data):
     if data is None:
         return ""
     try:
-        if not PY3:
+        if PY3:
+            if isinstance(data, QDateTime):
+                data = data.toString()
+        else:
             try:
                 data = data.toString()
             except AttributeError as e:
-- 
2.33.1

