diff --git a/Makefile b/Makefile
index 08b8cfc..cc03acd 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ CFLAGS += -I/opt/local/include
 OPTIMIZE=-O3
 #OPTIMIZE=-O0 -g
 
-COMPILEFLAGS=$(CFLAGS) $(OPTIMIZE) -DVERSION=\"$(VERSION)\" -Wall -Wextra -Wno-unused
+COMPILEFLAGS=$(CPPLAGS) $(CFLAGS) $(OPTIMIZE) -DVERSION=\"$(VERSION)\" -Wall -Wextra -Wno-unused
 LINKFLAGS=$(LDFLAGS) -ljack -lsndfile -lm -lpthread
 
 OS := $(shell uname)
