diff --git a/README b/README
index b813b59..8cb7024 100644
--- a/README
+++ b/README
@@ -160,7 +160,7 @@ Additional arguments:
  All hook options take a full-path to an executable as argument.
  The commands are executed in a fire-and-forget style upon internal events.
  All output of the hooks is discarded.
- Paramaters passed to the hook-scripts:
+ Parameters passed to the hook-scripts:
   open:   CMD <filename>
   close:  CMD <filename> <xrun-count> <io-error-count>
   rotate: CMD <filename> <xrun-count> <io-error-count> <new-filename> <seq-number>
diff --git a/jack_capture.c b/jack_capture.c
index b2d15a1..6a1971f 100644
--- a/jack_capture.c
+++ b/jack_capture.c
@@ -2276,7 +2276,7 @@ static const char *advanced_help =
   " All hook options take a full-path to an executable as argument.\n"
   " The commands are executed in a fire-and-forget style upon internal events.\n"
   " All output of the hooks is discarded.\n"
-  " Paramaters passed to the hook-scripts:\n"
+  " Parameters passed to the hook-scripts:\n"
   "  open:   CMD <filename>\n"
   "  close:  CMD <filename> <xrun-count> <io-error-count>\n"
   "  rotate: CMD <filename> <xrun-count> <io-error-count> <new-filename> <seq-number>\n"
