From 6855f6912aff21c5aec73777c504bfce9e62d37b Mon Sep 17 00:00:00 2001
From: hannes <hannes@fedoraproject.org>
Date: Wed, 16 Dec 2015 20:34:51 +0100
Subject: [PATCH] update_check

---
 configdialog.ui   | 10 ++++++++--
 configmanager.cpp |  2 +-
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/configdialog.ui b/configdialog.ui
index 2e7e131..a8f69b1 100644
--- a/configdialog.ui
+++ b/configdialog.ui
@@ -557,15 +557,21 @@
                  <number>1</number>
                 </property>
                 <property name="value">
-                 <number>7</number>
+			<number>7</number>
                 </property>
+		<property name="readOnly">
+			<bool>true</bool>
+		</property>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QCheckBox" name="checkBoxAutoUpdateCheck">
                 <property name="text">
                  <string>Automatically check every</string>
-                </property>
+	 </property>
+ 	<property name="checkable">
+	<bool>false</bool>
+	</property>
                </widget>
               </item>
               <item row="0" column="5">
diff --git a/configmanager.cpp b/configmanager.cpp
index e1ddbdb..443f4a7 100644
--- a/configmanager.cpp
+++ b/configmanager.cpp
@@ -467,7 +467,7 @@ ConfigManager::ConfigManager(QObject *parent): QObject (parent),
 	registerOption("Thesaurus/Database", &thesaurus_database, "<dic not found>", &pseudoDialog->comboBoxThesaurusFileName);
 
 	//updates
-	registerOption("Update/AutoCheck", &autoUpdateCheck, true, &pseudoDialog->checkBoxAutoUpdateCheck);
+	registerOption("Update/AutoCheck", &autoUpdateCheck, false, &pseudoDialog->checkBoxAutoUpdateCheck);
 	registerOption("Update/UpdateLevel", &updateLevel, 0, &pseudoDialog->comboBoxUpdateLevel);
 	registerOption("Update/AutoCheckInvervalDays", &autoUpdateCheckIntervalDays, 7, &pseudoDialog->spinBoxAutoUpdateCheckIntervalDays);
 	registerOption("Update/LastCheck", &lastUpdateCheck, QDateTime());
-- 
2.5.0

