From 571eb530162a7c3a370d30fbed8f73b725745ebb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ji=C5=99=C3=AD=20Kone=C4=8Dn=C3=BD?=
 <DragonLichcz@gmail.com>
Date: Wed, 13 Jan 2016 18:31:11 +0100
Subject: [PATCH] Change Makefile to install appdata.xml file

---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index a4fca7d..a24755a 100644
--- a/Makefile
+++ b/Makefile
@@ -59,11 +59,13 @@ install:
 	install -d $(execdir)
 	install -d $(sharedir)/pixmaps
 	install -d $(sharedir)/applications
+	install -d $(sharedir)/appdata
 	install -d $(sharedir)/playonlinux/bin
 	install -d $(sharedir)/man/man1
 	$(GZIP) -c ./doc/playonlinux-pkg.1 > $(sharedir)/man/man1/playonlinux-pkg.1.gz
 	$(GZIP) -c ./doc/playonlinux.1 > $(sharedir)/man/man1/playonlinux.1.gz
 	cp ./etc/PlayOnLinux.desktop $(sharedir)/applications/PlayOnLinux.desktop
+	cp ./etc/PlayOnLinux.appdata.xml $(sharedir)/appdata/PlayOnLinux.appdata.xml
 	cp ./etc/playonlinux.png $(sharedir)/pixmaps/playonlinux.png
 	cp ./etc/playonlinux16.png $(sharedir)/pixmaps/playonlinux16.png
 	cp ./etc/playonlinux32.png $(sharedir)/pixmaps/playonlinux32.png
-- 
2.5.0

