From 7edafce9c4dc19e3c323b73e62681001234ead81 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ji=C5=99=C3=AD=20Kone=C4=8Dn=C3=BD?=
 <DragonLichcz@gmail.com>
Date: Fri, 22 Jan 2016 21:03:39 +0100
Subject: [PATCH] Install lang files correctly

To use installed locales DEBIAN_PACKAGE value must be set to true.
---
 Makefile | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index a4fca7d..511f344 100644
--- a/Makefile
+++ b/Makefile
@@ -53,6 +53,8 @@ build:
 	echo -e '#!/bin/bash\n${sharedir}/playonlinux/playonlinux-pkg "$$@"\nexit 0' > ./bin/playonlinux-pkg
 	chmod +x ./bin/playonlinux
 	chmod +x ./bin/playonlinux-pkg
+	sed -i 's/\(\["DEBIAN_PACKAGE"\]\s*=\s*\)"FALSE"/\1"TRUE"/' \
+		./python/lib/Variables.py
 
 install:
 	install -d $(bindir)
@@ -61,6 +63,7 @@ install:
 	install -d $(sharedir)/applications
 	install -d $(sharedir)/playonlinux/bin
 	install -d $(sharedir)/man/man1
+	install -d $(sharedir)/locale
 	$(GZIP) -c ./doc/playonlinux-pkg.1 > $(sharedir)/man/man1/playonlinux-pkg.1.gz
 	$(GZIP) -c ./doc/playonlinux.1 > $(sharedir)/man/man1/playonlinux.1.gz
 	cp ./etc/PlayOnLinux.desktop $(sharedir)/applications/PlayOnLinux.desktop
@@ -70,7 +73,8 @@ install:
 	cp ./bin/{playonlinux,playonlinux-pkg} $(bindir)/
 	cp ./bin/playonlinux-check_dd $(execdir)/
 	cp ./{playonlinux*,README.md,TRANSLATORS,CHANGELOG.md,LICENCE} $(sharedir)/playonlinux/
-	cp -R ./{bash,etc,lang,lib,plugins,python,resources,tests} $(sharedir)/playonlinux/
+	cp -R ./{bash,etc,lib,plugins,python,resources,tests} $(sharedir)/playonlinux/
+	cp -R ./lang/locale/* $(sharedir)/locale/
 
 changelog:
 	(GIT_DIR=.git git log > .changelog.tmp && mv .changelog.tmp ChangeLog; rm -f .changelog.tmp) || (touch ChangeLog; echo 'git directory not found: installing possibly empty changelog.' >&2)
-- 
2.5.0

