From ef18eb7f0ed46bcde613e117bc74d25fa99fbb1a Mon Sep 17 00:00:00 2001
From: Ilikara <3435193369@qq.com>
Date: Mon, 29 Sep 2025 16:43:49 +0800
Subject: [PATCH 08/17] DEBIAN: Fix FTBFS with gcc-6.

Signed-off-by: Ilikara <3435193369@qq.com>
---
 htfuzzy/Accents.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/htfuzzy/Accents.cc b/htfuzzy/Accents.cc
index 619615a6..e489eb8e 100755
--- a/htfuzzy/Accents.cc
+++ b/htfuzzy/Accents.cc
@@ -43,7 +43,7 @@ extern int debug;
 | table iso-latin1 "minusculisee" et "de-accentuee" |
 `--------------------------------------------------*/
   
-static char MinusculeISOLAT1[256] = {
+static unsigned char MinusculeISOLAT1[256] = {
      0,   1,   2,   3,   4,   5,   6,   7,
      8,   9,  10,  11,  12,  13,  14,  15,
     16,  17,  18,  19,  20,  21,  22,  23,
-- 
2.51.0

