From 6a31678049151af83dadcedcdb6c3e849d3d59a4 Mon Sep 17 00:00:00 2001
From: Ilikara <3435193369@qq.com>
Date: Mon, 29 Sep 2025 17:02:20 +0800
Subject: [PATCH 10/17] DEBIAN: drop bogus assignment

Signed-off-by: Ilikara <3435193369@qq.com>
---
 htdig/Parsable.cc | 2 --
 1 file changed, 2 deletions(-)

diff --git a/htdig/Parsable.cc b/htdig/Parsable.cc
index 049362a8..a51b3667 100755
--- a/htdig/Parsable.cc
+++ b/htdig/Parsable.cc
@@ -74,7 +74,6 @@ Parsable::addString(Retriever& retriever, char *s, int& wordindex, int slot)
 	    retriever.got_word(w, wordindex++, slot); // slot for img_alt
 	w = HtWordToken(0);
     }
-    w = '\0';
 }
 
 //*****************************************************************************
@@ -92,5 +91,4 @@ Parsable::addKeywordString(Retriever& retriever, char *s, int& wordindex)
 	    retriever.got_word(w, wordindex++, 9);
 	w = HtWordToken(0);
     }
-    w = '\0';
 }
-- 
2.51.0

