--- htdig-3.2.0b6/htsearch/Collection.h.compile-fix	2006-02-09 18:36:26.000000000 +0100
+++ htdig-3.2.0b6/htsearch/Collection.h	2006-02-09 18:24:21.000000000 +0100
@@ -36,9 +36,9 @@
                const char *docExcerpt);
     ~Collection();
 
-    void Collection::Open();
+    void Open();
 
-    void Collection::Close(); 
+    void Close(); 
 
     char *getWordFile() { return wordFile.get(); }
     DocumentRef         *getDocumentRef(int id);
