--- htdig-3.2.0b6/installdir/rundig.incremental	2007-01-09 16:49:35.000000000 +0100
+++ htdig-3.2.0b6/installdir/rundig	2007-01-09 16:49:35.000000000 +0100
@@ -14,6 +14,7 @@
 
 # To help debug problems, try running rundig -vvv
 # This will add lots of debugging output to all of the programs
+incremental=0
 stats= opts= alt=
 while [ "$#" -gt 0 ]
 do
@@ -24,6 +25,7 @@
          conffile="$1"
 	 opts="$opts -c $conffile"
 	;;
+    +i) incremental=1;;
     *)   opts="$opts $1" ;;	# e.g. -v
     esac
     shift
@@ -53,7 +55,13 @@
 TMPDIR=$DBDIR
 export TMPDIR
 
-$BINDIR/htdig -i $opts $stats $alt
+if [ $incremental -ne 0 ]
+then
+    $BINDIR/htdig $opts $stats $alt
+else
+    $BINDIR/htdig -i $opts $stats $alt
+fi
+
 $BINDIR/htpurge $opts $alt
 case "$alt" in
 -a)
--- htdig-3.2.0b6/installdir/rundig.1.in.incremental	2007-01-09 16:50:11.000000000 +0100
+++ htdig-3.2.0b6/installdir/rundig.1.in	2007-01-09 16:52:17.000000000 +0100
@@ -23,6 +23,9 @@
 .TP
 .B \-c configfile
 Use alternate config file.
+.TP
+.B \+i
+Use incremental digging instead non-incremental
 .SH "SEE ALSO"
 Please refer to the HTML pages (in the htdig-doc package)
 .B /usr/share/doc/htdig-doc/html/index.html
