diff -Naur htdig-3.2.0b6.org/htfuzzy/Accents.cc htdig-3.2.0b6/htfuzzy/Accents.cc
--- htdig-3.2.0b6.org/htfuzzy/Accents.cc	2004-05-28 15:15:19.000000000 +0200
+++ htdig-3.2.0b6/htfuzzy/Accents.cc	2016-02-14 04:28:10.029793487 +0100
@@ -43,7 +43,7 @@
 | table iso-latin1 "minusculisee" et "de-accentuee" |
 `--------------------------------------------------*/
   
-static char MinusculeISOLAT1[256] = {
+static unsigned char MinusculeISOLAT1[256] = {
      0,   1,   2,   3,   4,   5,   6,   7,
      8,   9,  10,  11,  12,  13,  14,  15,
     16,  17,  18,  19,  20,  21,  22,  23,
