diff -Naur zita-resampler-1.6.0.old/apps/Makefile zita-resampler-1.6.0/apps/Makefile
--- a/apps/Makefile	2015-10-02 08:59:29.000000000 -0600
+++ b/apps/Makefile	2017-11-09 00:11:23.721763216 -0700
@@ -1,7 +1,7 @@
 # ----------------------------------------------------------------------------
 #
 #  Copyright (C) 2006-2011 Fons Adriaensen <fons@linuxaudio.org>
-#    
+#
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
 #  the Free Software Foundation; either version 3 of the License, or
@@ -26,8 +26,7 @@
 MANDIR = /usr/share/man/man1
 LDFLAGS += -L$(PREFIX)/$(LIBDIR)
 CPPFLAGS += -MMD -MP -DVERSION=\"$(VERSION)\"
-CXXFLAGS += -O3  -ffast-math -Wall
-CXXFLAGS += -march=native
+CXXFLAGS += -O3  -ffast-math -Wall -I../libs/
 
 
 all:	zresample zretune zresample.1.gz zretune.1.gz
@@ -62,7 +61,7 @@
 	install -Dm 644 zresample.1.gz $(DESTDIR)$(MANDIR)/zresample.1.gz
 	install -Dm 644 zretune.1.gz $(DESTDIR)$(MANDIR)/zretune.1.gz
 
-uninstall: 
+uninstall:
 	/bin/rm -f $(DESTDIR)$(PREFIX)/bin/zresample
 	/bin/rm -f $(DESTDIR)$(PREFIX)/bin/zretune
 	/bin/rm -f $(MANDIR)/zresample.1.gz
diff -Naur zita-resampler-1.6.0.old/libs/Makefile zita-resampler-1.6.0/libs/Makefile
--- a/libs/Makefile	2015-10-02 08:59:52.000000000 -0600
+++ b/libs/Makefile	2017-11-09 00:09:38.602722656 -0700
@@ -1,7 +1,7 @@
 # ----------------------------------------------------------------------------
 #
 #  Copyright (C) 2006-2012 Fons Adriaensen <fons@linuxaudio.org>
-#    
+#
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
 #  the Free Software Foundation; either version 3 of the License, or
@@ -32,8 +32,7 @@
 
 CPPFLAGS += -I. -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
 CXXFLAGS += -Wall -fPIC -O2 -ffast-math
-CXXFLAGS += -march=native
-LDFLAGS += 
+LDFLAGS +=
 LDLIBS +=
 
 
