--- a/libs/Makefile	2017-10-24 17:37:41.991867575 -0600
+++ b/libs/Makefile	2017-10-24 17:38:30.634674335 -0600
@@ -1,7 +1,7 @@
 # ----------------------------------------------------------------------------
 #
 #  Copyright (C) 2006-2012 Fons Adriaensen <fons@linuxaudio.org>
-#    
+#
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
 #  the Free Software Foundation; either version 3 of the License, or
@@ -32,8 +32,7 @@
 
 CPPFLAGS += -I. -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
 CXXFLAGS += -Wall -fPIC -O2 -ffast-math
-CXXFLAGS += -march=native
-LDFLAGS += 
+LDFLAGS +=
 LDLIBS +=
 
 
--- a/apps/Makefile	2017-10-24 17:37:17.849702362 -0600
+++ b/apps/Makefile	2017-10-24 17:53:35.019283908 -0600
@@ -1,7 +1,7 @@
 # ----------------------------------------------------------------------------
 #
 #  Copyright (C) 2006-2011 Fons Adriaensen <fons@linuxaudio.org>
-#    
+#
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
 #  the Free Software Foundation; either version 3 of the License, or
@@ -24,10 +24,9 @@
 DISTDIR = zresample-$(VERSION)
 VERSION = 1.6.0
 MANDIR = /usr/share/man/man1
-LDFLAGS += -L$(PREFIX)/$(LIBDIR)
+LDFLAGS += -L$(PREFIX)/$(LIBDIR) -I../libs/
 CPPFLAGS += -MMD -MP -DVERSION=\"$(VERSION)\"
 CXXFLAGS += -O3  -ffast-math -Wall
-CXXFLAGS += -march=native
 
 
 all:	zresample zretune zresample.1.gz zretune.1.gz
@@ -62,7 +61,7 @@
 	install -Dm 644 zresample.1.gz $(DESTDIR)$(MANDIR)/zresample.1.gz
 	install -Dm 644 zretune.1.gz $(DESTDIR)$(MANDIR)/zretune.1.gz
 
-uninstall: 
+uninstall:
 	/bin/rm -f $(DESTDIR)$(PREFIX)/bin/zresample
 	/bin/rm -f $(DESTDIR)$(PREFIX)/bin/zretune
 	/bin/rm -f $(MANDIR)/zresample.1.gz
