--- a/files/fail2ban.service.in 2022-09-21 06:38:37.576488068 +0000
+++ b/files/fail2ban.service.in       2022-09-21 06:41:12.297814704 +0000
@@ -8,9 +8,7 @@
 Type=simple
 Environment="PYTHONNOUSERSITE=1"
 ExecStartPre=/bin/mkdir -p /run/fail2ban
-ExecStart=@BINDIR@/fail2ban-server -xf start
-# if should be logged in systemd journal, use following line or set logtarget to sysout in fail2ban.local
-# ExecStart=@BINDIR@/fail2ban-server -xf --logtarget=sysout start
+ExecStart=@BINDIR@/fail2ban-server -xf --logtarget=sysout start
 ExecStop=@BINDIR@/fail2ban-client stop
 ExecReload=@BINDIR@/fail2ban-client reload
 PIDFile=/run/fail2ban/fail2ban.pid
