diff -Naur a/default b/default
--- a/default	2015-08-06 03:13:23.000000000 +0800
+++ b/default	2016-06-26 21:04:52.095562602 +0800
@@ -167,7 +167,7 @@
 # (should prevent accidential power on of hybrid graphics' discrete part).
 # Default is "radeon nouveau"; use "" to disable the feature completely.
 # Separate multiple drivers with spaces.
-RUNTIME_PM_DRIVER_BLACKLIST="radeon nouveau"
+# RUNTIME_PM_DRIVER_BLACKLIST="radeon nouveau"
 
 # Set to 0 to disable, 1 to enable USB autosuspend feature.
 USB_AUTOSUSPEND=1
