--- a/cmake/FindQCA.cmake	2019-09-13 07:11:09.000000000 -0500
+++ b/cmake/FindQCA.cmake	2020-09-19 00:23:28.682175990 -0500
@@ -44,9 +44,10 @@
       "$ENV{LIB_DIR}/include"
       $ENV{INCLUDE}
       /usr/local/include
-      PATH_SUFFIXES QtCrypto qt5/QtCrypto Qca-qt5/QtCrypto qt/Qca-qt5/QtCrypto
+      PATH_SUFFIXES QtCrypto qt5/QtCrypto Qca-qt5/QtCrypto qt/Qca-qt5/QtCrypto qt5/Qca-qt5/QtCrypto
   )
 
+  message(STATUS "${QCA_LIBRARY} ${QCA_INCLUDE_DIR}")
   if(QCA_LIBRARY AND QCA_INCLUDE_DIR)
     set(QCA_FOUND TRUE)
   endif()
