diff -Naur taisei/meson.build taisei.nostrip/meson.build
--- taisei/meson.build	2021-08-25 22:05:53.510311811 -0700
+++ taisei.nostrip/meson.build	2021-08-25 22:05:46.268256841 -0700
@@ -26,7 +26,7 @@
 
         # You may want to change these for a debug build dir
         'buildtype=release',
-        'strip=true',
+        'strip=false',
         'b_lto=true',
         'b_ndebug=if-release',
     ]
