diff -up qtbase-opensource-src-5.6.0-beta/configure.than qtbase-opensource-src-5.6.0-beta/configure
--- qtbase-opensource-src-5.6.0-beta/configure.than	2016-02-12 13:56:20.057741037 +0100
+++ qtbase-opensource-src-5.6.0-beta/configure	2016-02-12 14:10:10.267768256 +0100
@@ -4346,6 +4346,9 @@ if [ "$QMAKESPEC" != "$XQMAKESPEC" ]; th
 else
     # not cross compiling, host == target
     CFG_HOST_ARCH="$CFG_ARCH"
+    if [ "$CFG_ARCH" = "arm" ] ; then
+	CFG_CPUFEATURES="neon"
+    fi
     CFG_HOST_CPUFEATURES="$CFG_CPUFEATURES"
 fi
 unset OUTFILE

