diff -Naur rrdtool-1.7.1/etc/Makefile.am rrdtool-1.7.1.makefile/etc/Makefile.am
--- rrdtool-1.7.1/etc/Makefile.am	2019-02-04 11:39:15.000000000 -0600
+++ rrdtool-1.7.1.makefile/etc/Makefile.am	2019-02-19 10:27:41.207065102 -0600
@@ -3,6 +3,6 @@
 
 if HAVE_SYSTEMD
 systemdsystemunit_DATA = \
-	etc/rrdcached.socket \
-	etc/rrdcached.service
-endif
\ No newline at end of file
+	rrdcached.socket \
+	rrdcached.service
+endif
