--- htdig-3.2.0b5/htword/WordDBPage.cc.overflow	2003-06-24 21:57:27.000000000 +0200
+++ htdig-3.2.0b5/htword/WordDBPage.cc	2004-01-13 16:44:03.182583929 +0100
@@ -82,7 +82,7 @@
   	if(debuglevel>2)printf("TOTAL SIZE: %6d %8f\n",size,size/8.0);
 	// argh! compare failed somthing went wrong
 	// display the compress/decompress sequence and fail
-	if(cmp || size>8*1024*1000000000)
+	if(cmp || size>8*1024)
 	{
 	    if(size>8*1024)
 	    {
