--- htdig-3.2.0b6/htdig/ExternalParser.cc.orig	2004-05-28 08:15:14.000000000 -0500
+++ htdig-3.0.0b6/htdig/ExternalParser.cc	2005-09-15 11:41:05.000000000 -0500
@@ -227,7 +227,7 @@ ExternalParser::parse(Retriever &retriev
     int		get_file = (convertToType.length() != 0);
     String	newcontent;
 
-    StringList	cpargs(currentParser);
+    StringList	cpargs(currentParser, " \t");
     char   **parsargs = new char * [cpargs.Count() + 5];
     int    argi;
     for (argi = 0; argi < cpargs.Count(); argi++)
@@ -280,7 +280,11 @@ ExternalParser::parse(Retriever &retriev
 	// Call External Parser
 	execv(parsargs[0], parsargs);
 
-	exit(EXIT_FAILURE);
+	perror("execv");
+	write(STDERR_FILENO, "External parser error: Can't execute ", 37);
+	write(STDERR_FILENO, parsargs[0], strlen(parsargs[0]));
+	write(STDERR_FILENO, "\n", 1);
+	_exit(EXIT_FAILURE);
     }
 
     // Parent Process
@@ -424,7 +428,7 @@ ExternalParser::parse(Retriever &retriev
 		  {
 			metadatetags = new StringMatch();
 			metadatetags->IgnoreCase();
-			metadatetags->Pattern("date|dc.date|dc.date.created|dc.data.modified");
+			metadatetags->Pattern("date|dc.date|dc.date.created|dc.date.modified");
 		  }
     
 		  // <URL:http://www.w3.org/MarkUp/html-spec/html-spec_5.html#SEC5.2.5> 
