--- htdig-3.2.0b6/htfuzzy/Soundex.cc.keybug	2004-05-28 08:15:20.000000000 -0500
+++ htdig-3.2.0b6/htfuzzy/Soundex.cc	2008-02-27 16:04:51.000000000 -0600
@@ -153,6 +153,9 @@ Soundex::addWord(char *word)
 
     String	key;
     generateKey(word, key);
+    if (key.length() == 0 || mystrcasecmp(word, key.get()) == 0 ||
+				mystrcasecmp("0", key.get()) == 0) 
+	return;
 
     String	*s = (String *) dict->Find(key);
     if (s)
