--- a/languages.toml	2024-01-23 23:18:14.588275452 -0800
+++ b/languages.toml	2024-01-23 23:18:33.152880173 -0800
@@ -2998,7 +2998,7 @@
 
 [[grammar]]
 name = "gemini"
-source = { git = "https://git.sr.ht/~sfr/tree-sitter-gemini", rev = "3cc5e4bdf572d5df4277fc2e54d6299bd59a54b3" }
+source = { git = "https://git.sr.ht/~nbsp/tree-sitter-gemini", rev = "3cc5e4bdf572d5df4277fc2e54d6299bd59a54b3" }
 
 [[language]]
 name = "templ"
