--- a/runtime/grammars/sources/rescript/src/scanner.c	2024-01-23 23:36:15.736637895 -0800
+++ b/runtime/grammars/sources/rescript/src/scanner.c	2024-01-23 23:36:25.398954612 -0800
@@ -131,7 +131,7 @@
     const bool* valid_symbols
     ) {
   ScannerState* state = (ScannerState*)payload;
-  const in_string = state->in_quotes || state->in_backticks;
+  const bool in_string = state->in_quotes || state->in_backticks;
 
   if (valid_symbols[TEMPLATE_CHARS]) {
     lexer->result_symbol = TEMPLATE_CHARS;
--- a/runtime/grammars/sources/d/src/scanner.c	2024-01-23 23:37:27.176979712 -0800
+++ b/runtime/grammars/sources/d/src/scanner.c	2024-01-23 23:48:08.320623391 -0800
@@ -7,6 +7,7 @@
  * (See accompanying file LICENSE.txt or https://opensource.org/licenses/MIT)
  * SPDX-License-Identifier: MIT
  */
+#define _XOPEN_SOURCE 700
 #include "tree_sitter/parser.h"
 #include <assert.h>
 #include <ctype.h>
