diff -up xmms-1.2.11-20071117cvs/Input/mikmod/drv_xmms.c.fix xmms-1.2.11-20071117cvs/Input/mikmod/drv_xmms.c
--- xmms-1.2.11-20071117cvs/Input/mikmod/drv_xmms.c.fix	2014-02-11 10:33:47.944425895 -0500
+++ xmms-1.2.11-20071117cvs/Input/mikmod/drv_xmms.c	2014-02-11 10:34:39.000359857 -0500
@@ -124,7 +124,15 @@ MDRIVER drv_xmms =
 	"xmms output driver v1.0",
 	0, 255,
 #if (LIBMIKMOD_VERSION > 0x030106)
+/* Alias string */
         "xmms",
+#endif
+#if (LIBMIKMOD_VERSION >= 0x030200)
+/* CmdLineHelp string */
+        NULL,
+#endif
+#if (LIBMIKMOD_VERSION > 0x030106)
+/* CommandLine proc */
         NULL,
 #endif
         xmms_IsThere, 
