From 77575ee2afbe1ded0f0064bc43f92f52ecad071d Mon Sep 17 00:00:00 2001
From: linsy king <linsy_king@sjtu.edu.cn>
Date: Mon, 19 Sep 2022 18:28:22 +0800
Subject: [PATCH] fix: use contentItem instead of Text

---
 plugin/contents/ui/config.qml | 24 ++++++++++++++++++------
 1 file changed, 18 insertions(+), 6 deletions(-)

diff --git a/plugin/contents/ui/config.qml b/plugin/contents/ui/config.qml
index d2cfc69..865f480 100644
--- a/plugin/contents/ui/config.qml
+++ b/plugin/contents/ui/config.qml
@@ -130,16 +130,28 @@ ColumnLayout {
         id: bar
         implicitWidth: font.pixelSize*8 * 3
         PlasmaComponents.TabButton {
-            text: 'Wallpapers'
-            display: PlasmaComponents.TabButton.TextOnly
+            contentItem: Text {
+                text: "Wallpapers"
+                color: Theme.Theme.textColor
+                horizontalAlignment: Text.AlignHCenter
+                verticalAlignment: Text.AlignVCenter
+            }
         }
         PlasmaComponents.TabButton {
-            text: 'Settings'
-            display: PlasmaComponents.TabButton.TextOnly
+            contentItem: Text {
+                text: "Settings"
+                color: Theme.Theme.textColor
+                horizontalAlignment: Text.AlignHCenter
+                verticalAlignment: Text.AlignVCenter
+            }
         }
         PlasmaComponents.TabButton {
-            text: 'About'
-            display: PlasmaComponents.TabButton.TextOnly
+            contentItem: Text {
+                text: "About"
+                color: Theme.Theme.textColor
+                horizontalAlignment: Text.AlignHCenter
+                verticalAlignment: Text.AlignVCenter
+            }
         }
     }
 
-- 
2.39.0


