From 35318dde4cd2a40ffa9859ef6b474be22ada5bc6 Mon Sep 17 00:00:00 2001
From: Kay Lin <kaymw@aosc.io>
Date: Fri, 1 Jul 2022 19:04:00 -0700
Subject: [PATCH 1/2] dirs: use /var/lib/snapd/snap as host default snap mount
 directory

---
 dirs/dirs.go | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/dirs/dirs.go b/dirs/dirs.go
index 40e2525a7b..1d464b7f97 100644
--- a/dirs/dirs.go
+++ b/dirs/dirs.go
@@ -334,19 +334,8 @@ func SetRootDir(rootdir string) {
 	}
 	GlobalRootDir = rootdir
 
-	altDirDistros := []string{
-		"altlinux",
-		"antergos",
-		"arch",
-		"archlinux",
-		"fedora",
-		"gentoo",
-		"manjaro",
-		"manjaro-arm",
-	}
-
 	isInsideBase, _ := isInsideBaseSnap()
-	if !isInsideBase && release.DistroLike(altDirDistros...) {
+	if !isInsideBase {
 		SnapMountDir = filepath.Join(rootdir, "/var/lib/snapd/snap")
 	} else {
 		SnapMountDir = filepath.Join(rootdir, defaultSnapMountDir)
-- 
2.36.0

