From 3cc0f52f53bff7040a963a3f7a77a0b9ebe0a4fa Mon Sep 17 00:00:00 2001
From: Kay Lin <kaymw@aosc.io>
Date: Fri, 1 Jul 2022 19:06:30 -0700
Subject: [PATCH 2/2] dirs: use /usr/libexec as distro libexec path at default

---
 dirs/dirs.go | 28 +---------------------------
 1 file changed, 1 insertion(+), 27 deletions(-)

diff --git a/dirs/dirs.go b/dirs/dirs.go
index 1d464b7f97..0c8ae775d5 100644
--- a/dirs/dirs.go
+++ b/dirs/dirs.go
@@ -23,7 +23,6 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
-	"strconv"
 	"strings"
 
 	"github.com/snapcore/snapd/release"
@@ -429,32 +428,7 @@ func SetRootDir(rootdir string) {
 	LocaleDir = filepath.Join(rootdir, "/usr/share/locale")
 	ClassicDir = filepath.Join(rootdir, "/writable/classic")
 
-	opensuseTWWithLibexec := func() bool {
-		// XXX: this is pretty naive if openSUSE ever starts going back
-		// and forth about the change
-		if !release.DistroLike("opensuse-tumbleweed") {
-			return false
-		}
-		v, err := strconv.Atoi(release.ReleaseInfo.VersionID)
-		if err != nil {
-			// nothing we can do here
-			return false
-		}
-		// first seen on snapshot "20200826"
-		if v < 20200826 {
-			return false
-		}
-		return true
-	}
-
-	if release.DistroLike("fedora") || opensuseTWWithLibexec() {
-		// RHEL, CentOS, Fedora and derivatives, some more recent
-		// snapshots of openSUSE Tumbleweed;
-		// both RHEL and CentOS list "fedora" in ID_LIKE
-		DistroLibExecDir = filepath.Join(rootdir, "/usr/libexec/snapd")
-	} else {
-		DistroLibExecDir = filepath.Join(rootdir, "/usr/lib/snapd")
-	}
+	DistroLibExecDir = filepath.Join(rootdir, "/usr/libexec/snapd")
 
 	XdgRuntimeDirBase = filepath.Join(rootdir, "/run/user")
 	XdgRuntimeDirGlob = filepath.Join(XdgRuntimeDirBase, "*/")
-- 
2.36.0

