--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,9 +29,9 @@ if(buildcache_program)
   set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${buildcache_program}")
 endif()
 
-set(CMAKE_INSTALL_BINDIR bin CACHE PATH "directory for installing binary files")
-set(CMAKE_INSTALL_LIBDIR lib CACHE PATH "directory for installing library files")
-set(CMAKE_INSTALL_INCLUDEDIR include CACHE PATH "directory for installing header files")
+set(CMAKE_INSTALL_BINDIR bin CACHE STRING "directory for installing binary files")
+set(CMAKE_INSTALL_LIBDIR lib CACHE STRING "directory for installing library files")
+set(CMAKE_INSTALL_INCLUDEDIR include CACHE STRING "directory for installing header files")
 
 # Build lib3mf as shared by default, but do not implicitly switch third-party
 # dependencies (which often key off BUILD_SHARED_LIBS) to shared as well.
