From de8fc85d89ef659b44c7bb13a026d382a19c0ea3 Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Mon, 25 Aug 2025 14:29:42 +0800
Subject: [PATCH] fix std::find() undefine

---
 Source/Model/Writer/v100/NMR_ResourceDependencySorter.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Source/Model/Writer/v100/NMR_ResourceDependencySorter.cpp b/Source/Model/Writer/v100/NMR_ResourceDependencySorter.cpp
index b7e3967..4e5b42b 100644
--- a/Source/Model/Writer/v100/NMR_ResourceDependencySorter.cpp
+++ b/Source/Model/Writer/v100/NMR_ResourceDependencySorter.cpp
@@ -36,6 +36,7 @@ sorting them topologically.
 #include "Common/Graph/GraphAlgorithms.h"
 #include "Model/Classes/NMR_Model.h"
 #include "Model/Classes/NMR_ModelResource.h"
+#include <algorithm>
 
 namespace NMR
 {
-- 
2.40.1.windows.1

