diff --git a/HMCL/src/main/java/org/jackhuang/hmcl/ui/main/SettingsPage.java b/HMCL/src/main/java/org/jackhuang/hmcl/ui/main/SettingsPage.java
index ceff5dc8f..aede651dd 100644
--- a/HMCL/src/main/java/org/jackhuang/hmcl/ui/main/SettingsPage.java
+++ b/HMCL/src/main/java/org/jackhuang/hmcl/ui/main/SettingsPage.java
@@ -71,7 +71,6 @@ import static org.jackhuang.hmcl.util.logging.Logger.LOG;
 
 public final class SettingsPage extends ScrollPane {
     @SuppressWarnings("FieldCanBeLocal")
-    private final InvalidationListener updateListener;
 
     public SettingsPage() {
         this.setFitToWidth(true);
@@ -82,6 +81,7 @@ public final class SettingsPage extends ScrollPane {
         FXUtils.smoothScrolling(this);
 
         {
+            /*
             ComponentList updatePaneList = new ComponentList();
             {
                 ObjectProperty<UpdateChannel> updateChannel;
@@ -164,6 +164,7 @@ public final class SettingsPage extends ScrollPane {
 
                 rootPane.getChildren().addAll(ComponentList.createComponentListTitle(i18n("update")), updatePaneList);
             }
+            */
 
             {
                 ComponentList languagePaneList = new ComponentList();
