From 2924cce4df722031e4b2accba5ed19c9742619d1 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Sat, 28 Feb 2026 18:34:28 -0800
Subject: [PATCH 1/2] lua-shell: fix lua 5.4 dep detection on AOSC OS

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 lua-shell/meson.build | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lua-shell/meson.build b/lua-shell/meson.build
index fff53fbc..f72ad90e 100644
--- a/lua-shell/meson.build
+++ b/lua-shell/meson.build
@@ -6,7 +6,10 @@ config_h.set('BUILD_LUA_SHELL', '1')
 
 dep_lua = dependency('lua', version: '>= 5.4', required: false)
 if not dep_lua.found()
-	error('lua-shell requires Lua >= 5.4 which was not found. Or, you can use \'-Dshell-lua=false\'.')
+	dep_lua = dependency('lua5.4', version: '>= 5.4')
+	if not dep_lua.found()
+		error('lua-shell requires Lua >= 5.4 which was not found. Or, you can use \'-Dshell-lua=false\'.')
+	endif
 endif
 
 srcs_shell_lua = [
-- 
2.52.0

