--- a/src/cmd/ksh93/Makefile
+++ b/src/cmd/ksh93/Makefile
@@ -48,8 +48,8 @@
 SHOPT_SEVENBIT ==		/* strip the eigth bit from characters */
 SHOPT_SPAWN ==			/* use spawnveg for fork/exec */
 SHOPT_STATS ==	1		/* add .sh.stats variable */
-SHOPT_SUID_EXEC == 1		/* allow (safe) suid/sgid shell scripts */
-SHOPT_SYSRC ==			/* attempt . /etc/ksh.kshrc if interactive */
+SHOPT_SUID_EXEC == 0		/* allow (safe) suid/sgid shell scripts */
+SHOPT_SYSRC == 1		/* attempt . /etc/ksh.kshrc if interactive */
 SHOPT_TIMEOUT ==		/* number of seconds for shell timeout */
 SHOPT_TYPEDEF == 1		/* enable typeset type definitions */
 SHOPT_VSH == 1			/* vi edit mode */
