Description: Spelling mistake correction
 To keep lintian -I happy
Author: Nicholas Bamber <nicholas@periapt.co.uk>
Forwarded: no
Last-Update: 2015-10-29
--- a/src/lib/libcmd/head.c
+++ b/src/lib/libcmd/head.c
@@ -50,10 +50,10 @@
     "\anumber\a.]"
 "[n:lines?Copy \alines\a lines from each file.]#[lines:=10]"
 "[c:bytes?Copy \achars\a bytes from each file.]#[chars]"
-"[q:quiet|silent?Never ouput filename headers.]"
+"[q:quiet|silent?Never output filename headers.]"
 "[s:skip?Skip \askip\a characters or lines from each file before "
     "copying.]#[skip]"
-"[v:verbose?Always ouput filename headers.]"
+"[v:verbose?Always output filename headers.]"
     "\n\n"
 "[ file ... ]"
     "\n\n"
--- a/src/lib/libcmd/tail.c
+++ b/src/lib/libcmd/tail.c
@@ -86,7 +86,7 @@
 		"[+y?years]"
 		"[+S?scores]"
 	"}"
-"[v:verbose?Always ouput filename headers.]"
+"[v:verbose?Always output filename headers.]"
 
 "\n"
 "\n[file ...]\n"
--- a/src/cmd/ksh93/data/builtins.c
+++ b/src/cmd/ksh93/data/builtins.c
@@ -206,7 +206,7 @@
 	"invoked with a \b+\b the options will be written in a format "
 	"that can be reinput to the shell to restore the settings. "
 	"Options \b-o\b \aname\a can also be specified with \b--\b\aname\a "
-	"and \b+o \aname\a can be specifed with \b--no\b\aname\a  except that "
+	"and \b+o \aname\a can be specified with \b--no\b\aname\a  except that "
 	"options names beginning with \bno\b are turned on by omitting \bno\b."
 	"This option can be repeated to enable/disable multiple options. "
 	"The value of \aoption\a must be one of the following:]{"
