diff -Naur way-cooler-lock-0.2.1.old/src/pam/wrapper.c way-cooler-lock-0.2.1/src/pam/wrapper.c
--- way-cooler-lock-0.2.1.old/src/pam/wrapper.c 2017-12-11 02:00:56.000000000 +0000
+++ way-cooler-lock-0.2.1/src/pam/wrapper.c     2020-10-22 09:14:05.334106405 +0000
@@ -30,7 +30,7 @@
     if (!(responses[i].resp = malloc(len))) {
       goto error;
     }
-    strncpy(responses[i].resp, appdata_ptr, len);
+    strcpy(responses[i].resp, appdata_ptr);
   }
   *resp = responses;
   return PAM_SUCCESS;
