From c4cfe07e7efb2ae0c4c717848afabf6a5765d751 Mon Sep 17 00:00:00 2001
From: "Andreas B. Mundt" <andi@debian.org>
Date: Wed, 23 May 2018 00:00:00 +0000
Subject: [PATCH 4/4] DEBIAN: Add keyword to 'tilem2.desktop'

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 data/desktop/tilem2.desktop | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/desktop/tilem2.desktop b/data/desktop/tilem2.desktop
index 8118c97..86faa2b 100644
--- a/data/desktop/tilem2.desktop
+++ b/data/desktop/tilem2.desktop
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=TilEm
 Comment=Graphing calculator emulator
+Keywords=Texas Instruments;
 Exec=tilem2 %F
 Icon=tilem
 MimeType=application/x-tigroup;application/x-ti73-variables;application/x-ti73-program;application/x-ti73-backup;application/x-ti73-app;application/x-ti73-os;application/x-ti81-program;application/x-ti82-variables;application/x-ti82-program;application/x-ti82-backup;application/x-ti83-variables;application/x-ti83-program;application/x-ti83-backup;application/x-ti83plus-variables;application/x-ti83plus-program;application/x-ti83plus-app;application/x-ti83plus-os;application/x-ti85-variables;application/x-ti85-program;application/x-ti85-backup;application/x-ti86-variables;application/x-ti86-program;application/x-ti86-backup;
-- 
2.52.0

