From 70efbd3a1c18323c2e734e954d45067360935789 Mon Sep 17 00:00:00 2001
From: OriginCode <self@origincode.me>
Date: Sun, 12 Sep 2021 08:46:38 +0800
Subject: [PATCH] Add AOSC MC Custom MSA Client ID

---
 notsecrets/Secrets.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/notsecrets/Secrets.cpp b/notsecrets/Secrets.cpp
index 88995635..e7719009 100644
--- a/notsecrets/Secrets.cpp
+++ b/notsecrets/Secrets.cpp
@@ -28,7 +28,7 @@ namespace {
  * If you intend to base your own launcher on this code, take care and customize this to obfuscate the client ID, so it cannot be trivially found by casual attackers.
  */
 
-QString MSAClientID = "";
+QString MSAClientID = "81a207c0-a53c-46a3-be07-57d2b28c1643";
 }
 
 namespace Secrets {
-- 
2.30.2

