--- soxr-0.1.3-Source/CMakeLists.txt	2018-02-23 13:39:25.000000000 -0800
+++ soxr-0.1.3-Source.nostrip/CMakeLists.txt	2021-07-10 10:34:38.551483077 -0700
@@ -164,9 +164,6 @@
   # Can use std=c89, but gnu89 should give faster sinf, cosf, etc.:
   set (PROJECT_C_FLAGS "${PROJECT_CXX_FLAGS} \
        -std=gnu89 -Wnested-externs -Wmissing-prototypes -Wstrict-prototypes")
-  if (CMAKE_BUILD_TYPE STREQUAL "Release")
-    set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -s") # strip
-  endif ()
   cmake_dependent_option (VISIBILITY_HIDDEN
     "Build shared libraries with -fvisibility=hidden." ON
     "BUILD_SHARED_LIBS" OFF)
