diff -Naur linux-3.16.85/arch/mips/loongson/Platform linux-3.16.85.nommi/arch/mips/loongson/Platform
--- linux-3.16.85/arch/mips/loongson/Platform	2020-08-14 22:40:48.318806872 -0600
+++ linux-3.16.85.nommi/arch/mips/loongson/Platform	2020-08-14 22:40:09.910464556 -0600
@@ -8,6 +8,11 @@
 	$(call cc-option,-march=loongson2e,-march=r4600)
 cflags-$(CONFIG_CPU_LOONGSON2F) += \
 	$(call cc-option,-march=loongson2f,-march=r4600)
+
+# Some -march= flags enable MMI instructions, and GCC complains about that
+# support being enabled alongside -msoft-float. Thus explicitly disable MMI.
+cflags-y += $(call cc-option,-mno-loongson-mmi)
+
 # Enable the workarounds for Loongson2f
 ifdef CONFIG_CPU_LOONGSON2F_WORKAROUNDS
   ifeq ($(call as-option,-Wa$(comma)-mfix-loongson2f-nop,),)
