--- linux-3.16.85/arch/mips/loongson/common/cs5536/cs5536_ohci.c	2020-06-11 18:06:00.000000000 +0000
+++ linux-3.16.85.loongson/arch/mips/loongson/common/cs5536/cs5536_ohci.c	2020-08-15 04:56:30.991658167 +0000
@@ -138,8 +138,6 @@
 		break;
 	case PCI_OHCI_INT_REG:
 		_rdmsr(DIVIL_MSR_REG(PIC_YSEL_LOW), &hi, &lo);
-		if ((lo & 0x00000f00) == CS5536_USB_INTR)
-			conf_data = 1;
 		break;
 	default:
 		break;
