From 84ff2e2a9d5776036e28296acd29359bc5420720 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Tue, 8 Apr 2025 00:06:39 -0700
Subject: [PATCH] meson.build: make tests, code formatting and linting optional

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 meson.build       | 121 +++++++++++++++++++++++++---------------------
 meson_options.txt |  12 +++++
 2 files changed, 77 insertions(+), 56 deletions(-)

diff --git a/meson.build b/meson.build
index 9473274..eb782c5 100644
--- a/meson.build
+++ b/meson.build
@@ -13,6 +13,9 @@ ratbagd_api_version = 2
 
 # Options
 enable_runtime_dependency_checks = get_option('runtime-dependency-checks')
+enable_tests = get_option('tests')
+enable_formatting = get_option('format-code')
+enable_linting = get_option('lint-code')
 
 # Dependencies
 if enable_runtime_dependency_checks
@@ -117,61 +120,67 @@ else
     meson.add_install_script('meson_install.sh')
 endif
 
-env_test = environment()
-# See: https://github.com/mesonbuild/meson/issues/3306.
-env_test.set('MESON_SOURCE_ROOT', meson.current_source_dir())
-
-test_svg_files = find_program('tests/check-svg.py')
-test(
-  'check-svg',
-  test_svg_files,
-  args : join_paths(meson.current_source_dir(), 'data/svgs/'),
-  env : ['BASEDIR=@0@'.format(join_paths(meson.current_source_dir(), 'data/'))],
-)
-
-test(
-  'svg-lookup-check',
-  find_program('tests/svg-lookup-ini-test.py'),
-  args : [svg_mapping, join_paths(meson.current_source_dir(), 'data/svgs/')],
-)
-
-test(
-  'files-in-git',
-  find_program('tests/check-files-in-git.sh'),
-  args : [meson.current_source_dir()],
-  suite : ['all'],
-)
+if enable_tests
+    env_test = environment()
+    # See: https://github.com/mesonbuild/meson/issues/3306.
+    env_test.set('MESON_SOURCE_ROOT', meson.current_source_dir())
+
+    test_svg_files = find_program('tests/check-svg.py')
+    test(
+      'check-svg',
+      test_svg_files,
+      args : join_paths(meson.current_source_dir(), 'data/svgs/'),
+      env : ['BASEDIR=@0@'.format(join_paths(meson.current_source_dir(), 'data/'))],
+    )
+
+    test(
+      'svg-lookup-check',
+      find_program('tests/svg-lookup-ini-test.py'),
+      args : [svg_mapping, join_paths(meson.current_source_dir(), 'data/svgs/')],
+    )
+
+    test(
+      'files-in-git',
+      find_program('tests/check-files-in-git.sh'),
+      args : [meson.current_source_dir()],
+      suite : ['all'],
+    )
+endif
 
 #### code formatting #####
-run_target(
-  'python-black',
-  command : 'tools/python-black.sh',
-)
-
-python_black_check = find_program(join_paths(meson.current_source_dir(), 'tests/python-black-check.sh'))
-run_target(
-  'python-black-check',
-  command : python_black_check,
-)
-test(
-  'python-black-check',
-  python_black_check,
-  env: env_test,
-)
-
-#### code linting #####
-run_target(
-  'python-ruff',
-  command : 'tools/python-ruff.sh',
-)
-
-python_ruff_check = find_program(join_paths(meson.current_source_dir(), 'tests/python-ruff-check.sh'))
-run_target(
-  'python-ruff-check',
-  command : python_ruff_check,
-)
-test(
-  'python-ruff-check',
-  python_ruff_check,
-  env: env_test,
-)
+if enable_formatting
+    run_target(
+      'python-black',
+      command : 'tools/python-black.sh',
+    )
+
+    python_black_check = find_program(join_paths(meson.current_source_dir(), 'tests/python-black-check.sh'))
+    run_target(
+      'python-black-check',
+      command : python_black_check,
+    )
+    test(
+      'python-black-check',
+      python_black_check,
+      env: env_test,
+    )
+endif
+
+if enable_linting
+    #### code linting #####
+    run_target(
+      'python-ruff',
+      command : 'tools/python-ruff.sh',
+    )
+
+    python_ruff_check = find_program(join_paths(meson.current_source_dir(), 'tests/python-ruff-check.sh'))
+    run_target(
+      'python-ruff-check',
+      command : python_ruff_check,
+    )
+    test(
+      'python-ruff-check',
+      python_ruff_check,
+      env: env_test,
+    )
+endif
diff --git a/meson_options.txt b/meson_options.txt
index 594b3d6..992e37d 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -2,3 +2,15 @@ option('runtime-dependency-checks',
 	type: 'boolean',
 	value: true,
 	description: 'Check for runtime dependencies during installation (default=yes)')
+option('tests',
+	type: 'boolean',
+	value: false,
+	description: 'Run tests (default=no)')
+option('format-code',
+	type: 'boolean',
+	value: false,
+	description: 'Format code during building (default=no)')
+option('lint-code',
+	type: 'boolean',
+	value: false,
+	description: 'Lint code during building (default=no)')
-- 
2.49.0

