diff -Naur mate-tweak-19.10.4/layouts/contemporary.layout mate-tweak-19.10.4.aosc/layouts/contemporary.layout
--- mate-tweak-19.10.4/layouts/contemporary.layout	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/contemporary.layout	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1,70 @@
+[Toplevel top]
+expand=true
+orientation=top
+size=28
+
+[Toplevel bottom]
+expand=true
+orientation=bottom
+size=28
+
+[Object briskmenu]
+object-type=applet
+applet-iid=BriskMenuFactory::BriskMenu
+toplevel-id=top
+position=0
+locked=true
+
+[Object appmenuapplet]
+position=10
+toplevel-id=top
+applet-iid=AppmenuAppletFactory::AppmenuApplet
+object-type=applet
+panel-right-stick=false
+locked=true
+
+[Object notification-area]
+object-type=applet
+applet-iid=NotificationAreaAppletFactory::NotificationArea
+toplevel-id=top
+position=20
+panel-right-stick=true
+locked=true
+
+[Object clock]
+object-type=applet
+applet-iid=ClockAppletFactory::ClockApplet
+toplevel-id=top
+position=0
+panel-right-stick=true
+locked=true
+
+[Object show-desktop]
+object-type=applet
+applet-iid=WnckletFactory::ShowDesktopApplet
+toplevel-id=bottom
+position=0
+locked=true
+
+[Object window-list]
+object-type=applet
+applet-iid=WnckletFactory::WindowListApplet
+toplevel-id=bottom
+position=20
+locked=true
+
+[Object workspace-switcher]
+object-type=applet
+applet-iid=WnckletFactory::WorkspaceSwitcherApplet
+toplevel-id=bottom
+position=10
+panel-right-stick=true
+locked=true
+
+[Object trashapplet]
+locked=true
+position=0
+toplevel-id=bottom
+applet-iid=TrashAppletFactory::TrashApplet
+object-type=applet
+panel-right-stick=true
diff -Naur mate-tweak-19.10.4/layouts/eleven.dock mate-tweak-19.10.4.aosc/layouts/eleven.dock
--- mate-tweak-19.10.4/layouts/eleven.dock	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/eleven.dock	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1 @@
+plank
diff -Naur mate-tweak-19.10.4/layouts/eleven.layout mate-tweak-19.10.4.aosc/layouts/eleven.layout
--- mate-tweak-19.10.4/layouts/eleven.layout	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/eleven.layout	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1,34 @@
+[Toplevel top]
+expand=true
+orientation=top
+size=28
+
+[Object briskmenu]
+object-type=applet
+applet-iid=BriskMenuFactory::BriskMenu
+toplevel-id=top
+position=0
+locked=true
+
+[Object appmenuapplet]
+locked=true
+position=10
+toplevel-id=top
+applet-iid=AppmenuAppletFactory::AppmenuApplet
+object-type=applet
+
+[Object notification-area]
+object-type=applet
+applet-iid=NotificationAreaAppletFactory::NotificationArea
+toplevel-id=top
+position=20
+panel-right-stick=true
+locked=true
+
+[Object clock]
+object-type=applet
+applet-iid=ClockAppletFactory::ClockApplet
+toplevel-id=top
+position=0
+panel-right-stick=true
+locked=true
diff -Naur mate-tweak-19.10.4/layouts/familiar.layout mate-tweak-19.10.4.aosc/layouts/familiar.layout
--- mate-tweak-19.10.4/layouts/familiar.layout	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/familiar.layout	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1,70 @@
+[Toplevel top]
+expand=true
+orientation=top
+size=28
+
+[Toplevel bottom]
+expand=true
+orientation=bottom
+size=28
+
+[Object briskmenu]
+object-type=applet
+applet-iid=BriskMenuFactory::BriskMenu
+toplevel-id=top
+position=0
+locked=true
+
+[Object firefox]
+locked=true
+position=10
+launcher-location=/usr/share/applications/firefox.desktop
+toplevel-id=top
+object-type=launcher
+menu-path=applications:/
+
+[Object notification-area]
+object-type=applet
+applet-iid=NotificationAreaAppletFactory::NotificationArea
+toplevel-id=top
+position=20
+panel-right-stick=true
+locked=true
+
+[Object clock]
+object-type=applet
+applet-iid=ClockAppletFactory::ClockApplet
+toplevel-id=top
+position=0
+panel-right-stick=true
+locked=true
+
+[Object show-desktop]
+object-type=applet
+applet-iid=WnckletFactory::ShowDesktopApplet
+toplevel-id=bottom
+position=0
+locked=true
+
+[Object window-list]
+object-type=applet
+applet-iid=WnckletFactory::WindowListApplet
+toplevel-id=bottom
+position=20
+locked=true
+
+[Object workspace-switcher]
+object-type=applet
+applet-iid=WnckletFactory::WorkspaceSwitcherApplet
+toplevel-id=bottom
+position=10
+panel-right-stick=true
+locked=true
+
+[Object trashapplet]
+locked=true
+position=0
+toplevel-id=bottom
+applet-iid=TrashAppletFactory::TrashApplet
+object-type=applet
+panel-right-stick=true
diff -Naur mate-tweak-19.10.4/layouts/mutiny.layout mate-tweak-19.10.4.aosc/layouts/mutiny.layout
--- mate-tweak-19.10.4/layouts/mutiny.layout	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/mutiny.layout	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1,65 @@
+[Toplevel top]
+orientation=top
+expand=true
+size=28
+
+[Toplevel left]
+orientation=left
+expand=true
+size=66
+
+[Object windowbuttonsapplet]
+position=0
+toplevel-id=top
+applet-iid=WindowButtonsAppletFactory::WindowButtonsApplet
+object-type=applet
+panel-right-stick=false
+locked=true
+
+[Object appmenuapplet]
+position=64
+toplevel-id=top
+applet-iid=AppmenuAppletFactory::AppmenuApplet
+object-type=applet
+panel-right-stick=false
+locked=true
+
+[Object notification-area]
+object-type=applet
+applet-iid=NotificationAreaAppletFactory::NotificationArea
+toplevel-id=top
+position=20
+panel-right-stick=true
+locked=true
+
+[Object clock]
+object-type=applet
+applet-iid=ClockAppletFactory::ClockApplet
+toplevel-id=top
+position=0
+panel-right-stick=true
+locked=true
+
+[Object brisk-menu]
+locked=true
+position=0
+panel-right-stick=false
+toplevel-id=left
+applet-iid=BriskMenuFactory::BriskMenu
+object-type=applet
+
+[Object dockapplet]
+locked=true
+position=10
+panel-right-stick=false
+toplevel-id=left
+applet-iid=DockAppletFactory::DockApplet
+object-type=applet
+
+[Object trashapplet]
+locked=true
+position=0
+panel-right-stick=true
+toplevel-id=left
+applet-iid=TrashAppletFactory::TrashApplet
+object-type=applet
diff -Naur mate-tweak-19.10.4/layouts/mutiny.panel mate-tweak-19.10.4.aosc/layouts/mutiny.panel
--- mate-tweak-19.10.4/layouts/mutiny.panel	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/mutiny.panel	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1,89 @@
+[general]
+toplevel-id-list=['top', 'left']
+default-layout='mutiny'
+object-id-list=['windowbuttonsapplet', 'appmenuapplet', 'notification-area', 'clock', 'shutdown', 'brisk-menu', 'dockapplet', 'trashapplet']
+
+[toplevels/left]
+expand=true
+orientation='left'
+screen=0
+auto-hide=false
+size=66
+
+[toplevels/left/background]
+color='rgba(60,59,55,0.5)'
+type='color'
+
+[toplevels/top]
+expand=true
+orientation='top'
+screen=0
+size=28
+
+[objects/windowbuttonsapplet]
+applet-iid='WindowButtonsAppletFactory::WindowButtonsApplet'
+locked=true
+toplevel-id='top'
+position=0
+object-type='applet'
+panel-right-stick=false
+
+[objects/appmenuapplet]
+applet-iid='AppmenuAppletFactory::AppmenuApplet'
+locked=true
+toplevel-id='top'
+position=64
+object-type='applet'
+panel-right-stick=false
+
+[objects/notification-area]
+applet-iid='NotificationAreaAppletFactory::NotificationArea'
+locked=true
+toplevel-id='top'
+position=20
+object-type='applet'
+panel-right-stick=true
+
+[objects/clock]
+applet-iid='ClockAppletFactory::ClockApplet'
+locked=true
+toplevel-id='top'
+position=0
+object-type='applet'
+panel-right-stick=true
+
+[objects/brisk-menu]
+applet-iid='BriskMenuFactory::BriskMenu'
+locked=true
+toplevel-id='left'
+position=0
+object-type='applet'
+panel-right-stick=false
+
+[objects/dockapplet]
+applet-iid='DockAppletFactory::DockApplet'
+locked=true
+toplevel-id='left'
+position=10
+object-type='applet'
+panel-right-stick=false
+
+[objects/dockapplet/prefs]
+pinned-apps=['firefox.desktop', 'libreoffice-writer.desktop', 'libreoffice-calc.desktop', 'libreoffice-impress.desktop', 'matecc.desktop']
+first-run=false
+indicator-type=6
+app-spacing=4
+bg-type=1
+attention-type=1
+click-restore-last-active=true
+apps-from-all-workspaces=true
+multi-ind=true
+panel-act-list=true
+
+[objects/trashapplet]
+applet-iid='TrashAppletFactory::TrashApplet'
+locked=true
+toplevel-id='left'
+position=0
+object-type='applet'
+panel-right-stick=true
diff -Naur mate-tweak-19.10.4/layouts/netbook.layout mate-tweak-19.10.4.aosc/layouts/netbook.layout
--- mate-tweak-19.10.4/layouts/netbook.layout	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/netbook.layout	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1,44 @@
+[Toplevel top]
+orientation=top
+expand=true
+size=28
+
+[Object briskmenu]
+locked=true
+position=0
+panel-right-stick=false
+toplevel-id=top
+applet-iid=BriskMenuFactory::BriskMenu
+object-type=applet
+
+[Object dockapplet]
+locked=true
+position=10
+panel-right-stick=false
+toplevel-id=top
+applet-iid=DockAppletFactory::DockApplet
+object-type=applet
+
+[Object windowbuttonsapplet]
+locked=true
+position=30
+panel-right-stick=true
+toplevel-id=top
+applet-iid=WindowButtonsAppletFactory::WindowButtonsApplet
+object-type=applet
+
+[Object notificationarea]
+locked=true
+position=20
+panel-right-stick=true
+toplevel-id=top
+applet-iid=NotificationAreaAppletFactory::NotificationArea
+object-type=applet
+
+[Object clockapplet]
+locked=true
+position=0
+panel-right-stick=true
+toplevel-id=top
+applet-iid=ClockAppletFactory::ClockApplet
+object-type=applet
diff -Naur mate-tweak-19.10.4/layouts/netbook.panel mate-tweak-19.10.4.aosc/layouts/netbook.panel
--- mate-tweak-19.10.4/layouts/netbook.panel	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/netbook.panel	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1,65 @@
+[general]
+default-layout='netbook'
+object-id-list=['briskmenu', 'dockapplet', 'windowbuttonsapplet', 'notification-area', 'clockapplet']
+toplevel-id-list=['top']
+
+[toplevels/top]
+expand=true
+orientation='top'
+size=28
+
+[objects/briskmenu]
+applet-iid='BriskMenuFactory::BriskMenu'
+locked=true
+toplevel-id='top'
+position=0
+object-type='applet'
+panel-right-stick=false
+
+[objects/dockapplet]
+applet-iid='DockAppletFactory::DockApplet'
+locked=true
+toplevel-id='top'
+position=10
+object-type='applet'
+panel-right-stick=false
+
+[objects/dockapplet/prefs]
+pinned-apps=['firefox.desktop', 'matecc.desktop']
+first-run=false
+indicator-type=6
+app-spacing=4
+bg-type=1
+attention-type=1
+click-restore-last-active=true
+apps-from-all-workspaces=true
+multi-ind=true
+panel-act-list=true
+
+[objects/windowbuttonsapplet]
+applet-iid='WindowButtonsAppletFactory::WindowButtonsApplet'
+locked=true
+toplevel-id='top'
+position=30
+object-type='applet'
+panel-right-stick=true
+
+[objects/notification-area]
+applet-iid='NotificationAreaAppletFactory::NotificationArea'
+locked=true
+toplevel-id='top'
+position=20
+object-type='applet'
+panel-right-stick=true
+
+[objects/clockapplet]
+applet-iid='ClockAppletFactory::ClockApplet'
+locked=true
+toplevel-id='top'
+position=0
+object-type='applet'
+panel-right-stick=true
+
+[objects/clockapplet/prefs]
+format='24-hour'
+custom-format=''
diff -Naur mate-tweak-19.10.4/layouts/pantheon.dock mate-tweak-19.10.4.aosc/layouts/pantheon.dock
--- mate-tweak-19.10.4/layouts/pantheon.dock	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/pantheon.dock	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1 @@
+plank
diff -Naur mate-tweak-19.10.4/layouts/pantheon.layout mate-tweak-19.10.4.aosc/layouts/pantheon.layout
--- mate-tweak-19.10.4/layouts/pantheon.layout	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/pantheon.layout	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1,27 @@
+[Toplevel top]
+expand=true
+orientation=top
+size=28
+
+[Object briskmenu]
+object-type=applet
+applet-iid=BriskMenuFactory::BriskMenu
+toplevel-id=top
+position=0
+locked=true
+
+[Object notification-area]
+object-type=applet
+applet-iid=NotificationAreaAppletFactory::NotificationArea
+toplevel-id=top
+position=20
+panel-right-stick=true
+locked=true
+
+[Object clock]
+object-type=applet
+applet-iid=ClockAppletFactory::ClockApplet
+toplevel-id=top
+position=0
+panel-right-stick=true
+locked=true
diff -Naur mate-tweak-19.10.4/layouts/redmond.layout mate-tweak-19.10.4.aosc/layouts/redmond.layout
--- mate-tweak-19.10.4/layouts/redmond.layout	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/redmond.layout	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1,65 @@
+[Toplevel bottom]
+expand=true
+orientation=bottom
+size=28
+
+[Object mate-menu]
+locked=true
+position=0
+toplevel-id=bottom
+applet-iid=MateMenuAppletFactory::MateMenuApplet
+object-type=applet
+
+[Object showdesktopapplet]
+locked=true
+position=10
+toplevel-id=bottom
+applet-iid=WnckletFactory::ShowDesktopApplet
+object-type=applet
+
+[Object firefox]
+locked=true
+position=20
+launcher-location=/usr/share/applications/firefox.desktop
+toplevel-id=bottom
+object-type=launcher
+menu-path=applications:/
+
+[Object thunderbird]
+locked=true
+position=30
+launcher-location=/usr/share/applications/thunderbird.desktop
+toplevel-id=bottom
+object-type=launcher
+menu-path=applications:/
+
+[Object window-list]
+object-type=applet
+applet-iid=WnckletFactory::WindowListApplet
+toplevel-id=bottom
+position=40
+locked=true
+
+[Object drivemountapplet]
+object-type=applet
+applet-iid=DriveMountAppletFactory::DriveMountApplet
+toplevel-id=bottom
+position=30
+panel-right-stick=true
+locked=true
+
+[Object notification-area]
+object-type=applet
+applet-iid=NotificationAreaAppletFactory::NotificationArea
+toplevel-id=bottom
+position=20
+panel-right-stick=true
+locked=true
+
+[Object clock]
+object-type=applet
+applet-iid=ClockAppletFactory::ClockApplet
+toplevel-id=bottom
+position=0
+panel-right-stick=true
+locked=true
diff -Naur mate-tweak-19.10.4/layouts/ubuntu-mate.layout mate-tweak-19.10.4.aosc/layouts/ubuntu-mate.layout
--- mate-tweak-19.10.4/layouts/ubuntu-mate.layout	1970-01-01 00:00:00.000000000 +0000
+++ mate-tweak-19.10.4.aosc/layouts/ubuntu-mate.layout	2020-01-23 21:27:01.166749880 +0000
@@ -0,0 +1,69 @@
+[Toplevel top]
+expand=true
+orientation=top
+size=28
+
+[Toplevel bottom]
+expand=true
+orientation=bottom
+size=28
+
+[Object menu-bar]
+object-type=menu-bar
+toplevel-id=top
+position=0
+locked=true
+
+[Object firefox]
+locked=true
+position=10
+launcher-location=/usr/share/applications/firefox.desktop
+toplevel-id=top
+object-type=launcher
+menu-path=applications:/
+
+[Object notification-area]
+object-type=applet
+applet-iid=NotificationAreaAppletFactory::NotificationArea
+toplevel-id=top
+position=20
+panel-right-stick=true
+locked=true
+
+[Object clock]
+object-type=applet
+applet-iid=ClockAppletFactory::ClockApplet
+toplevel-id=top
+position=0
+panel-right-stick=true
+locked=true
+
+[Object show-desktop]
+object-type=applet
+applet-iid=WnckletFactory::ShowDesktopApplet
+toplevel-id=bottom
+position=0
+locked=true
+
+[Object window-list]
+object-type=applet
+applet-iid=WnckletFactory::WindowListApplet
+toplevel-id=bottom
+position=20
+locked=true
+
+[Object workspace-switcher]
+object-type=applet
+applet-iid=WnckletFactory::WorkspaceSwitcherApplet
+toplevel-id=bottom
+position=10
+panel-right-stick=true
+locked=true
+
+[Object trashapplet]
+locked=true
+position=0
+toplevel-id=bottom
+applet-iid=TrashAppletFactory::TrashApplet
+object-type=applet
+panel-right-stick=true
diff -Naur mate-tweak-19.10.4/mate-tweak mate-tweak-19.10.4.aosc/mate-tweak
--- mate-tweak-19.10.4/mate-tweak	2019-09-01 23:21:27.000000000 +0000
+++ mate-tweak-19.10.4.aosc/mate-tweak	2020-01-23 21:36:06.403446490 +0000
@@ -1149,7 +1149,6 @@
 
     def check_panel_features(self):
         # Determine what panel features are available
-        self.indicators_available = False
         self.mageia_cc_available = False
         self.mate_dock_available = False
         self.mate_menu_available = False
@@ -1159,16 +1158,6 @@
         self.brisk_menu_available = False
         self.appmenu_applet_available = False
 
-        if os.path.exists('/usr/lib/indicators3/7/libapplication.so') and \
-            os.path.exists('/usr/lib/' + self.multiarch + '/indicator-application/indicator-application-service') and \
-            os.path.exists('/usr/lib/' + self.multiarch + '/indicator-datetime/indicator-datetime-service') and \
-            os.path.exists('/usr/lib/' + self.multiarch + '/indicator-sound/indicator-sound-service') and \
-            os.path.exists('/usr/lib/' + self.multiarch + '/indicator-messages/indicator-messages-service') and \
-            os.path.exists('/usr/lib/' + self.multiarch + '/indicator-power/indicator-power-service') and \
-            os.path.exists('/usr/lib/' + self.multiarch + '/indicator-session/indicator-session-service') and \
-            os.path.exists('/usr/share/mate-panel/applets/org.ayatana.panel.IndicatorApplet.mate-panel-applet'):
-                self.indicators_available = True
-
         if os.path.exists('/usr/share/applications/mageia-drakconf.desktop'):
             self.mageia_cc_available = True
 
@@ -1188,7 +1177,7 @@
             self.appmenu_applet_available = True
 
         if os.path.exists('/usr/bin/mate-maximus') and \
-           os.path.exists('/usr/lib/mate-netbook/mate-window-picker-applet'):
+           os.path.exists('/usr/libexec/mate-window-picker-applet'):
             self.maximus_available = True
 
         if os.path.exists('/usr/lib/linuxmint/mintMenu/mintMenu.py'):
@@ -1234,36 +1223,28 @@
 
         if self.panel_layout_exists('contemporary') and \
            self.appmenu_applet_available and \
-           self.brisk_menu_available and \
-           self.indicators_available:
+           self.brisk_menu_available:
             self.add_to_panel_list(panels, "Contemporary", "contemporary")
 
-        # Prefer Indicator enabled Cupertino layout and fallback to non-Indicator version.
         if self.dock is not None and \
            self.appmenu_applet_available and \
-           self.brisk_menu_available and \
-           self.indicators_available:
+           self.brisk_menu_available:
             if self.panel_layout_exists('eleven'):
                 self.add_to_panel_list(panels, "Cupertino", "eleven")
-        elif self.dock is not None and \
-           self.appmenu_applet_available and \
-           self.brisk_menu_available and \
-           not self.indicators_available:
-            if self.panel_layout_exists('eleven-no-indicators'):
-                self.add_to_panel_list(panels, "Cupertino", "eleven-no-indicators")
 
         if self.panel_layout_exists('familiar') and \
-           self.brisk_menu_available and \
-           self.indicators_available:
+           self.brisk_menu_available:
             self.add_to_panel_list(panels, "Familiar", "familiar")
 
         if distro.id() != 'ubuntu' and \
             self.panel_layout_exists('fedora'):
             self.add_to_panel_list(panels, "Fedora", "fedora")
 
-        if self.panel_layout_exists('default') and \
-           not self.indicators_available:
-            self.add_to_panel_list(panels, "GNOME2", "default")
+        if self.panel_layout_exists('default'): \
+            panels.append([_("GNOME 2"), "default"])
+
+        if self.panel_layout_exists('aosc'): \
+            panels.append([_("AOSC OS"), "aosc"])
 
         if self.panel_layout_exists('linuxmint') and \
            self.mint_menu_available:
@@ -1277,57 +1258,35 @@
            self.brisk_menu_available:
             self.add_to_panel_list(panels, "Manjaro", "manjaro")
 
-        # Prefer Indicator enabled Mutiny layout and fallback to non-Indicator version.
         if self.panel_layout_exists('mutiny') and \
            self.mate_dock_available and \
            self.appmenu_applet_available and \
-           self.indicators_available and \
            self.brisk_menu_available:
             self.add_to_panel_list(panels, "Mutiny", "mutiny")
-        elif self.panel_layout_exists('mutiny-no-indicators') and \
-           self.mate_dock_available and \
-           self.appmenu_applet_available and \
-           not self.indicators_available:
-            self.add_to_panel_list(panels, "Mutiny", "mutiny-no-indicators")
 
-        # Prefer Indicator enabled Netbook layout and fallback to non-Indicator version.
         if self.panel_layout_exists('netbook') and \
            self.maximus_available and \
-           self.indicators_available and \
            self.brisk_menu_available and \
            self.mate_dock_available:
             self.add_to_panel_list(panels, "Netbook", "netbook")
-        elif self.panel_layout_exists('netbook-no-indicators') and \
-           self.maximus_available and \
-           not self.indicators_available:
-            self.add_to_panel_list(panels, "Netbook", "netbook-no-indicators")
 
-        if self.panel_layout_exists('opensuse') and \
-           not self.indicators_available:
+        if self.panel_layout_exists('opensuse'):
             self.add_to_panel_list(panels, "openSUSE", "opensuse")
 
         if self.dock is not None and \
-           self.brisk_menu_available and \
-           self.indicators_available:
+           self.brisk_menu_available:
             if self.panel_layout_exists('pantheon'):
                 self.add_to_panel_list(panels, "Pantheon", "pantheon")
 
-        # Prefer Indicator enabled Redmond layout and fallback to non-Indicator version.
         if self.panel_layout_exists('redmond') and \
-           self.indicators_available and \
            self.mate_menu_available:
             self.add_to_panel_list(panels, "Redmond", "redmond")
-        elif self.panel_layout_exists('redmond-no-indicators') and \
-           self.mate_menu_available and \
-           not self.indicators_available:
-            self.add_to_panel_list(panels, "Redmond", "redmond-no-indicators")
 
         if self.panel_layout_exists('solus') and \
            self.brisk_menu_available:
             self.add_to_panel_list(panels, "Solus", "solus")
 
-        if self.panel_layout_exists('ubuntu-mate') and \
-           self.indicators_available:
+        if self.panel_layout_exists('ubuntu-mate'):
             self.add_to_panel_list(panels, "Traditional", "ubuntu-mate")
 
         print("System installed layouts: ")
