diff -Naur mate-tweak-18.04.16/mate-tweak mate-tweak-18.04.16.aosc/mate-tweak
--- mate-tweak-18.04.16/mate-tweak	2018-03-28 23:12:44.000000000 +0000
+++ mate-tweak-18.04.16.aosc/mate-tweak	2018-04-12 01:32:29.777078645 +0000
@@ -1095,7 +1095,7 @@
             self.appmenu_applet_available = True
 
         if os.path.exists('/usr/bin/mate-maximus') and \
-           os.path.exists('/usr/lib/mate-netbook/mate-window-picker-applet'):
+           os.path.exists('/usr/libexec/mate-window-picker-applet'):
             self.maximus_available = True
 
         if os.path.exists('usr/lib/linuxmint/mintMenu/mintMenu.py'):
@@ -1174,7 +1174,10 @@
 
         if self.panel_layout_exists('default') and \
            not self.indicators_available:
-            panels.append([_("GNOME2"), "default"])
+            panels.append([_("GNOME 2"), "default"])
+
+        if self.panel_layout_exists('aosc'): \
+            panels.append([_("AOSC OS"), "aosc"])
 
         if self.panel_layout_exists('linuxmint') and \
            self.mint_menu_available:
