diff -Naur mate-tweak-19.10.4/marco-compton mate-tweak-19.10.4.compton/marco-compton
--- mate-tweak-19.10.4/marco-compton	2019-09-01 23:21:27.000000000 +0000
+++ mate-tweak-19.10.4.compton/marco-compton	2020-01-23 21:37:30.877057738 +0000
@@ -28,40 +28,5 @@
         --config ${HOME}/.config/${WM}-compton.conf &
 else
     compton \
-        --config /dev/null \
-        --backend ${BACKEND} \
-        --vsync ${VSYNC} \
-        --detect-rounded-corners \
-        --detect-client-leader \
-        --detect-transient \
-        --detect-client-opacity \
-        --paint-on-overlay \
-        --glx-no-stencil \
-        --glx-swap-method undefined \
-        --unredir-if-possible \
-        --unredir-if-possible-exclude "class_g = 'Mate-screensaver'" \
-        --inactive-opacity-override \
-        --mark-wmwin-focused \
-        --mark-ovredir-focused \
-        --use-ewmh-active-win \
-        -r 10 -o 0.225 -l -12 -t -12 \
-        -c -C -G \
-        --fading \
-        --fade-delta=4 \
-        --fade-in-step=0.03 \
-        --fade-out-step=0.03 \
-        --shadow-exclude "! name~=''" \
-        --shadow-exclude "name = 'Notification'" \
-        --shadow-exclude "name = 'Plank'" \
-        --shadow-exclude "name = 'Docky'" \
-        --shadow-exclude "name = 'Kupfer'" \
-        --shadow-exclude "name *= 'compton'" \
-        --shadow-exclude "class_g = 'albert'" \
-        --shadow-exclude "class_g = 'Conky'" \
-        --shadow-exclude "class_g = 'Kupfer'" \
-        --shadow-exclude "class_g = 'Synapse'" \
-        --shadow-exclude "class_g ?= 'Notify-osd'" \
-        --shadow-exclude "class_g ?= 'Cairo-dock'" \
-        --shadow-exclude "class_g = 'Cairo-clock'" \
-        --shadow-exclude "_GTK_FRAME_EXTENTS@:c" &
+        --config /etc/xdg/compton.conf &
 fi
