From 03865555b03b2258248e93be6aa99605fb2b5f80 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Wed, 6 Mar 2019 15:30:35 -0600
Subject: [PATCH] arch: adapt to current Core

---
 autobuild3/arch/_common.sh | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/autobuild3/arch/_common.sh b/autobuild3/arch/_common.sh
index 5deec22..f5a467a 100644
--- a/autobuild3/arch/_common.sh
+++ b/autobuild3/arch/_common.sh
@@ -2,7 +2,7 @@
 ##arch/_common.sh: Common arch defs for all ab arches, defines-mutable.
 ##@copyright GPL-2.0+
 AB_FLAGS_TYPES='_OPTI _ARCH _WEIRD '
-AB_FLAGS_FEATURES='LTO PERMISSIVE ANNOBIN '
+AB_FLAGS_FEATURES='LTO PERMISSIVE '
 # Naming, sadly. PORT_NOTICE!
 ARCH_TARGET['amd64']=x86_64-aosc-linux-gnu
 ARCH_TARGET['amd64/avx+']=x86_64-aosc-linux-gnu
@@ -28,8 +28,6 @@ ARCH_RPM['riscv64']=riscv64
 CFLAGS_COMMON='-fexceptions -pipe -Wno-error '
 CFLAGS_COMMON_OPTI='-O2 '
 CFLAGS_COMMON_DEBUG='-O0 '	# not that frequently used since autotools know it.
-CFLAGS_GCC="-fstack-clash-protection "
-CFLAGS_GCC_ANNOBIN="-fplugin=annobin "
 CFLAGS_GCC_OPTI="-fira-loop-pressure -fira-hoist-pressure -ftree-vectorize "
 CFLAGS_GCC_OPTI_LTO="-flto=$ABTHREADS "
 CFLAGS_GCC_DEBUG='-Og '		# note: not enabled for clang
-- 
2.18.1

