From 3ca2c57ae6791f21f29f41f933c9d3c1af4c7b3e Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Sat, 20 Apr 2019 14:16:02 -0500
Subject: [PATCH] arch: Core 5 compatibility

---
 arch/_common.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/autobuild3/arch/_common.sh b/autobuild3/arch/_common.sh
index 9d1c4e8..b7e5887 100644
--- a/autobuild3/arch/_common.sh
+++ b/autobuild3/arch/_common.sh
@@ -30,7 +30,7 @@ ARCH_RPM['i586']=i586
 CFLAGS_COMMON='-pipe -Wno-error '
 CFLAGS_COMMON_OPTI='-O2 '
 CFLAGS_COMMON_DEBUG='-O0 '	# not that frequently used since autotools know it.
-CFLAGS_GCC="-fstack-clash-protection "
+CFLAGS_GCC=" "
 CFLAGS_GCC_OPTI="-fira-loop-pressure -fira-hoist-pressure -ftree-vectorize "
 CFLAGS_GCC_OPTI_LTO="-flto=$ABTHREADS "
 CFLAGS_GCC_DEBUG='-Og '		# note: not enabled for clang
-- 
2.18.1

