From 09b784961fb7719ef6069c88378604c8118f8bc9 Mon Sep 17 00:00:00 2001
From: RawDiamondMC <RawDiamondMC@outlook.com>
Date: Wed, 31 Dec 2025 18:24:54 +0800
Subject: [PATCH] AOSCOS: remove ANSI C code to fix compliation

Signed-off-by: RawDiamondMC <RawDiamondMC@outlook.com>
---
 fileops.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fileops.c b/fileops.c
index 1e70af1..9017bd0 100644
--- a/fileops.c
+++ b/fileops.c
@@ -68,7 +68,6 @@ FILE *ftemp(char *templ, const char *mode)
 #else
     int fd = mkstemp(templ);
     if(fd >= 0) {
-        FILE* fdopen();                         /* in case -ansi is used */
         if(f = fdopen(fd, mode)) return f;
         close(fd);
 #endif
-- 
2.52.0

