From 78873ed78be6aed95f987feb298d04ef75d7c682 Mon Sep 17 00:00:00 2001
From: SkyBird233 <52884766+SkyBird233@users.noreply.github.com>
Date: Mon, 12 May 2025 15:47:18 +0800
Subject: [PATCH] fix: update apt-extracttemplates' path

---
 dpkg-preconfigure | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dpkg-preconfigure b/dpkg-preconfigure
index 6239603a..63a83786 100755
--- a/dpkg-preconfigure
+++ b/dpkg-preconfigure
@@ -140,8 +140,8 @@ elsif (! @debs) {
 }
 
 my $apt_extracttemplates;
-if (-x '/usr/lib/apt/apt-extracttemplates') {
-	$apt_extracttemplates = '/usr/lib/apt/apt-extracttemplates';
+if (-x '/usr/libexec/apt/apt-extracttemplates') {
+	$apt_extracttemplates = '/usr/libexec/apt/apt-extracttemplates';
 } elsif (Debconf::Path::find('apt-extracttemplates')) {
 	$apt_extracttemplates = 'apt-extracttemplates';
 } else {
-- 
2.49.0

