diff -Naur a/cinnamon-session/csm-logout-dialog.c b/cinnamon-session/csm-logout-dialog.c
--- a/cinnamon-session/csm-logout-dialog.c	2015-11-09 03:27:38.000000000 -0700
+++ b/cinnamon-session/csm-logout-dialog.c	2016-03-02 18:50:43.123146076 -0700
@@ -356,7 +356,6 @@
                 guint32             activate_time)
 {
         CsmLogoutDialog *logout_dialog;
-        GtkWidget       *dialog_image;
         GtkWidget       *hbox;
         const char      *primary_text;
         const char      *icon_name;
@@ -457,8 +456,6 @@
                 g_assert_not_reached ();
         }
         
-        dialog_image = gtk_message_dialog_get_image (GTK_MESSAGE_DIALOG (logout_dialog));
-        
         if (logout_dialog->priv->delay_toggle) {
             hbox = gtk_box_new (FALSE, 0);
             
@@ -471,8 +468,6 @@
             gtk_container_add (GTK_CONTAINER (gtk_dialog_get_content_area (GTK_DIALOG (logout_dialog))), hbox);
         }
 
-        gtk_image_set_from_icon_name (GTK_IMAGE (dialog_image),
-                                      icon_name, GTK_ICON_SIZE_DIALOG);
         gtk_window_set_icon_name (GTK_WINDOW (logout_dialog), icon_name);
         gtk_window_set_position (GTK_WINDOW (logout_dialog), GTK_WIN_POS_CENTER);
         gtk_message_dialog_set_markup (GTK_MESSAGE_DIALOG (logout_dialog), primary_text);
