--- a/sapi/fpm/php-fpm.conf.in	2015-12-01 06:36:27.000000000 -0700
+++ b/sapi/fpm/php-fpm.conf.in	2015-12-02 20:00:23.142854105 -0700
@@ -14,7 +14,7 @@
 ; Pid file
 ; Note: the default prefix is @EXPANDED_LOCALSTATEDIR@
 ; Default Value: none
-;pid = run/php-fpm.pid
+pid = /run/php-fpm/php-fpm.pid
 
 ; Error log file
 ; If it's set to "syslog", log is sent to syslogd instead of being written
@@ -122,4 +122,4 @@
 ; Relative path can also be used. They will be prefixed by:
 ;  - the global prefix if it's been set (-p argument)
 ;  - @prefix@ otherwise
-include=@php_fpm_sysconfdir@/php-fpm.d/*.conf
+include=@php_fpm_sysconfdir@/fpm.d/*.conf
