diff -Naur php-7.2.1/php.ini-production php-7.2.1.ini/php.ini-production
--- php-7.2.1/php.ini-production	2018-01-02 22:36:07.000000000 +0000
+++ php-7.2.1.ini/php.ini-production	2018-01-13 03:30:58.426133703 +0000
@@ -734,7 +734,7 @@
 
 ; Directory in which the loadable extensions (modules) reside.
 ; http://php.net/extension-dir
-; extension_dir = "./"
+extension_dir = "/usr/lib/php/modules/"
 ; On windows:
 ; extension_dir = "ext"
 
@@ -883,47 +883,47 @@
 ; deprecated in a future PHP major version. So, when it is possible, please
 ; move to the new ('extension=<ext>) syntax.
 ;
-; Notes for Windows environments :
-;
-; - ODBC support is built in, so no dll is needed for it.
-; - Many DLL files are located in the extensions/ (PHP 4) or ext/ (PHP 5+)
-;   extension folders as well as the separate PECL DLL download (PHP 5+).
-;   Be sure to appropriately set the extension_dir directive.
-;
-;extension=bz2
-;extension=curl
-;extension=fileinfo
-;extension=gd2
-;extension=gettext
-;extension=gmp
-;extension=intl
-;extension=imap
-;extension=interbase
-;extension=ldap
-;extension=mbstring
-;extension=exif      ; Must be after mbstring as it depends on it
-;extension=mysqli
-;extension=oci8_12c  ; Use with Oracle Database 12c Instant Client
-;extension=openssl
-;extension=pdo_firebird
-;extension=pdo_mysql
-;extension=pdo_oci
-;extension=pdo_odbc
-;extension=pdo_pgsql
-;extension=pdo_sqlite
-;extension=pgsql
-;extension=shmop
-
-; The MIBS data available in the PHP distribution must be installed.
-; See http://www.php.net/manual/en/snmp.installation.php
-;extension=snmp
-
-;extension=soap
-;extension=sockets
-;extension=sqlite3
-;extension=tidy
-;extension=xmlrpc
-;extension=xsl
+;extension=bcmath.so
+;extension=bz2.so
+;extension=calendar.so
+extension=curl.so
+;extension=dba.so
+;extension=enchant.so
+;extension=exif.so
+;extension=ftp.so
+;extension=gd.so
+;extension=gettext.so
+;extension=gmp.so
+;extension=iconv.so
+;extension=imap.so
+;extension=intl.so
+;extension=ldap.so
+;extension=mcrypt.so
+;extension=mysqli.so
+;extension=odbc.so
+;zend_extension=opcache.so
+;extension=openssl.so
+;extension=pdo_dblib.so
+;extension=pdo_mysql.so
+;extension=pdo_odbc.so
+;extension=pdo_pgsql.so
+;extension=pdo_sqlite.so
+;extension=pgsql.so
+;extension=phar.so
+;extension=posix.so
+;extension=pspell.so
+;extension=shmop.so
+;extension=snmp.so
+;extension=soap.so
+;extension=sockets.so
+;extension=sqlite3.so
+;extension=sysvmsg.so
+;extension=sysvsem.so
+;extension=sysvshm.so
+;extension=tidy.so
+;extension=xmlrpc.so
+;extension=xsl.so
+extension=zip.so
 
 ;;;;;;;;;;;;;;;;;;;
 ; Module Settings ;
