From b127120988f3d85ebc75c493eba08fc54c8b99dd Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Thu, 28 Dec 2023 22:11:42 -0800
Subject: [PATCH 12/22] 10_linux: do not count intel-ucode image as a valid
 initrd

---
 util/grub.d/10_linux.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index c9bde9182..e896c0c1a 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -263,7 +263,8 @@ for linux in ${reverse_sorted_list}; do
       initramfs=`grep CONFIG_INITRAMFS_SOURCE= "${config}" | cut -f2 -d= | tr -d \"`
   fi
 
-  if test -z "${initramfs}" && test -z "${initrd_real}" ; then
+  if ( test -z "${initramfs}" && test -z "${initrd_real}" ) || \
+    [[ "${initrd_real}" = "intel-ucode.img" ]]; then
     # "UUID=" and "ZFS=" magic is parsed by initrd or initramfs.  Since there's
     # no initrd or builtin initramfs, it can't work here.
     if [ "x${GRUB_DEVICE_PARTUUID}" = "x" ] \
-- 
2.43.4

