From f4bef0a70b24da1594b2957988578e01caaea3a1 Mon Sep 17 00:00:00 2001
From: "Lain \"Fearyncess\" Yang" <fsf@live.com>
Date: Mon, 28 Aug 2023 11:33:13 +0800
Subject: [PATCH 27/72] util/grub-mkimagexx: fix uncorrect section var in
 `make_reloc_section`

---
 util/grub-mkimagexx.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/util/grub-mkimagexx.c b/util/grub-mkimagexx.c
index 535fc9bb4..1ba5389db 100644
--- a/util/grub-mkimagexx.c
+++ b/util/grub-mkimagexx.c
@@ -2412,6 +2412,9 @@ make_reloc_section (Elf_Ehdr *e, struct grub_mkimage_layout *layout,
 			    smd->strtab + grub_le_to_cpu32 (s->sh_name));
 	    continue;
 	  }
+	symtab_section = (Elf_Shdr *) ((char *) smd->sections
+					+ (grub_target_to_host32 (s->sh_link)
+						* smd->section_entsize));
 
 	grub_util_info ("translating the relocation section %s",
 			smd->strtab + grub_le_to_cpu32 (s->sh_name));
-- 
2.52.0

