From cb1215b5499cb16020d9cca639d95ca0ba31b642 Mon Sep 17 00:00:00 2001
From: Henry Chen <chenx97@aosc.io>
Date: Mon, 7 Oct 2024 15:32:14 +0800
Subject: [PATCH 36/38] loader/mips64: fix warnings and style

---
 grub-core/loader/mips64/linux.c | 82 ++++++++++++++++++---------------
 1 file changed, 44 insertions(+), 38 deletions(-)

diff --git a/grub-core/loader/mips64/linux.c b/grub-core/loader/mips64/linux.c
index 514335e47..6b6cac273 100644
--- a/grub-core/loader/mips64/linux.c
+++ b/grub-core/loader/mips64/linux.c
@@ -124,11 +124,11 @@ grub_linux_boot (void)
 
   /* Boot the kernel.  */
   state.gpr[1] = entry_addr;
-  grub_dprintf("loongson", "entry_addr is %p\n", state.gpr[1]);
+  grub_dprintf("loongson", "entry_addr is %p\n", (void *) state.gpr[1]);
   state.gpr[4] = linux_argc;
-  grub_dprintf("loongson", "linux_argc is %d\n", state.gpr[4]);
+  grub_dprintf("loongson", "linux_argc is %d\n", (int) state.gpr[4]);
   state.gpr[5] = (grub_addr_t) linux_args_addr;
-  grub_dprintf("loongson", "args_addr is %p\n", state.gpr[5]);
+  grub_dprintf("loongson", "args_addr is %p\n", (void *) state.gpr[5]);
 
   if(grub_efi_is_loongson ())
   {
@@ -182,7 +182,7 @@ grub_linux_boot (void)
       grub_dprintf("loongson", "get old parameter interface\n");
     }
     state.gpr[6] = (grub_uint64_t)tmp_boot_params;
-    grub_dprintf("loongson", "boot_params is %p\n", state.gpr[6]);
+    grub_dprintf("loongson", "boot_params is %p\n", (void *) state.gpr[6]);
 
     mmap_size = find_mmap_size ();
     if (! mmap_size)
@@ -210,37 +210,43 @@ grub_linux_boot (void)
       {
         /* Recovery */
         if((lsdesc->type != GRUB_EFI_ACPI_RECLAIM_MEMORY) && \
-			(lsdesc->type != GRUB_EFI_ACPI_MEMORY_NVS) && \
-			(lsdesc->type != GRUB_EFI_RUNTIME_SERVICES_DATA) && \
-			(lsdesc->type != GRUB_EFI_RUNTIME_SERVICES_CODE) && \
-			(lsdesc->type != GRUB_EFI_RESERVED_MEMORY_TYPE) && \
-			(lsdesc->type != GRUB_EFI_PAL_CODE))
-	{
+	   (lsdesc->type != GRUB_EFI_ACPI_MEMORY_NVS) && \
+	   (lsdesc->type != GRUB_EFI_RUNTIME_SERVICES_DATA) && \
+	   (lsdesc->type != GRUB_EFI_RUNTIME_SERVICES_CODE) && \
+	   (lsdesc->type != GRUB_EFI_RESERVED_MEMORY_TYPE) && \
+	   (lsdesc->type != GRUB_EFI_PAL_CODE))
+	  {
+
+	    free_mem[free_index].memtype = GRUB_EFI_LOONGSON_SYSTEM_RAM_LOW;
+	    free_mem[free_index].memstart = (lsdesc->physical_start) & 0xffffffffffff;
+	    free_mem[free_index].memsize = lsdesc->num_pages * 4096;
+	    free_index++;
+	    /*ACPI*/
+	  }
+	else if(lsdesc->type == GRUB_EFI_ACPI_RECLAIM_MEMORY)
+	  {
+	    acpi_table_mem[acpi_table_index].memtype = GRUB_EFI_LOONGSON_ACPI_TABLE;
+	    acpi_table_mem[acpi_table_index].memstart = (lsdesc->physical_start) & 0xffffffffffff;
+	    acpi_table_mem[acpi_table_index].memsize = lsdesc->num_pages * 4096;
+	    acpi_table_index++;
 
-	  free_mem[free_index].memtype = GRUB_EFI_LOONGSON_SYSTEM_RAM_LOW;
-	  free_mem[free_index].memstart = (lsdesc->physical_start) & 0xffffffffffff;
-	  free_mem[free_index].memsize = lsdesc->num_pages * 4096;
-	  free_index++;
-	  /*ACPI*/
-	}else if((lsdesc->type == GRUB_EFI_ACPI_RECLAIM_MEMORY)){
-	  acpi_table_mem[acpi_table_index].memtype = GRUB_EFI_LOONGSON_ACPI_TABLE;
-	  acpi_table_mem[acpi_table_index].memstart = (lsdesc->physical_start) & 0xffffffffffff;
-	  acpi_table_mem[acpi_table_index].memsize = lsdesc->num_pages * 4096;
-	  acpi_table_index++;
-
-	}else if((lsdesc->type == GRUB_EFI_ACPI_MEMORY_NVS)){
-	  acpi_nvs_mem[acpi_nvs_index].memtype = GRUB_EFI_LOONGSON_ACPI_NVS;
-	  acpi_nvs_mem[acpi_nvs_index].memstart = (lsdesc->physical_start) & 0xffffffffffff;
-	  acpi_nvs_mem[acpi_nvs_index].memsize = lsdesc->num_pages * 4096;
-	  acpi_nvs_index++;
-
-	  /* Reserve */
-	}else{
-	  reserve_mem[reserve_index].memtype = GRUB_EFI_LOONGSON_MEMORY_RESERVED;
-	  reserve_mem[reserve_index].memstart = (lsdesc->physical_start) & 0xffffffffffff;
-	  reserve_mem[reserve_index].memsize = lsdesc->num_pages * 4096;
-	  reserve_index++;
-	}
+	  }
+	else if(lsdesc->type == GRUB_EFI_ACPI_MEMORY_NVS)
+	  {
+	    acpi_nvs_mem[acpi_nvs_index].memtype = GRUB_EFI_LOONGSON_ACPI_NVS;
+	    acpi_nvs_mem[acpi_nvs_index].memstart = (lsdesc->physical_start) & 0xffffffffffff;
+	    acpi_nvs_mem[acpi_nvs_index].memsize = lsdesc->num_pages * 4096;
+	    acpi_nvs_index++;
+
+	    /* Reserve */
+	  }
+	else
+	  {
+	    reserve_mem[reserve_index].memtype = GRUB_EFI_LOONGSON_MEMORY_RESERVED;
+	    reserve_mem[reserve_index].memstart = (lsdesc->physical_start) & 0xffffffffffff;
+	    reserve_mem[reserve_index].memsize = lsdesc->num_pages * 4096;
+	    reserve_index++;
+	  }
       }
 
       /* Recovery sort */
@@ -262,7 +268,7 @@ grub_linux_boot (void)
 	  new_interface_mem->map[tmp_index].memtype = GRUB_EFI_LOONGSON_SYSTEM_RAM_LOW;
 	new_interface_mem->map[tmp_index].memstart = free_mem[j].memstart;
 	new_interface_mem->map[tmp_index].memsize = tempMemsize;
-	grub_dprintf("loongson", "map[%d]:type %x, start 0x%llx, end 0x%llx\n",
+	grub_dprintf("loongson", "map[%d]:type %x, start 0x%lx, end 0x%lx\n",
 			tmp_index,
 			new_interface_mem->map[tmp_index].memtype,
 			new_interface_mem->map[tmp_index].memstart,
@@ -281,7 +287,7 @@ grub_linux_boot (void)
       new_interface_mem->mapcount = tmp_index;
       new_interface_mem->header.checksum = 0;
 
-      checksum = grub_efi_loongson_grub_calculatechecksum8(new_interface_mem, new_interface_mem->header.length);
+      checksum = grub_efi_loongson_grub_calculatechecksum8((const grub_uint8_t *) new_interface_mem, new_interface_mem->header.length);
       new_interface_mem->header.checksum = checksum;
     }
   }
@@ -346,10 +352,10 @@ grub_linux_load64 (grub_elf_t elf, const char *filename)
 
   /* Linux's entry point incorrectly contains a virtual address.  */
   entry_addr = elf->ehdr.ehdr64.e_entry;
-  grub_dprintf("loongson", "entry address = %p\n", entry_addr);
+  grub_dprintf("loongson", "entry address = %p\n", (void *) entry_addr);
 
   linux_size = grub_elf64_size (elf, &base, 0);
-  grub_dprintf("loongson", "base = %p\n", base);
+  grub_dprintf("loongson", "base = %p\n", (void *) base);
 
   if (linux_size == 0)
     return grub_errno;
-- 
2.47.0

