diff -Naur grub-2.12-rc1/util/grub-mkrescue.c grub-2.12-rc1.iso-level/util/grub-mkrescue.c
--- grub-2.12-rc1/util/grub-mkrescue.c	2023-09-19 07:56:32.823829975 +0000
+++ grub-2.12-rc1.iso-level/util/grub-mkrescue.c	2023-09-19 18:09:26.979596573 +0000
@@ -514,6 +514,8 @@
   xorriso_push ("-as");
   xorriso_push ("mkisofs");
   xorriso_push ("-graft-points");
+  xorriso_push ("-iso-level");
+  xorriso_push ("3");

   iso9660_dir = grub_util_make_temporary_dir ();
   grub_util_info ("temporary iso9660 dir is `%s'", iso9660_dir);
