From c06dea59cf9c10387c7fccd2fae532376d0ad414 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Sat, 16 Nov 2024 21:50:12 +0800
Subject: [PATCH 40/72] gitignore: reconsider PO files and grub.pot template

---
 .gitignore | 1 -
 1 file changed, 1 deletion(-)

diff --git a/.gitignore b/.gitignore
index db16295ad..f586577f9 100644
--- a/.gitignore
+++ b/.gitignore
@@ -256,7 +256,6 @@ widthspec.bin
 /po/POTFILES-shell.in
 /po/POTFILES.in
 /po/Rules-quot
-/po/grub.pot
 /po/remove-potcdate.sed
 /po/stamp-po
 /printf_test
-- 
2.52.0

