From d25128d6f3944aca66bdd8c3e9688c9b84a266dc Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Sat, 16 Nov 2024 21:50:12 +0800
Subject: [PATCH 40/68] gitignore: reconsider PO files and grub.pot template

---
 .gitignore | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/.gitignore b/.gitignore
index 524f2e6d0..3724645a4 100644
--- a/.gitignore
+++ b/.gitignore
@@ -249,8 +249,6 @@ widthspec.bin
 /pata_test
 /po/*.gmo
 /po/*.mo
-/po/*.po
-/po/LINGUAS
 /po/Makefile.in.in
 /po/Makevars
 /po/Makevars.template
@@ -258,7 +256,6 @@ widthspec.bin
 /po/POTFILES-shell.in
 /po/POTFILES.in
 /po/Rules-quot
-/po/grub.pot
 /po/remove-potcdate.sed
 /po/stamp-po
 /printf_test
-- 
2.51.1

