From 8f6d580d52d02b1187a98dab69b2635a5d6df187 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Sat, 16 Nov 2024 21:50:12 +0800
Subject: [PATCH 42/59] gitignore: reconsider PO files and grub.pot template

---
 .gitignore | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/.gitignore b/.gitignore
index 4d0dfb700..7523b9895 100644
--- a/.gitignore
+++ b/.gitignore
@@ -245,8 +245,6 @@ widthspec.bin
 /pata_test
 /po/*.gmo
 /po/*.mo
-/po/*.po
-/po/LINGUAS
 /po/Makefile.in.in
 /po/Makevars
 /po/Makevars.template
@@ -254,7 +252,6 @@ widthspec.bin
 /po/POTFILES-shell.in
 /po/POTFILES.in
 /po/Rules-quot
-/po/grub.pot
 /po/remove-potcdate.sed
 /po/stamp-po
 /printf_test
-- 
2.51.0

