From 8ca03b12cb6ae6fe378f15b1fbcad187d49c6efe Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Wed, 13 Aug 2025 09:26:51 +0800
Subject: [PATCH 51/57] 10_linux: lower the priority of RC kernels

---
 util/grub.d/10_linux.in | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 10df977cd..c24aec951 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -245,7 +245,11 @@ prioritize() {
 	for k in "${kernels[@]}" ; do
 		# This system probably don't have other page sizes.
 		if [ "${k%%-[0-9]*k}" = "$k" ] ; then
-			newkernels=("$k" "${newkernels[@]}")
+			if [ "${k%%-rc}" != "$k" ] ; then
+				newkernels+=("$k")
+			else
+				newkernels=("$k" "${newkernels[@]}")
+			fi
 			continue
 		fi
 		if [ "${k%%-$default_pgsize}" != "$k" ] ; then
-- 
2.50.1

