From bad4f1ec2a3fc23d4c6ee5fef5c2dec0844fc027 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Wed, 13 Aug 2025 09:28:17 +0800
Subject: [PATCH 52/57] 10_linux: fix the path of early initramfses (e.g.
 microcodes).

---
 util/grub.d/10_linux.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index c24aec951..85835e027 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -82,7 +82,7 @@ fi
 for rd in ${GRUB_EARLY_INITRD_LINUX_STOCK[@]} \
 	${GRUB_EARLY_INITRD_LINUX_CUSTOM[@]} ; do
 	if [ -e "$KERNEL_DIR"/"$rd" ] ; then
-		INITRD_EARLY="${INITRD_EARLY} ${rd}"
+		INITRD_EARLY="${INITRD_EARLY} ${KERNEL_DIR}/${rd}"
 	fi
 done
 # Returns the default page size suffix.
-- 
2.50.1

