From 0eb104c0e22760c382b380c462fb000c7ebb12ce Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Sat, 6 Sep 2025 00:17:44 +0800
Subject: [PATCH 52/78] 10_linux: show all kernels in the submenu

---
 util/grub.d/10_linux.in | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 46d8b9980..c3b5497ff 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -410,14 +410,13 @@ process_kernels() {
 			args="$ARGS"
 		fi
 		linux_entry "$title" "${GRUB_KERNEL_DIR}${kernel}" "$initrd" "$args" "${OS_ID}-${kernel}"
-		unset KERNELS["$kernel"]
 	done
 	# Now we processed the latest versions of the kernel, time to put
 	# other kernel versions into the submenu.
 	if [ "${#KERNELS[@]}" = "0" ] ; then
 		return
 	fi
-	submenu_title="$(gettext_printf "More kernel options for %s" "$OS")"
+	submenu_title="$(gettext_printf "All kernel options for %s" "$OS")"
 	# Sort other versions. However we can't prioritize them, it will
 	# mess up the order.
 	printf "submenu '%s' {\n" "$submenu_title"
-- 
2.52.0

