From 954f4f97a5d7826dc3593a1d84dec8f382a75636 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Wed, 13 Aug 2025 09:35:38 +0800
Subject: [PATCH 53/57] 10_linux: add missing `savedefault' to save the current
 selection

---
 util/grub.d/10_linux.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 85835e027..7d6150d3a 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -15,6 +15,7 @@ KERNEL_VERSIONS=()
 OS=$(gettext_printf "AOSC OS")
 OS_ID="aosc"
 KERNEL_DIR=${KERNEL_DIR:-/boot}
+SAVEDEFAULT=
 
 shopt -s extglob
 
@@ -78,6 +79,11 @@ else
 	_GFXPAYLOAD_LINUX="set gfxpayload=$GRUB_GFXPAYLOAD_LINUX"
 fi
 
+# `savedefault' for saving current selection upon boot.
+if [ "$GRUB_SAVEDEFAULT" = "true" ] ; then
+	SAVEDEFAULT="savedefault"
+fi
+
 # Handle microcodes.
 for rd in ${GRUB_EARLY_INITRD_LINUX_STOCK[@]} \
 	${GRUB_EARLY_INITRD_LINUX_CUSTOM[@]} ; do
@@ -185,6 +191,7 @@ linux_entry() {
 	printf "${indent}menuentry '%s' --class linux \$menuentry_id_option '%s' {\n" "$title" "$_id"
 	printf "$(echo "$MENUENTRY_PREP" | sed "s|^|\t${indent}|g")\n"
 	printf "${indent}\t$_GFXPAYLOAD_LINUX\n"
+	printf "${indent}\t$SAVEDEFAULT\n"
 	printf "${indent}\tinsmod	gzio\n"
 	printf "${indent}\techo	'%s'\n" "$(gettext_printf "Loading kernel $kernel ...")"
 	printf "${indent}\tlinux	%s %s\n" "$kernel" "$args"
-- 
2.50.1

