From ca4e8f14dd44fdcae1f441823ac079fdd9d8de53 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Wed, 13 Aug 2025 09:56:58 +0800
Subject: [PATCH 55/57] 10_linux: inform if no initrd is found for the kernel

---
 util/grub.d/10_linux.in | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 195590fb8..b5ac2b4c1 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -90,6 +90,7 @@ for rd in ${GRUB_EARLY_INITRD_LINUX_STOCK[@]} \
 	if [ -e "$KERNEL_DIR"/"$rd" ] ; then
 		INITRD_EARLY="${INITRD_EARLY} ${KERNEL_DIR}/${rd}"
 	fi
+	INITRD_EARLY="${INITRD_EARLY##[ 	]}"
 done
 # Returns the default page size suffix.
 # WARN $arch is the output of `uname -m`, which is not exactly the same as
@@ -297,14 +298,18 @@ get_initrd() {
 		initrd.img-"$1" \
 		initrd-"$1".gz ; do
 		if [ -e "$KERNEL_DIR/$rd" ] ; then
-			info "$(gettext_printf "Initial ramdisk image for kernel %s: %s" "$1" "${GRUB_KERNEL_DIR}${rd}")"
+			info "$(gettext_printf "Initial ramdisk image for kernel '%s': '%s'" "$1" "${GRUB_KERNEL_DIR}${rd}")"
 			ret="${ret} ${GRUB_KERNEL_DIR}${rd}"
 			break
 		else
 			continue
 		fi
 	done
-	printf "%s" "$ret"
+	if [ -n "$ret" ] ; then
+		printf "%s" "$ret"
+	else
+		info "$(gettext_printf "No initial ramdisk image found for kernel '%s'" "$1")"
+	fi
 }
 
 process_kernels() {
-- 
2.50.1

