From f18e95d1f7d690797035619772d39cfebec4c063 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Mon, 8 Sep 2025 16:53:09 +0800
Subject: [PATCH 57/70] 10_linux: add an arrow to the submenu entry

---
 util/grub.d/10_linux.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index bda28b435..704abcbaa 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -420,7 +420,7 @@ process_kernels() {
 	submenu_title="$(gettext_printf "All kernel options for %s" "$OS")"
 	# Sort other versions. However we can't prioritize them, it will
 	# mess up the order.
-	printf "submenu '%s' {\n" "$submenu_title"
+	printf "submenu '%s >>' {\n" "$submenu_title"
 	indent="$indent$tab"
 	for kernel in "${KERNELS[@]}" ; do
 		uname_r=${kernel/#@(vmlinu?|kernel)-/}
-- 
2.52.0

