From 0fe1e4e53122f71c874695c9af8a4401ddc1fd66 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Thu, 11 Sep 2025 15:35:46 +0800
Subject: [PATCH 59/78] 10_linux: support btrfs subvolumes for kernel/initrd
 paths

---
 util/grub.d/10_linux.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 00c6fe529..d46b835ab 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -69,6 +69,9 @@ if [ "$GRUB_DEVICE" == "$GRUB_DEVICE_BOOT" ] ; then
 else
 	GRUB_KERNEL_DIR="/"
 fi
+# Convert them into paths relative to the "actual" root, say, namespaces like
+# BtrFS subvolumes.
+GRUB_KERNEL_DIR="$(make_system_path_relative_to_its_root $GRUB_KERNEL_DIR)/"
 
 # search --fs-uuid commands.
 MENUENTRY_PREP="$(prepare_grub_to_access_device "$GRUB_DEVICE_BOOT")"
@@ -89,7 +92,7 @@ fi
 for rd in ${GRUB_EARLY_INITRD_LINUX_STOCK[@]} \
 	${GRUB_EARLY_INITRD_LINUX_CUSTOM[@]} ; do
 	if [ -e "$KERNEL_DIR"/"$rd" ] ; then
-		INITRD_EARLY="${INITRD_EARLY} ${KERNEL_DIR}/${rd}"
+		INITRD_EARLY="${INITRD_EARLY} ${GRUB_KERNEL_DIR}${rd}"
 	fi
 	INITRD_EARLY="${INITRD_EARLY##[ 	]}"
 done
-- 
2.52.0

