From 5476bfcd4a954ae5fd369a41a63b4c70323a8a0d Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Wed, 12 Nov 2025 15:24:19 +0800
Subject: [PATCH 75/78] 10_linux.in: Handle vanilla_rc kernel variant

---
 util/grub.d/10_linux.in | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index f27a09b1b..ba393094b 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -153,6 +153,9 @@ variant_str() {
 		lts)
 			gettext_printf "LTS Kernel"
 			;;
+		vanillarc)
+			gettext_printf "Vanilla RC Kernel"
+			;;
 		main)
 			: noop
 			;;
@@ -390,7 +393,7 @@ process_kernels() {
 	for ver in "${KERNEL_VERSIONS[@]}" ; do
 		kernels=($(echo "${KERNELS[*]}" | grep "$ver"))
 		for kernel in "${kernels[@]}" ; do
-			for variant in main lts rc ; do
+			for variant in main lts rc vanillarc ; do
 				# NOTE: Treat -main$ and -main- separately to
 				# handle partial name collisions.
 				if [ -z "${latest_variants[$variant]}" ] && (
@@ -405,7 +408,7 @@ process_kernels() {
 	# For each variant, add kernels with that version to the top level
 	# menu. This makes sure we see all configurations for this variant
 	# in the top level menu.
-	for variant in main lts rc ; do
+	for variant in main lts rc vanillarc ; do
 		if [ -z "${latest_variants[$variant]}" ] ; then
 			continue
 		fi
-- 
2.52.0

