diff --git a/util/grub-mkrescue.c b/util/grub-mkrescue.c
index ce2cbc4f1..b3108703f 100644
--- a/util/grub-mkrescue.c
+++ b/util/grub-mkrescue.c
@@ -52,6 +52,7 @@ static int xorriso_arg_alloc;
 static char **xorriso_argv;
 static char *iso_uuid;
 static char *iso9660_dir;
+static char *efidir;
 
 static void
 xorriso_push (const char *val)
@@ -268,11 +269,13 @@ make_image_abs (enum grub_install_plat plat,
 
   grub_install_push_module ("search");
   grub_install_push_module ("iso9660");
+  grub_install_push_module ("fat");
   grub_install_make_image_wrap (source_dirs[plat], "/boot/grub", output,
 				0, load_cfg,
 				mkimage_target, 0);
   grub_install_pop_module ();
   grub_install_pop_module ();
+  grub_install_pop_module ();
   grub_util_unlink (load_cfg);
 }
 
@@ -303,13 +306,19 @@ make_image_fwdisk_abs (enum grub_install_plat plat,
   load_cfg = grub_util_make_temporary_file ();
 
   load_cfg_f = grub_util_fopen (load_cfg, "wb");
+  fprintf (load_cfg_f, "search --file --set=root --hint=${root}, /boot/grub/grub.cfg\n");
+  fprintf (load_cfg_f, "set prefix=(${root})/boot/grub\n");
   write_part (load_cfg_f, source_dirs[plat]);
   fclose (load_cfg_f);
 
+  grub_install_push_module ("search");
   grub_install_push_module ("iso9660");
-  grub_install_make_image_wrap (source_dirs[plat], "()/boot/grub", output,
+  grub_install_push_module ("fat");
+  grub_install_make_image_wrap (source_dirs[plat], "/boot/grub", output,
 				0, load_cfg, mkimage_target, 0);
   grub_install_pop_module ();
+  grub_install_pop_module ();
+  grub_install_pop_module ();
   grub_util_unlink (load_cfg);
 }
 
@@ -741,7 +750,7 @@ main (int argc, char *argv[])
       || source_dirs[GRUB_INSTALL_PLATFORM_RISCV32_EFI]
       || source_dirs[GRUB_INSTALL_PLATFORM_RISCV64_EFI])
     {
-      char *efidir = grub_util_make_temporary_dir ();
+      efidir = grub_util_make_temporary_dir ();
       char *efidir_efi = grub_util_path_concat (2, efidir, "efi");
       char *efidir_efi_boot = grub_util_path_concat (3, efidir, "efi", "boot");
       char *imgname, *img32, *img64, *img_mac = NULL;
@@ -821,10 +830,8 @@ main (int argc, char *argv[])
       xorriso_push ("-efi-boot-part");
       xorriso_push ("--efi-boot-image");
 
-      grub_util_unlink_recursive (efidir);
       free (efiimgfat);
       free (efidir_efi);
-      free (efidir);
       grub_install_pop_module ();
       grub_install_pop_module ();
     }
@@ -969,6 +976,8 @@ main (int argc, char *argv[])
   xorriso_push (output_image);
   xorriso_push ("-r");
   xorriso_push (iso9660_dir);
+  if (efidir)
+    xorriso_push (efidir);
   xorriso_push ("--sort-weight");
   xorriso_push ("0");
   xorriso_push ("/");
@@ -986,6 +995,11 @@ main (int argc, char *argv[])
     grub_util_error ("`%s` invocation failed\n", "xorriso");
 
   grub_util_unlink_recursive (iso9660_dir);
+  if (efidir)
+    {
+      grub_util_unlink_recursive (efidir);
+      free(efidir);
+    }
 
   if (sysarea_img)
     grub_util_unlink (sysarea_img);
