From d43e19c47cee191687c1cad26f2f41858e50488c Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 17 Nov 2025 12:30:19 +0800
Subject: [PATCH 3/9] AOSCOS: fix(dkms.conf): drop unused options

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 dkms.conf | 2 --
 1 file changed, 2 deletions(-)

diff --git a/dkms.conf b/dkms.conf
index 80e9d9a..37d500c 100644
--- a/dkms.conf
+++ b/dkms.conf
@@ -1,8 +1,6 @@
 PACKAGE_NAME="arise"
 PACKAGE_VERSION=25.00.53
 AUTOINSTALL="yes"
-#REMAKE_INITRD="yes"
-SKIP_IMMD_MODPROBE=1
 
 BUILT_MODULE_NAME[0]="arise"
 BUILT_MODULE_LOCATION[0]=""
-- 
2.51.1

