diff --git a/core/e3k/vidsch/vidsch_setup_e3k.c b/core/e3k/vidsch/vidsch_setup_e3k.c
index 3728a69..7b7d227 100644
--- a/core/e3k/vidsch/vidsch_setup_e3k.c
+++ b/core/e3k/vidsch/vidsch_setup_e3k.c
@@ -207,7 +207,7 @@ static void vidschi_init_lookup_table_e3k(adapter_t  *adapter, vidsch_query_priv
     {
         query->pcie_segment_id         = SEGMENT_ID_LOCAL_E3K;
         query->local_segment_id        = SEGMENT_ID_LOCAL_E3K;
-#if defined(__mips64__) || defined(__loongarch__)
+#if defined(__mips64__)
         //PCIE cfg 0x68[14:12] defualt value is 010b, which means 512 Bytes maximum Read Request size
         //mips system BIOS(loonson PMON) modify this value to 000b, which means 128 Bytes maximum Read Request size
         //if fence buffer is in LOCAL, BIU will receive 8*256 bits Dummy Fence Read Request, which exceed 128 Bytes,
@@ -229,7 +229,7 @@ static void vidschi_init_lookup_table_e3k(adapter_t  *adapter, vidsch_query_priv
         }
         else
         {
-#if defined(__mips64__) || defined(__loongarch__)
+#if defined(__mips64__)
             //In mips or loongarch system, arise(1020,10C0), because of the HW issue, the FE/BE fence value may be not updated when fence buffer
             // is in pcie memory.
             query->fence_buffer_segment_id = SEGMENT_ID_LOCAL_E3K;
diff --git a/core/global/global.c b/core/global/global.c
index 90b5761..30fe72a 100644
--- a/core/global/global.c
+++ b/core/global/global.c
@@ -38,7 +38,7 @@ void glb_init_chip_id(adapter_t *adapter, krnl_adapter_init_info_t *info)
 
         adapter->hw_caps.support_snooping = TRUE;
         //from loongson, loonson 3A3000 not support snoop
-#if defined(__mips64__) || defined(__loongarch__)
+#if defined(__mips64__)
         adapter->hw_caps.support_snooping = FALSE;
 #endif
         adapter->hw_caps.page_4k_enable   = TRUE;
diff --git a/core/include/kernel_interface.h b/core/include/kernel_interface.h
index e12d152..9d82f7b 100644
--- a/core/include/kernel_interface.h
+++ b/core/include/kernel_interface.h
@@ -116,7 +116,7 @@ typedef struct
     void  (*init_adapter)(void *adp, int reserved_vmem, void *disp_info);
     void (*deinit_adapter)(void *data);
     void  (*get_adapter_info)(void* adp, adapter_info_t*  adapter_info);
-    void (*update_adapter_info)(void* adp, adapter_info_t*  adapter_info, krnl_adapter_init_info_t* a_info);
+    void (*update_adapter_info)(void* adp, adapter_info_t*  adapter_info, krnl_adapter_init_info_t* a_info, int force_enable_snoop);
     void (*dump_resource)(struct os_printer *p, void *data, int index, int iga_index);
     void (*debugfs_dump)(struct os_seq_file *seq_file, void *data, int type, void* arg);
     void (*final_cleanup)(void *data, unsigned int gpu_device);
diff --git a/core/kernel_interface.c b/core/kernel_interface.c
index ef9589a..ff14928 100644
--- a/core/kernel_interface.c
+++ b/core/kernel_interface.c
@@ -207,12 +207,16 @@ static  void krnl_get_adapter_info(void*  adp, adapter_info_t*  adapter_info)
     }
 }
 
-static  void  krnl_update_adapter_info(void* adp, adapter_info_t*  adapter_info, krnl_adapter_init_info_t* a_info)
+static  void  krnl_update_adapter_info(void* adp, adapter_info_t*  adapter_info, krnl_adapter_init_info_t* a_info, int force_enable_snoop)
 {
     adapter_t *adapter = (adapter_t *)adp;
     if(adapter && adapter_info)
     {
         adapter->hw_caps.snoop_only = (adapter_info->snoop_only)? 1 : 0;
+        if (force_enable_snoop)
+        {
+            adapter->hw_caps.snoop_only = 1;
+        }
         adapter->low_top_address = adapter_info->low_top_addr;
         adapter->hw_caps.miu_channel_num = adapter_info->chan_num;
         adapter->hw_caps.chip_slice_mask = adapter_info->chip_slice_mask;
diff --git a/linux/gf_disp.c b/linux/gf_disp.c
index 08424b2..9f9ae41 100644
--- a/linux/gf_disp.c
+++ b/linux/gf_disp.c
@@ -1471,7 +1471,7 @@ void disp_deinit_state_info(disp_info_t* disp_info)
     disp_info->state_info = NULL;
 }
 
-int  gf_init_modeset(struct drm_device *dev)
+int  gf_init_modeset(struct drm_device *dev, int force_enable_snoop)
 {
     gf_card_t*  gf_card = dev->dev_private;
     adapter_info_t* adapter_info = &gf_card->adapter_info;
@@ -1506,7 +1506,7 @@ int  gf_init_modeset(struct drm_device *dev)
 
     disp_cbios_get_slice_num(disp_info);
 
-    gf_core_interface->update_adapter_info(gf_card->adapter, adapter_info, a_info);
+    gf_core_interface->update_adapter_info(gf_card->adapter, adapter_info, a_info, force_enable_snoop);
 
     disp_cbios_get_crtc_resource(disp_info);
 
diff --git a/linux/gf_driver.c b/linux/gf_driver.c
index 832cb6d..da61d81 100644
--- a/linux/gf_driver.c
+++ b/linux/gf_driver.c
@@ -977,7 +977,7 @@ ATTRIBUTE_GROUPS(gf_hwmon);
 
 #endif
 
-int gf_card_init(gf_card_t *gf, void *pdev)
+int gf_card_init(gf_card_t *gf, void *pdev, int force_enable_snoop)
 {
     int ret = 0;
 
@@ -1003,11 +1003,11 @@ int gf_card_init(gf_card_t *gf, void *pdev)
     gf->cbios_flags = gf_modparams.gf_cbios_flags;
     if (gf_modparams.gf_virtual_display)
     {
-        gf_vkms_init_modeset(gf->drm_dev);
+        gf_vkms_init_modeset(gf->drm_dev, force_enable_snoop);
     }
     else
     {
-        gf_init_modeset(gf->drm_dev);
+        gf_init_modeset(gf->drm_dev, force_enable_snoop);
     }
 
     gf_disable_async_suspend(gf);
diff --git a/linux/gf_driver.h b/linux/gf_driver.h
index b129381..b1328a6 100644
--- a/linux/gf_driver.h
+++ b/linux/gf_driver.h
@@ -113,11 +113,11 @@ extern struct class *gf_class;
 extern struct drm_ioctl_desc gf_ioctls[];
 extern struct drm_ioctl_desc gf_ioctls_compat[];
 
-extern int  gf_card_init(gf_card_t *gf, void *pdev);
+extern int  gf_card_init(gf_card_t *gf, void *pdev, int force_enable_snoop);
 extern int  gf_card_deinit(gf_card_t *gf);
-extern int  gf_init_modeset(struct drm_device *dev);
+extern int  gf_init_modeset(struct drm_device *dev, int force_enable_snoop);
 extern void  gf_deinit_modeset(struct drm_device *dev);
-extern int gf_vkms_init_modeset(struct drm_device *ddev);
+extern int gf_vkms_init_modeset(struct drm_device *ddev, int force_enable_snoop);
 extern void gf_vkms_deinit_modeset(struct drm_device *ddev);
 
 extern int gf_debugfs_crtc_dump(struct seq_file* file, struct drm_device* dev, int index);
diff --git a/linux/gf_hw_null.c b/linux/gf_hw_null.c
index 4ef8dd0..4e3c42e 100644
--- a/linux/gf_hw_null.c
+++ b/linux/gf_hw_null.c
@@ -139,7 +139,7 @@ int gf_register_driver(void)
 
     gf_fb_mode = "640x480-32@60";
 
-    result = gf_card_init(gf, NULL);
+    result = gf_card_init(gf, NULL, FALSE);
     gf_dev = gf;
     return result;
 }
diff --git a/linux/gf_pcie.c b/linux/gf_pcie.c
index 19b30fa..cb6543d 100644
--- a/linux/gf_pcie.c
+++ b/linux/gf_pcie.c
@@ -371,7 +371,7 @@ void gf_selftest(gf_card_t *gf)
 }
 
 #define PCI_EN_IO_SPACE     1
-static int gf_drm_load_kms(struct drm_device *dev, unsigned long flags)
+static int gf_drm_load_kms(struct drm_device *dev, unsigned long flags, int force_enable_snoop)
 {
     struct pci_dev *pdev = to_pci_dev(dev->dev);
     struct device* device = &pdev->dev;
@@ -445,7 +445,7 @@ static int gf_drm_load_kms(struct drm_device *dev, unsigned long flags)
 
     gf_init_adapter_info_by_params(&gf->a_info, &gf_modparams);
     gf->a_info.minor_index = gf->index;
-    ret = gf_card_init(gf, pdev);
+    ret = gf_card_init(gf, pdev, force_enable_snoop);
     if(ret)
     {
         gf_error("%s_card_init() failed. ret:0x%x\n", STR(DRIVER_NAME), ret);
@@ -840,8 +840,26 @@ static int gf_pcie_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 #endif
 
     pci_set_drvdata(pdev, dev);
-
-    ret = gf_drm_load_kms(dev, flags);
+    int force_enable_snoop = FALSE;
+#if defined(__loongarch_lp64)
+    struct pci_bus * dev_bus = pdev->bus;
+    if (dev_bus)
+    {
+        struct pci_dev * parent_dev = dev_bus->self;
+        if (parent_dev)
+        {
+            gf_info("pci parent(vendor:0x%X, device:0x%X) \n", parent_dev->vendor, parent_dev->device);
+            // 3C6000/3B6000 pcie bridge device_id is 3c09 3c19 3c29
+            // 2K3000/3B6000M pcie bridge device_id is 7a99
+            if (parent_dev->vendor == PCI_VENDOR_ID_LOONGSON &&
+             ((parent_dev->device >= 0x3c00 && parent_dev->device <= 0x3cff) || (parent_dev->device == 0x7a99)))
+            {
+                force_enable_snoop = TRUE;
+            }
+        }
+    }
+#endif
+    ret = gf_drm_load_kms(dev, flags, force_enable_snoop);
     if (ret)
     {
         goto err_pci;
diff --git a/linux/gf_vkms.c b/linux/gf_vkms.c
index fe85434..d7c3727 100644
--- a/linux/gf_vkms.c
+++ b/linux/gf_vkms.c
@@ -516,13 +516,13 @@ static void gf_vkms_get_adapter_info(disp_info_t *disp_info)
     gf_core_interface->get_adapter_info(gf_card->adapter, disp_info->adp_info);
 }
 
-static void gf_vkms_update_adapter_info(disp_info_t *disp_info)
+static void gf_vkms_update_adapter_info(disp_info_t *disp_info, int force_enable_snoop)
 {
     gf_card_t *gf_card = (gf_card_t *)disp_info->gf_card;
     adapter_info_t* adapter_info = disp_info->adp_info;
     krnl_adapter_init_info_t* a_info = &gf_card->a_info;
 
-    gf_core_interface->update_adapter_info(gf_card->adapter, adapter_info, a_info);
+    gf_core_interface->update_adapter_info(gf_card->adapter, adapter_info, a_info, force_enable_snoop);
 
 }
 
@@ -555,7 +555,7 @@ static void gf_vkms_deinit_cbios(disp_info_t *disp_info)
     disp_info_deinit(disp_info);
 }
 
-int gf_vkms_init_modeset(struct drm_device *ddev)
+int gf_vkms_init_modeset(struct drm_device *ddev, int force_enable_snoop)
 {
     int ret, i;
     gf_card_t *gf_card = ddev->dev_private;
@@ -587,7 +587,7 @@ int gf_vkms_init_modeset(struct drm_device *ddev)
 
     gf_vkms_init_cbios(disp_info);
 
-    gf_vkms_update_adapter_info(disp_info);
+    gf_vkms_update_adapter_info(disp_info, force_enable_snoop);
 
     gf_vkms_mode_config_init(ddev);
 
@@ -675,7 +675,7 @@ void gf_vkms_deinit_modeset(struct drm_device *ddev)
 
 #else
 
-int gf_vkms_init_modeset(struct drm_device *ddev)
+int gf_vkms_init_modeset(struct drm_device *ddev, int force_enable_snoop)
 {
     gf_info("gf vkms only support drm version >= 4.19.0\n");
     return 0;
