From 56d9598e80c20eb5b8190c040910a00880311f33 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 15 Jul 2024 15:50:05 +0800
Subject: [PATCH 2/2] cargo: use default --release build settings

---
 Cargo.toml | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 2e8b947aa..9fddd588e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -480,19 +480,6 @@ ttf-parser = { opt-level = 3 }
 wasmtime-cranelift = { opt-level = 3 }
 wasmtime = { opt-level = 3 }
 
-[profile.release]
-debug = "limited"
-lto = "thin"
-codegen-units = 1
-
-[profile.release.package]
-zed = { codegen-units = 16 }
-
-[profile.release-fast]
-inherits = "release"
-lto = false
-codegen-units = 16
-
 [workspace.lints.clippy]
 dbg_macro = "deny"
 todo = "deny"
-- 
2.45.2

