From a173ec83f2d1704003502f84fbd392415356ed8a Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 15 Jul 2024 15:50:05 +0800
Subject: [PATCH] cargo: use default --release build settings

---
 Cargo.toml | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index f2f08506a1..743fc49c23 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -536,20 +536,6 @@ ttf-parser = { opt-level = 3 }
 wasmtime-cranelift = { opt-level = 3 }
 wasmtime = { opt-level = 3 }
 
-[profile.release]
-debug = "limited"
-lto = "thin"
-codegen-units = 1
-
-[profile.release.package]
-zed = { codegen-units = 16 }
-
-[profile.release-fast]
-inherits = "release"
-debug = "full"
-lto = false
-codegen-units = 16
-
 [workspace.lints.clippy]
 dbg_macro = "deny"
 todo = "deny"
-- 
2.46.0

