From 293bcf449d7de11dc992416653c436d53c4cb9ae Mon Sep 17 00:00:00 2001
From: Cyan <cyanoxygen@aosc.io>
Date: Thu, 29 Feb 2024 12:34:12 +0800
Subject: [PATCH] Makefile: use system libev and disable strip

---
 Makefile | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/Makefile b/Makefile
index 6e46d56..99f923a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,15 +1,11 @@
 CC = gcc
-CFLAGS = -std=c99 -Wall -Wextra -O2 -pthread
-LIBS = -lm
+CFLAGS = -std=c99 -Wall -Wextra -O2 -pthread -I/usr/include/libev
+LIBS = -lm -lev
 SRCS = ipt2socks.c lrucache.c netutils.c protocol.c
 OBJS = $(SRCS:.c=.o)
 MAIN = ipt2socks
 DESTDIR = /usr/local/bin
 
-EVCFLAGS = -O2 -fno-strict-aliasing
-EVSRCFILE = libev/ev.c
-EVOBJFILE = ev.o
-
 .PHONY: all install clean
 
 all: $(MAIN)
@@ -21,11 +17,8 @@ install: $(MAIN)
 clean:
 	$(RM) *.o $(MAIN)
 
-$(MAIN): $(EVOBJFILE) $(OBJS)
-	$(CC) $(CFLAGS) -s -o $(MAIN) $(OBJS) $(EVOBJFILE) $(LIBS)
+$(MAIN): $(OBJS)
+	$(CC) $(CFLAGS) -o $(MAIN) $(OBJS) $(LIBS)
 
 .c.o:
 	$(CC) $(CFLAGS) -c $< -o $@
-
-$(EVOBJFILE): $(EVSRCFILE)
-	$(CC) $(EVCFLAGS) -c $(EVSRCFILE) -o $(EVOBJFILE)
-- 
2.43.0

