--- a/Makefile	2021-01-21 19:00:42.742858732 +0800
+++ b/Makefile	2021-01-21 19:00:51.792305288 +0800
@@ -1,15 +1,11 @@
 CC = gcc
-CFLAGS = -std=c99 -Wall -Wextra -O2 -pthread
-LIBS = -lm
+CFLAGS = -std=c99 -Wall -Wextra -O2 -pthread -I/usr/include/libev
+LIBS = -lm -lev
 SRCS = ipt2socks.c lrucache.c netutils.c protocol.c
 OBJS = $(SRCS:.c=.o)
 MAIN = ipt2socks
 DESTDIR = /usr/local/bin
 
-EVCFLAGS = -w -O2
-EVSRCFILE = libev/ev.c
-EVOBJFILE = ev.o
-
 .PHONY: all install clean
 
 all: $(MAIN)
@@ -21,11 +17,8 @@
 clean:
 	$(RM) *.o $(MAIN)
 
-$(MAIN): $(EVOBJFILE) $(OBJS)
+$(MAIN): $(OBJS)
 	$(CC) $(CFLAGS) -s -o $(MAIN) $(OBJS) $(EVOBJFILE) $(LIBS)
 
 .c.o:
 	$(CC) $(CFLAGS) -c $< -o $@
-
-$(EVOBJFILE): $(EVSRCFILE)
-	$(CC) $(EVCFLAGS) -c $(EVSRCFILE) -o $(EVOBJFILE)
