From cf1236088b796a283c93554fee546cdfa7f25fe1 Mon Sep 17 00:00:00 2001
From: Canadew <110163919+chun-awa@users.noreply.github.com>
Date: Mon, 3 Aug 2026 19:56:59 +0800
Subject: [PATCH] Replace hard reset keybind with Ctrl+Shift+F12

---
 src/86box.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/86box.c b/src/86box.c
index 0f8793bd7..32263a9eb 100644
--- a/src/86box.c
+++ b/src/86box.c
@@ -292,7 +292,7 @@ struct accelKey def_acc_keys[NUM_ACCELS] = {
     {
         .name="hard_reset",
         .desc="Hard reset",
-        .seq="Ctrl+Alt+F12"
+        .seq="Ctrl+Shift+F12"
     },
     {
         .name="pause",
-- 
2.55.0

