--- fbterm-1.7/src/lib/vterm.cpp.orig	2016-02-29 14:06:59.979187259 +0900
+++ fbterm-1.7/src/lib/vterm.cpp	2016-02-29 14:10:53.852575092 +0900
@@ -68,13 +68,13 @@ u8 VTerm::control_map[MAX_CONTROL_CODE],
 
 void VTerm::init_state()
 {
-	for (u8 i = 1; control_sequences[i].code != (u16)-1; i++) {
+	for (u8 i = 1; control_sequences[i].code != (u16)0xffff; i++) {
 		control_map[control_sequences[i].code] = i;
 	}
 
 	u8 state = ESnormal;
 	for (u8 i = 1; ; i++) {
-		if (escape_sequences[i].code == (u16)-1) {
+		if (escape_sequences[i].code == (u16)0xffff) {
 			state++;
 			if (state == NR_STATES) break;
 		} else {
--- fbterm-1.7/src/lib/vterm_states.cpp.orig	2016-02-29 13:59:59.289083639 +0900
+++ fbterm-1.7/src/lib/vterm_states.cpp	2016-02-29 14:09:56.250972219 +0900
@@ -39,14 +39,14 @@ const VTerm::Sequence VTerm::control_seq
 	{ 0x1B, 0,	ESesc },
 	{ 0x7F, 0,	ESkeep },
 	{ 0x9B, 0,	ESsquare },
-	{ -1}
+	{ 0xffff }
 };
 
 const VTerm::Sequence VTerm::escape_sequences[] = {
 	{   0, 0, ESnormal },
 
 	// ESnormal
-	{ -1 },
+	{ 0xffff },
 
 	// ESesc
 	{ '[', &VTerm::clear_param,	ESsquare },
@@ -65,7 +65,7 @@ const VTerm::Sequence VTerm::escape_sequ
 	{ '8', &VTerm::restore_cursor,	ESnormal },
 	{ '>', &VTerm::keypad_numeric,	ESnormal },
 	{ '=', &VTerm::keypad_application,	ESnormal },
-	{ -1 },
+	{ 0xffff },
 
 	// ESsquare
 	{ '[', 0,	ESfunckey },
@@ -104,7 +104,7 @@ const VTerm::Sequence VTerm::escape_sequ
 	{ '`', &VTerm::cursor_position_col,	ESnormal },
 	{ ']', &VTerm::linux_specific, ESnormal },
 	{ '}', &VTerm::fbterm_specific, ESnormal },
-	{ -1 },
+	{ 0xffff },
 
 	// ESnonstd
 	{ '0' | ADDSAME(9), &VTerm::set_palette,    ESkeep },
@@ -112,25 +112,25 @@ const VTerm::Sequence VTerm::escape_sequ
 	{ 'a' | ADDSAME(5), &VTerm::set_palette,    ESkeep },
 	{ 'P', &VTerm::begin_set_palette, ESkeep },
 	{ 'R', &VTerm::reset_palette, ESnormal },
-	{ -1 },
+	{ 0xffff },
 
 	// ESpercent
 	{ '@', &VTerm::clear_utf8,	ESnormal },
 	{ 'G', &VTerm::set_utf8,	ESnormal },
 	{ '8', &VTerm::set_utf8,	ESnormal },
-	{ -1 },
+	{ 0xffff },
 
 	// EScharset
 	{ '0', &VTerm::set_charset, ESnormal },
 	{ 'B', &VTerm::set_charset, ESnormal },
 	{ 'U', &VTerm::set_charset, ESnormal },
 	{ 'K', &VTerm::set_charset, ESnormal },
-	{ -1 },
+	{ 0xffff },
 
 	// EShash
 	{ '8', &VTerm::screen_align,	ESnormal },
-	{ -1 },
+	{ 0xffff },
 
 	// ESfunckey
-	{ -1 },
+	{ 0xffff },
 };
