From af9219082e339b5f634689d9ae97ed1a7e61cc57 Mon Sep 17 00:00:00 2001
From: Yuannan <yuannan@users.noreply.github.com>
Date: Tue, 14 Oct 2025 09:34:48 +0000
Subject: [PATCH 4/4] Update CMakeLists.txt to have the proper version number

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5753aa5..36cad0c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,7 +43,7 @@ endif()
 
 set (VERSION_MAJOR 0)
 set (VERSION_MINOR 7)
-set (VERSION_RELEASE 2)
+set (VERSION_RELEASE 3)
 set (VERSION "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_RELEASE}")
 
 add_subdirectory (${PROJECT_SOURCE_DIR}/src)
-- 
2.52.0

