--- a/third_party/cpuinfo/src/api.c  2026-01-08 02:17:40.481979870 +0000
+++ b/third_party/cpuinfo/src/api.c  2026-01-08 02:22:12.354078180 +0000
@@ -9,11 +9,10 @@
 #include <linux/api.h>
 
 #include <sys/syscall.h>
+#define __USE_MISC 1
 #include <unistd.h>
-#if !defined(__NR_getcpu)
 #include <asm-generic/unistd.h>
 #endif
-#endif
 
 bool cpuinfo_is_initialized = false;
 
--- a/third_party/sleef/src/arch/helperpurec_scalar.h    2026-01-08 03:13:18.909853980 +0000
+++ b/third_party/sleef/src/arch/helperpurec_scalar.h    2026-01-08 03:15:19.989006880 +0000
@@ -54,7 +54,7 @@
 #define ENABLE_FMA_SP
 //@#define ENABLE_FMA_SP
 
-#if defined(__AVX2__) || defined(__aarch64__) || defined(__arm__) || defined(__powerpc64__) || defined(__zarch__) || defined(__riscv) || CONFIG == 3
+#if defined(__AVX2__) || defined(__aarch64__) || defined(__arm__) || defined(__powerpc64__) || defined(__zarch__) || defined(__riscv) || CONFIG == 3 || defined(__loongarch_lp64)
 #ifndef FP_FAST_FMA
 //@#ifndef FP_FAST_FMA
 #define FP_FAST_FMA
--- a/third_party/sleef/src/libm/CMakeLists.txt   2026-01-08 03:21:28.621613130 +0000
+++ b/third_party/sleef/src/libm/CMakeLists.txt   2026-01-08 03:21:57.444864690 +0000
@@ -69,6 +69,11 @@
     PUREC_SCALAR
     PURECFMA_SCALAR
     )
+else()
+  set(SLEEF_HEADER_LIST
+    PUREC_SCALAR
+    PURECFMA_SCALAR
+    )
 endif()
 
 # HEADER_PARAMS
