diff -up nut-2.7.4/scripts/systemd/nut-driver.service.in.tmpfiles nut-2.7.4/scripts/systemd/nut-driver.service.in
--- nut-2.7.4/scripts/systemd/nut-driver.service.in.tmpfiles	2016-08-11 16:58:00.759001400 +0200
+++ nut-2.7.4/scripts/systemd/nut-driver.service.in	2016-08-11 16:58:38.619248466 +0200
@@ -4,6 +4,7 @@ After=local-fs.target network.target
 StopWhenUnneeded=yes
 
 [Service]
+ExecStartPre=-/usr/bin/systemd-tmpfiles --create /usr/lib/tmpfiles.d/nut-client.conf
 ExecStart=@SBINDIR@/upsdrvctl start
 ExecStop=@SBINDIR@/upsdrvctl stop
 Type=forking
diff -up nut-2.7.4/scripts/systemd/nut-monitor.service.in.tmpfiles nut-2.7.4/scripts/systemd/nut-monitor.service.in
--- nut-2.7.4/scripts/systemd/nut-monitor.service.in.tmpfiles	2015-12-29 09:42:34.000000000 +0100
+++ nut-2.7.4/scripts/systemd/nut-monitor.service.in	2016-08-11 16:57:48.121918934 +0200
@@ -3,6 +3,7 @@ Description=Network UPS Tools - power de
 After=local-fs.target network.target nut-server.service
 
 [Service]
+ExecStartPre=-/usr/bin/systemd-tmpfiles --create /usr/lib/tmpfiles.d/nut-client.conf
 ExecStart=@SBINDIR@/upsmon
 PIDFile=@PIDPATH@/upsmon.pid
 Type=forking
