diff -up nut-2.7.3/configure.ac.quickfix nut-2.7.3/configure.ac
--- nut-2.7.3/configure.ac.quickfix	2015-04-23 12:05:21.219228670 +0200
+++ nut-2.7.3/configure.ac	2015-04-23 12:05:21.230228646 +0200
@@ -1065,7 +1065,7 @@ dnl Override installation directory, wit
 dnl prefix. This is needed for 'distcheck*' targets, otherwise
 dnl files will try to get intalled to the actual system directories
 if test -n "${systemdsystemunitdir}"; then
-	systemdsystemshutdowndir="${libdir}/systemd/system-shutdown"
+	systemdsystemshutdowndir="/lib/systemd/system-shutdown"
 	AC_MSG_RESULT(using ${systemdsystemunitdir})
 else
 	AC_MSG_RESULT(no)
