diff -urNp a/tools/nut-scanner/Makefile.am b/tools/nut-scanner/Makefile.am
--- a/tools/nut-scanner/Makefile.am	2020-09-02 10:16:27.304816091 +0200
+++ b/tools/nut-scanner/Makefile.am	2020-09-02 10:17:58.119811844 +0200
@@ -14,14 +14,14 @@ libnutscan_la_SOURCES = scan_nut.c scan_
 			scan_avahi.c scan_eaton_serial.c nutscan-serial.c \
 			../../drivers/serial.c \
 			../../drivers/bcmxcp_ser.c \
-			../../common/common.c
+			../../common/common.c ../../common/str.c
 libnutscan_la_LIBADD = $(NETLIBS) $(LIBLTDL_LIBS)
 libnutscan_la_LDFLAGS = $(SERLIBS) -version-info 1:0:0
 libnutscan_la_CFLAGS = -I$(top_srcdir)/clients -I$(top_srcdir)/include $(LIBLTDL_CFLAGS) -I$(top_srcdir)/drivers
 
 nut_scanner_SOURCES = nut-scanner.c
 nut_scanner_CFLAGS = -I$(top_srcdir)/clients -I$(top_srcdir)/include
-nut_scanner_LDADD = libnutscan.la ../../common/libcommon.la
+nut_scanner_LDADD = libnutscan.la
 
 if WITH_SSL
   libnutscan_la_CFLAGS += $(LIBSSL_CFLAGS)
