diff -up nut-2.7.4/drivers/snmp-ups.c.scratchdes nut-2.7.4/drivers/snmp-ups.c
--- nut-2.7.4/drivers/snmp-ups.c.scratchdes	2021-06-08 13:56:32.608702337 +0200
+++ nut-2.7.4/drivers/snmp-ups.c	2021-06-08 13:56:32.622702201 +0200
@@ -471,11 +471,7 @@ void nut_snmp_init(const char *type, con
 
 		privProtocol = testvar(SU_VAR_PRIVPROT) ? getval(SU_VAR_PRIVPROT) : "DES";
 
-		if (strcmp(privProtocol, "DES") == 0) {
-			g_snmp_sess.securityPrivProto = usmDESPrivProtocol;
-			g_snmp_sess.securityPrivProtoLen =  sizeof(usmDESPrivProtocol)/sizeof(oid);
-		}
-		else if (strcmp(privProtocol, "AES") == 0) {
+		if (strcmp(privProtocol, "AES") == 0) {
 			g_snmp_sess.securityPrivProto = usmAESPrivProtocol;
 			g_snmp_sess.securityPrivProtoLen =  sizeof(usmAESPrivProtocol)/sizeof(oid);
 		}
