--- libsigrok-0.5.2/Makefile.am	2019-12-25 12:29:19.000000000 -0800
+++ libsigrok-0.5.2-new/Makefile.am	2021-03-01 23:32:44.616325525 -0800
@@ -860,10 +860,10 @@
 		$(nodist_library_include_HEADERS) \
 		$(bindings_cxx_libsigrokcxx_la_include_HEADERS) \
 		$(nodist_bindings_cxx_libsigrokcxx_la_include_HEADERS)
-	$(AM_V_CXX)$(CXX) $(RBSIGROK_CFLAGS) -I. -Iinclude -Ibindings/cxx/include $(local_includes) -fPIC -o $@ -c $(RWRAP)
+	$(AM_V_CXX)$(CXX) $(RBSIGROK_CFLAGS) $(CXXFLAGS) -I. -Iinclude -Ibindings/cxx/include $(local_includes) -fPIC -o $@ -c $(RWRAP)
 
 $(REXT): $(ROBJ) @ORDER@ bindings/cxx/libsigrokcxx.la
-	$(AM_V_CXXLD)$(CXX) -shared -o $@ $(ROBJ) -lsigrokcxx -Lbindings/cxx/.libs $(RBSIGROK_LIBS)
+	$(AM_V_CXXLD)$(CXX) -shared $(CXXFLAGS) -o $@ $(ROBJ) -lsigrokcxx -Lbindings/cxx/.libs $(RBSIGROK_LIBS)
 
 ruby-build: $(REXT)
 
