--- pgadmin4-2.0/runtime/pgAdmin4.pro	2017-09-01 14:47:24.000000000 +0000
+++ pgadmin4-2.0.nopy2/runtime/pgAdmin4.pro	2017-11-16 03:25:06.367125851 +0000
@@ -92,11 +92,6 @@
     # Environment setting
     PYTHON_CONFIG = $$(PYTHON_CONFIG)
 
-    # Python 2?
-    isEmpty(PYTHON_CONFIG) {
-        PYTHON_CONFIG = $$system(which python-config)
-    }
-
     # Maybe Python 3?
     isEmpty(PYTHON_CONFIG) {
         PYTHON_CONFIG = $$system(which python3-config)
