diff -uNr anki-2.0.3.old/aqt/profiles.py anki-2.0.3/aqt/profiles.py
--- anki-2.0.3.old/aqt/profiles.py	2012-09-05 19:24:05.000000000 +0200
+++ anki-2.0.3/aqt/profiles.py	2012-12-27 19:43:11.369462377 +0100
@@ -18,11 +18,11 @@
 
 metaConf = dict(
     ver=0,
-    updates=True,
+    updates=False,
     created=intTime(),
     id=random.randrange(0, 2**63),
     lastMsg=-1,
-    suppressUpdate=False,
+    suppressUpdate=True,
     firstRun=True,
     defaultLang=None,
     disabledAddons=[],
