diff -Naur Mesa-7.11.2/configure.ac Mesa-7.11.2.glapi/configure.ac
--- Mesa-7.11.2/configure.ac	2011-10-15 00:43:58.000000000 +0000
+++ Mesa-7.11.2.glapi/configure.ac	2020-04-03 05:39:56.517674848 +0000
@@ -1125,7 +1125,7 @@
     fi
 
     # put all the necessary libs together, including possibly libdricore
-    DRI_LIB_DEPS="$DRI_LIB_DEPS $SELINUX_LIBS $LIBDRM_LIBS $EXPAT_LIB -lm -lpthread $DLOPEN_LIBS"
+    DRI_LIB_DEPS="-lglapi $DRI_LIB_DEPS $SELINUX_LIBS $LIBDRM_LIBS $EXPAT_LIB -lm -lpthread -lglapi $DLOPEN_LIBS"
 fi
 AC_SUBST([DRI_DIRS])
 AC_SUBST([EXPAT_INCLUDES])
