--- a/build/CMakeLists.txt	2020-11-30 09:50:15.000000000 -0800
+++ b/build/CMakeLists.txt	2021-02-19 22:11:23.719992309 -0800
@@ -3,6 +3,8 @@
 cmake_policy(VERSION 2.8.12)
 project("i2pd")
 
+set(CMAKE_EXE_LINKER_FLAGS "-latomic")
+
 # for debugging
 #set(CMAKE_VERBOSE_MAKEFILE on)
 
@@ -292,10 +294,6 @@
   message(WARNING "This build will NOT work on mainline i2p")
 endif()
 
-if(NOT MSYS)
-  include(CheckAtomic)
-endif()
-
 # show summary
 message(STATUS "---------------------------------------")
 message(STATUS "Build type         : ${CMAKE_BUILD_TYPE}")
