From 5355f1ad0e42047410739790f2d8c6040f5b8e2b Mon Sep 17 00:00:00 2001
From: Apollon Oikonomopoulos <apoikos@debian.org>
Date: Wed, 17 Jul 2019 13:08:45 -0300
Subject: [PATCH 1/2] DEBIAN: Use hardened LDFLAGS and CPPFLAGS

Forwarded: no
Last-Update: 2014-05-13

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 usr/Makefile | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/usr/Makefile b/usr/Makefile
index 394625b..efa1a89 100644
--- a/usr/Makefile
+++ b/usr/Makefile
@@ -64,14 +64,15 @@ TGTD_OBJS += tgtd.o mgmt.o target.o scsi.o log.o driver.o util.o work.o \
 
 TGTD_DEP = $(TGTD_OBJS:.o=.d)
 
-LDFLAGS = -Wl,-E,-rpath=$(libdir)
+TGTD_LDFLAGS = $(LDFLAGS)
+TGTD_LDFLAGS += -Wl,-E,-rpath=$(libdir)
 
 .PHONY:all
 all: $(PROGRAMS) $(MODULES)
 
 tgtd: $(TGTD_OBJS)
 	echo $(CC) $^ -o $@ $(LIBS)
-	$(CC) $^ -o $@ $(LDFLAGS) $(LIBS)
+	$(CC) $^ -o $@ $(TGTD_LDFLAGS) $(LIBS)
 
 -include $(TGTD_DEP)
 
@@ -79,7 +80,7 @@ TGTADM_OBJS = tgtadm.o concat_buf.o
 TGTADM_DEP = $(TGTADM_OBJS:.o=.d)
 
 tgtadm: $(TGTADM_OBJS)
-	$(CC) $^ -o $@
+	$(CC) $^ -o $@ $(LDFLAGS)
 
 -include $(TGTADM_DEP)
 
@@ -87,22 +88,25 @@ TGTIMG_OBJS = tgtimg.o libssc.o libcrc32c.o
 TGTIMG_DEP = $(TGTIMG_OBJS:.o=.d)
 
 tgtimg: $(TGTIMG_OBJS)
-	$(CC) $^ -o $@
+	$(CC) $^ -o $@ $(LDFLAGS)
 
 -include $(TGTIMG_DEP)
 
 %.o: %.c
-	$(CC) -c $(CFLAGS) $*.c -o $*.o
+	$(CC) $(CPPFLAGS) -c $(CFLAGS) $*.c -o $*.o
 	@$(CC) -MM $(CFLAGS) -MF $*.d -MT $*.o $*.c
 
 %.so: %.c
-	$(CC) -shared $(CFLAGS) $*.c -o $*.so
+	$(CC) $(CPPFLAGS) -shared $(CFLAGS) $*.c -o $*.so
 
 bs_rbd.so: bs_rbd.c
-	$(CC) -shared $(CFLAGS) bs_rbd.c -o bs_rbd.so -lrados -lrbd
+	$(CC) $(CPPFLAGS) -shared $(CFLAGS) bs_rbd.c -o bs_rbd.so \
+		-lrados -lrbd $(LDFLAGS)
 
 bs_glfs.so: bs_glfs.c
-	$(CC) -I/usr/include/glusterfs/api -shared $(CFLAGS) bs_glfs.c -o bs_glfs.so -lgfapi
+	$(CC) $(CPPFLAGS) -I/usr/include/glusterfs/api \
+		-shared $(CFLAGS) bs_glfs.c -o bs_glfs.so \
+		-lgfapi $(LDFLAGS)
 
 .PHONY: install
 install: $(PROGRAMS) $(MODULES)
-- 
2.52.0

