From b3e53e0c240a0b7dec8bd30c279b608626a13d86 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 28 Jun 2026 22:51:54 +0800
Subject: [PATCH 2/2] AOSCOS: scripts/tgtd.service: use /usr/bin for daemon
 paths

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 scripts/tgtd.service | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/scripts/tgtd.service b/scripts/tgtd.service
index e012541..1a52c84 100644
--- a/scripts/tgtd.service
+++ b/scripts/tgtd.service
@@ -12,17 +12,17 @@ ConditionPathExists=/etc/tgt/targets.conf
 Type=forking
 Environment=TGTD_CONFIG=/etc/tgt/targets.conf
 
-ExecStart=/usr/sbin/tgtd
-ExecStartPost=/usr/sbin/tgtadm --op update --mode sys --name State -v offline
-ExecStartPost=/usr/sbin/tgtadm --op update --mode sys --name State -v ready
-ExecStartPost=/usr/sbin/tgt-admin -e -c ${TGTD_CONFIG}
+ExecStart=/usr/bin/tgtd
+ExecStartPost=/usr/bin/tgtadm --op update --mode sys --name State -v offline
+ExecStartPost=/usr/bin/tgtadm --op update --mode sys --name State -v ready
+ExecStartPost=/usr/bin/tgt-admin -e -c ${TGTD_CONFIG}
 
-ExecReload=/usr/sbin/tgt-admin --update ALL -f -c ${TGTD_CONFIG}
+ExecReload=/usr/bin/tgt-admin --update ALL -f -c ${TGTD_CONFIG}
 
-ExecStop=/usr/sbin/tgtadm --op update --mode sys --name State -v offline
-ExecStop=/usr/sbin/tgt-admin --offline ALL
-ExecStop=/usr/sbin/tgt-admin --update ALL -c /dev/null -f
-ExecStop=/usr/sbin/tgtadm  --op delete --mode system
+ExecStop=/usr/bin/tgtadm --op update --mode sys --name State -v offline
+ExecStop=/usr/bin/tgt-admin --offline ALL
+ExecStop=/usr/bin/tgt-admin --update ALL -c /dev/null -f
+ExecStop=/usr/bin/tgtadm  --op delete --mode system
 
 # Exit code: 107 tgtd not running
 ProtectHostname=true
-- 
2.52.0

