From 0b50fcf4454bd700c2a2bf0f49aa96f2cfbeb33c Mon Sep 17 00:00:00 2001
From: Huacai Chen <chenhuacai@loongson.cn>
Date: Mon, 22 Feb 2021 13:57:38 +0800
Subject: [PATCH 015/137] MIPS: Loongson-3: Override __smp_store_release() for
 SFB

Signed-off-by: Huacai Chen <chenhuacai@loongson.cn>
---
 arch/mips/include/asm/barrier.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/mips/include/asm/barrier.h b/arch/mips/include/asm/barrier.h
index 9228f7386220..c020876f95e2 100644
--- a/arch/mips/include/asm/barrier.h
+++ b/arch/mips/include/asm/barrier.h
@@ -244,6 +244,16 @@
 #define __smp_mb__before_atomic()	__smp_mb__before_llsc()
 #define __smp_mb__after_atomic()	smp_llsc_mb()
 
+#ifdef CONFIG_CPU_LOONGSON3
+#define __smp_store_release(p, v)					\
+do {									\
+	compiletime_assert_atomic_type(*p);				\
+	__smp_mb();							\
+	WRITE_ONCE(*p, v);						\
+	__wbflush();							\
+} while (0)
+#endif
+
 /*
  * Some Loongson 3 CPUs have a bug wherein execution of a memory access (load,
  * store or prefetch) in between an LL & SC can cause the SC instruction to
-- 
2.37.1

