From 634a70e291cd2d66ed43e1fc756630a0589423de Mon Sep 17 00:00:00 2001
From: Huacai Chen <chenhc@lemote.com>
Date: Thu, 1 Dec 2016 09:51:35 +0800
Subject: [PATCH 069/137] Retry to configure USB device if needed

Signed-off-by: Huacai Chen <chenhc@lemote.com>
---
 drivers/usb/core/message.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/core/message.c b/drivers/usb/core/message.c
index 7ca908704777..c3f2bef6ffa9 100644
--- a/drivers/usb/core/message.c
+++ b/drivers/usb/core/message.c
@@ -1814,7 +1814,7 @@ int usb_set_configuration(struct usb_device *dev, int configuration)
 	struct usb_host_config *cp = NULL;
 	struct usb_interface **new_interfaces = NULL;
 	struct usb_hcd *hcd = bus_to_hcd(dev->bus);
-	int n, nintf;
+	int n, nintf, retried = 0;
 
 	if (dev->authorized == 0 || configuration == -1)
 		configuration = 0;
@@ -1967,6 +1967,7 @@ int usb_set_configuration(struct usb_device *dev, int configuration)
 	}
 	kfree(new_interfaces);
 
+retry:
 	ret = usb_control_msg(dev, usb_sndctrlpipe(dev, 0),
 			      USB_REQ_SET_CONFIGURATION, 0, configuration, 0,
 			      NULL, 0, USB_CTRL_SET_TIMEOUT);
@@ -1975,6 +1976,12 @@ int usb_set_configuration(struct usb_device *dev, int configuration)
 		 * All the old state is gone, so what else can we do?
 		 * The device is probably useless now anyway.
 		 */
+		if (!retried) {
+			retried = 1;
+			printk("Retry to configure %d-%s!\n", dev->bus->busnum, dev->devpath);
+			goto retry;
+		}
+
 		usb_hcd_alloc_bandwidth(dev, NULL, NULL, NULL);
 		for (i = 0; i < nintf; ++i) {
 			usb_disable_interface(dev, cp->interface[i], true);
-- 
2.37.1

