From 1dcf614d8816b2bb2b70eb32158121f3e069b39d Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Thu, 15 Sep 2022 18:36:31 +0800
Subject: [PATCH 137/137] mips: asm: serial: fix build

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 arch/mips/include/asm/serial.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/include/asm/serial.h b/arch/mips/include/asm/serial.h
index a780ee51a718..5c8bd9661ded 100644
--- a/arch/mips/include/asm/serial.h
+++ b/arch/mips/include/asm/serial.h
@@ -12,7 +12,7 @@
  */
 #define BASE_BAUD 0
 #else
-#define STD_COM_FLAGS (ASYNC_BOOT_AUTOCONF | ASYNC_SKIP_TEST)
+#define STD_COM_FLAGS (UPF_BOOT_AUTOCONF | UPF_SKIP_TEST)
 #include <asm-generic/serial.h>
 #endif
 
-- 
2.37.1

