From 5ab37c5ff729f30dfd04b762a09d305e5da88bad Mon Sep 17 00:00:00 2001
From: WhiredPlanck <whiredplanck@outlook.com>
Date: Wed, 6 Oct 2021 17:07:51 +0800
Subject: [PATCH] Fix Freetype and FontConfig Linkage

---
 libs/text/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libs/text/CMakeLists.txt b/libs/text/CMakeLists.txt
index 16d7e066fa2..429afe79704 100644
--- a/libs/text/CMakeLists.txt
+++ b/libs/text/CMakeLists.txt
@@ -152,11 +152,11 @@ if( SHOULD_BUILD_FEATURE_RDF )
 endif()
 
 if( FONTCONFIG_FOUND )
-    target_link_libraries(kotext PRIVATE ${FONTCONFIG_LIBRARIES})
+    target_link_libraries(kotext PRIVATE Fontconfig::Fontconfig)
 endif()
 
 if( FREETYPE_FOUND )
-    target_link_libraries(kotext PRIVATE ${FREETYPE_LIBRARIES})
+    target_link_libraries(kotext PRIVATE Freetype::Freetype)
 endif()
 
 
-- 
2.30.2

