diff -up calligra-3.2.0/CMakeLists.txt.disable_products calligra-3.2.0/CMakeLists.txt
--- calligra-3.2.0/CMakeLists.txt.disable_products	2020-04-24 10:58:02.572031256 -0500
+++ calligra-3.2.0/CMakeLists.txt	2020-04-24 10:58:33.687367280 -0500
@@ -1069,6 +1069,10 @@ calligra_drop_products_on_old_flag(words
 # calligra_disable_product(APP_FOO "isn't buildable at the moment")
 
 calligra_disable_product(APP_BRAINDUMP "Disabled, will (probably) be removed from Calligra")
+calligra_disable_product(APP_CSTESTER "not wanted in fedora packaging")
+calligra_disable_product(APP_DEVTOOLS "not wanted in fedora packaging")
+calligra_disable_product(APP_GEMINI "not wanted in fedora packaging")
+calligra_disable_product(PART_COMPONENTS "not wanted in fedora packaging")
 
 #############################################
 ####     Calculate buildable products    ####
