diff --git a/filters/sheets/excel/sidewinder/chartsubstreamhandler.cpp b/filters/sheets/excel/sidewinder/chartsubstreamhandler.cpp
index feac1c6..5739328 100644
--- a/filters/sheets/excel/sidewinder/chartsubstreamhandler.cpp
+++ b/filters/sheets/excel/sidewinder/chartsubstreamhandler.cpp
@@ -19,6 +19,7 @@
    Boston, MA 02110-1301, USA
  */
 
+#include <limits>
 #include "chartsubstreamhandler.h"
 #include "globalssubstreamhandler.h"
 #include "worksheetsubstreamhandler.h"
diff --git a/filters/sheets/excel/sidewinder/objects.cpp b/filters/sheets/excel/sidewinder/objects.cpp
index 7531dd7..6cbca37 100644
--- a/filters/sheets/excel/sidewinder/objects.cpp
+++ b/filters/sheets/excel/sidewinder/objects.cpp
@@ -17,6 +17,7 @@
    Boston, MA 02110-1301, USA
 */
 
+#include <limits>
 #include "objects.h"
 
 #include "generated/simpleParser.h"
diff --git a/filters/sheets/excel/sidewinder/utils.h b/filters/sheets/excel/sidewinder/utils.h
index 6e012b2..61d7624 100644
--- a/filters/sheets/excel/sidewinder/utils.h
+++ b/filters/sheets/excel/sidewinder/utils.h
@@ -23,6 +23,7 @@
 
 #include "value.h"
 #include <map>
+#include <limits>
 
 #include <QString>
 #include <QUuid>
diff --git a/filters/sheets/excel/sidewinder/value.cpp b/filters/sheets/excel/sidewinder/value.cpp
index bfa3351..3007058 100644
--- a/filters/sheets/excel/sidewinder/value.cpp
+++ b/filters/sheets/excel/sidewinder/value.cpp
@@ -17,6 +17,7 @@
  * Boston, MA 02110-1301, USA
 */
 
+#include <limits>
 #include "value.h"
 #include "utils.h"
 
