diff -Naur calligra-3.1.0/stage/part/KPrPresentationTool.cpp calligra-3.1.0.working/stage/part/KPrPresentationTool.cpp
--- calligra-3.1.0/stage/part/KPrPresentationTool.cpp	2018-01-26 17:24:44.000000000 +0800
+++ calligra-3.1.0.working/stage/part/KPrPresentationTool.cpp	2018-06-02 23:33:02.382029299 +0800
@@ -32,6 +32,7 @@
 #include <QDesktopServices>
 #include <QUrl>
 #include <QDBusConnection>
+#include <QFrame>
 
 #include <KoShape.h>
 #include <KoShapeManager.h>

