Index: gst-plugins-base0.10-0.10.29.2/ext/gnomevfs/gstgnomevfsuri.c
===================================================================
--- gst-plugins-base0.10-0.10.29.2.orig/ext/gnomevfs/gstgnomevfsuri.c	2009-05-21 20:44:11.000000000 +0200
+++ gst-plugins-base0.10-0.10.29.2/ext/gnomevfs/gstgnomevfsuri.c	2010-06-26 20:34:32.420174918 +0200
@@ -48,6 +48,7 @@
     "sftp://localhost/bla",
     "nfs://localhost/bla",
     "ssh://localhost/bla",
+    "dav://localhost/bla",
     "burn://"
   };
   GnomeVFSURI *uri;
