diff --git a/Makefile b/Makefile
index d59800f..fe67ea7 100644
--- a/Makefile
+++ b/Makefile
@@ -45,22 +45,22 @@ clean:
 	$(RM) $(BUILD)
 
 install: $(BUILD)
-	$(MKDIR) $(DESTDIR)$(PREFIX)/$(LIBDIR)
-	$(CP) $(SOFILENAME) $(SONAME) $(LIBNAME) $(ANAME) $(DESTDIR)$(PREFIX)/$(LIBDIR)
-	$(MKDIR) $(DESTDIR)$(PREFIX)/$(BINDIR)
-	$(CP) $(UTILITIES) $(DESTDIR)$(PREFIX)/$(BINDIR)
-	$(MKDIR) $(DESTDIR)$(PREFIX)/$(INCDIR)
-	$(CP) $(HEADERS) $(DESTDIR)$(PREFIX)/$(INCDIR)
-	$(MKDIR) $(DESTDIR)$(PREFIX)/$(PKGDIR)
-	$(PRINTF) 'prefix=%s\nexec_prefix=$${prefix}\nlibdir=$${exec_prefix}/%s\nincludedir=$${prefix}/%s\n\n' $(PREFIX) $(LIBDIR) $(INCDIR) > $(DESTDIR)$(PREFIX)/$(PKGDIR)/$(PCNAME)
-	$(PRINTF) 'Name: lib%s\nDescription: Open Source aptX codec library\nVersion: %u.%u.%u\n' $(NAME) $(MAJOR) $(MINOR) $(PATCH) >> $(DESTDIR)$(PREFIX)/$(PKGDIR)/$(PCNAME)
-	$(PRINTF) 'Libs: -Wl,-rpath=$${libdir} -L$${libdir} -l%s\nCflags: -I$${includedir}\n' $(NAME) >> $(DESTDIR)$(PREFIX)/$(PKGDIR)/$(PCNAME)
+	$(MKDIR) $(DESTDIR)$(LIBDIR)
+	$(CP) $(SOFILENAME) $(SONAME) $(LIBNAME) $(ANAME) $(DESTDIR)$(LIBDIR)
+	$(MKDIR) $(DESTDIR)$(BINDIR)
+	$(CP) $(UTILITIES) $(DESTDIR)$(BINDIR)
+	$(MKDIR) $(DESTDIR)$(INCDIR)
+	$(CP) $(HEADERS) $(DESTDIR)$(INCDIR)
+	$(MKDIR) $(DESTDIR)$(PKGDIR)
+	$(PRINTF) 'prefix=%s\nexec_prefix=$${prefix}\nlibdir=$${exec_prefix}/%s\nincludedir=$${prefix}/%s\n\n' $(PREFIX) $(LIBDIR) $(INCDIR) > $(DESTDIR)$(PKGDIR)/$(PCNAME)
+	$(PRINTF) 'Name: lib%s\nDescription: Open Source aptX codec library\nVersion: %u.%u.%u\n' $(NAME) $(MAJOR) $(MINOR) $(PATCH) >> $(DESTDIR)$(PKGDIR)/$(PCNAME)
+	$(PRINTF) 'Libs: -Wl,-rpath=$${libdir} -L$${libdir} -l%s\nCflags: -I$${includedir}\n' $(NAME) >> $(DESTDIR)$(PKGDIR)/$(PCNAME)
 
 uninstall:
-	for f in $(SOFILENAME) $(SONAME) $(LIBNAME) $(ANAME); do $(RM) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$$f; done
-	for f in $(UTILITIES); do $(RM) $(DESTDIR)$(PREFIX)/$(BINDIR)/$$f; done
-	for f in $(HEADERS); do $(RM) $(DESTDIR)$(PREFIX)/$(INCDIR)/$$f; done
-	$(RM) $(DESTDIR)$(PREFIX)/$(PKGDIR)/$(PCNAME)
+	for f in $(SOFILENAME) $(SONAME) $(LIBNAME) $(ANAME); do $(RM) $(DESTDIR)$(LIBDIR)/$$f; done
+	for f in $(UTILITIES); do $(RM) $(DESTDIR)$(BINDIR)/$$f; done
+	for f in $(HEADERS); do $(RM) $(DESTDIR)$(INCDIR)/$$f; done
+	$(RM) $(DESTDIR)$(PKGDIR)/$(PCNAME)
 
 $(UTILITIES): $(LIBNAME)
 
-- 
2.30.0

