From 4721ed83927d1eda1d6fdc7aabc7c4fbd552eb92 Mon Sep 17 00:00:00 2001
From: salieri <maliya355@outlook.com>
Date: Tue, 11 Jun 2024 18:02:38 +0800
Subject: [PATCH] enable debug symbol generation

---
 source/ddrescueview.lpi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/ddrescueview.lpi b/source/ddrescueview.lpi
index baa8e34..cfd8e90 100644
--- a/source/ddrescueview.lpi
+++ b/source/ddrescueview.lpi
@@ -98,8 +98,8 @@
           </CodeGeneration>
           <Linking>
             <Debugging>
-              <GenerateDebugInfo Value="False"/>
-              <StripSymbols Value="True"/>
+              <GenerateDebugInfo Value="True"/>
+              <StripSymbols Value="False"/>
             </Debugging>
             <LinkSmart Value="True"/>
             <Options>
-- 
2.34.1

