--- openmw-openmw-0.46.0/./components/detournavigator/offmeshconnectionsmanager.hpp	2020-06-08 02:11:24.000000000 -0700
+++ openmw-openmw-0.46.0-new/./components/detournavigator/offmeshconnectionsmanager.hpp	2021-02-01 21:40:08.698197205 -0800
@@ -12,7 +12,7 @@
 #include <osg/Vec3f>
 
 #include <boost/optional.hpp>
-
+#include <algorithm>
 #include <map>
 #include <mutex>
 #include <unordered_map>
