From 27c4189485f521f4cfd358e7e37b888919cbbce1 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Thu, 1 Feb 2024 17:54:10 -0800
Subject: [PATCH 1/3] Include btf_func_linkage for older kernels

---
 src/btf.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/btf.h b/src/btf.h
index db01f2c8..95ae2c45 100644
--- a/src/btf.h
+++ b/src/btf.h
@@ -3,6 +3,7 @@
 #include "types.h"
 #include <cstddef>
 #include <linux/types.h>
+#include <linux/version.h>
 #include <map>
 #include <optional>
 #include <regex>
@@ -25,6 +26,15 @@
 struct btf;
 struct btf_type;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,5,0)
+/* this is required to build with 5.4 and earlier kernel headers */
+enum btf_func_linkage {
+    BTF_FUNC_STATIC = 0,
+    BTF_FUNC_GLOBAL = 1,
+    BTF_FUNC_EXTERN = 2,
+};
+#endif
+
 namespace bpftrace {
 
 class BPFtrace;
-- 
2.45.2

