From b4cd094b42861b5877b44e507d3634dad091d266 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Thu, 1 Feb 2024 17:54:48 -0800
Subject: [PATCH 2/6] Fix missing include for cpp 17

---
 src/bpftrace.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/bpftrace.h b/src/bpftrace.h
index aa4b7d4f..8fa3864a 100644
--- a/src/bpftrace.h
+++ b/src/bpftrace.h
@@ -2,6 +2,7 @@
 
 #include <cstdint>
 #include <iostream>
+#include <iterator>
 #include <map>
 #include <memory>
 #include <optional>
-- 
2.44.0

