From 7720deee9a6f3f0bdee7c9ac6a732452d6b29d32 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Thu, 1 Feb 2024 17:54:48 -0800
Subject: [PATCH 2/2] Fix missing include for cpp 17

---
 src/bpftrace.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/bpftrace.h b/src/bpftrace.h
index 601003fd..a73a3a96 100644
--- a/src/bpftrace.h
+++ b/src/bpftrace.h
@@ -3,6 +3,7 @@
 #include <bcc/bcc_syms.h>
 #include <cstdint>
 #include <limits>
+#include <iterator>
 #include <map>
 #include <memory>
 #include <optional>
-- 
2.52.0

