From 660de4abfa042ad1d8812fe0ed8280e316f596cc Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Thu, 1 Feb 2024 17:54:48 -0800
Subject: [PATCH 2/2] Fix missing include for cpp 17

---
 src/bpftrace.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/bpftrace.h b/src/bpftrace.h
index 1e78aa45..b214f2e4 100644
--- a/src/bpftrace.h
+++ b/src/bpftrace.h
@@ -2,6 +2,7 @@
 
 #include <cstdint>
 #include <iostream>
+#include <iterator>
 #include <map>
 #include <memory>
 #include <optional>
-- 
2.43.0

