From e25a96ed8882262d67570cb234949efa7deffb5e Mon Sep 17 00:00:00 2001
From: Nianqing Yao <imbearchild@outlook.com>
Date: Sun, 5 Apr 2026 16:06:27 +0800
Subject: [PATCH] Disable fatal-warnings for linker, due to GCS change in GCC.

---
 meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meson.build b/meson.build
index 123c88e..8718f55 100644
--- a/meson.build
+++ b/meson.build
@@ -139,7 +139,6 @@ add_project_arguments(
 add_project_link_arguments(
   # --as-needed and --no-undefined are enabled by default
   cc.get_supported_link_arguments([
-    '-Wl,--fatal-warnings',
     '-Wl,--no-undefined-version',
     '-Wl,-O1',
   ]),
-- 
2.53.0

