diff -Naur transcode-1.1.7.orig/import/v4l/import_v4l2.c transcode-1.1.7/import/v4l/import_v4l2.c
--- transcode-1.1.7.orig/import/v4l/import_v4l2.c	2021-06-11 02:16:58.600561522 -0500
+++ transcode-1.1.7/import/v4l/import_v4l2.c	2021-06-11 02:17:04.688556525 -0500
@@ -455,7 +455,7 @@
         tc_log_warn(MOD_NAME,
                     "driver does not support cropping"
                     "(ioctl(VIDIOC_CROPCAP) returns \"%s\"), disabled",
-                    errno <= sys_nerr ? sys_errlist[errno] : "unknown");
+                    strerror(errno));
         return TC_ERROR;
     }
     if (verbose_flag > TC_INFO) {
@@ -485,7 +485,7 @@
         tc_log_warn(MOD_NAME,
                     "driver does not support inquiring cropping"
                     " parameters (ioctl(VIDIOC_G_CROP) returns \"%s\")",
-                    errno <= sys_nerr ? sys_errlist[errno] : "unknown");
+                    strerror(errno));
         return -1;
     }
 
@@ -513,7 +513,7 @@
             tc_log_warn(MOD_NAME,
                         "driver does not support inquering cropping"
                         " parameters (ioctl(VIDIOC_G_CROP) returns \"%s\")",
-                        errno <= sys_nerr ? sys_errlist[errno] : "unknown");
+                        strerror(errno));
             return -1;
         }
         if (verbose_flag > TC_INFO) {
@@ -613,7 +613,7 @@
     if (err < 0) {
         if (verbose_flag) {
             tc_log_warn(MOD_NAME, "driver does not support setting parameters (ioctl(VIDIOC_S_PARM) returns \"%s\")",
-                        errno <= sys_nerr ? sys_errlist[errno] : "unknown");
+                        strerror(errno));
         }
     }
     return TC_OK;
@@ -628,7 +628,7 @@
     err = ioctl(vs->video_fd, VIDIOC_G_STD, &stdid);
     if (err < 0) {
         tc_log_warn(MOD_NAME, "driver does not support get std (ioctl(VIDIOC_G_STD) returns \"%s\")",
-                    errno <= sys_nerr ? sys_errlist[errno] : "unknown");
+                    strerror(errno));
         memset(&stdid, 0, sizeof(v4l2_std_id));
     }
 
diff -Naur transcode-1.1.7.orig/import/v4l/import_v4lcam.c transcode-1.1.7/import/v4l/import_v4lcam.c
--- transcode-1.1.7.orig/import/v4l/import_v4lcam.c	2021-06-11 02:16:58.600561522 -0500
+++ transcode-1.1.7/import/v4l/import_v4lcam.c	2021-06-11 02:18:02.624508986 -0500
@@ -368,7 +368,7 @@
     if (err < 0) {
         tc_log_warn(MOD_NAME, "driver does not support setting parameters"
                               " (ioctl(VIDIOC_S_PARM) returns \"%s\")",
-                    errno <= sys_nerr ? sys_errlist[errno] : "unknown");
+	            strerror(errno));
     }
     return TC_OK;
 }
