From 0c7dfbc00c50b65ea81fb6b86ceb91851b694ffc Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Wed, 13 Mar 2024 06:25:29 -0700
Subject: [PATCH] Disable strip for debug symbol in AOSC

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 57cdd7e..cc1fcb2 100644
--- a/Makefile
+++ b/Makefile
@@ -319,7 +319,7 @@ clean:
 
 install: all
 	mkdir -p "$(DESTDIR)$(PREFIX)/bin"
-	$(STRIP) qjs qjsc
+	#$(STRIP) qjs qjsc
 	install -m755 qjs qjsc "$(DESTDIR)$(PREFIX)/bin"
 	ln -sf qjs "$(DESTDIR)$(PREFIX)/bin/qjscalc"
 	mkdir -p "$(DESTDIR)$(PREFIX)/lib/quickjs"
-- 
2.44.0

