From d82a26ca0950191546f546264e79ff72e2dbf09f Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <i.gnatenko.brain@gmail.com>
Date: Mon, 12 Oct 2015 18:52:16 +0200
Subject: [PATCH] ruby: make compatible with ruby 2.2

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>
---
 bindings/ruby/CMakeLists.txt | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/bindings/ruby/CMakeLists.txt b/bindings/ruby/CMakeLists.txt
index 6c3bd50..a7a3394 100644
--- a/bindings/ruby/CMakeLists.txt
+++ b/bindings/ruby/CMakeLists.txt
@@ -6,8 +6,17 @@ ELSE (USE_VENDORDIRS AND RUBY_VENDORARCH_DIR)
     SET (RUBY_INSTALL_DIR ${RUBY_SITEARCH_DIR})
 ENDIF (USE_VENDORDIRS AND RUBY_VENDORARCH_DIR)
 
+EXECUTE_PROCESS(COMMAND ${RUBY_EXECUTABLE} -r rbconfig -e "print RbConfig::CONFIG['arch']"
+   OUTPUT_VARIABLE RUBY_ARCH)
+
+EXECUTE_PROCESS(COMMAND ${RUBY_EXECUTABLE} -r rbconfig -e "print RbConfig::CONFIG['rubyhdrdir']"
+   OUTPUT_VARIABLE RUBY_HDR_DIR)
+
+SET(RUBY_INCLUDE_DIRS ${RUBY_HDR_DIR} ${RUBY_HDR_DIR}/${RUBY_ARCH})
+
 MESSAGE (STATUS "Ruby executable: ${RUBY_EXECUTABLE}")
 MESSAGE (STATUS "Ruby installation dir: ${RUBY_INSTALL_DIR}")
+MESSAGE (STATUS "Ruby include dir: ${RUBY_INCLUDE_DIRS}")
 
 ADD_CUSTOM_COMMAND (
     OUTPUT solv_ruby.c
@@ -17,7 +26,7 @@ ADD_CUSTOM_COMMAND (
 )
 
 ADD_DEFINITIONS(-Wno-unused)
-INCLUDE_DIRECTORIES (${RUBY_INCLUDE_PATH})
+INCLUDE_DIRECTORIES (${RUBY_INCLUDE_DIRS})
 
 ADD_LIBRARY (bindings_ruby MODULE solv_ruby.c)
 SET_TARGET_PROPERTIES (bindings_ruby PROPERTIES PREFIX "" OUTPUT_NAME "solv")
-- 
2.6.1

