From 18f93e12d10aa2a507ffa3689725e21a85a81e7f Mon Sep 17 00:00:00 2001
From: Michael Schroeder <mls@suse.de>
Date: Mon, 12 Oct 2015 14:04:54 +0200
Subject: [PATCH 4/4] Fix spelling, duh...

---
 ext/testcase.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/testcase.c b/ext/testcase.c
index 3c40451..b9fddef 100644
--- a/ext/testcase.c
+++ b/ext/testcase.c
@@ -164,7 +164,7 @@ static struct selflags2str {
 };
 
 static const char *features[] = {
-#ifdef ENABLE_LINKED_PACKAGES
+#ifdef ENABLE_LINKED_PKGS
   "linked_packages",
 #endif
 #ifdef ENABLE_COMPLEX_DEPS
-- 
2.6.1

