From e3028a7313976852cdd2f9b96a58262c8ac5887c Mon Sep 17 00:00:00 2001
From: hrh <hrhdeyouxiang@qq.com>
Date: Thu, 28 Dec 2023 16:25:15 +0800
Subject: [PATCH] Arch Linux: configure.ac: support libplist 2.2 and 2.3

---
 configure.ac          | 2 +-
 tools/ipod-lockdown.c | 8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index c4ffeaf..68e73c6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -42,7 +42,7 @@ AC_CHECK_FUNCS([localtime_r])
 AC_CHECK_MEMBERS([struct tm.tm_gmtoff],,,[#include <time.h>])
 dnl sqlite3 is needed for newer ipod models (nano5g), and libplist is needed 
 dnl by libgpod sqlite code
-PKG_CHECK_MODULES(LIBGPOD, glib-2.0 >= 2.16.0 gobject-2.0 sqlite3 libplist >= 1.0 gmodule-2.0)
+PKG_CHECK_MODULES(LIBGPOD, glib-2.0 >= 2.16.0 gobject-2.0 sqlite3 libplist-2.0 >= 2.2 gmodule-2.0)
 
 dnl ***************************************************
 dnl The following functions are only available starting
diff --git a/tools/ipod-lockdown.c b/tools/ipod-lockdown.c
index f345726..8503798 100644
--- a/tools/ipod-lockdown.c
+++ b/tools/ipod-lockdown.c
@@ -100,7 +100,7 @@ read_sysinfo_extended_by_uuid (const char *uuid)
 	plist_get_string_val(ptr, &str);
 	if (str != NULL) {
 	    ptr = plist_new_string(str);
-	    plist_dict_insert_item(value, "SerialNumber", ptr);
+	    plist_dict_set_item(value, "SerialNumber", ptr);
 	    free(str);
 	}
 
@@ -112,15 +112,15 @@ read_sysinfo_extended_by_uuid (const char *uuid)
 	plist_get_string_val(ptr, &str);
 	if (str != NULL) {
 	    ptr = plist_new_string(str);
-	    plist_dict_insert_item(value, "VisibleBuildID", ptr);
+	    plist_dict_set_item(value, "VisibleBuildID", ptr);
 	    free(str);
 	}
 
 	ptr = plist_new_string(uuid);
-	plist_dict_insert_item(value, "FireWireGUID", ptr);
+	plist_dict_set_item(value, "FireWireGUID", ptr);
 
 	ptr = plist_new_string(uuid);
-	plist_dict_insert_item(value, "UniqueDeviceID", ptr);
+	plist_dict_set_item(value, "UniqueDeviceID", ptr);
 
 	plist_to_xml(value, &xml, &xml_length);
 
-- 
2.43.0

