From db4ffde4bb266803a77eb915f99910728f2ad609 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sl=C3=A1vek=20Banko?= <slavek.banko@axis.cz>
Date: Fri, 9 Aug 2024 16:37:15 +0200
Subject: [PATCH] Fix compatibility with ffmpeg 7.x.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Slávek Banko <slavek.banko@axis.cz>
---
 plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp b/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
index 983c808..3e34c5c 100644
--- a/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
+++ b/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
@@ -217,8 +217,13 @@ bool K3bFFMpegFile::open() {
 #else
   d->sampleFormat = static_cast<::AVSampleFormat>(d->audio_stream->codecpar->format);
 #endif
+# if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(59, 24, 100)
   d->isSpacious = ::av_sample_fmt_is_planar(d->sampleFormat) &&
                   d->audio_stream->codecpar->channels > 1;
+# else
+  d->isSpacious = ::av_sample_fmt_is_planar(d->sampleFormat) &&
+                  d->audio_stream->codecpar->ch_layout.nb_channels > 1;
+# endif
 
   // dump some debugging info
   ::av_dump_format(d->formatContext, 0, m_filename.local8Bit(), 0);
@@ -255,7 +260,11 @@ int K3bFFMpegFile::sampleRate() const {
 }
 
 int K3bFFMpegFile::channels() const {
+# if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(59, 24, 100)
   return d->audio_stream->codecpar->channels;
+# else
+  return d->audio_stream->codecpar->ch_layout.nb_channels;
+# endif
 }
 
 int K3bFFMpegFile::type() const { return d->audio_stream->codecpar->codec_id; }
-- 
2.46.0

