From 93c4f89466166d79f1c8872043bcdf84221aecb7 Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@hadrons.org>
Date: Sat, 28 Aug 2021 01:07:03 +0200
Subject: [PATCH 4/4] build: Remove obsolete AC_TYPE_SIGNAL macro

---
 configure.ac | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index fd7856b9..be269852 100644
--- a/configure.ac
+++ b/configure.ac
@@ -651,9 +651,6 @@ fi
 
 ### Checks for types.
 
-# Checks return type of signal handlers
-AC_TYPE_SIGNAL
-
 AC_CHECK_TYPES(struct if_nameindex, , ,
   [IU_FLUSHLEFT([#include <sys/types.h>
                  #include <sys/socket.h>
-- 
2.33.0

