From 7877d2a987e6d9251d2c74f42a214a1f884fe2ad Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@hadrons.org>
Date: Sun, 5 Sep 2021 05:00:23 +0200
Subject: [PATCH] build: Use runstatedir for /run directory

---
 paths | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/paths b/paths
index f3e7d9fc..029457d4 100644
--- a/paths
+++ b/paths
@@ -79,12 +79,12 @@ PATH_FTPLOGINMESG /etc/motd
 PATH_FTPUSERS	$(sysconfdir)/ftpusers
 PATH_FTPCHROOT	$(sysconfdir)/ftpchroot
 PATH_FTPWELCOME $(sysconfdir)/ftpwelcome
-PATH_FTPDPID	$(localstatedir)/run/ftpd.pid
+PATH_FTPDPID	$(runstatedir)/ftpd.pid
 PATH_INETDCONF	$(sysconfdir)/inetd.conf
 PATH_INETDDIR	$(sysconfdir)/inetd.d
-PATH_INETDPID	$(localstatedir)/run/inetd.pid
-PATH_UTMP	<utmp.h> <utmp.h>:UTMP_FILE $(localstatedir)/run/utmp search:utmp:/var/run:/var/adm:/etc "/var/run/utx.active"
-PATH_UTMPX	<utmpx.h> <utmpx.h>:UTMPX_FILE $(localstatedir)/run/utmpx search:utmpx:/var/run:/var/adm:/etc "/var/run/utx.active"
+PATH_INETDPID	$(runstatedir)/inetd.pid
+PATH_UTMP	<utmp.h> <utmp.h>:UTMP_FILE $(runstatedir)/utmp search:utmp:/run:/var/run:/var/adm:/etc "/var/run/utx.active"
+PATH_UTMPX	<utmpx.h> <utmpx.h>:UTMPX_FILE $(runstatedir)/utmpx search:utmpx:/run:/var/run:/var/adm:/etc "/var/run/utx.active"
 PATH_WTMP	<utmp.h> <utmp.h>:WTMP_FILE $(localstatedir)/log/wtmp search:wtmp:/var/log:/var/adm:/etc "/var/log/utx.log"
 PATH_WTMPX	<utmpx.h> <utmpx.h>:WTMPX_FILE $(localstatedir)/log/wtmpx search:wtmpx:/var/log:/var/adm:/etc "/var/log/utx.log"
 PATH_LASTLOG	<utmp.h> $(localstatedir)/log/lastlog search:lastlog:/var/log:/var/adm:/etc "/var/log/utx.lastlogin"
@@ -93,7 +93,7 @@ PATH_KLOG	<syslog.h> /dev/klog no
 PATH_LOGCONF	$(sysconfdir)/syslog.conf
 PATH_LOGCONFD	$(sysconfdir)/syslog.d
 PATH_LOGIN	x $(bindir)/login search:login
-PATH_LOGPID	$(localstatedir)/run/syslog.pid
+PATH_LOGPID	$(runstatedir)/syslog.pid
 PATH_NOLOGIN	/etc/nologin
 PATH_RLOGIN	x $(bindir)/rlogin
 PATH_RSH	x $(bindir)/rsh
-- 
2.33.0

