From b94043665cabb9b9ca30054d8aa905f046d79081 Mon Sep 17 00:00:00 2001
From: Mats Erik Andersson <gnu@gisladisker.se>
Date: Thu, 28 Jan 2016 15:22:09 +0100
Subject: [PATCH 15/60] Amended changelog of last commit.

---
 ChangeLog | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index 78c783f4..120da370 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,6 +1,8 @@
 2016-01-23  Mats Erik Andersson  <gnu@gisladisker.se>
 
 	ftp: Allow remote user in command.
+	Ability implemented after a query by Andreas Adamcik, on the
+	legacy use of getlogin() versus the more admitting getuid().
 
 	* ftp/ftp.c (hookup): Check for an embedded at-sign in the input
 	variable `host', keeping the rightmost part.  New variable P.
-- 
2.26.0.292.g33ef6b2f38

