From 59a916153f08f3e53e7ed8d336ef734e2cc4ad69 Mon Sep 17 00:00:00 2001
From: Mats Erik Andersson <gnu@gisladisker.se>
Date: Tue, 9 Feb 2016 22:42:00 +0100
Subject: [PATCH 16/60] Properly disable the generation of ping6.

---
 ChangeLog    | 5 +++++
 configure.ac | 4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 120da370..27a031d3 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2016-02-09  Mats Erik Andersson  <gnu@gisladisker.se>
+
+	* configure.ac (missing ipv6 or icmp6): Disable `ping6' using
+	the macro IU_DISABLE_TARGET.
+
 2016-01-23  Mats Erik Andersson  <gnu@gisladisker.se>
 
 	ftp: Allow remote user in command.
diff --git a/configure.ac b/configure.ac
index b1ba09bc..aef32db6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -463,7 +463,7 @@ else
 fi
 
 if test "$ipv6" = "no" || test "$working_ipv6" = "no";then
-  ping6_BUILD=''
+  IU_DISABLE_TARGET([ping6])
 else
   working_icmp6=yes
   AC_CHECK_HEADER(netinet/icmp6.h, , working_icmp6=no,
@@ -552,7 +552,7 @@ else
 
   if test "$working_icmp6" = "no"; then
     AC_MSG_WARN([ICMPv6 support not available, disabling ping6])
-    ping6_BUILD=''
+    IU_DISABLE_TARGET([ping6])
   fi
 fi
 
-- 
2.26.0.292.g33ef6b2f38

