From 84ac94c03cb118bde0b7655d8fc1213d8adf0234 Mon Sep 17 00:00:00 2001
From: "Alfred M. Szmidt" <ams@gnu.org>
Date: Mon, 10 Jul 2017 19:59:44 +0000
Subject: [PATCH 24/60] ftpd/ftpd.c (options) <max-timeout>: Mention mandatory
 argument.

---
 ChangeLog   | 4 ++++
 ftpd/ftpd.c | 4 ++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 8ca7099a..72fa48bf 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2017-07-10  Omer Anson <oaanson@gmail.com>  (tiny change)
+
+	* ftpd/ftpd.c (options) <max-timeout>: Mention mandatory argument.
+
 2017-06-22  Omer Anson <oaanson@gmail.com>  (tiny change)
 
 	* ftp/main.c (main): Don't assume that uninitialized argv[argv] is
diff --git a/ftpd/ftpd.c b/ftpd/ftpd.c
index 5db88d0b..f77fcafe 100644
--- a/ftpd/ftpd.c
+++ b/ftpd/ftpd.c
@@ -294,8 +294,8 @@ static struct argp_option options[] = {
   { "timeout", 't', "TIMEOUT", 0,
     "set default idle timeout",
     GRID+1 },
-  { "max-timeout", 'T', NULL, 0,
-    "reset maximum value of timeout allowed",
+  { "max-timeout", 'T', "MAX_TIMEOUT", 0,
+    "set maximum value of timeout allowed",
     GRID+1 },
   { "non-rfc2577", OPT_NONRFC2577, NULL, 0,
     "neglect RFC 2577 by giving info on missing users",
-- 
2.26.0.292.g33ef6b2f38

