From fa7962cc9dda8aee39b5506373154fdbf46f3d47 Mon Sep 17 00:00:00 2001
From: Mats Erik Andersson <gnu@gisladisker.se>
Date: Fri, 9 Nov 2018 23:40:37 +0100
Subject: [PATCH 27/60] Gnulib requires autoconf 2.63, forced by libtool.

---
 ChangeLog    | 5 +++++
 configure.ac | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 64e8dad8..b5ced667 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2018-11-09  Mats Erik Andersson  <gnu@gisladisker.se>
+
+	* configure.ac (AC_PREREQ): Require autoconf 2.63.  Gnulib needs
+	a more recent version due to demands put forward by libtool.
+
 2017-07-10  Omer Anson <oaanson@gmail.com>  (tiny change)
 
 	* src/hostname.c (parse_file): Free name and allocate one extra
diff --git a/configure.ac b/configure.ac
index aef32db6..6f543987 100644
--- a/configure.ac
+++ b/configure.ac
@@ -18,7 +18,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see `http://www.gnu.org/licenses/'.
 
-AC_PREREQ(2.59)
+AC_PREREQ(2.63)
 
 AC_INIT([GNU inetutils],
  m4_esyscmd([build-aux/git-version-gen .tarball-version 's/inetutils-/v/;s/_/./g']),
-- 
2.26.0.292.g33ef6b2f38

