From 3441ac3bf1c116bc83b27e4f123560cbef982692 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sun, 16 Feb 2020 19:42:05 +0100
Subject: [PATCH 34/60] telnetd: Fix multiple definition of 'not42' (gcc 10)

---
 telnetd/utility.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/telnetd/utility.c b/telnetd/utility.c
index 5b8bd50a..20a74f38 100644
--- a/telnetd/utility.c
+++ b/telnetd/utility.c
@@ -63,7 +63,7 @@ static int ncc;
 static char ptyibuf[BUFSIZ], *ptyip;
 static int pcc;
 
-int not42;
+extern int not42;
 
 static int
 readstream (int p, char *ibuf, int bufsize)
-- 
2.26.0.292.g33ef6b2f38

