From 3365c5e234cb785655db8cd7cd1d1c2cf1e01b78 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sun, 16 Feb 2020 19:54:30 +0100
Subject: [PATCH 35/60] ftpd: Fix multiple definition of 'errcatch' (gcc 10)

---
 ftpd/extern.h | 2 +-
 ftpd/ftpd.c   | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/ftpd/extern.h b/ftpd/extern.h
index ab33cf38..91dbbeea 100644
--- a/ftpd/extern.h
+++ b/ftpd/extern.h
@@ -90,7 +90,7 @@ extern void user (const char *);
 extern char *sgetsave (const char *);
 
 /* Exported from ftpd.c.  */
-jmp_buf errcatch;
+extern jmp_buf errcatch;
 extern struct sockaddr_storage data_dest;
 extern socklen_t data_dest_len;
 extern struct sockaddr_storage his_addr;
diff --git a/ftpd/ftpd.c b/ftpd/ftpd.c
index f77fcafe..0ef455b0 100644
--- a/ftpd/ftpd.c
+++ b/ftpd/ftpd.c
@@ -146,6 +146,8 @@ char portstr[8];		/* Numeric port as string.  */
 /* Requester credentials.  */
 struct credentials cred;
 
+jmp_buf errcatch;
+
 static struct sockaddr_storage ctrl_addr;	/* Control address.  */
 static socklen_t ctrl_addrlen;
 static struct sockaddr_storage data_source;	/* Port address.  */
-- 
2.26.0.292.g33ef6b2f38

