From c9ae29073bee2ed34cc595a7557ceacfa810e50d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sat, 29 Feb 2020 18:40:41 +0100
Subject: [PATCH 39/60] ping, ping6: Silence -Wimplicit-fallthrough

---
 ping/ping.c  | 1 +
 ping/ping6.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/ping/ping.c b/ping/ping.c
index b6db5894..a0a386d5 100644
--- a/ping/ping.c
+++ b/ping/ping.c
@@ -255,6 +255,7 @@ parse_opt (int key, char *arg, struct argp_state *state)
     case ARGP_KEY_NO_ARGS:
       argp_error (state, "missing host operand");
 
+      /* FALLTHROUGH */
     default:
       return ARGP_ERR_UNKNOWN;
     }
diff --git a/ping/ping6.c b/ping/ping6.c
index 8bc92a8f..7992a104 100644
--- a/ping/ping6.c
+++ b/ping/ping6.c
@@ -215,6 +215,7 @@ parse_opt (int key, char *arg, struct argp_state *state)
     case ARGP_KEY_NO_ARGS:
       argp_error (state, "missing host operand");
 
+      /* FALLTHROUGH */
     default:
       return ARGP_ERR_UNKNOWN;
     }
-- 
2.26.0.292.g33ef6b2f38

