From 83e5393554320bf0c67187a64a5a1953dee18e5f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sat, 29 Feb 2020 19:05:18 +0100
Subject: [PATCH 43/60] ftp/cmds.c: Fix return value of remglob

---
 ftp/cmds.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ftp/cmds.c b/ftp/cmds.c
index f9731dc2..3bbaaed9 100644
--- a/ftp/cmds.c
+++ b/ftp/cmds.c
@@ -1037,7 +1037,7 @@ remglob (char **argv, int doswitch)
 	args = argv;
       if ((cp = *++args) == NULL)
 	args = NULL;
-      return cp ? 0 : strdup (cp);
+      return cp ? strdup (cp) : NULL;
     }
   if (ftemp == NULL)
     {
-- 
2.26.0.292.g33ef6b2f38

