From f1cd01e5cef55d10ac7dfe0dd8f0511f603efd3e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sat, 29 Feb 2020 19:11:04 +0100
Subject: [PATCH 44/60] ftp/cmds.c: Fix multipliers for M(ega) and G(iga)

---
 ftp/cmds.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/ftp/cmds.c b/ftp/cmds.c
index 3bbaaed9..2f1cc172 100644
--- a/ftp/cmds.c
+++ b/ftp/cmds.c
@@ -1242,10 +1242,12 @@ sethash (int argc _GL_UNUSED_PARAMETER, char **argv)
 	{
 	case 'g':
 	case 'G':
-	  hashbytes *= 1024;	/* Cascaded multiplication!  */
+	  hashbytes *= 1024 * 1024 * 1024;	/* Cascaded multiplication!  */
+	  break;
 	case 'm':
 	case 'M':
-	  hashbytes *= 1024;
+	  hashbytes *= 1024 * 1024;
+	  break;
 	case 'k':
 	case 'K':
 	  hashbytes *= 1024;
-- 
2.26.0.292.g33ef6b2f38

