From eb3e02fa6cec8f5204fa9d1748eec742a4088164 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sat, 29 Feb 2020 19:13:19 +0100
Subject: [PATCH 45/60] ftp/cmds.c: Silence -Wimplicit-fallthrough

---
 ftp/cmds.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ftp/cmds.c b/ftp/cmds.c
index 2f1cc172..7c40dbdd 100644
--- a/ftp/cmds.c
+++ b/ftp/cmds.c
@@ -2396,6 +2396,7 @@ domap (char *name)
 	      break;
 	    }
 	  /* Fall through, as '$' must be used verbatim.  */
+	  /* FALLTHROUGH */
 	default:
 	  if (*cp2 != *cp1)
 	    {
-- 
2.26.0.292.g33ef6b2f38

