From c9e616ab306593096d993068d3416c78184f219d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sat, 29 Feb 2020 19:14:32 +0100
Subject: [PATCH 46/60] ftp/domacro.c: Silence -Wimplicit-fallthrough

---
 ftp/domacro.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ftp/domacro.c b/ftp/domacro.c
index c9e8207a..d6933402 100644
--- a/ftp/domacro.c
+++ b/ftp/domacro.c
@@ -217,6 +217,7 @@ domacro (int argc, char *argv[])
 		   * use of '$' was detected.  Present input is the
 		   * dollar sign.
 		   */
+		  /* FALLTHROUGH */
 		default:
 		  *cp2++ = *cp1;	/* Copy present character.  */
 		  break;
-- 
2.26.0.292.g33ef6b2f38

