From 0ea7f2d9d0da7e87c8271bc6f0783581966f73e1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sat, 29 Feb 2020 19:18:58 +0100
Subject: [PATCH 47/60] telnetd: Silence unused warning for getterminaltype

---
 telnetd/utility.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/telnetd/utility.c b/telnetd/utility.c
index 20a74f38..db93c205 100644
--- a/telnetd/utility.c
+++ b/telnetd/utility.c
@@ -733,6 +733,7 @@ getterminaltype (char *uname, size_t len)
     }
 #else /* !AUTHENTICATION */
   (void) uname;	/* Silence warning.  */
+  (void) len;	/* Silence warning.  */
 #endif
 
 #ifdef	ENCRYPTION
-- 
2.26.0.292.g33ef6b2f38

