From 4eaf25595387881ca870e716a8c5473e24bae364 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Sat, 29 Feb 2020 19:19:20 +0100
Subject: [PATCH 48/60] telnetd/state.c: Silence -Wimplicit-fallthrough

---
 telnetd/state.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/telnetd/state.c b/telnetd/state.c
index 2184bca1..52ca9c7f 100644
--- a/telnetd/state.c
+++ b/telnetd/state.c
@@ -636,6 +636,7 @@ willoption (int option)
 	      slctab[SLC_XON].defset.flag |= SLC_DEFAULT;
 	      slctab[SLC_XOFF].defset.flag &= ~SLC_LEVELBITS;
 	      slctab[SLC_XOFF].defset.flag |= SLC_DEFAULT;
+	      /* FALLTHROUGH */
 	    case TELOPT_TTYPE:
 	    case TELOPT_SGA:
 	    case TELOPT_NAWS:
-- 
2.26.0.292.g33ef6b2f38

