From 9241033d409e2e519207181ca22cf4fb3d6b6ec4 Mon Sep 17 00:00:00 2001
From: "Alfred M. Szmidt" <ams@gnu.org>
Date: Sat, 4 Apr 2020 23:10:05 +0200
Subject: [PATCH 51/60] ChangeLog: Add missing entries.

---
 ChangeLog | 128 ++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 128 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index 39a545ff..e8d7d0a2 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,131 @@
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	telnet: Use strdup instead malloc+strcpy.
+
+	* telnet/commands.c (tn): Use strdup instead malloc/strcpy.
+
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	telnet: Fix -Wsign-compare in suboption.
+
+	* telnet/telnet.c (suboption): Explicit cast to int to silence
+	-Wsign-compare.
+
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	telnetd: Silence -Wimplicit-fallthrough.
+
+	* telnetd/state.c (willoption): Squeelch warning.
+
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	telnetd: Silence unused warning for getterminaltype.
+
+	* telnetd/utility.c (getterminaltype): Squeelch warning.
+
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	ftp: Silence -Wimplicit-fallthrough warning.
+
+	* ftp/domacro.c (domacro): Squeelch warning.
+
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	ftp: Silence -Wimplicit-fallthrough.
+
+	* ftp/cmds.c (domap): Squeelch warning.
+
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	ftp: Fix multipliers for M(ega) and G(iga).
+
+	This is *not* a fix as the comment accidentally states.  It is to
+	silence -Wimplicit-fallthrough and to avoid multiple
+	multiplications in case of a non-optimized build.  The HASHBYTES
+	type needs to be reviewed later as g/G only allows digits 1 and 2
+	without integer overflow, which is undefined behavior.
+
+	* ftp/cmds.c (sethash): Remove cascaded multiplications.
+
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	ftp: Fix return value of remglob.
+
+	* ftp/cmds.c (remglob): Turn around NULL check.
+
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	ftp: Fix possible buffer overflow.
+
+	* ftp/ftp.c (hookup): Terminate string after strncpy.
+
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	ftp: Fix 2x misuse of strncpy (read buffer overflow).
+
+	* ftp/ftp.c (pswitch): Correctly set the terminating NUL.
+
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	talk: Fix uninitialized variable NREADY.
+
+	* talk/ctl_transact.c (ctl_transact): Initialize NREADY to 0.
+
+2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	ping, ping6: Silence -Wimplicit-fallthrough.
+
+	* ping/ping.c (parse_opt): Squeelch warnings.
+	* ping/ping6.c (parse_opt): Likewise.
+
+2020-02-17  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	ping6: Fix memleak in ping_set_dest.
+
+	* ping/ping6.c (ping_set_dest): Rerrange code to avoid memory
+	leak.
+	* ping/ping6.h (ping_set_dest): Change type; update callees.
+
+2020-02-17  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	ping: Fix memleak in ping_set_dest.
+
+	* ping/libping.c (ping_set_dest): Rerrange code to avoid memory
+	leak.
+	* ping/ping.h (ping_set_dest): Change type; update callees.
+
+2020-02-16  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	libls: Remove unused variable kflag.
+
+	* libls/ls.c (ls_main): Remove unused variable.
+
+2020-02-16  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	ftpd: Fix multiple definition (GCC 10).
+
+	* ftpd/extern.h (errcatch): Declare as extern.
+	* ftpd/ftpd.c (errcatch): Define variable.
+
+2020-02-16  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	telnetd: Fix multiple definition (GCC 10).
+
+	* telnetd/utility.c (not42): Declare as extern.
+
+2020-02-16  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	* src/rcp.c (tolocal): Remove unused variable LEN.
+
+2020-02-16  Tim Rühsen <tim.ruehsen@gmx.de>
+
+	telnet: Fix silent truncation (off-by-one check)
+
+	If the DISPLAY variable had exactly 44 bytes, the SE byte (end sub
+	negotiation) was silently truncated.
+
+	* telnet/telnet.c (suboption): Use >= instead of >.
+
 2020-02-21  Mats Erik Andersson  <gnu@gisladisker.se>
 
 	Various compiler warnings, getting closer to `-Werror'.
-- 
2.26.0.292.g33ef6b2f38

