From b6aa17e9ea4e33f24d93402d33a975024d40fea7 Mon Sep 17 00:00:00 2001
From: "Alfred M. Szmidt" <ams@gnu.org>
Date: Sat, 4 Apr 2020 23:27:35 +0200
Subject: [PATCH 52/60] ping/ping6.c (print_packet_too_big): Use %u when
 printing MTU.

---
 ChangeLog    | 4 ++++
 ping/ping6.c | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index e8d7d0a2..30342bf7 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2020-04-04  Alfred M. Szmidt  <ams@gnu.org>
+
+	* ping/ping6.c (print_packet_too_big): Use %u when printing MTU.
+
 2020-02-29  Tim Rühsen <tim.ruehsen@gmx.de>
 
 	telnet: Use strdup instead malloc+strcpy.
diff --git a/ping/ping6.c b/ping/ping6.c
index 7992a104..114b9c0c 100644
--- a/ping/ping6.c
+++ b/ping/ping6.c
@@ -656,7 +656,7 @@ print_dst_unreach (struct icmp6_hdr *icmp6)
 static void
 print_packet_too_big (struct icmp6_hdr *icmp6)
 {
-  printf ("Packet too big, mtu=%d\n", icmp6->icmp6_mtu);
+  printf ("Packet too big, mtu=%u\n", icmp6->icmp6_mtu);
 }
 
 static struct icmp_code_descr icmp_time_exceeded_desc[] = {
-- 
2.26.0.292.g33ef6b2f38

