diff -rup pypy-pypy-f66246c46ca3/rpython/tool/ansi_print.py pypy-pypy-f66246c46ca3/rpython/tool/ansi_print.py
--- pypy-pypy-f66246c46ca3/rpython/tool/ansi_print.py	2013-05-27 10:35:37.648237156 +0200
+++ pypy-pypy-f66246c46ca3/rpython/tool/ansi_print.py	2013-05-28 10:15:58.200426205 +0200
@@ -25,7 +25,7 @@ class AnsiLog:
         self.kw_to_color = self.KW_TO_COLOR.copy()
         self.kw_to_color.update(kw_to_color)
         self.file = file
-        self.fancy = True
+        self.fancy = False
         self.isatty = getattr(sys.stderr, 'isatty', lambda: False)
         if self.fancy and self.isatty(): 
             self.mandelbrot_driver = Driver()

