--- a/mk_distinfo.py	2018-07-27 11:11:25.940734322 +0000
+++ b/mk_distinfo.py	2018-07-27 13:02:44.774293659 +0000
@@ -25,8 +25,8 @@
 
 
 # Parse the command line.
-if len(sys.argv) != 3:
-    error("usage: {0} dist-info installed".format(sys.argv[0]))
+if len(sys.argv) not in [3, 4]:
+    error("usage: {0} dist-info installed [install-root]".format(sys.argv[0]))
 
 distinfo_dir = sys.argv[1]
 installed_fn = sys.argv[2]
@@ -98,6 +98,8 @@
         else:
             fn_name = fn
 
+        if len(sys.argv) > 3 and not fn.startswith(sys.argv[3]):
+            fn = sys.argv[3] + os.path.sep + fn
         fn_f = open(fn, 'rb')
         data = fn_f.read()
         fn_f.close()
--- a/configure.py	2018-07-27 11:11:25.960735077 +0000
+++ b/configure.py	2018-07-27 13:06:53.600110981 +0000
@@ -1736,7 +1736,7 @@
         # The command to run to generate the .dist-info directory.
         distinfo_dir = os.path.join(target_config.pyqt_module_dir,
                 'PyQt5-' + PYQT_VERSION_STR + '.dist-info')
-        mk_distinfo = sys.executable + ' mk_distinfo.py $(INSTALL_ROOT)' + distinfo_dir + ' installed.txt'
+        mk_distinfo = sys.executable + ' mk_distinfo.py $(INSTALL_ROOT)' + distinfo_dir + ' installed.txt $(INSTALL_ROOT)'
 
         out_f.write('''
 distinfo.extra = %s
--- a/mk_distinfo.py	2018-07-27 14:09:42.669145472 +0000
+++ b/mk_distinfo.py	2018-07-27 14:36:19.808959563 +0000
@@ -78,6 +78,8 @@
 
 for name in installed:
     native_name = name.replace('/', os.sep)
+    if len(sys.argv) > 3 and not native_name.startswith(sys.argv[3]):
+            native_name = sys.argv[3] + os.path.sep + native_name
     if os.path.isdir(native_name):
         all_fns = []
 
