diff -Naur binutils-2.27/gas/config/tc-mips.c binutils-2.27.new/gas/config/tc-mips.c
--- binutils-2.27/gas/config/tc-mips.c	2017-01-19 22:57:05.485930117 +0800
+++ binutils-2.27.new/gas/config/tc-mips.c	2017-01-19 22:56:54.619223474 +0800
@@ -4046,8 +4046,12 @@
     static expressionS bak_offset_expr;
     static bfd_reloc_code_real_type bak_offset_reloc[3] ;
 
-    if ((strcmp (insn.insn_mo->name, "ll") == 0)
-        ||(strcmp (insn.insn_mo->name, "lld") == 0)) {
+    struct insn_label_list *is_label;
+    is_label = seg_info (now_seg)->label_list;
+
+    if (((strcmp (insn.insn_mo->name, "ll") == 0)
+        ||(strcmp (insn.insn_mo->name, "lld") == 0))
+	&& ( strcmp (history[0].insn_mo->name, "sync") || is_label)) {
            bak_imm_expr = imm_expr;
            bak_offset_expr = offset_expr;
 
