From 75d7e334224ca689eff8cfcdff9675f342beb3e1 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Thu, 27 Aug 2026 12:18:00 +0800
Subject: [PATCH 01/14] FEDORA: not build demos

---
 Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index ebd376a5..397e0f4d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -27,7 +27,7 @@ SUBDIRS	= bindings bitmaps \
              include \
              tools \
              clients \
-             doc \
-             demos
+             doc
+
 AUTOMAKE_OPTIONS = 1.4
 ACLOCAL_AMFLAGS = -I .
-- 
2.55.0

