From 4a9e64534e2c64709a5e589f976f6d8fe3cd0b79 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Thu, 4 Jul 2024 22:52:42 +0800
Subject: [PATCH] enable build id tagging

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 22051f190..257549255 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -933,6 +933,7 @@ set(ZIG_BUILD_ARGS
   "-Dcpu=${ZIG_TARGET_MCPU}"
 
   -Denable-llvm
+  "-Dbuild-id=sha1"
   "-Dconfig_h=${ZIG_CONFIG_H_OUT}"
 
   -Dno-langref
-- 
2.45.2.windows.1

