--- a/src/plugins/cppeditor/cppquickfixes.cpp
+++ b/src/plugins/cppeditor/cppquickfixes.cpp
@@ -4395,10 +4395,10 @@
     using Flag = GenerateGetterSetterOp::GenerateFlag;
     constexpr static Flag ColumnFlag[] = {
-        static_cast<Flag>(-1),
+        static_cast<Flag>(0xff),
         Flag::GenerateGetter,
         Flag::GenerateSetter,
         Flag::GenerateSignal,
         Flag::GenerateReset,
         Flag::GenerateProperty,
         Flag::GenerateConstantProperty,
     };
