From 04d16a52994a3b65e592718dd6a14fe4629501b6 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Tue, 25 Aug 2020 12:35:08 +0800
Subject: [PATCH] Let pkgconfig find `lua5.3`

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 356731e..0fa23b1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/lua5.4/lua.h")
   find_package(PkgConfig)
   if(PkgConfig_FOUND)
-    foreach(pkg lua lua54 lua53 lua52 luajit lua51)
+    foreach(pkg lua5.3 lua lua54 lua53 lua52 luajit lua51)
       pkg_check_modules(LUA IMPORTED_TARGET GLOBAL ${pkg})
       if(LUA_FOUND)
 	break()
-- 
2.45.0

