From 078343e75d79609498c879345500c625a05e32e3 Mon Sep 17 00:00:00 2001
From: Mag Mell <sakiiily@aosc.io>
Date: Mon, 13 Apr 2026 15:30:41 +0800
Subject: [PATCH] Let pkgconfig find `lua5.3`

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 42f2ebb..03518c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@ set(LUA_VERSION "lua" CACHE STRING "lua version")
 if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/lua5.4/lua.h")
   find_package(PkgConfig)
   if(PkgConfig_FOUND)
-    foreach(pkg ${LUA_VERSION} lua54 lua53 lua52 luajit lua51)
+    foreach(pkg ${LUA_VERSION} lua5.3 lua54 lua53 lua52 luajit lua51)
       pkg_check_modules(LUA IMPORTED_TARGET GLOBAL ${pkg})
       if(LUA_FOUND)
 	break()
-- 
2.52.0

