From c7a24fd5718f60de7930d89ef9045068bc6011db Mon Sep 17 00:00:00 2001
From: Misaka 0x4e21 <misaka4e21@gmail.com>
Date: Sun, 2 Nov 2025 00:53:10 +0800
Subject: [PATCH 1/1] Strip automatic update checking

---
 src/mainwindow.cpp | 260 +--------------------------------------------
 1 file changed, 2 insertions(+), 258 deletions(-)

diff --git a/src/mainwindow.cpp b/src/mainwindow.cpp
index f62844c7..f80b9b7c 100644
--- a/src/mainwindow.cpp
+++ b/src/mainwindow.cpp
@@ -384,7 +384,6 @@ Main::Main(QWidget *parent) : QMainWindow(parent)
     // After startup, schedule looking for updates AFTER
     // release notes have been downloaded
     // (avoid race condition with simultanously receiving cookies)
-    checkUpdatesAfterReleaseNotes = true;
 
 #if defined(VYM_DBUS)
     // Announce myself on DBUS
@@ -2947,14 +2946,6 @@ void Main::setupSettingsActions()
 
     QAction *a;
 
-    a = new QAction(
-        tr("Check for release notes and updates", "Settings action"), this);
-    a->setCheckable(true);
-    a->setChecked(settings.value("/downloads/enabled", true).toBool());
-    connect(a, SIGNAL(triggered()), this, SLOT(settingsToggleDownloads()));
-    settingsMenu->addAction(a);
-    actionSettingsToggleDownloads = a;
-
     a = new QAction(tr("Set author for new maps", "Settings action") + "...",
                     this);
     connect(a, SIGNAL(triggered()), this, SLOT(settingsDefaultMapAuthor()));
@@ -3149,15 +3140,6 @@ void Main::setupHelpActions()
     connect(a, SIGNAL(triggered()), this, SLOT(helpDemo()));
     helpMenu->addSeparator();
 
-    a = new QAction(tr("Download and show release notes", "Help action"), this);
-    helpMenu->addAction(a);
-    connect(a, SIGNAL(triggered()), this, SLOT(checkReleaseNotes()));
-
-    a = new QAction(tr("Check, if updates are available", "Help action"), this);
-    helpMenu->addAction(a);
-    connect(a, SIGNAL(triggered()), this, SLOT(checkUpdates()));
-    helpMenu->addSeparator();
-
     a = new QAction(tr("Show keyboard shortcuts", "Help action"), this);
     helpMenu->addAction(a);
     connect(a, SIGNAL(triggered()), this, SLOT(helpShortcuts()));
@@ -7202,280 +7184,42 @@ void Main::callMacro()
 
 void Main::downloadReleaseNotesFinished()
 {
-    DownloadAgent *agent = static_cast<DownloadAgent *>(sender());
-    QString s;
-
-    if (agent->isSuccess()) {
-        QString page;
-        if (agent->isSuccess()) {
-            if (loadStringFromDisk(agent->getDestination(), page)) {
-                ShowTextDialog dia(this);
-                dia.setText(page);
-                dia.exec();
-
-                // Don't load the release notes automatically again
-                settings.setValue("/downloads/releaseNotes/shownVersion",
-                                  vymVersion);
-            }
-        }
-    }
-    else {
-        statusMessage("Downloading release notes failed.");
-        if (debug) {
-            qDebug() << "Main::downloadReleaseNotesFinished ";
-            qDebug() << "  result: failed";
-            qDebug() << "     msg: " << agent->getResultMessage();
-        }
-    }
-    agent->deleteLater();
-
-    if (checkUpdatesAfterReleaseNotes)
-    {
-        // After startup we want to check also for updates, but only after
-        // releasenotes are there (and we have a cookie already)
-        checkUpdatesAfterReleaseNotes = false;
-        checkUpdates();
-    }
 }
 
 QUrl Main::serverUrl(const QString &scriptName)
 {
     // Local URL for testing only
     // QString("http://localhost/release-notes.php?vymVersion=%1") /
-    return QUrl(
-        QString("http://www.insilmaril.de/vym/%1?"
-                    "vymVersion=%2"
-                    "&config=darkTheme=%3+localeName=%4+buildDate=%5+codeQuality='%6'+codeName='%7'")
-            .arg(scriptName)
-            .arg(vymVersion)
-            .arg(usingDarkTheme)
-            .arg(localeName)
-            .arg(vymBuildDate)
-            .arg(vymCodeQuality)
-            .arg(vymCodeName)
-            );
+    return QUrl(QString("about:blank"));
 }
 
 void Main::checkReleaseNotesAndUpdates ()
 {
     // Called once after startup
     // checkUpdatesAfterReleaseNotes is already true then
-    checkReleaseNotes();
 }
 
 void Main::checkReleaseNotes ()
 {
-    bool userTriggered;
-    if (qobject_cast<QAction *>(sender()))
-        userTriggered = true;
-    else
-        userTriggered = false;
-
-    if (downloadsEnabled()) {
-        if (userTriggered ||
-            versionLowerThanVym(
-                settings.value("/downloads/releaseNotes/shownVersion", "0.0.1")
-                    .toString())) {
-            DownloadAgent *agent = new DownloadAgent(serverUrl("release-notes.php"));
-            connect(agent, SIGNAL(downloadFinished()), this,
-                    SLOT(downloadReleaseNotesFinished()));
-            QTimer::singleShot(0, agent, SLOT(execute()));
-        }
-    }
-    else {
-        // No downloads enabled
-        if (userTriggered) {
-            // Notification: vym could not download release notes
-            QMessageBox::warning(
-                0, tr("Warning"),
-                tr("Please allow vym to download release notes!"));
-            if (downloadsEnabled(userTriggered))
-                checkUpdates();
-        }
-    }
 }
 
 bool Main::downloadsEnabled(bool userTriggered)
 {
-    bool result;
-    if (!userTriggered &&
-        settings.value("/downloads/enabled", false).toBool()) {
-        result = true;
-    }
-    else {
-        QDate lastAsked =
-            settings.value("/downloads/permissionLastAsked", QDate(1970, 1, 1))
-                .toDate();
-        if (userTriggered ||
-            !settings.contains("/downloads/permissionLastAsked") ||
-            lastAsked.daysTo(QDate::currentDate()) > 7) {
-            QString infotext;
-            infotext =
-                tr("<html>"
-                   "<h3>Do you allow vym to check online for updates or "
-                   "release notes?</h3>"
-                   "If you allow, vym will "
-                   "<ul>"
-                   "<li>check once for release notes</li>"
-                   "<li>check regulary for updates and notify you in case you "
-                   "should update, e.g. if there are "
-                   "important bug fixes available</li>"
-                   "<li>receive a cookie with a random ID and send some anonymous data, like:"
-                   "<ul>"
-                   "<li>vym version</li>"
-                   "<li>platform name and the ID (e.g. \"Windows\" or \"Linux\")</li>"
-                   "<li>if you are using dark theme</li>"
-                   "</ul>"
-                   "This data is sent to me, Uwe Drechsel."
-                   "<p>As vym developer I am motivated to see "
-                   "many people using vym. Of course I am curious to see, on "
-                   "which system vym is used. Maintaining each "
-                   "of the systems requires a lot of my (spare) time.</p> "
-                   "<p>No other data than above will be sent, especially no "
-                   "private data will be collected or sent."
-                   "(Check the source code, if you don't believe.)"
-                   "</p>"
-                   "</li>"
-                   "</ul>"
-                   "If you do not allow, "
-                   "<ul>"
-                   "<li>nothing will be downloaded and especially I will "
-                   "<b>not be motivated</b> "
-                   "to spend some more thousands of hours on developing a free "
-                   "software tool."
-                   "</ul>"
-                   "Please allow vym to check for updates :-)");
-            QMessageBox mb(vymName, infotext, QMessageBox::Information,
-                           QMessageBox::Yes | QMessageBox::Default,
-                           QMessageBox::No | QMessageBox::Escape,
-                           QMessageBox::NoButton);
-
-            mb.setButtonText(QMessageBox::Yes, tr("Allow"));
-            mb.setButtonText(QMessageBox::No, tr("Do not allow"));
-            switch (mb.exec()) {
-            case QMessageBox::Yes: {
-                result = true;
-                QMessageBox msgBox;
-                msgBox.setText(tr("Thank you for enabling downloads!"));
-                msgBox.setStandardButtons(QMessageBox::Close);
-                msgBox.setIconPixmap(QPixmap(":/flag-face-smile.svg"));
-                msgBox.exec();
-                break;
-                                   }
-            default:
-                result = false;
-                QMessageBox msgBox;
-                msgBox.setText(tr("That's ok, though I would be happy to see many users working with vym and also on which platforms."));
-                msgBox.setStandardButtons(QMessageBox::Close);
-                msgBox.setIconPixmap(QPixmap(":/flag-face-sad.svg"));
-                msgBox.exec();
-                break;
-            }
-        }
-        else
-            result = false;
-        actionSettingsToggleDownloads->setChecked(result);
-        settings.setValue("/downloads/enabled", result);
-        settings.setValue("/downloads/permissionLastAsked",
-                          QDate::currentDate().toString(Qt::ISODate));
-    }
-    return result;
+    return false;
 }
 
 void Main::downloadUpdatesFinished(bool userTriggered)
 {
-    DownloadAgent *agent = static_cast<DownloadAgent *>(sender());
-    QString s;
-
-    if (agent->isSuccess()) {
-        ShowTextDialog dia;
-        dia.setWindowTitle(vymName + " - " + tr("Update information"));
-        QString page;
-        if (loadStringFromDisk(agent->getDestination(), page)) {
-            if (page.contains("vymisuptodate")) {
-                statusMessage(tr("vym is up to date.", "MainWindow"));
-                if (userTriggered) {
-                    // Notification: vym is up to date!
-                    dia.setHtml(page);
-                    dia.exec();
-                }
-            }
-            else if (page.contains("vymneedsupdate")) {
-                // Notification: updates available
-                dia.setHtml(page);
-                dia.exec();
-            }
-            else {
-                // Notification: Unknown version found
-                dia.setHtml(page);
-                dia.exec();
-            }
-
-            // Prepare to check again later
-            settings.setValue("/downloads/updates/lastChecked",
-                              QDate::currentDate().toString(Qt::ISODate));
-        }
-        else
-            statusMessage("Couldn't load update page from " +
-                          agent->getDestination());
-    }
-    else {
-        statusMessage("Check for updates failed.");
-        if (debug) {
-            qDebug() << "Main::downloadUpdatesFinished ";
-            qDebug() << "  result: failed";
-            qDebug() << "     msg: " << agent->getResultMessage();
-        }
-    }
-    agent->deleteLater();
 }
 
 void Main::downloadUpdatesFinishedInt() { downloadUpdatesFinished(true); }
 
 void Main::downloadUpdates(bool userTriggered)
 {
-    DownloadAgent *agent = new DownloadAgent(serverUrl("updates.php"));
-    if (userTriggered)
-        connect(agent, SIGNAL(downloadFinished()), this,
-                SLOT(downloadUpdatesFinishedInt()));
-    else
-        connect(agent, SIGNAL(downloadFinished()), this,
-                SLOT(downloadUpdatesFinished()));
-    statusMessage(tr("Checking for updates...", "MainWindow"));
-    QTimer::singleShot(0, agent, SLOT(execute()));
 }
 
 void Main::checkUpdates()
 {
-    bool userTriggered;
-    if (qobject_cast<QAction *>(sender()))
-        userTriggered = true;
-    else
-        userTriggered = false;
-
-    if (downloadsEnabled()) {
-        // Too much time passed since last update check?
-        QDate lastChecked =
-            settings.value("/downloads/updates/lastChecked", QDate(1970, 1, 1))
-                .toDate();
-        if (!lastChecked.isValid())
-            lastChecked = QDate(1970, 1, 1);
-        if (lastChecked.daysTo(QDate::currentDate()) >
-                settings.value("/downloads/updates/checkInterval", 3).toInt() ||
-            userTriggered == true) {
-            downloadUpdates(userTriggered);
-        }
-    }
-    else {
-        // No downloads enabled
-        if (userTriggered) {
-            // Notification: vym could not check for updates
-            QMessageBox::warning(0, tr("Warning"),
-                                 tr("Please allow vym to check for updates!"));
-            if (downloadsEnabled(userTriggered))
-                checkUpdates();
-        }
-    }
 }
 
 void Main::escapePressed()
-- 
2.51.1

