From: Hilko Bengen <bengen@debian.org>
Date: Tue, 14 Mar 2023 00:02:26 +0100
Subject: Deal with architectures that lack a OCaml native compiler

---
 Makefile.in | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 701a79e..fdd0b1b 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -20,6 +20,8 @@
 PACKAGE		= @PACKAGE_NAME@
 VERSION		= @PACKAGE_VERSION@
 
+OCAMLC          = @OCAMLC@
+OCAMLOPT        = @OCAMLOPT@
 OCAMLFIND	= @OCAMLFIND@
 OCAMLMKLIB	= @OCAMLMKLIB@
 OCAMLDOC	= @OCAMLDOC@
@@ -43,13 +45,18 @@ OCAMLDOCFLAGS	= -html -sort
 
 export LIBRARY_PATH = .
 
-all:	mlaugeas.cma mlaugeas.cmxa test_augeas
+all:	mlaugeas.cma test_augeas
+ifneq ($(OCAMLOPT),no)
+all: 	mlaugeas.cmxa
+endif
 
 mlaugeas.cma: augeas-c.o augeas.cmo
 	ocamlmklib $(LDFLAGS) $(LIBS) -o mlaugeas $^
 
+ifneq ($(OCAMLOPT),no)
 mlaugeas.cmxa: augeas-c.o augeas.cmx
 	ocamlmklib $(LDFLAGS) $(LIBS) -o mlaugeas $^
+endif
 
 # Test program.
 
@@ -88,7 +95,8 @@ install:
 	ocamlfind install \
 	  -ldconf ignore -destdir $(DESTDIR)$(OCAMLLIB) \
 	  augeas \
-	  META *.mli *.cmx *.cma *.cmxa *.a augeas.cmi *.so
+	  META *.mli *.cma *.a augeas.cmi *.so \
+	  $(if $(findstring no,$(OCAMLOPT)),,*.cmx *.cmxa)
 
 # Standard rules.
 
@@ -96,8 +104,10 @@ install:
 	$(OCAMLFIND) ocamlc $(OCAMLCFLAGS) $(OCAMLCPACKAGES) -c $<
 .ml.cmo:
 	$(OCAMLFIND) ocamlc $(OCAMLCFLAGS) $(OCAMLCPACKAGES) -c $<
+ifneq ($(OCAMLOPT),no)
 .ml.cmx:
 	$(OCAMLFIND) ocamlopt $(OCAMLOPTFLAGS) $(OCAMLOPTPACKAGES) -c $<
+endif
 
 depend: .depend
 
