From 993e9ee531fb4edd2dd486ee3c4590bd477d84db Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Tue, 20 Dec 2022 01:01:11 -0500
Subject: [PATCH 1/2] common/guided_filter: remove broken ifdefs on ppc

---
 src/common/guided_filter.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/common/guided_filter.h b/src/common/guided_filter.h
index 316a8f8bf7..23b99a3a58 100644
--- a/src/common/guided_filter.h
+++ b/src/common/guided_filter.h
@@ -23,12 +23,8 @@
 #ifdef NO_WARN_X86_INTRINSICS
 #include <xmmintrin.h>
 #else
-#define NO_WARN_X86_INTRINSICS 1
 #include <xmmintrin.h>
-#undef NO_WARN_X86_INTRINSICS
 #endif // NO_WARN_X86_INTRINSICS
-#else
-#include <xmmintrin.h>
 #endif // __PPC64__
 #endif
 
-- 
2.38.1

