--- darktable-3.4.0/src/common/guided_filter.h	2020-12-16 02:22:52.000000000 -0600
+++ darktable-3.4.0.nonx86/src/common/guided_filter.h	2021-01-31 01:06:15.738369423 -0600
@@ -22,12 +22,8 @@
 #ifdef NO_WARN_X86_INTRINSICS
 #include <xmmintrin.h>
 #else
-#define NO_WARN_X86_INTRINSICS 1
 #include <xmmintrin.h>
-#undef NO_WARN_X86_INTRINSICS
 #endif // NO_WARN_X86_INTRINSICS
-#else
-#include <xmmintrin.h>
 #endif // __PPC64__
 
 #include "common/darktable.h"
