From afd77d8ed47dfa8c1193277bf7f8b7e3941fd33a Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Fri, 30 Oct 2020 01:03:45 -0500
Subject: [PATCH] meta: allow compiling on mips64el

---
 src/is_supported_platform.h | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/is_supported_platform.h b/src/is_supported_platform.h
index e8b968776..8aab340a7 100644
--- a/src/is_supported_platform.h
+++ b/src/is_supported_platform.h
@@ -36,6 +36,12 @@
 #define DT_SUPPORTED_ARMv8A 0
 #endif
 
+#if defined(__mips__) && defined(__mips64)
+#define DT_SUPPORTED_MIPS64 1
+#else
+#define DT_SUPPORTED_MIPS64 0
+#endif
+
 #if defined(__PPC64__)
 #define DT_SUPPORTED_PPC64 1
 #else
@@ -46,8 +52,8 @@
 #error "Looks like hardware platform detection macros are broken?"
 #endif
 
-#if !DT_SUPPORTED_X86 && !DT_SUPPORTED_ARMv8A && !DT_SUPPORTED_PPC64
-#error "Unfortunately we only work on amd64, ARMv8-A and PPC64 (64-bit little-endian only)."
+#if !DT_SUPPORTED_X86 && !DT_SUPPORTED_ARMv8A && !DT_SUPPORTED_PPC64 && !DT_SUPPORTED_MIPS64
+#error "Unfortunately we only work on amd64, ARMv8-A, MIPS64 and PPC64 (64-bit little-endian only)."
 #endif
 
 #undef DT_SUPPORTED_PPC64
-- 
2.28.0

