--- bsd-finger-0.16/finger/util.c.pts	Tue Sep 28 18:53:58 1999
+++ bsd-finger-0.16/finger/util.c	Tue Dec 21 14:12:56 1999
@@ -71,8 +71,13 @@
 	}
 	snprintf(tbuf, TBUFLEN, "%s/%s", _PATH_DEV, w->tty);
 	if (stat(tbuf, &sb) < 0) {
+	    switch (errno) {
+	    case ENOENT:
+		break;
+	    default:
 		eprintf("finger: %s: %s\n", tbuf, strerror(errno));
 		return;
+	    }
 	}
 	w->idletime = now < sb.st_atime ? 0 : now - sb.st_atime;
 
