--- bsd-finger-0.16/finger/util.c.broken	Wed May 31 16:56:46 2000
+++ bsd-finger-0.16/finger/util.c	Wed May 31 16:57:45 2000
@@ -350,6 +350,7 @@
 		eprintf("finger: Out of space.\n");
 		exit(1);
 	}
+	memset(p, 0, sizeof(PERSON));
 	return(p);
 }
 
@@ -362,6 +363,7 @@
 		eprintf("finger: Out of space.\n");
 		exit(1);
 	}
+	memset(w, 0, sizeof(WHERE));
 	if (pn->whead == NULL)
 		pn->whead = pn->wtail = w;
 	else {
