--- bsd-finger-0.17-pre20000412/finger/finger.c.exact	Sat Dec 18 11:41:51 1999
+++ bsd-finger-0.17-pre20000412/finger/finger.c	Sun Aug 27 21:17:57 2000
@@ -238,15 +238,14 @@
 	 * traverse the list of possible login names and check the login name
 	 * and real name against the name specified by the user.
 	 */
-	if (mflag) {
-		for (i = 0; i < argc; i++)
-			if (used[i] >= 0 && (pw = getpwnam(argv[i]))) {
-				if (!check_nofinger(pw)) {
-					enter_person(pw);
-					used[i] = 1;
-				}
+	for (i = 0; i < argc; i++)
+		if (used[i] >= 0 && (pw = getpwnam(argv[i]))) {
+			if (!check_nofinger(pw)) {
+				enter_person(pw);
+				used[i] = 1;
 			}
+		}
-	} else for (pw = getpwent(); pw; pw = getpwent())
+	if(!mflag) for (pw = getpwent(); pw; pw = getpwent())
 		for (i = 0; i < argc; i++)
 			if (used[i] >= 0 &&
 			    (!strcasecmp(pw->pw_name, argv[i]) ||
