--- bsd-finger-0.17/finger/sprint.c	2009-11-02 13:11:20.000000000 +0530
+++ bsd-finger-0.17/finger/sprint.c.add-host-column	2009-11-02 13:10:43.000000000 +0530
@@ -89,7 +89,7 @@ void sflag_print(void) {
 	if (maxlname + maxrname < space-2) { maxlname++; maxrname++; }
 
 	(void)xprintf("%-*s %-*s %s\n", maxlname, "Login", maxrname,
-	    "Name", " Tty      Idle  Login Time   Office     Office Phone");
+	    "Name", " Tty      Idle  Login Time   Office     Office Phone   Host");
 	for (cnt = 0; cnt < entries; ++cnt) {
 		pn = list[cnt];
 		for (w = pn->whead; w != NULL; w = w->next) {
@@ -118,17 +118,18 @@ void sflag_print(void) {
 			else
 				(void)xprintf(" %.5s", p + 11);
 office:
-			if (w->host[0] != '\0') {
-				xprintf(" (%s)", w->host);
-			} else {
 			if (pn->office)
 				(void)xprintf(" %-10.10s", pn->office);
-			else if (pn->officephone)
+			else
 				(void)xprintf(" %-10.10s", " ");
 			if (pn->officephone)
-				(void)xprintf(" %-.14s",
+				(void)xprintf(" %-14.14s",
 				    prphone(pn->officephone));
-			}
+			else if (w->host[0] != '\0')
+				(void)xprintf(" %-14.14s", " ");
+
+			if (w->host[0] != '\0')
+				xprintf(" (%s)", w->host);
 			xputc('\n');
 		}
 	}
