--- bsd-finger-0.17-pre20000412/finger/lprint.c.orig	Tue Sep 14 06:51:11 1999
+++ bsd-finger-0.17-pre20000412/finger/lprint.c	Sun Feb  4 17:57:08 2001
@@ -48,7 +48,7 @@
 #include <sys/types.h>
 #include <sys/file.h>
 #include <sys/stat.h>
-#include <sys/time.h>
+#include <time.h>
 #include "finger.h"
 
 static void lprint(PERSON *pn);
--- bsd-finger-0.17-pre20000412/finger/sprint.c.orig	Sun Dec 12 13:59:33 1999
+++ bsd-finger-0.17-pre20000412/finger/sprint.c	Sun Feb  4 17:57:22 2001
@@ -40,7 +40,7 @@
 #endif /* not lint */
 
 #include <sys/types.h>
-#include <sys/time.h>
+#include <time.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
