diff -up ebtables-v2.0.10-4/ebtables.8.lockdirfix ebtables-v2.0.10-4/ebtables.8
--- ebtables-v2.0.10-4/ebtables.8.lockdirfix	2016-01-18 11:13:21.707069702 -0500
+++ ebtables-v2.0.10-4/ebtables.8	2016-01-18 11:13:40.554953365 -0500
@@ -1103,7 +1103,7 @@ arp message and the hardware address len
 .br
 .SH FILES
 .I /etc/ethertypes
-.I /var/lib/ebtables/lock
+.I /run/ebtables.lock
 .SH ENVIRONMENT VARIABLES
 .I EBTABLES_ATOMIC_FILE
 .SH MAILINGLISTS
diff -up ebtables-v2.0.10-4/INSTALL.lockdirfix ebtables-v2.0.10-4/INSTALL
--- ebtables-v2.0.10-4/INSTALL.lockdirfix	2016-01-18 11:15:31.458268826 -0500
+++ ebtables-v2.0.10-4/INSTALL	2016-01-18 11:15:53.890130367 -0500
@@ -31,7 +31,7 @@ WHAT GETS INSTALLED AND WHAT OPTIONS ARE
   copied to /etc/rc.d/init.d (change with option INITDIR)
 - The ebtables configuration file (ebtables-config) is copied to /etc/sysconfig
 - ebtables can use a lock file to enable concurrent execution of the ebtables
-  tool. The standard location of the lock file is /var/lib/ebtables/lock.
+  tool. The standard location of the lock file is /run/ebtables.lock.
   Include LOCKFILE=<<path-to-file>> if you want to use another file.
 
 That's all
diff -up ebtables-v2.0.10-4/libebtc.c.lockdirfix ebtables-v2.0.10-4/libebtc.c
--- ebtables-v2.0.10-4/libebtc.c.lockdirfix	2016-01-18 11:12:14.347485472 -0500
+++ ebtables-v2.0.10-4/libebtc.c	2016-01-18 11:13:06.515163472 -0500
@@ -134,8 +134,8 @@ void ebt_list_extensions()
 }
 
 #ifndef LOCKFILE
-#define LOCKDIR "/var/lib/ebtables"
-#define LOCKFILE LOCKDIR"/lock"
+#define LOCKDIR "/run"
+#define LOCKFILE LOCKDIR"/ebtables.lock"
 #endif
 static int lockfd = -1, locked;
 int use_lockfd;
diff -up ebtables-v2.0.10-4/Makefile.lockdirfix ebtables-v2.0.10-4/Makefile
--- ebtables-v2.0.10-4/Makefile.lockdirfix	2016-01-18 11:14:10.715767201 -0500
+++ ebtables-v2.0.10-4/Makefile	2016-01-18 11:15:20.506336425 -0500
@@ -5,7 +5,7 @@ PROGRELEASE:=4
 PROGVERSION_:=2.0.10
 PROGVERSION:=$(PROGVERSION_)-$(PROGRELEASE)
 PROGDATE:=December\ 2011
-LOCKFILE?=/var/lib/ebtables/lock
+LOCKFILE?=/run/ebtables.lock
 LOCKDIR:=$(shell echo $(LOCKFILE) | sed 's/\(.*\)\/.*/\1/')/
 
 # default paths
