diff --git a/js/src/threading/ExclusiveData.h b/js/src/threading/ExclusiveData.h
index 732ef91d6f..f21ca988b0 100644
--- a/js/src/threading/ExclusiveData.h
+++ b/js/src/threading/ExclusiveData.h
@@ -85,6 +85,8 @@ class ExclusiveData
 
     ExclusiveData(const ExclusiveData&) = delete;
     ExclusiveData& operator=(const ExclusiveData&) = delete;
+    ExclusiveData(ExclusiveData&&) = delete;
+    ExclusiveData& operator=(ExclusiveData&&) = delete;
 
     void acquire() const { lock_.lock(); }
     void release() const { lock_.unlock(); }
@@ -117,19 +119,6 @@ class ExclusiveData
         release();
     }
 
-    ExclusiveData(ExclusiveData&& rhs) :
-      lock_(mozilla::Move(rhs.lock))
-    {
-        MOZ_ASSERT(&rhs != this, "self-move disallowed!");
-        new (value_.addr()) T(mozilla::Move(*rhs.value_.addr()));
-    }
-
-    ExclusiveData& operator=(ExclusiveData&& rhs) {
-        this->~ExclusiveData();
-        new (this) ExclusiveData(mozilla::Move(rhs));
-        return *this;
-    }
-
     /**
      * An RAII class that provides exclusive access to a `ExclusiveData<T>`'s
      * protected inner `T` value.
