From 3ba9de31440d65f7f5a84c859ec1496d6549939f Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Wed, 17 Dec 2025 10:37:03 +0800
Subject: [PATCH 2/2] SConstruct: add loongarch64 support

---
 SConstruct | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/SConstruct b/SConstruct
index f80044d..67eeffb 100644
--- a/SConstruct
+++ b/SConstruct
@@ -350,7 +350,8 @@ def AppendEndianCheck(conf):
   || defined(__alpha__) || defined(__ARMEL__) \
   || defined(_MIPSEL)   || (defined(__sh__) && defined(__LITTLE_ENDIAN__)) \
   || defined(__riscv) \
-  || defined(__AARCH64EL__)
+  || defined(__AARCH64EL__) \
+  || defined(__loongarch_lp64)
 # undef WORDS_BIGENDIAN
 
 #else
-- 
2.52.0

