Description: Fix build failures with latest autotools
Author: Balint Reczey <balint@balintreczey.hu>
--- a/libgksuui/Makefile.am
+++ b/libgksuui/Makefile.am
@@ -4,7 +4,7 @@
 
 noinst_LTLIBRARIES = libgksuui1.0.la
 libgksuui1_0_la_SOURCES = gksuui-dialog.c
-libgksuui1_0_la_LDFLAGS = -Wl,-O1 `pkg-config --libs gtk+-2.0 gconf-2.0`
+libgksuui1_0_la_LDFLAGS = -Wl,-O1 `pkg-config --libs gtk+-2.0 gconf-2.0 x11`
 
 noinst_HEADERS = defines.h gksuui.h gksuui-dialog.h
 includedir = ${prefix}/include/$(PACKAGE)
@@ -12,4 +12,4 @@
 noinst_PROGRAMS = test-gksuui
 test_gksuui_SOURCES = test-gksuui.c
 test_gksuui_LDADD = libgksuui1.0.la
-test_gksuui_LDFLAGS = `pkg-config --libs glib-2.0`
+test_gksuui_LDFLAGS = `pkg-config --libs glib-2.0 x11`
--- a/libgksu/Makefile.am
+++ b/libgksu/Makefile.am
@@ -27,13 +27,13 @@
 pkgconfig_DATA = libgksu2.pc
 
 pkglibdir = ${libdir}/${PACKAGE}
-pkglib_PROGRAMS = gksu-run-helper
+pkglibexec_PROGRAMS = gksu-run-helper
 gksu_run_helper_LDFLAGS = `pkg-config --libs glib-2.0`
 gksu_run_helper_SOURCES = gksu-run-helper.c
 
 noinst_PROGRAMS = test-gksu
 test_gksu_SOURCES = test-gksu.c
 test_gksu_LDADD = libgksu2.la
-test_gksu_LDFLAGS = `pkg-config --libs glib-2.0`
+test_gksu_LDFLAGS = `pkg-config --libs glib-2.0 gtk+-2.0`
 
 EXTRA_DIST = libgksu.ver
