=== modified file 'libgksu/libgksu.c'
--- a/libgksu/libgksu.c
+++ b/libgksu/libgksu.c
@@ -1385,6 +1385,13 @@
 			       NULL, NULL);
   context->sn_context = sn_launcher_context_new (sn_display, gdk_screen_get_number (gdk_display_get_default_screen (gdk_display_get_default ())));
   sn_launcher_context_set_description (context->sn_context, _("Granting Rights"));
+  if (context->sudo_mode)
+  {
+    sn_launcher_context_set_description (context->sn_context,
+		                      _("Starting Administrative Application"));
+    sn_launcher_context_set_icon_name (context->sn_context, 
+		                       "applications-system");
+  }
   sn_launcher_context_set_name (context->sn_context, g_get_prgname ());
 }
 
