diff -up VTK-9.1.0.rc1/Common/Core/vtkDataArray.cxx.AllValues VTK-9.1.0.rc1/Common/Core/vtkDataArray.cxx
--- VTK-9.1.0.rc1/Common/Core/vtkDataArray.cxx.AllValues	2021-10-01 12:23:49.000000000 -0600
+++ VTK-9.1.0.rc1/Common/Core/vtkDataArray.cxx	2021-10-04 19:27:42.557502417 -0600
@@ -1674,7 +1674,7 @@ struct ScalarRangeDispatchWrapper
   void operator()(ArrayT* array)
   {
     this->Success = vtkDataArrayPrivate::DoComputeScalarRange(
-      array, this->Range, vtkDataArrayPrivate::AllValues());
+      array, this->Range, vtkDataArrayPrivate::vtkAllValues());
   }
 };
 
@@ -1693,7 +1693,7 @@ struct VectorRangeDispatchWrapper
   void operator()(ArrayT* array)
   {
     this->Success = vtkDataArrayPrivate::DoComputeVectorRange(
-      array, this->Range, vtkDataArrayPrivate::AllValues());
+      array, this->Range, vtkDataArrayPrivate::vtkAllValues());
   }
 };
 
diff -up VTK-9.1.0.rc1/Common/Core/vtkDataArray.h.AllValues VTK-9.1.0.rc1/Common/Core/vtkDataArray.h
--- VTK-9.1.0.rc1/Common/Core/vtkDataArray.h.AllValues	2021-10-01 12:23:49.000000000 -0600
+++ VTK-9.1.0.rc1/Common/Core/vtkDataArray.h	2021-10-04 19:27:42.557502417 -0600
@@ -599,7 +599,7 @@ vtkArrayDownCast_FastCastMacro(vtkDataAr
 // vtkGenericDataArray.h as well.
 namespace vtkDataArrayPrivate
 {
-struct AllValues
+struct vtkAllValues
 {
 };
 struct FiniteValues
diff -up VTK-9.1.0.rc1/Common/Core/vtkDataArrayPrivate.txx.AllValues VTK-9.1.0.rc1/Common/Core/vtkDataArrayPrivate.txx
--- VTK-9.1.0.rc1/Common/Core/vtkDataArrayPrivate.txx.AllValues	2021-10-01 12:23:49.000000000 -0600
+++ VTK-9.1.0.rc1/Common/Core/vtkDataArrayPrivate.txx	2021-10-04 19:27:42.559502435 -0600
@@ -310,7 +310,7 @@ template <int NumComps>
 struct ComputeScalarRange
 {
   template <class ArrayT, typename RangeValueType>
-  bool operator()(ArrayT* array, RangeValueType* ranges, AllValues)
+  bool operator()(ArrayT* array, RangeValueType* ranges, vtkAllValues)
   {
     AllValuesMinAndMax<NumComps, ArrayT> minmax(array);
     vtkSMPTools::For(0, array->GetNumberOfTuples(), minmax);
@@ -447,7 +447,7 @@ public:
 };
 
 template <class ArrayT, typename RangeValueType>
-bool GenericComputeScalarRange(ArrayT* array, RangeValueType* ranges, AllValues)
+bool GenericComputeScalarRange(ArrayT* array, RangeValueType* ranges, vtkAllValues)
 {
   AllValuesGenericMinAndMax<ArrayT> minmax(array);
   vtkSMPTools::For(0, array->GetNumberOfTuples(), minmax);
@@ -530,7 +530,7 @@ bool DoComputeScalarRange(ArrayT* array,
 //----------------------------------------------------------------------------
 // generic implementation that operates on ValueType.
 template <typename ArrayT, typename RangeValueType>
-bool DoComputeVectorRange(ArrayT* array, RangeValueType range[2], AllValues)
+bool DoComputeVectorRange(ArrayT* array, RangeValueType range[2], vtkAllValues)
 {
   range[0] = vtkTypeTraits<RangeValueType>::Max();
   range[1] = vtkTypeTraits<RangeValueType>::Min();
diff -up VTK-9.1.0.rc1/Common/Core/vtkGenericDataArray.h.AllValues VTK-9.1.0.rc1/Common/Core/vtkGenericDataArray.h
--- VTK-9.1.0.rc1/Common/Core/vtkGenericDataArray.h.AllValues	2021-10-01 12:23:49.000000000 -0600
+++ VTK-9.1.0.rc1/Common/Core/vtkGenericDataArray.h	2021-10-04 19:27:42.559502435 -0600
@@ -414,7 +414,7 @@ namespace vtkDataArrayPrivate
 template <typename A, typename R, typename T>
 bool DoComputeScalarRange(A*, R*, T);
 template <typename A, typename R>
-bool DoComputeVectorRange(A*, R[2], AllValues);
+bool DoComputeVectorRange(A*, R[2], vtkAllValues);
 template <typename A, typename R>
 bool DoComputeVectorRange(A*, R[2], FiniteValues);
 } // namespace vtkDataArrayPrivate
@@ -466,11 +466,11 @@ class vtkScaledSOADataArrayTemplate;
 
 #define VTK_INSTANTIATE_VALUERANGE_ARRAYTYPE(ArrayType, ValueType)                                 \
   template VTKCOMMONCORE_EXPORT bool DoComputeScalarRange(                                         \
-    ArrayType*, ValueType*, vtkDataArrayPrivate::AllValues);                                       \
+    ArrayType*, ValueType*, vtkDataArrayPrivate::vtkAllValues);                                       \
   template VTKCOMMONCORE_EXPORT bool DoComputeScalarRange(                                         \
     ArrayType*, ValueType*, vtkDataArrayPrivate::FiniteValues);                                    \
   template VTKCOMMONCORE_EXPORT bool DoComputeVectorRange(                                         \
-    ArrayType*, ValueType[2], vtkDataArrayPrivate::AllValues);                                     \
+    ArrayType*, ValueType[2], vtkDataArrayPrivate::vtkAllValues);                                     \
   template VTKCOMMONCORE_EXPORT bool DoComputeVectorRange(                                         \
     ArrayType*, ValueType[2], vtkDataArrayPrivate::FiniteValues);
 
@@ -516,18 +516,18 @@ namespace vtkDataArrayPrivate
 template <typename A, typename R, typename T>
 bool DoComputeScalarRange(A*, R*, T);
 template <typename A, typename R>
-bool DoComputeVectorRange(A*, R[2], AllValues);
+bool DoComputeVectorRange(A*, R[2], vtkAllValues);
 template <typename A, typename R>
 bool DoComputeVectorRange(A*, R[2], FiniteValues);
 } // namespace vtkDataArrayPrivate
 
 #define VTK_DECLARE_VALUERANGE_ARRAYTYPE(ArrayType, ValueType)                                     \
   extern template VTKCOMMONCORE_EXPORT bool DoComputeScalarRange(                                  \
-    ArrayType*, ValueType*, vtkDataArrayPrivate::AllValues);                                       \
+    ArrayType*, ValueType*, vtkDataArrayPrivate::vtkAllValues);                                       \
   extern template VTKCOMMONCORE_EXPORT bool DoComputeScalarRange(                                  \
     ArrayType*, ValueType*, vtkDataArrayPrivate::FiniteValues);                                    \
   extern template VTKCOMMONCORE_EXPORT bool DoComputeVectorRange(                                  \
-    ArrayType*, ValueType[2], vtkDataArrayPrivate::AllValues);                                     \
+    ArrayType*, ValueType[2], vtkDataArrayPrivate::vtkAllValues);                                     \
   extern template VTKCOMMONCORE_EXPORT bool DoComputeVectorRange(                                  \
     ArrayType*, ValueType[2], vtkDataArrayPrivate::FiniteValues);
 
diff -up VTK-9.1.0.rc1/Common/Core/vtkGenericDataArray.txx.AllValues VTK-9.1.0.rc1/Common/Core/vtkGenericDataArray.txx
--- VTK-9.1.0.rc1/Common/Core/vtkGenericDataArray.txx.AllValues	2021-10-01 12:23:49.000000000 -0600
+++ VTK-9.1.0.rc1/Common/Core/vtkGenericDataArray.txx	2021-10-04 19:27:42.558502426 -0600
@@ -1190,7 +1190,7 @@ bool vtkGenericDataArray<DerivedT, Value
   using namespace vtk_GDA_detail;
   using Supported = IsSupported<DerivedT, ValueTypeT>;
   return ComputeScalarValueRangeImpl(
-    static_cast<DerivedT*>(this), ranges, vtkDataArrayPrivate::AllValues{}, Supported{});
+    static_cast<DerivedT*>(this), ranges, vtkDataArrayPrivate::vtkAllValues{}, Supported{});
 }
 
 //-----------------------------------------------------------------------------
@@ -1200,7 +1200,7 @@ bool vtkGenericDataArray<DerivedT, Value
   using namespace vtk_GDA_detail;
   using Supported = IsSupported<DerivedT, ValueTypeT>;
   return ComputeVectorValueRangeImpl(
-    static_cast<DerivedT*>(this), range, vtkDataArrayPrivate::AllValues{}, Supported{});
+    static_cast<DerivedT*>(this), range, vtkDataArrayPrivate::vtkAllValues{}, Supported{});
 }
 
 //-----------------------------------------------------------------------------
