diff --color -aur VTK-9.1.0/ThirdParty/fmt/CMakeLists.txt VTK-9.1.0.aosc/ThirdParty/fmt/CMakeLists.txt
--- VTK-9.1.0/ThirdParty/fmt/CMakeLists.txt	2021-11-05 03:48:20.000000000 +0800
+++ VTK-9.1.0.aosc/ThirdParty/fmt/CMakeLists.txt	2021-12-19 11:05:46.520613605 +0800
@@ -8,7 +8,7 @@
     TARGETS fmt::fmt
     # 8.1.0 is a version which is unreleased.
     # The required PR from 8.1.0 is https://github.com/fmtlib/fmt/pull/2432
-    VERSION       "8.1.0"
+    VERSION       "8.0.1"
     STANDARD_INCLUDE_DIRS)
 
 configure_file(
diff --color -aur VTK-9.1.0/ThirdParty/libharu/CMakeLists.txt VTK-9.1.0.aosc/ThirdParty/libharu/CMakeLists.txt
--- VTK-9.1.0/ThirdParty/libharu/CMakeLists.txt	2021-11-05 03:48:20.000000000 +0800
+++ VTK-9.1.0.aosc/ThirdParty/libharu/CMakeLists.txt	2021-12-19 11:05:22.955778387 +0800
@@ -8,7 +8,7 @@
     # Unreleased. Requires these PRs:
     # https://github.com/libharu/libharu/pull/157
     # https://github.com/libharu/libharu/pull/187
-    VERSION 2.4.0
+    VERSION 2.3.0
     TARGETS LibHaru::LibHaru
     STANDARD_INCLUDE_DIRS)
 
