From 9701c70e88f9d220a1b4dd42ae182e6da3e587ff Mon Sep 17 00:00:00 2001
From: Michael Webster <miketwebster@gmail.com>
Date: Mon, 9 Mar 2026 12:25:35 -0400
Subject: [PATCH 2/2] wallpaper.c: Fix wrong return code.

---
 src/wallpaper.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/wallpaper.c b/src/wallpaper.c
index 45e15c7..5a7c772 100644
--- a/src/wallpaper.c
+++ b/src/wallpaper.c
@@ -145,7 +145,7 @@ on_file_copy_cb (GObject *source_object,
     if (set_gsettings (dest_uri))
         handle->response = 0;
     else
-        handle->response = 1;
+        handle->response = 2;
 
 out:
     send_response (handle);
-- 
2.52.0

