diff -up hplip-3.9.12/prnt/backend/hp.c.retry-open hplip-3.9.12/prnt/backend/hp.c
--- hplip-3.9.12/prnt/backend/hp.c.retry-open	2010-01-05 15:14:30.000000000 +0000
+++ hplip-3.9.12/prnt/backend/hp.c	2010-01-05 15:16:03.521911780 +0000
@@ -741,12 +741,6 @@ int main(int argc, char *argv[])
                      goto bugout;
                   }
 
-                  if (stat != HPMUD_R_DEVICE_BUSY)
-                  {
-                     BUG("ERROR: open device failed stat=%d: %s\n", stat, argv[0]);
-                     goto bugout;
-                  }
-
                   /* Display user error. */
                   device_event(argv[0], printer, 5000+stat, argv[2], argv[1], argv[3]);
 
