diff -up hplip-3.10.9/prnt/hpijs/hpijs.cpp.hpijs-marker-supply hplip-3.10.9/prnt/hpijs/hpijs.cpp
--- hplip-3.10.9/prnt/hpijs/hpijs.cpp.hpijs-marker-supply	2010-10-18 16:58:23.882993673 +0100
+++ hplip-3.10.9/prnt/hpijs/hpijs.cpp	2010-10-18 17:00:00.510869032 +0100
@@ -585,6 +585,7 @@ int main (int argc, char *argv[], char *
    char *raster = NULL, *k_raster = NULL;
    int status = EXIT_FAILURE;
    int ret, n, i, kn=0, width, k_width;
+   int low_marker = 0;
    char user_name[32]={0,};
         
    openlog("hpijs", LOG_PID,  LOG_DAEMON);
@@ -666,14 +667,17 @@ int main (int argc, char *argv[], char *
 			case WARN_LOW_INK_YELLOW:
 			case WARN_LOW_INK_MULTIPLE_PENS:
                         {
-                           fputs("STATE: +marker-supply-low-warning\n", stderr);
+			   low_marker = 1;
                            break;
                         }
 			default:
-                           fputs("STATE: +marker-supply-low-warning\n", stderr);
+			   low_marker = 1;
 		}
     }
 
+    fprintf(stderr, "STATE: %cmarker-supply-low-warning\n",
+	    low_marker ? '+' : '-');
+
 #if 0
    BUG("device model=%s\n", pSS->pPC->PrinterModel());
    BUG("device class=%s\n",  pSS->pPC->PrintertypeToString(pSS->pPC->SelectedDevice()));
