diff -up hplip-3.14.10/prnt/ps/hp-lj_300_400_color_m351_m451-ps.ppd.gz-deviceid hplip-3.14.10/prnt/ps/hp-lj_300_400_color_m351_m451-ps.ppd
--- hplip-3.14.10/prnt/ps/hp-lj_300_400_color_m351_m451-ps.ppd.gz-deviceid	2014-11-04 09:41:13.200005393 +0000
+++ hplip-3.14.10/prnt/ps/hp-lj_300_400_color_m351_m451-ps.ppd	2014-11-04 09:42:13.168312042 +0000
@@ -82,6 +82,7 @@
 *cupsManualCopies: False
 *cupsLanguages: "en da de es fi fr it ja ko nb nl pt ru sv zh_CN zh_TW"
 *cupsFilter: "application/vnd.cups-postscript 0 hpps"
+*1284DeviceID: "MFG:Hewlett-Packard;MDL:HP LaserJet 400 color M451dn;DES:HP LaserJet 400 color M451dn;"
 *UIConstraints: *HPOption_Duplexer False *Duplex
 *UIConstraints: *Duplex *HPOption_Duplexer False
 *UIConstraints: *HPOption_Tray3 False *InputSlot Tray3
