diff -up hplip-3.15.2/scan.py.scan-tmp hplip-3.15.2/scan.py
--- hplip-3.15.2/scan.py.scan-tmp	2015-02-04 15:27:20.423562999 +0100
+++ hplip-3.15.2/scan.py	2015-02-04 15:34:59.170978476 +0100
@@ -1080,7 +1080,7 @@ try:
         if ('editor' in dest or 'viewer' in dest or 'email' in dest or 'print' in dest) \
             and not file_saved:
 
-            output_fd, output = utils.make_temp_file(suffix='.png')
+            output_fd, output = utils.make_temp_file(dir='/var/tmp', suffix='.png')
             try:
                 im.save(output)
             except IOError as e:
diff -up hplip-3.15.2/scan/sane.py.scan-tmp hplip-3.15.2/scan/sane.py
--- hplip-3.15.2/scan/sane.py.scan-tmp	2015-01-29 13:20:21.000000000 +0100
+++ hplip-3.15.2/scan/sane.py	2015-02-04 15:27:20.424562984 +0100
@@ -425,7 +425,8 @@ class ScanThread(threading.Thread):
         self.dev = device
         self.update_queue = update_queue
         self.event_queue = event_queue
-        self.buffer_fd, self.buffer_path = utils.make_temp_file(prefix='hpscan')
+        self.buffer_fd, self.buffer_path = utils.make_temp_file(dir='/var/tmp',
+                                                                prefix='hpscan')
         self.buffer = os.fdopen(self.buffer_fd, "w+b")
         self.format = -1
         self.format_name = ''
