diff -up hplip-3.14.10/prnt/drv/hpcups.drv.in.dj990c-margin hplip-3.14.10/prnt/drv/hpcups.drv.in
--- hplip-3.14.10/prnt/drv/hpcups.drv.in.dj990c-margin	2014-12-23 15:55:09.779368939 +0000
+++ hplip-3.14.10/prnt/drv/hpcups.drv.in	2014-12-23 15:55:24.488454126 +0000
@@ -1531,9 +1531,9 @@ Manufacturer "HP"
          "<</cupsInteger0 2/PageSize[612 792]/ImagingBBox null>>setpagedevice"
   CustomMedia "Letter.Duplex/Letter AutoDuplex 8.5x11in" 612 783 18 27 18 36 "<</cupsInteger0 2/PageSize[612 783]/ImagingBBox null>>setpagedevice"
          "<</cupsInteger0 2/PageSize[612 783]/ImagingBBox null>>setpagedevice"
-  CustomMedia "A4/A4 210x297mm" 595.44 841.68 18 36.00 18 9.00 "<</cupsInteger0 26/PageSize[595.44 841.68]/ImagingBBox null>>setpagedevice"
+  CustomMedia "A4/A4 210x297mm" 595.44 841.68 9 36.00 9 9.00 "<</cupsInteger0 26/PageSize[595.44 841.68]/ImagingBBox null>>setpagedevice"
          "<</cupsInteger0 26/PageSize[595.44 841.68]/ImagingBBox null>>setpagedevice"
-  CustomMedia "A4.Duplex/A4 AutoDuplex 210x297mm" 595 833 18 27 18 36 "<</cupsInteger0 26/PageSize[595 833]/ImagingBBox null>>setpagedevice"
+  CustomMedia "A4.Duplex/A4 AutoDuplex 210x297mm" 595 833 9 27 9 36 "<</cupsInteger0 26/PageSize[595 833]/ImagingBBox null>>setpagedevice"
          "<</cupsInteger0 26/PageSize[595 833]/ImagingBBox null>>setpagedevice"
   CustomMedia "ExecutiveJIS/Executive (JIS) 8.5x12.986in" 612 936 18 36 18 9 "<</cupsInteger0 10/PageSize[612 936]/ImagingBBox null>>setpagedevice"
          "<</cupsInteger0 10/PageSize[612 936]/ImagingBBox null>>setpagedevice"
