diff -up hplip-3.15.11/config_usb_printer.py.no-write-bytecode hplip-3.15.11/config_usb_printer.py
--- hplip-3.15.11/config_usb_printer.py.no-write-bytecode	2015-11-16 10:50:03.127459222 +0100
+++ hplip-3.15.11/config_usb_printer.py	2015-11-16 10:53:52.266931490 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python -B
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2011-2015 HP Development Company, L.P.
diff -up hplip-3.15.11/fax/backend/hpfax.py.no-write-bytecode hplip-3.15.11/fax/backend/hpfax.py
--- hplip-3.15.11/fax/backend/hpfax.py.no-write-bytecode	2015-11-16 10:50:03.128459220 +0100
+++ hplip-3.15.11/fax/backend/hpfax.py	2015-11-16 10:53:49.587937659 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python -B
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -up hplip-3.15.11/prnt/filters/hpps.no-write-bytecode hplip-3.15.9/prnt/filters/hpps
--- hplip-3.15.11/prnt/filters/hpps.no-write-bytecode	2016-01-04 16:44:08.000000000 +0100
+++ hplip-3.15.11/prnt/filters/hpps	2016-01-04 16:50:11.272921194 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python -B
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
