diff -up hplip-3.17.4/Makefile.am.no-ernie hplip-3.17.4/Makefile.am
--- hplip-3.17.4/Makefile.am.no-ernie	2017-04-26 16:28:35.398099978 +0200
+++ hplip-3.17.4/Makefile.am	2017-04-26 16:29:02.235879035 +0200
@@ -519,7 +519,7 @@ hpcups_SOURCES = prnt/hpcups/HPCupsFilte
 	prnt/hpcups/Mode2.cpp prnt/hpcups/Mode2.h prnt/hpcups/Mode3.cpp prnt/hpcups/Mode3.h \
 	prnt/hpcups/Mode9.cpp prnt/hpcups/Mode9.h prnt/hpcups/Mode10.cpp prnt/hpcups/Mode10.h \
 	prnt/hpcups/ModeDeltaPlus.cpp prnt/hpcups/ModeDeltaPlus.h prnt/hpcups/ModeJbig.cpp prnt/hpcups/ModeJbig.h \
-	prnt/hpcups/ErnieFilter.cpp prnt/hpcups/ErnieFilter.h prnt/hpcups/EncapsulatorFactory.cpp prnt/hpcups/EncapsulatorFactory.h \
+	prnt/hpcups/EncapsulatorFactory.cpp prnt/hpcups/EncapsulatorFactory.h \
 	prnt/hpcups/Encapsulator.cpp prnt/hpcups/Encapsulator.h prnt/hpcups/Pcl3.cpp prnt/hpcups/Pcl3.h \
 	prnt/hpcups/Pcl3Gui.cpp prnt/hpcups/Pcl3Gui.h prnt/hpcups/Pcl3Gui2.cpp prnt/hpcups/Pcl3Gui2.h \
 	prnt/hpcups/LJMono.cpp prnt/hpcups/LJMono.h prnt/hpcups/LJColor.cpp prnt/hpcups/LJColor.h \
diff -up hplip-3.17.4/prnt/hpcups/Pcl3Gui2.cpp.no-ernie hplip-3.17.4/prnt/hpcups/Pcl3Gui2.cpp
--- hplip-3.17.4/prnt/hpcups/Pcl3Gui2.cpp.no-ernie	2017-04-26 16:28:39.525066003 +0200
+++ hplip-3.17.4/prnt/hpcups/Pcl3Gui2.cpp	2017-04-26 16:29:46.987510614 +0200
@@ -30,7 +30,6 @@
 
 #include "CommonDefinitions.h"
 #include "Pcl3Gui2.h"
-#include "ErnieFilter.h"
 #include "Mode10.h"
 #include "Mode9.h"
 #include "PrinterCommands.h"
@@ -38,7 +37,6 @@
 Pcl3Gui2::Pcl3Gui2() : Encapsulator()
 {
     speed_mech_enabled = true;
-    m_run_ernie_filter = true;
     crd_type = eCrd_both;
     strcpy(m_szLanguage, "PCL3GUI");
 }
@@ -59,21 +57,6 @@ DRIVER_ERROR Pcl3Gui2::Configure(Pipelin
     }
 
     width = m_pMA->printable_width;;
-    if (m_run_ernie_filter) {
-	    ErnieFilter    *pErnie;
-
-       // Normal: threshold = (resolution) * (0.0876) - 2
-       int threshold = ((m_pQA->horizontal_resolution * 876) / 10000) - 2;
-
-       pErnie = new ErnieFilter (width, eBGRPixelData, threshold);
-       p = new Pipeline (pErnie);
-       if (head) {
-          head->AddPhase (p);
-       }
-       else {
-           head = p;
-       }
-    }
 
     if (crd_type != eCrd_black_only) {
         Mode10    *pMode10;
diff -up hplip-3.17.4/prnt/hpcups/Pcl3Gui2.h.no-ernie hplip-3.17.4/prnt/hpcups/Pcl3Gui2.h
--- hplip-3.17.4/prnt/hpcups/Pcl3Gui2.h.no-ernie	2017-04-26 16:28:43.277035115 +0200
+++ hplip-3.17.4/prnt/hpcups/Pcl3Gui2.h	2017-04-26 16:30:05.826355522 +0200
@@ -60,7 +60,6 @@ private:
     DRIVER_ERROR encapsulateRaster(BYTE *input_raster, unsigned int num_bytes, COLORTYPE c_type);
     bool    speed_mech_enabled;
     int     page_number;
-    bool    m_run_ernie_filter;
     eCrdType    crd_type;
 };
 
