diff -up hplip-3.21.12/installer/pluginhandler.py.missing-links hplip-3.21.12/installer/pluginhandler.py
--- hplip-3.21.12/installer/pluginhandler.py.missing-links	2022-01-12 16:55:39.761934879 +0100
+++ hplip-3.21.12/installer/pluginhandler.py	2022-01-12 17:22:33.367435334 +0100
@@ -167,7 +167,7 @@ class PluginHandle(object):
                         continue
                     if link != "":
                         if not utils.check_library(link):
-                            self.__plugin_state = PLUGIN_FILES_CORRUPTED
+                            pass
 
 
     def __getPluginInformation(self, callback=None):
