diff --git a/ui5/fabwindow.py b/ui5/fabwindow.py
index 0b95c94..631727c 100644
--- a/ui5/fabwindow.py
+++ b/ui5/fabwindow.py
@@ -518,7 +518,7 @@ class FABWindow(QMainWindow,  Ui_MainWindow):
                          self.__tr("Import fax addresses from LDIF or vCard"),
                          #user_conf.workingDirectory(),
                          self.user_settings.working_dir,
-                         "vCard (*.vcf);;LDIF (*.ldif *.ldi)"))
+                         "vCard (*.vcf);;LDIF (*.ldif *.ldi)")[0])
 
         if result:
             working_directory = to_unicode(os.path.dirname(result))
