diff -up hplip-3.21.12/base/password.py.osname-mismatch hplip-3.21.12/base/password.py
--- hplip-3.21.12/base/password.py.osname-mismatch	2022-01-14 08:36:11.413431048 +0100
+++ hplip-3.21.12/base/password.py	2022-01-14 09:14:57.315230889 +0100
@@ -105,7 +105,7 @@ def get_distro_name():
         os_name = name.read().strip()
         name.close()
 
-    os_name = os_name.lower()
+    os_name = os_name.lower().replace(' ','')
     if 'MX' in distro_release_name:
         os_name = "mxlinux"
     if "redhatenterprise" in os_name:
