diff -Naur hplip-3.22.4.old/base/password.py hplip-3.22.4/base/password.py
--- hplip-3.22.4.old/base/password.py	2022-04-26 12:16:59.000000000 +0800
+++ hplip-3.22.4/base/password.py	2022-05-12 16:04:41.418980914 +0800
@@ -63,6 +63,7 @@
               'debiangnu/linux' : 'su',
               'mxlinux' : 'su',
               'elementaryos' : 'sudo',
+              'aoscos' : 'sudo',
               }
 
 
