diff -up hplip-3.16.5/prnt/hpcups/HPCupsFilter.h.ppd hplip-3.16.5/prnt/hpcups/HPCupsFilter.h
--- hplip-3.16.5/prnt/hpcups/HPCupsFilter.h.ppd	2016-07-11 13:37:19.224836291 +0200
+++ hplip-3.16.5/prnt/hpcups/HPCupsFilter.h	2016-07-11 13:38:05.290523517 +0200
@@ -37,6 +37,7 @@
 #include "Job.h"
 
 #include "dbuscomm.h"
+#include <cups/ppd.h>
 
 #define		DBITMAPFILEHEADER		14
 #define		DBITMAPINFOHEADER		40
diff -up hplip-3.16.5/prnt/hpijs/hpcupsfax.h.ppd hplip-3.16.5/prnt/hpijs/hpcupsfax.h
--- hplip-3.16.5/prnt/hpijs/hpcupsfax.h.ppd	2016-07-11 13:43:25.837347080 +0200
+++ hplip-3.16.5/prnt/hpijs/hpcupsfax.h	2016-07-11 13:45:14.590608672 +0200
@@ -32,6 +32,8 @@
 #ifndef HPCUPSFAX_H
 #define HPCUPSFAX_H
 
+#include <cups/ppd.h>
+
 /*
  * Raster data encoding methods
  */
