commit cdf5cb36e5d5a86f6a690c5861666cdc2da70593
Author: Tomas Korbar <tkorbar@redhat.com>
Date:   Wed Sep 19 15:33:03 2018 +0200

    Fix TypeError in hpfax

diff --git a/fax/backend/hpfax.py b/fax/backend/hpfax.py
index 9560e84..bc0912a 100755
--- a/fax/backend/hpfax.py
+++ b/fax/backend/hpfax.py
@@ -89,7 +89,7 @@ try:
     from base import utils
     from prnt import cups
 except ImportError as e:
-    bug("Error importing HPLIP modules: %s\n" % (pid, e))
+    bug("Error importing HPLIP modules: %s\n" % (e))
     sys.exit(1)
 
 def handle_sigpipe():
