diff --git a/uiscan.py b/uiscan.py
index f46ec5b..0ac993a 100755
--- a/uiscan.py
+++ b/uiscan.py
@@ -22,8 +22,8 @@ except ImportError as e:
 
 
 
-mod = module.Module(__mod__, __title__, __version__, __doc__, None, (INTERACTIVE_MODE,))
-
+mod = module.Module(__mod__, __title__, __version__, __doc__, None, (GUI_MODE,), (UI_TOOLKIT_QT4, UI_TOOLKIT_QT5))
+mod.setUsage(module.USAGE_FLAG_NONE, extra_options=None, see_also_list = ['hp-scan'])
 
 
 if __name__ == "__main__":
