diff --git a/installer/pluginhandler.py b/installer/pluginhandler.py
index 4d7c1b8..1af8d85 100755
--- a/installer/pluginhandler.py
+++ b/installer/pluginhandler.py
@@ -164,7 +164,7 @@ class PluginHandle(object):
                 for src, trg, link in copies:
                     if link != "":
                         if not utils.check_library(link):
-                            self.__plugin_state = PLUGIN_FILES_CORRUPTED
+                            pass
 
 
     def __getPluginInformation(self, callback=None):
