diff -up hplip-3.13.4/prnt/drv/hpcups.drv.in.snmp-quirks hplip-3.13.4/prnt/drv/hpcups.drv.in
--- hplip-3.13.4/prnt/drv/hpcups.drv.in.snmp-quirks	2013-04-09 12:11:58.447587815 +0200
+++ hplip-3.13.4/prnt/drv/hpcups.drv.in	2013-04-09 12:11:58.594585761 +0200
@@ -6757,6 +6757,8 @@ Manufacturer "HP"
       Attribute "NickName" "" "HP Officejet Pro 8500 a909a, hpcups $Version"
       Attribute "ShortNickName" "" "HP Officejet Pro 8500 a909a"
       Attribute "1284DeviceID" "" "MFG:HP;MDL:officejet pro 8500 a909a;DES:officejet pro 8500 a909a;"
+      // This device lies about its supplies capacity.
+      Attribute "cupsSNMPQuirks" "" "capacity"
       PCFileName "hp-officejet_pro_8500_a909a.ppd"
       Attribute "Product" "" "(HP Officejet Pro 8500 All-in-one Printer - a909a)"
     }
@@ -6773,6 +6775,8 @@ Manufacturer "HP"
       Attribute "NickName" "" "HP Officejet Pro 8500 a909g, hpcups $Version"
       Attribute "ShortNickName" "" "HP Officejet Pro 8500 a909g"
       Attribute "1284DeviceID" "" "MFG:HP;MDL:officejet pro 8500 a909g;DES:officejet pro 8500 a909g;"
+      // This device lies about its supplies capacity.
+      Attribute "cupsSNMPQuirks" "" "capacity"
       PCFileName "hp-officejet_pro_8500_a909g.ppd"
       Attribute "Product" "" "(HP Officejet Pro 8500 Wireless All-in-one Printer - a909g)"
     }
