diff -up hplip-3.16.8/prnt/filters/hpps.badwhitespace hplip-3.16.8/prnt/filters/hpps
--- hplip-3.16.8/prnt/filters/hpps.badwhitespace	2016-09-01 19:15:43.509068772 +0200
+++ hplip-3.16.8/prnt/filters/hpps	2016-09-01 19:32:47.861548645 +0200
@@ -197,11 +197,11 @@ if 'HPPinPrnt' in options and 'noHPPinPr
             msg("Setting Pin to default")
     else:
         szKeyInitials = ['HPFIDigit', 'HPSEDigit', 'HPTHDigit', 'HPFTDigit']                                 
-	for x in szKeyInitials:
+        for x in szKeyInitials:
            try:
-	        secpin += opts[x]
+                secpin += opts[x]
            except KeyError:
-	        secpin += '0'
+                secpin += '0'
 
     os.write(output_fd, to_bytes_utf8('@PJL SET HOLDKEY="%s"\x0a' % secpin))
 
