diff -up hplip-3.16.11/Makefile.am.no-ernie hplip-3.16.11/Makefile.am
--- hplip-3.16.11/Makefile.am.no-ernie	2016-11-25 18:10:12.191061577 +0100
+++ hplip-3.16.11/Makefile.am	2016-11-25 18:10:12.273060841 +0100
@@ -472,7 +472,7 @@ libapdk_la_SOURCES = prnt/hpijs/models.c
 	prnt/hpijs/compression.h prnt/hpijs/config.h prnt/hpijs/context.h prnt/hpijs/dj350.h prnt/hpijs/dj400.h prnt/hpijs/dj540.h \
 	prnt/hpijs/dj600.h prnt/hpijs/dj630.h prnt/hpijs/dj660.h prnt/hpijs/dj690.h prnt/hpijs/dj6xx.h prnt/hpijs/dj8x5.h prnt/hpijs/dj8xx.h \
 	prnt/hpijs/dj850.h prnt/hpijs/dj890.h prnt/hpijs/dj9xx.h prnt/hpijs/dj9xxvip.h \
-	prnt/hpijs/ernieplatform.h prnt/hpijs/filterhpa.h prnt/hpijs/global_types.h prnt/hpijs/halftoner.h prnt/hpijs/font.h prnt/hpijs/debug.h \
+	prnt/hpijs/filterhpa.h prnt/hpijs/global_types.h prnt/hpijs/halftoner.h prnt/hpijs/font.h prnt/hpijs/debug.h \
 	prnt/hpijs/halftoner_open.h prnt/hpijs/header.h prnt/hpijs/hpprint_c_api.h \
 	prnt/hpijs/hpprintapi.h prnt/hpijs/hptypes.h prnt/hpijs/htfed.h prnt/hpijs/internal.h \
 	prnt/hpijs/interp.h prnt/hpijs/interp_data_50.h prnt/hpijs/io_defs.h prnt/hpijs/job.h prnt/hpijs/models.h prnt/hpijs/modes.h prnt/hpijs/pmselect.h \
@@ -519,9 +519,9 @@ hpcups_SOURCES = prnt/hpcups/HPCupsFilte
 	prnt/hpcups/Mode2.cpp prnt/hpcups/Mode2.h prnt/hpcups/Mode3.cpp prnt/hpcups/Mode3.h \
 	prnt/hpcups/Mode9.cpp prnt/hpcups/Mode9.h prnt/hpcups/Mode10.cpp prnt/hpcups/Mode10.h \
 	prnt/hpcups/ModeDeltaPlus.cpp prnt/hpcups/ModeDeltaPlus.h prnt/hpcups/ModeJbig.cpp prnt/hpcups/ModeJbig.h \
-	prnt/hpcups/ErnieFilter.cpp prnt/hpcups/ErnieFilter.h prnt/hpcups/EncapsulatorFactory.cpp prnt/hpcups/EncapsulatorFactory.h \
+	prnt/hpcups/EncapsulatorFactory.cpp prnt/hpcups/EncapsulatorFactory.h \
 	prnt/hpcups/Encapsulator.cpp prnt/hpcups/Encapsulator.h prnt/hpcups/Pcl3.cpp prnt/hpcups/Pcl3.h \
-	prnt/hpcups/Pcl3Gui.cpp prnt/hpcups/Pcl3Gui.h prnt/hpcups/Pcl3Gui2.cpp prnt/hpcups/Pcl3Gui2.h \
+	prnt/hpcups/Pcl3Gui.cpp prnt/hpcups/Pcl3Gui.h \
 	prnt/hpcups/LJMono.cpp prnt/hpcups/LJMono.h prnt/hpcups/LJColor.cpp prnt/hpcups/LJColor.h \
 	prnt/hpcups/LJFastRaster.cpp prnt/hpcups/LJFastRaster.h prnt/hpcups/LJJetReady.cpp prnt/hpcups/LJJetReady.h \
 	prnt/hpcups/LJZjStream.cpp prnt/hpcups/LJZjStream.h \
diff -up hplip-3.16.11/Makefile.in.no-ernie hplip-3.16.11/Makefile.in
--- hplip-3.16.11/Makefile.in.no-ernie	2016-11-23 07:03:46.000000000 +0100
+++ hplip-3.16.11/Makefile.in	2016-11-25 18:17:06.466341822 +0100
@@ -436,13 +436,12 @@ am__hpcups_SOURCES_DIST = prnt/hpcups/HP
 	prnt/hpcups/Mode10.cpp prnt/hpcups/Mode10.h \
 	prnt/hpcups/ModeDeltaPlus.cpp prnt/hpcups/ModeDeltaPlus.h \
 	prnt/hpcups/ModeJbig.cpp prnt/hpcups/ModeJbig.h \
-	prnt/hpcups/ErnieFilter.cpp prnt/hpcups/ErnieFilter.h \
 	prnt/hpcups/EncapsulatorFactory.cpp \
 	prnt/hpcups/EncapsulatorFactory.h prnt/hpcups/Encapsulator.cpp \
 	prnt/hpcups/Encapsulator.h prnt/hpcups/Pcl3.cpp \
 	prnt/hpcups/Pcl3.h prnt/hpcups/Pcl3Gui.cpp \
-	prnt/hpcups/Pcl3Gui.h prnt/hpcups/Pcl3Gui2.cpp \
-	prnt/hpcups/Pcl3Gui2.h prnt/hpcups/LJMono.cpp \
+	prnt/hpcups/Pcl3Gui.h \
+	prnt/hpcups/LJMono.cpp \
 	prnt/hpcups/LJMono.h prnt/hpcups/LJColor.cpp \
 	prnt/hpcups/LJColor.h prnt/hpcups/LJFastRaster.cpp \
 	prnt/hpcups/LJFastRaster.h prnt/hpcups/LJJetReady.cpp \
@@ -490,12 +489,10 @@ am__hpcups_SOURCES_DIST = prnt/hpcups/HP
 @HPCUPS_INSTALL_TRUE@	hpcups-Mode10.$(OBJEXT) \
 @HPCUPS_INSTALL_TRUE@	hpcups-ModeDeltaPlus.$(OBJEXT) \
 @HPCUPS_INSTALL_TRUE@	hpcups-ModeJbig.$(OBJEXT) \
-@HPCUPS_INSTALL_TRUE@	hpcups-ErnieFilter.$(OBJEXT) \
 @HPCUPS_INSTALL_TRUE@	hpcups-EncapsulatorFactory.$(OBJEXT) \
 @HPCUPS_INSTALL_TRUE@	hpcups-Encapsulator.$(OBJEXT) \
 @HPCUPS_INSTALL_TRUE@	hpcups-Pcl3.$(OBJEXT) \
 @HPCUPS_INSTALL_TRUE@	hpcups-Pcl3Gui.$(OBJEXT) \
-@HPCUPS_INSTALL_TRUE@	hpcups-Pcl3Gui2.$(OBJEXT) \
 @HPCUPS_INSTALL_TRUE@	hpcups-LJMono.$(OBJEXT) \
 @HPCUPS_INSTALL_TRUE@	hpcups-LJColor.$(OBJEXT) \
 @HPCUPS_INSTALL_TRUE@	hpcups-LJFastRaster.$(OBJEXT) \
@@ -5076,7 +5073,7 @@ libapdk_la_SOURCES = prnt/hpijs/models.c
 	prnt/hpijs/compression.h prnt/hpijs/config.h prnt/hpijs/context.h prnt/hpijs/dj350.h prnt/hpijs/dj400.h prnt/hpijs/dj540.h \
 	prnt/hpijs/dj600.h prnt/hpijs/dj630.h prnt/hpijs/dj660.h prnt/hpijs/dj690.h prnt/hpijs/dj6xx.h prnt/hpijs/dj8x5.h prnt/hpijs/dj8xx.h \
 	prnt/hpijs/dj850.h prnt/hpijs/dj890.h prnt/hpijs/dj9xx.h prnt/hpijs/dj9xxvip.h \
-	prnt/hpijs/ernieplatform.h prnt/hpijs/filterhpa.h prnt/hpijs/global_types.h prnt/hpijs/halftoner.h prnt/hpijs/font.h prnt/hpijs/debug.h \
+	prnt/hpijs/filterhpa.h prnt/hpijs/global_types.h prnt/hpijs/halftoner.h prnt/hpijs/font.h prnt/hpijs/debug.h \
 	prnt/hpijs/halftoner_open.h prnt/hpijs/header.h prnt/hpijs/hpprint_c_api.h \
 	prnt/hpijs/hpprintapi.h prnt/hpijs/hptypes.h prnt/hpijs/htfed.h prnt/hpijs/internal.h \
 	prnt/hpijs/interp.h prnt/hpijs/interp_data_50.h prnt/hpijs/io_defs.h prnt/hpijs/job.h prnt/hpijs/models.h prnt/hpijs/modes.h prnt/hpijs/pmselect.h \
@@ -5115,9 +5112,8 @@ libapdk_la_CFLAGS = $(libapdk_la_CXXFLAG
 @HPCUPS_INSTALL_TRUE@	prnt/hpcups/Mode2.cpp prnt/hpcups/Mode2.h prnt/hpcups/Mode3.cpp prnt/hpcups/Mode3.h \
 @HPCUPS_INSTALL_TRUE@	prnt/hpcups/Mode9.cpp prnt/hpcups/Mode9.h prnt/hpcups/Mode10.cpp prnt/hpcups/Mode10.h \
 @HPCUPS_INSTALL_TRUE@	prnt/hpcups/ModeDeltaPlus.cpp prnt/hpcups/ModeDeltaPlus.h prnt/hpcups/ModeJbig.cpp prnt/hpcups/ModeJbig.h \
-@HPCUPS_INSTALL_TRUE@	prnt/hpcups/ErnieFilter.cpp prnt/hpcups/ErnieFilter.h prnt/hpcups/EncapsulatorFactory.cpp prnt/hpcups/EncapsulatorFactory.h \
 @HPCUPS_INSTALL_TRUE@	prnt/hpcups/Encapsulator.cpp prnt/hpcups/Encapsulator.h prnt/hpcups/Pcl3.cpp prnt/hpcups/Pcl3.h \
-@HPCUPS_INSTALL_TRUE@	prnt/hpcups/Pcl3Gui.cpp prnt/hpcups/Pcl3Gui.h prnt/hpcups/Pcl3Gui2.cpp prnt/hpcups/Pcl3Gui2.h \
+@HPCUPS_INSTALL_TRUE@	prnt/hpcups/Pcl3Gui.cpp prnt/hpcups/Pcl3Gui.h \
 @HPCUPS_INSTALL_TRUE@	prnt/hpcups/LJMono.cpp prnt/hpcups/LJMono.h prnt/hpcups/LJColor.cpp prnt/hpcups/LJColor.h \
 @HPCUPS_INSTALL_TRUE@	prnt/hpcups/LJFastRaster.cpp prnt/hpcups/LJFastRaster.h prnt/hpcups/LJJetReady.cpp prnt/hpcups/LJJetReady.h \
 @HPCUPS_INSTALL_TRUE@	prnt/hpcups/LJZjStream.cpp prnt/hpcups/LJZjStream.h \
@@ -5926,7 +5922,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-Compressor.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-Encapsulator.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-EncapsulatorFactory.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-ErnieFilter.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-HPCupsFilter.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-Halftoner.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-Hbpl1.Po@am__quote@
@@ -5948,7 +5943,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-ModeJpeg.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-Pcl3.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-Pcl3Gui.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-Pcl3Gui2.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-Pipeline.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-Processor.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hpcups-QuickConnect.Po@am__quote@
@@ -7381,20 +7375,6 @@ hpcups-ModeJbig.obj: prnt/hpcups/ModeJbi
 @AMDEP_TRUE@@am__fastdepCXX_FALSE@	DEPDIR=$(DEPDIR) $(CXXDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCXX_FALSE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -c -o hpcups-ModeJbig.obj `if test -f 'prnt/hpcups/ModeJbig.cpp'; then $(CYGPATH_W) 'prnt/hpcups/ModeJbig.cpp'; else $(CYGPATH_W) '$(srcdir)/prnt/hpcups/ModeJbig.cpp'; fi`
 
-hpcups-ErnieFilter.o: prnt/hpcups/ErnieFilter.cpp
-@am__fastdepCXX_TRUE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -MT hpcups-ErnieFilter.o -MD -MP -MF $(DEPDIR)/hpcups-ErnieFilter.Tpo -c -o hpcups-ErnieFilter.o `test -f 'prnt/hpcups/ErnieFilter.cpp' || echo '$(srcdir)/'`prnt/hpcups/ErnieFilter.cpp
-@am__fastdepCXX_TRUE@	$(am__mv) $(DEPDIR)/hpcups-ErnieFilter.Tpo $(DEPDIR)/hpcups-ErnieFilter.Po
-@AMDEP_TRUE@@am__fastdepCXX_FALSE@	source='prnt/hpcups/ErnieFilter.cpp' object='hpcups-ErnieFilter.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCXX_FALSE@	DEPDIR=$(DEPDIR) $(CXXDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCXX_FALSE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -c -o hpcups-ErnieFilter.o `test -f 'prnt/hpcups/ErnieFilter.cpp' || echo '$(srcdir)/'`prnt/hpcups/ErnieFilter.cpp
-
-hpcups-ErnieFilter.obj: prnt/hpcups/ErnieFilter.cpp
-@am__fastdepCXX_TRUE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -MT hpcups-ErnieFilter.obj -MD -MP -MF $(DEPDIR)/hpcups-ErnieFilter.Tpo -c -o hpcups-ErnieFilter.obj `if test -f 'prnt/hpcups/ErnieFilter.cpp'; then $(CYGPATH_W) 'prnt/hpcups/ErnieFilter.cpp'; else $(CYGPATH_W) '$(srcdir)/prnt/hpcups/ErnieFilter.cpp'; fi`
-@am__fastdepCXX_TRUE@	$(am__mv) $(DEPDIR)/hpcups-ErnieFilter.Tpo $(DEPDIR)/hpcups-ErnieFilter.Po
-@AMDEP_TRUE@@am__fastdepCXX_FALSE@	source='prnt/hpcups/ErnieFilter.cpp' object='hpcups-ErnieFilter.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCXX_FALSE@	DEPDIR=$(DEPDIR) $(CXXDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCXX_FALSE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -c -o hpcups-ErnieFilter.obj `if test -f 'prnt/hpcups/ErnieFilter.cpp'; then $(CYGPATH_W) 'prnt/hpcups/ErnieFilter.cpp'; else $(CYGPATH_W) '$(srcdir)/prnt/hpcups/ErnieFilter.cpp'; fi`
-
 hpcups-EncapsulatorFactory.o: prnt/hpcups/EncapsulatorFactory.cpp
 @am__fastdepCXX_TRUE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -MT hpcups-EncapsulatorFactory.o -MD -MP -MF $(DEPDIR)/hpcups-EncapsulatorFactory.Tpo -c -o hpcups-EncapsulatorFactory.o `test -f 'prnt/hpcups/EncapsulatorFactory.cpp' || echo '$(srcdir)/'`prnt/hpcups/EncapsulatorFactory.cpp
 @am__fastdepCXX_TRUE@	$(am__mv) $(DEPDIR)/hpcups-EncapsulatorFactory.Tpo $(DEPDIR)/hpcups-EncapsulatorFactory.Po
@@ -7451,20 +7431,6 @@ hpcups-Pcl3Gui.obj: prnt/hpcups/Pcl3Gui.
 @AMDEP_TRUE@@am__fastdepCXX_FALSE@	DEPDIR=$(DEPDIR) $(CXXDEPMODE) $(depcomp) @AMDEPBACKSLASH@
 @am__fastdepCXX_FALSE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -c -o hpcups-Pcl3Gui.obj `if test -f 'prnt/hpcups/Pcl3Gui.cpp'; then $(CYGPATH_W) 'prnt/hpcups/Pcl3Gui.cpp'; else $(CYGPATH_W) '$(srcdir)/prnt/hpcups/Pcl3Gui.cpp'; fi`
 
-hpcups-Pcl3Gui2.o: prnt/hpcups/Pcl3Gui2.cpp
-@am__fastdepCXX_TRUE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -MT hpcups-Pcl3Gui2.o -MD -MP -MF $(DEPDIR)/hpcups-Pcl3Gui2.Tpo -c -o hpcups-Pcl3Gui2.o `test -f 'prnt/hpcups/Pcl3Gui2.cpp' || echo '$(srcdir)/'`prnt/hpcups/Pcl3Gui2.cpp
-@am__fastdepCXX_TRUE@	$(am__mv) $(DEPDIR)/hpcups-Pcl3Gui2.Tpo $(DEPDIR)/hpcups-Pcl3Gui2.Po
-@AMDEP_TRUE@@am__fastdepCXX_FALSE@	source='prnt/hpcups/Pcl3Gui2.cpp' object='hpcups-Pcl3Gui2.o' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCXX_FALSE@	DEPDIR=$(DEPDIR) $(CXXDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCXX_FALSE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -c -o hpcups-Pcl3Gui2.o `test -f 'prnt/hpcups/Pcl3Gui2.cpp' || echo '$(srcdir)/'`prnt/hpcups/Pcl3Gui2.cpp
-
-hpcups-Pcl3Gui2.obj: prnt/hpcups/Pcl3Gui2.cpp
-@am__fastdepCXX_TRUE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -MT hpcups-Pcl3Gui2.obj -MD -MP -MF $(DEPDIR)/hpcups-Pcl3Gui2.Tpo -c -o hpcups-Pcl3Gui2.obj `if test -f 'prnt/hpcups/Pcl3Gui2.cpp'; then $(CYGPATH_W) 'prnt/hpcups/Pcl3Gui2.cpp'; else $(CYGPATH_W) '$(srcdir)/prnt/hpcups/Pcl3Gui2.cpp'; fi`
-@am__fastdepCXX_TRUE@	$(am__mv) $(DEPDIR)/hpcups-Pcl3Gui2.Tpo $(DEPDIR)/hpcups-Pcl3Gui2.Po
-@AMDEP_TRUE@@am__fastdepCXX_FALSE@	source='prnt/hpcups/Pcl3Gui2.cpp' object='hpcups-Pcl3Gui2.obj' libtool=no @AMDEPBACKSLASH@
-@AMDEP_TRUE@@am__fastdepCXX_FALSE@	DEPDIR=$(DEPDIR) $(CXXDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCXX_FALSE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -c -o hpcups-Pcl3Gui2.obj `if test -f 'prnt/hpcups/Pcl3Gui2.cpp'; then $(CYGPATH_W) 'prnt/hpcups/Pcl3Gui2.cpp'; else $(CYGPATH_W) '$(srcdir)/prnt/hpcups/Pcl3Gui2.cpp'; fi`
-
 hpcups-LJMono.o: prnt/hpcups/LJMono.cpp
 @am__fastdepCXX_TRUE@	$(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(hpcups_CXXFLAGS) $(CXXFLAGS) -MT hpcups-LJMono.o -MD -MP -MF $(DEPDIR)/hpcups-LJMono.Tpo -c -o hpcups-LJMono.o `test -f 'prnt/hpcups/LJMono.cpp' || echo '$(srcdir)/'`prnt/hpcups/LJMono.cpp
 @am__fastdepCXX_TRUE@	$(am__mv) $(DEPDIR)/hpcups-LJMono.Tpo $(DEPDIR)/hpcups-LJMono.Po
diff -up hplip-3.16.11/prnt/hpcups/EncapsulatorFactory.cpp.no-ernie hplip-3.16.11/prnt/hpcups/EncapsulatorFactory.cpp
--- hplip-3.16.11/prnt/hpcups/EncapsulatorFactory.cpp.no-ernie	2016-11-23 07:02:48.000000000 +0100
+++ hplip-3.16.11/prnt/hpcups/EncapsulatorFactory.cpp	2016-11-25 18:10:12.275060823 +0100
@@ -64,10 +64,6 @@ Encapsulator *EncapsulatorFactory::GetEn
     {
         return new Pcl3Gui();
     }
-    if (!strcmp (encap_tech, "pcl3gui2"))
-    {
-        return new Pcl3Gui2();
-    }
     if (!strcmp (encap_tech, "ljmono"))
     {
         return new LJMono();

