diff --git a/installer/dcheck.py b/installer/dcheck.py
index 3d0624e..1817819 100644
--- a/installer/dcheck.py
+++ b/installer/dcheck.py
@@ -366,11 +366,11 @@ def get_xsane_version():
 
 def get_pil_version():
     try:
-        from PIL import Image
+        import PIL
     except ImportError:
         return '-'
     else:
-         return Image.PILLOW_VERSION
+         return PIL.__version__
 
 def get_libpthread_version():
     try:
