diff -up hplip-3.17.4/installer/core_install.py.check-cups hplip-3.17.4/installer/core_install.py
--- hplip-3.17.4/installer/core_install.py.check-cups	2017-05-29 16:39:41.172572873 +0200
+++ hplip-3.17.4/installer/core_install.py	2017-05-29 16:42:12.199386100 +0200
@@ -338,9 +338,8 @@ class CoreInstall(object):
             # Required base packages
             'libjpeg':          (True,  ['base'], JPEG_STR, self.check_libjpeg, DEPENDENCY_RUN_AND_COMPILE_TIME,'-',None, GENERALDEP),
             'libtool':          (True,  ['base'], LIBTOOL_STR, self.check_libtool, DEPENDENCY_COMPILE_TIME,'-','libtool --version',COMPILEDEP),
-            'cups' :            (True,  ['base'], CUPS_STR, self.check_cups, DEPENDENCY_RUN_TIME,'1.1','cups-config --version', EXTERNALDEP),
+            'cups' :            (True,  ['base'], CUPS_STR, self.check_cups, DEPENDENCY_RUN_TIME,'1.1','rpm -qa cups', EXTERNALDEP),
             'cups-devel':       (True,  ['base'], CUPS_DEV_STR, self.check_cups_devel, DEPENDENCY_COMPILE_TIME,'-','cups-config --version', GENERALDEP),
-            'cups-image':       (True,  ['base'], CUPS_IMG_STR, self.check_cups_image, DEPENDENCY_COMPILE_TIME,'-','cups-config --version', GENERALDEP),
             'gcc' :             (True,  ['base'], GCC_STR, self.check_gcc, DEPENDENCY_COMPILE_TIME, '-','gcc --version',COMPILEDEP),
             'make' :            (True,  ['base'], MAKE_STR, self.check_make, DEPENDENCY_COMPILE_TIME,'3.0','make --version',COMPILEDEP),
             'libpthread' :      (True,  ['base'], THREAD_STR, self.check_libpthread, DEPENDENCY_RUN_AND_COMPILE_TIME, '-','FUNC#get_libpthread_version', GENERALDEP),
