From f8894ac2ed9c7e7e56d2318f032b36df92cf14d1 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 11 Aug 2025 19:09:41 +0800
Subject: [PATCH 1/2] AOSCOS: icecc-create-env: add ld.so.conf.d

---
 client/icecc-create-env.in | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/client/icecc-create-env.in b/client/icecc-create-env.in
index 4fea59c..96741c8 100755
--- a/client/icecc-create-env.in
+++ b/client/icecc-create-env.in
@@ -526,6 +526,12 @@ if test -f /etc/ld.so.conf; then
     echo "$directive $path"
   done </etc/ld.so.conf >$tmp_ld_so_conf
   add_file $tmp_ld_so_conf /etc/ld.so.conf
+  # populate ld.so.conf.d if it exists
+  if test -d /etc/ld.so.conf.d; then
+    for file in /etc/ld.so.conf.d/*; do
+      add_file "$file"
+    done
+  fi
 fi
 
 new_target_files=
-- 
2.50.1

