diff --git a/CMakeLists.txt b/CMakeLists.txt
index 34123b3..d2308e4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,7 @@ endif ()
 
 
 project (apitrace)
+find_package (PkgConfig REQUIRED)
 
 
 ##############################################################################
@@ -241,9 +215,12 @@ elseif (ENABLE_X11)
     endif ()
 endif ()
 
+
 if (ENABLE_EGL AND ENABLE_WAFFLE)
     # Use Waffle for eglretrace
-    find_package (Waffle REQUIRED)
+    # find_package (Waffle REQUIRED)
+    pkg_check_modules (Waffle REQUIRED waffle-1)
+
 endif ()
 
 if (ENABLE_EGL AND NOT ENABLE_X11)
