From b6a50bf04c70eb57329f6a3be598d367dc6c1112 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Thu, 13 Jul 2017 02:02:50 +0800
Subject: [PATCH] Add CMake build file

---
 CMakeLists.txt | 93 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 93 insertions(+)
 create mode 100644 CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..160a4f8
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,93 @@
+cmake_minimum_required(VERSION 2.8)
+
+PROJECT(dump1090) 
+
+SET(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/modules" ${CMAKE_MODULE_PATH})
+
+INCLUDE(CheckIncludeFiles)
+INCLUDE(FindPkgConfig)
+
+SET(CMAKE_C_STANDARD 11)
+SET(CMAKE_C_STANDARD_REQUIRED ON)
+
+SET(PROJECT_AUTHOR      "dump1090 developers")
+SET(PROJECT_URL         "https://github.com/flightaware/dump1090")
+SET(PROJECT_DESCRIPTION "Dump1090 is a simple Mode S decoder for RTLSDR devices")
+SET(VERSION_MAJOR       "3")
+SET(VERSION_MINOR       "6")
+SET(VERSION_RELEASE     "1")
+SET(VERSION             "${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_RELEASE}")
+
+OPTION(WANT_RTLSDR      "Include RTL SDR support"    ON)
+OPTION(WANT_BLADERF     "Include bladeRF support"    ON)
+OPTION(WANT_FAUP1090    "Build FlightAware uploader" OFF)
+
+SET(DUMP1090_VERSION    "${VERSION}")
+SET(DUMP1090_VARIANT    "dump1090-fa")
+
+find_package(Threads REQUIRED)
+find_package(Curses REQUIRED)
+
+pkg_check_modules(LIBUSB REQUIRED libusb) 
+IF(WANT_RTLSDR)
+    pkg_check_modules(RTLSDR librtlsdr)
+ELSE(WANT_RTLSDR)
+    SET(RTLSDR_FOUND FALSE)
+ENDIF(WANT_RTLSDR)
+IF(WANT_BLADERF)
+    pkg_check_modules(BLADERF libbladeRF)
+ELSE(WANT_BLADERF)
+    SET(BLADERF_FOUND FALSE)
+ENDIF(WANT_BLADERF)
+include_directories(AFTER ${LIBUSB_INCLUDE_DIRS})
+
+SET(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -O2 -g -Wall -W -D_DEFAULT_SOURCE -DMODES_DUMP1090_VERSION=\\\"${DUMP1090_VERSION}\\\" -DMODES_DUMP1090_VARIANT=\\\"${DUMP1090_VARIANT}\\\"")
+
+IF(HTMLPATH)
+    SET(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -DHTMLPATH=\\\"${HTMLPATH}\\\"")
+ENDIF()
+
+message(STATUS "Configuration Summary\n-- RTL SDR support: ${RTLSDR_FOUND} \n-- bladeRF support: ${BLADERF_FOUND}\n-- Build FlightAware uploader: ${WANT_FAUP1090} \n ")
+
+SET(COMMONSRC anet.c mode_ac.c mode_s.c net_io.c crc.c stats.c cpr.c icao_filter.c track.c util.c comm_b.c)
+SET(DUMP1090SRC dump1090.c interactive.c demod_2400.c convert.c sdr_ifile.c sdr.c)
+
+IF(RTLSDR_FOUND)
+    set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -DENABLE_RTLSDR ")
+    LIST(APPEND DUMP1090SRC sdr_rtlsdr.c)
+ENDIF(RTLSDR_FOUND)
+IF(BLADERF_FOUND)
+    set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -DENABLE_BLADERF ")
+    LIST(APPEND DUMP1090SRC sdr_bladerf.c)
+ENDIF(BLADERF_FOUND)
+add_library(dump1090_common OBJECT  ${COMMONSRC})
+add_executable(dump1090 $<TARGET_OBJECTS:dump1090_common> ${DUMP1090SRC})
+IF(RTLSDR_FOUND)
+    target_link_libraries(dump1090 ${RTLSDR_LIBRARIES})
+ENDIF(RTLSDR_FOUND)
+IF(BLADERF_FOUND)
+    target_link_libraries(dump1090 ${BLADERF_LIBRARIES})
+ENDIF(BLADERF_FOUND)
+if(THREADS_HAVE_PTHREAD_ARG)
+  target_compile_options(dump1090 PUBLIC "-pthread")
+endif()
+if(CMAKE_THREAD_LIBS_INIT)
+  target_link_libraries(dump1090 "${CMAKE_THREAD_LIBS_INIT}")
+endif()
+
+target_link_libraries(dump1090 rt m pthread ${CURSES_LIBRARIES})
+
+add_executable(view1090 $<TARGET_OBJECTS:dump1090_common> view1090.c interactive.c)
+target_link_libraries(view1090 rt m pthread ${CURSES_LIBRARIES})
+
+install(TARGETS dump1090 view1090
+        RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/bin/)
+
+IF(WANT_FAUP1090)
+    add_executable(faup1090 $<TARGET_OBJECTS:dump1090_common> faup1090.c)
+    target_link_libraries(faup1090 rt m pthread ${CURSES_LIBRARIES})
+    install(TARGETS faup1090
+            RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/piaware/helpers/)
+ENDIF()
+
+install(DIRECTORY public_html/ DESTINATION ${CMAKE_INSTALL_PREFIX}/share/dump1090/html/)
-- 
2.13.2

