From 7a9ad779796b57e8e7ac4af019f3b50319757223 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sat, 23 May 2020 04:33:53 -0500
Subject: [PATCH] Use python3

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 4a59191..a16dc09 100644
--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ $(FILENAME): $(FILENAME).gz
 	gzip -k -d $(FILENAME).gz

 zhwiki.raw: $(FILENAME)
-	./convert.py $(FILENAME) > zhwiki.raw
+	python3 ./convert.py $(FILENAME) > zhwiki.raw

 zhwiki.dict: zhwiki.raw
 	libime_pinyindict zhwiki.raw zhwiki.dict
--
2.26.2

