From: Roger Shimizu <rogershimizu@gmail.com>
Date: Wed, 25 May 2016 00:26:34 +0900
Subject: update test of python-cram according to libcork

---
 tests/CMakeLists.txt | 39 +++++++++++++++++----------------------
 tests/ccram          | 19 +++++++++++++++----
 2 files changed, 32 insertions(+), 26 deletions(-)

diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 41b778b..b7a7f4f 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -39,27 +39,22 @@ make_test(test-iterator)
 #-----------------------------------------------------------------------
 # Command-line tests
 
-if(TILERA)
-    configure_file(tile-test ${CMAKE_BINARY_DIR}/tile-test COPYONLY)
-endif(TILERA)
+find_package(PythonInterp)
 
-find_program(
-    CRAM_EXECUTABLE
-    NAMES cram
-    HINTS ENV CRAM_DIR
-    PATH_SUFFIXES bin
-    DOC "Cram test harness"
-)
-
-if (CRAM_EXECUTABLE)
+if (PYTHON_EXECUTABLE)
     configure_file(ccram ${CMAKE_BINARY_DIR}/ccram COPYONLY)
-    add_test(
-        cram-tests
-        ${CMAKE_COMMAND} -E chdir ${CMAKE_BINARY_DIR}
-        ${CMAKE_BINARY_DIR}/ccram
-            --cram ${CRAM_EXECUTABLE}
-            --tests ${CMAKE_CURRENT_SOURCE_DIR}
-    )
-else (CRAM_EXECUTABLE)
-    message(WARNING "Unable to find cram test harness; skipping tests.")
-endif (CRAM_EXECUTABLE)
+    file(GLOB_RECURSE TESTS "${CMAKE_CURRENT_SOURCE_DIR}/*.t")
+    foreach(TEST ${TESTS})
+        get_filename_component(TEST_NAME "${TEST}" NAME_WE)
+        add_test(
+            ${TEST_NAME}
+            ${CMAKE_COMMAND} -E chdir ${CMAKE_BINARY_DIR}
+            ${CMAKE_BINARY_DIR}/ccram
+                --python ${PYTHON_EXECUTABLE}
+                --root ${CMAKE_SOURCE_DIR}
+                --tests ${TEST}
+        )
+    endforeach(TEST)
+else (PYTHON_EXECUTABLE)
+    message(WARNING "Unable to find Python; skipping cram tests.")
+endif (PYTHON_EXECUTABLE)
diff --git a/tests/ccram b/tests/ccram
index 4054700..88ada7e 100755
--- a/tests/ccram
+++ b/tests/ccram
@@ -1,11 +1,19 @@
 #!/bin/sh
 
-if [ "$1" = "--cram" ]; then
+if [ "$1" = "--python" ]; then
     shift
-    CRAM="$1"
+    PYTHON="$1"
     shift
 else
-    CRAM=cram
+    PYTHON=python
+fi
+
+if [ "$1" = "--root" ]; then
+    shift
+    ROOT="$1"
+    shift
+else
+    ROOT=$(dirname $PWD)
 fi
 
 if [ "$1" = "--tests" ]; then
@@ -16,5 +24,8 @@ else
     TESTS=../tests
 fi
 
+export ROOT
+
+LD_LIBRARY_PATH="$PWD/src:$LD_LIBRARY_PATH" \
 PATH="$PWD/src:$PATH" \
-  "$CRAM" "$@" "$TESTS"
+  cram3 "$@" "$TESTS"
