From 25a777bb0ee2e2ee17d87006f76b3cea5d15a9f7 Mon Sep 17 00:00:00 2001
From: Enrico Scholz <enrico.scholz@informatik.tu-chemnitz.de>
Date: Mon, 25 Feb 2008 17:48:25 +0100
Subject: [PATCH 2/9] fixed broken format string modifiers

---
 examples/json.c      |    4 ++--
 examples/parse_xml.c |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/examples/json.c b/examples/json.c
index 89fe82b..91ea50d 100644
--- a/examples/json.c
+++ b/examples/json.c
@@ -41,7 +41,7 @@ printAsXml(xmlrpc_value * const valP) {
     printf("XML-RPC XML:\n");
 
     printf("%.*s\n",
-           XMLRPC_MEMBLOCK_SIZE(char, &out),
+           (int)XMLRPC_MEMBLOCK_SIZE(char, &out),
            XMLRPC_MEMBLOCK_CONTENTS(char, &out));
 
     XMLRPC_MEMBLOCK_CLEAN(char, &out);
@@ -70,7 +70,7 @@ printAsJson(xmlrpc_value * const valP) {
     printf("JSON:\n");
 
     printf("%.*s\n",
-           XMLRPC_MEMBLOCK_SIZE(char, &out),
+           (int)XMLRPC_MEMBLOCK_SIZE(char, &out),
            XMLRPC_MEMBLOCK_CONTENTS(char, &out));
 
     XMLRPC_MEMBLOCK_CLEAN(char, &out);
diff --git a/examples/parse_xml.c b/examples/parse_xml.c
index 2e6c508..4f6c308 100644
--- a/examples/parse_xml.c
+++ b/examples/parse_xml.c
@@ -58,7 +58,7 @@ describeXmlElement(const xml_element * const elemP,
     printf("%sXML element type:         '%s'\n",
            prefix, xml_element_name(elemP));
 
-    printf("%sNumber of child elements: %u\n",
+    printf("%sNumber of child elements: %zu\n",
            prefix, xml_element_children_size(elemP));
 
     for (i = 0; i < xml_element_children_size(elemP); ++i) {
-- 
1.7.6

