--- libgda-6.0.0/meson.build~	2021-01-31 18:29:25.129592000 -0600
+++ libgda-6.0.0/meson.build	2021-12-21 13:42:39.324465066 -0600
@@ -64,7 +64,7 @@
 libxslt_dep = dependency('libxslt', required: false)
 jsonglib_dep = dependency('json-glib-1.0',required : false)
 postgres_dep = dependency ('libpq', required: false)
-mysql_dep = dependency ('mysqlclient', required: false)
+mysql_dep = dependency ('mariadb', required: false)
 mysql_args = []
 if mysql_dep.found ()
 if mysql_dep.version ().version_compare ('>21.0.0')
