Index: b/tools/build/src/tools/gcc.jam
===================================================================
--- a/tools/build/src/tools/gcc.jam
+++ b/tools/build/src/tools/gcc.jam
@@ -451,17 +451,7 @@ rule setup-address-model ( targets * : s
         else
         {
             local arch = [ feature.get-values architecture : $(properties) ] ;
-            if $(arch) != arm
-            {
-                if $(model) = 32
-                {
-                    option = -m32 ;
-                }
-                else if $(model) = 64
-                {
-                    option = -m64 ;
-                }
-            }
+
             # For darwin, the model can be 32_64. darwin.jam will handle that
             # on its own.
         }
Index: b/tools/build/src/tools/pathscale.jam
===================================================================
--- a/tools/build/src/tools/pathscale.jam
+++ b/tools/build/src/tools/pathscale.jam
@@ -89,8 +89,6 @@ flags pathscale.compile OPTIONS <warning
 flags pathscale.compile OPTIONS <debug-symbols>on : -ggdb ;
 flags pathscale.compile OPTIONS <profiling>on     : -pg ;
 flags pathscale.compile OPTIONS <link>shared      : -fPIC ;
-flags pathscale.compile OPTIONS <address-model>32 : -m32 ;
-flags pathscale.compile OPTIONS <address-model>64 : -m64 ;
 
 flags pathscale.compile USER_OPTIONS <cflags> ;
 flags pathscale.compile.c++ USER_OPTIONS <cxxflags> ;
