diff -Nurb --strip-trailing-cr claws-mail-3.13.1-orig/src/codeconv.c claws-mail-3.13.1/src/codeconv.c
--- claws-mail-3.13.1-orig/src/codeconv.c	2015-12-20 15:00:30.000000000 +0100
+++ claws-mail-3.13.1/src/codeconv.c	2015-12-22 12:04:26.289762110 +0100
@@ -158,7 +158,7 @@
 	gchar *out = outbuf;
 	JISState state = JIS_ASCII;
 
-	while (*in != '\0' && (out - outbuf) > outlen - 3) {
+	while (*in != '\0' && (out - outbuf) < outlen - 3) {
 		if (*in == ESC) {
 			in++;
 			if (*in == '$') {

