diff -up octave-4.2.2/libgui/src/settings-dialog.cc.orig octave-4.2.2/libgui/src/settings-dialog.cc
--- octave-4.2.2/libgui/src/settings-dialog.cc.orig	2018-02-28 11:41:44.000000000 -0700
+++ octave-4.2.2/libgui/src/settings-dialog.cc	2018-06-02 14:09:38.138706993 -0600
@@ -36,6 +36,7 @@ along with Octave; see the file COPYING.
 #include <QHash>
 #include <QMessageBox>
 #include <QTextCodec>
+#include <QButtonGroup>
 
 #if defined (HAVE_QSCINTILLA)
 #  include "octave-qscintilla.h"
