--- octave-4.2.0/libinterp/corefcn/pr-output.cc	2016-11-13 09:16:10.000000000 -0600
+++ octave-4.2.0.gcc6/libinterp/corefcn/pr-output.cc	2016-11-29 08:16:02.580813004 -0600
@@ -33,6 +33,8 @@
 #include <sstream>
 #include <string>
 
+#include <cmath>
+
 #include "Array-util.h"
 #include "CMatrix.h"
 #include "Range.h"
@@ -3265,13 +3267,13 @@
             {
               int new_digits
                 = static_cast<int>
-                  (std::floor (log10 (double (abs (nda(i).value ()))) + 1.0));
+                  (std::floor (log10 (double (std::abs (nda(i).value ()))) + 1.0));
 
               if (new_digits > digits)
                 digits = new_digits;
 
               if (! isneg)
-                isneg = (abs (nda(i).value ()) != nda(i).value ());
+                isneg = (std::abs (nda(i).value ()) != nda(i).value ());
             }
 
           fw = digits + isneg;
