diff -up octave-4.0.0/etc/icons/octave.appdata.xml.in.appdata octave-4.0.0/etc/icons/octave.appdata.xml.in
--- octave-4.0.0/etc/icons/octave.appdata.xml.in.appdata	2015-05-23 08:21:53.000000000 -0600
+++ octave-4.0.0/etc/icons/octave.appdata.xml.in	2015-12-30 16:19:24.324969806 -0700
@@ -52,5 +52,4 @@
   </screenshots>
   <url type="homepage">http://www.octave.org</url>
   <updatecontact>maintainers@octave.org</updatecontact>
-  <project_group>GNU</project_group>
 </application>
