--- a/Make.defaults	2015-10-28 12:25:51.000000000 -0600
+++ b/Make.defaults	2016-03-28 17:25:15.879431636 -0600
@@ -35,14 +35,14 @@
 clang_cflags =
 gcc_cflags = -Wmaybe-uninitialized
 cflags	= $(CFLAGS) $(ARCH3264) \
-	-Wall -Werror -Wno-cpp  -Wsign-compare -Wno-unused-result \
+	-Wall -Wno-cpp  -Wsign-compare -Wno-unused-result \
 	-Wno-unused-function\
 	-std=gnu11 -fshort-wchar -fPIC -flto -fno-strict-aliasing \
 	-fno-merge-constants \
 	-D_GNU_SOURCE -DCONFIG_$(ARCH) -I${TOPDIR}/include \
 	$(if $(filter $(CC),clang),$(clang_cflags), ) \
 	$(if $(filter $(CC),gcc),$(gcc_cflags), ) \
-	$(call pkg-config-cflags) -Werror
+	$(call pkg-config-cflags)
 clang_ccldflags =
 gcc_ccldflags = -fno-merge-constants \
 	-Wl,--fatal-warnings,--no-allow-shlib-undefined,--default-symver \
