From 10a3991e7fa19c4dbfa05c761df1edb1077ad473 Mon Sep 17 00:00:00 2001
From: Patrick Poulain <petitchevalroux@free.fr>
Date: Tue, 7 Feb 2017 15:02:53 +0100
Subject: [PATCH] Fix ocaml 4.04 compatibility

---
 external/ocamlgtk/src/ml_glib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/external/ocamlgtk/src/ml_glib.c b/external/ocamlgtk/src/ml_glib.c
index 1dca0ae8c..df6fbb0ad 100644
--- a/external/ocamlgtk/src/ml_glib.c
+++ b/external/ocamlgtk/src/ml_glib.c
@@ -162,7 +162,7 @@ static value *lookup_exn_map (GQuark domain)
 static void ml_raise_gerror_exn(GError *, value *) Noreturn;
 static void ml_raise_gerror_exn(GError *err, value *exn)
 {
-  CAMLlocal2(b, msg);
+  value b, msg;
   g_assert (err && exn);
   msg = copy_string(err->message);
   b = alloc_small (3, 0);
