diff --git a/myldr/Makefile.PL b/myldr/Makefile.PL
index 6abee6e..b42f6fe 100644
--- a/myldr/Makefile.PL
+++ b/myldr/Makefile.PL
@@ -168,16 +168,7 @@ $static_ldflags =~ s/(^|\s)-l\S*perl\S*(\s|$)/ /g;
 $boot_ldflags .= " $static_ldflags";
 
 if ($dynperl) {
-    # on Debian derived distros make sure that the Debian package "libperl-dev"
-    # is installed (which contains the /usr/lib/libperl.so symlink)
-    die qq[You need to install the distro (Debian, Ubuntu etc) package "libperl-dev"\n] 
-       if $^O =~ /^(linux|gnukfreebsd)$/i
-          && -x "/usr/bin/dpkg"                 
-                # probably Debian or a derivative
-          && system("dpkg -S $^X >/dev/null 2>&1") == 0 
-                # we're building with the system (distro) perl
-          && system("dpkg -l libperl-dev >/dev/null 2>&1") != 0;
-                # check install status of libperl-dev
+    1;
 } else {
     my $file = $Config{libperl};
     $file = 'libperl.a' if $file eq 'libper'; # same redhat bug? Just making sure...
