From d79698d2566c63237fc1cc735002da7dd6af5fb9 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Sun, 19 May 2024 20:04:25 -0700
Subject: [PATCH] fix lto inject

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 0cfec4c3..1486ced3 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,7 +48,7 @@ required-features = ["winservice"]
 members = ["crates/shadowsocks", "crates/shadowsocks-service"]
 
 [profile.release]
-lto = "fat"
+lto = true
 codegen-units = 1
 incremental = false
 panic = "abort"
-- 
2.45.1

