--- a/libraries/source/spidermonkey/patch.sh	2021-08-10 20:38:38.697902748 -0700
+++ b/libraries/source/spidermonkey/patch.sh	2021-08-10 20:38:48.095052041 -0700
@@ -53,6 +53,3 @@
     # https://svnweb.freebsd.org/ports/head/lang/spidermonkey78/files/patch-third__party_rust_cc_src_lib.rs?view=log
     patch -p1 < ../FixFreeBSDRustThirdPartyOSDetection.diff
 fi
-
-# Correct build for newer rust
-patch -p4 < ../0ad-rust.patch
