From 3c8eec85fd6053e3f6829d8b246fc5be716a481a Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:43:40 +0800
Subject: [PATCH 204/348] MARKER: AOSCOS: Start of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 743f9d158cbf..ede7f6281eb1 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -548,6 +548,7 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
+	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.51.2

