From 689c22971c18887eddb0e6870d6b2e1b8ed472a1 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:45:08 +0800
Subject: [PATCH 296/305] MARKER: AOSCOS: End of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 28f2d974d38a3..c1c45056d492c 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -508,7 +508,6 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
-	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.49.0

