diff --git a/arch/loongarch/kernel/head.S b/arch/loongarch/kernel/head.S
index e3865e92a917..5b0a412e4bb0 100644
--- a/arch/loongarch/kernel/head.S
+++ b/arch/loongarch/kernel/head.S
@@ -47,6 +47,10 @@ SYM_CODE_START(kernel_entry)			# kernel entry point
 	SETUP_DMWINS	t0
 	JUMP_VIRT_ADDR	t0, t1
 
+	/* Workaround mysterious firmware bug */
+	li.d		t0, (1 << 32)
+	csrxchg		t0, t0, 0xf0
+
 	/* Enable PG */
 	li.w		t0, 0xb0		# PLV=0, IE=0, PG=1
 	csrwr		t0, LOONGARCH_CSR_CRMD
@@ -129,6 +133,11 @@ SYM_CODE_START(smpboot_entry)
 	li.d		t1, CSR_STFILL
 	csrxchg		t0, t1, LOONGARCH_CSR_IMPCTL1
 #endif
+
+	/* Workaround mysterious firmware bug */
+	li.d		t0, (1 << 32)
+	csrxchg		t0, t0, 0xf0
+
 	/* Enable PG */
 	li.w		t0, 0xb0		# PLV=0, IE=0, PG=1
 	csrwr		t0, LOONGARCH_CSR_CRMD
diff --git a/arch/loongarch/power/suspend_asm.S b/arch/loongarch/power/suspend_asm.S
index df0865df26fa..749bd075cbcd 100644
--- a/arch/loongarch/power/suspend_asm.S
+++ b/arch/loongarch/power/suspend_asm.S
@@ -78,6 +78,10 @@ SYM_INNER_LABEL(loongarch_wakeup_start, SYM_L_GLOBAL)
 	SETUP_DMWINS	t0
 	JUMP_VIRT_ADDR	t0, t1
 
+	/* Workaround mysterious firmware bug */
+	li.d		t0, (1 << 32)
+	csrxchg		t0, t0, 0xf0
+
 	/* Enable PG */
 	li.w		t0, 0xb0		# PLV=0, IE=0, PG=1
 	csrwr		t0, LOONGARCH_CSR_CRMD

