From 8ce6ca92c6c30549782afd60ae57e932c0b12ae0 Mon Sep 17 00:00:00 2001
From: Victor Toso <victortoso@redhat.com>
Date: Thu, 15 Sep 2016 18:38:47 +0200
Subject: [PATCH 05/55] buildsys: statis typo in configure option

Acked-by: Jonathon Jongsma <jjongsma@redhat.com>
Acked-by: Frediano Ziglio <fziglio@redhat.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 2106085..0c2cfd8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -73,7 +73,7 @@ AC_ARG_ENABLE([pciaccess],
               [enable_pciaccess="yes"])
 
 AC_ARG_ENABLE([static-uinput],
-              [AS_HELP_STRING([--enable-statis-uinput], [Enable use of a fixed, static uinput device for X-servers without hotplug support (default: no)])],
+              [AS_HELP_STRING([--enable-static-uinput], [Enable use of a fixed, static uinput device for X-servers without hotplug support (default: no)])],
               [enable_static_uinput="$enableval"],
               [enable_static_uinput="no"])
 
-- 
2.16.1

