From 2f6324d263c48e49faf99fc419465a1e9b58f8c3 Mon Sep 17 00:00:00 2001
From: Francois Gouget <fgouget@codeweavers.com>
Date: Fri, 7 Apr 2017 10:19:07 +0200
Subject: [PATCH 24/55] README: A couple of spelling fixes and slight rewording

Signed-off-by: Francois Gouget <fgouget@codeweavers.com>
Acked-by: Christophe Fergeau <cfergeau@redhat.com>
---
 README | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/README b/README
index 0cd42d0..2eefbc4 100644
--- a/README
+++ b/README
@@ -20,12 +20,12 @@ Features:
 * Automatic adjustment of the X-session resolution to the client resolution
 * Support of copy and paste (text and images) between the active X-session
   and the client. This supports both the primary selection and the clipboard.
-* Support for transfering files from the client to the agent
+* Support for transferring files from the client to the agent
 * Full support for multiple displays using Xrandr, this requires a new
   enough xorg-x11-drv-qxl driver, as well as a new enough host.
 * Limited support for multiple displays using Xinerama, prerequisites:
-  * A new enough Xorg-server. For Fedora atleast Fedora-17, for RHEL-6 atleast
-    xorg-x11-server-1.10.4-6.el6_2.3
+  * A new enough Xorg server: Fedora 17 or greater, for RHEL 6
+    xorg-x11-server-1.10.4-6.el6_2.3 or greater.
   * A vm configured with multiple qxl devices
   * A guest running the latest spice-vdagent
   Then connect to the vm with the multiple monitor client which you want to
-- 
2.16.1

