From 914574ba479cb318e1d41fe90426526633e68263 Mon Sep 17 00:00:00 2001
From: Francois Gouget <fgouget@codeweavers.com>
Date: Fri, 7 Apr 2017 10:20:59 +0200
Subject: [PATCH 25/55] udscs: A spelling fix in a comment

Signed-off-by: Francois Gouget <fgouget@codeweavers.com>
Acked-by: Christophe Fergeau <cfergeau@redhat.com>
---
 src/udscs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/udscs.h b/src/udscs.h
index 6160568..7b6bff0 100644
--- a/src/udscs.h
+++ b/src/udscs.h
@@ -53,7 +53,7 @@ typedef void (*udscs_read_callback)(struct udscs_connection **connp,
  *    this callback has completed!
  * 2) This callback is always called, even if the disconnect is initiated
  *    by the udscs user through returning -1 from a read callback, or
- *    by explictly calling udscs_destroy_connection.
+ *    by explicitly calling udscs_destroy_connection.
  */
 typedef void (*udscs_disconnect_callback)(struct udscs_connection *conn);
 
-- 
2.16.1

