diff -up choqok-1.6.0/cmake/modules/FindQtOAuth.cmake.qt5 choqok-1.6.0/cmake/modules/FindQtOAuth.cmake
--- choqok-1.6.0/cmake/modules/FindQtOAuth.cmake.qt5	2016-10-07 10:54:41.000000000 -0500
+++ choqok-1.6.0/cmake/modules/FindQtOAuth.cmake	2017-02-24 08:11:18.224609358 -0600
@@ -18,11 +18,11 @@ if (QTOAUTH_INCLUDE_DIR AND QTOAUTH_LIBR
 else (QTOAUTH_INCLUDE_DIR AND QTOAUTH_LIBRARY)
   if (NOT WIN32)
     find_package(PkgConfig)
-    pkg_check_modules(PC_QTOAUTH QUIET qoauth)
+    pkg_check_modules(PC_QTOAUTH QUIET qoauth-qt5)
     set(QTOAUTH_DEFINITIONS ${PC_QTOAUTH_CFLAGS_OTHER})
   endif(NOT WIN32)
 
-  find_library(QTOAUTH_LIBRARY NAMES qoauth
+  find_library(QTOAUTH_LIBRARY NAMES qoauth-qt5 qoauth
     HINTS ${PC_QTOAUTH_LIBDIR} ${PC_QTOAUTH_LIBRARY_DIRS}
   )
 
