diff -up choqok-1.6.0/libchoqok/uploader.h.me choqok-1.6.0/libchoqok/uploader.h
--- choqok-1.6.0/libchoqok/uploader.h.me	2020-02-17 12:33:45.694259010 +0100
+++ choqok-1.6.0/libchoqok/uploader.h	2020-02-17 12:33:28.350058895 +0100
@@ -25,6 +25,7 @@
 #define UPLOADER_H
 
 #include <QUrl>
+#include <QMap>
 
 #include "plugin.h"
 
diff -up choqok-1.6.0/plugins/betternotify/notify.h.me choqok-1.6.0/plugins/betternotify/notify.h
--- choqok-1.6.0/plugins/betternotify/notify.h.me	2020-02-17 12:26:17.371091306 +0100
+++ choqok-1.6.0/plugins/betternotify/notify.h	2020-02-17 12:26:35.392298515 +0100
@@ -27,6 +27,7 @@
 #include <QPoint>
 #include <QPointer>
 #include <QTimer>
+#include <QMap>
 
 #include "plugin.h"
 
diff -up choqok-1.6.0/plugins/untiny/untiny.h.me choqok-1.6.0/plugins/untiny/untiny.h
--- choqok-1.6.0/plugins/untiny/untiny.h.me	2020-02-17 12:28:19.693497785 +0100
+++ choqok-1.6.0/plugins/untiny/untiny.h	2020-02-17 12:28:36.138687302 +0100
@@ -30,6 +30,7 @@
 #include <QQueue>
 #include <QUrl>
 #include <QPointer>
+#include <QMap>
 
 namespace KIO {
 class Job;
