diff -Naur mpd-0.21.14/src/output/plugins/AlsaOutputPlugin.cxx mpd-0.21.14.build/src/output/plugins/AlsaOutputPlugin.cxx
--- mpd-0.21.14/src/output/plugins/AlsaOutputPlugin.cxx	2019-08-21 08:47:53.000000000 +0000
+++ mpd-0.21.14.build/src/output/plugins/AlsaOutputPlugin.cxx	2019-09-24 18:23:44.483389641 +0000
@@ -43,6 +43,7 @@
 
 #include <alsa/asoundlib.h>
 
+#include <boost/next_prior.hpp>
 #include <boost/lockfree/spsc_queue.hpp>
 
 #include <string>
