From 0d00d23c98db032f76dbb744dd2758d247a7619c Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Wed, 29 Jan 2025 15:24:28 -0800
Subject: [PATCH] systemd/system: run service with mpd user

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 systemd/system/mpd.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/systemd/system/mpd.service.in b/systemd/system/mpd.service.in
index fcb4b52c1..64ccddb36 100644
--- a/systemd/system/mpd.service.in
+++ b/systemd/system/mpd.service.in
@@ -4,6 +4,7 @@ Documentation=man:mpd(1) man:mpd.conf(5)
 After=network.target sound.target
 
 [Service]
+User=mpd
 Type=notify
 ExecStart=@prefix@/bin/mpd --systemd
 
-- 
2.49.0

