From 375c86a9f5270fa6b10b55dd4be1b8639d612111 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <wukaiyang@loongfans.cn>
Date: Mon, 20 Jul 2026 10:13:03 +0800
Subject: [PATCH] AOSCOS: chore: optimize description for desktop entry

Signed-off-by: Kaiyang Wu <wukaiyang@loongfans.cn>
---
 res/activate-linux.desktop | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/res/activate-linux.desktop b/res/activate-linux.desktop
index e308754..0e4621e 100644
--- a/res/activate-linux.desktop
+++ b/res/activate-linux.desktop
@@ -2,9 +2,9 @@
 Type=Application
 Version=1.0
 Name=Activate Linux
-Comment=a very cool software
+Comment=Meme program that displays an "Activate Windows"-like watermark on the desktop
 Exec=activate-linux
 Icon=activate-linux
 GenericName=Activate Linux
 Categories=Utility
-Terminal=false
\ No newline at end of file
+Terminal=false
-- 
2.52.0

