From efe228112804833a2d615e5ec1b9028c87837de2 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Mon, 12 May 2025 19:16:25 -0700
Subject: [PATCH 2/2] chore: better description and icon name for desktop entry

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 res/activate-linux.desktop | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/res/activate-linux.desktop b/res/activate-linux.desktop
index 27491a0..0e4621e 100644
--- a/res/activate-linux.desktop
+++ b/res/activate-linux.desktop
@@ -2,9 +2,9 @@
 Type=Application
 Version=1.0
 Name=Activate Linux
-Comment=a very cool software
+Comment=Meme program that displays an "Activate Windows"-like watermark on the desktop
 Exec=activate-linux
-Icon=icon
+Icon=activate-linux
 GenericName=Activate Linux
 Categories=Utility
-Terminal=false
\ No newline at end of file
+Terminal=false
-- 
2.49.0

