From 5e2fbdb671b2441d48c19f79281c4270a2e390f0 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Wed, 30 Nov 2022 16:31:46 -0500
Subject: [PATCH 3/3] dpdk 22.11: remove rxmode.split_hdr_size

---
 lib/netdev-dpdk.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/lib/netdev-dpdk.c b/lib/netdev-dpdk.c
index 647bc2bd7..9e1a1d4ce 100644
--- a/lib/netdev-dpdk.c
+++ b/lib/netdev-dpdk.c
@@ -166,7 +166,6 @@ typedef uint16_t dpdk_port_t;
 
 static const struct rte_eth_conf port_conf = {
     .rxmode = {
-        .split_hdr_size = 0,
         .offloads = 0,
     },
     .rx_adv_conf = {
-- 
2.38.1

