Description: Avoid qreal/double type clashes on ARM
Author: Colin Watson <cjwatson@ubuntu.com>
Forwarded: no
Last-Update: 2013-05-16

--- scribus-1.4.2.dfsg+r18267.orig/scribus/scribus.h
+++ scribus-1.4.2.dfsg+r18267/scribus/scribus.h
@@ -466,7 +466,7 @@ public slots:
 	/** Fragt nach den Farben */
 //	void GetBrushPen();
 	/** Erzeugt einen Rahmen */
-	void MakeFrame(int f, int c, double *vals);
+	void MakeFrame(int f, int c, qreal *vals);
 	//** Loescht ein Element */
 	//void DeleteObjekt();
 	/** Duplicate current item */
--- scribus-1.4.2.dfsg+r18267.orig/scribus/pageitem.cpp
+++ scribus-1.4.2.dfsg+r18267/scribus/pageitem.cpp
@@ -4597,7 +4597,7 @@ void PageItem::copyToCopyPasteBuffer(str
 
 
 //Moved from View
-void PageItem::SetFrameShape(int count, double *vals)
+void PageItem::SetFrameShape(int count, qreal *vals)
 {
 	PoLine.resize(0);
 	for (int a = 0; a < count-3; a += 4)
@@ -4620,7 +4620,7 @@ void PageItem::SetFrameShape(int count,
 
 void PageItem::SetRectFrame()
 {
-	static double rect[] = {  0.0,   0.0,   0.0,   0.0,
+	static qreal rect[] = {  0.0,   0.0,   0.0,   0.0,
 							100.0,   0.0, 100.0,   0.0,
 							100.0,   0.0, 100.0,   0.0,
 							100.0, 100.0, 100.0, 100.0,
@@ -4635,7 +4635,7 @@ void PageItem::SetRectFrame()
 
 void PageItem::SetOvalFrame()
 {
-	static double rect[] = {   100.0,  50.0, 100.0,       77.615235,
+	static qreal rect[] = {   100.0,  50.0, 100.0,       77.615235,
 								50.0, 100.0,  77.615235, 100.0,
 								50.0, 100.0,  22.385765, 100.0,
 								 0.0,  50.0,   0.0,       77.615235,
--- scribus-1.4.2.dfsg+r18267.orig/scribus/pageitem.h
+++ scribus-1.4.2.dfsg+r18267/scribus/pageitem.h
@@ -298,7 +298,7 @@ protected:
 	void DrawObj_PathText(ScPainter *p, double sc);
 public:
 	void DrawObj_Embedded(ScPainter *p, QRectF e, const CharStyle& style, PageItem* cembedded);
-	void SetFrameShape(int count, double *vals);
+	void SetFrameShape(int count, qreal *vals);
 	void SetRectFrame();
 	void SetOvalFrame();
 	void SetFrameRound();
--- scribus-1.4.2.dfsg+r18267.orig/scribus/propertiespalette.cpp
+++ scribus-1.4.2.dfsg+r18267/scribus/propertiespalette.cpp
@@ -4622,7 +4622,7 @@ void PropertiesPalette::MakeIrre(int f,
 			doc->setRedrawBounding(CurItem);
 			break;
 		default:
-			CurItem->SetFrameShape(c, (double*)vals);
+			CurItem->SetFrameShape(c, vals);
 			doc->setRedrawBounding(CurItem);
 			CurItem->FrameType = f+2;
 			break;
--- scribus-1.4.2.dfsg+r18267.orig/scribus/scribusdoc.h
+++ scribus-1.4.2.dfsg+r18267/scribus/scribusdoc.h
@@ -1005,7 +1005,7 @@ public: // Public attributes
 	DocumentInformation documentInfo;
 	int appMode;
 	int SubMode;
-	double *ShapeValues;
+	qreal *ShapeValues;
 	int ValCount;
 	QString DocName;
 	QMap<QString,int> UsedFonts;
--- scribus-1.4.2.dfsg+r18267.orig/scribus/canvasmode_create.cpp
+++ scribus-1.4.2.dfsg+r18267/scribus/canvasmode_create.cpp
@@ -122,7 +122,7 @@ void CreateMode::drawControls(QPainter*
 		{
 			FPointArray poly;
 			int valCount = m_doc->ValCount;
-			double *vals = m_doc->ShapeValues;
+			qreal *vals = m_doc->ShapeValues;
 			for (int a = 0; a < valCount-3; a += 4)
 			{
 				if (vals[a] < 0)
--- scribus-1.4.2.dfsg+r18267.orig/scribus/scribus.cpp
+++ scribus-1.4.2.dfsg+r18267/scribus/scribus.cpp
@@ -7599,7 +7599,7 @@ void ScribusMainWindow::GetBrushPen()
 }
 */
 //CB-->??
-void ScribusMainWindow::MakeFrame(int f, int c, double *vals)
+void ScribusMainWindow::MakeFrame(int f, int c, qreal *vals)
 {
 	PageItem *currItem = doc->m_Selection->itemAt(0);
 	switch (f)
--- scribus-1.4.2.dfsg+r18267.orig/scribus/plugins/imposition/imposition.cpp
+++ scribus-1.4.2.dfsg+r18267/scribus/plugins/imposition/imposition.cpp
@@ -542,7 +542,7 @@ void Imposition::booklet4p(QList<int>* p
 					}
 					groupcontrol->SetRectFrame();
 					
-					double points[32] = {
+					qreal points[32] = {
 					//left upper corner - left lower corner
 					targetDoc->Pages->at(0)->guides.vertical(1,targetDoc->Pages->at(0)->guides.Standard),
 					targetDoc->Pages->at(0)->guides.horizontal(0,targetDoc->Pages->at(i)->guides.Standard),
@@ -730,7 +730,7 @@ void Imposition::booklet4p(QList<int>* p
 					}
 					groupcontrol->SetRectFrame();
 					
-					double points[32] = {
+					qreal points[32] = {
 					//left upper corner - left lower corner
 					0,
 					0,
@@ -893,7 +893,7 @@ void Imposition::booklet4p(QList<int>* p
 					}
 					groupcontrol->SetRectFrame();
 					
-					double points[32] = {
+					qreal points[32] = {
 					//left upper corner - left lower corner
 					0,
 					0,
@@ -1047,7 +1047,7 @@ void Imposition::booklet4p(QList<int>* p
 					}
 					groupcontrol->SetRectFrame();
 					
-					double points[32] = {
+					qreal points[32] = {
 					//left upper corner - left lower corner
 					targetDoc->Pages->at(0)->guides.vertical(1,targetDoc->Pages->at(0)->guides.Standard),
 					targetDoc->Pages->at(0)->guides.horizontal(0,targetDoc->Pages->at(i)->guides.Standard),
