Index: pTk/mTk/additions/ClientWin.c
===================================================================
--- a/pTk/mTk/additions/ClientWin.c
+++ b/pTk/mTk/additions/ClientWin.c
@@ -30,7 +30,7 @@ in this Software without prior written a
 #include <X11/Xlib.h>
 #include <X11/Xatom.h>
 
-static Window TryChildren();
+static Window TryChildren (Display* dpy, Window win, Atom WM_STATE);
 
 /* Find a window with WM_STATE, else return win itself, as per ICCCM */
 
