--- nghttp2-1.51.0/python/Makefile.am	2022-11-12 22:53:06.000000000 -0800
+++ nghttp2-1.51.0.python/python/Makefile.am	2022-12-10 01:11:47.954361170 -0800
@@ -30,10 +30,10 @@
 if ENABLE_PYTHON_BINDINGS
 
 all-local: nghttp2.c
-	$(PYTHON) setup.py build
+	CC="gcc" $(PYTHON) setup.py build
 
 install-exec-local:
-	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
+	CC="gcc" $(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
 
 uninstall-local:
 	rm -f $(DESTDIR)$(libdir)/python*/site-packages/nghttp2.so
