diff -Naur libtraceevent-1.7.1/Documentation/Makefile libtraceevent-1.7.1.doc/Documentation/Makefile
--- libtraceevent-1.7.1/Documentation/Makefile	2023-03-16 13:10:06.083748216 -0400
+++ libtraceevent-1.7.1.doc/Documentation/Makefile	2023-03-16 13:09:48.662743559 -0400
@@ -13,8 +13,6 @@
 _MAN_HTML=$(patsubst %.txt,%.html,$(MAN_TXT))
 _DOC_MAN3=$(patsubst %.txt,%.m,$(MAN3_TXT))
 
-OUTPUT		:= $(obj)/Documentation/
-
 MAN_XML=$(addprefix $(OUTPUT),$(_MAN_XML))
 MAN_HTML=$(addprefix $(OUTPUT),$(_MAN_HTML))
 DOC_MAN3=$(addprefix $(OUTPUT),$(_DOC_MAN3))
