From bbaf03f6fdc9e9160f75626a8dd6c85824b6e186 Mon Sep 17 00:00:00 2001
From: WhiredPlanck <whiredplanck@outlook.com>
Date: Wed, 15 Sep 2021 20:59:12 +0800
Subject: [PATCH] Install binaries and icons after building

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 10a8fb6..b527d6d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,3 +51,6 @@ add_executable(
 target_include_directories(zdl PRIVATE ${PROJECT_SOURCE_DIR}/zdlconf)
 target_include_directories(zdl PRIVATE ${inih_SOURCE_DIR})
 target_link_libraries(zdl Qt5::Core Qt5::Widgets)
+
+install(TARGETS zdl DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(FILES res/zdl3.svg DESTINATION share/icons/hicolor/scalable/apps)
-- 
2.30.2

