diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3a521c8..2c0a323 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,6 +61,8 @@ set_package_properties(BISON PROPERTIES
     PURPOSE "Required for the Trader parser"
 )
 
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
+
 # create a Config.cmake and a ConfigVersion.cmake file and install them
 set(CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/KF5Service")
 
