diff -Naur mac-fdisk-0.1.orig/fdisklabel.c mac-fdisk-0.1.mips/fdisklabel.c
--- mac-fdisk-0.1.orig/fdisklabel.c	2021-08-02 13:59:54.286412944 -0700
+++ mac-fdisk-0.1.mips/fdisklabel.c	2021-08-02 14:01:17.438959658 -0700
@@ -378,7 +378,7 @@
     {
 #if defined (i386) || defined(__amd64)
       if (bsd_initlabel (bsd_part, &bsd_dlabel, bsd_part_index) == 1)
-#elif defined (__alpha__) || defined (__powerpc__) || defined (__mc68000__)
+#elif defined (__alpha__) || defined (__powerpc__) || defined (__mc68000__) || defined (__mips__)
       if (bsd_initlabel (NULL, &bsd_dlabel, 0) == 1)
 #endif
       {
diff -Naur mac-fdisk-0.1.orig/fdisklabel.h mac-fdisk-0.1.mips/fdisklabel.h
--- mac-fdisk-0.1.orig/fdisklabel.h	2021-08-02 13:59:54.286412944 -0700
+++ mac-fdisk-0.1.mips/fdisklabel.h	2021-08-02 14:01:38.110885136 -0700
@@ -58,6 +58,11 @@
 #define BSD_LABELOFFSET   0
 #define	BSD_BBSIZE        0
 #define	BSD_SBSIZE        0
+#elif defined (__mips__)
+#define BSD_LABELSECTOR   0
+#define BSD_LABELOFFSET   0
+#define BSD_BBSIZE        0
+#define BSD_SBSIZE        0
 #else
 #error unknown architecture
 #endif
