--- kaccounts-providers-20.12.1/CMakeLists.txt	2021-01-04 17:23:03.000000000 -0800
+++ kaccounts-providers-20.12.1.nowebengine/CMakeLists.txt	2021-01-09 17:01:35.310844640 -0800
@@ -19,7 +19,9 @@
 find_package(Intltool REQUIRED)
 find_package(KAccounts REQUIRED)
 
-find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Core Qml WebEngine)
+find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Core Qml)
+find_package(Qt5 ${QT_REQUIRED_VERSION} CONFIG REQUIRED Core Qml)
+find_package(Qt5WebEngine ${QT_REQUIRED_VERSION} CONFIG)
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED KIO I18n Declarative Package)
 
 include(KDEInstallDirs)
--- kaccounts-providers-20.12.1/plugins/CMakeLists.txt	2021-01-03 00:50:04.000000000 -0800
+++ kaccounts-providers-20.12.1.nowebengine/plugins/CMakeLists.txt	2021-01-09 17:00:42.971199524 -0800
@@ -1,2 +1,4 @@
 add_subdirectory(owncloud-ui)
-add_subdirectory(nextcloud-ui)
\ No newline at end of file
+if (Qt5WebEngineWidgets_FOUND)
+  add_subdirectory(nextcloud-ui)
+endif()
