diff -Naur source/Makefile zita-convolver-4.0.3.nonative/source/Makefile
--- source/Makefile	2018-08-26 08:35:37.000000000 -0500
+++ source/Makefile	2018-09-03 11:08:32.176091339 -0500
@@ -34,7 +34,6 @@
 CPPFLAGS += -I. -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
 CPPFLAGS += -DENABLE_VECTOR_MODE 
 CXXFLAGS += -fPIC -Wall -ffast-math -funroll-loops -O2
-CXXFLAGS += -march=native
 LDLFAGS += 
 LDLIBS +=
 
