From 4be7c8e24daa27eac9cb6db05b4ca1ae2287db2a Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Wed, 3 Nov 2021 00:44:36 -0500
Subject: [PATCH 13/21] x86-platform: add MAT0028 to ACPI identifiers for
 panasonic laptops

---
 drivers/platform/x86/panasonic-laptop.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/platform/x86/panasonic-laptop.c b/drivers/platform/x86/panasonic-laptop.c
index d9a095d2c0eb..1c44095c0b58 100644
--- a/drivers/platform/x86/panasonic-laptop.c
+++ b/drivers/platform/x86/panasonic-laptop.c
@@ -191,6 +191,7 @@ static const struct acpi_device_id pcc_device_ids[] = {
 	{ "MAT0013", 0},
 	{ "MAT0018", 0},
 	{ "MAT0019", 0},
+	{ "MAT0028", 0},
 	{ "", 0},
 };
 MODULE_DEVICE_TABLE(acpi, pcc_device_ids);
-- 
2.37.1

