From d611593e142719c721b874cc9617e767150ad716 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Wed, 24 Aug 2022 03:54:11 +0000
Subject: [PATCH 16/21] loongson64/init: suppress memcpy out-of-bound checking

---
 arch/mips/loongson64/init.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/mips/loongson64/init.c b/arch/mips/loongson64/init.c
index ee8de1735b7c..395405935a55 100644
--- a/arch/mips/loongson64/init.c
+++ b/arch/mips/loongson64/init.c
@@ -27,7 +27,10 @@ static void __init mips_nmi_setup(void)
 	void *base;
 
 	base = (void *)(CAC_BASE + 0x380);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Warray-bounds"
 	memcpy(base, except_vec_nmi, 0x80);
+#pragma GCC diagnostic pop
 	flush_icache_range((unsigned long)base, (unsigned long)base + 0x80);
 }
 
-- 
2.37.1

