--- a/src/e-sexp.h
+++ b/src/e-sexp.h
@@ -40,7 +40,7 @@
 		GPtrArray *ptrarray;
 		int number;
 		char *string;
-		int bool;
+		int bool_;
 		time_t time;
 	} value;
 };
@@ -78,7 +78,7 @@
 	union {
 		char *string;
 		int number;
-		int bool;
+		int bool_;
 		time_t time;
 		struct {
 			struct _ESExpSymbol *sym;
--- a/src/e-sexp.c
+++ b/src/e-sexp.c
@@ -185,7 +185,7 @@
 e_sexp_result_new_bool(ESExp *f, gboolean b)
 {
   ESExpResult *r = e_sexp_result_new (f, ESEXP_RES_BOOL);
-  r->value.bool = b;
+  r->value.bool_ = b;
   return r;
 }
 
@@ -257,12 +257,12 @@
 	GHashTable *ht = g_hash_table_new(g_str_hash, g_str_equal);
 	struct _glib_sux_donkeys lambdafoo;
 	int type=-1;
-	int bool = TRUE;
+	int bool_ = TRUE;
 	int i;
 	
 	r = e_sexp_result_new(f, ESEXP_RES_UNDEFINED);
 	
-	for (i=0;bool && i<argc;i++) {
+	for (i=0;bool_ && i<argc;i++) {
 		r1 = e_sexp_term_eval(f, argv[i]);
 		if (type == -1)
 			type = r1->type;
@@ -285,7 +285,7 @@
 				g_hash_table_insert(ht, a1[j], GINT_TO_POINTER(n+1));
 			}
 		} else if (r1->type == ESEXP_RES_BOOL) {
-			bool = bool && r1->value.bool;
+			bool_ = bool_ && r1->value.bool_;
 		}
 		e_sexp_result_free(f, r1);
 	}
@@ -298,7 +298,7 @@
 		r->value.ptrarray = lambdafoo.uids;
 	} else if (type == ESEXP_RES_BOOL) {
 		r->type = ESEXP_RES_BOOL;
-		r->value.bool = bool;
+		r->value.bool_ = bool_;
 	}
 	
 	g_hash_table_destroy(ht);
@@ -313,12 +313,12 @@
 	GHashTable *ht = g_hash_table_new(g_str_hash, g_str_equal);
 	struct _glib_sux_donkeys lambdafoo;
 	int type = -1;
-	int bool = FALSE;
+	int bool_ = FALSE;
 	int i;
 	
 	r = e_sexp_result_new(f, ESEXP_RES_UNDEFINED);
 	
-	for (i=0;!bool && i<argc;i++) {
+	for (i=0;!bool_ && i<argc;i++) {
 		r1 = e_sexp_term_eval(f, argv[i]);
 		if (type == -1)
 			type = r1->type;
@@ -337,7 +337,7 @@
 				g_hash_table_insert(ht, a1[j], (void *)1);
 			}
 		} else if (r1->type == ESEXP_RES_BOOL) {
-			bool |= r1->value.bool;				
+			bool_ |= r1->value.bool_;				
 		}
 		e_sexp_result_free(f, r1);
 	}
@@ -350,7 +350,7 @@
 		r->value.ptrarray = lambdafoo.uids;
 	} else if (type == ESEXP_RES_BOOL) {
 		r->type = ESEXP_RES_BOOL;
-		r->value.bool = bool;
+		r->value.bool_ = bool_;
 	}
 	g_hash_table_destroy(ht);
 	
@@ -365,11 +365,11 @@
 
 	if (argc>0) {
 		if (argv[0]->type == ESEXP_RES_BOOL
-		    && argv[0]->value.bool)
+		    && argv[0]->value.bool_)
 			res = FALSE;
 	}
 	r = e_sexp_result_new(f, ESEXP_RES_BOOL);
-	r->value.bool = res;
+	r->value.bool_ = res;
 	return r;
 }
 
@@ -391,13 +391,13 @@
 			e_sexp_fatal_error(f, _("Incompatible types in compare <"));
 		} else if (r1->type == ESEXP_RES_INT) {
 			r->type = ESEXP_RES_BOOL;
-			r->value.bool = r1->value.number < r2->value.number;
+			r->value.bool_ = r1->value.number < r2->value.number;
 		} else if (r1->type == ESEXP_RES_TIME) {
 			r->type = ESEXP_RES_BOOL;
-			r->value.bool = r1->value.time < r2->value.time;
+			r->value.bool_ = r1->value.time < r2->value.time;
 		} else if (r1->type == ESEXP_RES_STRING) {
 			r->type = ESEXP_RES_BOOL;
-			r->value.bool = strcmp(r1->value.string, r2->value.string) < 0;
+			r->value.bool_ = strcmp(r1->value.string, r2->value.string) < 0;
 		}
 		e_sexp_result_free(f, r1);
 		e_sexp_result_free(f, r2);
@@ -423,13 +423,13 @@
 			e_sexp_fatal_error(f, _("Incompatible types in compare >"));
 		} else if (r1->type == ESEXP_RES_INT) {
 			r->type = ESEXP_RES_BOOL;
-			r->value.bool = r1->value.number > r2->value.number;
+			r->value.bool_ = r1->value.number > r2->value.number;
 		} else if (r1->type == ESEXP_RES_TIME) {
 			r->type = ESEXP_RES_BOOL;
-			r->value.bool = r1->value.time > r2->value.time;
+			r->value.bool_ = r1->value.time > r2->value.time;
 		} else if (r1->type == ESEXP_RES_STRING) {
 			r->type = ESEXP_RES_BOOL;
-			r->value.bool = strcmp(r1->value.string, r2->value.string) > 0;
+			r->value.bool_ = strcmp(r1->value.string, r2->value.string) > 0;
 		}
 		e_sexp_result_free(f, r1);
 		e_sexp_result_free(f, r2);
@@ -449,15 +449,15 @@
 		r1 = e_sexp_term_eval(f, argv[0]);
 		r2 = e_sexp_term_eval(f, argv[1]);
 		if (r1->type != r2->type) {
-			r->value.bool = FALSE;
+			r->value.bool_ = FALSE;
 		} else if (r1->type == ESEXP_RES_INT) {
-			r->value.bool = r1->value.number == r2->value.number;
+			r->value.bool_ = r1->value.number == r2->value.number;
 		} else if (r1->type == ESEXP_RES_BOOL) {
-			r->value.bool = r1->value.bool == r2->value.bool;
+			r->value.bool_ = r1->value.bool_ == r2->value.bool_;
 		} else if (r1->type == ESEXP_RES_TIME) {
-			r->value.bool = r1->value.time == r2->value.time;
+			r->value.bool_ = r1->value.time == r2->value.time;
 		} else if (r1->type == ESEXP_RES_STRING) {
-			r->value.bool = strcmp(r1->value.string, r2->value.string) == 0;
+			r->value.bool_ = strcmp(r1->value.string, r2->value.string) == 0;
 		}
 		e_sexp_result_free(f, r1);
 		e_sexp_result_free(f, r2);
@@ -589,7 +589,7 @@
 		r->value.number = argv[0]->value.number;
 		break;
 	case ESEXP_RES_BOOL:
-		r->value.number = argv[0]->value.bool != 0;
+		r->value.number = argv[0]->value.bool_ != 0;
 		break;
 	case ESEXP_RES_STRING:
 		r->value.number = strtoul(argv[0]->value.string, 0, 10);
@@ -617,7 +617,7 @@
 		r->value.string = g_strdup_printf("%d", argv[0]->value.number);
 		break;
 	case ESEXP_RES_BOOL:
-		r->value.string = g_strdup_printf("%d", argv[0]->value.bool != 0);
+		r->value.string = g_strdup_printf("%d", argv[0]->value.bool_ != 0);
 		break;
 	case ESEXP_RES_STRING:
 		r->value.string = g_strdup(argv[0]->value.string);
@@ -639,7 +639,7 @@
 
 	if (argc >=2 && argc<=3) {
 		r = e_sexp_term_eval(f, argv[0]);
-		doit = (r->type == ESEXP_RES_BOOL && r->value.bool);
+		doit = (r->type == ESEXP_RES_BOOL && r->value.bool_);
 		e_sexp_result_free(f, r);
 		if (doit) {
 			return e_sexp_term_eval(f, argv[1]);
@@ -691,7 +691,7 @@
 		break;
 	case ESEXP_TERM_BOOL:
 		r = e_sexp_result_new(f, ESEXP_RES_BOOL);
-		r->value.bool = t->value.bool;
+		r->value.bool_ = t->value.bool_;
 		break;
 	case ESEXP_TERM_TIME:
 		r = e_sexp_result_new (f, ESEXP_RES_TIME);
@@ -751,7 +751,7 @@
 		g_print("string: '%s'\n", r->value.string);
 		break;
 	case ESEXP_RES_BOOL:
-		g_print("bool: %c\n", r->value.bool?'t':'f');
+		g_print("bool: %c\n", r->value.bool_?'t':'f');
 		break;
 	case ESEXP_RES_TIME:
 		g_print("time_t: %ld\n", (long) r->value.time);
@@ -784,7 +784,7 @@
 		g_print(" %d", t->value.number);
 		break;
 	case ESEXP_TERM_BOOL:
-		g_print(" #%c", t->value.bool?'t':'f');
+		g_print(" #%c", t->value.bool_?'t':'f');
 		break;
 	case ESEXP_TERM_TIME:
 		g_print(" %ld", (long) t->value.time);
@@ -936,7 +936,7 @@
 		}
 		
 		t = parse_term_new(f, ESEXP_TERM_BOOL);
-		t->value.bool = (str[0] == 't');
+		t->value.bool_ = (str[0] == 't');
 		break; }
 	case G_TOKEN_SYMBOL:
 		s = g_scanner_cur_value(gs).v_symbol;
--- a/src/logical.c
+++ b/src/logical.c
@@ -41,7 +41,7 @@
 
   r = e_sexp_result_new(f, ESEXP_RES_BOOL);
   /* TODO: decent UTF-8 string compare */
-  r->value.bool = strcmp(argv[0]->value.string, argv[1]->value.string) == 0;
+  r->value.bool_ = strcmp(argv[0]->value.string, argv[1]->value.string) == 0;
   return r;
 }
 
@@ -58,7 +58,7 @@
 
   r = e_sexp_result_new(f, ESEXP_RES_BOOL);
   /* TODO: decent UTF-8 string compare */
-  r->value.bool = strstr(argv[0]->value.string, argv[1]->value.string) != NULL;
+  r->value.bool_ = strstr(argv[0]->value.string, argv[1]->value.string) != NULL;
   return r;
 }
 
@@ -89,7 +89,7 @@
   }
 
   r = e_sexp_result_new(f, ESEXP_RES_BOOL);
-  r->value.bool = (regexec(&regex, string, 0, NULL, 0) == 0) ? TRUE : FALSE;
+  r->value.bool_ = (regexec(&regex, string, 0, NULL, 0) == 0) ? TRUE : FALSE;
   regfree (&regex);
   return r;
 }
--- a/src/actions.c
+++ b/src/actions.c
@@ -62,7 +62,7 @@
 
   switch (argv[0]->type) {
   case ESEXP_RES_BOOL:
-    s_arg0 = g_strdup((argv[0]->value.bool ? "TRUE" : "FALSE"));
+    s_arg0 = g_strdup((argv[0]->value.bool_ ? "TRUE" : "FALSE"));
     break;
 
   case ESEXP_RES_INT:
--- a/src/sexp-test.c
+++ b/src/sexp-test.c
@@ -79,7 +79,7 @@
         g_printerr("Invalid result type for test %s\n", name);
         return 1;
       }
-      if (result->value.bool != expected) {
+      if (result->value.bool_ != expected) {
         g_printerr("Incorrect result for test %s\n", name);
         return 1;
       }
