diff -ur mypaint-1.0.0.orig/mypaint.py mypaint-1.0.0/mypaint.py
--- mypaint-1.0.0.orig/mypaint.py	2011-11-22 21:27:03.000000000 +0100
+++ mypaint-1.0.0/mypaint.py	2012-09-26 16:46:20.507359964 +0200
@@ -59,7 +59,7 @@
         sys.argv_unicode = win32_unicode_argv()
     else:
         sys.argv_unicode = [s.decode(sys.getfilesystemencoding()) for s in sys.argv]
-    scriptdir=os.path.dirname(sys.argv_unicode[0])
+    scriptdir=os.path.dirname(os.path.realpath(sys.argv_unicode[0]))
 
     # this script is installed as $prefix/bin. We just need $prefix to continue.
     #pwd=os.getcwd() # why????

