--- mypaint-1.1.0.orig/SConstruct	2012-12-31 18:03:38.000000000 +0100
+++ mypaint-1.1.0/SConstruct	2013-06-06 07:44:48.780203865 +0200
@@ -1,6 +1,7 @@
 import os, sys
 from os.path import join, basename
 from SCons.Script.SConscript import SConsEnvironment
+import SCons.Util
 
 EnsureSConsVersion(1, 0)
 
@@ -45,6 +46,27 @@
 if sys.platform == "win32":
     # remove this mingw if trying VisualStudio
     env = Environment(tools=tools + ['mingw'], ENV=os.environ, options=opts)
+
+# Respect some standard build environment stuff
+# See http://cgit.freedesktop.org/mesa/mesa/tree/scons/gallium.py
+# See https://wiki.gentoo.org/wiki/SCons#Missing_CC.2C_CFLAGS.2C_LDFLAGS
+if os.environ.has_key('CC'):
+   env['CC'] = os.environ['CC']
+if os.environ.has_key('CFLAGS'):
+   env['CCFLAGS'] += SCons.Util.CLVar(os.environ['CFLAGS'])
+if os.environ.has_key('CXX'):
+   env['CXX'] = os.environ['CXX']
+if os.environ.has_key('CXXFLAGS'):
+   env['CXXFLAGS'] += SCons.Util.CLVar(os.environ['CXXFLAGS'])
+if os.environ.has_key('CPPFLAGS'):
+   env['CCFLAGS'] += SCons.Util.CLVar(os.environ['CPPFLAGS'])
+   env['CXXFLAGS'] += SCons.Util.CLVar(os.environ['CPPFLAGS'])
+if os.environ.has_key('LDFLAGS'):
+    # LDFLAGS is omitted in SHLINKFLAGS, which is derived from LINKFLAGS
+   env['LINKFLAGS'] += SCons.Util.CLVar(os.environ['LDFLAGS'])
+if "$CCFLAGS" in env['CXXCOM']:
+   env['CXXCOM'] = env['CXXCOM'].replace("$CCFLAGS","")
+
 opts.Update(env)
 
 env.Append(CXXFLAGS=' -Wall -Wno-sign-compare -Wno-write-strings')

