From d4b3f22299339f4b54f0013b5f86eff48db1e8c4 Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint.reczey@canonical.com>
Date: Fri, 9 Mar 2018 11:19:24 +0100
Subject: [PATCH 4/4] dpkg: Enable zstd uniform compression

---
 dpkg-deb/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dpkg-deb/main.c b/dpkg-deb/main.c
index 7f898210e..7a40ecb80 100644
--- a/dpkg-deb/main.c
+++ b/dpkg-deb/main.c
@@ -245,6 +245,7 @@ int main(int argc, const char *const *argv) {
   if (opt_uniform_compression &&
       (compress_params.type != COMPRESSOR_TYPE_NONE &&
        compress_params.type != COMPRESSOR_TYPE_GZIP &&
+       compress_params.type != COMPRESSOR_TYPE_ZSTD &&
        compress_params.type != COMPRESSOR_TYPE_XZ))
     badusage(_("unsupported compression type '%s' with uniform compression"),
              compressor_get_name(compress_params.type));
-- 
2.15.1

