From a4ebc1215209f1628b77f875320aa768e4bfb4ba Mon Sep 17 00:00:00 2001
From: Yanko Kaneti <yaneti@declera.com>
Date: Wed, 7 Mar 2018 22:32:33 +0200
Subject: [PATCH] Fix date_format function prototype

---
 src/date.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/date.h b/src/date.h
index a47bafaa..bf991dfb 100644
--- a/src/date.h
+++ b/src/date.h
@@ -34,7 +34,7 @@
  *
  * @returns a newly allocated formatted date string (encoded in UTF-8)
  */
-gchar * date_format (time_t date, const gchar *date_format);
+gchar * date_format (gint64 date, const gchar *date_format);
 
 /**
  * Parses a ISO8601 date.
-- 
2.16.2

