diff -up QScintilla_gpl-2.10.3/Qt4Qt5/qsciscintillabase.cpp.qt4 QScintilla_gpl-2.10.3/Qt4Qt5/qsciscintillabase.cpp
--- QScintilla_gpl-2.10.3/Qt4Qt5/qsciscintillabase.cpp.qt4	2018-02-27 04:50:56.000000000 -0600
+++ QScintilla_gpl-2.10.3/Qt4Qt5/qsciscintillabase.cpp	2018-03-07 11:51:50.579557114 -0600
@@ -90,7 +90,7 @@ QsciScintillaBase::QsciScintillaBase(QWi
         , clickCausedFocus(false)
 #endif
 {
-#if !defined(QT_NO_ACCESSIBILITY)
+#if QT_VERSION >= 0x050000 && !defined(QT_NO_ACCESSIBILITY)
     QsciAccessibleScintillaBase::initialise();
 #endif
 
diff -up QScintilla_gpl-2.10.3/Qt4Qt5/SciAccessibility.cpp.qt4 QScintilla_gpl-2.10.3/Qt4Qt5/SciAccessibility.cpp
--- QScintilla_gpl-2.10.3/Qt4Qt5/SciAccessibility.cpp.qt4	2018-02-27 04:50:56.000000000 -0600
+++ QScintilla_gpl-2.10.3/Qt4Qt5/SciAccessibility.cpp	2018-03-07 11:52:18.119754724 -0600
@@ -20,7 +20,7 @@
 
 #include <qglobal.h>
 
-#if !defined(QT_NO_ACCESSIBILITY)
+#if QT_VERSION >= 0x050000 && !defined(QT_NO_ACCESSIBILITY)
 
 #include "SciAccessibility.h"
 
diff -up QScintilla_gpl-2.10.3/Qt4Qt5/SciAccessibility.h.qt4 QScintilla_gpl-2.10.3/Qt4Qt5/SciAccessibility.h
--- QScintilla_gpl-2.10.3/Qt4Qt5/SciAccessibility.h.qt4	2018-02-27 04:50:56.000000000 -0600
+++ QScintilla_gpl-2.10.3/Qt4Qt5/SciAccessibility.h	2018-03-07 11:50:12.151850861 -0600
@@ -23,7 +23,7 @@
 
 #include <qglobal.h>
 
-#if !defined(QT_NO_ACCESSIBILITY)
+#if QT_VERSION >= 0x050000 && !defined(QT_NO_ACCESSIBILITY)
 
 #include <QAccessible>
 #include <QAccessibleEditableTextInterface>
diff -up QScintilla_gpl-2.10.3/Qt4Qt5/ScintillaQt.cpp.qt4 QScintilla_gpl-2.10.3/Qt4Qt5/ScintillaQt.cpp
--- QScintilla_gpl-2.10.3/Qt4Qt5/ScintillaQt.cpp.qt4	2018-02-27 04:50:56.000000000 -0600
+++ QScintilla_gpl-2.10.3/Qt4Qt5/ScintillaQt.cpp	2018-03-07 11:51:07.429247496 -0600
@@ -31,7 +31,7 @@
 
 #include "Qsci/qsciscintillabase.h"
 #include "ScintillaQt.h"
-#if !defined(QT_NO_ACCESSIBILITY)
+#if QT_VERSION >= 0x050000 && !defined(QT_NO_ACCESSIBILITY)
 #include "SciAccessibility.h"
 #endif
 #include "SciClasses.h"
@@ -386,7 +386,7 @@ void QsciScintillaQt::NotifyParent(SCNot
         {
             char *text;
 
-#if !defined(QT_NO_ACCESSIBILITY)
+#if QT_VERSION >= 0x050000 && !defined(QT_NO_ACCESSIBILITY)
             if ((scn.modificationType & SC_MOD_INSERTTEXT) != 0)
                 QsciAccessibleScintillaBase::textInserted(qsb, scn.position,
                         scn.text, scn.length);
@@ -442,7 +442,7 @@ void QsciScintillaQt::NotifyParent(SCNot
         break;
 
     case SCN_UPDATEUI:
-#if !defined(QT_NO_ACCESSIBILITY)
+#if QT_VERSION >= 0x050000 && !defined(QT_NO_ACCESSIBILITY)
         QsciAccessibleScintillaBase::updated(qsb);
 #endif
         emit qsb->SCN_UPDATEUI(scn.updated);
@@ -582,7 +582,7 @@ void QsciScintillaQt::ClaimSelection()
     else
         primarySelection = false;
 
-#if !defined(QT_NO_ACCESSIBILITY)
+#if QT_VERSION >= 0x050000 && !defined(QT_NO_ACCESSIBILITY)
     QsciAccessibleScintillaBase::selectionChanged(qsb, isSel);
 #endif
 
