diff -Naur dracut/modules.d/90dmsquash-live/dmsquash-live-root.sh dracut.splash/modules.d/90dmsquash-live/dmsquash-live-root.sh
--- dracut/modules.d/90dmsquash-live/dmsquash-live-root.sh	2023-05-20 00:46:52.962579493 -0700
+++ dracut.splash/modules.d/90dmsquash-live/dmsquash-live-root.sh	2023-05-20 00:47:25.942142821 -0700
@@ -67,7 +67,6 @@
 fi
 getarg rd.live.check -d check || check=""
 if [ -n "$check" ]; then
-    type plymouth > /dev/null 2>&1 && plymouth --hide-splash
     if [ -n "$DRACUT_SYSTEMD" ]; then
         p=$(dev_unit_name "$check_dev")
         systemctl start checkisomd5@"${p}".service
@@ -75,10 +74,10 @@
         checkisomd5 --verbose "$check_dev"
     fi
     if [ $? -eq 1 ]; then
+        type plymouth > /dev/null 2>&1 && plymouth --hide-splash
         die "CD check failed!"
         exit 1
     fi
-    type plymouth > /dev/null 2>&1 && plymouth --show-splash
 fi
 
 ln -s "$livedev" /run/initramfs/livedev
