From 2d4fc8d7e261deb290ae696acf0c79dd04743a04 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Mon, 22 Jan 2024 16:47:15 +0800
Subject: [PATCH] example: use AOSC OS's own NTP Pool vendor zone

---
 examples/chrony.conf.example3 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/examples/chrony.conf.example3 b/examples/chrony.conf.example3
index 6d84c01..4295429 100644
--- a/examples/chrony.conf.example3
+++ b/examples/chrony.conf.example3
@@ -31,7 +31,7 @@
 ! server ntp2.example.net iburst
 ! server ntp3.example.net iburst
 
-! pool pool.ntp.org iburst
+! pool aoscos.pool.ntp.org iburst
  
 #######################################################################
 ### AVOIDING POTENTIALLY BOGUS CHANGES TO YOUR CLOCK
-- 
2.43.0.windows.1

