--- a/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h	2023-11-28 16:52:28.000000000 +0800
+++ b/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h	2024-01-27 20:06:39.141815271 +0800
@@ -32,7 +32,12 @@
   // given thread.
   static std::unique_ptr<NativeRegisterContextLinux>
   CreateHostNativeRegisterContextLinux(const ArchSpec &target_arch,
-                                       NativeThreadLinux &native_thread);
+                                       NativeThreadLinux &native_thread)
+#ifdef __mips64
+  { llvm_unreachable("mips64 unsupported"); }
+#else
+  ;
+#endif
 
   // Determine the architecture of the thread given by its ID.
   static llvm::Expected<ArchSpec> DetermineArchitecture(lldb::tid_t tid);
