diff --git a/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h b/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
index b66f6907d8ae..e18011855149 100644
--- a/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
+++ b/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
@@ -28,7 +28,12 @@ public:
   // variant should be compiled into the final executable.
   static std::unique_ptr<NativeRegisterContextLinux>
   CreateHostNativeRegisterContextLinux(const ArchSpec &target_arch,
-                                       NativeThreadLinux &native_thread);
+                                       NativeThreadLinux &native_thread)
+#ifdef __mips64
+  { llvm_unreachable("mips64 unsupported"); }
+#else
+  ;
+#endif
 
   // Invalidates cached values in register context data structures
   virtual void InvalidateAllRegisters(){}
