From 98a38ba8d9c3146b55c31f8ba320d68b227ccb64 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Sun, 28 Jan 2024 04:49:26 -0800
Subject: [PATCH 3/9] [mips] Add stub for mips64 LLDB impl

---
 .../Plugins/Process/Linux/NativeRegisterContextLinux.h     | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h b/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
index 08f19d374e28..a042dd3dd677 100644
--- a/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
+++ b/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
@@ -32,7 +32,12 @@ public:
   // given thread.
   static std::unique_ptr<NativeRegisterContextLinux>
   CreateHostNativeRegisterContextLinux(const ArchSpec &target_arch,
-                                       NativeThreadLinux &native_thread);
+                                       NativeThreadLinux &native_thread)
+#ifdef __mips64
+  { llvm_unreachable("mips64 unsupported"); }
+#else
+  ;
+#endif
 
   // Determine the architecture of the thread given by its ID.
   static llvm::Expected<ArchSpec> DetermineArchitecture(lldb::tid_t tid);
-- 
2.45.2

