From 8070451190bf8b440b09ad1d384e2743f8fc5297 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Sun, 28 Jan 2024 04:52:51 -0800
Subject: [PATCH 4/9] [mips] Add stub for lldb arch functions on loongson3

---
 .../Plugins/Process/Linux/NativeRegisterContextLinux.h    | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h b/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
index a042dd3dd677..ee4901aa7cd3 100644
--- a/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
+++ b/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
@@ -40,7 +40,15 @@ public:
 #endif
 
   // Determine the architecture of the thread given by its ID.
+#ifdef __mips64
+  static llvm::Expected<ArchSpec> DetermineArchitecture(lldb::tid_t tid) {
+    return llvm::createStringError(
+        llvm::inconvertibleErrorCode(),
+        "Architecture does not support this function");
+  }
+#else
   static llvm::Expected<ArchSpec> DetermineArchitecture(lldb::tid_t tid);
+#endif
 
   // Invalidates cached values in register context data structures
   virtual void InvalidateAllRegisters(){}
-- 
2.45.2

