From eb271125e8f6b5186a31e2a5a17ac3d4beb51d62 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Sun, 28 Jan 2024 04:53:18 -0800
Subject: [PATCH 5/9] [mips] HACK: force enable cacheflush function

---
 compiler-rt/lib/builtins/clear_cache.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/compiler-rt/lib/builtins/clear_cache.c b/compiler-rt/lib/builtins/clear_cache.c
index 2ac99b25c243..f055d3fcee4d 100644
--- a/compiler-rt/lib/builtins/clear_cache.c
+++ b/compiler-rt/lib/builtins/clear_cache.c
@@ -43,6 +43,7 @@ uintptr_t GetCurrentProcess(void);
 #endif
 
 #if defined(__linux__) && defined(__mips__)
+#define __USE_MISC
 #include <sys/cachectl.h>
 #include <sys/syscall.h>
 #include <unistd.h>
-- 
2.45.2

