From 783da8c5897b82d44677cdf8ebb4f73333b2bad9 Mon Sep 17 00:00:00 2001
From: Nikita Popov <npopov@redhat.com>
Date: Thu, 10 Feb 2022 09:59:11 +0100
Subject: [PATCH 1005/1013] [rust] Work around ICE in GCC 5.4

---
 lld/ELF/SyntheticSections.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lld/ELF/SyntheticSections.cpp b/lld/ELF/SyntheticSections.cpp
index b359c2e7bcea..5abf0a979b98 100644
--- a/lld/ELF/SyntheticSections.cpp
+++ b/lld/ELF/SyntheticSections.cpp
@@ -1662,8 +1662,8 @@ void RelocationBaseSection::partitionRels() {
   if (!combreloc)
     return;
   const RelType relativeRel = target->relativeRel;
-  numRelativeRelocs =
-      llvm::partition(relocs, [=](auto &r) { return r.type == relativeRel; }) -
+  numRelativeRelocs = llvm::partition(
+      relocs, [=](const DynamicReloc &r) { return r.type == relativeRel; }) -
       relocs.begin();
 }
 
-- 
2.40.0

