diff --git a/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h b/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
index 08f19d374e28..95424b6f8cc3 100644
--- a/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
+++ b/lldb/source/Plugins/Process/Linux/NativeRegisterContextLinux.h
@@ -35,7 +35,15 @@ public:
                                        NativeThreadLinux &native_thread);
 
   // Determine the architecture of the thread given by its ID.
+#ifdef __mips64
+  static llvm::Expected<ArchSpec> DetermineArchitecture(lldb::tid_t tid) {
+    return llvm::createStringError(
+        llvm::inconvertibleErrorCode(),
+        "Architecture does not support this function");
+  }
+#else
   static llvm::Expected<ArchSpec> DetermineArchitecture(lldb::tid_t tid);
+#endif
 
   // Invalidates cached values in register context data structures
   virtual void InvalidateAllRegisters(){}
