diff --git a/compiler-rt/lib/builtins/clear_cache.c b/compiler-rt/lib/builtins/clear_cache.c
index 8993761eb3d4..fee877630531 100644
--- a/compiler-rt/lib/builtins/clear_cache.c
+++ b/compiler-rt/lib/builtins/clear_cache.c
@@ -43,6 +43,7 @@ uintptr_t GetCurrentProcess(void);
 #endif
 
 #if defined(__linux__) && defined(__mips__)
+#define __USE_MISC
 #include <sys/cachectl.h>
 #include <sys/syscall.h>
 #include <unistd.h>
