diff -Naur fontconfig-2.14.0/conf.d/60-latin.conf fontconfig-2.14.0.noto/conf.d/60-latin.conf
--- fontconfig-2.14.0/conf.d/60-latin.conf	2021-12-03 10:48:08.000000000 +0000
+++ fontconfig-2.14.0.noto/conf.d/60-latin.conf	2022-09-13 20:38:31.019489102 +0000
@@ -35,6 +35,7 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
+			<family>Source Code Pro</family>
 			<family>Noto Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
 			<family>Inconsolata</family>
@@ -77,8 +78,8 @@
 	<alias>
 		<family>system-ui</family>
 		<prefer>
-			<family>Cantarell</family>
 			<family>Noto Sans UI</family>
+			<family>Cantarell</family>
 			<family>Segoe UI</family>
 			<family>Segoe UI Historic</family>
 			<family>Segoe UI Symbol</family>
