From 1eead504cff6d88f12ec8508040276cf3018305b Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 22 May 2024 23:15:43 +0800
Subject: [PATCH 3/3] fix(conf.d): disable non-emoji bitmap fonts by default

---
 conf.d/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/conf.d/meson.build b/conf.d/meson.build
index b6d07ee..771fee9 100644
--- a/conf.d/meson.build
+++ b/conf.d/meson.build
@@ -66,6 +66,7 @@ conf_links = [
   '65-fonts-persian.conf',
   '65-nonlatin.conf',
   '69-unifont.conf',
+  '70-no-bitmaps-except-emoji.conf',
   '80-delicious.conf',
   '90-synthetic.conf',
 ]
-- 
2.46.0

