diff --git a/apps/zxtune-qt/ui/desktop/mainwindow.cpp b/apps/zxtune-qt/ui/desktop/mainwindow.cpp
index 0260bb276..654f93b0b 100644
--- a/apps/zxtune-qt/ui/desktop/mainwindow.cpp
+++ b/apps/zxtune-qt/ui/desktop/mainwindow.cpp
@@ -106,15 +106,7 @@ namespace
       Require(connect(actionReportBug, SIGNAL(triggered()), SLOT(ReportIssue())));
       Require(connect(actionAboutQt, SIGNAL(triggered()), SLOT(ShowAboutQt())));
       Require(connect(actionPreferences, SIGNAL(triggered()), SLOT(ShowPreferences())));
-      if (Update::CheckOperation* op = Update::CheckOperation::Create(*this))
-      {
-        Require(op->connect(actionCheckUpdates, SIGNAL(triggered()), SLOT(Execute())));
-        Require(connect(op, SIGNAL(ErrorOccurred(const Error&)), SLOT(ShowError(const Error&))));
-      }
-      else
-      {
-        actionCheckUpdates->setEnabled(false);
-      }
+      actionCheckUpdates->setEnabled(false);
 
       Require(MultiPlaylist->connect(Controls, SIGNAL(OnPrevious()), SLOT(Prev())));
       Require(MultiPlaylist->connect(Controls, SIGNAL(OnNext()), SLOT(Next())));
diff --git a/apps/zxtune-qt/Makefile b/apps/zxtune-qt/Makefile
index 36fd6c41c..a07d17811 100644
--- a/apps/zxtune-qt/Makefile
+++ b/apps/zxtune-qt/Makefile
@@ -41,8 +40,8 @@ libraries.boost = filesystem system
 libraries.qt = Gui Core Network
 
 libraries.common += l10n debug
-libraries += ui_desktop playlist_ui_desktop ui_conversion ui_preferences check_update
-depends += $(addprefix apps/zxtune-qt/,ui/desktop playlist/ui/desktop ui/conversion ui/preferences update)
+libraries += ui_desktop playlist_ui_desktop ui_conversion ui_preferences
+depends += $(addprefix apps/zxtune-qt/,ui/desktop playlist/ui/desktop ui/conversion ui/preferences)
 libraries.boost += locale
 libraries.mingw += iconv
 libraries.darwin += iconv

