diff -Naur a/lookandfeel/contents/lockscreen/LockScreen.qml b/lookandfeel/contents/lockscreen/LockScreen.qml
--- a/lookandfeel/contents/lockscreen/LockScreen.qml	2016-09-29 06:35:01.000000000 -0500
+++ b/lookandfeel/contents/lockscreen/LockScreen.qml	2016-10-08 11:42:58.426396249 -0500
@@ -23,7 +23,7 @@
 import org.kde.plasma.private.sessions 2.0
 import "../components"
 
-Item {
+Image {
     id: root
     property bool viewVisible: false
     property bool debug: false
@@ -31,6 +31,10 @@
     property int interfaceVersion: org_kde_plasma_screenlocker_greeter_interfaceVersion ? org_kde_plasma_screenlocker_greeter_interfaceVersion : 0
     signal clearPassword()
 
+    source: "/usr/share/plasma/components/artwork/background.png"
+    fillMode: Image.PreserveAspectCrop
+    asynchronous: false
+
     LayoutMirroring.enabled: Qt.application.layoutDirection === Qt.RightToLeft
     LayoutMirroring.childrenInherit: true
 
diff -Naur a/lookandfeel/contents/splash/Splash.qml b/lookandfeel/contents/splash/Splash.qml
--- a/lookandfeel/contents/splash/Splash.qml	2016-09-29 06:35:01.000000000 -0500
+++ b/lookandfeel/contents/splash/Splash.qml	2016-10-08 11:38:17.880440481 -0500
@@ -19,9 +19,11 @@
 
 import QtQuick 2.5
 
-Rectangle {
+Image {
     id: root
-    color: "black"
+
+    source: "/usr/share/plasma/components/artwork/background.png"
+    fillMode: Image.PreserveAspectCrop
 
     property int stage
 
diff -Naur a/sddm-theme/theme.conf b/sddm-theme/theme.conf
--- a/sddm-theme/theme.conf	2016-09-29 06:35:01.000000000 -0500
+++ b/sddm-theme/theme.conf	2016-10-08 11:38:02.893765830 -0500
@@ -1,2 +1,2 @@
 [General]
-background=
+background=/usr/share/plasma/components/artwork/background.png
