diff -Naur plasma-workspace-5.9.0/lookandfeel/contents/splash/Splash.qml plasma-workspace-5.9.0.aosc/lookandfeel/contents/splash/Splash.qml
--- plasma-workspace-5.9.0/lookandfeel/contents/splash/Splash.qml	2017-01-26 06:28:32.000000000 -0600
+++ plasma-workspace-5.9.0.aosc/lookandfeel/contents/splash/Splash.qml	2017-02-02 21:19:58.077144869 -0600
@@ -19,9 +19,11 @@
 
 import QtQuick 2.5
 
-Rectangle {
+Image {
     id: root
-    color: "black"
+
+    source: "/usr/share/plasma/components/artwork/background.png"
+    fillMode: Image.PreserveAspectCrop
 
     property int stage
 
diff -Naur plasma-workspace-5.9.0/startkde/startkde.cmake plasma-workspace-5.9.0.aosc/startkde/startkde.cmake
--- plasma-workspace-5.9.0/startkde/startkde.cmake	2017-01-26 06:28:32.000000000 -0600
+++ plasma-workspace-5.9.0.aosc/startkde/startkde.cmake	2017-02-02 21:23:32.832387940 -0600
@@ -75,7 +75,7 @@
 [General]
 XftAntialias=true
 XftHintStyle=hintmedium
-XftSubPixel=none
+XftSubPixel=rgb
 EOF
 }
 
