diff -Naur plasma-workspace-5.9.2/startkde/startkde.cmake plasma-workspace-5.9.2.aosc/startkde/startkde.cmake
--- plasma-workspace-5.9.2/startkde/startkde.cmake	2017-02-14 09:56:43.000000000 -0600
+++ plasma-workspace-5.9.2.aosc/startkde/startkde.cmake	2017-02-19 21:33:07.975932626 -0600
@@ -75,7 +75,7 @@
 [General]
 XftAntialias=true
 XftHintStyle=hintmedium
-XftSubPixel=none
+XftSubPixel=rgb
 EOF
 }
 
