diff -Naur plasma-workspace-5.9.2/lookandfeel/contents/splash/Splash.qml plasma-workspace-5.9.2.aosc/lookandfeel/contents/splash/Splash.qml
--- plasma-workspace-5.9.2/lookandfeel/contents/splash/Splash.qml	2017-02-14 09:56:43.000000000 -0600
+++ plasma-workspace-5.9.2.aosc/lookandfeel/contents/splash/Splash.qml	2017-02-19 21:33:07.975932626 -0600
@@ -19,9 +19,11 @@
 
 import QtQuick 2.5
 
-Rectangle {
+Image {
     id: root
-    color: "black"
+
+    source: "/usr/share/backgrounds/core5/Plain-5120x2880.png"
+    fillMode: Image.PreserveAspectCrop
 
     property int stage
 
diff -Naur plasma-workspace-5.9.2/sddm-theme/theme.conf plasma-workspace-5.9.2.aosc/sddm-theme/theme.conf
--- plasma-workspace-5.9.2/sddm-theme/theme.conf	2017-02-14 09:56:43.000000000 -0600
+++ plasma-workspace-5.9.2.aosc/sddm-theme/theme.conf	2017-02-19 21:33:14.259508810 -0600
@@ -1,4 +1,4 @@
 [General]
-type=color
+type=image
 color=#1d99f3
-background=
+background=/usr/share/backgrounds/core5/Plain-5120x2880.png
diff -Naur plasma-workspace-5.9.2/startkde/startkde.cmake plasma-workspace-5.9.2.aosc/startkde/startkde.cmake
--- plasma-workspace-5.9.2/startkde/startkde.cmake	2017-02-14 09:56:43.000000000 -0600
+++ plasma-workspace-5.9.2.aosc/startkde/startkde.cmake	2017-02-19 21:33:07.975932626 -0600
@@ -75,7 +75,7 @@
 [General]
 XftAntialias=true
 XftHintStyle=hintmedium
-XftSubPixel=none
+XftSubPixel=rgb
 EOF
 }
 
