diff -Naur plasma-workspace-5.12.4/startkde/startkde.cmake plasma-workspace-5.12.4.qtpath/startkde/startkde.cmake
--- plasma-workspace-5.12.4/startkde/startkde.cmake	2018-03-27 11:26:39.000000000 +0000
+++ plasma-workspace-5.12.4.qtpath/startkde/startkde.cmake	2018-04-14 04:24:01.814604130 +0000
@@ -294,7 +294,7 @@
   exit 1
 fi
 
-qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit &
+/usr/lib/qt5/bin/qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit &
 
 # finally, give the session control to the session manager
 # see kdebase/ksmserver for the description of the rest of the startup sequence
@@ -327,13 +327,13 @@
     # wait for remaining drkonqi instances with timeout (in seconds)
     wait_drkonqi_timeout=`kreadconfig5 --file startkderc --group WaitForDrKonqi --key Timeout --default 900`
     wait_drkonqi_counter=0
-    while qdbus | grep "^[^w]*org.kde.drkonqi" > /dev/null ; do
+    while /usr/lib/qt5/bin/qdbus | grep "^[^w]*org.kde.drkonqi" > /dev/null ; do
         sleep 5
         wait_drkonqi_counter=$((wait_drkonqi_counter+5))
         if test "$wait_drkonqi_counter" -ge "$wait_drkonqi_timeout" ; then
             # ask remaining drkonqis to die in a graceful way
-            qdbus | grep 'org.kde.drkonqi-' | while read address ; do
-                qdbus "$address" "/MainApplication" "quit"
+            /usr/lib/qt5/bin/qdbus | grep 'org.kde.drkonqi-' | while read address ; do
+                /usr/lib/qt5/bin/qdbus "$address" "/MainApplication" "quit"
             done
             break
         fi
