--- kde-workspace-4.11.20/startkde.cmake	2015-05-29 01:30:26.000000000 +0000
+++ kde-workspace-4.11.20-patched/startkde.cmake	2015-06-06 14:18:15.367994151 +0000
@@ -3,6 +3,12 @@
 #  DEFAULT KDE STARTUP SCRIPT ( @KDE4WORKSPACE_VERSION@ )
 #
 
+if [ -e /etc/locale.conf ]; then
+   export $(cat /etc/locale.conf)
+else
+   echo "Locale not set, it is all up to KDE."
+fi
+
 if test "x$1" = x--failsafe; then
     KDE_FAILSAFE=1 # General failsafe flag
     KWIN_COMPOSE=N # Disable KWin's compositing
