From 459a6fbc3b5d0d6a66b658b9939829223740b902 Mon Sep 17 00:00:00 2001
From: MutsukiC <mutsuki@aosc.io>
Date: Thu, 16 Apr 2026 10:13:42 +0800
Subject: [PATCH 1/3] Remove helper installation from CMakeLists

---
 linux/CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/linux/CMakeLists.txt b/linux/CMakeLists.txt
index 04b44a85..49fedd82 100644
--- a/linux/CMakeLists.txt
+++ b/linux/CMakeLists.txt
@@ -123,9 +123,6 @@ if(NOT CMAKE_BUILD_TYPE MATCHES "Debug")
     COMPONENT Runtime)
 endif()
 
-# Copy the Helper
-install(DIRECTORY "../build/linux/helper" DESTINATION "${BUILD_BUNDLE_DIR}" USE_SOURCE_PERMISSIONS)
-
 # Copy Linux support files
 install(DIRECTORY "../resources/linux/linux_support" DESTINATION "${BUILD_BUNDLE_DIR}" USE_SOURCE_PERMISSIONS)
 install(FILES "../resources/linux/README.adoc" DESTINATION "${BUILD_BUNDLE_DIR}")
-- 
2.52.0

