From a7b6d2275fa0273764d841fcc1e1176542270b88 Mon Sep 17 00:00:00 2001
From: RyogiMutsuki <me@mutsuki.ink>
Date: Sun, 31 Aug 2025 16:08:03 +0800
Subject: [PATCH 1/2] remove helper installation from CMakeLists

---
 linux/CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/linux/CMakeLists.txt b/linux/CMakeLists.txt
index 9e1670b6..2dd260e1 100644
--- a/linux/CMakeLists.txt
+++ b/linux/CMakeLists.txt
@@ -122,9 +122,6 @@ if(NOT CMAKE_BUILD_TYPE MATCHES "Debug")
     COMPONENT Runtime)
 endif()
 
-# Copy the Helper
-install(DIRECTORY "../build/linux/helper" DESTINATION "${BUILD_BUNDLE_DIR}" USE_SOURCE_PERMISSIONS)
-
 # Copy Linux support files
 install(DIRECTORY "../resources/linux/linux_support" DESTINATION "${BUILD_BUNDLE_DIR}" USE_SOURCE_PERMISSIONS)
 install(FILES "../resources/linux/README.adoc" DESTINATION "${BUILD_BUNDLE_DIR}")
-- 
2.51.0

