Index: rarian-0.8.1/librarian/rarian-language.c
===================================================================
--- rarian-0.8.1.orig/librarian/rarian-language.c	2010-07-04 17:24:46.881069651 +0200
+++ rarian-0.8.1/librarian/rarian-language.c	2010-07-04 17:25:32.305067458 +0200
@@ -89,6 +89,9 @@ rrn_language_init (char *lang)
       loc = getenv ("LC_ALL");
     }
     if (!loc || !strcmp (loc, "")) {
+      loc = getenv ("LC_MESSAGES");
+    }
+    if (!loc || !strcmp (loc, "")) {
       loc = getenv ("LANG");
     }
     
