Index: pyside-setup-opensource-src-5.15.2/build_scripts/options.py
===================================================================
--- pyside-setup-opensource-src-5.15.2.orig/build_scripts/options.py
+++ pyside-setup-opensource-src-5.15.2/build_scripts/options.py
@@ -167,6 +167,7 @@ OPTION = {
     # Legacy, not used any more.
     "JOM": has_option('jom'),
     "MACOS_USE_LIBCPP": has_option("macos-use-libc++"),
+    "REUSE_BUILD": has_option("reuse-build"),
     "QUIET": has_option('quiet', remove=False),
     "SNAPSHOT_BUILD": has_option("snapshot-build"),
     "LIMITED_API": option_value("limited-api"),
