=== modified file 'gst/gstregistry.c'
Index: trunk/gst/gstregistry.c
===================================================================
--- trunk.orig/gst/gstregistry.c
+++ trunk/gst/gstregistry.c
@@ -1503,7 +1503,7 @@
    * path, and the plugins installed in the user's home directory */
   plugin_path = g_getenv ("GST_PLUGIN_SYSTEM_PATH");
   if (plugin_path == NULL) {
-    char *home_plugins;
+    char *home_plugins, *backwards_compat_plugins;
 
     GST_DEBUG ("GST_PLUGIN_SYSTEM_PATH not set");
 
@@ -1519,6 +1519,13 @@
     GST_DEBUG ("scanning main plugins %s", PLUGINDIR);
     changed |= gst_registry_scan_path_internal (&context, PLUGINDIR);
 
+    /* finally, look in the /usr/lib backwards-compatibility path. */
+    backwards_compat_plugins = g_build_filename ("/usr/lib",
+        "gstreamer-" GST_MAJORMINOR, NULL);
+    GST_DEBUG ("scanning plugins %s", backwards_compat_plugins);
+    changed |= gst_registry_scan_path_internal (&context, backwards_compat_plugins);
+    g_free (backwards_compat_plugins);
+
 #ifdef G_OS_WIN32
     {
       char *base_dir;
