diff -up dbus-sharp-0.8.1/src/dbus-sharp.csproj.fixframework dbus-sharp-0.8.1/src/dbus-sharp.csproj
--- dbus-sharp-0.8.1/src/dbus-sharp.csproj.fixframework	2014-06-23 23:26:42.000000000 -0400
+++ dbus-sharp-0.8.1/src/dbus-sharp.csproj	2017-02-27 10:50:36.349148867 -0500
@@ -13,7 +13,7 @@
     <RootNamespace>DBus</RootNamespace>
     <SignAssembly>true</SignAssembly>
     <AssemblyOriginatorKeyFile>..\dbus-sharp.snk</AssemblyOriginatorKeyFile>
-    <TargetFrameworkVersion>v3.5</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <DebugSymbols>True</DebugSymbols>
