diff -up dbus-sharp-0.8.1/src/Connection.cs.nocheckexists dbus-sharp-0.8.1/src/Connection.cs
--- dbus-sharp-0.8.1/src/Connection.cs.nocheckexists	2017-02-27 12:51:27.263165005 -0500
+++ dbus-sharp-0.8.1/src/Connection.cs	2017-02-27 12:51:42.732757110 -0500
@@ -399,9 +399,6 @@ namespace DBus
 
 		public object GetObject (Type type, string bus_name, ObjectPath path)
 		{
-			if (!CheckBusNameExists (bus_name))
-				return null;
-
 			//if the requested type is an interface, we can implement it efficiently
 			//otherwise we fall back to using a transparent proxy
 			if (type.IsInterface || type.IsAbstract) {
