diff -up dbus-sharp-0.8.1/src/Protocol/MessageWriter.cs.fixarray dbus-sharp-0.8.1/src/Protocol/MessageWriter.cs
--- dbus-sharp-0.8.1/src/Protocol/MessageWriter.cs.fixarray	2017-02-27 12:48:19.916104885 -0500
+++ dbus-sharp-0.8.1/src/Protocol/MessageWriter.cs	2017-02-27 12:48:56.738133979 -0500
@@ -341,7 +341,12 @@ namespace DBus.Protocol
 
 			Type type = val.GetType ();
 
-			WriteVariant (type, val);
+			if (type.IsArray) {
+				Write(type, val);
+			}
+			else{
+				WriteVariant (type, val);
+			}
 		}
 
 		public void WriteVariant (Type type, object val)
