diff -aur dssi-vst-0.9.2/Makefile dssi-vst-0.9.2.patched/Makefile
--- dssi-vst-0.9.2/Makefile
+++ dssi-vst-0.9.2.patched/Makefile
@@ -1,6 +1,6 @@
 
-DSSIDIR		= /usr/local/lib/dssi
-LADSPADIR	= /usr/local/lib/ladspa
-BINDIR		= /usr/local/bin
+DSSIDIR		= $(DESTDIR)/usr/lib/dssi
+LADSPADIR	= $(DESTDIR)/usr/lib/lasspa
+BINDIR		= $(DESTDIR)/usr/bin

 # To compile with the VeSTige compatibility header:
