From 6a3669fabe6b796e1fbd2aa61cf4de10273b6f5a Mon Sep 17 00:00:00 2001
From: Seren541 <gholmann16@gmail.com>
Date: Mon, 17 Jul 2023 18:58:55 -0400
Subject: [PATCH 2/3] Necessary to compile on Ubuntu 14

---
 src/libappimage/utils/IconHandleCairoRsvg.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libappimage/utils/IconHandleCairoRsvg.cpp b/src/libappimage/utils/IconHandleCairoRsvg.cpp
index 8a20ef5..20b5979 100644
--- a/src/libappimage/utils/IconHandleCairoRsvg.cpp
+++ b/src/libappimage/utils/IconHandleCairoRsvg.cpp
@@ -1,6 +1,7 @@
 // libraries
 #include <glib-object.h>
 #include <fstream>
+#include <cstring>
 
 // local
 #include "IconHandle.h"
-- 
2.42.0

