From: Stef Walter <stefw@gnome.org>
Subject: data: Fix desktop-centric polkit policy.
Origin: http://cgit.freedesktop.org/accountsservice/commit/?id=74aa92e3f30deb455e1fe9b543ea8a8b12f82d8b
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=78279
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1349813


diff -ruN a/data/org.freedesktop.accounts.policy.in b/data/org.freedesktop.accounts.policy.in
--- a/data/org.freedesktop.accounts.policy.in	2014-03-20 23:47:55.000000000 +0500
+++ b/data/org.freedesktop.accounts.policy.in	2014-07-30 17:58:04.487538363 +0500
@@ -11,8 +11,8 @@
     <_description>Change your own user data</_description>
     <_message>Authentication is required to change your own user data</_message>
     <defaults>
-      <allow_any>no</allow_any>
-      <allow_inactive>no</allow_inactive>
+      <allow_any>auth_self</allow_any>
+      <allow_inactive>auth_self</allow_inactive>
       <allow_active>yes</allow_active>
     </defaults>
   </action>
@@ -21,8 +21,8 @@
     <_description>Manage user accounts</_description>
     <_message>Authentication is required to change user data</_message>
     <defaults>
-      <allow_any>no</allow_any>
-      <allow_inactive>no</allow_inactive>
+      <allow_any>auth_admin</allow_any>
+      <allow_inactive>auth_admin</allow_inactive>
       <allow_active>auth_admin_keep</allow_active>
     </defaults>
   </action>
@@ -31,8 +31,8 @@
     <_description>Change the login screen configuration</_description>
     <_message>Authentication is required to change the login screen configuration</_message>
     <defaults>
-      <allow_any>no</allow_any>
-      <allow_inactive>no</allow_inactive>
+      <allow_any>auth_admin</allow_any>
+      <allow_inactive>auth_admin</allow_inactive>
       <allow_active>auth_admin_keep</allow_active>
     </defaults>
     <annotate key="org.gnome.gconf.defaults.set-mandatory.prefix">/apps/gdm/simple-greeter</annotate>
