From ca76d291af2234bd6ca561485b7b0629bfaedb16 Mon Sep 17 00:00:00 2001
From: "mingang.he" <dustgle@gmail.com>
Date: Tue, 19 Oct 2021 20:53:33 +0800
Subject: [PATCH 2/5] Fix Makefile

---
 local/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/local/Makefile b/local/Makefile
index 57d2b4e..c295ece 100644
--- a/local/Makefile
+++ b/local/Makefile
@@ -2,7 +2,7 @@ ifneq ($(shell echo $(VERSION) | head -c 1), v)
 	VERSION=v0.4
 endif
 
-ifneq (,$(findstring arm,$(CC)))
+ifeq (,$(findstring arm,$(CC)))
 	GO_BUILD_PREFIX_FLAGS :=
 	CGO_FLAGS :=
 else
-- 
2.36.1

