From 4386d8f83cbbbc9d47e0970f5db09dc95c14cec3 Mon Sep 17 00:00:00 2001
From: "mingang.he" <dustgle@gmail.com>
Date: Sat, 27 Nov 2021 14:24:31 +0000
Subject: [PATCH 4/5] Update Makefile

---
 local/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/local/Makefile b/local/Makefile
index 50e0d42..2f8e7cc 100644
--- a/local/Makefile
+++ b/local/Makefile
@@ -25,7 +25,7 @@ graftcp-local: $(wildcard *.go cmd/graftcp-local/*.go)
 	${GO_BUILD_PREFIX_FLAGS} go build -ldflags "-s -w -X main.version=${VERSION}" ./cmd/graftcp-local
 
 mgraftcp: $(wildcard *.go cmd/mgraftcp/*.go) ../libgraftcp.a
-	${GO_BUILD_PREFIX_FLAGS} ${CGO_FLAGS} go build -ldflags "-s -w -X main.version=${VERSION}" ./cmd/mgraftcp
+	${GO_BUILD_PREFIX_FLAGS} ${CGO_FLAGS} go build -ldflags "-s -w -X main.version=${VERSION}" -ldflags '-linkmode "external" -extldflags "-static"' ./cmd/mgraftcp
 
 ../libgraftcp.a:
 	$(MAKE) -C .. libgraftcp.a
-- 
2.36.1

