From d3cc5395d217d383696a8aeb2192e36dfdd3d56d Mon Sep 17 00:00:00 2001
From: ilikara <3435193369@qq.com>
Date: Fri, 19 Sep 2025 20:53:38 +0800
Subject: [PATCH] AOSCOS: turn autotools to unselected version

Signed-off-by: ilikara <3435193369@qq.com>
---
 autogen.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/autogen.sh b/autogen.sh
index ce41821..e4154c8 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -7,9 +7,9 @@
 # tools and you shouldn't use this script. Just call ./configure
 # directly.
 
-ACLOCAL=${ACLOCAL-aclocal-1.16}
+ACLOCAL=${ACLOCAL-aclocal}
 AUTOCONF=${AUTOCONF-autoconf}
-AUTOMAKE=${AUTOMAKE-automake-1.16}
+AUTOMAKE=${AUTOMAKE-automake}
 
 AUTOCONF_REQUIRED_VERSION=2.62
 AUTOMAKE_REQUIRED_VERSION=1.13
-- 
2.51.0

