Author: Aaron M. Ucko
Last-Update: 2023-01-15 20:56:15
Description: link vdb dynamically

Index: b/build/env.cmake
===================================================================
--- a/build/env.cmake
+++ b/build/env.cmake
@@ -644,13 +644,13 @@ endif()
 
 if( NOT SINGLE_CONFIG )
     set( COMMON_LINK_LIBRARIES kapp tk-version )
-    set( COMMON_LIBS_READ  $<$<CONFIG:Debug>:${NCBI_VDB_LIBDIR_DEBUG}>$<$<CONFIG:Release>:${NCBI_VDB_LIBDIR_RELEASE}>/${LIBPFX}ncbi-vdb.${STLX} ${MBEDTLS_LIBS} )
-    set( COMMON_LIBS_WRITE $<$<CONFIG:Debug>:${NCBI_VDB_LIBDIR_DEBUG}>$<$<CONFIG:Release>:${NCBI_VDB_LIBDIR_RELEASE}>/${LIBPFX}ncbi-wvdb.${STLX} ${MBEDTLS_LIBS} )
+    set( COMMON_LIBS_READ  $<$<CONFIG:Debug>:${NCBI_VDB_LIBDIR_DEBUG}>$<$<CONFIG:Release>:${NCBI_VDB_LIBDIR_RELEASE}>/${LIBPFX}ncbi-vdb.${SHLX} ${MBEDTLS_LIBS} )
+    set( COMMON_LIBS_WRITE $<$<CONFIG:Debug>:${NCBI_VDB_LIBDIR_DEBUG}>$<$<CONFIG:Release>:${NCBI_VDB_LIBDIR_RELEASE}>/${LIBPFX}ncbi-wvdb.${SHLX} ${MBEDTLS_LIBS} )
 else()
     # single-config generators need full path to ncbi-vdb libraries in order to handle the dependency correctly
     set( COMMON_LINK_LIBRARIES ${NCBI_VDB_LIBDIR}/libkapp.${STLX} tk-version )
-    set( COMMON_LIBS_READ   ${NCBI_VDB_LIBDIR}/libncbi-vdb.${STLX} pthread dl m ${MBEDTLS_LIBS} )
-    set( COMMON_LIBS_WRITE  ${NCBI_VDB_LIBDIR}/libncbi-wvdb.${STLX} pthread dl m ${MBEDTLS_LIBS} )
+    set( COMMON_LIBS_READ   ${NCBI_VDB_LIBDIR}/libncbi-vdb.${SHLX} pthread dl m ${MBEDTLS_LIBS} )
+    set( COMMON_LIBS_WRITE  ${NCBI_VDB_LIBDIR}/libncbi-wvdb.${SHLX} pthread dl m ${MBEDTLS_LIBS} )
 endif()
 
 if( WIN32 )
Index: b/tools/loaders/general-loader/test/CMakeLists.txt
===================================================================
--- a/tools/loaders/general-loader/test/CMakeLists.txt
+++ b/tools/loaders/general-loader/test/CMakeLists.txt
@@ -66,7 +66,7 @@ add_executable ( makeinputs
 )
 target_link_libraries ( makeinputs
     general-writer
-    ncbi-vdb-static
+    ncbi-vdb
 )
 
 if ( NOT WIN32)
