Author: Aaron M. Ucko
Last-Update: 2023-01-15 20:56:15
Description: use Debian packaged compression libs

Index: b/tools/loaders/bam-loader/CMakeLists.txt
===================================================================
--- a/tools/loaders/bam-loader/CMakeLists.txt
+++ b/tools/loaders/bam-loader/CMakeLists.txt
@@ -62,13 +62,13 @@ if( NOT WIN32)
             target_compile_options( bam-load PUBLIC -msse4.2 -DBMSSE42OPT)
         endif()
         target_include_directories( bam-load PRIVATE ${CMAKE_SOURCE_DIR}/libs/inc)
-        target_link_libraries( bam-load  loader OpenSSL::SSL ${COMMON_LINK_LIBRARIES}  ${COMMON_LIBS_WRITE})
+        target_link_libraries( bam-load  loader OpenSSL::SSL ${COMMON_LINK_LIBRARIES}  ${COMMON_LIBS_WRITE} -lz )
 
         MakeLinksExe( bam-load false )
 	endif()
 
 	# Internal
-	GenerateExecutableWithDefs( samview "bam;sam;samview" "" "" "${COMMON_LINK_LIBRARIES};${COMMON_LIBS_READ}" )
+	GenerateExecutableWithDefs( samview "bam;sam;samview" "" "" "${COMMON_LINK_LIBRARIES};${COMMON_LIBS_READ};-lz" )
 	MakeLinksExe( samview false )
 
 endif()
Index: b/tools/loaders/sra-sort/CMakeLists.txt
===================================================================
--- a/tools/loaders/sra-sort/CMakeLists.txt
+++ b/tools/loaders/sra-sort/CMakeLists.txt
@@ -64,8 +64,8 @@ set( SRA_SORT_SRC
 	csra-pair
 )
 
-GenerateExecutableWithDefs( sra-sort "${SRA_SORT_SRC}" "" "" "${COMMON_LINK_LIBRARIES};${COMMON_LIBS_WRITE}" )
+GenerateExecutableWithDefs( sra-sort "${SRA_SORT_SRC}" "" "" "${COMMON_LINK_LIBRARIES};${COMMON_LIBS_WRITE};-lbz2;-llzma" )
 MakeLinksExe( sra-sort false )
 
-GenerateExecutableWithDefs( sra-sort-cg "${SRA_SORT_SRC}" "SEQUENCE_BEFORE_SECONDARY=0" "" "${COMMON_LINK_LIBRARIES};${COMMON_LIBS_WRITE}" )
+GenerateExecutableWithDefs( sra-sort-cg "${SRA_SORT_SRC}" "SEQUENCE_BEFORE_SECONDARY=0" "" "${COMMON_LINK_LIBRARIES};${COMMON_LIBS_WRITE};-lbz2;-llzma" )
 MakeLinksExe( sra-sort-cg false )
