Author: Aaron M. Ucko
Last-Update: 2023-01-22 13:15:28
Description: use C locale for comma

Index: b/tools/loaders/bam-loader/loader-imp.c
===================================================================
--- a/tools/loaders/bam-loader/loader-imp.c
+++ b/tools/loaders/bam-loader/loader-imp.c
@@ -4014,7 +4014,7 @@ static rc_t ArchiveBAM(VDBManager *mgr,
                        bool *has_alignments,
                        bool continuing)
 {
-    std::locale::global(std::locale("en_US.UTF-8")); // enable comma as thousand separator
+    std::locale::global(std::locale("C")); // enable comma as thousand separator
     if (G.hasExtraLogging) {
         auto logger = spdlog::stderr_logger_mt("stderr"); // send log to stderr
         spdlog::set_default_logger(logger);
Index: b/tools/loaders/sharq/fastq_parse.cpp
===================================================================
--- a/tools/loaders/sharq/fastq_parse.cpp
+++ b/tools/loaders/sharq/fastq_parse.cpp
@@ -567,7 +567,7 @@ void CFastqParseApp::xCheckErrorLimits(f
 int main(int argc, const char* argv[])
 {
     ios_base::sync_with_stdio(false);   // turn off synchronization with standard C streams
-    std::locale::global(std::locale("en_US.UTF-8")); // enable comma as thousand separator
+    std::locale::global(std::locale("C")); // enable comma as thousand separator
     auto stderr_logger = spdlog::stderr_logger_mt("stderr"); // send log to stderr
     spdlog::set_default_logger(stderr_logger);
 
