Author: Aaron M. Ucko
Last-Update: 2023-01-22 13:16:48
Description: Fulfill baseline requirement

Index: b/test/loaders/sharq/CMakeLists.txt
===================================================================
--- a/test/loaders/sharq/CMakeLists.txt
+++ b/test/loaders/sharq/CMakeLists.txt
@@ -31,7 +31,7 @@ if( SINGLE_CONFIG )
     if ( ZLIB_FOUND AND BZip2_FOUND AND Threads_FOUND )
 
         set(CMAKE_CXX_STANDARD 17)
-        add_definitions(-msse4.2 -DBMSSE42OPT)
+        # add_definitions(-msse4.2 -DBMSSE42OPT)
 
         # assume tools/sharq built libre2
         set(LOCAL_BUILD_DIR ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY})
Index: b/tools/loaders/bam-loader/CMakeLists.txt
===================================================================
--- a/tools/loaders/bam-loader/CMakeLists.txt
+++ b/tools/loaders/bam-loader/CMakeLists.txt
@@ -58,9 +58,9 @@ if( NOT WIN32)
 
         GenerateExecutableWithDefs( bam-load "${SRC}" "" "${CMAKE_SOURCE_DIR}/libs/inc" "loader;${COMMON_LINK_LIBRARIES};${COMMON_LIBS_WRITE}" )
         target_compile_features(bam-load PUBLIC cxx_std_17)
-        if (COMPILER_OPTION_SSE42_SUPPORTED)
-            target_compile_options( bam-load PUBLIC -msse4.2 -DBMSSE42OPT)
-        endif()
+        #if (COMPILER_OPTION_SSE42_SUPPORTED)
+        #    target_compile_options( bam-load PUBLIC -msse4.2 -DBMSSE42OPT)
+        #endif()
         target_include_directories( bam-load PRIVATE ${CMAKE_SOURCE_DIR}/libs/inc)
         target_link_libraries( bam-load  loader OpenSSL::SSL ${COMMON_LINK_LIBRARIES}  ${COMMON_LIBS_WRITE} -lz )
 
Index: b/tools/loaders/sharq/CMakeLists.txt
===================================================================
--- a/tools/loaders/sharq/CMakeLists.txt
+++ b/tools/loaders/sharq/CMakeLists.txt
@@ -52,10 +52,10 @@ if ( ZLIB_FOUND AND BZip2_FOUND AND Thre
 #set(CMAKE_VERBOSE_MAKEFILE ON)
 set(CMAKE_CXX_STANDARD 17)
 
-#specific to BitMagic
-if (COMPILER_OPTION_SSE42_SUPPORTED)
-    add_definitions(-msse4.2 -DBMSSE42OPT)
-endif()
+##specific to BitMagic
+#if (COMPILER_OPTION_SSE42_SUPPORTED)
+#    add_definitions(-msse4.2 -DBMSSE42OPT)
+#endif()
 
 set(LOCAL_BUILD_DIR ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY})
 set(LOCAL_INCDIR ${LOCAL_BUILD_DIR}/include)
