diff -Naur sra-tools-2.9.2-1/setup/konfigure.perl sra-tools-2.9.2-1.ngs/setup/konfigure.perl
--- sra-tools-2.9.2-1/setup/konfigure.perl	2019-02-28 09:53:33.717537330 -0600
+++ sra-tools-2.9.2-1.ngs/setup/konfigure.perl	2019-02-28 09:53:52.774939792 -0600
@@ -463,6 +463,7 @@
     }
 }
 
+$OPT{'debug'} = 1;
 my %DEPEND_OPTIONS;
 foreach my $href (DEPENDS()) {
     $_ = $href->{name};
@@ -970,7 +971,7 @@
 
     if ($PKG{LNG} eq 'JAVA') {
         L($F,
-            "INCPATHS = \$(SRCDIR):\$(SRCDIR)/itf:\$(TOP)/gov/nih/nlm/ncbi/ngs")
+            "INCPATHS = \$(SRCDIR):/usr/include/ngs/itf:/usr/include/ngs")
     }
 
     print $F <<EndText;
@@ -1351,7 +1352,9 @@
         print "\tlibraries... " unless ($AUTORUN);
         if ($lib) {
             my $builddir = File::Spec->catdir($dir, $OS, $TOOLS, $ARCH, $BUILD);
-            my $libdir  = File::Spec->catdir($builddir, 'lib');
+            my $buildarch = `dpkg-architecture -qDEB_BUILD_MULTIARCH`;
+            chomp $buildarch;
+            my $libdir  = File::Spec->catdir($dir, 'lib');
             my $ilibdir = File::Spec->catdir($builddir, 'ilib');
             my $f = File::Spec->catdir($libdir, $lib);
             print "\n\t\tchecking $f\n\t" if ($OPT{'debug'});
diff -Naur sra-tools-2.9.2-1/setup/package.prl sra-tools-2.9.2-1.ngs/setup/package.prl
--- sra-tools-2.9.2-1/setup/package.prl	2019-02-28 09:53:33.714203866 -0600
+++ sra-tools-2.9.2-1.ngs/setup/package.prl	2019-02-28 09:54:33.283067029 -0600
@@ -8,7 +8,7 @@
           ( LNG   => 'C',
             OUT   => 'ncbi-outdir',
             LOCOUT=> '../../OUTDIR',
-            PATH  => '/usr/local/ncbi/sra-tools',
+            PATH  => '/usr/bin/sra-tools',
             UPATH =>      '$HOME/ncbi/sra-tools',
             OPT   => \@OPT,
          ) }
@@ -22,7 +22,7 @@
               option  => 'with-ngs-sdk-prefix',
               origin  => 'I',
               type    => 'L',
-              srcpath => '../ngs/ngs-sdk',
+              srcpath => '/usr/include',
               pkgpath => '/usr/local/ngs/ngs-sdk',
               usrpath =>      '$HOME/ngs/ngs-sdk',
               bldpath => '$HOME/ncbi-outdir/ngs-sdk',
