Author: Andreas Tille <tille@debian.org>
Last-Update: Mon, 06 Jun 2016 11:24:46 +0200
Description: ngs-sdk header files are in /usr/include

--- a/setup/konfigure.perl
+++ b/setup/konfigure.perl
@@ -435,6 +435,7 @@ if ($TOOLS =~ /gcc$/) {
 
 my @dependencies;
 
+$OPT{'debug'} = 1;
 my %DEPEND_OPTIONS;
 foreach my $href (DEPENDS()) {
     $_ = $href->{name};
@@ -931,7 +932,7 @@ EndText
 
     if ($PKG{LNG} eq 'JAVA') {
         L($F,
-            "INCPATHS = \$(SRCDIR):\$(SRCDIR)/itf:\$(TOP)/gov/nih/nlm/ncbi/ngs")
+            "INCPATHS = \$(SRCDIR):/usr/include/ngs/itf:/usr/include/ngs")
     }
 
     print $F <<EndText;
@@ -1309,7 +1310,9 @@ sub find_in_dir {
         print "\tlibraries... " unless ($AUTORUN);
         if ($lib) {
             my $builddir = File::Spec->catdir($dir, $OS, $TOOLS, $ARCH, $BUILD);
-            my $libdir  = File::Spec->catdir($builddir, 'lib');
+            my $buildarch = `dpkg-architecture -qDEB_BUILD_MULTIARCH`;
+            chomp $buildarch;
+            my $libdir  = File::Spec->catdir($dir, 'lib');
             my $ilibdir = File::Spec->catdir($builddir, 'ilib');
             my $f = File::Spec->catdir($libdir, $lib);
             print "\n\t\tchecking $f\n\t" if ($OPT{'debug'});
--- a/setup/package.prl
+++ b/setup/package.prl
@@ -7,7 +7,7 @@ sub CONFIG_OUT   { 'build' }
 sub PKG { ( LNG   => 'C',
             OUT   => 'ncbi-outdir',
             LOCOUT=> '../../OUTDIR',
-            PATH  => '/usr/local/ncbi/sra-tools',
+            PATH  => '/usr/bin/sra-tools',
             UPATH =>      '$HOME/ncbi/sra-tools', ) }
 sub DEPENDS { ( { name => 'fuse' , Include => '/usr/include'        , },
                 { name => 'hdf5' , Include => '/usr/include'        , },
@@ -19,7 +19,7 @@ sub REQ { (
               option  => 'with-ngs-sdk-prefix',
               origin  => 'I',
               type    => 'L',
-              srcpath => '../ngs/ngs-sdk',
+              srcpath => '/usr/include',
               pkgpath => '/usr/local/ngs/ngs-sdk',
               usrpath =>      '$HOME/ngs/ngs-sdk',
               bldpath => '$HOME/ncbi-outdir/ngs-sdk',

