Author: Andreas Tille <tille@debian.org>
Last-Update: Mon, 06 Jun 2016 11:24:46 +0200
Description: Make sure libncbi-vdb will be found

--- a/setup/package.prl
+++ b/setup/package.prl
@@ -33,10 +33,10 @@ sub REQ { (
               boption => 'with-ncbi-vdb-build',
               origin  => 'I',
               type    => 'SB',
-              srcpath => '../ncbi-vdb',
-              pkgpath => '/usr/local/ncbi/ncbi-vdb',
+              srcpath => '/usr/include',
+              pkgpath => '/usr',
               usrpath =>      '$HOME/ncbi/ncbi-vdb',
-              bldpath => '$HOME/ncbi-outdir/ncbi-vdb',
+              bldpath => '/usr',
               locbldpath=>'$Bin/../../OUTDIR/ncbi-vdb',
               include => 'klib/rc.h',
               src     => 'libs/kfg/default.kfg',
--- a/setup/konfigure.perl
+++ b/setup/konfigure.perl
@@ -1301,6 +1301,9 @@ sub find_in_dir {
         } elsif (-e "$dir/interfaces/$include") {
             println $dir unless ($AUTORUN);
             $found_inc = "$dir/interfaces";
+        } elsif (-e "$dir/include/ncbi-vdb") {
+            println $dir unless ($AUTORUN);
+            $found_inc = "$dir/include/ncbi-vdb";
         } else {
             print "$dir: " if ($OPT{'debug'});
             println 'no' unless ($AUTORUN);
