From: =?utf-8?b?0L3QsNCx?= <nabijaczleweli@nabijaczleweli.xyz>
Date: Tue, 24 Sep 2024 17:38:04 +0200
Subject: Format MAC addresses correctly (Closes: #1082673)

---
 ether-wake.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ether-wake.c b/ether-wake.c
index 5839346..fcfeac5 100644
--- a/ether-wake.c
+++ b/ether-wake.c
@@ -210,7 +210,7 @@ int main(int argc, char *argv[])
 
 		if (verbose) {
 			printf("The hardware address (SIOCGIFHWADDR) of %s is type %d  "
-				   "%2.2x:%2.2x:%2.2x:%2.2x:%2.2x:%2.2x.\n", ifname,
+				   "%02hhx:%02hhx:%02hhx:%02hhx:%02hhx:%02hhx.\n", ifname,
 				   if_hwaddr.ifr_hwaddr.sa_family, hwaddr[0], hwaddr[1],
 				   hwaddr[2], hwaddr[3], hwaddr[4], hwaddr[5]);
 		}
