From e5a01df4f075ddb96c393748d37a16498aa37c0a Mon Sep 17 00:00:00 2001
From: Shawn Landden <shawnlandden@gmail.com>
Date: Wed, 18 Jul 2012 18:21:14 -0700
Subject: [PATCH] fix Makefile -- allows build of examples, multi-arch: same

---
 examples/Makefile.in |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: b/examples/Makefile.in
===================================================================
--- a/examples/Makefile.in
+++ b/examples/Makefile.in
@@ -27,12 +27,12 @@
 datarootdir = @datarootdir@
 
 bindir = @bindir@
-srcdir = @srcdir@
+srcdir = /usr/share/doc/libreadline6/examples
 datadir = @datadir@
-VPATH = @srcdir@
-top_srcdir = @top_srcdir@
+VPATH = .:/usr/share/doc/libreadline6/examples
+top_srcdir = /usr/include/readline
 #BUILD_DIR = .
-BUILD_DIR = @BUILD_DIR@
+BUILD_DIR = /usr/src/readline6/build
 installdir = $(datadir)/readline
 
 INSTALL = @INSTALL@
