From 97d8a824bd37b70c3aef4a7be96d4a0a00f1a848 Mon Sep 17 00:00:00 2001
From: Lion Yang <lion@aosc.io>
Date: Thu, 7 Dec 2017 22:33:44 +0800
Subject: [PATCH] CMakeLists.txt: fix libbase PIC error and filename issue

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git MaiAT3PlusDecoder/CMakeLists.txt MaiAT3PlusDecoder/CMakeLists.txt
index bf8b073..1b46d5d 100644
--- MaiAT3PlusDecoder/CMakeLists.txt
+++ MaiAT3PlusDecoder/CMakeLists.txt
@@ -200,9 +200,11 @@ add_library(base STATIC
 	src/base/MaiString.cc
 	src/base/MaiThread.cc
 	src/base/Mai_Sleep.cc
-	src/base/Mai_Mem.cc
+	src/base/Mai_mem.cc
 	src/base/Unicode.cc)
 
+set_property(TARGET base PROPERTY POSITION_INDEPENDENT_CODE ON)
+
 add_library(at3plusdecoder SHARED
 	maiatrac3plus.cpp
 	src/MaiAT3PlusCoreDecoder.cpp
-- 
2.14.2


