From dc5a0dd419aaedbe836b501f289625e3f0e408d0 Mon Sep 17 00:00:00 2001
From: Devin Lin <espidev@gmail.com>
Date: Sun, 5 Mar 2023 12:53:09 -0800
Subject: [PATCH] Disable QtFeedback

---
 .../mobileshell/qml/components/HapticsEffectLoader.qml   | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/components/mobileshell/qml/components/HapticsEffectLoader.qml b/components/mobileshell/qml/components/HapticsEffectLoader.qml
index ebb42e88b..a25f5ba25 100644
--- a/components/mobileshell/qml/components/HapticsEffectLoader.qml
+++ b/components/mobileshell/qml/components/HapticsEffectLoader.qml
@@ -6,14 +6,15 @@ import QtQuick 2.15
 import org.kde.plasma.private.mobileshell 1.0 as MobileShell
 
 Loader {
-    source: "qrc:/org/kde/plasma/private/mobileshell/qml/components/HapticsEffectWrapper.qml"
+    // source: "qrc:/org/kde/plasma/private/mobileshell/qml/components/HapticsEffectWrapper.qml"
     property bool valid: item !== null
     
     function buttonVibrate() {
         if (valid && MobileShell.MobileShellSettings.vibrationsEnabled) {
-            item.intensity = MobileShell.MobileShellSettings.vibrationIntensity;
-            item.duration = MobileShell.MobileShellSettings.vibrationDuration;
-            item.start();
+            // TODO we need a haptics stack for Qt 6
+            // item.intensity = MobileShell.MobileShellSettings.vibrationIntensity;
+            // item.duration = MobileShell.MobileShellSettings.vibrationDuration;
+            // item.start();
         }
     }
 }
-- 
2.39.1

