From a559bc2791d427c5b2728c17aa56d04dc4198704 Mon Sep 17 00:00:00 2001
From: wxiwnd <wxiwnd@outlook.com>
Date: Mon, 15 Sep 2025 03:48:43 +0800
Subject: [PATCH] fix: disable iconv check

Signed-off-by: wxiwnd <wxiwnd@outlook.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f2c410088..f6c1fd2c2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -259,8 +259,8 @@ if ((CMAKE_BUILD_TYPE STREQUAL "Debug") OR enable_lex_debug)
   set(LEX_FLAGS "${LEX_FLAGS} -d")
 endif()
 
-find_package(Iconv REQUIRED)
-include_directories(${Iconv_INCLUDE_DIRS})
+# find_package(Iconv REQUIRED)
+# include_directories(${Iconv_INCLUDE_DIRS})
 
 
 #set(DOXYDOCS ${PROJECT_SOURCE_DIR}/doc CACHE INTERNAL "Path to doxygen docs")
-- 
2.51.0

