diff --git a/rvslib/CMakeLists.txt b/rvslib/CMakeLists.txt
index 79b1bc2..13e7766 100644
--- a/rvslib/CMakeLists.txt
+++ b/rvslib/CMakeLists.txt
@@ -168,7 +168,6 @@ set(SOURCES_UT
 ## define rvslib library
 add_library(${RVS_TARGET} SHARED ${SOURCES})

-target_compile_options(${RVS_TARGET} PRIVATE -mf16c)
 target_compile_definitions(${RVS_TARGET} PRIVATE ROCM_USE_FLOAT16)
 set_target_properties(${RVS_TARGET}
  PROPERTIES SUFFIX .so.${LIB_VERSION_STRING}
diff --git a/pbqt.so/CMakeLists.txt b/pbqt.so/CMakeLists.txt
index ccaea42..597b1f4 100644
--- a/pbqt.so/CMakeLists.txt
+++ b/pbqt.so/CMakeLists.txt
@@ -42,8 +42,6 @@ message(STATUS "MODULE: ${RVS}")
 add_definitions(-D__linux__)
 add_definitions(-DUNIX_OS)
 add_definitions(-DLINUX)
-add_definitions(-D__AMD64__)
-add_definitions(-D__x86_64__)
 add_definitions(-DAMD_INTERNAL_BUILD)
 add_definitions(-DLITTLEENDIAN_CPU=1)
 add_definitions(-DHSA_LARGE_MODEL=)
diff --git a/pebb.so/CMakeLists.txt b/pebb.so/CMakeLists.txt
index 7f624fd..75346e3 100644
--- a/pebb.so/CMakeLists.txt
+++ b/pebb.so/CMakeLists.txt
@@ -42,8 +42,6 @@ message(STATUS "MODULE: ${RVS}")
 add_definitions(-D__linux__)
 add_definitions(-DUNIX_OS)
 add_definitions(-DLINUX)
-add_definitions(-D__AMD64__)
-add_definitions(-D__x86_64__)
 add_definitions(-DAMD_INTERNAL_BUILD)
 add_definitions(-DLITTLEENDIAN_CPU=1)
 add_definitions(-DHSA_LARGE_MODEL=)
