diff -Naur kvirc-4.9.3~git20180404+dfsg/CMakeLists.txt kvirc-4.9.3~git20180404+dfsg.threads/CMakeLists.txt
--- kvirc-4.9.3~git20180404+dfsg/CMakeLists.txt	2018-04-03 18:42:53.000000000 +0000
+++ kvirc-4.9.3~git20180404+dfsg.threads/CMakeLists.txt	2018-07-16 21:02:11.212239525 +0000
@@ -354,30 +354,6 @@
 
 # ignore sigalarm
 option(WANT_IGNORE_SIGALARM "Ignore SIGALARM signal" OFF)
-# Check for threads
-find_package(Threads REQUIRED)
-if(CMAKE_USE_WIN32_THREADS_INIT OR CMAKE_USE_PTHREADS_INIT)
-	if(CMAKE_USE_WIN32_THREADS_INIT)
-		# No need to link to any specific library
-		set(COMPILE_THREADS_USE_WIN32 1)
-		set(CMAKE_STATUS_THREADS_SUPPORT "Win32")
-		list(APPEND CMAKE_KVIRC_BUILD_FLAGS Threads=Win32)
-	else()
-		if(CMAKE_HAVE_THREADS_LIBRARY)
-			list(APPEND LIBS ${CMAKE_THREAD_LIBS_INIT})
-		endif()
-		set(COMPILE_THREADS_USE_POSIX 1)
-		list(APPEND CMAKE_KVIRC_BUILD_FLAGS Threads=POSIX)
-		if(WANT_IGNORE_SIGALARM)
-			set(CMAKE_STATUS_THREADS_SUPPORT "POSIX (ignoring SIGALARM)")
-			set(COMPILE_IGNORE_SIGALARM 1)
-		else()
-			set(CMAKE_STATUS_THREADS_SUPPORT "POSIX")
-		endif()
-	endif()
-else()
-	set(CMAKE_STATUS_THREADS_SUPPORT "Not used")
-endif()
 
 # Check for ZLib
 find_package(ZLIB REQUIRED)
