From fb56c95129b169cf5d1e86e426cddfe6657f630c Mon Sep 17 00:00:00 2001
From: ShellWen <me@shellwen.com>
Date: Thu, 10 Jul 2025 19:16:13 +0800
Subject: [PATCH 2/2] fix build error in AOSC

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 3b69a131f9..e19b3ddb4b 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -12,7 +12,7 @@ name = "breezy"
 maintainers = [{name = "Breezy Developers", email = "team@breezy-vcs.org"}]
 description = "Friendly distributed version control system"
 readme = "README.rst"
-license = "GPL-2.0-or-later"
+license = { text = "GPL-2.0-or-later" }
 classifiers = [
     "Development Status :: 6 - Mature",
     "Environment :: Console",
-- 
2.50.0

