diff -Naur fox-1.7.67/lib/FXRex.cpp fox-1.7.67.REXDEBUG/lib/FXRex.cpp
--- fox-1.7.67/lib/FXRex.cpp	2019-08-23 02:46:13.000000000 +0000
+++ fox-1.7.67.REXDEBUG/lib/FXRex.cpp	2020-02-19 04:48:42.999813251 +0000
@@ -503,7 +503,7 @@
 */
 
 // Debugging regex code
-#define REXDEBUG 1
+// #define REXDEBUG 1
 
 // As close to infinity as we're going to get; this seems big enough.  We can not make
 // it too large as this may wrap around when added to something else!
@@ -4656,9 +4656,9 @@
 
 /*******************************************************************************/
 
-#ifdef REXDEBUG
-#include "fxrexdbg.h"
-#endif
+// #ifdef REXDEBUG
+// #include "fxrexdbg.h"
+// #endif
 
 
 // Parse pattern
@@ -4698,9 +4698,9 @@
 
                 // Install new program
                 code=prog;
-#ifdef REXDEBUG
-                if(fxTraceLevel>50){ dump(adjustedpattern,code); }
-#endif
+// #ifdef REXDEBUG
+//                if(fxTraceLevel>50){ dump(adjustedpattern,code); }
+// #endif
                 freeElms(adjustedpattern);
                 return ErrOK;
                 }
