From a33188abd3370b0b57fa247571342384de94f64e Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sun, 27 Sep 2020 02:56:33 +0800
Subject: [PATCH] Set default proxy as shadowsocks or v2ray
 (socks5://127.0.0.1:1080)

---
 src/proxychains.conf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/proxychains.conf b/src/proxychains.conf
index 83f7d90..e68e0a6 100644
--- a/src/proxychains.conf
+++ b/src/proxychains.conf
@@ -111,6 +111,6 @@ tcp_connect_time_out 8000
 [ProxyList]
 # add proxy here ...
 # meanwile
-# defaults set to "tor"
-socks4 	127.0.0.1 9050
+# defaults set to "Shadowsocks" or "v2ray"
+socks5 	127.0.0.1 1080
 
-- 
2.27.0

