diff --git a/site_scons/site_tools/ninja.py b/site_scons/site_tools/ninja.py
index a55205fc..ae3a94f3 100644
--- a/site_scons/site_tools/ninja.py
+++ b/site_scons/site_tools/ninja.py
@@ -1052,6 +1057,8 @@ def get_command_env(env, target, source):
 
     It will be compatible with the default shell of the operating system.
     """
+    def get_default_env(env, target, source):
+        return get_default_ENV(env)
     try:
         return env["NINJA_ENV_VAR_CACHE"]
     except KeyError:
@@ -1061,7 +1068,7 @@ def get_command_env(env, target, source):
     # os.environ or differ from it. We assume if it's a new or
     # differing key from the process environment then it's
     # important to pass down to commands in the Ninja file.
-    ENV = env.get('SHELL_ENV_GENERATOR', get_default_ENV)(env, target, source)
+    ENV = env.get('SHELL_ENV_GENERATOR', get_default_env)(env, target, source)
     scons_specified_env = {
         key: value
         for key, value in ENV.items() if key not in os.environ or os.environ.get(key, None) != value
