diff --git a/modules/parse_amd.c b/modules/parse_amd.c
index 163174c..27a2368 100644
--- a/modules/parse_amd.c
+++ b/modules/parse_amd.c
@@ -27,7 +27,6 @@
 #include <sys/utsname.h>
 #include <netinet/in.h>
 #include <sys/mount.h>
-#include <linux/fs.h>
 
 #define MODULE_PARSE
 #include "automount.h"
diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index d9ac0c9..5c858fd 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -30,7 +30,6 @@
 #include <sys/utsname.h>
 #include <netinet/in.h>
 #include <sys/mount.h>
-#include <linux/fs.h>
 
 #define MODULE_PARSE
 #include "automount.h"
