From f278d9a5ac7ad31eb16bd120708b07e5019f3a0e Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Wed, 15 Oct 2025 19:18:48 +0800
Subject: [PATCH 21/22] FEDORA: fix configure c99

---
 configure  | 1 +
 m4/sane.m4 | 1 +
 2 files changed, 2 insertions(+)

diff --git a/configure b/configure
index 965e41d..677309d 100755
--- a/configure
+++ b/configure
@@ -6672,6 +6672,7 @@ else
 
 #include <sane/sane.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 int
 main ()
diff --git a/m4/sane.m4 b/m4/sane.m4
index d0dc89c..c55bf0c 100644
--- a/m4/sane.m4
+++ b/m4/sane.m4
@@ -44,6 +44,7 @@ dnl
       AC_TRY_RUN([
 #include <sane/sane.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 int 
 main ()
-- 
2.51.1

