diff -Naur lshw-B.02.19.2/src/core/version.cc lshw-B.02.19.2.nocheckupdate/src/core/version.cc
--- lshw-B.02.19.2/src/core/version.cc	2020-03-22 10:50:25.000000000 -0700
+++ lshw-B.02.19.2.nocheckupdate/src/core/version.cc	2022-12-15 21:40:40.837273321 -0800
@@ -84,13 +84,3 @@
 
   return txt;
 }
-
-const char * checkupdates()
-{
-  static char *latest = NULL;
-
-  if(!latest)
-    latest = txtquery(PACKAGE, "ezix.org", NULL);
-
-  return latest;
-}
diff -Naur lshw-B.02.19.2/src/core/version.h lshw-B.02.19.2.nocheckupdate/src/core/version.h
--- lshw-B.02.19.2/src/core/version.h	2020-03-22 10:50:25.000000000 -0700
+++ lshw-B.02.19.2.nocheckupdate/src/core/version.h	2022-12-15 21:40:42.606296315 -0800
@@ -13,8 +13,6 @@
 
 const char * getpackageversion();
 
-const char * checkupdates();
-
 #ifdef __cplusplus
 }
 #endif
diff -Naur lshw-B.02.19.2/src/gui/callbacks.c lshw-B.02.19.2.nocheckupdate/src/gui/callbacks.c
--- lshw-B.02.19.2/src/gui/callbacks.c	2020-03-22 10:50:25.000000000 -0700
+++ lshw-B.02.19.2.nocheckupdate/src/gui/callbacks.c	2022-12-15 21:41:05.813597862 -0800
@@ -53,35 +53,7 @@
 on_version_realize                     (GtkWidget       *widget,
 gpointer         user_data)
 {
-  const char *latest = checkupdates();
-
   gtk_label_set_text(GTK_LABEL(widget), getpackageversion());
-
-  if(latest)
-  {
-    if(strcmp(latest, getpackageversion()) != 0)
-    {
-      static GtkWidget *dialog = NULL;
-
-      if(!GTK_IS_WIDGET(dialog))
-      {
-        dialog = gtk_message_dialog_new_with_markup (GTK_WINDOW(mainwindow),
-                                  GTK_DIALOG_DESTROY_WITH_PARENT,
-                                  GTK_MESSAGE_INFO,
-                                  GTK_BUTTONS_CLOSE,
-                                  "The latest version is <tt>%s</tt>.\n\nYou can visit <span foreground=\"blue\"><u>http://www.ezix.org/</u></span> for more information.",
-                                  latest);
-
-        gtk_window_set_title(GTK_WINDOW(dialog), "Update available");
-        /* Destroy the dialog when the user responds to it (e.g. clicks a button) */
-        g_signal_connect_swapped (dialog, "response",
-                           G_CALLBACK (gtk_widget_destroy),
-                           dialog);
-      }
-
-      gtk_widget_show(dialog);
-    }
-  }
 }
 
 
diff -Naur lshw-B.02.19.2/src/lshw.cc lshw-B.02.19.2.nocheckupdate/src/lshw.cc
--- lshw-B.02.19.2/src/lshw.cc	2020-03-22 10:50:25.000000000 -0700
+++ lshw-B.02.19.2.nocheckupdate/src/lshw.cc	2022-12-15 21:40:37.045224042 -0800
@@ -117,13 +117,7 @@
 
     if (strcmp(argv[1], "-version") == 0)
     {
-      const char *latest = checkupdates();
       printf("%s\n", getpackageversion());
-      if(latest)
-      {
-        if(strcmp(latest, getpackageversion()) != 0)
-          fprintf(stderr, _("the latest version is %s\n"), latest);
-      }
       exit(0);
     }
 
