From 119149f44e765fdbb80f28e2a629ef2b9855ef86 Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 31 May 2025 06:23:11 +0800
Subject: [PATCH 2/4] AOSCOS: Use Python 3

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 3959dca1d087..250b3e87af9a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -41,7 +41,7 @@ if test "$GLIB_LIBS" = ""; then
 fi
 
 dnl == check for python ==
-AM_PATH_PYTHON(2.4)
+AM_PATH_PYTHON(3.8)
 
 dnl == Library dependencies ==
 PYGOBJECT_REQUIRED=2.90.3
-- 
2.49.0

