diff -Naur nvidia-515.65.01/kernel/nvidia-drm/nvidia-drm-helper.c nvidia-515.65.01.modded/kernel/nvidia-drm/nvidia-drm-helper.c
--- nvidia-515.65.01/kernel/nvidia-drm/nvidia-drm-helper.c	2022-09-13 18:41:10.235610460 -0400
+++ nvidia-515.65.01.modded/kernel/nvidia-drm/nvidia-drm-helper.c	2022-09-13 19:06:23.599321167 -0400
@@ -41,6 +41,11 @@
 #include <drm/drm_atomic_uapi.h>
 #endif
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 0, 0)
+#include <drm/drm_framebuffer.h>
+#endif
+
 static void __nv_drm_framebuffer_put(struct drm_framebuffer *fb)
 {
 #if defined(NV_DRM_FRAMEBUFFER_GET_PRESENT)
