diff -Naur azcvpng-1.4-src/Makefile azcvpng-1.4-src.working/Makefile
--- azcvpng-1.4-src/Makefile	2013-01-04 09:00:00.000000000 -0600
+++ azcvpng-1.4-src.working/Makefile	2019-03-31 19:52:17.815357256 -0500
@@ -12,7 +12,7 @@
 CXXFLAGS = -O2 -Wall
 INCLUDE = -I./azxclass/include -I/usr/include/freetype2
 LFLAGS	= -s
-LINKS	= -lrt -lX11 -lXext -lXft -lpthread -lz
+LINKS	= -lrt -lX11 -lXext -lXft -lpthread -lz -lfontconfig
 MACRO	= -D_AZCVPNG_RESDIR="\"$(datadir)\""
 RM	= rm -f
 
