From 84802fe3dd17ce904f8a4293f49bcdfd7e98564a Mon Sep 17 00:00:00 2001
From: Thomas Kluyver <thomas@kluyver.me.uk>
Date: Mon, 31 Oct 2022 11:09:53 +0000
Subject: [PATCH] Convert pyproject.toml to PEP-621 [project] style metadata

---
 pyproject.toml | 26 +++++++++++++++-----------
 1 file changed, 15 insertions(+), 11 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 9f407a3..e61bb92 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,24 +1,28 @@
 [build-system]
-requires = ["flit_core >=2,<4"]
+requires = ["flit_core >=3.2"]
 build-backend = "flit_core.buildapi"
 
-[tool.flit.metadata]
-module = "pep517"
-author = "Thomas Kluyver"
-author-email = "thomas@kluyver.me.uk"
-home-page = "https://github.com/pypa/pep517"
-description-file = "README.rst"
-requires = [
-    "tomli >=1.1.0;python_version<'3.11'",
-    "importlib_metadata;python_version<'3.8'",
-    "zipp;python_version<'3.8'",
+[project]
+name = "pep517"
+authors = [
+    {name = "Thomas Kluyver", email = "thomas@kluyver.me.uk"},
 ]
+readme = "README.rst"
 requires-python = ">=3.6"
+dependencies = [
+    "tomli >=1.1.0 ; python_version<'3.11'",
+    "importlib_metadata ; python_version<'3.8'",
+    "zipp ; python_version<'3.8'",
+]
 classifiers = [
     "License :: OSI Approved :: MIT License",
     "Programming Language :: Python :: 3",
     "Programming Language :: Python :: 3 :: Only",
 ]
+dynamic = ["version", "description"]
+
+[project.urls]
+Homepage = "https://github.com/pypa/pep517"
 
 [tool.isort]
 profile = "black"
