diff -Naur a/js/misc/config.js b/js/misc/config.js
--- a/js/misc/config.js	2016-05-11 04:21:54.000000000 +0800
+++ b/js/misc/config.js	2016-06-28 20:05:51.718838442 +0800
@@ -11,7 +11,7 @@
 /* gettext package */
 const GETTEXT_PACKAGE = 'gnome-shell';
 /* locale dir */
-const LOCALEDIR = '/home/fmuellner/opt/gnome/share/locale';
+const LOCALEDIR = '/usr/share/locale';
 /* other standard directories */
-const LIBEXECDIR = '/home/fmuellner/opt/gnome/libexec';
-const SYSCONFDIR = '/home/fmuellner/opt/gnome/etc';
+const LIBEXECDIR = '/usr/libexec';
+const SYSCONFDIR = '/etc';
