--- a/js/misc/config.js	2017-06-24 21:35:40.059169313 +0800
+++ b/js/misc/config.js	2017-06-24 21:37:30.418656835 +0800
@@ -11,9 +11,9 @@
 /* gettext package */
 const GETTEXT_PACKAGE = 'gnome-shell';
 /* locale dir */
-const LOCALEDIR = '/home/fmuellner/opt/gnome/share/locale';
+const LOCALEDIR = '/usr/share/locale';
 /* other standard directories */
-const LIBEXECDIR = '/home/fmuellner/opt/gnome/libexec';
-const SYSCONFDIR = '/home/fmuellner/opt/gnome/etc';
+const LIBEXECDIR = '/usr/libexec';
+const SYSCONFDIR = '/etc';
 /* g-i package versions */
 const LIBMUTTER_API_VERSION = '0'
