diff -ru afpfs-ng-0.8.1.old/cmdline/cmdline_testafp.c afpfs-ng-0.8.1.new/cmdline/cmdline_testafp.c
--- afpfs-ng-0.8.1.old/cmdline/cmdline_testafp.c	2025-02-06 15:22:10.187795481 +0100
+++ afpfs-ng-0.8.1.new/cmdline/cmdline_testafp.c	2025-02-06 15:30:13.226751326 +0100
@@ -31,7 +31,7 @@
 	snprintf(valid_url.path,sizeof(valid_url.path),"%s",path);
 	snprintf(valid_url.username,sizeof(valid_url.username),"%s",username);
 	snprintf(valid_url.password,sizeof(valid_url.password),"%s",password);
-	snprintf(valid_url.uamname,(valid_url.uamname),"%s",uamname);
+	snprintf(valid_url.uamname,sizeof(valid_url.uamname),"%s",uamname);
 	valid_url.port=port;
 
 	if (afp_url_validate(url_string,&valid_url)) 
@@ -42,7 +42,7 @@
 	return 0;
 }
 
-int test_urls(void)
+int test_urls(char * arg)
 {
 
 	printf("Testing URL parsing\n");
diff -ru afpfs-ng-0.8.1.old/cmdline/cmdline_testafp.h afpfs-ng-0.8.1.new/cmdline/cmdline_testafp.h
--- afpfs-ng-0.8.1.old/cmdline/cmdline_testafp.h	2008-01-05 06:00:57.000000000 +0100
+++ afpfs-ng-0.8.1.new/cmdline/cmdline_testafp.h	2025-02-06 15:30:51.577987329 +0100
@@ -1,4 +1,4 @@
 #ifndef __CMDLINE_TESTAFP_H_
-int test_urls(void);
+int test_urls(char * arg);
 #endif
 
