diff --git a/src/main/services/AppUpdater.ts b/src/main/services/AppUpdater.ts
index ec0f1b97e..f1c0e5345 100644
--- a/src/main/services/AppUpdater.ts
+++ b/src/main/services/AppUpdater.ts
@@ -129,6 +129,7 @@ export default class AppUpdater {
   }
 
   public setAutoUpdate(isActive: boolean) {
+    isActive = false
     autoUpdater.autoDownload = isActive
     autoUpdater.autoInstallOnAppQuit = isActive
   }
diff --git a/src/main/services/ConfigManager.ts b/src/main/services/ConfigManager.ts
index 3cab0bf91..0ab582aa8 100644
--- a/src/main/services/ConfigManager.ts
+++ b/src/main/services/ConfigManager.ts
@@ -141,7 +141,7 @@ export class ConfigManager {
   }
 
   getAutoUpdate(): boolean {
-    return this.get<boolean>(ConfigKeys.AutoUpdate, true)
+    return this.get<boolean>(ConfigKeys.AutoUpdate, false)
   }
 
   setAutoUpdate(value: boolean) {
diff --git a/src/renderer/src/pages/settings/AboutSettings.tsx b/src/renderer/src/pages/settings/AboutSettings.tsx
index dcf95324d..14d0acc55 100644
--- a/src/renderer/src/pages/settings/AboutSettings.tsx
+++ b/src/renderer/src/pages/settings/AboutSettings.tsx
@@ -216,26 +216,9 @@ const AboutSettings: FC = () => {
               </Tag>
             </VersionWrapper>
           </Row>
-          {!isPortable && (
-            <CheckUpdateButton
-              onClick={onCheckUpdate}
-              loading={update.checking}
-              disabled={update.downloading || update.checking}>
-              {update.downloading
-                ? t('settings.about.downloading')
-                : update.available
-                  ? t('settings.about.checkUpdate.available')
-                  : t('settings.about.checkUpdate.label')}
-            </CheckUpdateButton>
-          )}
         </AboutHeader>
         {!isPortable && (
           <>
-            <SettingDivider />
-            <SettingRow>
-              <SettingRowTitle>{t('settings.general.auto_check_update.title')}</SettingRowTitle>
-              <Switch value={autoCheckUpdate} onChange={(v) => setAutoCheckUpdate(v)} />
-            </SettingRow>
             <SettingDivider />
             <SettingRow>
               <SettingRowTitle>{t('settings.general.test_plan.title')}</SettingRowTitle>
