diff -up libqaccessibilityclient-0.1.1/src/qaccessibilityclient/accessibleobject.cpp.me libqaccessibilityclient-0.1.1/src/qaccessibilityclient/accessibleobject.cpp
diff -up libqaccessibilityclient-0.1.1/src/qaccessibilityclient/accessibleobject.h.me libqaccessibilityclient-0.1.1/src/qaccessibilityclient/accessibleobject.h
--- libqaccessibilityclient-0.1.1/src/qaccessibilityclient/accessibleobject.h.me	2018-02-16 19:04:19.323834631 +0100
+++ libqaccessibilityclient-0.1.1/src/qaccessibilityclient/accessibleobject.h	2018-02-16 19:40:17.604383933 +0100
@@ -39,6 +39,7 @@ namespace QAccessibleClient {
 
 class AccessibleObjectPrivate;
 class RegistryPrivate;
+QACCESSIBILITYCLIENT_EXPORT QDebug operator<<(QDebug, const AccessibleObject &);
 
 /**
     This class represents an accessible object.
