--- ldc-1.24.0-src/runtime/druntime/src/ldc/intrinsics.di	2020-10-24 05:26:27.000000000 -0700
+++ ldc-1.24.0-src.llvm1110/runtime/druntime/src/ldc/intrinsics.di	2021-02-18 12:49:23.773724867 -0800
@@ -26,6 +26,7 @@
 else version (LDC_LLVM_900)  enum LLVM_version =  900;
 else version (LDC_LLVM_1000) enum LLVM_version = 1000;
 else version (LDC_LLVM_1100) enum LLVM_version = 1100;
+else version (LDC_LLVM_1101) enum LLVM_version = 1101;
 else static assert(false, "LDC LLVM version not supported");
 
 enum LLVM_atleast(int major) = (LLVM_version >= major * 100);
