diff -Naur yuzu/externals/SDL/CMakeLists.txt yuzu.nowayland/externals/SDL/CMakeLists.txt
--- yuzu/externals/SDL/CMakeLists.txt	2022-08-09 00:13:58.251794087 +0000
+++ yuzu.nowayland/externals/SDL/CMakeLists.txt	2022-08-09 00:14:17.518126285 +0000
@@ -390,7 +390,7 @@
 set_option(RPATH               "Use an rpath when linking SDL" ${UNIX_SYS})
 set_option(CLOCK_GETTIME       "Use clock_gettime() instead of gettimeofday()" OFF)
 set_option(VIDEO_X11           "Use X11 video driver" ${UNIX_SYS})
-set_option(VIDEO_WAYLAND       "Use Wayland video driver" ${UNIX_SYS})
+set_option(VIDEO_WAYLAND       "Use Wayland video driver" OFF)
 dep_option(WAYLAND_SHARED      "Dynamically load Wayland support" ON "VIDEO_WAYLAND" OFF)
 dep_option(WAYLAND_LIBDECOR    "Use client-side window decorations on Wayland" ON "VIDEO_WAYLAND" ON)
 dep_option(LIBDECOR_SHARED     "Dynamically load libdecor support" ON "WAYLAND_LIBDECOR" OFF)
