From 61a7d570beb9d7266de3c35b210fd04dca223005 Mon Sep 17 00:00:00 2001
From: Bart Martens <bartm@knars.be>
Date: Sun, 20 May 2007 11:00:41 +0200
Subject: [PATCH 3/8] FEDORA: Do not symlink zssh to ztelnet

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 69f68d7..51f60c0 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -76,7 +76,7 @@ ztelnet : $(NAME)
 install :
 	strip $(NAME)
 	$(INSTALL) -m 0711 zssh $(DESTDIR)/@bindir@
-	ln -f $(DESTDIR)/@bindir@/zssh  $(DESTDIR)/@bindir@/ztelnet
+	#ln -f $(DESTDIR)/@bindir@/zssh  $(DESTDIR)/@bindir@/ztelnet
 	$(INSTALL) -m 0644 $(MANS) $(DESTDIR)/@mandir@/man1
 
 uninstall :
-- 
2.52.0

