From 2f95975f861b78e6c2029842645eff0c57e1e499 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Thu, 22 Aug 2024 00:41:52 -0700
Subject: [PATCH] vt,terminal: fix utf8proc.h location on AOSC OS

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 terminal.h | 2 +-
 vt.c       | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/terminal.h b/terminal.h
index a87a125b..f749e43b 100644
--- a/terminal.h
+++ b/terminal.h
@@ -8,7 +8,7 @@
 #include <semaphore.h>
 
 #if defined(FOOT_GRAPHEME_CLUSTERING)
- #include <utf8proc.h>
+ #include <libutf8proc/utf8proc.h>
 #endif
 
 #include <tllist.h>
diff --git a/vt.c b/vt.c
index 572dd2af..1501ea06 100644
--- a/vt.c
+++ b/vt.c
@@ -5,7 +5,7 @@
 #include <unistd.h>
 
 #if defined(FOOT_GRAPHEME_CLUSTERING)
- #include <utf8proc.h>
+ #include <libutf8proc/utf8proc.h>
 #endif
 
 #define LOG_MODULE "vt"
-- 
2.46.0

