diff --git a/src/memberdef.cpp b/src/memberdef.cpp
index d37efb9..26001a1 100644
--- a/src/memberdef.cpp
+++ b/src/memberdef.cpp
@@ -3143,8 +3143,7 @@ void MemberDef::writeMemberDocSimple(OutputList &ol, Definition *container)
           scope,                   // scope
           getBodyDef(),            // fileScope
           this,                    // self
-          ts,                      // text
-          TRUE                     // autoBreak
+          ts                       // text
           );
     }
     ol.endDoxyAnchor(cfname,memAnchor);
