--- azdrawing-1.5/Makefile	2015-06-30 10:00:00.000000000 -0500
+++ azdrawing-1.5.pthread/Makefile	2021-01-31 16:41:06.694159984 -0600
@@ -26,7 +26,7 @@
 CXX := g++ -pthread
 CXXFLAGS := -Wall -O2
 LFLAGS := -s
-LINKS := -lX11 -lXext -lXft -lfontconfig -lfreetype -lz -ljpeg -lrt
+LINKS := -lX11 -lXext -lXft -lfontconfig -lfreetype -lz -ljpeg -lrt -lpthread
 MACRO := -D_AZDRAW_RESDIR="\"$(datadir)\""
 MACRO_OPT :=
 INCLUDE := -Iazxclass/include -I$(freetype_dir)
