From a084f2f782ad12ca6923a84b702ef1aeac28a5a7 Mon Sep 17 00:00:00 2001
From: Eric Koegel <eric.koegel@gmail.com>
Date: Mon, 10 Aug 2015 17:52:55 +0300
Subject: Don't redraw the icons on focus change events (Bug #12071)


diff --git a/src/xfdesktop-icon-view.c b/src/xfdesktop-icon-view.c
index c598171..46de229 100644
--- a/src/xfdesktop-icon-view.c
+++ b/src/xfdesktop-icon-view.c
@@ -739,7 +739,7 @@ xfdesktop_icon_view_init(XfdesktopIconView *icon_view)
                      G_CALLBACK(xfdesktop_icon_view_show_tooltip), NULL);
     
     gtk_widget_set_has_window(GTK_WIDGET(icon_view), FALSE);
-    gtk_widget_set_can_focus(GTK_WIDGET(icon_view), TRUE);
+    gtk_widget_set_can_focus(GTK_WIDGET(icon_view), FALSE);
 }
 
 static void
-- 
cgit v0.10.1

