From 0ff4392507037fa840e70a324a07465c21890cd0 Mon Sep 17 00:00:00 2001
From: miwu04 <mail@alanlin.icu>
Date: Tue, 17 Dec 2024 00:54:50 +0800
Subject: [PATCH 6/7] Do not find Mihomo(Clash Meta) binary in sidecar dir

---
 src/main/utils/dirs.ts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/main/utils/dirs.ts b/src/main/utils/dirs.ts
index 0ea3566..8ba99d4 100644
--- a/src/main/utils/dirs.ts
+++ b/src/main/utils/dirs.ts
@@ -58,7 +58,8 @@ export function themesDir(): string {
 }
 
 export function mihomoCoreDir(): string {
-  return path.join(resourcesDir(), 'sidecar')
+  const isLinux = process.platform === 'linux'
+  return isLinux ? '/usr/bin' : path.join(resourcesDir(), 'sidecar')
 }
 
 export function mihomoCorePath(core: string): string {
-- 
2.48.1

