diff -Naur pingus/src/util/json_writer_impl.cpp pingus.gcc9/src/util/json_writer_impl.cpp
--- pingus/src/util/json_writer_impl.cpp	2020-02-02 01:56:40.365805177 +0000
+++ pingus.gcc9/src/util/json_writer_impl.cpp	2020-02-02 01:56:53.406355546 +0000
@@ -19,6 +19,7 @@
 #include <sstream>
 #include <ostream>
 #include <assert.h>
+#include <memory>
 
 #include "math/color.hpp"
 #include "math/size.hpp"
