From 3bbe75f79eb81c45e51ec8a0705547bc3d4c3921 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Fri, 24 Oct 2025 09:59:04 +0800
Subject: [PATCH 2/2] fix: weird version mismatch in CMakeLists.txt

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b7083e..6f85bc5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.16)
 project(
     hotspot
-    VERSION 1.5.80
+    VERSION 1.5.1
     LANGUAGES CXX C
 )
 
-- 
2.51.0

