From ec869a6edea6dbe12452f1f9c37dee810da9c1b0 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Thu, 13 Aug 2020 12:27:24 +0800
Subject: [PATCH] Let pkgconfig find `lua5.3`

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 37acf38..5453b27 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/lua5.3/lua.h")
   find_package(PkgConfig)
   if(PkgConfig_FOUND)
-    foreach(pkg luajit lua lua53 lua52 lua51)
+    foreach(pkg luajit lua lua5.3 lua53 lua52 lua51)
       pkg_check_modules(LUA ${pkg})
       if(LUA_FOUND)
 	break()
-- 
2.27.0

