diff --git a/gnome-flashback/libinput-sources/gf-ibus-manager.c b/gnome-flashback/libinput-sources/gf-ibus-manager.c
index ce41132..9ee48aa 100644
--- a/gnome-flashback/libinput-sources/gf-ibus-manager.c
+++ b/gnome-flashback/libinput-sources/gf-ibus-manager.c
@@ -163,8 +163,6 @@ list_engines_cb (GObject      *source_object,
     {
       g_warning ("Failed to list ibus engines - %s", error->message);
       g_error_free (error);
-
-      clear (manager);
     }
 }
 
