diff --git a/configure b/configure
index 82053df..cf6ceea 100755
--- a/configure
+++ b/configure
@@ -1,3 +1,3 @@
 #!/bin/sh
 srcdir=`dirname $0`
-exec $srcdir/configure.orig --without-debug --with-strip --with-openmp --with-mt --without-vdb --without-gnutls --without-gcrypt --with-build-root=$srcdir/ReleaseMT --with-experimental=Int8GI ${1+"$@"}
+exec $srcdir/configure.orig --with-openmp --with-mt --without-vdb --without-gnutls --without-gcrypt --with-build-root=$srcdir/ReleaseMT --with-experimental=Int8GI ${1+"$@"}
