From: Josselin Mouette <joss@debian.org>
Subject: Add support for switching tabs with alt+scroll
Bug-Debian: https://bugs.debian.org/640980
Index: webkitgtk/Source/WebKit/gtk/webkit/webkitwebview.cpp
===================================================================
--- webkitgtk.orig/Source/WebKit/gtk/webkit/webkitwebview.cpp
+++ webkitgtk/Source/WebKit/gtk/webkit/webkitwebview.cpp
@@ -813,6 +813,12 @@ static gboolean webkit_web_view_scroll_e
     if (!frame.view())
         return FALSE;
 
+#ifndef GTK_API_VERSION_2
+    /* Ignore Alt-scroll events in GTK3 so that GtkNotebook can use it to switch tabs */
+    if (event->state & GDK_MOD1_MASK)
+        return FALSE;
+#endif
+
     PlatformWheelEvent wheelEvent(event);
     return frame.eventHandler().handleWheelEvent(wheelEvent);
 }
