From: Alberto Garcia <berto@igalia.com>
Subject: Disable JIT on x86 CPUs without SSE2
Bug: https://bugs.webkit.org/show_bug.cgi?id=119190
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=783293
Index: webkitgtk/Source/JavaScriptCore/runtime/VM.cpp
===================================================================
--- webkitgtk.orig/Source/JavaScriptCore/runtime/VM.cpp
+++ webkitgtk/Source/JavaScriptCore/runtime/VM.cpp
@@ -134,6 +134,11 @@ static bool enableAssembler(ExecutableAl
         return false;
     }
 
+#if CPU(X86)
+    if (!MacroAssembler::supportsFloatingPoint())
+        return false;
+#endif
+
 #if USE(CF)
 #if COMPILER(GCC) && !COMPILER(CLANG)
     // FIXME: remove this once the EWS have been upgraded to LLVM.
