From: Michael Catanzaro <mcatanzaro@igalia.com>
Subject: Fix FTBFS with GCC 6
Bug: https://bugs.webkit.org/show_bug.cgi?id=159124
Bug-Debian: https://bugs.debian.org/828198
Index: webkitgtk/Source/WebCore/platform/gtk/GtkClickCounter.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/platform/gtk/GtkClickCounter.cpp
+++ webkitgtk/Source/WebCore/platform/gtk/GtkClickCounter.cpp
@@ -85,8 +85,8 @@ int GtkClickCounter::clickCountForGdkBut
     guint32 eventTime = getEventTime(event);
 
     if ((event->type == GDK_2BUTTON_PRESS || event->type == GDK_3BUTTON_PRESS)
-        || ((abs(buttonEvent->x - m_previousClickPoint.x()) < doubleClickDistance)
-            && (abs(buttonEvent->y - m_previousClickPoint.y()) < doubleClickDistance)
+        || ((fabs(buttonEvent->x - m_previousClickPoint.x()) < doubleClickDistance)
+            && (fabs(buttonEvent->y - m_previousClickPoint.y()) < doubleClickDistance)
             && (eventTime - m_previousClickTime < static_cast<guint>(doubleClickTime))
             && (buttonEvent->button == m_previousClickButton)))
         m_currentClickCount++;
Index: webkitgtk/Source/WebCore/platform/gtk/LocalizedStringsGtk.cpp
===================================================================
--- webkitgtk.orig/Source/WebCore/platform/gtk/LocalizedStringsGtk.cpp
+++ webkitgtk/Source/WebCore/platform/gtk/LocalizedStringsGtk.cpp
@@ -659,7 +659,7 @@ String localizedMediaTimeDescription(flo
     if (!std::isfinite(time))
         return String::fromUTF8(_("indefinite time"));
 
-    int seconds = static_cast<int>(abs(time));
+    int seconds = static_cast<int>(fabs(time));
     int days = seconds / (60 * 60 * 24);
     int hours = seconds / (60 * 60);
     int minutes = (seconds / 60) % 60;
