From: Alberto Garcia <berto@igalia.com>
Subject: Fix FTBFS in armhf
Bug: https://bugs.webkit.org/show_bug.cgi?id=130507
Index: webkitgtk/Source/JavaScriptCore/GNUmakefile.am
===================================================================
--- webkitgtk.orig/Source/JavaScriptCore/GNUmakefile.am
+++ webkitgtk/Source/JavaScriptCore/GNUmakefile.am
@@ -80,9 +80,11 @@ javascriptcore_cppflags += \
 	-fno-omit-frame-pointer
 
 if COMPILER_GCC
+if !TARGET_ARMHF
 javascriptcore_cppflags += \
 	-fno-tree-dce
 endif
+endif
 
 javascriptcore_cflags += \
 	$(global_cflags) \
Index: webkitgtk/Source/autotools/SetupAutomake.m4
===================================================================
--- webkitgtk.orig/Source/autotools/SetupAutomake.m4
+++ webkitgtk/Source/autotools/SetupAutomake.m4
@@ -67,3 +67,5 @@ AM_CONDITIONAL([ENABLE_SPELLCHECK],[test
 
 # Introspection conditional.
 AM_CONDITIONAL([ENABLE_INTROSPECTION],[test "$enable_introspection" = "yes"])
+
+AM_CONDITIONAL([TARGET_ARMHF],[test "$build" = "arm-unknown-linux-gnueabihf"])
