From: Alberto Garcia <berto@igalia.com>
Subject: Install the MiniBrowser binary
Forwarded: not-needed
Index: webkitgtk/Tools/MiniBrowser/gtk/main.c
===================================================================
--- webkitgtk.orig/Tools/MiniBrowser/gtk/main.c
+++ webkitgtk/Tools/MiniBrowser/gtk/main.c
@@ -265,8 +265,6 @@ int main(int argc, char *argv[])
     }
     g_option_context_free (context);
 
-    g_setenv("WEBKIT_INJECTED_BUNDLE_PATH", WEBKIT_INJECTED_BUNDLE_PATH, FALSE);
-
     // Enable the favicon database, by specifying the default directory.
     webkit_web_context_set_favicon_database_directory(webkit_web_context_get_default(), NULL);
 
Index: webkitgtk/Tools/MiniBrowser/gtk/GNUmakefile.am
===================================================================
--- webkitgtk.orig/Tools/MiniBrowser/gtk/GNUmakefile.am
+++ webkitgtk/Tools/MiniBrowser/gtk/GNUmakefile.am
@@ -1,5 +1,5 @@
 if ENABLE_WEBKIT2
-noinst_PROGRAMS += \
+libexec_PROGRAMS += \
 	Programs/MiniBrowser
 endif
 
@@ -44,9 +44,6 @@ Programs_MiniBrowser_LDADD = \
 	$(GTK_LIBS) \
 	$(LIBSOUP_LIBS)
 
-Programs_MiniBrowser_LDFLAGS = \
-	-no-install
-
 CLEANFILES += \
 	$(top_builddir)/Programs/MiniBrowser \
 	$(minibrowser_built_sources) \
