From 70dc73695af00883ef85e966f764e69910a5c6bc Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 9 May 2022 15:07:06 +0800
Subject: [PATCH] disks: fix new_partition_table in ppc64/el

---
 src/disks.rs | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/src/disks.rs b/src/disks.rs
index 768ee02..880a4b7 100644
--- a/src/disks.rs
+++ b/src/disks.rs
@@ -204,13 +204,15 @@ pub fn new_partition_table() -> Result<()> {
     use crate::network;
 
     let arch_name = network::get_arch_name();
-    if libparted::Disk::new(&mut i).is_err() && arch_name == Some("ppc64el") {
-        let disk = libparted::Disk::new_fresh(
-            &mut i,
-            DiskType::get("gpt").ok_or_else(|| anyhow!("Unsupport partition table type!"))?,
-        );
-        if let Ok(mut disk) = disk {
-            disk.commit_to_dev().ok();
+    for mut i in libparted::Device::devices(true) {
+        if libparted::Disk::new(&mut i).is_err() && arch_name == Some("ppc64el") {
+            let disk = libparted::Disk::new_fresh(
+                &mut i,
+                DiskType::get("gpt").ok_or_else(|| anyhow!("Unsupport partition table type!"))?,
+            );
+            if let Ok(mut disk) = disk {
+                disk.commit_to_dev().ok();
+            }
         }
     }
 
-- 
2.35.1

