From 5a86932eb2cb86cf5b44e968ca4272be5174ac2c Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sun, 1 May 2022 22:26:18 +0800
Subject: [PATCH] network: fix nix::libc import

---
 src/network.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/network.rs b/src/network.rs
index d16167b..7905e1b 100644
--- a/src/network.rs
+++ b/src/network.rs
@@ -97,6 +97,7 @@ pub fn fetch_mirrors(recipe: &Recipe) -> Vec<Mirror> {
 #[cfg(target_arch = "powerpc64")]
 #[inline]
 pub(crate) fn get_arch_name() -> Option<&'static str> {
+    use nix::libc;
     let mut endian: libc::c_int = -1;
     let result;
     unsafe {
