From 2c73b0efc52ed3055ffe0b99d073384136454980 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 25 Nov 2024 15:09:30 +0800
Subject: [PATCH 85/88] libiberty: also generate a PIC version of libiberty.a

---
 libiberty/configure | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/libiberty/configure b/libiberty/configure
index 860f981fa18..770e6fa52fd 100755
--- a/libiberty/configure
+++ b/libiberty/configure
@@ -5263,10 +5263,6 @@ if [ "${enable_host_shared}" = "yes" ]; then
   shared=yes
 fi
 
-if [ "${shared}" != "yes" ]; then
-  PICFLAG=
-fi
-
 
 NOASANFLAG=
 case " ${CFLAGS} " in
-- 
2.48.1

