diff --git a/src/Cutter.cpp b/src/Cutter.cpp
index 09cc33de..39b76140 100644
--- a/src/Cutter.cpp
+++ b/src/Cutter.cpp
@@ -111,7 +111,7 @@ QList<QString> CutterCore::sdbListKeys(QString path)
         SdbList *l = sdb_foreach_list(root, false);
         ls_foreach(l, iter, vsi) {
             SdbKv *nsi = (SdbKv *)vsi;
-            list << nsi->key;
+            list << nsi->base.key;
         }
     }
     return list;
