diff -up qwtplot3d/include/qwt3d_io_gl2ps.h.qt48 qwtplot3d/include/qwt3d_io_gl2ps.h
--- qwtplot3d/include/qwt3d_io_gl2ps.h.qt48	2005-07-10 09:54:52.000000000 -0500
+++ qwtplot3d/include/qwt3d_io_gl2ps.h	2012-06-27 13:10:09.056248812 -0500
@@ -6,7 +6,7 @@
 #if QT_VERSION < 0x040000
 #include <qgl.h>
 #else
-#include <QtOpenGL/qgl.h>
+//#include <QtOpenGL/qgl.h>
 #endif
 
 #include "qwt3d_types.h"
diff -up qwtplot3d/include/qwt3d_openglhelper.h.qt48 qwtplot3d/include/qwt3d_openglhelper.h
--- qwtplot3d/include/qwt3d_openglhelper.h.qt48	2005-07-19 10:40:28.000000000 -0500
+++ qwtplot3d/include/qwt3d_openglhelper.h	2012-06-27 13:11:47.914012939 -0500
@@ -7,6 +7,7 @@
 #else
 #include <QtOpenGL/qgl.h>
 #endif
+#include <GL/glu.h>
 
 namespace Qwt3D
 {
