diff -up lapack-3.7.1/INSTALL/make.inc.gfortran.shared lapack-3.7.1/INSTALL/make.inc.gfortran
--- lapack-3.7.1/INSTALL/make.inc.gfortran.shared	2017-08-01 11:46:10.665067382 -0400
+++ lapack-3.7.1/INSTALL/make.inc.gfortran	2017-08-01 11:46:10.667067346 -0400
@@ -83,3 +83,4 @@ CBLASLIB     = ../../libcblas.a
 LAPACKLIB    = liblapack.a
 TMGLIB       = libtmglib.a
 LAPACKELIB   = liblapacke.a
+SHLIB_LAPACKE = liblapacke.so
diff -up lapack-3.7.1/LAPACKE/Makefile.shared lapack-3.7.1/LAPACKE/Makefile
--- lapack-3.7.1/LAPACKE/Makefile.shared	2017-08-01 11:46:10.668067328 -0400
+++ lapack-3.7.1/LAPACKE/Makefile	2017-08-01 11:47:23.649755395 -0400
@@ -51,6 +51,9 @@ lapacke: include/lapacke_mangling.h
 include/lapacke_mangling.h: include/lapacke_mangling_with_flags.h.in
 	cp $< $@
 
+shlib: lapacke
+	$(CC) $(CFLAGS) -shared -Wl,-soname,$(SHLIB_LAPACKE).@SHORTVER@ -o $(SHLIB_LAPACKE).@LONGVER@ src/*.o utils/*.o $(LIBS) -lgfortran -lc
+
 lapacke_example: lapacke
 	$(MAKE) -C example
 
