From 3d58172f9352bfd2b5089bc3299e3ac3150eb9b1 Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Fri, 2 Aug 2024 22:29:46 +0800
Subject: [PATCH] Makefile: Add option to skip mandb refresh

Signed-off-by: Bingwu Zhang <xtexchooser@duck.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3330a1ded823..8d40c5de176a 100644
--- a/Makefile
+++ b/Makefile
@@ -41,7 +41,7 @@ install: snag_docs #fixup_locale
 	ln -sf "$(bindir)/git-filter-repo" "$(DESTDIR)/$(pythondir)/git_filter_repo.py"
 	$(INSTALL) -Dm0644 Documentation/man1/git-filter-repo.1 "$(DESTDIR)/$(mandir)/man1/git-filter-repo.1"
 	$(INSTALL) -Dm0644 Documentation/html/git-filter-repo.html "$(DESTDIR)/$(htmldir)/git-filter-repo.html"
-	if which mandb > /dev/null; then mandb; fi
+	$(if $(no-refresh-mandb),,if which mandb > /dev/null; then mandb; fi)
 
 
 #

base-commit: e1b7c3d1eae4b8e724ad81b584d20fffe78dd092
-- 
2.46.0

