diff --git a/man/man1/Makefile.am b/man/man1/Makefile.am
index f2274d7..d98ae8c 100644
--- a/man/man1/Makefile.am
+++ b/man/man1/Makefile.am
@@ -1,3 +1,12 @@
-man1_MANS=pkcsconf.1 pkcsicsf.1 pkcsep11_migrate.1 pkcscca.1
+man1_MANS=pkcsconf.1 pkcsicsf.1
+
+if ENABLE_PKCSEP11_MIGRATE
+man1_MANS += pkcsep11_migrate.1
+endif
+
+if ENABLE_CCATOK
+man1_MANS += pkcscca.1
+endif
+
 EXTRA_DIST = $(man1_MANS)
 CLEANFILES = $(man1_MANS)
