--- a/runtime/bin/ffi_test/ffi_test_functions.cc
+++ b/runtime/bin/ffi_test/ffi_test_functions.cc
@@ -15,6 +15,7 @@
 
 #include <cinttypes>
 #include <cmath>
+#include <cstdint>
 #include <iostream>
 #include <limits>
 #include <thread>  // NOLINT
diff --git a/runtime/bin/ffi_test/ffi_test_functions_generated.cc b/runtime/bin/ffi_test/ffi_test_functions_generated.cc
index 3c58f727..7022a30f 100644
--- a/runtime/bin/ffi_test/ffi_test_functions_generated.cc
+++ b/runtime/bin/ffi_test/ffi_test_functions_generated.cc
@@ -12,6 +12,7 @@
 #include <sys/types.h>
 
 #include <cmath>
+#include <cstdint>
 #include <iostream>
 #include <limits>
 
