From 368190b4ea9458e0ae95e3243d7cc0f61d3136fe Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Wed, 1 Oct 2025 00:49:09 -0400
Subject: [PATCH 2/2] remoteSearch.js: move DesktopAppInfo to GioUnix

---
 js/ui/remoteSearch.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/js/ui/remoteSearch.js b/js/ui/remoteSearch.js
index 8ebb55052..54b494003 100644
--- a/js/ui/remoteSearch.js
+++ b/js/ui/remoteSearch.js
@@ -1,7 +1,7 @@
 // -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
 /* exported loadRemoteSearchProviders */
 
-const { GdkPixbuf, Gio, GLib, Shell, St } = imports.gi;
+const { GdkPixbuf, Gio, GioUnix, GLib, Shell, St } = imports.gi;
 
 const FileUtils = imports.misc.fileUtils;
 
@@ -89,7 +89,7 @@ function loadRemoteSearchProviders(searchSettings, callback) {
             let appInfo = null;
             try {
                 let desktopId = keyfile.get_string(group, 'DesktopId');
-                appInfo = Gio.DesktopAppInfo.new(desktopId);
+                appInfo = GioUnix.DesktopAppInfo.new(desktopId);
                 if (!appInfo.should_show())
                     return;
             } catch (e) {
-- 
2.51.0

