diff -up telepathy-qt-0.9.5/TelepathyQt/CMakeLists.txt.static_fpic telepathy-qt-0.9.5/TelepathyQt/CMakeLists.txt
--- telepathy-qt-0.9.5/TelepathyQt/CMakeLists.txt.static_fpic	2015-02-23 11:42:16.889403849 -0600
+++ telepathy-qt-0.9.5/TelepathyQt/CMakeLists.txt	2015-02-23 11:41:54.782360989 -0600
@@ -943,6 +943,7 @@ if(ENABLE_SERVICE_SUPPORT)
     #endif (ENABLE_COMPILER_COVERAGE)
     # lets build a static only library until we have a stable API/ABI
     add_library(telepathy-qt${QT_VERSION_MAJOR}-service STATIC ${telepathy_qt_service_SRCS})
+    set_target_properties(telepathy-qt${QT_VERSION_MAJOR}-service PROPERTIES COMPILE_FLAGS "${CMAKE_SHARED_LIBRARY_C_FLAGS}")
 
     # generate service moc files
     foreach(moc_src ${telepathy_qt_service_MOC_SRCS})

