--- libofa-0.9.3/examples/wavefile.cpp.orig	2006-05-10 20:05:56.000000000 +0200
+++ libofa-0.9.3/examples/wavefile.cpp	2006-08-22 15:12:07.000000000 +0200
@@ -42,7 +42,11 @@
     int srate = 0;
     int channels = 0;
 
-    int fd = open(file, O_RDONLY | 0x8000);
+#ifdef WIN32
+    int fd = open(file, O_RDONLY | O_BINARY);
+#else
+    int fd = open(file, O_RDONLY);
+#endif
     if (fd == -1)
 	return 0;
 
