diff -Naur treefrog-framework-1.19.0/configure treefrog-framework-1.19.0.nothird/configure
--- treefrog-framework-1.19.0/configure	2017-09-20 07:20:16.000000000 +0000
+++ treefrog-framework-1.19.0.nothird/configure	2017-11-18 18:19:42.312548591 +0000
@@ -165,21 +165,6 @@
 $QMAKE -recursive $OPT target.path=\"$BINDIR\" header.path=\"$INCLUDEDIR\" datadir=\"$DATADIR\" lib.path=\"$LIBDIR\"
 make qmake
 
-# compile MongoDB files
-cd "$BASEDIR/3rdparty/mongo-c-driver"
-echo
-echo -n "Compiling MongoDB driver library ... "
-$QMAKE $OPT mongo-c-driver.pro
-make clean >/dev/null 2>&1
-make >/dev/null 2>&1
-RET=$?
-if [ $RET != 0 ]; then
-  echo "failed"
-  echo "MongoDB driver not available."
-  exit $RET
-fi
-echo "OK"
-
 echo
 echo "First, run \"make\" and \"sudo make install\" in src directory."
 echo "Next, run \"make\" and \"sudo make install\" in tools directory."
