diff -rupN dieharder-3.31.1/include/dieharder/libdieharder.h dieharder-3.31.1_patch/include/dieharder/libdieharder.h
--- dieharder-3.31.1/include/dieharder/libdieharder.h	2011-10-14 15:41:37.000000000 +0200
+++ dieharder-3.31.1_patch/include/dieharder/libdieharder.h	2014-07-14 16:17:28.501427405 +0200
@@ -14,6 +14,7 @@
 #include <stdarg.h>
 #include <string.h>
 #include <sys/time.h>
+#include <stdint.h>
 
 /* This turns on uint macro in c99 */
 #define __USE_MISC 1
