diff -Naur mariadb-10.5.9.orig/libmariadb/cmake/ConnectorName.cmake mariadb-10.5.9/libmariadb/cmake/ConnectorName.cmake
--- mariadb-10.5.9.orig/libmariadb/cmake/ConnectorName.cmake	2021-09-13 16:37:38.857516544 +0800
+++ mariadb-10.5.9/libmariadb/cmake/ConnectorName.cmake	2021-09-13 16:37:48.819644531 +0800
@@ -22,7 +22,7 @@
     SET(MACHINE_NAME "x64")
   ELSE()
     SET(MACHINE_NAME "32")
-  END()
+  ENDIF()
 ENDIF()
 
 SET(product_name "mysql-connector-c-${CPACK_PACKAGE_VERSION}-${PLATFORM_NAME}${CONCAT_SIGN}${MACHINE_NAME}")
