From 340ef7eeae56cb0caea8d6bb6bc373e290f2b477 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Fri, 4 Apr 2025 13:32:15 +0800
Subject: [PATCH 1/4] Remove unnecessary `nix` and `libc` dependency

---
 Cargo.lock | 1 -
 Cargo.toml | 1 -
 2 files changed, 2 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 95e9449..730008f 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -143,7 +143,6 @@ version = "1.0.5"
 dependencies = [
  "anyhow",
  "interfaces",
- "libc",
  "log",
  "log4rs",
  "once_cell",
diff --git a/Cargo.toml b/Cargo.toml
index ebf71e9..7e47838 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -30,7 +30,6 @@ windows-service = "0.8.0"
 tokio = { version = "1.44.0", features = ["full"] }
 serde = { version = "1.0.217", features = ["derive"] }
 sysinfo = "0.33.1"
-libc = "0.2.171"
 
 [target.'cfg(target_os = "linux")'.dependencies]
 openssl = { version = "0.10.71", features = ["vendored"] }
-- 
2.49.0

