From 3518d307f47c79db06bd0e9430f39b91488673dc Mon Sep 17 00:00:00 2001
From: Hyunjin Song <tteu.ingog@gmail.com>
Date: Tue, 24 May 2022 09:57:09 +0900
Subject: [PATCH] Fix Carla version detction in CarlaBase (#6403)

---
 plugins/carlabase/Carla.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plugins/carlabase/carla.h b/plugins/carlabase/carla.h
index da36147a3..98662014b 100644
--- a/plugins/carlabase/carla.h
+++ b/plugins/carlabase/carla.h
@@ -36,6 +36,7 @@
 
 // carla/source/includes
 #include "carlabase_export.h"
+#include "CarlaDefines.h"
 #if CARLA_VERSION_HEX >= 0x010911
     #include "CarlaNativePlugin.h"
 #else
-- 
2.36.0

