From 32b2beaf2197f841b76358e1fa5f5868d0de46e1 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Mon, 1 Jan 2018 21:28:24 -0700
Subject: [PATCH] LadspaEffect/tap: add linking flags

---
 plugins/LadspaEffect/tap/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/LadspaEffect/tap/CMakeLists.txt b/plugins/LadspaEffect/tap/CMakeLists.txt
index 309283da6..aaebc8827 100644
--- a/plugins/LadspaEffect/tap/CMakeLists.txt
+++ b/plugins/LadspaEffect/tap/CMakeLists.txt
@@ -16,7 +16,7 @@ FOREACH(_item ${PLUGIN_SOURCES})
 		SET_TARGET_PROPERTIES("${_plugin}" PROPERTIES LINK_FLAGS "${LINK_FLAGS} -shared -Wl,-no-undefined -Wl,-Bsymbolic -lm")
 	ENDIF(LMMS_BUILD_APPLE)
 	IF(LMMS_BUILD_LINUX OR LMMS_BUILD_HAIKU)
-		SET_TARGET_PROPERTIES("${_plugin}" PROPERTIES LINK_FLAGS "${LINK_FLAGS}")
+		SET_TARGET_PROPERTIES("${_plugin}" PROPERTIES LINK_FLAGS "${LINK_FLAGS} -lmvec")
 	ENDIF(LMMS_BUILD_LINUX OR LMMS_BUILD_HAIKU)
 ENDFOREACH(_item ${PLUGIN_SOURCES})
 
-- 
2.15.1

