diff -Naur goom2k4-0/configure.in goom2k4-0.noppc/configure.in
--- goom2k4-0/configure.in	2005-02-07 13:46:41.000000000 +0000
+++ goom2k4-0.noppc/configure.in	2018-02-13 03:37:47.960962417 +0000
@@ -62,14 +62,6 @@
 	AC_DEFINE(CPU_X86)
 	HAVE_MMX="yes"
 	;;
-
-powerpc-*-*)
-  	CCASFLAGS=-force_cpusubtype_ALL
-  	AC_SUBST(CCASFLAGS)
-	AC_DEFINE(CPU_POWERPC)
-	HAVE_PPC="yes"
-	;;
-
 esac
 AM_CONDITIONAL(HAVE_MMX,test "x$HAVE_MMX" = "xyes")
 AM_CONDITIONAL(HAVE_PPC,test "x$HAVE_PPC" = "xyes")
