From 346c1daaee72de38fdbb07461e4d39cf1363989b Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sun, 8 Nov 2020 23:48:08 -0600
Subject: [PATCH] fix arm64 build

---
 Cargo.toml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Cargo.toml b/Cargo.toml
index eccd733..3a12da8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,3 +23,6 @@ opt-level = 'z' # Optimize for size
 codegen-units = 1
 lto = true
 opt-level = 'z' # Optimize for size
+
+[patch.crates-io]
+rusty_v8 = { path = "../rusty_v8" }
-- 
2.28.0

