From e9d0acae1506a79274d37a1b6e02bbfd432e6298 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 11 Aug 2024 12:16:31 +0800
Subject: [PATCH 1/3] [Debian] fix: burn{K6,K7,P5,P6}: define variables as
 .data section

From Debian: cpuburn (1.4-20) unstable.
---
 burnK6.S | 1 +
 burnK7.S | 1 +
 burnP5.S | 1 +
 burnP6.S | 1 +
 4 files changed, 4 insertions(+)

diff --git a/burnK6.S b/burnK6.S
index ad86042..a7425bf 100644
--- a/burnK6.S
+++ b/burnK6.S
@@ -68,6 +68,7 @@ int_exit:
 	push	%eax
 	int	$0x80
 #endif
+.data				# Data allocation
 .align 32,0
 half:	.long 0x7fffffff,0
 e:	.long 0xffffffff,0x3fdfffff
diff --git a/burnK7.S b/burnK7.S
index dd08a30..9d41871 100644
--- a/burnK7.S
+++ b/burnK7.S
@@ -74,6 +74,7 @@ int_exit:
 	push	%eax
 	int	$0x80
 #endif
+.data				# Data allocation
 .align 32,0
 	.fill 64
 half:	.long 0x7fffffff,0
diff --git a/burnP5.S b/burnP5.S
index a47af74..c55dbc8 100644
--- a/burnP5.S
+++ b/burnP5.S
@@ -77,6 +77,7 @@ crunch:
 #else
 	int	$0x80
 #endif
+.data				# Data allocation
 .align 32,0
 half:	.long 0xffffffff,0x3fdfffff
 one:	.long 0xffffffff,0x3fefffff
diff --git a/burnP6.S b/burnP6.S
index eaf331f..017c481 100644
--- a/burnP6.S
+++ b/burnP6.S
@@ -69,6 +69,7 @@ int_exit:			# error abort
 	push	%eax		# *BSD syscall
 	int	$0x80
 #endif
+.data				# Data allocation
 .align 32,0
 half:	.long 0x7fffffff,0
 e:	.long 0xffffffff,0x3fdfffff
-- 
2.46.0

