From c869026639d38eab8342e428788f2076a657ca48 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Fri, 21 Oct 2022 18:06:45 -0700
Subject: [PATCH] neofetch: (AOSC OS) print dpkg architecture where possible

---
 neofetch | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/neofetch b/neofetch
index 48b96d21..341aca51 100755
--- a/neofetch
+++ b/neofetch
@@ -4893,7 +4893,11 @@ cache_uname() {
 
     kernel_name="${uname[0]}"
     kernel_version="${uname[1]}"
-    kernel_machine="${uname[2]}"
+    if command -v dpkg > /dev/null; then
+        kernel_machine=$(dpkg --print-architecture)
+    else
+        kernel_machine="${uname[2]}"
+    fi
 
     if [[ "$kernel_name" == "Darwin" ]]; then
         # macOS can report incorrect versions unless this is 0.
-- 
2.37.1
