From 5b7bf6cf969429b5068dc151018882ecfb129a56 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Fri, 21 Oct 2022 18:06:45 -0700
Subject: [PATCH 1/2] neofetch: (AOSC OS) print dpkg architecture where
 possible

---
 neofetch | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/neofetch b/neofetch
index c2f56f04..a07a8e68 100755
--- a/neofetch
+++ b/neofetch
@@ -6195,7 +6195,11 @@ cache_uname() {
 
     kernel_name="${uname[0]}"
     kernel_version="${uname[1]}"
-    kernel_machine="${uname[2]}"
+    if command -v dpkg > /dev/null; then
+        kernel_machine=$(dpkg --print-architecture)
+    else
+        kernel_machine="${uname[2]}"
+    fi
 
     if [[ "$kernel_name" == "Darwin" ]] ||
        [[ "$kernel_name" == "FreeBSD" && -f /System/Library/CoreServices/SystemVersion.plist ]]; then
-- 
2.44.0

