From 6916763b8e9be69cac3cd20eb4344d39460f45df Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Fri, 21 Oct 2022 18:16:03 -0700
Subject: [PATCH 2/2] neofetch: (AOSC OS) add quotation marks around
 architecture name

---
 neofetch | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/neofetch b/neofetch
index a07a8e68..0dc4bc97 100755
--- a/neofetch
+++ b/neofetch
@@ -1496,7 +1496,7 @@ get_distro() {
     esac
 
     [[ $os_arch == on ]] && \
-        distro+=" $machine_arch"
+        distro+=" ($machine_arch)"
 
     [[ ${ascii_distro:-auto} == auto ]] && \
         ascii_distro=$(trim "$distro")
-- 
2.44.0

