From 3c3c74f5abc7f72612c9b708372b4f22c4429782 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Fri, 21 Oct 2022 18:16:03 -0700
Subject: [PATCH 2/2] neofetch: (AOSC OS) add quotation marks around
 architecture name

---
 neofetch | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/neofetch b/neofetch
index 9e30ddef..0be4f4e1 100755
--- a/neofetch
+++ b/neofetch
@@ -1538,7 +1538,7 @@ get_distro() {
     esac
 
     [[ $os_arch == on ]] && \
-        distro+=" $machine_arch"
+        distro+=" ($machine_arch)"
 
     [[ ${ascii_distro:-auto} == auto ]] && \
         ascii_distro=$(trim "$distro")
-- 
2.51.0

