diff -Naur bleachbit-3.0/bleachbit/GUI.py bleachbit-3.0.noupdate/bleachbit/GUI.py
--- bleachbit-3.0/bleachbit/GUI.py	2019-10-24 02:12:57.000000000 +0000
+++ bleachbit-3.0.noupdate/bleachbit/GUI.py	2020-01-21 20:36:17.677437284 +0000
@@ -744,12 +744,6 @@
         # expand tree view
         self.view.expand_all()
 
-        # Check for online updates.
-        if bleachbit.online_update_notification_enabled and options.get("check_online_updates") and \
-                not hasattr(self, 'checked_for_updates'):
-            self.checked_for_updates = True
-            self.check_online_updates()
-
         # remove from idle loop (see GObject.idle_add)
         return False
 
@@ -1042,18 +1036,3 @@
         # done
         self.show_all()
         self.progressbar.hide()
-
-    @threaded
-    def check_online_updates(self):
-        """Check for software updates in background"""
-        from bleachbit import Update
-        try:
-            updates = Update.check_updates(options.get('check_beta'),
-                                           options.get('update_winapp2'),
-                                           self.append_text,
-                                           lambda: GLib.idle_add(self.cb_refresh_operations))
-            if updates:
-                GLib.idle_add(
-                    lambda: Update.update_dialog(self, updates))
-        except Exception:
-            logger.exception(_("Error when checking for updates: "))
diff -Naur bleachbit-3.0/bleachbit/GuiPreferences.py bleachbit-3.0.noupdate/bleachbit/GuiPreferences.py
--- bleachbit-3.0/bleachbit/GuiPreferences.py	2019-10-24 02:12:57.000000000 +0000
+++ bleachbit-3.0.noupdate/bleachbit/GuiPreferences.py	2020-01-21 20:36:52.095642702 +0000
@@ -26,7 +26,6 @@
 
 from __future__ import absolute_import
 
-from bleachbit import _, _p, online_update_notification_enabled
 from bleachbit.Options import options
 from bleachbit import GuiBasic
 
@@ -84,10 +83,6 @@
     def __toggle_callback(self, cell, path):
         """Callback function to toggle option"""
         options.toggle(path)
-        if online_update_notification_enabled:
-            self.cb_beta.set_sensitive(options.get('check_online_updates'))
-            if 'nt' == os.name:
-                self.cb_winapp2.set_sensitive(options.get('check_online_updates'))
         if 'auto_hide' == path:
             self.refresh_operations = True
         if 'dark_mode' == path:
@@ -101,37 +96,6 @@
 
         vbox = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
 
-        if online_update_notification_enabled:
-            cb_updates = Gtk.CheckButton.new_with_label(
-                _("Check periodically for software updates via the Internet"))
-            cb_updates.set_active(options.get('check_online_updates'))
-            cb_updates.connect(
-                'toggled', self.__toggle_callback, 'check_online_updates')
-            cb_updates.set_tooltip_text(
-                _("If an update is found, you will be given the option to view information about it.  Then, you may manually download and install the update."))
-            vbox.pack_start(cb_updates, False, True, 0)
-
-            updates_box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL)
-            updates_box.set_border_width(10)
-
-            self.cb_beta = Gtk.CheckButton.new_with_label(label=_("Check for new beta releases"))
-            self.cb_beta.set_active(options.get('check_beta'))
-            self.cb_beta.set_sensitive(options.get('check_online_updates'))
-            self.cb_beta.connect(
-                'toggled', self.__toggle_callback, 'check_beta')
-            updates_box.pack_start(self.cb_beta, False, True, 0)
-
-            if 'nt' == os.name:
-                self.cb_winapp2 = Gtk.CheckButton.new_with_label(
-                    _("Download and update cleaners from community (winapp2.ini)"))
-                self.cb_winapp2.set_active(options.get('update_winapp2'))
-                self.cb_winapp2.set_sensitive(
-                    options.get('check_online_updates'))
-                self.cb_winapp2.connect(
-                    'toggled', self.__toggle_callback, 'update_winapp2')
-                updates_box.pack_start(self.cb_winapp2, False, True, 0)
-            vbox.pack_start(updates_box, False, True, 0)
-
         # TRANSLATORS: This means to hide cleaners which would do
         # nothing.  For example, if Firefox were never used on
         # this system, this option would hide Firefox to simplify
diff -Naur bleachbit-3.0/bleachbit/Options.py bleachbit-3.0.noupdate/bleachbit/Options.py
--- bleachbit-3.0/bleachbit/Options.py	2019-10-24 02:12:57.000000000 +0000
+++ bleachbit-3.0.noupdate/bleachbit/Options.py	2020-01-21 20:37:23.467044661 +0000
@@ -41,7 +41,6 @@
 
 boolean_keys = ['auto_hide',
                 'check_beta',
-                'check_online_updates',
                 'dark_mode',
                 'delete_confirmation',
                 'debug',
@@ -268,7 +267,6 @@
         # set defaults
         self.__set_default("auto_hide", True)
         self.__set_default("check_beta", False)
-        self.__set_default("check_online_updates", True)
         self.__set_default("dark_mode", True)
         self.__set_default("delete_confirmation", True)
         self.__set_default("debug", False)
diff -Naur bleachbit-3.0/bleachbit/__init__.py bleachbit-3.0.noupdate/bleachbit/__init__.py
--- bleachbit-3.0/bleachbit/__init__.py	2019-10-24 02:12:57.000000000 +0000
+++ bleachbit-3.0.noupdate/bleachbit/__init__.py	2020-01-21 20:37:36.384288435 +0000
@@ -49,10 +49,6 @@
 
 logger = Log.init_log()
 
-# Setting below value to false disables update notification (useful
-# for packages in repositories).
-online_update_notification_enabled = True
-
 #
 # Paths
 #
