From 09791ff0aa5d006d602f593d65c357da25faad84 Mon Sep 17 00:00:00 2001
From: qzed <qzed@users.noreply.github.com>
Date: Tue, 17 Sep 2019 17:16:23 +0200
Subject: [PATCH 08/60] hid: register Surface hid devices

---
 drivers/hid/hid-ids.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/hid/hid-ids.h b/drivers/hid/hid-ids.h
index 9a0f8f6b39cd..6e85b3b41169 100644
--- a/drivers/hid/hid-ids.h
+++ b/drivers/hid/hid-ids.h
@@ -945,6 +945,18 @@
 #define USB_DEVICE_ID_MS_TYPE_COVER_2    0x07a9
 #define USB_DEVICE_ID_MS_POWER_COVER     0x07da
 #define USB_DEVICE_ID_MS_SURFACE3_COVER		0x07de
+#define USB_DEVICE_ID_MS_TYPE_COVER_3          0x07de
+#define USB_DEVICE_ID_MS_TYPE_COVER_PRO_3      0x07dc
+#define USB_DEVICE_ID_MS_TYPE_COVER_PRO_3_1    0x07de
+#define USB_DEVICE_ID_MS_TYPE_COVER_PRO_3_2    0x07e2
+#define USB_DEVICE_ID_MS_TYPE_COVER_PRO_3_JP   0x07dd
+#define USB_DEVICE_ID_MS_TYPE_COVER_PRO_4      0x07e8
+#define USB_DEVICE_ID_MS_TYPE_COVER_PRO_4_1    0x07e4
+#define USB_DEVICE_ID_MS_SURFACE_BOOK          0x07cd
+#define USB_DEVICE_ID_MS_SURFACE_BOOK_2        0x0922
+#define USB_DEVICE_ID_MS_SURFACE_GO            0x096f
+#define USB_DEVICE_ID_MS_SURFACE_VHF           0xf001
+
 /*
  * For a description of the Xbox controller models, refer to:
  * https://en.wikipedia.org/wiki/Xbox_Wireless_Controller#Summary
-- 
2.45.1

