From 2ff4dd3b69e2cd86558cb162ffc8c23771c270e0 Mon Sep 17 00:00:00 2001
From: Jiaxun Yang <jiaxun.yang@flygoat.com>
Date: Mon, 24 Apr 2023 11:50:52 +0100
Subject: [PATCH] cip-eng: MIPS: Select select WEAK_REORDERING_BEYOND_LLSC for
 generic platform

Signed-off-by: Jiaxun Yang <jiaxun.yang@flygoat.com>
---
 arch/mips/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 7629e671381b..89731365d679 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -178,6 +178,7 @@ config MIPS_GENERIC_KERNEL
 	select USB_UHCI_BIG_ENDIAN_DESC if CPU_BIG_ENDIAN
 	select USB_UHCI_BIG_ENDIAN_MMIO if CPU_BIG_ENDIAN
 	select USE_OF
+	select WEAK_REORDERING_BEYOND_LLSC
 	help
 	  Select this to build a kernel which aims to support multiple boards,
 	  generally using a flattened device tree passed from the bootloader
-- 
2.39.1

