From ddbaa565c0a7bf702bf4009e504aa11914492cee Mon Sep 17 00:00:00 2001
From: Jiaxun Yang <jiaxun.yang@flygoat.com>
Date: Mon, 24 Apr 2023 11:57:58 +0100
Subject: [PATCH] cip-eng: MIPS: Select ARCH_SUPPORTS_ATOMIC_RMW when possible

Signed-off-by: Jiaxun Yang <jiaxun.yang@flygoat.com>
---
 arch/mips/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 89731365d679..5d7bc30ec721 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -101,6 +101,7 @@ config MIPS
 	select TRACE_IRQFLAGS_SUPPORT
 	select ARCH_HAS_ELFCORE_COMPAT
 	select HAVE_ARCH_KCSAN if 64BIT
+	select ARCH_SUPPORTS_ATOMIC_RMW if TARGET_ISA_REV >= 1
 
 config MIPS_FIXUP_BIGPHYS_ADDR
 	bool
-- 
2.39.1

