From 81399b8514b542c2578483a794fd8223cb745256 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:43:01 +0800
Subject: [PATCH 096/333] MARKER: FROMLIST: End of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index e857b79e77d26..6b38e6da99fd4 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -507,7 +507,6 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
-	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.49.0

