From b40b722e445b622a660fd857ec5e4631be7c7dec Mon Sep 17 00:00:00 2001
From: Miao Wang <shankerwangmiao@gmail.com>
Date: Sun, 11 Aug 2024 02:50:05 +0800
Subject: [PATCH 100/101] fix acpi

---
 arch/loongarch/kernel/legacy_boot.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/arch/loongarch/kernel/legacy_boot.c b/arch/loongarch/kernel/legacy_boot.c
index 48c13e1ad220..3ec69ffd8b94 100644
--- a/arch/loongarch/kernel/legacy_boot.c
+++ b/arch/loongarch/kernel/legacy_boot.c
@@ -355,6 +355,7 @@ static void __init init_acpi_arch_os_table_override (struct acpi_table_header *e
 	int entry_count = 0;
 	int local_apic_count = 0;
 	int io_apic_count = 0;
+	u64 node_map = 0;
 	while((void *)entry < madt_end) {
 		unsigned int ent_len = entry->length;
 		if (ent_len < sizeof(struct acpi_subtable_header)) {
@@ -368,6 +369,8 @@ static void __init init_acpi_arch_os_table_override (struct acpi_table_header *e
 		switch(entry->type) {
 		case ACPI_MADT_TYPE_LOCAL_APIC:
 			local_apic_count++;
+			struct acpi_madt_local_apic *lapic = (struct acpi_madt_local_apic *)entry;
+			node_map |= 1 << (lapic->id / CORES_PER_EIO_NODE);
 			break;
 		case ACPI_MADT_TYPE_IO_APIC:
 			io_apic_count++;
@@ -488,10 +491,10 @@ static void __init init_acpi_arch_os_table_override (struct acpi_table_header *e
 			eio_pics[eio_idx].cascade = 3 + eio_idx;
 			eio_pics[eio_idx].node = ioapic->id;
 			if(eio_idx == 0){
-				eio_pics[eio_idx].node_map = 1;
+				eio_pics[eio_idx].node_map = node_map;
 			}else{
-				eio_pics[0].node_map = 0x1DF;
-				eio_pics[eio_idx].node_map = 0xFE20;
+				eio_pics[0].node_map = node_map & 0x0f0f0f0f0f0f0f0full;
+				eio_pics[eio_idx].node_map = node_map & 0xf0f0f0f0f0f0f0f0ull;
 			}
 
 			unsigned long addr = ioapic->address;
@@ -516,7 +519,7 @@ static void __init init_acpi_arch_os_table_override (struct acpi_table_header *e
 			bio_pics[eio_idx].id = ioapic->id;
 			bio_pics[eio_idx].gsi_base = ioapic->global_irq_base;
 
-			mcfg_entry->address = mcfg_addr_init(eio_idx);
+			mcfg_entry->address = mcfg_addr_init(ioapic->id);
 			mcfg_entry->pci_segment = eio_idx;
 			mcfg_entry->start_bus_number = 0;
 			mcfg_entry->end_bus_number = 0xFF; // Who knows?
-- 
2.46.0

