From 364883d434939ad58f1c96bb16be8cb9125a46dc Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:43:40 +0800
Subject: [PATCH 211/334] MARKER: AOSCOS: Start of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index aae3b2d2b9fdd..a104c742a0959 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -507,6 +507,7 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
+	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.50.0

