From 84b2a0687915222f87208052e8e4f1990d9c29bb Mon Sep 17 00:00:00 2001
From: Xiangyi Zeng <xiangyi.zeng@linux.alibaba.com>
Date: Sun, 1 Sep 2024 01:49:37 +0800
Subject: [PATCH 249/418] XUANTIE: drivers: pwm: fix pwm enable status check
 error

From: https://github.com/revyos/th1520-linux-kernel/commit/e40c8171f8c1116d31e3eb856e2c7334119eac3f

Signed-off-by: Xiangyi Zeng <xiangyi.zeng@linux.alibaba.com>
Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 drivers/pwm/pwm-thead.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/pwm/pwm-thead.c b/drivers/pwm/pwm-thead.c
index 31590f91aad7..01db22918e39 100644
--- a/drivers/pwm/pwm-thead.c
+++ b/drivers/pwm/pwm-thead.c
@@ -51,6 +51,7 @@
 #define THEAD_PWM_FP(n)			(THEAD_PWM_CHN_BASE(n) + 0x0c)
 #define THEAD_PWM_STATUS(n)		(THEAD_PWM_CHN_BASE(n) + 0x10)
 #define  THEAD_PWM_STATUS_CYCLE		GENMASK(7, 0)
+#define  THEAD_PWM_STATUS_BUSY		BIT(8)
 
 struct thead_pwm_chip {
 	void __iomem *mmio_base;
@@ -165,8 +166,8 @@ static int thead_pwm_get_state(struct pwm_chip *chip, struct pwm_device *pwm,
 	 */
 	state->period = DIV64_U64_ROUND_UP((u64)val * NSEC_PER_SEC, rate);
 
-	val = readl(priv->mmio_base + THEAD_PWM_FP(pwm->hwpwm));
-	state->enabled = !!val;
+	val = readl(priv->mmio_base + THEAD_PWM_STATUS(pwm->hwpwm));
+	state->enabled = !!(val & THEAD_PWM_STATUS_BUSY);
 	/*
 	 * val 32 bits, multiply NSEC_PER_SEC, won't overflow.
 	 */
-- 
2.52.0

