From 09b92499b88c585ff385abdb7a59d536a62a9a2a Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:43:40 +0800
Subject: [PATCH 255/373] MARKER: AOSCOS: Start of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index bb116b29b698..b78c3c912df6 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -533,6 +533,7 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
+	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.51.0

