From 83c33044da5bf0a1c45ed9fb97b2923935f30b4a Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:43:01 +0800
Subject: [PATCH 304/587] MARKER: FROMLIST: End of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/mips/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 0820276782ab..256240ff86f7 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -550,7 +550,6 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
-	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.54.0

