From 00e7c15adf589e2deeec62823467c95839420139 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:45:08 +0800
Subject: [PATCH 367/373] MARKER: AOSCOS: End of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 067581196f94..750096f0e8b2 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -533,7 +533,6 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
-	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.51.0

