From 1a780172a0f1d4db24932debfabe02e9f3e69162 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:45:08 +0800
Subject: [PATCH 541/566] MARKER: AOSCOS: End of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/mips/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 290cecf89dcc3..f637eecd8a532 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -547,7 +547,6 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
-	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.52.0

