From 5543e52d041cdf2d3e57e1778006a93f18013ad5 Mon Sep 17 00:00:00 2001
From: Junhao Xie <bigfoot@radxa.com>
Date: Wed, 15 Oct 2025 12:58:41 +0800
Subject: [PATCH 596/820] RADXA: kbuild: Force all dtbs generated symbols to
 use overlay

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 scripts/Makefile.dtbs | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/Makefile.dtbs b/scripts/Makefile.dtbs
index 8c03bd8a9bc43..0acc282d20b14 100644
--- a/scripts/Makefile.dtbs
+++ b/scripts/Makefile.dtbs
@@ -100,6 +100,9 @@ $(call multi_depend, $(multi-dtb-y), .dtb, -dtbs)
 DTC ?= $(objtree)/scripts/dtc/dtc
 DTC_FLAGS += -Wno-unique_unit_address
 
+# Force generation of symbols even if there are no overlays
+DTC_FLAGS += -@
+
 # Disable noisy checks by default
 ifeq ($(findstring 1,$(KBUILD_EXTRA_WARN)),)
 DTC_FLAGS += -Wno-unit_address_vs_reg \
-- 
2.52.0

