From bd32b6c4db2aca36622dc81e15a05a5a5aaaf9ab Mon Sep 17 00:00:00 2001
From: SongJun Li <lisongjun@radxa.com>
Date: Thu, 29 Jan 2026 09:28:17 +0800
Subject: [PATCH 615/820] RADXA: media: i2c: imx214: add reset GPIO handling in
 power-on sequence

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 drivers/media/i2c/imx214.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/drivers/media/i2c/imx214.c b/drivers/media/i2c/imx214.c
index d4945b192776b..bc2413f5cfac6 100644
--- a/drivers/media/i2c/imx214.c
+++ b/drivers/media/i2c/imx214.c
@@ -271,6 +271,7 @@ struct imx214 {
 
 	struct regulator_bulk_data	supplies[IMX214_NUM_SUPPLIES];
 
+	struct gpio_desc *reset_gpio;
 	struct gpio_desc *enable_gpio;
 };
 
@@ -548,6 +549,16 @@ static int __maybe_unused imx214_power_on(struct device *dev)
 		return ret;
 	}
 
+	if (!IS_ERR(imx214->reset_gpio))
+		gpiod_set_value_cansleep(imx214->reset_gpio, 1);
+
+	usleep_range(500, 1000);
+
+	if (!IS_ERR(imx214->reset_gpio))
+		gpiod_set_value_cansleep(imx214->reset_gpio, 0);
+
+	usleep_range(500, 1000);
+
 	gpiod_set_value_cansleep(imx214->enable_gpio, 1);
 	usleep_range(12000, 15000);
 
@@ -1405,6 +1416,11 @@ static int imx214_probe(struct i2c_client *client)
 	if (ret < 0)
 		return dev_err_probe(dev, ret, "failed to get regulators\n");
 
+	imx214->reset_gpio = devm_gpiod_get(dev, "reset", GPIOD_OUT_LOW);
+	if (IS_ERR(imx214->reset_gpio))
+		return dev_err_probe(dev, PTR_ERR(imx214->reset_gpio),
+				     "failed to get reset gpio\n");
+
 	imx214->enable_gpio = devm_gpiod_get(dev, "enable", GPIOD_OUT_LOW);
 	if (IS_ERR(imx214->enable_gpio))
 		return dev_err_probe(dev, PTR_ERR(imx214->enable_gpio),
-- 
2.52.0

