From 8041b4d53d8f95f7e4f7aa10e40de107e3b765a6 Mon Sep 17 00:00:00 2001
From: Xilin Wu <sophon@radxa.com>
Date: Tue, 13 Jan 2026 21:31:48 +0800
Subject: [PATCH 630/823] RADXA: firmware: qcom: scm: Allow QSEECOM for Radxa
 Dragon Q8B

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 drivers/firmware/qcom/qcom_scm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/firmware/qcom/qcom_scm.c b/drivers/firmware/qcom/qcom_scm.c
index 3edfbc766bf68..d944d67b4e0ac 100644
--- a/drivers/firmware/qcom/qcom_scm.c
+++ b/drivers/firmware/qcom/qcom_scm.c
@@ -2339,6 +2339,7 @@ static const struct of_device_id qcom_scm_qseecom_allowlist[] __maybe_unused = {
 	{ .compatible = "qcom,x1e80100-qcp" },
 	{ .compatible = "qcom,x1p42100-crd" },
 	{ .compatible = "radxa,dragon-q6a" },
+	{ .compatible = "radxa,dragon-q8b" },
 	{ }
 };
 
-- 
2.52.0

