From 16fc0487df719a9008580ac9656109db14e421f1 Mon Sep 17 00:00:00 2001
From: Jiali Chen <chenjiali@radxa.com>
Date: Mon, 27 Apr 2026 18:10:44 +0800
Subject: [PATCH 702/820] RADXA: firmware: qcom: scm: Allow QSEECOM for Radxa
 CM-Q64

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 drivers/firmware/qcom/qcom_scm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/firmware/qcom/qcom_scm.c b/drivers/firmware/qcom/qcom_scm.c
index d944d67b4e0ac..b53d9b7ebb6e3 100644
--- a/drivers/firmware/qcom/qcom_scm.c
+++ b/drivers/firmware/qcom/qcom_scm.c
@@ -2338,6 +2338,7 @@ static const struct of_device_id qcom_scm_qseecom_allowlist[] __maybe_unused = {
 	{ .compatible = "qcom,x1e80100-crd" },
 	{ .compatible = "qcom,x1e80100-qcp" },
 	{ .compatible = "qcom,x1p42100-crd" },
+	{ .compatible = "radxa,cm-q64" },
 	{ .compatible = "radxa,dragon-q6a" },
 	{ .compatible = "radxa,dragon-q8b" },
 	{ }
@@ -2604,6 +2605,7 @@ EXPORT_SYMBOL_GPL(qcom_scm_storage_send_cmd);
  */
 static const struct of_device_id qcom_scm_storage_allowlist[] = {
 	{ .compatible = "radxa,dragon-q6a" },
+	{ .compatible = "radxa,cm-q64" },
 	{ .compatible = "radxa,dragon-q8b" },
 	{ }
 };
-- 
2.52.0

