From 6e8afca5d15f9c198d012e1f1e5621f28276350d Mon Sep 17 00:00:00 2001
From: Jiali Chen <chenjiali@radxa.com>
Date: Tue, 30 Dec 2025 09:41:46 +0000
Subject: [PATCH 713/823] RADXA: usb: typec: fusb302: workaround for fixed VBUS
 hardware

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 drivers/usb/typec/tcpm/fusb302.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/usb/typec/tcpm/fusb302.c b/drivers/usb/typec/tcpm/fusb302.c
index 5b1f2750cfc3b..673fa3a5d3923 100644
--- a/drivers/usb/typec/tcpm/fusb302.c
+++ b/drivers/usb/typec/tcpm/fusb302.c
@@ -99,6 +99,7 @@ struct fusb302_chip {
 	bool intr_togdone;
 	bool intr_bc_lvl;
 	bool intr_comp_chng;
+	bool force_vbus_toggle;
 
 	/* port status */
 	bool vconn_on;
@@ -776,6 +777,13 @@ static int tcpm_set_vbus(struct tcpc_dev *dev, bool on, bool charge)
 		}
 		chip->vbus_on = on;
 		fusb302_log(chip, "vbus := %s", str_on_off(on));
+		if (on && chip->force_vbus_toggle) {
+			fusb302_log(chip, "WORKAROUND: Forcing VBUS toggle for TCPM state machine");
+			chip->vbus_present = false;
+			tcpm_vbus_change(chip->tcpm_port);
+			chip->vbus_present = true;
+			tcpm_vbus_change(chip->tcpm_port);
+		}
 	}
 	if (chip->charge_on == charge)
 		fusb302_log(chip, "charge is already %s", str_on_off(charge));
@@ -1723,6 +1731,8 @@ static int fusb302_probe(struct i2c_client *client)
 	if (IS_ERR(chip->vbus))
 		return PTR_ERR(chip->vbus);
 
+	chip->force_vbus_toggle = device_property_read_bool(chip->dev, "fcs,force-vbus-toggle");
+
 	chip->wq = create_singlethread_workqueue(dev_name(chip->dev));
 	if (!chip->wq)
 		return -ENOMEM;
-- 
2.52.0

