From 6dae40cb9deb377349a5ddb33b18a3dd8ab437c0 Mon Sep 17 00:00:00 2001
From: Jiali Chen <chenjiali@radxa.com>
Date: Tue, 30 Dec 2025 09:50:31 +0000
Subject: [PATCH 715/823] RADXA: arm64: dts: meson: radxa-zero: add status LED

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 .../dts/amlogic/meson-g12a-radxa-zero.dts     | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts b/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts
index 199dead5847a3..1dbe725afb147 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12a-radxa-zero.dts
@@ -7,6 +7,7 @@
 
 #include "meson-g12a.dtsi"
 #include <dt-bindings/gpio/meson-g12a-gpio.h>
+#include <dt-bindings/leds/common.h>
 #include <dt-bindings/sound/meson-g12a-tohdmitx.h>
 #include <dt-bindings/usb/pd.h>
 
@@ -130,6 +131,24 @@ vddcpu: regulator-vddcpu {
 		regulator-always-on;
 	};
 
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: led-power {
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_STATUS;
+			gpios = <&gpio_ao GPIOAO_8 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = LED_FUNCTION_HEARTBEAT;
+		};
+
+		led_power_151: led-power-151 {
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_STATUS;
+			gpios = <&gpio_ao GPIOAO_10 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = LED_FUNCTION_HEARTBEAT;
+		};
+	};
+
 	typec_vbus: regulator-typec-vbus {
 		compatible = "regulator-fixed";
 		regulator-name = "TYPEC_VBUS";
-- 
2.52.0

