From 9ac10acdbba677afef92498252a0891455aa0850 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Thu, 18 Dec 2025 10:00:02 +0800
Subject: [PATCH 9991/9994] extract variable

---
 .../media/platform/verisilicon/hantro_drv.c   | 38 ++++++++++---------
 1 file changed, 20 insertions(+), 18 deletions(-)

diff --git a/drivers/media/platform/verisilicon/hantro_drv.c b/drivers/media/platform/verisilicon/hantro_drv.c
index 60b95b5d8565..11cb7fb68b17 100644
--- a/drivers/media/platform/verisilicon/hantro_drv.c
+++ b/drivers/media/platform/verisilicon/hantro_drv.c
@@ -1039,19 +1039,21 @@ static int hantro_probe(struct platform_device *pdev)
 {
 	const struct of_device_id *match;
 	struct hantro_dev *vpu;
+	struct device *vpu_dev;
 	int num_bases;
 	int i, ret;
 
-	vpu = devm_kzalloc(&pdev->dev, sizeof(*vpu), GFP_KERNEL);
+	vpu_dev = &pdev->dev;
+	vpu = devm_kzalloc(vpu_dev, sizeof(*vpu), GFP_KERNEL);
 	if (!vpu)
 		return -ENOMEM;
 
-	vpu->dev = &pdev->dev;
+	vpu->dev = vpu_dev;
 	vpu->pdev = pdev;
 	mutex_init(&vpu->vpu_mutex);
 	spin_lock_init(&vpu->irqlock);
 
-	match = of_match_node(of_hantro_match, pdev->dev.of_node);
+	match = of_match_node(of_hantro_match, vpu_dev->of_node);
 	vpu->variant = match->data;
 
 	ret = hantro_disable_multicore(vpu);
@@ -1063,13 +1065,13 @@ static int hantro_probe(struct platform_device *pdev)
 	 * but it's deprecated. Please update your DTS file to use
 	 * nxp,imx8mq-vpu-g1 or nxp,imx8mq-vpu-g2 instead.
 	 */
-	if (of_device_is_compatible(pdev->dev.of_node, "nxp,imx8mq-vpu"))
-		dev_warn(&pdev->dev, "%s compatible is deprecated\n",
+	if (of_device_is_compatible(vpu_dev->of_node, "nxp,imx8mq-vpu"))
+		dev_warn(vpu_dev, "%s compatible is deprecated\n",
 			 match->compatible);
 
 	INIT_DELAYED_WORK(&vpu->watchdog_work, hantro_watchdog);
 
-	vpu->clocks = devm_kcalloc(&pdev->dev, vpu->variant->num_clocks,
+	vpu->clocks = devm_kcalloc(vpu_dev, vpu->variant->num_clocks,
 				   sizeof(*vpu->clocks), GFP_KERNEL);
 	if (!vpu->clocks)
 		return -ENOMEM;
@@ -1078,7 +1080,7 @@ static int hantro_probe(struct platform_device *pdev)
 		for (i = 0; i < vpu->variant->num_clocks; i++)
 			vpu->clocks[i].id = vpu->variant->clk_names[i];
 
-		ret = devm_clk_bulk_get(&pdev->dev, vpu->variant->num_clocks,
+		ret = devm_clk_bulk_get(vpu_dev, vpu->variant->num_clocks,
 					vpu->clocks);
 		if (ret)
 			return ret;
@@ -1087,17 +1089,17 @@ static int hantro_probe(struct platform_device *pdev)
 		 * If the driver has a single clk, chances are there will be no
 		 * actual name in the DT bindings.
 		 */
-		vpu->clocks[0].clk = devm_clk_get(&pdev->dev, NULL);
+		vpu->clocks[0].clk = devm_clk_get(vpu_dev, NULL);
 		if (IS_ERR(vpu->clocks[0].clk))
 			return PTR_ERR(vpu->clocks[0].clk);
 	}
 
-	vpu->resets = devm_reset_control_array_get_optional_exclusive(&pdev->dev);
+	vpu->resets = devm_reset_control_array_get_optional_exclusive(vpu_dev);
 	if (IS_ERR(vpu->resets))
 		return PTR_ERR(vpu->resets);
 
 	num_bases = vpu->variant->num_regs ?: 1;
-	vpu->reg_bases = devm_kcalloc(&pdev->dev, num_bases,
+	vpu->reg_bases = devm_kcalloc(vpu_dev, num_bases,
 				      sizeof(*vpu->reg_bases), GFP_KERNEL);
 	if (!vpu->reg_bases)
 		return -ENOMEM;
@@ -1122,7 +1124,7 @@ static int hantro_probe(struct platform_device *pdev)
 		dev_err(vpu->dev, "Could not set DMA coherent mask.\n");
 		return ret;
 	}
-	vb2_dma_contig_set_max_seg_size(&pdev->dev, DMA_BIT_MASK(32));
+	vb2_dma_contig_set_max_seg_size(vpu_dev, DMA_BIT_MASK(32));
 
 	for (i = 0; i < vpu->variant->num_irqs; i++) {
 		const char *irq_name;
@@ -1158,7 +1160,7 @@ static int hantro_probe(struct platform_device *pdev)
 	if (vpu->variant->init) {
 		ret = vpu->variant->init(vpu);
 		if (ret) {
-			dev_err(&pdev->dev, "Failed to init VPU hardware\n");
+			dev_err(vpu_dev, "Failed to init VPU hardware\n");
 			return ret;
 		}
 	}
@@ -1169,19 +1171,19 @@ static int hantro_probe(struct platform_device *pdev)
 
 	ret = reset_control_deassert(vpu->resets);
 	if (ret) {
-		dev_err(&pdev->dev, "Failed to deassert resets\n");
+		dev_err(vpu_dev, "Failed to deassert resets\n");
 		goto err_pm_disable;
 	}
 
 	ret = clk_bulk_prepare(vpu->variant->num_clocks, vpu->clocks);
 	if (ret) {
-		dev_err(&pdev->dev, "Failed to prepare clocks\n");
+		dev_err(vpu_dev, "Failed to prepare clocks\n");
 		goto err_rst_assert;
 	}
 
-	ret = v4l2_device_register(&pdev->dev, &vpu->v4l2_dev);
+	ret = v4l2_device_register(vpu_dev, &vpu->v4l2_dev);
 	if (ret) {
-		dev_err(&pdev->dev, "Failed to register v4l2 device\n");
+		dev_err(vpu_dev, "Failed to register v4l2 device\n");
 		goto err_clk_unprepare;
 	}
 	platform_set_drvdata(pdev, vpu);
@@ -1201,13 +1203,13 @@ static int hantro_probe(struct platform_device *pdev)
 
 	ret = hantro_add_enc_func(vpu);
 	if (ret) {
-		dev_err(&pdev->dev, "Failed to register encoder\n");
+		dev_err(vpu_dev, "Failed to register encoder\n");
 		goto err_m2m_rel;
 	}
 
 	ret = hantro_add_dec_func(vpu);
 	if (ret) {
-		dev_err(&pdev->dev, "Failed to register decoder\n");
+		dev_err(vpu_dev, "Failed to register decoder\n");
 		goto err_rm_enc_func;
 	}
 
-- 
2.52.0

