From 03f6bc9fba3895ac9342e2a330a426b1703bb6d8 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Thu, 18 Dec 2025 09:58:40 +0800
Subject: [PATCH 9993/9994] ls3 pci table

---
 drivers/media/platform/verisilicon/hantro_drv.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/media/platform/verisilicon/hantro_drv.c b/drivers/media/platform/verisilicon/hantro_drv.c
index f6d9f9bf1c0a..68e625eb93c4 100644
--- a/drivers/media/platform/verisilicon/hantro_drv.c
+++ b/drivers/media/platform/verisilicon/hantro_drv.c
@@ -737,11 +737,22 @@ static const struct of_device_id of_hantro_match[] = {
 #ifdef CONFIG_VIDEO_HANTRO_STM32MP25
 	{ .compatible = "st,stm32mp25-vdec", .data = &stm32mp25_vdec_variant, },
 	{ .compatible = "st,stm32mp25-venc", .data = &stm32mp25_venc_variant, },
+#endif
+#ifdef CONFIG_VIDEO_HANTRO_LOONGSON3
+	{ .compatible = "loongson,ls-vpu", .data = &ls2k_vpu_g1_variant, },
 #endif
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, of_hantro_match);
 
+#ifdef CONFIG_VIDEO_HANTRO_SUPPORT_PCI_DEVICE
+static const struct pci_device_id pci_hantro_match[] = {
+	{ PCI_DEVICE(PCI_VENDOR_ID_LOONGSON, 0x7A16), .driver_data = (kernel_ulong_t)&ls2k_vpu_g1_variant },
+	{ /* sentinel */ }
+};
+MODULE_DEVICE_TABLE(pci, pci_hantro_match);
+#endif
+
 static int hantro_register_entity(struct media_device *mdev,
 				  struct media_entity *entity,
 				  const char *entity_name,
-- 
2.52.0

