diff --git a/drivers/media/platform/verisilicon/hantro_drv.c b/drivers/media/platform/verisilicon/hantro_drv.c
index 68e625eb93c4..6db1529a364f 100644
--- a/drivers/media/platform/verisilicon/hantro_drv.c
+++ b/drivers/media/platform/verisilicon/hantro_drv.c
@@ -1072,6 +1072,7 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 	if (is_pci) {
 		pci_match = pci_match_id(pci_hantro_match, to_pci_dev(vpu_dev));
 		vpu->variant = (const struct hantro_variant *)pci_match->driver_data;
+		printk("init 1\n");
 	} else {
 		of_match = of_match_node(of_hantro_match, vpu_dev->of_node);
 		vpu->variant = of_match->data;
@@ -1089,7 +1090,7 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 			dev_warn(vpu_dev, "%s compatible is deprecated\n",
 				 of_match->compatible);
 	}
-
+  printk("init 2\n");
 	INIT_DELAYED_WORK(&vpu->watchdog_work, hantro_watchdog);
 
 	vpu->clocks = devm_kcalloc(vpu_dev, vpu->variant->num_clocks,
@@ -1105,7 +1106,7 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 					vpu->clocks);
 		if (ret)
 			return ret;
-	} else {
+	} else if (vpu->variant->num_clocks > 0) {
 		/*
 		 * If the driver has a single clk, chances are there will be no
 		 * actual name in the DT bindings.
@@ -1114,10 +1115,11 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 		if (IS_ERR(vpu->clocks[0].clk))
 			return PTR_ERR(vpu->clocks[0].clk);
 	}
-
+  printk("init 3\n");
 	vpu->resets = devm_reset_control_array_get_optional_exclusive(vpu_dev);
 	if (IS_ERR(vpu->resets))
 		return PTR_ERR(vpu->resets);
+	printk("init 4\n");
 
 	num_bases = vpu->variant->num_regs ?: 1;
 	vpu->reg_bases = devm_kcalloc(vpu_dev, num_bases,
@@ -1141,6 +1143,7 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 			return PTR_ERR(io_resources);
 		vpu->reg_bases[i] = io_resources;
 	}
+	printk("init 5\n");
 	vpu->enc_base = vpu->reg_bases[0] + vpu->variant->enc_offset;
 	vpu->dec_base = vpu->reg_bases[0] + vpu->variant->dec_offset;
 
@@ -1154,6 +1157,7 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 		dev_err(vpu->dev, "Could not set DMA coherent mask.\n");
 		return ret;
 	}
+	printk("init 6\n");
 	vb2_dma_contig_set_max_seg_size(vpu_dev, DMA_BIT_MASK(32));
 
 	for (i = 0; i < vpu->variant->num_irqs; i++) {
@@ -1181,7 +1185,7 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 			return irq;
 
 		ret = devm_request_irq(vpu->dev, irq,
-				       vpu->variant->irqs[i].handler, 0,
+				       vpu->variant->irqs[i].handler, IRQF_SHARED,
 				       dev_name(vpu->dev), vpu);
 		if (ret) {
 			dev_err(vpu->dev, "Could not request %s IRQ.\n",
@@ -1189,6 +1193,7 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 			return ret;
 		}
 	}
+	printk("init 6\n");
 
 	if (vpu->variant->init) {
 		ret = vpu->variant->init(vpu);
@@ -1201,6 +1206,7 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 	pm_runtime_set_autosuspend_delay(vpu->dev, 100);
 	pm_runtime_use_autosuspend(vpu->dev);
 	pm_runtime_enable(vpu->dev);
+	printk("init 7\n");
 
 	ret = reset_control_deassert(vpu->resets);
 	if (ret) {
@@ -1213,6 +1219,7 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 		dev_err(vpu_dev, "Failed to prepare clocks\n");
 		goto err_rst_assert;
 	}
+	printk("init 8\n");
 
 	ret = v4l2_device_register(vpu_dev, &vpu->v4l2_dev);
 	if (ret) {
@@ -1236,6 +1243,7 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 	media_device_init(&vpu->mdev);
 	vpu->mdev.ops = &hantro_m2m_media_ops;
 	vpu->v4l2_dev.mdev = &vpu->mdev;
+	printk("init 9\n");
 
 	ret = hantro_add_enc_func(vpu);
 	if (ret) {
@@ -1254,6 +1262,7 @@ static int hantro_init(struct device *vpu_dev, bool is_pci)
 		v4l2_err(&vpu->v4l2_dev, "Failed to register mem2mem media device\n");
 		goto err_rm_dec_func;
 	}
+	printk("vpu finish\n");
 
 	return 0;
 
diff --git a/drivers/media/platform/verisilicon/ls3_vpu_hw.c b/drivers/media/platform/verisilicon/ls3_vpu_hw.c
index 0f5349b87264..44eb8ab12f97 100644
--- a/drivers/media/platform/verisilicon/ls3_vpu_hw.c
+++ b/drivers/media/platform/verisilicon/ls3_vpu_hw.c
@@ -99,5 +99,5 @@ const struct hantro_variant ls2k_vpu_g1_variant = {
 	.postproc_ops = &hantro_g1_postproc_ops,
 	.num_irqs = ARRAY_SIZE(ls2k_g1_irqs),
 	.clk_names = ls2k_clk_names,
-	.num_clocks = ARRAY_SIZE(ls2k_clk_names),
+	.num_clocks = 0,
 };
