--- a/CMakeLists.txt	2022-08-05 15:55:17.000000000 +0800
+++ b/CMakeLists.txt	2025-03-12 20:48:27.471296264 +0800
@@ -737,7 +737,7 @@
 # news-to-appdata
 # ******************************
 
-set(CMAKE_REQUIRED_FLAGS ${GNOME_PLATFORM_CFLAGS})
+set(CMAKE_REQUIRED_DEFINITIONS ${GNOME_PLATFORM_CFLAGS})
 set(CMAKE_REQUIRED_INCLUDES ${GNOME_PLATFORM_INCLUDE_DIRS})
 set(CMAKE_REQUIRED_LIBRARIES ${GNOME_PLATFORM_LDFLAGS})
 file(TO_NATIVE_PATH "${CMAKE_BINARY_DIR}/appdata-releases.txt" _output_filename)
@@ -752,7 +752,7 @@
 unset(_output_filename)
 unset(CMAKE_REQUIRED_LIBRARIES)
 unset(CMAKE_REQUIRED_INCLUDES)
-unset(CMAKE_REQUIRED_FLAGS)
+unset(CMAKE_REQUIRED_DEFINITIONS)
 
 if(NOT "${_news_to_appdata_result}" EQUAL "1")
 	message(FATAL_ERROR "Failed to run news-to-appdata")
