From 14cf9c2d9e85f1dd3cf564c7e9bb1233c3bf774a Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Tue, 20 Dec 2022 01:01:11 -0500
Subject: [PATCH] common/guided_filter: remove broken ifdefs on ppc

---
 src/common/sse.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/common/sse.h b/src/common/sse.h
index f7609add9a..45d9c8e461 100644
--- a/src/common/sse.h
+++ b/src/common/sse.h
@@ -17,7 +17,9 @@
 #include "common/darktable.h"
 
 #ifdef __SSE__
-
+#ifdef __PPC64__
+#define NO_WARN_X86_INTRINSICS 1
+#endif
 #include <xmmintrin.h>
 
 
-- 
2.45.2

