From 913ccb6e186a412c75dc27cf3d1a3365479fd1b2 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Tue, 20 Dec 2022 01:04:49 -0500
Subject: [PATCH] fix building for mips64

---
 src/is_supported_platform.h | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/is_supported_platform.h b/src/is_supported_platform.h
index 03a77f704d..9d418d3c5f 100644
--- a/src/is_supported_platform.h
+++ b/src/is_supported_platform.h
@@ -36,6 +36,12 @@
 #define DT_SUPPORTED_ARMv8A 0
 #endif
 
+#if defined(__mips__) && defined(__mips64)
+#define DT_SUPPORTED_MIPS64 1
+#else
+#define DT_SUPPORTED_MIPS64 0
+#endif
+
 #if defined(__PPC64__)
 #define DT_SUPPORTED_PPC64 1
 #else
@@ -48,16 +54,17 @@
 #define DT_SUPPORTED_RISCV64 0
 #endif
 
-#if (DT_SUPPORTED_X86 + DT_SUPPORTED_ARMv8A + DT_SUPPORTED_PPC64 + DT_SUPPORTED_RISCV64) > 1
+#if (DT_SUPPORTED_X86 + DT_SUPPORTED_ARMv8A + DT_SUPPORTED_MIPS64 + DT_SUPPORTED_PPC64 + DT_SUPPORTED_RISCV64) > 1
 #error "Looks like hardware platform detection macros are broken?"
 #endif
 
-#if !DT_SUPPORTED_X86 && !DT_SUPPORTED_ARMv8A && !DT_SUPPORTED_PPC64 && !DT_SUPPORTED_RISCV64
+#if !DT_SUPPORTED_X86 && !DT_SUPPORTED_ARMv8A && !DT_SUPPORTED_MIPS64 && !DT_SUPPORTED_PPC64 && !DT_SUPPORTED_RISCV64
 #error "Unfortunately we only work on amd64, ARMv8-A, PPC64 (64-bit little-endian only) and riscv64"
 #endif
 
 #undef DT_SUPPORTED_RISCV64
 #undef DT_SUPPORTED_PPC64
+#undef DT_SUPPORTED_MIPS64
 #undef DT_SUPPORTED_ARMv8A
 #undef DT_SUPPORTED_X86
 
-- 
2.39.1

