From 27269de3ff9d78b5fa25678d3d8bc54d538b2576 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Tue, 20 Dec 2022 01:04:49 -0500
Subject: [PATCH] fix building for mips64 and loongarch64

---
 src/is_supported_platform.h | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/is_supported_platform.h b/src/is_supported_platform.h
index 8fc8c941f3..abeadc1918 100644
--- a/src/is_supported_platform.h
+++ b/src/is_supported_platform.h
@@ -36,6 +36,12 @@
 #define DT_SUPPORTED_ARMv8A 0
 #endif
 
+#if defined(__mips__) && defined(__mips64)
+#define DT_SUPPORTED_MIPS64 1
+#else
+#define DT_SUPPORTED_MIPS64 0
+#endif
+
 #if defined(__PPC64__)
 #define DT_SUPPORTED_PPC64 1
 #else
@@ -54,17 +60,18 @@
 #define DT_SUPPORTED_LOONGARCH64 0
 #endif
 
-#if (DT_SUPPORTED_X86 + DT_SUPPORTED_ARMv8A + DT_SUPPORTED_PPC64 + DT_SUPPORTED_RISCV64 + DT_SUPPORTED_LOONGARCH64) > 1
+#if (DT_SUPPORTED_X86 + DT_SUPPORTED_ARMv8A + DT_SUPPORTED_PPC64 + DT_SUPPORTED_RISCV64 + DT_SUPPORTED_LOONGARCH64 + DT_SUPPORTED_MIPS64) > 1
 #error "Looks like hardware platform detection macros are broken?"
 #endif
 
-#if !DT_SUPPORTED_X86 && !DT_SUPPORTED_ARMv8A && !DT_SUPPORTED_PPC64 && !DT_SUPPORTED_RISCV64 && !DT_SUPPORTED_LOONGARCH64
+#if !DT_SUPPORTED_X86 && !DT_SUPPORTED_ARMv8A && !DT_SUPPORTED_PPC64 && !DT_SUPPORTED_RISCV64 && !DT_SUPPORTED_LOONGARCH64 && !DT_SUPPORTED_MIPS64
 #error "Unfortunately we only work on amd64, ARMv8-A, PPC64 (64-bit little-endian only), riscv64 and loongarch64"
 #endif
 
 #undef DT_SUPPORTED_LOONGARCH64
 #undef DT_SUPPORTED_RISCV64
 #undef DT_SUPPORTED_PPC64
+#undef DT_SUPPORTED_MIPS64
 #undef DT_SUPPORTED_ARMv8A
 #undef DT_SUPPORTED_X86
 
-- 
2.45.2

