From fe2144cd2647757ce96a326872b47cea41f8e9e6 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Tue, 20 Dec 2022 01:04:49 -0500
Subject: [PATCH 2/2] fix building for mips64 and loongarch64

---
 src/is_supported_platform.h | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/src/is_supported_platform.h b/src/is_supported_platform.h
index 03a77f704d..275c12a28d 100644
--- a/src/is_supported_platform.h
+++ b/src/is_supported_platform.h
@@ -36,6 +36,18 @@
 #define DT_SUPPORTED_ARMv8A 0
 #endif
 
+#if defined(__loongarch__) && __loongarch_grlen == 64
+#define DT_SUPPORTED_LOONGARCH64 1
+#else
+#define DT_SUPPORTED_LOONGARCH64 0
+#endif
+
+#if defined(__mips__) && defined(__mips64)
+#define DT_SUPPORTED_MIPS64 1
+#else
+#define DT_SUPPORTED_MIPS64 0
+#endif
+
 #if defined(__PPC64__)
 #define DT_SUPPORTED_PPC64 1
 #else
@@ -48,16 +60,17 @@
 #define DT_SUPPORTED_RISCV64 0
 #endif
 
-#if (DT_SUPPORTED_X86 + DT_SUPPORTED_ARMv8A + DT_SUPPORTED_PPC64 + DT_SUPPORTED_RISCV64) > 1
+#if (DT_SUPPORTED_X86 + DT_SUPPORTED_ARMv8A + DT_SUPPORTED_LOONGARCH64 + DT_SUPPORTED_MIPS64 + DT_SUPPORTED_PPC64 + DT_SUPPORTED_RISCV64) > 1
 #error "Looks like hardware platform detection macros are broken?"
 #endif
 
-#if !DT_SUPPORTED_X86 && !DT_SUPPORTED_ARMv8A && !DT_SUPPORTED_PPC64 && !DT_SUPPORTED_RISCV64
+#if !DT_SUPPORTED_X86 && !DT_SUPPORTED_ARMv8A && !DT_SUPPORTED_LOONGARCH64 && !DT_SUPPORTED_MIPS64 && !DT_SUPPORTED_PPC64 && !DT_SUPPORTED_RISCV64
 #error "Unfortunately we only work on amd64, ARMv8-A, PPC64 (64-bit little-endian only) and riscv64"
 #endif
 
 #undef DT_SUPPORTED_RISCV64
 #undef DT_SUPPORTED_PPC64
+#undef DT_SUPPORTED_MIPS64
 #undef DT_SUPPORTED_ARMv8A
 #undef DT_SUPPORTED_X86
 
-- 
2.44.0

