diff --git a/CMakeLists.txt b/CMakeLists.txt
index 248ddd8..38f4780 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,7 +42,6 @@ endif()

 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-add_compile_options(-Wall -Wno-error=ignored-attributes -Werror)
 # To set addition RUNPATH in libraries
 # installed in /opt/rocm-ver/lib/roctracer
 set(ROCM_APPEND_PRIVLIB_RPATH "$ORIGIN/..")
@@ -60,11 +59,6 @@ endif()
 ## Build libraries
 add_subdirectory(src)
 
-## Build tests
-if(${LIBRARY_TYPE} STREQUAL SHARED)
-  add_subdirectory(test)
-endif()
-
 ## Build Plugins
 add_subdirectory(plugin)
 
