diff -Naur tdebase-trinity-14.0.10/ksysguard/ksysguardd/Linux/CMakeLists.txt tdebase-trinity-14.0.10.libbsd/ksysguard/ksysguardd/Linux/CMakeLists.txt
--- tdebase-trinity-14.0.10/ksysguard/ksysguardd/Linux/CMakeLists.txt	2021-04-23 21:28:04.000000000 -0700
+++ tdebase-trinity-14.0.10.libbsd/ksysguard/ksysguardd/Linux/CMakeLists.txt	2021-06-26 11:38:20.840292911 -0700
@@ -25,6 +25,7 @@
   ${CMAKE_SOURCE_DIR}/ksysguard/ksysguardd
 )
 
+LINK_LIBRARIES(bsd)
 
 ##### ksysguardd (static) #######################
 
