From 744a6dd979b90148fac7db9168a190354b496ec9 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 13 Jan 2025 15:14:51 +0800
Subject: [PATCH 06/13] feat: always not portable

---
 src-tauri/src/utils/dirs.rs | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/src-tauri/src/utils/dirs.rs b/src-tauri/src/utils/dirs.rs
index 2e986e1a..f06055f1 100644
--- a/src-tauri/src/utils/dirs.rs
+++ b/src-tauri/src/utils/dirs.rs
@@ -26,16 +26,6 @@ pub static PROFILE_YAML: &str = "profiles.yaml";
 
 /// init portable flag
 pub fn init_portable_flag() -> Result<()> {
-    use tauri::utils::platform::current_exe;
-
-    let app_exe = current_exe()?;
-    if let Some(dir) = app_exe.parent() {
-        let dir = PathBuf::from(dir).join(".config/PORTABLE");
-
-        if dir.exists() {
-            PORTABLE_FLAG.get_or_init(|| true);
-        }
-    }
     PORTABLE_FLAG.get_or_init(|| false);
     Ok(())
 }
-- 
2.52.0

