From 53a3453ddc958c1949533aa42649db7de7f274ca Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 13 Jan 2025 14:36:28 +0800
Subject: [PATCH 07/23] fix(check.mjs): do not check Mihomo (Clash Meta)
 platform support

---
 scripts/prebuild.mjs | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/scripts/prebuild.mjs b/scripts/prebuild.mjs
index e5b29146..49ec5987 100644
--- a/scripts/prebuild.mjs
+++ b/scripts/prebuild.mjs
@@ -148,21 +148,6 @@ async function getLatestReleaseVersion() {
   }
 }
 
-/*
- * check available
- */
-if (!META_MAP[`${platform}-${arch}`]) {
-  throw new Error(
-    `clash meta alpha unsupported platform "${platform}-${arch}"`,
-  );
-}
-
-if (!META_ALPHA_MAP[`${platform}-${arch}`]) {
-  throw new Error(
-    `clash meta alpha unsupported platform "${platform}-${arch}"`,
-  );
-}
-
 /**
  * core info
  */
-- 
2.51.0

