From 7fecfc360b907b5378496e8b796685834009b643 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Tue, 14 Jan 2025 10:16:54 +0800
Subject: [PATCH 12/19] Drop "Upgrade core" button

---
 .../setting/mods/clash-core-viewer.tsx        | 28 -------------------
 1 file changed, 28 deletions(-)

diff --git a/src/components/setting/mods/clash-core-viewer.tsx b/src/components/setting/mods/clash-core-viewer.tsx
index f409acbf..56969915 100644
--- a/src/components/setting/mods/clash-core-viewer.tsx
+++ b/src/components/setting/mods/clash-core-viewer.tsx
@@ -82,22 +82,6 @@ export function ClashCoreViewer({ ref }: { ref?: Ref<DialogRef> }) {
     }
   });
 
-  const onUpgrade = useLockFn(async () => {
-    try {
-      setUpgrading(true);
-      await upgradeCore();
-      setUpgrading(false);
-      showNotice("success", t(`Core Version Updated`));
-    } catch (err: any) {
-      setUpgrading(false);
-      const errMsg = err.response?.data?.message || err.toString();
-      const showMsg = errMsg.includes("already using latest version")
-        ? "Already Using Latest Core Version"
-        : errMsg;
-      showNotice("error", t(showMsg));
-    }
-  });
-
   return (
     <BaseDialog
       open={open}
@@ -105,18 +89,6 @@ export function ClashCoreViewer({ ref }: { ref?: Ref<DialogRef> }) {
         <Box display="flex" justifyContent="space-between">
           {t("Clash Core")}
           <Box>
-            <LoadingButton
-              variant="contained"
-              size="small"
-              startIcon={<SwitchAccessShortcutRounded />}
-              loadingPosition="start"
-              loading={upgrading}
-              disabled={restarting || changingCore !== null}
-              sx={{ marginRight: "8px" }}
-              onClick={onUpgrade}
-            >
-              {t("Upgrade")}
-            </LoadingButton>
             <LoadingButton
               variant="contained"
               size="small"
-- 
2.51.1

