From 7513e00ef23c0314a2dd54c944a404d2148a2629 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 13 Jan 2025 16:52:29 +0800
Subject: [PATCH 13/23] Unset all release profile

---
 src-tauri/Cargo.toml | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src-tauri/Cargo.toml b/src-tauri/Cargo.toml
index 5b385104..a22703af 100755
--- a/src-tauri/Cargo.toml
+++ b/src-tauri/Cargo.toml
@@ -111,13 +111,6 @@ custom-protocol = ["tauri/custom-protocol"]
 verge-dev = []
 tokio-trace = ["console-subscriber"]
 
-[profile.release]
-panic = "abort"
-codegen-units = 1
-lto = true
-opt-level = "s"
-strip = true
-
 [profile.dev]
 incremental = true
 codegen-units = 256 # 增加编译单元，提升编译速度
-- 
2.51.0

