From 4359d3f0d5457b51c1ef0d7497c371a839c434b9 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 13 Jan 2025 16:52:29 +0800
Subject: [PATCH 13/22] Unset all release profile

---
 src-tauri/Cargo.toml | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src-tauri/Cargo.toml b/src-tauri/Cargo.toml
index c83b2c02..8a1da496 100755
--- a/src-tauri/Cargo.toml
+++ b/src-tauri/Cargo.toml
@@ -94,13 +94,6 @@ default = ["custom-protocol"]
 custom-protocol = ["tauri/custom-protocol"]
 verge-dev = []
 
-[profile.release]
-panic = "abort"
-codegen-units = 1
-lto = true
-opt-level = "s"
-strip = true
-
 [profile.dev]
 incremental = true
 
-- 
2.49.0

