From 7c735a2d85027354b75a4cb5d92951669c107e53 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 13 Jan 2025 16:52:29 +0800
Subject: [PATCH 13/23] Unset all release profile

---
 src-tauri/Cargo.toml | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src-tauri/Cargo.toml b/src-tauri/Cargo.toml
index 080451f8..4962b2e5 100755
--- a/src-tauri/Cargo.toml
+++ b/src-tauri/Cargo.toml
@@ -108,13 +108,6 @@ default = ["custom-protocol"]
 custom-protocol = ["tauri/custom-protocol"]
 verge-dev = []
 
-[profile.release]
-panic = "abort"
-codegen-units = 1
-lto = true
-opt-level = "s"
-strip = true
-
 [profile.dev]
 incremental = true
 codegen-units = 256 # 增加编译单元，提升编译速度
-- 
2.50.1

