From 8c0884617736271bdf697ac480deb88fbd5927a9 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Tue, 14 Jan 2025 10:16:54 +0800
Subject: [PATCH 15/23] Drop "Upgrade core" button

---
 .../setting/mods/clash-core-viewer.tsx        | 29 -------------------
 1 file changed, 29 deletions(-)

diff --git a/src/components/setting/mods/clash-core-viewer.tsx b/src/components/setting/mods/clash-core-viewer.tsx
index 8e95945d..21da1cd8 100644
--- a/src/components/setting/mods/clash-core-viewer.tsx
+++ b/src/components/setting/mods/clash-core-viewer.tsx
@@ -20,7 +20,6 @@ import {
 import { changeClashCore, restartCore } from "@/services/cmds";
 import {
   closeAllConnections,
-  upgradeCore,
   forceRefreshClashConfig,
 } from "@/services/cmds";
 import { showNotice } from "@/services/noticeService";
@@ -86,22 +85,6 @@ export const ClashCoreViewer = forwardRef<DialogRef>((props, ref) => {
     }
   });
 
-  const onUpgrade = useLockFn(async () => {
-    try {
-      setUpgrading(true);
-      await upgradeCore();
-      setUpgrading(false);
-      showNotice("success", t(`Core Version Updated`));
-    } catch (err: any) {
-      setUpgrading(false);
-      const errMsg = err.response?.data?.message || err.toString();
-      const showMsg = errMsg.includes("already using latest version")
-        ? "Already Using Latest Core Version"
-        : errMsg;
-      showNotice("error", t(showMsg));
-    }
-  });
-
   return (
     <BaseDialog
       open={open}
@@ -109,18 +92,6 @@ export const ClashCoreViewer = forwardRef<DialogRef>((props, ref) => {
         <Box display="flex" justifyContent="space-between">
           {t("Clash Core")}
           <Box>
-            <LoadingButton
-              variant="contained"
-              size="small"
-              startIcon={<SwitchAccessShortcutRounded />}
-              loadingPosition="start"
-              loading={upgrading}
-              disabled={restarting || changingCore !== null}
-              sx={{ marginRight: "8px" }}
-              onClick={onUpgrade}
-            >
-              {t("Upgrade")}
-            </LoadingButton>
             <LoadingButton
               variant="contained"
               size="small"
-- 
2.51.0

