From 0e3cf6975f96291fa203332aad9e44086ff8f2f2 Mon Sep 17 00:00:00 2001
From: Hisham <hisham@gobolinux.org>
Date: Fri, 12 Feb 2016 23:33:53 -0200
Subject: [PATCH] Fix crash when emptying column. Closes #381.

---
 MetersPanel.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/MetersPanel.c b/MetersPanel.c
index bdbe83e..302b6d1 100644
--- a/MetersPanel.c
+++ b/MetersPanel.c
@@ -50,7 +50,10 @@ static void MetersPanel_delete(Object* object) {
 void MetersPanel_setMoving(MetersPanel* this, bool moving) {
    Panel* super = (Panel*) this;
    this->moving = moving;
-   ((ListItem*)Panel_getSelected(super))->moving = moving;
+   ListItem* selected = (ListItem*)Panel_getSelected(super);
+   if (selected) {
+      selected->moving = moving;
+   }
    if (!moving) {
       Panel_setSelectionColor(super, CRT_colors[PANEL_SELECTION_FOCUS]);
       Panel_setDefaultBar(super);

