diff -Naur glib-1.2.10/gmain.c glib-1.2.10.glibc/gmain.c
--- glib-1.2.10/gmain.c	2001-02-27 00:00:21.000000000 -0600
+++ glib-1.2.10.glibc/gmain.c	2020-04-10 21:16:37.763224729 -0500
@@ -37,7 +37,7 @@
 /* #define G_MAIN_POLL_DEBUG */
 
 
-
+#include <poll.h>
 #include "glib.h"
 #include <sys/types.h>
 #include <time.h>
diff -Naur glib-1.2.10/gstrfuncs.c glib-1.2.10.glibc/gstrfuncs.c
--- glib-1.2.10/gstrfuncs.c	2001-02-27 00:00:22.000000000 -0600
+++ glib-1.2.10.glibc/gstrfuncs.c	2020-04-10 21:19:11.924142624 -0500
@@ -28,6 +28,8 @@
  * MT safe
  */
 
+#include <limits.h>
+
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
