diff --git a/panel-plugin/CMakeLists.txt b/panel-plugin/CMakeLists.txt
index d0bb976..95eefde 100644
--- a/panel-plugin/CMakeLists.txt
+++ b/panel-plugin/CMakeLists.txt
@@ -112,9 +112,6 @@ if(CMAKE_BUILD_TYPE_TOLOWER MATCHES "release|minsizerel|relwithdebinfo")
 	endif()
 
 	target_compile_definitions(whiskermenu PRIVATE G_DISABLE_CAST_CHECKS G_DISABLE_ASSERT)
-
-	add_custom_command(TARGET whiskermenu
-		POST_BUILD COMMAND ${CMAKE_STRIP} $<TARGET_FILE:whiskermenu>)
 endif()
 
 install(TARGETS whiskermenu LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}/xfce4/panel/plugins)
