From dd440dfb25e7bbff7ddf3a4c8c9cc29d609b1b67 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Thu, 7 Mar 2024 21:43:04 -0500
Subject: [PATCH] openToNanoVDB: take ConstPtr

---
 nanovdb/nanovdb/util/CreateNanoGrid.h | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/nanovdb/nanovdb/util/CreateNanoGrid.h b/nanovdb/nanovdb/util/CreateNanoGrid.h
index 7ad71c57..df6e9461 100644
--- a/nanovdb/nanovdb/util/CreateNanoGrid.h
+++ b/nanovdb/nanovdb/util/CreateNanoGrid.h
@@ -118,10 +118,10 @@ template <typename> struct MapToNano;
 /// @return Handle to the destination NanoGrid
 template<typename BufferT = HostBuffer>
 GridHandle<BufferT>
-openToNanoVDB(const openvdb::GridBase::Ptr& base,
-              StatsMode                     sMode = StatsMode::Default,
-              ChecksumMode                  cMode = ChecksumMode::Default,
-              int                           verbose = 0);
+openToNanoVDB(const openvdb::GridBase::ConstPtr& base,
+              StatsMode                          sMode = StatsMode::Default,
+              ChecksumMode                       cMode = ChecksumMode::Default,
+              int                                verbose = 0);
 #endif
 
 //================================================================================================
@@ -2026,10 +2026,10 @@ createNanoGrid(const SrcGridT &srcGrid,
 #if defined(NANOVDB_USE_OPENVDB) && !defined(__CUDACC__)
 template<typename BufferT>
 GridHandle<BufferT>
-openToNanoVDB(const openvdb::GridBase::Ptr& base,
-              StatsMode                     sMode,
-              ChecksumMode                  cMode,
-              int                           verbose)
+openToNanoVDB(const openvdb::GridBase::ConstPtr& base,
+              StatsMode                          sMode,
+              ChecksumMode                       cMode,
+              int                                verbose)
 {
     // We need to define these types because they are not defined in OpenVDB
     using openvdb_Vec4fTree = typename openvdb::tree::Tree4<openvdb::Vec4f, 5, 4, 3>::Type;
-- 
2.44.0

