From 9fc4b8e0fe1fb24c48373b632eca1e66ff1ea1b4 Mon Sep 17 00:00:00 2001
From: Jonathan Swartz <jonathan@jswartz.info>
Date: Mon, 22 Dec 2025 18:56:36 +1300
Subject: [PATCH] Fix python nanoToOpenVDB bindings

Signed-off-by: Jonathan Swartz <jonathan@jswartz.info>
---
 nanovdb/nanovdb/python/PyNanoToOpenVDB.cc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/nanovdb/nanovdb/python/PyNanoToOpenVDB.cc b/nanovdb/nanovdb/python/PyNanoToOpenVDB.cc
index 49dc464a05..07fb5285ef 100644
--- a/nanovdb/nanovdb/python/PyNanoToOpenVDB.cc
+++ b/nanovdb/nanovdb/python/PyNanoToOpenVDB.cc
@@ -20,9 +20,9 @@ template<typename BufferT> void defineNanoToOpenVDB(nb::module_& m)
 #ifdef NANOVDB_USE_OPENVDB
     // Wrap nanoToOpenVDB into a lambda to workaround an MSVC compiler bug
     m.def(
-        "nanoToOpenVDB", [](GridHandle<BufferT>& handle, int verbose, uint32_t n){
-            return tools::nanoToOpenVDB<BufferT>(handle, verbose, n);
-        }, "handle"_a, "verbose"_a = 0, "n"_a = 0);
+        "nanoToOpenVDB", [](GridHandle<BufferT>& handle, uint32_t n){
+            return tools::nanoToOpenVDB<BufferT>(handle, n);
+        }, "handle"_a, "n"_a = 0);
 #endif
 }
 

