From ee44aa5f9ce235d8b0f74d22de386c85684cab9a Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Sat, 19 Oct 2024 16:26:16 +0000
Subject: [PATCH] add slock support

---
 scripts/xdg-screensaver.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/scripts/xdg-screensaver.in b/scripts/xdg-screensaver.in
index e70942f..db0f8d0 100644
--- a/scripts/xdg-screensaver.in
+++ b/scripts/xdg-screensaver.in
@@ -110,6 +110,11 @@ perform_action()
     cinnamon)
       screensaver_cinnamon_screensaver "$1"
       ;;
+    
+    slock)
+      slock
+      exit 0
+      ;;
 
     xscreensaver)
       screensaver_xscreensaver "$1"
@@ -884,6 +889,8 @@ dbus-send --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.fr
 # Consider "cinnamon-screensaver" a separate DE
 dbus-send --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.GetNameOwner string:org.cinnamon.ScreenSaver > /dev/null 2>&1 && DE="cinnamon"
 # Consider "xautolock" a separate DE
+slock -v > /dev/null 2>&1 && DE="slock"
+# Consider "xautolock" a separate DE
 xautolock -enable > /dev/null 2>&1 && DE="xautolock_screensaver"
 
 if [ "$action" = "resume" ] ; then
-- 
2.47.0

