From d26524b96d56963d09459312e741f32e81aeb487 Mon Sep 17 00:00:00 2001
From: Elimar Riesebieter <riesebie@lxtec.de>
Date: Sun, 27 Nov 2016 12:19:29 +0100
Subject: [PATCH 02/11] [Debian] Adapted the opus patch to config.example.in

---
 config.example.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.example.in b/config.example.in
index 2dde585..55e5329 100644
--- a/config.example.in
+++ b/config.example.in
@@ -399,7 +399,7 @@
 #PreferredDecoders += ogg(vorbis,*,ffmpeg):oga(vorbis,*,ffmpeg):ogv(ffmpeg)
 #PreferredDecoders += application/ogg(vorbis):audio/ogg(vorbis)
 #PreferredDecoders += flac(flac,*,ffmpeg)
-#PreferredDecoders += opus(ffmpeg)
+#PreferredDecoders += opus(opus,ffmpeg)
 #PreferredDecoders += spx(speex)
 
 # Which resampling method to use.  There are a few methods of resampling
-- 
2.46.0

