From 15d7e4adf7750a79a9576c492b4a7e1bed904b1d Mon Sep 17 00:00:00 2001
From: Elimar Riesebieter <riesebie@lxtec.de>
Date: Mon, 30 Jul 2018 07:31:13 +0200
Subject: [PATCH 04/11] [Debian] Assign "Switch on/off play time percentage"

---
 keys.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/keys.c b/keys.c
index 7ff052e..c1a7dfa 100644
--- a/keys.c
+++ b/keys.c
@@ -204,8 +204,8 @@ static struct command commands[] = {
 		"toggle_percent",
 		"Switch on/off play time percentage",
 		CON_MENU,
-		{ -1 },
-		0
+		{CTRL('p'), -1 },
+		1
 	},
 	{
 		KEY_CMD_PLIST_ADD_FILE,
-- 
2.46.0

