From d59b6c79373ddf4d91f3787e3c091efa5cd229c1 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Mon, 26 May 2025 09:06:03 +0800
Subject: [PATCH 12/12] AOSCOS fix configure.ac add missing Gettext macro

---
 configure.ac | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure.ac b/configure.ac
index eb71bdf..cd80e59 100644
--- a/configure.ac
+++ b/configure.ac
@@ -381,6 +381,8 @@ dnl Decoder plugins
 m4_include(decoder_plugins/decoders.m4)
 
 dnl Require with iconv for charset translation.
+AM_GNU_GETTEXT([external])
+AM_GNU_GETTEXT_VERSION(0.25)
 AM_ICONV
 if test "x$am_cv_func_iconv" != xyes; then
 	AC_MSG_ERROR([No iconv library found.])
-- 
2.49.0

