--- a/IGC/VectorCompiler/cmake/spirv.cmake	2020-12-12 01:29:34.213085053 -0800
+++ b/IGC/VectorCompiler/cmake/spirv.cmake	2020-12-12 01:35:05.382723646 -0800
@@ -147,47 +147,18 @@
     message(FATAL_ERROR "[VC] Found unsupported version of LLVM (LLVM_VERSION_MAJOR is set to ${LLVM_VERSION_MAJOR})")
   endif()
 
-  if(NOT EXISTS ${SPIRV_COPY})
-    message(STATUS "[VC] : Copying stock SPIRV-Translator sources to ${SPIRV_COPY}")
-    execute_process(COMMAND ${CMAKE_COMMAND} -E copy_directory ${SPIRV_SOURCES} ${SPIRV_COPY})
-  endif()
-
-  apply_patches(${SPIRV_COPY}
-  ${SPRIV_PATCHES}
-  ${SPIRV_REV_PATCH}
-  ${SPRIV_BRANCH_PATCH}
-  )
-
   if(IGC_OPTION__FORCE_SYSTEM_LLVM)
-
-    ExternalProject_Add(SPIRVDLL_EX
-        PREFIX ${CMAKE_CURRENT_BINARY_DIR}/SPIRVDLL
-        SOURCE_DIR ${SPIRV_COPY}
-        CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/spirv-install
-        BUILD_COMMAND ${MAKE_EXEC} SPIRVDLL
-        INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/spirv-install
-      )
-
+    add_subdirectory(${SPIRV_COPY})
   else()
-
-    ExternalProject_Add(SPIRVDLL_EX
-        PREFIX ${CMAKE_CURRENT_BINARY_DIR}/SPIRVDLL
-        SOURCE_DIR ${SPIRV_COPY}
-        CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/spirv-install -DLLVM_DIR=${LLVM_DIR}
-        BUILD_COMMAND ${MAKE_EXEC} SPIRVDLL
-        INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/spirv-install
-      )
-
+    add_subdirectory(${SPIRV_SOURCES} "${SPIRV_COPY}")
+    add_dependencies(SPIRVDLL VCCodeGen)
   endif(IGC_OPTION__FORCE_SYSTEM_LLVM)
 
-  add_dependencies(SPIRVDLL_EX VCCodeGen)
-
   install(FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/spirv-install/lib/libSPIRVDLL.so
-    DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
-    COMPONENT igc-core
-  )
-
+      $<TARGET_FILE:SPIRVDLL>
+      DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
+      COMPONENT igc-core
+    )
 endif(DEFINED SPIRVDLL_SRC)
 
 elseif(NOT TARGET SPIRVDLL)
