From 7451912d6c34b1f967128ac96ddc8f57bfd54fb1 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Tue, 25 May 2021 17:21:11 -0600
Subject: [PATCH 2/3] use cmake chaining

---
 IGC/VectorCompiler/cmake/spirv.cmake | 40 +++++++---------------------
 1 file changed, 9 insertions(+), 31 deletions(-)

diff --git a/IGC/VectorCompiler/cmake/spirv.cmake b/IGC/VectorCompiler/cmake/spirv.cmake
index 65edffc83..b3307b0a0 100644
--- a/IGC/VectorCompiler/cmake/spirv.cmake
+++ b/IGC/VectorCompiler/cmake/spirv.cmake
@@ -154,49 +154,27 @@ else()
     set(SPIRV_REV_PATCH f9d9902e4bcbfb207d6d41fe9f7327ee64cfcb1b)
     set(SPRIV_PATCHES ${CMAKE_CURRENT_SOURCE_DIR}/spirv-patches-11/)
     set(SPRIV_BRANCH_PATCH spirvdll_110)
+  elseif(${LLVM_VERSION_MAJOR} EQUAL 12)
+    message(STATUS "[VC] Found LLVM version 12")
   else()
     message(FATAL_ERROR "[VC] Found unsupported version of LLVM (LLVM_VERSION_MAJOR is set to ${LLVM_VERSION_MAJOR})")
   endif()
 
-  if(NOT EXISTS ${SPIRV_COPY})
-    message(STATUS "[VC] : Copying stock SPIRV-Translator sources to ${SPIRV_COPY}")
-    execute_process(COMMAND ${CMAKE_COMMAND} -E copy_directory ${SPIRV_SOURCES} ${SPIRV_COPY})
-  endif()
-
-  apply_patches(${SPIRV_COPY}
-  ${SPRIV_PATCHES}
-  ${SPIRV_REV_PATCH}
-  ${SPRIV_BRANCH_PATCH}
-  )
-
   if(IGC_BUILD__LLVM_PREBUILDS)
 
-    ExternalProject_Add(SPIRVDLL_EX
-        PREFIX ${CMAKE_CURRENT_BINARY_DIR}/SPIRVDLL
-        SOURCE_DIR ${SPIRV_COPY}
-        CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/spirv-install
-        BUILD_COMMAND ${MAKE_EXEC} SPIRVDLL
-        INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/spirv-install
-      )
+    add_subdirectory(${SPIRV_SOURCES} "${SPIRV_COPY}")
+    add_dependencies(SPIRVDLL VCCodeGen)
 
   else()
 
-    ExternalProject_Add(SPIRVDLL_EX
-        PREFIX ${CMAKE_CURRENT_BINARY_DIR}/SPIRVDLL
-        SOURCE_DIR ${SPIRV_COPY}
-        CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/spirv-install -DLLVM_DIR=${LLVM_DIR}
-        BUILD_COMMAND ${MAKE_EXEC} SPIRVDLL
-        INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/spirv-install
-      )
-
+    add_subdirectory(${SPIRV_SOURCES} "${SPIRV_COPY}")
+    add_dependencies(SPIRVDLL VCCodeGen)
   endif(IGC_BUILD__LLVM_PREBUILDS)
 
-  add_dependencies(SPIRVDLL_EX VCCodeGen)
-
   install(FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/spirv-install/lib/libSPIRVDLL.so
-    DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
-    COMPONENT igc-core
+      $<TARGET_FILE:SPIRVDLL>
+      DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
+      COMPONENT igc-core
   )
 
 endif(DEFINED SPIRVDLL_SRC)
-- 
2.32.0

