--- intel-graphics-compiler-igc-1.0.9441/IGC/VectorCompiler/cmake/spirv.cmake  2021-11-12 09:27:55.000000000 +0000
+++ intel-graphics-compiler-igc-1.0.9441-new/IGC/VectorCompiler/cmake/spirv.cmake      2021-12-06 08:43:33.913487984 +0000
@@ -168,43 +168,20 @@
     message(FATAL_ERROR "[VC] Found unsupported version of LLVM (LLVM_VERSION_MAJOR is set to ${LLVM_VERSION_MAJOR})")
   endif()
 
-  if(NOT EXISTS ${SPIRV_COPY})
-    message(STATUS "[VC] : Copying stock SPIRV-Translator sources to ${SPIRV_COPY}")
-    execute_process(COMMAND ${CMAKE_COMMAND} -E copy_directory ${SPIRV_SOURCES} ${SPIRV_COPY})
-  endif()
-
-  apply_patches(${SPIRV_COPY}
-  ${SPRIV_PATCHES}
-  ${SPIRV_REV_PATCH}
-  ${SPRIV_BRANCH_PATCH}
-  )
-
   if(IGC_BUILD__LLVM_PREBUILDS)
 
-    ExternalProject_Add(SPIRVDLL_EX
-        PREFIX ${CMAKE_CURRENT_BINARY_DIR}/SPIRVDLL
-        SOURCE_DIR ${SPIRV_COPY}
-        CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/spirv-install -DBASE_LLVM_VERSION=${IGC_OPTION__LLVM_PREFERRED_VERSION}
-        BUILD_COMMAND ${MAKE_EXEC} SPIRVDLL
-        INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/spirv-install
-      )
+  add_subdirectory(${SPIRV_SOURCES} "${SPIRV_COPY}")
+  add_dependencies(SPIRVDLL VCCodeGen)
 
   else()
 
-    ExternalProject_Add(SPIRVDLL_EX
-        PREFIX ${CMAKE_CURRENT_BINARY_DIR}/SPIRVDLL
-        SOURCE_DIR ${SPIRV_COPY}
-        CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/spirv-install -DLLVM_DIR=${LLVM_DIR}
-        BUILD_COMMAND ${MAKE_EXEC} SPIRVDLL
-        INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/spirv-install
-      )
+  add_subdirectory(${SPIRV_SOURCES} "${SPIRV_COPY}")
+  add_dependencies(SPIRVDLL VCCodeGen)
 
   endif(IGC_BUILD__LLVM_PREBUILDS)
 
-  add_dependencies(SPIRVDLL_EX VCCodeGen)
-
   install(FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/spirv-install/lib/libSPIRVDLL.so
+    $<TARGET_FILE:SPIRVDLL>
     DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
     COMPONENT igc-core
   )
