--- intel-graphics-compiler-igc-1.0.9441/IGC/CMakeLists.txt	2021-11-12 09:27:55.000000000 +0000
+++ intel-graphics-compiler-igc-1.0.9441-new/IGC/CMakeLists.txt	2021-12-07 00:24:23.424842607 +0000
@@ -1497,7 +1497,7 @@
 if(IGC_OPTION__LINK_KHRONOS_SPIRV_TRANSLATOR)
   target_link_libraries("${IGC_BUILD__PROJ__igc_dll}" PRIVATE LLVMSPIRVLib)
 endif()
-
+target_link_libraries(${IGC_BUILD__PROJ__igc_dll} PRIVATE lldELF lldCommon z)
 if(IGC_BUILD__LLVM_SOURCES AND IGC_OPTION__LLVM_LLD)
   if(UNIX)
     if(NOT ${CMAKE_BUILD_TYPE} EQUAL "Release")
