From 6aa2a6bac25a3a0f8fed240bb956c89aee68ee16 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Sat, 10 Jul 2021 20:54:31 -0600
Subject: [PATCH] IGC: linking to pthread

---
 IGC/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/IGC/CMakeLists.txt b/IGC/CMakeLists.txt
index fd05f0ba4..92644d8e2 100644
--- a/IGC/CMakeLists.txt
+++ b/IGC/CMakeLists.txt
@@ -1493,7 +1493,7 @@ target_include_directories(${IGC_BUILD__PROJ__igc_dll} INTERFACE
   "${CMAKE_CURRENT_SOURCE_DIR}/AdaptorOCL/"
   "${IGC_BUILD__IGC_BIN_DIR}/AdaptorOCL/"
 )
-
+target_link_libraries(${IGC_BUILD__PROJ__igc_dll} PRIVATE pthread)
 
 #IGC_DLL
 igc_target_flag_property_add_once_config_var("${IGC_BUILD__PROJ__igc_dll}" LINK_FLAGS IGC_BUILD__EXPORT_SYMBOLS_LINK_FLAGS)
-- 
2.32.0

