From 8e976aaa21dd6ca3328bdb2aebe5c4211a5a38d8 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Wed, 14 Jul 2021 02:29:04 -0600
Subject: [PATCH] igc: fix issues with intel-compute-runtime

---
 IGC/AdaptorOCL/UnifyIROCL.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/IGC/AdaptorOCL/UnifyIROCL.cpp b/IGC/AdaptorOCL/UnifyIROCL.cpp
index e9f31b3a0..676d551e2 100644
--- a/IGC/AdaptorOCL/UnifyIROCL.cpp
+++ b/IGC/AdaptorOCL/UnifyIROCL.cpp
@@ -169,6 +169,7 @@ static void CommonOCLBasedPasses(
 #endif
 
     COMPILER_TIME_START(pContext, TIME_UnificationPasses);
+    initializeScalarOpts(*PassRegistry::getPassRegistry());
 
     unify_opt_PreProcess(pContext);
 
-- 
2.32.0

