From 4bc3019f8c70fb94881d0ac83f959e50ef59bba0 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Wed, 14 Jul 2021 02:29:04 -0600
Subject: [PATCH 5/5] igc: fix issues with intel-compute-runtime

---
 IGC/AdaptorOCL/UnifyIROCL.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/IGC/AdaptorOCL/UnifyIROCL.cpp b/IGC/AdaptorOCL/UnifyIROCL.cpp
index ea2ff95e3..43f8578c2 100644
--- a/IGC/AdaptorOCL/UnifyIROCL.cpp
+++ b/IGC/AdaptorOCL/UnifyIROCL.cpp
@@ -170,6 +170,7 @@ static void CommonOCLBasedPasses(
 #endif
 
     COMPILER_TIME_START(pContext, TIME_UnificationPasses);
+    initializeScalarOpts(*PassRegistry::getPassRegistry());
 
     unify_opt_PreProcess(pContext);
 
-- 
2.32.0

