diff --git a/Makefile b/Makefile
index 458c524..c8181e7 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ OBJ = $(SRC:.c=.o)
 all: dmenu stest
 
 .c.o:
-	$(CC) -c $(CFLAGS) $<
+	$(CC) -c $(CFLAGS_) $<
 
 config.h:
 	cp config.def.h $@
@@ -17,10 +17,10 @@ config.h:
 $(OBJ): arg.h config.h config.mk drw.h
 
 dmenu: dmenu.o drw.o util.o
-	$(CC) -o $@ dmenu.o drw.o util.o $(LDFLAGS)
+	$(CC) -o $@ dmenu.o drw.o util.o $(LDFLAGS_)
 
 stest: stest.o
-	$(CC) -o $@ stest.o $(LDFLAGS)
+	$(CC) -o $@ stest.o $(LDFLAGS_)
 
 clean:
 	rm -f dmenu stest $(OBJ) dmenu-$(VERSION).tar.gz
diff --git a/config.mk b/config.mk
index 137f7c8..2a9130b 100644
--- a/config.mk
+++ b/config.mk
@@ -25,8 +25,10 @@ LIBS = -L$(X11LIB) -lX11 $(XINERAMALIBS) $(FREETYPELIBS)
 
 # flags
 CPPFLAGS = -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_XOPEN_SOURCE=700 -D_POSIX_C_SOURCE=200809L -DVERSION=\"$(VERSION)\" $(XINERAMAFLAGS)
-CFLAGS   = -std=c99 -pedantic -Wall -Os $(INCS) $(CPPFLAGS)
-LDFLAGS  = $(LIBS)
 
+CFLAGS   ?= -std=c99 -pedantic -Wall -Os
+CFLAGS_  = $(CFLAGS) $(INCS) $(CPPFLAGS)
+LDFLAGS  ?=
+LDFLAGS_ = $(LDFLAGS) $(LIBS)
 # compiler and linker
 CC = cc
