From 5a3c91ad60060b65a19af8a34c0b2682f49673c4 Mon Sep 17 00:00:00 2001
From: Adrian Bunk <bunk@debian.org>
Date: Wed, 25 Aug 2021 17:24:07 +0300
Subject: [PATCH] x86: Don't pass -msse -msse2 with --disable-sse

Signed-off-by: Camber Huang <camber@poi.science>
---
 build/ck.build.x86 | 2 +-
 configure          | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/build/ck.build.x86 b/build/ck.build.x86
index 6e12783..4d07fa4 100644
--- a/build/ck.build.x86
+++ b/build/ck.build.x86
@@ -1,2 +1,2 @@
-CFLAGS+=-m32 -D__x86__ -msse -msse2
+CFLAGS+=-m32 -D__x86__
 LDFLAGS+=-m32
diff --git a/configure b/configure
index 6a22dca..742adfe 100755
--- a/configure
+++ b/configure
@@ -424,6 +424,9 @@ case $PLATFORM in
 		LDFLAGS="-m64 $LDFLAGS"
 		;;
 	i386|i486|i586|i686|i586_i686|pentium*|athlon*|k5|k6|k6_2|k6_3)
+		if test "${SSE_DISABLE}" = "CK_MD_SSE_ENABLE"; then
+			CFLAGS="$CFLAGS -msse -msse2"
+		fi
 		LSE_ENABLE="CK_MD_LSE_DISABLE"
 		MM="${MM:-"CK_MD_TSO"}"
 		case $SYSTEM in
-- 
2.37.1

