diff -Naur oxygen-icons5-5.39.0/CMakeLists.txt oxygen-icons5-5.39.0.notest/CMakeLists.txt
--- oxygen-icons5-5.39.0/CMakeLists.txt	2017-10-07 11:24:55.000000000 +0000
+++ oxygen-icons5-5.39.0.notest/CMakeLists.txt	2017-10-25 04:26:19.488372360 +0000
@@ -20,8 +20,6 @@
 
 set( oxygen_icon_dirs  8x8 16x16 22x22 32x32 48x48 64x64 128x128 256x256 )
 
-add_subdirectory(autotests)
-
 ########### install files ###############
 
 set(OXYGEN_INSTALL_DIR ${KDE_INSTALL_ICONDIR}/oxygen/base)

