Description: Don't list volumes option on LTSP configs
Bug: https://launchpad.net/bugs/220564

Index: nautilus-3.8.2/libnautilus-private/nautilus-file.c
===================================================================
--- nautilus-3.8.2.orig/libnautilus-private/nautilus-file.c	2014-01-15 14:31:31.473606184 +1300
+++ nautilus-3.8.2/libnautilus-private/nautilus-file.c	2014-01-15 14:31:31.465606184 +1300
@@ -1018,6 +1018,10 @@
 {
 	g_return_val_if_fail (NAUTILUS_IS_FILE (file), FALSE);
 
+	if (getenv("LTSP_CLIENT")) {
+		return FALSE;
+	}
+
 	return file->details->can_unmount ||
 		(file->details->mount != NULL &&
 		 g_mount_can_unmount (file->details->mount));
@@ -1028,6 +1032,10 @@
 {
 	g_return_val_if_fail (NAUTILUS_IS_FILE (file), FALSE);
 
+	if (getenv("LTSP_CLIENT")) {
+		return FALSE;
+	}
+
 	return file->details->can_eject ||
 		(file->details->mount != NULL &&
 		 g_mount_can_eject (file->details->mount));
