Index: nautilus-3.5.5/data/nautilus-autostart.desktop.in
===================================================================
--- nautilus-3.5.5.orig/data/nautilus-autostart.desktop.in	2012-05-08 06:28:08.000000000 +0200
+++ nautilus-3.5.5/data/nautilus-autostart.desktop.in	2012-08-08 09:41:29.991206859 +0200
@@ -4,3 +4,4 @@
 Exec=nautilus -n
 OnlyShowIn=GNOME;Unity;
 AutostartCondition=GSettings org.gnome.desktop.background show-desktop-icons
+NoDisplay=true
