Index: nautilus-3.15.90/src/nautilus-progress-ui-handler.c
===================================================================
--- nautilus-3.15.90.orig/src/nautilus-progress-ui-handler.c
+++ nautilus-3.15.90/src/nautilus-progress-ui-handler.c
@@ -156,6 +156,7 @@ static void
 progress_ui_handler_ensure_window (NautilusProgressUIHandler *self)
 {
 	GtkWidget *progress_dialog;
+	const gchar *desktop_environment = g_getenv ("DESKTOP_SESSION");
 	
 	if (self->priv->progress_dialog != NULL) {
 		return;
@@ -175,6 +176,10 @@ progress_ui_handler_ensure_window (Nauti
 				 GTK_WIN_POS_CENTER);
 	gtk_window_set_icon_name (GTK_WINDOW (progress_dialog),
 				"system-file-manager");
+	if ((!g_strcmp0(desktop_environment, "ubuntu")) ||
+	       (!g_strcmp0(desktop_environment, "ubuntu-2d")))
+	    gtk_window_set_skip_taskbar_hint (GTK_WINDOW (progress_window),
+				TRUE);
 
 	self->priv->content_area = gtk_dialog_get_content_area (GTK_DIALOG (self->priv->progress_dialog));
 
