Index: nautilus-3.8.2/data/org.gnome.Nautilus.desktop.in.in
===================================================================
--- nautilus-3.8.2.orig/data/org.gnome.Nautilus.desktop.in.in	2014-01-15 14:31:44.521605732 +1300
+++ nautilus-3.8.2/data/org.gnome.Nautilus.desktop.in.in	2014-01-15 14:31:44.517605732 +1300
@@ -15,3 +15,9 @@
 X-GNOME-Bugzilla-Component=general
 X-GNOME-Bugzilla-Version=@VERSION@
 X-GNOME-UsesNotifications=true
+Actions=Window;
+
+[Desktop Action Window]
+_Name=Open a New Window
+Exec=nautilus --new-window
+OnlyShowIn=Unity;
