Index: nautilus-3.8.1/data/org.gnome.Nautilus.desktop.in.in
===================================================================
--- nautilus-3.8.1.orig/data/org.gnome.Nautilus.desktop.in.in	2013-06-04 22:27:44.308397907 -0400
+++ nautilus-3.8.1/data/org.gnome.Nautilus.desktop.in.in	2013-06-04 22:27:44.304397907 -0400
@@ -16,6 +16,7 @@
 X-GNOME-Bugzilla-Version=@VERSION@
 X-GNOME-UsesNotifications=true
 Actions=Window;
+X-Unity-IconBackgroundColor=#af4853
 
 [Desktop Action Window]
 _Name=Open a New Window
