From a1b42a6756d478492303a89273236b2a8c56f387 Mon Sep 17 00:00:00 2001
From: Nelson Benítez León <nbenitezl+gnome@gmail.com>
Date: Sun, 10 Nov 2013 15:34:29 +0000
Subject: properties: fix crash when entering an unsupported URI

https://bugzilla.gnome.org/show_bug.cgi?id=711480
---
diff --git a/src/nautilus-properties-window.c b/src/nautilus-properties-window.c
index 4cf7244..df8d69f 100644
--- a/src/nautilus-properties-window.c
+++ b/src/nautilus-properties-window.c
@@ -5406,7 +5406,11 @@ custom_icon_file_chooser_response_cb (GtkDialog *dialog,
 
 	case GTK_RESPONSE_OK:
 		uri = gtk_file_chooser_get_uri (GTK_FILE_CHOOSER (dialog));
-		set_icon (uri, window);
+		if (uri != NULL) {
+			set_icon (uri, window);
+		} else {
+			reset_icon (window);
+		}
 		g_free (uri);
 		break;
 
--
cgit v0.9.2
