Index: nautilus-3.10.1/src/nautilus-trash-bar.c
===================================================================
--- nautilus-3.10.1.orig/src/nautilus-trash-bar.c	2014-02-24 13:47:09.832935047 +0100
+++ nautilus-3.10.1/src/nautilus-trash-bar.c	2014-02-24 13:47:09.828935047 +0100
@@ -229,6 +229,6 @@
 {
 	return g_object_new (NAUTILUS_TYPE_TRASH_BAR,
 			     "view", view,
-			     "message-type", GTK_MESSAGE_QUESTION,
+			     "message-type", GTK_MESSAGE_OTHER,
 			     NULL);
 }
Index: nautilus-3.10.1/src/nautilus-x-content-bar.c
===================================================================
--- nautilus-3.10.1.orig/src/nautilus-x-content-bar.c	2014-02-24 13:47:09.832935047 +0100
+++ nautilus-3.10.1/src/nautilus-x-content-bar.c	2014-02-24 13:47:09.828935047 +0100
@@ -368,7 +368,7 @@
 			    const char **x_content_types)
 {
 	return g_object_new (NAUTILUS_TYPE_X_CONTENT_BAR,
-			     "message-type", GTK_MESSAGE_QUESTION,
+			     "message-type", GTK_MESSAGE_OTHER,
 			     "mount", mount,
 			     "x-content-types", x_content_types,
 			     NULL);
Index: nautilus-3.10.1/src/nautilus-special-location-bar.c
===================================================================
--- nautilus-3.10.1.orig/src/nautilus-special-location-bar.c	2013-04-02 15:48:28.000000000 +0200
+++ nautilus-3.10.1/src/nautilus-special-location-bar.c	2014-02-24 13:47:59.308936695 +0100
@@ -163,7 +163,7 @@
 nautilus_special_location_bar_new (NautilusSpecialLocation location)
 {
 	return g_object_new (NAUTILUS_TYPE_SPECIAL_LOCATION_BAR,
-			     "message-type", GTK_MESSAGE_QUESTION,
+			     "message-type", GTK_MESSAGE_OTHER,
 			     "special-location", location,
 			     NULL);
 }
