Description: Only run tracker search engine on GNOME sessions
Author: Tim Lunn <tim@feathertop.org>

Index: nautilus-3.10.1/libnautilus-private/nautilus-search-engine.c
===================================================================
--- nautilus-3.10.1.orig/libnautilus-private/nautilus-search-engine.c
+++ nautilus-3.10.1/libnautilus-private/nautilus-search-engine.c
@@ -66,7 +66,8 @@ nautilus_search_engine_set_query (Nautil
 {
 	NautilusSearchEngine *engine = NAUTILUS_SEARCH_ENGINE (provider);
 #ifdef ENABLE_TRACKER
-	nautilus_search_provider_set_query (NAUTILUS_SEARCH_PROVIDER (engine->details->tracker), query);
+	if (g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "GNOME") == 0)
+		nautilus_search_provider_set_query (NAUTILUS_SEARCH_PROVIDER (engine->details->tracker), query);
 #endif
 	nautilus_search_provider_set_query (NAUTILUS_SEARCH_PROVIDER (engine->details->model), query);
 	nautilus_search_provider_set_query (NAUTILUS_SEARCH_PROVIDER (engine->details->simple), query);
@@ -86,8 +87,10 @@ search_engine_start_real (NautilusSearch
 	g_object_ref (engine);
 
 #ifdef ENABLE_TRACKER
-	nautilus_search_provider_start (NAUTILUS_SEARCH_PROVIDER (engine->details->tracker));
-	engine->details->providers_running++;
+	if (g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "GNOME") == 0) {
+		nautilus_search_provider_start (NAUTILUS_SEARCH_PROVIDER (engine->details->tracker));
+		engine->details->providers_running++;
+	}
 #endif
 	if (nautilus_search_engine_model_get_model (engine->details->model)) {
 		nautilus_search_provider_start (NAUTILUS_SEARCH_PROVIDER (engine->details->model));
@@ -134,7 +137,8 @@ nautilus_search_engine_stop (NautilusSea
 	DEBUG ("Search engine stop");
 
 #ifdef ENABLE_TRACKER
-	nautilus_search_provider_stop (NAUTILUS_SEARCH_PROVIDER (engine->details->tracker));
+	if (g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "GNOME") == 0)
+		nautilus_search_provider_stop (NAUTILUS_SEARCH_PROVIDER (engine->details->tracker));
 #endif
 	nautilus_search_provider_stop (NAUTILUS_SEARCH_PROVIDER (engine->details->model));
 	nautilus_search_provider_stop (NAUTILUS_SEARCH_PROVIDER (engine->details->simple));
@@ -258,7 +262,8 @@ nautilus_search_engine_finalize (GObject
 	g_hash_table_destroy (engine->details->uris);
 
 #ifdef ENABLE_TRACKER
-	g_clear_object (&engine->details->tracker);
+	if (g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "GNOME") == 0)
+		g_clear_object (&engine->details->tracker);
 #endif
 	g_clear_object (&engine->details->model);
 	g_clear_object (&engine->details->simple);
@@ -288,8 +293,10 @@ nautilus_search_engine_init (NautilusSea
 	engine->details->uris = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
 
 #ifdef ENABLE_TRACKER
-	engine->details->tracker = nautilus_search_engine_tracker_new ();
-	connect_provider_signals (engine, NAUTILUS_SEARCH_PROVIDER (engine->details->tracker));
+	if (g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "GNOME") == 0) {
+		engine->details->tracker = nautilus_search_engine_tracker_new ();
+		connect_provider_signals (engine, NAUTILUS_SEARCH_PROVIDER (engine->details->tracker));
+	}
 #endif
 	engine->details->model = nautilus_search_engine_model_new ();
 	connect_provider_signals (engine, NAUTILUS_SEARCH_PROVIDER (engine->details->model));
