From b97f085565bac06d90a8577c3e3b9c2969ab3456 Mon Sep 17 00:00:00 2001
From: Sebastien Noël <sebastien@twolife.be>
Date: Tue, 28 Jan 2014 19:08:31 +0000
Subject: Restore backspace for going to parent folder

This restores the behaviour from nautilus <= 3.4.
---
Index: nautilus-3.14.0/src/nautilus-list-view.c
===================================================================
--- nautilus-3.14.0.orig/src/nautilus-list-view.c
+++ nautilus-3.14.0/src/nautilus-list-view.c
@@ -1904,6 +1904,7 @@ create_and_set_up_tree_view (NautilusLis
 {
 	GtkCellRenderer *cell;
 	GtkTreeViewColumn *column;
+	GtkBindingSet *binding_set;
 	AtkObject *atk_obj;
 	GList *nautilus_columns;
 	GList *l;
@@ -1916,6 +1917,10 @@ create_and_set_up_tree_view (NautilusLis
 							NULL);
 	gtk_tree_view_set_enable_search (view->details->tree_view, FALSE);
 
+	/* Don't handle backspace key. It's used to open the parent folder. */
+	binding_set = gtk_binding_set_by_class (GTK_WIDGET_GET_CLASS (view->details->tree_view));
+	gtk_binding_entry_remove (binding_set, GDK_KEY_BackSpace, 0);
+
 	view->details->drag_dest = 
 		nautilus_tree_view_drag_dest_new (view->details->tree_view);
 
Index: nautilus-3.14.0/src/nautilus-query-editor.c
===================================================================
--- nautilus-3.14.0.orig/src/nautilus-query-editor.c
+++ nautilus-3.14.0/src/nautilus-query-editor.c
@@ -122,7 +122,7 @@ nautilus_query_editor_handle_event (Naut
 	}
 
 	/* never handle these events */
-	if (event->keyval == GDK_KEY_slash || event->keyval == GDK_KEY_Delete) {
+	if (event->keyval == GDK_KEY_slash || event->keyval == GDK_KEY_Delete || event->keyval == GDK_KEY_BackSpace) {
 		return FALSE;
 	}
 
Index: nautilus-3.14.0/src/nautilus-shell-ui.xml
===================================================================
--- nautilus-3.14.0.orig/src/nautilus-shell-ui.xml
+++ nautilus-3.14.0/src/nautilus-shell-ui.xml
@@ -16,6 +16,7 @@
 <accelerator action="PromptLocationSlashAccel"/>
 <accelerator action="PromptLocationTildeAccel"/>
 <accelerator action="ReloadAccel"/>
+<accelerator action="GoUpAccel"/>
 <popup name="ViewMenu" accelerators="true">
   <placeholder name="Zoom Items Placeholder">
     <menuitem name="Zoom In" action="Zoom In"/>
Index: nautilus-3.14.0/src/nautilus-window-menus.c
===================================================================
--- nautilus-3.14.0.orig/src/nautilus-window-menus.c
+++ nautilus-3.14.0/src/nautilus-window-menus.c
@@ -447,6 +447,13 @@ action_prompt_for_location_tilde_callbac
 }
 
 static void
+action_window_go_up_callback (GtkAction *action,
+				     NautilusWindow *window)
+{
+	nautilus_window_slot_go_up (nautilus_window_get_active_slot (window), 0);
+}
+
+static void
 action_view_radio_changed (GtkRadioAction *action,
 			   GtkRadioAction *current,
 			   NautilusWindow *window)
@@ -495,6 +502,10 @@ static const GtkActionEntry main_entries
   /* label, accelerator */       "ReloadAccel", "F5",
   /* tooltip */                  NULL,
                                  G_CALLBACK (action_reload_callback) },
+  /* name, stock id */         { "GoUpAccel", NULL,
+  /* label, accelerator */       "GoUpAccel", "BackSpace",
+  /* tooltip */                  NULL,
+                                 G_CALLBACK (action_window_go_up_callback) },
   /* name, stock id */         { NAUTILUS_ACTION_HELP, NULL,
   /* label, accelerator */       N_("_All Topics"), "F1",
   /* tooltip */                  N_("Display Nautilus help"),
