--- geary-40.alpha/meson.build	2021-01-18 03:04:45.728322700 -0800
+++ geary-40.alpha.ayatana/meson.build	2021-03-27 17:13:19.840641739 -0700
@@ -133,7 +133,7 @@
 # Optional dependencies
 appstream_util = find_program('appstream-util', required: false)
 desktop_file_validate = find_program('desktop-file-validate', required: false)
-libmessagingmenu_dep = dependency('messaging-menu', version: '>= 12.10', required: false)
+libmessagingmenu_dep = dependency('messaging-menu', version: '>= 0.8.2', required: false)
 
 #
 # Internal build configuration
