From 9bb053194b2b94b323fcdcf98b23d2a7c7a7e7f1 Mon Sep 17 00:00:00 2001
From: Mikura Kyouka <13207280@qq.com>
Date: Fri, 5 Dec 2025 13:12:56 +0800
Subject: [PATCH] Make it not to strip while building

---
 Makefile                 | 3 ---
 config-libkrunfw_aarch64 | 2 +-
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index faa443e..a3ebdd9 100644
--- a/Makefile
+++ b/Makefile
@@ -122,9 +122,6 @@ endif
 
 $(KRUNFW_BINARY_$(OS)): $(KERNEL_C_BUNDLE) $(QBOOT_C_BUNDLE) $(INITRD_C_BUNDLE)
 	$(CC) -fPIC -DABI_VERSION=$(ABI_VERSION) -shared $(SONAME_$(OS)) -o $@ $(KERNEL_C_BUNDLE) $(QBOOT_C_BUNDLE) $(INITRD_C_BUNDLE)
-ifeq ($(OS),Linux)
-	$(STRIP) $(KRUNFW_BINARY_$(OS))
-endif
 
 install:
 	install -d $(DESTDIR)$(PREFIX)/$(LIBDIR_$(OS))/
diff --git a/config-libkrunfw_aarch64 b/config-libkrunfw_aarch64
index 2a0efb6..50615e1 100644
--- a/config-libkrunfw_aarch64
+++ b/config-libkrunfw_aarch64
@@ -3223,7 +3223,7 @@ CONFIG_DEBUG_BUGVERBOSE=y
 #
 CONFIG_AS_HAS_NON_CONST_LEB128=y
 CONFIG_FRAME_WARN=1024
-CONFIG_STRIP_ASM_SYMS=y
+CONFIG_STRIP_ASM_SYMS=n
 CONFIG_HEADERS_INSTALL=y
 CONFIG_DEBUG_SECTION_MISMATCH=y
 CONFIG_SECTION_MISMATCH_WARN_ONLY=y
-- 
2.52.0

