Make g++ happy.

--- a/src/frontend.cpp	2016-07-10 19:36:14.000000000 +0200
+++ b/src/frontend.cpp	2016-07-12 13:32:08.154269378 +0200
@@ -378,7 +378,7 @@ static void startCampaignSelector()
 	QList<CAMPAIGN_FILE> list = readCampaignFiles();
 	for (int i = 0; i < list.size(); i++)
 	{
-		ssprintf(hackList[i], list[i].name.toUtf8().constData()); // since widget system is crazy and takes pointers not copies
+		ssprintf(hackList[i], "%s", list[i].name.toUtf8().constData()); // since widget system is crazy and takes pointers not copies
 		addTextButton(FRONTEND_CAMPAIGN_1 + i,  FRONTEND_POS1X, FRONTEND_POS2Y + 40 * i, hackList[i], WBUT_TXTCENTRE);
 	}
 	addSideText(FRONTEND_SIDETEXT, FRONTEND_SIDEX, FRONTEND_SIDEY, _("CAMPAIGNS"));

