diff --git a/wallpapers/image/plugin/finder/imagefinder.cpp b/wallpapers/image/plugin/finder/imagefinder.cpp
index 7de3f525e..be7fbbf76 100644
--- a/wallpapers/image/plugin/finder/imagefinder.cpp
+++ b/wallpapers/image/plugin/finder/imagefinder.cpp
@@ -33,11 +33,11 @@ void ImageFinder::run()
         return info.baseName() != QLatin1String("screenshot") && !path.contains(QLatin1String("contents/images/"))
             && !path.contains(QLatin1String("contents/images_dark/"));
     };
-    int i;
 
-    for (i = 0; i < m_paths.size(); ++i) {
+    for (int i = 0; i < m_paths.size(); ++i) {
         const QString &path = m_paths.at(i);
-        const QFileInfo info(findSymlinkTarget(QFileInfo(path)));
+        const QFileInfo originalInfo = QFileInfo(path);
+        const QFileInfo info(findSymlinkTarget(originalInfo));
         const QString target = info.absoluteFilePath();
 
         if (!info.exists() || !filterCondition(info)) {
@@ -46,7 +46,7 @@ void ImageFinder::run()
         }
 
         if (info.isFile()) {
-            if (isAcceptableSuffix(info.suffix()) && !info.isSymLink()) {
+            if (isAcceptableSuffix(info.suffix()) && !originalInfo.isSymLink()) {
                 images.append(target);
             }
 
@@ -60,7 +60,7 @@ void ImageFinder::run()
             const QFileInfo realwp(findSymlinkTarget(wp));
 
             if (realwp.isFile()) {
-                if (filterCondition(realwp) && !realwp.isSymLink()) {
+                if (filterCondition(realwp) && !wp.isSymLink()) {
                     images.append(realwp.filePath());
                 }
             } else if (realwp.isDir()) {
