From bb3d116b4835e7df7f613aa98b1f1844a52996dc Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Mon, 31 Mar 2025 05:54:38 -0700
Subject: [PATCH 1/4] Fix Ruby not found on AOSC OS

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 34ddb30f8..4d66a22f3 100644
--- a/meson.build
+++ b/meson.build
@@ -635,7 +635,7 @@ conf_data.set('CONFIG_PERL_POPPX_WITHOUT_N_A', 1)
 
 rubydeps = []
 if conf_data.get('CONFIG_SCRIPTING_RUBY')
-    rubydeps = dependency('ruby', static: st)
+    rubydeps = dependency('Ruby', static: st)
     deps += rubydeps
 endif
 
-- 
2.49.0

