From d998510753602a7169e7ffa3d1e33d8c8dc0eddd Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Mon, 5 Sep 2022 17:45:10 +0200
Subject: [PATCH 10/21] daemon: Don't count hidden devices

---
 src/up-daemon.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/up-daemon.c b/src/up-daemon.c
index 333f73b..c2dd14b 100644
--- a/src/up-daemon.c
+++ b/src/up-daemon.c
@@ -111,7 +111,8 @@ up_daemon_get_number_devices_of_type (UpDaemon *daemon, UpDeviceKind type)
 		g_object_get (device,
 			      "type", &type_tmp,
 			      NULL);
-		if (type == type_tmp)
+		if (type == type_tmp &&
+		    up_device_get_object_path (device) != NULL)
 			count++;
 	}
 	g_ptr_array_unref (array);
-- 
2.39.1

