From 7f3a31447faf2006120abfad4802dbde7cd11e8b Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 16 Nov 2022 16:16:34 +0100
Subject: [PATCH 21/21] rules: Update hwdb from upstream NUT

---
 rules/95-upower-hid.hwdb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rules/95-upower-hid.hwdb b/rules/95-upower-hid.hwdb
index b0b7fbf..c5f0bda 100644
--- a/rules/95-upower-hid.hwdb
+++ b/rules/95-upower-hid.hwdb
@@ -70,6 +70,7 @@ usb:v050Dp1100*
 usb:v051Dp0000*
 usb:v051Dp0002*
 usb:v051Dp0003*
+usb:v051Dp0004*
  UPOWER_BATTERY_TYPE=ups
  UPOWER_VENDOR=APC
 
-- 
2.39.1

