diff -Naur qtwebkit-5.212.0-alpha2/Source/Qt5WebKitConfig.cmake.in qtwebkit-5.212.0-alpha2.noexact/Source/Qt5WebKitConfig.cmake.in
--- qtwebkit-5.212.0-alpha2/Source/Qt5WebKitConfig.cmake.in	2017-06-04 15:16:06.000000000 -0500
+++ qtwebkit-5.212.0-alpha2.noexact/Source/Qt5WebKitConfig.cmake.in	2017-07-07 02:08:51.540296281 -0500
@@ -1,8 +1,8 @@
 @PACKAGE_INIT@
 
-find_dependency(Qt5Core    @Qt5_VERSION@ EXACT)
-find_dependency(Qt5Gui     @Qt5_VERSION@ EXACT)
-find_dependency(Qt5Network @Qt5_VERSION@ EXACT)
+find_dependency(Qt5Core    @Qt5_VERSION@)
+find_dependency(Qt5Gui     @Qt5_VERSION@)
+find_dependency(Qt5Network @Qt5_VERSION@)
 
 include("${CMAKE_CURRENT_LIST_DIR}/WebKitTargets.cmake")
 
diff -Naur qtwebkit-5.212.0-alpha2/Source/Qt5WebKitWidgetsConfig.cmake.in qtwebkit-5.212.0-alpha2.noexact/Source/Qt5WebKitWidgetsConfig.cmake.in
--- qtwebkit-5.212.0-alpha2/Source/Qt5WebKitWidgetsConfig.cmake.in	2017-06-04 15:16:06.000000000 -0500
+++ qtwebkit-5.212.0-alpha2.noexact/Source/Qt5WebKitWidgetsConfig.cmake.in	2017-07-07 02:09:09.203629812 -0500
@@ -1,9 +1,9 @@
 @PACKAGE_INIT@
 
-find_dependency(Qt5Core    @Qt5_VERSION@ EXACT)
-find_dependency(Qt5Gui     @Qt5_VERSION@ EXACT)
-find_dependency(Qt5Network @Qt5_VERSION@ EXACT)
-find_dependency(Qt5Widgets @Qt5_VERSION@ EXACT)
+find_dependency(Qt5Core    @Qt5_VERSION@)
+find_dependency(Qt5Gui     @Qt5_VERSION@)
+find_dependency(Qt5Network @Qt5_VERSION@)
+find_dependency(Qt5Widgets @Qt5_VERSION@)
 
 find_dependency(Qt5WebKit @PROJECT_VERSION_STRING@ EXACT)
 
