--- qt-5/.gitmodules	2022-05-27 23:49:59.000000000 +0000
+++ qt-5.webengine/.gitmodules	2022-05-29 19:05:08.318062961 +0000
@@ -271,3 +271,12 @@
 	url = ../qtquick3d.git
 	branch = 5.15
 	status = addon
+[submodule "qtwebengine"]
+	depends = qtdeclarative
+	recommends = qtquickcontrols qtquickcontrols2 qtlocation qtwebchannel qttools
+	path = qtwebengine
+	url = ../qtwebengine.git
+	branch = 5.15
+	status = addon
+	priority = 10
+
