--- qt-5/qtwebengine/src/3rdparty/chromium/third_party/skia/src/utils/SkParseColor.cpp	2022-12-03 23:43:50.000000000 -0800
+++ qt-5-bak/qtwebengine/src/3rdparty/chromium/third_party/skia/src/utils/SkParseColor.cpp	2022-12-29 16:26:29.927925068 -0800
@@ -8,6 +8,12 @@
 
 #include "include/utils/SkParse.h"
 
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <iterator>
+#include <string>
+
 static constexpr const char* gColorNames[] = {
     "aliceblue",
     "antiquewhite",
