From c99d490707a8ccdc2f89d1dc062986b3d65649c1 Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Wed, 12 Mar 2014 08:27:45 +0100
Subject: [PATCH 5/5] Resolves: rhbz#953071 - GUI cannot be opened on some OS

---
 src/Volume_Tab_View.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Volume_Tab_View.py b/src/Volume_Tab_View.py
index 6fad930..3e176e1 100644
--- a/src/Volume_Tab_View.py
+++ b/src/Volume_Tab_View.py
@@ -546,6 +546,8 @@ class Volume_Tab_View:
   
   def on_best_fit(self, obj):
       if (self.try_not_best_fit == True):
+          if self.display_view.display != None:
+              self.display_view.display.respect_smallest_selectable_width(False)
           return
 
       self.on_resize_drawing_area(None, None)
-- 
2.1.4

