--- a/thirdparty/fetch-thirdparty-deps.sh	2016-10-17 16:56:37.785972066 -0500
+++ b/thirdparty/fetch-thirdparty-deps.sh	2016-10-17 16:56:52.139369864 -0500
@@ -86,8 +86,8 @@
 if [ ! -f Open.Nat.dll ]; then
 	echo "Fetching Open.Nat from NuGet"
 	get Open.NAT 2.1.0
-	cp ./Open.NAT/lib/net45/Open.Nat.dll .
-	rm -rf Open.NAT
+	cp ./Open.Nat/lib/net45/Open.Nat.dll .
+	rm -rf Open.Nat
 fi
 
 if [ ! -f FuzzyLogicLibrary.dll ]; then
