From 6c757b3356d1d334572eb9c12e774721a2eb1afe Mon Sep 17 00:00:00 2001
From: SignKirigami <prcups@krgm.moe>
Date: Fri, 24 Jul 2026 16:17:30 +0800
Subject: [PATCH 2/2] fix unexpected reactivation when delay

---
 src/SpectacleCore.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/SpectacleCore.cpp b/src/SpectacleCore.cpp
index 63bd6f34..8aee2776 100644
--- a/src/SpectacleCore.cpp
+++ b/src/SpectacleCore.cpp
@@ -1200,8 +1200,6 @@ void SpectacleCore::takeNewScreenshot(ImagePlatform::GrabMode grabMode, int time
     m_delayAnimation->setDuration(timeout);
     m_delayAnimation->start();
 
-    // skip minimize animation.
-    SpectacleWindow::setVisibilityForAll(QWindow::Hidden);
     SpectacleWindow::setVisibilityForAll(QWindow::Minimized);
 }
 
-- 
2.55.0

