From 6f801ab7aedde69c097ae19cc2276c1d4f921828 Mon Sep 17 00:00:00 2001
From: SignKirigami <prcups@krgm.moe>
Date: Fri, 24 Jul 2026 16:17:30 +0800
Subject: [PATCH 2/3] fix unexpected reactivation when delay

---
 src/SpectacleCore.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/SpectacleCore.cpp b/src/SpectacleCore.cpp
index 8c8bfaf7..5e47c6bd 100644
--- a/src/SpectacleCore.cpp
+++ b/src/SpectacleCore.cpp
@@ -1204,8 +1204,6 @@ void SpectacleCore::takeNewScreenshot(ImagePlatform::GrabMode grabMode, int time
     m_delayAnimation->setDuration(timeout);
     m_delayAnimation->start();
 
-    // skip minimize animation.
-    SpectacleWindow::setVisibilityForAll(QWindow::Hidden);
     SpectacleWindow::setVisibilityForAll(QWindow::Minimized);
 }
 
-- 
2.55.0

