From 4845cb8e5814cef86670d2f338ffec9515a56231 Mon Sep 17 00:00:00 2001
From: SignKirigami <prcups@krgm.moe>
Date: Fri, 24 Jul 2026 16:17:30 +0800
Subject: [PATCH 3/4] AOSCOS: fix unexpected reactivation when delay

Signed-off-by: SignKirigami <prcups@krgm.moe>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 src/SpectacleCore.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/SpectacleCore.cpp b/src/SpectacleCore.cpp
index 8c8bfaf7..5e47c6bd 100644
--- a/src/SpectacleCore.cpp
+++ b/src/SpectacleCore.cpp
@@ -1204,8 +1204,6 @@ void SpectacleCore::takeNewScreenshot(ImagePlatform::GrabMode grabMode, int time
     m_delayAnimation->setDuration(timeout);
     m_delayAnimation->start();
 
-    // skip minimize animation.
-    SpectacleWindow::setVisibilityForAll(QWindow::Hidden);
     SpectacleWindow::setVisibilityForAll(QWindow::Minimized);
 }
 
-- 
2.55.0

