diff -Naur EasyRSA-3.0.1/easyrsa EasyRSA-3.0.1.dirs/easyrsa
--- EasyRSA-3.0.1/easyrsa	2015-09-09 23:18:15.000000000 +0000
+++ EasyRSA-3.0.1.dirs/easyrsa	2016-12-11 16:07:56.809523897 +0000
@@ -8,6 +8,11 @@
 # This code released under version 2 of the GNU GPL; see COPYING and the
 # Licensing/ directory of this project for full licensing details.
 
+# Some package-defined variables
+EASYRSA=/etc/easy-rsa
+EASYRSA_PKI=/etc/easy-rsa/pki
+EASYRSA_SSL_CONF=/etc/easy-rsa/openssl-easyrsa.cnf
+
 # Help/usage output to stdout
 usage() {
 	# command help:
