--- EasyRSA-3.2.0.orig/easyrsa	2024-07-05 21:35:19.196130168 +0800
+++ EasyRSA-3.2.0/easyrsa	2024-07-05 21:35:51.738133488 +0800
@@ -10,6 +10,11 @@
 # and the Licensing/ directory of this project for full licensing
 # details.
 
+# Some package-defined variables
+EASYRSA=/etc/easy-rsa
+EASYRSA_PKI=/etc/easy-rsa/pki
+EASYRSA_SSL_CONF=/etc/easy-rsa/openssl-easyrsa.cnf
+
 # Help/usage output to stdout
 usage() {
 	# command help:
