diff --git a/panel-plugin/notification-plugin.desktop.in b/panel-plugin/notification-plugin.desktop.in
index 2393376..f4672cc 100644
--- a/panel-plugin/notification-plugin.desktop.in
+++ b/panel-plugin/notification-plugin.desktop.in
@@ -5,6 +5,6 @@ Name=Notification Plugin
 Comment=Notification plugin for the Xfce panel
 Icon=org.xfce.notification
 X-XFCE-Module=notification-plugin
-X-XFCE-Internal=false
+X-XFCE-Internal=true
 X-XFCE-Unique=false
 X-XFCE-API=2.0
