From e736188655daf02576b590a19a76bc42fcf001bb Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Mon, 30 Mar 2026 21:36:15 +0800
Subject: [PATCH 3/5] AOSCOS: fix FTBFS with -fno-common

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 src/shar-std.def | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shar-std.def b/src/shar-std.def
index 3461e5f..23b2d14 100755
--- a/src/shar-std.def
+++ b/src/shar-std.def
@@ -41,7 +41,7 @@ no-misuse-usage;
 usage-message;
 die-code;
 
-export  = '#include "local.h"'"\nchar const * const program_name;";
+export  = '#include "local.h"'"\nextern char const * const program_name;";
 
 #shell
 echo "include = 'char const * const program_name = \"${progname}\";';"
-- 
2.53.0

