From 83f04e3850532c3963090f7467a4e96713eee7ec Mon Sep 17 00:00:00 2001
From: Alexandre Julliard <julliard@winehq.org>
Date: Sat, 6 Jul 2019 20:05:53 +0200
Subject: [PATCH 3/5] include: Make sure the JUMP_BUFFER structure is correctly
 aligned on x86-64

Signed-off-by: Alexandre Julliard <julliard@winehq.org>
---
 include/msvcrt/setjmp.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/msvcrt/setjmp.h b/include/msvcrt/setjmp.h
index 2a4af49f10..26dc4a72d7 100644
--- a/include/msvcrt/setjmp.h
+++ b/include/msvcrt/setjmp.h
@@ -47,12 +47,12 @@ typedef struct __JUMP_BUFFER
 
 #elif defined(__x86_64__)
 
-typedef struct _SETJMP_FLOAT128
+typedef DECLSPEC_ALIGN(16) struct _SETJMP_FLOAT128
 {
-    unsigned __int64 DECLSPEC_ALIGN(16) Part[2];
+    unsigned __int64 Part[2];
 } SETJMP_FLOAT128;
 
-typedef struct _JUMP_BUFFER
+typedef DECLSPEC_ALIGN(16) struct _JUMP_BUFFER
 {
     unsigned __int64 Frame;
     unsigned __int64 Rbx;
-- 
2.22.0

