From ff1512ec1162ce6b230fedc6150eb7c88c19bd16 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 19 Jul 2024 14:20:16 +0800
Subject: [PATCH 1/7] [Fedora] fix(libpisock): fix return value variable name

---
 libpisock/dlp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libpisock/dlp.c b/libpisock/dlp.c
index 2514605..03e3987 100644
--- a/libpisock/dlp.c
+++ b/libpisock/dlp.c
@@ -2285,7 +2285,7 @@ dlp_ReadFeature(int sd, unsigned long creator, int num,
 		if (errCode) {
 			LOG((PI_DBG_DLP, PI_DBG_LVL_INFO,
 			    "DLP ReadFeature FtrGet error 0x%8.8lX\n",
-			    res));
+			    errCode));
 			pi_set_palmos_error(sd, (int)errCode);
 			return pi_set_error(sd, PI_ERR_DLP_PALMOS);
 		}
-- 
2.45.2

