From 86bea19f1333493017f0f1ae4fe8b9841389004d Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 19 Jul 2024 14:26:55 +0800
Subject: [PATCH 2/7] [Fedora] fix(libpisock): fix data->impl.open usage

---
 libpisock/serial.c | 4 ++--
 libpisock/usb.c    | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/libpisock/serial.c b/libpisock/serial.c
index 37d292a..5b48aec 100644
--- a/libpisock/serial.c
+++ b/libpisock/serial.c
@@ -286,7 +286,7 @@ pi_serial_connect(pi_socket_t *ps, struct sockaddr *addr,
 		data->establishrate = data->rate = 57600;
 	}
 
-	if ((err = data->impl.open(ps, pa, addrlen)) < 0)
+	if ((err = (data->impl.open)(ps, pa, addrlen)) < 0)
 		return err;	/* errno already set */
 
 	ps->raddr 	= malloc(addrlen);
@@ -362,7 +362,7 @@ pi_serial_bind(pi_socket_t *ps, struct sockaddr *addr, size_t addrlen)
 	}
 
 begin:
-	if ((err = data->impl.open(ps, pa, addrlen)) < 0) {
+	if ((err = (data->impl.open)(ps, pa, addrlen)) < 0) {
 		int 	save_errno = errno;
 #ifdef MAXPATHLEN
 		char	realport[MAXPATHLEN];
diff --git a/libpisock/usb.c b/libpisock/usb.c
index 050e52f..e86823f 100644
--- a/libpisock/usb.c
+++ b/libpisock/usb.c
@@ -267,7 +267,7 @@ pi_usb_connect(pi_socket_t *ps, struct sockaddr *addr, size_t addrlen)
 		data->establishrate = data->rate = 57600;
 	}
 
-	result = data->impl.open(ps, pa, addrlen);
+	result = (data->impl.open)(ps, pa, addrlen);
 	if (result < 0)
 		goto fail;
 
@@ -343,7 +343,7 @@ pi_usb_bind(pi_socket_t *ps, struct sockaddr *addr, size_t addrlen)
 		data->establishrate = data->rate = 57600;
 	}
 
-	result = data->impl.open(ps, pa, addrlen);
+	result = (data->impl.open)(ps, pa, addrlen);
 	if (result < 0)
 		return result;
 
-- 
2.45.2

