From f01fe9761e43e7de4491f9212d79d3d00f0d2ae9 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 19 Jul 2024 14:48:36 +0800
Subject: [PATCH 7/7] [Fedora] fix: fix build against Fedora 19

---
 libpisock/md5.c | 2 +-
 popt/popthelp.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libpisock/md5.c b/libpisock/md5.c
index cc8213b..d7fd52b 100644
--- a/libpisock/md5.c
+++ b/libpisock/md5.c
@@ -125,7 +125,7 @@ void MD5Final(UINT8 digest[16], struct MD5Context *ctx)
 
 	byteSwap(ctx->buf, 4);
 	memcpy(digest, ctx->buf, 16);
-	memset(ctx, 0, sizeof(ctx));	/* In case it's sensitive */
+	memset(ctx, 0, sizeof(*ctx));	/* In case it's sensitive */
 }
 
 #ifndef ASM_MD5
diff --git a/popt/popthelp.c b/popt/popthelp.c
index 4d8f0ef..d4830f9 100644
--- a/popt/popthelp.c
+++ b/popt/popthelp.c
@@ -683,7 +683,7 @@ static int showShortOptions(const struct poptOption * opt, FILE * fp,
     s[0] = '\0';
     /*@-branchstate@*/		/* FIX: W2DO? */
     if (str == NULL) {
-	memset(s, 0, sizeof(s));
+	memset(s, 0, sizeof(*s));
 	str = s;
     }
     /*@=branchstate@*/
-- 
2.45.2

