diff -up pilot-link-0.12.5/libpisock/md5.c.ftbfs-f19 pilot-link-0.12.5/libpisock/md5.c
--- pilot-link-0.12.5/libpisock/md5.c.ftbfs-f19	2013-02-25 15:17:31.708780677 +0100
+++ pilot-link-0.12.5/libpisock/md5.c	2013-02-25 15:15:07.771861922 +0100
@@ -125,7 +125,7 @@ void MD5Final(UINT8 digest[16], struct M
 
 	byteSwap(ctx->buf, 4);
 	memcpy(digest, ctx->buf, 16);
-	memset(ctx, 0, sizeof(ctx));	/* In case it's sensitive */
+	memset(ctx, 0, sizeof(*ctx));	/* In case it's sensitive */
 }
 
 #ifndef ASM_MD5
diff -up pilot-link-0.12.5/popt/popthelp.c.ftbfs-f19 pilot-link-0.12.5/popt/popthelp.c
--- pilot-link-0.12.5/popt/popthelp.c.ftbfs-f19	2004-08-25 17:06:39.000000000 +0200
+++ pilot-link-0.12.5/popt/popthelp.c	2013-02-25 15:16:34.593606516 +0100
@@ -683,7 +683,7 @@ static int showShortOptions(const struct
     s[0] = '\0';
     /*@-branchstate@*/		/* FIX: W2DO? */
     if (str == NULL) {
-	memset(s, 0, sizeof(s));
+	memset(s, 0, sizeof(*s));
 	str = s;
     }
     /*@=branchstate@*/
