diff --git a/src/gprs.c b/src/gprs.c
index 2d3a99e6..2d414b5a 100644
--- a/src/gprs.c
+++ b/src/gprs.c
@@ -1245,8 +1245,19 @@ static DBusMessage *pri_set_property(DBusConnection *conn,
 
 		dbus_message_iter_get_basic(&var, &value);
 
-		if (ctx->active == (ofono_bool_t) value)
+		if (ctx->active == (ofono_bool_t) value) {
+			if (ctx->context_driver) {
+				pri_context_signal_settings(ctx,
+					ctx->context_driver->settings->ipv4 != NULL,
+					ctx->context_driver->settings->ipv6 != NULL);
+			}
+
+			ofono_dbus_signal_property_changed(conn, ctx->path,
+					OFONO_CONNECTION_CONTEXT_INTERFACE,
+					"Active", DBUS_TYPE_BOOLEAN, &value);
+
 			return dbus_message_new_method_return(msg);
+		}
 
 		if (value && !ctx->gprs->attached)
 			return __ofono_error_not_attached(msg);
