From f66ec6a378e39982ded339587fc314d2e723290e Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Wed, 22 Oct 2008 08:36:22 +0300
Subject: [PATCH] Gcc 4.3 include header fix


Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 src/statistics.hpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/statistics.hpp b/src/statistics.hpp
index d771dc6..c916ecb 100644
--- a/src/statistics.hpp
+++ b/src/statistics.hpp
@@ -3,6 +3,7 @@
 
 #include <list>
 #include <string>
+#include <sys/time.h>
 
 namespace statistics {
     
-- 
1.6.5

