diff --git a/clients/benchmarks/CMakeLists.txt b/clients/benchmarks/CMakeLists.txt
index 99ec31f3d..93d7ab299 100644
--- a/clients/benchmarks/CMakeLists.txt
+++ b/clients/benchmarks/CMakeLists.txt
@@ -45,11 +45,6 @@ target_include_directories( hipblaslt-bench
 
 target_link_libraries( hipblaslt-bench PRIVATE ${BLAS_LIBRARY} roc::hipblaslt )
 
-if( CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  # GCC or hip-clang needs specific flags to turn on f16c intrinsics
-  target_compile_options( hipblaslt-bench PRIVATE -mf16c )
-endif( )
-
 target_compile_definitions( hipblaslt-bench PRIVATE HIPBLASLT_BENCH ROCM_USE_FLOAT16 HIPBLASLT_INTERNAL_API ${TENSILE_DEFINES} )
 if ( NOT BUILD_FORTRAN_CLIENTS )
   target_compile_definitions( hipblaslt-bench PRIVATE CLIENTS_NO_FORTRAN )
@@ -172,7 +167,7 @@ foreach( exe ${ext_bench_list_all} )
 
   if( CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     # GCC or hip-clang needs specific flags to turn on f16c intrinsics
-    target_compile_options( ${exe} PRIVATE -mf16c -Wno-unused-result )
+    target_compile_options( ${exe} PRIVATE -Wno-unused-result )
     target_compile_definitions( ${exe} PRIVATE ROCBLASLT_INTERNAL_API )
   endif( )
 endforeach( )
diff --git a/clients/gtest/CMakeLists.txt b/clients/gtest/CMakeLists.txt
index cbedcfa41..7bf1ba576 100644
--- a/clients/gtest/CMakeLists.txt
+++ b/clients/gtest/CMakeLists.txt
@@ -74,11 +74,6 @@ else()
   target_link_libraries( hipblaslt-test PRIVATE ${CUDA_LIBRARIES} )
 endif()
 
-if( CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  # GCC or hip-clang needs specific flag to turn on f16c intrinsics
-  target_compile_options( hipblaslt-test PRIVATE -mf16c )
-endif( )
-
 target_compile_definitions( hipblaslt-test PRIVATE ROCM_USE_FLOAT16 HIPBLASLT_INTERNAL_API )
 
 target_compile_options(hipblaslt-test PRIVATE $<$<COMPILE_LANGUAGE:CXX>:${COMMON_CXX_OPTIONS}>)
diff --git a/clients/samples/CMakeLists.txt b/clients/samples/CMakeLists.txt
index 9c145b9a8..585c4ca2c 100644
--- a/clients/samples/CMakeLists.txt
+++ b/clients/samples/CMakeLists.txt
@@ -130,7 +130,7 @@ foreach( exe ${sample_list_all} )
 
   if( CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     # GCC or hip-clang needs specific flags to turn on f16c intrinsics
-    target_compile_options( ${exe} PRIVATE -mf16c -Wno-unused-result )
+    target_compile_options( ${exe} PRIVATE -Wno-unused-result )
     target_compile_definitions( ${exe} PRIVATE ROCBLASLT_INTERNAL_API )
   endif( )
   if( LEGACY_HIPBLAS_DIRECT )
diff --git a/tensilelite/Tensile/Source/lib/CMakeLists.txt b/tensilelite/Tensile/Source/lib/CMakeLists.txt
index d055b635d..1023ead68 100644
--- a/tensilelite/Tensile/Source/lib/CMakeLists.txt
+++ b/tensilelite/Tensile/Source/lib/CMakeLists.txt
@@ -96,11 +96,11 @@ if(TENSILE_USE_LLVM OR TENSILE_USE_MSGPACK)
 endif()

 if(TENSILE_USE_MSGPACK)
-    find_package(msgpack REQUIRED)
+    find_package(msgpack-cxx REQUIRED)
     target_compile_definitions(TensileHost PUBLIC -DTENSILE_MSGPACK=1)

-    if(TARGET msgpackc-cxx)
-        get_target_property(msgpack_inc msgpackc-cxx INTERFACE_INCLUDE_DIRECTORIES)
+    if(TARGET msgpack-cxx)
+        get_target_property(msgpack_inc msgpack-cxx INTERFACE_INCLUDE_DIRECTORIES)
     elseif(TARGET msgpackc)
         get_target_property(msgpack_inc msgpackc INTERFACE_INCLUDE_DIRECTORIES)
     endif()
diff --git a/clients/include/hipblaslt_math.hpp b/clients/include/hipblaslt_math.hpp
index f25941b7f..02cbe1c64 100644
--- a/clients/include/hipblaslt_math.hpp
+++ b/clients/include/hipblaslt_math.hpp
@@ -30,7 +30,7 @@
 #include <hip/hip_runtime.h>
 #include <hipblaslt/hipblaslt.h>
 #include <hipblaslt/hipblaslt_xfloat32.h>
-#include <immintrin.h>
+//#include <immintrin.h>
 #include <type_traits>
 
 /* ============================================================================================ */
