Author: Geoffrey R. Hutchison
Reviewed-By: Daniel Leidert <dleidert@debian.org>
Description: Fix a potential crash reported by the Mayhem project.
Origin: https://github.com/openbabel/openbabel/commit/7b7c716a1c48a7c5d39ba14d30c691f1894fd6fb#diff-0
Bug-Debian: http://bugs.debian.org/715668
Forwarded: http://forums.openbabel.org/Fwd-Bug-report-on-openbabel-obprobe-crashes-with-exit-status-139-td4656453.html

--- a/tools/babel.cpp
+++ b/tools/babel.cpp
@@ -133,6 +133,9 @@
                   if(!*oext)
                     oext = argv[++arg]; //space left after -i: use next argument
 					
+                  if (arg >= argc)
+                    usage(); // error in parsing command-line
+			
                   if (strncasecmp(oext, "MIME", 4) == 0)
                     {
                       // get the MIME type from the next argument
