From 6ffcad1894446189037209b61dd027a8bfd8357a Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Sat, 2 Jul 2016 11:15:46 +0800
Subject: [PATCH] translations: correct usage of data dir

---
 translations/CMakeLists.txt | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/translations/CMakeLists.txt b/translations/CMakeLists.txt
index f0dc4df..d036cda 100644
--- a/translations/CMakeLists.txt
+++ b/translations/CMakeLists.txt
@@ -51,9 +51,4 @@ FOREACH(_item ${qm_targets})
 	ADD_DEPENDENCIES(finalize-locales "${_item}")
 ENDFOREACH(_item ${qm_targets})
 
-IF(LMMS_BUILD_WIN32)
-	FILE(GLOB QT_QM_FILES "${QT_TRANSLATIONS_DIR}/qt*[^h].qm")
-	LIST(SORT QT_QM_FILES)
-ENDIF(LMMS_BUILD_WIN32)
-
-INSTALL(FILES ${QM_FILES} ${QT_QM_FILES} DESTINATION "${LMMS_DATA_DIR}/locale")
+INSTALL(FILES ${QM_FILES} ${QT_QM_FILES} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/locale")
