From c238724eec8f2ea707b23ffd9b6e0bfc67e8b149 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Sat, 2 Jul 2016 11:18:33 +0800
Subject: [PATCH] src: silent a warning

---
 src/CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ba49b0f..7d3c083 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -45,6 +45,4 @@ target_link_libraries(qwx
     Qt5::Xml
 )
 
-add_dependencies(qwx translation)
-
 install(TARGETS qwx RUNTIME DESTINATION bin)
