From: Roger Shimizu <rogershimizu@gmail.com>
Date: Thu, 21 Jul 2016 22:19:11 +0900
Subject: fix typo in src/CMakeLists.txt

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b61ff83..e35dc0e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -96,7 +96,7 @@ set(IPSETDOT_SRC
     ipsetdot/ipsetdot.c
 )
 
-add_executable(ipsetdot ${IPSETBUILD_SRC})
+add_executable(ipsetdot ${IPSETDOT_SRC})
 target_link_libraries(ipsetdot libcorkipset)
 install(TARGETS ipsetdot DESTINATION bin)
 
