diff -Naur latrace-0.5.11/src/config-flex.l latrace-0.5.11.flex/src/config-flex.l
--- latrace-0.5.11/src/config-flex.l	2011-07-25 11:18:18.000000000 +0000
+++ latrace-0.5.11.flex/src/config-flex.l	2018-09-03 07:07:12.445592144 +0000
@@ -108,15 +108,13 @@
 
 %%
 
-#ifndef yywrap
-int yywrap()
+int lt_config_wrap()
 {
 	return 1;
 	/* XXX not to get the compiler 'not used' warning */
 	yyunput(0, NULL);
 	input();
 }
-#endif
 
 void lt_config_error(const char *m)
 {
