diff -Nur partimage-0.6.8/src/shared/common.cpp partimage-0.6.8-nmu//src/shared/common.cpp
--- partimage-0.6.8/src/shared/common.cpp	2009-09-12 10:06:22.000000000 +0200
+++ partimage-0.6.8-nmu//src/shared/common.cpp	2011-11-18 23:03:31.278838500 +0100
@@ -79,7 +79,7 @@
   addText(i18n("\n* Version is %s [%s].\n"), PACKAGE_VERSION, __VERSION__);
 
   addText(i18n("\n* Supported file systems:\n  - "));
-  addText(g_szSupportedFileSystems);
+  addText("%s", g_szSupportedFileSystems);
   addText(i18n("\n"));
 
 #ifdef LIBRARY_EXT2FS
