Index: partimage-0.6.8/src/client/netclient.cpp
===================================================================
--- partimage-0.6.8.orig/src/client/netclient.cpp	2009-09-24 21:21:42.000000000 +0200
+++ partimage-0.6.8/src/client/netclient.cpp	2011-04-03 16:55:38.379221262 +0200
@@ -43,9 +43,8 @@
     {
       showDebug(3, "initializing client ssl\n");
       SSLeay_add_ssl_algorithms();
-      meth = SSLv2_client_method();
       SSL_load_error_strings();
-      ctx = SSL_CTX_new(meth);
+      ctx = SSL_CTX_new(SSLv23_client_method());
       if (!ctx)
         THROW(ERR_SSL_CTX);
       m_bUseSSL = (ctx != NULL);
Index: partimage-0.6.8/src/client/netclient.h
===================================================================
--- partimage-0.6.8.orig/src/client/netclient.h	2009-09-24 21:21:42.000000000 +0200
+++ partimage-0.6.8/src/client/netclient.h	2011-04-03 16:56:42.858410648 +0200
@@ -35,7 +35,6 @@
 #ifdef HAVE_SSL
   SSL_CTX * ctx;
   X509 * server_cert;
-  SSL_METHOD * meth;
 #endif
   bool m_bUseSSL;
   bool m_bMustLogin;
Index: partimage-0.6.8/src/server/netserver.h
===================================================================
--- partimage-0.6.8.orig/src/server/netserver.h	2009-09-12 10:06:22.000000000 +0200
+++ partimage-0.6.8/src/server/netserver.h	2011-04-03 16:56:33.614526858 +0200
@@ -41,7 +41,6 @@
 #ifdef HAVE_SSL
   SSL_CTX * ctx;
   X509 * client_cert;
-  SSL_METHOD * meth; 
   int err;
 #endif
 
Index: partimage-0.6.8/src/server/netserver.cpp
===================================================================
--- partimage-0.6.8.orig/src/server/netserver.cpp	2011-04-03 16:55:48.707091421 +0200
+++ partimage-0.6.8/src/server/netserver.cpp	2011-04-03 16:56:18.462717343 +0200
@@ -39,8 +39,7 @@
     {
       SSL_load_error_strings();
       SSLeay_add_ssl_algorithms();
-      meth = SSLv23_server_method();
-      ctx = SSL_CTX_new(meth);
+      ctx = SSL_CTX_new(SSLv23_server_method());
       if (!ctx)
         {
           ERR_print_errors_fp(stderr);
