diff -ur pilot-link-0.12.5.orig/configure pilot-link-0.12.5/configure
--- pilot-link-0.12.5.orig/configure	2010-02-07 04:42:29.000000000 +0100
+++ pilot-link-0.12.5/configure	2014-05-09 15:36:30.466127800 +0200
@@ -2604,12 +2604,12 @@
 
 
 enable_werror=no
-save_CFLAGS="$CFLAGS"
-CFLAGS=`echo $save_CFLAGS | sed -e s/-Werror//g`
-CXXFLAGS=`echo $save_CXXFLAGS | sed -e s/-Werror//g`
-if test "x$CFLAGS" != "x$save_CFLAGS"; then
-		enable_werror="yes"
-fi
+#save_CFLAGS="$CFLAGS"
+#CFLAGS=`echo $save_CFLAGS | sed -e s/-Werror//g`
+#CXXFLAGS=`echo $save_CXXFLAGS | sed -e s/-Werror//g`
+#if test "x$CFLAGS" != "x$save_CFLAGS"; then
+#		enable_werror="yes"
+#fi
 
 
 ac_aux_dir=
