diff -up pilot-link-0.12.5/libpisock/serial.c.pom pilot-link-0.12.5/libpisock/serial.c
--- pilot-link-0.12.5/libpisock/serial.c.pom	2006-10-12 16:21:22.000000000 +0200
+++ pilot-link-0.12.5/libpisock/serial.c	2010-06-24 10:35:48.000000000 +0200
@@ -286,7 +286,7 @@ pi_serial_connect(pi_socket_t *ps, struc
 		data->establishrate = data->rate = 57600;
 	}
 
-	if ((err = data->impl.open(ps, pa, addrlen)) < 0)
+	if ((err = (data->impl.open)(ps, pa, addrlen)) < 0)
 		return err;	/* errno already set */
 
 	ps->raddr 	= malloc(addrlen);
@@ -362,7 +362,7 @@ pi_serial_bind(pi_socket_t *ps, struct s
 	}
 
 begin:
-	if ((err = data->impl.open(ps, pa, addrlen)) < 0) {
+	if ((err = (data->impl.open)(ps, pa, addrlen)) < 0) {
 		int 	save_errno = errno;
 #ifdef MAXPATHLEN
 		char	realport[MAXPATHLEN];
diff -up pilot-link-0.12.5/libpisock/usb.c.pom pilot-link-0.12.5/libpisock/usb.c
--- pilot-link-0.12.5/libpisock/usb.c.pom	2010-06-24 09:24:56.000000000 +0200
+++ pilot-link-0.12.5/libpisock/usb.c	2010-06-24 10:36:08.000000000 +0200
@@ -267,7 +267,7 @@ pi_usb_connect(pi_socket_t *ps, struct s
 		data->establishrate = data->rate = 57600;
 	}
 
-	result = data->impl.open(ps, pa, addrlen);
+	result = (data->impl.open)(ps, pa, addrlen);
 	if (result < 0)
 		goto fail;
 
@@ -343,7 +343,7 @@ pi_usb_bind(pi_socket_t *ps, struct sock
 		data->establishrate = data->rate = 57600;
 	}
 
-	result = data->impl.open(ps, pa, addrlen);
+	result = (data->impl.open)(ps, pa, addrlen);
 	if (result < 0)
 		return result;
 
