--- pilot-link-0.12.1/libpisock/dlp.c.pom	2006-08-27 18:20:17.000000000 +0200
+++ pilot-link-0.12.1/libpisock/dlp.c	2006-11-30 13:39:29.000000000 +0100
@@ -2268,7 +2268,7 @@
 		if (errCode) {
 			LOG((PI_DBG_DLP, PI_DBG_LVL_INFO,
 			    "DLP ReadFeature FtrGet error 0x%8.8lX\n",
-			    res));
+			    errCode));
 			pi_set_palmos_error(sd, (int)errCode);
 			return pi_set_error(sd, PI_ERR_DLP_PALMOS);
 		}
