diff -up pilot-link-0.12.4/libpisock/usb.c.clio pilot-link-0.12.4/libpisock/usb.c
--- pilot-link-0.12.4/libpisock/usb.c.clio	2009-05-28 14:55:43.000000000 +0200
+++ pilot-link-0.12.4/libpisock/usb.c	2009-05-28 14:56:57.000000000 +0200
@@ -655,6 +655,7 @@ pi_usb_dev_t known_devices[] = {
 		.vendor 	= 0x054c,
 		.product 	= 0x0066,
 		.idstr 		= "Sony T, SJ series, and other Palm OS 4.0 devices",
+		.flags      = USB_INIT_SONY_CLIE,
 	},
 
 	{
