From b3408465b452b605d8bd0df712092baa1c434ec4 Mon Sep 17 00:00:00 2001
From: Cryolitia <cryolitia@deepin.org>
Date: Fri, 11 Jul 2025 17:43:09 +0800
Subject: [PATCH] fix: add executable bit for `deepin-greater`

Log:
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c33ff56f..559ba306 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -419,7 +419,7 @@ if (DISABLE_DSS_SNIPE)
     install(FILES files/wayland/lightdm-deepin-greeter-wayland.desktop DESTINATION ${CMAKE_INSTALL_DATADIR}/xgreeters/)
 else () # snipe
     install(TARGETS dde-lock lightdm-deepin-greeter DESTINATION ${CMAKE_INSTALL_BINDIR})
-    install(FILES files/snipe/deepin-greeter DESTINATION ${CMAKE_INSTALL_BINDIR})
+    install(PROGRAMS files/snipe/deepin-greeter DESTINATION ${CMAKE_INSTALL_BINDIR})
 
     file(GLOB SCRIPTS  files/snipe/scripts/*)
     install(PROGRAMS ${SCRIPTS} DESTINATION ${CMAKE_INSTALL_SYSCONFDIR}/deepin/greeters.d)
@@ -490,4 +490,4 @@ endif ()
 install(FILES files/deepin-debug-config/org.deepin.dde.session-shell.json
     DESTINATION ${CMAKE_INSTALL_DATADIR}/deepin-debug-config/deepin-debug-config.d)
 install(FILES files/deepin-log-config/org.deepin.dde.session-shell.json
-    DESTINATION ${CMAKE_INSTALL_DATADIR}/deepin-log-viewer/deepin-log.conf.d)
\ No newline at end of file
+    DESTINATION ${CMAKE_INSTALL_DATADIR}/deepin-log-viewer/deepin-log.conf.d)
