From 0f2357963daefc51faeb6e5bece86a13c847b75d Mon Sep 17 00:00:00 2001
From: xffish <xffish@126.com>
Date: Sun, 28 Dec 2025 20:09:11 +0800
Subject: [PATCH] use fat lto

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 62882c8..de67be9 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -30,7 +30,7 @@ default = ["cli"]
 yaml = ["dep:serde_yaml"]
 
 [profile.release]
-lto = "thin"
+lto = true
 panic = "abort"
 
 [[bin]]
-- 
2.52.0
