From b0b9a86bce605ba402476ac9408c725eea4f0aff Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Fri, 7 Nov 2025 09:32:27 -0800
Subject: [PATCH] Use tomli instead of tomllib for Python<3.11

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 build-aux/crates-version.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build-aux/crates-version.py b/build-aux/crates-version.py
index 80a7321..3f4df9d 100755
--- a/build-aux/crates-version.py
+++ b/build-aux/crates-version.py
@@ -1,6 +1,6 @@
 #!/usr/bin/python3
 
-import tomllib
+import tomli
 import sys
 import os.path
 
@@ -8,7 +8,7 @@ crate = sys.argv[1]
 gnome_version = False if len(sys.argv) < 3 else sys.argv[2] == 'gnome'
 
 path = os.path.dirname(__file__)
-data = tomllib.load(open(f'{path}/../{crate}/Cargo.toml', 'rb'))
+data = tomli.load(open(f'{path}/../{crate}/Cargo.toml', 'rb'))
 version = data['package']['version']
 
 if gnome_version:
-- 
2.51.1

