From b36e05caae8a152f01edec7e36f7f1b826657315 Mon Sep 17 00:00:00 2001
From: Sam Lantinga <slouken@libsdl.org>
Date: Mon, 11 Feb 2013 20:51:13 -0800
Subject: [PATCH 6/6] BACKPORT: Fixed missing return value, thanks to marius

Signed-off-by: ilikara <3435193369@qq.com>
---
 music_mad.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/music_mad.c b/music_mad.c
index 81a7cbf9..fcb0b9d6 100644
--- a/music_mad.c
+++ b/music_mad.c
@@ -224,7 +224,7 @@ mad_getSamples(mad_data *mp3_mad, Uint8 *stream, int len) {
   if ((mp3_mad->status & MS_playing) == 0) {
 	/* We're not supposed to be playing, so send silence instead. */
 	memset(stream, 0, len);
-	return;
+	return 0;
   }
 
   out = stream;
-- 
2.51.0

