From df7b9dfb5d4f73714250f4cdf1d8d20df233d2af Mon Sep 17 00:00:00 2001
From: Christos Zoulas <christos@zoulas.com>
Date: Thu, 20 Mar 2025 14:57:41 +0000
Subject: [PATCH 2/2] Fix termios handling on linux/ppc (Dr. Werner Fink)

---
 src/seccomp.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/seccomp.c b/src/seccomp.c
index 9e00034a..f05c30f9 100644
--- a/src/seccomp.c
+++ b/src/seccomp.c
@@ -27,15 +27,20 @@
 #include "file.h"
 
 #ifndef	lint
-FILE_RCSID("@(#)$File: seccomp.c,v 1.30 2024/11/28 14:04:24 christos Exp $")
+FILE_RCSID("@(#)$File: seccomp.c,v 1.31 2025/03/20 14:57:41 christos Exp $")
 #endif	/* lint */
 
 #if HAVE_LIBSECCOMP
 #include <seccomp.h> /* libseccomp */
 #include <sys/prctl.h> /* prctl */
-#include <sys/ioctl.h>
 #include <sys/socket.h>
-#include <termios.h>
+#ifdef __powerpc64__
+// See: https://sourceware.org/bugzilla/show_bug.cgi?id=32806
+# include <asm/termbits.h>
+#else
+# include <termios.h>
+#endif
+#include <sys/ioctl.h>
 #include <fcntl.h>
 #include <stdlib.h>
 #include <errno.h>
-- 
2.52.0

