diff --git a/data/Makefile.am b/data/Makefile.am
index fe0f1d5..18693fe 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -36,6 +36,11 @@ pkgconfig_DATA = polkit-gobject-1.pc polkit-agent-1.pc
 
 # ----------------------------------------------------------------------------------------------------
 
+itsdir = $(datadir)/gettext/its
+its_DATA = polkit.loc polkit.its
+
+# ----------------------------------------------------------------------------------------------------
+
 systemdservice_in_files = polkit.service.in
 
 if HAVE_SYSTEMD
diff --git a/data/polkit.its b/data/polkit.its
new file mode 100644
index 0000000..1c37e6b
--- /dev/null
+++ b/data/polkit.its
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<its:rules xmlns:its="http://www.w3.org/2005/11/its"
+           version="2.0">
+  <its:translateRule selector="//*" translate="no"/>
+  <its:translateRule selector="//action/description |
+                               //action/message"
+                     translate="yes"/>
+</its:rules>
diff --git a/data/polkit.loc b/data/polkit.loc
new file mode 100644
index 0000000..c7427ec
--- /dev/null
+++ b/data/polkit.loc
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
+<locatingRules>
+  <locatingRule name="polkit policy" pattern="*.policy">
+    <documentRule localName="policyconfig" target="polkit.its"/>
+  </locatingRule>
+</locatingRules>
