--- rustc/compiler/rustc_target/src/spec/mod.rs	2023-04-25 00:04:35.137209117 +0000
+++ rustc.i486/compiler/rustc_target/src/spec/mod.rs	2023-04-25 00:06:42.093674104 +0000
@@ -1021,6 +1021,7 @@
     ("x86_64-unknown-linux-gnux32", x86_64_unknown_linux_gnux32),
     ("i686-unknown-linux-gnu", i686_unknown_linux_gnu),
     ("i586-unknown-linux-gnu", i586_unknown_linux_gnu),
+    ("i486-unknown-linux-gnu", i486_unknown_linux_gnu),
     ("loongarch64-unknown-linux-gnu", loongarch64_unknown_linux_gnu),
     ("m68k-unknown-linux-gnu", m68k_unknown_linux_gnu),
     ("mips-unknown-linux-gnu", mips_unknown_linux_gnu),
