--- rustc-1.61.0-src/compiler/rustc_target/src/spec/mod.rs	2022-05-18 01:29:36.000000000 +0000
+++ rustc-1.61.0-src.i486/compiler/rustc_target/src/spec/mod.rs	2022-07-17 06:20:40.334197962 +0000
@@ -812,6 +812,7 @@
     ("x86_64-unknown-linux-gnux32", x86_64_unknown_linux_gnux32),
     ("i686-unknown-linux-gnu", i686_unknown_linux_gnu),
     ("i586-unknown-linux-gnu", i586_unknown_linux_gnu),
+    ("i486-unknown-linux-gnu", i486_unknown_linux_gnu),
     ("m68k-unknown-linux-gnu", m68k_unknown_linux_gnu),
     ("mips-unknown-linux-gnu", mips_unknown_linux_gnu),
     ("mips64-unknown-linux-gnuabi64", mips64_unknown_linux_gnuabi64),
