From c86ce4ffc04573b293b1bd9892e904843c1ace04 Mon Sep 17 00:00:00 2001
From: Henry Chen <chenx97@aosc.io>
Date: Sat, 9 Aug 2025 22:48:09 +0800
Subject: [PATCH 4/4] AOSCOS: bootstrap: temporarily replace lib64 with lib

---
 src/bootstrap/src/lib.rs | 21 ++++++++++++++-------
 1 file changed, 14 insertions(+), 7 deletions(-)

diff --git a/src/bootstrap/src/lib.rs b/src/bootstrap/src/lib.rs
index f44fe4548a1..417260c70ed 100644
--- a/src/bootstrap/src/lib.rs
+++ b/src/bootstrap/src/lib.rs
@@ -382,13 +382,20 @@ impl Build {
         let in_tree_llvm_info = config.in_tree_llvm_info.clone();
         let in_tree_gcc_info = config.in_tree_gcc_info.clone();
 
-        let initial_target_libdir = command(&config.initial_rustc)
-            .run_always()
-            .args(["--print", "target-libdir"])
-            .run_capture_stdout(&config)
-            .stdout()
-            .trim()
-            .to_owned();
+        let initial_target_libdir = {
+            let s = command(&config.initial_rustc)
+                .run_always()
+                .args(["--print", "target-libdir"])
+                .run_capture_stdout(&config)
+                .stdout()
+                .trim()
+                .to_owned();
+            if s.contains("lib64") {
+                s.replace("lib64", "lib")
+            } else {
+                panic!("AOSCOS: No lib64 to be replaced. Please remove this patch.")
+            }
+        };
 
         let initial_target_dir = Path::new(&initial_target_libdir)
             .parent()
-- 
2.50.1

