From 6c31a432119a53d6594e4fc5f6478943a5a3ca0d Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Fri, 5 Jul 2024 21:21:23 +0800
Subject: [PATCH 1/2] hack: fix IME input text

---
 bin_steam.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/bin_steam.sh b/bin_steam.sh
index 9ef1732..29bf3e4 100755
--- a/bin_steam.sh
+++ b/bin_steam.sh
@@ -310,4 +310,10 @@ fi
 # go to the install directory and run the client
 cd "$LAUNCHSTEAMDIR"
 
+# Setting IME to fcitx5 to fix input
+export GTK_IM_MODULE=fcitx5
+export QT_IM_MODULE=fcitx5
+export QT5_IM_MODULE=fcitx5
+export XMODIFIERS=@im=fcitx5
+
 exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" ${log_opened+-srt-logger-opened} "$@"
-- 
2.51.1

