From a09277be7571d4a3b408d47aa067e2cfb833c54a Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 4 Mar 2024 14:59:34 +0800
Subject: [PATCH 1/2] fix(steam.desktop): make sure Steam recognises our 32-bit
 x86 environment (optenv32)

---
 steam.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/steam.desktop b/steam.desktop
index 77314e0..bdaea30 100644
--- a/steam.desktop
+++ b/steam.desktop
@@ -27,7 +27,7 @@ Comment[th]=โปรแกรมสำหรับจัดการและ
 Comment[tr]=Steam üzerinden oyun oynama ve düzenleme uygulaması
 Comment[uk]=Програма для керування іграми та запуску ігор у Steam
 Comment[vi]=Ứng dụng để quản lý và chơi trò chơi trên Steam
-Exec=/usr/bin/steam %U
+Exec=env LD_LIBRARY_PATH=/opt/32/lib:/usr/lib /usr/bin/steam %U
 Icon=steam
 Terminal=false
 Type=Application
-- 
2.45.2

