From 9271c715f151af9d4cb99f99d5dd23a8f5322382 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Fri, 5 Jul 2024 21:21:23 +0800
Subject: [PATCH 2/2] hack: fix IME input text

---
 bin_steam.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/bin_steam.sh b/bin_steam.sh
index 91d2519..eabc1ae 100755
--- a/bin_steam.sh
+++ b/bin_steam.sh
@@ -229,4 +229,10 @@ fi
 # go to the install directory and run the client
 cd "$LAUNCHSTEAMDIR"
 
+# Setting IME to fcitx5 to fix input
+export GTK_IM_MODULE=fcitx5
+export QT_IM_MODULE=fcitx5
+export QT5_IM_MODULE=fcitx5
+export XMODIFIERS=@im=fcitx5
+
 exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@"
-- 
2.46.2

