From ff7d1ffe305eb0bd28f604ea4f40994a090ee06e Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Fri, 5 Jul 2024 21:21:23 +0800
Subject: [PATCH 2/2] hack: fix IME input text

---
 bin_steam.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/bin_steam.sh b/bin_steam.sh
index fb81d47..730407e 100755
--- a/bin_steam.sh
+++ b/bin_steam.sh
@@ -202,4 +202,10 @@ fi
 cp "$LAUNCHSTEAMBOOTSTRAPFILE" "$LAUNCHSTEAMDIR/bootstrap.tar.xz"
 cd "$LAUNCHSTEAMDIR"
 
+# Setting IME to fcitx5 to fix input
+export GTK_IM_MODULE=fcitx5
+export QT_IM_MODULE=fcitx5
+export QT5_IM_MODULE=fcitx5
+export XMODIFIERS=@im=fcitx5
+
 exec "$LAUNCHSTEAMDIR/$STEAMBOOTSTRAP" "$@"
-- 
2.45.2

