--- a/pastebinit
+++ b/pastebinit
@@ -38,9 +38,15 @@ defaultPB = "pastebin.com"
 
 # Now try to override it with a distributor pastebin
 try:
-    import platform
-    release = platform.linux_distribution()[0].lower()
-    if release == 'debian':
+    if sys.version_info[0] == 3 and sys.version_info[1] > 7:
+        import distro
+        release = distro.linux_distribution()[0].lower()
+    else:
+        import platform
+        release = platform.linux_distribution()[0].lower()
+    if release == 'aosc os':
+        defaultPB = "paste.aosc.io"
+    elif release == 'debian':
         defaultPB = "paste.debian.net"
     elif release == 'fedora':
         defaultPB = "fpaste.org"
