From f460c8ca8791a460c0ee71f7fac8b7deb0ba24bc Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Wed, 3 Dec 2025 21:49:13 +0800
Subject: [PATCH 01/63] loonggpu: fix build without CONFIG_MMU_NOTIFIER

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 lgkcd/kcd_process.c    | 4 ++++
 loonggpu/loonggpu_mn.c | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/lgkcd/kcd_process.c b/lgkcd/kcd_process.c
index d373bb8..94e4456 100644
--- a/lgkcd/kcd_process.c
+++ b/lgkcd/kcd_process.c
@@ -1061,12 +1061,14 @@ static void kcd_process_notifier_release_internal(struct kcd_process *p)
 	mm = p->mm;
 	p->mm = NULL;
 
+#ifdef CONFIG_MMU_NOTIFIER
 #if defined(LG_MMU_NOTIFIER_UNREGISTER_NO_RELEASE)
 	mmu_notifier_unregister_no_release(&p->mmu_notifier, mm);
 	mmu_notifier_call_srcu(&p->rcu, &kcd_process_destroy_delayed);
 #else
 	mmu_notifier_put(&p->mmu_notifier);
 #endif
+#endif
 }
 
 static void kcd_process_notifier_release(struct mmu_notifier *mn,
@@ -1232,11 +1234,13 @@ static struct kcd_process *create_process(const struct task_struct *thread)
 	 */
 	kref_get(&process->ref);
 
+#ifdef CONFIG_MMU_NOTIFIER
 	/* Must be last, have to use release destruction after this */
 	process->mmu_notifier.ops = &kcd_process_mmu_notifier_ops;
 	err = mmu_notifier_register(&process->mmu_notifier, process->mm);
 	if (err)
 		goto err_register_notifier;
+#endif
 
 	kcd_unref_process(process);
 	get_task_struct(process->lead_thread);
diff --git a/loonggpu/loonggpu_mn.c b/loonggpu/loonggpu_mn.c
index f38ef45..4bbcbec 100644
--- a/loonggpu/loonggpu_mn.c
+++ b/loonggpu/loonggpu_mn.c
@@ -1,3 +1,4 @@
+#ifdef CONFIG_MMU_NOTIFIER
 #include <linux/firmware.h>
 #include <linux/module.h>
 #include <linux/mmu_notifier.h>
@@ -512,3 +513,4 @@ void loonggpu_mn_unregister(struct loonggpu_bo *bo)
 	mutex_unlock(&adev->mn_lock);
 }
 
+#endif
-- 
2.52.0

