From 5a01159f90249c17dd9acafe01b42ddd27f06e15 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Sun, 22 Jun 2025 16:05:51 +0800
Subject: [PATCH 26/61] loonggpu: bridge: Adapt for recent kernel API changes

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 gsgpu-bridge/bridge_phy.c | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/gsgpu-bridge/bridge_phy.c b/gsgpu-bridge/bridge_phy.c
index 921583e..353a481 100644
--- a/gsgpu-bridge/bridge_phy.c
+++ b/gsgpu-bridge/bridge_phy.c
@@ -48,9 +48,15 @@ static int bridge_phy_connector_get_modes(struct drm_connector *connector)
 	return count;
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 15, 0)
+#define MODE_VALID_CONST const
+#else
+#define MODE_VALID_CONST
+#endif
+
 static enum drm_mode_status
 bridge_phy_connector_mode_valid(struct drm_connector *connector,
-				struct drm_display_mode *mode)
+				MODE_VALID_CONST struct drm_display_mode *mode)
 {
 	struct gsgpu_device *adev = connector->dev->dev_private;
 	struct gsgpu_bridge_phy *phy =
@@ -169,7 +175,16 @@ void bridge_phy_mode_set(struct gsgpu_bridge_phy *phy,
 	__bridge_phy_mode_set(phy, mode, adj_mode);
 }
 
-static int bridge_phy_attach (lg_bridge_phy_attach_args)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 16, 0)
+#define lg_bridge_phy_attach_args_1 \
+	struct drm_bridge *bridge, \
+	struct drm_encoder *encoder, \
+	enum drm_bridge_attach_flags flags
+#else
+#define lg_bridge_phy_attach_args_1 lg_bridge_phy_attach_args
+#endif
+
+static int bridge_phy_attach (lg_bridge_phy_attach_args_1)
 {
 	struct gsgpu_bridge_phy *phy = to_bridge_phy(bridge);
 	struct gsgpu_connector *lconnector;
-- 
2.52.0

