From 958483d34fc9420a5001b67fbeb1922fadaafef6 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Tue, 24 Jun 2025 16:10:04 +0800
Subject: [PATCH 44/63] loonggpu: Handle the different drm_client_setup.h
 location in 6.12

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu/loonggpu_drv.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/loonggpu/loonggpu_drv.c b/loonggpu/loonggpu_drv.c
index 3f08044..5979dc0 100644
--- a/loonggpu/loonggpu_drv.c
+++ b/loonggpu/loonggpu_drv.c
@@ -6,7 +6,13 @@
 #include <linux/delay.h>
 #include <linux/pci.h>
 #include <linux/delay.h>
+
+#if __has_include(<drm/clients/drm_client_setup.h>)
 #include <drm/clients/drm_client_setup.h>
+#else
+#include <drm/drm_client_setup.h>
+#endif
+
 #include <drm/drm_file.h>
 #include <drm/drm_drv.h>
 #include <drm/drm_device.h>
-- 
2.52.0

