From 65dbc94a1b54a95afbac38af1fc693ebbac0a97d Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Fri, 12 Jun 2026 20:25:29 +0800
Subject: [PATCH 48/49] lgkcd: only build kcd_debugfs.c if CONFIG_DEBUG_FS is
 enabled

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 lgkcd/lgkcd.Kbuild | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lgkcd/lgkcd.Kbuild b/lgkcd/lgkcd.Kbuild
index 8cbb8d6..02ecd80 100644
--- a/lgkcd/lgkcd.Kbuild
+++ b/lgkcd/lgkcd.Kbuild
@@ -4,7 +4,6 @@ LGKCD_SOURCES += lgkcd/kcd_ipc.c
 LGKCD_SOURCES += lgkcd/kcd_chardev.c
 LGKCD_SOURCES += lgkcd/kcd_crat.c
 LGKCD_SOURCES += lgkcd/kcd_debug.c
-LGKCD_SOURCES += lgkcd/kcd_debugfs.c
 LGKCD_SOURCES += lgkcd/kcd_device.c
 LGKCD_SOURCES += lgkcd/kcd_device_queue_manager.c
 LGKCD_SOURCES += lgkcd/kcd_device_queue_manager_lg2xx.c
@@ -30,3 +29,7 @@ ifeq ($(CONFIG_VDD_LOONGSON_SVM),1)
 LGKCD_SOURCES += lgkcd/kcd_svm.c
 LGKCD_SOURCES += lgkcd/kcd_migrate.c
 endif
+
+ifeq ($(CONFIG_DEBUG_FS),y)
+LGKCD_SOURCES += lgkcd/kcd_debugfs.c
+endif
-- 
2.54.0

