From 44216a176d0561fafa5c7d3a34e8f842e94a8f64 Mon Sep 17 00:00:00 2001
From: Thomas Otto <th1000s@posteo.net>
Date: Thu, 9 Jan 2025 22:39:47 +0100
Subject: [PATCH 3/6] Upgrade unicode-width to v0.1.14 (but still < 0.2.0)

---
 Cargo.lock | 4 ++--
 Cargo.toml | 4 +++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 051cdba..8fd8d14 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1576,9 +1576,9 @@ checksum = "1dd624098567895118886609431a7c3b8f516e41d30e0643f03d94592a147e36"
 
 [[package]]
 name = "unicode-width"
-version = "0.1.12"
+version = "0.1.14"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "68f5e5f3158ecfd4b8ff6fe086db7c8467a2dfdac97fe420f2b7c4aa97af66d6"
+checksum = "7dd6e30e90baa6f72411720665d41d89b9a3d039dc45b8faea1ddd07f617f6af"
 
 [[package]]
 name = "unsafe-libyaml"
diff --git a/Cargo.toml b/Cargo.toml
index 28982ff..efdec1e 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -55,7 +55,9 @@ syntect = "5.0.0"
 sysinfo = { version = "0.29.0", default-features = false, features = [] }
 terminal-colorsaurus = "0.4.1"
 unicode-segmentation = "1.10.1"
-unicode-width = "=0.1.12"
+# 0.2.0 (and 0.1.13) treats \n as width 1. Lines processed by delta have, lose,
+# and re-gain \n in various stages, which complicates upgrading.
+unicode-width = ">=0.1.14, <0.2.0"
 xdg = "2.4.1"
 
 [lints.rust]
-- 
2.50.1

