diff -up trousers-0.3.13/src/include/tcsps.h.noinline trousers-0.3.13/src/include/tcsps.h
--- trousers-0.3.13/src/include/tcsps.h.noinline	2014-04-24 20:05:44.000000000 +0200
+++ trousers-0.3.13/src/include/tcsps.h	2015-05-26 16:36:20.685075185 +0200
@@ -27,8 +27,8 @@ void		   ps_destroy();
 TSS_RESULT  read_data(int, void *, UINT32);
 TSS_RESULT  write_data(int, void *, UINT32);
 #else
-inline TSS_RESULT  read_data(int, void *, UINT32);
-inline TSS_RESULT  write_data(int, void *, UINT32);
+TSS_RESULT  read_data(int, void *, UINT32);
+TSS_RESULT  write_data(int, void *, UINT32);
 #endif
 int		   write_key_init(int, UINT32, UINT32, UINT32);
 TSS_RESULT	   cache_key(UINT32, UINT16, TSS_UUID *, TSS_UUID *, UINT16, UINT32, UINT32);
diff -up trousers-0.3.13/src/include/tspps.h.noinline trousers-0.3.13/src/include/tspps.h
--- trousers-0.3.13/src/include/tspps.h.noinline	2014-04-24 20:05:44.000000000 +0200
+++ trousers-0.3.13/src/include/tspps.h	2015-05-26 16:36:31.730325291 +0200
@@ -18,8 +18,8 @@
 
 TSS_RESULT	   get_file(int *);
 int		   put_file(int);
-inline TSS_RESULT  read_data(int, void *, UINT32);
-inline TSS_RESULT  write_data(int, void *, UINT32);
+TSS_RESULT  read_data(int, void *, UINT32);
+TSS_RESULT  write_data(int, void *, UINT32);
 UINT32		   psfile_get_num_keys(int);
 TSS_RESULT	   psfile_get_parent_uuid_by_uuid(int, TSS_UUID *, TSS_UUID *);
 TSS_RESULT	   psfile_remove_key_by_uuid(int, TSS_UUID *);
diff -up trousers-0.3.13/src/tcs/ps/ps_utils.c.noinline trousers-0.3.13/src/tcs/ps/ps_utils.c
--- trousers-0.3.13/src/tcs/ps/ps_utils.c.noinline	2014-04-24 20:05:44.000000000 +0200
+++ trousers-0.3.13/src/tcs/ps/ps_utils.c	2015-05-26 16:38:33.626085483 +0200
@@ -45,7 +45,7 @@ struct key_disk_cache *key_disk_cache_he
 #ifdef SOLARIS
 TSS_RESULT
 #else
-inline TSS_RESULT
+TSS_RESULT
 #endif
 read_data(int fd, void *data, UINT32 size)
 {
@@ -67,7 +67,7 @@ read_data(int fd, void *data, UINT32 siz
 #ifdef SOLARIS
 TSS_RESULT
 #else
-inline TSS_RESULT
+TSS_RESULT
 #endif
 write_data(int fd, void *data, UINT32 size)
 {
diff -up trousers-0.3.13/src/tspi/ps/ps_utils.c.noinline trousers-0.3.13/src/tspi/ps/ps_utils.c
--- trousers-0.3.13/src/tspi/ps/ps_utils.c.noinline	2014-04-24 20:05:44.000000000 +0200
+++ trousers-0.3.13/src/tspi/ps/ps_utils.c	2015-05-26 16:39:30.881381965 +0200
@@ -22,7 +22,7 @@
 #include "tspps.h"
 #include "tsplog.h"
 
-inline TSS_RESULT
+TSS_RESULT
 read_data(int fd, void *data, UINT32 size)
 {
 	int rc;
@@ -39,7 +39,7 @@ read_data(int fd, void *data, UINT32 siz
 	return TSS_SUCCESS;
 }
 
-inline TSS_RESULT
+TSS_RESULT
 write_data(int fd, void *data, UINT32 size)
 {
 	int rc;

