diff -Naur a/netfilter/iptcrdr.c b/netfilter/iptcrdr.c
--- a/netfilter/iptcrdr.c	2016-04-20 05:01:06.000000000 +0800
+++ b/netfilter/iptcrdr.c	2016-07-27 15:02:43.254894532 +0800
@@ -4,6 +4,9 @@
  * (c) 2006-2016 Thomas Bernard
  * This software is subject to the conditions detailed
  * in the LICENCE file provided within the distribution */
+
+#define _LINUX_IF_H
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -Naur a/netfilter/iptpinhole.c b/netfilter/iptpinhole.c
--- a/netfilter/iptpinhole.c	2016-01-19 18:27:00.000000000 +0800
+++ b/netfilter/iptpinhole.c	2016-07-27 15:02:35.594864888 +0800
@@ -5,6 +5,8 @@
  * This software is subject to the conditions detailed
  * in the LICENCE file provided within the distribution */
 
+#define _LINUX_IF_H
+
 #include <stdlib.h>
 #include <string.h>
 #include <syslog.h>
