From 60850cb774280cab958b67538c9067108b8e0b1d Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Thu, 5 Mar 2026 17:15:13 +0800
Subject: [PATCH 1/2] don't exit when dotnet sdk version mismatch

---
 build.psm1 | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build.psm1 b/build.psm1
index 871526cff..9edc4e574 100644
--- a/build.psm1
+++ b/build.psm1
@@ -432,7 +432,6 @@ Fix steps:
 3. Start-PSBuild -Clean
 `n
 "@
-        return
     }
 
     # set output options
-- 
2.52.0

