From 1177ffe1a0f1cda5b397a940b2de16a14765553a Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Fri, 27 Mar 2026 21:43:05 +0800
Subject: [PATCH 1/4] Support build on LoongArch64 architecture

---
 build.psm1 | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/build.psm1 b/build.psm1
index 42a202d94..da4551162 100644
--- a/build.psm1
+++ b/build.psm1
@@ -425,6 +425,8 @@ function Start-PSBuild {
                      "linux-arm",
                      "linux-arm64",
                      "linux-x64",
+                     "linux-loongarch64",
+                     "linux-musl-loongarch64",
                      "osx-arm64",
                      "osx-x64",
                      "win-arm",
@@ -1220,6 +1222,8 @@ function New-PSOptions {
                      "linux-arm",
                      "linux-arm64",
                      "linux-x64",
+                     "linux-loongarch64",
+                     "linux-musl-loongarch64",
                      "osx-arm64",
                      "osx-x64",
                      "win-arm",
@@ -4691,6 +4695,9 @@ function Clear-NativeDependencies
         '.*-arm64' {
             $diasymFileName = $diasymFileNamePattern -f 'arm64'
         }
+        '.*-loongarch64' {
+            $diasymFileName = $diasymFileNamePattern -f 'loongarch64'
+        }
         'fxdependent.*' {
             Write-Verbose -Message "$($script:Options.Runtime) is a fxdependent runtime, no cleanup needed in pwsh.deps.json" -Verbose
             return
-- 
2.52.0

