From 86d1f6bf614146d96535f55e257dc4b407e8b71e Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Sat, 12 Oct 2024 03:02:32 +0800
Subject: [PATCH] AOSCOS: allow known security vulnerabilities to fix building

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 PowerShell.Common.props | 1 +
 1 file changed, 1 insertion(+)

diff --git a/PowerShell.Common.props b/PowerShell.Common.props
index 72fbf3c86..1e3a6a741 100644
--- a/PowerShell.Common.props
+++ b/PowerShell.Common.props
@@ -142,6 +142,7 @@
     <LangVersion>11.0</LangVersion>
 
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <WarningsNotAsErrors>NU1901;NU1902;NU1903;NU1904</WarningsNotAsErrors>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <SuppressNETCoreSdkPreviewMessage>true</SuppressNETCoreSdkPreviewMessage>
-- 
2.47.0.windows.1

