diff --git a/build.psm1 b/build.psm1
index 936296588..bda161be5 100644
--- a/build.psm1
+++ b/build.psm1
@@ -192,6 +192,7 @@ function Get-EnvironmentInformation
         $environment += @{'IsSUSEFamily' = $environment.IsSLES -or $environment.IsOpenSUSE}
         $environment += @{'IsAlpine' = $LinuxInfo.ID -match 'alpine'}
         $environment += @{'IsMariner' = $LinuxInfo.ID -match 'mariner'}
+        $environment += @{'IsAOSC' = $LinuxInfo.ID -match 'aosc'}
 
         # Workaround for temporary LD_LIBRARY_PATH hack for Fedora 24
         # https://github.com/PowerShell/PowerShell/issues/2511
@@ -206,7 +207,8 @@ function Get-EnvironmentInformation
             $environment.IsRedHatFamily -or
             $environment.IsSUSEFamily -or
             $environment.IsAlpine -or
-            $environment.IsMariner)
+            $environment.IsMariner -or
+            $environment.IsAOSC)
         ) {
             if ($SkipLinuxDistroCheck) {
                 Write-Warning "The current OS : $($LinuxInfo.ID) is not supported for building PowerShell."
