From 6a6f5eb5e08be30fd0ffbc8c968af2b7f61a0440 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Thu, 5 Mar 2026 17:15:13 +0800
Subject: [PATCH 2/4] don't exit when dotnet sdk version mismatch

---
 build.psm1 | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build.psm1 b/build.psm1
index da4551162..d796e523d 100644
--- a/build.psm1
+++ b/build.psm1
@@ -535,7 +535,6 @@ Fix steps:
 3. Start-PSBuild -Clean
 `n
 "@
-        return
     }
 
     # set output options
-- 
2.52.0

