From 0695495e61f6cdca0ea44a138dbdcbe40a8824a6 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Thu, 26 Mar 2026 01:18:16 +0800
Subject: [PATCH 2/2] downgrade dotnet sdk to 105

---
 global.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/global.json b/global.json
index ce67766bb..0370976a3 100644
--- a/global.json
+++ b/global.json
@@ -1,5 +1,5 @@
 {
   "sdk": {
-    "version": "10.0.201"
+    "version": "10.0.105"
   }
 }
-- 
2.52.0

