From 625c7f39f636e9234211c43db630cd09112ecea8 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Sun, 31 May 2026 01:04:08 +0800
Subject: [PATCH 4/4] disable nuget audit

---
 PowerShell.Common.props | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/PowerShell.Common.props b/PowerShell.Common.props
index 0b40b8d71..4a7b504cc 100644
--- a/PowerShell.Common.props
+++ b/PowerShell.Common.props
@@ -22,6 +22,10 @@
       <Output TaskParameter="ConsoleOutput" PropertyName="PowerShellVersion" />
     </Exec>
 
+    <PropertyGroup>
+        <NuGetAudit>false</NuGetAudit>
+    </PropertyGroup>
+
     <PropertyGroup Condition = "'$(ReleaseTag)' != ''">
       <!--
          DO NOT UPDATE without reviewing with the Microsoft update plan.
-- 
2.52.0

