diff --git a/src/main.rs b/src/main.rs
index 8ffd39c..1401d30 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -180,22 +180,24 @@ For more information about this issue see https://askubuntu.com/questions/110969
     #[cfg(target_os = "linux")]
     let distribution = linux::Distribution::detect();
 
-    #[cfg(target_os = r#"linux"#)]
-    {
-        match &distribution {
-            Ok(distribution) => {
-                runner.execute(Step::System, "System update", || distribution.upgrade(&ctx))?;
-            }
-            Err(e) => {
-                println!("Error detecting current distribution: {e}");
-            }
-        }
-        runner.execute(Step::ConfigUpdate, "config-update", || linux::run_config_update(&ctx))?;
-
-        runner.execute(Step::BrewFormula, "Brew", || {
-            unix::run_brew_formula(&ctx, unix::BrewVariant::Path)
-        })?;
-    }
+    // Lines below are for system upgrades
+    // Commented out because it may be a bit too aggressive
+    // #[cfg(target_os = r#"linux"#)]
+    // {
+    //     match &distribution {
+    //         Ok(distribution) => {
+    //             runner.execute(Step::System, "System update", || distribution.upgrade(&ctx))?;
+    //         }
+    //         Err(e) => {
+    //             println!("Error detecting current distribution: {e}");
+    //         }
+    //     }
+    //     runner.execute(Step::ConfigUpdate, "config-update", || linux::run_config_update(&ctx))?;
+
+    //     runner.execute(Step::BrewFormula, "Brew", || {
+    //         unix::run_brew_formula(&ctx, unix::BrewVariant::Path)
+    //     })?;
+    // }
 
     #[cfg(windows)]
     {
@@ -439,16 +441,18 @@ For more information about this issue see https://askubuntu.com/questions/110969
         }
     }
 
-    #[cfg(target_os = "linux")]
-    {
-        runner.execute(Step::System, "pihole", || {
-            linux::run_pihole_update(ctx.sudo().as_ref(), run_type)
-        })?;
-        runner.execute(Step::Firmware, "Firmware upgrades", || linux::run_fwupdmgr(&ctx))?;
-        runner.execute(Step::Restarts, "Restarts", || {
-            linux::run_needrestart(ctx.sudo().as_ref(), run_type)
-        })?;
-    }
+    // Lines below are for firmware upgrades
+    // Commented out because it may be a bit too aggressive
+    // #[cfg(target_os = "linux")]
+    // {
+    //     runner.execute(Step::System, "pihole", || {
+    //         linux::run_pihole_update(ctx.sudo().as_ref(), run_type)
+    //     })?;
+    //     runner.execute(Step::Firmware, "Firmware upgrades", || linux::run_fwupdmgr(&ctx))?;
+    //     runner.execute(Step::Restarts, "Restarts", || {
+    //         linux::run_needrestart(ctx.sudo().as_ref(), run_type)
+    //     })?;
+    // }
 
     #[cfg(target_os = "macos")]
     {
