--- DarkRadiant-2.6.0/libs/os/fs.h	2018-05-10 12:35:33.000000000 -0500
+++ DarkRadiant-2.6.0.working/libs/os/fs.h	2020-04-28 23:23:59.869908790 -0500
@@ -16,8 +16,8 @@
 // If C++17 <filesystem> is available, use that one
 #ifdef HAVE_STD_FILESYSTEM
 
-#include <filesystem>
-namespace fs = std::filesystem;
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
 #define DR_USE_STD_FILESYSTEM
 
 // At the time of writing C++17 is still in draft state, but some compilers
