--- opencl-clang-11.0.0/CMakeLists.txt	2021-02-18 11:53:20.535247530 -0800
+++ opencl-clang-11.0.0.llvm1110/CMakeLists.txt	2021-02-18 11:52:37.799674930 -0800
@@ -20,7 +20,7 @@
     add_definitions(-DUSE_PREBUILT_LLVM)
 
     if(NOT PREFERRED_LLVM_VERSION)
-        set(PREFERRED_LLVM_VERSION "11.0.0")
+        set(PREFERRED_LLVM_VERSION "11.1.0")
     endif(NOT PREFERRED_LLVM_VERSION)
     message(STATUS "Looking for LLVM version ${PREFERRED_LLVM_VERSION}")
     find_package(LLVM ${PREFERRED_LLVM_VERSION} REQUIRED)
@@ -123,7 +123,7 @@
 
 
     set(SPIRV_BASE_REVISION llvm_release_110)
-    set(TARGET_BRANCH "ocl-open-110")
+    set(TARGET_BRANCH "ocl-open-111")
     get_filename_component(LLVM_MONOREPO_DIR ${LLVM_SOURCE_DIR} DIRECTORY)
     set(LLVM_PATCHES_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/patches/llvm
                           ${CMAKE_CURRENT_SOURCE_DIR}/patches/clang)
