From: Anreas Jochens <aj@andaco.de>
Subject: Fix FTBS on AMD64

diff -Naur agrep-4.17.orig/Makefile agrep-4.17/Makefile
--- agrep-4.17.orig/Makefile	2003-10-14 09:58:05.000000000 +0000
+++ agrep-4.17/Makefile	2005-12-27 15:55:52.872184392 +0000
@@ -19,7 +19,7 @@
 ISO_CHAR_SET = 1
 
 # You might have to change this depending on your machine configuration.
-CC	      = gcc -march=i486
+CC	      = gcc
 
 SHELL = /bin/sh
 
