From 3c51bd95659a40754fdd5adae6337ded9ca15c7c Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Wed, 27 Aug 2025 06:23:38 +0800
Subject: [PATCH] fix std::sort() undefine

---
 tests/testsynth.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/testsynth.cc b/tests/testsynth.cc
index ba1e440..b9a295a 100644
--- a/tests/testsynth.cc
+++ b/tests/testsynth.cc
@@ -10,6 +10,7 @@
 
 #include "liquidsfz.hh"
 #include "config.h"
+#include <algorithm>
 
 #if HAVE_FFTW
 #include <fftw3.h>
-- 
2.51.0

