From cc4a9399837cf387f2aae28ae8737b0b0947c13b Mon Sep 17 00:00:00 2001
From: Weijia Wang <9713184+wegank@users.noreply.github.com>
Date: Sun, 15 Jun 2025 16:14:09 +0200
Subject: [PATCH 2/2] UPSTREAM: Use fq_nmod_mat_entry instead of row pointer
 (take 2) (#1280)

Signed-off-by: Weijia Wang <9713184+wegank@users.noreply.github.com>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 libpolys/polys/flintconv.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libpolys/polys/flintconv.cc b/libpolys/polys/flintconv.cc
index 89152f5e4..277a62bf1 100644
--- a/libpolys/polys/flintconv.cc
+++ b/libpolys/polys/flintconv.cc
@@ -331,7 +331,7 @@ void convSingMFlintFq_nmod_mat(matrix m, fq_nmod_mat_t M, const fq_nmod_ctx_t fq
   {
     for(j=MATCOLS(m);j>0;j--)
     {
-      convSingPFlintnmod_poly_t (M->rows[i-1]+j-1, MATELEM(m,i,j),r);
+      convSingPFlintnmod_poly_t (fq_nmod_mat_entry(M, i-1, j-1), MATELEM(m,i,j),r);
     }
   }
 }
-- 
2.51.0

