From 459f2bd8fdd6c00b28a725d795c68f2017303fad Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Tue, 20 May 2025 13:27:16 +0800
Subject: [PATCH 1/5] Add other architecture support

---
 src/simd/memchr2.rs | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/simd/memchr2.rs b/src/simd/memchr2.rs
index be2980e..afdd1de 100644
--- a/src/simd/memchr2.rs
+++ b/src/simd/memchr2.rs
@@ -28,6 +28,9 @@ unsafe fn memchr2_raw(needle1: u8, needle2: u8, beg: *const u8, end: *const u8)
 
     #[cfg(target_arch = "aarch64")]
     return unsafe { memchr2_neon(needle1, needle2, beg, end) };
+
+    #[cfg(not(any(target_arch = "x86", target_arch = "x86_64", target_arch = "aarch64")))]
+    return unsafe { memchr2_fallback(needle1, needle2, beg, end) };
 }
 
 unsafe fn memchr2_fallback(
-- 
2.49.0

