From 08b139a1eee8246553ad38a7ee32a62dac319c17 Mon Sep 17 00:00:00 2001
From: MutsukiC <mutsuki@aosc.io>
Date: Thu, 30 Apr 2026 09:41:07 +0800
Subject: [PATCH] Fix: enable stdarch_loongarch feature for stdext crate

---
 crates/stdext/src/lib.rs | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/crates/stdext/src/lib.rs b/crates/stdext/src/lib.rs
index c85378b..121b10a 100644
--- a/crates/stdext/src/lib.rs
+++ b/crates/stdext/src/lib.rs
@@ -3,6 +3,11 @@
 
 //! Arena allocators. Small and fast.
 
+#![cfg_attr(
+    target_arch = "loongarch64",
+    feature(stdarch_loongarch),
+)]
+
 pub mod alloc;
 pub mod arena;
 pub mod collections;
-- 
2.52.0

