From e1f3cb774435d11e8f68192eee17decdc6e2b661 Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Mon, 6 Mar 2023 01:23:13 +0100
Subject: [PATCH] libdpms/wayland: Do not create dpms interfaces for
 placeholder QScreens

BUG: 466674
---
 src/libdpms/waylanddpmshelper.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/libdpms/waylanddpmshelper.cpp b/src/libdpms/waylanddpmshelper.cpp
index 021dbf5c..454d5272 100644
--- a/src/libdpms/waylanddpmshelper.cpp
+++ b/src/libdpms/waylanddpmshelper.cpp
@@ -112,6 +112,13 @@ public:
 private:
     void addScreen(QScreen *screen)
     {
+        // We can't rely on checking the wl_output being null yet
+        // https://codereview.qt-project.org/c/qt/qtwayland/+/464669
+        const bool fake = screen->geometry().isEmpty() || screen->name().isEmpty();
+        if (fake) {
+            return;
+        }
+
         QPlatformNativeInterface *native = qGuiApp->platformNativeInterface();
         wl_output *output = reinterpret_cast<wl_output *>(native->nativeResourceForScreen(QByteArrayLiteral("output"), screen));
         if (output) {
-- 
GitLab

