From 68fa3e667f7d82455a6a2a6fef7960af812437aa Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 31 May 2025 19:57:37 +0800
Subject: [PATCH 2/4] AOSCOS: Use CFLAGS with Swig

---
 Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 3b68142f..390ddf45 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1032,10 +1032,10 @@ $(ROBJ): $(RWRAP) \
 		$(nodist_library_include_HEADERS) \
 		$(bindings_cxx_libsigrokcxx_la_include_HEADERS) \
 		$(nodist_bindings_cxx_libsigrokcxx_la_include_HEADERS)
-	$(AM_V_CXX)$(CXX) $(RBSIGROK_CFLAGS) -I. -Iinclude -Ibindings/cxx/include $(local_includes) -fPIC -o $@ -c $(RWRAP)
+	$(AM_V_CXX)$(CXX) $(RBSIGROK_CFLAGS) $(CXXFLAGS) -I. -Iinclude -Ibindings/cxx/include $(local_includes) -fPIC -o $@ -c $(RWRAP)
 
 $(REXT): $(ROBJ) @ORDER@ bindings/cxx/libsigrokcxx.la
-	$(AM_V_CXXLD)$(CXX) -shared -o $@ $(ROBJ) -lsigrokcxx -Lbindings/cxx/.libs $(RBSIGROK_LIBS)
+	$(AM_V_CXXLD)$(CXX) -shared $(CXXFLAGS) -o $@ $(ROBJ) -lsigrokcxx -Lbindings/cxx/.libs $(RBSIGROK_LIBS)
 
 ruby-build: $(REXT)
 
-- 
2.52.0

