From 07a06df1ed0a558a6738a8d1142187df0b36c217 Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 31 May 2025 19:58:20 +0800
Subject: [PATCH 3/4] AOSCOS: Fix generation with Swig 4

---
 bindings/swig/templates.i | 1 +
 1 file changed, 1 insertion(+)

diff --git a/bindings/swig/templates.i b/bindings/swig/templates.i
index 1baaf0fe..a2e98f1d 100644
--- a/bindings/swig/templates.i
+++ b/bindings/swig/templates.i
@@ -22,6 +22,7 @@
 using namespace std;
 %}
 
+%include "attribute.i"
 %include "std_string.i"
 %include "std_shared_ptr.i"
 %include "std_vector.i"
-- 
2.52.0

