From 9423166af120b991f59c2263dba2ea1df14fc2eb Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 31 May 2025 19:58:20 +0800
Subject: [PATCH 3/3] AOSCOS: Fix generation with Swig 4

---
 bindings/swig/templates.i | 1 +
 1 file changed, 1 insertion(+)

diff --git a/bindings/swig/templates.i b/bindings/swig/templates.i
index 1baaf0fe3a6c..a2e98f1ddb69 100644
--- a/bindings/swig/templates.i
+++ b/bindings/swig/templates.i
@@ -22,6 +22,7 @@
 using namespace std;
 %}
 
+%include "attribute.i"
 %include "std_string.i"
 %include "std_shared_ptr.i"
 %include "std_vector.i"
-- 
2.49.0

