From 86644efc04e2c9948c270d11e262fbef79e515be Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 31 Dec 2025 16:51:06 +0800
Subject: [PATCH 4/4] AOSCOS: python: fix build with SWIG 4.4

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 bindings/python/sigrok/core/classes.i | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bindings/python/sigrok/core/classes.i b/bindings/python/sigrok/core/classes.i
index d22db1c1..037f694f 100644
--- a/bindings/python/sigrok/core/classes.i
+++ b/bindings/python/sigrok/core/classes.i
@@ -85,7 +85,7 @@ typedef guint pyg_flags_type;
     if (!GLib) {
         fprintf(stderr, "Import of gi.repository.GLib failed.\n");
 #if PY_VERSION_HEX >= 0x03000000
-        return nullptr;
+        return 0;
 #else
         return;
 #endif
-- 
2.52.0

