Index: xalan/CMakeLists.txt
===================================================================
--- xalan.orig/CMakeLists.txt	2025-10-22 12:12:38.956073414 -0400
+++ xalan/CMakeLists.txt	2025-10-22 12:13:06.080002178 -0400
@@ -19,7 +19,7 @@
 
 # Run "cmake" to generate the build files for your platform
 
-cmake_minimum_required(VERSION 3.2.0)
+cmake_minimum_required(VERSION 3.10.0)
 
 # Use new variable expansion policy.
 if (POLICY CMP0053)
