Index: xalan/src/xalanc/XMLSupport/XalanOtherEncodingWriter.hpp
===================================================================
--- xalan.orig/src/xalanc/XMLSupport/XalanOtherEncodingWriter.hpp	2025-10-22 08:05:45.403631397 -0400
+++ xalan/src/xalanc/XMLSupport/XalanOtherEncodingWriter.hpp	2025-10-22 08:18:14.821542155 -0400
@@ -320,13 +320,13 @@
                 {
                     XalanUnicodeChar    value = decodeUTF16SurrogatePair(ch, theChars[i+1],  getMemoryManager());
 
-                    if (this->m_isPresentable(value))
+                    if (m_predicate(value))
                     {
                         write(value);
                     }
                     else
                     {
-                        this->writeNumberedEntityReference(value);
+                        writeNumericCharacterReference(value);
                     }
 
                     ++i;
