From 8eb74c470b7839cf8516aa986aff4a251b9a41a4 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 14 Jan 2014 16:15:46 -0500
Subject: [PATCH 36/74] Don't print during fdt loading method.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 util/grub.d/10_linux.in | 2 --
 1 file changed, 2 deletions(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 379a577..794f04b 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -162,9 +162,7 @@ EOF
     fi
   fi
   if test -n "${fdt}" ; then
-    message="$(gettext_printf "Loading fdt ...")"
     sed "s/^/$submenu_indentation/" << EOF
-	echo	'$(echo "$message" | grub_quote)'
 	devicetree	${rel_dirname}/${fdt}
 EOF
   fi
-- 
2.4.3

