From 2f7da057da45a039313a5a905f58f45f7837da6e Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Mon, 30 Jun 2014 14:16:46 -0400
Subject: [PATCH 38/74] Don't munge raw spaces when we're doing our cmdline
 escaping (#923374)

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 grub-core/lib/cmdline.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/grub-core/lib/cmdline.c b/grub-core/lib/cmdline.c
index 0a5b2af..970ea86 100644
--- a/grub-core/lib/cmdline.c
+++ b/grub-core/lib/cmdline.c
@@ -97,16 +97,7 @@ int grub_create_loader_cmdline (int argc, char *argv[], char *buf,
 
       while (*c)
 	{
-	  if (*c == ' ')
-	    {
-	      *buf++ = '\\';
-	      *buf++ = 'x';
-	      *buf++ = '2';
-	      *buf++ = '0';
-	      c++;
-	      continue;
-	    }
-	  else if (*c == '\\' && *(c+1) == 'x' &&
+	  if (*c == '\\' && *(c+1) == 'x' &&
 		   is_hex(*(c+2)) && is_hex(*(c+3)))
 	    {
 	      *buf++ = *c++;
-- 
2.4.3

