From 753e72ea12dfeb7a1374b9840325795cf65897a5 Mon Sep 17 00:00:00 2001
From: Fedora Ninjas <grub2-owner@fedoraproject.org>
Date: Mon, 10 Feb 2014 16:13:10 -0500
Subject: [PATCH 48/74] Switch to use APM Mustang device tree, for hardware
 testing.

Signed-off-by: David A. Marlin <d.marlin@redhat.com>
---
 util/grub.d/10_linux.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
index 137af42..c97226a 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -230,8 +230,8 @@ while [ "x$list" != "x" ] ; do
 
   fdt=
   for i in "dtb-${version}" "dtb-${alt_version}"; do
-    if test -e "${dirname}/${i}/foundation-v8.dtb" ; then
-      fdt="${i}/foundation-v8.dtb"
+    if test -e "${dirname}/${i}/apm-mustang.dtb" ; then
+      fdt="${i}/apm-mustang.dtb"
       break
     fi
   done
-- 
2.4.3

