diff --git a/10_linux.orig b/10_linux
old mode 100644
new mode 100755
index b82ecee..e79af9c
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -217,7 +217,7 @@ while [ "x$list" != "x" ] ; do
   dirname=`dirname $linux`
   rel_dirname=`make_system_path_relative_to_its_root $dirname`
   localname=`echo $basename | sed -e "s,vmlinuz-,,g" | sed -e "s,-[0-9].*,,g"`
-  version=`echo $basename | sed -e "s,^[^0-9]*-,,g"`
+  version=`echo $basename | sed -e "s,.*-,,g"`
   alt_version=`echo $version | sed -e "s,\.old$,,g"`
   linux_root_device_thisversion="${LINUX_ROOT_DEVICE}"
