diff --git a/10_linux.orig b/10_linux
index 008b627..b82ecee 100644
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -216,6 +216,7 @@ while [ "x$list" != "x" ] ; do
   basename=`basename $linux`
   dirname=`dirname $linux`
   rel_dirname=`make_system_path_relative_to_its_root $dirname`
+  localname=`echo $basename | sed -e "s,vmlinu[z,x]-,,g" | sed -e "s,-[0-9].*,,g"`
   version=`echo $basename | sed -e "s,^[^0-9]*-,,g"`
   alt_version=`echo $version | sed -e "s,\.old$,,g"`
   linux_root_device_thisversion="${LINUX_ROOT_DEVICE}"
@@ -229,9 +230,7 @@ while [ "x$list" != "x" ] ; do
 	   "initramfs-genkernel-${alt_version}" \
 	   "initramfs-genkernel-${GENKERNEL_ARCH}-${version}" \
 	   "initramfs-genkernel-${GENKERNEL_ARCH}-${alt_version}" \
-	   "initramfs-${version}-aosc-main.img" "initramfs-${version}-oracle-uek.img" \
-	   "initramfs-${version}-ckt9-debian.img" "initramfs-${version}-fedora.img" \
-	   "initramfs-${version}-linux-rt.img"; do
+	   "initramfs-${version}-${localname}.img"; do
     if test -e "${dirname}/${i}" ; then
       initrd="$i"
       break
