From 22ba237a041b493cd0b7bb3e1c3fcd0a34dbdc2d Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Sun, 28 Jun 2015 17:57:24 -0400
Subject: [PATCH 4/6] Don't tell the compiler to do annoying things with -f.

---
 configure.ac | 34 ++--------------------------------
 1 file changed, 2 insertions(+), 32 deletions(-)

diff --git a/configure.ac b/configure.ac
index 8d780e3..5e2529a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -548,8 +548,8 @@ LDFLAGS="$TARGET_LDFLAGS"
 LIBS=""
 
 # debug flags.
-TARGET_CFLAGS="$TARGET_CFLAGS $WARN_FLAGS -g -Wredundant-decls -Wmissing-prototypes -Wmissing-declarations"
-TARGET_CCASFLAGS="$TARGET_CCASFLAGS -g"
+TARGET_CFLAGS="$TARGET_CFLAGS $WARN_FLAGS -g3 -Wredundant-decls -Wmissing-prototypes -Wmissing-declarations"
+TARGET_CCASFLAGS="$TARGET_CCASFLAGS -g3"
 
 if test "x$target_cpu" != xi386 && test "x$target_cpu" != xx86_64; then
 TARGET_CFLAGS="$TARGET_CFLAGS -Wcast-align"
@@ -809,21 +809,6 @@ if test x"$target_cpu" = xsparc64 ; then
   TARGET_LDFLAGS="$TARGET_LDFLAGS $grub_cv_target_cc_mno_relax"
 fi
 
-# By default, GCC 4.4 generates .eh_frame sections containing unwind
-# information in some cases where it previously did not. GRUB doesn't need
-# these and they just use up vital space. Restore the old compiler
-# behaviour.
-AC_CACHE_CHECK([whether -fno-dwarf2-cfi-asm works], [grub_cv_cc_fno_dwarf2_cfi_asm], [
-  CFLAGS="$TARGET_CFLAGS -fno-dwarf2-cfi-asm"
-  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[]])],
-      [grub_cv_cc_fno_dwarf2_cfi_asm=yes],
-      [grub_cv_cc_fno_dwarf2_cfi_asm=no])
-])
-
-if test "x$grub_cv_cc_fno_dwarf2_cfi_asm" = xyes; then
-  TARGET_CFLAGS="$TARGET_CFLAGS -fno-dwarf2-cfi-asm"
-fi
-
 if test x"$target_os" = xcygwin; then
   AC_CACHE_CHECK([whether option -fno-reorder-functions works], grub_cv_cc_no_reorder_functions, [
     CFLAGS="$TARGET_CFLAGS -fno-reorder-functions"
@@ -837,21 +822,6 @@ if test x"$target_os" = xcygwin && test "x$grub_cv_cc_no_reorder_functions" = xy
   TARGET_CFLAGS="$TARGET_CFLAGS -fno-reorder-functions"
 fi
 
-# By default, GCC 4.6 generates .eh_frame sections containing unwind
-# information in some cases where it previously did not. GRUB doesn't need
-# these and they just use up vital space. Restore the old compiler
-# behaviour.
-AC_CACHE_CHECK([whether -fno-asynchronous-unwind-tables works], [grub_cv_cc_fno_asynchronous_unwind_tables], [
-  CFLAGS="$TARGET_CFLAGS -fno-dwarf2-cfi-asm"
-  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[]])],
-      [grub_cv_cc_fno_asynchronous_unwind_tables=yes],
-      [grub_cv_cc_fno_asynchronous_unwind_tables=no])
-])
-
-if test "x$grub_cv_cc_fno_asynchronous_unwind_tables" = xyes; then
-  TARGET_CFLAGS="$TARGET_CFLAGS -fno-asynchronous-unwind-tables"
-fi
-
 AC_ARG_ENABLE([efiemu],
 	      [AS_HELP_STRING([--enable-efiemu],
                              [build and install the efiemu runtimes (default=guessed)])])
-- 
2.4.3

