From 4b5d7fb44ea120c8f2f52920964f57b3a9cf1540 Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Fri, 8 Aug 2025 16:12:57 +0800
Subject: [PATCH 2/2] bootblocks module fix function define

---
 bootblocks/makeboot.c | 399 +++++++++++++++++++++---------------------
 1 file changed, 200 insertions(+), 199 deletions(-)

diff --git a/bootblocks/makeboot.c b/bootblocks/makeboot.c
index 4b55068..1d9b045 100644
--- a/bootblocks/makeboot.c
+++ b/bootblocks/makeboot.c
@@ -183,190 +183,7 @@ unsigned char bpb_buffer[100];
 unsigned char bpb_flags[100];
 int has_bpb_overrides = 0;
 
-main(argc, argv)
-int argc;
-char ** argv;
-{
-   FILE * fd;
-   struct bblist *ptr = bblocks;
-   int i;
-   int ar, opton=1;
-   char *p;
-   char * bbname = 0;
-   char * devname = 0;
-
-   progname = argv[0];
-
-   /* Traditional option processing. */
-   for(ar=1; ar<argc; ar++) 
-      if(opton && argv[ar][0] == '-' && argv[ar][1] != 0) 
-	 for(p=argv[ar]+1;*p;p++)
-	    switch(*p) {
-	       char ch, * ap;
-	       case '-': opton = 0; break;
-	       case 'f': force++; break;
-	       case 'H': set_superfield("drive=0x80");
-	       case 'g': set_superfield("spt=63");
-                         set_superfield("heads=255");
-			 break;
-	       default: 
-		  ch = *p;
-	          if (p[1]) { ap = p+1; p=" "; }
-		  else {
-		     if (ar+1>=argc) Usage();
-		     ap = argv[++ar];
-		  }
-		  switch(ch) {
-		     case 'S': set_superfield(ap); break;
-		     default:  Usage();
-		  }
-		  break;
-	    }
-      else {
-         if (bbname) Usage();
-	 bbname = devname;
-	 devname = argv[ar];
-      }
-
-   if( devname == 0 ) Usage();
-   /* First bootblock is default */
-   if( bbname == 0 ) bbname = bblocks->name;
-
-   boot_id = strchr(bbname, '=');
-   if( boot_id )
-      *boot_id++ = '\0';
-
-   if( (i=strlen(bbname)) < 2 ) Usage();
-   for(ptr = bblocks; ptr->name; ptr++)
-      if( strncmp(bbname, ptr->name, i) == 0 ) break;
-   if( ptr->name == 0 ) Usage();
-
-   if (has_bpb_overrides && ptr != bblocks &&
-       ptr->fstype != FS_DOS && ptr->fstype != FS_ADOS)
-      Usage();
-
-   open_disk(devname);
-   if( read_sector(0, buffer) != 0 )
-      exit(1);
-   read_sector(1, buffer+512);
-
-   write_zero = (ptr->size >= 512);
-   write_one =  (ptr->size >= 1024);
-
-   switch(ptr->fstype)
-   {
-   case FS_NONE:	/* override */
-   case FS_STAT:
-   case FS_ADOS:
-      if ( has_bpb_overrides )
-         check_msdos();
-      break;
-   case FS_DOS:
-      check_msdos();
-      if(ptr->fsmod) check_simpledos(ptr->fsmod);
-      break;
-   case FS_TAR:
-      check_tar();
-      break;
-   case FS_ZERO:
-      check_zapped();
-      break;
-   case FS_MBR:
-      check_mbr();
-      break;
-
-   default:
-      fprintf(stderr, "Program error, unknown filesystem requirement\n");
-      exit(2);
-   }
-
-   switch(ptr->fstype)
-   {
-   case FS_DOS:
-   case FS_ADOS:
-      if (ptr->fsmod == 12 || ptr->fsmod == 16)
-      {
-	 for(i=0; i<sysboot16_dosfs_stat; i++)
-	    buffer[i] = ptr->data[i];
-	 for(i=sysboot16_codestart; i<512; i++)
-	    buffer[i] = ptr->data[i];
-      }
-      else
-      {
-	 for(i=0; i<sysboot_dosfs_stat; i++)
-	    buffer[i] = ptr->data[i];
-	 for(i=sysboot_codestart; i<512; i++)
-	    buffer[i] = ptr->data[i];
-      }
-      break;
-
-   case FS_TAR:
-      copy_tarblock();
-      break;
-
-   case FS_MBR:
-      copy_mbr(ptr->data, ptr->size);
-      break;
-
-   case FS_ZERO:
-   case FS_NONE:
-      if( ptr->data )
-	 memcpy(buffer, ptr->data, ptr->size);
-      else
-      {
-	 memset(buffer, '\0', 1024);
-	 write_one = 1;
-      }
-      break;
-   }
-
-   if ( has_bpb_overrides )
-   {
-      copy_superfields(buffer);
-      write_zero = 1;
-   }
-
-   switch(ptr->fstype)
-   {
-   case FS_STAT:
-      if( strcmp(ptr->name, "copy") == 0 )
-         save_super(buffer);
-      else
-         print_super(buffer);
-
-      if ( !has_bpb_overrides )
-      {
-         close_disk();
-         exit(0);
-      }
-      break;
-   }
-
-   if( boot_id ) switch(ptr->name_type)
-   {
-   case 1:
-      set_dosname(ptr->boot_name);
-      break;
-   case 2:
-      set_asciz(ptr->boot_name);
-      break;
-   default:
-      fprintf(stderr, "Cannot specify boot name for this block\n");
-      exit(1);
-   }
-
-   if( write_zero ) {
-      write_sector(0, buffer);
-      if (copy_zero)
-	 write_sector(copy_zero, buffer); /* FAT32 backup */
-   }
-   if( write_one )  write_sector(1, buffer+512);
-
-   close_disk();
-   exit(0);
-}
-
-Usage()
+void Usage()
 {
    struct bblist *ptr = bblocks;
 
@@ -421,7 +238,7 @@ char * diskname;
    return 0;
 }
 
-close_disk()
+void close_disk()
 {
    if( diskfd && disktype == 0 ) fclose(diskfd);
    diskfd = 0;
@@ -707,7 +524,7 @@ dos_write_err:
 #endif
 /**************************************************************************/
 
-check_zapped()
+void check_zapped()
 {
    int i;
    for(i=0; i<512; i++)
@@ -754,7 +571,7 @@ char *s;
    return val;
 }
 
-check_tar()
+int check_tar()
 {
    char vbuf[100];
    unsigned char *p;
@@ -797,7 +614,7 @@ not_zapped:
    return 0;
 }
 
-copy_tarblock()
+void copy_tarblock()
 {
    char lbuf[20];
    unsigned char * p;
@@ -929,7 +746,7 @@ struct bootfields {
    { 0, -1,0,0}
 };
 
-print_super(bootsect)
+void print_super(bootsect)
 char * bootsect;
 {
 static char * fieldnames[] = {
@@ -1018,7 +835,7 @@ static char * fieldnames[] = {
    }
 }
 
-decode_super(bootsect)
+void decode_super(bootsect)
 char * bootsect;
 {
    int i;
@@ -1052,7 +869,7 @@ char * bootsect;
      copy_zero = 0;
 }
 
-save_super(bootsect)
+void save_super(bootsect)
 char * bootsect;
 {
    FILE * fd;
@@ -1065,7 +882,7 @@ char * bootsect;
    fclose(fd);
 }
 
-set_superfield(setstr)
+void set_superfield(setstr)
 char * setstr;
 {
    int i, l;
@@ -1085,7 +902,7 @@ char * setstr;
    }
 }
 
-copy_superfields(bootsect)
+void copy_superfields(bootsect)
 char * bootsect;
 {
    int i, j;
@@ -1122,7 +939,7 @@ char * bootsect;
 
 /**************************************************************************/
 
-check_msdos()
+void check_msdos()
 {
    decode_super(buffer);
    if( dosflds[DOS_CLUST].value == 0 )	/* MSDOS v1.0 */
@@ -1148,7 +965,7 @@ check_msdos()
    if(!force) exit(2);
 }
 
-check_simpledos(bb_fatbits)
+void check_simpledos(bb_fatbits)
 int bb_fatbits;
 {
    long numclust = 0xFFFF;
@@ -1224,7 +1041,7 @@ int bb_fatbits;
    }
 }
 
-set_dosname(boot_name)
+void set_dosname(boot_name)
 int boot_name;
 {
    char dos_name[20];
@@ -1246,7 +1063,7 @@ int boot_name;
    memcpy(buffer+boot_name, dos_name, 11);
 }
 
-set_asciz(boot_name)
+void set_asciz(boot_name)
 int boot_name;
 {
    int i, j;
@@ -1288,7 +1105,7 @@ int boot_name;
    }
 }
 
-check_mbr()
+void check_mbr()
 {
    int i = 0;
 
@@ -1319,7 +1136,7 @@ check_mbr()
    }
 }
 
-copy_mbr(boot_data, boot_size)
+void copy_mbr(boot_data, boot_size)
 char * boot_data;
 int boot_size;
 {
@@ -1338,3 +1155,187 @@ int boot_size;
    buffer[511] = 0xAA;
    write_zero = 1;
 }
+
+int main(argc, argv)
+int argc;
+char ** argv;
+{
+   FILE * fd;
+   struct bblist *ptr = bblocks;
+   int i;
+   int ar, opton=1;
+   char *p;
+   char * bbname = 0;
+   char * devname = 0;
+
+   progname = argv[0];
+
+   /* Traditional option processing. */
+   for(ar=1; ar<argc; ar++) 
+      if(opton && argv[ar][0] == '-' && argv[ar][1] != 0) 
+	 for(p=argv[ar]+1;*p;p++)
+	    switch(*p) {
+	       char ch, * ap;
+	       case '-': opton = 0; break;
+	       case 'f': force++; break;
+	       case 'H': set_superfield("drive=0x80");
+	       case 'g': set_superfield("spt=63");
+                         set_superfield("heads=255");
+			 break;
+	       default: 
+		  ch = *p;
+	          if (p[1]) { ap = p+1; p=" "; }
+		  else {
+		     if (ar+1>=argc) Usage();
+		     ap = argv[++ar];
+		  }
+		  switch(ch) {
+		     case 'S': set_superfield(ap); break;
+		     default:  Usage();
+		  }
+		  break;
+	    }
+      else {
+         if (bbname) Usage();
+	 bbname = devname;
+	 devname = argv[ar];
+      }
+
+   if( devname == 0 ) Usage();
+   /* First bootblock is default */
+   if( bbname == 0 ) bbname = bblocks->name;
+
+   boot_id = strchr(bbname, '=');
+   if( boot_id )
+      *boot_id++ = '\0';
+
+   if( (i=strlen(bbname)) < 2 ) Usage();
+   for(ptr = bblocks; ptr->name; ptr++)
+      if( strncmp(bbname, ptr->name, i) == 0 ) break;
+   if( ptr->name == 0 ) Usage();
+
+   if (has_bpb_overrides && ptr != bblocks &&
+       ptr->fstype != FS_DOS && ptr->fstype != FS_ADOS)
+      Usage();
+
+   open_disk(devname);
+   if( read_sector(0, buffer) != 0 )
+      exit(1);
+   read_sector(1, buffer+512);
+
+   write_zero = (ptr->size >= 512);
+   write_one =  (ptr->size >= 1024);
+
+   switch(ptr->fstype)
+   {
+   case FS_NONE:	/* override */
+   case FS_STAT:
+   case FS_ADOS:
+      if ( has_bpb_overrides )
+         check_msdos();
+      break;
+   case FS_DOS:
+      check_msdos();
+      if(ptr->fsmod) check_simpledos(ptr->fsmod);
+      break;
+   case FS_TAR:
+      check_tar();
+      break;
+   case FS_ZERO:
+      check_zapped();
+      break;
+   case FS_MBR:
+      check_mbr();
+      break;
+
+   default:
+      fprintf(stderr, "Program error, unknown filesystem requirement\n");
+      exit(2);
+   }
+
+   switch(ptr->fstype)
+   {
+   case FS_DOS:
+   case FS_ADOS:
+      if (ptr->fsmod == 12 || ptr->fsmod == 16)
+      {
+	 for(i=0; i<sysboot16_dosfs_stat; i++)
+	    buffer[i] = ptr->data[i];
+	 for(i=sysboot16_codestart; i<512; i++)
+	    buffer[i] = ptr->data[i];
+      }
+      else
+      {
+	 for(i=0; i<sysboot_dosfs_stat; i++)
+	    buffer[i] = ptr->data[i];
+	 for(i=sysboot_codestart; i<512; i++)
+	    buffer[i] = ptr->data[i];
+      }
+      break;
+
+   case FS_TAR:
+      copy_tarblock();
+      break;
+
+   case FS_MBR:
+      copy_mbr(ptr->data, ptr->size);
+      break;
+
+   case FS_ZERO:
+   case FS_NONE:
+      if( ptr->data )
+	 memcpy(buffer, ptr->data, ptr->size);
+      else
+      {
+	 memset(buffer, '\0', 1024);
+	 write_one = 1;
+      }
+      break;
+   }
+
+   if ( has_bpb_overrides )
+   {
+      copy_superfields(buffer);
+      write_zero = 1;
+   }
+
+   switch(ptr->fstype)
+   {
+   case FS_STAT:
+      if( strcmp(ptr->name, "copy") == 0 )
+         save_super(buffer);
+      else
+         print_super(buffer);
+
+      if ( !has_bpb_overrides )
+      {
+         close_disk();
+         exit(0);
+      }
+      break;
+   }
+
+   if( boot_id ) switch(ptr->name_type)
+   {
+   case 1:
+      set_dosname(ptr->boot_name);
+      break;
+   case 2:
+      set_asciz(ptr->boot_name);
+      break;
+   default:
+      fprintf(stderr, "Cannot specify boot name for this block\n");
+      exit(1);
+   }
+
+   if( write_zero ) {
+      write_sector(0, buffer);
+      if (copy_zero)
+	 write_sector(copy_zero, buffer); /* FAT32 backup */
+   }
+   if( write_one )  write_sector(1, buffer+512);
+
+   close_disk();
+   exit(0);
+   return 0;
+}
\ No newline at end of file
-- 
2.40.1.windows.1

