From 492676ac9777aca55dd711328f3f9affb3f05ed2 Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Sat, 9 Aug 2025 10:19:35 +0800
Subject: [PATCH] use c89 mode

---
 makefile.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/makefile.in b/makefile.in
index b586da1..d11abf8 100644
--- a/makefile.in
+++ b/makefile.in
@@ -16,7 +16,7 @@ ARFLAGS =r
 CC      =%CC%
 CFLAGS	=%CFLAGS%
 LDFLAGS	=
-MAKEARG	=CC='$(CC)' CFLAGS='$(CFLAGS)' LDFLAGS='$(LDFLAGS)' \
+MAKEARG	=CC='$(CC)' CFLAGS='$(CFLAGS) -std=gnu89' LDFLAGS='$(LDFLAGS)' \
 	 PREFIX=$(PREFIX) LIBDIR='$(LIBDIR)' \
 	 BINDIR='$(BINDIR)' ANSI='$(ANSI)'
 MAKEC=$(MAKE) -C
@@ -81,7 +81,7 @@ CFLAGS += -no-cpp-precomp
 
 # Install files with the userid of the currently running process.
 INDAT=-m 644
-INEXE=-m 755 -s
+INEXE=-m 755
 INSCR=-m 755
 
 #ifdef __CYGWIN__
-- 
2.50.1

