From e06c824beb4f490d9d6bc4d824f80f64b916e1ce Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 24 Jul 2024 23:18:19 +0800
Subject: [PATCH 2/3] fix(goldendict.pro): respect Autobuild flags

---
 goldendict.pro | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/goldendict.pro b/goldendict.pro
index 11cb5652..7a573c62 100644
--- a/goldendict.pro
+++ b/goldendict.pro
@@ -2,6 +2,10 @@ TEMPLATE = app
 TARGET = goldendict
 VERSION = 1.5.0+git
 
+QMAKE_CXXFLAGS += $$(CXXFLAGS)
+QMAKE_CFLAGS += $$(CFLAGS)
+QMAKE_LFLAGS += $$(LDFLAGS)
+
 # Generate version file. We do this here and in a build rule described later.
 # The build rule is required since qmake isn't run each time the project is
 # rebuilt; and doing it here is required too since any other way the RCC
-- 
2.46.0

