From 7299cf3d403b7713029b41e4e582b9b5eb98ce6a Mon Sep 17 00:00:00 2001
From: Abs62 <ottomann@yandex.ru>
Date: Sat, 11 Oct 2025 21:27:23 +0300
Subject: [PATCH 3/3] Fix compilation with FFMpeg 8.0 (issue #1729)

---
 ffmpegaudio.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ffmpegaudio.cc b/ffmpegaudio.cc
index 5ef602ad..b7cd7ddf 100644
--- a/ffmpegaudio.cc
+++ b/ffmpegaudio.cc
@@ -350,7 +350,9 @@ void DecoderContext::closeCodec()
   if ( audioStream_ && codecContext_ && codec_ )
   {
     audioStream_->discard = AVDISCARD_ALL;
+#if LIBAVCODEC_VERSION_MAJOR < 61
     avcodec_close( codecContext_ );
+#endif
 #if LIBAVCODEC_VERSION_MAJOR > 57 || ( LIBAVCODEC_VERSION_MAJOR == 57 && LIBAVCODEC_VERSION_MINOR >= 33 )
     avcodec_free_context( &codecContext_ );
 #endif
-- 
2.53.0

