--- goldendict-1.5.0-RC2/goldendict.pro	2016-04-26 16:32:50.000000000 +0000
+++ goldendict-1.5.0-RC2.systemflags/goldendict.pro	2022-03-03 06:16:26.779487079 +0000
@@ -2,6 +2,10 @@
 TARGET = goldendict
 VERSION = 1.5.0-RC2+git
 
+QMAKE_CXXFLAGS += $$(CXXFLAGS)
+QMAKE_CFLAGS += $$(CFLAGS)
+QMAKE_LFLAGS += $$(LDFLAGS)
+
 # Generate version file. We do this here and in a build rule described later.
 # The build rule is required since qmake isn't run each time the project is
 # rebuilt; and doing it here is required too since any other way the RCC
