From 3d21aa95105c71bbf037f990c9579392e7398b51 Mon Sep 17 00:00:00 2001
From: Mikhail Mikhailov <26198-yohanson@users.noreply.gitlab.xfce.org>
Date: Sat, 22 Feb 2025 08:02:47 +0000
Subject: [PATCH] systray: Fix legacy icons being drawn with offset

Fixes: #909
---
 plugins/systray/systray.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/plugins/systray/systray.c b/plugins/systray/systray.c
index fb838d449..de1977d21 100644
--- a/plugins/systray/systray.c
+++ b/plugins/systray/systray.c
@@ -261,6 +261,7 @@ systray_plugin_box_draw_icon (GtkWidget *child,
 {
   cairo_t *cr = user_data;
   GtkAllocation alloc;
+  GtkAllocation box_alloc;
 
   if (systray_socket_is_composited (SYSTRAY_SOCKET (child)))
     {
@@ -269,7 +270,10 @@ systray_plugin_box_draw_icon (GtkWidget *child,
       /* skip hidden (see offscreen in box widget) icons */
       if (alloc.x > -1 && alloc.y > -1)
         {
-          // FIXME
+          GtkWidget *box = gtk_widget_get_parent (child);
+          gtk_widget_get_allocation (box, &box_alloc);
+          alloc.x -= box_alloc.x;
+          alloc.y -= box_alloc.y;
           gdk_cairo_set_source_window (cr, gtk_widget_get_window (child),
                                        alloc.x, alloc.y);
           cairo_paint (cr);
-- 
GitLab

