From b790db0519f2cc7c2236b41f48c27045fd4f2103 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ga=C3=ABl=20Bonithon?= <gael@xfce.org>
Date: Thu, 13 Mar 2025 18:27:28 +0100
Subject: [PATCH] systray: Use correct gtk functions to show/hide items

Fixes: #921
---
 plugins/systray/sn-box.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/plugins/systray/sn-box.c b/plugins/systray/sn-box.c
index 2bb281ad6..54645d02c 100644
--- a/plugins/systray/sn-box.c
+++ b/plugins/systray/sn-box.c
@@ -245,12 +245,12 @@ sn_box_list_changed (SnBox *box,
           if (!sn_config_is_hidden (box->config, SN_ITEM_TYPE_DEFAULT,
                                     sn_button_get_name (button)))
             {
-              gtk_widget_map (GTK_WIDGET (button));
+              gtk_widget_show (GTK_WIDGET (button));
               n_visible_children++;
             }
           else
             {
-              gtk_widget_set_mapped (GTK_WIDGET (button), box->show_hidden);
+              gtk_widget_set_visible (GTK_WIDGET (button), box->show_hidden);
               n_hidden_children++;
             }
         }
@@ -407,11 +407,11 @@ sn_box_measure_and_allocate (GtkWidget *widget,
               n_hidden_children++;
               if (!box->show_hidden)
                 {
-                  gtk_widget_unmap (GTK_WIDGET (button));
+                  gtk_widget_hide (GTK_WIDGET (button));
                   continue;
                 }
             }
-          gtk_widget_map (GTK_WIDGET (button));
+          gtk_widget_show (GTK_WIDGET (button));
           n_visible_children++;
 
           gtk_widget_get_preferred_size (GTK_WIDGET (button), NULL, &child_req);
-- 
GitLab

