diff --git a/migrate/default.xml.in b/migrate/default.xml.in
index 24980455..2ed52e60 100644
--- a/migrate/default.xml.in
+++ b/migrate/default.xml.in
@@ -2,6 +2,7 @@
 
 <channel name="xfce4-panel" version="1.0">
   <property name="configver" type="int" value="@XFCE4_PANEL_CONFIG_VERSION@"/>
+  <property name="force-all-internal" type="bool" value="true"/>
   <property name="panels" type="array">
     <value type="int" value="1"/>
     <value type="int" value="2"/>
