From 556b770ebe3af0b5c91839208be0b3d077d62de7 Mon Sep 17 00:00:00 2001
From: Michael Jeanson <mjeanson@efficios.com>
Date: Mon, 30 Nov 2015 13:05:06 -0500
Subject: [PATCH] Fix: disable sys_futex on the mips architecture

Signed-off-by: Michael Jeanson <mjeanson@efficios.com>
---
 configure.ac | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -234,11 +234,26 @@
 	]])
 ],[
 	AC_MSG_RESULT([yes])
-	AC_DEFINE([CONFIG_RCU_HAVE_FUTEX], [1])
-	compat_futex_test=0
+	have_sys_futex="yes"
 ],[
 	AC_MSG_RESULT([no])
-	compat_futex_test=1
+	have_sys_futex="no"
+])
+
+# Disable sys_futex detection on MIPS,
+# the implementation is broken.
+AS_IF([test "x$ARCHTYPE" = "xmips"],[
+	have_sys_futex="no"
+])
+
+# Disable sys_futex detection on HPPA,
+# the implementation is broken.
+AS_IF([test "x$ARCHTYPE" = "xhppa"],[
+	have_sys_futex="no"
+])
+
+AS_IF([test "x$have_sys_futex" = "xyes"],[
+	AC_DEFINE([CONFIG_RCU_HAVE_FUTEX], [1])
 ])
 
 # Search for clock_gettime
@@ -247,7 +262,6 @@
 	config_rcu_have_clock_gettime=yes
 ], [])
 
-AM_CONDITIONAL([COMPAT_FUTEX], [test "x$compat_futex_test" = "x1"])
 AM_CONDITIONAL([COMPAT_ARCH], [test "x$SUBARCHTYPE" = "xx86compat"])
 AM_CONDITIONAL([NO_SHARED], [test "x$enable_shared" = "xno"])
 
@@ -425,7 +439,7 @@
 PPRINT_PROP_BOOL([Memory fence instructions], $value)
 
 # Futex support available
-test "x$compat_futex_test" = "x0" && value=1 || value=0
+test "x$have_sys_futex" = "xyes" && value=1 || value=0
 PPRINT_PROP_BOOL([Futex support], $value)
 
 # TLS
