--- scons-4.1.0/site_scons/SConsRevision.py	2021-01-19 12:10:56.000000000 -0800
+++ scons-4.1.0.bootstrap/site_scons/SConsRevision.py	2021-04-12 16:40:09.021381498 -0700
@@ -23,7 +23,6 @@
         contents = contents.replace('__DATE'      + '__', env['DATE'])
         contents = contents.replace('__DEB_DATE'  + '__', env['DEB_DATE'])
 
-        contents = contents.replace('__DEVELOPER' + '__', env['DEVELOPER'])
         contents = contents.replace('__FILE'      + '__', str(source[0]).replace('\\', '/'))
         contents = contents.replace('__MONTH_YEAR'+ '__', env['MONTH_YEAR'])
         contents = contents.replace('__REVISION'  + '__', env['REVISION'])
@@ -39,4 +38,4 @@
             of.write(contents)
 
 
-    os.chmod(t, os.stat(s)[0])
\ No newline at end of file
+    os.chmod(t, os.stat(s)[0])
