From 7ee7a5b9b656e309e68694d76f1de0dcfc272abb Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Mon, 23 Dec 2024 16:55:35 -0800
Subject: [PATCH] Disable SIGSTKFLT on mips

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 src/signal.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/signal.rs b/src/signal.rs
index 3c4e2dd5a..46e011e37 100644
--- a/src/signal.rs
+++ b/src/signal.rs
@@ -417,7 +417,7 @@ macro_rules! signal_entry {
     #[cfg(any(apple, bsd))]
     signal_entry!(SIGINFO, SIGINFO_DESC),
 
-    #[cfg(target_os = "linux")]
+    #[cfg(all(target_os = "linux", not(target_arch = "mips64")))]
     signal_entry!(SIGSTKFLT, SIGSTKFLT_DESC),
 
     #[cfg(target_os = "linux")]
-- 
2.51.0

