diff --git a/compiler/GHC/CmmToAsm/LA64/CodeGen.hs b/compiler/GHC/CmmToAsm/LA64/CodeGen.hs
index a27b03b4e3..460a14dbfe 100644
--- a/compiler/GHC/CmmToAsm/LA64/CodeGen.hs
+++ b/compiler/GHC/CmmToAsm/LA64/CodeGen.hs
@@ -513,10 +513,19 @@ getRegister' config plat expr =
               format = cmmTypeFormat rep
               width = typeWidth rep
           (off_r, _off_format, off_code) <- getSomeReg $ CmmLit (CmmInt (fromIntegral off) width)
-          return (Any format (\dst -> off_code `snocOL`
-                                                LD format (OpReg (formatToWidth format) dst) op `snocOL`
-                                                ADD (OpReg W64 dst) (OpReg width dst) (OpReg width off_r)
-                             ))
+          case width of
+            width | width `elem` [W8, W16] ->
+              return (Any format (\dst -> off_code `appOL`
+                signExtend width W64 dst dst `appOL`
+                signExtend width W64 off_r off_r `snocOL`
+                LD format (OpReg (formatToWidth format) dst) op `snocOL`
+                ADD (OpReg W64 dst) (OpReg W64 dst) (OpReg W64 off_r)
+                ))
+            _ ->
+              return (Any format (\dst -> off_code `snocOL`
+                LD format (OpReg (formatToWidth format) dst) op `snocOL`
+                ADD (OpReg W64 dst) (OpReg width dst) (OpReg width off_r)
+                ))
 
         CmmLabelDiffOff {} -> pprPanic "getRegister' (CmmLit:CmmLabelOff): " (pdoc plat expr)
         CmmBlock _ -> pprPanic "getRegister' (CmmLit:CmmLabelOff): " (pdoc plat expr)
@@ -548,13 +557,22 @@ getRegister' config plat expr =
       where
         width = typeWidth (cmmRegType reg)
     CmmRegOff reg off -> do
-      (off_r, _off_format, off_code) <- getSomeReg $ CmmLit (CmmInt (fromIntegral off) width)
-      (reg, _format, code) <- getSomeReg $ CmmReg reg
-      return $ Any (intFormat width) ( \dst ->
-                                        off_code `appOL`
-                                        code `snocOL`
-                                        ADD (OpReg W64 dst) (OpReg width reg) (OpReg width off_r)
-                                     )
+      (off_r, off_format, off_code) <- getSomeReg $ CmmLit (CmmInt (fromIntegral off) width)
+      (reg, format, code) <- getSomeReg $ CmmReg reg
+      case width of
+        width | width `elem` [W8, W16] ->
+          return (Any (intFormat width) (\dst -> off_code `appOL`
+            code `appOL`
+            signExtend (formatToWidth format) W64 reg reg `appOL`
+            signExtend (formatToWidth off_format) W64 off_r off_r `snocOL`
+            ADD (OpReg W64 dst) (OpReg W64 reg) (OpReg W64 off_r)
+            ))
+        _ ->
+          return $ Any (intFormat width) ( \dst ->
+            off_code `appOL`
+            code `snocOL`
+            ADD (OpReg W64 dst) (OpReg width reg) (OpReg width off_r)
+            )
       where
         width = typeWidth (cmmRegType reg)
 
