From 683bdf335ae5132f1d7fca581697eb0d82a52d94 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 9 Mar 2025 01:50:16 +0800
Subject: [PATCH 1/2] AOSCOS: fix(ldb/wscript): allow building with ldb 2.9

---
 lib/ldb/wscript | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/ldb/wscript b/lib/ldb/wscript
index 60bb7cf48b3..5d04cfe4c2c 100644
--- a/lib/ldb/wscript
+++ b/lib/ldb/wscript
@@ -2,7 +2,7 @@
 
 APPNAME = 'ldb'
 # For Samba 4.17.x
-VERSION = '2.6.1'
+VERSION = '2.9.1'
 
 import sys, os
 
-- 
2.48.1

