From 52cb522862e265c88bc8b338bc62ab81c2555156 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 9 Mar 2025 01:51:19 +0800
Subject: [PATCH 2/2] AOSCOS: fix(network.h): drop conflicting stropts.h

---
 lib/replace/system/network.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/lib/replace/system/network.h b/lib/replace/system/network.h
index a84bfd453b1..3062055819c 100644
--- a/lib/replace/system/network.h
+++ b/lib/replace/system/network.h
@@ -87,10 +87,6 @@
 #include <sys/uio.h>
 #endif
 
-#ifdef HAVE_STROPTS_H
-#include <stropts.h>
-#endif
-
 #ifndef HAVE_SOCKLEN_T
 #define HAVE_SOCKLEN_T
 typedef int socklen_t;
-- 
2.48.1

