--- cmix-11/makefile	2016-07-13 00:16:51.000000000 -0500
+++ cmix-11.fix/makefile	2017-03-31 14:40:01.370955897 -0500
@@ -4,8 +4,8 @@
 
 OBJS = build/preprocessor.o build/encoder.o build/decoder.o build/predictor.o build/logistic.o build/mixer-input.o build/mixer.o build/byte-mixer.o build/byte-model.o build/sse.o build/manager.o build/direct.o build/direct-hash.o build/indirect.o build/nonstationary.o build/run-map.o build/byte-run.o build/match.o build/dmc.o build/ppm.o build/ppmd.o build/bracket.o build/paq8l.o build/paq8hp.o build/bracket-context.o build/context-hash.o build/sparse.o build/indirect-hash.o build/interval.o build/interval-hash.o build/bit-context.o
 
-all: CFLAGS += -Ofast -march=native -s
-all: LFLAGS += -Ofast -march=native -s
+all: CFLAGS += -Ofast -s
+all: LFLAGS += -Ofast -s
 all: build cmix
 
 debug: CFLAGS += -ggdb
