diff -Naur dhcp-4.3.6-old/bind/Makefile.in dhcp-4.3.6/bind/Makefile.in
--- dhcp-4.3.6-old/bind/Makefile.in	2017-07-25 22:26:41.000000000 +0800
+++ dhcp-4.3.6/bind/Makefile.in	2018-03-04 14:40:12.178601606 +0800
@@ -64,6 +64,7 @@
 		echo Configuring BIND Export libraries for DHCP. ;      \
 		rm -rf ${cleandirs} ${cleanfiles} ;                     \
 		(cd ${bindsrcdir} &&                                    \
+		 cp /usr/bin/config.guess /usr/bin/config.sub . &&      \
                  ./configure ${bindconfig} > ${binddir}/configure.log); \
 	fi
 
