diff -Naur libpfm-4.12.0/python/Makefile libpfm-4.12.0.python3/python/Makefile
--- libpfm-4.12.0/python/Makefile	2022-09-16 23:21:31.000000000 -0700
+++ libpfm-4.12.0.python3/python/Makefile	2022-12-29 21:09:50.230740367 -0800
@@ -28,10 +28,10 @@
 PYTHON_PREFIX=$(PREFIX)
 
 all:
-	CFLAGS="-O2 -g" ./setup.py build
+	CFLAGS="-O2 -g" python3 ./setup.py build
 
 install:
-	 CFLAGS="-O2 -g" ./setup.py install --prefix=$(DESTDIR)$(PYTHON_PREFIX)
+	CFLAGS="-O2 -g" python3 ./setup.py install --prefix=$(DESTDIR)$(PYTHON_PREFIX)
 
 clean:
 	$(RM) src/perfmon_int_wrap.c src/perfmon_int.py src/*.pyc
