diff -Naur a/etc/mpv.conf b/etc/mpv.conf
--- a/etc/mpv.conf	2016-10-19 16:10:13.000000000 -0500
+++ b/etc/mpv.conf	2016-10-22 16:30:48.048946914 -0500
@@ -53,7 +53,7 @@
 #ontop=yes
 
 # Specify default video driver (see --vo=help for a list).
-#vo=opengl
+vo=opengl
 
 # Specify high quality video rendering preset (for OpenGL VO only)
 # Can cause performance problems with some drivers and GPUs.
@@ -68,14 +68,14 @@
 # video outputs, but should work well with default settings on most systems.
 # If performance or energy usage is an issue, forcing the vdpau or vaapi VOs
 # may or may not help.
-#hwdec=auto
+hwdec=auto
 
 ##################
 # audio settings #
 ##################
 
 # Specify default audio driver (see --ao=help for a list).
-#ao=alsa
+ao=pulse,alsa
 
 # Disable softvol usage, and always use the system mixer if available.
 #softvol=no
