diff -Naur mpv-0.15.0/etc/mpv.conf mpv-0.15.0-conf/etc/mpv.conf
--- mpv-0.15.0/etc/mpv.conf	2016-01-17 23:28:12.000000000 -0700
+++ mpv-0.15.0-conf/etc/mpv.conf	2016-01-18 09:22:52.821524762 -0700
@@ -27,7 +27,7 @@
 ##################
 
 # Specify default video driver (see --vo=help for a list).
-#vo=opengl-hq
+vo=opengl
 
 # Start in fullscreen mode by default.
 #fs=yes
@@ -49,7 +49,7 @@
 ##################
 
 # Specify default audio driver (see --ao=help for a list).
-#ao=alsa
+ao=pulse,alsa
 
 # Disable softvol usage, and always use the system mixer if available.
 #softvol=no
@@ -102,7 +102,7 @@
 
 # Enable hardware decoding if available. Often, this requires using an certain
 # video output, otherwise no hardware decoding will be used.
-#hwdec=auto
+hwdec=auto
 
 ############
 # Profiles #
