--- a/etc/mpv.conf	2017-06-04 00:10:50.129556543 +0800
+++ b/etc/mpv.conf	2017-06-04 00:36:30.137994671 +0800
@@ -61,11 +61,14 @@
 # with some drivers and desktop environments.
 #video-sync=display-resample
 
+# Specify default video driver (see --vo=help for a list).
+vo=gpu
+
 # Enable hardware decoding if available. Often, this does not work with all
 # video outputs, but should work well with default settings on most systems.
 # If performance or energy usage is an issue, forcing the vdpau or vaapi VOs
 # may or may not help.
-#hwdec=auto
+hwdec=auto
 
 ##################
 # audio settings #
