diff -Naur a/etc/mpv.conf b/etc/mpv.conf
--- a/etc/mpv.conf	2016-02-28 16:22:04.000000000 -0700
+++ b/etc/mpv.conf	2016-02-29 10:55:19.786539105 -0700
@@ -55,7 +55,7 @@
 # Specify default video driver (see --vo=help for a list).
 # This one selects high quality video scaling etc. - can cause problems with
 # some drivers and GPUs.
-#vo=opengl-hq
+vo=opengl
 
 # Force video to lock on the display's refresh rate, and change video and audio
 # speed to some degree to ensure synchronous playback - can cause problems
@@ -66,14 +66,14 @@
 # video outputs, but should work well with default settings on most systems.
 # If performance or energy usage is an issue, forcing the vdpau or vaapi VOs
 # may or may not help.
-#hwdec=auto
+hwdec=auto
 
 ##################
 # audio settings #
 ##################
 
 # Specify default audio driver (see --ao=help for a list).
-#ao=alsa
+ao=pulse,alsa
 
 # Disable softvol usage, and always use the system mixer if available.
 #softvol=no
