From 01169ffe71ddf88345b897d576a65d724dd3dd5c Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sat, 10 Aug 2024 22:12:38 +0800
Subject: [PATCH] fix(Makefile): do not strip during build-time

---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6fd4442..b4ffc64 100644
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,6 @@ INSTALLDIR=/usr/local/bin
 
 unarj: unarj.o decode.o environ.o
 	$(CC) $(CFLAGS) -o unarj unarj.o decode.o environ.o
-	strip unarj
 
 clean:
 	rm -f core unarj *.o *.out *~
-- 
2.46.0

