diff -up ./include/wayland-client.hpp.orig ./include/wayland-client.hpp
--- ./include/wayland-client.hpp.orig	2022-04-25 18:04:33.000000000 +0200
+++ ./include/wayland-client.hpp	2023-03-26 11:58:50.003952654 +0200
@@ -29,6 +29,7 @@
 /** \file */
 
 #include <atomic>
+#include <cstdint>
 #include <functional>
 #include <memory>
 #include <string>
diff -up ./scanner/scanner.cpp.orig ./scanner/scanner.cpp
--- ./scanner/scanner.cpp.orig	2022-04-25 18:04:33.000000000 +0200
+++ ./scanner/scanner.cpp	2023-03-26 11:49:29.853386221 +0200
@@ -23,6 +23,7 @@
 #include <vector>
 #include <cctype>
 #include <cmath>
+#include <cstdint>
 #include <stdexcept>
 
 #include "pugixml.hpp"
@@ -1106,6 +1107,7 @@ int main(int argc, char *argv[])
   wayland_hpp << "#pragma once" << std::endl
               << std::endl
               << "#include <array>" << std::endl
+              << "#include <cstdint>" << std::endl
               << "#include <functional>" << std::endl
               << "#include <memory>" << std::endl
               << "#include <string>" << std::endl
@@ -1125,6 +1127,7 @@ int main(int argc, char *argv[])
     wayland_server_hpp << "#pragma once" << std::endl
                        << std::endl
                        << "#include <array>" << std::endl
+                       << "#include <cstdint>" << std::endl
                        << "#include <functional>" << std::endl
                        << "#include <memory>" << std::endl
                        << "#include <string>" << std::endl
