diff -Naur gc-7.6.0-old/include/private/gcconfig.h gc-7.6.0/include/private/gcconfig.h
--- gc-7.6.0-old/include/private/gcconfig.h	2016-08-03 03:36:14.000000000 +0800
+++ gc-7.6.0/include/private/gcconfig.h	2018-02-19 02:53:57.308291164 +0800
@@ -196,6 +196,10 @@
 #    define OR1K        /* OpenRISC/or1k */
 #    define mach_type_known
 # endif
+# if defined(__riscv)
+#    define RISCV
+#    define mach_type_known
+# endif
 # if defined(DGUX) && (defined(i386) || defined(__i386__))
 #    define I386
 #    ifndef _USING_DGUX
@@ -1787,6 +1791,24 @@
 #   endif /* Linux */
 # endif
 
+# ifdef RISCV
+#   define CPP_WORDSZ 64
+#   define MACH_TYPE "RISCV"
+#   ifdef LINUX
+#     define OS_TYPE "LINUX"
+#     define DYNAMIC_LOADING
+      extern int _end[];
+      extern int __data_start[];
+#     define DATASTART ((ptr_t)(__data_start))
+#     define DATAEND ((ptr_t)(_end))
+#     define ALIGNMENT 8
+#     ifndef HBLKSIZE
+#       define HBLKSIZE 4096
+#     endif
+#     define LINUX_STACKBOTTOM
+#   endif /* Linux */
+# endif
+
 # ifdef HP_PA
 #   define MACH_TYPE "HP_PA"
 #   ifdef __LP64__
