From ffbf600ad078297e8516890393428e5e703473d7 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Mon, 13 Nov 2023 23:51:15 -0500
Subject: [PATCH 03/13] update-m1n1: catch mount_sys_esp when efi is not
 available

---
 update-m1n1 | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/update-m1n1 b/update-m1n1
index 2bc2984..89a1b72 100755
--- a/update-m1n1
+++ b/update-m1n1
@@ -53,7 +53,9 @@ if [ -e "$CONFIG" ]; then
 fi
 
 if [ -z "$TARGET" ]; then
-    mount_sys_esp /efi
+    if ! mount_sys_esp /efi; then
+        return 0;
+    fi
     TARGET="/efi/m1n1/boot.bin"
 fi
 
@@ -72,3 +74,5 @@ fi
 mv -f "${TARGET}.new" "$TARGET"
 
 echo "m1n1 updated at ${TARGET}"
+
+# :vim set sts=4 ts=4 sw=4 expandtab:
-- 
2.51.0

