diff -Naur a/data/gdm.service.in b/data/gdm.service.in
--- a/data/gdm.service.in	2015-11-04 18:49:51.000000000 -0600
+++ b/data/gdm.service.in	2016-09-07 22:47:32.945288346 -0500
@@ -13,7 +13,7 @@
 # pulled from getty@.service and @PLYMOUTH_QUIT_SERVICE@
 # (except for plymouth-quit-wait.service since it waits until
 # plymouth is quit, which we do)
-After=rc-local.service plymouth-start.service systemd-user-sessions.service
+After=rc-local.service plymouth-start.service systemd-user-sessions.service multi-user.target
 
 # GDM takes responsibility for stopping plymouth, so if it fails
 # for any reason, make sure plymouth still stops
