Description: Use recent (>=3.3.02) ispell hash format
Author: Agustin Martin Domingo <agmartin@debian.org>
Forwarded: no
Bug-Debian: http://bugs.debian.org/640788

Index: enchant-1.6.0/src/ispell/ispell.h
===================================================================
--- enchant-1.6.0.orig/src/ispell/ispell.h	2011-09-07 18:32:36.212371501 +0200
+++ enchant-1.6.0/src/ispell/ispell.h	2011-09-07 18:33:55.152091750 +0200
@@ -297,7 +297,7 @@
 ** character counts as two!
 */
 #ifndef MAXSTRINGCHARS
-#define MAXSTRINGCHARS 128
+#define MAXSTRINGCHARS 512
 #endif /* MAXSTRINGCHARS */
 
 /*
