--- kdeartwork-15.04.3/CMakeLists.txt	2013-10-19 00:04:24.000000000 +0800
+++ kdeartwork-15.04.3-changed/CMakeLists.txt	2015-07-25 23:39:36.306880864 +0800
@@ -17,10 +17,6 @@
 macro_optional_add_subdirectory(ColorSchemes)
 macro_optional_add_subdirectory(IconThemes)
 macro_optional_add_subdirectory(emoticons)
-macro_optional_add_subdirectory(kscreensaver)
-if (Q_WS_X11)
-   macro_optional_add_subdirectory(kwin-styles)
-endif (Q_WS_X11)
 macro_optional_add_subdirectory(styles)
 macro_optional_add_subdirectory(wallpapers)
 macro_optional_add_subdirectory(HighResolutionWallpapers)
