diff -Naur rlvm-release-0.14/src/systems/base/graphics_object.cc rlvm-release-0.14.gcc6/src/systems/base/graphics_object.cc
--- rlvm-release-0.14/src/systems/base/graphics_object.cc	2014-10-04 12:12:24.000000000 -0500
+++ rlvm-release-0.14.gcc6/src/systems/base/graphics_object.cc	2017-01-15 09:49:36.634927056 -0600
@@ -40,6 +40,7 @@
 #include <sstream>
 #include <string>
 #include <vector>
+#include <iostream>
 
 #include "systems/base/graphics_object_data.h"
 #include "systems/base/object_mutator.h"
