--- a/SConstruct
+++ b/SConstruct
@@ -105,6 +105,11 @@
   env['RANLIBCOMSTR'] = 'Indexing $TARGET ...'
   env['SHLINKCOMSTR'] = 'Linking $TARGET ...'
 
+if ARGUMENTS.get('CFLAGS') != None:
+  env['CPPFLAGS'] += ARGUMENTS.get('CFLAGS').split()
+if ARGUMENTS.get('LDFLAGS') != None:
+  env['LINKFLAGS'] = ARGUMENTS.get('LDFLAGS').split()
+
 #########################################################################
 ## Platform Specific Locations
 #########################################################################
