--- a/src/VBox/Runtime/common/ldr/ldrkStuff.cpp	2014-07-20 17:50:52.918200154 +0200
+++ b/src/VBox/Runtime/common/ldr/ldrkStuff.cpp	2014-07-20 22:59:50.360692743 +0200
@@ -368,7 +368,7 @@
  * We provide our own based on IPRT instead of using the kLdr ones.
  */
 extern "C" const KRDROPS g_kLdrRdrFileOps;
-extern "C" const KRDROPS g_kLdrRdrFileOps =
+const KRDROPS g_kLdrRdrFileOps __attribute__((section(".rodata"))) =
 {
     /* .pszName = */        "IPRT",
     /* .pNext = */          NULL,

