--- VirtualBox-5.0.0/configure	2015-07-09 09:50:42.000000000 +0000
+++ VirtualBox-5.0.0-changed/configure	2015-07-24 10:06:14.329671511 +0000
@@ -102,12 +102,12 @@
 BUILD_LIBSSL=
 BUILD_LIBVPX=
 PASSIVE_MESA=0
-CC="gcc"
-CC32=""
-CC64=""
-CXX="g++"
-CXX32=""
-CXX64=""
+CC="gcc-multilib-wrapper"
+CC32="i686-pc-linux-gnu-gcc"
+CC64="gcc"
+CXX="g++-multilib-wrapper"
+CXX32="i686-pc-linux-gnu-g++"
+CXX64="g++"
 YASM="yasm"
 IASL="iasl"
 XSLTPROC="xsltproc"
