--- vboxhost-5.2.8_OSE.bak/vboxdrv/Makefile	2018-03-22 20:52:00.399177106 +0100
+++ vboxhost-5.2.8_OSE/vboxdrv/Makefile	2018-03-22 20:51:14.143176300 +0100
@@ -132,7 +132,7 @@
  MOD_OBJS += SUPDrvDTrace.o
 endif
 
-MOD_INCL  = $(addprefix -I$(KBUILD_EXTMOD)/vboxdrv,/ /include /r0drv/linux)
+MOD_INCL  = $(addprefix -I$(KBUILD_EXTMOD),/ /include /r0drv/linux)
 ifdef VBOX_WITH_NATIVE_DTRACE
  MOD_INCL += -I/usr/include/linux -I/usr/include
 endif
@@ -158,7 +158,7 @@
 endif
 
 # build defs
-MOD_CFLAGS = -include $(KBUILD_EXTMOD)/vboxdrv/include/VBox/SUPDrvMangling.h \
+MOD_CFLAGS = -include $(KBUILD_EXTMOD)/include/VBox/SUPDrvMangling.h \
 	-fno-omit-frame-pointer -fno-pie
 
 include $(obj)/Makefile.include.footer
--- vboxhost-5.2.8_OSE.bak/vboxnetflt/Makefile	2018-03-22 20:52:00.408177106 +0100
+++ vboxhost-5.2.8_OSE/vboxnetflt/Makefile	2018-03-22 20:53:01.180178164 +0100
@@ -46,7 +46,7 @@
 	math/gcc/umoddi3.o
 endif
 
-MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD)/vboxnetflt,/ /include /r0drv/linux)
+MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD),/ /include /r0drv/linux)
 MOD_DEFS = -DRT_OS_LINUX -DIN_RING0 -DIN_RT_R0 \
 	-DIN_SUP_R0 -DVBOX -DRT_WITH_VBOX -DVBOX_WITH_HARDENING \
 	-Wno-declaration-after-statement
@@ -71,6 +71,6 @@
 endif
 
 # build defs
-MOD_CFLAGS = -include $(KBUILD_EXTMOD)/vboxnetflt/include/VBox/SUPDrvMangling.h -fno-pie
+MOD_CFLAGS = -include $(KBUILD_EXTMOD)/include/VBox/SUPDrvMangling.h -fno-pie
 
 include $(obj)/Makefile.include.footer
--- vboxhost-5.2.8_OSE.bak/vboxnetadp/Makefile	2018-03-22 20:52:00.405177106 +0100
+++ vboxhost-5.2.8_OSE/vboxnetadp/Makefile	2018-03-22 20:53:53.839179082 +0100
@@ -42,7 +42,7 @@
 	math/gcc/umoddi3.o
 endif
 
-MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD)/vboxnetadp,/ /include /r0drv/linux)
+MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD),/ /include /r0drv/linux)
 MOD_DEFS = -DRT_OS_LINUX -DIN_RING0 -DIN_RT_R0 -DIN_SUP_R0 -DVBOX \
 	    -DRT_WITH_VBOX -DVBOX_WITH_HARDENING \
 	    -Wno-declaration-after-statement
@@ -67,6 +67,6 @@
 endif
 
 # build defs
-MOD_CFLAGS = -include $(KBUILD_EXTMOD)/vboxnetadp/include/VBox/SUPDrvMangling.h -fno-pie
+MOD_CFLAGS = -include $(KBUILD_EXTMOD)/include/VBox/SUPDrvMangling.h -fno-pie
 
 include $(obj)/Makefile.include.footer
--- vboxhost-5.2.8_OSE.bak/vboxpci/Makefile	2018-03-22 20:52:00.398177106 +0100
+++ vboxhost-5.2.8_OSE/vboxpci/Makefile	2018-03-22 20:54:30.291179716 +0100
@@ -46,7 +46,7 @@
 	math/gcc/umoddi3.o
 endif
 
-MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD)/vboxpci,/ /include /r0drv/linux)
+MOD_INCL = $(addprefix -I$(KBUILD_EXTMOD),/ /include /r0drv/linux)
 MOD_DEFS = -DRT_OS_LINUX -DIN_RING0 -DIN_RT_R0 -DIN_SUP_R0 -DVBOX \
 	-DRT_WITH_VBOX -DVBOX_WITH_HARDENING
 ifeq ($(BUILD_TARGET_ARCH),amd64)
@@ -68,6 +68,6 @@
 endif
 
 # build defs
-MOD_CFLAGS = -include $(KBUILD_EXTMOD)/vboxpci/include/VBox/SUPDrvMangling.h -fno-pie
+MOD_CFLAGS = -include $(KBUILD_EXTMOD)/include/VBox/SUPDrvMangling.h -fno-pie
 
 include $(obj)/Makefile.include.footer
