From a37e92d1ba996c3d4748a04a691c6bddb8b5e129 Mon Sep 17 00:00:00 2001
From: Oliver Giles <ohw.giles@gmail.com>
Date: Wed, 30 May 2018 08:11:04 +0300
Subject: [PATCH] Check count of MAPIProperty array before allocating

Resolves CVE-2017-9473
---
 lib/ytnef.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/ytnef.c b/lib/ytnef.c
index 5eb4204..f48c4bb 100644
--- a/lib/ytnef.c
+++ b/lib/ytnef.c
@@ -421,6 +421,11 @@ int TNEFFillMapi(TNEFStruct *TNEF, BYTE *data, DWORD size, MAPIProps *p) {
   d = data;
   p->count = SwapDWord((BYTE*)data, 4);
   d += 4;
+  // Arbitrary limit on the amount of properties
+  if (((size_t) p->count) > 1000) {
+     printf("ERROR: suspecting a corrupt file in MAPI allocation\n");
+     return -1;
+  }
   p->properties = calloc(p->count, sizeof(MAPIProperty));
   ALLOCCHECK(p->properties);
   mp = p->properties;
