From 9517e81372aca815c6a463f742534e926b45ee09 Mon Sep 17 00:00:00 2001
From: Cyan <cyanoxygen@aosc.io>
Date: Thu, 29 Feb 2024 20:48:38 +0800
Subject: [PATCH] Makefile: disable strip

To allow autobuild4 to split debug symbols.
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 46f11a4..efb3eb9 100644
--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@ ifdef DEBUG
 	endif
 else
 	CFLAGS = -pipe -std=c99 -Wall -Wextra -Wvla -O3 $(LTOFLAGS) -fno-pie -fno-PIE -DNDEBUG
-	LDFLAGS = -pipe -no-pie -O3 $(LTOFLAGS) -s
+	LDFLAGS = -pipe -no-pie -O3 $(LTOFLAGS)
 endif
 
 ifdef STATIC
-- 
2.43.0

