diff -Naur ispc-1.13.0/examples/cmake/AddISPCExample.cmake ispc-1.13.0.modded/examples/cmake/AddISPCExample.cmake
--- ispc-1.13.0/examples/cmake/AddISPCExample.cmake	2020-04-23 18:29:06.000000000 -0500
+++ ispc-1.13.0.modded/examples/cmake/AddISPCExample.cmake	2020-09-02 00:43:54.120504658 -0500
@@ -155,15 +155,15 @@
     # Install example
     # We do not need to include examples binaries to the package
     if (NOT ISPC_PREPARE_PACKAGE)
-        install(TARGETS ${example_NAME} RUNTIME DESTINATION examples/${example_NAME})
+        install(TARGETS ${example_NAME} RUNTIME DESTINATION share/ispc/examples/${example_NAME})
         if (example_DATA_FILES)
             install(FILES ${example_DATA_FILES}
-                    DESTINATION examples/${example_NAME})
+                    DESTINATION share/ispc/examples/${example_NAME})
         endif()
 
         if (example_DATA_DIR)
             install(DIRECTORY ${example_DATA_DIR}
-                    DESTINATION examples/${example_NAME})
+                    DESTINATION share/ispc/examples/${example_NAME})
         endif()
     endif()
 
