From bb9035efdec6f6f468615269be32ed229026a0cc Mon Sep 17 00:00:00 2001
From: Henry Chen <>
Date: Sun, 2 Aug 2020 02:13:14 -0700
Subject: [PATCH] variables.sh: omit errors from commands not found

---
 variables.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/variables.sh b/variables.sh
index 25b451b..024d4b3 100644
--- a/variables.sh
+++ b/variables.sh
@@ -38,9 +38,9 @@ export default_config="default"
 export SHELLRC="$HOME/.bashrc"
 
 case "$SHELL" in
-    "$(which bash)") SHELLRC="$HOME/.bashrc"
+    "$(which bash 2> /dev/null)") SHELLRC="$HOME/.bashrc"
                      ;;
-    "$(which zsh)") SHELLRC="$HOME/.zshrc"
+    "$(which zsh 2> /dev/null)") SHELLRC="$HOME/.zshrc"
                     ;;
 esac
 
-- 
2.27.0

