diff -Naur mediastreamer2-2.16.1/src/base/msfactory.c mediastreamer2-2.16.1.workaround/src/base/msfactory.c
--- mediastreamer2-2.16.1/src/base/msfactory.c	2017-07-21 08:00:47.000000000 -0500
+++ mediastreamer2-2.16.1.workaround/src/base/msfactory.c	2017-09-13 08:23:23.308649372 -0500
@@ -17,17 +17,8 @@
 Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
 
-#ifdef HAVE_CONFIG_H
 #include "mediastreamer-config.h"
 #include "gitversion.h"
-#else
-#   ifndef MEDIASTREAMER_VERSION
-#   define MEDIASTREAMER_VERSION "unknown"
-#   endif
-#	ifndef MS2_GIT_VERSION
-#	define MS2_GIT_VERSION "unknown"
-#	endif
-#endif
 
 #include "mediastreamer2/msfilter.h"
 #include "mediastreamer2/mseventqueue.h"
@@ -174,7 +165,7 @@
 		ortp_set_log_level_mask(ORTP_LOG_DOMAIN, ORTP_MESSAGE|ORTP_WARNING|ORTP_ERROR|ORTP_FATAL);
 	}
 
-	ms_message("Mediastreamer2 factory " MEDIASTREAMER_VERSION " (git: " MS2_GIT_VERSION ") initialized.");
+	ms_message("Mediastreamer2 factory " MEDIASTREAMER_VERSION " initialized.");
 	/* register builtin MSFilter's */
 	for (i=0;ms_base_filter_descs[i]!=NULL;i++){
 		ms_factory_register_filter(obj,ms_base_filter_descs[i]);
