From c50f49e3164c0d5507dad079ece8642369b47e64 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Fri, 6 Mar 2020 11:45:32 +0000
Subject: [PATCH] install arch config file use /etc/pacman.conf.pacstrap

---
 pacstrap.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/pacstrap.in b/pacstrap.in
index 703df11..204ee20 100644
--- a/pacstrap.in
+++ b/pacstrap.in
@@ -44,6 +44,10 @@ fi

 (( EUID == 0 )) || die 'This script must be run with root privileges'

+# We ship `pacstrap' with an extra configuration file that complies with Arch Linux,
+# as the standard package is meant for *any* Pacman distributions/repositories.
+pacman_config="/etc/pacman.conf.pacstrap"
+
 while getopts ':C:cdGiMU' flag; do
   case $flag in
     C)
--
2.24.1
