From 4817abcc72a4fab605017dd71ce07220ffcdfd02 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Fri, 6 Mar 2020 11:52:32 +0000
Subject: [PATCH] copy archlinux mirror /etc/pacman.d/mirrorlist.pacstrap to
 new archlinux

---
 pacstrap.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pacstrap.in b/pacstrap.in
index 204ee20..2eb537b 100644
--- a/pacstrap.in
+++ b/pacstrap.in
@@ -122,7 +122,7 @@ fi

 if (( copymirrorlist )); then
   # install the host's mirrorlist onto the new root
-  cp -a /etc/pacman.d/mirrorlist "$newroot/etc/pacman.d/"
+  cp -a /etc/pacman.d/mirrorlist.pacstrap "$newroot/etc/pacman.d/mirrorlist"
 fi

 # vim: et ts=2 sw=2 ft=sh:
--
2.24.1

