diff -Naur xorg-server-1.20.6/hw/xfree86/Makefile.am xorg-server-1.20.6.vbe/hw/xfree86/Makefile.am
--- xorg-server-1.20.6/hw/xfree86/Makefile.am	2020-04-18 20:57:13.389636570 -0500
+++ xorg-server-1.20.6.vbe/hw/xfree86/Makefile.am	2020-04-18 20:58:11.372261229 -0500
@@ -32,17 +32,9 @@
 VGAHW_SUBDIR = vgahw
 endif
 
-if VBE
-VBE_SUBDIR = vbe
-endif
-
-if INT10MODULE
-INT10_SUBDIR = int10
-endif
-
-SUBDIRS = common ddc x86emu $(INT10_SUBDIR) os-support parser \
+SUBDIRS = common ddc x86emu int10 os-support parser \
 	  ramdac $(VGAHW_SUBDIR) loader modes $(DRI_SUBDIR) \
-	  $(DRI2_SUBDIR) . $(VBE_SUBDIR) $(XAA_SUBDIR) i2c dixmods xkb \
+	  $(DRI2_SUBDIR) . vbe $(XAA_SUBDIR) i2c dixmods xkb \
 	  fbdevhw shadowfb exa $(XF86UTILS_SUBDIR) doc man \
 	  $(GLAMOR_EGL_SUBDIR) drivers
 
