diff -Naur id3lib-3.8.3.old/doc/Doxyfile.in id3lib-3.8.3/doc/Doxyfile.in
--- id3lib-3.8.3.old/doc/Doxyfile.in	2019-09-26 21:18:57.184798907 +0200
+++ id3lib-3.8.3/doc/Doxyfile.in	2019-09-26 21:19:24.557297128 +0200
@@ -263,7 +263,7 @@
 
 INPUT                  = ../src \
                          ../include/id3/field.h \
-                         ../include/id3/frame.h \
+                         ../include/id3/id3lib_frame.h \
                          ../include/id3/globals.h \
                          ../include/id3/misc_support.h \
                          ../include/id3/reader.h \
