Description: Use GCC Visibility
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: no

--- charls-1.1.0+dfsg.orig/src/interface.cpp
+++ charls-1.1.0+dfsg/src/interface.cpp
@@ -7,7 +7,7 @@
 #if defined(WIN32) && defined(CHARLS_DLL)
 #define CHARLS_IMEXPORT(returntype) __declspec(dllexport) returntype __stdcall
 #else
-#define CHARLS_IMEXPORT(returntype) returntype
+#define CHARLS_IMEXPORT(returntype) __attribute__ ((visibility ("default"))) returntype
 #endif
 
 #include "config.h"
--- charls-1.1.0+dfsg.orig/src/interface.h
+++ charls-1.1.0+dfsg/src/interface.h
@@ -20,7 +20,7 @@
 
 /* windows dll */
 #if !defined(CHARLS_IMEXPORT) && defined(CHARLS_DLL)
-#  define CHARLS_IMEXPORT(returntype) __declspec(dllimport) returntype __stdcall
+#define CHARLS_IMEXPORT(returntype) __attribute__ ((visibility ("default"))) returntype
 #endif
 
 #if !defined(CHARLS_IMEXPORT)
