diff -Naur tdesktop-4.9.7-full/Telegram/SourceFiles/window/window.style tdesktop-4.9.7-full.360px/Telegram/SourceFiles/window/window.style
--- tdesktop-4.9.7-full/Telegram/SourceFiles/window/window.style	2023-09-13 10:16:03.000000000 -0700
+++ tdesktop-4.9.7-full.360px/Telegram/SourceFiles/window/window.style	2023-09-14 11:24:48.935409202 -0700
@@ -10,7 +10,7 @@
 using "chat_helpers/chat_helpers.style";
 using "boxes/boxes.style"; // UserpicButton
 
-windowMinWidth: 380px;
+windowMinWidth: 360px;
 windowMinHeight: 480px;
 windowDefaultWidth: 800px;
 windowDefaultHeight: 600px;
@@ -19,7 +19,7 @@
 
 columnMinimalWidthLeft: 260px;
 columnMaximalWidthLeft: 540px;
-columnMinimalWidthMain: 380px;
+columnMinimalWidthMain: 360px;
 columnMinimalWidthThird: 292px;
 columnMaximalWidthThird: 392px;
 
