diff -Naur tdesktop/Telegram/lib_ui/ui/style/style_core_custom_font.cpp tdesktop.customfonts/Telegram/lib_ui/ui/style/style_core_custom_font.cpp
--- tdesktop/Telegram/lib_ui/ui/style/style_core_custom_font.cpp	2023-09-23 10:51:31.452034690 -0700
+++ tdesktop.customfonts/Telegram/lib_ui/ui/style/style_core_custom_font.cpp	2023-09-23 10:56:59.741015400 -0700
@@ -30,45 +30,23 @@
 	static auto Database = QFontDatabase();
 
 	const auto bold = ((flags & FontBold) || (flags & FontSemibold));
-	const auto italic = (flags & FontItalic);
-	const auto &custom = bold ? BoldFont : RegularFont;
-	const auto useCustom = !custom.family.isEmpty();
+	const QString FontFamilyName = qEnvironmentVariable("TELEGRAM_DESKTOP_UI_FONT");
+	const QString FontFamilyNameMono = qEnvironmentVariable("TELEGRAM_DESKTOP_UI_FONT_MONO");
+	const QString FontFamilyBoldStyle = qEnvironmentVariable("TELEGRAM_DESKTOP_UI_FONT_BOLD_STYLE");
 
-	auto result = QFont(QGuiApplication::font().family());
-	if (!familyOverride.isEmpty()) {
-		result.setFamily(familyOverride);
-		if (bold) {
-			result.setBold(true);
-		}
-	} else if (flags & FontMonospace) {
-		result.setFamily(MonospaceFont());
-	} else if (useCustom) {
-		const auto sizes = Database.smoothSizes(custom.family, custom.style);
-		const auto good = sizes.isEmpty()
-			? Database.pointSizes(custom.family, custom.style)
-			: sizes;
-		const auto point = good.isEmpty() ? size : good.front();
-		result = Database.font(custom.family, custom.style, point);
-	} else {
-		result.setFamily(GetFontOverride(flags));
-		if (bold) {
-#ifdef LIB_UI_USE_PACKAGED_FONTS
-			result.setWeight(QFont::DemiBold);
-#else // LIB_UI_USE_PACKAGED_FONTS
-			result.setBold(true);
-#endif // !LIB_UI_USE_PACKAGED_FONTS
-
-			if (flags & FontItalic) {
-				result.setStyleName("Semibold Italic");
-			} else {
-				result.setStyleName("Semibold");
-			}
-		}
+	auto result = QFont(FontFamilyName.isEmpty() ? QGuiApplication::font().family() : FontFamilyName);
+	if (flags & FontMonospace) {
+		result.setFamily(FontFamilyNameMono.isEmpty() ? MonospaceFont() : FontFamilyNameMono);
 	}
-	if (italic) {
-		result.setItalic(true);
+	if (bold) {
+		if (FontFamilyBoldStyle.isEmpty()){
+			result.setWeight(QFont::Medium);
+		}else{
+			result.setStyleName(FontFamilyBoldStyle);
+		}
 	}
 
+	result.setItalic(flags & FontItalic);
 	result.setUnderline(flags & FontUnderline);
 	result.setStrikeOut(flags & FontStrikeOut);
 	result.setPixelSize(size);
