From 775accdb5f21db7991269abdfe88e9da8b49689e Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Mon, 6 May 2024 00:06:32 -0700
Subject: [PATCH 5/6] fix(settings): use system fonts by default

---
 Telegram/SourceFiles/core/core_settings.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Telegram/SourceFiles/core/core_settings.h b/Telegram/SourceFiles/core/core_settings.h
index 6c1ba990..f38ab653 100644
--- a/Telegram/SourceFiles/core/core_settings.h
+++ b/Telegram/SourceFiles/core/core_settings.h
@@ -11,6 +11,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "media/media_common.h"
 #include "window/themes/window_themes_embedded.h"
 #include "ui/chat/attach/attach_send_files_way.h"
+#include "ui/style/style_core_font.h"
 #include "base/flags.h"
 #include "emoji.h"
 
@@ -1039,7 +1040,7 @@ private:
 	rpl::variable<bool> _storiesClickTooltipHidden = false;
 	rpl::variable<bool> _ttlVoiceClickTooltipHidden = false;
 	WindowPosition _ivPosition;
-	QString _customFontFamily;
+	QString _customFontFamily = style::SystemFontTag();
 	bool _systemUnlockEnabled = false;
 	std::optional<bool> _weatherInCelsius;
 	QByteArray _tonsiteStorageToken;
-- 
2.46.0

