From 7126598f074e8123c462ef68665b3a71edc46a3a Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Mon, 6 May 2024 00:06:32 -0700
Subject: [PATCH 05/15] fix(settings): use system fonts by default

---
 Telegram/SourceFiles/core/core_settings.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Telegram/SourceFiles/core/core_settings.h b/Telegram/SourceFiles/core/core_settings.h
index 2e59e02c6f..1331b60ceb 100644
--- a/Telegram/SourceFiles/core/core_settings.h
+++ b/Telegram/SourceFiles/core/core_settings.h
@@ -12,6 +12,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "dialogs/ui/dialogs_quick_action.h"
 #include "window/themes/window_themes_embedded.h"
 #include "ui/chat/attach/attach_send_files_way.h"
+#include "ui/style/style_core_font.h"
 #include "base/flags.h"
 #include "emoji.h"
 
@@ -1085,7 +1086,7 @@ private:
 	rpl::variable<bool> _storiesClickTooltipHidden = false;
 	rpl::variable<bool> _ttlVoiceClickTooltipHidden = false;
 	WindowPosition _ivPosition;
-	QString _customFontFamily;
+	QString _customFontFamily = style::SystemFontTag();
 	bool _systemUnlockEnabled = false;
 	std::optional<bool> _weatherInCelsius;
 	QByteArray _tonsiteStorageToken;
-- 
2.52.0

