From 1c0ed089e514fecf325e0b3b1e89b0c114c00c07 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Fri, 1 Nov 2024 23:28:52 -0700
Subject: [PATCH 08/15] fix(lib_webview): add missing cstdint include

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 Telegram/lib_webview/webview/webview_interface.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Telegram/lib_webview/webview/webview_interface.h b/Telegram/lib_webview/webview/webview_interface.h
index 7477811ad4..a175bd69aa 100644
--- a/Telegram/lib_webview/webview/webview_interface.h
+++ b/Telegram/lib_webview/webview/webview_interface.h
@@ -12,6 +12,7 @@
 #include <string>
 #include <optional>
 #include <functional>
+#include <cstdint>
 
 #include <rpl/never.h>
 #include <rpl/producer.h>
-- 
2.52.0

