From 25e92738b6a59ad59122129323730d14930e88aa Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Fri, 1 Nov 2024 23:28:52 -0700
Subject: [PATCH 08/10] fix(lib_webview): add missing cstdint include

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 Telegram/lib_webview/webview/webview_interface.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Telegram/lib_webview/webview/webview_interface.h b/Telegram/lib_webview/webview/webview_interface.h
index 4a92fe3020..3cf6125358 100644
--- a/Telegram/lib_webview/webview/webview_interface.h
+++ b/Telegram/lib_webview/webview/webview_interface.h
@@ -12,6 +12,7 @@
 #include <string>
 #include <optional>
 #include <functional>
+#include <cstdint>
 
 #include <rpl/never.h>
 #include <rpl/producer.h>
-- 
2.48.1

