From 536c7a090af56fd77288176eff437405d12d3712 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Sat, 10 May 2025 16:24:38 -0700
Subject: [PATCH 11/12] fix(cmake): disable CMP0160 to workaround KF5 cmake
 errors

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc7b239b03..d9f278a357 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,12 @@
 
 cmake_minimum_required(VERSION 3.25...3.31)
 
+# Workaround for KDE Frameworks 5 components cmake errors
+# `IMPORTED property is read-only for target("KF5CoreAddons_QCH")`
+if (POLICY CMP0160)
+    cmake_policy(SET CMP0160 OLD)
+endif()
+
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
 include(cmake/validate_special_target.cmake)
-- 
2.49.0

