From f81c30c19ef812aa4ecd900177aab8a616c2eef8 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <origincode@aosc.io>
Date: Thu, 3 Jul 2025 19:04:36 -0700
Subject: [PATCH 12/15] fix(credits_amount.h): include cmath for std::floor

Signed-off-by: Kaiyang Wu <origincode@aosc.io>
---
 Telegram/SourceFiles/core/credits_amount.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Telegram/SourceFiles/core/credits_amount.h b/Telegram/SourceFiles/core/credits_amount.h
index 098266d887..a5d329c417 100644
--- a/Telegram/SourceFiles/core/credits_amount.h
+++ b/Telegram/SourceFiles/core/credits_amount.h
@@ -7,6 +7,8 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 */
 #pragma once
 
+#include <cmath>
+
 #include "base/algorithm.h"
 #include "base/basic_types.h"
 
-- 
2.52.0

