From 22649337d820bee2c1e53e3b89947dc0be3d8a71 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Sat, 10 May 2025 21:38:50 -0700
Subject: [PATCH 12/13] fix(cmake): use correct namespace for tde2d

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 cmake/external/td/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/external/td/CMakeLists.txt b/cmake/external/td/CMakeLists.txt
index 448d10b4b6..72ae501808 100644
--- a/cmake/external/td/CMakeLists.txt
+++ b/cmake/external/td/CMakeLists.txt
@@ -8,8 +8,8 @@ add_library(external_td INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_td ALIAS external_td)
 
 if (DESKTOP_APP_USE_PACKAGED)
-    find_package(Td REQUIRED)
-    target_link_libraries(external_td INTERFACE Td::tde2e)
+    find_package(tde2e REQUIRED)
+    target_link_libraries(external_td INTERFACE tde2e::tde2e)
     return()
 endif()
 
-- 
2.49.0

