diff -Naur tdesktop/Telegram/lib_webview/CMakeLists.txt tdesktop.qt5/Telegram/lib_webview/CMakeLists.txt
--- tdesktop/Telegram/lib_webview/CMakeLists.txt	2023-09-23 10:39:13.974421475 -0700
+++ tdesktop.qt5/Telegram/lib_webview/CMakeLists.txt	2023-09-23 10:49:22.309717624 -0700
@@ -7,6 +7,7 @@
 add_library(lib_webview STATIC)
 add_library(desktop-app::lib_webview ALIAS lib_webview)
 init_target(lib_webview)
+set_target_properties(lib_webview PROPERTIES AUTOMOC TRUE)
 
 get_filename_component(src_loc . REALPATH)
 
diff -Naur tdesktop/Telegram/lib_webview/webview/platform/linux/webview_linux_compositor.cpp tdesktop.qt5/Telegram/lib_webview/webview/platform/linux/webview_linux_compositor.cpp
--- tdesktop/Telegram/lib_webview/webview/platform/linux/webview_linux_compositor.cpp	2023-09-23 10:39:13.975421509 -0700
+++ tdesktop.qt5/Telegram/lib_webview/webview/platform/linux/webview_linux_compositor.cpp	2023-09-23 10:50:13.671434361 -0700
@@ -39,6 +39,7 @@
 };
 
 class Output : public QWaylandQuickOutput {
+	Q_OBJECT
 public:
 	Output(
 			QWaylandCompositor *compositor,
@@ -231,3 +232,5 @@
 }
 
 } // namespace Webview
+
+#include "webview_linux_compositor.moc"
diff -Naur tdesktop/Telegram/lib_webview/webview/platform/linux/webview_linux_webkitgtk.cpp tdesktop.qt5/Telegram/lib_webview/webview/platform/linux/webview_linux_webkitgtk.cpp
--- tdesktop/Telegram/lib_webview/webview/platform/linux/webview_linux_webkitgtk.cpp	2023-09-23 10:39:13.975421509 -0700
+++ tdesktop.qt5/Telegram/lib_webview/webview/platform/linux/webview_linux_webkitgtk.cpp	2023-09-23 10:50:46.200521783 -0700
@@ -110,6 +110,7 @@
 Instance::Instance(bool remoting)
 : _remoting(remoting) {
 	if (_remoting) {
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
 		if ((_wayland = ProvidesQWidget())) {
 			[[maybe_unused]] static const auto Inited = [] {
 				const auto backend = Ui::GL::ChooseBackendDefault(
@@ -127,7 +128,7 @@
 				return true;
 			}();
 		}
-
+#endif
 		startProcess();
 	}
 }
--- a/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp	2023-09-23 12:37:24.217402847 -0700
+++ b/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp	2023-09-23 12:37:35.245776842 -0700
@@ -13,8 +13,10 @@
 #include <QtGui/QGuiApplication>
 #include <QtGui/QWindow>
 #include <qpa/qplatformintegration.h>
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
 #include <private/qguiapplication_p.h>
 #include <private/qgenericunixservices_p.h>
+#endif
 
 namespace base::Platform::XDP {
 
