Submodule Telegram/lib_base contains modified content
diff --git a/Telegram/lib_base/base/qt/qt_compare.h b/Telegram/lib_base/base/qt/qt_compare.h
index ca03fa2..3160943 100644
--- a/Telegram/lib_base/base/qt/qt_compare.h
+++ b/Telegram/lib_base/base/qt/qt_compare.h
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <compare>
+#include <variant>
 #include <gsl/pointers>
 
 #include <QString>
Submodule Telegram/lib_ui contains modified content
diff --git a/Telegram/lib_ui/ui/text/text_entity.cpp b/Telegram/lib_ui/ui/text/text_entity.cpp
index 383b3d0..f1fff9b 100644
--- a/Telegram/lib_ui/ui/text/text_entity.cpp
+++ b/Telegram/lib_ui/ui/text/text_entity.cpp
@@ -2349,6 +2349,8 @@ EntityInText::EntityInText(
 , _data(data) {
 }
 
+EntityInText::EntityInText() {}
+
 int EntityInText::FirstMonospaceOffset(
 		const EntitiesInText &entities,
 		int textLength) {
diff --git a/Telegram/lib_ui/ui/text/text_entity.h b/Telegram/lib_ui/ui/text/text_entity.h
index 2aa8385..dacff68 100644
--- a/Telegram/lib_ui/ui/text/text_entity.h
+++ b/Telegram/lib_ui/ui/text/text_entity.h
@@ -70,6 +70,8 @@ public:
 		int length,
 		const QString &data = QString());
 
+	EntityInText();
+
 	[[nodiscard]] EntityType type() const {
 		return _type;
 	}
Submodule Telegram/lib_webview contains modified content
diff --git a/Telegram/lib_webview/CMakeLists.txt b/Telegram/lib_webview/CMakeLists.txt
index 18ed682..e54ced0 100644
--- a/Telegram/lib_webview/CMakeLists.txt
+++ b/Telegram/lib_webview/CMakeLists.txt
@@ -7,6 +7,7 @@
 add_library(lib_webview STATIC)
 add_library(desktop-app::lib_webview ALIAS lib_webview)
 init_target(lib_webview)
+set_target_properties(lib_webview PROPERTIES AUTOMOC TRUE)
 
 get_filename_component(src_loc . REALPATH)
 
diff --git a/Telegram/lib_webview/webview/platform/linux/webview_linux_compositor.cpp b/Telegram/lib_webview/webview/platform/linux/webview_linux_compositor.cpp
index 60bed44..68635d2 100644
--- a/Telegram/lib_webview/webview/platform/linux/webview_linux_compositor.cpp
+++ b/Telegram/lib_webview/webview/platform/linux/webview_linux_compositor.cpp
@@ -39,6 +39,7 @@ private:
 };
 
 class Output : public QWaylandQuickOutput {
+	Q_OBJECT
 public:
 	Output(
 			QWaylandCompositor *compositor,
@@ -232,3 +233,5 @@ void Compositor::setWidget(QQuickWidget *widget) {
 }
 
 } // namespace Webview
+
+#include "webview_linux_compositor.moc"
