--- a/build.gradle	2021-04-27 23:09:19.690565730 -0500
+++ b/build.gradle	2021-04-27 23:10:55.604657717 -0500
@@ -16,7 +16,7 @@
                      .startsWith('mac os x') ? 'macosx' :\
                      System.getProperty('os.name').split(' ')[0].toLowerCase()
 ext.targetOSArch   = ["i386":"x86", "i486":"x86", "i586":"x86", "i686":"x86", "x86":"x86",
-                      "amd64":"x86_64", "x86-64":"x86_64", "x86_64":"x86_64"]\
+                      "amd64":"x86_64", "x86-64":"x86_64", "x86_64":"x86_64", "ppc64le": "ppc64le"]\
                      [System.getProperty('os.arch').toLowerCase()]
 ext.targetOS       = "${project.ext.targetOSName}-${project.ext.targetOSArch}"
 println "targetOS=${project.ext.targetOS}"
