From b1c44f27e590df54d921d72af7ae413e3ef37f6a Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Mon, 17 Jun 2024 22:41:46 -0700
Subject: [PATCH] javautils: detect Java paths on AOSC OS

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 launcher/java/JavaUtils.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/launcher/java/JavaUtils.cpp b/launcher/java/JavaUtils.cpp
index b12461f44..9eeabd8bb 100644
--- a/launcher/java/JavaUtils.cpp
+++ b/launcher/java/JavaUtils.cpp
@@ -377,7 +377,13 @@ QList<QString> JavaUtils::FindJavaPaths()
         QDir dir(dirPath);
         if (!dir.exists())
             return;
-        auto entries = dir.entryInfoList(QDir::Dirs | QDir::NoDotAndDotDot);
+        QFileInfoList entries;
+        if (dirPath == "/usr/lib") {
+            QStringList javaDirs = { "java*" };
+            entries = dir.entryInfoList(javaDirs, QDir::Dirs | QDir::NoDotAndDotDot);
+        } else {
+            entries = dir.entryInfoList(QDir::Dirs | QDir::NoDotAndDotDot);
+        }
         for (auto& entry : entries) {
             QString prefix;
             prefix = entry.canonicalFilePath();
@@ -399,6 +405,7 @@ QList<QString> JavaUtils::FindJavaPaths()
     scanJavaDirs("/usr/lib/jvm");
     scanJavaDirs("/usr/lib64/jvm");
     scanJavaDirs("/usr/lib32/jvm");
+    scanJavaDirs("/usr/lib");
     // javas stored in Prism Launcher's folder
     scanJavaDirs("java");
     // manually installed JDKs in /opt
-- 
2.45.2

