From ed82cd3473d06083b0a7c3a3298a5989a991f25e Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Mon, 17 Jun 2024 22:41:46 -0700
Subject: [PATCH 1/2] javautils: detect Java paths on AOSC OS

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 launcher/java/JavaUtils.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/launcher/java/JavaUtils.cpp b/launcher/java/JavaUtils.cpp
index 3b32b54c7..3bd2161b6 100644
--- a/launcher/java/JavaUtils.cpp
+++ b/launcher/java/JavaUtils.cpp
@@ -393,7 +393,13 @@ QList<QString> JavaUtils::FindJavaPaths()
         QDir dir(dirPath);
         if (!dir.exists())
             return;
-        auto entries = dir.entryInfoList(QDir::Dirs | QDir::NoDotAndDotDot);
+        QFileInfoList entries;
+        if (dirPath == "/usr/lib") {
+            QStringList javaDirs = { "java*" };
+            entries = dir.entryInfoList(javaDirs, QDir::Dirs | QDir::NoDotAndDotDot);
+        } else {
+            entries = dir.entryInfoList(QDir::Dirs | QDir::NoDotAndDotDot);
+        }
         for (auto& entry : entries) {
             QString prefix;
             prefix = entry.canonicalFilePath();
@@ -415,6 +421,7 @@ QList<QString> JavaUtils::FindJavaPaths()
     scanJavaDirs("/usr/lib/jvm");
     scanJavaDirs("/usr/lib64/jvm");
     scanJavaDirs("/usr/lib32/jvm");
+    scanJavaDirs("/usr/lib");
     // javas stored in Prism Launcher's folder
     scanJavaDirs("java");
     // manually installed JDKs in /opt
-- 
2.46.0

