From b4b2343e9a656257e59de5b6063c16ce66d44ed5 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Mon, 17 Jun 2024 22:41:46 -0700
Subject: [PATCH] javautils: add a filter for detecting Java paths on AOSC OS

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 launcher/java/JavaUtils.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/launcher/java/JavaUtils.cpp b/launcher/java/JavaUtils.cpp
index bc8026348..507afdbe0 100644
--- a/launcher/java/JavaUtils.cpp
+++ b/launcher/java/JavaUtils.cpp
@@ -440,9 +440,15 @@ QList<QString> JavaUtils::FindJavaPaths()
         QString fileName = info.fileName();
         return fileName.startsWith("openjdk-") || fileName.startsWith("openj9-");
     };
+    // AOSC OS's locations for openjdk
+    auto aoscFilter = [](const QFileInfo& info) {
+        QString fileName = info.fileName();
+        return fileName.startsWith("java");
+    };
     scanJavaDir("/usr/lib64", gentooFilter);
     scanJavaDir("/usr/lib", gentooFilter);
     scanJavaDir("/opt", gentooFilter);
+    scanJavaDir("/usr/lib", aoscFilter);
     // javas stored in Prism Launcher's folder
     scanJavaDirs("java");
     // manually installed JDKs in /opt
-- 
2.47.0

