diff -Naur megasync.orig/src/MEGASync/gui/MegaSpeedGraph.h megasync.modded/src/MEGASync/gui/MegaSpeedGraph.h
--- megasync.orig/src/MEGASync/gui/MegaSpeedGraph.h	2022-01-14 21:50:09.126251853 -0600
+++ megasync.modded/src/MEGASync/gui/MegaSpeedGraph.h	2022-01-14 21:53:15.734308236 -0600
@@ -2,6 +2,7 @@
 #define MEGASPEEDGRAPH_H
 
 #include <QWidget>
+#include <QPainterPath>
 #include "megaapi.h"
 
 namespace Ui {
