--- a/build/moz.configure/toolchain.configure	2021-01-25 01:47:24.497409188 -0800
+++ b/build/moz.configure/toolchain.configure	2021-01-25 01:47:38.225698491 -0800
@@ -1869,7 +1869,7 @@
 # ==============================================================
 
 option(env='RUSTFLAGS',
-       nargs=1,
+       nargs='*',
        help='Rust compiler flags')
 set_config('RUSTFLAGS', depends('RUSTFLAGS')(lambda flags: flags))
 
