From ceea43dc0976ca0b716638bb1cee8ce24b10f319 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Wed, 15 May 2024 12:53:58 -0700
Subject: [PATCH] do not show nm connection editor in KDE

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 nm-connection-editor.desktop.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/nm-connection-editor.desktop.in b/nm-connection-editor.desktop.in
index bc13e9d9..24d576f1 100644
--- a/nm-connection-editor.desktop.in
+++ b/nm-connection-editor.desktop.in
@@ -7,3 +7,4 @@ Terminal=false
 StartupNotify=true
 Type=Application
 Categories=GNOME;GTK;Settings;X-GNOME-NetworkSettings;X-GNOME-Utilities;
+NotShowIn=KDE;
-- 
2.45.1

