diff -Naur network-manager-applet-1.24.0/nm-connection-editor.desktop.in network-manager-applet-1.24.0.desktop/nm-connection-editor.desktop.in
--- network-manager-applet-1.24.0/nm-connection-editor.desktop.in	2021-08-17 08:10:34.000000000 +0000
+++ network-manager-applet-1.24.0.desktop/nm-connection-editor.desktop.in	2022-09-14 05:44:19.885871247 +0000
@@ -10,3 +10,4 @@
 X-GNOME-Bugzilla-Product=NetworkManager
 X-GNOME-Bugzilla-Component=nm-connection-editor
 Categories=GNOME;GTK;Settings;X-GNOME-NetworkSettings;X-GNOME-Utilities;
+NotShowIn=KDE;
