From bfc4bfcbfcc265273358ad57fb8a767aad891190 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Sun, 3 Mar 2019 21:27:40 -0600
Subject: [PATCH] greeter: remove mlockall

---
 src/lightdm-gtk-greeter.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/lightdm-gtk-greeter.c b/src/lightdm-gtk-greeter.c
index aa1c4cb..c59fb3f 100644
--- a/src/lightdm-gtk-greeter.c
+++ b/src/lightdm-gtk-greeter.c
@@ -2748,9 +2748,6 @@ main (int argc, char **argv)
     guint            fallback_css_priority = GTK_STYLE_PROVIDER_PRIORITY_APPLICATION;
     GtkIconTheme    *icon_theme;
 
-    /* Prevent memory from being swapped out, as we are dealing with passwords */
-    mlockall (MCL_CURRENT | MCL_FUTURE);
-
     g_message ("Starting %s (%s, %s)", PACKAGE_STRING, __DATE__, __TIME__);
 
     /* Disable global menus */
-- 
2.20.1

