diff -up gperftools-2.7/src/base/linux_syscall_support.h.gcc9 gperftools-2.7/src/base/linux_syscall_support.h
--- gperftools-2.7/src/base/linux_syscall_support.h.gcc9	2019-01-08 13:02:24.410813132 -0500
+++ gperftools-2.7/src/base/linux_syscall_support.h	2019-01-08 13:05:37.608511076 -0500
@@ -1485,7 +1485,7 @@ struct kernel_stat {
                                "d"(LSS_SYSCALL_ARG(parent_tidptr)),
                                "r"(LSS_SYSCALL_ARG(newtls)),
                                "r"(LSS_SYSCALL_ARG(child_tidptr))
-                             : "rsp", "memory", "r8", "r10", "r11", "rcx");
+                             : "memory", "r8", "r10", "r11", "rcx");
       }
       LSS_RETURN(int, __res);
     }
