From b52e3f3139810f1d28f62df0f5a27f7709f1d4ed Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <wukaiyang@loongfans.cn>
Date: Mon, 8 Jun 2026 17:14:12 +0800
Subject: [PATCH] gvedit: force use linux-g++ spec

Signed-off-by: Kaiyang Wu <wukaiyang@loongfans.cn>
---
 cmd/gvedit/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmd/gvedit/Makefile.am b/cmd/gvedit/Makefile.am
index 9b1a8b918..01e53dbd6 100644
--- a/cmd/gvedit/Makefile.am
+++ b/cmd/gvedit/Makefile.am
@@ -63,7 +63,7 @@ mocables ui_settings.h: qMakefile
 
 # in case the user has both Qt4 and Qt5 installed, set QT_SELECT to force Qt5
 qMakefile: gvedit.pro
-	QT_SELECT=5 $(QMAKE) -o qMakefile gvedit.pro
+	QT_SELECT=5 $(QMAKE) -spec linux-g++ -o qMakefile gvedit.pro
 
 .1.1.pdf:
 	rm -f $@; pdffile=$@; psfile=$${pdffile%pdf}ps; \
-- 
2.52.0

