From 4701665477ab0baadc721d849cfd245dce3994a3 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Sun, 23 Nov 2025 05:19:37 -0800
Subject: [PATCH] chore: workaround lua 5.1 pkg-config suffix

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ce9529b79..57e633fb3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -737,7 +737,7 @@ else
             if test "$PKG_CONFIG" != ""; then
 	        AC_MSG_CHECKING(for Lua headers and libraries with pkg-config)
 	        echo
-	        for l in "$lua_suffix" "" "54" "5.4" "53" "5.3" "52" "5.2" "51" "5.1" ; do
+	        for l in "$lua_suffix" "" "54" "5.4" "53" "5.3" "52" "5.2" "51" "5.1" "-5.1" ; do
 	            pkgconfig_lua_found=`$PKG_CONFIG --exists lua$l 2>/dev/null`
 	            if test "$?" = "0" ; then
                         LUA_INCLUDES="$LUA_CFLAGS "`$PKG_CONFIG --cflags-only-I lua$l`
-- 
2.52.0

