diff -urN a/Jamfile b/Jamfile
--- a/Jamfile	2023-12-18 21:07:44.000000000 +0800
+++ b/Jamfile	2024-01-22 16:11:28.193486960 +0800
@@ -16,8 +16,8 @@
 }
 
 # add some additional files to package
-Package README README-Unlimited README-WaterWell RELNOTES TODO COPYING COPYING-data.txt COPYING-fonts.txt CREDITS ;
-InstallDoc README README-Unlimited README-WaterWell RELNOTES TODO COPYING COPYING-data.txt COPYING-fonts.txt CREDITS ;
+Package README README-Unlimited README-WaterWell RELNOTES TODO COPYING COPYING-data.txt COPYING-fonts.txt ;
+InstallDoc README README-Unlimited README-WaterWell RELNOTES TODO COPYING COPYING-data.txt COPYING-fonts.txt ;
 
 InstallDesktop lincity-ng.desktop ;
 InstallPixmap data/lincity-ng.png ;
