From 14b928f0c9425ec3872a516bed0125847ade994e Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 28 Feb 2024 21:52:54 +0800
Subject: [PATCH 01/21] [Fedora] QtBase: Fix build with libxkbcommon >= 1.6.0

---
 .../platformsupport/input/xkbcommon/qxkbcommon.cpp   | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/qtbase/src/platformsupport/input/xkbcommon/qxkbcommon.cpp b/qtbase/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
index 1c7c15d5d7..19e8f99565 100644
--- a/qtbase/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
+++ b/qtbase/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
@@ -281,7 +281,17 @@ static constexpr const auto KeyTbl = qMakeArray(
         Xkb2Qt<XKB_KEY_dead_aboveverticalline,  Qt::Key_Dead_Aboveverticalline>,
         Xkb2Qt<XKB_KEY_dead_belowverticalline,  Qt::Key_Dead_Belowverticalline>,
         Xkb2Qt<XKB_KEY_dead_longsolidusoverlay, Qt::Key_Dead_Longsolidusoverlay>,
-#endif
+#else
+        // These were removed in libxkbcommon 1.6.0
+        // XK_dead_lowline
+        Xkb2Qt<0xfe90,                          Qt::Key_Dead_Lowline>,
+        // XK_dead_aboveverticalline
+        Xkb2Qt<0xfe91,                          Qt::Key_Dead_Aboveverticalline>,
+        // XK_dead_belowverticalline
+        Xkb2Qt<0xfe92,                          Qt::Key_Dead_Belowverticalline>,
+        // XK_dead_longsolidusoverlay
+        Xkb2Qt<0xfe93,                          Qt::Key_Dead_Longsolidusoverlay>,
+#endif /* XKB_KEY_dead_lowline */
 
         // Special keys from X.org - This include multimedia keys,
         // wireless/bluetooth/uwb keys, special launcher keys, etc.
-- 
2.46.0

