diff -Naur sonic-visualiser-4.3/main/MainWindow.cpp sonic-visualiser-4.3.noupdate/main/MainWindow.cpp
--- sonic-visualiser-4.3/main/MainWindow.cpp	2021-08-19 21:58:40.659886576 +0800
+++ sonic-visualiser-4.3.noupdate/main/MainWindow.cpp	2021-08-19 22:04:59.624603669 +0800
@@ -89,7 +89,6 @@
 
 #include "Surveyer.h"
 #include "NetworkPermissionTester.h"
-#include "framework/VersionTester.h"
 
 // For version information
 #include <vamp/vamp.h>
@@ -339,15 +338,8 @@
         TransformFactory::getInstance()->startPopulationThread();

         m_surveyer = nullptr;
-
-        SVDEBUG << "MainWindow: Creating version tester" << endl;
-        m_versionTester = new VersionTester
-            ("sonicvisualiser.org", "latest-version.txt", SV_VERSION);
-        connect(m_versionTester, SIGNAL(newerVersionAvailable(QString)),
-                this, SLOT(newerVersionAvailable(QString)));
     } else {
         m_surveyer = nullptr;
-        m_versionTester = nullptr;
     }

     if (withOSCSupport && networkPermission) {
@@ -379,7 +371,6 @@
     delete m_unitConverter;
     delete m_preferencesDialog;
     delete m_layerTreeDialog;
-    delete m_versionTester;
     delete m_surveyer;
     Profiles::getInstance()->dump();
 //    SVDEBUG << "MainWindow::~MainWindow finishing" << endl;
@@ -5344,10 +5335,6 @@
     QTextEdit *textEdit = new QTextEdit;
     layout->addWidget(textEdit, row++, 1, 1, 2);
 
-    if (m_newerVersionIs != "") {
-        layout->addWidget(new QLabel(tr("<b>Note:</b> A newer version of Sonic Visualiser is available.<br>(Version %1 is available; you are using version %2)").arg(m_newerVersionIs).arg(SV_VERSION)), row++, 1, 1, 2);
-    }
-    
     QDialogButtonBox *bb = new QDialogButtonBox(QDialogButtonBox::Ok);
     layout->addWidget(bb, row++, 0, 1, 3);
     connect(bb, SIGNAL(accepted()), d, SLOT(accept()));
@@ -5595,22 +5582,3 @@
 {
     m_keyReference->show();
 }
-
-void
-MainWindow::newerVersionAvailable(QString version)
-{
-    m_newerVersionIs = version;
-    
-    QSettings settings;
-    settings.beginGroup("NewerVersionWarning");
-    QString tag = QString("version-%1-available-show").arg(version);
-    if (settings.value(tag, true).toBool()) {
-        QString title(tr("Newer version available"));
-        QString text(tr("<h3>Newer version available</h3><p>You are using version %1 of Sonic Visualiser, but version %2 is now available.</p><p>Please see the <a href=\"http://sonicvisualiser.org/\">Sonic Visualiser website</a> for more information.</p>").arg(SV_VERSION).arg(version));
-        QMessageBox::information(this, title, text);
-        settings.setValue(tag, false);
-    }
-    settings.endGroup();
-}
-
-
diff -Naur sonic-visualiser-4.3/main/MainWindow.h sonic-visualiser-4.3.noupdate/main/MainWindow.h
--- sonic-visualiser-4.3/main/MainWindow.h	2021-08-19 22:06:26.315378031 +0800
+++ sonic-visualiser-4.3.noupdate/main/MainWindow.h	2021-08-19 22:07:41.115197359 +0800
@@ -18,7 +18,6 @@
 
 #include "framework/MainWindowBase.h"
 
-class VersionTester;
 class Surveyer;
 class LayerTreeDialog;
 class ActivityLog;
@@ -169,7 +168,6 @@
     virtual void about();
     virtual void whatsNew();
     virtual void keyReference();
-    void newerVersionAvailable(QString) override;
 
 protected:
     Overview                *m_overview;
@@ -234,8 +232,6 @@
     QFileSystemWatcher      *m_templateWatcher;
 
     Surveyer                *m_surveyer;
-    VersionTester           *m_versionTester;
-    QString                  m_newerVersionIs;
 
     struct LayerConfiguration {
         LayerConfiguration(LayerFactory::LayerType _layer
diff -Naur sonic-visualiser-4.3/svapp/framework/MainWindowBase.h sonic-visualiser-4.3.noupdate/svapp/framework/MainWindowBase.h
--- sonic-visualiser-4.3/svapp/framework/MainWindowBase.h	2021-08-19 22:08:45.899049940 +0800
+++ sonic-visualiser-4.3.noupdate/svapp/framework/MainWindowBase.h	2021-08-19 22:09:55.935898800 +0800
@@ -388,8 +388,6 @@
 
     virtual void emitHideSplash();
 
-    virtual void newerVersionAvailable(QString) { }
-
     virtual void menuActionMapperInvoked(QObject *);
 
 protected:
