From 5b7db8a9e7220f421d77de7fcd483b12178c3f3f Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Wed, 24 Dec 2025 15:52:22 +0800
Subject: [PATCH] fix build on arm64

---
 main/segfault.c | 22 +++++++++++++++-------
 1 file changed, 15 insertions(+), 7 deletions(-)

diff --git a/main/segfault.c b/main/segfault.c
index 1ed18f4..40f2aeb 100644
--- a/main/segfault.c
+++ b/main/segfault.c
@@ -47,12 +47,11 @@
 static inline void
 print_str(int fd, const char *str)
 {
-
-	if (str == NULL) {
-		write(fd, NULLSTR, strlen(NULLSTR));
-	} else {
-		write(fd, str, strlen(str));
-	}
+    if (str == NULL) {
+        (void)write(fd, NULLSTR, strlen(NULLSTR));
+    } else {
+        (void)write(fd, str, strlen(str));
+    }
 }
 
 static void
@@ -69,13 +68,21 @@ static int
 print_stack_trace(ucontext_t *context)
 {
 	unw_cursor_t cursor;
+	unw_context_t uc;
 	unw_word_t backtrace[BACKTRACE_DEPTH];
 	unw_word_t ip, off;
 	char buf[1024];
 	unsigned int i, level;
 	int ret;
 
-	if ((ret = unw_init_local(&cursor, context)) != 0) {
+	(void)context;
+
+	if (unw_getcontext(&uc) != 0) {
+		print_str(STDERR_FILENO, "unw_getcontext failed\n");
+		return (1);
+	}
+
+	if ((ret = unw_init_local(&cursor, &uc)) != 0) {
 		print_unw_error("unw_init_local", ret);
 		return (1);
 	}
@@ -83,6 +90,7 @@ print_stack_trace(ucontext_t *context)
 	print_str(STDERR_FILENO, "   thread frame     IP       function\n");
 	level = 0;
 	ret = 0;
+
 	for (;;) {
 		char name[128];
 
-- 
2.52.0

