diff -rupN wxWidgets-3.0.2-orig/src/gtk/spinbutt.cpp wxWidgets-3.0.2/src/gtk/spinbutt.cpp
--- wxWidgets-3.0.2-orig/src/gtk/spinbutt.cpp	2014-10-06 17:33:44.000000000 -0400
+++ wxWidgets-3.0.2/src/gtk/spinbutt.cpp	2015-08-22 10:44:00.743403904 -0400
@@ -92,6 +92,10 @@ bool wxSpinButton::Create(wxWindow *pare
     g_object_ref(m_widget);
 
     gtk_entry_set_width_chars(GTK_ENTRY(m_widget), 0);
+#if GTK_CHECK_VERSION(3,12,0)
+    if (gtk_check_version(3,12,0) == NULL)
+        gtk_entry_set_max_width_chars(GTK_ENTRY(m_widget), 0);
+#endif
     gtk_spin_button_set_wrap( GTK_SPIN_BUTTON(m_widget),
                               (int)(m_windowStyle & wxSP_WRAP) );
 

