From 2f4b2d12fece5a046ccece256f73728604516326 Mon Sep 17 00:00:00 2001
From: Chris Kerr <cjk34@cam.ac.uk>
Date: Sun, 13 Oct 2013 19:11:10 +0100
Subject: [PATCH] Fixes build error introduced by previous commit

---
 src/sql/Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/sql/Makefile.am b/src/sql/Makefile.am
index e5fa05d..94b0838 100644
--- a/src/sql/Makefile.am
+++ b/src/sql/Makefile.am
@@ -1,8 +1,9 @@
+BUILT_SOURCES = parser.h
+AM_YFLAGS = -d
 lib_LTLIBRARIES	=	libmdbsql.la
 libmdbsql_la_SOURCES=	mdbsql.c parser.y lexer.l 
 libmdbsql_la_LDFLAGS = -version-info 2:0:0 -export-symbols-regex '^mdb_sql_'
 CLEANFILES = parser.c parser.h lexer.c
-BUILT_SOURCES = parser.h
 AM_CFLAGS	=	-I$(top_srcdir)/include $(GLIB_CFLAGS)
 LIBS	=	$(GLIB_LIBS)
 libmdbsql_la_LIBADD =	../libmdb/libmdb.la
