--- gnuradio-3.9.0.0/gr-uhd/CMakeLists.txt	2021-01-17 11:27:28.000000000 -0800
+++ gnuradio-3.9.0.0.atomic/gr-uhd/CMakeLists.txt	2021-02-12 16:57:29.777320720 -0800
@@ -17,6 +17,8 @@
     set(ENABLE_UHD_RFNOC FALSE)
 endif()
 
+set(CMAKE_EXE_LINKER_FLAGS "-latomic")
+
 ########################################################################
 # Register component
 ########################################################################
