--- gnuradio-3.7.11/gnuradio-runtime/lib/CMakeLists.txt.old	2017-07-10 11:19:48.210035847 +0000
+++ gnuradio-3.7.11/gnuradio-runtime/lib/CMakeLists.txt	2017-07-10 11:20:15.581316624 +0000
@@ -154,7 +154,7 @@
 
 #need to link with librt on ubuntu 11.10 for shm_*
 if(LINUX)
-    list(APPEND gnuradio_runtime_libs rt)
+    list(APPEND gnuradio_runtime_libs rt "-lpthread")
 endif()
 
 ########################################################################
