diff --git a/Services/NextTraceWrapper.cs b/Services/NextTraceWrapper.cs
index b638ea8..6c018a7 100644
--- a/Services/NextTraceWrapper.cs
+++ b/Services/NextTraceWrapper.cs
@@ -194,32 +194,6 @@ namespace OpenTrace.Services
             {
                 Console.WriteLine($"Using NextTrace: {nexttracePath}");
 
-#if NET8_0_OR_GREATER
-                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && 
-                    Array.Find(extraArgs, e => e == "-T" || e == "-U") != null &&
-                    Environment.UserName != "root")
-                {
-                    FileSystemInfo fa = new FileInfo(nexttracePath);
-                    if ((fa.UnixFileMode & UnixFileMode.SetUser) == 0) 
-                    {
-                        App.app.Invoke(() => {
-                            DialogResult dr = MessageBox.Show(Resources.MISSING_COMP_PRIV_TEXT, Resources.TCP_UDP_REQUIREMENTS_TITLE, MessageBoxButtons.YesNo);
-                            if (dr == DialogResult.Yes)
-                            {
-                                platformService.RestartAsAdministrator(host, () => {
-                                    MessageBox.Show(Resources.RESTART_AS_ADMIN_FAILED, Resources.TCP_UDP_REQUIREMENTS_TITLE, MessageBoxButtons.OK);
-                                    Process.Start(new ProcessStartInfo("https://github.com/Archeb/opentrace/wiki/How-to-manually-set-the-required-permissions-for-TCP-UDP-traceroute-on-macOS-and-Linux") { UseShellExecute = true });
-                                });
-                            } else {
-                                Process.Start(new ProcessStartInfo("https://github.com/Archeb/opentrace/wiki/How-to-manually-set-the-required-permissions-for-TCP-UDP-traceroute-on-macOS-and-Linux") { UseShellExecute = true });
-                            }
-                        });
-                        Status = AppStatus.Quit;
-                        AppQuit?.Invoke(this, new AppQuitEventArgs(0));
-                        return;
-                    }
-                }
-#endif
                 if (stopRequested)
                 {
                     Status = AppStatus.Quit;

