diff --git a/OpenTrace.csproj b/OpenTrace.csproj
index 185809d..1f5a216 100644
--- a/OpenTrace.csproj
+++ b/OpenTrace.csproj
@@ -9,7 +9,7 @@
   <PropertyGroup>
     <TargetFramework Condition="'$(RuntimeIdentifier)' == 'win-x64'">net48</TargetFramework>
     <TargetFramework Condition="'$(RuntimeIdentifier)' == 'win-arm64'">net481</TargetFramework>
-    <TargetFramework Condition="'$(TargetFramework)' == ''">net8.0</TargetFramework>
+    <TargetFramework Condition="'$(TargetFramework)' == ''">net10.0</TargetFramework>
     
     <GeneratePackageOnBuild>False</GeneratePackageOnBuild>
     <OutputType>Exe</OutputType>
