diff --git a/lib/srchilite/fileutil.cc b/lib/srchilite/fileutil.cc
index 59a6d64..b7aa3a9 100644
--- a/lib/srchilite/fileutil.cc
+++ b/lib/srchilite/fileutil.cc
@@ -48,7 +48,7 @@ void set_file_util_verbose(bool b) {
 // FIXME avoid using a global variable
 std::string start_path;
 
-string readFile(const string &fileName) throw (IOException) {
+string readFile(const string &fileName) noexcept(false) {
     ifstream file(fileName.c_str());
 
     if (!file.is_open()) {
diff --git a/lib/srchilite/fileutil.h b/lib/srchilite/fileutil.h
index 7335a9b..456386c 100644
--- a/lib/srchilite/fileutil.h
+++ b/lib/srchilite/fileutil.h
@@ -27,7 +27,7 @@ extern std::string start_path;
  * @return the contents of the file
  * @throw IOException
  */
-string readFile(const string &fileName) throw (IOException);
+string readFile(const string &fileName) noexcept(false);
 
 //char *read_file(const string &fileName);
 
