From 4d23cc82f039497ef280232a358ec15f21b17a59 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Fri, 8 Aug 2025 12:57:53 +0800
Subject: [PATCH] fix path of libev

---
 sslh-ev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sslh-ev.c b/sslh-ev.c
index 8498d2a..8e1b648 100644
--- a/sslh-ev.c
+++ b/sslh-ev.c
@@ -21,7 +21,7 @@
 */
 
 #include <stdlib.h>
-#include <ev.h>
+#include <libev/ev.h>
 #include "gap.h"
 #include "log.h"
 #include "udp-listener.h"
-- 
2.50.1

