From 9710d1cd34f6a1a42e7103abd267e282f3a7ba7c Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Mon, 29 Jun 2026 13:49:07 +0800
Subject: [PATCH] AOSCOS: utils/padsp: undef _TIME_BITS if _FILE_OFFSET_BITS is
 defined

---
 src/utils/padsp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/utils/padsp.c b/src/utils/padsp.c
index 9d1cb5cad..2c986aa20 100644
--- a/src/utils/padsp.c
+++ b/src/utils/padsp.c
@@ -23,6 +23,10 @@
 #endif
 
 #ifdef _FILE_OFFSET_BITS
+/* _TIME_BITS requires _FILE_OFFSET_BITS. */
+#ifdef _TIME_BITS
+#undef _TIME_BITS
+#endif
 #undef _FILE_OFFSET_BITS
 #endif
 
-- 
2.52.0

