From 8a81071faccba365506261bb12f46bc87ba4bed0 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Wed, 19 Feb 2025 08:58:50 -0800
Subject: [PATCH 1/5] AOSCOS: Remove check for update button

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 chrome/content/zotero/zoteroPane.xhtml | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/chrome/content/zotero/zoteroPane.xhtml b/chrome/content/zotero/zoteroPane.xhtml
index ba1049efa..6cc045728 100644
--- a/chrome/content/zotero/zoteroPane.xhtml
+++ b/chrome/content/zotero/zoteroPane.xhtml
@@ -848,10 +848,6 @@
 									data-l10n-id="restart-in-troubleshooting-mode-menuitem"
 									oncommand="ZoteroStandalone.promptForRestartInTroubleshootingMode();"/>
 							<menuseparator/>
-							<menuitem id="checkForUpdates"
-									  accesskey="&helpCheckForUpdates.accesskey;"
-									  label="&helpCheckForUpdates.label;"
-									  oncommand="Zotero.openCheckForUpdatesWindow();"/>
 							<menuitem id="aboutName"
 									  accesskey="&aboutProduct.accesskey;"
 									  label="&aboutProduct.label;"
-- 
2.52.0

