diff -Naur primus/primusrun primus.glvnd/primusrun
--- primus/primusrun	2018-04-28 19:24:28.200404397 +0000
+++ primus.glvnd/primusrun	2018-04-28 19:24:41.823738287 +0000
@@ -26,7 +26,7 @@
 # export PRIMUS_libGLd=${PRIMUS_libGLd:-'/usr/$LIB/libGL.so.1'}
 
 # Directory containing primus libGL
-PRIMUS_libGL=${PRIMUS_libGL:-$(dirname `readlink -ne $0`)/'$LIB'}
+PRIMUS_libGL=/usr/\$LIB/primus
 
 # On some distributions, e.g. on Ubuntu, libnvidia-tls.so is not available
 # in default search paths.  Add its path manually after the primus library
@@ -39,4 +39,5 @@
 export LD_LIBRARY_PATH=${PRIMUS_libGL}${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
 
 # And go!
+export __GLVND_DISALLOW_PATCHING=1
 exec "$@"
