From ef89dbce5e17c141ac3b2e5da3ea72716bbcbb16 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:43:01 +0800
Subject: [PATCH 158/437] MARKER: FROMLIST: End of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 2032375611b8b..d2bb0d55c3ef6 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -548,7 +548,6 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
-	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.52.0

