From dab9371105ae1c8807dfbe81440d5643e9d27b7b Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:42:45 +0800
Subject: [PATCH 175/466] MARKER: FROMLIST: Start of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/mips/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 3ab48b34b8024..8abc9ece3738d 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -545,6 +545,7 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
+	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.52.0

