From 0fdb65ae88eff044db2a21794cb8c9907192e8e0 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:43:01 +0800
Subject: [PATCH 185/467] MARKER: FROMLIST: End of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/mips/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 887935175127d..58f2089c1d91f 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -548,7 +548,6 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
-	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.52.0

