From 2cb40e1ae80d9d1bc1e7f52e15c5a3aa81860477 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:43:40 +0800
Subject: [PATCH 224/345] MARKER: AOSCOS: Start of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 3775d2b066519..dae56c1544345 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -532,6 +532,7 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
+	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.51.0

