From 5bc0c7627fa5430e5faab94103e5ea5fc7513288 Mon Sep 17 00:00:00 2001
From: Patrick Yavitz <pyavitz@gmail.com>
Date: Mon, 23 Dec 2024 10:14:57 -0500
Subject: [PATCH 408/587] ARMBIAN: Enable creation of __symbols__ node

Signed-off-by: Patrick Yavitz <pyavitz@gmail.com>
Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 scripts/Makefile.dtbs | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/scripts/Makefile.dtbs b/scripts/Makefile.dtbs
index c4e4663902844..8c03bd8a9bc43 100644
--- a/scripts/Makefile.dtbs
+++ b/scripts/Makefile.dtbs
@@ -111,6 +111,11 @@ else
 DTC_FLAGS += -Wunique_unit_address_if_enabled
 endif
 
+ifeq ($(CONFIG_OF_OVERLAY),y)
+# enable creation of __symbols__ node
+DTC_FLAGS += -@
+endif
+
 ifneq ($(findstring 2,$(KBUILD_EXTRA_WARN)),)
 DTC_FLAGS += -Wnode_name_chars_strict \
              -Wproperty_name_chars_strict \
-- 
2.52.0

