From 93160ee8fd841b17c4eacb2004718dedd758cdd1 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 25 Feb 2025 17:45:08 +0800
Subject: [PATCH 412/418] MARKER: AOSCOS: End of Lemote patches

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/mips/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index c80f0c740a0b..145fb6cc2f3e 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -548,7 +548,6 @@ config MACH_LOONGSON64
 	select USE_OF
 	select BUILTIN_DTB
 	select PCI_HOST_GENERIC
-	depends on BROKEN
 	help
 	  This enables the support of Loongson-2/3 family of machines.
 
-- 
2.52.0

