diff -Naur libwnck-2.31.0/doc/Makefile.am libwnck-2.31.0.gtkdoc/doc/Makefile.am
--- libwnck-2.31.0/doc/Makefile.am	2012-03-05 03:29:53.000000000 -0600
+++ libwnck-2.31.0.gtkdoc/doc/Makefile.am	2020-06-23 19:26:32.485049168 -0500
@@ -10,7 +10,7 @@
 DOC_SOURCE_DIR=../libwnck
 
 # Extra options to pass to gtkdoc-scangobj
-SCANGOBJ_OPTIONS=--nogtkinit --type-init-func="g_type_init ()"
+SCANGOBJ_OPTIONS=--type-init-func="g_type_init ()"
 
 # Extra options to supply to gtkdoc-scan
 SCAN_OPTIONS=--deprecated-guards="WNCK_DISABLE_DEPRECATED"
diff -Naur libwnck-2.31.0/doc/Makefile.in libwnck-2.31.0.gtkdoc/doc/Makefile.in
--- libwnck-2.31.0/doc/Makefile.in	2012-03-05 03:30:54.000000000 -0600
+++ libwnck-2.31.0.gtkdoc/doc/Makefile.in	2020-06-23 19:28:32.861941422 -0500
@@ -257,7 +257,7 @@
 DOC_SOURCE_DIR = ../libwnck
 
 # Extra options to pass to gtkdoc-scangobj
-SCANGOBJ_OPTIONS = --nogtkinit --type-init-func="g_type_init ()"
+SCANGOBJ_OPTIONS = --type-init-func="g_type_init ()"
 
 # Extra options to supply to gtkdoc-scan
 SCAN_OPTIONS = --deprecated-guards="WNCK_DISABLE_DEPRECATED"
