From 97b06641533bf4bb52e0d6f963888c137653efbc Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Wed, 14 Jul 2021 23:11:12 -0600
Subject: [PATCH] mips64: fix build

---
 src/arch/mips64.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/arch/mips64.cpp b/src/arch/mips64.cpp
index b63d1ce5..cd7f0bdc 100644
--- a/src/arch/mips64.cpp
+++ b/src/arch/mips64.cpp
@@ -161,5 +161,11 @@ std::string name()
   return std::string("mips64");
 }
 
+std::vector<std::string> invalid_watchpoint_modes()
+{
+  throw std::runtime_error(
+      "Watchpoints are not supported on this architecture");
+}
+
 } // namespace arch
 } // namespace bpftrace
-- 
2.32.0

