Author: Daniel Leidert <dleidert@debian.org>
Description: Fix several FTBFSes by cmake and make.
Forwarded: not yet

--- a/scripts/CMakeLists.txt
+++ b/scripts/CMakeLists.txt
@@ -242,7 +242,7 @@
         # bindings_perl as the WORKING_DIRECTORY needs to already exist!
         COMMAND ${CMAKE_COMMAND} -E make_directory ${openbabel_BINARY_DIR}/scripts/perl
         )
-      add_custom_target(bindings_perl ALL
+      add_custom_target(all_bindings_perl ALL
         COMMAND ${CMAKE_COMMAND} -E copy ${openbabel_SOURCE_DIR}/scripts/perl/Makefile.PL ${openbabel_BINARY_DIR}/scripts/perl
         COMMAND ${CMAKE_COMMAND} -E copy ${openbabel_SOURCE_DIR}/scripts/perl/OpenBabel.pm ${openbabel_BINARY_DIR}/scripts/perl
         COMMAND ${CMAKE_COMMAND} -E copy ${openbabel_SOURCE_DIR}/scripts/perl/openbabel-perl.cpp ${openbabel_BINARY_DIR}/scripts/perl
@@ -251,7 +251,7 @@
             DEPENDS openbabel ${openbabel_SOURCE_DIR}/scripts/perl/openbabel-perl.cpp
             WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/perl
         )
-      add_dependencies(bindings_perl pre_bindings_perl openbabel)
+      add_dependencies(all_bindings_perl pre_bindings_perl openbabel)
       install(CODE "execute_process(COMMAND make install DESTDIR=\$ENV{DESTDIR} WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/perl)")
 
     else (NOT MSVC)
--- a/scripts/python/setup.py
+++ b/scripts/python/setup.py
@@ -24,7 +24,7 @@
                  )
 
 if "build" in sys.argv:
-    shutil.copyfile(os.path.join(srcdir, "pybel_py%dx.py" % sys.version_info[0]), "pybel.py")
+    shutil.copyfile(os.path.join(srcdir, "pybel.py"), "pybel.py")
     shutil.copyfile(os.path.join(srcdir, "openbabel.py"), "openbabel.py")
 
 setup(name='openbabel',
