Index: b/include/openbabel/shared_ptr.h
===================================================================
--- a/include/openbabel/shared_ptr.h
+++ b/include/openbabel/shared_ptr.h
@@ -21,7 +21,7 @@ GNU General Public License for more deta
   #define shared_ptr boost::shared_ptr
 #else
   #include <memory>
-  #if __GNUC__ == 4  //&& __GNUC_MINOR__ < 3  removed at the suggestion of Konstantin Tokarev
+  #if __GNUC__ >= 4  //&& __GNUC_MINOR__ < 3  removed at the suggestion of Konstantin Tokarev
     #include <tr1/memory>
   #endif
   using std::tr1::shared_ptr;
