--- xdm-1.1.12/xdm.service.in	2019-03-02 16:06:13.000000000 -0600
+++ xdm-1.1.12.systemd/xdm.service.in	2020-06-19 21:38:08.792737850 -0500
@@ -1,11 +1,15 @@
 [Unit]
-Description=X-Window Display Manager
-After=systemd-user-sessions.service
+Description=X11 Display Manager
+Documentation=man:xdm(1)
+Conflicts=getty@tty1.service
+After=systemd-user-sessions.service getty@tty1.service plymouth-quit.service
 
 [Service]
 ExecStart=BINDIR/xdm -nodaemon
-Type=notify
-NotifyAccess=all
+Type=simple
+Restart=always
+IgnoreSIGPIPE=no
 
 [Install]
 Alias=graphical.target.wants/xdm.service
+Alias=display-manager.service
