diff -Naur mono-upnp-0.1.2/Makefile.am mono-upnp-0.1.2.nonunit/Makefile.am
--- mono-upnp-0.1.2/Makefile.am	2012-05-31 19:01:03.000000000 +0000
+++ mono-upnp-0.1.2.nonunit/Makefile.am	2020-03-20 20:50:46.844547222 +0000
@@ -5,13 +5,8 @@
 
 #Warning: This is an automatically generated file, do not edit!
 if ENABLE_DEBUG
- SUBDIRS =  src/Mono.Ssdp/Mono.Ssdp src/Mono.Upnp src/Mono.Upnp.Dcp/Mono.Upnp.Dcp.MediaServer1 tests
+ SUBDIRS =  src/Mono.Ssdp/Mono.Ssdp src/Mono.Upnp src/Mono.Upnp.Dcp/Mono.Upnp.Dcp.MediaServer1
 endif
 if ENABLE_RELEASE
- SUBDIRS =  src/Mono.Ssdp/Mono.Ssdp src/Mono.Upnp src/Mono.Upnp.Dcp/Mono.Upnp.Dcp.MediaServer1 tests
+ SUBDIRS =  src/Mono.Ssdp/Mono.Ssdp src/Mono.Upnp src/Mono.Upnp.Dcp/Mono.Upnp.Dcp.MediaServer1
 endif
-
-.PHONY: test
-
-test: all
-	(cd bin; $(NUNIT_CMD) *.Tests.dll)
diff -Naur mono-upnp-0.1.2/configure.ac mono-upnp-0.1.2.nonunit/configure.ac
--- mono-upnp-0.1.2/configure.ac	2012-05-31 19:01:03.000000000 +0000
+++ mono-upnp-0.1.2.nonunit/configure.ac	2020-03-20 20:32:57.374087333 +0000
@@ -51,11 +51,6 @@
 PKG_CHECK_MODULES([GLADE_SHARP_20], [glade-sharp-2.0])
 PKG_CHECK_MODULES([MONO_ADDINS], [mono-addins])
 PKG_CHECK_MODULES([TAGLIB_SHARP], [taglib-sharp])
-PKG_CHECK_MODULES([NUNIT], [nunit])
-
-AC_PATH_PROG(NUNIT_CMD, nunit-console2, nunit-console)
-AC_SUBST(NUNIT_CMD)
-
 
 AC_CONFIG_FILES([
 src/Mono.Ssdp/Mono.Ssdp/mono.ssdp.pc
@@ -76,11 +71,6 @@
 src/Mono.Upnp.Dcp/Mono.Upnp.Dcp.MediaServer1/Mono.Upnp.Dcp.MediaServer1.GtkClient/mono.upnp.dcp.mediaserver1.gtkclient.pc
 src/Mono.Upnp.Dcp/Mono.Upnp.Dcp.MediaServer1/Mono.Upnp.Dcp.MediaServer1.GtkClient/Makefile
 src/Mono.Upnp.Dcp/Mono.Upnp.Dcp.MediaServer1/Makefile
-tests/Mono.Ssdp.Tests/Makefile
-tests/Mono.Upnp.Tests/Makefile
-tests/Mono.Upnp.Dcp.MediaServer1.Tests/Makefile
-tests/Mono.Upnp.Dcp.MediaServer1.FileSystem.Tests/Makefile
-tests/Makefile
 Makefile
 
 ])
