From 41bb721f9ee878b31151c03340f4739fe904f19d Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 12 Oct 2024 13:15:01 +0000
Subject: [PATCH] FROM AOSC: Make pyuno work with system-wide module install

---
 pyuno/source/module/uno.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/pyuno/source/module/uno.py b/pyuno/source/module/uno.py
index b2a75bd039825..dde88af9ac417 100644
--- a/pyuno/source/module/uno.py
+++ b/pyuno/source/module/uno.py
@@ -16,8 +16,14 @@
 #   except in compliance with the License. You may obtain a copy of
 #   the License at http://www.apache.org/licenses/LICENSE-2.0 .
 #
-import pyuno
+
+# Special modification to make uno load on Arch
+import os
 import sys
+sys.path.append("/usr/lib/libreoffice/program/")
+os.putenv("URE_BOOTSTRAP", "vnd.sun.star.pathname:/usr/lib/libreoffice/program/fundamentalrc")
+
+import pyuno
 import traceback
 import warnings
 
