diff --git a/configure.ac b/configure.ac
index cb786847f11d..48788a176310 100644
--- a/configure.ac
+++ b/configure.ac
@@ -8576,7 +8576,7 @@ if test "$ENABLE_JAVA" != ""; then
         JAVAINTERPRETER=`win_short_path_for_make "$JAVAINTERPRETER"`
     elif test "$cross_compiling" != "yes"; then
         case $CPUNAME in
-            AARCH64|AXP|X86_64|IA64|POWERPC64|S390X|SPARC64|MIPS64|RISCV64)
+            AARCH64|AXP|X86_64|IA64|POWERPC64|S390X|SPARC64|MIPS64|RISCV64|LOONGARCH64)
                 if test -f "$JAVAINTERPRETER" -a "`$JAVAINTERPRETER -version 2>&1 | $GREP -i 64-bit`" = "" >/dev/null; then
                     AC_MSG_WARN([You are building 64-bit binaries but the JDK $JAVAINTERPRETER is 32-bit])
                     AC_MSG_ERROR([You should pass the --with-jdk-home option pointing to a 64-bit JDK])
diff --git a/desktop/source/deployment/misc/dp_platform.cxx b/desktop/source/deployment/misc/dp_platform.cxx
index 879e617b7258..a6b700dbd0ac 100644
--- a/desktop/source/deployment/misc/dp_platform.cxx
+++ b/desktop/source/deployment/misc/dp_platform.cxx
@@ -127,6 +127,8 @@ namespace
             ret = checkOSandCPU(u"Linux", u"ALPHA");
         else if (token == u"linux_aarch64")
             ret = checkOSandCPU(u"Linux", u"AARCH64");
+        else if (token == u"linux_loongarch64")
+            ret = checkOSandCPU(u"Linux", u"LOONGARCH64");
         else if (token == u"freebsd_x86")
             ret = checkOSandCPU(u"FreeBSD", u"x86");
         else if (token == u"freebsd_x86_64")
