diff -Naur plasma-angelfish/CMakeLists.txt plasma-angelfish.feedback/CMakeLists.txt
--- plasma-angelfish/CMakeLists.txt	2022-07-17 05:18:45.815207915 +0000
+++ plasma-angelfish.feedback/CMakeLists.txt	2022-07-17 05:12:09.355591197 +0000
@@ -47,7 +47,7 @@
 
 ################# Find dependencies #################
 
-find_package(Qt5 ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Core Quick Test Gui Svg QuickControls2 Sql Feedback)
+find_package(Qt5 ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Core Quick Test Gui Svg QuickControls2 Sql)
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS Kirigami2 Purpose I18n Config CoreAddons DBusAddons WindowSystem Notifications)
 
 # Necessary to support QtWebEngine installed in a different prefix than the rest of Qt (e.g flatpak)
