From 627b1c16fe9a4018e0f488ed54afd77feb651ea2 Mon Sep 17 00:00:00 2001
From: Qing Yun <kf@aosc.io>
Date: Thu, 5 Mar 2026 14:54:17 +0800
Subject: [PATCH 3/4] igc: compatable with gcc15

---
 IGC/Compiler/Optimizer/InstructionHoistingOptimization.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/IGC/Compiler/Optimizer/InstructionHoistingOptimization.cpp b/IGC/Compiler/Optimizer/InstructionHoistingOptimization.cpp
index 82276bdd8..3e591ed84 100644
--- a/IGC/Compiler/Optimizer/InstructionHoistingOptimization.cpp
+++ b/IGC/Compiler/Optimizer/InstructionHoistingOptimization.cpp
@@ -147,7 +147,7 @@ private:
   }
 
   inline void InsertHoistableInstruction(Instruction *hoistInstr) {
-    auto isDuplicate = [](std::vector<Instruction *> hoistedInstr, Instruction *newInstr) -> bool {
+    auto isDuplicate = [](const std::vector<Instruction *> &hoistedInstr, Instruction *newInstr) -> bool {
       auto it = std::find(hoistedInstr.begin(), hoistedInstr.end(), newInstr);
       return it != hoistedInstr.end();
     };
-- 
2.52.0

