diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4ea34f9d..eff7e5bd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,7 +42,7 @@ set(DEFAULT_PREFS_FILE "" CACHE STRING "Set system wide preferences file")
 #server options
 set(SERVER_UID "" CACHE STRING "User id of the user who runs wesnothd")
 set(SERVER_GID "" CACHE STRING "Group id of the user who runs wesnothd")
-set(FIFO_DIR "/var/run/wesnothd" CACHE STRING "Directory for the wesnothd fifo socket file")
+set(FIFO_DIR "/run/wesnothd" CACHE STRING "Directory for the wesnothd fifo socket file")
 
 #build options
 option(ENABLE_GAME "Enable compilation of the game" ON)
