--- a/src/gettext.cpp	2023-10-09 15:16:58.387860268 -0700
+++ b/src/gettext.cpp	2023-10-09 15:24:42.758873198 -0700
@@ -169,7 +169,11 @@
 			return msg;
 		}
 
+#if BOOST_VERSION < 108300
 		const char* get(int domain_id, const char* ctx, const char* sid, int n) const override
+#else
+		const char* get(int domain_id, const char* ctx, const char* sid, bl::count_type n) const override
+#endif
 		{
 			auto& base = get_base();
 			const char* msg = base.get(domain_id, ctx, sid, n);
