--- teeworlds-0.7.5-src/bam.lua	2020-04-19 15:23:41.000000000 -0500
+++ teeworlds-0.7.5-src.everywhere/bam.lua	2021-09-06 00:55:25.231578947 -0500
@@ -174,21 +174,6 @@
 end
 
 function GenerateLinuxSettings(settings, conf, arch, compiler)
-	if arch == "x86" then
-		if config.buildwithoutsseflag.value == false then
-			settings.cc.flags:Add("-msse2") -- for the _mm_pause call
-		end
-		settings.cc.flags:Add("-m32")
-		settings.link.flags:Add("-m32")
-	elseif arch == "x86_64" then
-		settings.cc.flags:Add("-m64")
-		settings.link.flags:Add("-m64")
-	elseif arch == "armv7l" then
-		-- arm 32 bit
-	else
-		print("Unknown Architecture '" .. arch .. "'. Supported: x86, x86_64")
-		os.exit(1)
-	end
 	settings.link.libs:Add("pthread")
 
 	GenerateCommonSettings(settings, conf, arch, compiler)
