--- a/setup.py	2020-03-27 20:11:10.080375697 +0000
+++ b/setup.py	2020-03-27 20:11:48.333174940 +0000
@@ -157,7 +157,7 @@
 print("setup.py: platform.system() => %r" % SYSTEM)
 print("setup.py: platform.architecture() => %r" % (platform.architecture(),))
 if SYSTEM == 'Linux':
-    print("setup.py: platform.linux_distribution() => %r" % (platform.linux_distribution(),))
+    print("setup.py: platform.linux_distribution() => %r" % (('AOSC OS', '', ''),))
 if SYSTEM != 'Windows':
     print("setup.py: platform.libc_ver() => %r" % (platform.libc_ver(),))
 
