From: Andrej Shadura <andrewsh@debian.org>
Date: Thu, 18 Jul 2019 13:00:01 -0300
Subject: Add $(CPPFLAGS), $(CFLAGS) and $(LDFLAGS) to $(COMPILE_FOR_BUILD) so that helpers are built with hardening

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -14,6 +14,7 @@
 COMPILE_FOR_BUILD = \
 	$(CC_FOR_BUILD) $(DEFAULT_INCLUDES) $(AM_CPPFLAGS_FOR_BUILD) \
 	$(CPPFLAGS_FOR_BUILD) \
+	$(CPPFLAGS) $(CFLAGS) $(LDFLAGS) \
 	$(AM_CFLAGS_FOR_BUILD) $(CFLAGS_FOR_BUILD) 
 
 bin_PROGRAMS = dash
