diff --git a/Cargo.toml.in b/Cargo.toml.in
index 6ec4c04..e5d8b53 100644
--- a/Cargo.toml.in
+++ b/Cargo.toml.in
@@ -23,14 +23,14 @@ rustc_less_1_36 = []
 
 # Dependencies which don't change based on build flags
 [dependencies.cairo-sys-rs]
-version = ""
+version = "*"
 
 [dependencies.glib-sys]
-version = ""
+version = "*"
 features = ["v2_44"]
 
 [dependencies.gtk-sys]
-version = ""
+version = "*"
 features = ["v3_22"]
 
 [dependencies]
