Description: Add support for devices with long names
Bug-Ubuntu: http://launchpad.net/bugs/1130923
Index: consolekit/src/ck-sysdeps-linux.c
===================================================================
--- consolekit.orig/src/ck-sysdeps-linux.c	2013-03-04 10:53:26.597712000 -0800
+++ consolekit/src/ck-sysdeps-linux.c	2013-03-04 10:57:33.612308825 -0800
@@ -93,7 +93,7 @@
         guint major_number;
         guint minor_first;
         guint minor_last;
-        char name[16];
+        char name[32];
         char devfs_type;
 } tty_map_node;
 
