From 170156aed17caa65c63f17b49b6b01e336b78e18 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Tue, 25 Aug 2020 12:35:08 +0800
Subject: [PATCH]  Let pkgconfig find `lua5.3`

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff75973..57b0c05 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/lua5.4/lua.h")
   find_package(PkgConfig)
   if(PkgConfig_FOUND)
-    foreach(pkg lua lua54 lua53 lua52 luajit lua51)
+    foreach(pkg lua lua5.3 lua54 lua53 lua52 luajit lua51)
       pkg_check_modules(LUA ${pkg})
       if(LUA_FOUND)
 	break()
-- 
2.27.0

