From d986a39ec6f59eda4363bd83846d7afdbc17352d Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sat, 23 Apr 2022 12:43:46 +0800
Subject: [PATCH] config: do not check update

---
 config/src/config.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/src/config.rs b/config/src/config.rs
index 02ed12b..b5ff37d 100644
--- a/config/src/config.rs
+++ b/config/src/config.rs
@@ -508,7 +508,7 @@ pub struct Config {
     #[serde(default = "default_true")]
     pub automatically_reload_config: bool,
 
-    #[serde(default = "default_true")]
+    #[serde(default)]
     pub check_for_updates: bool,
     #[serde(default)]
     pub show_update_window: bool,
-- 
2.35.1

