--- a/CMake/Documentation.cmake	2016-09-20 16:53:51.000000000 -0500
+++ b/CMake/Documentation.cmake	2016-09-24 00:27:49.836352597 -0500
@@ -31,7 +31,7 @@
          # OpenSUSE
          /usr/share/xml/docbook/stylesheet/nwalsh/current
          # Arch
-          /usr/share/xml/docbook/xsl-stylesheets
+          /usr/share/xml/docbook/xsl-stylesheets-1.79.2
          # Fedora
          /usr/share/sgml/docbook/xsl-stylesheets
          # Fink
