diff -Naur stalonetray-old/configure.ac stalonetray/configure.ac
--- stalonetray-old/configure.ac	2022-06-13 21:43:42.083281711 +0800
+++ stalonetray/configure.ac	2022-06-13 21:47:17.628579402 +0800
@@ -177,7 +177,7 @@
 AC_DEFUN([AX_CHECK_DOCBOOK], [
     XSLTPROC_FLAGS=--nonet
     DOCBOOK_ROOT=
-    for i in /usr/share/sgml/docbook/stylesheet/xsl/nwalsh /usr/share/xml/docbook/stylesheet/docbook-xs /usr/share/sgml/docbook/xsl-stylesheets/ /usr/local/share/xsl/docbook;
+    for i in /usr/share/sgml/docbook/stylesheet/xsl/nwalsh /usr/share/xml/docbook/stylesheet/docbook-xs /usr/share/sgml/docbook/xsl-stylesheets/ /usr/local/share/xsl/docbook /usr/share/xml/docbook/xsl-stylesheets-1.79.2;
     do
         if test -d "$i"; then
             DOCBOOK_ROOT=$i
