diff --git a/drivers/pinctrl/pinctrl-amd.c b/drivers/pinctrl/pinctrl-amd.c
index d69e357a7a98..f2b3ed33c65a 100644
--- a/drivers/pinctrl/pinctrl-amd.c
+++ b/drivers/pinctrl/pinctrl-amd.c
@@ -29,6 +29,7 @@
 #include <linux/interrupt.h>
 #include <linux/list.h>
 #include <linux/bitops.h>
+#include <linux/dmi.h>
 #include <linux/pinctrl/pinconf.h>
 #include <linux/pinctrl/pinconf-generic.h>

@@ -739,6 +740,26 @@ static struct pinctrl_desc amd_pinctrl_desc = {
        .owner = THIS_MODULE,
 };

+static bool amd_gpio_is_gigabyte_am4(void)
+{
+ const char *board_name;
+
+ if (!dmi_match(DMI_BOARD_VENDOR, "Gigabyte Technology Co., Ltd."))
+ return false;
+
+ board_name = dmi_get_system_info(DMI_BOARD_NAME);
+
+ if (!board_name)
+ return false;
+
+ if (strstr(board_name, "A320") ||
+ strstr(board_name, "B350") ||
+ strstr(board_name, "X370"))
+ return true;
+
+ return false;
+}
+
 static int amd_gpio_probe(struct platform_device *pdev)
 {
        int ret = 0;
@@ -746,6 +767,10 @@
 	struct resource *res;
 	struct amd_gpio *gpio_dev;
 
+	/* Don't load this module if it's a Gigabyte AM4 board */
+	if (amd_gpio_is_gigabyte_am4())
+		return -ENODEV;
+
 	gpio_dev = devm_kzalloc(&pdev->dev,
 				sizeof(struct amd_gpio), GFP_KERNEL);
 	if (!gpio_dev)
