diff -Naur linux-5.8-rc1/drivers/platform/x86/panasonic-laptop.c linux-5.8-rc1.panasonic/drivers/platform/x86/panasonic-laptop.c
--- linux-5.8-rc1/drivers/platform/x86/panasonic-laptop.c	2020-06-14 14:45:04.000000000 -0500
+++ linux-5.8-rc1.panasonic/drivers/platform/x86/panasonic-laptop.c	2020-07-07 22:35:42.957089381 -0500
@@ -169,6 +169,7 @@
 	{ "MAT0013", 0},
 	{ "MAT0018", 0},
 	{ "MAT0019", 0},
+	{ "MAT0028", 0},
 	{ "", 0},
 };
 MODULE_DEVICE_TABLE(acpi, pcc_device_ids);
