From 7ff36c3146fc88600ae65cecfac440fcd4bb646e Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Sun, 3 Jan 2021 01:10:53 -0600
Subject: [PATCH] bugfix btusb: dirty quirk on ath3k firmware loading

---
 drivers/bluetooth/btusb.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/bluetooth/btusb.c b/drivers/bluetooth/btusb.c
index 80468745d5c5..dff1b7b65965 100644
--- a/drivers/bluetooth/btusb.c
+++ b/drivers/bluetooth/btusb.c
@@ -3692,9 +3692,10 @@ static int btusb_setup_qca(struct hci_dev *hdev)
 		if (ver_rom == qca_devices_table[i].rom_version)
 			info = &qca_devices_table[i];
 	}
+
 	if (!info) {
 		bt_dev_err(hdev, "don't support firmware rome 0x%x", ver_rom);
-		return -ENODEV;
+		return (ver_rom == 0x11020000) ? 0 : -ENODEV;
 	}
 
 	err = btusb_qca_send_vendor_req(udev, QCA_CHECK_STATUS, &status,
-- 
2.28.0

