diff --git a/RenderSystems/GLES2/src/OgreGLES2PixelFormat.cpp b/RenderSystems/GLES2/src/OgreGLES2PixelFormat.cpp
index 0e3f88e785..d43c47a2fa 100644
--- a/RenderSystems/GLES2/src/OgreGLES2PixelFormat.cpp
+++ b/RenderSystems/GLES2/src/OgreGLES2PixelFormat.cpp
@@ -276,6 +276,9 @@ namespace Ogre {
                 return GL_UNSIGNED_SHORT_5_5_5_1;
 
 #if OGRE_ENDIAN == OGRE_ENDIAN_BIG
+#ifndef GL_UNSIGNED_INT_8_8_8_8_REV
+#define GL_UNSIGNED_INT_8_8_8_8_REV	0x8367
+#endif
             case PF_X8B8G8R8:
             case PF_X8R8G8B8:
             case PF_A8B8G8R8:
