diff --git a/OgreMain/CMakeLists.txt b/OgreMain/CMakeLists.txt
index d4ad520ddd..df959a96ae 100644
--- a/OgreMain/CMakeLists.txt
+++ b/OgreMain/CMakeLists.txt
@@ -481,7 +481,12 @@ use_precompiled_header(${OGRE_NEXT}Main
 # install ${OGRE_NEXT}Main
 ogre_config_lib(${OGRE_NEXT}Main TRUE)
 foreach(HEADER_FILE ${HEADER_FILES})
-	string(REGEX REPLACE "((${CMAKE_CURRENT_SOURCE_DIR}|${OGRE_BINARY_DIR})/)?(include|src)/" "" RELATIVE_HEADER_FILE ${HEADER_FILE})
+  string(REGEX REPLACE "((${CMAKE_CURRENT_SOURCE_DIR}|${OGRE_BINARY_DIR})/)?(include|src)/" "" RELATIVE_HEADER_FILE ${HEADER_FILE})
+  cmake_path(GET HEADER_FILE PARENT_PATH FILE_DIR)
+  cmake_path(COMPARE ${OGRE_BINARY_DIR}/include EQUAL ${FILE_DIR} IN_OGRE_BINARY_DIR)
+  if (IS_ABSOLUTE ${HEADER_FILE} AND NOT IN_OGRE_BINARY_DIR)
+    file(RELATIVE_PATH RELATIVE_HEADER_FILE ${CMAKE_CURRENT_SOURCE_DIR}/include ${HEADER_FILE})
+  endif()
 	string(REGEX MATCH "((.*)/)+" HEADER_SUBDIRECTORY ${RELATIVE_HEADER_FILE})
 	install(FILES ${HEADER_FILE} DESTINATION include/${OGRE_NEXT_PREFIX}/${HEADER_SUBDIRECTORY})
 endforeach()
