From 2b2f51ae55f653df672c9db7ab4e766b220d85b5 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sat, 12 Apr 2025 12:49:44 +0800
Subject: [PATCH 2/5] fix(build_cli.sh): drop hard-coded cross compiler setting

ARM != cross compile.
---
 build_cli.sh | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/build_cli.sh b/build_cli.sh
index 4e2056b..daa77c9 100755
--- a/build_cli.sh
+++ b/build_cli.sh
@@ -65,12 +65,6 @@ else
   if [[ "${VSCODE_ARCH}" == "arm64" ]]; then
     VSCODE_CLI_TARGET="aarch64-unknown-linux-gnu"
 
-    if [[ "${CI_BUILD}" != "no" ]]; then
-      export CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-linux-gnu-gcc
-      export CC_aarch64_unknown_linux_gnu=aarch64-linux-gnu-gcc
-      export CXX_aarch64_unknown_linux_gnu=aarch64-linux-gnu-g++
-      export PKG_CONFIG_ALLOW_CROSS=1
-    fi
   elif [[ "${VSCODE_ARCH}" == "armhf" ]]; then
     VSCODE_CLI_TARGET="armv7-unknown-linux-gnueabihf"
 
-- 
2.52.0

