From dc9e1b01edb196d6e48bc363b4d14b336a176b19 Mon Sep 17 00:00:00 2001
From: darkyzhou <me@zqy.io>
Date: Fri, 26 Sep 2025 00:24:11 +0800
Subject: [PATCH 4/4] fix: replace tsgo package when SHOULD_REPLACE_TSGO is set

---
 build.sh | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/build.sh b/build.sh
index bebcbba..49def40 100755
--- a/build.sh
+++ b/build.sh
@@ -12,6 +12,15 @@ if [[ "${SHOULD_BUILD}" == "yes" ]]; then
 
   cd vscode || { echo "'vscode' dir not found"; exit 1; }
 
+  if [[ "${SHOULD_REPLACE_TSGO,,}" == "true" ]]; then
+    jq '
+      del(.devDependencies."@typescript/native-preview")
+      | .devDependencies."@loongdotjs/typescript-native-preview" = "7.0.0-dev.20250926.1"
+    ' package.json > package.json.tmp
+    mv package.json.tmp package.json
+    npm i
+  fi
+
   export NODE_OPTIONS="--max-old-space-size=8192"
 
   npm run monaco-compile-check
-- 
2.50.1

