From 987284e565aa9bfca56f68b1ec7e8b13ae7b20f2 Mon Sep 17 00:00:00 2001
From: Kay Lin <i@v2bv.net>
Date: Sat, 27 Mar 2021 19:56:34 -0700
Subject: [PATCH] [client] shader.h: Include stddef.h

Signed-off-by: Kay Lin <i@v2bv.net>
---
 client/renderers/egl/shader.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/client/renderers/egl/shader.h b/client/renderers/egl/shader.h
index 6b5ae65..642f152 100644
--- a/client/renderers/egl/shader.h
+++ b/client/renderers/egl/shader.h
@@ -19,6 +19,7 @@ Place, Suite 330, Boston, MA 02111-1307 USA
 
 #pragma once
 
+#include <stddef.h>
 #include <stdbool.h>
 
 #include <GL/gl.h>
@@ -33,4 +34,4 @@ bool egl_shader_compile(EGL_Shader * model, const char * vertex_code, size_t ver
 void egl_shader_use    (EGL_Shader * shader);
 
 void egl_shader_associate_textures(EGL_Shader * shader, const int count);
-GLint egl_shader_get_uniform_location(EGL_Shader * shader, const char * name);
\ No newline at end of file
+GLint egl_shader_get_uniform_location(EGL_Shader * shader, const char * name);
-- 
2.30.2

