From: Iain Lane <laney@debian.org>
Date: Fri, 1 Apr 2016 00:18:20 +0800
Subject: Find 4.5 versions of required assemblies

---
 build/m4/shamrock/mono.m4 | 5 +++++
 configure.ac              | 4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/build/m4/shamrock/mono.m4 b/build/m4/shamrock/mono.m4
index c40ecbf..4d8a13f 100644
--- a/build/m4/shamrock/mono.m4
+++ b/build/m4/shamrock/mono.m4
@@ -87,6 +87,11 @@ AC_DEFUN([SHAMROCK_CHECK_MONO_4_0_GAC_ASSEMBLIES],
 	_SHAMROCK_CHECK_MONO_GAC_ASSEMBLIES(mono, 4.0, $*)
 ])
 
+AC_DEFUN([SHAMROCK_CHECK_MONO_4_5_GAC_ASSEMBLIES],
+[
+	_SHAMROCK_CHECK_MONO_GAC_ASSEMBLIES(mono, 4.5, $*)
+])
+
 AC_DEFUN([SHAMROCK_CHECK_MONO2_4_0_GAC_ASSEMBLIES],
 [
 	_SHAMROCK_CHECK_MONO_GAC_ASSEMBLIES(mono-2, 4.0, $*)
diff --git a/configure.ac b/configure.ac
index 7926526..28d0380 100644
--- a/configure.ac
+++ b/configure.ac
@@ -65,9 +65,9 @@ SHAMROCK_CHECK_GNOME_DOC_UTILS(0.17.3)
 
 dnl Mono and gmcs
 SHAMROCK_CHECK_MONO_MODULE(2.4.3)
-SHAMROCK_FIND_MONO_2_0_COMPILER
+SHAMROCK_FIND_MONO_1_0_COMPILER
 SHAMROCK_FIND_MONO_RUNTIME
-SHAMROCK_CHECK_MONO_2_0_GAC_ASSEMBLIES([Mono.Cairo Mono.Posix ICSharpCode.SharpZipLib])
+SHAMROCK_CHECK_MONO_4_5_GAC_ASSEMBLIES([Mono.Cairo Mono.Posix ICSharpCode.SharpZipLib])
 
 dnl In MeeGo we want the full name, but not in GNOME 3 (see bgo#596242)
 PKG_CHECK_MODULES(MEEGO_PANEL, meego-panel, HAVE_MEEGO_PANEL=yes, HAVE_MEEGO_PANEL=no)
