diff -Naur a/configure.in b/configure.in
--- a/configure.in	2016-03-10 20:12:57.354722626 -0700
+++ b/configure.in	2016-03-10 20:13:37.598091365 -0700
@@ -9,7 +9,6 @@
 AC_INIT([libglade],[2.6.4],
         [http://bugzilla.gnome.org/enter_bug.cgi?product=libglade])
 
-AC_CONFIG_SRCDIR([test-libglade.c])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_HEADERS([config.h])
 
@@ -169,7 +168,6 @@
 glade/Makefile
 doc/Makefile
 doc/version.xml
-tests/Makefile
 libglade-2.0.pc
 libglade-2.0-uninstalled.pc
 ])
diff -Naur a/Makefile.am b/Makefile.am
--- a/Makefile.am	2016-03-10 20:12:57.354722626 -0700
+++ b/Makefile.am	2016-03-10 20:13:16.531387928 -0700
@@ -1,4 +1,4 @@
-SUBDIRS = glade doc tests
+SUBDIRS = glade doc
 
 ACLOCAL_AMFLAGS = -I m4
 
@@ -14,11 +14,6 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libglade-2.0.pc
 
-noinst_PROGRAMS = test-libglade
-
-test_libglade_LDADD = glade/libglade-2.0.la $(LIBGLADE_LIBS)
-test_libglade_SOURCES = test-libglade.c
-
 # handling of the XML catalog
 
 xmldir = $(datadir)/xml/libglade
