--- a/src/utilities/file.h	2023-10-09 13:34:30.041157143 -0700
+++ b/src/utilities/file.h	2023-10-09 13:34:47.023739092 -0700
@@ -32,6 +32,7 @@
 
 #include <iosfwd>
 #include <string>
+#include <memory>
 #include <vector>
 
 class Gameexe;
--- a/src/systems/base/system.h	2023-10-09 13:33:19.217729263 -0700
+++ b/src/systems/base/system.h	2023-10-09 13:33:44.946611436 -0700
@@ -36,6 +36,7 @@
 #include <sstream>
 #include <string>
 #include <utility>
+#include <memory>
 #include <vector>
 
 class GraphicsSystem;
