From: Ying-Chun Liu (PaulLiu) <paulliu@debian.org>
Subject: Use system library instead of internal static link
Last-Update: 2011-02-06
Index: rlvm-0.14/SConscript.luarlvm
===================================================================
--- rlvm-0.14.orig/SConscript.luarlvm
+++ rlvm-0.14/SConscript.luarlvm
@@ -53,7 +53,7 @@ if env['BUILD_LUA_TESTS'] == True:
   test_env.Append(CPPPATH = [ "/usr/include/lua5.1" ] )
 
   # Build our included copy of luabind.
-  test_env.BuildSubcomponent("luabind")
+  test_env.ParseConfig ("pkg-config luabind --cflags --libs")
 
   test_env.RlvmProgram("lua_rlvm", ['test/lua_rlvm.cc', script_machine_files],
                        use_lib_set = ["SDL", "LUA"],
