From 80c6c553207b6debabf3aef064a542512330076e Mon Sep 17 00:00:00 2001
From: Konstantin <ria.freelander@gmail.com>
Date: Thu, 12 Mar 2020 01:27:11 +0300
Subject: [PATCH] vapi: fix libxfce4panel vapi

---
 vapi/libxfce4panel-2.0.vapi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vapi/libxfce4panel-2.0.vapi b/vapi/libxfce4panel-2.0.vapi
index 6dccb65..f1700a1 100644
--- a/vapi/libxfce4panel-2.0.vapi
+++ b/vapi/libxfce4panel-2.0.vapi
@@ -101,7 +101,7 @@ namespace Xfce {
         public virtual signal void mode_changed (Xfce.PanelPluginMode mode);
         public virtual signal void nrows_changed (uint rows);
         public virtual signal void orientation_changed (Gtk.Orientation orientation);
-        public virtual signal bool remote_event (string name, GLib.Value value);
+        public virtual signal bool remote_event (string name, GLib.Value value, uint handle);
         public virtual signal void removed ();
         public virtual signal void save ();
         public virtual signal void screen_position_changed (Xfce.ScreenPosition position);
-- 
2.26.2

