diff --git a/CorsixTH/Lua/config_finder.lua b/CorsixTH/Lua/config_finder.lua
index 25d505f3..0b98500c 100644
--- a/CorsixTH/Lua/config_finder.lua
+++ b/CorsixTH/Lua/config_finder.lua
@@ -372,7 +372,7 @@ if needs_rewrite then
 -- Specify a font file here if you wish to play the game in a language not
 -- present in the original game. Examples include Russian, Chinese and Polish.
 --
-unicode_font = nil -- [[X:\ThemeHospital\font.ttc]]
+unicode_font = [[/usr/share/fonts/TTF/wqy-microhei.ttc]]
 
 -------------------------------------------------------------------------------------------------------------------------
 -- Savegames. By default, the "Saves" directory alongside this config file will
