--- linphone/configure.ac	2017-09-13 14:53:38.876212665 -0500
+++ linphone.fixver/configure.ac	2017-09-13 14:54:59.759667352 -0500
@@ -1,6 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
 
-AC_INIT([linphone],[3.11.2],[linphone-developers@nongnu.org])
+AC_INIT([linphone],[3.12.0],[linphone-developers@nongnu.org])
 AC_CANONICAL_SYSTEM
 AC_CONFIG_SRCDIR([coreapi/linphonecore.c])
 
