From 0dd900a5af074ea5db8ef3d5521bd40bef379253 Mon Sep 17 00:00:00 2001
From: "maple@max" <maple@max>
Date: Mon, 27 Jan 2025 18:42:39 +0800
Subject: [PATCH 2/3] support build from source on risc64

---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 6df88f2c..f34e7eeb 100644
--- a/Makefile
+++ b/Makefile
@@ -78,6 +78,8 @@ else ifeq ($(LOCAL_ARCH),s390x)
     TARGET_ARCH ?= s390x
 else ifeq ($(LOCAL_ARCH),loongarch64)
     TARGET_ARCH ?= loong64
+else ifeq ($(LOCAL_ARCH),riscv64)
+    TARGET_ARCH ?= riscv64
 else
     $(error This system's architecture $(LOCAL_ARCH) isn't supported)
 endif
-- 
2.50.0

