From ca358a46e9ff59a6a86b8487e506f67876cd8bfb Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Sat, 28 Jun 2025 02:34:50 +0000
Subject: [PATCH 3/3] add ppc64le and mips64 architecture support

---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index f34e7eeb..27086ced 100644
--- a/Makefile
+++ b/Makefile
@@ -80,6 +80,10 @@ else ifeq ($(LOCAL_ARCH),loongarch64)
     TARGET_ARCH ?= loong64
 else ifeq ($(LOCAL_ARCH),riscv64)
     TARGET_ARCH ?= riscv64
+else ifeq ($(LOCAL_ARCH),mips64)
+    TARGET_ARCH ?= mips64
+else ifeq ($(LOCAL_ARCH),ppc64le)
+    TARGET_ARCH ?= ppc64le
 else
     $(error This system's architecture $(LOCAL_ARCH) isn't supported)
 endif
-- 
2.50.0

