From ea00c6c3d1d03791268addab8e014915b7efa961 Mon Sep 17 00:00:00 2001
From: Neurognostic <neurognostic@astranetics.com>
Date: Fri, 8 Aug 2025 23:22:35 -0600
Subject: [PATCH] Install in to the appropriate man page section

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e8605cb..56ec81f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,7 +63,7 @@ target_include_directories(httping PUBLIC "${PROJECT_BINARY_DIR}")
 
 install(TARGETS httping DESTINATION bin)
 install(FILES README.md LICENSE plot-json.py DESTINATION ${CMAKE_INSTALL_DOCDIR})
-install(FILES httping.1 DESTINATION ${CMAKE_INSTALL_MANDIR})
+install(FILES httping.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
 
 # setup summary
 message(STATUS "CMake version: ${CMAKE_VERSION}")
-- 
2.52.0

