From 4b91cdba76b511dcc7267e511a7b1db3cfb86f6b Mon Sep 17 00:00:00 2001
From: Stefan Ritter <xeno@thehappy.de>
Date: Sat, 4 Apr 2026 21:10:44 +0800
Subject: [PATCH 01/12] DEBIAN: Fixes to make it debian compatible

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 Makefile     | 25 ++++++++++++-------------
 doc/Makefile | 16 ++++++++--------
 2 files changed, 20 insertions(+), 21 deletions(-)

diff --git a/Makefile b/Makefile
index db019cc..1c710cb 100644
--- a/Makefile
+++ b/Makefile
@@ -14,7 +14,7 @@
 # MAKE = make
 
 # make NetHack
-PREFIX	 = /usr
+PREFIX	 = $(DESTDIR)/usr
 GAME     = glhack
 # GAME     = glhack.prg
 GAMEUID  = root
@@ -38,8 +38,8 @@ VARDIRPERM  = 02775
 # (if there is, you'll have to do the installation by hand or modify the
 # instructions)
 GAMEDIR  = $(PREFIX)/lib/games/$(GAME)
-VARDIR   = /var/lib/games/glhack
-SHELLDIR = $(PREFIX)/bin
+VARDIR   = $(DESTDIR)/var/games/$(GAME)
+SHELLDIR = $(PREFIX)/games
 MANDIR   = $(PREFIX)/share/man
 
 # per discussion in Install.X11 and Install.Qt
@@ -54,10 +54,7 @@ MANDIR   = $(PREFIX)/share/man
 # for GL
 VARDATND = 
 
-DATGL = glfont8.png glfont14.png glfont20.png glfont22.png  \
-    gltile16.png gltile32.png gltile64.png  \
-    gllogo.png glrip.png  \
-    glface16.lst glface32.lst
+DATGL = glface16.lst glface32.lst
 
 VARDATD = data oracles options quest.dat rumors
 VARDAT = $(VARDATD) $(VARDATND)
@@ -89,7 +86,7 @@ SPEC_LEVS = asmodeus.lev baalz.lev bigrm-?.lev castle.lev fakewiz?.lev \
 	astral.lev air.lev earth.lev fire.lev water.lev
 QUEST_LEVS = ???-goal.lev ???-fil?.lev ???-loca.lev ???-strt.lev
 
-DATNODLB = $(VARDATND) license
+DATNODLB = $(VARDATND)
 # GL Hack
 DATNODLB += $(DATGL)
 
@@ -202,11 +199,13 @@ dofiles:
 	cp util/recover_glhack $(GAMEDIR)
 	-rm -f $(SHELLDIR)/$(GAME)
 # [AJA] remove any old glhack binary (location changed in 1.2)
-	-rm -f $(PREFIX)/games/$(GAME)
-	sed -e 's;/usr/games/lib/nethackdir;$(GAMEDIR);' \
-		-e 's;HACKDIR/nethack;HACKDIR/$(GAME);' \
-		< win/gl/glhack.sh \
-		> $(SHELLDIR)/$(GAME)
+#	-rm -f $(PREFIX)/games/$(GAME)
+#	sed -e 's;/usr/games/lib/nethackdir;$(GAMEDIR);' \
+#		-e 's;HACKDIR/nethack;HACKDIR/$(GAME);' \
+#		< win/gl/glhack.sh \
+#		> $(SHELLDIR)/$(GAME)
+	cp win/gl/glhack.sh $(SHELLDIR)/$(GAME)
+
 # set up their permissions
 	-( cd $(GAMEDIR) ; $(CHOWN) $(GAMEUID) $(GAME) recover_glhack ; \
 			$(CHGRP) $(GAMEGRP) $(GAME) recover_glhack )
diff --git a/doc/Makefile b/doc/Makefile
index 076b2ba..76967f8 100644
--- a/doc/Makefile
+++ b/doc/Makefile
@@ -20,7 +20,7 @@ PSCMD = groff
 
 # Use the "cat" GUIDECMD if nroff and/or tbl and/or col are not installed
 # GUIDECMD = cat Guidebook.txt
-GUIDECMD = tbl tmac.n Guidebook.mn | nroff | $(COLCMD)
+GUIDECMD = tbl tmac.n Guidebook.mn | nroff -Tascii | $(COLCMD)
 
 # the basic guidebook
 Guidebook:	Guidebook.mn
@@ -38,7 +38,7 @@ Guidebook.dvi:	Guidebook.tex
 
 
 GAME	= nethack
-MANDIR	= /usr/man/man6
+MANDIR	= $(DESTDIR)$(PREFIX)/share/man/man$(MANEXT)
 MANEXT	= 6
 
 # manual installation for most BSD-style systems
@@ -68,17 +68,17 @@ distrib: $(DISTRIB)
 	@echo "Plain text documentation is up to date."
 
 Guidebook.txt	: Guidebook.mn tmac.n
-	tbl tmac.n Guidebook.mn | nroff | $(COLCMD) > Guidebook.txt
+	tbl tmac.n Guidebook.mn | nroff -Tascii | $(COLCMD) > Guidebook.txt
 nethack.txt	: nethack.6
-	nroff -man nethack.6 | $(COLCMD) > nethack.txt
+	nroff -Tascii -man nethack.6 | $(COLCMD) > nethack.txt
 lev_comp.txt	: lev_comp.6
-	nroff -man lev_comp.6 | $(COLCMD) > lev_comp.txt
+	nroff -Tascii -man lev_comp.6 | $(COLCMD) > lev_comp.txt
 dgn_comp.txt	: dgn_comp.6
-	nroff -man dgn_comp.6 | $(COLCMD) > dgn_comp.txt
+	nroff -Tascii -man dgn_comp.6 | $(COLCMD) > dgn_comp.txt
 recover.txt	: recover.6
-	nroff -man recover.6 | $(COLCMD) > recover.txt
+	nroff -Tascii -man recover.6 | $(COLCMD) > recover.txt
 dlb.txt		: dlb.6
-	nroff -man dlb.6 | $(COLCMD) > dlb.txt
+	nroff -Tascii -man dlb.6 | $(COLCMD) > dlb.txt
 
 
 clean:
-- 
2.52.0

