From d5a3ca32a540ff6fa8528c207fb584c933bcf1d2 Mon Sep 17 00:00:00 2001
From: Stefan Ritter <xeno@thehappy.de>
Date: Sat, 4 Apr 2026 21:10:44 +0800
Subject: [PATCH 02/12] DEBIAN: We're using the .txt variant instead of a
 broken manpage

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1c710cb..fea11f8 100644
--- a/Makefile
+++ b/Makefile
@@ -108,7 +108,7 @@ Guidebook:
 manpages:
 	cp doc/glhack.6 $(MANDIR)/man6
 	cp doc/glhackrc.5 $(MANDIR)/man5
-	cp doc/nethack-guidebook.6 $(MANDIR)/man6
+	# cp doc/nethack-guidebook.6 $(MANDIR)/man6
 	( cd doc ; $(MAKE) manpages )
 
 data: $(GAME)
-- 
2.52.0

