From c963941cb62c6a35044e79bad8c77e0eb0af6b2a Mon Sep 17 00:00:00 2001
From: Stefan Ritter <xeno@thehappy.de>
Date: Sat, 4 Apr 2026 21:10:44 +0800
Subject: [PATCH 03/12] DEBIAN: The old script was weird

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 win/gl/glhack.sh | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/win/gl/glhack.sh b/win/gl/glhack.sh
index 5d661b0..d71cfb2 100644
--- a/win/gl/glhack.sh
+++ b/win/gl/glhack.sh
@@ -1,14 +1,3 @@
 #!/bin/sh
-#	SCCS Id: @(#)nethack.sh	3.3	90/02/26
 
-HACKDIR=/usr/games/lib/nethackdir
-export HACKDIR
-HACK=$HACKDIR/nethack
-
-# -AJA- removed HACKPAGER stuff, unused in glHack.
-# 
-# Note: the above HACKDIR directory gets modified to the actual
-#       directory by the top-level Makefile during installation.
-
-cd $HACKDIR
-exec $HACK "$@"
+exec /usr/lib/games/glhack/glhack
-- 
2.52.0

