From 469f131da800070b074750f8259f8d22879b91f7 Mon Sep 17 00:00:00 2001
From: Stefan Ritter <xeno@thehappy.de>
Date: Sat, 4 Apr 2026 21:10:44 +0800
Subject: [PATCH 05/12] DEBIAN: Fixed path to vardir

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 include/unixconf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/unixconf.h b/include/unixconf.h
index 3899644..d0bc0b3 100644
--- a/include/unixconf.h
+++ b/include/unixconf.h
@@ -102,7 +102,7 @@
  * If you want the static parts of your playground on a read-only file
  * system, define VAR_PLAYGROUND to be where the variable parts are kept.
  */
-#define VAR_PLAYGROUND "/var/lib/games/glhack"
+#define VAR_PLAYGROUND "/var/games/glhack"
 
 
 /*
-- 
2.52.0

