From 75d5da7094cf4f781f2b41fd20a703da59466376 Mon Sep 17 00:00:00 2001
From: Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
Date: Sat, 4 Apr 2026 21:10:44 +0800
Subject: [PATCH 08/12] DEBIAN: FTBFS by libpng 1.5 (Closes: #649948)

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 win/gl/gl_image.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/win/gl/gl_image.c b/win/gl/gl_image.c
index e2316e0..fba8b14 100644
--- a/win/gl/gl_image.c
+++ b/win/gl/gl_image.c
@@ -486,7 +486,11 @@ unsigned char *sdlgl_load_png_file(const char *filename,
   /* set error handling since we are using the setjmp/longjmp method
    * (this is the normal method of doing things with libpng).
    */
+#if PNG_LIBPNG_VER_MAJOR >= 1 && PNG_LIBPNG_VER_MINOR >= 4
+  if ( setjmp(png_jmpbuf(png_ptr)) )
+#else
   if (setjmp(png_ptr->jmpbuf))
+#endif
   {
     sdlgl_warning("Problem within LibPNG (unknown)\n");
     goto failed;
-- 
2.52.0

