From fcbe27d17f0750c640d8f0489a461fb44ec6a505 Mon Sep 17 00:00:00 2001
From: Santiago Vila <sanvila@debian.org>
Date: Sat, 4 Apr 2026 21:10:44 +0800
Subject: [PATCH 09/12] DEBIAN: Do not define or use the build date

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 util/makedefs.c | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/util/makedefs.c b/util/makedefs.c
index 80eb06d..b3d1f1f 100644
--- a/util/makedefs.c
+++ b/util/makedefs.c
@@ -167,7 +167,7 @@ extern void NDECL(objects_init);	/* objects.c */
 
 static void NDECL(make_version);
 static char *FDECL(version_string, (char *));
-static char *FDECL(version_id_string, (char *,const char *));
+static char *FDECL(version_id_string, (char *));
 static char *FDECL(xcrypt, (const char *));
 static int FDECL(check_control, (char *));
 static char *FDECL(without_control, (char *));
@@ -524,9 +524,8 @@ char *outbuf;
 }
 
 static char *
-version_id_string(outbuf, build_date)
+version_id_string(outbuf)
 char *outbuf;
-const char *build_date;
 {
     char subbuf[64], versbuf[64];
 
@@ -539,8 +538,8 @@ const char *build_date;
     Strcat(subbuf, " Beta");
 #endif
 
-    Sprintf(outbuf, "%s NetHack%s Version %s - last build %s.",
-	    PORT_ID, subbuf, version_string(versbuf), build_date);
+    Sprintf(outbuf, "%s NetHack%s Version %s.",
+	    PORT_ID, subbuf, version_string(versbuf));
     return outbuf;
 }
 
@@ -572,8 +571,6 @@ do_date()
 #endif
 	for (c = cbuf; *c; c++) if (*c == '\n') break;
 	*c = '\0';	/* strip off the '\n' */
-	Fprintf(ofp,"#define BUILD_DATE \"%s\"\n", cbuf);
-	Fprintf(ofp,"#define BUILD_TIME (%ldL)\n", clocktim);
 	Fprintf(ofp,"\n");
 #ifdef NHSTDC
 	ul_sfx = "UL";
@@ -595,7 +592,7 @@ do_date()
 	Fprintf(ofp,"\n");
 	Fprintf(ofp,"#define VERSION_STRING \"%s\"\n", version_string(buf));
 	Fprintf(ofp,"#define VERSION_ID \\\n \"%s\"\n",
-		version_id_string(buf, cbuf));
+		version_id_string(buf));
 	Fprintf(ofp,"\n");
 #ifdef AMIGA
 	{
-- 
2.52.0

