diff -Naur glhack-1.2/win/gl/gl_image.c glhack-1.2.patched/win/gl/gl_image.c
--- glhack-1.2/win/gl/gl_image.c	2002-12-15 19:21:35.000000000 +0800
+++ glhack-1.2.patched/win/gl/gl_image.c	2015-12-10 19:23:31.603781175 +0800
@@ -486,7 +486,7 @@
   /* set error handling since we are using the setjmp/longjmp method
    * (this is the normal method of doing things with libpng).
    */
-  if (setjmp(png_ptr->jmpbuf))
+  if (setjmp(png_jmpbuf(png_ptr)))
   {
     sdlgl_warning("Problem within LibPNG (unknown)\n");
     goto failed;
