diff --git a/core/utilities/setup/downloader/filesdownloader.cpp b/core/utilities/setup/downloader/filesdownloader.cpp
index bfe2792236..98bcdb0a71 100644
--- a/core/utilities/setup/downloader/filesdownloader.cpp
+++ b/core/utilities/setup/downloader/filesdownloader.cpp
@@ -438,8 +438,7 @@ void FilesDownloader::slotDownloadProgress(qint64 bytesReceived, qint64 bytesTot
 
 QString FilesDownloader::getFacesEnginePath() const
 {
-    QString appPath = QStandardPaths::writableLocation(QStandardPaths::AppDataLocation);
-    QUrl    appUrl  = QUrl::fromLocalFile(appPath).adjusted(QUrl::RemoveFilename);
+    QUrl    appUrl  = QUrl::fromLocalFile(QLatin1String("/usr/share/"));
     appUrl.setPath(appUrl.path() + QLatin1String("digikam/facesengine"));
 
     return appUrl.toLocalFile();
