From ba355ed44a541876d2cb3323dd309ce8ac48fd13 Mon Sep 17 00:00:00 2001
From: Anjia Wang <anjiawang@gmail.com>
Date: Wed, 14 Aug 2024 05:28:47 +0000
Subject: [PATCH 3/4] fix(bootstrap): remove automake version restriction

---
 bootstrap | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bootstrap b/bootstrap
index 39480ed..c6c1002 100644
--- a/bootstrap
+++ b/bootstrap
@@ -8,7 +8,7 @@ aclocal
 autoheader
 #add --include-deps if you want to bootstrap with any other compiler than gcc
 #automake --add-missing --copy --include-deps
-automake-1.9 --add-missing --force --copy
+automake --add-missing --force --copy
 #autoconf2.50
 autoconf
 rm -f config.cache
-- 
2.46.0

