From 79e2e6a8e73669d82bd0743d56c451d9e72126b8 Mon Sep 17 00:00:00 2001
From: Anjia Wang <anjiawang@gmail.com>
Date: Wed, 14 Aug 2024 06:09:20 +0000
Subject: [PATCH 4/4] fix(configure.ac): remove unneeded -m64 flag

---
 configure.ac | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index ddfed18..ad31f10 100644
--- a/configure.ac
+++ b/configure.ac
@@ -201,11 +201,7 @@ AC_ARG_ENABLE(64-bit, [  --enable-64-bit          Generate code for a 64-bit env
               b64="$enableval", b64="no")
 if test x"$b64" != "xno"; then
     AC_MSG_RESULT(Configured to Building 64-bit)
-    if test x"$GCC" = xyes; then
-      CFLAGS="-DJU_64BIT -m64 $CFLAGS"
-    else
-      CFLAGS="-DJU_64BIT $CFLAGS"
-    fi
+    CFLAGS="-DJU_64BIT $CFLAGS"
 fi
 
 
-- 
2.46.0

