diff -Naur kube-0.8.0/CMakeLists.txt kube-0.8.0.notest/CMakeLists.txt
--- kube-0.8.0/CMakeLists.txt	2019-09-15 09:58:13.000000000 +0000
+++ kube-0.8.0.notest/CMakeLists.txt	2019-11-09 20:31:35.061340492 +0000
@@ -66,7 +66,6 @@
 add_subdirectory(applications)
 add_subdirectory(views)
 add_subdirectory(accounts)
-add_subdirectory(tests)
 if (${ENABLE_EXTENSIONS})
     add_subdirectory(extensions)
 endif()
diff -Naur kube-0.8.0/framework/src/CMakeLists.txt kube-0.8.0.notest/framework/src/CMakeLists.txt
--- kube-0.8.0/framework/src/CMakeLists.txt	2019-09-15 09:58:13.000000000 +0000
+++ kube-0.8.0.notest/framework/src/CMakeLists.txt	2019-11-09 20:30:36.282143801 +0000
@@ -102,7 +102,6 @@
 
 set(BUILD_TESTING ON)
 
-add_subdirectory(tests)
 add_subdirectory(domain/mime)
 add_subdirectory(domain/mime/tests)
 add_subdirectory(domain/mime/mimetreeparser)
