diff -Naur icewm-1.4.2/lib/keys.in icewm-1.4.2.aosc/lib/keys.in
--- icewm-1.4.2/lib/keys.in	2017-07-30 03:53:35.524267864 -0500
+++ icewm-1.4.2.aosc/lib/keys.in	2020-04-12 00:56:54.401801125 -0500
@@ -9,7 +9,7 @@
 # You'll have to omit XK_ prefixs and to replace XF86XK_ prefixes by
 # XF86. Valid modifiers are Alt, Ctrl, Shift, Meta, Super and Hyper.
 #
-key "Alt+Ctrl+t"			xterm
+key "Alt+Ctrl+t"			urxvt
 key "Alt+Ctrl+b"			xdg-open about:blank
 key "Alt+Ctrl+s"			xdg-open http://www.google.com
 
diff -Naur icewm-1.4.2/lib/menu.in icewm-1.4.2.aosc/lib/menu.in
--- icewm-1.4.2/lib/menu.in	2017-07-30 03:53:35.524267864 -0500
+++ icewm-1.4.2.aosc/lib/menu.in	2020-04-12 00:55:14.305466543 -0500
@@ -4,14 +4,10 @@
 # since modifications to this file will be discarded when you
 # (re)install icewm.
 #
-prog xterm xterm xterm
-prog rxvt xterm rxvt -bg black -cr green -fg white -C -fn 9x15 -sl 500
-prog fte fte fte
-prog NEdit nedit nedit
-prog Mozilla mozilla mozilla
-prog XChat xchat xchat
-prog Gimp gimp gimp
+prog "W_eb Browser" netsurf netsurf-gtk
+prog "_File Manager" xfe xfe
+prog "_Media Player" xmms xmms
 separator
-menuprog "Desktop Apps" folder icewm-menu-fdo
-menufile Programs folder programs
-menufile Tool_bar folder toolbar
+menufile "_Programs" folder programs
+menufile "_System Settings" folder sysutils
+menufile "Tool_bar" folder toolbar
diff -Naur icewm-1.4.2/lib/toolbar.in icewm-1.4.2.aosc/lib/toolbar.in
--- icewm-1.4.2/lib/toolbar.in	2017-07-30 03:53:35.524267864 -0500
+++ icewm-1.4.2.aosc/lib/toolbar.in	2020-04-12 00:57:57.089384502 -0500
@@ -4,6 +4,4 @@
 # since modifications to this file will be discarded when you
 # (re)install icewm.
 #
-prog xterm utilities-terminal xterm
-prog "Web browser" web-browser xdg-open about:blank
-
+prog "Web Browser" netsurf netsurf-gtk
diff -Naur icewm-1.4.2/src/default.h icewm-1.4.2.aosc/src/default.h
--- icewm-1.4.2/src/default.h	2017-07-30 03:53:35.524267864 -0500
+++ icewm-1.4.2.aosc/src/default.h	2020-04-12 01:00:52.675173519 -0500
@@ -29,8 +29,8 @@
 XIV(bool, beepOnNewMail,                        false)
 XIV(bool, warpPointer,                          false)
 XIV(bool, warpPointerOnEdgeSwitch,              false)
-XIV(bool, opaqueMove,                           true)
-XIV(bool, opaqueResize,                         true)
+XIV(bool, opaqueMove,                           false)
+XIV(bool, opaqueResize,                         false)
 XIV(bool, hideTitleBarWhenMaximized,            false)
 XSV(const char *, winMenuItems,                 "rmsnxfhualyticw")
 #ifdef CONFIG_TASKBAR
@@ -42,10 +42,10 @@
 XIV(bool, taskBarShowApmAuto,                   true)
 XIV(bool, taskBarShowApmTime,                   true) // mschy
 XIV(bool, taskBarShowApmGraph,                  true) // hatred
-XIV(bool, taskBarShowMailboxStatus,             true)
+XIV(bool, taskBarShowMailboxStatus,             false)
 XIV(bool, taskBarShowStartMenu,                 true)
-XIV(bool, taskBarShowWindowListMenu,            true)
-XIV(bool, taskBarShowWorkspaces,                true)
+XIV(bool, taskBarShowWindowListMenu,            false)
+XIV(bool, taskBarShowWorkspaces,                false)
 XIV(bool, taskBarShowWindows,                   true)
 XIV(bool, taskBarShowShowDesktopButton,         true)
 
@@ -69,14 +69,14 @@
 XIV(bool, pagerShowMinimized,                   true)
 XIV(bool, pagerShowBorders,                     true)
 XIV(bool, pagerShowNumbers,                     true)
-XIV(bool, taskBarShowCPUStatus,                 true)
+XIV(bool, taskBarShowCPUStatus,                 false)
 XIV(bool, cpustatusShowRamUsage,                true)
 XIV(bool, cpustatusShowSwapUsage,               true)
 XIV(bool, cpustatusShowAcpiTemp,                true)
 XIV(bool, cpustatusShowAcpiTempInGraph,         false)
 XIV(bool, cpustatusShowCpuFreq,                 true)
-XIV(bool, taskBarShowMEMStatus,                 true)
-XIV(bool, taskBarShowNetStatus,                 true)
+XIV(bool, taskBarShowMEMStatus,                 false)
+XIV(bool, taskBarShowNetStatus,                 false)
 XIV(bool, taskBarLaunchOnSingleClick,           true)
 XIV(bool, taskBarShowCollapseButton,            false)
 #endif
