diff --git a/src/include/miopen/serializable.hpp b/src/include/miopen/serializable.hpp
index f7563f50be..ccd4ecc24a 100644
--- a/src/include/miopen/serializable.hpp
+++ b/src/include/miopen/serializable.hpp
@@ -27,7 +27,7 @@
 #ifndef GUARD_MLOPEN_SERIALIZABLE_HPP
 #define GUARD_MLOPEN_SERIALIZABLE_HPP
 
-#include <ciso646>
+#include <version>
 #include <miopen/config.h>
 #include <iostream>
 #include <sstream>
