From 5d756620437062ea45b90f40a77dc2f4918190f9 Mon Sep 17 00:00:00 2001
From: Tristan Isham <tristan@isham.co>
Date: Mon, 26 Jan 2026 16:22:42 -0500
Subject: [PATCH] UPSTREAM: upped version

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 cli/meta/version.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cli/meta/version.go b/cli/meta/version.go
index acb84e1..e824d1a 100644
--- a/cli/meta/version.go
+++ b/cli/meta/version.go
@@ -9,7 +9,7 @@ import (
 )
 
 const (
-	VERSION = "v0.8.12"
+	VERSION = "v0.8.13"
 
 	// VERSION = "v0.0.0" // For testing zvm upgrade
 
-- 
2.52.0

