From c28f2df990b317c662e145b27fc132ccbbeb37e5 Mon Sep 17 00:00:00 2001
From: Xeonacid <h.dwwwwww@gmail.com>
Date: Fri, 20 Jun 2025 06:09:24 +0800
Subject: [PATCH 1/2] UPSTREAM: Only enable -march=native and SSE on x86_64
 (#37)

Fixes #35

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 build.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/build.py b/build.py
index 658ada2..64dc979 100644
--- a/build.py
+++ b/build.py
@@ -2,6 +2,7 @@ from __future__ import annotations
 
 import os
 import shutil
+import platform
 
 from pathlib import Path
 
@@ -10,8 +11,11 @@ from setuptools import Distribution
 from setuptools import Extension
 from setuptools.command.build_ext import build_ext
 
+COMPILE_ARGS = ["-O3"]
+# x86_64 on linux, AMD64 on Windows
+if platform.machine() in ['x86_64', 'AMD64']:
+    COMPILE_ARGS += ["-march=native", "-msse", "-msse2", "-mfma", "-mfpmath=sse"]
 
-COMPILE_ARGS = ["-march=native", "-O3", "-msse", "-msse2", "-mfma", "-mfpmath=sse"]
 LINK_ARGS: list[str] = []
 INCLUDE_DIRS: list[str] = []
 LIBRARIES: list[str] = []
-- 
2.52.0

