From 07e5d47fb56102bef51f56fd4e57cf24e73053ac Mon Sep 17 00:00:00 2001
From: Andrew Resch <andrewresch@gmail.com>
Date: Sun, 22 Jun 2025 09:50:26 -0700
Subject: [PATCH 2/2] UPSTREAM: Change default COMPILE_ARGS to just '-O3'

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 build.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/build.py b/build.py
index 64dc979..be03ada 100644
--- a/build.py
+++ b/build.py
@@ -12,9 +12,6 @@ from setuptools import Extension
 from setuptools.command.build_ext import build_ext
 
 COMPILE_ARGS = ["-O3"]
-# x86_64 on linux, AMD64 on Windows
-if platform.machine() in ['x86_64', 'AMD64']:
-    COMPILE_ARGS += ["-march=native", "-msse", "-msse2", "-mfma", "-mfpmath=sse"]
 
 LINK_ARGS: list[str] = []
 INCLUDE_DIRS: list[str] = []
-- 
2.52.0

