--- comm-esr38/mozilla/configure	2015-11-19 07:22:51.000000000 -0700
+++ comm-esr38-post-patch/mozilla/configure	2015-11-25 18:43:33.082176726 -0700
@@ -9158,7 +9158,7 @@
         MOZ_OPTIMIZE_FLAGS="-O2"
     elif test "$GNU_CC" -o "$GNU_CXX"; then
         MOZ_PGO_OPTIMIZE_FLAGS="-O3"
-        MOZ_OPTIMIZE_FLAGS="-Os"
+        MOZ_OPTIMIZE_FLAGS="-O2"
         if test -z "$CLANG_CC"; then
            MOZ_OPTIMIZE_FLAGS="-freorder-blocks $MOZ_OPTIMIZE_FLAGS"
         fi
--- comm-esr38/mozilla/configure.in	2015-11-19 07:22:39.000000000 -0700
+++ comm-esr38-post-patch/mozilla/configure.in	2015-11-25 18:43:43.822177349 -0700
@@ -2139,7 +2139,7 @@
         MOZ_OPTIMIZE_FLAGS="-O2"
     elif test "$GNU_CC" -o "$GNU_CXX"; then
         MOZ_PGO_OPTIMIZE_FLAGS="-O3"
-        MOZ_OPTIMIZE_FLAGS="-Os"
+        MOZ_OPTIMIZE_FLAGS="-O2"
         if test -z "$CLANG_CC"; then
            MOZ_OPTIMIZE_FLAGS="-freorder-blocks $MOZ_OPTIMIZE_FLAGS"
         fi
