diff -up thunderbird-60.7.0/servo/components/style/gecko/regen_atoms.py.rust-1.33-build thunderbird-60.7.0/servo/components/style/gecko/regen_atoms.py
--- thunderbird-60.7.0/servo/components/style/gecko/regen_atoms.py.rust-1.33-build	2019-05-17 14:05:38.000000000 +0200
+++ thunderbird-60.7.0/servo/components/style/gecko/regen_atoms.py	2019-05-20 15:01:06.284881858 +0200
@@ -202,6 +202,7 @@ RULE_TEMPLATE = ('("{atom}") =>\n  '
                  ' }}}};')
 
 MACRO = '''
+/// Returns a static atom by passing the literal string it represents.
 #[macro_export]
 macro_rules! atom {{
 {}
diff -up thunderbird-60.7.0/servo/components/style/properties/properties.mako.rs.rust-1.33-build thunderbird-60.7.0/servo/components/style/properties/properties.mako.rs
--- thunderbird-60.7.0/servo/components/style/properties/properties.mako.rs.rust-1.33-build	2019-05-17 14:05:38.000000000 +0200
+++ thunderbird-60.7.0/servo/components/style/properties/properties.mako.rs	2019-05-20 15:01:06.286881852 +0200
@@ -55,6 +55,7 @@ use style_adjuster::StyleAdjuster;
 
 pub use self::declaration_block::*;
 
+/// Generated
 #[cfg(feature = "gecko")]
 #[macro_export]
 macro_rules! property_name {
@@ -3852,7 +3853,7 @@ impl fmt::Debug for AliasId {
     }
 }
 
-// NOTE(emilio): Callers are responsible to deal with prefs.
+/// Callers are responsible to deal with prefs.
 #[macro_export]
 macro_rules! css_properties_accessors {
     ($macro_name: ident) => {
@@ -3875,6 +3876,7 @@ macro_rules! css_properties_accessors {
     }
 }
 
+/// Neco
 #[macro_export]
 macro_rules! longhand_properties_idents {
     ($macro_name: ident) => {
