--- a/src/intio.c.orig	2009-06-28 08:47:34.000000000 -0600
+++ b/src/intio.c	2017-02-17 13:13:01.393089605 -0700
@@ -10,6 +10,7 @@
 
 #include "meataxe.h"
 #include "config.h"
+#include <stdint.h>
 
 MTX_DEFINE_FILE_INFO
 
@@ -43,7 +44,6 @@ MTX_DEFINE_FILE_INFO
 
 int SysReadLong32(FILE *f, long *buf, int n)
 {
-    unsigned char a[4];
     int nread;
 
     /* Check the arguments
@@ -57,24 +57,28 @@ int SysReadLong32(FILE *f, long *buf, in
 
     /* Read <n> long integers
        ---------------------- */
-    if (Mtx_IsX86)
+#if MTX_CONFIG_BIG_ENDIAN
+    for (nread = 0; nread < n; ++nread)
     {
-	nread = (int)fread((char *)buf,sizeof(long),(size_t)n,f);
-	/* printf("nread=%d, error=%d, eof=%d, GetLastError()=%d\n",
-	   nread,ferror(f),feof(f),GetLastError());
-	*/
+	uint32_t a = 0;
+	if (fread(&a,1,sizeof(a),f) != 4) break;
+	*buf = (long)(int32_t)__builtin_bswap32(a);
+	++buf;
     }
-    else
+#elif MTX_CONFIG_LONG64
+    for (nread = 0; nread < n; ++nread)
     {
-    	for (nread = 0; nread < n; ++nread)
-    	{
-	    if (fread(a,1,4,f) != 4) break;
-	    *buf = ((unsigned long)a[0]|((unsigned long)a[1] << 8)|
-		((unsigned long)a[2] << 16)|((long)(char)a[3] << 24));
-	    ++buf;
-    	}
+	int32_t a = 0;
+	if (fread(&a,1,sizeof(int32_t),f) != 4) break;
+	*buf = (long)a;
+	++buf;
     }
-
+#else
+    nread = (int)fread((char *)buf,sizeof(long),(size_t)n,f);
+    /* printf("nread=%d, error=%d, eof=%d, GetLastError()=%d\n",
+       nread,ferror(f),feof(f),GetLastError());
+     */
+#endif
     /* Check for errors
        ---------------- */
     if (ferror(f) && !feof(f))
@@ -104,69 +108,51 @@ int SysWriteLong32(FILE *f, const long *
     unsigned char a[4];
     int nwritten;
 
-    if (Mtx_IsX86)
-	nwritten = fwrite((char *)buf,sizeof(long),n,f);
-    else
-    {
-    	for (nwritten = 0; nwritten < n; ++nwritten)
-    	{
-    	    a[0] = (unsigned char) *buf;
-    	    a[1] = (unsigned char) (*buf >> 8);
-    	    a[2] = (unsigned char) (*buf >> 16);
-    	    a[3] = (unsigned char) (*buf >> 24);
-    	    if (fwrite(a,1,4,f) != 4) break;
-	    ++buf;
-    	}
-    }
-    return nwritten;
-}
-
-
 #if MTX_CONFIG_BIG_ENDIAN
-static void Swap(long *dest, const long *src, int n)
-{
-    for (; n > 0; --n)
+    for (nwritten = 0; nwritten < n; ++nwritten)
     {
-	register long x = *src++;
-#if MTX_CONFIG_LONG32
-	*dest++ =
-	      (x << 24)
-	    + ((x << 8) & 0x00FF0000L)
-	    + ((x >> 8) & 0x0000FF00L)
-	    + (x >> 24);
+	uint32_t a = __builtin_bswap32((uint32_t)(int32_t)*buf);
+	if (fwrite(&a,1,sizeof(a),f) != 4) break;
+	++buf;
+    }
 #elif MTX_CONFIG_LONG64
-	*dest++ =
-	      (x << 56)
-	    + ((x << 40) & 0x00FF000000000000L)
-	    + ((x << 24) & 0x0000FF0000000000L)
-	    + ((x <<  8) & 0x000000FF00000000L)
-	    + ((x >>  8) & 0x00000000FF000000L)
-	    + ((x >> 24) & 0x0000000000FF0000L)
-	    + ((x >> 40) & 0x000000000000FF00L)
-	    + (x >> 56);
+    for (nwritten = 0; nwritten < n; ++nwritten)
+    {
+	int32_t a = (int32_t)*buf;
+	if (fwrite(&a,1,sizeof(a),f) != 4) break;
+	++buf;
+    }
 #else
-#error "sizeof(long) not known - check config.h"
+    nwritten = fwrite((char *)buf,sizeof(long),n,f);
 #endif
-    }
+    return nwritten;
 }
-#endif
 
-#define BLK_SIZE 4096
 
 int SysReadLongX(FILE *f, long *buf, int n_bytes)
 {
-    int n_read = 0;		/* Bytes read so far */
+    int n_read;			/* Bytes read so far */
 #if MTX_CONFIG_BIG_ENDIAN
-    unsigned long tmp[BLK_SIZE];
-    while (n_read < num_elem)
+    size_t to_read;
+    for (n_read = 0; n_read < n_bytes; n_read += to_read)
     {
-	int n_blk = sizeof(tmp);
-	if (n_read + n_blk > n_bytes)
-	    n_blk = n_bytes - n_read;
-	if (fread(tmp,1,n_blk,f) != n_blk)
-	    break;
-	Swap(buf + n_read / sizeof(long),tmp,(n_blk - 1)/sizeof(long) + 1);
-	n_written += n_blk;
+	unsigned long a = 0;
+	if (n_bytes - n_read < sizeof(long))
+	{
+	    to_read = n_bytes - n_read;
+	}
+	else
+	{
+	    to_read = sizeof(long);
+	}
+	to_read = fread(&a,1,to_read,f);
+	if (to_read == 0) break;
+#if MTX_CONFIG_LONG32
+	*buf = (long)__builtin_bswap32(a);
+#else
+	*buf = (long)__builtin_bswap64(a);
+#endif
+	++buf;
     }
 #else
     n_read = fread(buf,1,n_bytes,f);
@@ -177,18 +163,26 @@ int SysReadLongX(FILE *f, long *buf, int
 
 int SysWriteLongX(FILE *f, const long *buf, int n_bytes)
 {
-    int n_written = 0;
+    int n_written;
 #if MTX_CONFIG_BIG_ENDIAN
-    unsigned long tmp[BLK_SIZE];
-    while (n_written < n_bytes)
+    size_t to_write;
+    for (n_written = 0; n_written < n_bytes; n_written += to_write)
     {
-	int n_blk = sizeof(tmp);
-	if (n_written + n_blk > n_bytes)
-	    n_blk = n_bytes - n_written;
-	Swap(tmp,buf + n_written / sizeof(long),(n_blk - 1) / sizeof(long) + 1);
-	if (fwrite(tmp,1,n_blk,f) != n_blk)
-	    break;
-	n_written += n_blk;
+	if (n_bytes - n_written < sizeof(long))
+	{
+	    to_write = n_bytes - n_written;
+	}
+	else
+	{
+	    to_write = sizeof(long);
+	}
+#if MTX_CONFIG_LONG32
+	unsigned long a = __builtin_bswap32((unsigned long)*buf);
+#else
+	unsigned long a = __builtin_bswap64((unsigned long)*buf);
+#endif
+	if (fwrite(&a,1,to_write,f) != to_write) break;
+	++buf;
     }
 #else
     n_written = fwrite(buf,1,n_bytes,f);
