From 179ff837b017d71139193be181d2dbaf355b68ff Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Tue, 16 Dec 2025 10:52:07 +0800
Subject: [PATCH] add missing header

---
 src/arg_int.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/arg_int.c b/src/arg_int.c
index 29c20e5..bc10012 100644
--- a/src/arg_int.c
+++ b/src/arg_int.c
@@ -30,6 +30,7 @@ USA.
 
 #include "argtable2.h"
 #include <limits.h>
+#include <ctype.h>
 
 /* local error codes */
 enum {EMINCOUNT=1,EMAXCOUNT,EBADINT,EOVERFLOW};
-- 
2.52.0

