Bug: https://bugzilla.gnome.org/show_bug.cgi?id=623223
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=651571
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=669423

Index: gnome-settings-daemon-3.15.4/data/org.gnome.settings-daemon.plugins.media-keys.gschema.xml.in.in
===================================================================
--- gnome-settings-daemon-3.15.4.orig/data/org.gnome.settings-daemon.plugins.media-keys.gschema.xml.in.in
+++ gnome-settings-daemon-3.15.4/data/org.gnome.settings-daemon.plugins.media-keys.gschema.xml.in.in
@@ -90,6 +90,11 @@
       <_summary>Stop playback</_summary>
       <_description>Binding to stop playback.</_description>
     </key>
+    <key name="video-out" type="s">
+      <default>'&lt;Super&gt;p'</default>
+      <_summary>Video output</_summary>
+      <_description>Binding to switch the video output device.</_description>
+    </key>
     <key name="volume-down" type="s">
       <default>'XF86AudioLowerVolume'</default>
       <_summary>Volume down</_summary>
Index: gnome-settings-daemon-3.15.4/plugins/media-keys/shortcuts-list.h
===================================================================
--- gnome-settings-daemon-3.15.4.orig/plugins/media-keys/shortcuts-list.h
+++ gnome-settings-daemon-3.15.4/plugins/media-keys/shortcuts-list.h
@@ -81,7 +81,7 @@ static struct {
         { FORWARD_KEY, NULL, N_("Forward"), "XF86AudioForward", SHELL_ACTION_MODE_ALL },
         { REPEAT_KEY, NULL, N_("Repeat"), "XF86AudioRepeat", SHELL_ACTION_MODE_ALL },
         { RANDOM_KEY, NULL, N_("Random Play"), "XF86AudioRandomPlay", SHELL_ACTION_MODE_ALL },
-        { VIDEO_OUT_KEY, NULL, N_("Video Out"), "<Super>p", SHELL_ACTION_MODE_ALL },
+        { VIDEO_OUT_KEY, "video-out", N_("Video Out"), NULL, SHELL_ACTION_MODE_ALL },
         /* Key code of the XF86Display key (Fn-F7 on Thinkpads, Fn-F4 on HP machines, etc.) */
         { VIDEO_OUT_KEY, NULL, N_("Video Out"), "XF86Display", SHELL_ACTION_MODE_ALL },
         /* Key code of the XF86RotateWindows key (present on some tablets) */
