diff --git a/plugins/power/gsd-power-manager.c b/plugins/power/gsd-power-manager.c
index c500fa38..bdbb523e 100644
--- a/plugins/power/gsd-power-manager.c
+++ b/plugins/power/gsd-power-manager.c
@@ -2536,7 +2536,7 @@ iio_proxy_changed (GsdPowerManager *manager)
         /* calculate exponential moving average */
         brightness = manager->ambient_last_absolute * 100.f / manager->ambient_norm_value;
         brightness = MIN (brightness, 100.f);
-        brightness = MAX (brightness, 0.f);
+        brightness = MAX (brightness, 30.f);
         manager->ambient_accumulator = (GSD_AMBIENT_SMOOTH * brightness) +
                 (1.0 - GSD_AMBIENT_SMOOTH) * manager->ambient_accumulator;
 
