diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2016-04-13 06:28:55.000000000 -0600
+++ b/configure.ac	2016-04-16 21:20:02.000000000 -0600
@@ -276,7 +276,7 @@
 
 case $host_os in
   linux*)
-    if test "$host_cpu" = s390 -o "$host_cpu" = s390x; then
+    if test "$host_cpu" = s390 -o "$host_cpu" = s390x -o "$host_cpu" = armv7a -o "$host_cpu" = armv7l -o "$host_cpu" = armv8a -o "$host_cpu" = armv8l; then
       have_wacom=no
     else
       if test x$enable_gudev != xno; then
