Index: gnome-settings-daemon-3.8.6.1/plugins/keyboard/gsd-keyboard-manager.c
===================================================================
--- gnome-settings-daemon-3.8.6.1.orig/plugins/keyboard/gsd-keyboard-manager.c	2014-05-06 00:39:44.000000000 -0400
+++ gnome-settings-daemon-3.8.6.1/plugins/keyboard/gsd-keyboard-manager.c	2014-05-06 00:43:24.886984876 -0400
@@ -583,8 +583,8 @@
         XkbSelectEventDetails (dpy,
                                XkbUseCoreKbd,
                                XkbStateNotify,
-                               XkbModifierLockMask,
-                               XkbModifierLockMask);
+                               XkbModifierLockMask | XkbGroupLockMask,
+                               XkbModifierLockMask | XkbGroupLockMask);
 }
 
 static unsigned
@@ -651,6 +651,11 @@
 		}
 	}
 
+	if (xkbev->state.changed & XkbGroupLockMask) {
+		/* Fix the locked group to the first group if it was changed by a grp: option. */
+		XkbLockGroup (GDK_DISPLAY_XDISPLAY (gdk_display_get_default ()), XkbUseCoreKbd, 0);
+	}
+
         return GDK_FILTER_CONTINUE;
 }
 
