diff -Naur tdm-orig/sys/cmake/SvnVersion.cmake tdm/sys/cmake/SvnVersion.cmake
--- tdm-orig/sys/cmake/SvnVersion.cmake	2020-06-27 14:11:35.327533300 -0600
+++ tdm/sys/cmake/SvnVersion.cmake	2021-02-05 18:10:41.176519311 -0700
@@ -1,18 +1,4 @@
 macro(get_svn_revision OutVar)
-	message("Determining SVN revision")
-	find_program(SVNVERSION_PROGRAM NAMES svnversion)
-	if (SVNVERSION_PROGRAM)
-		execute_process(COMMAND "${SVNVERSION_PROGRAM}" "-c"
-				WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
-				RESULT_VARIABLE SVNVERSION_EXITCODE
-				OUTPUT_VARIABLE ${OutVar}
-				OUTPUT_STRIP_TRAILING_WHITESPACE)
-		if (NOT SVNVERSION_EXITCODE EQUAL 0)
-			message("svnversion failed")
-			set(${OutVar} "NOTFOUND")
-		endif()
-	else()
-		message("svnversion not found")
-		set(${OutVar} "NOTFOUND")
-	endif()
+        message("SVN version skipped")
+        set(${OutVar} "NOTFOUND")
 endmacro()
