diff -Naur thedarkmod/CMakeLists.txt thedarkmod.nostrip/CMakeLists.txt
--- thedarkmod/CMakeLists.txt	2021-05-18 09:20:47.000000000 -0700
+++ thedarkmod.nostrip/CMakeLists.txt	2021-08-27 20:48:14.154896310 -0700
@@ -249,19 +249,6 @@
 	set_target_properties(TheDarkMod PROPERTIES WIN32_EXECUTABLE TRUE)
 elseif (UNIX)
 	target_link_libraries(TheDarkMod pthread dl stdc++fs X11 Xext Xxf86vm)
-	# strip debug symbols from executable and put them into a separate symbols file
-	add_custom_command(
-			TARGET TheDarkMod POST_BUILD
-			COMMAND objcopy --only-keep-debug "$<TARGET_FILE:TheDarkMod>" "$<TARGET_FILE:TheDarkMod>.debug"
-	)
-	add_custom_command(
-			TARGET TheDarkMod POST_BUILD
-			COMMAND strip --strip-debug --strip-unneeded "$<TARGET_FILE:TheDarkMod>"
-	)
-	add_custom_command(
-			TARGET TheDarkMod POST_BUILD
-			COMMAND objcopy "--add-gnu-debuglink=$<TARGET_FILE:TheDarkMod>.debug" "$<TARGET_FILE:TheDarkMod>"
-	)
 endif ()
 
 if (COPY_EXE)
