From 81c169330771049220c3fa1a53787b0ef5a94c45 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Fri, 1 May 2020 20:49:30 +0800
Subject: [PATCH] Fix another #define None in X11/X.h

---
 src/lib/fcitx-utils/keysym.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/lib/fcitx-utils/keysym.h b/src/lib/fcitx-utils/keysym.h
index 3b4cce3..1c1c013 100644
--- a/src/lib/fcitx-utils/keysym.h
+++ b/src/lib/fcitx-utils/keysym.h
@@ -30,6 +30,11 @@
 #include <fcitx-utils/macros.h>

 namespace fcitx {
+
+#ifdef None
+#undef None
+#endif
+
 /// \brief KeyState to represent modifier keys.
 enum class KeyState : uint32_t {
     None = 0,
--
2.26.2

