From bd83a9841e1dc3e7296174d32f5ee7655f680689 Mon Sep 17 00:00:00 2001
From: Weng Xuetian <wengxt@gmail.com>
Date: Sun, 24 May 2020 13:39:25 -0700
Subject: [PATCH] Fix the wrong order when set rgb (was rbg).

Fix #65
---
 src/ui/classic/theme.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ui/classic/theme.h b/src/ui/classic/theme.h
index dd044bd..67b10dc 100644
--- a/src/ui/classic/theme.h
+++ b/src/ui/classic/theme.h
@@ -165,7 +165,7 @@ private:
 };
 
 inline void cairoSetSourceColor(cairo_t *cr, const Color &color) {
-    cairo_set_source_rgba(cr, color.redF(), color.blueF(), color.greenF(),
+    cairo_set_source_rgba(cr, color.redF(), color.greenF(), color.blueF(),
                           color.alphaF());
 }
 } // namespace classicui
-- 
2.27.0

