From c20a5119b8e908b0fc9de39d8cf982765f5af21f Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 20 Feb 2024 12:32:22 +0800
Subject: [PATCH 1/2] AUR: hack to disable floating shadow for WeChat

---
 dlls/user32/win.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/dlls/user32/win.c b/dlls/user32/win.c
index b8c04047333..0aaf027b0db 100644
--- a/dlls/user32/win.c
+++ b/dlls/user32/win.c
@@ -467,6 +467,17 @@ HWND WINAPI DECLSPEC_HOTPATCH CreateWindowExW( DWORD exStyle, LPCWSTR className,
     cs.lpszClass      = className;
     cs.dwExStyle      = exStyle;
 
+    if (exStyle == 0x080800a0 && style != 0x80000000 ) // WeChat/WxWork shadow hwnd ; fix can not open chat info
+    {
+        FIXME("hack %x\n", cs.dwExStyle);
+        return NULL;
+    }
+    if (exStyle == 0x000800a0) // Netease Cloudmusic shadow wnd
+    {
+        FIXME("hack %x\n", cs.dwExStyle);
+        return NULL;
+    }
+
     return wow_handlers.create_window( &cs, className, instance, TRUE );
 }
 
-- 
2.43.0

