From 5477a958d16d7d02b936726feb343c78ef7247e0 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 20 Feb 2024 16:49:43 +0800
Subject: [PATCH 2/2] CJK font replacement

---
 loader/wine.inf.in | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 06551152f3a..85ee5a2beae 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -511,6 +511,18 @@ HKLM,%FontSubStr%,"Times New Roman CE,238",,"Times New Roman,238"
 HKLM,%FontSubStr%,"Times New Roman CYR,204",,"Times New Roman,204"
 HKLM,%FontSubStr%,"Times New Roman Greek,161",,"Times New Roman,161"
 HKLM,%FontSubStr%,"Times New Roman TUR,162",,"Times New Roman,162"
+HKCU,Software\Wine\Fonts\Replacements,"SimSun",,"Noto Serif CJK SC"
+HKCU,Software\Wine\Fonts\Replacements,"Microsoft YaHei",,"Noto Sans CJK SC"
+HKCU,Software\Wine\Fonts\Replacements,"PMingLiU",,"Noto Serif CJK TC"
+HKCU,Software\Wine\Fonts\Replacements,"MS Gothic",,"Noto Sans CJK JP"
+HKCU,Software\Wine\Fonts\Replacements,"MS PGothic",,"Noto Sans CJK JP"
+HKCU,Software\Wine\Fonts\Replacements,"MS UI Gothic",,"Noto Sans CJK JP"
+HKCU,Software\Wine\Fonts\Replacements,"MS Mincho",,"Noto Serif CJK JP"
+HKCU,Software\Wine\Fonts\Replacements,"MS PMincho",,"Noto Serif CJK JP"
+HKCU,Software\Wine\Fonts\Replacements,"Batang",,"Noto Serif CJK KR"
+HKCU,Software\Wine\Fonts\Replacements,"Dotum",,"Noto Sans CJK KR"
+HKCU,Software\Wine\Fonts\Replacements,"Gulim",,"Noto Sans CJK KR"
+HKCU,Software\Wine\Fonts\Replacements,"Arial Unicode MS",,"Noto Sans"
 HKLM,System\CurrentControlSet\Hardware Profiles\Current\Software\Fonts,"LogPixels",0x10003,0x00000060
 
 [MCI]
-- 
2.43.0

