diff --git a/configure.ac b/configure.ac
index 0633284..fa7c1d0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -143,11 +143,11 @@ fi
 
 # Filter out -Werror temporarily, otherwise library checks can fail
 CFLAGS_nowerror="`echo $CFLAGS | sed 's/-Werror\([^=]\|$\)//'`"
-if test "$CFLAGS" = "$CFLAGS_nowerror"; then
-  CFLAGS_werror=
-else
+if echo $CFLAGS | grep -q -- "Werror"; then
   CFLAGS_werror=" -Werror"
-  CFLAGS="$CFLAGS_nowerror"
+  CFLAGS="$CFLAGS_noweror"
+else
+  CFLAGS_werror=
 fi
 
 # Checks for libraries.
