diff -ruN a/subprojects/magpie/platform-linux/Cargo.toml b/subprojects/magpie/platform-linux/Cargo.toml
--- a/subprojects/magpie/platform-linux/Cargo.toml	2025-05-14 17:30:00.214703570 +0800
+++ b/subprojects/magpie/platform-linux/Cargo.toml	2025-05-14 17:32:28.554693871 +0800
@@ -14,7 +14,7 @@
 libc = "0.2"
 libloading = "0.8"
 log = "0.4"
-sysinfo = "0.33"
+sysinfo = "0.35.1"
 thiserror = "2.0"
 trim-in-place = "0.1"
 udisks2 = "0.3"
