diff -ruN a/subprojects/magpie/platform-linux/src/cpu/cpufreq.rs b/subprojects/magpie/platform-linux/src/cpu/cpufreq.rs
--- a/subprojects/magpie/platform-linux/src/cpu/cpufreq.rs	2025-05-17 18:33:25.031795029 +0800
+++ b/subprojects/magpie/platform-linux/src/cpu/cpufreq.rs	2025-05-17 18:43:15.651773600 +0800
@@ -102,7 +102,7 @@
         let mut result = 0;
         for line in cpuinfo
             .split('\n')
-            .filter(|line| line.starts_with("cpu MHz\t") || line.starts_with("clock\t"))
+            .filter(|line| line.starts_with("cpu MHz\t") || line.starts_with("CPU MHz\t") || line.starts_with("clock\t"))
         {
             result = line
                 .split(':')
