From d890518535ec08cb506c57523ef90a893e171f96 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Fri, 10 May 2019 14:21:42 +0800
Subject: [PATCH] Fix Cargo.toml

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 06bb570..3d3c76f 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -53,7 +53,7 @@ gtk_3_22_30 = ["gtk_3_18", "gtk/v3_22_30", "gio/v2_56"]
 gtk_3_24 = ["gtk_3_22_30", "gtk/v3_24", "gio/v2_58"]
 
 [dependencies.mpris-player]
-version = "0.2.0"
+version = "0.3.0"
 git = "https://gitlab.gnome.org/World/Rust/mpris-player"
 
 [package.metadata.rpm.cargo]
-- 
2.21.0

