From c7596e371244a33f9a713b6a1b7b26e0b016d1e7 Mon Sep 17 00:00:00 2001
From: Fangrui Song <i@maskray.me>
Date: Tue, 11 Jan 2022 19:25:21 -0800
Subject: [PATCH 09/11] initialize: Add offsetEncoding to InitializeResult

---
 src/messages/initialize.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/messages/initialize.cc b/src/messages/initialize.cc
index 1601db70..ea07006d 100644
--- a/src/messages/initialize.cc
+++ b/src/messages/initialize.cc
@@ -226,9 +226,10 @@ struct InitializeResult {
     const char *name = "ccls";
     const char *version = CCLS_VERSION;
   } serverInfo;
+  const char *offsetEncoding = "utf-32";
 };
 REFLECT_STRUCT(InitializeResult::ServerInfo, name, version);
-REFLECT_STRUCT(InitializeResult, capabilities, serverInfo);
+REFLECT_STRUCT(InitializeResult, capabilities, serverInfo, offsetEncoding);
 
 struct FileSystemWatcher {
   std::string globPattern = "**/*";
-- 
2.35.1

