From b28cec18721bf550b0ba9b70cb46233647ee4632 Mon Sep 17 00:00:00 2001
From: augiedoggie <augiedoggie@users.noreply.github.com>
Date: Sun, 23 Jan 2022 01:22:29 -0700
Subject: [PATCH 10/11] platform_posix: add support for Haiku (#863)

---
 src/platform_posix.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/platform_posix.cc b/src/platform_posix.cc
index 5a937c65..885e9557 100644
--- a/src/platform_posix.cc
+++ b/src/platform_posix.cc
@@ -1,7 +1,7 @@
 // Copyright 2017-2018 ccls Authors
 // SPDX-License-Identifier: Apache-2.0
 
-#if defined(__unix__) || defined(__APPLE__)
+#if defined(__unix__) || defined(__APPLE__) || defined(__HAIKU__)
 #include "platform.hh"
 
 #include "utils.hh"
-- 
2.35.1

