diff -U4 -r modem-manager-gui-0.0.20--orig/appdata/meson.build modem-manager-gui-0.0.20--patched/appdata/meson.build
--- modem-manager-gui-0.0.20--orig/appdata/meson.build	2020-08-01 23:12:39.000000000 +0200
+++ modem-manager-gui-0.0.20--patched/appdata/meson.build	2022-07-22 12:04:18.848304669 +0200
@@ -1,7 +1,7 @@
 i18n = import('i18n')
 
-i18n.merge_file('appdata-file',
+i18n.merge_file(
 	input: 'modem-manager-gui.appdata.xml.in',
 	output: 'modem-manager-gui.appdata.xml',
 	type: 'xml',
 	data_dirs: join_paths(meson.source_root(), 'appdata'),
@@ -9,9 +9,9 @@
 	install: true,
 	install_dir: join_paths(get_option('prefix'), get_option('datadir'), 'metainfo')
 )
 
-i18n.merge_file('desktop-file',
+i18n.merge_file(
 	input: 'modem-manager-gui.desktop.in',
 	output: 'modem-manager-gui.desktop',
 	type: 'desktop',
 	po_dir: join_paths(meson.source_root(), 'appdata'),
diff -U4 -r modem-manager-gui-0.0.20--orig/polkit/meson.build modem-manager-gui-0.0.20--patched/polkit/meson.build
--- modem-manager-gui-0.0.20--orig/polkit/meson.build	2020-08-01 23:12:39.000000000 +0200
+++ modem-manager-gui-0.0.20--patched/polkit/meson.build	2022-07-22 12:08:28.855487055 +0200
@@ -1,7 +1,7 @@
 i18n = import('i18n')
 
-i18n.merge_file('polkit-file',
+i18n.merge_file(
 	input: 'ru.linuxonly.modem-manager-gui.policy.in',
 	output: 'ru.linuxonly.modem-manager-gui.policy',
 	type: 'xml',
 	data_dirs: join_paths(meson.source_root(), 'polkit'),
