diff -up avogadro-1.1.1/libavogadro/tests/moleculefiletest.cpp.qt avogadro-1.1.1/libavogadro/tests/moleculefiletest.cpp
--- avogadro-1.1.1/libavogadro/tests/moleculefiletest.cpp.qt	2013-12-06 16:50:04.000000000 +0100
+++ avogadro-1.1.1/libavogadro/tests/moleculefiletest.cpp	2016-02-21 23:54:57.391691551 +0100
@@ -151,7 +151,7 @@ void MoleculeFileTest::readFile()
   OBConversion conv;
   conv.SetOutFormat("sdf");
   std::ofstream ofs(filename.toAscii().data());
-  QVERIFY( ofs );
+  QVERIFY( ofs.good() );
   // write the molecule 4 times...
   conv.Write(&mol, &ofs);
   conv.Write(&mol, &ofs);
@@ -171,7 +171,7 @@ void MoleculeFileTest::readFile()
 
 
   ofs.open(filename.toAscii().data());
-  QVERIFY( ofs );
+  QVERIFY( ofs.good() );
   // write the molecule 4 times...
   conv.Write(&mol, &ofs);
   conv.Write(&mol, &ofs);
