--- avogadro-1.2.0/CMakeLists.txt.omv~	2014-01-30 13:58:22.999443609 +0100
+++ avogadro-1.2.0/CMakeLists.txt	2014-01-30 13:59:31.086106925 +0100
@@ -231,13 +231,15 @@
   message(WARNING " Qt4 Linguist not found, please install it if you want Avogadro translations")
 endif()
 
-find_package(Eigen3) # find and setup Eigen3 if available
+find_package(Eigen3 REQUIRED) # find and setup Eigen3 if available
 if(NOT EIGEN3_FOUND)
    message(STATUS "Cannot find Eigen3, trying Eigen2")
-   find_package(Eigen2 REQUIRED) # Some version is required
+   find_package(Eigen2) # Some version is required
 else()
 # Use Stage10 Eigen3 support
    set (EIGEN2_SUPPORT_STAGE10_FULL_EIGEN2_API TRUE)
+   add_definitions (-DEIGEN2_SUPPORT_STAGE10_FULL_EIGEN2_API)
+   set (EIGEN2_INCLUDE_DIR ${EIGEN3_INCLUDE_DIR})
 endif()
 
 find_package(ZLIB REQUIRED)
--- avogadro-1.2.0/libavogadro/src/extensions/qtaim/CMakeLists.txt.omv~	2014-01-30 14:12:33.859401753 +0100
+++ avogadro-1.2.0/libavogadro/src/extensions/qtaim/CMakeLists.txt	2014-01-30 14:15:07.652727522 +0100
@@ -4,6 +4,8 @@
 set(PLUGIN_LABEL extensions)
 set(PLUGIN_TARGET extensions)
 
+add_definitions(-DEIGEN2_SUPPORT_STAGE20_RESOLVE_API_CONFLICTS -std=gnu++11)
+
 set(qtaimextension_SRCS
     qtaimextension.cpp
 #    qtaimdialog.cpp
--- avogadro-1.2.0/libavogadro/src/extensions/surfaces/openqube/CMakeLists.txt.omv~	2014-01-30 13:59:05.889441500 +0100
+++ avogadro-1.2.0/libavogadro/src/extensions/surfaces/openqube/CMakeLists.txt	2014-01-30 13:59:08.832774688 +0100
@@ -2,7 +2,7 @@
 find_package(Eigen3)
 if(NOT EIGEN3_FOUND)
   message(STATUS "Cannot find Eigen3, trying Eigen2")
-  find_package(Eigen2 REQUIRED)
+  find_package(Eigen3 REQUIRED)
 else()
   set (EIGEN2_SUPPORT_STAGE10_FULL_EIGEN2_API TRUE)
 endif()
--- avogadro-1.2.0/cmake/modules/AvogadroUse.cmake.orig	2014-10-13 20:13:59.909828034 +0100
+++ avogadro-1.2.0/cmake/modules/AvogadroUse.cmake	2014-10-13 20:15:09.492822908 +0100
@@ -20,7 +20,7 @@
 # If Eigen3 wasn't found, find Eigen2. It will be required.
 else(EIGEN3_FOUND)
   message("Eigen3 not found. Trying to find Eigen2...")
-  find_package(Eigen2 REQUIRED)
+  find_package(Eigen3 REQUIRED)
   include_directories(${EIGEN2_INCLUDE_DIR})
 endif(EIGEN3_FOUND)
 
