diff -up avogadro-1.1.1/libavogadro/src/pythonengine_p.h.Q_MOC_RUN avogadro-1.1.1/libavogadro/src/pythonengine_p.h
--- avogadro-1.1.1/libavogadro/src/pythonengine_p.h.Q_MOC_RUN	2013-12-06 09:50:04.000000000 -0600
+++ avogadro-1.1.1/libavogadro/src/pythonengine_p.h	2016-01-25 13:29:52.576196449 -0600
@@ -27,7 +27,9 @@
 
 #include <avogadro/global.h>
 #include <avogadro/engine.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 namespace Avogadro {
 
diff -up avogadro-1.1.1/libavogadro/src/pythonextension_p.h.Q_MOC_RUN avogadro-1.1.1/libavogadro/src/pythonextension_p.h
--- avogadro-1.1.1/libavogadro/src/pythonextension_p.h.Q_MOC_RUN	2013-12-06 09:50:04.000000000 -0600
+++ avogadro-1.1.1/libavogadro/src/pythonextension_p.h	2016-01-25 13:29:52.576196449 -0600
@@ -29,7 +29,9 @@
 #include <avogadro/extension.h>
 #include <avogadro/primitive.h>
 #include <avogadro/glwidget.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 #include <QWidget>
 #include <QList>
diff -up avogadro-1.1.1/libavogadro/src/pythoninterpreter.h.Q_MOC_RUN avogadro-1.1.1/libavogadro/src/pythoninterpreter.h
--- avogadro-1.1.1/libavogadro/src/pythoninterpreter.h.Q_MOC_RUN	2013-12-06 09:50:04.000000000 -0600
+++ avogadro-1.1.1/libavogadro/src/pythoninterpreter.h	2016-01-25 13:34:36.911051943 -0600
@@ -26,7 +26,9 @@
 #define PYTHONINTERPRETER_H
 
 #include <avogadro/global.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 #include <avogadro/primitive.h>
 #include <QString>
 
diff -up avogadro-1.1.1/libavogadro/src/pythonscript.h.Q_MOC_RUN avogadro-1.1.1/libavogadro/src/pythonscript.h
--- avogadro-1.1.1/libavogadro/src/pythonscript.h.Q_MOC_RUN	2013-12-06 09:50:04.000000000 -0600
+++ avogadro-1.1.1/libavogadro/src/pythonscript.h	2016-01-25 13:34:27.048952883 -0600
@@ -27,7 +27,9 @@
 #define PYTHONSCRIPT_H
 
 #include <avogadro/global.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 #include "pythonerror.h"
 
diff -up avogadro-1.1.1/libavogadro/src/pythontool_p.h.Q_MOC_RUN avogadro-1.1.1/libavogadro/src/pythontool_p.h
--- avogadro-1.1.1/libavogadro/src/pythontool_p.h.Q_MOC_RUN	2013-12-06 09:50:04.000000000 -0600
+++ avogadro-1.1.1/libavogadro/src/pythontool_p.h	2016-01-25 13:29:52.576196449 -0600
@@ -27,7 +27,9 @@
 
 #include <avogadro/global.h>
 #include <avogadro/tool.h>
+#ifndef Q_MOC_RUN
 #include <boost/python.hpp>
+#endif
 
 #include <QObject>
 #include <QAction>
