From 5d26650f0da9d47ce69cd9b739ad7a6990071f39 Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Wed, 30 Dec 2020 16:56:17 +0100
Subject: Fix syntax of namespace documentation block

---
 libzeitgeist/log.vala | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libzeitgeist/log.vala b/libzeitgeist/log.vala
index 9b488af..53cd4f5 100644
--- a/libzeitgeist/log.vala
+++ b/libzeitgeist/log.vala
@@ -29,6 +29,7 @@
  *
  * Its main component is the Zeitgeist engine, a D-Bus service that logs
  * any events other applications send to it. An event may be anything like:
+ *
  *  - The user opened/created/modified/closed a file, or visited a website.
  *  - The user received an e-mail, a phone call or an IM notification.
  *  - Someone modified a remote (eg. Google Drive) document owned by the user.
-- 
cgit v1.1

