From 1bcc8585275b4a8a9d47652cfe2ee4d64d155c1f Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Mon, 27 Feb 2017 16:27:36 +0100
Subject: Fix placeholder in systemd service template


diff --git a/extensions/fts++/zeitgeist-fts.service.in b/extensions/fts++/zeitgeist-fts.service.in
index 93bf8f6..6459c0c 100644
--- a/extensions/fts++/zeitgeist-fts.service.in
+++ b/extensions/fts++/zeitgeist-fts.service.in
@@ -4,4 +4,4 @@ Description=Zeitgeist full-text search indexer
 [Service]
 Type=dbus
 BusName=org.gnome.zeitgeist.SimpleIndexer
-ExecStart=@libexecdir@/zeitgeist-fts
+ExecStart=@pkglibexecdir@/zeitgeist-fts
-- 
cgit v0.10.2

