From ddb715967b332ebcfa65b4842d5778a0cf17bad2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20K=2E=20H=C3=BCttel?= <dilfridge@gentoo.org>
Date: Sat, 25 Jul 2026 09:21:33 +0900
Subject: [PATCH 02/63] UPSTREAM: NEWS: start 2.44.1 section
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas K. Hüttel <dilfridge@gentoo.org>
---
 NEWS | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/NEWS b/NEWS
index ee28fadf49..1043343d70 100644
--- a/NEWS
+++ b/NEWS
@@ -5,6 +5,10 @@ See the end for copying conditions.
 Please send GNU C library bug reports via <https://sourceware.org/bugzilla/>
 using `glibc' in the "product" field.
 
+Version 2.44.1
+
+The following bugs are resolved with this release:
+
 Version 2.44
 
 Major new features:
-- 
2.55.0

