diff -Naur astrometry.net-0.73/util/makefile.common astrometry.net-0.73.nonative/util/makefile.common
--- astrometry.net-0.73/util/makefile.common	2017-11-17 20:03:49.000000000 +0000
+++ astrometry.net-0.73.nonative/util/makefile.common	2017-11-19 23:54:44.424713814 +0000
@@ -162,58 +162,6 @@
 #ARG := -std=c99
 #FLAGS_DEF += $(shell $(CCTEST))
 
-## Below, we try to guess good compiler flags for this CPU / architecture.
-## If we make an incorrect choice -- ie, the code fails in some way on the
-## machine on which it was compiled, please let us know by posting at
-##    astrometry.net/group
-## If you want to override this setting, you can set the
-##      ARCH_FLAGS
-## environment variable in your shell.
-## With gcc 4.2 and later, gcc should be able to make the optimal choice on its own.
-## See the README section "I wanna go fast!" for more details.
-ifneq ($(origin ARCH_FLAGS),undefined)
-  FLAGS_DEF += $(ARCH_FLAGS)
-else
-  # Try to guess some good compiler flags for this CPU.
-
-  # Use -march=native if it's available (gcc 4.2 and above)
-  ARG := -march=native
-  X := $(shell $(CCTEST))
-  ifneq ($(X),)
-    FLAGS_DEF += $(X)
-  else
-    # Try to guess -march
-    MACHINE:=$(shell uname -m)
-    ifeq ($(MACHINE), i686)
-      # gcc before version 3.1 doesn't support "pentium4"; use "i686" instead.
-
-      ARG := -march=pentium4
-      X := $(shell $(CCTEST))
-      ifneq ($(X),)
-        STR := \#define __tune_pentium4__ 1
-        FLAGS_DEF += $(shell $(DEFTEST) && echo "-march=pentium4" \
-                                        || echo "-march=i686")
-      else
-        ARG := -march=i686
-        FLAGS_DEF += $(shell $(CCTEST))
-      endif
-
-    else
-      # make 3.79 doesn't allow multiple "else" statements, so nest 'em.
-      ifeq ($(MACHINE), x86_64)
-        STR := \#define __tune_k8__ 1
-        FLAGS_DEF += $(shell $(DEFTEST) && echo "-march=k8")
-          FLAGS_DEF += -m64
-      else # ppc, ...
-        FLAGS_DEF += -DNOT_686
-
-      endif
-
-    endif
-  endif
-
-endif
-
 STR := __APPLE__
 X := $(shell $(DEFTEST) && echo "-DNOBOOL")
 

