diff --git a/libkd/pyspherematch.c b/libkd/pyspherematch.c
index 5f944bdcd..4eb23c665 100644
--- a/libkd/pyspherematch.c
+++ b/libkd/pyspherematch.c
@@ -15,6 +15,7 @@
 
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include <numpy/arrayobject.h>
+#undef I
 
 #include "os-features.h"
 #include "keywords.h"
diff --git a/util/util.i b/util/util.i
index ea20c0c46..941980395 100644
--- a/util/util.i
+++ b/util/util.i
@@ -164,7 +164,7 @@ void log_set_level(int lvl);
         printf("  descr kind: '%c'\n", desc->kind);
         printf("  descr type: '%c'\n", desc->type);
         printf("  descr byteorder: '%c'\n", desc->byteorder);
-        printf("  descr elsize: %i\n", desc->elsize);
+        printf("  descr elsize: %i\n", (int)PyArray_ITEMSIZE(obj));
     }
 
 
