From 252e4f78bfb18d515c99c55afbe47a3987deb2d7 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Sun, 13 Apr 2025 22:11:41 -0700
Subject: [PATCH] fix: support libcava 0.10.4

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 include/modules/cava.hpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/include/modules/cava.hpp b/include/modules/cava.hpp
index b4d2325b..1a88c7b7 100644
--- a/include/modules/cava.hpp
+++ b/include/modules/cava.hpp
@@ -5,7 +5,16 @@
 
 namespace cava {
 extern "C" {
+// Need sdl_glsl output feature to be enabled on libcava
+#ifndef SDL_GLSL
+#define SDL_GLSL
+#endif
+
 #include <cava/common.h>
+
+#ifdef SDL_GLSL
+#undef SDL_GLSL
+#endif
 }
 }  // namespace cava
 
-- 
2.49.0

