diff --git a/kasten/controllers/view/structures/allprimitivetypes.cpp b/kasten/controllers/view/structures/allprimitivetypes.cpp
index c4f6185..60c3fd2 100644
--- a/kasten/controllers/view/structures/allprimitivetypes.cpp
+++ b/kasten/controllers/view/structures/allprimitivetypes.cpp
@@ -6,6 +6,7 @@
     SPDX-License-Identifier: LGPL-2.1-only OR LGPL-3.0-only OR LicenseRef-KDE-Accepted-LGPL
 */
 
+#include <limits>
 #include "allprimitivetypes.hpp"
 #include "datatypes/primitive/primitivedatainformation.hpp"
 
diff --git a/kasten/controllers/view/structures/datatypes/primitive/enumdefinition.cpp b/kasten/controllers/view/structures/datatypes/primitive/enumdefinition.cpp
index 384e0f7..cafedc1 100644
--- a/kasten/controllers/view/structures/datatypes/primitive/enumdefinition.cpp
+++ b/kasten/controllers/view/structures/datatypes/primitive/enumdefinition.cpp
@@ -6,11 +6,11 @@
     SPDX-License-Identifier: LGPL-2.1-only OR LGPL-3.0-only OR LicenseRef-KDE-Accepted-LGPL
 */
 
+#include <limits>
 #include "enumdefinition.hpp"
 
 #include <QScriptValue>
 #include <QScriptValueIterator>
-#include <limits>
 
 #include "../../script/scriptlogger.hpp"
 
