--- a/aclocal.m4
+++ b/aclocal.m4
@@ -878,7 +878,7 @@
   dnl Determine whether the user wants rpath handling at all.
   AC_ARG_ENABLE(rpath,
     [  --disable-rpath         do not hardcode runtime library paths],
-    :, enable_rpath=yes)
+    :, enable_rpath=no)
 ])
 
 dnl AC_LIB_LINKFLAGS_BODY(name [, dependencies]) searches for libname and
--- a/configure.in
+++ b/configure.in
@@ -92,17 +92,18 @@
 
 # Set runtime shared library paths as needed 
 
-case "$target" in
-    *-*-linux*)
-	SMPEG_RLD_FLAGS="-Wl,-rpath,\${exec_prefix}/lib"
-	;;
-    *-*-freebsd*)
-	SMPEG_RLD_FLAGS="-Wl,-rpath,\${exec_prefix}/lib"
-	;;
-    *-*-solaris*)
-	SMPEG_RLD_FLAGS="-R\${exec_prefix}/lib"
-	;;
-esac
+dnl Disable in Debian
+dnl case "$target" in
+dnl     *-*-linux*)
+dnl 	SMPEG_RLD_FLAGS="-Wl,-rpath,\${exec_prefix}/lib"
+dnl 	;;
+dnl     *-*-freebsd*)
+dnl 	SMPEG_RLD_FLAGS="-Wl,-rpath,\${exec_prefix}/lib"
+dnl 	;;
+dnl     *-*-solaris*)
+dnl 	SMPEG_RLD_FLAGS="-R\${exec_prefix}/lib"
+dnl 	;;
+dnl esac
 
 AC_SUBST(SMPEG_RLD_FLAGS)
 
