Index: smpeg-0.4.5+cvs20030824/configure.in
===================================================================
--- smpeg-0.4.5+cvs20030824.orig/configure.in	2007-06-21 04:06:54.000000000 +0200
+++ smpeg-0.4.5+cvs20030824/configure.in	2007-06-21 04:07:11.000000000 +0200
@@ -210,7 +210,7 @@
               , enable_gtk_player=yes)
 have_gtk=no
 if test x$enable_gtk_player = xyes; then
-    AM_PATH_GTK(1.2.1, have_gtk=yes)
+    AM_PATH_GTK_2_0(2.0.0, have_gtk=yes)
     if test x$have_gtk = xyes; then
         CFLAGS="$CFLAGS $GTK_CFLAGS"
     fi
