--- smpeg-0.4.5+cvs20030824.orig/smpeg-config.in
+++ smpeg-0.4.5+cvs20030824/smpeg-config.in
@@ -52,6 +52,14 @@
       fi
       echo $libdirs -lsmpeg `@SDL_CONFIG@ --libs`
       ;;
+    --library-libs)
+      echo "Please update to a new smpeg.m4. --library-libs is no longer"
+      echo "supported."
+      ;;
+    --plugin-libs)
+      echo "Please update to a new smpeg.m4. --plugin-libs is no longer"
+      echo "supported."
+      ;;
     *)
       echo "${usage}" 1>&2
       exit 1
