--- smpeg-0.4.5+cvs20030824.orig/video/mmxflags_asm.S
+++ smpeg-0.4.5+cvs20030824/video/mmxflags_asm.S
@@ -1,4 +1,6 @@
 
+.section .note.GNU-stack,"",%progbits
+
 #if defined(i386) && defined(USE_MMX)
 
 .data
--- smpeg-0.4.5+cvs20030824.orig/video/mmxidct_asm.S
+++ smpeg-0.4.5+cvs20030824/video/mmxidct_asm.S
@@ -1,4 +1,5 @@
 
+.section .note.GNU-stack,"",%progbits
 
 #if defined(i386) && defined(USE_MMX)
 
