From ae33d96b5a66f55dced018d922e8b502cd31c5c3 Mon Sep 17 00:00:00 2001
From: sena <sena@aosc-kp920.lan>
Date: Thu, 11 Sep 2025 09:55:41 +0800
Subject: [PATCH] FEXRootFSFetcher: Use -help-all to check unsquashfs

---
 Source/Tools/FEXRootFSFetcher/Main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/Tools/FEXRootFSFetcher/Main.cpp b/Source/Tools/FEXRootFSFetcher/Main.cpp
index 3d809dc73..6446ab5e5 100644
--- a/Source/Tools/FEXRootFSFetcher/Main.cpp
+++ b/Source/Tools/FEXRootFSFetcher/Main.cpp
@@ -248,7 +248,7 @@ void CheckSquashfuse() {
 void CheckUnsquashfs() {
   const std::array<const char*, 3> ExecveArgs = {
     "unsquashfs",
-    "--help",
+    "-help-all",
     nullptr,
   };

--
2.51.0
