--- djview4-4.12/src/Makefile.am	2020-11-20 18:20:06.000000000 +0000
+++ djview4-4.12.debuginfo/src/Makefile.am	2022-03-08 07:01:28.132032933 +0000
@@ -64,7 +64,9 @@
 		QMAKE_CXXFLAGS+='$(CXXFLAGS) $(qmake_allflags)' \
 		QMAKE_CPPFLAGS+='$(CPPFLAGS) $(qmake_allflags)' \
 		LIBS+='$(QMAKELIBS)' \
-		VPATH='$(srcdir)' -o $@0 djview.pro
+		VPATH='$(srcdir)' \
+		CONFIG+=force_debug_info \
+		-o $@0 djview.pro
 	$(AWK) -f $(top_srcdir)/config/qmsilent.awk < $@0 > $@ && ${RM} $@0
 
 # cleaning
