--- src/Makefile	2016-11-19 10:09:55.000000000 -0800
+++ src/Makefile	2021-08-25 22:17:47.881327984 -0700
@@ -23,6 +23,7 @@
 LIB=-lasound lib/libgameptr.a
 
 GENERAL_CFLAGS=-DINSTALL_DIR="\"$(INSTALL_DIR)\""
+CFLAGS=
 GTK_CFLAGS = `pkg-config --cflags gtk+-2.0` -DGTK_DISABLE_DEPRECATED=1 -DGTK_MULTIHEAD_SAFE=1
 GTK_LFLAGS = `pkg-config --cflags --libs gtk+-2.0`  -DGTK_DISABLE_DEPRECATED=1 -DGTK_MULTIHEAD_SAFE=1
 XML_LFLAGS=`xml2-config --cflags --libs`
@@ -36,9 +37,9 @@
 		$(XML_LFLAGS) $(INC) $(GTK_LFLAGS) $(LIB) $(GTHREAD_LFLAGS) #lib/dartlib.o
 
 debug:
-	g++ interface.cpp -c -g $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS)
-	g++ support.cpp -c -g $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS)
-	g++ events.cpp -c -g $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS)
+	g++ interface.cpp -c -g $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS) $(CFLAGS)
+	g++ support.cpp -c -g $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS) $(CFLAGS)
+	g++ events.cpp -c -g $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS) $(CFLAGS)
 	(cd lib; ${MAKE} "INSTALL_DIR=${INSTALL_DIR}" debug)
 	for i in ${GAMES}; do \
 		(cd $$i; ${MAKE} "INSTALL_DIR=${INSTALL_DIR}" debug); \
@@ -47,13 +48,13 @@
 		$(GAME_LIBS) $(XML_LFLAGS) $(INC) $(GTK_LFLAGS) $(LIB) $(GTHREAD_LFLAGS)
 
 interface.o: touchmegames.h interface.cpp
-	g++ interface.cpp -c $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS)
+	g++ interface.cpp -c $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS) $(CFLAGS)
 	
 support.o: touchmegames.h support.cpp	
-	g++ support.cpp -c $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS)
+	g++ support.cpp -c $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS) $(CFLAGS)
 	
 events.o: touchmegames.h events.cpp
-	g++ events.cpp -c $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS)
+	g++ events.cpp -c $(INC) $(GTK_CFLAGS) $(GENERAL_CFLAGS) $(CFLAGS)
 	
 lib:
 	(cd lib; ${MAKE} "INSTALL_DIR=${INSTALL_DIR}")
