From e93aef00283b701ed1ec64a5f9bf570744db1bab Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Thu, 7 Mar 2024 00:17:31 -0800
Subject: [PATCH] CMakeLists: fix conf file path

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ac6b60..88ac1d2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,7 @@ include(CheckCCompilerFlag)
 include(CheckIncludeFile)
 include(CheckSymbolExists)
 include(ExternalProject)
+include(GNUInstallDirs)
 find_package(PkgConfig REQUIRED)
 
 set(binary replay-sorcery)
@@ -224,7 +225,7 @@ endif()
 install(TARGETS ${binary} DESTINATION bin PERMISSIONS ${permissions})
 
 # Install config
-install(FILES sys/replay-sorcery.conf DESTINATION etc)
+install(FILES sys/replay-sorcery.conf DESTINATION /etc)
 
 # Install services
 set(RS_SYSTEMD_DIR /usr/lib/systemd CACHE STRING "Where to install the systemd services")
-- 
2.44.0

