diff -Naur palemoon/platform/js/src/jsapi-tests/moz.build palemoon.gcc9/platform/js/src/jsapi-tests/moz.build
--- palemoon/platform/js/src/jsapi-tests/moz.build	2020-02-05 04:40:00.350539144 +0000
+++ palemoon.gcc9/platform/js/src/jsapi-tests/moz.build	2020-02-05 04:41:54.018853712 +0000
@@ -144,7 +144,7 @@
 OS_LIBS += CONFIG['MOZ_ZLIB_LIBS']
 
 if CONFIG['GNU_CXX']:
-    CXXFLAGS += ['-Wno-shadow', '-Werror=format', '-fno-strict-aliasing']
+    CXXFLAGS += ['-Wno-shadow', '-fno-strict-aliasing']
 
 # This is intended as a temporary workaround to enable VS2015.
 if CONFIG['_MSC_VER']:
diff -Naur palemoon/platform/js/src/moz.build palemoon.gcc9/platform/js/src/moz.build
--- palemoon/platform/js/src/moz.build	2020-02-05 04:39:59.473833978 +0000
+++ palemoon.gcc9/platform/js/src/moz.build	2020-02-05 04:41:38.121491555 +0000
@@ -785,7 +785,7 @@
 if CONFIG['GNU_CXX']:
     # Disable strict-aliasing for GCC, which is enabled by default
     # starting with version 7.1, see Mozilla bug 1363009.
-    CXXFLAGS += ['-Wno-shadow', '-Werror=format', '-fno-strict-aliasing']
+    CXXFLAGS += ['-Wno-shadow', '-fno-strict-aliasing']
 
 # Suppress warnings in third-party code.
 if CONFIG['CLANG_CXX'] or CONFIG['GNU_CXX']:
diff -Naur palemoon/platform/js/src/shell/moz.build palemoon.gcc9/platform/js/src/shell/moz.build
--- palemoon/platform/js/src/shell/moz.build	2020-02-05 04:40:00.360539583 +0000
+++ palemoon.gcc9/platform/js/src/shell/moz.build	2020-02-05 04:41:24.170880935 +0000
@@ -51,7 +51,7 @@
 ]
 
 if CONFIG['GNU_CXX']:
-    CXXFLAGS += ['-Wno-shadow', '-Werror=format']
+    CXXFLAGS += ['-Wno-shadow']
 
 # This is intended as a temporary workaround to enable VS2015.
 if CONFIG['_MSC_VER']:
