--- palemoon/js/src/moz.build	2018-09-29 23:08:22.923009076 -0500
+++ palemoon.nonx86/js/src/moz.build	2018-09-29 23:13:36.247768505 -0500
@@ -491,9 +491,6 @@
         UNIFIED_SOURCES += [
             'jit/arm/Simulator-arm.cpp'
         ]
-        # Configuration used only for testing.
-        if CONFIG['OS_ARCH'] == 'Linux' or CONFIG['OS_ARCH'] == 'Darwin':
-            CXXFLAGS += [ '-msse2', '-mfpmath=sse' ]
     elif CONFIG['OS_ARCH'] == 'Darwin':
         SOURCES += [
             'jit/arm/llvm-compiler-rt/arm/aeabi_idivmod.S',
--- /var/cache/acbs/tarballs/palemoon/build/autoconf/compiler-opts.m4	2018-09-29 23:00:07.852236124 -0500
+++ palemoon.nonx86/build/autoconf/compiler-opts.m4	2018-09-29 23:31:53.987664713 -0500
@@ -176,8 +176,8 @@
         CFLAGS="$CFLAGS -ffunction-sections -fdata-sections"
         CXXFLAGS="$CXXFLAGS -ffunction-sections -fdata-sections"
     fi
-    CFLAGS="$CFLAGS -fno-math-errno -msse2 -mfpmath=sse"
-    CXXFLAGS="$CXXFLAGS -fno-exceptions -fno-math-errno -msse2 -mfpmath=sse"
+    CFLAGS="$CFLAGS -fno-math-errno"
+    CXXFLAGS="$CXXFLAGS -fno-exceptions -fno-math-errno"
 
     if test -z "$CLANG_CC"; then
         case "$CC_VERSION" in
