--- palemoon/gfx/skia/moz.build 2017-10-31 21:44:58.054443201 +0000
+++ palemoon.arm/gfx/skia/moz.build     2017-10-31 21:47:51.864967129 +0000
@@ -960,9 +960,6 @@
     SOURCES['trunk/src/opts/SkBitmapProcState_opts_SSSE3.cpp'].flags += ['-mssse3']
     SOURCES['trunk/src/opts/SkBlurImage_opts_SSE4.cpp'].flags += ['-msse4.1']
 
-if CONFIG['GNU_CXX'] and CONFIG['CPU_ARCH'] == 'arm':
-    SOURCES['skia/src/opts/SkBlitRow_opts_arm.cpp'].flags += ['-fomit-frame-pointer']
-
 DEFINES['SKIA_IMPLEMENTATION'] = 1
 DEFINES['GR_IMPLEMENTATION'] = 1
 

