From 88421344973970d5a685737c0b968d23cc857ee8 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Thu, 24 Nov 2016 10:41:28 -0600
Subject: [PATCH 1/5] aboutDialog: mroe architectural detection

---
 browser/base/content/aboutDialog.xul | 54 ++++++++++++++++++++++++++++++++++--
 1 file changed, 51 insertions(+), 3 deletions(-)

diff --git a/browser/base/content/aboutDialog.xul b/browser/base/content/aboutDialog.xul
index 434758e..f2f42d1 100644
--- a/browser/base/content/aboutDialog.xul
+++ b/browser/base/content/aboutDialog.xul
@@ -42,10 +42,58 @@
     <hbox id="PMclientBox">
       <vbox id="PMleftBox" flex="1"/>
       <vbox id="PMrightBox" flex="1">
-#ifdef HAVE_64BIT_BUILD
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (x64)</label>
+#if defined(__alpha__)
+#ifdef HAVE_64BIT_OS
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Alpha 64-bit)</label>
 #else
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (x86)</label>
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Alpha 32-bit)</label>
+#endif
+#elif defined(__arm__)
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (ARM 32-bit)</label>
+#elif defined(__aarch64__)
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (ARM 64-bit)</label>
+#elif defined(__i386__) || defined(__i486__) || defined(__i586__) || defined(__i686__)
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Intel x86, 32-bit)</label>
+#elif defined(__ia64__)
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Intel IA-64)</label>
+#elif defined(__m68k__)
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Motorola M68k)</label>
+#elif defined(__mips__)
+#ifdef HAVE_64BIT_OS
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (MIPS 64-bit)</label>
+#else
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (MIPS 32-bit)</label>
+#endif
+#elif defined(__powerpc__) || defined(__ppc__)
+#ifdef HAVE_64BIT_OS
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (PowerPC 64-bit)</label>
+#else
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (PowerPC 32-bit)</label>
+#endif
+#elif defined(__sparc__)
+#ifdef HAVE_64BIT_OS
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SPARC 64-bit)</label>
+#else
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SPARC 32-bit)</label>
+#endif
+#elif defined(__sh__)
+#ifdef HAVE_64BIT_OS
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SuperH 64-bit)</label>
+#else
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SuperH 32-bit)</label>
+#endif
+#elif defined(__s390__)
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (S390)</label>
+#elif defined(__s390x__)
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (S390x)</label>
+#elif defined(__x86_64__)
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Intel x86, 64-bit)</label>
+#else
+#ifdef HAVE_64BIT_OS
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (64-bit)</label>
+#else
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (32-bit)</label>
+#endif
 #endif
         <label id="distribution" class="text-blurb"/>
         <label id="distributionId" class="text-blurb"/>

From 7cd01ee6e04f4fa303dce056f5f21cfa58acca6d Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Thu, 24 Nov 2016 10:43:42 -0600
Subject: [PATCH 2/5] aboutDialog: HAVE_64BIT_BUILD not HAVE_64BIT_OS

---
 browser/base/content/aboutDialog.xul | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/browser/base/content/aboutDialog.xul b/browser/base/content/aboutDialog.xul
index f2f42d1..e3455f3 100644
--- a/browser/base/content/aboutDialog.xul
+++ b/browser/base/content/aboutDialog.xul
@@ -37,13 +37,13 @@
         >
 
   <script type="application/javascript" src="chrome://browser/content/aboutDialog.js"/>
-
+a
   <vbox id="aboutPMDialogContainer">
     <hbox id="PMclientBox">
       <vbox id="PMleftBox" flex="1"/>
       <vbox id="PMrightBox" flex="1">
 #if defined(__alpha__)
-#ifdef HAVE_64BIT_OS
+#ifdef HAVE_64BIT_BUILD
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Alpha 64-bit)</label>
 #else
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Alpha 32-bit)</label>
@@ -59,25 +59,25 @@
 #elif defined(__m68k__)
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Motorola M68k)</label>
 #elif defined(__mips__)
-#ifdef HAVE_64BIT_OS
+#ifdef HAVE_64BIT_BUILD
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (MIPS 64-bit)</label>
 #else
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (MIPS 32-bit)</label>
 #endif
 #elif defined(__powerpc__) || defined(__ppc__)
-#ifdef HAVE_64BIT_OS
+#ifdef HAVE_64BIT_BUILD
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (PowerPC 64-bit)</label>
 #else
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (PowerPC 32-bit)</label>
 #endif
 #elif defined(__sparc__)
-#ifdef HAVE_64BIT_OS
+#ifdef HAVE_64BIT_BUILD
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SPARC 64-bit)</label>
 #else
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SPARC 32-bit)</label>
 #endif
 #elif defined(__sh__)
-#ifdef HAVE_64BIT_OS
+#ifdef HAVE_64BIT_BUILD
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SuperH 64-bit)</label>
 #else
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SuperH 32-bit)</label>
@@ -89,7 +89,7 @@
 #elif defined(__x86_64__)
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Intel x86, 64-bit)</label>
 #else
-#ifdef HAVE_64BIT_OS
+#ifdef HAVE_64BIT_BUILD
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (64-bit)</label>
 #else
 #expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (32-bit)</label>

From cc06e7eaec0ce9e3b764daec0f9e3fcb910b815e Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Thu, 24 Nov 2016 10:44:40 -0600
Subject: [PATCH 3/5] aboutDialog: typo

---
 browser/base/content/aboutDialog.xul | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/browser/base/content/aboutDialog.xul b/browser/base/content/aboutDialog.xul
index e3455f3..833c041 100644
--- a/browser/base/content/aboutDialog.xul
+++ b/browser/base/content/aboutDialog.xul
@@ -37,7 +37,7 @@
         >
 
   <script type="application/javascript" src="chrome://browser/content/aboutDialog.js"/>
-a
+
   <vbox id="aboutPMDialogContainer">
     <hbox id="PMclientBox">
       <vbox id="PMleftBox" flex="1"/>

From 6f3b358050e4d95f43df526471611f33783b8f44 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Sun, 27 Nov 2016 10:53:51 -0600
Subject: [PATCH 4/5] aboutDialog: generalize to bits

---
 browser/base/content/aboutDialog.xul | 52 ++----------------------------------
 1 file changed, 2 insertions(+), 50 deletions(-)

diff --git a/browser/base/content/aboutDialog.xul b/browser/base/content/aboutDialog.xul
index 833c041..4f2a458 100644
--- a/browser/base/content/aboutDialog.xul
+++ b/browser/base/content/aboutDialog.xul
@@ -42,58 +42,10 @@
     <hbox id="PMclientBox">
       <vbox id="PMleftBox" flex="1"/>
       <vbox id="PMrightBox" flex="1">
-#if defined(__alpha__)
 #ifdef HAVE_64BIT_BUILD
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Alpha 64-bit)</label>
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (64-Bit)</label>
 #else
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Alpha 32-bit)</label>
-#endif
-#elif defined(__arm__)
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (ARM 32-bit)</label>
-#elif defined(__aarch64__)
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (ARM 64-bit)</label>
-#elif defined(__i386__) || defined(__i486__) || defined(__i586__) || defined(__i686__)
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Intel x86, 32-bit)</label>
-#elif defined(__ia64__)
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Intel IA-64)</label>
-#elif defined(__m68k__)
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Motorola M68k)</label>
-#elif defined(__mips__)
-#ifdef HAVE_64BIT_BUILD
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (MIPS 64-bit)</label>
-#else
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (MIPS 32-bit)</label>
-#endif
-#elif defined(__powerpc__) || defined(__ppc__)
-#ifdef HAVE_64BIT_BUILD
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (PowerPC 64-bit)</label>
-#else
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (PowerPC 32-bit)</label>
-#endif
-#elif defined(__sparc__)
-#ifdef HAVE_64BIT_BUILD
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SPARC 64-bit)</label>
-#else
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SPARC 32-bit)</label>
-#endif
-#elif defined(__sh__)
-#ifdef HAVE_64BIT_BUILD
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SuperH 64-bit)</label>
-#else
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (SuperH 32-bit)</label>
-#endif
-#elif defined(__s390__)
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (S390)</label>
-#elif defined(__s390x__)
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (S390x)</label>
-#elif defined(__x86_64__)
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (Intel x86, 64-bit)</label>
-#else
-#ifdef HAVE_64BIT_BUILD
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (64-bit)</label>
-#else
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (32-bit)</label>
-#endif
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (32-Bit)</label>
 #endif
         <label id="distribution" class="text-blurb"/>
         <label id="distributionId" class="text-blurb"/>

From 8df44203a67e8accde1fa80e83137045c995e520 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Sun, 27 Nov 2016 11:08:18 -0600
Subject: [PATCH 5/5] aboutDialog: do not capitalize bit

---
 browser/base/content/aboutDialog.xul | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/browser/base/content/aboutDialog.xul b/browser/base/content/aboutDialog.xul
index 4f2a458..eca387b 100644
--- a/browser/base/content/aboutDialog.xul
+++ b/browser/base/content/aboutDialog.xul
@@ -43,9 +43,9 @@
       <vbox id="PMleftBox" flex="1"/>
       <vbox id="PMrightBox" flex="1">
 #ifdef HAVE_64BIT_BUILD
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (64-Bit)</label>
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (64-bit)</label>
 #else
-#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (32-Bit)</label>
+#expand <label id="PMversion">Version: __MOZ_APP_VERSION__ (32-bit)</label>
 #endif
         <label id="distribution" class="text-blurb"/>
         <label id="distributionId" class="text-blurb"/>
