diff -urNr keepassxc-2.7.12-orig/src/main.cpp keepassxc-2.7.12/src/main.cpp
--- keepassxc-2.7.12-orig/src/main.cpp	2026-03-10 01:27:46.000000000 +0100
+++ keepassxc-2.7.12/src/main.cpp	2026-03-12 17:00:05.780444169 +0100
@@ -49,8 +49,24 @@
 #include <windows.h>
 #endif
 
+#ifdef Q_OS_LINUX
+void wayland_hacks()
+{
+    QByteArray currentDesktop = qgetenv("XDG_CURRENT_DESKTOP").toLower();
+    QByteArray sessionDesktop = qgetenv("XDG_SESSION_DESKTOP").toLower();
+    QByteArray sessionType = qgetenv("XDG_SESSION_TYPE").toLower();
+    if ((sessionType.contains("wayland") && qEnvironmentVariableIsEmpty("QT_QPA_PLATFORM")) && (currentDesktop.contains("gnome") || sessionDesktop.contains("gnome")))
+    {
+        qputenv("QT_QPA_PLATFORM", "xcb");
+    }
+}
+#endif
+
 int main(int argc, char** argv)
 {
+#ifdef Q_OS_LINUX
+    wayland_hacks();
+#endif
     QT_REQUIRE_VERSION(argc, argv, QT_VERSION_STR)
 
 #ifdef Q_OS_WIN
