From 01011754629b015d7955d1665dce6593d971dea3 Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Thu, 3 Apr 2025 16:47:56 +0800
Subject: [PATCH 2/2] fix up

---
 src/draco/io/gltf_utils.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/draco/io/gltf_utils.h b/src/draco/io/gltf_utils.h
index befbacb..4547d42 100644
--- a/src/draco/io/gltf_utils.h
+++ b/src/draco/io/gltf_utils.h
@@ -21,6 +21,7 @@
 #include <iomanip>
 #include <sstream>
 #include <string>
+#include <cstdint>
 
 namespace draco {
 
-- 
2.49.0

