From d2032d5f85bce6658d911284458118fd6c712d60 Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Sat, 26 Apr 2025 19:29:10 +0800
Subject: [PATCH] disable static library install

---
 cmake/draco_install.cmake | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmake/draco_install.cmake b/cmake/draco_install.cmake
index b9dda26..2bc306a 100644
--- a/cmake/draco_install.cmake
+++ b/cmake/draco_install.cmake
@@ -66,10 +66,10 @@ macro(draco_setup_install_target)
         ARCHIVE DESTINATION "${libs_path}"
         LIBRARY DESTINATION "${libs_path}")
     else()
-      install(
-        TARGETS draco_static
-        EXPORT dracoExport
-        DESTINATION "${libs_path}")
+      #install(
+      #  TARGETS draco_static
+      #  EXPORT dracoExport
+      #  DESTINATION "${libs_path}")
 
       if(BUILD_SHARED_LIBS)
         install(
-- 
2.49.0

