diff -urNp --exclude-from=/home/mdomsch/excludes --minimal aiksaurus-1.2.1.orig/base/aiksaurus.t.cpp aiksaurus-1.2.1/base/aiksaurus.t.cpp
--- aiksaurus-1.2.1.orig/base/aiksaurus.t.cpp	2008-09-10 13:40:45.000000000 -0500
+++ aiksaurus-1.2.1/base/aiksaurus.t.cpp	2008-09-10 13:40:54.000000000 -0500
@@ -24,6 +24,7 @@
 #include "config.h"
 #include <iostream>
 #include <string>
+#include <cstdlib>
 using namespace std;
 using AiksaurusImpl::AsciiCompare;
 
diff -urNp --exclude-from=/home/mdomsch/excludes --minimal aiksaurus-1.2.1.orig/gtk/src/Display.cpp aiksaurus-1.2.1/gtk/src/Display.cpp
--- aiksaurus-1.2.1.orig/gtk/src/Display.cpp	2003-05-27 18:56:54.000000000 -0500
+++ aiksaurus-1.2.1/gtk/src/Display.cpp	2008-09-10 14:01:56.000000000 -0500
@@ -18,6 +18,7 @@
  * 02111-1307, USA.
  */
 
+#include <string.h>
 #include "Display.h"
 #include "Meaning.h"
 #include "DialogMediator.h"
diff -urNp --exclude-from=/home/mdomsch/excludes --minimal aiksaurus-1.2.1.orig/gtk/src/gtkAiksaur.t.cpp aiksaurus-1.2.1/gtk/src/gtkAiksaur.t.cpp
--- aiksaurus-1.2.1.orig/gtk/src/gtkAiksaur.t.cpp	2002-01-11 20:02:12.000000000 -0600
+++ aiksaurus-1.2.1/gtk/src/gtkAiksaur.t.cpp	2008-09-10 13:51:47.000000000 -0500
@@ -23,6 +23,7 @@
 #include <gtk/gtk.h>
 #include <iostream>
 #include <string>
+#include <cstdlib>
 using namespace std;
 
 int main(int argc, char** argv)
