diff -Naur rustc-1.34.0-src/src/bootstrap/bootstrap.py rustc-1.34.0-src.i586/src/bootstrap/bootstrap.py
--- rustc-1.34.0-src/src/bootstrap/bootstrap.py	2019-04-10 18:45:55.000000000 +0000
+++ rustc-1.34.0-src.i586/src/bootstrap/bootstrap.py	2019-04-21 07:25:19.089672723 +0000
@@ -240,6 +240,7 @@
         'arm64': 'aarch64',
         'i386': 'i686',
         'i486': 'i686',
+        'i586': 'i686',
         'i686': 'i686',
         'i786': 'i686',
         'powerpc': 'powerpc',
