--- rustc-1.53.0-src/compiler/rustc_target/src/spec/mod.rs	2021-06-17 03:53:51.000000000 +0000
+++ rustc-1.53.0-src.i486/compiler/rustc_target/src/spec/mod.rs	2022-01-29 07:02:03.616045440 +0000
@@ -705,6 +705,7 @@
     ("x86_64-unknown-linux-gnux32", x86_64_unknown_linux_gnux32),
     ("i686-unknown-linux-gnu", i686_unknown_linux_gnu),
     ("i586-unknown-linux-gnu", i586_unknown_linux_gnu),
+    ("i486-unknown-linux-gnu", i486_unknown_linux_gnu),
     ("mips-unknown-linux-gnu", mips_unknown_linux_gnu),
     ("mips64-unknown-linux-gnuabi64", mips64_unknown_linux_gnuabi64),
     ("mips64el-unknown-linux-gnuabi64", mips64el_unknown_linux_gnuabi64),
