--- rustc-1.2.0/mk/install.mk	2015-08-04 13:28:47.000000000 -0600
+++ rustc-1.2.0+1/mk/install.mk	2015-08-08 05:47:27.421115562 -0600
@@ -9,12 +9,7 @@
 # except according to those terms.
 
 install:
-ifeq (root user, $(USER) $(patsubst %,user,$(SUDO_USER)))
-# Build the dist as the original user
-	$(Q)sudo -u "$$SUDO_USER" $(MAKE) prepare_install
-else
 	$(Q)$(MAKE) prepare_install
-endif
 ifeq ($(CFG_DISABLE_DOCS),)
 	$(Q)cd tmp/empty_dir && sh ../../tmp/dist/$(DOC_PKG_NAME)-$(CFG_BUILD)/install.sh --prefix="$(DESTDIR)$(CFG_PREFIX)" --libdir="$(DESTDIR)$(CFG_LIBDIR)" --mandir="$(DESTDIR)$(CFG_MANDIR)"
 endif
