--- AssaultCube_v1.2.0.2.source/source/src/Makefile	2013-11-09 15:12:11.000000000 -0800
+++ AssaultCube_v1.2.0.2.debundle/source/src/Makefile	2021-08-23 22:44:49.571373207 -0700
@@ -5,7 +5,7 @@
 PLATFORM= $(shell uname -s)
 PLATFORM_PREFIX=native
 
-INCLUDES= -I. -Ibot -I../enet/include
+INCLUDES= -I. -Ibot
 
 STRIP=
 ifeq (,$(findstring -g,$(CXXFLAGS)))
@@ -23,7 +23,7 @@
 USRLIB=$(shell if [ -e /usr/lib64 ]; then echo "/usr/lib64"; else echo "/usr/lib"; fi)
 # override CXXFLAGS+= -rdynamic		# clang++ doesn't use this...
 CLIENT_INCLUDES= $(INCLUDES) -I/usr/include `sdl-config --cflags` -idirafter ../include
-CLIENT_LIBS= -L../enet/.libs -lenet -L$(USRLIB) -lX11 `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile -lcurl
+CLIENT_LIBS= -lenet -L$(USRLIB) -lX11 `sdl-config --libs` -lSDL_image -lz -lGL -lopenal -lvorbisfile -lcurl
 endif
 
 CLIENT_OBJS= \
@@ -88,7 +88,7 @@
 SERVER_LIBS= -L../lib -lzdll -lenet -llibintl -lws2_32 -lwinmm
 else
 SERVER_INCLUDES= -DSTANDALONE $(INCLUDES)
-SERVER_LIBS= -L../enet/.libs -lenet -lz
+SERVER_LIBS= -lenet -lz
 endif
 
 SERVER_OBJS= \
@@ -115,18 +115,8 @@
 
 all: client server
 
-../enet/Makefile:
-	cd ../enet; ./configure --enable-shared=no --enable-static=yes
-
-libenet: ../enet/Makefile
-	-$(MAKE) -C ../enet all
-
-clean-enet: ../enet/Makefile
-	$(MAKE) -C ../enet/ clean
-
 clean:
 	-$(RM) $(CLIENT_PCH) $(CLIENT_OBJS) $(SERVER_OBJS) $(MASTER_OBJS) ac_client ac_server ac_master
-#	-$(MAKE) -C ../enet/ clean
 
 %.h.gch: %.h
 	$(CXX) $(CXXFLAGS) -o $@.tmp $(subst .h.gch,.h,$@)
@@ -155,12 +145,12 @@
 server_install: server
 
 else
-client: libenet $(CLIENT_OBJS)
+client: $(CLIENT_OBJS)
 	$(CXX) $(CXXFLAGS) -o ac_client $(CLIENT_OBJS) $(CLIENT_LIBS)
 
-server: libenet $(SERVER_OBJS)
+server: $(SERVER_OBJS)
 	$(CXX) $(CXXFLAGS) -o ac_server $(SERVER_OBJS) $(SERVER_LIBS)
-master: libenet $(MASTER_OBJS)
+master: $(MASTER_OBJS)
 	$(CXX) $(CXXFLAGS) -o ac_master $(MASTER_OBJS) $(SERVER_LIBS)
 
 client_install: client
