diff -Naur a/scripts/xinput_calibrator.desktop b/scripts/xinput_calibrator.desktop
--- a/scripts/xinput_calibrator.desktop	2010-09-12 14:36:05.000000000 -0500
+++ b/scripts/xinput_calibrator.desktop	2016-09-11 13:55:02.000000000 -0500
@@ -3,8 +3,8 @@
 Name[de]=Touchscreen Kalibrieren
 Name[nl]=Touchscreen Kalibreren
 Comment=Run the calibration tool for touchscreens
-Exec=/bin/sh -c "xinput_calibrator; cat"
-Terminal=true
+Exec=xinput_calibrator
+Terminal=false
 Type=Application
 Icon=xinput_calibrator
 StartupNotify=true
