Author: Dane Springmeyer <dane.springmeyer@gmail.com>
Description: Add missing includes for fix compile. (LP #393923)

diff -pruN -x '*~' agg-2.5+dfsg1.orig/include/agg_rasterizer_cells_aa.h agg-2.5+dfsg1/include/agg_rasterizer_cells_aa.h
--- agg-2.5+dfsg1.orig/include/agg_rasterizer_cells_aa.h	2007-10-11 00:06:16.000000000 +0200
+++ agg-2.5+dfsg1/include/agg_rasterizer_cells_aa.h	2010-08-23 01:51:07.000000000 +0200
@@ -40,7 +40,8 @@
 #define AGG_RASTERIZER_CELLS_AA_INCLUDED
 
 #include <string.h>
-#include <math.h>
+#include <cstdlib>
+#include <limits>
 #include "agg_math.h"
 #include "agg_array.h"
 
