From 1a21da04703c2c02e71ac7a630877c51a96dd5f3 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Wed, 30 Apr 2025 19:30:15 +0800
Subject: [PATCH] enable specifying build version

---
 cmd/main.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cmd/main.go b/cmd/main.go
index 9a898d3..c6fc6e7 100644
--- a/cmd/main.go
+++ b/cmd/main.go
@@ -41,6 +41,7 @@ var (
 	buildDebugOutput = os.Getenv("XCADDY_DEBUG") == "1"
 	buildFlags       = os.Getenv("XCADDY_GO_BUILD_FLAGS")
 	modFlags         = os.Getenv("XCADDY_GO_MOD_FLAGS")
+	Version 		 string
 )

 func Main() {
@@ -208,6 +209,9 @@ func splitWith(arg string) (module, version, replace string, err error) {
 // xcaddyVersion returns a detailed version string, if available.
 func xcaddyVersion() string {
 	mod := goModule()
+	if Version != "" {
+		mod.Version = Version
+	}
 	ver := mod.Version
 	if mod.Sum != "" {
 		ver += " " + mod.Sum
--
2.49.0
