From 5430b534a7fc5abaf581646ee474191b0f628ab5 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Tue, 16 Dec 2025 14:07:06 +0800
Subject: [PATCH] add missing header

---
 src/xconnection.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/xconnection.cpp b/src/xconnection.cpp
index e2129db..f889939 100644
--- a/src/xconnection.cpp
+++ b/src/xconnection.cpp
@@ -10,6 +10,7 @@
 #include <climits>
 #include <cstring>
 #include <iostream>
+#include <cstdint>
 
 #include "globals.h"
 
-- 
2.52.0

