From a53e661b4bbd61d869f156d214187d2eb764f691 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 13 Jan 2025 16:52:29 +0800
Subject: [PATCH 14/18] Unset all release profile

---
 src-tauri/Cargo.toml | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src-tauri/Cargo.toml b/src-tauri/Cargo.toml
index 683180bb..e36781a5 100755
--- a/src-tauri/Cargo.toml
+++ b/src-tauri/Cargo.toml
@@ -91,13 +91,6 @@ default = ["custom-protocol"]
 custom-protocol = ["tauri/custom-protocol"]
 verge-dev = []
 
-[profile.release]
-panic = "abort"
-codegen-units = 1
-lto = true
-opt-level = "s"
-strip = true
-
 [profile.dev]
 incremental = true
 
-- 
2.48.1

