From e4093dcd4e78bfcd4813a8338b0a7462f9f8bf75 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 13 Jan 2025 16:52:29 +0800
Subject: [PATCH 14/19] Unset all release profile

---
 src-tauri/Cargo.toml | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src-tauri/Cargo.toml b/src-tauri/Cargo.toml
index 35239f7c..0e4bed4d 100755
--- a/src-tauri/Cargo.toml
+++ b/src-tauri/Cargo.toml
@@ -95,13 +95,6 @@ default = ["custom-protocol"]
 custom-protocol = ["tauri/custom-protocol"]
 verge-dev = []
 
-[profile.release]
-panic = "abort"
-codegen-units = 1
-lto = true
-opt-level = "s"
-strip = true
-
 [profile.dev]
 incremental = true
 
-- 
2.49.0

