From 835e19b0d842eaa7e2cbcddfe4e4ec65da6add9c Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 13 Jan 2025 16:52:29 +0800
Subject: [PATCH 14/20] Unset all release profile

---
 src-tauri/Cargo.toml | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src-tauri/Cargo.toml b/src-tauri/Cargo.toml
index 6ff5d76c..564ed393 100755
--- a/src-tauri/Cargo.toml
+++ b/src-tauri/Cargo.toml
@@ -96,13 +96,6 @@ default = ["custom-protocol"]
 custom-protocol = ["tauri/custom-protocol"]
 verge-dev = []
 
-[profile.release]
-panic = "abort"
-codegen-units = 1
-lto = true
-opt-level = "s"
-strip = true
-
 [profile.dev]
 incremental = true
 
-- 
2.49.0

