From 679d56b7d1bbff4141cf30d7cab9d97d619d897f Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <origincode@aosc.io>
Date: Tue, 11 Apr 2023 18:04:25 -0700
Subject: [PATCH 1/3] sleep.conf: AllowHibernation=no by default

Signed-off-by: Kaiyang Wu <origincode@aosc.io>
---
 src/sleep/sleep.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sleep/sleep.conf b/src/sleep/sleep.conf
index 98430348a7bf..41534e6c15d3 100644
--- a/src/sleep/sleep.conf
+++ b/src/sleep/sleep.conf
@@ -18,7 +18,7 @@
 
 [Sleep]
 #AllowSuspend=yes
-#AllowHibernation=yes
+AllowHibernation=no
 #AllowSuspendThenHibernate=yes
 #AllowHybridSleep=yes
 #SuspendState=mem standby freeze
-- 
2.47.0

