From aa081fb940258ad35a088af1dabd6ee25ad6dd43 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Sat, 27 Jul 2024 14:43:07 +0800
Subject: [PATCH 2/3] main.c: revert part of systemd/systemd#31442

To make the user happy.
---
 src/core/main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/core/main.c b/src/core/main.c
index 4b8a315d86..830cd4a801 100644
--- a/src/core/main.c
+++ b/src/core/main.c
@@ -3110,8 +3110,8 @@ int main(int argc, char *argv[]) {
                                 goto finish;
                         }
                         if (r > 0) {
-                                log_full(LOG_CRIT, "Legacy cgroup v1 support selected. This is no longer supported. Will proceed anyway after 30s.");
-                                (void) usleep_safe(30 * USEC_PER_SEC);
+                                log_full(LOG_CRIT, "Legacy cgroup v1 support selected. This is no longer supported. Will proceed anyway.");
+                                (void) usleep_safe(0 * USEC_PER_SEC);
                         }
                 }
 
-- 
2.46.0

