From abc758fbac474598c10a393338c27362f94aedea Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <origincode@aosc.io>
Date: Tue, 11 Apr 2023 18:04:25 -0700
Subject: [PATCH 2/2] sleep.conf: AllowHibernation=no by default

Signed-off-by: Kaiyang Wu <origincode@aosc.io>
---
 src/sleep/sleep.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sleep/sleep.conf b/src/sleep/sleep.conf
index fad95b3897..a5c05dda26 100644
--- a/src/sleep/sleep.conf
+++ b/src/sleep/sleep.conf
@@ -18,7 +18,7 @@
 
 [Sleep]
 #AllowSuspend=yes
-#AllowHibernation=yes
+AllowHibernation=no
 #AllowSuspendThenHibernate=yes
 #AllowHybridSleep=yes
 #SuspendState=mem standby freeze
-- 
2.45.1

