From 0cf9b8ebccc3992988630711d4af9f3d5c51e000 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Sun, 10 Nov 2024 13:38:51 +0800
Subject: [PATCH 3/5] AOSCOS: mount-setup.c: do not sleep during boot

To make the user happy.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 src/shared/mount-setup.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/shared/mount-setup.c b/src/shared/mount-setup.c
index d5009fb59e..9e20aa7b7c 100644
--- a/src/shared/mount-setup.c
+++ b/src/shared/mount-setup.c
@@ -522,10 +522,10 @@ int mount_cgroup_legacy_controllers(bool loaded_policy) {
                 else if (cg_is_legacy_enabled()) {
                         log_full(LOG_CRIT,
                                  "Legacy cgroup v1 configured. This will stop being supported soon.\n"
-                                 "Will proceed with cgroup v2 after 30 s.\n"
+                                 "Will proceed with cgroup v2\n"
                                  "Set systemd.unified_cgroup_hierarchy=1 to switch to cgroup v2 "
                                  "or set SYSTEMD_CGROUP_ENABLE_LEGACY_FORCE=1 to reenable v1 temporarily.");
-                        (void) usleep_safe(30 * USEC_PER_SEC);
+                        (void) usleep_safe(0 * USEC_PER_SEC);
 
                         return 0;
                 }
-- 
2.43.0

