From b2104c594758cada4597478b0322e8eccf647f2f Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <origincode@aosc.io>
Date: Tue, 11 Apr 2023 18:04:25 -0700
Subject: [PATCH 3/3] sleep.conf: AllowHibernation=no by default

Signed-off-by: Kaiyang Wu <origincode@aosc.io>
---
 src/sleep/sleep.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sleep/sleep.conf b/src/sleep/sleep.conf
index 4c8e8b9680..3bd3ad1a08 100644
--- a/src/sleep/sleep.conf
+++ b/src/sleep/sleep.conf
@@ -14,7 +14,7 @@
 
 [Sleep]
 #AllowSuspend=yes
-#AllowHibernation=yes
+AllowHibernation=no
 #AllowSuspendThenHibernate=yes
 #AllowHybridSleep=yes
 #SuspendMode=
-- 
2.43.0

