From b74ccdc68ad585ad03a98b037ad0795276647cc7 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <origincode@aosc.io>
Date: Tue, 11 Apr 2023 18:04:25 -0700
Subject: [PATCH] sleep.conf: AllowHibernation=no by default

Signed-off-by: Kaiyang Wu <origincode@aosc.io>
---
 src/sleep/sleep.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sleep/sleep.conf b/src/sleep/sleep.conf
index a3d31140d8..bd15ebf50b 100644
--- a/src/sleep/sleep.conf
+++ b/src/sleep/sleep.conf
@@ -14,7 +14,7 @@
 
 [Sleep]
 #AllowSuspend=yes
-#AllowHibernation=yes
+AllowHibernation=no
 #AllowSuspendThenHibernate=yes
 #AllowHybridSleep=yes
 #SuspendMode=
-- 
2.39.1

