From f79b87d23b828cc47aa1d79dba37fc9442289970 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sun, 28 Jul 2019 00:35:07 +0800
Subject: [PATCH] AOSC disable version check

---
 src/winetricks | 51 --------------------------------------------------
 1 file changed, 51 deletions(-)

diff --git a/src/winetricks b/src/winetricks
index e5c5155..b3262b2 100755
--- a/src/winetricks
+++ b/src/winetricks
@@ -3036,55 +3036,6 @@ winetricks_get_platform()
     fi
 }
 
-winetricks_latest_version_check()
-{
-    if [ "$WINETRICKS_LATEST_VERSION_CHECK" = 'disabled' ] || [ -f "${WINETRICKS_CONFIG}/disable-latest-version-check" ] ; then
-        w_info "winetricks latest version check update disabled"
-        return
-    # Used by ./src/release.sh, not for end users. Silently disables update check, without using $WINETRICKS_SUPER_QUIET
-    elif [ "$WINETRICKS_LATEST_VERSION_CHECK" = 'development' ] ; then
-        return
-    fi
-
-    latest_version="$(winetricks_dl_url_to_stdout https://raw.githubusercontent.com/Winetricks/winetricks/master/files/LATEST)"
-
-    # Check that $latest_version is an actual number in case github is down
-    if ! echo "${latest_version}" | grep -q -E "[0-9]{8}" || [ -z "${latest_version}" ] ; then
-        case $LANG in
-            pl*) w_warn "GitHub nie działa? Wersja '${latest_version}' nie wydaje się być prawdiłową wersją" ;;
-            ru*) w_warn "Отсутствует подключение к Github? версия '${latest_version}' может быть неактуальной" ;;
-            *) w_warn "Github down? version '${latest_version}' doesn't appear to be a valid version" ;;
-        esac
-
-        # If we can't get the latest version, no reason to go further:
-        return
-    fi
-
-    if [ ! "$WINETRICKS_VERSION" = "${latest_version}" ] && [ ! "$WINETRICKS_VERSION" = "${latest_version}-next" ]; then
-        if [ -f "${WINETRICKS_CONFIG}/enable-auto-update" ] ; then
-            w_info "You are running winetricks-${WINETRICKS_VERSION}."
-            w_info "New upstream release winetricks-${latest_version} is available."
-            w_info "auto-update enabled: running winetricks_selfupdate"
-            winetricks_selfupdate
-        else
-            case $LANG in
-                pl*)
-                    w_warn "Korzystasz z winetricks-${WINETRICKS_VERSION}, a najnowszą wersją winetricks-${latest_version}!"
-                    w_warn "Zalecana jest aktualizacja z użyciem menedżera pakietów Twojej dystrybucji, --self-update lub ręczna aktualizacja."
-                    ;;
-                ru*)
-                    w_warn "Запущен winetricks-${WINETRICKS_VERSION}, последняя версия winetricks-${latest_version}!"
-                    w_warn "Вы можете ее обновить с помощью менеджера пакетов, --self-update или вручную."
-                    ;;
-                *)
-                    w_warn "You are running winetricks-${WINETRICKS_VERSION}, latest upstream is winetricks-${latest_version}!"
-                    w_warn "You should update using your distribution's package manager, --self-update, or manually."
-                    ;;
-            esac
-        fi
-    fi
-}
-
 winetricks_print_version()
 {
     # Normally done by winetricks_init, but we don't want to set up the WINEPREFIX
@@ -5182,8 +5133,6 @@ winetricks_init()
     if [ -z "$WINETRICKS_SUPER_QUIET" ] ; then
         echo "Using winetricks $(winetricks_print_version) with ${WINETRICKS_ORIG_WINE_VERSION} and WINEARCH=${W_ARCH}"
     fi
-
-    winetricks_latest_version_check
 }
 
 winetricks_usage()
-- 
2.22.0

