diff -Naur winetricks-20210206/src/winetricks winetricks-20210206.versioncheck/src/winetricks
--- winetricks-20210206/src/winetricks	2021-02-06 12:45:09.000000000 -0800
+++ winetricks-20210206.versioncheck/src/winetricks	2021-06-21 17:38:48.125258239 -0700
@@ -3362,65 +3362,6 @@
     fi
 }
 
-winetricks_latest_version_check()
-{
-    if [ "${WINETRICKS_LATEST_VERSION_CHECK}" = 'disabled' ] || [ -f "${WINETRICKS_CONFIG}/disable-latest-version-check" ] ; then
-        w_info "winetricks latest version check update disabled"
-        return
-    # Used by ./src/release.sh, not for end users. Silently disables update check, without using $WINETRICKS_SUPER_QUIET
-    elif [ "${WINETRICKS_LATEST_VERSION_CHECK}" = 'development' ] ; then
-        return
-    fi
-
-    latest_version="$(winetricks_dl_url_to_stdout https://raw.githubusercontent.com/Winetricks/winetricks/master/files/LATEST)"
-
-    # Check that $latest_version is an actual number in case github is down
-    if ! echo "${latest_version}" | grep -q -E "[0-9]{8}" || [ -z "${latest_version}" ] ; then
-        case ${LANG} in
-            pl*) w_warn "GitHub nie działa? Wersja '${latest_version}' nie wydaje się być prawdiłową wersją" ;;
-            ru*) w_warn "Отсутствует подключение к Github? версия '${latest_version}' может быть неактуальной" ;;
-            zh_CN*) w_warn "Github 提供的下载？${latest_version} 似乎不是个有效的版本号。" ;;
-            zh_TW*|zh_HK*) w_warn "Github 提供的下載？${latest_version} 似乎不是個有效的版本號。" ;;
-            *) w_warn "Github down? version '${latest_version}' doesn't appear to be a valid version" ;;
-        esac
-
-        # If we can't get the latest version, no reason to go further:
-        return
-    fi
-
-    if [ ! "${WINETRICKS_VERSION}" = "${latest_version}" ] && [ ! "${WINETRICKS_VERSION}" = "${latest_version}-next" ]; then
-        if [ -f "${WINETRICKS_CONFIG}/enable-auto-update" ] ; then
-            w_info "You are running winetricks-${WINETRICKS_VERSION}."
-            w_info "New upstream release winetricks-${latest_version} is available."
-            w_info "auto-update enabled: running winetricks_selfupdate"
-            winetricks_selfupdate
-        else
-            case ${LANG} in
-                pl*)
-                    w_warn "Korzystasz z winetricks-${WINETRICKS_VERSION}, a najnowszą wersją winetricks-${latest_version}!"
-                    w_warn "Zalecana jest aktualizacja z użyciem menedżera pakietów Twojej dystrybucji, --self-update lub ręczna aktualizacja."
-                    ;;
-                ru*)
-                    w_warn "Запущен winetricks-${WINETRICKS_VERSION}, последняя версия winetricks-${latest_version}!"
-                    w_warn "Вы можете ее обновить с помощью менеджера пакетов, --self-update или вручную."
-                    ;;
-                zh_CN*)
-                    w_warn "你正在使用 winetricks-${WINETRICKS_VERSION}，最新版本是 winetricks-${latest_version}!"
-                    w_warn "你应该使用你的发行版软件管理器、--self-update 或者手动更新。"
-                    ;;
-                zh_TW*|zh_HK*)
-                    w_warn "你正在使用 winetricks-${WINETRICKS_VERSION}，最新版本是 winetricks-${latest_version}!"
-                    w_warn "你應該使用你的發行版軟體管理者、--self-update 或者手動更新。"
-                    ;;
-                *)
-                    w_warn "You are running winetricks-${WINETRICKS_VERSION}, latest upstream is winetricks-${latest_version}!"
-                    w_warn "You should update using your distribution's package manager, --self-update, or manually."
-                    ;;
-            esac
-        fi
-    fi
-}
-
 winetricks_print_version()
 {
     # Normally done by winetricks_init, but we don't want to set up the WINEPREFIX
@@ -5416,8 +5357,6 @@
         return
     fi
 
-    winetricks_latest_version_check
-
     # Overridden for windows
     W_ISO_MOUNT_ROOT=/mnt/winetricks
     W_ISO_USER_MOUNT_ROOT="${HOME}"/winetricks-iso
