From cc277f2e2439d3d1eb09cd505081bc0a0dc96266 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 12 Jul 2026 17:35:28 +0800
Subject: [PATCH 1/5] AOSCOS: fix: add prefix to ppm.h include path

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 spudec.c       | 2 +-
 subtitle2pgm.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/spudec.c b/spudec.c
index d13d542..4ddb2e9 100644
--- a/spudec.c
+++ b/spudec.c
@@ -37,7 +37,7 @@ Revision:
 #endif
 
 #ifdef _HAVE_LIB_PPM_
-#include <ppm.h>
+#include <netpbm/ppm.h>
 #endif
 
 #ifdef _HAVE_PNG_
diff --git a/subtitle2pgm.c b/subtitle2pgm.c
index d77f2b1..e17abd2 100644
--- a/subtitle2pgm.c
+++ b/subtitle2pgm.c
@@ -28,7 +28,7 @@
 #include <unistd.h>  
 
 #ifdef _HAVE_LIB_PPM_
-#include <ppm.h>
+#include <netpbm/ppm.h>
 #endif
 
 #define READ_BUF_SIZE (64*1024)
-- 
2.52.0

