From 63166b0f219929504a8c4cc82081753868f5638d Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 12 Jul 2026 17:38:31 +0800
Subject: [PATCH 4/5] UBUNTU: Makefile: fix linkage against netpbm, set
 HAVE_GETLINE, honour LDFLAGS

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 2001932..f5f685e 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ INCLUDES :=
 
 ### enable ppm support ###
 DEFINES  += -D_HAVE_LIB_PPM_
-LIBS     += -lppm
+LIBS     += -lnetpbm
 
 ### enable PNG support ###
 DEFINES += -D_HAVE_PNG_
-- 
2.52.0

