--- subtitleripper-0.3.4.orig/pgm2txt	2004-11-18 13:59:04.000000000 +0100
+++ subtitleripper-0.3.4/pgm2txt	2004-11-18 14:00:25.000000000 +0100
@@ -16,7 +16,7 @@
 # Adjust this path to where the files
 # gocrfilter_en.sed  gocrfilter_fr.sed  gocrfilter_none.sed
 # are located
-PATH_TO_LANGUAGE_FILTER=~/sourceforge/subtitleripper/src/
+PATH_TO_LANGUAGE_FILTER=/usr/share/subtitleripper
 
 # set your options for gocr. Please read the gocr
 # docs to find out more about this
@@ -35,8 +35,8 @@
 # Choose your favorite image viewer if you want to see
 # the current pgm image while gocr is running.
 # Enable the viewer with -v command line option.
-IMAGE_VIWER=xv
-IMAGE_VIEWER_OPTIONS=-q
+IMAGE_VIEWER=display
+IMAGE_VIEWER_OPTIONS=
 
 ###### End of configuration section #########
 
@@ -108,8 +108,8 @@
 shift $(($OPTIND - 1))
 PGM_BASE_NAME=$1
     
-if [ -f ${PATH_TO_LANGUAGE_FILTER}gocrfilter_${LANGUAGE}.sed ]; then
-	FILTER_SCRIPT=${PATH_TO_LANGUAGE_FILTER}gocrfilter_${LANGUAGE}.sed
+if [ -f ${PATH_TO_LANGUAGE_FILTER}/gocrfilter_${LANGUAGE}.sed ]; then
+	FILTER_SCRIPT=${PATH_TO_LANGUAGE_FILTER}/gocrfilter_${LANGUAGE}.sed
 	echo "Using ${FILTER_SCRIPT} to filter gocr output"
 else
 	echo "    ------------------------------------------------"
@@ -153,7 +153,7 @@
 
     echo "Converting $i into text"
     if [ ! "$DISPLAY_PGM" = "false" ]; then
-	${IMAGE_VIWER} ${IMAGE_VIEWER_OPTIONS} $i &
+	${IMAGE_VIEWER} ${IMAGE_VIEWER_OPTIONS} $i &
     fi
     if [ "none" = "${LANGUAGE}" ]; then
 	gocr ${GOCR_OPTIONS} -p ${DB_PATH}  $i > $i.txt
@@ -163,7 +163,7 @@
 
     # close the viewer
     if [ ! "$DISPLAY_PGM" = "false" ]; then
-	killall ${IMAGE_VIWER}
+	killall ${IMAGE_VIEWER}
     fi
 
 done
