Index: subtitleripper-0.3.4/Makefile
===================================================================
--- subtitleripper-0.3.4.orig/Makefile	2012-07-27 15:09:43.324754534 +0200
+++ subtitleripper-0.3.4/Makefile	2012-07-27 15:10:51.803406418 +0200
@@ -9,7 +9,7 @@
 
 ### enable ppm support ###
 DEFINES  += -D_HAVE_LIB_PPM_
-LIBS     += -lppm
+LIBS     += -lnetpbm
 
 ### enable PNG support ###
 DEFINES += -D_HAVE_PNG_
 @@ -19,6 +19,9 @@
 DEFINES += -D_HAVE_ZLIB_
 LIBS    += -lz
 
+### getline is defined for us
+DEFINES += -DHAVE_GETLINE
+
 CC = gcc
 WARN     = -Wall -Wstrict-prototypes 
 COPT     = -g -O2
@@ -57,7 +57,7 @@
 
 %.o:%.c
 	@echo Compiling $<
-	@$(CC) -c $(CFLAGS) $<
+	@$(CC) -c $(CFLAGS) $(CPPFLAGS) $<
 
 # Dependencies
 subtitle2pgm.o: subtitle2pgm.c spudec.h subtitle2pgm.h
@@ -70,19 +70,19 @@
 # Target
 subtitle2pgm: subtitle2pgm.o spudec.o
 	@echo "Linking $@"
-	@$(CC) $(LIBS) $^ -o $@  
+	@$(CC) $(LIBS) $(LDFLAGS) $^ -o $@
 
 subtitle2vobsub: subtitle2vobsub.o vobsub.o 
 	@echo "Linking $@"
-	@$(CC) $(LIBS) $^ -o $@  
+	@$(CC) $(LIBS) $(LDFLAGS) $^ -o $@
 
 srttool: srttool.o
 	@echo "Linking $@"
-	@$(CC) $(LIBS) -g $^ -o $@  
+	@$(CC) $(LIBS) $(LDFLAGS) -g $^ -o $@
 
 vobsub2pgm: vobsub2pgm.o vobsub.o spudec.o
 	@echo "Linking $@"
-	@$(CC) $(LIBS) -g $^ -o $@  
+	@$(CC) $(LIBS) $(LDFLAGS) -g $^ -o $@
 
 .PHONY: clean dist rpm
 clean:
@@ -107,10 +107,3 @@
 	@rm $(PACKAGE)-$(VERSION)-$(RELEASE).tgz
 	sudo cp subtitleripper.spec  /usr/src/packages/SPECS/
 	sudo rpm -ba /usr/src/packages/SPECS/subtitleripper.spec
-
-
-
-
-
-
-
