From af4901abfbdce7e6865a62de2cdc25c62698c910 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Tue, 10 Mar 2026 19:27:38 +0800
Subject: [PATCH] aosc: use mirrorlist.pacstrap for bootstrapping Arch Linux
 sysroot

---
 etc/pacman.conf.in | 37 +++++++++++++------------------------
 1 file changed, 13 insertions(+), 24 deletions(-)

diff --git a/etc/pacman.conf.in b/etc/pacman.conf.in
index 73ec1c68..be6803d1 100644
--- a/etc/pacman.conf.in
+++ b/etc/pacman.conf.in
@@ -1,5 +1,5 @@
 #
-# @sysconfdir@/pacman.conf
+# /etc/pacman.conf
 #
 # See the pacman.conf(5) manpage for option and repository directives
 
@@ -9,12 +9,12 @@
 [options]
 # The following paths are commented out with their default values listed.
 # If you wish to use different paths, uncomment and update the paths.
-#RootDir     = @ROOTDIR@
-#DBPath      = @localstatedir@/lib/pacman/
-#CacheDir    = @localstatedir@/cache/pacman/pkg/
-#LogFile     = @localstatedir@/log/pacman.log
-#GPGDir      = @sysconfdir@/pacman.d/gnupg/
-#HookDir     = @sysconfdir@/pacman.d/hooks/
+#RootDir     = /
+#DBPath      = /var/lib/pacman/
+#CacheDir    = /var/cache/pacman/pkg/
+#LogFile     = /var/log/pacman.log
+#GPGDir      = /etc/pacman.d/gnupg/
+#HookDir     = /etc/pacman.d/hooks/
 HoldPkg     = pacman glibc
 #XferCommand = /usr/bin/curl -L -C - -f -o %o %u
 #XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
@@ -40,8 +40,8 @@ ParallelDownloads = 5
 #DisableSandboxSyscalls
 
 # PGP signature checking
-#SigLevel = Optional
-#LocalFileSigLevel = Optional
+SigLevel = Required DatabaseOptional
+LocalFileSigLevel = Optional
 #RemoteFileSigLevel = Optional
 
 #
@@ -61,20 +61,9 @@ ParallelDownloads = 5
 #
 # The header [repo-name] is crucial - it must be present and
 # uncommented to enable the repo.
-#
 
-# An example of a disabled remote package repository with multiple servers
-# available. To enable, uncomment the following lines. You can add preferred
-# servers immediately after the header and they will be used before the
-# default mirrors.
-#[core]
-#SigLevel = Required
-#Server = ftp://ftp.example.com/foobar/$repo/os/$arch/
-# The file referenced here should contain a list of 'Server = ' lines.
-#Include = @sysconfdir@/pacman.d/mirrorlist
+[core]
+Include = /etc/pacman.d/mirrorlist
 
-# An example of a custom package repository.  See the pacman manpage for
-# tips on creating your own repositories.
-#[custom]
-#SigLevel = Optional TrustAll
-#Server = file:///home/packages
+[extra]
+Include = /etc/pacman.d/mirrorlist
-- 
2.52.0

