From 274ba4df31cdea7bccecb5a70dbb904dfb388201 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Mon, 12 Jul 2021 15:29:25 -0400
Subject: [PATCH 004/105] ephyr: Don't open-code glamor_compile_glsl_prog

Reviewed-by: Emma Anholt <emma@anholt.net>
---
 hw/kdrive/ephyr/ephyr_glamor.c | 35 ++--------------------------------
 1 file changed, 2 insertions(+), 33 deletions(-)

diff --git a/hw/kdrive/ephyr/ephyr_glamor.c b/hw/kdrive/ephyr/ephyr_glamor.c
index 1e22cbd02..44e48ff59 100644
--- a/hw/kdrive/ephyr/ephyr_glamor.c
+++ b/hw/kdrive/ephyr/ephyr_glamor.c
@@ -69,37 +69,6 @@ struct ephyr_glamor {
     GLuint vao, vbo;
 };
 
-static GLint
-ephyr_glamor_compile_glsl_prog(GLenum type, const char *source)
-{
-    GLint ok;
-    GLint prog;
-
-    prog = glCreateShader(type);
-    glShaderSource(prog, 1, (const GLchar **) &source, NULL);
-    glCompileShader(prog);
-    glGetShaderiv(prog, GL_COMPILE_STATUS, &ok);
-    if (!ok) {
-        GLchar *info;
-        GLint size;
-
-        glGetShaderiv(prog, GL_INFO_LOG_LENGTH, &size);
-        info = malloc(size);
-        if (info) {
-            glGetShaderInfoLog(prog, size, NULL, info);
-            ErrorF("Failed to compile %s: %s\n",
-                   type == GL_FRAGMENT_SHADER ? "FS" : "VS", info);
-            ErrorF("Program source:\n%s", source);
-            free(info);
-        }
-        else
-            ErrorF("Failed to get shader compilation info.\n");
-        FatalError("GLSL compile failure\n");
-    }
-
-    return prog;
-}
-
 static GLuint
 ephyr_glamor_build_glsl_prog(GLuint vs, GLuint fs)
 {
@@ -156,8 +125,8 @@ ephyr_glamor_setup_texturing_shader(struct ephyr_glamor *glamor)
 
     GLuint fs, vs, prog;
 
-    vs = ephyr_glamor_compile_glsl_prog(GL_VERTEX_SHADER, vs_source);
-    fs = ephyr_glamor_compile_glsl_prog(GL_FRAGMENT_SHADER, fs_source);
+    vs = glamor_compile_glsl_prog(GL_VERTEX_SHADER, vs_source);
+    fs = glamor_compile_glsl_prog(GL_FRAGMENT_SHADER, fs_source);
     prog = ephyr_glamor_build_glsl_prog(vs, fs);
 
     glamor->texture_shader = prog;
-- 
2.45.2

