From 8e5124bb116c3c800692950ec10eb390d343aa94 Mon Sep 17 00:00:00 2001
From: nerdopolis <bluescreen_avenger@verizon.net>
Date: Fri, 8 Oct 2021 18:15:29 -0400
Subject: [PATCH 008/105] os: Try to discover the current seat with the
 XDG_SEAT var first

---
 os/utils.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/os/utils.c b/os/utils.c
index f0d022fba..3e4a4fa4f 100644
--- a/os/utils.c
+++ b/os/utils.c
@@ -685,6 +685,7 @@ ProcessCommandLine(int argc, char *argv[])
                     ErrorF("Failed to disable listen for %s transport",
                            defaultNoListenList[i]);
     }
+    SeatId = getenv("XDG_SEAT");
 
     for (i = 1; i < argc; i++) {
         /* call ddx first, so it can peek/override if it wants */
-- 
2.45.2

