From 278e743de7ba296242b0b58fa6b9687d44270330 Mon Sep 17 00:00:00 2001
From: JiangWu <wujiang@kylinos.cn>
Date: Tue, 9 Aug 2022 07:17:07 +0000
Subject: [PATCH 015/105] randr: Correctly get physical size for screen with
 RandR 1.5

---
 randr/rrmonitor.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/randr/rrmonitor.c b/randr/rrmonitor.c
index ec9cc3ef4..a0fb820d7 100644
--- a/randr/rrmonitor.c
+++ b/randr/rrmonitor.c
@@ -156,8 +156,8 @@ RRMonitorGetGeometry(RRMonitorPtr monitor, RRMonitorGeometryPtr geometry)
 
         /* Adjust physical sizes to account for total area */
         if (active_crtcs > 1 && first.box.x2 != first.box.x1 && first.box.y2 != first.box.y1) {
-            geometry->mmWidth = (this.box.x2 - this.box.x1) / (first.box.x2 - first.box.x1) * first.mmWidth;
-            geometry->mmHeight = (this.box.y2 - this.box.y1) / (first.box.y2 - first.box.y1) * first.mmHeight;
+            geometry->mmWidth = ((double)(geometry->box.x2 - geometry->box.x1) / (first.box.x2 - first.box.x1)) * first.mmWidth;
+            geometry->mmHeight = ((double)(geometry->box.y2 - geometry->box.y1) / (first.box.y2 - first.box.y1)) * first.mmHeight;
         }
     } else {
         *geometry = monitor->geometry;
-- 
2.45.2

