From 22e4c8970e62c14649a38bd62c9c9db2f6e7ee20 Mon Sep 17 00:00:00 2001
From: salieri <maliya355@outlook.com>
Date: Tue, 12 Mar 2024 17:08:33 +0800
Subject: [PATCH] configure.ac: link libwebp with -lwebp

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index b5684ea..e181a9b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -997,7 +997,7 @@ AC_ARG_ENABLE(webp, AS_HELP_STRING([--disable-webp],[Disable WebP file support.]
         fi
 ])
 if test x$fim_handle_webp = xyes; then
-	LIBS="$LIBS `pkg-config libwebpdemux --libs`";
+	LIBS="$LIBS `pkg-config libwebpdemux --libs` -lwebp";
 	CXXFLAGS="$CXXFLAGS `pkg-config libwebpdemux --cflags`";
 	AC_MSG_NOTICE([Enabling WebP file format support.])
 	AC_DEFINE(FIM_WITH_LIBWEBP, 1, [WebP file support.])
-- 
2.34.1

