From: xinmu <xinmu@xinmu.moe>
From: gray <3059305632@qq.com>
Subject: [PATCH] ghidra: architecture support (loongarch64)
---
diff --git a/GPL/DemanglerGnu/build.gradle b/GPL/DemanglerGnu/build.gradle
index 4b079cce79..ad6b36c169 100644
--- a/GPL/DemanglerGnu/build.gradle
+++ b/GPL/DemanglerGnu/build.gradle
@@ -83,6 +83,7 @@ model {
 			targetPlatform "mac_arm_64"
 			targetPlatform "freebsd_x86_64"
 			targetPlatform "freebsd_arm_64"
+                        targetPlatform "linux_loongarch_64"
 			sources {
 				c {
 					source {
@@ -106,7 +107,8 @@ model {
 			targetPlatform "mac_arm_64"
 			targetPlatform "freebsd_x86_64"
 			targetPlatform "freebsd_arm_64"
-			sources {
+			targetPlatform "linux_loongarch_64"
+                        sources {
 				c {
 					source {
 						srcDir srcVersion24 + "/c"
diff --git a/GPL/nativePlatforms.gradle b/GPL/nativePlatforms.gradle
index 0a2c02a2d2..1efffc17c2 100644
--- a/GPL/nativePlatforms.gradle
+++ b/GPL/nativePlatforms.gradle
@@ -23,6 +23,7 @@ project.ext.PLATFORMS = [
 	[name: "win_x86_64", os: "windows", arch: "x86_64"],
 	[name: "linux_x86_64", os: "linux", arch: "x86_64"],
 	[name: "linux_arm_64", os: "linux", arch: "arm64"],
+        [name: "linux_loongarch_64", os: "linux", arch: "loongarch64"],
 	[name: "mac_x86_64", os: "osx", arch: "x86_64"],
 	[name: "mac_arm_64", os: "osx", arch: "arm64"],
 	[name: "freebsd_x86_64", os: "freebsd", arch: "x86_64"],
@@ -76,6 +77,9 @@ ext.getCurrentPlatformName = {
 		case "arm64":
 			arch = "arm_64"
 			break
+                case "loongarch64":
+                        arch = "loongarch_64"
+                        break
 		default:
 			throw new GradleException("Unrecognized platform architecture: $arch" )
 	}
@@ -171,6 +175,20 @@ ext.isCurrentArm_64 = {
 	return isArm_64(getCurrentPlatformName())
 }
 
+/*********************************************************************************
+ * Returns true if the given platform is loongarch64.
+ *********************************************************************************/
+ ext.isloongarch64 = { platform_name ->
+	return platform_name.contains("loongarch64")
+ }
+
+ /*********************************************************************************
+ * Returns true if the current platform is loongarch64.
+ *********************************************************************************/
+ext.isCurrentloongarch64 = {
+	return isloongarch64(getCurrentPlatformName())
+}
+
 /****************************************************************************
  * Establish Visual Studio configuration environment for Windows native builds
  * NOTE: vsconfig.gradle path is relative to each GPL project module
diff --git a/Ghidra/Features/Decompiler/buildNatives.gradle b/Ghidra/Features/Decompiler/buildNatives.gradle
index 5b7edf8af2..3cea865d45 100644
--- a/Ghidra/Features/Decompiler/buildNatives.gradle
+++ b/Ghidra/Features/Decompiler/buildNatives.gradle
@@ -40,6 +40,7 @@ model {
 			targetPlatform "win_x86_64"
 			targetPlatform "linux_x86_64"
 			targetPlatform "linux_arm_64"
+			targetPlatform "linux_loongarch_64"
 			targetPlatform "mac_x86_64"
 			targetPlatform "mac_arm_64"
 			targetPlatform "freebsd_x86_64"
@@ -154,6 +155,7 @@ model {
 			targetPlatform "mac_arm_64"
 			targetPlatform "freebsd_x86_64"
 			targetPlatform "freebsd_arm_64"
+                        targetPlatform "linux_loongarch_64"
 			sources {
 				cpp {
 					// NOTE: The bison/flex generated files are assumed to be up-to-date.
diff --git a/Ghidra/Features/FileFormats/buildNatives.gradle b/Ghidra/Features/FileFormats/buildNatives.gradle
index 37bba0fdff..7e8f55439e 100644
--- a/Ghidra/Features/FileFormats/buildNatives.gradle
+++ b/Ghidra/Features/FileFormats/buildNatives.gradle
@@ -31,6 +31,7 @@ model {
 			targetPlatform "win_x86_64"
 			targetPlatform "linux_x86_64"
 			targetPlatform "linux_arm_64"
+			targetPlatform "linux_loongarch_64"
 			targetPlatform "mac_x86_64"
 			targetPlatform "mac_arm_64"
 			targetPlatform "freebsd_x86_64"
diff --git a/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Architecture.java b/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Architecture.java
index 1b0c7c37cd..f31dc4673b 100644
--- a/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Architecture.java
+++ b/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Architecture.java
@@ -21,6 +21,7 @@ public enum Architecture {
 	POWERPC("ppc"),
 	POWERPC_64("ppc64"),
 	ARM_64("aarch64", "arm64"),
+	LOONGARCH_64("loongarch64", "loong64"),
 	UNKNOWN("Unknown Architecture");
 
 	/**
diff --git a/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Platform.java b/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Platform.java
index 06e37c2afe..4336f1ccdf 100644
--- a/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Platform.java
+++ b/Ghidra/Framework/Generic/src/main/java/ghidra/framework/Platform.java
@@ -55,6 +55,11 @@ public enum Platform {
 	 */
 	LINUX_ARM_64(OperatingSystem.LINUX, Architecture.ARM_64, "linux_arm_64", ".so", ""),
 
+        /**
+         * Identifies a Linux LoongArch 64-bit OS.
+         */
+        LINUX_LOONGARCH_64(OperatingSystem.LINUX, Architecture.LOONGARCH_64, "linux_loongarch_64", ".so", ""),
+
 	/**
 	 * Identifies a macOS x86 32-bit OS.
 	 */
