diff -Naur gegl-0.4.32/tools/gegl-tester.c gegl-0.4.32.DIRTY.HACK/tools/gegl-tester.c
--- gegl-0.4.32/tools/gegl-tester.c	2022-01-02 00:19:35.457910034 -0600
+++ gegl-0.4.32.DIRTY.HACK/tools/gegl-tester.c	2022-01-02 00:19:19.358901818 -0600
@@ -144,7 +144,7 @@
   g_free (input_path);
   g_free (aux_path);
   g_free (output_path);
-  g_object_unref (composition);
+//g_object_unref (composition);
 }
 
 static gchar *
