diff -Naur gdal-3.0.2.old/swig/python/setup.py gdal-3.0.2/swig/python/setup.py
--- gdal-3.0.2.old/swig/python/setup.py	2022-02-07 15:08:27.735252241 +0800
+++ gdal-3.0.2/swig/python/setup.py	2022-02-07 15:08:32.814926976 +0800
@@ -129,9 +129,9 @@
         all_fixers = set(get_fixers_from_package('lib2to3.fixes'))
         exclude_fixers = sorted(all_fixers.difference(fixer_names))
 
-        extra['use_2to3'] = True
-        extra['use_2to3_fixers'] = []
-        extra['use_2to3_exclude_fixers'] = exclude_fixers
+        # extra['use_2to3'] = True
+        # extra['use_2to3_fixers'] = []
+        # extra['use_2to3_exclude_fixers'] = exclude_fixers
 
 
 class gdal_config_error(Exception):
