diff -Naur udisks-1.0.5/configure.ac udisks-1.0.5.mod/configure.ac
--- udisks-1.0.5/configure.ac	2014-03-10 03:57:48.000000000 -0500
+++ udisks-1.0.5.mod/configure.ac	2021-08-17 22:52:40.468787258 -0500
@@ -197,13 +197,8 @@
   LDFLAGS="-Wl,--as-needed $LDFLAGS"
 fi
 
-if test "$prefix" = "/usr" -o "$prefix" = "/usr/local" ; then
-  slashlibdir=/lib
-  slashsbindir=/sbin
-else
-  slashlibdir=$prefix/lib
-  slashsbindir=$prefix/sbin
-fi
+slashlibdir=$prefix/lib
+slashsbindir=$prefix/bin
 AC_SUBST(slashlibdir)
 AC_SUBST(slashsbindir)
 
