From 0ac94380b34bb2fc268909e43455a30e440c2eea Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Mon, 22 Dec 2025 16:34:48 +0800
Subject: [PATCH 6/6] [FEDORA] xine-lib 1.2.13 fix build with gcc_15 w32dll

---
 src/libw32dll/wine/win32.c | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/src/libw32dll/wine/win32.c b/src/libw32dll/wine/win32.c
index c481d61..87e336f 100644
--- a/src/libw32dll/wine/win32.c
+++ b/src/libw32dll/wine/win32.c
@@ -2517,10 +2517,10 @@ static int WINAPI expMonitorFromPoint(void *p, int flags)
 }
 
 static int WINAPI expEnumDisplayMonitors(void *dc, RECT *r, 
-    int WINAPI (*callback_proc)(), void *callback_param)
+    int WINAPI (*callback_proc) (int n, void *dc, RECT *r, void *param), void *callback_param)
 {
     dbgprintf("EnumDisplayMonitors(%p, %p, %p, %p) => ?\n",
-	dc, r, callback_proc, callback_param);
+        dc, (void *)r, (void *)callback_proc, callback_param);
     return callback_proc(0, dc, r, callback_param);
 }
 
@@ -2617,10 +2617,10 @@ static int WINAPI expCreateRectRgn(int x, int y, int width, int height)
     return 0;
 }
 
-static int WINAPI expEnumWindows(int (*callback_func)(), void *callback_param)
+static int WINAPI expEnumWindows(int (*callback_func) (int n, void *param), void *callback_param)
 {
     int i, i2;
-    dbgprintf("EnumWindows(%p, %p) => 1\n", callback_func, callback_param);
+    dbgprintf("EnumWindows(%p, %p) => 1\n", (void *)callback_func, callback_param);
     i = callback_func(0, callback_param);
     i2 = callback_func(1, callback_param);
     return i && i2;
@@ -3506,13 +3506,14 @@ static HANDLE WINAPI expCreateFileA(LPCSTR cs1,DWORD i1,DWORD i2,
     }
     if(strstr(cs1, ".qtx"))
     {
-	int result;
+	int result = -1;
 	char* x=strrchr(cs1,'\\');
 	char* tmp;
-	asprintf(&tmp,"%s/%s",win32_def_path,x?(x+1):cs1);
-//	printf("### Open: %s -> %s\n",cs1,tmp);
-	result=open(tmp, O_RDONLY);
-	free(tmp);
+        if (asprintf (&tmp,"%s/%s",win32_def_path,x?(x+1):cs1) >= 0) {
+            // printf("### Open: %s -> %s\n",cs1,tmp);
+            result = open (tmp, O_RDONLY);
+            free (tmp);
+        }
 	return result;
     }
 #endif
-- 
2.52.0

