diff -Naur tdenetwork-trinity-14.0.7/kppp/opener.cpp tdenetwork-trinity-14.0.7.gcc10/kppp/opener.cpp
--- tdenetwork-trinity-14.0.7/kppp/opener.cpp	2018-05-20 11:44:14.000000000 -0700
+++ tdenetwork-trinity-14.0.7.gcc10/kppp/opener.cpp	2020-12-02 22:13:39.064467994 -0800
@@ -342,7 +342,7 @@
 //
 int Opener::sendFD(int fd, struct ResponseHeader *response) {
 
-  struct { struct cmsghdr cmsg; int fd; } control;
+  struct { int fd; struct cmsghdr cmsg; } control;
   struct msghdr	msg;
   struct iovec iov;
 
diff -Naur tdenetwork-trinity-14.0.7/kppp/requester.cpp tdenetwork-trinity-14.0.7.gcc10/kppp/requester.cpp
--- tdenetwork-trinity-14.0.7/kppp/requester.cpp	2018-05-20 11:44:14.000000000 -0700
+++ tdenetwork-trinity-14.0.7.gcc10/kppp/requester.cpp	2020-12-02 22:15:00.619264072 -0800
@@ -80,7 +80,7 @@
 // Receive file name and file descriptors from envoy
 //
 int Requester::recvFD() {
-  struct { struct cmsghdr cmsg; int fd; } control;
+  struct { int fd; struct cmsghdr cmsg; } control;
   struct msghdr	msg;
   struct ResponseHeader response;
 
