diff -Naur kdeconnect-kde-22.04.3/app/org.kde.kdeconnect.app.desktop kdeconnect-kde-22.04.3.desktop/app/org.kde.kdeconnect.app.desktop
--- kdeconnect-kde-22.04.3/app/org.kde.kdeconnect.app.desktop	2022-07-02 03:16:09.000000000 +0000
+++ kdeconnect-kde-22.04.3.desktop/app/org.kde.kdeconnect.app.desktop	2022-08-27 06:29:13.174223075 +0000
@@ -143,4 +143,4 @@
 Icon=kdeconnect
 Type=Application
 Terminal=false
-Categories=Qt;KDE;Network
+Categories=Qt;KDE;Utility;
diff -Naur kdeconnect-kde-22.04.3/data/org.kde.kdeconnect_open.desktop kdeconnect-kde-22.04.3.desktop/data/org.kde.kdeconnect_open.desktop
--- kdeconnect-kde-22.04.3/data/org.kde.kdeconnect_open.desktop	2022-07-02 03:16:09.000000000 +0000
+++ kdeconnect-kde-22.04.3.desktop/data/org.kde.kdeconnect_open.desktop	2022-08-27 06:29:24.284421067 +0000
@@ -124,5 +124,5 @@
 Icon=kdeconnect
 Type=Service
 Terminal=false
-Categories=Qt;KDE;Network;
+Categories=Qt;KDE;Utility;
 X-KDE-FormFactor=desktop;tablet;handset;
diff -Naur kdeconnect-kde-22.04.3/indicator/org.kde.kdeconnect.nonplasma.desktop kdeconnect-kde-22.04.3.desktop/indicator/org.kde.kdeconnect.nonplasma.desktop
--- kdeconnect-kde-22.04.3/indicator/org.kde.kdeconnect.nonplasma.desktop	2022-07-02 03:16:09.000000000 +0000
+++ kdeconnect-kde-22.04.3.desktop/indicator/org.kde.kdeconnect.nonplasma.desktop	2022-08-27 06:29:37.418656067 +0000
@@ -92,6 +92,6 @@
 Icon=kdeconnect
 Type=Application
 Terminal=false
-Categories=Qt;KDE;Network;
+Categories=Qt;KDE;Utility;
 NotShowIn=KDE;
 
diff -Naur kdeconnect-kde-22.04.3/smsapp/org.kde.kdeconnect.sms.desktop kdeconnect-kde-22.04.3.desktop/smsapp/org.kde.kdeconnect.sms.desktop
--- kdeconnect-kde-22.04.3/smsapp/org.kde.kdeconnect.sms.desktop	2022-07-02 03:16:09.000000000 +0000
+++ kdeconnect-kde-22.04.3.desktop/smsapp/org.kde.kdeconnect.sms.desktop	2022-08-27 06:29:18.377315705 +0000
@@ -125,7 +125,7 @@
 Icon=kdeconnect
 Type=Application
 Terminal=false
-Categories=Qt;KDE;Network;InstantMessaging
+Categories=Qt;KDE;InstantMessaging;Utility;
 SingleMainWindow=true
 
 X-DBUS-StartupType=Unique
