diff -up telepathy-python-0.15.19/src/Makefile.am.autotools telepathy-python-0.15.19/src/Makefile.am
--- telepathy-python-0.15.19/src/Makefile.am.autotools	2010-06-09 20:28:59.000000000 +0200
+++ telepathy-python-0.15.19/src/Makefile.am	2011-06-14 21:19:42.692499952 +0200
@@ -13,10 +13,12 @@ telepathy_PYTHON = \
 spec_dir = $(top_srcdir)/spec
 spec_files := $(patsubst $(spec_dir)%.xml,_generated%.py,$(wildcard $(spec_dir)/*.xml))
 
+# We leave _generated/errors.py out of here because there exists a
+# spec/errors.xml file, so that means there'll be a
+# _generated/errors.py in $(spec_files). See fd.o#32526
 BUILT_SOURCES = \
 	_generated/interfaces.py \
 	_generated/constants.py \
-	_generated/errors.py \
 	_generated/__init__.py \
 	$(spec_files)
 
diff -up telepathy-python-0.15.19/src/Makefile.in.autotools telepathy-python-0.15.19/src/Makefile.in
--- telepathy-python-0.15.19/src/Makefile.in.autotools	2010-12-03 17:09:48.000000000 +0100
+++ telepathy-python-0.15.19/src/Makefile.in	2011-06-14 21:19:42.693499955 +0200
@@ -219,7 +219,6 @@ spec_files := $(patsubst $(spec_dir)%.xm
 BUILT_SOURCES = \
 	_generated/interfaces.py \
 	_generated/constants.py \
-	_generated/errors.py \
 	_generated/__init__.py \
 	$(spec_files)
 
