diff --git a/renderdoc/3rdparty/plthook/plthook_elf.c b/renderdoc/3rdparty/plthook/plthook_elf.c
index 6fc2ac6e3..04fc602b1 100644
--- a/renderdoc/3rdparty/plthook/plthook_elf.c
+++ b/renderdoc/3rdparty/plthook/plthook_elf.c
@@ -107,6 +107,9 @@
 #elif 0 /* disabled because not tested */ && (defined __ia64 || defined __ia64__)
 #define R_JUMP_SLOT   R_IA64_IPLTMSB
 #define Elf_Plt_Rel   Elf_Rela
+#elif defined __loongarch64
+#define R_JUMP_SLOT   R_LARCH_JUMP_SLOT
+#define Elf_Plt_Rel   Elf_Rela
 #else
 #error unsupported OS
 #endif
diff --git a/renderdoc/os/posix/linux/linux_process.cpp b/renderdoc/os/posix/linux/linux_process.cpp
index aa9c7393c..ec612910e 100644
--- a/renderdoc/os/posix/linux/linux_process.cpp
+++ b/renderdoc/os/posix/linux/linux_process.cpp
@@ -236,6 +236,24 @@ static uint64_t get_nanotime()
 #define BREAK_INST_BYTES_SIZE 4
 #define BREAK_INST_INST_PTR_ADJUST 4
 
+#elif defined(__powerpc64__)
+
+#define user_regs_struct struct pt_regs
+
+#define INST_PTR_REG nip
+
+#define BREAK_INST 0x0800e07fULL
+#define BREAK_INST_BYTES_SIZE 4
+#define BREAK_INST_INST_PTR_ADJUST 4
+
+#elif defined(__loongarch64)
+
+#define INST_PTR_REG csr_era
+
+#define BREAK_INST 0x002a0000ULL
+#define BREAK_INST_BYTES_SIZE 4
+#define BREAK_INST_INST_PTR_ADJUST 4
+
 #else
 
 #define BREAK_INST 0xccULL
