diff --git a/CMakeLists.txt b/CMakeLists.txt
index 536a859..1044177 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,7 +34,7 @@ endif ()
 
 # Find dependencies
 # Eigen3
-find_package(Eigen3 3.2 QUIET REQUIRED NO_MODULE)
+find_package(Eigen3 3.2...5 QUIET REQUIRED NO_MODULE)
 if (ENABLE_MKL)
     find_package(MKL)
     if (MKL_FOUND)
diff --git a/cmake/teaserppConfig.cmake b/cmake/teaserppConfig.cmake
index 554eab2..6c41de8 100644
--- a/cmake/teaserppConfig.cmake
+++ b/cmake/teaserppConfig.cmake
@@ -1,7 +1,7 @@
 get_filename_component(TEASERPP_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
 include(CMakeFindDependencyMacro)
 
-find_dependency(Eigen3 3.3 REQUIRED)
+find_dependency(Eigen3 3.3...5 REQUIRED)
 find_dependency(OpenMP REQUIRED)
 
 include("${TEASERPP_CMAKE_DIR}/teaserppTargets.cmake")
