--- VirtualBox-5.0.0/src/VBox/Devices/Makefile.kmk	2015-07-09 17:53:59.000000000 +0800
+++ VirtualBox-5.0.0-modified/src/VBox/Devices/Makefile.kmk	2015-07-24 17:18:51.726867542 +0800
@@ -24,9 +24,6 @@
 endif
 
 # Include sub-makefiles.
-include $(PATH_SUB_CURRENT)/testcase/Makefile.kmk
-include $(PATH_SUB_CURRENT)/Audio/testcase/Makefile.kmk
-include $(PATH_SUB_CURRENT)/Input/testcase/Makefile.kmk
 if defined(VBOX_WITH_INTEL_PXE) || defined(VBOX_ONLY_EXTPACKS)
  include $(PATH_SUB_CURRENT)/PC/PXE/Makefile.kmk
 else if defined(VBOX_WITH_PXE_ROM)
