--- a/configure	2016-04-29 13:09:09.064617982 -0600
+++ b/configure	2016-04-29 14:55:02.232572048 -0600
@@ -436,9 +436,9 @@
              -o \( $cc_maj -eq 4 -a $cc_min -lt 1 -a "$OS" != "darwin" \) \
              -o \( $cc_maj -eq 4 -a $cc_min -gt 9 \) \
              -o \( $cc_maj -eq 5 -a $cc_min -gt 3 \) \
-             -o \( $cc_maj -eq 6 -a $cc_min -gt 0 \) \
+             -o \( $cc_maj -eq 6 -a $cc_min -gt 9 \) \
              -o $cc_maj -gt 6 ]; then
-        log_failure "gcc version $cc_maj.$cc_min found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<10 or gcc 5.3"
+        log_failure "gcc version $cc_maj.$cc_min found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<10 or gcc 6.x 0<x<10"
         fail really
       else
         log_success "found version $cc_ver"
