--- a/src/interface/aboutdialog.cpp	2026-04-24 11:30:02.000000000 +0800
+++ b/src/interface/aboutdialog.cpp	2026-07-09 15:51:45.328116011 +0800
@@ -67,13 +67,13 @@
 
 		inner->Add(new wxStaticText(box, nullID, _("Compiled with:")));
 		wxString cc = CBuildInfo::GetCompiler();
-		WrapText(this, cc, 300);
+		WrapText(this, cc, 640);
 		inner->Add(new wxStaticText(box, nullID, cc));
 
 		wxString compilerFlags = CBuildInfo::GetCompilerFlags();
 		if (!compilerFlags.empty()) {
 			inner->Add(new wxStaticText(box, nullID, _("Compiler flags:")));
-			WrapText(this, compilerFlags, 300);
+			WrapText(this, compilerFlags, 640);
 			inner->Add(new wxStaticText(box, nullID, compilerFlags));
 		}
 	}
@@ -139,7 +139,7 @@
 		wxString cpuCaps = CBuildInfo::GetCPUCaps(' ');
 		if (!cpuCaps.empty()) {
 			inner->Add(new wxStaticText(box, nullID, _("CPU features:")));
-			WrapText(this, cpuCaps, 300);
+			WrapText(this, cpuCaps, 640);
 			inner->Add(new wxStaticText(box, nullID, cpuCaps));
 		}
 
