From e995fb0fbb83673a60568c371d58de0b0a67c950 Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Sun, 21 Jan 2024 23:42:47 -0800
Subject: [PATCH] Increase max text width on the about dialog

---
 src/interface/aboutdialog.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/interface/aboutdialog.cpp b/src/interface/aboutdialog.cpp
index 9b79edf..3e1f55a 100644
--- a/src/interface/aboutdialog.cpp
+++ b/src/interface/aboutdialog.cpp
@@ -63,13 +63,13 @@ bool CAboutDialog::Create(wxWindow* parent)
 
 		inner->Add(new wxStaticText(box, -1, _("Compiled with:")));
 		wxString cc = CBuildInfo::GetCompiler();
-		WrapText(this, cc, 300);
+		WrapText(this, cc, 640);
 		inner->Add(new wxStaticText(box, -1, cc));
 
 		wxString compilerFlags = CBuildInfo::GetCompilerFlags();
 		if (!compilerFlags.empty()) {
 			inner->Add(new wxStaticText(box, -1, _("Compiler flags:")));
-			WrapText(this, compilerFlags, 300);
+			WrapText(this, compilerFlags, 640);
 			inner->Add(new wxStaticText(box, -1, compilerFlags));
 		}
 	}
@@ -111,7 +111,7 @@ bool CAboutDialog::Create(wxWindow* parent)
 		wxString cpuCaps = CBuildInfo::GetCPUCaps(' ');
 		if (!cpuCaps.empty()) {
 			inner->Add(new wxStaticText(box, -1, _("CPU features:")));
-			WrapText(this, cpuCaps, 300);
+			WrapText(this, cpuCaps, 640);
 			inner->Add(new wxStaticText(box, -1, cpuCaps));
 		}
 
-- 
2.39.1

