diff -Naur e2fsprogs-1.43.4/misc/badblocks.c e2fsprogs-1.43.4.wcswidth/misc/badblocks.c
--- e2fsprogs-1.43.4/misc/badblocks.c	2016-09-01 23:17:32.000000000 -0500
+++ e2fsprogs-1.43.4.wcswidth/misc/badblocks.c	2017-04-23 12:18:20.180273737 -0500
@@ -229,7 +229,9 @@
 		       num_write_errors,
 		       num_corruption_errors);
 #ifdef HAVE_MBSTOWCS
-	len = mbstowcs(NULL, line_buf, sizeof(line_buf));
+	wchar_t wline_buf[128];
+	mbstowcs(wline_buf, line_buf, sizeof(line_buf));
+	len = wcswidth(wline_buf);
 #endif
 	fputs(line_buf, stderr);
 	memset(line_buf, '\b', len);
