diff -Naur boinc-client_release-7.16-7.16.16/configure.ac boinc-client_release-7.16-7.16.16.nonx86/configure.ac
--- boinc-client_release-7.16-7.16.16/configure.ac	2021-01-31 23:38:58.000000000 +0000
+++ boinc-client_release-7.16-7.16.16.nonx86/configure.ac	2023-08-15 10:19:36.735657509 +0000
@@ -674,11 +674,7 @@
 
 save_cxxflags="${CXXFLAGS}"
 save_cppflags="${CPPFLAGS}"
-CXXFLAGS="${CXXFLAGS} -msse3"
-CPPFLAGS="${CPPFLAGS} -msse3"
 AC_CHECK_HEADERS([intrin.h x86intrin.h pmmintrin.h xmmintrin.h emmintrin.h])
-CXXFLAGS="${save_cxxflags} -mavx"
-CPPFLAGS="${save_cppflags} -mavx"
 AC_CHECK_HEADERS([immintrin.h avxintrin.h])
 
 AC_CHECK_DECLS([_xgetbv, xgetbv, __xgetbv, cpuid, _cpuid, __cpuid],
