diff --git a/CMakeLists.txt b/CMakeLists.txt
index 56f4cc3..50f8f99 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -181,6 +181,7 @@ set (COMPILE_OPTIONS_INC opencl_clang_options.inc)
 
 find_program(LLVM_TABLEGEN_EXE "llvm-tblgen" ${LLVM_TOOLS_BINARY_DIR})
 set(LLVM_TARGET_DEFINITIONS ${COMPILE_OPTIONS_TD})
+set(LLVM_INCLUDE_DIRS "/usr/include")
 if(USE_PREBUILT_LLVM)
   set(TABLEGEN_ADDITIONAL -I ${LLVM_INCLUDE_DIRS})
 else(USE_PREBUILT_LLVM)
