From 4476f253c57c261d0c068a8b799ffc309fe22164 Mon Sep 17 00:00:00 2001
From: Wenwan Xing <wenwan.xing@intel.com>
Date: Wed, 16 Aug 2023 08:12:44 +0800
Subject: [PATCH] add 'Visibility' field to Option (#474)

---
 options.h           |  4 ++--
 options_compile.cpp | 12 ++++++------
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/options.h b/options.h
index 3e3ed38..5be4e1c 100644
--- a/options.h
+++ b/options.h
@@ -33,8 +33,8 @@ Copyright (c) Intel Corporation (2009-2017).
 enum COMPILE_OPT_ID {
   OPT_COMPILE_INVALID = 0, // This is not an option ID.
 #define PREFIX(NAME, VALUE)
-#define OPTION(PREFIX, NAME, ID, KIND, GROUP, ALIAS, ALIASARGS, FLAGS, PARAM,  \
-               HELPTEXT, METAVAR, VALUES)                                      \
+#define OPTION(PREFIX, NAME, ID, KIND, GROUP, ALIAS, ALIASARGS, FLAGS,         \
+               VISIBILITY, PARAM, HELPTEXT, METAVAR, VALUES)                   \
   OPT_COMPILE_##ID,
 #include "opencl_clang_options.inc"
   OPT_COMPILE_LAST_OPTION
diff --git a/options_compile.cpp b/options_compile.cpp
index 7b7aeb5..26693fb 100644
--- a/options_compile.cpp
+++ b/options_compile.cpp
@@ -35,8 +35,8 @@ Copyright (c) Intel Corporation (2009-2017).
   static constexpr llvm::StringLiteral NAME##_init[] = VALUE;                  \
   static constexpr llvm::ArrayRef<llvm::StringLiteral> NAME(                   \
       NAME##_init, std::size(NAME##_init) - 1);
-#define OPTION(PREFIX, NAME, ID, KIND, GROUP, ALIAS, ALIASARGS, FLAGS, PARAM,  \
-               HELPTEXT, METAVAR, VALUES)
+#define OPTION(PREFIX, NAME, ID, KIND, GROUP, ALIAS, ALIASARGS, FLAGS, VISIBILITY, \
+               PARAM, HELPTEXT, METAVAR, VALUES)
 #include "opencl_clang_options.inc"
 #undef OPTION
 #undef PREFIX
@@ -47,12 +47,12 @@ extern llvm::ManagedStatic<llvm::sys::SmartMutex<true>> compileMutex;
 
 static constexpr OptTable::Info ClangOptionsInfoTable[] = {
 #define PREFIX(NAME, VALUE)
-#define OPTION(PREFIX, NAME, ID, KIND, GROUP, ALIAS, ALIASARGS, FLAGS, PARAM,  \
-               HELPTEXT, METAVAR, VALUES)                                      \
+#define OPTION(PREFIX, NAME, ID, KIND, GROUP, ALIAS, ALIASARGS, FLAGS, VISIBILITY, \
+               PARAM, HELPTEXT, METAVAR, VALUES)                               \
   {                                                                            \
     PREFIX, NAME, HELPTEXT, METAVAR, OPT_COMPILE_##ID,                         \
-        llvm::opt::Option::KIND##Class, PARAM, FLAGS, OPT_COMPILE_##GROUP,     \
-        OPT_COMPILE_##ALIAS, ALIASARGS, VALUES                                 \
+        llvm::opt::Option::KIND##Class, VISIBILITY, PARAM, FLAGS,              \
+        OPT_COMPILE_##GROUP, OPT_COMPILE_##ALIAS, ALIASARGS, VALUES            \
   }                                                                            \
   ,
 #include "opencl_clang_options.inc"
-- 
2.44.0

