--- a/deps/llvm/CMakeLists.txt	2017-12-14 16:32:27.920598598 -0500
+++ b/deps/llvm/CMakeLists.txt	2017-12-14 16:32:48.923878133 -0500
@@ -207,8 +207,8 @@
 endif()
 
 if(UNIX)
-	target_link_libraries(llvm INTERFACE debug     ${ZLIB_LIBRARIES} rt dl tinfo)
-	target_link_libraries(llvm INTERFACE optimized ${ZLIB_LIBRARIES} rt dl tinfo)
+	target_link_libraries(llvm INTERFACE debug     ${ZLIB_LIBRARIES} rt dl ncurses)
+	target_link_libraries(llvm INTERFACE optimized ${ZLIB_LIBRARIES} rt dl ncurses)
 endif()
 
 # Set include directories.
--- a/deps/fnc-patterns/deps/fileformat/deps/llvm/CMakeLists.txt	2017-12-14 16:32:27.920598598 -0500
+++ b/deps/fnc-patterns/deps/fileformat/deps/llvm/CMakeLists.txt	2017-12-14 16:32:48.923878133 -0500
@@ -207,8 +207,8 @@
 endif()
 
 if(UNIX)
-	target_link_libraries(llvm INTERFACE debug     ${ZLIB_LIBRARIES} rt dl tinfo)
-	target_link_libraries(llvm INTERFACE optimized ${ZLIB_LIBRARIES} rt dl tinfo)
+	target_link_libraries(llvm INTERFACE debug     ${ZLIB_LIBRARIES} rt dl ncurses)
+	target_link_libraries(llvm INTERFACE optimized ${ZLIB_LIBRARIES} rt dl ncurses)
 endif()
 
 # Set include directories.
--- a/deps/capstone2llvmir/deps/llvm/CMakeLists.txt	2017-12-14 16:32:27.920598598 -0500
+++ b/deps/capstone2llvmir/deps/llvm/CMakeLists.txt	2017-12-14 16:32:48.923878133 -0500
@@ -207,8 +207,8 @@
 endif()
 
 if(UNIX)
-	target_link_libraries(llvm INTERFACE debug     ${ZLIB_LIBRARIES} rt dl tinfo)
-	target_link_libraries(llvm INTERFACE optimized ${ZLIB_LIBRARIES} rt dl tinfo)
+	target_link_libraries(llvm INTERFACE debug     ${ZLIB_LIBRARIES} rt dl ncurses)
+	target_link_libraries(llvm INTERFACE optimized ${ZLIB_LIBRARIES} rt dl ncurses)
 endif()
 
 # Set include directories.
--- a/deps/fileformat/deps/llvm/CMakeLists.txt	2017-12-14 16:32:27.920598598 -0500
+++ b/deps/fileformat/deps/llvm/CMakeLists.txt	2017-12-14 16:32:48.923878133 -0500
@@ -207,8 +207,8 @@
 endif()
 
 if(UNIX)
-	target_link_libraries(llvm INTERFACE debug     ${ZLIB_LIBRARIES} rt dl tinfo)
-	target_link_libraries(llvm INTERFACE optimized ${ZLIB_LIBRARIES} rt dl tinfo)
+	target_link_libraries(llvm INTERFACE debug     ${ZLIB_LIBRARIES} rt dl ncurses)
+	target_link_libraries(llvm INTERFACE optimized ${ZLIB_LIBRARIES} rt dl ncurses)
 endif()
 
 # Set include directories.
