--- a/deps/llvm/CMakeLists.txt	2018-02-08 02:07:16.763365489 -0700
+++ b/deps/llvm/CMakeLists.txt	2018-02-08 02:07:34.304622903 -0700
@@ -156,8 +156,8 @@
 	target_link_libraries(llvm INTERFACE optimized ${ZLIB_LIBRARIES} dl)
 
 	if(NOT APPLE)
-		target_link_libraries(llvm INTERFACE debug     rt tinfo)
-		target_link_libraries(llvm INTERFACE optimized rt tinfo)
+		target_link_libraries(llvm INTERFACE debug     rt ncurses)
+		target_link_libraries(llvm INTERFACE optimized rt ncurses)
 	else()
 		target_link_libraries(llvm INTERFACE debug     curses)
 		target_link_libraries(llvm INTERFACE optimized curses)
