From fb5acc2d1bb633a42bf3bf8ef33752c9d0b059d1 Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 17 May 2025 09:02:32 +0800
Subject: AOSCOS: Use system FreeType

---
 meson.options | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.options b/meson.options
index 7e03ff405f85..4dede215814e 100644
--- a/meson.options
+++ b/meson.options
@@ -5,7 +5,7 @@
 # Qhull. You may use the following options to instead link against a system
 # FreeType/Qhull. As an exception, Matplotlib defaults to the system version of
 # FreeType on AIX.
-option('system-freetype', type: 'boolean', value: false,
+option('system-freetype', type: 'boolean', value: true,
        description: 'Build against system version of FreeType')
 option('system-libraqm', type: 'boolean', value: false,
        description: 'Build against system version of libraqm')
-- 
2.34.1

