--- a/Cbc/src/CbcModel.hpp	2017-06-19 14:21:06.165941996 +0800
+++ b/Cbc/src/CbcModel.hpp	2017-06-19 13:58:45.759538950 +0800
@@ -2394,8 +2394,8 @@
     void redoWalkBack();
     //@}
     
-    void setMIPStart( const std::vector< std::pair< std::string, double > > &mips ) {
-       this->mipStart_ = mips;
+    void setMIPStart( const std::vector< std::pair< std::string, double > > &Mips ) {
+       this->mipStart_ = Mips;
     }
 
     const std::vector< std::pair< std::string, double > > &getMIPStart() {
