diff -Naur budgie-desktop-10.5.1/src/wm/meson.build budgie-desktop-10.5.1.mutter36/src/wm/meson.build
--- budgie-desktop-10.5.1/src/wm/meson.build	2020-03-19 14:58:23.124533499 +0000
+++ budgie-desktop-10.5.1.mutter36/src/wm/meson.build	2020-03-19 14:58:48.959351147 +0000
@@ -26,8 +26,8 @@
         vapi_mutter = 'libmutter-4'
     else
         dep_mutter = dependency('libmutter-5', version: gnome_minimum_version, required: false)
-        budgie_wm_status_vala_args += ['-D', 'HAVE_MUTTER_5']
         if dep_mutter.found()
+            budgie_wm_status_vala_args += ['-D', 'HAVE_MUTTER_5']
             message('Using new libmutter-5 ABI from GNOME 3.34')
             vapi_mutter = 'libmutter-5'
         else
diff -Naur budgie-desktop-10.5.1/src/wm/shim.vala budgie-desktop-10.5.1.mutter36/src/wm/shim.vala
--- budgie-desktop-10.5.1/src/wm/shim.vala	2019-10-03 13:50:03.000000000 +0000
+++ budgie-desktop-10.5.1.mutter36/src/wm/shim.vala	2020-03-19 14:59:50.647915705 +0000
@@ -193,7 +193,7 @@
         osd_proxy = null;
     }
 
-#if HAVE_MUTTER_5
+#if HAVE_MUTTER_5 || HAVE_MUTTER_6
     private void on_accelerator_activated(uint action, Clutter.InputDevice dev, uint device_id)
 #else
     private void on_accelerator_activated(uint action, uint device_id)
diff -Naur budgie-desktop-10.5.1/vapi/libmutter-6.vapi budgie-desktop-10.5.1.mutter36/vapi/libmutter-6.vapi
--- budgie-desktop-10.5.1/vapi/libmutter-6.vapi	2020-03-19 14:58:23.125533492 +0000
+++ budgie-desktop-10.5.1.mutter36/vapi/libmutter-6.vapi	2020-03-19 14:58:48.959351147 +0000
@@ -368,7 +368,6 @@
 		public uint add_idle_watch (uint64 interval_msec, owned Meta.IdleMonitorWatchFunc? callback);
 		public uint add_user_active_watch (owned Meta.IdleMonitorWatchFunc? callback);
 		public static unowned Meta.IdleMonitor get_core ();
-		public static unowned Meta.IdleMonitor get_for_device (Clutter.InputDevice device);
 		public int64 get_idletime ();
 		public void remove_watch (uint id);
 		[NoAccessorMethod]
diff -Naur budgie-desktop-10.5.1/vapi/mutter-clutter-6.vapi budgie-desktop-10.5.1.mutter36/vapi/mutter-clutter-6.vapi
--- budgie-desktop-10.5.1/vapi/mutter-clutter-6.vapi	2020-03-19 14:58:23.127533478 +0000
+++ budgie-desktop-10.5.1.mutter36/vapi/mutter-clutter-6.vapi	2020-03-19 14:58:48.960351140 +0000
@@ -6731,6 +6731,7 @@
 		public virtual unowned Clutter.InputDevice get_pointer ();
 		public void get_pointer_a11y_settings (Clutter.PointerA11ySettings settings);
 		public virtual Clutter.VirtualDeviceType get_supported_virtual_device_types ();
+		public bool get_touch_mode ();
 		public virtual GLib.List<weak Clutter.InputDevice> list_devices ();
 		public void set_kbd_a11y_settings (Clutter.KbdA11ySettings settings);
 		public void set_pointer_a11y_dwell_click_type (Clutter.PointerA11yDwellClickType click_type);
@@ -6738,6 +6739,7 @@
 		public virtual void warp_pointer (int x, int y);
 		[NoAccessorMethod]
 		public Clutter.Backend backend { owned get; construct; }
+		public bool touch_mode { get; }
 		public signal void device_added (Clutter.InputDevice object);
 		public signal void device_removed (Clutter.InputDevice object);
 		public signal void kbd_a11y_flags_changed (uint settings_flags, uint changed_mask);
diff -Naur budgie-desktop-10.5.1/vapi/mutter-cogl-6.vapi budgie-desktop-10.5.1.mutter36/vapi/mutter-cogl-6.vapi
--- budgie-desktop-10.5.1/vapi/mutter-cogl-6.vapi	2020-03-19 14:58:23.128533471 +0000
+++ budgie-desktop-10.5.1.mutter36/vapi/mutter-cogl-6.vapi	2020-03-19 14:58:48.960351140 +0000
@@ -1046,18 +1046,9 @@
 	public static void set_depth_test_enabled (bool setting);
 	[CCode (cheader_filename = "cogl/cogl.h")]
 	[Version (deprecated = true, deprecated_since = "1.16")]
-	public static void shader_compile (Cogl.Handle handle);
-	[CCode (cheader_filename = "cogl/cogl.h")]
-	[Version (deprecated = true, deprecated_since = "1.16")]
-	public static string shader_get_info_log (Cogl.Handle handle);
-	[CCode (cheader_filename = "cogl/cogl.h")]
-	[Version (deprecated = true, deprecated_since = "1.16")]
 	public static Cogl.ShaderType shader_get_type (Cogl.Handle handle);
 	[CCode (cheader_filename = "cogl/cogl.h")]
 	[Version (deprecated = true, deprecated_since = "1.16")]
-	public static bool shader_is_compiled (Cogl.Handle handle);
-	[CCode (cheader_filename = "cogl/cogl.h")]
-	[Version (deprecated = true, deprecated_since = "1.16")]
 	public static void shader_source (Cogl.Handle shader, string source);
 	[CCode (cheader_filename = "cogl/cogl.h")]
 	[Version (deprecated = true, deprecated_since = "1.18", since = "1.0")]
