Description: Do not package non-free files
Author: Thomas Goirand <zigo@debian.org>
Forwarded: not-needed
Last-Update: 2015-10-15

--- python-pulp-1.6.0+dfsg1.orig/setup.py
+++ python-pulp-1.6.0+dfsg1/setup.py
@@ -44,27 +44,8 @@ problems.
       #ext_modules = [pulpCOIN],
       package_dir={'':'src'},
       #need the cbc directories here as the executable bit is set
-      packages = ['pulp', 
-      'pulp.solverdir',
-      'pulp.solverdir.cbc.linux.32',
-      'pulp.solverdir.cbc.linux.64',
-      'pulp.solverdir.cbc.win.32',
-      'pulp.solverdir.cbc.win.64',
-      'pulp.solverdir.cbc.osx.64'],
-      package_data = {'pulp' : ["AUTHORS","LICENSE",
-                                "pulp.cfg.linux",
-                                "pulp.cfg.win",
-                                "pulp.cfg.osx",
-                                "LICENSE.CoinMP.txt",
-                                "AUTHORS.CoinMP.txt",
-                                "README.CoinMP.txt",
-                                ],
-                      'pulp.solverdir.cbc.linux.32' : ['*','*.*'],
-                      'pulp.solverdir.cbc.linux.64' : ['*','*.*'],
-                      'pulp.solverdir.cbc.win.32' : ['*','*.*'],
-                      'pulp.solverdir.cbc.win.64' : ['*','*.*'],
-                      'pulp.solverdir.cbc.osx.64' : ['*','*.*'],
-                      },
+      packages = ['pulp'],
+      package_data = {'pulp' : ["pulp.cfg.linux"]},
       install_requires = [pyparsing_ver],
       entry_points = ("""
       [console_scripts]

