From 88645e977b42c4d47d733f9a434cf94c31cb0cff Mon Sep 17 00:00:00 2001
From: relictMarauder <relict_marauder@bonelis.de>
Date: Wed, 29 Jun 2016 21:42:08 +0200
Subject: [PATCH] sync to attr from 2015-04-14(Remove <attr/xattr.h> and the
 syscall wrappers)

---
 src/xattr.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/xattr.h b/src/xattr.h
index 182040a..1b61e43 100644
--- a/src/xattr.h
+++ b/src/xattr.h
@@ -25,7 +25,10 @@
  * Linux
  */
 #if PLATFORM == PLATFORM_LINUX
-#include <attr/xattr.h>
+#ifndef ENOATTR
+# define ENOATTR ENODATA
+#endif
+#include <sys/xattr.h>
 
 #define EXTATTR_NAMESPACE_USER 0
 #define EXTATTR_NAMESPACE_SYSTEM 0

