From a7002bb23f91466c7d18f8cfc1800b2e68a78f58 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Fri, 13 Sep 2024 23:53:58 +0800
Subject: [PATCH 1/2] Use `lurk` git source to support aarch64

---
 Cargo.lock | 3 +--
 Cargo.toml | 2 +-
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/Cargo.lock b/Cargo.lock
index 16a0d1c..2d0c5b6 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -965,8 +965,7 @@ dependencies = [
 [[package]]
 name = "lurk-cli"
 version = "0.3.6"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "bbf13337addf721f044c89275e9afda5387a56282869fe859c4fccf31b353bce"
+source = "git+https://github.com/JakWai01/lurk?rev=f28e066dc5da831d2b3db66eb43d4677864d0c54#f28e066dc5da831d2b3db66eb43d4677864d0c54"
 dependencies = [
  "anyhow",
  "atty",
diff --git a/Cargo.toml b/Cargo.toml
index 708ee9e..cb92184 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -30,7 +30,7 @@ tui-input = "0.10.1"
 tui-popup = "0.5.0"
 unicode-width = "0.1.11"
 textwrap = "0.16.1"
-lurk-cli = "0.3.6"
+lurk-cli = { git = "https://github.com/JakWai01/lurk", rev = "f28e066dc5da831d2b3db66eb43d4677864d0c54" }
 nix = { version = "0.29.0", features = ["ptrace", "signal"] }
 ansi-to-tui = "6.0.0"
 console = "0.15.8"
-- 
2.46.0

