diff -Naur hardinfo-0.5.1/Makefile.in hardinfo-0.5.1.gmodule/Makefile.in
--- hardinfo-0.5.1/Makefile.in	2009-04-06 08:43:15.000000000 -0500
+++ hardinfo-0.5.1.gmodule/Makefile.in	2020-04-16 15:12:14.735400745 -0500
@@ -15,7 +15,7 @@
 
 all:	$(OBJECTS) $(MODULES)
 	$(CC) $(CCFLAGS) -o hardinfo -Wl,-export-dynamic $(OBJECTS) $(GTK_LIBS) \
-		$(SOUP_LIBS)
+		$(SOUP_LIBS) -lgmodule-2.0 -lm
 
 md5.o:
 	$(CCSLOW) $(CCFLAGS) $(CFLAGS) -c md5.c -o $@
