--- hardinfo-0.5.1.orig/Makefile.in
+++ hardinfo-0.5.1/Makefile.in
@@ -1,6 +1,6 @@
 
 CCFLAGS = -fPIC -pipe -Wall -g 
-CFLAGS = $(GTK_CFLAGS) $(SOUP_CFLAGS) -I.
+CFLAGS = $(GTK_CFLAGS) $(SOUP_CFLAGS) -I. -std=gnu89
 CC = gcc $(ARCHOPTS) -g
 CCSLOW = gcc -O0 -g
 
--- hardinfo-0.5.1.orig/binreloc.c
+++ hardinfo-0.5.1/binreloc.c
@@ -594,7 +594,7 @@
     }
 
 #ifdef ARCH_x86_64
-    dir = g_build_filename(prefix, "lib64", NULL);
+    dir = g_build_filename(prefix, "lib", NULL);
 #else
     dir = g_build_filename(prefix, "lib", NULL);
 #endif
--- hardinfo-0.5.1.orig/config.h
+++ hardinfo-0.5.1/config.h
@@ -0,0 +1,17 @@
+#ifndef __CONFIG_H__
+#define __CONFIG_H__
+
+#define VERSION "0.5.1"
+#define ARCH_i386
+#define ARCH     "ARCH_i386"
+#define PLATFORM "Linux"
+#define KERNEL   "2.6.30-1-686"
+#define HOSTNAME "comanche"
+#define PREFIX "/usr/share/hardinfo/"
+#define LIBPREFIX "/usr/lib/hardinfo/"
+#define HAS_LINUX_WE
+#define DEBUG(...)
+#define ENABLE_BINRELOC 1
+#define RELEASE 1
+
+#endif	/* __CONFIG_H__ */
--- hardinfo-0.5.1.orig/configure
+++ hardinfo-0.5.1/configure
@@ -56,7 +56,7 @@
 	x86_64)
 		ln -sf linux/x86_64 arch/this
 		ARCH="ARCH_x86_64"
-		LIBDIR="/usr/lib64" ;;
+		LIBDIR="/usr/lib" ;;
 	mips*)
 		ln -sf linux/mips arch/this
 		ARCH="ARCH_MIPS" ;;
--- hardinfo-0.5.1.orig/debian/changelog
+++ hardinfo-0.5.1/debian/changelog
@@ -0,0 +1,280 @@
+hardinfo (0.5.1-1.5) unstable; urgency=medium
+
+  * Non-maintainer upload.
+  * Fix "FTBFS with clang instead of gcc":
+    patch Makefile* to add "-std=gnu89" to CFLAGS.
+    (Closes: #757525)
+
+ -- gregor herrmann <gregoa@debian.org>  Wed, 22 Jul 2015 17:51:45 +0200
+
+hardinfo (0.5.1-1.4) unstable; urgency=medium
+
+  * Non-maintainer upload.
+  * fix FTBFS "ld: cannot find -lselinux" by adding a build-dep on
+    libselinux1-dev [!kfreebsd-i386 !kfreebsd-amd64 !hurd-i386]
+    (Closes: #747059).
+
+ -- HIGUCHI Daisuke (VDR dai) <dai@debian.org>  Wed, 07 May 2014 21:12:18 +0900
+
+hardinfo (0.5.1-1.3) unstable; urgency=medium
+
+  * Non-maintainer upload.
+  * Fix FTBFS "ld: cannot find -llzma" by adding a build-dep on liblzma-dev.
+    (Closes: #713717)
+
+ -- Vincent Cheng <vcheng@debian.org>  Tue, 25 Mar 2014 15:47:36 -0700
+
+hardinfo (0.5.1-1.2) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Fix "FTBFS: ld: cannot find -lffi":
+    add patch from Ubuntu / Julien Lavergne:
+    + debian/control:
+      - Build-depends on libffi-dev to fix FTBFS (LP: #831185)
+    (Closes: #652179)
+
+ -- gregor herrmann <gregoa@debian.org>  Thu, 05 Jan 2012 16:14:54 +0100
+
+hardinfo (0.5.1-1.1) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * [cd677eda] Move files from /usr/lib64 to /usr/lib. (Closes: #557569)
+    - thanks to Kurt Roeckx
+
+ -- Bernd Zeimetz <bzed@debian.org>  Fri, 22 Jan 2010 21:22:24 +0100
+
+hardinfo (0.5.1-1) unstable; urgency=low
+
+  * New upstream version. (Closes: #530290, #517591)
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Thu, 24 Sep 2009 15:13:54 -0300
+
+hardinfo (0.5c-1) unstable; urgency=low
+
+  * New upstream version. 
+     (Closes: #517591, #511237, #457703, #519256, #449250, #457820, #497758) 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Sat, 28 Mar 2009 22:55:02 -0300
+
+hardinfo (0.4.2.3-5) unstable; urgency=low
+
+  * Fixed location on menu. (Closes: #487569) 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Sun, 06 Jul 2008 19:01:51 -0300
+
+hardinfo (0.4.2.3-4) unstable; urgency=low
+
+  * Fixed LSPCIPATH on configure. (Closes: #482381) 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Tue, 27 May 2008 21:17:23 -0300
+
+hardinfo (0.4.2.3-3) unstable; urgency=low
+
+  * changed modules install dir for amd64 
+    (thanks to ucko@d.o). (Closes: #468105) 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Tue, 26 Feb 2008 23:52:45 -0300
+
+hardinfo (0.4.2.3-2) unstable; urgency=low 
+
+  * fixed desktop file, thanks to Kjell Braden. (Closes: #466587) 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Sun, 24 Feb 2008 23:35:44 -0300
+
+hardinfo (0.4.2.3-1) unstable; urgency=low
+
+  * New upstream release.(Closes: #454947) 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Sun, 09 Dec 2007 05:03:40 -0400
+
+hardinfo (0.4.2.2-3) unstable; urgency=low
+
+  * Fixed FTBFS on s390. (Closes: #443293) 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Thu, 20 Sep 2007 16:59:20 -0400
+
+hardinfo (0.4.2.2-2) unstable; urgency=low
+
+  * Fixed FTBFS. (Closes: #442594) 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Mon, 17 Sep 2007 15:51:12 -0300
+
+hardinfo (0.4.2.2-1) unstable; urgency=low
+
+  * New upstream release.
+  * hardinfo now suggests mesa-utils (Closes: #435793)
+
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Tue, 11 Sep 2007 14:53:15 -0300
+
+hardinfo (0.4.2.1.svn30042007-1) unstable; urgency=low
+
+  * Fixed FTBFS on SVN version (Closes: #419587)
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Mon, 30 Apr 2007 14:52:35 -0300
+
+hardinfo (0.4.2.1-1) unstable; urgency=low
+
+  * New upstream release.
+  * Fixed segfault on device entry (Closes: #387955)
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Mon, 09 Apr 2007 11:45:31 -0300
+
+hardinfo (0.4.1-9) unstable; urgency=low
+
+  * Fixed segfaults on powerpc (Closes: #408050)
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Wed, 21 Feb 2007 21:10:57 -0200
+
+hardinfo (0.4.1-8) unstable; urgency=low
+
+  * Fixed errors on processor.h on ia64 and alpha. 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Tue, 12 Dec 2006 09:30:53 -0300
+
+hardinfo (0.4.1-7) unstable; urgency=low
+
+  * Fixed FTBFS on s390, ia64 and alpha. 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Mon, 11 Dec 2006 20:40:14 -0200
+
+hardinfo (0.4.1-6) unstable; urgency=low
+
+  * Changed configure script to reconize m68k machines 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Sun, 10 Dec 2006 18:38:14 -0200
+
+hardinfo (0.4.1-3) unstable; urgency=low
+
+  * Changed libz path (Closes: #391963)
+  * added support to alpha arm ia64 m68k s390
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Wed, 11 Oct 2006 23:56:11 -0300
+
+hardinfo (0.4.1-2) unstable; urgency=low
+
+  * fixed debian/watch file.
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Fri, 23 Jun 2006 00:35:08 -0300
+
+hardinfo (0.4.1-1) unstable; urgency=low
+
+  * New upstream release. 
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Wed, 21 Jun 2006 22:18:24 -0300
+
+hardinfo (0.4-3) unstable; urgency=low
+
+  * Updated project's homepage, old is unreachable. (Closes: #370797)
+  * upated printers.h (Closes: #370363)
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Fri,  9 Jun 2006 08:51:12 -0300
+
+hardinfo (0.4-2) unstable; urgency=low
+
+  * Limited supported architectures. (Closes: #368582)
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Thu, 25 May 2006 22:13:49 -0300
+
+hardinfo (0.4-1) unstable; urgency=low
+
+  * new upstream release
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Mon, 22 May 2006 19:43:53 -0300
+
+hardinfo (0.3.7pre-4) unstable; urgency=low
+
+  * added ia64 support. (Closes: #362175)
+
+ -- Agney Lopes Roth Ferraz <agney@debian.org>  Sat, 20 May 2006 16:31:36 -0300
+
+hardinfo (0.3.7pre-3) unstable; urgency=low
+
+  * added mips64 support. (Closes: #361640)
+
+ -- Agney Lopes Roth Ferraz <agney@users.sourceforge.net>  Sun,  9 Apr 2006 15:34:14 -0300
+
+hardinfo (0.3.7pre-2) unstable; urgency=low
+
+  * Fixed FTBFS (file intl.c updated). Thanks to  Andreas Jochens. (Closes: #326033) 
+
+ -- Agney Lopes Roth Ferraz <agney@users.sourceforge.net>  Fri,  9 Sep 2005 18:30:09 -0300
+
+hardinfo (0.3.7pre-1) unstable; urgency=low
+
+  * New upstream release. (Closes: #280694, #300114, #233033)
+
+ -- Agney Lopes Roth Ferraz <agney@users.sourceforge.net>  Mon, 13 Jun 2005 18:34:20 -0300
+
+hardinfo (0.3.6-5) unstable; urgency=high
+
+  * Add Amd64 support (closes: #253935).
+    Thanks to Kurt Roeckx <Q@ping.be>
+  * Close duplicate "doesn't work with newer pciutils" bug (closes: #254018).
+
+ -- Leandro Pereira <leandro@linuxmag.com.br>  Sat, 14 Jun 2004 21:33:25 -0300
+
+hardinfo (0.3.6-4) unstable; urgency=high
+
+  * Fixed segfault on startup (closes: #242843).
+    Thanks to Remco van de Meent <remco@debian.org>
+
+ -- Leandro Pereira <leandro@linuxmag.com.br>  Sat, 05 Jun 2004 20:35:12 -0300
+
+hardinfo (0.3.6-3) unstable; urgency=high
+
+  * Added Debian menu entry icon.
+  * Fixed some misc packaging bugs.
+  * Changed package description.
+
+ -- Leandro Pereira <leandro@linuxmag.com.br>  Sun, 17 Dec 2003 10:52:45 -0300
+
+hardinfo (0.3.6-2) unstable; urgency=low
+
+  * Sync with upstream sources.
+  * Disabled "Network" tab.
+
+ -- Leandro Pereira <leandro@linuxmag.com.br>  Sun, 17 Dec 2003 10:31:23 -0300
+
+hardinfo (0.3.6-1) unstable; urgency=high
+
+  * Sync with upstream sources.
+
+ -- Leandro Pereira <leandro@linuxmag.com.br>  Sun, 18 Jul 2003 10:07:58 -0300
+
+hardinfo (0.3.5-1) unstable; urgency=high
+
+  * Sync with upstream sources.
+
+ -- Leandro Pereira <leandro@linuxmag.com.br>  Sun, 23 Jun 2003 15:55:22 -0300
+
+hardinfo (0.3.4-1) unstable; urgency=high
+
+  * Sync with upstream sources.
+
+ -- Leandro Pereira <leandro@linuxmag.com.br>  Sun, 22 Jun 2003 21:45:15 -0300
+
+hardinfo (0.3.3-1) unstable; urgency=low
+
+  * Sync with upstream sources.
+
+ -- Leandro Pereira <leandro@linuxmag.com.br>  Sun, 18 Jun 2003 22:15:34 -0300
+
+hardinfo (0.3.2-1) unstable; urgency=low
+
+  * Sync with upstream sources.
+
+ -- Leandro Pereira <leandro@linuxmag.com.br>  Sun, 17 Jun 2003 10:50:34 -0300
+
+hardinfo (0.3.1-1) unstable; urgency=low
+
+  * Sync with upstream sources.
+
+ -- Leandro Pereira <leandro@linuxmag.com.br>  Sun, 15 Jun 2003 21:32:14 -0300
+
+hardinfo (0.3-1) unstable; urgency=low
+
+  * Initial Release.
+
+ -- Leandro Pereira <leandro@linuxmag.com.br>  Sun, 15 Jun 2003 11:13:14 -0300
+
--- hardinfo-0.5.1.orig/debian/compat
+++ hardinfo-0.5.1/debian/compat
@@ -0,0 +1 @@
+5
--- hardinfo-0.5.1.orig/debian/configure-debian.desktop
+++ hardinfo-0.5.1/debian/configure-debian.desktop
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Configure-Debian
+Comment=Reconfigure Your Packages
+Exec=/usr/sbin/su-to-root -p root -c /usr/sbin/configure-debian
+Icon=/usr/share/pixmaps/configure-debian-48x48.png
+Terminal=true
+MultipleArgs=false
+Type=Application
+Categories=Application;System;
--- hardinfo-0.5.1.orig/debian/control
+++ hardinfo-0.5.1/debian/control
@@ -0,0 +1,15 @@
+Source: hardinfo
+Section: x11
+Priority: optional
+Maintainer: Agney Lopes Roth Ferraz <agney@debian.org>
+Build-Depends: debhelper (>> 5.0.0), libgtk2.0-dev, pciutils (>=1:2.1.11-10), libsoup2.4-dev, libffi-dev, liblzma-dev, libselinux1-dev [!kfreebsd-i386 !kfreebsd-amd64 !hurd-i386]
+Standards-Version: 3.8.3.0
+
+Package: hardinfo
+Architecture: any 
+Depends: ${shlibs:Depends}, pciutils (>=1:2.1.11-10)
+Suggests: mesa-utils
+Description: Displays system information
+ HardInfo is a small application that displays information about your
+ hardware and operating system. Currently it knows about PCI, ISA PnP, USB,
+ IDE, SCSI, Serial and parallel port devices.
--- hardinfo-0.5.1.orig/debian/copyright
+++ hardinfo-0.5.1/debian/copyright
@@ -0,0 +1,13 @@
+This package was debianized by Agney Lopes Roth Ferraz <agney@debian.org> on
+Fri,  9 Jun 2006 08:49:25 -0300.
+
+It was downloaded from http://hardinfo.berlios.de/
+
+Upstream Author: Leandro Pereira <leandro@linuxmag.com.br>
+
+Copyright (c) 2003, Leandro Pereira.
+
+You are free to distribute and/or modify this software under the terms of
+the GNU General Public License version 2.
+On Debian systems, the complete text of the GNU General Public License can
+be found in /usr/share/common-licenses/GPL-2 file.
--- hardinfo-0.5.1.orig/debian/dirs
+++ hardinfo-0.5.1/debian/dirs
@@ -0,0 +1,2 @@
+usr/bin
+usr/share/hardinfo/pixmaps
--- hardinfo-0.5.1.orig/debian/hardinfo.1
+++ hardinfo-0.5.1/debian/hardinfo.1
@@ -0,0 +1,35 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH HARDINFO 1 "June 15, 2003"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+hardinfo \- shows hardware information in a GTK+ window
+.SH SYNOPSIS
+.B hardinfo
+.SH DESCRIPTION
+.B hardinfo
+parses some files in /proc and displays their information in a
+user-friendly window.
+
+This manual page was written for the Debian distribution
+because the original program does not have a manual page.
+.PP
+.\" TeX users may be more comfortable with the \fB<whatever>\fP and
+.\" \fI<whatever>\fP escape sequences to invode bold face and italics, 
+.\" respectively.
+.SH AUTHOR
+This manual page was written by Leandro Pereira <leandro@linuxmag.com.br>,
+for the Debian project (but may be used by others).
--- hardinfo-0.5.1.orig/debian/hardinfo.desktop
+++ hardinfo-0.5.1/debian/hardinfo.desktop
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Name=System Profiler and Benchmark
+Name[pt_BR]=Informações e Testes do Sistema
+Exec=hardinfo
+Icon=/usr/share/hardinfo/pixmaps/logo.png
+Terminal=false
+Type=Application
+StartupNotify=true
+Categories=System;
+
--- hardinfo-0.5.1.orig/debian/hardinfo.xpm
+++ hardinfo-0.5.1/debian/hardinfo.xpm
@@ -0,0 +1,349 @@
+/* XPM */
+static char * hardinfo_xpm[] = {
+"32 32 314 2",
+"  	c None",
+". 	c #F3F3F3",
+"+ 	c #F2F2F2",
+"@ 	c #F1F1F1",
+"# 	c #F0F0F0",
+"$ 	c #EFEFEF",
+"% 	c #EEEEEE",
+"& 	c #ECECEC",
+"* 	c #EAEAEA",
+"= 	c #E8E8E8",
+"- 	c #E7E7E7",
+"; 	c #E3E3E3",
+"> 	c #AEAEAE",
+", 	c #F4F4F4",
+"' 	c #DCDCDC",
+") 	c #D8D8D8",
+"! 	c #D6D6D6",
+"~ 	c #D5D5D5",
+"{ 	c #D3D3D3",
+"] 	c #D1D1D1",
+"^ 	c #D0D0D0",
+"/ 	c #CFCFCF",
+"( 	c #CECECE",
+"_ 	c #CCCCCC",
+": 	c #CBCBCB",
+"< 	c #CDCDCD",
+"[ 	c #626262",
+"} 	c #898989",
+"| 	c #383838",
+"1 	c #373737",
+"2 	c #363636",
+"3 	c #353535",
+"4 	c #343434",
+"5 	c #424242",
+"6 	c #E2E2E2",
+"7 	c #D9D9D9",
+"8 	c #3F3F3F",
+"9 	c #6E6E6E",
+"0 	c #888682",
+"a 	c #E2E2DF",
+"b 	c #EDEAE6",
+"c 	c #DFDED9",
+"d 	c #C9CAC7",
+"e 	c #BFC0BD",
+"f 	c #C4C1BA",
+"g 	c #C1BCB2",
+"h 	c #B8B4AA",
+"i 	c #ABA69C",
+"j 	c #919089",
+"k 	c #7D7C76",
+"l 	c #282724",
+"m 	c #E5E5E5",
+"n 	c #D7D7D7",
+"o 	c #9A9793",
+"p 	c #EEF0F1",
+"q 	c #FFFFFF",
+"r 	c #EDEDEC",
+"s 	c #D8DADA",
+"t 	c #D0D2D2",
+"u 	c #CAC7C1",
+"v 	c #C1BDB3",
+"w 	c #AAA49B",
+"x 	c #797E7F",
+"y 	c #89846E",
+"z 	c #2C2B27",
+"A 	c #E4E4E4",
+"B 	c #6D6D6D",
+"C 	c #96948F",
+"D 	c #F7F6F5",
+"E 	c #FAF9F8",
+"F 	c #EEECEA",
+"G 	c #EBE9E6",
+"H 	c #E7E5E1",
+"I 	c #D0CDC5",
+"J 	c #C0BCB2",
+"K 	c #B8B3A9",
+"L 	c #A8A399",
+"M 	c #9D9685",
+"N 	c #8C7D76",
+"O 	c #2B2926",
+"P 	c #EBEBEB",
+"Q 	c #6C6C6C",
+"R 	c #908E89",
+"S 	c #F3F2F0",
+"T 	c #F4F2F1",
+"U 	c #ECEAE7",
+"V 	c #E9E7E4",
+"W 	c #E6E3DF",
+"X 	c #CECBC4",
+"Y 	c #BEBAB0",
+"Z 	c #B7B2A8",
+"` 	c #A6A197",
+" .	c #937162",
+"..	c #8B6863",
+"+.	c #2A2826",
+"@.	c #D2D2D2",
+"#.	c #464C52",
+"$.	c #49525D",
+"%.	c #3F4954",
+"&.	c #3E4852",
+"*.	c #4D545B",
+"=.	c #8E8F90",
+"-.	c #C8C8C6",
+";.	c #E5E3DF",
+">.	c #E5E2DF",
+",.	c #E2DFDB",
+"'.	c #CAC7BE",
+").	c #BAB5AC",
+"!.	c #B3ADA4",
+"~.	c #A39E94",
+"{.	c #928E84",
+"].	c #928E86",
+"^.	c #2A2825",
+"/.	c #E1E1E1",
+"(.	c #6F7A86",
+"_.	c #8397AB",
+":.	c #7D96B2",
+"<.	c #7B95B1",
+"[.	c #758CA7",
+"}.	c #637282",
+"|.	c #5B5F63",
+"1.	c #BAB9B6",
+"2.	c #D8D6D3",
+"3.	c #DAD8D3",
+"4.	c #C3BFB7",
+"5.	c #B3AEA4",
+"6.	c #ACA79D",
+"7.	c #9D988E",
+"8.	c #8F8A80",
+"9.	c #908B84",
+"0.	c #292724",
+"a.	c #E0E0E0",
+"b.	c #41464D",
+"c.	c #7D8A98",
+"d.	c #8199B4",
+"e.	c #7590AE",
+"f.	c #849CB7",
+"g.	c #A3B5C9",
+"h.	c #94A9C0",
+"i.	c #7892AF",
+"j.	c #778BA1",
+"k.	c #585D60",
+"l.	c #A6A4A1",
+"m.	c #C7C4C0",
+"n.	c #B3AFA7",
+"o.	c #A5A097",
+"p.	c #9E998F",
+"q.	c #928D83",
+"r.	c #858077",
+"s.	c #8A867F",
+"t.	c #272623",
+"u.	c #DFDFDF",
+"v.	c #606A75",
+"w.	c #8097B1",
+"x.	c #D1DAE4",
+"y.	c #627182",
+"z.	c #676868",
+"A.	c #B4B2AF",
+"B.	c #AFABA5",
+"C.	c #A6A29A",
+"D.	c #A09C94",
+"E.	c #97938B",
+"F.	c #908C85",
+"G.	c #8D8A83",
+"H.	c #272522",
+"I.	c #DEDEDE",
+"J.	c #C9C9C9",
+"K.	c #222528",
+"L.	c #7C8EA2",
+"M.	c #7892B0",
+"N.	c #B2C1D2",
+"O.	c #6B84A0",
+"P.	c #394048",
+"Q.	c #54514C",
+"R.	c #5D5A55",
+"S.	c #5C5A54",
+"T.	c #5A5852",
+"U.	c #54514B",
+"V.	c #4F4C47",
+"W.	c #474540",
+"X.	c #151413",
+"Y.	c #C6C6C6",
+"Z.	c #33373C",
+"`.	c #889FB8",
+" +	c #708AA7",
+".+	c #3D4752",
+"++	c #838383",
+"@+	c #989898",
+"#+	c #9C9C9C",
+"$+	c #9B9B9B",
+"%+	c #C4C4C4",
+"&+	c #393F44",
+"*+	c #89A0BA",
+"=+	c #E0E6ED",
+"-+	c #718BA8",
+";+	c #35404B",
+">+	c #B0B0B0",
+",+	c #D4D4D4",
+"'+	c #D7D4C3",
+")+	c #E2DAA9",
+"!+	c #B5B5B5",
+"~+	c #BCBCBC",
+"{+	c #3E3E3E",
+"]+	c #5F5F5F",
+"^+	c #3B3F45",
+"/+	c #6F89A6",
+"(+	c #999999",
+"_+	c #CACACA",
+":+	c #C0BDAC",
+"<+	c #A09766",
+"[+	c #A5A5A5",
+"}+	c #7D7D7D",
+"|+	c #323232",
+"1+	c #9E9E9E",
+"2+	c #8394A6",
+"3+	c #7A94B1",
+"4+	c #5B7188",
+"5+	c #262B31",
+"6+	c #3A3A3A",
+"7+	c #404040",
+"8+	c #454545",
+"9+	c #434343",
+"0+	c #2E2E2E",
+"a+	c #1E1E1E",
+"b+	c #797979",
+"c+	c #565F69",
+"d+	c #8198B1",
+"e+	c #738DAB",
+"f+	c #3D4B5A",
+"g+	c #16181A",
+"h+	c #1D1D1D",
+"i+	c #222222",
+"j+	c #212121",
+"k+	c #262626",
+"l+	c #242424",
+"m+	c #252525",
+"n+	c #202020",
+"o+	c #090909",
+"p+	c #6B7580",
+"q+	c #748FAD",
+"r+	c #6A829E",
+"s+	c #465668",
+"t+	c #0B0D10",
+"u+	c #000000",
+"v+	c #535353",
+"w+	c #515A65",
+"x+	c #76889C",
+"y+	c #7D95B0",
+"z+	c #748EAA",
+"A+	c #6E87A4",
+"B+	c #5F758D",
+"C+	c #3E4A58",
+"D+	c #575A5F",
+"E+	c #808080",
+"F+	c #969696",
+"G+	c #A2A2A2",
+"H+	c #A3A3A3",
+"I+	c #F5F5F5",
+"J+	c #5B5B5B",
+"K+	c #7F7F80",
+"L+	c #7E8083",
+"M+	c #494D52",
+"N+	c #464B51",
+"O+	c #484B4F",
+"P+	c #7A7A7A",
+"Q+	c #A1A1A1",
+"R+	c #BBBBBB",
+"S+	c #C5C5C5",
+"T+	c #A9A9A9",
+"U+	c #C0C0C0",
+"V+	c #B6B6B6",
+"W+	c #606060",
+"X+	c #727272",
+"Y+	c #7E7E7E",
+"Z+	c #ABABAB",
+"`+	c #AFAFAF",
+" @	c #B4B4B4",
+".@	c #C3C3C3",
+"+@	c #B8B8B8",
+"@@	c #A0A0A0",
+"#@	c #929292",
+"$@	c #525252",
+"%@	c #515151",
+"&@	c #DADADA",
+"*@	c #BDBDBD",
+"=@	c #555555",
+"-@	c #111111",
+";@	c #B7B7B7",
+">@	c #BFBFBF",
+",@	c #DDDDDD",
+"'@	c #9D9D9D",
+")@	c #2B2B2B",
+"!@	c #9A9A9A",
+"~@	c #979797",
+"{@	c #939393",
+"]@	c #919191",
+"^@	c #959595",
+"/@	c #848484",
+"(@	c #7C7C7C",
+"_@	c #757575",
+":@	c #8E8E8E",
+"<@	c #646464",
+"[@	c #6A6A6A",
+"}@	c #696969",
+"|@	c #686868",
+"1@	c #676767",
+"2@	c #666666",
+"3@	c #656565",
+"4@	c #636363",
+"5@	c #5E5E5E",
+"6@	c #5D5D5D",
+"7@	c #5A5A5A",
+"8@	c #585858",
+"9@	c #575757",
+"                                                                ",
+"                                                                ",
+"                                                                ",
+"            . . . + @ @ # $ % % & & * * = - ; >                 ",
+"            , @ ' ) ! ~ { { ] ^ / ( _ : < ! ' : [               ",
+"            , @ } | 1 1 2 2 3 3 3 3 3 4 1 5 6 7 8               ",
+"            . $ 9 0 a b c d e f g h i j k l m n |               ",
+"            . $ 9 o p q r s t u v h w x y z A ! |               ",
+"            . % B C D E F G H I J K L M N O ; { |               ",
+"            @ P Q R S T U V W X Y Z `  ...+.6 @.|               ",
+"          #.$.%.&.*.=.-.;.>.,.'.).!.~.{.].^././ |               ",
+"        (._.:.:.<.[.}.|.1.2.3.4.5.6.7.8.9.0.a.< |               ",
+"    b.c.d.e.f.g.h.e.i.j.k.l.m.n.o.p.q.r.s.t.u.: |               ",
+"    v.w.e.e.g.q x.e.e.i.y.z.A.B.C.D.E.F.G.H.I.J.|               ",
+"  K.L.M.e.e.h.x.N.e.e.e.O.P.Q.R.S.T.U.V.W.X.' Y.|               ",
+"  Z.`.e.e.e.x.x.N.e.e.e. +.+++@+#+#+$+$+$+@+' %+|               ",
+"  &+*+e.e.e.=+q x.e.e.e.-+;+>+: ,+@.( _ '+)+!+~+{+]+            ",
+"  ^+`.e.e.e.g.q x.e.e.e./+$.(+~+@.^ < _+:+<+[+}+|+3   1+        ",
+"    2+3+e.e.g.q x.e.e.e.4+5+6+7+8+8+5 8+9+9+0+a+      1+b+      ",
+"    c+d+e.e.g.q x.e.e.e+f+g+h+i+j+k+j+l+m+n+o+        1+b+      ",
+"      p+d+3+e.e.e.q+r+s+t+u+u+u+u+u+u+u+u+u+u+        1+v+      ",
+"        w+x+y+z+A+B+C+D+E+F+G+H+H+H+H+H+H+H+H+@.I+#+  1+J+      ",
+"        K+L+M+N+N+O+P+Q+R+! @.~ S+%+! { Y._ T+U+V+W+X+Y+        ",
+"        : 7 Z+`+ @Z+.@R+) _ S+%+Y.] ~++@T+`+F+Z+@@#@$@%@        ",
+"        @ /.) ~ ) { &@^ { *@n J._ ( ^ S+_+/ !+R+ @#+=@-@        ",
+"      I+# _ ^ &@a.= ' - ,+a.! > ~+;@>@S+,@; _+'@T+U+[ )@        ",
+"      % R+.@= @.[+H+'@!@~@{@]@`+@. @G+~+^@/@(@_@T+:@Z+<@        ",
+"      (@[@[@[@}@}@|@1@1@2@3@<@<@4@[ W+]+5@6@J+7@8@9@=@1 u+      ",
+"            u+u+u+u+u+u+u+u+u+u+u+u+u+u+u+u+u+u+u+u+u+u+        ",
+"                                                                ",
+"                                                                ",
+"                                                                "};
--- hardinfo-0.5.1.orig/debian/menu
+++ hardinfo-0.5.1/debian/menu
@@ -0,0 +1,5 @@
+?package(hardinfo): \
+needs="X11" \
+section="Applications/System/Administration" \
+title="System Information" command="/usr/bin/hardinfo" \
+icon="/usr/share/pixmaps/hardinfo.xpm"
--- hardinfo-0.5.1.orig/debian/postinst
+++ hardinfo-0.5.1/debian/postinst
@@ -0,0 +1,48 @@
+#! /bin/sh
+# postinst script for hardinfo
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <postinst> `configure' <most-recently-configured-version>
+#        * <old-postinst> `abort-upgrade' <new version>
+#        * <conflictor's-postinst> `abort-remove' `in-favour' <package>
+#          <new-version>
+#        * <deconfigured's-postinst> `abort-deconfigure' `in-favour'
+#          <failed-install-package> <version> `removing'
+#          <conflicting-package> <version>
+# for details, see http://www.debian.org/doc/debian-policy/ or
+# the debian-policy package
+#
+# quoting from the policy:
+#     Any necessary prompting should almost always be confined to the
+#     post-installation script, and should be protected with a conditional
+#     so that unnecessary prompting doesn't happen if a package's
+#     installation fails and the `postinst' is called with `abort-upgrade',
+#     `abort-remove' or `abort-deconfigure'.
+
+case "$1" in
+    configure)
+
+    ;;
+
+    abort-upgrade|abort-remove|abort-deconfigure)
+
+    ;;
+
+    *)
+        echo "postinst called with unknown argument \`$1'" >&2
+        exit 1
+    ;;
+esac
+
+# dh_installdeb will replace this with shell code automatically
+# generated by other debhelper scripts.
+
+#DEBHELPER#
+
+exit 0
+
+
--- hardinfo-0.5.1.orig/debian/postrm
+++ hardinfo-0.5.1/debian/postrm
@@ -0,0 +1,38 @@
+#! /bin/sh
+# postrm script for hardinfo
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <postrm> `remove'
+#        * <postrm> `purge'
+#        * <old-postrm> `upgrade' <new-version>
+#        * <new-postrm> `failed-upgrade' <old-version>
+#        * <new-postrm> `abort-install'
+#        * <new-postrm> `abort-install' <old-version>
+#        * <new-postrm> `abort-upgrade' <old-version>
+#        * <disappearer's-postrm> `disappear' <r>overwrit>r> <new-version>
+# for details, see http://www.debian.org/doc/debian-policy/ or
+# the debian-policy package
+
+
+case "$1" in
+       purge|remove|upgrade|failed-upgrade|abort-install|abort-upgrade|disappear)
+
+
+        ;;
+
+    *)
+        echo "postrm called with unknown argument \`$1'" >&2
+        exit 1
+
+esac
+
+# dh_installdeb will replace this with shell code automatically
+# generated by other debhelper scripts.
+
+#DEBHELPER#
+
+exit 0
--- hardinfo-0.5.1.orig/debian/rules
+++ hardinfo-0.5.1/debian/rules
@@ -0,0 +1,102 @@
+#!/usr/bin/make -f
+# Sample debian/rules that uses debhelper.
+# GNU copyright 1997 to 1999 by Joey Hess.
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+
+
+
+CFLAGS = -Wall -g
+
+ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
+	CFLAGS += -O0
+else
+	CFLAGS += -O2
+endif
+ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
+	INSTALL_PROGRAM += -s
+endif
+
+configure: configure-stamp
+configure-stamp:
+	dh_testdir
+	./configure
+
+	touch configure-stamp
+
+
+build: build-stamp
+
+build-stamp: configure-stamp 
+	dh_testdir
+
+	$(MAKE)
+
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp configure-stamp
+
+	# Add here commands to clean up after the build process.
+	[ ! -f Makefile ] || $(MAKE) clean
+	rm -f arch/this
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+
+	# Add here commands to install the package into debian/hardinfo.
+	$(MAKE) install DESTDIR=$(CURDIR)/debian/hardinfo
+	install -d `pwd`/debian/hardinfo/usr/share/pixmaps
+	install -m 644 debian/hardinfo.xpm \
+		`pwd`/debian/hardinfo/usr/share/pixmaps
+	install -d `pwd`/debian/hardinfo/usr/share/applications
+	install -m 644 debian/hardinfo.desktop  \
+		`pwd`/debian/hardinfo/usr/share/applications
+
+# Build architecture-independent files here.
+binary-indep: build install
+# We have nothing to do by default.
+
+# Build architecture-dependent files here.
+binary-arch: build install
+	dh_testdir
+	dh_testroot
+	#dh_installchangelogs ChangeLog
+	dh_installchangelogs
+	dh_installdocs
+#	dh_installexamples
+	dh_install
+	dh_installmenu
+#	dh_desktop
+#	dh_installdebconf	
+#	dh_installlogrotate
+#	dh_installemacsen
+#	dh_installpam
+#	dh_installmime
+#	dh_installinit
+#	dh_installcron
+#	dh_installinfo
+	dh_installman debian/hardinfo.1
+	dh_link
+	dh_strip
+	dh_compress
+	dh_fixperms
+#	dh_perl
+#	dh_python
+#	dh_makeshlibs
+	dh_installdeb
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install configure
--- hardinfo-0.5.1.orig/debian/watch
+++ hardinfo-0.5.1/debian/watch
@@ -0,0 +1,6 @@
+version=3
+opts=downloadurlmangle=s/prdownload/download/ \
+   http://developer.berlios.de/project/showfiles.php?group_id=5897 \
+   http://prdownload.berlios.de/hardinfo/hardinfo-(.*)\.tar\.gz \
+   debian uupdate
+
