diff -Naur Leanify-0.4.3/Makefile Leanify-0.4.3.ports/Makefile
--- Leanify-0.4.3/Makefile	2015-11-26 06:11:59.000000000 +0000
+++ Leanify-0.4.3.ports/Makefile	2017-03-11 17:20:17.864407914 +0000
@@ -6,7 +6,7 @@
 ZOPFLI_SRC      := lib/zopfli/hash.c lib/zopfli/squeeze.c lib/zopfli/gzip_container.c lib/zopfli/katajainen.c lib/zopfli/zopfli_lib.c lib/zopfli/cache.c lib/zopfli/zlib_container.c lib/zopfli/util.c lib/zopfli/tree.c lib/zopfli/deflate.c lib/zopfli/blocksplitter.c lib/zopfli/lz77.c
 ZOPFLIPNG_SRC   := lib/zopflipng/lodepng/lodepng.cpp lib/zopflipng/lodepng/lodepng_util.cpp lib/zopflipng/zopflipng_lib.cc
 
-CFLAGS      += -Wall -O3 -msse2 -mfpmath=sse
+CFLAGS      += -Wall -O3
 ifneq ($(CC),clang)
     CFLAGS  += -flto
 endif
