diff --git a/tests/Makefile.am b/tests/Makefile.am
index 481f168..aabd7ef 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -38,7 +38,7 @@ endif
 noinst_PROGRAMS = $(AUTOMATED_TESTS) $(MANUAL_TESTS)
 
 DEPS = $(top_builddir)/gmime/libgmime-$(GMIME_API_VERSION).la
-LDADDS = $(top_builddir)/gmime/libgmime-$(GMIME_API_VERSION).la $(GLIB_LIBS)
+LDADDS = $(top_builddir)/gmime/libgmime-$(GMIME_API_VERSION).la -lgpg-error $(GLIB_LIBS)
 
 test_best_SOURCES = test-best.c
 test_best_LDFLAGS = 
diff --git a/tests/test-pkcs7.c b/tests/test-pkcs7.c
index 7b49128..4175ff2 100644
--- a/tests/test-pkcs7.c
+++ b/tests/test-pkcs7.c
@@ -32,6 +32,7 @@
 
 #include <gmime/gmime.h>
 #include <gmime/gmime-pkcs7-context.h>
+#include <gpg-error.h>
 
 #include <gpg-error.h>
 
