diff -uprN mono-addins-mono-addins-1.3.old/Makefile.include mono-addins-mono-addins-1.3/Makefile.include
--- mono-addins-mono-addins-1.3.old/Makefile.include	2016-07-27 01:26:26.000000000 +0300
+++ mono-addins-mono-addins-1.3/Makefile.include	2017-05-03 02:49:41.690295653 +0300
@@ -14,8 +14,14 @@ $(POLICY_ASSEMBLIES) : policy.%.$(ASSEMB
 	$(AL) -link:policy.$*.config -out:$@ -keyfile:$(top_srcdir)/mono-addins.snk

 gac-install: $(POLICY_ASSEMBLIES)
+	sn -v '$(ASSEMBLY)' || if [[ $$? = 1 ]]; then \
+		sn -R '$(ASSEMBLY)' ../mono-addins.snk; \
+	fi
 	$(GACUTIL) /i $(ASSEMBLY) /f $(GACUTIL_FLAGS) || exit 1;
 	@for p in $(POLICY_ASSEMBLIES); do \
+		sn -v "$$p" || if [[ $$? = 1 ]]; then \
+			sn -R "$$p" ../mono-addins.snk; \
+		fi; \
 		echo $(GACUTIL) /i $$p /f $(GACUTIL_POLICY_FLAGS) || exit 1; \
 		$(GACUTIL) /i $$p /f $(GACUTIL_POLICY_FLAGS) || exit 1; \
 	done
