diff -aur keepassxc-2.6.6/CMakeLists.txt keepassxc-2.6.6.aosc/CMakeLists.txt
--- keepassxc-2.6.6/CMakeLists.txt	2021-06-12 10:13:52.000000000 +0800
+++ keepassxc-2.6.6.aosc/CMakeLists.txt	2022-01-01 15:58:47.630105552 +0800
@@ -180,6 +180,15 @@
     set(IS_32BIT TRUE)
 endif()
 
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.14.0")
+    cmake_policy(SET CMP0083 NEW)
+    include(CheckPIESupported)
+    check_pie_supported()
+endif()
+
+# Create position independent code for shared libraries and executables
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
 set(CLANG_COMPILER_ID_REGEX "^(Apple)?[Cc]lang$")
 if("${CMAKE_C_COMPILER}" MATCHES "clang$"
         OR "${CMAKE_EXTRA_GENERATOR_C_SYSTEM_DEFINED_MACROS}" MATCHES "__clang__"
@@ -308,7 +317,6 @@
 
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
     check_add_gcc_compiler_flag("-Qunused-arguments")
-    add_gcc_compiler_flags("-pie -fPIE")
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--no-add-needed -Wl,--as-needed -Wl,--no-undefined")
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,relro,-z,now")
     set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--no-add-needed -Wl,--as-needed")
