diff -Naur xmms-1.2.11/xmms/main.c xmms-1.2.11.srandom/xmms/main.c
--- xmms-1.2.11/xmms/main.c	2006-07-16 08:40:04.000000000 -0500
+++ xmms-1.2.11.srandom/xmms/main.c	2020-04-10 20:33:02.569797309 -0500
@@ -4199,9 +4199,6 @@
 
 	parse_cmd_line(argc, argv, &options);
 
-#ifdef HAVE_SRANDOMDEV
-	srandomdev();
-#else
 	{
 		int fd = open("/dev/urandom", O_RDONLY);
 		if (fd != -1)
@@ -4214,7 +4211,6 @@
 		else
 			srandom(time(NULL));
 	}
-#endif
 
 	read_config();
 
