--- xmms-1.2.11-20071117cvs/xmms/xmms.desktop	2002-02-11 23:15:56.000000000 +0000
+++ xmms-1.2.11-20071117cvs/xmms/xmms.desktop-new	2009-09-02 14:19:57.000000000 +0100
@@ -14,7 +14,7 @@
 comment[tr]=X Multimedya Sistem
 comment[zh_TW]=X 多媒體系統
 Encoding=Legacy-Mixed
-Exec=xmms
+Exec=xmms -e -p %F
 Icon=xmms_mini.xpm
 MimeType=audio/x-scpls;audio/x-mpegurl;audio/mpegurl;audio/mp3;audio/x-mp3;audio/mpeg;audio/x-mpeg;audio/x-wav;application/x-ogg
 Terminal=0
