--- xmms-1.2.11-20071117cvs/configure	2008-09-02 19:22:09.000000000 +0100
+++ xmms-1.2.11-20071117cvs/configure-new	2008-09-02 21:02:33.000000000 +0100
@@ -30090,8 +30090,6 @@
 #define HAVE_LIBASOUND 1
 _ACEOF
 
-  LIBS="-lasound $LIBS"
-
 else
 
 	 alsa_found=no
@@ -30104,7 +30102,6 @@
    have_alsa=yes
    LIBS=`echo $LIBS | sed 's/-lasound//g'`
    LIBS=`echo $LIBS | sed 's/  //'`
-   LIBS="-lasound $LIBS"
 fi
 if test "x$alsa_found" = "xno" ; then
    have_alsa=no
