--- xmms-1.2.11-20071117cvs/xmms/Makefile.am	2003-12-01 22:45:23.000000000 +0000
+++ xmms-1.2.11-20071117cvs/xmms/Makefile-new.am	2010-02-20 23:23:26.000000000 +0000
@@ -6,7 +6,8 @@
 
 xmms_LDFLAGS = -export-dynamic
 xmms_LDADD = @GTK_LIBS@ @PTHREAD_LIBS@ @SM_LIBS@ @VM_LIBS@ \
-@POSIX_LIBS@ $(top_builddir)/libxmms/libxmms.la @LTLIBINTL@
+@POSIX_LIBS@ $(top_builddir)/libxmms/libxmms.la @LTLIBINTL@ \
+-ldl
 
 INCLUDES = @GTK_CFLAGS@ @XMMS_DEFINES@ @ARCH_DEFINES@ \
 -I$(top_builddir)/intl -I$(top_srcdir)
--- xmms-1.2.11-20071117cvs/xmms/Makefile.in	2007-11-17 22:45:34.000000000 +0000
+++ xmms-1.2.11-20071117cvs/xmms/Makefile-new.in	2010-02-20 23:24:28.000000000 +0000
@@ -307,7 +307,8 @@
 xmmsincludedir = $(includedir)/xmms
 xmms_LDFLAGS = -export-dynamic
 xmms_LDADD = @GTK_LIBS@ @PTHREAD_LIBS@ @SM_LIBS@ @VM_LIBS@ \
-@POSIX_LIBS@ $(top_builddir)/libxmms/libxmms.la @LTLIBINTL@
+@POSIX_LIBS@ $(top_builddir)/libxmms/libxmms.la @LTLIBINTL@ \
+-ldl
 
 INCLUDES = @GTK_CFLAGS@ @XMMS_DEFINES@ @ARCH_DEFINES@ \
 -I$(top_builddir)/intl -I$(top_srcdir)
