diff -Naur ksquirrel-trinity-14.0.10/ksquirrel/ksquirrelpart/sq_glselectionpainter.cpp ksquirrel-trinity-14.0.10.redefinition/ksquirrel/ksquirrelpart/sq_glselectionpainter.cpp
--- ksquirrel-trinity-14.0.10/ksquirrel/ksquirrelpart/sq_glselectionpainter.cpp 2021-02-27 16:48:53.000000000 -0800
+++ ksquirrel-trinity-14.0.10.redefinition/ksquirrel/ksquirrelpart/sq_glselectionpainter.cpp    2021-06-28 15:50:11.556179206 -0700
@@ -27,7 +27,6 @@
 static const int len2 = 10;
 static const int lenc = 2;
 static const int lenc2 = lenc / 2;
-static const double rad_const = 3.14159265358979323846 / 180.0;
 
 /* ******************************************************************* */
 
diff -Naur ksquirrel-trinity-14.0.10/ksquirrel/sq_glwidget.cpp ksquirrel-trinity-14.0.10.redefinition/ksquirrel/sq_glwidget.cpp
--- ksquirrel-trinity-14.0.10/ksquirrel/sq_glwidget.cpp 2021-02-27 16:48:53.000000000 -0800
+++ ksquirrel-trinity-14.0.10.redefinition/ksquirrel/sq_glwidget.cpp    2021-06-28 15:43:55.294908960 -0700
@@ -94,7 +94,6 @@
 SQ_GLWidget * SQ_GLWidget::m_instance = 0;
 
 static const int timer_delay_file = 5;
-static const double rad_const = 3.14159265358979323846 / 180.0;
 
 static const float SQ_WINDOW_BACKGROUND_POS = -1000.0;
 static const float SQ_IMAGE_CHECKER_POS =     -999.0;
