--- gtk+-1.2.10/gtk/gtkstyle.c.pixmapref	Tue Feb 20 11:46:58 2001
+++ gtk+-1.2.10/gtk/gtkstyle.c	Sun Aug 12 15:42:06 2001
@@ -348,8 +348,9 @@
       new_style->bg[i] = style->bg[i];
       new_style->text[i] = style->text[i];
       new_style->base[i] = style->base[i];
-      
-      new_style->bg_pixmap[i] = style->bg_pixmap[i];
+
+      if (style->bg_pixmap[i] && !(style->rc_style && style->rc_style->bg_pixmap_name[i]))
+	new_style->bg_pixmap[i] = gdk_pixmap_ref (style->bg_pixmap[i]);
     }
   
   gdk_font_unref (new_style->font);
