--- a/gtk/gtkthemes.c
+++ b/gtk/gtkthemes.c
@@ -90,7 +90,7 @@ gtk_theme_engine_get (const gchar *name)
        g_free(engine_path);
        if (!library)
 	 {
-	   g_warning (g_module_error());
+	   g_warning ("%s", g_module_error());
 	   return NULL;
 	 }
        else
@@ -107,7 +107,7 @@ gtk_theme_engine_get (const gchar *name)
 		!g_module_symbol (library, "theme_exit", 
 				  (gpointer *)&result->exit))
 	      {
-		g_warning (g_module_error());
+		g_warning ("%s", g_module_error());
 		g_free (result);
 		return NULL;
 	      }
