--- gtk+-1.2.10/configure.libtool	Wed Jan 15 12:44:35 2003
+++ gtk+-1.2.10/configure	Wed Jan 15 12:45:29 2003
@@ -1389,11 +1389,7 @@
   echo "$ac_t""no" 1>&6
 fi
 
-
-case "$target" in
-NONE) lt_target="$host" ;;
-*) lt_target="$target" ;;
-esac
+lt_target="$host"
 
 # Check for any special flags to pass to ltconfig.
 libtool_flags="--cache-file=$cache_file"
--- gtk+-1.2.10/aclocal.m4.libtool	Wed Jan 15 12:44:47 2003
+++ gtk+-1.2.10/aclocal.m4	Wed Jan 15 12:45:42 2003
@@ -56,10 +56,7 @@
 AC_REQUIRE([AC_PROG_LN_S])dnl
 dnl
 
-case "$target" in
-NONE) lt_target="$host" ;;
-*) lt_target="$target" ;;
-esac
+lt_target="$host"
 
 # Check for any special flags to pass to ltconfig.
 libtool_flags="--cache-file=$cache_file"
--- gtk+-1.2.10/ltconfig.libtool	Wed Jan 15 12:44:58 2003
+++ gtk+-1.2.10/ltconfig	Wed Jan 15 12:46:15 2003
@@ -447,16 +447,16 @@
   host_alias=$host
 fi
 
+host_cpu=`echo $host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\1/'`
+host_vendor=`echo $host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\2/'`
+host_os=`echo $host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\3/'`
+
 # Transform linux* to *-*-linux-gnu*, to support old configure scripts.
 case "$host_os" in
 linux-gnu*) ;;
 linux*) host=`echo $host | sed 's/^\(.*-.*-linux\)\(.*\)$/\1-gnu\2/'`
 esac
 
-host_cpu=`echo $host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\1/'`
-host_vendor=`echo $host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\2/'`
-host_os=`echo $host | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\3/'`
-
 case "$host_os" in
 aix3*)
   # AIX sometimes has problems with the GCC collect2 program.  For some
