--- gtk+-1.2.10/gdk/gdkselection.c.localecrash	Thu Aug 22 16:50:01 2002
+++ gtk+-1.2.10/gdk/gdkselection.c	Thu Aug 22 17:50:46 2002
@@ -238,12 +238,14 @@
   if (res != Success)
     {
       property.encoding = None;
-      property.format = None;
+      property.format = 8;
       property.value = NULL;
       property.nitems = 0;
-    }
 
-  g_assert (property.encoding == gdk_atom_intern ("COMPOUND_TEXT", FALSE) && property.format == 8);
+      g_warning ("Error converting string to compound text.\n"
+		 "This might mean that your locale setting is supported\n"
+		 "by the C library but not by Xlib.");
+    }
 
   if (encoding)
     *encoding = property.encoding;
