--- gtk+-1.2.10/gtk/Makefile.am.no_undefined	2006-04-08 20:58:18.000000000 -0500
+++ gtk+-1.2.10/gtk/Makefile.am	2006-04-08 20:58:59.000000000 -0500
@@ -31,6 +31,7 @@
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)	\
 	-release $(LT_RELEASE) 					\
 	-export-dynamic						\
+	-no-undefined -Wl,--no-undefined			\
 	@GLIB_DEPLIBS@ 						\
 	@x_ldflags@						\
 	@x_libs@ 						\
--- gtk+-1.2.10/gdk/Makefile.am.no_undefined	2000-10-21 13:20:40.000000000 -0500
+++ gtk+-1.2.10/gdk/Makefile.am	2006-04-08 20:58:18.000000000 -0500
@@ -21,6 +21,7 @@
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-release $(LT_RELEASE) \
 	-export-dynamic \
+	-no-undefined -Wl,--no-undefined \
 	@GLIB_DEPLIBS@	\
 	@x_ldflags@	\
 	@x_libs@	\
