Description: enable linux features on armel
Bug: http://bugzilla.gnome.org/show_bug.cgi?id=588338
Bug-Debian: http://bugs.debian.org/459936

--- a/configure.in.old	2009-03-26 10:05:38.000000000 +0100
+++ a/configure.in	2009-03-26 10:10:35.000000000 +0100
@@ -176,7 +176,7 @@
 
 dnl check what OS we're on
 #AM_CONDITIONAL(HAVE_LINUX, test x$target_os = xlinux-gnu)
-if test x$target_os = xlinux-gnu; then
+if test x$target_os = xlinux-gnu -o x$target_os = xlinux-gnueabi; then
 	AC_DEFINE([HAVE_LINUX],[],[Whether we are using linux or not])
 fi
 
@@ -223,7 +223,7 @@
 dnl check if inotify backend is enabled
 AM_CONDITIONAL(ENABLE_INOTIFY, test x$inotify = xtrue)
 
-if test x$os = xlinux-gnu; then
+if test x$os = xlinux-gnu -o x$os = xlinux-gnueabi; then
 	AC_ARG_ENABLE(dnotify,
 		AC_HELP_STRING([--disable-dnotify], [Disable the DNotify backend]),
 		[dnotify="${enableval}"], [dnotify=true])
@@ -297,10 +297,12 @@
 fi
 
 dnl Use weak symbols on linux/gcc to avoid imposing libpthreads to apps
-if test x$os = xlinux-gnu -a x$WITH_THREADS = x1 ; then
-    if test "${CC}" = "gcc" ; then
-        echo Use weak symbols !
-        THREAD_LIBS=
+if test x$os = xlinux-gnu -o x$os = xlinux-gnueabi ; then
+    if test x$WITH_THREADS = x1 ; then
+      if test "${CC}" = "gcc" ; then
+          echo Use weak symbols !
+          THREAD_LIBS=
+      fi
     fi
 fi	
 AC_SUBST(THREAD_LIBS)
