From 571deb291492f08d02b47be727f1da491eab380d Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sat, 9 Aug 2025 13:06:35 +0800
Subject: [PATCH 04/32] fix gethostbyname

---
 socket.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/socket.c b/socket.c
index c659b16..a219882 100644
--- a/socket.c
+++ b/socket.c
@@ -52,7 +52,8 @@ static struct hostent *gethostbyname_dot(name)
 char   *name;
 {
     char    dot_name[MAXHOSTNAMELEN + 1];
-
+    struct hostent *hp;
+    
     /*
      * Don't append dots to unqualified names. Such names are likely to come
      * from local hosts files or from NIS.
@@ -61,8 +62,12 @@ char   *name;
     if (strchr(name, '.') == 0 || strlen(name) >= MAXHOSTNAMELEN - 1) {
 	return (gethostbyname(name));
     } else {
-	sprintf(dot_name, "%s.", name);
-	return (gethostbyname(dot_name));
+        sprintf(dot_name, "%s.", name);
+        hp = gethostbyname(dot_name);
+	if (hp)
+	    return hp;
+	else
+	    return (gethostbyname(name));
     }
 }
 
-- 
2.50.1

