From de7e932ed7f4c49a0f9cfb11e81d0c033ca0268a Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sat, 9 Aug 2025 13:18:20 +0800
Subject: [PATCH 16/32] have strerror

---
 percent_m.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/percent_m.c b/percent_m.c
index bb11b22..da4473a 100644
--- a/percent_m.c
+++ b/percent_m.c
@@ -29,11 +29,15 @@ char   *ibuf;
 
     while (*bp = *cp)
 	if (*cp == '%' && cp[1] == 'm') {
+#ifdef HAVE_STRERROR
+	    strcpy(bp, strerror(errno));
+#else
 	    if (errno < sys_nerr && errno > 0) {
 		strcpy(bp, sys_errlist[errno]);
 	    } else {
 		sprintf(bp, "Unknown error %d", errno);
 	    }
+#endif
 	    bp += strlen(bp);
 	    cp += 2;
 	} else {
-- 
2.50.1

