From 58d86db91bdc70b87405ca851a897ee66846de48 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sat, 9 Aug 2025 13:24:26 +0800
Subject: [PATCH 25/32] size_t

---
 fix_options.c | 5 +++++
 socket.c      | 8 ++++++++
 2 files changed, 13 insertions(+)

diff --git a/fix_options.c b/fix_options.c
index 2eb206a..a0c5dec 100644
--- a/fix_options.c
+++ b/fix_options.c
@@ -38,7 +38,12 @@ struct request_info *request;
 #ifdef IP_OPTIONS
     unsigned char optbuf[BUFFER_SIZE / 3], *cp;
     char    lbuf[BUFFER_SIZE], *lp;
+#ifdef __GLIBC__
+    socklen_t optsize = sizeof(optbuf);
+    int     ipproto;
+#else
     int     optsize = sizeof(optbuf), ipproto;
+#endif
     struct protoent *ip;
     int     fd = request->fd;
     unsigned int opt;
diff --git a/socket.c b/socket.c
index 4b2c575..a0d2987 100644
--- a/socket.c
+++ b/socket.c
@@ -95,7 +95,11 @@ struct request_info *request;
     static struct sockaddr_in client;
     static struct sockaddr_in server;
 #endif
+#ifdef __GLIBC__
+    socklen_t  len;
+#else
     int     len;
+#endif
     char    buf[BUFSIZ];
     int     fd = request->fd;
 
@@ -426,7 +430,11 @@ int     fd;
 #else
     struct sockaddr_in sin;
 #endif
+#ifdef __GLIBC__
+    socklen_t  size = sizeof(sin);
+#else
     int     size = sizeof(sin);
+#endif
 
     /*
      * Eat up the not-yet received datagram. Some systems insist on a
-- 
2.50.1

