From 4570e3aebaa97ef8885e07ebdcbc886b49a85613 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sat, 9 Aug 2025 13:25:31 +0800
Subject: [PATCH 27/32] fix static

---
 workarounds.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/workarounds.c b/workarounds.c
index b22b378..59600f4 100644
--- a/workarounds.c
+++ b/workarounds.c
@@ -8,7 +8,7 @@
   */
 
 #ifndef lint
-char    sccsid[] = "@(#) workarounds.c 1.6 96/03/19 16:22:25";
+static char    sccsid[] = "@(#) workarounds.c 1.6 96/03/19 16:22:25";
 #endif
 
 #include <sys/types.h>
-- 
2.50.1

