From f790d9e8c715b38487168411f2eaf12ded2c70c1 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sat, 9 Aug 2025 13:27:21 +0800
Subject: [PATCH 30/32] ignore missing inetdconf

---
 inetcf.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/inetcf.c b/inetcf.c
index 75017f2..49aa2f7 100644
--- a/inetcf.c
+++ b/inetcf.c
@@ -90,9 +90,7 @@ char   *conf;
 	for (i = 0; inet_files[i] && (fp = fopen(inet_files[i], "r")) == 0; i++)
 	     /* void */ ;
 	if (fp == 0) {
-	    fprintf(stderr, "Cannot find your inetd.conf or tlid.conf file.\n");
-	    fprintf(stderr, "Please specify its location.\n");
-	    exit(1);
+	    return inet_files[i];
 	}
 	conf = inet_files[i];
 	check_path(conf, &st);
-- 
2.50.1

