From 43e7b031187ee2ee1f20271bb4da4681f2b5c81e Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sat, 9 Aug 2025 13:33:20 +0800
Subject: [PATCH 32/32] fix No rule to make target 'cflags'

---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index fe97fe3..61d58cf 100644
--- a/Makefile
+++ b/Makefile
@@ -733,6 +733,8 @@ all other: config-check tcpd tcpdmatch try-from safe_finger tcpdchk $(LIB)
 
 # Invalidate all object files when the compiler options (CFLAGS) have changed.
 
+cflags: config-check
+
 config-check:
 	@set +e; test -n "$(REAL_DAEMON_DIR)" || { make; exit 1; }
 	@set +e; echo $(CFLAGS) >cflags.new ; \
-- 
2.50.1

