diff -Naur notmuch-0.29.1/configure notmuch-0.29.1.gmime/configure
--- notmuch-0.29.1/configure	2019-06-11 23:16:48.000000000 +0000
+++ notmuch-0.29.1.gmime/configure	2019-11-16 02:35:24.849637895 +0000
@@ -492,7 +492,7 @@
 GMIME_MINVER=3.0.3
 
 printf "Checking for GMime development files... "
-if pkg-config --exists "gmime-3.0 > $GMIME_MINVER"; then
+if pkg-config --exists "gmime-3.0"; then
     printf "Yes.\n"
     have_gmime=1
     gmime_cflags=$(pkg-config --cflags gmime-3.0)
