--- log4cpp-1.0.orig/doc/Makefile.am	2007-09-03 05:50:42.000000000 -0400
+++ log4cpp-1.0/doc/Makefile.am	2008-12-12 12:14:24.000000000 -0500
@@ -1,7 +1,7 @@
 SUBDIRS = html
 
 man3dir = $(mandir)/man3
-docdir  = $(prefix)/doc/@PACKAGE_TARNAME@-@PACKAGE_VERSION@
+docdir  = $(prefix)/share/doc/@PACKAGE_TARNAME@-@PACKAGE_VERSION@
 
 EXTRA_DIST = \
 	mainPage.txt
@@ -22,23 +22,23 @@
 
 
 install-data-local:
-	$(mkinstalldirs) $(man3dir)
+	$(mkinstalldirs) $(DESTDIR)/$(man3dir)
 	@for i in ./man/man3/log4cpp.3 ./man/man3/log4cpp_*.3; do \
 		inst=`basename $$i | sed 's/_/::/g'`; \
 		echo "$(INSTALL_DATA) $$i$(man3dir)/$$inst"; \
-		$(INSTALL_DATA) $$i $(man3dir)/$$inst; \
+		$(INSTALL_DATA) $$i $(DESTDIR)/$(man3dir)/$$inst; \
 	done
-	$(mkinstalldirs) $(docdir)
-	cp -r html/. $(docdir)
-	$(RM) -r -f $(docdir)/CVS \
-		$(docdir)/Makefile.am \
-		$(docdir)/Makefile.in \
-		$(docdir)/Makefile
+	$(mkinstalldirs) $(DESTDIR)/$(docdir)
+	cp -r html/. $(DESTDIR)/$(docdir)
+	$(RM) -r -f $(DESTDIR)/$(docdir)/CVS \
+		$(DESTDIR)/$(docdir)/Makefile.am \
+		$(DESTDIR)/$(docdir)/Makefile.in \
+		$(DESTDIR)/$(docdir)/Makefile
 
 uninstall-local:
-	$(RM) $(man3dir)/log4cpp.3
-	$(RM) $(man3dir)/log4cpp::*.3
-	$(RM) -r -f $(docdir)
+	$(RM) $(DESTDIR)/$(man3dir)/log4cpp.3
+	$(RM) $(DESTDIR)/$(man3dir)/log4cpp::*.3
+	$(RM) -r -f $(DESTDIR)/$(docdir)
 
 clean-local:
 	$(RM) -r latex
