diff -up log4cpp/src/Makefile.am.no-snprintf log4cpp/src/Makefile.am
--- log4cpp/src/Makefile.am.no-snprintf	2017-07-12 23:39:11.000000000 -0600
+++ log4cpp/src/Makefile.am	2019-11-20 09:24:22.934246996 -0700
@@ -2,7 +2,7 @@ lib_LTLIBRARIES = liblog4cpp.la
 
 INCLUDES = -I$(top_srcdir)/include
 
-noinst_HEADERS = snprintf.c Localtime.hh
+noinst_HEADERS = Localtime.hh
 
 liblog4cpp_la_SOURCES = \
 	Appender.cpp \
diff -up log4cpp/src/Makefile.in.no-snprintf log4cpp/src/Makefile.in
--- log4cpp/src/Makefile.in.no-snprintf	2019-11-20 09:24:22.936246987 -0700
+++ log4cpp/src/Makefile.in	2019-11-20 09:25:05.758058149 -0700
@@ -297,7 +297,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 lib_LTLIBRARIES = liblog4cpp.la
 INCLUDES = -I$(top_srcdir)/include
-noinst_HEADERS = snprintf.c Localtime.hh
+noinst_HEADERS = Localtime.hh
 liblog4cpp_la_SOURCES = Appender.cpp AppenderSkeleton.cpp \
 	AppendersFactory.cpp BufferingAppender.cpp FactoryParams.cpp \
 	LayoutsFactory.cpp LevelEvaluator.cpp Localtime.cpp \
