--- log4cpp-1.0.orig/log4cpp.pc.in	2008-12-12 11:37:04.000000000 -0500
+++ log4cpp-1.0/log4cpp.pc.in	2008-12-12 12:07:24.000000000 -0500
@@ -6,5 +6,5 @@
 Name: @PACKAGE@
 Description: C++ library for flexible logging, modeled after Log4j
 Version: @VERSION@
-Libs: -L${libdir} @log4cpp_libs@
+Libs: -L${libdir}
 Cflags: -I${includedir} @log4cpp_cflags@
--- log4cpp-1.0.orig/m4/ac_config_pkgconfig_in.m4	2004-01-27 05:27:38.000000000 -0500
+++ log4cpp-1.0/m4/ac_config_pkgconfig_in.m4	2008-12-12 12:10:47.000000000 -0500
@@ -121,7 +121,7 @@
 fi
 echo 'Version: @VERSION@' >>$PKGCONFIG_FILE
 echo 'Libs: -L${libdir} @MODULE_DOWN[]_libs@' >>$PKGCONFIG_FILE
-echo 'Cflags: -I${includedir} @MODULE_DOWN[]_cflags@' >>$PKGCONFIG_FILE
+echo 'Cflags: -I${includedir}' >>$PKGCONFIG_FILE
 m4_pushdef([PKGCONFIG_UP], [m4_translit([$1], [a-z], [A-Z])])dnl
 PKGCONFIG_UP[]_PKGCONFIG="PKGCONFIG_DIR[]$1-config"
 AC_SUBST(PKGCONFIG_UP[]_PKGCONFIG)
