From ebbb9e22afa6b2646d8cd9c359d62b604fc7f9d4 Mon Sep 17 00:00:00 2001
From: Roland Rosenfeld <roland@debian.org>
Date: Tue, 26 Sep 2023 16:49:58 +0200
Subject: [PATCH 03/11] Adapt testsuite to cope with ghostscript 10.02

---
 fig2dev/tests/bitmaps.at | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/fig2dev/tests/bitmaps.at b/fig2dev/tests/bitmaps.at
index 4b08454..bfb30af 100644
--- a/fig2dev/tests/bitmaps.at
+++ b/fig2dev/tests/bitmaps.at
@@ -245,8 +245,9 @@ FIG_FILE_TOP
 4 0 0 50 -1 16 12 4.7124 4 210 2100 7225 1575 Included postscript file\001
 EOF`
 # $blackpixels are 1488 on archlinux with ghostscript 9.20,
-#		1433 on ubuntu 16.04.2 with ghostscript 9.18
-AT_CHECK([test $blackpixels -gt 1431 && test $blackpixels -lt 1498])
+#		1433 on ubuntu 16.04.2 with ghostscript 9.18,
+#               1498 on Debian with ghostscript 10.02
+AT_CHECK([test $blackpixels -gt 1431 && test $blackpixels -lt 1500])
 AT_CLEANUP
 
 AT_SETUP([find /MediaBox in pdf file])
-- 
2.34.1

