From a22a904e73fa00511820861193ea68e5f74b3706 Mon Sep 17 00:00:00 2001
From: Thomas Loimer <thomas.loimer@tuwien.ac.at>
Date: Wed, 27 Dec 2023 18:10:22 +0100
Subject: [PATCH 9/9] Fix the header search path for out-of-tree builds

---
 fig2dev/Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fig2dev/Makefile.am b/fig2dev/Makefile.am
index 1756cf9..557ca42 100644
--- a/fig2dev/Makefile.am
+++ b/fig2dev/Makefile.am
@@ -32,6 +32,8 @@ config.vc: $(srcdir)/config.vc.in $(top_builddir)/config.status
 
 bin_PROGRAMS = fig2dev
 
+AM_CPPFLAGS = -I"$(top_srcdir)/fig2dev/dev"
+
 fig2dev_SOURCES = alloc.h bool.h bound.h bound.c colors.h colors.c \
     creationdate.h creationdate.c drivers.h fig2dev.h fig2dev.c free.h free.c \
     iso2tex.c localmath.h localmath.c messages.h messages.c object.h read1_3.c \
-- 
2.34.1

