From 9b80e795069a2531891ffd13cc6cd1746103a68a Mon Sep 17 00:00:00 2001
From: fujiwarat <takao.fujiwara1@gmail.com>
Date: Fri, 24 Jan 2025 09:40:31 +0900
Subject: [PATCH 2/3] UPSTREAM: src: Fix declaration errors with GCC 15

BUG=rhbz#2340629
---
 src/ibusenginesimple.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ibusenginesimple.c b/src/ibusenginesimple.c
index 83b04ebc..ebaa3150 100644
--- a/src/ibusenginesimple.c
+++ b/src/ibusenginesimple.c
@@ -477,7 +477,7 @@ check_hex (IBusEngineSimple *simple,
 }
 
 static IBusEngineDict *
-load_emoji_dict ()
+load_emoji_dict (void)
 {
     IBusEngineDict *emoji_dict;
     GList *keys;
@@ -520,7 +520,7 @@ check_emoji_table (IBusEngineSimple       *simple,
         g_object_ref_sink (priv->lookup_table);
     }
     if (emoji_dict == NULL)
-        emoji_dict = priv->emoji_dict = load_emoji_dict (simple);
+        emoji_dict = priv->emoji_dict = load_emoji_dict ();
 
     if (emoji_dict == NULL || emoji_dict->dict == NULL)
         return FALSE;
-- 
2.52.0

