Last-Update: 2011-08-06
Description: Always require posix

diff --git a/html2ps b/html2ps
index 18d4d4d..2876275 100755
--- a/html2ps
+++ b/html2ps
@@ -353,8 +353,8 @@ A:link, HR { color: black }
 
 EOR
 
-eval "require POSIX";
-$posix = !$@;
+use POSIX;
+$posix = 1;
 
 %extend=('quote',1, 'font',1, 'colour',1, 'hyphenation',1);
 %fal=("serif","times", "sans_serif","helvetica", "monospace","courier");
