From d74abae63e977d1e813f9209e964e0162ff8559d Mon Sep 17 00:00:00 2001
From: Kay Lin <kaymw@aosc.io>
Date: Tue, 14 Feb 2023 01:01:55 -0800
Subject: [PATCH] configure: remove the second AM_INIT_AUTOMAKE

... Otherwise autoreconf will fail
---
 configure.ac | 1 -
 1 file changed, 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 2842f04..a3b41bc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -31,4 +31,3 @@ AC_FUNC_FORK
 
 AC_CONFIG_FILES([Makefile config.h])
 AC_OUTPUT
-AM_INIT_AUTOMAKE
-- 
2.38.1

