diff -up a2ps-4.14/etc/a2ps_cfg.in.etc a2ps-4.14/etc/a2ps_cfg.in
--- a2ps-4.14/etc/a2ps_cfg.in.etc	2007-12-29 02:38:58.000000000 +0100
+++ a2ps-4.14/etc/a2ps_cfg.in	2008-04-27 01:27:55.000000000 +0200
@@ -73,6 +73,7 @@ LibraryPath: @libpath@
 # It may be useful to extend it so that a2ps can see some
 # TeX or X11 resources: it likes AFM files and PF[AB] files.
 #AppendLibraryPath: foo
+AppendLibraryPath: @apppath@
 
 
 #################################################################
diff -up a2ps-4.14/etc/Makefile.am.etc a2ps-4.14/etc/Makefile.am
--- a2ps-4.14/etc/Makefile.am.etc	2002-07-19 14:07:27.000000000 +0200
+++ a2ps-4.14/etc/Makefile.am	2008-04-27 01:27:55.000000000 +0200
@@ -2,7 +2,7 @@
 ## Makefile for a2ps' etc library directory.
 
 ogonkifydir=$(datadir)/ogonkify
-libpath = $(pkgdatadir)/sheets:$(pkgdatadir)/ps:$(pkgdatadir)/encoding:$(pkgdatadir)/afm:$(ogonkifydir)/afm:$(pkgdatadir)/ppd:$(pkgdatadir)/fonts:$(ogonkifydir)/fonts:$(pkgdatadir)
+libpath = $(sysconfdir)/$(PACKAGE):$(pkgdatadir)/sheets:$(pkgdatadir)/ps:$(pkgdatadir)/encoding:$(pkgdatadir)/afm:$(ogonkifydir)/afm:$(pkgdatadir)/ppd:$(pkgdatadir)/fonts:$(ogonkifydir)/fonts:$(pkgdatadir)
 
 pkgdata_DATA = README
 
@@ -40,7 +40,9 @@ DISTCLEANFILES = a2ps.cfg README
 
 # Building the correct a2ps.cfg
 a2ps.cfg: a2ps_cfg Makefile
-	sed "s!@libpath@!$(libpath)!" a2ps_cfg > a2ps.cfg
+	sed -e "s!@libpath@!$(libpath)!" \
+	    -e "s!@apppath@!$(PSFONT_PATH)!" \
+	    a2ps_cfg > a2ps.cfg
 
 # Building a time stamp to know the version.
 README: README.in Makefile
diff -up a2ps-4.14/etc/Makefile.in.etc a2ps-4.14/etc/Makefile.in
--- a2ps-4.14/etc/Makefile.in.etc	2007-12-29 04:28:07.000000000 +0100
+++ a2ps-4.14/etc/Makefile.in	2008-04-27 01:30:55.000000000 +0200
@@ -262,7 +262,7 @@ sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 tex = @tex@
 ogonkifydir = $(datadir)/ogonkify
-libpath = $(pkgdatadir)/sheets:$(pkgdatadir)/ps:$(pkgdatadir)/encoding:$(pkgdatadir)/afm:$(ogonkifydir)/afm:$(pkgdatadir)/ppd:$(pkgdatadir)/fonts:$(ogonkifydir)/fonts:$(pkgdatadir)
+libpath = $(sysconfdir)/$(PACKAGE):$(pkgdatadir)/sheets:$(pkgdatadir)/ps:$(pkgdatadir)/encoding:$(pkgdatadir)/afm:$(ogonkifydir)/afm:$(pkgdatadir)/ppd:$(pkgdatadir)/fonts:$(ogonkifydir)/fonts:$(pkgdatadir)
 pkgdata_DATA = README
 dsysconf = $(DESTDIR)$(sysconfdir)
 EXTRA_DIST = README.in
@@ -482,7 +482,9 @@ uninstall-local:
 
 # Building the correct a2ps.cfg
 a2ps.cfg: a2ps_cfg Makefile
-	sed "s!@libpath@!$(libpath)!" a2ps_cfg > a2ps.cfg
+	sed -e "s!@libpath@!$(libpath)!" \
+	    -e "s!@apppath@!$(PSFONT_PATH)!" \
+	    a2ps_cfg > a2ps.cfg
 
 # Building a time stamp to know the version.
 README: README.in Makefile
diff -up a2ps-4.14/Makefile.am.etc a2ps-4.14/Makefile.am
diff -up a2ps-4.14/Makefile.in.etc a2ps-4.14/Makefile.in
