diff -up a2ps-4.13/encoding/iso5.edf.iso5-minus a2ps-4.13/encoding/iso5.edf
--- a2ps-4.13/encoding/iso5.edf.iso5-minus	2007-09-25 17:57:32.000000000 +0100
+++ a2ps-4.13/encoding/iso5.edf	2007-09-25 17:57:55.000000000 +0100
@@ -69,7 +69,7 @@ Vector:
 .notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef
 .notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef space
 exclam quotedbl numbersign dollar percent ampersand quoteright
-parenleft parenright asterisk plus comma minus period slash zero one
+parenleft parenright asterisk plus comma hyphen period slash zero one
 two three four five six seven eight nine colon semicolon less equal
 greater question at A B C D E F G H I J K L M N O P Q R S T U V W X Y
 Z bracketleft backslash bracketright asciicircum underscore quoteleft
