--- a2ps-4.13/lib/fonts.c.attr	Wed Jan  9 14:41:27 2002
+++ a2ps-4.13/lib/fonts.c	Wed Jan  9 14:42:01 2002
@@ -2173,7 +2173,7 @@
  * in the encoding specific dictionary
  */
 int
-font_is_to_reencode (PARAM_UNUSED struct a2ps_job * job, const char * key)
+font_is_to_reencode (struct a2ps_job * job PARAM_UNUSED, const char * key)
 {
   return !strequ (key, "Symbol");
 }
--- a2ps-4.13/lib/lister.c.attr	Wed Jan  9 14:41:30 2002
+++ a2ps-4.13/lib/lister.c	Wed Jan  9 14:42:28 2002
@@ -333,7 +333,7 @@
 }
 
 void
-lister_fprint_vertical (struct lister * lister, PARAM_UNUSED FILE *unused,
+lister_fprint_vertical (struct lister * lister, FILE *unused PARAM_UNUSED,
 			void **items, size_t item_number,
 			lister_width_t item_width_fn,
 			lister_print_t item_print_fn)
@@ -483,7 +483,7 @@
 /* FIXME: document */
 
 void
-lister_fprint_horizontal (struct lister * lister, PARAM_UNUSED FILE *unused,
+lister_fprint_horizontal (struct lister * lister, FILE *unused PARAM_UNUSED,
 			  void **items, size_t item_number,
 			  lister_width_t item_width_fn,
 			  lister_print_t item_print_fn)
@@ -577,7 +577,7 @@
 */
 
 void
-lister_fprint_separated (struct lister * lister,PARAM_UNUSED FILE *unused,
+lister_fprint_separated (struct lister * lister,FILE *unused PARAM_UNUSED,
 			 void **items, size_t item_number,
 			 lister_width_t item_width_fn,
 			 lister_print_t item_print_fn)
--- a2ps-4.13/lib/media.c.attr	Wed Jan  9 14:41:33 2002
+++ a2ps-4.13/lib/media.c	Wed Jan  9 14:42:35 2002
@@ -190,7 +190,7 @@
 
 /* Is libpaper used? */
 bool
-a2ps_medium_libpaper_p (PARAM_UNUSED a2ps_job * job, const char * name)
+a2ps_medium_libpaper_p (a2ps_job * job PARAM_UNUSED, const char * name)
 {
   return strequ (name, LIBPAPER_MEDIUM) ? true : false;
 }
--- a2ps-4.13/lib/pathwalk.c.attr	Wed Jan  9 14:41:38 2002
+++ a2ps-4.13/lib/pathwalk.c	Wed Jan  9 14:42:47 2002
@@ -425,14 +425,14 @@
 /* Helping functions for pw_glob. */
 
 static bool
-pw_filter_fnmatch (PARAM_UNUSED const char * dir, const char *file,
+pw_filter_fnmatch (const char * dir PARAM_UNUSED, const char *file,
 		   const char *pattern)
 {
   return !fnmatch (pattern, file, 0);
 }
 
 static void
-pw_filter_da_append (PARAM_UNUSED const char * dir, const char *file,
+pw_filter_da_append (const char * dir PARAM_UNUSED, const char *file,
 		     struct darray *da)
 {
   da_append (da, xstrdup (file));
--- a2ps-4.13/lib/routines.c.attr	Wed Jan  9 14:41:42 2002
+++ a2ps-4.13/lib/routines.c	Wed Jan  9 14:42:54 2002
@@ -235,7 +235,7 @@
  * Unlink the file FILENAME.
  */
 void
-unlink2 (PARAM_UNUSED void * dummy, const char * filename)
+unlink2 (void * dummy PARAM_UNUSED, const char * filename)
 {
   message (msg_tool | msg_file, (stderr, "Unlinking file `%s'\n", filename));
 
--- a2ps-4.13/lib/fonts.l.attr	Wed Jan  9 14:58:38 2002
+++ a2ps-4.13/lib/fonts.l	Wed Jan  9 14:58:52 2002
@@ -585,7 +585,7 @@
  * in the encoding specific dictionary
  */
 int
-font_is_to_reencode (PARAM_UNUSED struct a2ps_job * job, const char * key)
+font_is_to_reencode (struct a2ps_job * job PARAM_UNUSED, const char * key)
 {
   return !strequ (key, "Symbol");
 }
