From a9c154d38f632537b54c95793d5ea111d8d35c3f Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Sun, 19 May 2024 16:24:36 +0800
Subject: [PATCH] Fix libunwind library detection on loongarch64 and riscv

---
 cmake/FindUnwind.cmake | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cmake/FindUnwind.cmake b/cmake/FindUnwind.cmake
index abe1a20ee..8c0171906 100644
--- a/cmake/FindUnwind.cmake
+++ b/cmake/FindUnwind.cmake
@@ -29,6 +29,10 @@ elseif (CMAKE_SYSTEM_PROCESSOR MATCHES "^hppa")
     set (Unwind_ARCH "unwind-hppa")
 elseif (CMAKE_SYSTEM_PROCESSOR MATCHES "^ia64")
     set (Unwind_ARCH "unwind-ia64")
+elseif (CMAKE_SYSTEM_PROCESSOR MATCHES "^loongarch64")
+    set (Unwind_ARCH "unwind-loongarch64")
+elseif (CMAKE_SYSTEM_PROCESSOR MATCHES "^riscv")
+    set (Unwind_ARCH "unwind-riscv")
 endif (CMAKE_SYSTEM_PROCESSOR MATCHES "^arm")
 
 find_library (Unwind_PLATFORM_LIBRARY NAMES ${Unwind_ARCH}
-- 
2.45.1

