From 913e794f2783309d9fef59f9f454ffbd90acaab5 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Fri, 27 Mar 2026 21:43:05 +0800
Subject: [PATCH 1/4] Remove architecture restriction in build.psm1

---
 build.psm1 | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/build.psm1 b/build.psm1
index 9c593e49d..c47aecf57 100644
--- a/build.psm1
+++ b/build.psm1
@@ -419,7 +419,7 @@ function Start-PSBuild {
         # These runtimes must match those in project.json
         # We do not use ValidateScript since we want tab completion
         # If this parameter is not provided it will get determined automatically.
-        [ValidateSet("linux-musl-x64",
+        [ArgumentCompletions("linux-musl-x64",
                      "fxdependent",
                      "fxdependent-noopt-linux-musl-x64",
                      "fxdependent-linux-x64",
@@ -1246,7 +1246,7 @@ function New-PSOptions {
 
         # These are duplicated from Start-PSBuild
         # We do not use ValidateScript since we want tab completion
-        [ValidateSet("",
+        [ArgumentCompletions("",
                      "linux-musl-x64",
                      "fxdependent",
                      "fxdependent-noopt-linux-musl-x64",
@@ -4788,7 +4788,10 @@ function Clear-NativeDependencies
             return
         }
         Default {
-            throw "Unknown runtime $($script:Options.Runtime)"
+            $parts = $($script:Options.Runtime) -split "-"
+            $arch = $parts[-1]
+            $diasymFileName = $diasymFileNamePattern -f $arch
+            Write-Warning "Unknown runtime $($script:Options.Runtime)"
         }
     }
 
-- 
2.55.0

