From 661b690e95448bbac9fd1f36a9325a0a299f1ac0 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Thu, 5 Mar 2026 17:15:13 +0800
Subject: [PATCH 2/4] don't exit when dotnet sdk version mismatch

---
 build.psm1 | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build.psm1 b/build.psm1
index c47aecf57..36817a7ad 100644
--- a/build.psm1
+++ b/build.psm1
@@ -536,7 +536,6 @@ Fix steps:
 3. Start-PSBuild -Clean
 `n
 "@
-        return
     }
 
     # set output options
-- 
2.55.0

