From 9ac32b821313f50d2d5cc405a040d3244383f882 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Thu, 26 Mar 2026 01:18:16 +0800
Subject: [PATCH 3/4] use dotnet sdk 10.0.110

---
 global.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/global.json b/global.json
index 14b50cf15..cc056fe1d 100644
--- a/global.json
+++ b/global.json
@@ -1,5 +1,5 @@
 {
   "sdk": {
-    "version": "10.0.302"
+    "version": "10.0.110"
   }
 }
-- 
2.55.0

