From 945f5a11d37ccc16da25981edcfb681692539138 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Thu, 26 Mar 2026 01:18:16 +0800
Subject: [PATCH 3/4] use dotnet sdk 10.0.111

---
 global.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/global.json b/global.json
index 00eac7708..1a5b39c8b 100644
--- a/global.json
+++ b/global.json
@@ -1,5 +1,5 @@
 {
   "sdk": {
-    "version": "10.0.303"
+    "version": "10.0.111"
   }
 }
-- 
2.55.0

