diff -Naur OpenSubdiv-3_5_0/examples/CMakeLists.txt OpenSubdiv-3_5_0.GLX/examples/CMakeLists.txt
--- OpenSubdiv-3_5_0/examples/CMakeLists.txt	2022-09-26 14:17:44.000000000 -0400
+++ OpenSubdiv-3_5_0.GLX/examples/CMakeLists.txt	2023-04-20 19:58:28.300650187 -0400
@@ -26,6 +26,8 @@
 
 if (OPENGL_FOUND AND GLFW_FOUND)
 
+    link_libraries("-lGLX")
+
     add_subdirectory(glViewer)
     add_subdirectory(glEvalLimit)
     add_subdirectory(glShareTopology)
