diff -up ratpoints-2.1.3/Makefile.orig ratpoints-2.1.3/Makefile
--- ratpoints-2.1.3/Makefile.orig	2012-05-01 20:38:42.990234312 -0300
+++ ratpoints-2.1.3/Makefile	2012-05-01 20:38:44.851234426 -0300
@@ -21,17 +21,22 @@
 #
 #    Michael Stoll, September 21, 2009
 
+MAJOR=0
 PRIME_SIZE = 7
 
 CC = gcc
 RM = rm -f
 INSTALL = cp
 
-INSTALL_DIR = /usr/local
+PREFIX=/usr
+BINDIR=/usr/bin
+INCDIR=/usr/include
+LIBDIR=/usr/lib
+INSTALL=install
 
-CCFLAGS1 = -Wall -O2 -fomit-frame-pointer -DRATPOINTS_MAX_BITS_IN_PRIME=${PRIME_SIZE} -DUSE_SSE
+CCFLAGS1 = -Wall -O2 -fomit-frame-pointer -DRATPOINTS_MAX_BITS_IN_PRIME=${PRIME_SIZE} -fPIC
 # for gcc on Apple, may have to add '-fnested-functions' to CCFLAGS1
-CCFLAGS2 = -lgmp -lgcc -lc -lm
+CCFLAGS2 = -lgmp -lm
 CCFLAGS3 = -L. -lratpoints
 CCFLAGS = 
 
@@ -47,9 +52,9 @@ TEMPFILES = sift.o init.o sturm.o find_p
             gen_find_points_h gen_init_sieve_h \
             rptest.out sift-debug.o find_points-debug.o main.o
 
-TARGETFILES = ratpoints libratpoints.a rptest ratpoints-debug
+TARGETFILES = ratpoints libratpoints.so.$(MAJOR) rptest ratpoints-debug
 
-all: ratpoints libratpoints.a
+all: ratpoints libratpoints.so.$(MAJOR)
 
 test: rptest testbase
 	time ./rptest > rptest.out
@@ -57,14 +62,13 @@ test: rptest testbase
 	cmp -s testbase rptest.out || echo "Test failed!"
 
 install-bin: ratpoints
-	${INSTALL} ratpoints ${INSTALL_DIR}/bin/
-	chmod 755 ${INSTALL_DIR}/bin/ratpoints
+	${INSTALL} -D -m755 ratpoints ${DESTDIR}${BINDIR}/ratpoints
 
-install-lib: ratpoints.h libratpoints.a
-	${INSTALL} ratpoints.h ${INSTALL_DIR}/include/
-	chmod 644 ${INSTALL_DIR}/include/ratpoints.h
-	${INSTALL} libratpoints.a ${INSTALL_DIR}/lib/
-	chmod 644 ${INSTALL_DIR}/lib/libratpoints.a
+install-lib: ratpoints.h libratpoints.so.$(MAJOR)
+	${INSTALL} -D -m644 ratpoints.h ${DESTDIR}${INCDIR}/ratpoints.h
+	${INSTALL} -d ${DESTDIR}${LIBDIR}
+	${INSTALL} -m755 libratpoints.so.$(MAJOR) ${DESTDIR}${LIBDIR}
+	ln -s libratpoints.so.$(MAJOR)  ${DESTDIR}${LIBDIR}/libratpoints.so
 
 install: install-bin install-lib
 
@@ -83,10 +87,11 @@ distclean: clean
 
 debug: ratpoints-debug
 
-libratpoints.a: sift.o init.o sturm.o find_points.o
-	ar rs libratpoints.a sift.o init.o sturm.o find_points.o
+libratpoints.so.$(MAJOR): sift.o init.o sturm.o find_points.o
+	gcc -shared -o libratpoints.so.$(MAJOR) -Wl,-soname=libratpoints.so.$(MAJOR) sift.o init.o sturm.o find_points.o
+	ln -s libratpoints.so.$(MAJOR) libratpoints.so
 
-ratpoints: libratpoints.a main.c ratpoints.h
+ratpoints: libratpoints.so.$(MAJOR) main.c ratpoints.h
 	${CC} main.c -o ratpoints ${CCFLAGS1} ${CCFLAGS2} ${CCFLAGS3} ${CCFLAGS}
 
 main.o: main.c ratpoints.h
@@ -123,7 +128,7 @@ find_points.o: find_points.c ratpoints.h
 find_points-debug.o: find_points.c ratpoints.h rp-private.h primes.h find_points.h
 	${CC} find_points.c -c -o find_points-debug.o ${CCFLAGS1} -DDEBUG ${CCFLAGS}
 
-rptest: libratpoints.a rptest.c ratpoints.h testdata.h
+rptest: libratpoints.so.$(MAJOR) rptest.c ratpoints.h testdata.h
 	${CC} rptest.c -o rptest ${CCFLAGS1} ${CCFLAGS2} ${CCFLAGS3} ${CCFLAGS}
 
 gen_init_sieve_h: gen_init_sieve_h.c ratpoints.h rp-private.h primes.h
