diff -Naur avahi-0.8/avahi-ui/bssh.desktop.in.in avahi-0.8.desktop/avahi-ui/bssh.desktop.in.in
--- avahi-0.8/avahi-ui/bssh.desktop.in.in	2019-02-26 02:32:23.428104313 +0000
+++ avahi-0.8.desktop/avahi-ui/bssh.desktop.in.in	2022-08-27 06:21:31.383902827 +0000
@@ -6,5 +6,5 @@
 Terminal=false
 Type=Application
 Icon=network-wired
-Categories=GNOME;Network;
+Categories=GNOME;System;
 StartupNotify=false
diff -Naur avahi-0.8/avahi-ui/bvnc.desktop.in.in avahi-0.8.desktop/avahi-ui/bvnc.desktop.in.in
--- avahi-0.8/avahi-ui/bvnc.desktop.in.in	2019-02-26 02:32:23.428104313 +0000
+++ avahi-0.8.desktop/avahi-ui/bvnc.desktop.in.in	2022-08-27 06:21:53.977205758 +0000
@@ -6,5 +6,5 @@
 Terminal=false
 Type=Application
 Icon=network-wired
-Categories=GNOME;Network;
+Categories=GNOME;System;
 StartupNotify=false
