From 1f1cbab23530784dac1a47d7b04750e3ca4cf806 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Thu, 27 Jan 2022 18:29:05 +0100
Subject: [PATCH 16/39] speaker-test: remove sample_map.csv from Makefile

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 speaker-test/samples/Makefile.am | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/speaker-test/samples/Makefile.am b/speaker-test/samples/Makefile.am
index 8cc3546..d024baa 100644
--- a/speaker-test/samples/Makefile.am
+++ b/speaker-test/samples/Makefile.am
@@ -10,11 +10,6 @@ wav_files = Front_Left.wav \
 	Side_Left.wav  
 sound_DATA = $(wav_files)
 
-alsadir = $(datadir)/alsa/speaker-test
-cfg_files = sample_map.csv
-
-alsa_DATA = $(cfg_files)
-
 EXTRA_DIST = \
 	$(wav_files) \
 	$(cfg_files)
-- 
2.36.0

