From 329cc772af09eb984b778f7670ab5a3be70097db Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Fri, 28 Jan 2022 09:29:34 +0100
Subject: [PATCH 19/39] topology: fix the comment for improved -D option

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 topology/topology.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/topology/topology.c b/topology/topology.c
index 026e1a4..c0c5270 100644
--- a/topology/topology.c
+++ b/topology/topology.c
@@ -58,6 +58,7 @@ _("Usage: %s [OPTIONS]...\n"
 "-o, --output=FILE       set output file\n"
 #if SND_LIB_VER(1, 2, 5) < SND_LIB_VERSION
 "-D, --define=ARGS       define variables (VAR1=VAL1[,VAR2=VAL2] ...)\n"
+"                        (may be used multiple times)\n"
 "-I, --inc-dir=DIR       set include path\n"
 #endif
 "-s, --sort              sort the identifiers in the normalized output\n"
-- 
2.36.0

