From 3afa353c0a116695fe5454948d161233b3a517a2 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Fri, 28 Jan 2022 17:50:33 +0100
Subject: [PATCH 23/39] topology: the includes should be deleted not removed

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 topology/pre-processor.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/topology/pre-processor.c b/topology/pre-processor.c
index ddfbecd..a5ef778 100644
--- a/topology/pre-processor.c
+++ b/topology/pre-processor.c
@@ -462,8 +462,8 @@ static int pre_process_includes(struct tplg_pre_processor *tplg_pp, snd_config_t
 		}
 	}
 
-	/* remove all includes from current top */
-	snd_config_remove(includes);
+	/* delete all includes from current top */
+	snd_config_delete(includes);
 
 	return 0;
 }
-- 
2.36.0

