From 31d4afd8cad0220ca18e3624b3c55dc163514147 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Fri, 28 Jan 2022 18:01:03 +0100
Subject: [PATCH 24/39] topology: create Define subtree for command line
 defines only

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 topology/pre-processor.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/topology/pre-processor.c b/topology/pre-processor.c
index a5ef778..5278240 100644
--- a/topology/pre-processor.c
+++ b/topology/pre-processor.c
@@ -253,6 +253,12 @@ static int pre_process_add_defines(struct tplg_pre_processor *tplg_pp, snd_confi
 	int ret;
 
 	ret = snd_config_search(from, "Define", &conf_defines);
+	if (ret == -ENOENT) {
+		if (tplg_pp->input_cfg == from && tplg_pp->define_cfg) {
+			conf_defines = NULL;
+			goto create;
+		}
+	}
 	if (ret < 0)
 		return ret;
 
@@ -265,6 +271,7 @@ static int pre_process_add_defines(struct tplg_pre_processor *tplg_pp, snd_confi
 		tplg_pp->define_cfg_merged = conf_defines;
 
 	if (tplg_pp->define_cfg_merged == NULL) {
+create:
 		ret = snd_config_make_compound(&tplg_pp->define_cfg_merged, "Define", 0);
 		if (ret < 0)
 			return ret;
-- 
2.36.0

