From ccb6d4612bdfeb2b68c118549a9e717489f5d70c Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Tue, 1 Feb 2022 20:01:08 +0100
Subject: [PATCH 29/39] alsactl: remove unused argument for state_lock_()

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 alsactl/lock.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/alsactl/lock.c b/alsactl/lock.c
index 5b47462..aa77254 100644
--- a/alsactl/lock.c
+++ b/alsactl/lock.c
@@ -30,7 +30,7 @@
 #include <sys/stat.h>
 #include "alsactl.h"
 
-static int state_lock_(const char *file, int lock, int timeout, int _fd)
+static int state_lock_(int lock, int timeout, int _fd)
 {
 	int fd = -1, err = 0;
 	struct flock lck;
@@ -124,7 +124,7 @@ int state_lock(const char *file, int timeout)
 {
 	int err;
 
-	err = state_lock_(file, 1, timeout, -1);
+	err = state_lock_(1, timeout, -1);
 	if (err < 0)
 		error("file %s lock error: %s", file, strerror(-err));
 	return err;
@@ -134,7 +134,7 @@ int state_unlock(int _fd, const char *file)
 {
 	int err;
 
-	err = state_lock_(file, 0, 10, _fd);
+	err = state_lock_(0, 10, _fd);
 	if (err < 0)
 		error("file %s unlock error: %s", file, strerror(-err));
 	return err;
-- 
2.36.0

