From 3862452e4eb58c36b6912e082538785c0b804b9b Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Tue, 3 May 2022 14:06:13 +0200
Subject: [PATCH 37/39] topology: add missing pre-process-external.h to
 Makefile.am

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 topology/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/topology/Makefile.am b/topology/Makefile.am
index 237eb3c..61055d8 100644
--- a/topology/Makefile.am
+++ b/topology/Makefile.am
@@ -13,7 +13,7 @@ endif
 alsatplg_SOURCES = topology.c pre-processor.c pre-process-class.c pre-process-object.c \
 		    pre-process-dapm.c pre-process-dai.c
 
-noinst_HEADERS = topology.h pre-processor.h
+noinst_HEADERS = topology.h pre-processor.h pre-process-external.h
 
 AM_CPPFLAGS = \
          -Wall -I$(top_srcdir)/include -DALSA_TOPOLOGY_PLUGIN_DIR=\"@ALSA_TOPOLOGY_PLUGIN_DIR@\"
-- 
2.36.0

