From 904bc1dfd4a97002e1bb05058cfb15a7f486d493 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Tue, 3 May 2022 14:16:43 +0200
Subject: [PATCH 38/39] topology: add missing nhlt headers to Makefile.am

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 topology/nhlt/Makefile.am | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/topology/nhlt/Makefile.am b/topology/nhlt/Makefile.am
index 5ef6a92..434e22e 100644
--- a/topology/nhlt/Makefile.am
+++ b/topology/nhlt/Makefile.am
@@ -12,5 +12,19 @@ libalsatplg_module_nhlt_la_SOURCES = nhlt-processor.c \
 				intel/ssp-nhlt.c \
 				intel/ssp/ssp-debug.c intel/ssp/ssp-process.c
 
+noinst_HEADERS = nhlt.h \
+		intel/intel-nhlt.h \
+		intel/dmic-nhlt.h \
+		intel/dmic/dmic-intel.h \
+		intel/dmic/dmic-process.h \
+		intel/dmic/dmic-debug.h \
+		intel/dmic/dmic-internal.h \
+		intel/dmic/pdm-decim-fir.h \
+		intel/ssp-nhlt.h \
+		intel/ssp/ssp-debug.h \
+		intel/ssp/ssp-intel.h \
+		intel/ssp/ssp-process.h \
+		intel/ssp/ssp-internal.h
+
 libalsatplg_module_nhlt_la_LDFLAGS = -shared
 libalsatplg_module_nhlt_la_LIBADD = -lasound
-- 
2.36.0

