diff -up a2ps-4.14/configure.conf a2ps-4.14/configure
--- a2ps-4.14/configure.conf	2008-04-27 01:01:04.000000000 +0200
+++ a2ps-4.14/configure	2008-04-27 01:04:12.000000000 +0200
@@ -30758,9 +30758,13 @@ fi
 echo $ECHO_N "checking for PostScript fonts path... $ECHO_C" >&6; }
 # Try to find some PostScript fonts.
 # Find out if ghostscript is installed
+# for ac_dir in /usr/local/ghostscript/fonts \
+#  /usr/local/share/ghostscript/fonts;
 ac_psfont_path=
-for ac_dir in /usr/local/ghostscript/fonts \
-  /usr/local/share/ghostscript/fonts;
+for ac_dir in /usr/share/fonts/default/Type1 \
+    /usr/share/fonts/default/ghostscript \
+    /usr/share/X11/fonts/Type1 \
+    /usr/share/fonts/culmus ;
 do
   if test "cd $ac_dir 2>/dev/null && echo *.afm"; then
     ac_psfont_path="$ac_psfont_path:$ac_dir";
diff -up a2ps-4.14/configure.in.conf a2ps-4.14/configure.in
--- a2ps-4.14/configure.in.conf	2008-04-27 01:00:54.000000000 +0200
+++ a2ps-4.14/configure.in	2008-04-27 01:03:52.000000000 +0200
@@ -207,8 +207,10 @@ AC_MSG_CHECKING(for PostScript fonts pat
 # Try to find some PostScript fonts.
 # Find out if ghostscript is installed
 ac_psfont_path=
-for ac_dir in /usr/local/ghostscript/fonts \
-  /usr/local/share/ghostscript/fonts;
+for ac_dir in /usr/share/fonts/default/Type1 \
+    /usr/share/fonts/default/ghostscript \
+    /usr/share/X11/fonts/Type1 \
+    /usr/share/fonts/culmus ;
 do
   if test "cd $ac_dir 2>/dev/null && echo *.afm"; then
     ac_psfont_path="$ac_psfont_path:$ac_dir";
