diff -up a2ps-4.13/sheets/perl.ssh.perl a2ps-4.13/sheets/perl.ssh
--- a2ps-4.13/sheets/perl.ssh.perl	2007-09-26 10:30:43.000000000 +0100
+++ a2ps-4.13/sheets/perl.ssh	2007-09-26 10:47:47.000000000 +0100
@@ -106,9 +106,9 @@ operators are
     (/(s|tr)/			# 1. The command: `tr' or `s'
      /(([[:punct:]])/		# 2. The delimiter
      /(/			# 3. The *two* regexps
-     /(.|\\\\.)*/		#   First regexp
+     /(\\\\.)*/			#   First regexp
      /\\3/			#   Separator
-     /(.|\\\\.)*/		#   Second Regexp
+     /(\\\\.)*/			#   Second Regexp
      /)/			#
      /\\3)/			# Closer
      \1 Keyword_strong, \2 Plain),
