--- a2ps-4.13/encoding/koi8.edf.tilde	2002-06-21 12:59:41.000000000 +0100
+++ a2ps-4.13/encoding/koi8.edf	2002-06-21 13:00:02.000000000 +0100
@@ -60,7 +60,7 @@
 greater question at A B C D E F G H I J K L M N O P Q R S T U V W X Y
 Z bracketleft backslash bracketright asciicircum underscore quoteleft
 a b c d e f g h i j k l m n o p q r s t u v w x y z braceleft bar
-braceright tilde .notdef .notdef .notdef .notdef .notdef guillmotleft
+braceright asciitilde .notdef .notdef .notdef .notdef .notdef guillmotleft
 guillmotright afii61352 .notdef .notdef .notdef .notdef .notdef
 .notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef
 .notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef
