diff -up a2ps-4.14/encoding/encoding.map.euckr a2ps-4.14/encoding/encoding.map
--- a2ps-4.14/encoding/encoding.map.euckr	2008-04-27 01:46:59.000000000 +0200
+++ a2ps-4.14/encoding/encoding.map	2008-04-27 01:46:59.000000000 +0200
@@ -100,6 +100,12 @@ japanese		euc-jp
 ja			euc-jp
 jp			euc-jp
 
+euc-kr			euc-kr
+euc-kr			euc-kr
+korean			euc-kr
+ko			euc-kr
+kr			euc-kr
+
 ########################################################################
 # Some architectures specific char sets
 ########################################################################
diff -up /dev/null a2ps-4.14/encoding/euc-kr.edf
--- /dev/null	2008-04-26 20:48:40.298012400 +0200
+++ a2ps-4.14/encoding/euc-kr.edf	2008-04-27 01:46:59.000000000 +0200
@@ -0,0 +1,104 @@
+# Description of the EUC-KR encoding
+# Copyright (c) 1988, 89, 90, 91, 92, 93 Miguel Santana
+# Copyright (c) 1995, 96, 97, 98 Akim Demaille, Miguel Santana
+# Copyright (c) 2003 Red Hat Inc.
+#
+
+#
+# This file is part of a2ps.
+# 
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; see the file COPYING.  If not, write to
+# the Free Software Foundation, 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+#
+
+#
+# This file is derived from euc-jp.edf.
+# Just font is renamed.
+# The bug should go into our bugilla. http://bugzilla.redhat.com/
+#
+
+Name: EUC-KR
+
+Documentation
+EUC-KR is Korean charset.
+EndDocumentation
+
+# If there is an unknown font requested, use Courier
+Default: Courier
+
+# Automatic spell checking :)
+Substitute: 	Times			Times-Roman
+Substitute: 	Helvetica-Italic	Helvetica-Oblique
+Substitute: 	Helvetica-BoldItalic	Helvetica-BoldOblique
+
+
+# SlantFont for Korean font
+#		new font			source font		value
+SlantFont:	Baekmuk-Gulim--KSC-EUC-H-Italic	Baekmuk-Gulim--KSC-EUC-H	0.2
+SlantFont:	Baekmuk-Dotum--KSC-EUC-H-Italic	Baekmuk-Dotum--KSC-EUC-H	0.2
+
+# Compose font for Korean EUC code
+#		target font		additional font			width	size rate
+DefaultComposite:			Baekmuk-Gulim--KSC-EUC-H		1.0	1.1
+
+# 1:2 fixed
+#DefaultComposite:			Baekmuk-Gulim--KSC-EUC-H		1.0	0.8333
+
+# 1:1 fixed
+#DefaultComposite:			Baekmuk-Gulim--KSC-EUC-H		1.0	1.6666
+
+Composite:	Helvetica		Baekmuk-Dotum--KSC-EUC-H		1.0	1.1
+Composite:	Helvetica-Oblique	Baekmuk-Dotum--KSC-EUC-H-Italic	1.0 	1.1
+Composite:	Helvetica-Bold		Baekmuk-Dotum--KSC-EUC-H  	1.0 	1.1
+Composite:	Helvetica-BoldOblique	Baekmuk-Dotum--KSC-EUC-H-Italic	1.0	1.1
+
+Composite:	Courier-Oblique		Baekmuk-Gulim--KSC-EUC-H-Italic	1.0	1.1
+Composite:	Courier-Bold		Baekmuk-Dotum--KSC-EUC-H		1.0	1.1
+Composite:	Courier-BoldOblique	Baekmuk-Dotum--KSC-EUC-H-Italic	1.0	1.1
+Composite:	Times-Bold		Baekmuk-Dotum--KSC-EUC-H		1.0	1.1
+
+Vector:
+.notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef
+.notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef
+.notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef
+.notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef
+space exclam quotedbl numbersign dollar percent ampersand quoteright
+parenleft parenright asterisk plus comma minus period slash
+zero one two three four five six seven
+eight nine colon semicolon less equal greater question
+at A B C D E F G
+H I J K L M N O
+P Q R S T U V W
+X Y Z bracketleft backslash bracketright asciicircum underscore
+quoteleft a b c d e f g
+h i j k l m n o
+p q r s t u v w
+x y z braceleft bar braceright asciitilde .notdef
+.notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef
+.notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef
+.notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef
+.notdef .notdef .notdef .notdef .notdef .notdef .notdef .notdef
+space exclamdown cent sterling currency yen brokenbar section
+dieresis copyright ordfeminine guillemotleft logicalnot hyphen registered macron
+degree plusminus twosuperior threesuperior acute mu paragraph bullet
+cedilla onesuperior ordmasculine guillemotright onequarter onehalf threequarters questiondown
+Agrave Aacute Acircumflex Atilde Adieresis Aring AE Ccedilla
+Egrave Eacute Ecircumflex Edieresis Igrave Iacute Icircumflex Idieresis
+Eth Ntilde Ograve Oacute Ocircumflex Otilde Odieresis multiply
+Oslash Ugrave Uacute Ucircumflex Udieresis Yacute Thorn germandbls
+agrave aacute acircumflex atilde adieresis aring ae ccedilla
+egrave eacute ecircumflex edieresis igrave iacute icircumflex idieresis
+eth ntilde ograve oacute ocircumflex otilde odieresis divide
+oslash ugrave uacute ucircumflex udieresis yacute thorn ydieresis
diff -up a2ps-4.14/encoding/Makefile.am.euckr a2ps-4.14/encoding/Makefile.am
--- a2ps-4.14/encoding/Makefile.am.euckr	2008-04-27 01:46:59.000000000 +0200
+++ a2ps-4.14/encoding/Makefile.am	2008-04-27 01:47:58.000000000 +0200
@@ -16,7 +16,7 @@ major_encodings = ascii.edf iso1.edf
 minor_encodings = \
 ms-cp1250.edf hp.edf mac.edf ibm-cp437.edf ibm-cp850.edf iso2.edf	\
 iso3.edf iso4.edf iso5.edf iso7.edf iso9.edf iso10.edf iso13.edf	\
-iso15.edf koi8.edf ms-cp1251.edf euc-jp.edf
+iso15.edf koi8.edf ms-cp1251.edf euc-jp.edf euc-kr.edf
 
 if EXTENSIONS
 encodings_DATA = encoding.map $(major_encodings) $(minor_encodings)
diff -up a2ps-4.14/encoding/Makefile.in.euckr a2ps-4.14/encoding/Makefile.in
--- a2ps-4.14/encoding/Makefile.in.euckr	2008-04-27 01:46:59.000000000 +0200
+++ a2ps-4.14/encoding/Makefile.in	2008-04-27 01:48:09.000000000 +0200
@@ -270,7 +270,7 @@ major_encodings = ascii.edf iso1.edf
 minor_encodings = \
 ms-cp1250.edf hp.edf mac.edf ibm-cp437.edf ibm-cp850.edf iso2.edf	\
 iso3.edf iso4.edf iso5.edf iso7.edf iso9.edf iso10.edf iso13.edf	\
-iso15.edf koi8.edf ms-cp1251.edf euc-jp.edf
+iso15.edf koi8.edf ms-cp1251.edf euc-jp.edf euc-kr.edf
 
 @EXTENSIONS_FALSE@encodings_DATA = encoding.map $(major_encodings)
 @EXTENSIONS_TRUE@encodings_DATA = encoding.map $(major_encodings) $(minor_encodings)
