diff -up a2ps-4.14/lib/output.c.format-security a2ps-4.14/lib/output.c
--- a2ps-4.14/lib/output.c.format-security	2007-12-29 01:58:21.000000000 +0000
+++ a2ps-4.14/lib/output.c	2014-04-03 18:24:35.259901356 +0100
@@ -525,7 +525,7 @@ output_file (struct output * out, a2ps_j
 		     expand_user_string (job, FIRST_FILE (job),
 					 (const uchar *) "Expand: requirement",
 					 (const uchar *) token));
-	output (dest, expansion);
+	output (dest, "%s", expansion);
 	continue;
       }
 
diff -up a2ps-4.14/lib/parseppd.y.format-security a2ps-4.14/lib/parseppd.y
--- a2ps-4.14/lib/parseppd.y.format-security	2007-12-29 01:58:21.000000000 +0000
+++ a2ps-4.14/lib/parseppd.y	2014-04-03 18:24:35.259901356 +0100
@@ -154,7 +154,7 @@ font_clause :
 void
 yyerror (const char *msg)
 {
-  error_at_line (1, 0, ppdfilename, ppdlineno, msg);
+  error_at_line (1, 0, ppdfilename, ppdlineno, "%s", msg);
 }
 
 /*
diff -up a2ps-4.14/lib/psgen.c.format-security a2ps-4.14/lib/psgen.c
--- a2ps-4.14/lib/psgen.c.format-security	2014-04-03 18:24:35.241901276 +0100
+++ a2ps-4.14/lib/psgen.c	2014-04-03 18:24:35.259901356 +0100
@@ -232,7 +232,7 @@ output_marker (a2ps_job * job, const cha
     default:
       *buf = '\0';
       ps_escape_char (job, cp[i], buf);
-      output (jdiv, (char *) buf);
+      output (jdiv, "%s", (char *) buf);
       break;
     }
   }
diff -up a2ps-4.14/src/parsessh.y.format-security a2ps-4.14/src/parsessh.y
--- a2ps-4.14/src/parsessh.y.format-security	2014-04-03 18:25:56.011259069 +0100
+++ a2ps-4.14/src/parsessh.y	2014-04-03 18:26:04.725297585 +0100
@@ -740,7 +740,7 @@ exception_def_opt:
 void
 yyerror (const char *msg)
 {
-  error_at_line (1, 0, sshfilename, sshlineno, msg);
+  error_at_line (1, 0, sshfilename, sshlineno, "%s", msg);
 }
 
 /*
