--- a2ps-4.13/afm/make_fonts_map.sh.make-fonts-map	1999-01-20 14:28:27.000000000 +0000
+++ a2ps-4.13/afm/make_fonts_map.sh	2004-12-09 12:34:45.517150083 +0000
@@ -58,23 +58,15 @@
 do
   # Extract the font name.
   name=`sed -n -e '/^FontName/{
-s/FontName[ ]*\([-a-zA-Z]*\).*/\1/p
+s/FontName[ ]*\([-a-zA-Z0-9]*\).*/\1/p
 q
 }' $file`
   shortname=`basename $file | sed -e 's/\.[^\.]*$//g'`
   if test x$name = x; then :; else
-    # This is probably not a correct AFM file.
+    # If name was empty this is probably not a correct AFM file.
     # (For instance Ogonkify's pseudo AFMs that define the encodings)
     # Forget it.
-    col1=`echo "$name$many_spaces" | sed -e 's/^\('$many_dots'\).*$/\1/g'`
-    # Make sure the name has not been cut
-    case "$col1" in
-      $name*) ;;
-      *) echo "A name has been cut ($name -> $col1)." 1>&2
-	 exit 1 ;;
-    esac
-    col2="$shortname"
-    echo "$col1$col2" >> fonts.map.new
+    printf "%-30s %s\n" $name $shortname >> fonts.map.new
   fi
 done
 
