--- a2ps-4.13/contrib/pdiff.m4.wdiff	2005-03-23 15:43:47.000000000 +0000
+++ a2ps-4.13/contrib/pdiff.m4	2005-03-23 15:46:13.000000000 +0000
@@ -15,7 +15,7 @@
 a2ps=${A2PS:-a2ps}
 a2ps_options=
 debug=
-diff_on=words
+diff_on=lines
 diff_prog=${DIFF:-diff}
 diff_options='-u'
 file=
@@ -40,8 +40,6 @@
  -h, --help           display this help and exit
  -v, --version        display version information and exit
  -q, --quiet          don't print informational messages
- -l, --lines          search for line differences (\`diff')
- -w, --words          search for word differences (\`wdiff')
  -o, --output=FILE    save the output in FILE
 
 Options for a2ps are given after \`--', for instance
--- a2ps-4.13/contrib/pdiff.in.wdiff	2005-03-23 15:43:50.000000000 +0000
+++ a2ps-4.13/contrib/pdiff.in	2005-03-23 15:46:16.000000000 +0000
@@ -29,7 +29,7 @@
 a2ps=${A2PS:-a2ps}
 a2ps_options=
 debug=
-diff_on=words
+diff_on=lines
 diff_prog=${DIFF:-diff}
 diff_options='-u'
 file=
@@ -54,8 +54,6 @@
  -h, --help           display this help and exit
  -v, --version        display version information and exit
  -q, --quiet          don't print informational messages
- -l, --lines          search for line differences (\`diff')
- -w, --words          search for word differences (\`wdiff')
  -o, --output=FILE    save the output in FILE
 
 Options for a2ps are given after \`--', for instance
