--- a2ps-4.13/contrib/pdiff.in.U	2005-05-05 12:46:05.000000000 +0100
+++ a2ps-4.13/contrib/pdiff.in	2005-05-05 12:47:34.000000000 +0100
@@ -182,7 +182,7 @@
     # We need the total number of lines
     lines=`wc -l $file1 $file2 | sed -n 3p`
     lines=`set -- $lines && echo $1`
-    $diff_prog $diff_options -$lines $file1 $file2	\
+    $diff_prog $diff_options -U$lines $file1 $file2	\
        | $a2ps -gEudiff $a2ps_options || exit 1
     ;;
 esac
--- a2ps-4.13/contrib/pdiff.m4.U	2005-05-05 12:47:54.000000000 +0100
+++ a2ps-4.13/contrib/pdiff.m4	2005-05-05 12:53:10.000000000 +0100
@@ -100,7 +100,7 @@
     # We need the total number of lines
     lines=`wc -l $file1 $file2 | sed -n 3p`
     lines=`set -- $lines && echo $1`
-    $diff_prog $diff_options -$lines $file1 $file2	\
+    $diff_prog $diff_options -U$lines $file1 $file2	\
        | $a2ps -gEudiff $a2ps_options || exit 1
     ;;
 esac
