diff -up a2ps-4.14/configure.external-libtool a2ps-4.14/configure
--- a2ps-4.14/configure.external-libtool	2008-09-24 15:28:13.000000000 +0100
+++ a2ps-4.14/configure	2008-09-24 15:28:13.000000000 +0100
@@ -21864,7 +21864,8 @@ fi
 LIBTOOL_DEPS="$ac_aux_dir/ltmain.sh"
 
 # Always use our own libtool.
-LIBTOOL='$(SHELL) $(top_builddir)/libtool'
+#LIBTOOL='$(SHELL) $(top_builddir)/libtool'
+LIBTOOL='/usr/bin/libtool'
 
 # Prevent multiple expansion
 
