diff -up a2ps-4.14/lib/darray.c.overrun-dynamic a2ps-4.14/lib/darray.c
--- a2ps-4.14/lib/darray.c.overrun-dynamic	2011-08-10 16:16:49.607014904 +0100
+++ a2ps-4.14/lib/darray.c	2011-08-10 16:17:10.965625881 +0100
@@ -430,7 +430,7 @@ da_qsort (struct darray * arr)
 	  jstack += 2;
 	  /* Push pointers to larger subarry on stack.
 	   * Process smaller subarrays now	*/
-	  if (jstack > QSORT_STACK)
+	  if (jstack >= QSORT_STACK)
 	    error (da_exit_error, 0, "da_qsort: QSORT_STACK too small (%d)",
 			   QSORT_STACK);
 	  if (ir - i + 1 >= j - l) {
@@ -509,7 +509,7 @@ da_qsort_with_arg (struct darray * arr, 
 	  jstack += 2;
 	  /* Push pointers to larger subarry on stack.
 	   * Process smaller subarrays now	*/
-	  if (jstack > QSORT_STACK)
+	  if (jstack >= QSORT_STACK)
 	    error (da_exit_error, 0, "da_qsort: QSORT_STACK too small (%d)",
 			   QSORT_STACK);
 	  if (ir - i + 1 >= j - l) {
