--- a2ps-4.13/i18n-fonts-0.1/make_fonts_map.sh.sort	2006-04-04 13:49:44.000000000 +0100
+++ a2ps-4.13/i18n-fonts-0.1/make_fonts_map.sh	2006-04-04 13:52:00.000000000 +0100
@@ -80,7 +80,7 @@
 
 # Sort them by name, and keep a unique file for each font
 echo "Sorting entries..." 1>&2
-sort -u -t' ' +0 -1 fonts.map.new > fonts.map.tmp
+sort -u -t' ' -k1,2 fonts.map.new > fonts.map.tmp
 
 echo "Finishing." 1>&2
 cat > fonts.map.new <<EOF
--- a2ps-4.13/afm/make_fonts_map.sh.sort	2006-04-04 13:52:32.000000000 +0100
+++ a2ps-4.13/afm/make_fonts_map.sh	2006-04-04 13:52:40.000000000 +0100
@@ -72,7 +72,7 @@
 
 # Sort them by name, and keep a unique file for each font
 echo "Sorting entries..." 1>&2
-sort -u -t' ' +0 -1 fonts.map.new > fonts.map.tmp
+sort -u -t' ' -k1,2 fonts.map.new > fonts.map.tmp
 
 echo "Finishing." 1>&2
 cat > fonts.map.new <<EOF
