diff -up a2ps-4.14/lib/metaseq.c.overrun-static a2ps-4.14/lib/metaseq.c
--- a2ps-4.14/lib/metaseq.c.overrun-static	2011-08-10 16:18:44.924915695 +0100
+++ a2ps-4.14/lib/metaseq.c	2011-08-10 16:19:04.739555258 +0100
@@ -334,7 +334,7 @@ grow_user_string_obstack (struct obstack
 	      {
 		/* `%D{}' format run date with strftime() */
 		for (j = 0, i += 2;
-		     j < sizeof (buf2) && str[i] && str[i] != '}';
+		     j < sizeof (buf2) - 1 && str[i] && str[i] != '}';
 		     i++, j++)
 		  buf2[j] = str[i];
 		if (str[i] != '}')
