diff -up a2ps-4.14/lib/Makefile.am.lm a2ps-4.14/lib/Makefile.am
--- a2ps-4.14/lib/Makefile.am.lm	2007-12-29 01:38:03.000000000 +0000
+++ a2ps-4.14/lib/Makefile.am	2012-04-04 09:25:00.709845857 +0100
@@ -32,7 +32,7 @@ YFLAGS = -dtv
 GPERFFLAGS = -t -D -L ANSI-C
 BUILT_SOURCES = parseppd.c parseppd.h liba2ps.h confg.c
 
-liba2ps_la_LIBADD = @LTALLOCA@ @LTLIBOBJS@
+liba2ps_la_LIBADD = @LTALLOCA@ @LTLIBOBJS@ -lm
 liba2ps_la_LDFLAGS = -version-info @LIBVERSION@
 
 liba2psheaders = encoding.h media.h jobs.h output.h		\
diff -up a2ps-4.14/lib/Makefile.in.lm a2ps-4.14/lib/Makefile.in
--- a2ps-4.14/lib/Makefile.in.lm	2012-04-04 09:25:25.091842586 +0100
+++ a2ps-4.14/lib/Makefile.in	2012-04-04 09:25:32.889841559 +0100
@@ -322,7 +322,7 @@ INCLUDES = -I$(top_builddir) -I$(top_bui
 # Use a struct, handle duplicates, produce ANSI-C
 GPERFFLAGS = -t -D -L ANSI-C
 BUILT_SOURCES = parseppd.c parseppd.h liba2ps.h confg.c
-liba2ps_la_LIBADD = @LTALLOCA@ @LTLIBOBJS@
+liba2ps_la_LIBADD = @LTALLOCA@ @LTLIBOBJS@ -lm
 liba2ps_la_LDFLAGS = -version-info @LIBVERSION@
 liba2psheaders = encoding.h media.h jobs.h output.h		\
 routines.h psgen.h prolog.h faces.h confg.h useropt.h		\
diff -up a2ps-4.14/Makefile.in.lm a2ps-4.14/Makefile.in
