--- a2ps-4.13/src/main.c.gnusource	2003-10-17 17:07:05.000000000 +0100
+++ a2ps-4.13/src/main.c	2003-10-17 17:07:17.000000000 +0100
@@ -30,6 +30,7 @@
 /*                      I n c l u d e   f i l e s                       */
 /*                                                                      */
 /************************************************************************/
+#define _GNU_SOURCE
 #include <assert.h>
 #include <string.h>
 #include <langinfo.h>
@@ -105,7 +106,6 @@
    defined twice, see lib/confg.gperf, handling of `Options:'. */
 
 char *program_name;
-const char *program_invocation_name;
 
 /* Stores the data of liba2ps.  */
 
