Bug: https://bugzilla.gnome.org/show_bug.cgi?id=616738

If I say $PYTHON is foo, just use it, kthxbye.

--- a/configure.in
+++ b/configure.in
@@ -467,7 +467,7 @@
 	    echo Found python in $with_python
 	    PYTHON="$with_python"
 	else
-	    if test -x "$PYTHON"
+	    if test -n "$PYTHON"
 	    then
 	        echo Found python in environment PYTHON=$PYTHON
 		with_python=`$PYTHON -c "import sys; print sys.exec_prefix"`
