diff -Naur a/nss/coreconf/Linux.mk b/nss/coreconf/Linux.mk
--- a/nss/coreconf/Linux.mk	2016-02-26 13:51:11.000000000 -0700
+++ b/nss/coreconf/Linux.mk	2016-04-24 15:23:01.187076164 -0600
@@ -16,9 +16,9 @@
 	IMPL_STRATEGY = _PTH
 endif
 
-CC			= gcc
-CCC			= g++
-RANLIB			= ranlib
+CC			= i686-pc-linux-gnu-gcc
+CCC			= i686-pc-linux-gnu-g++
+RANLIB			= i686-pc-linux-gnu-ranlib
 
 DEFAULT_COMPILER = gcc
 
