diff -Naur a/src/desktoptheme/breeze/metadata.desktop b/src/desktoptheme/breeze/metadata.desktop
--- a/src/desktoptheme/breeze/metadata.desktop	2016-10-02 05:03:44.000000000 -0500
+++ b/src/desktoptheme/breeze/metadata.desktop	2016-10-08 11:54:16.994191289 -0500
@@ -53,7 +53,7 @@
 X-Plasma-API=5.0
 
 [Wallpaper]
-defaultWallpaperTheme=Next
+defaultWallpaperTheme=B
 defaultFileSuffix=.png
 defaultWidth=1920
 defaultHeight=1080
diff -Naur a/src/desktoptheme/breeze-dark/metadata.desktop b/src/desktoptheme/breeze-dark/metadata.desktop
--- a/src/desktoptheme/breeze-dark/metadata.desktop	2016-10-02 05:03:44.000000000 -0500
+++ b/src/desktoptheme/breeze-dark/metadata.desktop	2016-10-08 11:54:08.924273858 -0500
@@ -53,7 +53,7 @@
 X-Plasma-API=5.0
 
 [Wallpaper]
-defaultWallpaperTheme=Next
+defaultWallpaperTheme=B
 defaultFileSuffix=.png
 defaultWidth=1920
 defaultHeight=1080
diff -Naur a/src/desktoptheme/breeze-light/metadata.desktop b/src/desktoptheme/breeze-light/metadata.desktop
--- a/src/desktoptheme/breeze-light/metadata.desktop	2016-10-02 05:03:44.000000000 -0500
+++ b/src/desktoptheme/breeze-light/metadata.desktop	2016-10-08 11:54:13.047565034 -0500
@@ -43,7 +43,7 @@
 X-Plasma-API=5.0
 
 [Wallpaper]
-defaultWallpaperTheme=Next
+defaultWallpaperTheme=B
 defaultFileSuffix=.png
 defaultWidth=1920
 defaultHeight=1080
