diff -Naur kamoso-22.04.3/src/org.kde.kamoso.desktop kamoso-22.04.3.desktop/src/org.kde.kamoso.desktop
--- kamoso-22.04.3/src/org.kde.kamoso.desktop	2022-09-14 01:30:15.688120083 +0000
+++ kamoso-22.04.3.desktop/src/org.kde.kamoso.desktop	2022-09-14 01:30:45.729534663 +0000
@@ -133,4 +133,4 @@
 Type=Application
 X-DocPath=kamoso/index.html
 Terminal=false
-Categories=Qt;KDE;AudioVideo;
+Categories=Qt;KDE;Utility;
