diff -Naur drkonqi-5.26.4/CMakeLists.txt drkonqi-5.26.4.userfeedback/CMakeLists.txt
--- drkonqi-5.26.4/CMakeLists.txt	2022-11-29 03:21:46.000000000 -0800
+++ drkonqi-5.26.4.userfeedback/CMakeLists.txt	2023-01-12 00:44:13.435162497 -0800
@@ -58,9 +58,6 @@
         SyntaxHighlighting
 )
 
-find_package(KUserFeedback)
-set_package_properties(KUserFeedback PROPERTIES TYPE REQUIRED PURPOSE "Checking whether feedback is enabled or not")
-
 ecm_find_qmlmodule(org.kde.kirigami 2.19)
 ecm_find_qmlmodule(org.kde.kitemmodels 1.0)
 ecm_find_qmlmodule(org.kde.kcm 1.6)
diff -Naur drkonqi-5.26.4/src/CMakeLists.txt drkonqi-5.26.4.userfeedback/src/CMakeLists.txt
--- drkonqi-5.26.4/src/CMakeLists.txt	2022-11-29 03:21:46.000000000 -0800
+++ drkonqi-5.26.4.userfeedback/src/CMakeLists.txt	2023-01-12 00:44:10.717217767 -0800
@@ -103,7 +103,6 @@
     drkonqi_backtrace_parser
     qbugzilla
     KF5::Declarative
-    KUserFeedbackCore
 )
 
 if(Systemd_FOUND)
diff -Naur drkonqi-5.26.4/src/bugzillaintegration/reportinterface.cpp drkonqi-5.26.4.userfeedback/src/bugzillaintegration/reportinterface.cpp
--- drkonqi-5.26.4/src/bugzillaintegration/reportinterface.cpp	2022-11-29 03:21:46.000000000 -0800
+++ drkonqi-5.26.4.userfeedback/src/bugzillaintegration/reportinterface.cpp	2023-01-12 00:43:53.523576485 -0800
@@ -14,7 +14,6 @@
 
 #include <KIO/TransferJob>
 #include <KLocalizedString>
-#include <KUserFeedback/Provider>
 
 #include "backtracegenerator.h"
 #include "bugzillalib.h"
@@ -58,13 +57,6 @@
         m_sentryUserFeedbackSent = true;
         maybeDone();
     });
-    if (KUserFeedback::Provider provider; provider.isEnabled() && provider.telemetryMode() == KUserFeedback::Provider::DetailedUsageStatistics
-        && !DrKonqi::isTestingBugzilla() && qgetenv("DRKONQI_KDE_BUGZILLA_URL").isEmpty() && !DrKonqi::crashedApplication()->hasDeletedFiles()) {
-        metaObject()->invokeMethod(this, [this] {
-            // Send crash event ASAP, if applicable. Trace quality doesn't matter for it.
-            sendCrashEvent();
-        });
-    }
 }
 
 void ReportInterface::setBugAwarenessPageData(bool rememberSituation, Reproducible reproducible, bool actions, bool unusual, bool configuration)
