diff -Naur drkonqi-5.27.0/CMakeLists.txt drkonqi-5.27.0.userfeedback/CMakeLists.txt
--- drkonqi-5.27.0/CMakeLists.txt	2023-02-09 03:58:46.000000000 -0800
+++ drkonqi-5.27.0.userfeedback/CMakeLists.txt	2023-02-17 13:02:59.790166311 -0800
@@ -57,9 +57,6 @@
         SyntaxHighlighting
 )
 
-find_package(KUserFeedback)
-set_package_properties(KUserFeedback PROPERTIES TYPE REQUIRED PURPOSE "Checking whether feedback is enabled or not")
-
 ecm_find_qmlmodule(org.kde.kirigami 2.19)
 ecm_find_qmlmodule(org.kde.kitemmodels 1.0)
 ecm_find_qmlmodule(org.kde.kcm 1.6)
diff -Naur drkonqi-5.27.0/src/CMakeLists.txt drkonqi-5.27.0.userfeedback/src/CMakeLists.txt
--- drkonqi-5.27.0/src/CMakeLists.txt	2023-02-09 03:58:46.000000000 -0800
+++ drkonqi-5.27.0.userfeedback/src/CMakeLists.txt	2023-02-17 13:03:02.779192419 -0800
@@ -103,7 +103,6 @@
     KF5::SyntaxHighlighting # Backtrace Highlighting
     drkonqi_backtrace_parser
     qbugzilla
-    KUserFeedbackCore
 )
 
 if(Systemd_FOUND)
diff -Naur drkonqi-5.27.0/src/bugzillaintegration/reportinterface.cpp drkonqi-5.27.0.userfeedback/src/bugzillaintegration/reportinterface.cpp
--- drkonqi-5.27.0/src/bugzillaintegration/reportinterface.cpp	2023-02-09 03:58:46.000000000 -0800
+++ drkonqi-5.27.0.userfeedback/src/bugzillaintegration/reportinterface.cpp	2023-02-17 13:02:50.542086819 -0800
@@ -14,7 +14,6 @@
 
 #include <KIO/TransferJob>
 #include <KLocalizedString>
-#include <KUserFeedback/Provider>
 
 #include "backtracegenerator.h"
 #include "bugzillalib.h"
@@ -58,13 +57,6 @@
         m_sentryUserFeedbackSent = true;
         maybeDone();
     });
-    if (KUserFeedback::Provider provider; provider.isEnabled() && provider.telemetryMode() == KUserFeedback::Provider::DetailedUsageStatistics
-        && !DrKonqi::isTestingBugzilla() && qgetenv("DRKONQI_KDE_BUGZILLA_URL").isEmpty() && !DrKonqi::crashedApplication()->hasDeletedFiles()) {
-        metaObject()->invokeMethod(this, [this] {
-            // Send crash event ASAP, if applicable. Trace quality doesn't matter for it.
-            sendCrashEvent();
-        });
-    }
 }
 
 void ReportInterface::setBugAwarenessPageData(bool rememberSituation, Reproducible reproducible, bool actions, bool unusual, bool configuration)
