diff -up mozjs-31.2.0/mfbt/Endian.h.orig mozjs-31.2.0/mfbt/Endian.h
--- mozjs-31.2.0/mfbt/Endian.h.orig	2016-05-20 11:03:47.242434797 -0300
+++ mozjs-31.2.0/mfbt/Endian.h	2016-05-20 11:05:24.553438523 -0300
@@ -148,7 +148,11 @@
 #  endif
 #elif defined(__GNUC__)
 #  if MOZ_GCC_VERSION_AT_LEAST(4, 8, 0)
+/* FIXME Fedora does not make mandatory NEON and rawhide gcc
+ * intrinsic needs it for __builtin_bswap16 */
+#    ifndef __arm__
 #    define MOZ_HAVE_BUILTIN_BYTESWAP16 __builtin_bswap16
+#    endif
 #  endif
 #elif defined(_MSC_VER)
 #    define MOZ_HAVE_BUILTIN_BYTESWAP16 _byteswap_ushort
