From 3beb66b96f89396316ed2ee08efe16c13abccbed Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=93=BA=E7=9B=96=E5=B4=BD?= <i@pugai.life>
Date: Tue, 8 Apr 2025 09:00:44 +0800
Subject: [PATCH] use gtk portal for file chooser dialogues

---
 resources/niri-portals.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/resources/niri-portals.conf b/resources/niri-portals.conf
index 3f13d669a..a79c5aa58 100644
--- a/resources/niri-portals.conf
+++ b/resources/niri-portals.conf
@@ -1,5 +1,6 @@
 [preferred]
 default=gnome;gtk;
 org.freedesktop.impl.portal.Access=gtk;
+org.freedesktop.impl.portal.FileChooser=gtk;
 org.freedesktop.impl.portal.Notification=gtk;
 org.freedesktop.impl.portal.Secret=gnome-keyring;
