From bbf82b52a84f5db749c97f08d48a50d8c97f8e91 Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Sat, 14 Jun 2025 19:38:24 +0800
Subject: [PATCH] cmake build shared lib

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 45b9125..5bc53ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,7 +71,7 @@ file(GLOB AP4_HEADERS
   ${SOURCE_METADATA}/*.h
 )
 
-add_library(ap4 STATIC ${AP4_SOURCES})
+add_library(ap4 SHARED ${AP4_SOURCES})
 target_include_directories(ap4 PUBLIC
   ${AP4_INCLUDE_DIRS}
 )
-- 
2.49.0

