From c3ecdcb1dc676c88f2b6167d4a25a9196d60b633 Mon Sep 17 00:00:00 2001
From: BillZhou233 <46444349+BillZhou233@users.noreply.github.com>
Date: Sat, 4 Apr 2026 19:41:21 +0800
Subject: [PATCH 2/2] AOSCOS: fix mime xml path

---
 cmake/platforms/linux.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/platforms/linux.cmake b/cmake/platforms/linux.cmake
index 0f2f4e4..1c23c28 100644
--- a/cmake/platforms/linux.cmake
+++ b/cmake/platforms/linux.cmake
@@ -4,7 +4,7 @@ include(GNUInstallDirs)
 
 set(LEMON_LINUX_ICON_DIMENSIONS 16 22 24 32 36 44 48 64 72 96 128 150 192 256 310 512 1024)
 install(FILES assets/lemon-lime.metainfo.xml.in DESTINATION "${CMAKE_INSTALL_DATADIR}/metainfo" RENAME lemon-lime.metainfo.xml)
-install(FILES assets/x-lemon-contest.xml.in DESTINATION "${CMAKE_INSTALL_DATADIR}/mime/application" RENAME x-lemon-contest.xml)
+install(FILES assets/x-lemon-contest.xml.in DESTINATION "${CMAKE_INSTALL_DATADIR}/mime/packages" RENAME x-lemon-contest.xml)
 install(FILES assets/lemon-lime.desktop.in DESTINATION "${CMAKE_INSTALL_DATADIR}/applications" RENAME lemon-lime.desktop)
 #install(FILES assets/icons/lemon-lime.svg DESTINATION "${CMAKE_INSTALL_DATADIR}/icons/hicolor/scalable/apps")
 foreach(LEMON_LINUX_ICON_DIMENSION ${LEMON_LINUX_ICON_DIMENSIONS})
-- 
2.50.1 (Apple Git-155)

