diff -Naur gstreamer-old/gst/gstutils.h gstreamer/gst/gstutils.h
--- gstreamer-old/gst/gstutils.h	2020-09-25 09:34:16.000000000 +0800
+++ gstreamer/gst/gstutils.h	2022-08-23 16:28:30.150194228 +0800
@@ -136,7 +136,7 @@
   /* The typedef for GType may be gulong or gsize, depending on the	\
    * system and whether the compiler is c++ or not. The g_once_init_*	\
    * functions always take a gsize * though ... */			\
-  static volatile gsize gonce_data = 0;					\
+  static gsize gonce_data = 0;						\
   if (g_once_init_enter (&gonce_data)) {				\
     GType _type;							\
     _type = gst_type_register_static_full (parent_type_macro,           \
