Author: Petr Salinger <Petr.Salinger@seznam.cz>
Description: Fixes FTBFS on kFreeBSD (Closes: #497013).

diff -Naurp libcdaudio.orig/src/cdaudio.c libcdaudio/src/cdaudio.c
--- libcdaudio.orig/src/cdaudio.c	2008-06-21 14:49:54.000000000 +0200
+++ libcdaudio/src/cdaudio.c	2008-08-29 10:41:07.000000000 +0200
@@ -118,8 +118,7 @@ cd_init_device(char *device_name)
 #elif defined(HAVE_GETMNTENT)
   FILE *mounts;
   struct mntent *mnt;
-#endif
-#ifdef HAVE_GETMNTINFO
+#elif defined(HAVE_GETMNTINFO)
   int mounts;
   struct statfs *mnt;
 #endif
@@ -163,8 +162,7 @@ cd_init_device(char *device_name)
     }
   }
   endmntent(mounts);
-#endif
-#ifdef HAVE_GETMNTINFO
+#elif defined(HAVE_GETMNTINFO)
   for ( (mounts = getmntinfo(&mnt, 0)); mounts > 0;)
     {
       mounts--;
