diff --git a/conf/main.cf b/conf/main.cf
index 2b51635c7..9ed867407 100644
--- a/conf/main.cf
+++ b/conf/main.cf
@@ -407,9 +407,9 @@ unknown_local_recipient_reject_code = 550
 # "postfix reload" to eliminate the delay.
 #
 #alias_maps = dbm:/etc/aliases
-#alias_maps = hash:/etc/aliases
 #alias_maps = hash:/etc/aliases, nis:mail.aliases
 #alias_maps = netinfo:/aliases
+alias_maps = hash:/etc/aliases
 
 # The alias_database parameter specifies the alias database(s) that
 # are built with "newaliases" or "sendmail -bi".  This is a separate
@@ -420,6 +420,7 @@ unknown_local_recipient_reject_code = 550
 #alias_database = dbm:/etc/mail/aliases
 #alias_database = hash:/etc/aliases
 #alias_database = hash:/etc/aliases, hash:/opt/majordomo/aliases
+alias_database = hash:/etc/aliases
 
 # ADDRESS EXTENSIONS (e.g., user+foo)
 #
