From a1962846e8d170e3bcd5f5fb61c432bbdbaa3855 Mon Sep 17 00:00:00 2001
From: rvalue <i@rvalue.moe>
Date: Sat, 29 Jan 2022 20:33:04 +0800
Subject: [PATCH] Update `elfhack.h` to support RISC-V architecture

---
 glinject/elfhacks.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/glinject/elfhacks.h b/glinject/elfhacks.h
index 3c351e2..a23f5d7 100644
--- a/glinject/elfhacks.h
+++ b/glinject/elfhacks.h
@@ -44,6 +44,11 @@ extern "C" {
 #ifdef __i386__
 # define __elf32
 #endif
+#if __riscv_xlen == 64
+# define __elf64
+#elif __riscv_xlen == 32
+# define __elf32
+#endif
 
 #if defined(__elf64)
 # define ELFW_R_SYM ELF64_R_SYM
-- 
2.39.1

