From 5253ffc920879bce6fc71e6d7dd9a7df9dd97bff Mon Sep 17 00:00:00 2001
From: grover92000 <96498785+grover92000@users.noreply.github.com>
Date: Sat, 13 Apr 2024 15:23:38 -0400
Subject: [PATCH 2/4] Update AudioEncoder.cpp for ffmpeg-7.0

---
 src/AV/Output/AudioEncoder.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/AV/Output/AudioEncoder.cpp b/src/AV/Output/AudioEncoder.cpp
index 34d015c..77d6cde 100644
--- a/src/AV/Output/AudioEncoder.cpp
+++ b/src/AV/Output/AudioEncoder.cpp
@@ -69,7 +69,11 @@ AVSampleFormat AudioEncoder::GetSampleFormat() {
 }
 
 unsigned int AudioEncoder::GetChannels() {
+#if LIBAVCODEC_VERSION_MAJOR < 61
 	return GetCodecContext()->channels;
+#else
+	return GetCodecContext()->ch_layout.nb_channels;
+#endif
 }
 
 unsigned int AudioEncoder::GetSampleRate() {
@@ -106,8 +110,13 @@ void AudioEncoder::PrepareStream(AVStream* stream, AVCodecContext* codec_context
 	}
 
 	codec_context->bit_rate = bit_rate;
+#if LIBAVCODEC_VERSION_MAJOR < 61
 	codec_context->channels = channels;
 	codec_context->channel_layout = (channels == 1)? AV_CH_LAYOUT_MONO : AV_CH_LAYOUT_STEREO;
+#else
+	codec_context->ch_layout.nb_channels = channels;
+	codec_context->ch_layout.u.mask = (channels == 1)? AV_CH_LAYOUT_MONO : AV_CH_LAYOUT_STEREO;
+#endif
 	codec_context->sample_rate = sample_rate;
 	codec_context->time_base.num = 1;
 	codec_context->time_base.den = sample_rate;
-- 
2.46.0

