From 1ab1e8b270b7d533e381b3185c87ff86ca9b600d Mon Sep 17 00:00:00 2001
From: grover92000 <96498785+grover92000@users.noreply.github.com>
Date: Sat, 13 Apr 2024 15:25:35 -0400
Subject: [PATCH 3/4] Update Synchronizer.cpp for ffmpeg-7.0

---
 src/AV/Output/Synchronizer.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/AV/Output/Synchronizer.cpp b/src/AV/Output/Synchronizer.cpp
index b109dc6..b4f9159 100644
--- a/src/AV/Output/Synchronizer.cpp
+++ b/src/AV/Output/Synchronizer.cpp
@@ -180,7 +180,11 @@ static std::unique_ptr<AVFrameWrapper> CreateAudioFrame(unsigned int channels, u
 	frame->GetFrame()->nb_samples = samples;
 #endif
 #if SSR_USE_AVFRAME_CHANNELS
+#if LIBAVCODEC_VERSION_MAJOR < 61
 	frame->GetFrame()->channels = channels;
+#else
+	frame->GetFrame()->ch_layout.nb_channels = channels;
+#endif
 #endif
 #if SSR_USE_AVFRAME_SAMPLE_RATE
 	frame->GetFrame()->sample_rate = sample_rate;
-- 
2.46.0

