From 5dfe4918fa178a5870b2320e4c04bef346d87c08 Mon Sep 17 00:00:00 2001
From: Halla Rempt <halla@valdyas.org>
Date: Thu, 14 Mar 2024 11:57:12 +0100
Subject: [PATCH] Re-enable the workaround for plasma global menu

BUG:483170
---
 libs/ui/opengl/KisOpenGLModeProber.cpp | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/libs/ui/opengl/KisOpenGLModeProber.cpp b/libs/ui/opengl/KisOpenGLModeProber.cpp
index f76ded8b114..fba1b8c0e19 100644
--- a/libs/ui/opengl/KisOpenGLModeProber.cpp
+++ b/libs/ui/opengl/KisOpenGLModeProber.cpp
@@ -160,7 +160,6 @@ KisOpenGLModeProber::probeFormat(const KisOpenGL::RendererConfig &rendererConfig
     char *argv = probeAppName.data();
 
 
-
     if (adjustGlobalState) {
         sharedContextSetter.reset(new AppAttributeSetter(Qt::AA_ShareOpenGLContexts, false));
 
@@ -175,9 +174,19 @@ KisOpenGLModeProber::probeFormat(const KisOpenGL::RendererConfig &rendererConfig
 
         // Disable this workaround for plasma (BUG:408015), because it causes 
         // a crash on Windows with Qt 5.15.7
-        //QGuiApplication::setDesktopSettingsAware(false);
+        const bool runningInKDE =  qEnvironmentVariableIsSet("KDE_FULL_SESSION");
+        const bool isInAppimage = qEnvironmentVariableIsSet("APPIMAGE");
+
+        if (runningInKDE && !isInAppimage) {
+            QGuiApplication::setDesktopSettingsAware(false);
+        }
+
         application.reset(new QGuiApplication(argc, &argv));
-        //QGuiApplication::setDesktopSettingsAware(true);
+
+        if (runningInKDE && !isInAppimage) {
+            QGuiApplication::setDesktopSettingsAware(true);
+        }
+
     }
 
     QWindow surface;
-- 
GitLab

