--- a/src/Makefile.orig	2017-06-17 19:41:54.816138857 +0800
+++ b/src/Makefile	2017-06-17 19:43:20.644727915 +0800
@@ -104,8 +104,8 @@
 
 ifeq ($(MALLOC),jemalloc)
 	DEPENDENCY_TARGETS+= jemalloc
-	FINAL_CFLAGS+= -DUSE_JEMALLOC -I../deps/jemalloc/include
-	FINAL_LIBS+= ../deps/jemalloc/lib/libjemalloc.a
+	FINAL_CFLAGS+= -DUSE_JEMALLOC -DJEMALLOC_NO_DEMANGLE -I/usr/include/jemalloc
+	FINAL_LIBS+= -ljemalloc
 endif
 
 REDIS_CC=$(QUIET_CC)$(CC) $(FINAL_CFLAGS)
