From 8efb6f2c26673c4490f3a1512330322644a1d83a Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 17 Jun 2024 22:41:41 +0800
Subject: [PATCH] feat(src/default.h): set AOSC OS defaults

---
 src/default.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/default.h b/src/default.h
index 6adbe95d..9e71afbf 100644
--- a/src/default.h
+++ b/src/default.h
@@ -41,9 +41,9 @@ XIV(bool, taskBarShowApm,                       false)
 XIV(bool, taskBarShowApmAuto,                   true)
 XIV(bool, taskBarShowApmTime,                   true)
 XIV(bool, taskBarShowApmGraph,                  true)
-XIV(bool, taskBarShowMailboxStatus,             true)
+XIV(bool, taskBarShowMailboxStatus,             false)
 XIV(bool, taskBarShowStartMenu,                 true)
-XIV(bool, taskBarShowWindowListMenu,            true)
+XIV(bool, taskBarShowWindowListMenu,            false)
 XIV(bool, taskBarShowWorkspaces,                true)
 XIV(bool, taskBarShowWindows,                   true)
 XIV(int, taskBarShowShowDesktopButton,          1)
@@ -72,14 +72,14 @@ XIV(bool, pagerShowMinimized,                   true)
 XIV(bool, pagerShowBorders,                     true)
 XIV(bool, pagerShowLabels,                      true)
 XIV(bool, pagerShowNumbers,                     false)
-XIV(bool, taskBarShowCPUStatus,                 true)
+XIV(bool, taskBarShowCPUStatus,                 false)
 XIV(bool, cpustatusShowRamUsage,                true)
 XIV(bool, cpustatusShowSwapUsage,               true)
 XIV(bool, cpustatusShowAcpiTemp,                true)
 XIV(bool, cpustatusShowAcpiTempInGraph,         false)
 XIV(bool, cpustatusShowCpuFreq,                 true)
 XIV(bool, netstatusShowOnlyRunning,             false)
-XIV(bool, taskBarShowMEMStatus,                 true)
+XIV(bool, taskBarShowMEMStatus,                 false)
 XIV(bool, taskBarShowNetStatus,                 true)
 XIV(bool, taskBarLaunchOnSingleClick,           true)
 XIV(bool, taskBarShowCollapseButton,            false)
-- 
2.45.1

