diff --git a/lutris/startup.py b/lutris/startup.py
index 5504a5b0..24c15ecd 100644
--- a/lutris/startup.py
+++ b/lutris/startup.py
@@ -112,24 +112,6 @@ def check_libs(all_components=False):
                     missing_vulkan_libs.append(arch)
                 logger.error("%s %s missing (needed by %s)", arch, lib, req.lower())
 
-    if missing_vulkan_libs:
-        setting = "dismiss-missing-vulkan-library-warning"
-        if settings.read_setting(setting) != "True":
-            DontShowAgainDialog(
-                setting,
-                _("Missing vulkan libraries"),
-                secondary_message=_(
-                    "Lutris was unable to detect Vulkan support for "
-                    "the %s architecture.\n"
-                    "This will prevent many games and programs from working.\n"
-                    "To install it, please use the following guide: "
-                    "<a href='%s'>Installing Graphics Drivers</a>"
-                ) % (
-                    _(" and ").join(missing_vulkan_libs),
-                    settings.DRIVER_HOWTO_URL,
-                )
-            )
-
 
 def check_vulkan():
     """Reports if Vulkan is enabled on the system"""
diff --git a/lutris/util/linux.py b/lutris/util/linux.py
index 78a0882b..e4ea199e 100644
--- a/lutris/util/linux.py
+++ b/lutris/util/linux.py
@@ -95,6 +95,7 @@ class LinuxSystem:  # pylint: disable=too-many-public-methods
         ("/usr/lib32", "/usr/lib64"),
         ("/lib/i386-linux-gnu", "/lib/x86_64-linux-gnu"),
         ("/usr/lib/i386-linux-gnu", "/usr/lib/x86_64-linux-gnu"),
+        ("/usr/lib", "/opt/32/lib"),
     ]
 
     soundfont_folders = [
