--- a/src-ILUT/src/ilut_opengl.c
+++ b/src-ILUT/src/ilut_opengl.c
@@ -84,7 +84,7 @@ void *aglGetProcAddress( const GLubyte *
 
 static ILboolean HasCubemapHardware = IL_FALSE;
 static ILboolean HasNonPowerOfTwoHardware = IL_FALSE;
-#if defined(_WIN32) || defined(_WIN64) || defined(linux) || defined(__APPLE__)
+#if defined(_WIN32) || defined(_WIN64) || defined(linux) || defined(__APPLE__) || defined(__GLIBC__)
 	ILGLTEXIMAGE3DARBPROC			ilGLTexImage3D = NULL;
 	ILGLTEXSUBIMAGE3DARBPROC		ilGLTexSubImage3D = NULL;
 	ILGLCOMPRESSEDTEXIMAGE2DARBPROC	ilGLCompressed2D = NULL;
