--- a/src-IL/src/il_jp2.c	2016-12-10 12:54:27.033747667 +0800
+++ b/src-IL/src/il_jp2.c	2016-12-10 12:54:37.157241391 +0800
@@ -459,7 +459,7 @@
 			/* The buffer must be large enough to accommodate maximum
 			  putback. */
 			assert(bufsize > JAS_STREAM_MAXPUTBACK);
-			stream->bufbase_ = JAS_CAST(uchar *, buf);
+			stream->bufbase_ = JAS_CAST(unsigned char *, buf);
 			stream->bufsize_ = bufsize - JAS_STREAM_MAXPUTBACK;
 		}
 	} else {
