diff -Naur open-vm-tools-11.0.1-14773994/configure.ac open-vm-tools-11.0.1-14773994.nowerror/configure.ac
--- open-vm-tools-11.0.1-14773994/configure.ac	2019-10-01 22:29:15.000000000 +0000
+++ open-vm-tools-11.0.1-14773994.nowerror/configure.ac	2020-01-21 20:55:30.714582717 +0000
@@ -1161,7 +1161,7 @@
 
 ### General flags / actions
 CFLAGS="$CFLAGS -Wall"
-CFLAGS="$CFLAGS -Werror"
+CFLAGS="$CFLAGS"
 
 # -Wno-unknown-pragmas is due to gcc not understanding '#pragma ident'
 # in Xlib.h on OpenSolaris.
