diff -Naur chromium-69.0.3497.100/components/policy/core/common/policy_map.cc components/policy/core/common/policy_map.cc
--- a/components/policy/core/common/policy_map.cc	2018-09-17 19:10:03.000000000 +0000
+++ b/components/policy/core/common/policy_map.cc	2018-09-22 02:00:04.562815850 +0000
@@ -16,7 +16,7 @@
 PolicyMap::Entry::~Entry() = default;
 
 PolicyMap::Entry::Entry(Entry&&) noexcept = default;
-PolicyMap::Entry& PolicyMap::Entry::operator=(Entry&&) noexcept = default;
+PolicyMap::Entry& PolicyMap::Entry::operator=(Entry&&) = default;
 
 PolicyMap::Entry PolicyMap::Entry::DeepCopy() const {
   Entry copy;
diff -Naur chromium-69.0.3497.100/gpu/config/gpu_info.h gpu/config/gpu_info.h
--- a/gpu/config/gpu_info.h	2018-09-17 19:10:11.000000000 +0000
+++ b/gpu/config/gpu_info.h	2018-09-22 01:54:29.982767940 +0000
@@ -114,7 +114,7 @@
     GPUDevice(GPUDevice&& other) noexcept;
     ~GPUDevice() noexcept;
     GPUDevice& operator=(const GPUDevice& other);
-    GPUDevice& operator=(GPUDevice&& other) noexcept;
+    GPUDevice& operator=(GPUDevice&& other);
 
     // The DWORD (uint32_t) representing the graphics card vendor id.
     uint32_t vendor_id;
