--- chromium-72.0.3626.81/third_party/webrtc/modules/desktop_capture/linux/window_capturer_x11.cc	2019-01-30 01:20:37.000000000 +0000
+++ chromium-72.0.3626.81.working/third_party/webrtc/modules/desktop_capture/linux/window_capturer_x11.cc	2019-01-30 05:07:05.608286258 +0000
@@ -17,6 +17,7 @@
 #include <memory>
 #include <string>
 #include <utility>
+#include <string.h>
 
 #include "modules/desktop_capture/desktop_frame.h"
 #include "modules/desktop_capture/linux/shared_x_display.h"
