diff --git a/components/bookmarks/browser/bookmark_storage.cc b/components/bookmarks/browser/bookmark_storage.cc
index 1633ba1..3ae0c62 100644
--- a/components/bookmarks/browser/bookmark_storage.cc
+++ b/components/bookmarks/browser/bookmark_storage.cc
@@ -158,6 +158,10 @@
   url_index_ = std::make_unique<UrlIndex>(std::move(root_node_));
 }
 
+std::unique_ptr<UrlIndex> BookmarkLoadDetails::owned_url_index() {
+  return std::move(url_index_);
+}
+
 BookmarkPermanentNode* BookmarkLoadDetails::CreatePermanentNode(
     BookmarkClient* client,
     BookmarkNode::Type type) {
diff --git a/components/bookmarks/browser/bookmark_storage.h b/components/bookmarks/browser/bookmark_storage.h
index 08df5bb..0a1b1a1 100644
--- a/components/bookmarks/browser/bookmark_storage.h
+++ b/components/bookmarks/browser/bookmark_storage.h
@@ -104,7 +104,7 @@
   bool ids_reassigned() const { return ids_reassigned_; }
 
   void CreateUrlIndex();
-  std::unique_ptr<UrlIndex> owned_url_index() { return std::move(url_index_); }
+  std::unique_ptr<UrlIndex> owned_url_index();
 
  private:
   // Creates one of the possible permanent nodes (bookmark bar node, other node

