diff -up chromium-54.0.2840.59/third_party/icu/icu.gyp.armfix chromium-54.0.2840.59/third_party/icu/icu.gyp
--- chromium-54.0.2840.59/third_party/icu/icu.gyp.armfix	2016-10-13 14:57:47.148707100 -0400
+++ chromium-54.0.2840.59/third_party/icu/icu.gyp	2016-10-13 14:59:24.294837176 -0400
@@ -44,7 +44,7 @@
       }],
       ['(OS=="linux" or OS=="freebsd" or OS=="openbsd" or OS=="solaris" \
          or OS=="netbsd" or OS=="mac" or OS=="android" or OS=="qnx") and \
-        (target_arch=="arm" or target_arch=="ia32" or \
+        (target_arch=="ia32" or \
          target_arch=="mipsel" or target_arch=="mips")', {
         'target_conditions': [
           ['_toolset=="host"', {
