diff --git a/components/signin/core/browser/account_info.cc b/components/signin/core/browser/account_info.cc
--- a/components/signin/core/browser/account_info.cc	2018-12-15 12:50:40.734981198 -0600
+++ b/components/signin/core/browser/account_info.cc	2018-12-15 12:50:57.191567263 -0600
@@ -28,11 +28,11 @@
 
 AccountInfo::AccountInfo(const AccountInfo& other) = default;
 
-AccountInfo::AccountInfo(AccountInfo&& other) noexcept = default;
+AccountInfo::AccountInfo(AccountInfo&& other) = default;
 
 AccountInfo& AccountInfo::operator=(const AccountInfo& other) = default;
 
-AccountInfo& AccountInfo::operator=(AccountInfo&& other) noexcept = default;
+AccountInfo& AccountInfo::operator=(AccountInfo&& other) = default;
 
 bool AccountInfo::IsEmpty() const {
   return account_id.empty() && email.empty() && gaia.empty() &&
