diff -Naur a/chrome/common/media_router/media_sink.cc b/chrome/common/media_router/media_sink.cc
--- a/chrome/common/media_router/media_sink.cc	2018-12-15 11:57:24.523870184 -0600
+++ b/chrome/common/media_router/media_sink.cc	2018-12-15 11:58:00.293701597 -0600
@@ -19,12 +19,12 @@
       provider_id_(provider_id) {}
 
 MediaSink::MediaSink(const MediaSink& other) = default;
-MediaSink::MediaSink(MediaSink&& other) noexcept = default;
+MediaSink::MediaSink(MediaSink&& other) = default;
 MediaSink::MediaSink() = default;
 MediaSink::~MediaSink() = default;
 
 MediaSink& MediaSink::operator=(const MediaSink& other) = default;
-MediaSink& MediaSink::operator=(MediaSink&& other) noexcept = default;
+MediaSink& MediaSink::operator=(MediaSink&& other) = default;
 
 bool MediaSink::Equals(const MediaSink& other) const {
   return sink_id_ == other.sink_id_;

