diff -up chromium-54.0.2840.59/build/toolchain/linux/BUILD.gn.nullfix chromium-54.0.2840.59/build/toolchain/linux/BUILD.gn
--- chromium-54.0.2840.59/build/toolchain/linux/BUILD.gn.nullfix	2016-10-13 16:49:26.844221023 -0400
+++ chromium-54.0.2840.59/build/toolchain/linux/BUILD.gn	2016-10-13 16:50:49.882605484 -0400
@@ -31,6 +31,7 @@ gcc_toolchain("arm") {
   ld = cxx
   readelf = "${toolprefix}readelf"
   nm = "${toolprefix}nm"
+  extra_cppflags = "-fno-delete-null-pointer-checks"
 
   toolchain_args = {
     current_cpu = "arm"
@@ -70,6 +71,7 @@ gcc_toolchain("x86") {
   nm = "nm"
   ar = "ar"
   ld = cxx
+  extra_cppflags = "-fno-delete-null-pointer-checks"
 
   toolchain_args = {
     current_cpu = "x86"
@@ -109,6 +111,7 @@ gcc_toolchain("x64") {
   nm = "nm"
   ar = "ar"
   ld = cxx
+  extra_cppflags = "-fno-delete-null-pointer-checks"
 
   toolchain_args = {
     current_cpu = "x64"
@@ -131,6 +134,7 @@ gcc_toolchain("mipsel") {
   ld = cxx
   readelf = "mipsel-linux-gnu-readelf"
   nm = "mipsel-linux-gnu-nm"
+  extra_cppflags = "-fno-delete-null-pointer-checks"
 
   toolchain_args = {
     cc_wrapper = ""
