diff -up chromium-54.0.2840.59/third_party/BUILD.gn.jpegfix chromium-54.0.2840.59/third_party/BUILD.gn
--- chromium-54.0.2840.59/third_party/BUILD.gn.jpegfix	2016-10-17 11:45:44.995340495 -0400
+++ chromium-54.0.2840.59/third_party/BUILD.gn	2016-10-17 11:46:35.254289872 -0400
@@ -20,6 +20,7 @@ declare_args() {
 config("system_libjpeg_config") {
   libs = [ "jpeg" ]
   defines = [ "USE_SYSTEM_LIBJPEG" ]
+  include_dirs = [ "/usr/include/" ]
 }
 
 config("libjpeg_turbo_config") {
