diff -up chromium-77.0.3865.75/third_party/zlib/zconf.h.nozmangle chromium-77.0.3865.75/third_party/zlib/zconf.h
--- chromium-77.0.3865.75/third_party/zlib/zconf.h.nozmangle	2019-09-12 09:36:37.924086850 +0200
+++ chromium-77.0.3865.75/third_party/zlib/zconf.h	2019-09-12 09:53:01.623958551 +0200
@@ -9,18 +9,6 @@
 #define ZCONF_H
 
 /*
- * This library is also built as a part of AOSP, which does not need to include
- * chromeconf.h. This config does not want chromeconf.h, so it can set this
- * macro to opt out. While this works today, there's no guarantee that building
- * zlib outside of Chromium keeps working in the future.
- */
-#if !defined(CHROMIUM_ZLIB_NO_CHROMECONF)
-/* This include does prefixing as below, but with an updated set of names.  Also
- * sets up export macros in component builds. */
-#include "chromeconf.h"
-#endif
-
-/*
  * If you *really* need a unique prefix for all types and library functions,
  * compile with -DZ_PREFIX. The "standard" zlib should be compiled without it.
  * Even better than compiling with -DZ_PREFIX would be to use configure to set
