diff -Naur chromium-90.0.4430.72/components/translate/core/language_detection/ngram_hash_ops_utils.cc chromium-90.0.4430.72.llvm11/components/translate/core/language_detection/ngram_hash_ops_utils.cc
--- chromium-90.0.4430.72/components/translate/core/language_detection/ngram_hash_ops_utils.cc	2021-04-14 11:41:01.000000000 -0700
+++ chromium-90.0.4430.72.llvm11/components/translate/core/language_detection/ngram_hash_ops_utils.cc	2021-04-15 22:15:54.123535101 -0700
@@ -8,6 +8,8 @@
 
 #include "third_party/utf/src/include/utf.h"
 
+#include <string.h>
+
 namespace translate {
 
 constexpr char kPrefix[] = "^";
diff -Naur chromium-90.0.4430.72/components/translate/core/language_detection/quantization_utils.h chromium-90.0.4430.72.llvm11/components/translate/core/language_detection/quantization_utils.h
--- chromium-90.0.4430.72/components/translate/core/language_detection/quantization_utils.h	2021-04-14 11:41:01.000000000 -0700
+++ chromium-90.0.4430.72.llvm11/components/translate/core/language_detection/quantization_utils.h	2021-04-15 22:14:35.914579340 -0700
@@ -7,6 +7,7 @@
 
 #include <algorithm>
 #include <cmath>
+#include <stdint.h>
 
 namespace translate {
 
