diff -up chromium-96.0.4664.45/remoting/signaling/message_tracker.cc.gcc-remoting-constexpr chromium-96.0.4664.45/remoting/signaling/message_tracker.cc
--- chromium-96.0.4664.45/remoting/signaling/message_tracker.cc.gcc-remoting-constexpr	2021-11-19 16:53:23.197177348 -0500
+++ chromium-96.0.4664.45/remoting/signaling/message_tracker.cc	2021-11-19 17:03:16.354910826 -0500
@@ -9,7 +9,7 @@
 namespace remoting {
 
 // static
-const base::TimeDelta MessageTracker::kCleanupInterval = base::Minutes(2);
+constexpr base::TimeDelta MessageTracker::kCleanupInterval;
 
 MessageTracker::MessageTracker() = default;
 
diff -up chromium-96.0.4664.45/remoting/signaling/message_tracker.h.gcc-remoting-constexpr chromium-96.0.4664.45/remoting/signaling/message_tracker.h
--- chromium-96.0.4664.45/remoting/signaling/message_tracker.h.gcc-remoting-constexpr	2021-11-12 05:24:33.000000000 -0500
+++ chromium-96.0.4664.45/remoting/signaling/message_tracker.h	2021-11-19 17:03:00.112809836 -0500
@@ -40,7 +40,8 @@ class MessageTracker final {
 
   // All IDs older than now - kCleanupInterval will be eventually removed, but
   // they are not guaranteed to be immediately removed after the interval.
-  static const base::TimeDelta kCleanupInterval;
+  static constexpr base::TimeDelta kCleanupInterval =
+      base::Minutes(2);
 
   void RemoveExpiredIds();
 
