diff -up chromium-91.0.4472.77/third_party/closure_compiler/compiler.py.java-allowed chromium-91.0.4472.77/third_party/closure_compiler/compiler.py
--- chromium-91.0.4472.77/third_party/closure_compiler/compiler.py.java-allowed	2021-06-02 17:14:48.445064647 +0000
+++ chromium-91.0.4472.77/third_party/closure_compiler/compiler.py	2021-06-02 17:15:12.994836949 +0000
@@ -13,8 +13,7 @@ import subprocess
 
 
 _CURRENT_DIR = os.path.join(os.path.dirname(__file__))
-_JAVA_PATH = os.path.join(_CURRENT_DIR, "..", "jdk", "current", "bin", "java")
-assert os.path.isfile(_JAVA_PATH), "java only allowed in android builds"
+_JAVA_PATH = "java"
 
 class Compiler(object):
   """Runs the Closure compiler on given source files to typecheck them
