diff -up chromium-96.0.4664.45/net/test/embedded_test_server/http2_connection.cc.no-const chromium-96.0.4664.45/net/test/embedded_test_server/http2_connection.cc
--- chromium-96.0.4664.45/net/test/embedded_test_server/http2_connection.cc.no-const	2021-11-23 18:56:57.700174009 +0000
+++ chromium-96.0.4664.45/net/test/embedded_test_server/http2_connection.cc	2021-11-23 18:57:39.405201162 +0000
@@ -24,10 +24,9 @@ namespace net {
 
 namespace {
 
-std::vector<const http2::adapter::Header> GenerateHeaders(
-    HttpStatusCode status,
-    base::StringPairs headers) {
-  std::vector<const http2::adapter::Header> response_vector;
+std::vector<http2::adapter::Header> GenerateHeaders(HttpStatusCode status,
+                                                    base::StringPairs headers) {
+  std::vector<http2::adapter::Header> response_vector;
   response_vector.emplace_back(
       http2::adapter::HeaderRep(std::string(":status")),
       http2::adapter::HeaderRep(base::NumberToString(status)));
