diff -up chromium-108.0.5359.124/content/public/common/content_features.cc.me chromium-108.0.5359.124/content/public/common/content_features.cc
--- chromium-108.0.5359.124/content/public/common/content_features.cc.me	2022-12-24 11:02:56.417773922 +0100
+++ chromium-108.0.5359.124/content/public/common/content_features.cc	2022-12-24 11:04:41.641696681 +0100
@@ -1479,7 +1479,7 @@ BASE_FEATURE(kRetryGetVideoCaptureDevice
 // Wayland display server.
 BASE_FEATURE(kWebRtcPipeWireCapturer,
              "WebRTCPipeWireCapturer",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 #endif  // defined(WEBRTC_USE_PIPEWIRE)
 
 enum class VideoCaptureServiceConfiguration {
