diff -up chromium-109.0.5414.74/build/config/compiler/BUILD.gn.link-error-clang14 chromium-109.0.5414.74/build/config/compiler/BUILD.gn
--- chromium-109.0.5414.74/build/config/compiler/BUILD.gn.link-error-clang14	2023-01-11 18:00:34.692512606 +0100
+++ chromium-109.0.5414.74/build/config/compiler/BUILD.gn	2023-01-11 18:30:10.905410122 +0100
@@ -601,12 +601,7 @@ config("compiler") {
         cflags_cc += [ "-fno-trigraphs" ]
       }
     } else if (is_clang) {
-      if (is_chromeos_device) {
-        # TODO(crbug.com/1392471): Support C++20 in CrOS toolchain.
-        cflags_cc += [ "-std=${standard_prefix}++17" ]
-      } else {
-        cflags_cc += [ "-std=${standard_prefix}++20" ]
-      }
+      cflags_cc += [ "-std=${standard_prefix}++17" ]
     } else {
       # The gcc bots are currently using GCC 9, which is not new enough to
       # support "c++20"/"gnu++20".
