diff -up chromium-90.0.4430.93/third_party/llvm/lldb/test/API/functionalities/postmortem/netbsd-core/1lwp_SIGSEGV.aarch64 chromium-90.0.4430.93/third_party/llvm/lldb/test/API/functionalities/postmortem/netbsd-core/1lwp_SIGSEGV
diff -up chromium-90.0.4430.93/third_party/llvm/lldb/test/API/functionalities/postmortem/netbsd-core/2lwp_process_SIGSEGV.aarch64 chromium-90.0.4430.93/third_party/llvm/lldb/test/API/functionalities/postmortem/netbsd-core/2lwp_process_SIGSEGV
diff -up chromium-90.0.4430.93/third_party/llvm/lldb/test/API/functionalities/postmortem/netbsd-core/2lwp_t2_SIGSEGV.aarch64 chromium-90.0.4430.93/third_party/llvm/lldb/test/API/functionalities/postmortem/netbsd-core/2lwp_t2_SIGSEGV
diff -up chromium-90.0.4430.93/third_party/pdfium/third_party/libpng16/pngprefix.h.aarch64 chromium-90.0.4430.93/third_party/pdfium/third_party/libpng16/pngprefix.h
--- chromium-90.0.4430.93/third_party/pdfium/third_party/libpng16/pngprefix.h.aarch64	2021-05-03 14:08:47.184438786 -0400
+++ chromium-90.0.4430.93/third_party/pdfium/third_party/libpng16/pngprefix.h	2021-05-03 14:09:59.289797317 -0400
@@ -458,4 +458,11 @@
 #define png_zlib_inflate PDFIUM_png_zlib_inflate
 #define png_zstream_error PDFIUM_png_zstream_error
 
+#if PNG_ARM_NEON_IMPLEMENTATION == 1
+#define png_riffle_palette_neon PDFIUM_png_riffle_palette_neon
+#define png_do_expand_palette_rgba8_neon PDFIUM_png_do_expand_palette_rgba8_neon
+#define png_do_expand_palette_rgb8_neon PDFIUM_png_do_expand_palette_rgb8_neon
+#endif
+
+
 #endif  // PNGPREFIX_H
