diff -up chromium-67.0.3396.62/third_party/crashpad/crashpad/util/misc/capture_context_linux.S.aarch64fix chromium-67.0.3396.62/third_party/crashpad/crashpad/util/misc/capture_context_linux.S
--- chromium-67.0.3396.62/third_party/crashpad/crashpad/util/misc/capture_context_linux.S.aarch64fix	2018-06-04 14:58:04.730083967 -0400
+++ chromium-67.0.3396.62/third_party/crashpad/crashpad/util/misc/capture_context_linux.S	2018-06-04 14:58:24.896572816 -0400
@@ -288,7 +288,7 @@ CAPTURECONTEXT_SYMBOL2:
 #elif defined(__aarch64__)
 
   // Zero out fault_address, which is unused.
-  str x31, [x0, #0xb0]  // context->uc_mcontext.fault_address
+  str xzr, [x0, #0xb0]  // context->uc_mcontext.fault_address
 
   // Save general purpose registers in context->uc_mcontext.regs[i].
   // The original x0 can't be recovered.
diff -up chromium-67.0.3396.62/third_party/skia/src/opts/SkRasterPipeline_opts.h.aarch64fix chromium-67.0.3396.62/third_party/skia/src/opts/SkRasterPipeline_opts.h
