diff -Naur a/chromium-launcher-6/launcher.c b/chromium-launcher-6/launcher.c
--- a/chromium-launcher-6/launcher.c	2020-03-04 13:30:44.323726089 +0800
+++ b/chromium-launcher-6/launcher.c	2020-03-04 13:43:45.817310109 +0800
@@ -21,12 +21,16 @@
 static GSList *get_user_flags(const char *conf_path) {
   GSList *flags = NULL;
 
+#if defined(__aarch64__)
+  flags = g_slist_append(flags, g_strdup("--use-gl=egl"));
+#endif
+
   if (!conf_path)
-    return NULL;
+    return flags;
 
   FILE *file = fopen(conf_path, "r");
   if (!file)
-    return NULL;
+    return flags;
 
   {
     char buf[LINE_MAX];
