diff -up chromium-52.0.2743.82/printing/backend/print_backend_cups.cc.cups22 chromium-52.0.2743.82/printing/backend/print_backend_cups.cc
--- chromium-52.0.2743.82/printing/backend/print_backend_cups.cc.cups22	2016-07-22 09:20:15.794340886 -0400
+++ chromium-52.0.2743.82/printing/backend/print_backend_cups.cc	2016-07-22 09:20:26.338261197 -0400
@@ -16,6 +16,7 @@
 #include "base/synchronization/lock.h"
 #include "base/values.h"
 #include "printing/backend/cups_helper.h"
+#include <cups/ppd.h>
 #include "printing/backend/print_backend_consts.h"
 #include "url/gurl.h"
 
diff -up chromium-52.0.2743.82/printing/BUILD.gn.cups22 chromium-52.0.2743.82/printing/BUILD.gn
--- chromium-52.0.2743.82/printing/BUILD.gn.cups22	2016-07-21 11:27:05.597507544 -0400
+++ chromium-52.0.2743.82/printing/BUILD.gn	2016-07-21 14:36:45.574137758 -0400
@@ -147,12 +147,13 @@ component("printing") {
                                  ],
                                  "trim string")
 
-      if (cups_version == "1.6" || cups_version == "1.7") {
+      if (cups_version == "1.6" || cups_version == "1.7" || cups_version == "2.2") {
         cflags += [
           # CUPS 1.6 deprecated the PPD APIs, but we will stay with this
           # API for now as supported Linux and Mac OS'es are still using
           # older versions of CUPS. More info: crbug.com/226176
           "-Wno-deprecated-declarations",
+          "-D_PPD_DEPRECATED=\"\"",
           # CUPS 1.7 deprecates httpConnectEncrypt(), see the mac section
           # below.
         ]
diff -up chromium-52.0.2743.82/printing/printing.gyp.cups22 chromium-52.0.2743.82/printing/printing.gyp
--- chromium-52.0.2743.82/printing/printing.gyp.cups22	2016-07-21 11:26:26.622746918 -0400
+++ chromium-52.0.2743.82/printing/printing.gyp	2016-07-21 14:36:07.390373060 -0400
@@ -142,12 +142,13 @@
             'cups_version': '<!(python cups_config_helper.py --api-version <(sysroot))',
           },
           'conditions': [
-            ['cups_version in ["1.6", "1.7"]', {
+            ['cups_version in ["1.6", "1.7", "2.2"]', {
               'cflags': [
                 # CUPS 1.6 deprecated the PPD APIs, but we will stay with this
                 # API for now as supported Linux and Mac OS'es are still using
                 # older versions of CUPS. More info: crbug.com/226176
                 '-Wno-deprecated-declarations',
+                '-D_PPD_DEPRECATED=""',
                 # CUPS 1.7 deprecates httpConnectEncrypt(), see the mac section
                 # below.
               ],
