diff -up chromium-81.0.4044.92/base/test/icu_test_util.h.missing-memory chromium-81.0.4044.92/base/test/icu_test_util.h
--- chromium-81.0.4044.92/base/test/icu_test_util.h.missing-memory	2020-04-15 07:52:11.239815139 -0400
+++ chromium-81.0.4044.92/base/test/icu_test_util.h	2020-04-15 07:52:19.982632306 -0400
@@ -5,6 +5,7 @@
 #ifndef BASE_TEST_ICU_TEST_UTIL_H_
 #define BASE_TEST_ICU_TEST_UTIL_H_
 
+#include <memory>
 #include <string>
 
 #include "base/macros.h"
