From 6b0b6fca64cb0bc707a7f4d629a93efa9e967d13 Mon Sep 17 00:00:00 2001
From: Jan Tojnar <jtojnar@gmail.com>
Date: Fri, 1 Oct 2021 23:11:26 +0200
Subject: [PATCH] gir: Fix build with WebkitGTK 4.1

---
 devhelp/meson.build | 2 +-
 meson.build         | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/devhelp/meson.build b/devhelp/meson.build
index 010e563f..0afdbe76 100644
--- a/devhelp/meson.build
+++ b/devhelp/meson.build
@@ -126,7 +126,7 @@ libdevhelp_gir = GNOME.generate_gir(
   header: 'devhelp/devhelp.h',
   identifier_prefix: 'Dh',
   include_directories: ROOT_INCLUDE_DIR,
-  includes: ['Gtk-3.0', 'WebKit2-4.0'],
+  includes: ['Gtk-3.0', 'WebKit2-' + webkit_abi],
   install: true,
   namespace: 'Devhelp',
   nsversion: LIBDEVHELP_API_VERSION_FULL,
diff --git a/meson.build b/meson.build
index 47ba0b3d..990762e4 100644
--- a/meson.build
+++ b/meson.build
@@ -47,7 +47,9 @@ lt_age = 0
 LIBDEVHELP_LT_VERSION = '@0@.@1@.@2@'.format(lt_current, lt_revision, lt_age)
 
 webkit_dep = dependency('webkit2gtk-4.1', required : false)
+webkit_abi = '4.1'
 if not webkit_dep.found()
+  webkit_abi = '4.0'
   webkit_dep = dependency('webkit2gtk-4.0')
 endif
 
-- 
2.36.0

