diff --git a/common.pri b/common.pri
index 924c57c..3e104f5 100644
--- a/common.pri
+++ b/common.pri
@@ -10,5 +10,9 @@ greaterThan(QT_MAJOR_VERSION, 5)|\
 }
 TEMPLATE -= fakelib
 
-QTSINGLEAPPLICATION_LIBDIR = $$PWD/lib
+PREFIX = $$PREFIX
+isEmpty(PREFIX) {
+    PREFIX = /usr
+}
+QTSINGLEAPPLICATION_LIBDIR = $$PREFIX/lib
 unix:qtsingleapplication-uselib:!qtsingleapplication-buildlib:QMAKE_RPATHDIR += $$QTSINGLEAPPLICATION_LIBDIR
-- 
2.30.2

