From 0c64305824ea58c75ffaec8d7d5885ceb2eccda1 Mon Sep 17 00:00:00 2001
From: WhiredPlanck <whiredplanck@outlook.com>
Date: Wed, 27 Oct 2021 11:32:09 +0800
Subject: [PATCH] QtSingleApplication: allow to use custom prefix

---
 qtsingleapplication/common.pri | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/qtsingleapplication/common.pri b/qtsingleapplication/common.pri
index 924c57c..3e104f5 100644
--- a/qtsingleapplication/common.pri
+++ b/qtsingleapplication/common.pri
@@ -10,5 +10,9 @@ greaterThan(QT_MAJOR_VERSION, 5)|\
 }
 TEMPLATE -= fakelib
 
-QTSINGLEAPPLICATION_LIBDIR = $$PWD/lib
+PREFIX = $$PREFIX
+isEmpty(PREFIX) {
+    PREFIX = /usr
+}
+QTSINGLEAPPLICATION_LIBDIR = $$PREFIX/lib
 unix:qtsingleapplication-uselib:!qtsingleapplication-buildlib:QMAKE_RPATHDIR += $$QTSINGLEAPPLICATION_LIBDIR
-- 
2.30.2

