--- rstudio-1.4.1103/src/cpp/session/CMakeLists.txt	2021-01-06 15:46:50.000000000 -0800
+++ rstudio-1.4.1103.pandoc/src/cpp/session/CMakeLists.txt	2021-02-16 23:13:42.291716937 -0800
@@ -25,10 +25,6 @@
 if(NOT EXISTS "${RSTUDIO_DEPENDENCIES_DIR}/common/mathjax-27")
   message(FATAL_ERROR "Mathjax 2.7 not found (re-run install-dependencies script to install)")
 endif()
-set(PANDOC_VERSION "2.11.2" CACHE INTERNAL "Pandoc version")
-if(NOT EXISTS "${RSTUDIO_DEPENDENCIES_DIR}/common/pandoc/${PANDOC_VERSION}")
-  message(FATAL_ERROR "pandoc ${PANDOC_VERSION} not found (re-run install-dependencies script to install)")
-endif()
 
 # verify embedded packages are available
 foreach(PKG ${RSTUDIO_EMBEDDED_PACKAGES})
