diff -ruN a/usb-modeswitch-2.6.0/usb_modeswitch.conf b/usb-modeswitch-2.6.0/usb_modeswitch.conf
--- a/usb-modeswitch-2.6.0/usb_modeswitch.conf	2019-11-28 05:24:38.000000000 +0800
+++ b/usb-modeswitch-2.6.0/usb_modeswitch.conf	2025-05-27 14:03:10.007219409 +0800
@@ -1,7 +1,7 @@
 # Configuration for the usb_modeswitch package, a mode switching tool for
 # USB devices providing multiple states or modes
 #
-# Evaluated by the wrapper script /usr/sbin/usb_modeswitch_dispatcher
+# Evaluated by the wrapper script /usr/bin/usb_modeswitch_dispatcher
 #
 # To enable an option, set it to "1", "yes" or "true" (case doesn't matter)
 # Everything else counts as "disable"
diff -ruN a/usb-modeswitch-2.6.0/usb_modeswitch_dispatcher.tcl b/usb-modeswitch-2.6.0/usb_modeswitch_dispatcher.tcl
--- a/usb-modeswitch-2.6.0/usb_modeswitch_dispatcher.tcl	2019-11-29 04:14:34.000000000 +0800
+++ b/usb-modeswitch-2.6.0/usb_modeswitch_dispatcher.tcl	2025-05-27 14:03:10.010198299 +0800
@@ -285,7 +285,7 @@
 	if [CheckMBIM] {
 		Log " driver for MBIM devices is available"
 		Log "Find MBIM configuration number ..."
-		if [catch {set cfgno [exec /usr/sbin/usb_modeswitch -j -Q $busParam $devParam -v $usb(idVendor) -p $usb(idProduct)]} err] {
+		if [catch {set cfgno [exec /usr/bin/usb_modeswitch -j -Q $busParam $devParam -v $usb(idVendor) -p $usb(idProduct)]} err] {
 			Log "Error when trying to find MBIM configuration, switch to legacy modem mode"
 		} else {
 			set cfgno [string trim $cfgno]
@@ -321,7 +321,7 @@
 	# Now we are actually switching
 	if $flags(logging) {
 		Log "Command line:\nusb_modeswitch -W -D $configParam $busParam $devParam -v $usb(idVendor) -p $usb(idProduct) -f \$flags(config)"
-		catch {set report [exec /usr/sbin/usb_modeswitch -W -D $configParam $busParam $devParam -v $usb(idVendor) -p $usb(idProduct) -f "$flags(config)" 2>@1]} report
+		catch {set report [exec /usr/bin/usb_modeswitch -W -D $configParam $busParam $devParam -v $usb(idVendor) -p $usb(idProduct) -f "$flags(config)" 2>@1]} report
 		Log "\nVerbose debug output of usb_modeswitch and libusb follows"
 		Log "(Note that some USB errors are to be expected in the process)"
 		Log "--------------------------------"
@@ -329,7 +329,7 @@
 		Log "--------------------------------"
 		Log "(end of usb_modeswitch output)\n"
 	} else {
-		catch {set report [exec /usr/sbin/usb_modeswitch -Q -D $configParam $busParam $devParam -v $usb(idVendor) -p $usb(idProduct) -f "$flags(config)" 2>@1]} report
+		catch {set report [exec /usr/bin/usb_modeswitch -Q -D $configParam $busParam $devParam -v $usb(idVendor) -p $usb(idProduct) -f "$flags(config)" 2>@1]} report
 	}
 }
 
diff -ruN a/usb-modeswitch-2.6.0/usb_modeswitch.sh b/usb-modeswitch-2.6.0/usb_modeswitch.sh
--- a/usb-modeswitch-2.6.0/usb_modeswitch.sh	2019-11-28 04:04:55.000000000 +0800
+++ b/usb-modeswitch-2.6.0/usb_modeswitch.sh	2025-05-27 14:03:10.009456515 +0800
@@ -46,7 +46,7 @@
 	--symlink-name)
 		device_in "link_list" $v_id $p_id
 		if [ "$?" = "1" ]; then
-			if [ -e "/usr/sbin/usb_modeswitch_dispatcher" ]; then
+			if [ -e "/usr/bin/usb_modeswitch_dispatcher" ]; then
 				exec usb_modeswitch_dispatcher $1 $2 2>>/dev/null
 			fi
 		fi
