diff -ruN a/usb-modeswitch-data-20170120/40-usb_modeswitch.rules b/usb-modeswitch-data-20170120/40-usb_modeswitch.rules
--- a/usb-modeswitch-data-20170120/40-usb_modeswitch.rules	2017-01-21 05:41:17.000000000 +0800
+++ b/usb-modeswitch-data-20170120/40-usb_modeswitch.rules	2025-05-27 05:34:00.897289788 +0800
@@ -395,6 +395,9 @@
 # Olivetti Olicard 500
 ATTR{idVendor}=="0b3c", ATTR{idProduct}=="f017", RUN+="usb_modeswitch '/%k'"
 
+# Realtek Semiconductor Corp. RTL8188GU 802.11n WLAN Adapter
+ATTR{idVendor}=="0bda", ATTR{idProduct}=="1a2b", RUN+="usb_modeswitch '/%k'"
+
 # Ericsson F5521gw
 ATTR{idVendor}=="0bdb", ATTR{idProduct}=="190d", RUN+="usb_modeswitch '/%k'"
 
diff -ruN a/usb-modeswitch-data-20170120/usb_modeswitch.d/0bda:1a2b b/usb-modeswitch-data-20170120/usb_modeswitch.d/0bda:1a2b
--- a/usb-modeswitch-data-20170120/usb_modeswitch.d/0bda:1a2b	1970-01-01 08:00:00.000000000 +0800
+++ b/usb-modeswitch-data-20170120/usb_modeswitch.d/0bda:1a2b	2025-05-27 06:00:23.611517886 +0800
@@ -0,0 +1,4 @@
+# Realtek Semiconductor Corp. RTL8188GU 802.11n WLAN Adapter
+TargetVendor=0x0bda
+TargetProduct=0xb711
+StandardEject=1
