diff --git a/assets/packaging/micro.desktop b/assets/packaging/micro.desktop
index 35772518..3066f82e 100644
--- a/assets/packaging/micro.desktop
+++ b/assets/packaging/micro.desktop
@@ -12,5 +12,4 @@ Keywords=text;editor;syntax;terminal;
 Exec=micro %F
 StartupNotify=false
 Terminal=true
-NoDisplay=true
 MimeType=text/plain;text/x-chdr;text/x-csrc;text/x-c++hdr;text/x-c++src;text/x-java;text/x-dsrc;text/x-pascal;text/x-perl;text/x-python;application/x-php;application/x-httpd-php3;application/x-httpd-php4;application/x-httpd-php5;application/xml;text/html;text/css;text/x-sql;text/x-diff;
