diff -Naur AppStream-0.12.9/meson.build AppStream-0.12.9.gcc8/meson.build
--- AppStream-0.12.9/meson.build	2019-09-24 13:19:49.000000000 -0500
+++ AppStream-0.12.9.gcc8/meson.build	2019-11-09 13:35:22.510115190 -0600
@@ -54,7 +54,7 @@
 
 # a few compiler warning flags we always want enabled
 add_global_arguments('-Werror=implicit-function-declaration', '-Wno-unused-parameter', language: 'c')
-add_global_arguments('-Wno-unused-parameter', '-Wno-error=deprecated-copy', language: 'cpp')
+add_global_arguments('-Wno-unused-parameter', language: 'cpp')
 add_global_arguments('-DAS_COMPILATION', language : 'c')
 
 #
