From f585cc28c9ee67ec8a9290eb958d18d1a1db4563 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Wed, 21 Feb 2024 08:54:36 +0800
Subject: [PATCH] Add loongarch support

---
 avs_core/include/avs/config.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/avs_core/include/avs/config.h b/avs_core/include/avs/config.h
index bdabf17f..34ebfd83 100644
--- a/avs_core/include/avs/config.h
+++ b/avs_core/include/avs/config.h
@@ -63,6 +63,8 @@
 #   define SPARC
 #elif defined(__mips__)
 #   define MIPS
+#elif defined(__loongarch__)
+#   define LOONGARCH
 #else
 #   error Unsupported CPU architecture.
 #endif
-- 
2.43.0

