diff -Naur node-10.10.0/deps/openssl/openssl/crypto/init.c node-10.10.0.openssl/deps/openssl/openssl/crypto/init.c
--- node-10.10.0/deps/openssl/openssl/crypto/init.c	2018-09-06 20:10:39.000000000 +0000
+++ node-10.10.0.openssl/deps/openssl/openssl/crypto/init.c	2018-10-10 16:11:25.998554930 +0000
@@ -118,7 +118,8 @@
 #ifdef OPENSSL_INIT_DEBUG
     fprintf(stderr, "OPENSSL_INIT: ossl_init_load_crypto_nodelete()\n");
 #endif
-#if !defined(OPENSSL_NO_DSO) && !defined(OPENSSL_USE_NODELETE)
+#if defined(OPENSSL_NO_STATIC_ENGINE) && \
+    !defined(OPENSSL_NO_DSO) && !defined(OPENSSL_USE_NODELETE)
 # ifdef DSO_WIN32
     {
         HMODULE handle = NULL;
@@ -663,7 +664,8 @@
 {
     OPENSSL_INIT_STOP *newhand;
 
-#if !defined(OPENSSL_NO_DSO) && !defined(OPENSSL_USE_NODELETE)
+#if defined(OPENSSL_NO_STATIC_ENGINE) && \
+    !defined(OPENSSL_NO_DSO) && !defined(OPENSSL_USE_NODELETE)
     {
         union {
             void *sym;
