Description: on mipsel, mips64el, node --jitless segfaults
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2020-06-03
Forwarded: https://github.com/nodejs/node/issues/33703
--- a/test/parallel/test-cli-node-options.js
+++ b/test/parallel/test-cli-node-options.js
@@ -68,7 +68,8 @@
 expect('--abort_on-uncaught_exception', 'B\n');
 expect('--disallow-code-generation-from-strings', 'B\n');
 expect('--huge-max-old-generation-size', 'B\n');
-expect('--jitless', 'B\n');
+if (!['mipsel', 'mips64el'].includes(process.arch))
+  expect('--jitless', 'B\n');
 expect('--max-old-space-size=0', 'B\n');
 expect('--stack-trace-limit=100',
        /(\s*at f \(\[(eval|worker eval)\]:1:\d*\)\r?\n)/,
