diff -Naur a/src/calibre/gui2/ui.py b/src/calibre/gui2/ui.py
--- a/src/calibre/gui2/ui.py	2016-09-15 21:04:34.000000000 -0500
+++ b/src/calibre/gui2/ui.py	2016-09-18 11:10:06.886055982 -0500
@@ -334,9 +334,6 @@
         self.location_manager.update_device_metadata.connect(self.update_metadata_on_device)
         self.eject_action.triggered.connect(self.device_manager.umount_device)
 
-        # ################### Update notification ###################
-        UpdateMixin.init_update_mixin(self, opts)
-
         # ###################### Search boxes ########################
         SearchRestrictionMixin.init_search_restriction_mixin(self)
         SavedSearchBoxMixin.init_saved_seach_box_mixin(self)
