diff --git a/src/gallium/targets/dri/meson.build b/src/gallium/targets/dri/meson.build
index 7640ec1216d..3930c6f2d1d 100644
--- a/src/gallium/targets/dri/meson.build
+++ b/src/gallium/targets/dri/meson.build
@@ -115,7 +115,10 @@ foreach d : [[with_gallium_kmsro, [
              [with_gallium_zink, 'zink_dri.so'],
              [with_gallium_d3d12, 'd3d12_dri.so'],
              [with_gallium_asahi, 'asahi_dri.so'],
-             [with_gallium_pvr, 'pvr_dri.so'],
+             [with_gallium_pvr, [
+               'pvr_dri.so',
+               'starfive_dri.so',
+             ]],
              [with_gallium_pvr_alias, gallium_pvr_alias + '_dri.so']]
   if d[0]
     gallium_dri_drivers += d[1]
diff --git a/src/gallium/targets/dri/target.c b/src/gallium/targets/dri/target.c
index 1e660ecb080..359c9b4f285 100644
--- a/src/gallium/targets/dri/target.c
+++ b/src/gallium/targets/dri/target.c
@@ -149,6 +149,7 @@ DEFINE_LOADER_DRM_ENTRYPOINT(d3d12);
 
 #if defined(GALLIUM_PVR)
 DEFINE_LOADER_PVR_ENTRYPOINT(pvr);
+DEFINE_LOADER_PVR_ENTRYPOINT(starfive);
 #endif
 
 #if defined(GALLIUM_PVR_ALIAS)
