From 019542ff5e9b9e101738350c4a7b5e43fd9ceb36 Mon Sep 17 00:00:00 2001
From: Laurent Montel <montel@kde.org>
Date: Sun, 13 Oct 2019 14:48:24 +0200
Subject: [PATCH 10/12] Now we depend against kf5.58

---
 CMakeLists.txt | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c04a052..0d64f0c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,10 +45,6 @@ add_subdirectory(plasmoid)
 add_subdirectory(tests)
 add_subdirectory(console)
 
-if (${ECM_VERSION} STRGREATER "5.58.0")
-    install(FILES kscreen.categories  DESTINATION  ${KDE_INSTALL_LOGGINGCATEGORIESDIR})
-else()
-    install(FILES kscreen.categories  DESTINATION ${KDE_INSTALL_CONFDIR})
-endif()
+install(FILES kscreen.categories  DESTINATION  ${KDE_INSTALL_LOGGINGCATEGORIESDIR})
 
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
-- 
2.21.0

