diff -Naur musicbox-0.2.4.3/NEMbox/menu.py musicbox-0.2.4.3.noupdate/NEMbox/menu.py
--- musicbox-0.2.4.3/NEMbox/menu.py	2018-05-21 17:21:34.000000000 +0800
+++ musicbox-0.2.4.3.noupdate/NEMbox/menu.py	2018-12-26 16:33:51.683706341 +0800
@@ -139,14 +139,6 @@
         curses.endwin()
         sys.exit()
 
-    def update_alert(self, version):
-        latest = Menu().check_version()
-        if latest != version and latest != 0:
-            notify('MusicBox Update is available', 1)
-            time.sleep(0.5)
-            notify('NetEase-MusicBox installed version:' + version +
-                   '\nNetEase-MusicBox latest version:' + latest, 0)
-
     def check_version(self):
         # 检查更新 && 签到
         try:
@@ -165,10 +157,7 @@
 
     def start_fork(self, version):
         pid = os.fork()
-        if pid == 0:
-            Menu().update_alert(version)
-        else:
-            Menu().start()
+        Menu().start()
 
     def _is_playlist_empty(self):
         return len(self.storage.database['player_info']['player_list']) == 0
