--- xreader-3.4.4/meson.build	2022-08-13 10:29:42.000000000 +0000
+++ xreader-3.4.4.webkit2/meson.build	2022-08-16 04:20:26.950713831 +0000
@@ -111,7 +111,7 @@
 thumbnailer_mime_types = xreader_mime_types
 
 if get_option('epub')
-    webkit = dependency('webkit2gtk-4.0', version: '>= 2.4.3')
+    webkit = dependency('webkit2gtk-4.1', version: '>= 2.4.3')
     backend_subdirs += 'epub'
     xreader_mime_types += 'application/epub+zip;'
 endif
