From 95a443e0ae7da69e9165856f9405fd9db471f50b Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Thu, 18 Dec 2025 15:38:40 +0800
Subject: [PATCH] AOSCOS: include/f2fs_fs.h: avoid defining bool for c23

---
 include/f2fs_fs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/f2fs_fs.h b/include/f2fs_fs.h
index f890634..1365de4 100644
--- a/include/f2fs_fs.h
+++ b/include/f2fs_fs.h
@@ -103,7 +103,7 @@ typedef uint16_t	u16;
 typedef uint8_t		u8;
 typedef u32		block_t;
 typedef u32		nid_t;
-#ifndef bool
+#if !defined(bool) && (__STDC_VERSION__ < 202311)
 typedef u8		bool;
 #endif
 typedef unsigned long	pgoff_t;
-- 
2.52.0

