From b434d6e43b173ed2d3145b30119d1974287aff88 Mon Sep 17 00:00:00 2001
From: intrigeri <intrigeri@boum.org>
Date: Fri, 28 Dec 2018 16:20:24 +0000
Subject: [PATCH 1/5] Enable writing cache.

Forwarded: not-needed
---
 parser/rc.apparmor.functions | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parser/rc.apparmor.functions b/parser/rc.apparmor.functions
index 2e017e601..489b8e49f 100644
--- a/parser/rc.apparmor.functions
+++ b/parser/rc.apparmor.functions
@@ -32,7 +32,7 @@
 # Some nice defines that we use
 
 PARSER=/sbin/apparmor_parser
-PARSER_OPTS=
+PARSER_OPTS=--write-cache
 # Suppress warnings when booting in quiet mode
 if [ "${QUIET:-no}" = yes ] || [ "${quiet:-n}" = y ]; then
 	PARSER_OPTS="$PARSER_OPTS --quiet"
-- 
2.50.1

