From: John Johansen <john.johansen@canonical.com>
Date: Mon, 21 Nov 2022 17:09:28 -0800
Subject: libapparmor: fix expect library version to bump in 3.0.8 release

Signed-off-by: John Johansen <john.johansen@canonical.com>
---
 libraries/libapparmor/src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libraries/libapparmor/src/Makefile.am b/libraries/libapparmor/src/Makefile.am
index 4511bba..ef8a45e 100644
--- a/libraries/libapparmor/src/Makefile.am
+++ b/libraries/libapparmor/src/Makefile.am
@@ -29,7 +29,7 @@ INCLUDES = $(all_includes)
 AA_LIB_CURRENT = 9
 AA_LIB_REVISION = 4
 AA_LIB_AGE = 8
-EXPECTED_SO_NAME = libapparmor.so.1.8.3
+EXPECTED_SO_NAME = libapparmor.so.1.8.4
 
 SUFFIXES = .pc.in .pc
 
