From a96074bd1bc2cb1d5795b242595616cc0d4718e4 Mon Sep 17 00:00:00 2001
From: Jan Grulich <jgrulich@redhat.com>
Date: Wed, 1 Aug 2018 09:31:24 +0200
Subject: Search also for libpipewire-0.2


diff --git a/cmake/modules/FindPipeWire.cmake b/cmake/modules/FindPipeWire.cmake
index d267e23..4a85b0a 100644
--- a/cmake/modules/FindPipeWire.cmake
+++ b/cmake/modules/FindPipeWire.cmake
@@ -61,7 +61,7 @@
 # Use pkg-config to get the directories and then use these values
 # in the FIND_PATH() and FIND_LIBRARY() calls
 find_package(PkgConfig QUIET)
-pkg_check_modules(PKG_PipeWire QUIET libpipewire-0.1)
+pkg_check_modules(PKG_PipeWire QUIET libpipewire-0.1 libpipewire-0.2)
 
 set(PipeWire_DEFINITIONS "${PKG_PipeWire_CFLAGS_OTHER}")
 set(PipeWire_VERSION "${PKG_PipeWire_VERSION}")
@@ -75,7 +75,7 @@ find_path(PipeWire_INCLUDE_DIRS
 
 find_library(PipeWire_LIBRARIES
     NAMES
-        pipewire-0.1
+        pipewire-0.1 pipewire-0.2
     HINTS
         ${PKG_PipeWire_LIBRARIES_DIRS}
 )
