From 8125ae8b6bfddaa6a7e929406ca260558a4f7ddf Mon Sep 17 00:00:00 2001
From: Slava Aseev <ptrnine@altlinux.org>
Date: Thu, 12 May 2022 12:02:51 +0300
Subject: [PATCH 3/3] Do not try to rename label twice in entryRenamed()

---
 src/runtime/kwalletd/kwalletfreedesktopservice.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/runtime/kwalletd/kwalletfreedesktopservice.cpp b/src/runtime/kwalletd/kwalletfreedesktopservice.cpp
index 17c66ea..ed6421f 100644
--- a/src/runtime/kwalletd/kwalletfreedesktopservice.cpp
+++ b/src/runtime/kwalletd/kwalletfreedesktopservice.cpp
@@ -7,6 +7,7 @@
 #include "kwalletfreedesktopservice.h"
 
 #include "kwalletd.h"
+#include "kwalletd_debug.h"
 #include "kwalletfreedesktopcollection.h"
 #include "kwalletfreedesktopitem.h"
 #include "kwalletfreedesktopprompt.h"
@@ -510,7 +511,11 @@ void KWalletFreedesktopService::entryRenamed(const QString &walletName, const QS
 
     auto *item = collection->findItemByEntryLocation(oldLocation);
     if (!item) {
-        item = collection->findItemByEntryLocation(newLocation);
+        /* Warn if label not found and not yet renamed */
+        if (!collection->findItemByEntryLocation(newLocation)) {
+            qCWarning(KWALLETD_LOG) << "Cannot rename secret service label:" << FdoUniqueLabel::fromEntryLocation(oldLocation).label;
+        }
+        return;
     }
 
     if (item) {
-- 
2.36.0

