From 4cb55eecaf939f8cc66f44d70684f9f1ed5a61a3 Mon Sep 17 00:00:00 2001
From: stydxm <stydxm@outlook.com>
Date: Tue, 2 Sep 2025 12:11:05 +0800
Subject: [PATCH] Update glog import logic for glog 0.7.1

---
 CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8b3a75e..f0f03b7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,8 +50,7 @@ set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake"
   "${CMAKE_CURRENT_SOURCE_DIR}/build/fbcode_builder/CMake"
   ${CMAKE_MODULE_PATH})
 
-find_package(Glog MODULE REQUIRED)
-include_directories(${GLOG_INCLUDE_DIR})
+find_package(glog CONFIG QUIET)
 
 find_package(Gflags REQUIRED)
 include_directories(${GFLAGS_INCLUDE_DIR})
-- 
2.52.0

