From a983ff617eb9d074ed7e5138bdd31eaec7a0828e Mon Sep 17 00:00:00 2001
From: Eric Berquist <eric.berquist@gmail.com>
Date: Mon, 10 Oct 2016 11:15:20 -0400
Subject: [PATCH] Fix for GCC 6 compatibility.

---
 cmake/GenerateExportHeader.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/GenerateExportHeader.cmake b/cmake/GenerateExportHeader.cmake
index a91da06..3dd14ed 100644
--- a/cmake/GenerateExportHeader.cmake
+++ b/cmake/GenerateExportHeader.cmake
@@ -164,12 +164,12 @@ macro(_test_compiler_hidden_visibility)
   if(CMAKE_COMPILER_IS_GNUCXX)
     exec_program(${CMAKE_C_COMPILER} ARGS --version
       OUTPUT_VARIABLE _gcc_version_info)
-    string(REGEX MATCH "[345]\\.[0-9]\\.[0-9]"
+    string(REGEX MATCH "[3456]\\.[0-9]\\.[0-9]"
       _gcc_version "${_gcc_version_info}")
     # gcc on mac just reports: "gcc (GCC) 3.3 20030304 ..." without the
     # patch level, handle this here:
     if(NOT _gcc_version)
-      string(REGEX REPLACE ".*\\(GCC\\).* ([34]\\.[0-9]) .*" "\\1.0"
+      string(REGEX REPLACE ".*\\(GCC\\).* ([3456]\\.[0-9]) .*" "\\1.0"
         _gcc_version "${_gcc_version_info}")
     endif()
 
