--- onscripter-jh/Makefile.Linux	2019-04-16 22:27:36.049027727 -0500
+++ onscripter-jh.noavi/Makefile.Linux	2019-04-16 22:27:19.845568851 -0500
@@ -41,13 +41,6 @@
 # optional: support CD audio
 DEFS += -DUSE_CDROM
 
-# optional: avifile
-DEFS += -DUSE_AVIFILE
-INCS += `avifile-config --cflags`
-LIBS += `avifile-config --libs`
-TARGET += simple_aviplay$(EXESUFFIX)
-EXT_OBJS += AVIWrapper$(OBJSUFFIX)
-
 # optional: lua
 DEFS += -DUSE_LUA
 INCS += -I/usr/include/lua5.1
