diff -Naur zita-alsa-pcmi-0.3.2/source/Makefile zita-alsa-pcmi-0.3.2.nonative/source/Makefile
--- zita-alsa-pcmi-0.3.2/source/Makefile        2018-08-24 13:41:30.000000000 -0500
+++ zita-alsa-pcmi-0.3.2.nonative/source/Makefile       2019-06-26 22:01:31.604363512 -0500
@@ -31,7 +31,6 @@
 
 CPPFLAGS += -DVERSION=\"$(VERSION)\" -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
 CXXFLAGS += -O2 -Wall -fPIC
-CXXFLAGS += -march=native
 LDLFAGS +=
 LDLIBS +=
 

