--- a/libs/Makefile	2012-03-20 14:23:07.000000000 -0600
+++ b/libs/Makefile	2017-10-24 16:19:50.681046080 -0600
@@ -52,7 +52,7 @@
 	install -Dm 644 $(ZITA-ALSA-PCMI_H)   $(DESTDIR)$(PREFIX)/include/$(ZITA-ALSA-PCMI_H)
 	install -Dm 644 $(ZITA-ALSA-PCMI_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(ZITA-ALSA-PCMI_MIN)
 	ln -sf $(ZITA-ALSA-PCMI_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(ZITA-ALSA-PCMI_SO)
-	ldconfig
+	ldconfig -N $(DESTDIR)$(PREFIX)/$(LIBDIR)
 
 uninstall:
 	rm -rf $(DESTDIR)$(PREFIX)/include/zita-alsa-pcmi.h
