Author: Michael Karcher <debian@mkarcher.dialup.fu-berlin.de>
Description: Fixes a possible security issue by initialising a local variable.

--- unace-nonfree-2.5.orig/source/base/all/dcpr/comments/comments.c
+++ unace-nonfree-2.5/source/base/all/dcpr/comments/comments.c
@@ -62,6 +62,8 @@ INT       CommentPos,
 
   if (BASE_COMMENTS.CompressedSize)
   {
+    MatchPos = 0;
+  
     CommentPos = 0;
     memset(&Hash, 0, sizeof(Hash));
     BASE_DCPR_COMMENTS_Init();
