From c2e01c45a2cdd77a94872a3e9e4ea8ac29f96170 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Fri, 20 Jun 2025 10:35:18 -0700
Subject: [PATCH 2/2] Fix build with Qt 5

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 gui/pinyindictmanager/pinyindictmanager.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gui/pinyindictmanager/pinyindictmanager.cpp b/gui/pinyindictmanager/pinyindictmanager.cpp
index 85d5678..cb876dd 100644
--- a/gui/pinyindictmanager/pinyindictmanager.cpp
+++ b/gui/pinyindictmanager/pinyindictmanager.cpp
@@ -248,7 +248,7 @@ void PinyinDictManager::importFromSogou() {
 
     auto tempFile = prepareTempFile(fullname + "_XXXXXX");
 
-    QDir runtimeDir(runtimeDirectory);
+    QDir runtimeDir(QString::fromStdWString(runtimeDirectory.wstring()));
     auto txtFile = prepareTempFile(runtimeDir.filePath("scel_txt_XXXXXX"));
     if (tempFile.isEmpty() || txtFile.isEmpty()) {
         if (!tempFile.isEmpty()) {
@@ -356,7 +356,7 @@ void PinyinDictManager::removeAllDict() {
             index.data(Qt::UserRole).toString().toLocal8Bit().constData();
         auto fullPath = StandardPaths::global().locate(
             StandardPathsType::PkgData, "pinyin/dictionaries/" + fileName);
-        QFile::remove(fullPath);
+        QFile::remove(QString::fromStdWString(fullPath.wstring()));
     }
     reload();
 }
@@ -379,7 +379,7 @@ void PinyinDictManager::removeDict() {
         QMessageBox::Ok | QMessageBox::Cancel);
 
     if (ret == QMessageBox::Ok) {
-        bool ok = QFile::remove(fullPath);
+        bool ok = QFile::remove(QString::fromStdWString(fullPath.wstring()));
         if (!ok) {
             QMessageBox::warning(
                 this, _("File Operation Failed"),
-- 
2.49.0

