diff -Naur curl-7.68.0/src/tool_cb_hdr.c curl-7.68.0.8177/src/tool_cb_hdr.c
--- curl-7.68.0/src/tool_cb_hdr.c	2020-01-04 15:41:58.000000000 -0600
+++ curl-7.68.0.8177/src/tool_cb_hdr.c	2020-08-26 01:34:49.635769944 -0500
@@ -188,25 +188,11 @@
       filename = parse_filename(p, len);
       if(filename) {
         if(outs->stream) {
-          int rc;
-          /* already opened and possibly written to */
-          if(outs->fopened)
-            fclose(outs->stream);
-          outs->stream = NULL;
-
-          /* rename the initial file name to the new file name */
-          rc = rename(outs->filename, filename);
-          if(rc != 0) {
-            warnf(outs->config->global, "Failed to rename %s -> %s: %s\n",
-                  outs->filename, filename, strerror(errno));
-          }
-          if(outs->alloc_filename)
-            Curl_safefree(outs->filename);
-          if(rc != 0) {
-            free(filename);
-            return failure;
-          }
+          /* indication of problem, get out! */
+          free(filename);
+          return failure;
         }
+
         outs->is_cd_filename = TRUE;
         outs->s_isreg = TRUE;
         outs->fopened = FALSE;
diff -Naur curl-7.68.0/src/tool_getparam.c curl-7.68.0.8177/src/tool_getparam.c
--- curl-7.68.0/src/tool_getparam.c	2020-01-06 12:53:35.000000000 -0600
+++ curl-7.68.0.8177/src/tool_getparam.c	2020-08-26 01:33:16.229286973 -0500
@@ -1800,6 +1800,11 @@
       }
       break;
     case 'i':
+      if(config->content_disposition) {
+        warnf(global,
+              "--include and --remote-header-name cannot be combined.\n");
+        return PARAM_BAD_USE;
+      }
       config->show_headers = toggle; /* show the headers as well in the
                                         general output stream */
       break;
