diff -ur NVIDIA-Linux-x86_64-367.35/kernel/nvidia-drm/nvidia-drm-fb.c b/kernel/nvidia-drm/nvidia-drm-fb.c
--- NVIDIA-Linux-x86_64-367.27/kernel/nvidia-drm/nvidia-drm-fb.c	2016-06-10 02:38:43.000000000 +0200
+++ b/kernel/nvidia-drm/nvidia-drm-fb.c	2016-06-14 02:45:44.263506669 +0200
@@ -114,7 +114,7 @@
      * We don't support any planar format, pick up first buffer only.
      */
 
-    gem = drm_gem_object_lookup(dev, file, cmd->handles[0]);
+    gem = drm_gem_object_lookup(file, cmd->handles[0]);
 
     if (gem == NULL)
     {
diff -ur NVIDIA-Linux-x86_64-367.27/kernel/nvidia-drm/nvidia-drm-gem.c b/kernel/nvidia-drm/nvidia-drm-gem.c
--- NVIDIA-Linux-x86_64-367.27/kernel/nvidia-drm/nvidia-drm-gem.c	2016-06-10 02:38:43.000000000 +0200
+++ b/kernel/nvidia-drm/nvidia-drm-gem.c	2016-06-14 02:45:44.263506669 +0200
@@ -408,7 +408,7 @@
 
     mutex_lock(&dev->struct_mutex);
 
-    gem = drm_gem_object_lookup(dev, file, handle);
+    gem = drm_gem_object_lookup(file, handle);
 
     if (gem == NULL)
     {
diff -ur NVIDIA-Linux-x86_64-367.27/kernel/nvidia-uvm/uvm_linux.h b/kernel/nvidia-uvm/uvm_linux.h
--- NVIDIA-Linux-x86_64-367.27/kernel/nvidia-uvm/uvm_linux.h	2016-06-10 02:37:08.000000000 +0200
+++ b/kernel/nvidia-uvm/uvm_linux.h	2016-06-14 02:49:35.214495381 +0200
@@ -563,12 +563,13 @@
     INIT_RADIX_TREE(tree, GFP_NOWAIT);
 }
 
+/*
 static bool radix_tree_empty(struct radix_tree_root *tree)
 {
     void *dummy;
     return radix_tree_gang_lookup(tree, &dummy, 0, 1) == 0;
 }
-
+*/
 
 #if !defined(NV_USLEEP_RANGE_PRESENT)
 static void __sched usleep_range(unsigned long min, unsigned long max)
