From da141349a50483e4822e9cf0872d8ac91551f4dc Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Fri, 19 Mar 2021 18:25:30 -0700
Subject: [PATCH] Fix missing include

---
 Telegram/ThirdParty/tgcalls/tgcalls/group/StreamingPart.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Telegram/ThirdParty/tgcalls/tgcalls/group/StreamingPart.h b/Telegram/ThirdParty/tgcalls/tgcalls/group/StreamingPart.h
index 08859d6..1228cff 100644
--- a/Telegram/ThirdParty/tgcalls/tgcalls/group/StreamingPart.h
+++ b/Telegram/ThirdParty/tgcalls/tgcalls/group/StreamingPart.h
@@ -4,6 +4,8 @@
 #include "absl/types/optional.h"
 #include <vector>
 
+#include <stdint.h>
+
 namespace tgcalls {
 
 class StreamingPartState;

diff --git a/webview/platform/linux/webview_linux_webkit_gtk.h b/webview/platform/linux/webview_linux_webkit_gtk.h
index a7f0f97..7e9fe26 100644
--- a/Telegram/lib_webview/webview/platform/linux/webview_linux_webkit_gtk.h
+++ b/Telegram/lib_webview/webview/platform/linux/webview_linux_webkit_gtk.h
@@ -12,6 +12,7 @@ extern "C" {
 #undef signals
 #include <JavaScriptCore/JavaScript.h>
 #include <gtk/gtk.h>
+#include <gdk/gdkx.h>
 #include <webkit2/webkit2.h>
 #include <X11/Xlib.h>
 #define signals public

-- 
2.30.2

