diff -Naur tdesktop-2.9.8-full/Telegram/SourceFiles/api/api_media.cpp tdesktop-2.9.8-full-new/Telegram/SourceFiles/api/api_media.cpp
--- tdesktop-2.9.8-full/Telegram/SourceFiles/api/api_media.cpp	2021-08-23 08:30:14.000000000 -0700
+++ tdesktop-2.9.8-full-new/Telegram/SourceFiles/api/api_media.cpp	2021-08-25 03:46:39.237261572 -0700
@@ -83,7 +83,7 @@
 	return MTP_inputMediaUploadedPhoto(
 		MTP_flags(flags),
 		file,
-		MTP_vector<MTPInputDocument>(ranges::to<QVector>(attachedStickers)),
+		MTP_vector<MTPInputDocument>(QVector<MTPInputDocument>::fromStdVector(attachedStickers)),
 		MTP_int(0));
 }
 
@@ -108,7 +108,7 @@
 		thumb.value_or(MTPInputFile()),
 		MTP_string(document->mimeString()),
 		ComposeSendingDocumentAttributes(document),
-		MTP_vector<MTPInputDocument>(ranges::to<QVector>(attachedStickers)),
+		MTP_vector<MTPInputDocument>(QVector<MTPInputDocument>::fromStdVector(attachedStickers)),
 		MTP_int(0));
 }
 
