--- a/cmake/external/webrtc/CMakeLists.txt	2020-08-20 08:59:10.000000000 -0500
+++ b/cmake/external/webrtc/CMakeLists.txt	2020-08-27 00:45:52.559408181 -0500
@@ -4,18 +4,11 @@
 # For license and copyright information please follow this link:
 # https://github.com/desktop-app/legal/blob/master/LEGAL
 
-add_library(external_webrtc INTERFACE IMPORTED GLOBAL)
-add_library(desktop-app::external_webrtc ALIAS external_webrtc)
+add_subdirectory(${DESKTOP_APP_WEBRTC_LOCATION} "${CMAKE_CURRENT_BINARY_DIR}/tg_owt" EXCLUDE_FROM_ALL)
+add_library(external_webrtc ALIAS tg_owt)
+add_library(desktop-app::external_webrtc ALIAS tg_owt)
 
-if (DESKTOP_APP_WEBRTC_LOCATION)
-    set(webrtc_loc ${DESKTOP_APP_WEBRTC_LOCATION})
-else()
-    set(webrtc_loc ${libs_loc}/tg_owt/src)
-endif()
-
-set(webrtc_build_loc ${libs_loc}/tg_owt/out/$<CONFIG>)
-
-target_compile_definitions(external_webrtc
+target_compile_definitions(tg_owt
 INTERFACE
     WEBRTC_ENABLE_PROTOBUF=0
     WEBRTC_INCLUDE_INTERNAL_AUDIO_DEVICE
@@ -28,44 +21,35 @@
 )
 
 if (WIN32)
-    set(webrtc_lib_prefix "")
-    set(webrtc_lib_suffix .lib)
-
-    target_compile_definitions(external_webrtc
+    target_compile_definitions(tg_owt
     INTERFACE
         WEBRTC_WIN
     )
 else()
-    set(webrtc_lib_prefix lib)
-    set(webrtc_lib_suffix .a)
-
     if (APPLE)
-        target_compile_definitions(external_webrtc
+        target_compile_definitions(tg_owt
         INTERFACE
             WEBRTC_MAC
         )
     else()
-        target_compile_definitions(external_webrtc
+        target_compile_definitions(tg_owt
         INTERFACE
             WEBRTC_LINUX
         )
     endif()
 
-    target_compile_definitions(external_webrtc
+    target_compile_definitions(tg_owt
     INTERFACE
         WEBRTC_POSIX
     )
 endif()
 
-target_include_directories(external_webrtc
+target_include_directories(tg_owt
 INTERFACE
     ${webrtc_loc}
     ${webrtc_loc}/third_party/abseil-cpp
 )
 
-set(webrtc_libs
-    ${webrtc_lib_prefix}tg_owt
-)
 if (APPLE)
     target_include_directories(external_webrtc
     INTERFACE
@@ -75,12 +59,9 @@
         ${webrtc_loc}/third_party/libyuv/include
     )
 endif()
-foreach (lib ${webrtc_libs})
-    list(APPEND webrtc_libs_list "${webrtc_build_loc}/${lib}${webrtc_lib_suffix}")
-endforeach()
 
 if (WIN32)
-    target_link_libraries(external_webrtc
+    target_link_libraries(tg_owt
     INTERFACE
         Secur32.lib # Required for rtc_base/http_common.cc
         dmoguids.lib # Required for the built-in WASAPI AEC, see modules/audio_device/BUILD.gn
@@ -89,14 +70,13 @@
         msdmo.lib
     )
 elseif (APPLE)
-    target_link_libraries(external_webrtc
+    target_link_libraries(tg_owt
     INTERFACE
         -ObjC
     )
 endif()
 
-target_link_libraries(external_webrtc
+target_link_libraries(tg_owt
 INTERFACE
-    ${webrtc_libs_list}
-    desktop-app::external_openssl
+    desktop-app::external_openssl jpeg
 )
