diff --git a/base/platform/linux/base_linux_gtk_integration.cpp b/base/platform/linux/base_linux_gtk_integration.cpp
index bafd075..b9242f6 100644
--- a/Telegram/lib_base/base/platform/linux/base_linux_gtk_integration.cpp
+++ b/Telegram/lib_base/base/platform/linux/base_linux_gtk_integration.cpp
@@ -203,7 +203,8 @@ void GtkIntegration::prepareEnvironment() {
 #ifdef DESKTOP_APP_USE_PACKAGED // static binary doesn't contain qgtk3/qgtk2
 	// if gtk integration and qgtk3/qgtk2 platformtheme (or qgtk2 style)
 	// is used at the same time, the app will crash
-	if (!qEnvironmentVariableIsSet(
+	if (qEnvironmentVariable("XDG_SESSION_DESKTOP") != "KDE" &&
+                !qEnvironmentVariableIsSet(
 			kIgnoreGtkIncompatibility.utf8().constData())) {
 		g_warning(
 			"Unfortunately, GTK integration "
