diff -Naur tdesktop-2.9.8-full/Telegram/SourceFiles/api/api_media.cpp tdesktop-2.9.8-full-new/Telegram/SourceFiles/api/api_media.cpp
--- tdesktop-2.9.8-full/Telegram/SourceFiles/api/api_media.cpp	2021-08-23 08:30:14.000000000 -0700
+++ tdesktop-2.9.8-full-new/Telegram/SourceFiles/api/api_media.cpp	2021-08-25 03:46:39.237261572 -0700
@@ -83,7 +83,7 @@
 	return MTP_inputMediaUploadedPhoto(
 		MTP_flags(flags),
 		file,
-		MTP_vector<MTPInputDocument>(ranges::to<QVector>(attachedStickers)),
+		MTP_vector<MTPInputDocument>(QVector<MTPInputDocument>::fromStdVector(attachedStickers)),
 		MTP_int(0));
 }
 
@@ -108,7 +108,7 @@
 		thumb.value_or(MTPInputFile()),
 		MTP_string(document->mimeString()),
 		ComposeSendingDocumentAttributes(document),
-		MTP_vector<MTPInputDocument>(ranges::to<QVector>(attachedStickers)),
+		MTP_vector<MTPInputDocument>(QVector<MTPInputDocument>::fromStdVector(attachedStickers)),
 		MTP_int(0));
 }
 
diff -Naur tdesktop-2.9.8-full/Telegram/SourceFiles/media/streaming/media_streaming_video_track.cpp tdesktop-2.9.8-full-new/Telegram/SourceFiles/media/streaming/media_streaming_video_track.cpp
--- tdesktop-2.9.8-full/Telegram/SourceFiles/media/streaming/media_streaming_video_track.cpp	2021-08-23 08:30:14.000000000 -0700
+++ tdesktop-2.9.8-full-new/Telegram/SourceFiles/media/streaming/media_streaming_video_track.cpp	2021-08-25 12:59:35.208249381 -0700
@@ -14,9 +14,11 @@
 
 #include "zlib.h"
 
+#ifdef _MSC_VER
 extern "C" {
 extern int __isa_available;
 }
+#endif
 
 #define TO_LOG(x) debugLog(QString x)
 
@@ -212,7 +214,13 @@
 		).arg(options.waitForMarkAsShown ? "true" : "false"
 		).arg(_stream.duration
 		).arg(_shared->initialized() ? "true" : "false"
-		).arg(__isa_available));
+		).arg(
+#ifdef _MSC_VER
+__isa_available
+#else
+"invalid"
+#endif
+                ));
 }
 
 rpl::producer<> VideoTrackObject::checkNextFrame() const {
diff -Naur tdesktop-2.9.8-full/Telegram/lib_webview/webview/platform/linux/webview_linux_webkit2gtk.cpp tdesktop-2.9.8-full-new/Telegram/lib_webview/webview/platform/linux/webview_linux_webkit2gtk.cpp
--- tdesktop-2.9.8-full/Telegram/lib_webview/webview/platform/linux/webview_linux_webkit2gtk.cpp	2021-08-10 02:07:04.000000000 -0700
+++ tdesktop-2.9.8-full-new/Telegram/lib_webview/webview/platform/linux/webview_linux_webkit2gtk.cpp	2021-08-25 03:00:31.689587199 -0700
@@ -10,9 +10,9 @@
 #error "WebKitGTK support depends on D-Bus integration."
 #endif // DESKTOP_APP_DISABLE_DBUS_INTEGRATION
 
+#include "base/platform/linux/base_linux_dbus_utilities.h"
 #include "webview/platform/linux/webview_linux_webkit_gtk.h"
 #include "base/platform/linux/base_linux_glibmm_helper.h"
-#include "base/platform/linux/base_linux_dbus_utilities.h"
 #include "base/platform/base_platform_info.h"
 #include "base/const_string.h"
 #include "base/integration.h"
