--- a/Telegram/SourceFiles/core/core_settings.h	2020-08-27 02:55:36.557601966 -0500
+++ b/Telegram/SourceFiles/core/core_settings.h	2020-08-27 02:56:01.425080202 -0500
@@ -533,7 +533,7 @@
 	rpl::variable<float64> _dialogsWidthRatio; // per-window
 	rpl::variable<int> _thirdColumnWidth = kDefaultThirdColumnWidth; // p-w
 	bool _notifyFromAll = true;
-	rpl::variable<bool> _nativeWindowFrame = false;
+	rpl::variable<bool> _nativeWindowFrame = true;
 	rpl::variable<std::optional<bool>> _systemDarkMode = std::nullopt;
 	rpl::variable<bool> _systemDarkModeEnabled = false;
 	rpl::variable<Window::ControlsLayout> _windowControlsLayout;
