diff -Naur tdesktop-1.9.3-full/Telegram/lib_spellcheck/spellcheck/platform/linux/linux_enchant.cpp tdesktop-1.9.3-full.noenchant1/Telegram/lib_spellcheck/spellcheck/platform/linux/linux_enchant.cpp
--- tdesktop-1.9.3-full/Telegram/lib_spellcheck/spellcheck/platform/linux/linux_enchant.cpp	2019-12-31 15:06:06.000000000 +0000
+++ tdesktop-1.9.3-full.noenchant1/Telegram/lib_spellcheck/spellcheck/platform/linux/linux_enchant.cpp	2020-01-07 08:31:04.341755847 +0000
@@ -217,9 +217,7 @@
 	static enum { NotLoadedYet, LoadSuccessful, LoadFailed = -1 } load_status;
 	if (load_status == NotLoadedYet) {
 		load_status = LoadFailed;
-		void * handle = dlopen ("libenchant.so.1", RTLD_NOW)
-				?: dlopen ("libenchant-2.so.2", RTLD_NOW)
-				?: dlopen ("libenchant.so.2", RTLD_NOW);
+		void * handle = dlopen ("libenchant-2.so.2", RTLD_NOW);
 		if (!handle) {
 			// logs ?
 			return false;
