diff --git a/Telegram/lib_base/base/build_config.h b/Telegram/lib_base/base/build_config.h
index 17316303..480096b8 100644
--- a/Telegram/lib_base/base/build_config.h
+++ b/Telegram/lib_base/base/build_config.h
@@ -60,8 +60,6 @@ Copyright (c) 2014-2017 John Preston, https://desktop.telegram.org
 #define ARCH_CPU_X86_FAMILY 1
 #define ARCH_CPU_X86 1
 #define ARCH_CPU_32_BITS 1
-#else
-#error Please add support for your architecture in base/build_config.h
 #endif
 
 #if defined(COMPILER_GCC) || defined(COMPILER_CLANG)
diff --git a/Telegram/cmake/lib_tgvoip.cmake b/Telegram/cmake/lib_tgvoip.cmake
index 38fa632a40..2801fbec2d 100644
--- a/Telegram/cmake/lib_tgvoip.cmake
+++ b/Telegram/cmake/lib_tgvoip.cmake
@@ -756,7 +756,7 @@ else()
         -Wno-error=sequence-point
         -Wno-error=unused-result
     )
-    if (build_linux32)
+    if (CMAKE_SYSTEM_PROCESSOR MATCHES "(i686|i386|x86)")
         target_compile_options(lib_tgvoip PRIVATE -msse2)
     endif()
     target_compile_definitions(lib_tgvoip
