diff -Naur telegram-desktop/Telegram/SourceFiles/base/build_config.h telegram-desktop.noarchdetect/Telegram/SourceFiles/base/build_config.h
--- telegram-desktop/Telegram/SourceFiles/base/build_config.h	2017-06-30 10:54:35.258096597 +0000
+++ telegram-desktop.noarchdetect/Telegram/SourceFiles/base/build_config.h	2017-06-30 10:55:07.998072956 +0000
@@ -60,8 +60,6 @@
 #define ARCH_CPU_X86_FAMILY 1
 #define ARCH_CPU_X86 1
 #define ARCH_CPU_32_BITS 1
-#else
-#error Please add support for your architecture in base/build_config.h
 #endif
 
 #if defined(COMPILER_GCC) || defined(COMPILER_CLANG)
