diff -Naur telegram-desktop/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp telegram-desktop.voip/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp
--- telegram-desktop/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp        2020-01-04 13:46:32.120644114 +0000
+++ telegram-desktop.voip/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp   2020-01-04 13:46:05.496128734 +0000
@@ -17,6 +17,7 @@
         },
         'include_dirs': [
           '<(tgvoip_src_loc)/webrtc_dsp',
+          '/usr/include/openssl'
         ],
         'direct_dependent_settings': {
           'include_dirs': [
