diff -Naur telegram-desktop/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp telegram-desktop.portable/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp
--- telegram-desktop/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp	2017-06-30 08:26:40.154394013 +0000
+++ telegram-desktop.portable/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp	2017-06-30 08:27:07.264111059 +0000
@@ -376,7 +376,7 @@
                 'WEBRTC_POSIX',
               ],
               'cflags_cc': [
-                '-msse2',
+
               ],
               'direct_dependent_settings': {
                 'libraries': [
