diff -Naur a/Telegram/gyp/telegram_linux.gypi b/Telegram/gyp/telegram_linux.gypi
--- a/Telegram/gyp/telegram_linux.gypi	2016-10-03 10:45:52.000000000 -0500
+++ b/Telegram/gyp/telegram_linux.gypi	2016-10-06 17:32:23.959989524 -0500
@@ -65,18 +65,10 @@
     'configurations': {
       'Release': {
         'cflags': [
-          '-Ofast',
-          '-flto',
-          '-fno-strict-aliasing',
         ],
         'cflags_cc': [
-          '-Ofast',
-          '-flto',
-          '-fno-strict-aliasing',
         ],
         'ldflags': [
-          '-Ofast',
-          '-flto',
         ],
       },
     },
