diff --git a/Telegram/lib_lottie/lottie/lottie_cache.cpp b/Telegram/lib_lottie/lottie/lottie_cache.cpp
index 10eb55b35..71c1213ae 100644
--- a/Telegram/lib_lottie/lottie/lottie_cache.cpp
+++ b/Telegram/lib_lottie/lottie/lottie_cache.cpp
@@ -148,7 +148,7 @@ void DecodeYUV2RGB(
 		dst,
 		dstLineSize);
 
-	Ensures(lines == to.height());
+	//Ensures(lines == to.height());
 }
 
 void DecodeAlpha(QImage &to, const EncodedStorage &from) {
@@ -225,7 +225,7 @@ void EncodeRGB2YUV(
 		dst,
 		dstLineSize);
 
-	Ensures(lines == from.height());
+	//Ensures(lines == from.height());
 }
 
 void EncodeAlpha(EncodedStorage &to, const QImage &from) {
