--- telegram-desktop/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp	2020-01-04 13:57:47.816810455 +0000
+++ telegram-desktop.ffmpeg/Telegram/SourceFiles/ffmpeg/ffmpeg_utility.cpp	2020-01-04 14:00:01.425697736 +0000
@@ -392,6 +392,12 @@
 }
 
 void UnPremultiply(QImage &to, const QImage &from) {
+       to = from.convertToFormat(QImage::Format_ARGB32);
+        return;
+        //// below is original tdesktop code, we shortcut them by using Qt public API
+        //// see https://github.com/telegramdesktop/tdesktop/issues/6219
+        /*
+
 	// This creates QImage::Format_ARGB32_Premultiplied, but we use it
 	// as an image in QImage::Format_ARGB32 format.
 	if (!GoodStorageForFrame(to, from.size())) {
@@ -412,9 +418,15 @@
 	} else {
 		UnPremultiplyLine(toBytes, fromBytes, width * height);
 	}
+	*/
 }
 
 void PremultiplyInplace(QImage &image) {
+        image = image.convertToFormat(QImage::Format_ARGB32_Premultiplied);
+        return;
+        //// below is original tdesktop code, we shortcut them by using Qt public API
+        //// see https://github.com/telegramdesktop/tdesktop/issues/6219
+        /*
 	const auto perLine = image.bytesPerLine();
 	const auto width = image.width();
 	const auto height = image.height();
@@ -427,6 +439,7 @@
 	} else {
 		PremultiplyLine(bytes, bytes, width * height);
 	}
+	*/
 }
 
 } // namespace FFmpeg
