--- a/pastebinit
+++ b/pastebinit
@@ -31,7 +31,17 @@ if sys.version[0] == "2":
 else:
     from configparser import ConfigParser, NoOptionError
     from urllib.parse import urlencode
-    from urllib.request import FancyURLopener
+    from urllib import request
+    class FancyURLopener(dict):
+        version = "FancyURLopener_wrapper 1.0 (Report bug if you see me)"
+        def addheader(self,k,v):
+            self[k] = v
+        def open(self,url,data):
+            tmp = request.Request(url,data.encode('ASCII'))
+            for k in self:
+                tmp.add_header(k,self[k])
+            tmp.add_header('User-agent',self.version)
+            return request.urlopen(tmp)
 
 # Set the default pastebin
 defaultPB = "paste.aosc.io"
