Index: xamarin-pkg-nuget/src/CommandLine/CommandLine.csproj
===================================================================
--- xamarin-pkg-nuget.orig/src/CommandLine/CommandLine.csproj	2015-01-15 19:30:44.346864746 +0000
+++ xamarin-pkg-nuget/src/CommandLine/CommandLine.csproj	2015-02-26 11:10:38.114087193 +0000
@@ -32,7 +32,7 @@
     <Reference Include="Microsoft.CSharp" />
     <Reference Include="Microsoft.Web.XmlTransform, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\lib\Microsoft.Web.XmlTransform.dll</HintPath>
+      <HintPath>..\..\xdt\XmlTransform\bin\Debug\Microsoft.Web.XmlTransform.dll</HintPath>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.ComponentModel.Composition" />
Index: xamarin-pkg-nuget/src/Core/Core.csproj
===================================================================
--- xamarin-pkg-nuget.orig/src/Core/Core.csproj	2015-02-26 11:01:59.642076980 +0000
+++ xamarin-pkg-nuget/src/Core/Core.csproj	2015-02-26 11:10:26.654086967 +0000
@@ -220,7 +220,7 @@
     <Compile Include="Utility\WebRequestEventArgs.cs" />
     <Compile Include="FileModifiers\XdtTransformer.cs" />
     <Reference Include="Microsoft.Web.XmlTransform">
-      <HintPath>..\..\lib\Microsoft.Web.XmlTransform.dll</HintPath>
+      <HintPath>..\..\xdt\XmlTransform\bin\Debug\Microsoft.Web.XmlTransform.dll</HintPath>
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.ComponentModel.DataAnnotations" />
@@ -321,4 +321,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
\ No newline at end of file
+</Project>
