diff -Naur a/src/CommandLine/CommandLine.csproj b/src/CommandLine/CommandLine.csproj
--- a/src/CommandLine/CommandLine.csproj	2015-08-25 10:06:56.000000000 -0500
+++ b/src/CommandLine/CommandLine.csproj	2016-10-01 01:49:08.413971526 -0500
@@ -30,10 +30,6 @@
     <Reference Include="Microsoft.Build.Engine" />
     <Reference Include="Microsoft.Build.Framework" />
     <Reference Include="Microsoft.CSharp" />
-    <Reference Include="Microsoft.Web.XmlTransform, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\lib\Microsoft.Web.XmlTransform.dll</HintPath>
-    </Reference>
     <Reference Include="System" />
     <Reference Include="System.ComponentModel.Composition" />
     <Reference Include="System.ComponentModel.DataAnnotations" />
@@ -41,6 +37,7 @@
     <Reference Include="System.Xml.Linq" />
     <Reference Include="System.Xml" />
     <Reference Include="WindowsBase" />
+    <Reference Include="Microsoft.Web.XmlTransform, Version=2.8.1.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756" />
   </ItemGroup>
   <ItemGroup>
     <Compile Include="..\..\Common\MsBuildProjectUtility.cs">
diff -Naur a/src/Core/Core.csproj b/src/Core/Core.csproj
--- a/src/Core/Core.csproj	2015-08-25 10:06:56.000000000 -0500
+++ b/src/Core/Core.csproj	2016-10-01 01:50:10.544367448 -0500
@@ -221,9 +221,7 @@
     <Compile Include="Utility\VersionSpec.cs" />
     <Compile Include="Utility\WebRequestEventArgs.cs" />
     <Compile Include="FileModifiers\XdtTransformer.cs" />
-    <Reference Include="Microsoft.Web.XmlTransform">
-      <HintPath>..\..\lib\Microsoft.Web.XmlTransform.dll</HintPath>
-    </Reference>
+    <Reference Include="Microsoft.Web.XmlTransform, Version=2.8.1.0, Culture=neutral, PublicKeyToken=0738eb9f132ed756" />
     <Reference Include="System" />
     <Reference Include="System.ComponentModel.DataAnnotations" />
     <Reference Include="System.Core" />
@@ -323,4 +321,4 @@
   <Target Name="AfterBuild">
   </Target>
   -->
-</Project>
+</Project>
