diff --git a/setup.py b/setup.py
index 9c751f3244..e691038086 100755
--- a/setup.py
+++ b/setup.py
@@ -321,7 +321,7 @@ def run(self):
           },
           # Set upper bound when making the release branch.
           install_requires=[
-              'mpmath >= 1.1.0, < 1.4',
+              'mpmath >= 1.1.0, < 1.5',
           ],
           py_modules=['isympy'],
           packages=['sympy'] + modules + tests,

