Author: Bartosz Fenski <fenio@debian.org>
Description: Upstream build system should not run update-rc.d.

diff -Naurp fuse.orig/util/Makefile.am fuse/util/Makefile.am
--- fuse.orig/util/Makefile.am
+++ fuse/util/Makefile.am
@@ -41,7 +41,7 @@ install-exec-local:
 	$(INSTALL_SCRIPT) $(srcdir)/init_script $(DESTDIR)$(INIT_D_PATH)/fuse
 	@if test -x /usr/sbin/update-rc.d; then \
 		echo "/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true"; \
-		/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true; \
+		#/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true; \
 	fi
 
 install-data-local:
