Author: Daniel Baumann <mail@daniel-baumann.ch>
Description: Don't exclude fioc.h from examples (Closes: #626522).

diff -Naurp fuse.orig/example/Makefile.am fuse/example/Makefile.am
--- fuse.orig/example/Makefile.am
+++ fuse/example/Makefile.am
@@ -1,7 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
 AM_CPPFLAGS = -I$(top_srcdir)/include -D_FILE_OFFSET_BITS=64 -D_REENTRANT
-noinst_HEADERS = fioc.h
 noinst_PROGRAMS = fusexmp fusexmp_fh null hello hello_ll fioc fioclient \
 		  fsel fselclient cusexmp
 
