From b4396b8fc0a01b88496392df25d1852970877849 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 25 Apr 2024 01:10:22 -0700
Subject: [PATCH 1/4] [Fedora] feat(cups-pdf.c): use document title where
 possible

---
 src/cups-pdf.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/cups-pdf.c b/src/cups-pdf.c
index 4c814d7..2068f9d 100644
--- a/src/cups-pdf.c
+++ b/src/cups-pdf.c
@@ -651,6 +651,13 @@ static int preparetitle(char *title) {
       title[strlen(title)-1]='\0';
       memmove(title, title+1, strlen(title));
     }
+    cut=strchr(title, '\r');
+    if (!cut)
+      cut=strchr(title, '\n');
+    if (cut) {
+      *cut = 0;
+      log_event(CPDEBUG, "removing newlines from title", title);
+    }
   }
   cut=strrchr(title, '/');
   if (cut != NULL) {
@@ -740,11 +747,13 @@ static int preparespoolfile(FILE *fpsrc, char *spoolfile, char *title, char *cmd
   (void) fputs(buffer, fpdest);
   while (fgets2(buffer, BUFSIZE, fpsrc) != NULL) {
     (void) fputs(buffer, fpdest);
-    if (!is_title && !rec_depth)
+    if (!is_title && !rec_depth) {
+      memset(title, 0, BUFSIZE);
       if (sscanf(buffer, "%%%%Title: %"TBUFSIZE"c", title)==1) {
         log_event(CPDEBUG, "found title in ps code: %s", title);
         is_title=1;
       }
+    }
     if (!strncmp(buffer, "%!", 2)) {
       log_event(CPDEBUG, "found embedded (e)ps code: %s", buffer);
       rec_depth++;
-- 
2.44.0

