From 8563d160a8bdc3dfa546367149b6fc8a27414198 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 16 Jul 2024 11:56:40 +0800
Subject: [PATCH] fix(krusaderapp.h): fix compatibility with newer libX11

---
 krusader/krusaderapp.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/krusader/krusaderapp.h b/krusader/krusaderapp.h
index a50f5c9..16f3d30 100644
--- a/krusader/krusaderapp.h
+++ b/krusader/krusaderapp.h
@@ -4,6 +4,18 @@
 #include <tdeapplication.h>
 #include "X11/Xlib.h"
 
+#ifdef KeyPress
+#undef KeyPress
+#endif
+
+#ifdef GrayScale
+#undef GrayScale
+#endif
+
+#ifdef Status
+#undef Status
+#endif
+
 // declare a dummy kapplication, just to get the X focusin focusout events
 class KrusaderApp: public TDEApplication {
 	TQ_OBJECT
-- 
2.45.2

