From ed061bb68156aac464e6fe109016378e24b8c3cc Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Tue, 11 Nov 2025 10:16:25 +0800
Subject: [PATCH] fix(krusaderapp.h): fix compatibility with newer libX11

---
 src/app/krusaderapp.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/app/krusaderapp.h b/src/app/krusaderapp.h
index a50f5c9..16f3d30 100644
--- a/src/app/krusaderapp.h
+++ b/src/app/krusaderapp.h
@@ -4,6 +4,18 @@
 #include <tdeapplication.h>
 #include "X11/Xlib.h"
 
+#ifdef KeyPress
+#undef KeyPress
+#endif
+
+#ifdef GrayScale
+#undef GrayScale
+#endif
+
+#ifdef Status
+#undef Status
+#endif
+
 // declare a dummy kapplication, just to get the X focusin focusout events
 class KrusaderApp: public TDEApplication {
 	TQ_OBJECT
-- 
2.51.1

