From bb4f29885ccffd1b205648ccb0ea2e98324defd2 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Sun, 12 Jul 2026 20:37:49 +0800
Subject: [PATCH 2/2] Fix build with JavaScript enabled

---
 src/app/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index e9e722c..429a0f4 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -32,7 +32,7 @@ include_directories(
 set( LIB_KRJS "" ) 
 if( HAVE_JAVASCRIPT )
   add_definitions( -D__KJSEMBED__ )
-  set( LIB_KRJS "KrJS-static kjsembed kjs-shared" ) 
+  set( LIB_KRJS KrJS-static kjsembed kjs-shared )
 endif( )
 
 set( LIB_KONQ "" ) 
-- 
2.52.0

