diff -Naur VirtualBox-6.1.36/src/VBox/Frontends/VirtualBox/src/extradata/UIExtraDataManager.cpp VirtualBox-6.1.36-no-update-check/src/VBox/Frontends/VirtualBox/src/extradata/UIExtraDataManager.cpp
--- VirtualBox-6.1.36/src/VBox/Frontends/VirtualBox/src/extradata/UIExtraDataManager.cpp	2022-07-19 16:58:28.000000000 -0400
+++ VirtualBox-6.1.36-no-update-check/src/VBox/Frontends/VirtualBox/src/extradata/UIExtraDataManager.cpp	2022-08-20 21:25:13.627144817 -0400
@@ -2348,8 +2348,7 @@
 #ifdef VBOX_GUI_WITH_NETWORK_MANAGER
 bool UIExtraDataManager::applicationUpdateEnabled()
 {
-    /* 'True' unless 'restriction' feature allowed: */
-    return !isFeatureAllowed(GUI_PreventApplicationUpdate);
+    return false;
 }
 
 QString UIExtraDataManager::applicationUpdateData()
