From c7ce104923c8219425aa30fa46e5b8f7acff51db Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Mon, 30 Sep 2024 21:48:02 +0800
Subject: [PATCH 09/11] FROM AOSC: Add gitignore

---
 .gitignore | 5 +++++
 1 file changed, 5 insertions(+)
 create mode 100644 .gitignore

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 000000000..8610ed868
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,5 @@
+/AutoConfig.kmk
+/LocalConfig.kmk
+/configure.log
+/env.sh
+/out
-- 
2.51.0

