From cc2d555396b782dd6b00d5fa6d237bd251b67e2b Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Mon, 30 Sep 2024 21:48:02 +0800
Subject: [PATCH 09/10] FROM AOSC: Add gitignore

---
 .gitignore | 5 +++++
 1 file changed, 5 insertions(+)
 create mode 100644 .gitignore

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 000000000000..8610ed8689b9
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,5 @@
+/AutoConfig.kmk
+/LocalConfig.kmk
+/configure.log
+/env.sh
+/out
-- 
2.46.2

