diff -Naur VirtualBox-7.0.8.wheel/debian/postinst VirtualBox-7.0.8/debian/postinst
--- VirtualBox-7.0.8.wheel/debian/postinst	2023-04-25 18:52:06.680165716 -0700
+++ VirtualBox-7.0.8/debian/postinst	2023-04-25 18:53:08.987224751 -0700
@@ -59,9 +59,9 @@
 
   # create users groups (disable with INSTALL_NO_GROUP=1 in /etc/default/virtualbox)
   if [ "$INSTALL_NO_GROUP" != "1" ]; then
-    db_input low virtualbox/group-vboxusers || true
+    db_input low virtualbox/group-wheel || true
     db_go || true
-    addgroup --system vboxusers || true
+    addgroup --system wheel || true
   fi
 
   # The starters need to be Suid root. They drop the privileges before starting
diff -Naur VirtualBox-7.0.8.wheel/debian/templates VirtualBox-7.0.8/debian/templates
--- VirtualBox-7.0.8.wheel/debian/templates	2023-04-25 18:52:06.679165683 -0700
+++ VirtualBox-7.0.8/debian/templates	2023-04-25 18:53:08.992224916 -0700
@@ -1,9 +1,9 @@
-Template: virtualbox/group-vboxusers
+Template: virtualbox/group-wheel
 Type: note
-Description: Creating group 'vboxusers'
+Description: Creating group 'wheel'
  Users of VirtualBox must be members of this group. Host network interfaces
  will be assigned to this group.
-Description-de.UTF-8: Erzeuge Gruppe 'vboxusers'
+Description-de.UTF-8: Erzeuge Gruppe 'wheel'
  Nutzer von VirtualBox müssen Mitglied dieser Gruppe sein. Hostinterfaces
  werden dieser Gruppe zugewiesen.
 
diff -Naur VirtualBox-7.0.8.wheel/doc/manual/en_US/user_BasicConcepts.xml VirtualBox-7.0.8/doc/manual/en_US/user_BasicConcepts.xml
--- VirtualBox-7.0.8.wheel/doc/manual/en_US/user_BasicConcepts.xml	2023-04-25 18:52:11.121312457 -0700
+++ VirtualBox-7.0.8/doc/manual/en_US/user_BasicConcepts.xml	2023-04-25 18:53:08.998225115 -0700
@@ -2479,7 +2479,7 @@
         On supported Linux hosts, &product-name; accesses USB devices
         through special files in the file system. When &product-name; is
         installed, these are made available to all users in the
-        <literal>vboxusers</literal> system group. In order to be able
+        <literal>wheel</literal> system group. In order to be able
         to access USB from guest systems, make sure that you are a
         member of this group.
       </para>
diff -Naur VirtualBox-7.0.8.wheel/doc/manual/en_US/user_Installation.xml VirtualBox-7.0.8/doc/manual/en_US/user_Installation.xml
--- VirtualBox-7.0.8.wheel/doc/manual/en_US/user_Installation.xml	2023-04-25 18:52:11.120312424 -0700
+++ VirtualBox-7.0.8/doc/manual/en_US/user_Installation.xml	2023-04-25 18:53:09.004225313 -0700
@@ -824,7 +824,7 @@
           <listitem>
             <para>
               Creates a new system group called
-              <literal>vboxusers</literal>.
+              <literal>wheel</literal>.
             </para>
           </listitem>
 
@@ -846,7 +846,7 @@
               <filename>/etc/udev/rules.d/60-vboxdrv.rules</filename>, a
               description file for udev, if that is present, which makes
               the USB devices accessible to all users in the
-              <literal>vboxusers</literal> group.
+              <literal>wheel</literal> group.
             </para>
           </listitem>
 
@@ -876,12 +876,12 @@
 
         <para>
           Add every user who needs to access USB devices from a
-          VirtualBox guests to the group <literal>vboxusers</literal>.
+          VirtualBox guests to the group <literal>wheel</literal>.
           Either use the OS user management tools or run the following
           command as root:
         </para>
 
-<screen>sudo usermod -a -G vboxusers username</screen>
+<screen>sudo usermod -a -G wheel username</screen>
 
         <note>
           <para>
@@ -893,7 +893,7 @@
             <command>groups</command> command and add all these groups
             in a comma-separated list to the command line after the
             <option>-G</option> option. For example: <command>usermod -G
-            <replaceable>group1</replaceable>,<replaceable>group2</replaceable>,vboxusers
+            <replaceable>group1</replaceable>,<replaceable>group2</replaceable>,wheel
             <replaceable>username</replaceable></command>.
           </para>
         </note>
@@ -1107,7 +1107,7 @@
 <screen>INSTALL_NO_UDEV=1</screen>
 
         <para>
-          The creation of the group vboxusers can be prevented as
+          The creation of the group wheel can be prevented as
           follows:
         </para>
 
@@ -1125,20 +1125,20 @@
 
     </sect2>
 
-    <sect2 id="install-linux-vboxusers">
+    <sect2 id="install-linux-wheel">
 
-      <title>The vboxusers Group</title>
+      <title>The wheel Group</title>
 
       <para>
         The Linux installers create the system user group
-        <literal>vboxusers</literal> during installation. Any system
+        <literal>wheel</literal> during installation. Any system
         user who is going to use USB devices from &product-name; guests
         must be a member of that group. A user can be made a member of
-        the group <literal>vboxusers</literal> either by using the
+        the group <literal>wheel</literal> either by using the
         desktop user and group tools, or with the following command:
       </para>
 
-<screen>sudo usermod -a -G vboxusers username</screen>
+<screen>sudo usermod -a -G wheel username</screen>
 
     </sect2>
 
diff -Naur VirtualBox-7.0.8.wheel/doc/manual/en_US/user_Security.xml VirtualBox-7.0.8/doc/manual/en_US/user_Security.xml
--- VirtualBox-7.0.8.wheel/doc/manual/en_US/user_Security.xml	2023-04-25 18:52:11.110312094 -0700
+++ VirtualBox-7.0.8/doc/manual/en_US/user_Security.xml	2023-04-25 18:53:09.008225445 -0700
@@ -173,7 +173,7 @@
         and Linux hosts it is necessary to configure the proper
         permissions for users executing VMs and who should be able to
         access certain host resources. For instance, Linux users must be
-        member of the <emphasis>vboxusers</emphasis> group to be able to
+        member of the <emphasis>wheel</emphasis> group to be able to
         pass USB devices to a guest. If a serial host interface should
         be accessed from a VM, the proper permissions must be granted to
         the user to be able to access that device. The same applies to
diff -Naur VirtualBox-7.0.8.wheel/doc/manual/en_US/user_Troubleshooting.xml VirtualBox-7.0.8/doc/manual/en_US/user_Troubleshooting.xml
--- VirtualBox-7.0.8.wheel/doc/manual/en_US/user_Troubleshooting.xml	2023-04-25 18:52:11.120312424 -0700
+++ VirtualBox-7.0.8/doc/manual/en_US/user_Troubleshooting.xml	2023-04-25 18:53:09.014225644 -0700
@@ -1863,7 +1863,7 @@
 
       <para>
         If USB is not working on your Linux host, make sure that the
-        current user is a member of the <literal>vboxusers</literal>
+        current user is a member of the <literal>wheel</literal>
         group. Please keep in mind that group membership does not take
         effect immediately but rather at the next login. If available,
         the <command>newgrp</command> command may avoid the need for a
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_bg.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_bg.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_bg.ts	2023-04-25 18:52:10.571294282 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_bg.ts	2023-04-25 18:53:09.029226139 -0700
@@ -16948,8 +16948,8 @@
         <translation>Не може да се зареди услугата Host USB Proxy (VERR_FILE_NOT_FOUND). Услугата може да не е инсталирана на реалния компютър</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>Текущо не е позволено VirtualBox да достъпва USB устройства. Можете да промените това с добавяне на вашия потребител към групата &apos;vboxusers&apos;.  Моля, погледнете потребителското ръководство за по-подробно описание</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>Текущо не е позволено VirtualBox да достъпва USB устройства. Можете да промените това с добавяне на вашия потребител към групата &apos;wheel&apos;.  Моля, погледнете потребителското ръководство за по-подробно описание</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ca.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ca.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ca.ts	2023-04-25 18:52:10.594295043 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ca.ts	2023-04-25 18:53:09.040226503 -0700
@@ -11679,8 +11679,8 @@
         <translation>No s&apos;ha pogut carregar el servei de servidor intermediari USB (VERR_FILE_NOT_FOUND). Pot ser que el servei no estigui instal·lat a l&apos;ordinador amfitrió</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>El VirtualBox no té permisos per a accedir als dispositius USB. Podeu canviar això afegint el vostre usuari al grup «vboxusers». Vegeu el manual de l&apos;usuari per a una explicació més detallada</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>El VirtualBox no té permisos per a accedir als dispositius USB. Podeu canviar això afegint el vostre usuari al grup «wheel». Vegeu el manual de l&apos;usuari per a una explicació més detallada</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ca_VA.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ca_VA.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ca_VA.ts	2023-04-25 18:52:10.592294977 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ca_VA.ts	2023-04-25 18:53:09.055226999 -0700
@@ -16169,8 +16169,8 @@
         <translation>No s&apos;ha pogut carregar el servei de servidor intermediari USB (VERR_FILE_NOT_FOUND). Pot ser que el servei no estiga instal·lat a l&apos;ordinador amfitrió</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>El VirtualBox no té permisos per accedir als dispositius USB. Podeu canviar això afegint el vostre usuari al grup «vboxusers». Vegeu el manual de l&apos;usuari per a una explicació més detallada</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>El VirtualBox no té permisos per accedir als dispositius USB. Podeu canviar això afegint el vostre usuari al grup «wheel». Vegeu el manual de l&apos;usuari per a una explicació més detallada</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_cs.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_cs.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_cs.ts	2023-04-25 18:52:10.559293886 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_cs.ts	2023-04-25 18:53:09.071227527 -0700
@@ -16985,8 +16985,8 @@
         <translation>Nelze spustit službu USB Proxy (VERR_FILE_NOT_FOUND). Zřejmě na tomto hostutelském počítači není služba nainstalována</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>Aplikace VirtualBox nemůže nyní přistupovat k USB zařízením. Můžete to změnit přidáním uživatele do skupiny &apos;vboxusers&apos;. Více informací najdete v uživatelské příručce</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>Aplikace VirtualBox nemůže nyní přistupovat k USB zařízením. Můžete to změnit přidáním uživatele do skupiny &apos;wheel&apos;. Více informací najdete v uživatelské příručce</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_da.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_da.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_da.ts	2023-04-25 18:52:10.564294051 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_da.ts	2023-04-25 18:53:09.083227924 -0700
@@ -12556,8 +12556,8 @@
         <translation>Kunne ikke starte værtens USB-proxytjeneste (VERR_FILE_NOT_FOUND). Tjenesten er muligvis ikke installeret på værtsmaskinen</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox har ikke rettigheder til at tilgå USB-enheder. Du kan ændre dette ved at tilføje din bruger til gruppen &apos;vboxusers&apos;. Se dokumentationen for yderligere detaljer</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox har ikke rettigheder til at tilgå USB-enheder. Du kan ændre dette ved at tilføje din bruger til gruppen &apos;wheel&apos;. Se dokumentationen for yderligere detaljer</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_de.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_de.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_de.ts	2023-04-25 18:52:10.567294150 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_de.ts	2023-04-25 18:53:09.096228354 -0700
@@ -13887,8 +13887,8 @@
         <translation>Der USB-Proxy-Dienst konnte nicht gestarted werden (VERR_FILE_NOT_FOUND). Dieser Dienst ist möglicherweise nicht auf diesem Computer installiert</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox kann nicht auf USB-Geräte zugreifen. Dieses Problem kann dadurch gelöst werden, indem der aktuelle Nutzer Mitglied der Gruppe &apos;vboxusers&apos; wird. In der Dokumentation finden Sie weitere Erläuterungen</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox kann nicht auf USB-Geräte zugreifen. Dieses Problem kann dadurch gelöst werden, indem der aktuelle Nutzer Mitglied der Gruppe &apos;wheel&apos; wird. In der Dokumentation finden Sie weitere Erläuterungen</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_el.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_el.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_el.ts	2023-04-25 18:52:10.582294646 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_el.ts	2023-04-25 18:53:09.109228783 -0700
@@ -13779,8 +13779,8 @@
         <translation>Δεν φορτώθηκε η υπηρεσία διαμεσολάβησης USB του οικοδεσπότη (VERR_FILE_NOT_FOUND). Η υπηρεσία μπορεί να μην έχει εγκατασταθεί στον υπολογιστή</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>Το VirtualBox δεν έχει αυτή τη στιγμή πρόσβαση στις συσκευές USB. Μπορείτε να το αλλάξετε προσθέτοντας τον χρήστη στην ομάδα &apos;vboxusers&apos;. Δείτε το εγχειρίδιο για λεπτομερέστερη εξήγηση</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>Το VirtualBox δεν έχει αυτή τη στιγμή πρόσβαση στις συσκευές USB. Μπορείτε να το αλλάξετε προσθέτοντας τον χρήστη στην ομάδα &apos;wheel&apos;. Δείτε το εγχειρίδιο για λεπτομερέστερη εξήγηση</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_es.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_es.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_es.ts	2023-04-25 18:52:10.616295770 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_es.ts	2023-04-25 18:53:09.126229345 -0700
@@ -18826,8 +18826,8 @@
         <translation>No se puede cargar el servicio Host USB Proxy (VERR_FILE_NOT_FOUND). El servicio podría no estar instalado en la máquina anfitrión</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox no tiene permisos para acceder a los dispositivos  USB. Puede cambiar esto agregando su usuario al grupo «vboxusers».  Mire el manual de usuario para una explicación más detallada</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox no tiene permisos para acceder a los dispositivos  USB. Puede cambiar esto agregando su usuario al grupo «wheel».  Mire el manual de usuario para una explicación más detallada</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_eu.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_eu.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_eu.ts	2023-04-25 18:52:10.584294712 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_eu.ts	2023-04-25 18:53:09.140229808 -0700
@@ -18338,8 +18338,8 @@
     </message>
     <message>
         <location line="+1"/>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox-ek ez du sarbiderik USB gailuetara.  Hau zure erabiltzailea &apos;vboxusers&apos; taldera gehituz aldatu dezakezu.  Mesedez ikusi erabiltzaile eskuliburua azalpen xehetuagorako</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox-ek ez du sarbiderik USB gailuetara.  Hau zure erabiltzailea &apos;wheel&apos; taldera gehituz aldatu dezakezu.  Mesedez ikusi erabiltzaile eskuliburua azalpen xehetuagorako</translation>
     </message>
     <message>
         <location line="+1"/>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fa.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fa.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fa.ts	2023-04-25 18:52:10.614295704 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fa.ts	2023-04-25 18:53:09.154230271 -0700
@@ -15366,8 +15366,8 @@
         <translation>قادر به بارگیری سرویس پراکسی یو اِس بی میزبان نیست (VERR_FILE_NOT_FOUND). شاید سرویس در رایانه میزبان نصب نشده است</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>ویرچوال باکس اکنون اجازه دسترسی به دستگاه یو ایس بی را ندارد.  این را میتوانید با افزودن کاربر خود به گروه &apos;vboxusers&apos; تغییر دهید.  لطفا برای توضیحات جزئی تر راهنمای کاربری را ببینید</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>ویرچوال باکس اکنون اجازه دسترسی به دستگاه یو ایس بی را ندارد.  این را میتوانید با افزودن کاربر خود به گروه &apos;wheel&apos; تغییر دهید.  لطفا برای توضیحات جزئی تر راهنمای کاربری را ببینید</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fi.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fi.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fi.ts	2023-04-25 18:52:10.579294547 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fi.ts	2023-04-25 18:53:09.167230700 -0700
@@ -13415,7 +13415,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fr.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fr.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fr.ts	2023-04-25 18:52:10.565294084 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_fr.ts	2023-04-25 18:53:09.182231196 -0700
@@ -17142,8 +17142,8 @@
         <translation>Le service de relais de périphériques USB n&apos;a pas pû être activé (erreur VERR_FILE_NOT_FOUND). Il se peut que le service ne soit pas installé sur l&apos;ordinateur hôte</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox n&apos;est pas autorisé à accéder aux périphériques USB actuellement. Vous pouvez changer ceci en ajoutant votre nom d&apos;utilisateur au groupe «vboxusers». Veuillez consulter le guide de l&apos;utilisateur pour des explications plus détaillées</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox n&apos;est pas autorisé à accéder aux périphériques USB actuellement. Vous pouvez changer ceci en ajoutant votre nom d&apos;utilisateur au groupe «wheel». Veuillez consulter le guide de l&apos;utilisateur pour des explications plus détaillées</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_gl.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_gl.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_gl.ts	2023-04-25 18:52:10.604295373 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_gl.ts	2023-04-25 18:53:09.194231593 -0700
@@ -13370,7 +13370,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_he.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_he.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_he.ts	2023-04-25 18:52:10.590294910 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_he.ts	2023-04-25 18:53:09.204231923 -0700
@@ -10816,7 +10816,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_hr_HR.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_hr_HR.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_hr_HR.ts	2023-04-25 18:52:10.566294117 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_hr_HR.ts	2023-04-25 18:53:09.214232254 -0700
@@ -8386,8 +8386,8 @@
         <translation>Nisam uspio učitati Domaćinov USB Posrednički Servis (VERR_FILE_NOT_FOUND). Servis možda nije instaliran na računalo domaćina</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBoxu trenutačno nije dozvoljen pristup USB uređajima.  Možete promijeniti to dodavajući svog korisnika &apos;vboxusers&apos; grupi.  Pogledajte korisnička uputstva za detaljnije objašnjenje</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBoxu trenutačno nije dozvoljen pristup USB uređajima.  Možete promijeniti to dodavajući svog korisnika &apos;wheel&apos; grupi.  Pogledajte korisnička uputstva za detaljnije objašnjenje</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_hu.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_hu.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_hu.ts	2023-04-25 18:52:10.562293985 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_hu.ts	2023-04-25 18:53:09.229232750 -0700
@@ -18297,8 +18297,8 @@
         <translation>Nem tölthető be a gazda USB Proxy szolgáltatás (FÁJL_NEM_TALÁLHATÓ). A szolgáltatás talán nincs telepítve a gazda számítógépre</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>A VirtualBox jelenleg nem férhet hozzá az USB eszközökhöz.  Ezen változtathatsz, ha a felhasználód hozzáadod a &apos;vboxusers&apos; csoporthoz.  Részletes magyarázatért nézd meg a Felhasználói kézikönyvet</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>A VirtualBox jelenleg nem férhet hozzá az USB eszközökhöz.  Ezen változtathatsz, ha a felhasználód hozzáadod a &apos;wheel&apos; csoporthoz.  Részletes magyarázatért nézd meg a Felhasználói kézikönyvet</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_id.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_id.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_id.ts	2023-04-25 18:52:10.573294349 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_id.ts	2023-04-25 18:53:09.245233278 -0700
@@ -17334,8 +17334,8 @@
         <translation>Tak bisa memuat Layanan Proksi USB Host (VERR_FILE_NOT_FOUND). Layanan mungkin tak terinstal pada komputer host</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox kini tak diijinkan mengakses peranti USB. Anda dapat mengubah ini dengan cara menambahkan pengguna Anda ke grup &apos;vboxusers&apos;.  Silakan lihat manual pengguna untuk penjelasan lebih terinci</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox kini tak diijinkan mengakses peranti USB. Anda dapat mengubah ini dengan cara menambahkan pengguna Anda ke grup &apos;wheel&apos;.  Silakan lihat manual pengguna untuk penjelasan lebih terinci</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_it.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_it.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_it.ts	2023-04-25 18:52:10.601295274 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_it.ts	2023-04-25 18:53:09.262233840 -0700
@@ -18953,8 +18953,8 @@
         <translation>Impossibile caricare il servizio Host USB Proxy (VERR_FILE_NOT_FOUND). Il servizio potrebbe non essere installato sul computer host</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>Attualmente VirtualBox non ha il permesso di accedere ai dispositivi USB. Puoi cambiare questa situazione aggiungendo il tuo utente al gruppo &apos;vboxusers&apos;. Vedi il manuale utente per una spiegazione più dettagliata</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>Attualmente VirtualBox non ha il permesso di accedere ai dispositivi USB. Puoi cambiare questa situazione aggiungendo il tuo utente al gruppo &apos;wheel&apos;. Vedi il manuale utente per una spiegazione più dettagliata</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ja.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ja.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ja.ts	2023-04-25 18:52:10.587294811 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ja.ts	2023-04-25 18:53:09.278234369 -0700
@@ -19276,8 +19276,8 @@
         <translation>USBプロキシーサービスが読み込めません (VERR_FILE_NOT_FOUND)。このサービスはホストコンピューターにインストールされている必要があります</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBoxは現在USBデバイスへのアクセスが許可されていません。ユーザーを&apos;vboxusers&apos;グループに追加してください。詳細な説明はユーザーマニュアルをお読みください</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBoxは現在USBデバイスへのアクセスが許可されていません。ユーザーを&apos;wheel&apos;グループに追加してください。詳細な説明はユーザーマニュアルをお読みください</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ka.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ka.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ka.ts	2023-04-25 18:52:10.577294481 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ka.ts	2023-04-25 18:53:09.288234700 -0700
@@ -8356,7 +8356,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_km_KH.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_km_KH.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_km_KH.ts	2023-04-25 18:52:10.595295076 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_km_KH.ts	2023-04-25 18:53:09.300235096 -0700
@@ -13297,7 +13297,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ko.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ko.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ko.ts	2023-04-25 18:52:10.585294745 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ko.ts	2023-04-25 18:53:09.311235460 -0700
@@ -11147,8 +11147,8 @@
         <translation>호스트 USB 프록시 서비스를 불러올 수 없습니다(VERR_FILE_NOT_FOUND). 호스트 컴퓨터에 서비스가 설치되어 있지 않을 수도 있습니다</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox에서 USB 장치에 접근할 수 없습니다. 현재 사용자를 &apos;vboxusers&apos; 그룹에 추가하십시오. 더 자세한 설명은 사용자 설명서를 참고하십시오</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox에서 USB 장치에 접근할 수 없습니다. 현재 사용자를 &apos;wheel&apos; 그룹에 추가하십시오. 더 자세한 설명은 사용자 설명서를 참고하십시오</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_lt.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_lt.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_lt.ts	2023-04-25 18:52:10.597295142 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_lt.ts	2023-04-25 18:53:09.325235922 -0700
@@ -15751,8 +15751,8 @@
         <translation>Nepavyksta įkelti pagr. kompiuterio USB įgaliotojo serverio paslaugos (VERR_FILE_NOT_FOUND). Galbūt ši paslauga neįdiegta pagrindiniame kompiueryje</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox programai neleidote pasiekti USB įrenginių. Šią parinktį galite pakeisti savo naudotoją priskirdami „vboxusers“ grupei. Daugiau informacijos rasite naudotojo žinyne</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox programai neleidote pasiekti USB įrenginių. Šią parinktį galite pakeisti savo naudotoją priskirdami „wheel“ grupei. Daugiau informacijos rasite naudotojo žinyne</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_nl.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_nl.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_nl.ts	2023-04-25 18:52:10.619295869 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_nl.ts	2023-04-25 18:53:09.338236352 -0700
@@ -13740,8 +13740,8 @@
         <translation>Kon de Host USB Proxy Service (VERR_FILE_NOT_FOUND) niet laden. De service is mogelijk niet geïnstalleerd op de hostcomputer</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox wordt momenteel niet toegestaan USB-apparaten te benaderen.  U kunt dit veranderen door uw gebruiker toe te voegen aan de groep &apos;vboxusers&apos;.  Lees de gebruikershandleiding voor een meer gedetailleerde uitleg</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox wordt momenteel niet toegestaan USB-apparaten te benaderen.  U kunt dit veranderen door uw gebruiker toe te voegen aan de groep &apos;wheel&apos;.  Lees de gebruikershandleiding voor een meer gedetailleerde uitleg</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pl.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pl.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pl.ts	2023-04-25 18:52:10.606295439 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pl.ts	2023-04-25 18:53:09.352236815 -0700
@@ -16550,8 +16550,8 @@
         <translation>Nie można załadować usługi Host USB Proxy (VERR_FILE_NOT_FOUND). Usługa ta być może nie została zainstalowana na komputerze gospodarza</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox aktualnie nie ma dostępu do urządzeń USB. Można to zmienić, dodając użytkownika do grupy vboxusers&apos;. Proszę zapoznać się z instrukcją obsługi w celu bardziej szczegółowego wyjaśnienia</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox aktualnie nie ma dostępu do urządzeń USB. Można to zmienić, dodając użytkownika do grupy wheel&apos;. Proszę zapoznać się z instrukcją obsługi w celu bardziej szczegółowego wyjaśnienia</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pt.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pt.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pt.ts	2023-04-25 18:52:10.618295836 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pt.ts	2023-04-25 18:53:09.366237277 -0700
@@ -14831,7 +14831,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pt_BR.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pt_BR.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pt_BR.ts	2023-04-25 18:52:10.611295604 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_pt_BR.ts	2023-04-25 18:53:09.383237839 -0700
@@ -18899,8 +18899,8 @@
         <translation>Não foi possível carregar o Serviço de Proxy USB do Hospedeiro (VERR_FILE_NOT_FOUND). O serviço pode não estar instalado no computador hospedeiro</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>O VirtualBox não está conseguindo acessar os dispositivos USB.  Você pode alterar este comportamento acrescentando o seu usuário ao grupo &apos;vboxusers&apos;.  Leia o manual do usuário para uma explicação mais detalhada</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>O VirtualBox não está conseguindo acessar os dispositivos USB.  Você pode alterar este comportamento acrescentando o seu usuário ao grupo &apos;wheel&apos;.  Leia o manual do usuário para uma explicação mais detalhada</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ro.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ro.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ro.ts	2023-04-25 18:52:10.568294183 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ro.ts	2023-04-25 18:53:09.396238269 -0700
@@ -13526,7 +13526,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ru.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ru.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ru.ts	2023-04-25 18:52:10.613295671 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_ru.ts	2023-04-25 18:53:09.411238765 -0700
@@ -18414,8 +18414,8 @@
         <translation>Не удалось загрузить USB Proxy службу хоста (VERR_FILE_NOT_FOUND). Возможно, данная служба не установлена на данном компьютере</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>В данный момент VirtualBox не может использовать устройства USB. Вы можете исправить данную проблему, добавив текущего пользователя в группу &apos;vboxusers&apos;. Для получения более детальных объяснений следуйте указаниям руководства пользователя</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>В данный момент VirtualBox не может использовать устройства USB. Вы можете исправить данную проблему, добавив текущего пользователя в группу &apos;wheel&apos;. Для получения более детальных объяснений следуйте указаниям руководства пользователя</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sk.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sk.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sk.ts	2023-04-25 18:52:10.608295505 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sk.ts	2023-04-25 18:53:09.425239227 -0700
@@ -13872,7 +13872,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sl.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sl.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sl.ts	2023-04-25 18:52:10.577294481 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sl.ts	2023-04-25 18:53:09.438239657 -0700
@@ -15651,8 +15651,8 @@
     </message>
     <message>
         <location line="+1"/>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox trenutno ne sme dostopati do naprav USB. To lahko spremenite z dodajanjem svojega uporabnika v skupino &apos;vboxusers&apos;. Za bolj podrobno razlago glejte uporabniški priročnik.</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox trenutno ne sme dostopati do naprav USB. To lahko spremenite z dodajanjem svojega uporabnika v skupino &apos;wheel&apos;. Za bolj podrobno razlago glejte uporabniški priročnik.</translation>
     </message>
     <message>
         <location line="+1"/>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sr.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sr.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sr.ts	2023-04-25 18:52:10.589294877 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sr.ts	2023-04-25 18:53:09.451240087 -0700
@@ -14588,8 +14588,8 @@
         <translation>Не могу да учитам Host USB Proxy Service (VERR_FILE_NOT_FOUND). Сервис можда није инсталиран на хосту</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox нема приступа USB уређајима. Можете то да промените тако што додате корисник у групи &apos;vboxusers&apos;.  Видети приручник за више детаље</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox нема приступа USB уређајима. Можете то да промените тако што додате корисник у групи &apos;wheel&apos;.  Видети приручник за више детаље</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sv.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sv.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sv.ts	2023-04-25 18:52:10.609295538 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_sv.ts	2023-04-25 18:53:09.465240550 -0700
@@ -16184,8 +16184,8 @@
         <translation>Kunde inte läsa in USB-proxytjänsten för värden (VERR_FILE_NOT_FOUND). Tjänsten kanske inte är installerad på värddatorn</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox tillåts för närvarande inte att komma åt USB-enheter.  Du kan ändra detta genom att lägga till din användare till gruppen &quot;vboxusers&quot;.  Se användarhandboken för ytterligare information</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox tillåts för närvarande inte att komma åt USB-enheter.  Du kan ändra detta genom att lägga till din användare till gruppen &quot;wheel&quot;.  Se användarhandboken för ytterligare information</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_th.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_th.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_th.ts	2023-04-25 18:52:10.574294382 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_th.ts	2023-04-25 18:53:09.477240946 -0700
@@ -12431,8 +12431,8 @@
         <translation>ไม่สามารถโหลดบริการพรอกซี USB ของโฮสต์ได้  (VERR_FILE_NOT_FOUND) บริการนี้อาจไม่ได้ถูกติดตั้งไว้บนคอมพิวเตอร์โฮสต์</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>เวอร์ชวลบอกซ์ไม่ได้รับอนุญาตให้เข้าถึงอุปกรณ์ USB คุณสามารถแก้ไขได้ด้วยการเพิ่มผู้ใช้ของคุณไปยังกลุ่ม &apos;vboxusers&apos; คำอธิบายโดยละเอียดสามารถดูเพิ่มเติมได้จากคู่มือผู้ใช้</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>เวอร์ชวลบอกซ์ไม่ได้รับอนุญาตให้เข้าถึงอุปกรณ์ USB คุณสามารถแก้ไขได้ด้วยการเพิ่มผู้ใช้ของคุณไปยังกลุ่ม &apos;wheel&apos; คำอธิบายโดยละเอียดสามารถดูเพิ่มเติมได้จากคู่มือผู้ใช้</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_tr.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_tr.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_tr.ts	2023-04-25 18:52:10.599295208 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_tr.ts	2023-04-25 18:53:09.493241475 -0700
@@ -18288,8 +18288,8 @@
         <translation>Anamakine USB Vekil Hizmeti (VERR_FILE_NOT_FOUND) yüklenemedi. Hizmet anamakine bilgisayarına yüklenememiştir</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox şu anda USB aygıtlara erişime izin vermiyor. Bunu kullanıcınızı &apos;vboxusers&apos; grubuna ekleyerek değiştirebilirsiniz. Lütfen daha fazla ayrıntılı açıklama için kullanım kılavuzuna bakın</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox şu anda USB aygıtlara erişime izin vermiyor. Bunu kullanıcınızı &apos;wheel&apos; grubuna ekleyerek değiştirebilirsiniz. Lütfen daha fazla ayrıntılı açıklama için kullanım kılavuzuna bakın</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_uk.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_uk.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_uk.ts	2023-04-25 18:52:10.581294613 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_uk.ts	2023-04-25 18:53:09.507241938 -0700
@@ -16343,8 +16343,8 @@
         <translation>Неможливо завантажити головний проксі сервер USB (VERR_FILE_NOT_FOUND). Службу, можливо, не встановили на головному комп&apos;ютері</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox зараз не дозволяє доступитись до пристроїв USB. Ви моежете змінити це, додавши користувача до групи «vboxusers». Будь ласка, ознайомтесь з користувацьким підручником для докладнішого пояснення</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox зараз не дозволяє доступитись до пристроїв USB. Ви моежете змінити це, додавши користувача до групи «wheel». Будь ласка, ознайомтесь з користувацьким підручником для докладнішого пояснення</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_xx_YY.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_xx_YY.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_xx_YY.ts	2023-04-25 18:52:10.557293820 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_xx_YY.ts	2023-04-25 18:53:09.517242268 -0700
@@ -8342,7 +8342,7 @@
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_zh_CN.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_zh_CN.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_zh_CN.ts	2023-04-25 18:52:10.603295340 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_zh_CN.ts	2023-04-25 18:53:09.530242698 -0700
@@ -16821,8 +16821,8 @@
         <translation>无法载入主机 USB 代理服务 (VERR_FILE_NOT_FOUND)。主机电脑中可能没有安装服务</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>VirtualBox 当前不允许访问 USB 设备。您可以将您的用户添加到“vboxusers”组来更改以上的问题。请查阅用户手册以获得更多详细的说明</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>VirtualBox 当前不允许访问 USB 设备。您可以将您的用户添加到“wheel”组来更改以上的问题。请查阅用户手册以获得更多详细的说明</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_zh_TW.ts VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_zh_TW.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/nls/VirtualBox_zh_TW.ts	2023-04-25 18:52:10.603295340 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/nls/VirtualBox_zh_TW.ts	2023-04-25 18:53:09.540243028 -0700
@@ -8342,8 +8342,8 @@
         <translation>無法載入主機 USB Proxy 服務  (VERR_FILE_NOT_FOUND)。 服務可能未安裝在主機上</translation>
     </message>
     <message>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>目前不允許 VirtualBox 存取 USB 裝置。 您可以透過將使用者加入到「vboxusers」群組來變更這個設定。 有關更詳細的說明，請參閱使用者手冊</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>目前不允許 VirtualBox 存取 USB 裝置。 您可以透過將使用者加入到「wheel」群組來變更這個設定。 有關更詳細的說明，請參閱使用者手冊</translation>
     </message>
     <message>
         <source>VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the &apos;usbfs&apos; folder and files.  Please see the user manual for a more detailed explanation</source>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp
--- VirtualBox-7.0.8.wheel/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp	2023-04-25 18:52:10.909305452 -0700
+++ VirtualBox-7.0.8/src/VBox/Frontends/VirtualBox/src/globals/UIMessageCenter.cpp	2023-04-25 18:53:09.546243227 -0700
@@ -2262,7 +2262,7 @@
     /* Translations for Main.
      * Please make sure they corresponds to the strings coming from Main one-by-one symbol! */
     tr("Could not load the Host USB Proxy Service (VERR_FILE_NOT_FOUND). The service might not be installed on the host computer");
-    tr("VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the 'vboxusers' group.  Please see the user manual for a more detailed explanation");
+    tr("VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the 'wheel' group.  Please see the user manual for a more detailed explanation");
     tr("VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the 'usbfs' folder and files.  Please see the user manual for a more detailed explanation");
     tr("The USB Proxy Service has not yet been ported to this host");
     tr("Could not load the Host USB Proxy service");
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/HostDrivers/linux/loadall.sh VirtualBox-7.0.8/src/VBox/HostDrivers/linux/loadall.sh
--- VirtualBox-7.0.8.wheel/src/VBox/HostDrivers/linux/loadall.sh	2023-04-25 18:52:08.303219337 -0700
+++ VirtualBox-7.0.8/src/VBox/HostDrivers/linux/loadall.sh	2023-04-25 18:53:09.550243359 -0700
@@ -36,13 +36,13 @@
 
 TARGET=`readlink -e -- "${0}"` || exit 1 # The GNU-specific way.
 MY_DIR="${TARGET%/[!/]*}"
-MY_GROUPNAME="vboxusers"
+MY_GROUPNAME="wheel"
 
 #
-# vboxusers membership check.
+# wheel membership check.
 #
 if ! (id -Gn | grep -w ${MY_GROUPNAME}); then
-    echo "loadall.sh: you're not a member of vboxusers...";
+    echo "loadall.sh: you're not a member of wheel...";
     # Create the group.
     if ! getent group ${MY_GROUPNAME}; then
         set -e
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/VBoxCreateUSBNode.sh VirtualBox-7.0.8/src/VBox/Installer/linux/VBoxCreateUSBNode.sh
--- VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/VBoxCreateUSBNode.sh	2023-04-25 18:52:08.342220625 -0700
+++ VirtualBox-7.0.8/src/VBox/Installer/linux/VBoxCreateUSBNode.sh	2023-04-25 18:53:09.555243524 -0700
@@ -46,7 +46,7 @@
   then
       exit 0
   fi
-  case "$group" in "") group="vboxusers";; esac
+  case "$group" in "") group="wheel";; esac
   mkdir /dev/vboxusb -m 0750 2>/dev/null
   chown root:$group /dev/vboxusb 2>/dev/null
   mkdir "$devdir" -m 0750 2>/dev/null
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/debian/postinst VirtualBox-7.0.8/src/VBox/Installer/linux/debian/postinst
--- VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/debian/postinst	2023-04-25 18:52:08.342220625 -0700
+++ VirtualBox-7.0.8/src/VBox/Installer/linux/debian/postinst	2023-04-25 18:53:09.560243689 -0700
@@ -59,9 +59,9 @@
 
   # create users groups (disable with INSTALL_NO_GROUP=1 in /etc/default/virtualbox)
   if [ "$INSTALL_NO_GROUP" != "1" ]; then
-    db_input low virtualbox/group-vboxusers || true
+    db_input low virtualbox/group-wheel || true
     db_go || true
-    addgroup --system vboxusers || true
+    addgroup --system wheel || true
   fi
 
   # The starters need to be Suid root. They drop the privileges before starting
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/debian/templates VirtualBox-7.0.8/src/VBox/Installer/linux/debian/templates
--- VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/debian/templates	2023-04-25 18:52:08.341220592 -0700
+++ VirtualBox-7.0.8/src/VBox/Installer/linux/debian/templates	2023-04-25 18:53:09.564243822 -0700
@@ -1,9 +1,9 @@
-Template: virtualbox/group-vboxusers
+Template: virtualbox/group-wheel
 Type: note
-Description: Creating group 'vboxusers'
+Description: Creating group 'wheel'
  Users of VirtualBox must be members of this group. Host network interfaces
  will be assigned to this group.
-Description-de.UTF-8: Erzeuge Gruppe 'vboxusers'
+Description-de.UTF-8: Erzeuge Gruppe 'wheel'
  Nutzer von VirtualBox müssen Mitglied dieser Gruppe sein. Hostinterfaces
  werden dieser Gruppe zugewiesen.
 
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/install.sh VirtualBox-7.0.8/src/VBox/Installer/linux/install.sh
--- VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/install.sh	2023-04-25 18:52:08.340220559 -0700
+++ VirtualBox-7.0.8/src/VBox/Installer/linux/install.sh	2023-04-25 18:53:09.569243987 -0700
@@ -51,7 +51,7 @@
 CONFIG="vbox.cfg"
 CONFIG_FILES="filelist"
 DEFAULT_FILES=`pwd`/deffiles
-GROUPNAME="vboxusers"
+GROUPNAME="wheel"
 INSTALLATION_DIR="_INSTALLATION_DIR_"
 LICENSE_ACCEPTED=""
 PREV_INSTALLATION=""
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/rpm/VirtualBox.tmpl.spec VirtualBox-7.0.8/src/VBox/Installer/linux/rpm/VirtualBox.tmpl.spec
--- VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/rpm/VirtualBox.tmpl.spec	2023-04-25 18:52:08.343220658 -0700
+++ VirtualBox-7.0.8/src/VBox/Installer/linux/rpm/VirtualBox.tmpl.spec	2023-04-25 18:53:09.574244152 -0700
@@ -299,9 +299,9 @@
 # create users groups (disable with INSTALL_NO_GROUP=1 in /etc/default/virtualbox)
 if [ "$INSTALL_NO_GROUP" != "1" ]; then
   echo
-  echo "Creating group 'vboxusers'. VM users must be member of that group!"
+  echo "Creating group 'wheel'. VM users must be member of that group!"
   echo
-  groupadd -r -f vboxusers 2> /dev/null
+  groupadd -r -f wheel 2> /dev/null
 fi
 
 %if %{?rpm_mdv:1}%{!?rpm_mdv:0}
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/vboxdrv.sh VirtualBox-7.0.8/src/VBox/Installer/linux/vboxdrv.sh
--- VirtualBox-7.0.8.wheel/src/VBox/Installer/linux/vboxdrv.sh	2023-04-25 18:52:08.342220625 -0700
+++ VirtualBox-7.0.8/src/VBox/Installer/linux/vboxdrv.sh	2023-04-25 18:53:09.579244317 -0700
@@ -92,7 +92,7 @@
     GROUP=root
     DEVICE_MODE=0600
 else
-    GROUP=vboxusers
+    GROUP=wheel
     DEVICE_MODE=0660
 fi
 
@@ -555,7 +555,7 @@
     # of USB access.  The USB code checks for the existance of that path.
     if grep -q usb_device /proc/devices; then
         mkdir -p -m 0750 /dev/vboxusb 2>/dev/null
-        chown root:vboxusers /dev/vboxusb 2>/dev/null
+        chown root:wheel /dev/vboxusb 2>/dev/null
     fi
     # Remove any kernel modules left over from previously installed kernels.
     cleanup only_old
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Installer/solaris/vbox-ips.mog VirtualBox-7.0.8/src/VBox/Installer/solaris/vbox-ips.mog
--- VirtualBox-7.0.8.wheel/src/VBox/Installer/solaris/vbox-ips.mog	2023-04-25 18:52:08.343220658 -0700
+++ VirtualBox-7.0.8/src/VBox/Installer/solaris/vbox-ips.mog	2023-04-25 18:53:09.584244483 -0700
@@ -7,7 +7,7 @@
 set name=variant.opensolaris.zone value=global value=nonglobal
 set name=info.classification value=org.opensolaris.category.2008:System/Virtualization
 license LICENSE license="GPLv2 (virtualbox)"
-group groupname=vboxusers gid=86
+group groupname=wheel gid=86
 # The data-xkb was renamed to xkeyboard-config in S11.4, so adjust this when
 # we have moved our builds away from S11.3.
 depend fmri=pkg:/x11/keyboard/data-xkb type=require
@@ -62,7 +62,7 @@
 <transform file path=platform/i86pc/kernel/drv/amd64/vboxdrv$ -> emit driver name=vboxdrv perms="* 0600 root sys" perms="vboxdrvu 0666 root sys" devlink=type=ddi_pseudo;name=vboxdrv;minor=vboxdrv\t\D devlink=type=ddi_pseudo;name=vboxdrv;minor=vboxdrvu\t\M0>
 <transform file path=platform/i86pc/kernel/drv/amd64/vboxnet$ -> emit driver name=vboxnet>
 <transform file path=platform/i86pc/kernel/drv/amd64/vboxbow$ -> emit driver name=vboxbow>
-<transform file path=platform/i86pc/kernel/drv/amd64/vboxusbmon$ -> emit driver name=vboxusbmon perms="* 0660 root vboxusers" devlink=type=ddi_pseudo;name=vboxusbmon\t\D>
+<transform file path=platform/i86pc/kernel/drv/amd64/vboxusbmon$ -> emit driver name=vboxusbmon perms="* 0660 root wheel" devlink=type=ddi_pseudo;name=vboxusbmon\t\D>
 <transform file path=platform/i86pc/kernel/drv/amd64/vboxusb$ -> emit driver name=vboxusb>
 <transform file path=platform/i86pc/kernel/drv/.*$ -> set group sys>
 
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Main/nls/VirtualBoxAPI_ru.ts VirtualBox-7.0.8/src/VBox/Main/nls/VirtualBoxAPI_ru.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Main/nls/VirtualBoxAPI_ru.ts	2023-04-25 18:52:07.941207374 -0700
+++ VirtualBox-7.0.8/src/VBox/Main/nls/VirtualBoxAPI_ru.ts	2023-04-25 18:53:09.592244747 -0700
@@ -5022,8 +5022,8 @@
     </message>
     <message>
         <location filename="../src-server/HostImpl.cpp" line="3552"/>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
-        <translation>В настоящее время VirtualBox не имеет доступа к USB устройствам. Вы можете изменить это путем добавления пользователя в группу &apos;vboxusers&apos;. Пожалуйста, обратитесь к руководству пользователя за более подробными объяснениями</translation>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <translation>В настоящее время VirtualBox не имеет доступа к USB устройствам. Вы можете изменить это путем добавления пользователя в группу &apos;wheel&apos;. Пожалуйста, обратитесь к руководству пользователя за более подробными объяснениями</translation>
     </message>
     <message>
         <location filename="../src-server/HostImpl.cpp" line="3555"/>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Main/nls/VirtualBoxAPI_tr.ts VirtualBox-7.0.8/src/VBox/Main/nls/VirtualBoxAPI_tr.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Main/nls/VirtualBoxAPI_tr.ts	2023-04-25 18:52:07.940207341 -0700
+++ VirtualBox-7.0.8/src/VBox/Main/nls/VirtualBoxAPI_tr.ts	2023-04-25 18:53:09.599244978 -0700
@@ -4955,7 +4955,7 @@
     </message>
     <message>
         <location filename="../src-server/HostImpl.cpp" line="3552"/>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Main/nls/VirtualBoxAPI_xx_YY.ts VirtualBox-7.0.8/src/VBox/Main/nls/VirtualBoxAPI_xx_YY.ts
--- VirtualBox-7.0.8.wheel/src/VBox/Main/nls/VirtualBoxAPI_xx_YY.ts	2023-04-25 18:52:07.940207341 -0700
+++ VirtualBox-7.0.8/src/VBox/Main/nls/VirtualBoxAPI_xx_YY.ts	2023-04-25 18:53:09.607245243 -0700
@@ -4955,7 +4955,7 @@
     </message>
     <message>
         <location filename="../src-server/HostImpl.cpp" line="3552"/>
-        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;vboxusers&apos; group.  Please see the user manual for a more detailed explanation</source>
+        <source>VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the &apos;wheel&apos; group.  Please see the user manual for a more detailed explanation</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Main/src-client/ConsoleImpl2.cpp VirtualBox-7.0.8/src/VBox/Main/src-client/ConsoleImpl2.cpp
--- VirtualBox-7.0.8.wheel/src/VBox/Main/src-client/ConsoleImpl2.cpp	2023-04-25 18:52:07.948207606 -0700
+++ VirtualBox-7.0.8/src/VBox/Main/src-client/ConsoleImpl2.cpp	2023-04-25 18:53:09.614245474 -0700
@@ -6163,7 +6163,7 @@
                                                              "Failed to open '/dev/%s' for read/write access.  Please check the "
                                                              "permissions of that node, and that the net.link.tap.user_open "
                                                              "sysctl is set.  Either run 'chmod 0666 /dev/%s' or change the "
-                                                             "group of that node to vboxusers and make yourself a member of "
+                                                             "group of that node to wheel and make yourself a member of "
                                                              "that group.  Make sure that these changes are permanent."),
                                                              pszBridgedIfName, pszBridgedIfName);
                             default:
diff -Naur VirtualBox-7.0.8.wheel/src/VBox/Main/src-server/HostImpl.cpp VirtualBox-7.0.8/src/VBox/Main/src-server/HostImpl.cpp
--- VirtualBox-7.0.8.wheel/src/VBox/Main/src-server/HostImpl.cpp	2023-04-25 18:52:07.977208564 -0700
+++ VirtualBox-7.0.8/src/VBox/Main/src-server/HostImpl.cpp	2023-04-25 18:53:09.620245672 -0700
@@ -3610,7 +3610,7 @@
                                   tr("Could not load the Host USB Proxy Service (VERR_FILE_NOT_FOUND).  The service might not be installed on the host computer"));
             case VERR_VUSB_USB_DEVICE_PERMISSION:
                 return setWarning(E_FAIL,
-                                  tr("VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the 'vboxusers' group.  Please see the user manual for a more detailed explanation"));
+                                  tr("VirtualBox is not currently allowed to access USB devices.  You can change this by adding your user to the 'wheel' group.  Please see the user manual for a more detailed explanation"));
             case VERR_VUSB_USBFS_PERMISSION:
                 return setWarning(E_FAIL,
                                   tr("VirtualBox is not currently allowed to access USB devices.  You can change this by allowing your user to access the 'usbfs' folder and files.  Please see the user manual for a more detailed explanation"));
