From c4759c8bfcbdefbd9487a37aba9a2cf7bf8097a9 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Tue, 25 Mar 2025 21:50:02 +0800
Subject: [PATCH 1/2] install binary to bin instead of sbin

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index bffac53..7a47972 100644
--- a/Makefile
+++ b/Makefile
@@ -145,12 +145,12 @@ install: $(TARGET)
 	$(E) STRIP $(TARGET)
 	$(Q)$(STRIP) $(TARGET)
 	$(E) INSTALL $(TARGET)
-	$(Q)$(INSTALL) -D -m 0755 $(TARGET) $(PREFIX)/sbin/$(TARGET)
+	$(Q)$(INSTALL) -D -m 0755 $(TARGET) $(PREFIX)/bin/$(TARGET)
 	$(Q)$(INSTALL) -D -m 0644 iotop.8 $(PREFIX)/share/man/man8/iotop.8
 
 uninstall:
 	$(E) UNINSTALL $(TARGET)
-	$(Q)rm -f $(PREFIX)/sbin/$(TARGET)
+	$(Q)rm -f $(PREFIX)/bin/$(TARGET)
 	$(Q)rm -f $(PREFIX)/share/man/man8/iotop.8
 
 bld/.mkdir:
-- 
2.49.0

