From 1684ef249726ab9765c8ab8384cfbd51dd8a6a0d Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Tue, 25 Mar 2025 21:50:02 +0800
Subject: [PATCH 1/2] install binary to bin instead of sbin

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 14d7f75..ac04685 100644
--- a/Makefile
+++ b/Makefile
@@ -36,7 +36,7 @@ CFLAGS+=-fanalyzer
 endif
 
 PREFIX?=$(DESTDIR)/usr
-BINDIR?=$(PREFIX)/sbin
+BINDIR?=$(PREFIX)/bin
 INSTALL?=install
 STRIP?=strip
 
-- 
2.49.0

