From 869061812a2e983c4a9ba8ff46b97cee961f6937 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Tue, 25 Mar 2025 21:50:14 +0800
Subject: [PATCH 2/2] do not strip during compilation

---
 Makefile | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Makefile b/Makefile
index ac04685..148f64d 100644
--- a/Makefile
+++ b/Makefile
@@ -143,8 +143,6 @@ clean:
 	$(Q)rm -rf ./bld $(TARGET)
 
 install: $(TARGET)
-	$(E) STRIP $(TARGET)
-	$(Q)$(STRIP) $(TARGET)
 	$(E) INSTALL $(TARGET)
 	$(Q)$(INSTALL) -D -m 0755 $(TARGET) $(BINDIR)/$(TARGET)
 	$(Q)$(INSTALL) -D -m 0644 iotop.8 $(PREFIX)/share/man/man8/iotop.8
-- 
2.49.0

