From 53dc9022b54212b49e9e236de5bccb681b9221c1 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Tue, 25 Mar 2025 21:50:14 +0800
Subject: [PATCH 2/2] do not strip during compilation

---
 Makefile | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/Makefile b/Makefile
index 7a47972..89abe01 100644
--- a/Makefile
+++ b/Makefile
@@ -37,7 +37,6 @@ endif
 
 PREFIX?=$(DESTDIR)/usr
 INSTALL?=install
-STRIP?=strip
 
 PKG_CONFIG?=pkg-config
 NCCC?=$(shell $(PKG_CONFIG) --cflags ncursesw)
@@ -142,8 +141,6 @@ clean:
 	$(Q)rm -rf ./bld $(TARGET)
 
 install: $(TARGET)
-	$(E) STRIP $(TARGET)
-	$(Q)$(STRIP) $(TARGET)
 	$(E) INSTALL $(TARGET)
 	$(Q)$(INSTALL) -D -m 0755 $(TARGET) $(PREFIX)/bin/$(TARGET)
 	$(Q)$(INSTALL) -D -m 0644 iotop.8 $(PREFIX)/share/man/man8/iotop.8
-- 
2.49.0

