diff --color -aur nixnote2-2.1.6/src/html/enmlformatter.cpp nixnote2-2.1.6.aosc/src/html/enmlformatter.cpp
--- nixnote2-2.1.6/src/html/enmlformatter.cpp	2020-05-05 16:19:52.000000000 +0800
+++ nixnote2-2.1.6.aosc/src/html/enmlformatter.cpp	2021-12-14 00:28:16.875208611 +0800
@@ -34,13 +34,8 @@
 #include "src/logger/qslog.h"
 #include "src/utilities/NixnoteStringUtils.h"
 
-#ifdef Q_OS_MACOS
 #include <tidy.h>
 #include <tidybuffio.h>
-#else
-#include <tidy/tidy.h>
-#include <tidy/tidybuffio.h>
-#endif
 
 #define ENML_MODULE_LOGPREFIX "enml-cleanup: "
 
@@ -916,4 +911,4 @@
 void EnmlFormatter::setContent(QString &contentStr) {
     this->content.clear();
     this->content.append(contentStr.toUtf8());
-}
\ 文件尾没有换行符
+}
diff --color -aur nixnote2-2.1.6/src/main.cpp nixnote2-2.1.6.aosc/src/main.cpp
--- nixnote2-2.1.6/src/main.cpp	2020-05-05 16:19:52.000000000 +0800
+++ nixnote2-2.1.6.aosc/src/main.cpp	2021-12-14 00:26:11.283139886 +0800
@@ -46,13 +46,8 @@
 
 #include "src/application.h"
 
-#ifdef Q_OS_MACOS
 #include <tidy.h>
 #include <tidybuffio.h>
-#else
-#include <tidy/tidy.h>
-#include <tidy/tidybuffio.h>
-#endif
 
 
 NixNote *w;
