From 66a98316e39542489e3afc4bb19467b60769acbe Mon Sep 17 00:00:00 2001
From: ShellWen <me@shellwen.com>
Date: Sat, 19 Jul 2025 16:55:25 +0800
Subject: [PATCH 1/1] use system CFLAGS env var

---
 autobahn/nvx/_utf8validator.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/autobahn/nvx/_utf8validator.py b/autobahn/nvx/_utf8validator.py
index 008ec9e4..55331740 100644
--- a/autobahn/nvx/_utf8validator.py
+++ b/autobahn/nvx/_utf8validator.py
@@ -51,11 +51,12 @@ else:
 
 with open(os.path.join(os.path.dirname(__file__), '_utf8validator.c')) as fd:
     c_source = fd.read()
+    cflags = os.environ.get('CFLAGS', '-std=c99 -Wall -Wno-strict-prototypes -O3')
     ffi.set_source(
         "_nvx_utf8validator",
         c_source,
         libraries=[],
-        extra_compile_args=['-std=c99', '-Wall', '-Wno-strict-prototypes', '-O3', '-march=native'],
+        extra_compile_args=cflags.split(" "),
         optional=optional
     )
 
-- 
2.50.0

