From 59d9bcd468aab0d97d763595fd4e934044dc7590 Mon Sep 17 00:00:00 2001
From: Rob Browning <rlb@defaultvalue.org>
Date: Sat, 21 Jul 2018 14:30:35 -0500
Subject: gc.test: after-gc-hook - mark unresolved on failure for mips(el)

Bug: https://debbugs.gnu.org/31776
Bug-Debian: https://bugs.debian.org/900652
---
 test-suite/tests/gc.test | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/test-suite/tests/gc.test b/test-suite/tests/gc.test
index 04f353984..69f48ebbf 100644
--- a/test-suite/tests/gc.test
+++ b/test-suite/tests/gc.test
@@ -67,7 +67,10 @@
       (add-hook! after-gc-hook thunk)
       (gc)
       (remove-hook! after-gc-hook thunk)
-      foo))
+      (let ((cpu (substring %host-type 0 (string-index %host-type #\-))))
+        (cond ((string=? cpu "mips") (maybe-gc-flakiness foo))
+              ((string=? cpu "mipsel") (maybe-gc-flakiness foo))
+              (else foo)))))
 
   (pass-if "Unused modules are removed"
     (let* ((guard (make-guardian))
