diff -Nur orbit2-2.14.3/src/services/name/Makefile.am orbit2-2.14.3.new/src/services/name/Makefile.am
--- orbit2-2.14.3/src/services/name/Makefile.am	2006-08-30 00:33:12.000000000 +0200
+++ orbit2-2.14.3.new/src/services/name/Makefile.am	2006-10-24 15:58:44.000000000 +0200
@@ -8,7 +8,7 @@
 
 lib_LTLIBRARIES=libORBitCosNaming-2.la
 
-noinst_PROGRAMS = name-client-2 orbit-name-server-2
+bin_PROGRAMS = name-client-2 orbit-name-server-2
 
 lib_LIBRARIES = libname-server-2.a
 
