Author: Roland Rosenfeld <roland@debian.org>
Description: fix locale settings.

--- a/fig2dev/dev/genps.c
+++ b/fig2dev/dev/genps.c
@@ -42,6 +42,7 @@
 #include "bound.h"
 #include "psencode.h"
 #include "psfonts.h"
+#include <locale.h>
 
 /* for the xpm package */
 #ifdef USE_XPM
--- a/fig2dev/fig2dev.c
+++ b/fig2dev/fig2dev.c
@@ -24,6 +24,7 @@
 #include <sys/types.h>
 #endif
 #include <sys/file.h>
+#include <locale.h>
 #include "fig2dev.h"
 #include "alloc.h"
 #include "object.h"
@@ -380,6 +381,7 @@
 	F_compound	objects;
 	int		status;
 
+	setlocale(LC_CTYPE, "");
 #ifdef HAVE_SETMODE
 	setmode(1,O_BINARY); /* stdout is binary */
 #endif
