Author: Roland Rosenfeld <roland@debian.org>
Description: Fix generation of postscript manual.

--- a/doc/manual/Makefile
+++ b/doc/manual/Makefile
@@ -2,13 +2,20 @@
 # TransFig makefile
 #
 
-all: trans.tex 
+all: trans.tex manual.ps
 
 # translation into latex
 
 trans.tex: trans.fig Makefile
 	../../fig2dev/fig2dev -L latex trans.fig > trans.tex
 clean::
-	rm -f trans.tex
+	rm -f trans.tex *.aux *.log *.dvi *.ps
 
 cleandir::
+
+manual.ps: manual.dvi
+	dvips -o $@ $^
+
+manual.dvi: manual.tex trans.tex
+	latex manual.tex
+	latex manual.tex
--- a/doc/manual/manual.tex
+++ b/doc/manual/manual.tex
@@ -6,7 +6,7 @@
 % Document Type: LaTeX
 %
 \input{setup12.tex}
-\input{transfig}
+%\input{transfig}
 %
 \newcommand{\version}{2.1.5}
 %
