Author: Roland Rosenfeld <roland@debian.org>
Description: Remove unnecessary dependencies on libz and libX11.

--- a/fig2dev/Imakefile
+++ b/fig2dev/Imakefile
@@ -154,12 +154,12 @@
 
 #ifdef USEPNG
 DUSEPNG = -DUSE_PNG
-PNGLIBS = -L$(PNGLIBDIR) -lpng -L$(ZLIBDIR) -lz
+PNGLIBS = -L$(PNGLIBDIR) -lpng
 #endif /* USEPNG */
 
 #ifdef USEXPM
 DUSEXPM = -DUSE_XPM
-XPMLIBS = -L$(XPMLIBDIR) -lXpm -lX11
+XPMLIBS = -L$(XPMLIBDIR) -lXpm
 #endif /* USEXPM */
 
 #if defined(NOSTRSTR) 
