From 73263f21ae4527b0337bad7b2e5a0142ba456526 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 25 Nov 2024 15:09:30 +0800
Subject: [PATCH] libiberty: also generate a PIC version of libiberty.a

---
 libiberty/configure | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/libiberty/configure b/libiberty/configure
index 5c69fee56c1..bc335693b5c 100755
--- a/libiberty/configure
+++ b/libiberty/configure
@@ -5410,10 +5410,6 @@ if [ "${enable_host_shared}" = "yes" ] || [ "${enable_host_pie}" = "yes" ]; then
   shared=yes
 fi
 
-if [ "${shared}" != "yes" ]; then
-  PICFLAG=
-fi
-
 
 NOASANFLAG=
 case " ${CFLAGS} " in
-- 
2.49.0

