diff --git a/src/opt/apps/@PKG_NAME@/files/amber-ce-tools/container-init/init.sh b/src/opt/apps/@PKG_NAME@/files/amber-ce-tools/container-init/init.sh
--- a/src/opt/apps/@PKG_NAME@/files/amber-ce-tools/container-init/init.sh
+++ b/src/opt/apps/@PKG_NAME@/files/amber-ce-tools/container-init/init.sh
@@ -135,3 +135,9 @@ ln -sfv ../../usr/share/icons/ .
 rm -vf ./mime
 update-desktop-database /usr/share/applications || true
 update-mime-database /usr/share/mime || true
+
+# Write locale-gen
+cat << EOF > /etc/locale.gen
+zh_CN.UTF-8 UTF-8
+en_US.UTF-8 UTF-8
+EOF

diff --git a/src/opt/apps/@PKG_NAME@/files/bin/ace-init b/src/opt/apps/@PKG_NAME@/files/bin/ace-init
--- a/src/opt/apps/@PKG_NAME@/files/bin/ace-init
+++ b/src/opt/apps/@PKG_NAME@/files/bin/ace-init
@@ -83,7 +83,7 @@ bookworm-run bash /amber-ce-tools/container-init/init.sh
 rm $chrootEnvPath/etc/localtime
 cp $(realpath /etc/localtime) $chrootEnvPath/etc/localtime
 chmod 777 $chrootEnvPath/etc/localtime 
-bookworm-run cp /host/etc/locale.gen /etc/locale.gen && locale-gen
+# bookworm-run cp /host/etc/locale.gen /etc/locale.gen && locale-gen
 bookworm-run touch /finish.flag
 bookworm-run apt clean
 bookworm-run chown -R $(get_current_user)  /usr/lib/locale/
