diff -Naur skanpage-22.04.3/org.kde.skanpage.desktop skanpage-22.04.3.desktop/org.kde.skanpage.desktop
--- skanpage-22.04.3/org.kde.skanpage.desktop	2022-07-03 03:32:33.000000000 +0000
+++ skanpage-22.04.3.desktop/org.kde.skanpage.desktop	2022-09-14 01:18:52.583365513 +0000
@@ -2,7 +2,7 @@
 #
 # SPDX-License-Identifier: CC0-1.0
 [Desktop Entry]
-Categories=Qt;KDE;Graphics;Scanning;
+Categories=Qt;KDE;Office;Scanning;
 Comment=Scan and save images
 Comment[ar]=امسح الصّور ضوئيًّا واحفظها
 Comment[ca]=Escaneja i desa imatges
