--- a/src/corelib/io/qfilesystemengine_unix.cpp	2018-04-05 11:35:55.799968461 -0600
+++ b/src/corelib/io/qfilesystemengine_unix.cpp	2018-04-05 11:39:06.260236237 -0600
@@ -97,6 +97,9 @@
 #  define FICLONE       _IOW(0x94, 9, int)
 #endif
 
+#undef SYS_statx
+#undef STATX_BASIC_STATS
+
 #  if !QT_CONFIG(renameat2) && defined(SYS_renameat2)
 static int renameat2(int oldfd, const char *oldpath, int newfd, const char *newpath, unsigned flags)
 { return syscall(SYS_renameat2, oldfd, oldpath, newfd, newpath, flags); }
@@ -109,7 +112,7 @@
 #  endif
 #endif
 
-#ifndef STATX_BASIC_STATS
+#ifndef STATX_ALL
 struct statx { mode_t stx_mode; };
 #endif
 
