--- qtbase-everywhere-src-5.10.1/src/corelib/io/qfilesystemengine_unix.cpp  2018-02-08 18:24:48.000000000 +0000
+++ qtbase-everywhere-src-5.10.1.working/src/corelib/io/qfilesystemengine_unix.cpp   2018-04-05 16:26:24.765025874 +0000
@@ -91,25 +91,20 @@
 #  include <sys/syscall.h>
 #  include <sys/sendfile.h>
 #  include <linux/fs.h>
+#  include <linux/stat.h>
 
 // in case linux/fs.h is too old and doesn't define it:
 #ifndef FICLONE
 #  define FICLONE       _IOW(0x94, 9, int)
 #endif
 
-#  if !QT_CONFIG(renameat2) && defined(SYS_renameat2)
-static int renameat2(int oldfd, const char *oldpath, int newfd, const char *newpath, unsigned flags)
-{ return syscall(SYS_renameat2, oldfd, oldpath, newfd, newpath, flags); }
-#  endif
+#undef SYS_renameat2 105
+#undef SYS_statx 106
+#undef STATX_BASIC_STATS
 
-#  if !QT_CONFIG(statx) && defined(SYS_statx) && QT_HAS_INCLUDE(<linux/stat.h>)
-#    include <linux/stat.h>
-static int statx(int dirfd, const char *pathname, int flag, unsigned mask, struct statx *statxbuf)
-{ return syscall(SYS_statx, dirfd, pathname, flag, mask, statxbuf); }
-#  endif
 #endif
 
-#ifndef STATX_BASIC_STATS
+#ifndef STATX_ALL
 struct statx { mode_t stx_mode; };
 #endif
 

