--- qt-everywhere-opensource-src-5.9.1/qtspeech/src/plugins/tts/speechdispatcher/qtexttospeech_speechd.cpp	2017-05-10 17:07:32.000000000 +0000
+++ qt-everywhere-opensource-src-5.9.1.qttext/qtspeech/src/plugins/tts/speechdispatcher/qtexttospeech_speechd.cpp	2017-07-19 01:32:15.068971740 +0000
@@ -40,10 +40,6 @@
 #include <qdebug.h>
 #include <speech-dispatcher/libspeechd.h>
 
-#if LIBSPEECHD_MAJOR_VERSION > 0 || LIBSPEECHD_MINOR_VERSION >= 9
-  #define HAVE_SPD_090
-#endif
-
 QT_BEGIN_NAMESPACE
 
 QString dummyModule = QStringLiteral("dummy");
