--- qt-everywhere-opensource-src-5.8.0/qtspeech/src/plugins/tts/speechdispatcher/qtexttospeech_speechd.cpp	2016-10-31 09:30:18.000000000 -0500
+++ qt-everywhere-opensource-src-5.8.0.tts/qtspeech/src/plugins/tts/speechdispatcher/qtexttospeech_speechd.cpp	2017-02-01 14:07:12.285668000 -0600
@@ -40,10 +40,6 @@
 #include <qdebug.h>
 #include <speech-dispatcher/libspeechd.h>
 
-#if LIBSPEECHD_MAJOR_VERSION > 0 || LIBSPEECHD_MINOR_VERSION >= 9
-  #define HAVE_SPD_090
-#endif
-
 QT_BEGIN_NAMESPACE
 
 QString dummyModule = QStringLiteral("dummy");
