diff -Naur ngs-1.3.0.install-fix/ngs-sdk/Makefile.install ngs-1.3.0/ngs-sdk/Makefile.install
--- ngs-1.3.0.install-fix/ngs-sdk/Makefile.install	2016-12-08 18:50:56.789385438 -0600
+++ ngs-1.3.0/ngs-sdk/Makefile.install	2016-12-08 18:52:19.512809470 -0600
@@ -73,7 +73,6 @@
 ifeq (true, $(LINUX_ROOT))
 	@ # install symlink to includes 
 	@ echo "Updating $(INCLUDE_SYMLINK)"
-	@ rm -f $(INCLUDE_SYMLINK)
 	@ ln -s $(INST_INCDIR)/ngs $(INCLUDE_SYMLINK)
 	@ #
 	@ echo "Updating $(PROFILE_FILE).[c]sh"
