diff -Naur ngs-2.10.2/ngs-bam/setup/konfigure.perl ngs-2.10.2.aarch64/ngs-bam/setup/konfigure.perl
--- ngs-2.10.2/ngs-bam/setup/konfigure.perl	2020-01-15 15:17:22.000000000 -0600
+++ ngs-2.10.2.aarch64/ngs-bam/setup/konfigure.perl	2020-05-30 12:47:27.053893444 -0500
@@ -223,7 +223,7 @@
 
 print "checking machine architecture... " unless ($AUTORUN);
 println $MARCH unless ($AUTORUN);
-unless ($MARCH =~ /x86_64/i || $MARCH =~ /i?86/i) {
+unless ($MARCH =~ /x86_64/i || $MARCH =~ /i?86/i || $MARCH =~ /aarch64/i) {
     println "configure: error: unsupported architecture '$OSTYPE'";
     exit 1;
 }
@@ -314,6 +314,8 @@
     $BITS = '32_64';
 } elsif ($MARCH =~ /i?86/i) {
     $BITS = 32;
+} elsif ($MARCH =~ /aarch64/i) {
+    $BITS = 64;
 } else {
     die "unrecognized Architecture '$ARCH'";
 }
diff -Naur ngs-2.10.2/ngs-java/setup/konfigure.perl ngs-2.10.2.aarch64/ngs-java/setup/konfigure.perl
--- ngs-2.10.2/ngs-java/setup/konfigure.perl	2020-01-15 15:17:22.000000000 -0600
+++ ngs-2.10.2.aarch64/ngs-java/setup/konfigure.perl	2020-05-30 12:48:01.340898662 -0500
@@ -223,7 +223,7 @@
 
 print "checking machine architecture... " unless ($AUTORUN);
 println $MARCH unless ($AUTORUN);
-unless ($MARCH =~ /x86_64/i || $MARCH =~ /i?86/i) {
+unless ($MARCH =~ /x86_64/i || $MARCH =~ /i?86/i || $MARCH =~ /aarch64/i) {
     println "configure: error: unsupported architecture '$OSTYPE'";
     exit 1;
 }
@@ -314,6 +314,8 @@
     $BITS = '32_64';
 } elsif ($MARCH =~ /i?86/i) {
     $BITS = 32;
+} elsif ($MARCH =~ /aarch64/i) {
+    $BITS = 64;
 } else {
     die "unrecognized Architecture '$ARCH'";
 }
diff -Naur ngs-2.10.2/ngs-python/setup/konfigure.perl ngs-2.10.2.aarch64/ngs-python/setup/konfigure.perl
--- ngs-2.10.2/ngs-python/setup/konfigure.perl	2020-01-15 15:17:22.000000000 -0600
+++ ngs-2.10.2.aarch64/ngs-python/setup/konfigure.perl	2020-05-30 12:48:44.611140395 -0500
@@ -223,7 +223,7 @@
 
 print "checking machine architecture... " unless ($AUTORUN);
 println $MARCH unless ($AUTORUN);
-unless ($MARCH =~ /x86_64/i || $MARCH =~ /i?86/i) {
+unless ($MARCH =~ /x86_64/i || $MARCH =~ /i?86/i || $MARCH =~ /aarch64/i) {
     println "configure: error: unsupported architecture '$OSTYPE'";
     exit 1;
 }
@@ -314,6 +314,8 @@
     $BITS = '32_64';
 } elsif ($MARCH =~ /i?86/i) {
     $BITS = 32;
+} elsif ($MARCH =~ /aarch64/i) {
+    $BITS = 64;
 } else {
     die "unrecognized Architecture '$ARCH'";
 }
diff -Naur ngs-2.10.2/ngs-sdk/setup/konfigure.perl ngs-2.10.2.aarch64/ngs-sdk/setup/konfigure.perl
--- ngs-2.10.2/ngs-sdk/setup/konfigure.perl	2020-01-15 15:17:22.000000000 -0600
+++ ngs-2.10.2.aarch64/ngs-sdk/setup/konfigure.perl	2020-05-30 12:49:17.969078641 -0500
@@ -223,7 +223,7 @@
 
 print "checking machine architecture... " unless ($AUTORUN);
 println $MARCH unless ($AUTORUN);
-unless ($MARCH =~ /x86_64/i || $MARCH =~ /i?86/i) {
+unless ($MARCH =~ /x86_64/i || $MARCH =~ /i?86/i || $MARCH =~ /aarch64/i) {
     println "configure: error: unsupported architecture '$OSTYPE'";
     exit 1;
 }
@@ -314,6 +314,8 @@
     $BITS = '32_64';
 } elsif ($MARCH =~ /i?86/i) {
     $BITS = 32;
+} elsif ($MARCH =~ /aarch64/i) {
+    $BITS = 64;
 } else {
     die "unrecognized Architecture '$ARCH'";
 }
