--- a/configure.in	2012-01-14 20:44:08.000000000 -0800
+++ b/configure.in	2023-10-27 17:32:50.406931629 -0700
@@ -94,6 +94,7 @@
 dnl
 dnl Get the cflags and libraries from the freetype-config script
 dnl
+PKG_CHECK_MODULES(FreeType2, [freetype2], [ft2_pc=yes])
 AC_ARG_WITH(freetype-prefix,[  --with-freetype-prefix=PFX   Prefix where FREETYPE is 
 installed (optional)],
             freetype_prefix="$withval", freetype_prefix="")
@@ -113,7 +114,11 @@
         FREETYPE_CONFIG=$freetype_prefix/bin/freetype-config
      fi
 fi
-AC_PATH_PROG(FREETYPE_CONFIG, freetype-config, no)
+if test "x$ft2_pc" = xyes; then
+     FREETYPE_CONFIG="pkg-config freetype2"
+else
+     AC_PATH_PROG(FREETYPE_CONFIG, freetype-config, no)
+fi
 no_freetype=""
 if test "$FREETYPE_CONFIG" = "no" ; then
     AC_MSG_ERROR([
