From be46a928a1e15a0cc7e7a9168c18702d5acd200f Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Mon, 26 May 2025 10:42:41 +0800
Subject: [PATCH 1/4] AOSCOS: Hide from KDE

---
 config/firewall-config.desktop.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/config/firewall-config.desktop.in b/config/firewall-config.desktop.in
index f8bcb81fccb3..f66e3cf08289 100644
--- a/config/firewall-config.desktop.in
+++ b/config/firewall-config.desktop.in
@@ -9,3 +9,4 @@ Exec=/usr/bin/firewall-config
 Type=Application
 StartupNotify=true
 Terminal=false
+NotShowIn=KDE;

base-commit: 72e1808a38629cc9da518f0ba8a5ebf064b8220a
-- 
2.49.0

