From c80d8f12797d155cb9f6eabe6cefda39f388410a Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Mon, 26 May 2025 10:42:58 +0800
Subject: [PATCH 2/4] AOSCOS: Introduce AOSC OS default zone

---
 config/firewalld.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/firewalld.conf b/config/firewalld.conf
index 800b255c77e0..c12d932b521d 100644
--- a/config/firewalld.conf
+++ b/config/firewalld.conf
@@ -3,7 +3,7 @@
 # default zone
 # The default zone used if an empty zone string is used.
 # Default: public
-DefaultZone=public
+DefaultZone=default
 
 # Clean up on exit
 # If set to no or false the firewall configuration will not get cleaned up
-- 
2.49.0

