diff -Naur firewalld-1.2.0/config/firewalld.conf firewalld-1.2.0.conf/config/firewalld.conf
--- firewalld-1.2.0/config/firewalld.conf	2022-07-01 13:38:03.000000000 +0000
+++ firewalld-1.2.0.conf/config/firewalld.conf	2022-09-15 22:12:50.604428522 +0000
@@ -3,7 +3,7 @@
 # default zone
 # The default zone used if an empty zone string is used.
 # Default: public
-DefaultZone=public
+DefaultZone=default
 
 # Clean up on exit
 # If set to no or false the firewall configuration will not get cleaned up
