diff -Naur gstreamer.old/subprojects/gst-plugins-bad/ext/fluidsynth/gstfluiddec.c gstreamer/subprojects/gst-plugins-bad/ext/fluidsynth/gstfluiddec.c
--- gstreamer.old/subprojects/gst-plugins-bad/ext/fluidsynth/gstfluiddec.c	2024-10-07 18:11:09.362975197 +0800
+++ gstreamer/subprojects/gst-plugins-bad/ext/fluidsynth/gstfluiddec.c	2024-10-07 18:10:48.896756153 +0800
@@ -73,7 +73,7 @@
   LAST_SIGNAL
 };
 
-#define DEFAULT_SOUNDFONT       NULL
+#define DEFAULT_SOUNDFONT       "/usr/share/soundfonts/default.sf2"
 #define DEFAULT_SYNTH_CHORUS    TRUE
 #define DEFAULT_SYNTH_REVERB    TRUE
 #define DEFAULT_SYNTH_GAIN      0.2
