diff --git a/subprojects/gst-devtools/dots-viewer/meson.build b/subprojects/gst-devtools/dots-viewer/meson.build
index 9802fd5..1eca920 100644
--- a/subprojects/gst-devtools/dots-viewer/meson.build
+++ b/subprojects/gst-devtools/dots-viewer/meson.build
@@ -11,6 +11,8 @@ endif
 
 cargo_wrapper = find_program('cargo_wrapper.py')
 extra_env = {'RUSTC': ' '.join(rustc.cmd_array())}
+rustflags_cmd = run_command('sh', '-c', 'echo $RUSTFLAGS')
+extra_env += {'RUSTFLAGS': rustflags_cmd.stdout().strip() }
 
 system = host_machine.system()
 exe_suffix = ''
