--- a/configure.in.orig	2017-06-10 14:12:27.803102198 +0800
+++ b/configure.in	2017-06-10 14:18:28.137756673 +0800
@@ -2119,9 +2119,6 @@
 esac
 
 case "$target" in
-mips*-*)
-     AC_DEFINE(JS_NUNBOX32)
-     ;;
 *)
     if test "$HAVE_64BIT_BUILD" ; then
         AC_DEFINE(JS_PUNBOX64)
--- a/configure.orig	2017-06-10 14:33:44.408818535 +0800
+++ b/configure	2017-06-10 14:34:16.853808877 +0800
@@ -8493,15 +8493,6 @@
 esac
 
 case "$target" in
-mips*-*)
-     cat >> confdefs.pytmp <<\EOF
-    (''' JS_NUNBOX32 ''', ' 1 ')
-EOF
-cat >> confdefs.h <<\EOF
-#define JS_NUNBOX32 1
-EOF
-
-     ;;
 *)
     if test "$HAVE_64BIT_BUILD" ; then
         cat >> confdefs.pytmp <<\EOF
