From 8399155243521e1a792c921d453e8834ce366bfb Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Sat, 21 Aug 2021 22:25:16 -0500
Subject: [PATCH 3/3] auto sink: try pulse first

Signed-off-by: Tianhao Chai <cth451@gmail.com>
---
 akode/lib/auto_sink.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/akode/lib/auto_sink.cpp b/akode/lib/auto_sink.cpp
index bee17b1..5cf6295 100644
--- a/akode/lib/auto_sink.cpp
+++ b/akode/lib/auto_sink.cpp
@@ -64,9 +64,8 @@ AutoSink::~AutoSink()
 
 bool AutoSink::open()
 {
-    // Try Polypaudio
-    if (getenv("PULSE_SERVER"))
-        if (m_data->tryOpen("pulse")) return true;
+    // Try Pulseaudio
+    if (m_data->tryOpen("pulse")) return true;
     // Try Jack
     if (m_data->tryOpen("jack")) return true;
     // Try ALSA
-- 
2.30.2

