diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2016-09-18 01:39:23.000000000 -0500
+++ b/configure.ac	2016-09-25 00:52:48.453752862 -0500
@@ -329,7 +329,7 @@
 else
         AC_MSG_RESULT(no)
 	AC_MSG_CHECKING(if libvala is available)
-	m4_foreach([VERSION], [[0.18], [0.20], [0.22], [0.24], [0.26], [0.28], [0.30], [0.32]],
+	m4_foreach([VERSION], [[0.18], [0.20], [0.22], [0.24], [0.26], [0.28], [0.30], [0.32], [0.34]],
 		[PKG_CHECK_EXISTS([ libvala-VERSION ],
 			 [ valaver="-VERSION" ])
 			 ])
