--- game-music-emu/gme/Data_Reader.cpp.old	2015-09-19 18:17:13.210080507 +0200
+++ game-music-emu/gme/Data_Reader.cpp	2015-09-19 18:19:23.034944871 +0200
@@ -290,13 +290,13 @@
 
 long Gzip_File_Reader::size() const { return size_; }
 
-long Gzip_File_Reader::read_avail( void* p, long s ) { return gzread( file_, p, s ); }
+long Gzip_File_Reader::read_avail( void* p, long s ) { return gzread( (gzFile) file_, p, s ); }
 
-long Gzip_File_Reader::tell() const { return gztell( file_ ); }
+long Gzip_File_Reader::tell() const { return gztell( (gzFile) file_ ); }
 
 blargg_err_t Gzip_File_Reader::seek( long n )
 {
-	if ( gzseek( file_, n, SEEK_SET ) >= 0 )
+	if ( gzseek( (gzFile) file_, n, SEEK_SET ) >= 0 )
 		return 0;
 	if ( n > size_ )
 		return eof_error;
@@ -307,7 +307,7 @@
 {
 	if ( file_ )
 	{
-		gzclose( file_ );
+		gzclose( (gzFile) file_ );
 		file_ = 0;
 	}
 }
