diff --git a/src/options.h b/src/options.h
index eec6f17e..090218ce 100644
--- a/src/options.h
+++ b/src/options.h
@@ -1,9 +1,10 @@
 #pragma once
 
+#include <string>
 #include <unordered_map>
 
 std::unordered_map<std::string, std::string> ParseOptions(int argc,
                                                           char** argv);
 
 bool HasOption(const std::unordered_map<std::string, std::string>& options,
-               const std::string& option);
\ No newline at end of file
+               const std::string& option);
