From c48b291a28ae4a9118b958dbef979f35caed6e00 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 17 Nov 2025 12:30:19 +0800
Subject: [PATCH 3/8] AOSCOS: fix(dkms.conf): drop unused options

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 dkms.conf | 2 --
 1 file changed, 2 deletions(-)

diff --git a/dkms.conf b/dkms.conf
index 0a0ce16..080d376 100644
--- a/dkms.conf
+++ b/dkms.conf
@@ -1,8 +1,6 @@
 PACKAGE_NAME="arise"
 PACKAGE_VERSION=25.00.46
 AUTOINSTALL="yes"
-#REMAKE_INITRD="yes"
-SKIP_IMMD_MODPROBE=1
 
 BUILT_MODULE_NAME[0]="arise"
 BUILT_MODULE_LOCATION[0]=""
-- 
2.51.1

