From 1087f3c89f780b8c472f14bf03c884895f245197 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Sat, 5 Oct 2024 03:56:14 +0800
Subject: [PATCH 2/2] Ignore case when parse /proc/cpuinfo

---
 platform/linuxbsd/os_linuxbsd.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/platform/linuxbsd/os_linuxbsd.cpp b/platform/linuxbsd/os_linuxbsd.cpp
index 6355562feb..25902bf1be 100644
--- a/platform/linuxbsd/os_linuxbsd.cpp
+++ b/platform/linuxbsd/os_linuxbsd.cpp
@@ -184,7 +184,7 @@ String OS_LinuxBSD::get_processor_name() const {
 
 	while (!f->eof_reached()) {
 		const String line = f->get_line();
-		if (line.contains("model name")) {
+		if (line.to_lower().contains("model name")) {
 			return line.split(":")[1].strip_edges();
 		}
 	}
-- 
2.46.2

