From bdae84f744b560eee2bd398e55162119c1714277 Mon Sep 17 00:00:00 2001
From: Student Main <studentmain@aosc.io>
Date: Thu, 19 Mar 2026 05:54:38 +0800
Subject: [PATCH 3/3] Allow mono module build on RISC-V

---
 modules/mono/config.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/modules/mono/config.py b/modules/mono/config.py
index 0573fc662a..24d028a9d0 100644
--- a/modules/mono/config.py
+++ b/modules/mono/config.py
@@ -1,7 +1,4 @@
 def can_build(env, platform):
-    if env["arch"].startswith("rv"):
-        return False
-
     if env.editor_build:
         env.module_add_dependencies("mono", ["regex"])
 
-- 
2.52.0

