diff -up lightdm-1.18.0/configure.ac.qtchooser lightdm-1.18.0/configure.ac
--- lightdm-1.18.0/configure.ac.qtchooser	2016-03-22 21:12:54.000000000 -0500
+++ lightdm-1.18.0/configure.ac	2016-04-02 15:23:04.781774155 -0500
@@ -116,9 +116,9 @@ if test x"$enable_liblightdm_qt5" != "xn
         Qt5Gui
     ],
     [compile_liblightdm_qt5=yes
-     AC_CHECK_TOOLS(MOC5, [moc])
+     AC_CHECK_TOOLS(MOC5, [moc-qt5 moc])
      if test x"$(readlink $(which $MOC5))" = xqtchooser; then
-         MOC5="$MOC5 --qt=qt5"
+         MOC5="$MOC5 --qt=5"
      fi
     ],
     [if test "x$enable_liblightdm_qt5" != xauto; then
diff -up lightdm-1.18.0/configure.qtchooser lightdm-1.18.0/configure
--- lightdm-1.18.0/configure.qtchooser	2016-03-22 21:28:47.000000000 -0500
+++ lightdm-1.18.0/configure	2016-04-02 15:24:31.062312669 -0500
@@ -17693,7 +17693,7 @@ esac
 fi
 
      if test x"$(readlink $(which $MOC4))" = xqtchooser; then
-         MOC4="$MOC4 --qt=qt4"
+         MOC4="$MOC4 --qt=4"
      fi
 
 fi
@@ -17875,7 +17875,7 @@ fi
 fi
 if test -z "$MOC5"; then
   ac_ct_MOC5=$MOC5
-  for ac_prog in moc
+  for ac_prog in moc-qt5 moc
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
@@ -17931,7 +17931,7 @@ esac
 fi
 
      if test x"$(readlink $(which $MOC5))" = xqtchooser; then
-         MOC5="$MOC5 --qt=qt5"
+         MOC5="$MOC5 --qt=5"
      fi
 
 fi
