diff -Naur lightdm-1.30.0/data/lightdm.conf lightdm-1.30.0.aosc/data/lightdm.conf
--- lightdm-1.30.0/data/lightdm.conf	2018-09-05 01:33:31.000000000 +0000
+++ lightdm-1.30.0.aosc/data/lightdm.conf	2019-10-05 18:55:58.400850959 +0000
@@ -22,7 +22,7 @@
 #start-default-seat=true
 #greeter-user=lightdm
 #minimum-display-number=0
-#minimum-vt=7
+minimum-vt=1
 #lock-memory=true
 #user-authority-in-system-dir=false
 #guest-account-script=guest-account
diff -Naur lightdm-1.30.0/data/users.conf lightdm-1.30.0.aosc/data/users.conf
--- lightdm-1.30.0/data/users.conf	2018-09-05 01:33:31.000000000 +0000
+++ lightdm-1.30.0.aosc/data/users.conf	2019-10-05 18:56:57.136577091 +0000
@@ -9,6 +9,6 @@
 # hidden-shells = Shells that indicate a user cannot login
 #
 [UserList]
-minimum-uid=500
+minimum-uid=1000
 hidden-users=nobody nobody4 noaccess
 hidden-shells=/bin/false /usr/sbin/nologin /sbin/nologin
