diff -Naur a/data/lightdm.conf b/data/lightdm.conf
--- a/data/lightdm.conf	2016-04-05 08:05:09.000000000 +0800
+++ b/data/lightdm.conf	2016-07-07 20:31:43.061832904 +0800
@@ -21,13 +21,13 @@
 #start-default-seat=true
 #greeter-user=lightdm
 #minimum-display-number=0
-#minimum-vt=7
+minimum-vt=1
 #lock-memory=true
 #user-authority-in-system-dir=false
 #guest-account-script=guest-account
 #logind-check-graphical=false
 #log-directory=/var/log/lightdm
-#run-directory=/var/run/lightdm
+run-directory=/run/lightdm
 #cache-directory=/var/cache/lightdm
 #sessions-directory=/usr/share/lightdm/sessions:/usr/share/xsessions:/usr/share/wayland-sessions
 #remote-sessions-directory=/usr/share/lightdm/remote-sessions
@@ -110,7 +110,7 @@
 #allow-user-switching=true
 #allow-guest=true
 #guest-session=
-#session-wrapper=lightdm-session
+session-wrapper=/etc/lightdm/Xsession
 #greeter-wrapper=
 #guest-wrapper=
 #display-setup-script=
diff -Naur a/data/users.conf b/data/users.conf
--- a/data/users.conf	2016-04-05 08:05:09.000000000 +0800
+++ b/data/users.conf	2016-07-07 20:31:07.446316392 +0800
@@ -9,6 +9,6 @@
 # hidden-shells = Shells that indicate a user cannot login
 #
 [UserList]
-minimum-uid=500
+minimum-uid=1000
 hidden-users=nobody nobody4 noaccess
-hidden-shells=/bin/false /usr/sbin/nologin
+hidden-shells=/bin/false /usr/bin/nologin
