diff -up cantor-16.08.0/cmake/FindLuaJIT.cmake.than cantor-16.08.0/cmake/FindLuaJIT.cmake
--- cantor-16.08.0/cmake/FindLuaJIT.cmake.than	2016-09-06 16:07:30.539296387 +0200
+++ cantor-16.08.0/cmake/FindLuaJIT.cmake	2016-09-06 16:08:06.383945859 +0200
@@ -1,6 +1,6 @@
 SET (LuaJIT_FOUND FALSE)
 SET (LUAJIT_LIBRARY_NAME luajit-5.1)
-SET (LUAJIT_INCLUDE_DIRS /usr/include/luajit-2.0 /usr/local/include/luajit-2.0)
+SET (LUAJIT_INCLUDE_DIRS /usr/include/luajit-2.1 /usr/local/include/luajit-2.1)
 
 FIND_PATH    (LUAJIT_INCLUDE_DIR lua.h ${LUAJIT_INCLUDE_DIRS} )
 FIND_LIBRARY (LUAJIT_LIBRARY NAMES ${LUAJIT_LIBRARY_NAME} PATHS ${CMAKE_LIBRARY_PATH})
diff -up cantor-16.08.0/src/backends/lua/luaexpression.cpp.than cantor-16.08.0/src/backends/lua/luaexpression.cpp
--- cantor-16.08.0/src/backends/lua/luaexpression.cpp.than	2016-09-06 16:22:52.745555302 +0200
+++ cantor-16.08.0/src/backends/lua/luaexpression.cpp	2016-09-06 16:23:14.702728153 +0200
@@ -26,7 +26,7 @@
 #include "imageresult.h"
 #include "helpresult.h"
 
-#include <luajit-2.0/lua.hpp> // need the luajit-2.0 prefix to avoid conflicts with Lua 5.2
+#include <luajit-2.1/lua.hpp> // need the luajit-2.0 prefix to avoid conflicts with Lua 5.2
 
 #include <QDebug>
 #include <QString>
diff -up cantor-16.08.0/src/backends/lua/luahelper.cpp.than cantor-16.08.0/src/backends/lua/luahelper.cpp
--- cantor-16.08.0/src/backends/lua/luahelper.cpp.than	2016-09-06 16:23:45.007586540 +0200
+++ cantor-16.08.0/src/backends/lua/luahelper.cpp	2016-09-06 16:23:56.292161439 +0200
@@ -20,7 +20,7 @@
 
 #include "luahelper.h"
 
-#include <luajit-2.0/lua.hpp> // need the luajit-2.0 prefix to avoid conflicts with Lua 5.2
+#include <luajit-2.1/lua.hpp> // need the luajit-2.0 prefix to avoid conflicts with Lua 5.2
 #include <QString>
 #include <QStringList>
 
diff -up cantor-16.08.0/src/backends/lua/luasession.h.than cantor-16.08.0/src/backends/lua/luasession.h
--- cantor-16.08.0/src/backends/lua/luasession.h.than	2016-09-06 16:23:27.272254647 +0200
+++ cantor-16.08.0/src/backends/lua/luasession.h	2016-09-06 16:23:36.916891324 +0200
@@ -22,7 +22,7 @@
 #define _LUASESSION_H
 
 #include "session.h"
-#include <luajit-2.0/lua.hpp> // need the luajit-2.0 prefix to avoid conflicts with Lua 5.2
+#include <luajit-2.1/lua.hpp> // need the luajit-2.0 prefix to avoid conflicts with Lua 5.2
 
 class LuaExpression;
 

