--- network-manager-applet-0.6.5/src/applet.c.orig	2007-10-02 20:53:36.000000000 +0200
+++ network-manager-applet-0.6.5/src/applet.c	2007-10-02 20:54:46.000000000 +0200
@@ -132,7 +132,7 @@ static void nma_init (NMApplet *applet)
 	applet->animation_step = 0;
 	glade_gnome_init ();
 
-	if (!nma_icons_init (applet))
+	if ((!nma_icons_init (applet)) || (g_getenv("LTSP_CLIENT")))
 		return;
 
 /*	gtk_window_set_default_icon_from_file (ICONDIR"/NMApplet/wireless-applet.png", NULL); */
