---
 src/applet.c |    6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

Index: b/src/applet.c
===================================================================
--- a/src/applet.c
+++ b/src/applet.c
@@ -1794,8 +1794,6 @@ nma_menu_add_vpn_submenu (GtkWidget *men
 	GSList *list, *iter;
 	int num_vpn_active = 0;
 
-	nma_menu_add_separator_item (menu);
-
 	vpn_menu = GTK_MENU (gtk_menu_new ());
 
 	item = GTK_MENU_ITEM (gtk_menu_item_new_with_mnemonic (_("_VPN Connections")));
@@ -1985,14 +1983,14 @@ static void nma_menu_show_cb (GtkWidget
 	}
 
 	nma_menu_add_devices (menu, applet);
-	nma_menu_add_vpn_submenu (menu, applet);
 
 	if (has_usable_wifi (applet)) {
 		/* Add the "Hidden Wi-Fi network..." entry */
-		nma_menu_add_separator_item (menu);
 		nma_menu_add_hidden_network_item (menu, applet);
 		nma_menu_add_create_network_item (menu, applet);
+		nma_menu_add_separator_item (menu);
 	}
+	nma_menu_add_vpn_submenu (menu, applet);
 
 	gtk_widget_show_all (menu);
 
