---
 src/applet.c |   38 +++++++++++++++++---------------------
 1 file changed, 17 insertions(+), 21 deletions(-)

Index: b/src/applet.c
===================================================================
--- a/src/applet.c
+++ b/src/applet.c
@@ -839,17 +839,6 @@ applet_menu_item_create_device_item_help
 	return item;
 }
 
-static void
-applet_clear_notify (NMApplet *applet)
-{
-	if (applet->notification == NULL)
-		return;
-
-	notify_notification_close (applet->notification, NULL);
-	g_object_unref (applet->notification);
-	applet->notification = NULL;
-}
-
 static gboolean
 applet_notify_server_has_actions (void)
 {
@@ -900,19 +889,28 @@ applet_do_notify (NMApplet *applet,
 	if (!applet->agent)
 		return;
 
-	applet_clear_notify (applet);
-
 	escaped = utils_escape_notify_message (message);
-	notify = notify_notification_new (summary,
-	                                  escaped,
-	                                  icon ? icon : GTK_STOCK_NETWORK
+
+	if (applet->notification == NULL) {
+		notify = notify_notification_new (summary,
+		                                  escaped,
+		                                  icon ? icon : GTK_STOCK_NETWORK
 #if HAVE_LIBNOTIFY_07
-	                                  );
+		                                  );
 #else
-	                                  , NULL);
+		                                 , NULL);
 #endif
+
+		applet->notification = notify;
+	} else {
+		notify = applet->notification;
+		notify_notification_update (notify,
+		                            summary,
+		                            escaped,
+		                            icon ? icon : GTK_STOCK_NETWORK);
+	}
+
 	g_free (escaped);
-	applet->notification = notify;
 
 #if HAVE_LIBNOTIFY_07
 	notify_notification_set_hint (notify, "transient", g_variant_new_boolean (TRUE));
@@ -3588,7 +3586,6 @@ status_icon_activate_cb (GtkStatusIcon *
 	/* Have clicking on the applet act also as acknowledgement
 	 * of the notification. 
 	 */
-	applet_clear_notify (applet);
 
 	/* Kill any old menu */
 	if (applet->menu)
@@ -3618,7 +3615,6 @@ status_icon_popup_menu_cb (GtkStatusIcon
 	/* Have clicking on the applet act also as acknowledgement
 	 * of the notification. 
 	 */
-	applet_clear_notify (applet);
 
 	nma_context_menu_update (applet);
 	gtk_menu_popup (GTK_MENU (applet->context_menu), NULL, NULL,
