From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: set IPv6 to Ignore by default for creating new ad-hoc wifi networks.
Bug-Ubuntu: https://bugs.launchpad.net/bugs/830178

---
 src/libnm-gtk/nm-wifi-dialog.c |    5 +++++
 1 file changed, 5 insertions(+)

Index: b/src/libnm-gtk/nm-wifi-dialog.c
===================================================================
--- a/src/libnm-gtk/nm-wifi-dialog.c
+++ b/src/libnm-gtk/nm-wifi-dialog.c
@@ -1210,12 +1210,17 @@ nma_wifi_dialog_get_connection (NMAWifiD
 
 		if (priv->adhoc_create) {
 			NMSettingIP4Config *s_ip4;
+			NMSettingIP6Config *s_ip6;
 
 			g_object_set (s_wireless, NM_SETTING_WIRELESS_MODE, "adhoc", NULL);
 
 			s_ip4 = (NMSettingIP4Config *) nm_setting_ip4_config_new ();
 			g_object_set (s_ip4, NM_SETTING_IP4_CONFIG_METHOD, NM_SETTING_IP4_CONFIG_METHOD_SHARED, NULL);
 			nm_connection_add_setting (connection, (NMSetting *) s_ip4);
+
+			s_ip6 = (NMSettingIP6Config *) nm_setting_ip6_config_new ();
+			g_object_set (s_ip6, NM_SETTING_IP6_CONFIG_METHOD, NM_SETTING_IP6_CONFIG_METHOD_IGNORE, NULL);
+			nm_connection_add_setting (connection, (NMSetting *) s_ip6);
 		}
 
 		nm_connection_add_setting (connection, (NMSetting *) s_wireless);
