From: Antti Kaijanmäki <antti.kaijanmaki@canonical.com>
Subject: Always center dialogs on the screen

---
 src/applet-dialogs.c |    4 ++++
 src/utils/utils.c    |    2 ++
 2 files changed, 6 insertions(+)

Index: b/src/applet-dialogs.c
===================================================================
--- a/src/applet-dialogs.c
+++ b/src/applet-dialogs.c
@@ -55,6 +55,7 @@ info_dialog_show_error (const char *err)
 
 	dialog = gtk_message_dialog_new_with_markup (NULL, 0, GTK_MESSAGE_ERROR, GTK_BUTTONS_OK,
 			"<span weight=\"bold\" size=\"larger\">%s</span>\n\n%s", _("Error displaying connection information:"), err);
+	gtk_window_set_position (GTK_WINDOW (dialog), GTK_WIN_POS_CENTER_ALWAYS);
 	gtk_window_present (GTK_WINDOW (dialog));
 	g_signal_connect_swapped (dialog, "response", G_CALLBACK (gtk_widget_destroy), dialog);
 }
@@ -972,6 +973,7 @@ applet_info_dialog_show (NMApplet *apple
 	g_signal_connect (dialog, "delete-event", G_CALLBACK (gtk_widget_hide_on_delete), dialog);
 	g_signal_connect_swapped (dialog, "response", G_CALLBACK (gtk_widget_hide), dialog);
 	gtk_widget_realize (dialog);
+	gtk_window_set_position (GTK_WINDOW(dialog), GTK_WIN_POS_CENTER_ALWAYS);
 	gtk_window_present_with_time (GTK_WINDOW (dialog),
 		gdk_x11_get_server_time (gtk_widget_get_window (dialog)));
 }
@@ -1026,6 +1028,7 @@ applet_mobile_password_dialog_new (NMCon
 
 	dialog = GTK_DIALOG (gtk_dialog_new ());
 	gtk_window_set_modal (GTK_WINDOW (dialog), TRUE);
+	gtk_window_set_position (GTK_WINDOW (dialog), GTK_WIN_POS_CENTER_ALWAYS);
 	gtk_window_set_title (GTK_WINDOW (dialog), _("Mobile broadband network password"));
 
 	gtk_dialog_add_button (dialog, GTK_STOCK_CANCEL, GTK_RESPONSE_REJECT);
@@ -1362,6 +1365,7 @@ applet_mobile_pin_dialog_new (const char
 	} else
 		g_assert_not_reached ();
 
+	gtk_window_set_position (GTK_WINDOW(dialog), GTK_WIN_POS_CENTER_ALWAYS);
 	gtk_window_set_title (GTK_WINDOW (dialog), title);
 
 	widget = GTK_WIDGET (gtk_builder_get_object (builder, "header_label"));
Index: b/src/utils/utils.c
===================================================================
--- a/src/utils/utils.c
+++ b/src/utils/utils.c
@@ -192,6 +192,8 @@ utils_show_error_dialog (const char *tit
 	                                     "%s",
 	                                     text1);
 
+	gtk_window_set_position (GTK_WINDOW (err_dialog), GTK_WIN_POS_CENTER_ALWAYS);
+
 	if (text2)
 		gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (err_dialog), "%s", text2);
 	if (title)
