From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: Force a menu update 30 seconds after the initialization is done.

---
 src/applet.c |    2 ++
 1 file changed, 2 insertions(+)

Index: b/src/applet.c
===================================================================
--- a/src/applet.c
+++ b/src/applet.c
@@ -3952,6 +3952,8 @@ setup_indicator_menu (NMApplet *applet)
 
 	app_indicator_set_menu(applet->app_indicator, GTK_MENU(applet->menu));
 
+	applet_schedule_update_menu (applet);
+
 	return TRUE;
 }
 #endif /* ENABLE_INDICATOR */
