diff -Naur damo-1.0.4/src/damo/damo.py damo-1.0.4.mod/src/damo/damo.py
--- damo-1.0.4/src/damo/damo.py	2021-11-26 02:53:46.000000000 -0600
+++ damo-1.0.4.mod/src/damo/damo.py	2021-11-28 00:46:40.153653151 -0600
@@ -15,6 +15,8 @@
 import schemes
 import validate
 
+from damo_version import __version__
+
 class SubCmdHelpFormatter(argparse.RawDescriptionHelpFormatter):
     def _format_action(self, action):
         parts = super(argparse.RawDescriptionHelpFormatter,
@@ -83,9 +85,7 @@
     elif args.command == 'validate':
         validate.main(args)
     elif args.command == 'version':
-        bindir = os.path.dirname(os.path.abspath(__file__))
-        with open(os.path.join(bindir, 'damo_version.py'), 'r') as f:
-            print(f.read().strip())
+        print(__version__)
 
 if __name__ == '__main__':
     main()
diff -Naur damo-1.0.4/src/damo/damo_version.py damo-1.0.4.mod/src/damo/damo_version.py
--- damo-1.0.4/src/damo/damo_version.py	2021-11-26 02:53:46.000000000 -0600
+++ damo-1.0.4.mod/src/damo/damo_version.py	2021-11-28 00:45:59.515802719 -0600
@@ -1 +1 @@
-1.0.4
+__version__ = "1.0.4"
