--- a/Makefile
+++ b/Makefile
@@ -73,8 +73,10 @@
 DYNAMIC_LINK= libiw.so
 
 # Install directories
+ARCH=$(shell gcc -print-multiarch)
 INSTALL_DIR= $(PREFIX)/sbin
-INSTALL_LIB= $(PREFIX)/lib
+INSTALL_LIB= $(PREFIX)/lib/$(ARCH)
+INSTALL_USR_LIB= $(PREFIX)/usr/lib/$(ARCH)
 INSTALL_INC= $(PREFIX)/usr/include
 INSTALL_MAN= $(PREFIX)/usr/share/man
 
@@ -120,7 +122,7 @@
 %.o: %.c wireless.h
 	$(CC) $(XCFLAGS) -c $<
 %.so: %.c wireless.h
-	$(CC) $(XCFLAGS) $(PICFLAG) -c -o $@ $<
+	$(CC) $(XCFLAGS) $(LDFLAGS) $(PICFLAG) -c -o $@ $<
 
 iwconfig: iwconfig.o $(IWLIB)
 
@@ -148,7 +150,7 @@
 
 # Compilation of the dynamic library
 $(DYNAMIC): $(OBJS:.o=.so)
-	$(CC) -shared -o $@ -Wl,-soname,$@ $(STRIPFLAGS) $(LIBS) -lc $^
+	$(CC) -shared -o $@ -Wl,-soname,$@ $(STRIPFLAGS) $(LIBS) $(LDFLAGS) -lc $^
 
 # Compilation of the static library
 $(STATIC): $(OBJS:.o=.so)
@@ -164,14 +166,15 @@
 install-dynamic:: $(DYNAMIC)
 	install -m 755 -d $(INSTALL_LIB)
 	install -m 755 $(DYNAMIC) $(INSTALL_LIB)
-	ln -sfn $(DYNAMIC) $(INSTALL_LIB)/$(DYNAMIC_LINK)
+	install -m 755 -d $(INSTALL_USR_LIB)
+	ln -sfn /lib/$(ARCH)/$(DYNAMIC) $(INSTALL_USR_LIB)/$(DYNAMIC_LINK)
 	@echo "*** Don't forget to add $(INSTALL_LIB) to /etc/ld.so.conf, and run ldconfig as root. ***"
 	@$(LDCONFIG) || echo "*** Could not run ldconfig ! ***"
 
 # Install the static library
 install-static:: $(STATIC)
-	install -m 755 -d $(INSTALL_LIB)
-	install -m 644 $(STATIC) $(INSTALL_LIB)
+	install -m 755 -d $(INSTALL_USR_LIB)
+	install -m 644 $(STATIC) $(INSTALL_USR_LIB)
 
 # All the binaries. Careful, no dependancy on install-dynamic
 install-bin:: all
