From 4588853ad10bd9ad30b253a3f17fea72b56e261a Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 11 Aug 2024 17:05:24 +0800
Subject: [PATCH 1/2] fix(Makefile.in): honor system CC variable

From Ubuntu: wavemon (0.8.1-1) unstable (not sure where the patch originated
though but they did refresh it in this revisison).

--- Credit ---

From: Helmut Grohne <helmut@subdivi.de>
Subject: honour the CC variable from configure

A ?= assignment to CC is a noop, because CC has a make default of "cc". This
breaks e.g. cross compilation.
---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index acab13e..c8e520c 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -13,7 +13,7 @@ CC_DEFAULT := @CC@
 ifeq ($(origin CC), default)
 CC	 := $(CC_DEFAULT)
 else
-CC	 ?= $(CC_DEFAULT)
+CC	 = $(CC_DEFAULT)
 endif
 CFLAGS	 ?= @CFLAGS@ @LIBNL3_CLI_CFLAGS@
 CPPFLAGS ?= @CPPFLAGS@
-- 
2.46.0

