From 44d68781fff4eed17ced9f1ee301780b1d10f434 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 11 Aug 2024 18:34:12 +0800
Subject: [PATCH 2/2] fix(Makefile.in): honor Autobuild flags

---
 Makefile.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index c8e520c..171d4d7 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -15,9 +15,9 @@ CC	 := $(CC_DEFAULT)
 else
 CC	 = $(CC_DEFAULT)
 endif
-CFLAGS	 ?= @CFLAGS@ @LIBNL3_CLI_CFLAGS@
-CPPFLAGS ?= @CPPFLAGS@
-LDFLAGS  ?= @LDFLAGS@
+CFLAGS	 = @CFLAGS@ @LIBNL3_CLI_CFLAGS@
+CPPFLAGS = @CPPFLAGS@
+LDFLAGS  = @LDFLAGS@
 DEFS	 ?= @DEFS@
 LDLIBS	 ?= @LIBS@ @LIBNL3_CLI_LIBS@
 
-- 
2.46.0

