diff -Naur fcitx5-qt-5.0.16/qt5/platforminputcontext/qtkey.h fcitx5-qt-5.0.16.gcc13/qt5/platforminputcontext/qtkey.h
--- fcitx5-qt-5.0.16/qt5/platforminputcontext/qtkey.h	2022-11-20 07:08:10.000000000 +0000
+++ fcitx5-qt-5.0.16.gcc13/qt5/platforminputcontext/qtkey.h	2023-06-22 11:57:55.560000000 +0000
@@ -9,6 +9,7 @@
 #define _PLATFORMINPUTCONTEXT_QTKEY_H_
 
 #include <QString>
+#include <cstdint>
 
 int keysymToQtKey(uint32_t keysym, const QString &text);
 
