Description: Fix FTBFS in Debian GNU/Hurd.
Author: Samuel Thibault <sthibault@debian.org>
Last-Update: 2012-06-22

--- libcdaudio-0.99.12p2.orig/configure.ac
+++ libcdaudio-0.99.12p2/configure.ac
@@ -97,7 +97,7 @@
 
 AC_HEADER_DIRENT
 AC_HEADER_STDC
-AC_CHECK_HEADERS(fcntl.h sys/ioctl.h unistd.h linux/cdrom.h linux/ucdrom.h sys/cdio.h io/cam/cdrom.h stdarg.h mntent.h sys/mntent.h sys/types.h sys/param.h sys/ucred.h sys/mount.h strings.h,,,
+AC_CHECK_HEADERS(fcntl.h sys/ioctl.h unistd.h linux/cdrom.h linux/ucdrom.h sys/cdrom.h sys/cdio.h io/cam/cdrom.h stdarg.h mntent.h sys/mntent.h sys/types.h sys/param.h sys/ucred.h sys/mount.h strings.h,,,
 [[#if defined (HAVE_SYS_TYPES_H)
 # include <sys/types.h>
 #endif
--- libcdaudio-0.99.12p2.orig/src/cdaudio.c
+++ libcdaudio-0.99.12p2/src/cdaudio.c
@@ -70,6 +70,11 @@
 #include <linux/ucdrom.h>
 #endif
 
+#ifdef HAVE_SYS_CDROM_H
+#include <sys/cdrom.h>
+#define NON_BLOCKING
+#endif
+
 /* For FreeBSD, OpenBSD, and Solaris */
 #ifdef HAVE_SYS_CDIO_H
 #include <sys/cdio.h>
--- libcdaudio-0.99.12p2.orig/src/changer.c
+++ libcdaudio-0.99.12p2/src/changer.c
@@ -36,6 +36,10 @@
 #include <linux/ucdrom.h>
 #endif
 
+#ifdef HAVE_SYS_CDROM_H
+#include <sys/cdrom.h>
+#endif
+
 /* Choose a particular disc from the CD changer */
 int
 cd_changer_select_disc(int cd_desc, int disc)
