--- lrzsz-0.12.21.orig/src/lrz.c
+++ lrzsz-0.12.21/src/lrz.c
@@ -44,7 +44,6 @@
 #ifndef STRICT_PROTOTYPES
 extern time_t time();
 extern char *strerror();
-extern char *strstr();
 #endif
 
 #ifndef HAVE_ERRNO_DECLARATION
--- lrzsz-0.12.21.orig/src/lsz.c
+++ lrzsz-0.12.21/src/lsz.c
@@ -53,7 +53,6 @@
 #ifndef STRICT_PROTOTYPES
 extern time_t time();
 extern char *strerror();
-extern char *strstr();
 #endif
 
 #ifndef HAVE_ERRNO_DECLARATION
