diff --git a/library/CMakeLists.txt b/library/CMakeLists.txt
index 7513bb6..bcd344a 100644
--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -28,11 +28,6 @@ if(HIPSPARSELT_ENABLE_HIP)
     )
 endif()

-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    # Enable F16C intrinsics
-    target_compile_options(hipsparselt PRIVATE -mf16c)
-endif()
-
 if(HIPSPARSELT_ENABLE_ASAN)
     target_compile_options(hipsparselt PRIVATE -fsanitize=address -shared-libasan)
     target_link_options(hipsparselt PRIVATE -fsanitize=address -shared-libasan -fuse-ld=lld)
