From c56ed6e202be715d0819ea8f811d42365e229178 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Wed, 14 Aug 2024 10:55:59 +0800
Subject: [PATCH 01/35] (LoongArchLinux) add loongarch64 support

---
 configure | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/configure b/configure
index d4882c0..539f6e5 100755
--- a/configure
+++ b/configure
@@ -1853,6 +1853,7 @@ if test -z "$_target" ; then
       nios2) host_arch=nios2 ;;
       vax) host_arch=vax ;;
       xtensa*) host_arch=xtensa ;;
+      loongarch64) host_arch=loongarch64 ;;
       *) host_arch=UNKNOWN ;;
   esac
   }
@@ -2835,6 +2836,18 @@ EOF
     arch='arc'
     iproc='arc'
     ;;
+  loongarch64)
+    arch='loongarch'
+    subarch='loongarch64'
+    def_fast_unaligned='#define HAVE_FAST_UNALIGNED 1'
+    def_av_fast_unaligned='#define AV_HAVE_FAST_UNALIGNED 1'
+    def_local_aligned_8='#define HAVE_LOCAL_ALIGNED_8 1'
+    def_local_aligned_16='#define HAVE_LOCAL_ALIGNED_16 1'
+    def_local_aligned_32='#define HAVE_LOCAL_ALIGNED_32 1'
+    def_fast_64bit='#define HAVE_FAST_64BIT 1'
+    _march='-march=loongarch64'
+    iproc='loongarch'
+    ;;
 
   *)
     echo "The architecture of your CPU ($host_arch) is not supported by this configure script"
-- 
2.46.0

