From e61237ced28b6173558b25f954aca0578a757cc5 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Sep 2024 23:27:02 +0800
Subject: [PATCH 07/35] [r38360] configure: fix compiling against latest
 FFmpeg.

(From upstream SVN r38360)
---
 configure | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/configure b/configure
index d104656..926eb9d 100755
--- a/configure
+++ b/configure
@@ -1667,6 +1667,9 @@ libavmuxers=$(filter_out_component muxer 'CHROMAPRINT LIB[A-Z0-9_]* RTP RTSP SAP
 libavprotocols=$(filter_out_component protocol 'BLURAY FFRTMPCRYPT HTTPS LIB[A-Z0-9_]* TLS TLS_GNUTLS TLS_OPENSSL TLS_SECURETRANSPORT TLS_SCHANNEL')
 libavfilters=$(filter_out_component filter 'VF_FREI0R[A-Z0-9_]* LIB[A-Z0-9_]* MP VF_OCV')
 
+libavwrappers_all="DEFLATE_WRAPPER INFLATE_WRAPPER"
+libavwrappers="$libavwrappers_all"
+
 # second pass command line parsing for options needing local FFmpeg checkout
 for ac_option do
   case "$ac_option" in
@@ -6803,6 +6806,7 @@ else
   def_zlib='#define CONFIG_ZLIB 0'
   libavdecoders=$(filter_out_component decoder 'APNG FLASHSV FLASHSV2 PNG ZMBV ZLIB DXA EXR G2M LSCR MSCC MVHA MWSC RASC RSCC SCREENPRESSO SRGC TDSC TSCC ZEROCODEC WCMV')
   libavencoders=$(filter_out_component encoder 'APNG EXR FLASHSV FLASHSV2 PNG ZMBV ZLIB')
+  libavwrappers=$(filter_out_component wrapper 'DEFLATE INFLATE')
 fi
 echores "$_zlib"
 
@@ -9071,6 +9075,7 @@ $(mak_enable "$libavbsfs_all"      "$libavbsfs"      CONFIG)
 $(mak_enable "$libavcbs_all"       "$libavcbs"       CONFIG)
 $(mak_enable "$libavhwaccels_all"  "$libavhwaccels"  CONFIG)
 $(mak_enable "$libavfilters_all"   "$libavfilters"   CONFIG)
+$(mak_enable "$libavwrappers_all"  "$libavwrappers"   CONFIG)
 EOF
 
 #############################################################################
@@ -9724,6 +9729,7 @@ $(ff_config_enable "$libavbsfs_all"      "$libavbsfs" "#")
 $(ff_config_enable "$libavcbs_all"       "$libavcbs" "#")
 $(ff_config_enable "$libavhwaccels_all"  "$libavhwaccels" "#")
 $(ff_config_enable "$libavfilters_all"   "$libavfilters" "#")
+$(ff_config_enable "$libavwrappers_all"   "$libavwrappers" "#")
 
 #endif /* MPLAYER_CONFIG_H */
 EOF
@@ -9782,6 +9788,7 @@ cp ffmpeg/ffbuild/config.mak ffmpeg/config.mak
 cat > ffmpeg/config.h << EOF
 #include "../config.h"
 EOF
+cp ffmpeg/config.h ffmpeg/config_components.h
 touch ffmpeg/.config
 
 # generate the lists of enabled components for ffmpeg
-- 
2.46.0

