From cf761ae539541f0dc3c14a918c3d0834e97b33dd Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Sep 2024 23:27:25 +0800
Subject: [PATCH 09/35] [r38365] configure: Fix compilation against latest
 FFmpeg.

(From upstream SVN r38365)
---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index 1a19081..0aa102a 100755
--- a/configure
+++ b/configure
@@ -9022,6 +9022,7 @@ CONFIG_SINEWIN = yes
 CONFIG_SNAPPY = yes
 CONFIG_STARTCODE = yes
 CONFIG_TEXTUREDSP = yes
+CONFIG_TEXTUREDSPENC = yes
 CONFIG_VC1DSP = yes
 CONFIG_VIDEODSP = yes
 CONFIG_VP3DSP  = yes
@@ -9699,6 +9700,7 @@ $def_struct_pollfd
 #define CONFIG_SNAPPY 1
 #define CONFIG_STARTCODE 1
 #define CONFIG_TEXTUREDSP 1
+#define CONFIG_TEXTUREDSPENC 1
 #define CONFIG_VC1DSP 1
 #define CONFIG_VIDEODSP 1
 #define CONFIG_VP56DSP 1
-- 
2.46.0

