From 0b6857eb47e198b6a0feb37aef57001a43dee665 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Sep 2024 23:28:00 +0800
Subject: [PATCH 11/35] [r38375] configure: update for latest FFmpeg

(From upstream SVN r38375)
---
 configure | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/configure b/configure
index 94c8825..0e8ebea 100755
--- a/configure
+++ b/configure
@@ -9014,6 +9014,8 @@ CONFIG_MPEGAUDIOHEADER = yes
 CONFIG_MPEGVIDEO    = yes
 CONFIG_MPEGVIDEODEC = yes
 CONFIG_MPEGVIDEOENC = yes
+CONFIG_MSMPEG4DEC = yes
+CONFIG_MSMPEG4ENC = yes
 CONFIG_MSS34DSP = yes
 CONFIG_PIXBLOCKDSP  = yes
 CONFIG_RANGECODER   = yes
@@ -9598,6 +9600,7 @@ $def_gnutls
 $def_openssl
 #define CONFIG_GPL 1
 #define CONFIG_GRAY 0
+#define CONFIG_LCMS2 0
 #define CONFIG_LIBMODPLUG 0
 #define CONFIG_LIBVORBIS 0
 #define CONFIG_LINUX_PERF 0
@@ -9612,6 +9615,16 @@ $def_openssl
 #define SWS_MAX_FILTER_SIZE 256
 #define CONFIG_QSV 0
 #define CONFIG_VAAPI 0
+#define CONFIG_VALGRIND_BACKTRACE 0
+
+/* related to HW accelarations */
+#define CONFIG_CUDA 0
+#define CONFIG_D3D11VA 0
+#define CONFIG_LIBDRM 0
+#define CONFIG_OPENCL 0
+#define CONFIG_VIDEOTOOLBOX 0
+#define CONFIG_MEDIACODEC 0
+#define CONFIG_VULKAN 0
 
 #define HAVE_ALIGNED_STACK 1
 #define HAVE_AS_ARCH_DIRECTIVE 1
@@ -9625,6 +9638,7 @@ $def_as_func
 #define HAVE_SYNC_VAL_COMPARE_AND_SWAP 1
 #define HAVE_ATOMICS_NATIVE 1
 #define HAVE_ATTRIBUTE_PACKED 1
+#define HAVE_FAST_FLOAT16 0
 #define HAVE_GETHRTIME 0
 $def_inline_asm
 #define HAVE_INLINE_ASM_NONLOCAL_LABELS 1
@@ -9652,6 +9666,7 @@ $def_struct_pollfd
 #define HAVE_SYMVER_GNU_ASM 0
 #define HAVE_SYNC_SYNCHRONIZE 1
 #define HAVE_SYS_SELECT_H 0
+#define HAVE_VALGRIND_VALGRIND_H 0
 #define HAVE_VIRTUALALLOC 0
 
 /* Some FFmpeg codecs depend on these. Enable them unconditionally for now. */
@@ -9695,6 +9710,8 @@ $def_struct_pollfd
 #define CONFIG_MPEGVIDEO 1
 #define CONFIG_MPEGVIDEODEC 1
 #define CONFIG_MPEGVIDEOENC 1
+#define CONFIG_MSMPEG4DEC 1
+#define CONFIG_MSMPEG4ENC 1
 #define CONFIG_MSS34DSP 1
 #define CONFIG_PIXBLOCKDSP 1
 #define CONFIG_RDFT 1
-- 
2.46.0

