From af5619a8c5fa0fdff1c9461e59405cae3cb234cb Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Sep 2024 23:33:24 +0800
Subject: [PATCH 16/35] [r38407] configure: update to fix compiling latest
 FFmpeg.

(From upstream SVN r38407)
---
 configure | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/configure b/configure
index 85d5a35..169158c 100755
--- a/configure
+++ b/configure
@@ -1661,7 +1661,7 @@ libavprotocols=$(echo $libavprotocols_all)
 libavfilters=$(echo $libavfilters_all)
 
 libavdecoders=$(filter_out_component decoder 'LIB[A-Z0-9_]* [A-Z0-9]*_QSV [A-Z0-9]*_MMAL [A-Z0-9]*_MEDIACODEC [A-Z0-9]*_CUVID [A-Z0-9_]*_AT [A-Z0-9]*_RKMPP [A-Z0-9]*_V4L2M2M')
-libavencoders=$(filter_out_component encoder 'LIB[A-Z0-9_]* [A-Z0-9]*_QSV [A-Z0-9]*_MMAL NVENC[A-Z0-9_]* AV1_NVENC[A-Z0-9_]* H264_NVENC[A-Z0-9_]* HEVC_NVENC[A-Z0-9_]* HAP [A-Z0-9]*_VIDEOTOOLBOX [A-Z0-9]*_VAAPI [A-Z0-9]*_OMX [A-Z0-9_]*_AT [A-Z0-9]*_V4L2M2M [A-Z0-9]*_AMF [A-Z0-9]*_MF')
+libavencoders=$(filter_out_component encoder 'LIB[A-Z0-9_]* [A-Z0-9]*_QSV [A-Z0-9]*_MMAL [A-Z0-9]*_MEDIACODEC NVENC[A-Z0-9_]* AV1_NVENC[A-Z0-9_]* H264_NVENC[A-Z0-9_]* HEVC_NVENC[A-Z0-9_]* HAP [A-Z0-9]*_VIDEOTOOLBOX [A-Z0-9]*_VAAPI [A-Z0-9]*_OMX [A-Z0-9_]*_AT [A-Z0-9]*_V4L2M2M [A-Z0-9]*_AMF [A-Z0-9]*_MF')
 libavbsfs=$(filter_out_component bsf 'TRACE_HEADERS [A-Z0-9_]*_METADATA H264_REDUNDANT_PPS FILTER_UNITS')
 libavdemuxers=$(filter_out_component demuxer 'AVISYNTH LIB[A-Z0-9_]* REDIR VAPOURSYNTH')
 libavmuxers=$(filter_out_component muxer 'CHROMAPRINT LIB[A-Z0-9_]* RTP RTSP SAP')
@@ -8983,7 +8983,9 @@ CONFIG_H264DSP = yes
 CONFIG_H264PARSE = yes
 CONFIG_H264PRED= yes
 CONFIG_H264QPEL= yes
+CONFIG_H264_SEI = yes
 CONFIG_HEVCPARSE = yes
+CONFIG_HEVC_SEI = yes
 CONFIG_HPELDSP = yes
 CONFIG_IIRFILTER = yes
 CONFIG_ISO_MEDIA = yes
@@ -9696,7 +9698,9 @@ $def_struct_pollfd
 #define CONFIG_H264DSP 1
 #define CONFIG_H264PRED 1
 #define CONFIG_H264QPEL 1
+#define CONFIG_H264_SEI 1
 #define CONFIG_HEVCPARSE 1
+#define CONFIG_HEVC_SEI 1
 #define CONFIG_HUFFMAN 1
 #define CONFIG_IDCTDSP 1
 #define CONFIG_IVIDSP 1
-- 
2.46.0

