From cc7a977662d5c87ad8e6477cd628f294090d8844 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Sep 2024 23:39:12 +0800
Subject: [PATCH 21/35] [r38426] configure: some FFmpeg compile fixes.

(From upstream SVN r38426)
---
 configure | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index da1ad6a..94cc2d9 100755
--- a/configure
+++ b/configure
@@ -3425,7 +3425,7 @@ if arm ; then
 fi
 
 intrinsics="none"
-cpuexts_all='ALTIVEC XOP AVX AVX2 AVX512 AVX512ICL FMA3 FMA4 MMX MMX2 MMXEXT AMD3DNOW AMD3DNOWEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE42 FAST_CMOV I686 FAST_CLZ ARMV5TE ARMV6 ARMV6T2 ARMV8 VFP VFPV3 SETEND NEON INTRINSICS_NEON IWMMXT MMI VIS MVI LASX LSX'
+cpuexts_all='ALTIVEC XOP AVX AVX2 AVX512 AVX512ICL FMA3 FMA4 MMX MMX2 MMXEXT AMD3DNOW AMD3DNOWEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE42 FAST_CMOV I686 FAST_CLZ ARMV5TE ARMV6 ARMV6T2 ARMV8 VFP VFPV3 SETEND NEON INTRINSICS_NEON I8MM IWMMXT MMI VIS MVI LASX LSX'
 test "$_altivec"   = yes && cpuexts="ALTIVEC $cpuexts"
 test "$_mmx"       = yes && cpuexts="MMX $cpuexts"
 test "$_mmxext"    = yes && cpuexts="MMX2 $cpuexts"
@@ -6807,7 +6807,7 @@ if test "$_zlib" = yes ; then
   mplayer_encoders="$mplayer_encoders PNG_ENCODER"
 else
   def_zlib='#define CONFIG_ZLIB 0'
-  libavdecoders=$(filter_out_component decoder 'APNG FLASHSV FLASHSV2 PNG ZMBV ZLIB DXA EXR G2M LSCR MSCC MVHA MWSC RASC RSCC SCREENPRESSO SRGC TDSC TSCC ZEROCODEC WCMV')
+  libavdecoders=$(filter_out_component decoder 'APNG FLASHSV FLASHSV2 PNG ZMBV ZLIB DXA EXR G2M LSCR MSCC MVHA MWSC PDV RASC RSCC SCREENPRESSO SRGC TDSC TSCC ZEROCODEC WCMV')
   libavencoders=$(filter_out_component encoder 'APNG EXR FLASHSV FLASHSV2 PNG ZMBV ZLIB')
   libavwrappers=$(filter_out_component wrapper 'DEFLATE INFLATE')
 fi
@@ -9629,6 +9629,7 @@ $def_openssl
 
 #define HAVE_ALIGNED_STACK 1
 #define HAVE_AS_ARCH_DIRECTIVE 1
+#define AS_ARCH_LEVEL armv8-a
 #define HAVE_AS_DN_DIRECTIVE 1
 #define HAVE_AS_FPU_DIRECTIVE 1
 $def_as_func
-- 
2.46.0

