From 8763fe82f24a4d5da2ba9ed6102250934559c24b Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Sep 2024 23:55:33 +0800
Subject: [PATCH 23/35] [r38441] mp_image.c: Add missing limits.h include.

(From upstream SVN r38441)
---
 libmpcodecs/mp_image.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libmpcodecs/mp_image.c b/libmpcodecs/mp_image.c
index 27e7dbc..2b2271f 100644
--- a/libmpcodecs/mp_image.c
+++ b/libmpcodecs/mp_image.c
@@ -18,6 +18,7 @@
 
 #include "config.h"
 
+#include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-- 
2.46.0

