From c960263e97a5c52d93e74b0f9e60df59f8ea4aa3 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Sep 2024 23:40:08 +0800
Subject: [PATCH 24/35] [r38449] configure: Fixes for latest FFmpeg

(From upstream SVN r38449)
---
 configure | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure b/configure
index 9a346d5..1fdd98f 100755
--- a/configure
+++ b/configure
@@ -8445,6 +8445,7 @@ if test "$_dvdnav" = yes ; then
 else
   def_dvdnav='#undef CONFIG_DVDNAV'
   noinputmodules="dvdnav $noinputmodules"
+  libavdemuxers=$(filter_out_component demuxer 'DVDVIDEO')
 fi
 echores "$_dvdnav"
 
@@ -8995,6 +8996,8 @@ CONFIG_QPELDSP = yes
 CONFIG_QSV     = no
 CONFIG_HUFFYUVDSP = yes
 CONFIG_HUFFYUVENCDSP = yes
+CONFIG_IAMFDEC = yes
+CONFIG_IAMFENC = yes
 CONFIG_IDCTDSP = yes
 CONFIG_TPELDSP = yes
 CONFIG_HUFFMAN = yes
@@ -9704,6 +9707,8 @@ $def_struct_pollfd
 #define CONFIG_HEVCPARSE 1
 #define CONFIG_HEVC_SEI 1
 #define CONFIG_HUFFMAN 1
+#define CONFIG_IAMFDEC 1
+#define CONFIG_IAMFENC 1
 #define CONFIG_IDCTDSP 1
 #define CONFIG_IVIDSP 1
 #define CONFIG_LPC 1
-- 
2.46.0

