From ac02a16ec9fe94c361efdbd60128125c90ecc0d3 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Fri, 10 Apr 2026 14:30:03 +0800
Subject: [PATCH 43/66] [r38476] Make cosmetic changes.

---
 gui/util/bitmap.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gui/util/bitmap.c b/gui/util/bitmap.c
index bc04639..2244c6c 100644
--- a/gui/util/bitmap.c
+++ b/gui/util/bitmap.c
@@ -82,8 +82,8 @@ static int pngRead(const char *fname, guiImage *img, int *pix_fmt)
     void *data;
     int decode_ok, bpl;
     AVCodecContext *avctx;
-    AVFrame *frame;
     AVPacket *pkt;
+    AVFrame *frame;
 
     file = fopen(fname, "rb");
 
@@ -188,8 +188,8 @@ static int pngRead(const char *fname, guiImage *img, int *pix_fmt)
     }
 
     avcodec_send_packet(avctx, NULL);   // flush the decoder
-
     avcodec_close(avctx);
+
     av_frame_free(&frame);
     av_packet_free(&pkt);
     avcodec_free_context(&avctx);
-- 
2.54.0

