diff -rupN libffado.old/doc/reference.doxygen.in libffado/doc/reference.doxygen.in
--- libffado.old/doc/reference.doxygen.in	2009-12-18 13:52:13.000000000 -0500
+++ libffado/doc/reference.doxygen.in	2010-06-05 21:00:54.000000000 -0400
@@ -568,9 +568,7 @@ WARN_LOGFILE           =
 # directories like "/usr/src/myproject". Separate the files or directories
 # with spaces.
 
-INPUT                  = $top_srcdir/src \
-                         $top_srcdir/doc \
-                         $top_srcdir/libffado
+INPUT                  = $top_srcdir/libffado
 
 # This tag can be used to specify the character encoding of the source files
 # that doxygen parses. Internally doxygen uses the UTF-8 encoding, which is
diff -rupN libffado.old/SConstruct libffado/SConstruct
--- libffado.old/SConstruct	2010-05-10 15:34:07.000000000 -0400
+++ libffado/SConstruct	2010-06-05 21:29:43.000000000 -0400
@@ -583,7 +583,7 @@ if not env.GetOption('clean'):
     Default( 'support' )
     if env['BUILD_TESTS']:
         Default( 'tests' )
-
+    Default( 'doc' )
 #
 # Deal with the DESTDIR vs. xdg-tools conflict (which is basicely that the
 # xdg-tools can't deal with DESTDIR, so the packagers have to deal with this
