From 8c2399ff1ca95f448c0f442cd3bd71635039a436 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 18 Dec 2025 22:28:52 +0800
Subject: [PATCH 2/3] AOSCOS: fix(binfmt-dispatcher.toml): set use_muvm for
 box64

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 data/binfmt-dispatcher.toml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/binfmt-dispatcher.toml b/data/binfmt-dispatcher.toml
index ba8826a..1bb72b1 100644
--- a/data/binfmt-dispatcher.toml
+++ b/data/binfmt-dispatcher.toml
@@ -7,6 +7,7 @@ path = "/usr/bin/muvm"
 
 [interpreters.box64]
 path = "/usr/bin/box64"
+use_muvm = true
 
 [interpreters.fex]
 name = "FEX"
-- 
2.52.0

