From 940a3b713cf58b2af94a8b8aceaca386b3f88bfd Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 18 Dec 2025 22:51:47 +0800
Subject: [PATCH 3/3] AOSCOS: fix(Makefile): do not confuse Cargo parameters
 with RUSTFLAGS

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index f723429..be79661 100644
--- a/Makefile
+++ b/Makefile
@@ -6,17 +6,17 @@ DATADIR ?= $(PREFIX)/share
 POLKITACTIONSDIR ?= $(DATADIR)/polkit-1/actions
 SYSCONFIGDIR ?= /etc
 
-RUSTFLAGS ?= --release
+CARGOFLAGS ?= --release
 
 ROOTDIR := $(dir $(realpath $(lastword $(MAKEFILE_LIST))))
 
 all: build
 
 build:
-	cargo build $(RUSTFLAGS)
+	cargo build $(CARGOFLAGS)
 
 check:
-	cargo test $(RUSTFLAGS)
+	cargo test $(CARGOFLAGS)
 
 clean:
 	rm -rf target
-- 
2.52.0

