diff -Nur libunwind-1.1/tests/Makefile.am libunwind-1.1~/tests/Makefile.am
--- libunwind-1.1/tests/Makefile.am	2014-01-20 11:26:56.229924564 -0500
+++ libunwind-1.1~/tests/Makefile.am	2014-01-20 11:25:39.318794342 -0500
@@ -189,3 +189,5 @@
 Lia64_test_readonly_LDADD = $(LIBUNWIND_local)
 ia64_test_dyn1_LDADD = $(LIBUNWIND)
 ia64_test_sig_LDADD = $(LIBUNWIND)
+
+ppc64_test_altivec_LDADD = $(LIBUNWIND)
