diff -up libunwind-1.1/src/arm/Gglobal.c.default-to-exidx libunwind-1.1/src/arm/Gglobal.c
--- libunwind-1.1/src/arm/Gglobal.c.default-to-exidx	2015-06-02 10:38:39.733587918 -0400
+++ libunwind-1.1/src/arm/Gglobal.c	2015-06-02 10:38:53.086500142 -0400
@@ -29,7 +29,10 @@ HIDDEN define_lock (arm_lock);
 HIDDEN int tdep_init_done;
 
 /* Unwinding methods to use. See UNW_METHOD_ enums */
-HIDDEN int unwi_unwind_method = UNW_ARM_METHOD_ALL;
+/* UNW_ARM_METHOD_ALL starts with UNW_ARM_METHOD_DWARF */
+/* which is never right on Fedora ARM. Default instead */
+/* to UNW_ARM_METHOD_EXIDX. */
+HIDDEN int unwi_unwind_method = UNW_ARM_METHOD_EXIDX;
 
 HIDDEN void
 tdep_init (void)
