From: Daigo Moriwaki <daigo@debian.org>
Subject: rpath should not be used. See http://wiki.debian.org/RpathIssue

--- a/configure.ac
+++ b/configure.ac
@@ -21,6 +21,16 @@ LT_INIT
 AM_PROG_AS
 AM_PROG_CC_C_O
 
+case ${host} in
+  *-pc-linux-gnu)
+    AC_MSG_RESULT([Fixing libtool for -rpath problems.])
+    sed < libtool > libtool-2 \
+    's/^hardcode_libdir_flag_spec.*$'/'hardcode_libdir_flag_spec=" -D__LIBTOOL_IS_A_FOOL__ "/'
+    mv libtool-2 libtool
+    chmod 755 libtool
+  ;;
+esac
+
 dnl Checks for libraries.
 AC_CHECK_LIB(uca, __uc_get_grs)
 OLD_LIBS=${LIBS}
