Index: libunwind/tests/test-ptrace.c
===================================================================
--- libunwind.orig/tests/test-ptrace.c	2012-07-30 10:05:35.271168000 +0100
+++ libunwind/tests/test-ptrace.c	2012-07-30 11:39:37.309158623 +0100
@@ -182,6 +182,11 @@
 
       /* automated test case */
       argv = args;
+
+      /* Unless the args array is 'walked' the child
+         process is unable to access it and dies with a segfault */
+      fprintf(stderr, "Automated test (%s,%s,%s,%s)\n",
+              args[0],args[1],args[2],args[3]);
     }
   else if (argc > 1)
     while (argv[optind][0] == '-')
