Index: pkg-libunwind/tests/Makefile.am
===================================================================
--- pkg-libunwind.orig/tests/Makefile.am	2013-08-04 16:25:11.271314176 +0900
+++ pkg-libunwind/tests/Makefile.am	2013-08-04 16:30:32.666240614 +0900
@@ -47,7 +47,7 @@
 			test-async-sig test-flush-cache test-init-remote \
 			test-mem test-setjmp test-ptrace Ltest-varargs	 \
 			Ltest-nomalloc Ltest-nocalloc Lrs-race
- noinst_PROGRAMS_cdep = forker crasher mapper test-ptrace-misc		 \
+ noinst_PROGRAMS_cdep = forker mapper test-ptrace-misc		 \
 			Gperf-simple Lperf-simple
 
  noinst_PROGRAMS_cdep += Gperf-trace Lperf-trace
@@ -59,7 +59,7 @@
 if OS_LINUX
 if BUILD_COREDUMP
  check_SCRIPTS_cdep += run-coredump-unwind
- noinst_PROGRAMS_cdep += test-coredump-unwind
+ noinst_PROGRAMS_cdep += crasher test-coredump-unwind
 
 if HAVE_LZMA
  check_SCRIPTS_cdep += run-coredump-unwind-mdi
