diff -rupN uriparser-0.9.4/CMakeLists.txt uriparser-0.9.4-new/CMakeLists.txt
--- uriparser-0.9.4/CMakeLists.txt	2020-05-31 17:11:30.000000000 +0200
+++ uriparser-0.9.4-new/CMakeLists.txt	2020-06-02 09:25:06.950566977 +0200
@@ -271,13 +271,14 @@ if(URIPARSER_BUILD_TESTS)
         ${LIBRARY_CODE_FILES}
     )
 
+    set_property(TARGET testrunner PROPERTY POSITION_INDEPENDENT_CODE TRUE)
     target_compile_definitions(testrunner PRIVATE URI_STATIC_BUILD)
 
     if(MSVC)
         target_compile_definitions(testrunner PRIVATE -D_CRT_SECURE_NO_WARNINGS)
     endif()
 
-    target_include_directories(testrunner SYSTEM PRIVATE
+    target_include_directories(testrunner PRIVATE
         ${GTEST_INCLUDE_DIRS}
     )
 
