diff -Naur userspace-rcu-0.9.5-old/configure.ac userspace-rcu-0.9.5/configure.ac
--- userspace-rcu-0.9.5-old/configure.ac	2018-01-23 20:19:49.000000000 +0000
+++ userspace-rcu-0.9.5/configure.ac	2018-03-04 08:10:20.000000000 +0000
@@ -137,6 +137,7 @@
 	[mips*], [ARCHTYPE="mips"],
 	[tile*], [ARCHTYPE="tile"],
 	[hppa*], [ARCHTYPE="hppa"],
+	[riscv*], [ARCHTYPE="riscv"],
 	[ARCHTYPE="unknown"]
 )
 
diff -Naur userspace-rcu-0.9.5-old/urcu/arch/riscv.h userspace-rcu-0.9.5/urcu/arch/riscv.h
--- userspace-rcu-0.9.5-old/urcu/arch/riscv.h	1970-01-01 00:00:00.000000000 +0000
+++ userspace-rcu-0.9.5/urcu/arch/riscv.h	2018-03-04 08:10:48.000000000 +0000
@@ -0,0 +1,52 @@
+#ifndef _URCU_ARCH_RISCV_H
+#define _URCU_ARCH_RISCV_H
+
+/*
+ * arch/riscv.h: definitions for RISC-V architecture
+ *
+ * Copyright (c) 2010 Paul E. McKenney, IBM Corporation.
+ * Copyright (c) 2009-2015 Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
+ * Copyright (C) 2018 Icenowy Zheng <icenowy@aosc.io>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <urcu/compiler.h>
+#include <urcu/config.h>
+#include <urcu/syscall-compat.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdlib.h>
+#include <sys/time.h>
+
+/*
+ * On Linux, define the membarrier system call number if not yet available in
+ * the system headers. riscv implements asm-generic/unistd.h system call
+ * numbers.
+ */
+#if (defined(__linux__) && !defined(__NR_membarrier))
+#define __NR_membarrier		283
+#endif
+
+#ifdef __cplusplus
+}
+#endif
+
+#include <urcu/arch/generic.h>
+
+#endif /* _URCU_ARCH_RISCV_H */
diff -Naur userspace-rcu-0.9.5-old/urcu/uatomic/riscv.h userspace-rcu-0.9.5/urcu/uatomic/riscv.h
--- userspace-rcu-0.9.5-old/urcu/uatomic/riscv.h	1970-01-01 00:00:00.000000000 +0000
+++ userspace-rcu-0.9.5/urcu/uatomic/riscv.h	2018-03-04 08:11:05.000000000 +0000
@@ -0,0 +1,43 @@
+#ifndef _URCU_ARCH_UATOMIC_RISCV_H
+#define _URCU_ARCH_UATOMIC_RISCV_H
+
+/*
+ * Copyright (c) 1991-1994 by Xerox Corporation.  All rights reserved.
+ * Copyright (c) 1996-1999 by Silicon Graphics.  All rights reserved.
+ * Copyright (c) 1999-2004 Hewlett-Packard Development Company, L.P.
+ * Copyright (c) 2009-2015 Mathieu Desnoyers
+ * Copyright (c) 2010      Paul E. McKenney, IBM Corporation
+ *			   (Adapted from uatomic_arch_ppc.h)
+ * Copyright (c) 2018      Icenowy Zheng
+ *                         (Adapted from uatomic/aarch64.h)
+ *
+ * THIS MATERIAL IS PROVIDED AS IS, WITH ABSOLUTELY NO WARRANTY EXPRESSED
+ * OR IMPLIED.  ANY USE IS AT YOUR OWN RISK.
+ *
+ * Permission is hereby granted to use or copy this program
+ * for any purpose,  provided the above notices are retained on all copies.
+ * Permission to modify the code and to distribute modified code is granted,
+ * provided the above notices are retained, and a notice that the code was
+ * modified is included with the above copyright notice.
+ *
+ * Code inspired from libuatomic_ops-1.2, inherited in part from the
+ * Boehm-Demers-Weiser conservative garbage collector.
+ */
+
+#include <urcu/compiler.h>
+#include <urcu/system.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define UATOMIC_HAS_ATOMIC_BYTE
+#define UATOMIC_HAS_ATOMIC_SHORT
+
+#ifdef __cplusplus
+}
+#endif
+
+#include <urcu/uatomic/generic.h>
+
+#endif /* _URCU_ARCH_UATOMIC_RISCV_H */
