diff -Naur xpra-4.2.2/setup.py xpra-4.2.2.no-default-doc/setup.py
--- xpra-4.2.2/setup.py	2021-08-09 11:39:02.000000000 -0500
+++ xpra-4.2.2.no-default-doc/setup.py	2021-08-21 13:50:41.314329243 -0500
@@ -202,7 +202,7 @@
 example_ENABLED         = DEFAULT
 
 #Cython / gcc / packaging build options:
-docs_ENABLED            = DEFAULT
+docs_ENABLED            = False
 annotate_ENABLED        = DEFAULT
 warn_ENABLED            = True
 strict_ENABLED          = True
