diff --git a/src/bin/keactrl/kea-ctrl-agent.conf.pre b/src/bin/keactrl/kea-ctrl-agent.conf.pre
index 9569919..fb3f297 100644
--- a/src/bin/keactrl/kea-ctrl-agent.conf.pre
+++ b/src/bin/keactrl/kea-ctrl-agent.conf.pre
@@ -107,11 +107,12 @@
                 // - syslog (logs to syslog)
                 // - syslog:name (logs to syslog using specified name)
                 // Any other value is considered a name of the file
-                "output": "kea-ctrl-agent.log"
+                // "output": "kea-ctrl-agent.log"
+                "output": "syslog",
 
                 // Shorter log pattern suitable for use with systemd,
                 // avoids redundant information
-                // "pattern": "%-5p %m\n"
+                "pattern": "%-5p %m\n"
 
                 // This governs whether the log output is flushed to disk after
                 // every write.
diff --git a/src/bin/keactrl/kea-dhcp-ddns.conf.pre b/src/bin/keactrl/kea-dhcp-ddns.conf.pre
index 867f018..f065470 100644
--- a/src/bin/keactrl/kea-dhcp-ddns.conf.pre
+++ b/src/bin/keactrl/kea-dhcp-ddns.conf.pre
@@ -44,11 +44,12 @@
                 // - syslog (logs to syslog)
                 // - syslog:name (logs to syslog using specified name)
                 // Any other value is considered a name of the file
-                "output": "kea-ddns.log"
+                // "output": "kea-ddns.log"
+                "output": "syslog",
 
                 // Shorter log pattern suitable for use with systemd,
                 // avoids redundant information
-                // "pattern": "%-5p %m\n"
+                "pattern": "%-5p %m\n"
 
                 // This governs whether the log output is flushed to disk after
                 // every write.
diff --git a/src/bin/keactrl/kea-dhcp4.conf.pre b/src/bin/keactrl/kea-dhcp4.conf.pre
index 07511cc..4965e83 100644
--- a/src/bin/keactrl/kea-dhcp4.conf.pre
+++ b/src/bin/keactrl/kea-dhcp4.conf.pre
@@ -436,11 +436,12 @@
                 // - syslog (logs to syslog)
                 // - syslog:name (logs to syslog using specified name)
                 // Any other value is considered a name of the file
-                "output": "kea-dhcp4.log"
+                // "output": "kea-dhcp4.log"
+                "output": "syslog",
 
                 // Shorter log pattern suitable for use with systemd,
                 // avoids redundant information
-                // "pattern": "%-5p %m\n",
+                "pattern": "%-5p %m\n"
 
                 // This governs whether the log output is flushed to disk after
                 // every write.
diff --git a/src/bin/keactrl/kea-dhcp6.conf.pre b/src/bin/keactrl/kea-dhcp6.conf.pre
index 221381e..ff5b12f 100644
--- a/src/bin/keactrl/kea-dhcp6.conf.pre
+++ b/src/bin/keactrl/kea-dhcp6.conf.pre
@@ -395,11 +395,12 @@
                 // - syslog (logs to syslog)
                 // - syslog:name (logs to syslog using specified name)
                 // Any other value is considered a name of the file
-                "output": "kea-dhcp6.log"
+                // "output": "kea-dhcp6.log"
+                "output": "syslog",
 
                 // Shorter log pattern suitable for use with systemd,
                 // avoids redundant information
-                // "pattern": "%-5p %m\n",
+                "pattern": "%-5p %m\n"
 
                 // This governs whether the log output is flushed to disk after
                 // every write.
diff --git a/src/bin/keactrl/kea-netconf.conf.pre b/src/bin/keactrl/kea-netconf.conf.pre
index b559604..9f61162 100644
--- a/src/bin/keactrl/kea-netconf.conf.pre
+++ b/src/bin/keactrl/kea-netconf.conf.pre
@@ -69,11 +69,12 @@
                 // - syslog (logs to syslog)
                 // - syslog:name (logs to syslog using specified name)
                 // Any other value is considered a name of a time
-                "output": "kea-netconf.log"
+                // "output": "kea-netconf.log"
+                "output": "syslog",
 
                 // Shorter log pattern suitable for use with systemd,
                 // avoids redundant information
-                // "pattern": "%-5p %m\n"
+                "pattern": "%-5p %m\n"
 
                 // This governs whether the log output is flushed to disk after
                 // every write.
