diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2016-04-05 08:48:42.000000000 +0800
+++ b/configure.ac	2016-08-03 21:11:37.205777544 +0800
@@ -89,7 +89,7 @@
 # Check for python.
 AS_IF([test "x$enable_python" != "xno"], [
     AM_PATH_PYTHON(3)
-    PKG_CHECK_MODULES([PYTHON], [python3 >= 0.28])
+    PKG_CHECK_MODULES([PYTHON], [python-3.5 >= 0.28])
 ])
 AM_CONDITIONAL([BUILD_PYTHON], [test "x$enable_python" != "xno"])
 AM_CONDITIONAL([OSX], [test "x$_os" = "xDarwin"])
