diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8546c055..4546f93a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,7 @@
 #
 # -DGAMMARAY_INSTALL_QT_LAYOUT=[true|false]
 #  Install GammaRay in Qt folder using Qt layout folders:
-#    - plugins will go in <qt_folder>/plugins/gammaray
+#    - plugins will go in <qt_folder>/lib/qt/plugins/gammaray
 #    - all the other libs, including probe library will go in <qt_folder>/lib
 #    - qmake mkspecs file
 #  Default=false
@@ -621,7 +621,7 @@ if(APPLE)
     set(RESOURCES_INSTALL_DIR "${BUNDLE_INSTALL_DIR}/${BUNDLE_APP_NAME}/Contents/Resources")
     set(BIN_INSTALL_DIR "${BUNDLE_INSTALL_DIR}/${BUNDLE_APP_NAME}/Contents/MacOS")
     set(LIB_INSTALL_DIR "${BUNDLE_INSTALL_DIR}/${BUNDLE_APP_NAME}/Contents/Frameworks")
-    set(PLUGIN_INSTALL_DIR "${BUNDLE_INSTALL_DIR}/${BUNDLE_APP_NAME}/Contents/PlugIns/gammaray")
+    set(PLUGIN_INSTALL_DIR "${BUNDLE_INSTALL_DIR}/${BUNDLE_APP_NAME}/Contents/lib/qt/plugins/gammaray")
     set(LIBEXEC_INSTALL_DIR "${BIN_INSTALL_DIR}")
     set(MAN_INSTALL_DIR "${RESOURCES_INSTALL_DIR}/man/man1")
     set(DOC_INSTALL_DIR "${RESOURCES_INSTALL_DIR}/docs")
@@ -656,7 +656,7 @@ else()
   set(QCH_INSTALL_DIR "${CMAKE_INSTALL_DOCDIR}" CACHE PATH "Install location of Qt Assistant help files.")
   gammaray_convert_to_relative_path(QCH_INSTALL_DIR)
   if(WIN32)
-    set(PLUGIN_INSTALL_DIR "plugins/gammaray")
+    set(PLUGIN_INSTALL_DIR "lib/qt/plugins/gammaray")
     set(LIBEXEC_INSTALL_DIR "${BIN_INSTALL_DIR}")
     set(DOC_INSTALL_DIR .)
     set(TRANSLATION_INSTALL_DIR "translations")
@@ -680,9 +680,9 @@ if(GAMMARAY_INSTALL_QT_LAYOUT)
   else()
     set(PROBE_INSTALL_DIR ${LIB_INSTALL_DIR})
   endif()
-  set(PROBE_PLUGIN_INSTALL_DIR "plugins/gammaray")
+  set(PROBE_PLUGIN_INSTALL_DIR "lib/qt/plugins/gammaray")
   set(PLUGIN_INSTALL_DIR ${PROBE_PLUGIN_INSTALL_DIR})
-  set(TARGET_PLUGIN_INSTALL_DIR "plugins/gammaray-target")
+  set(TARGET_PLUGIN_INSTALL_DIR "lib/qt/plugins/gammaray-target")
 else()
   set(PROBE_INSTALL_DIR "${PLUGIN_INSTALL_DIR}/${GAMMARAY_PLUGIN_VERSION}/${GAMMARAY_PROBE_ABI}${GAMMARAY_PROBE_ABI_POSTFIX}")
   set(PROBE_PLUGIN_INSTALL_DIR ${PROBE_INSTALL_DIR})
