--- boringssl/include/openssl/base.h	2018-08-08 13:40:52.283719779 -0500
+++ boringssl.ppc/include/openssl/base.h	2018-08-08 13:40:30.383735109 -0500
@@ -110,6 +110,10 @@
 #define OPENSSL_32_BIT
 #elif defined(__myriad2__)
 #define OPENSSL_32_BIT
+#elif defined(__PPC__)
+#define OPENSSL_32_BIT
+#elif defined(__PPC64__)
+#define OPENSSL_64_BIT
 #else
 // Note BoringSSL only supports standard 32-bit and 64-bit two's-complement,
 // little-endian architectures. Functions will not produce the correct answer
