Description: fix other bugs

--- cpio-2.11.orig/po/Makefile.in.in
+++ cpio-2.11/po/Makefile.in.in
@@ -1,5 +1,3 @@
-# -*- buffer-read-only: t -*- vi: set ro:
-# DO NOT EDIT! GENERATED AUTOMATICALLY!
 # Makefile for PO directory in any package using GNU gettext.
 # Copyright (C) 1995-1997, 2000-2007 by Ulrich Drepper <drepper@gnu.ai.mit.edu>
 #
--- cpio-2.11.orig/lib/sysdep.h
+++ cpio-2.11/lib/sysdep.h
@@ -20,6 +20,8 @@
    You should have received a copy of the GNU General Public License
    along with GNU cpiio.  If not, see <http://www.gnu.org/licenses/>. */
 
+#ifndef LIB_SYSDEP_H
+#define LIB_SYSDEP_H 1
 
 #ifdef HAVE_PROCESS_H
 # include <process.h>
@@ -141,4 +143,4 @@ int link (const char *oldname,const char
 int chown (const char *filename,int owner,int group);
 #endif
 
-
+#endif
--- cpio-2.11.orig/lib/rtapelib.c
+++ cpio-2.11/lib/rtapelib.c
@@ -425,7 +425,9 @@ rmt_open__ (const char *file_name, int o
   }
 
   /* FIXME: Should somewhat validate the decoding, here.  */
+#ifndef __WIN32__
   if (gethostbyname (remote_host) == NULL)
+#endif
     error (EXIT_ON_EXEC_ERROR, 0, _("Cannot connect to %s: resolve failed"),
 	   remote_host);
 	  
--- cpio-2.11.orig/lib/sysdep.c
+++ cpio-2.11/lib/sysdep.c
@@ -187,4 +187,12 @@ chown ( const char *filename __attribute
   return -1;
 }
 #endif
-
+#ifdef __WIN32__
+# warning "Providing stub placeholder for readlink function"
+int
+readlink ( const char *path __attribute__ ((unused)) ,  char *buf __attribute__ ((unused)) ,  size_t bufsiz __attribute__ ((unused)) )
+{
+  errno = ENOSYS;
+  return -1;
+}
+#endif

--- cpio-2.11.orig/src/filetypes.h
+++ cpio-2.11/src/filetypes.h
@@ -81,5 +81,8 @@
 #ifndef S_ISLNK
 #define lstat stat
 #endif
+
+#ifndef __WIN32__
 int lstat ();
 int stat ();
+#endif

--- cpio-2.11.orig/src/main.c
+++ cpio-2.11/src/main.c
@@ -734,6 +734,8 @@ main (int argc, char *argv[])
   
   initialize_buffers ();
 
+  SET_BINARY_MODE (1);
+
   (*copy_function) ();
 
   if (archive_des >= 0 && rmtclose (archive_des) == -1)

