From efaf55ebe16ff633b2d010643a662763d715916a Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 14 May 2025 23:02:24 +0800
Subject: [PATCH 2/2] AOSCOS: fix(configure.ac): use sdl12_compat to provide
 SDL 1.x

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 75fb9eaa..1e53a904 100644
--- a/configure.ac
+++ b/configure.ac
@@ -396,7 +396,7 @@ fi
 if test  x$lv_tool_enabled = xyes -o x$examples = xyes ; then
   # Find SDL 1.x.x
   # On success, this will set and substitute SDL_CFLAGS and SDL_LIBS
-  PKG_CHECK_MODULES([SDL], [sdl >= 1.2.0], [], [
+  PKG_CHECK_MODULES([SDL], [sdl12_compat >= 1.2.0], [], [
     AC_MSG_ERROR([*** SDL 1.2.x not installed - please install first])
   ])
   if test x$examples = xyes ; then
-- 
2.49.0

