--- trace-cmd-v2.8.3/tracecmd/trace-hist.c	2019-07-25 00:41:28.000000000 +0000
+++ trace-cmd-v2.8.3.gcc10/tracecmd/trace-hist.c	2022-03-03 19:56:25.036689444 +0000
@@ -27,7 +27,7 @@
 
 static int long_size;
 
-struct tep_format_field *common_type_field;
+struct tep_format_field *common_type_hist;
 struct tep_format_field *common_pid_field;
 struct tep_format_field *sched_wakeup_comm_field;
 struct tep_format_field *sched_wakeup_new_comm_field;
@@ -568,7 +568,7 @@
 	unsigned long long val;
 	int type;
 
-	tep_read_number_field(common_type_field, record->data, &val);
+	tep_read_number_field(common_type_hist, record->data, &val);
 	type = val;
 
 	if (type == function_type)
@@ -956,8 +956,8 @@
 
 	long_size = tracecmd_long_size(handle);
 
-	common_type_field = tep_find_common_field(event, "common_type");
-	if (!common_type_field)
+	common_type_hist = tep_find_common_field(event, "common_type");
+	if (!common_type_hist)
 		die("Can't find a 'type' field?");
 
 	common_pid_field = tep_find_common_field(event, "common_pid");
--- trace-cmd-v2.8.3/tracecmd/trace-mem.c	2019-07-25 00:41:28.000000000 +0000
+++ trace-cmd-v2.8.3.gcc10/tracecmd/trace-mem.c	2022-03-03 19:55:51.895079900 +0000
@@ -30,7 +30,7 @@
 static int kmem_cache_alloc_node_type;
 static int kmem_cache_free_type;
 
-struct tep_format_field *common_type_field;
+struct tep_format_field *common_type_mem;
 
 struct tep_format_field *kmalloc_callsite_field;
 struct tep_format_field *kmalloc_bytes_req_field;
@@ -369,7 +369,7 @@
 	unsigned long long val;
 	int type;
 
-	tep_read_number_field(common_type_field, record->data, &val);
+	tep_read_number_field(common_type_mem, record->data, &val);
 	type = val;
 
 	if (type == kmalloc_type)
@@ -490,8 +490,8 @@
 	ret = tep_data_type(pevent, record);
 	event = tep_find_event(pevent, ret);
 
-	common_type_field = tep_find_common_field(event, "common_type");
-	if (!common_type_field)
+	common_type_mem = tep_find_common_field(event, "common_type");
+	if (!common_type_mem)
 		die("Can't find a 'type' field?");
 
 	update_kmalloc(pevent);
