--- a/GNUmakefile	2016-09-05 01:21:51.000000000 -0500
+++ b/GNUmakefile	2016-09-07 20:06:35.335369822 -0500
@@ -26,18 +26,6 @@
   BSD_LIBS:=	 $(shell $(PKG_CONFIG) --libs libbsd-overlay)
 endif
 
-# Test is some required libraries are installed. Rather bummer that
-# they are also required to run make clean or uninstall. Oh well... Who
-# does that?
-ifeq ($(BSD_LIBS),)
-  $(error You probably need to install "libbsd-dev" or "libbsd-devel" or something like that.)
-endif
-
-CURSES_LIBS:= $(shell $(PKG_CONFIG) --libs ncurses)
-ifeq ($(CURSES_LIBS),)
-  $(error You probably need to install "libncurses5-dev" or "libnnurses6-devel" or something like that.)
-endif
-
 CC?=		gcc
 CFLAGS?=	-O2 -pipe
 CFLAGS+=	-g -Wall
