From ab191f903dec4641e34fe5ea6b3048a63d8ed98e Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Tue, 11 Mar 2025 01:45:32 -0400
Subject: [PATCH] cc-input-chooser: fix potential nullptr hash table lookup

---
 panels/keyboard/cc-input-chooser.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/panels/keyboard/cc-input-chooser.c b/panels/keyboard/cc-input-chooser.c
index 728eb6d40..042199b0c 100644
--- a/panels/keyboard/cc-input-chooser.c
+++ b/panels/keyboard/cc-input-chooser.c
@@ -831,6 +831,9 @@ add_locale_to_table (GHashTable  *table,
 
   language = gnome_get_language_from_code (lang_code, NULL);
 
+  if (!language)
+    return;
+
   set = g_hash_table_lookup (table, language);
   if (!set)
     {
-- 
2.48.1

