--- a/Makefile.in	2002-12-28 13:57:37.000000000 +0800
+++ b/Makefile.in	2016-08-14 22:46:57.649882949 +0800
@@ -59,11 +59,9 @@
 	@make "PURIFY=$(PURE_ARGS)"
 
 install: $(TARGET) $(TARGET).6
-	touch sweeptimes
-	./install-sh -c -m 2555 -o root -g games -s $(TARGET) $(bindir)/$(TARGET)
-	./install-sh -c -m 0664 -o root -g games sweeptimes @SCORESDIR@/sweeptimes
-	./install-sh -c -m 0644 -o root -g games sweeprc @PREFSDIR@/sweeprc
-	./install-sh -c -m 0444 -o root -g games $(TARGET).6 @mandir@/man6/$(TARGET).6
+	./install-sh -c -m 2555 -o root -g games $(TARGET) $(DESTDIR)$(bindir)/$(TARGET)
+	./install-sh -c -m 0644 -o root -g games sweeprc $(DESTDIR)@PREFSDIR@/sweeprc
+	./install-sh -c -m 0444 -o root -g games $(TARGET).6 $(DESTDIR)@mandir@/man6/$(TARGET).6
 
 distclean:
 	@make sterile
