Index: freesweep-0.90/gpl.c
===================================================================
--- freesweep-0.90.orig/gpl.c
+++ freesweep-0.90/gpl.c
@@ -22,7 +22,7 @@
 "		       Version 2, June 1991",
 "",
 " Copyright (C) 1989, 1991 Free Software Foundation, Inc.",
-"     59 Temple Place, Suite 330, Boston, MA  02111-1307  USA",
+" 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA",
 " Everyone is permitted to copy and distribute verbatim copies",
 " of this license document, but changing it is not allowed.",
 "",
@@ -326,7 +326,8 @@
 "",
 "    You should have received a copy of the GNU General Public License",
 "    along with this program; if not, write to the Free Software",
-"    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA",
+"    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,",
+"    MA 02110-1301 USA.",
 "",
 "",
 "Also add information on how to contact you by electronic and paper mail.",
