--- a/peach.pas	2021-01-15 15:04:15.869899477 +0800
+++ b/peach.pas	2021-01-15 17:11:49.610374735 +0800
@@ -38660,7 +38660,7 @@
 
 procedure TForm_peach.baboutlocalhelpClick(Sender: TObject);
 begin
-cp_open(executable_path+'peazip_help.pdf',desk_env);
+cp_open('/usr/share/doc/peazip/peazip_help.pdf',desk_env);
 end;
 
 procedure TForm_peach.baboutremoveadditionalClick(Sender: TObject);
@@ -59635,7 +59635,7 @@
 
 procedure TForm_peach.mpeahelpofflineClick(Sender: TObject);
 begin
-cp_open(executable_path+'peazip_help.pdf',desk_env);
+cp_open('/usr/share/doc/peazip/peazip_help.pdf',desk_env);
 end;
 
 procedure TForm_peach.mRecent9Click(Sender: TObject);
