From 71fbfe13dbf8af61bbe1786b7529320c9ac81c6b Mon Sep 17 00:00:00 2001
From: XiaoTan <tocker@16iot.cn>
Date: Tue, 2 Sep 2025 13:03:11 +0800
Subject: [PATCH] angie.conf: adapt to AOSC OS defaults

---
 conf/angie.conf | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/conf/angie.conf b/conf/angie.conf
index 6627092fd..4218ada96 100644
--- a/conf/angie.conf
+++ b/conf/angie.conf
@@ -1,6 +1,6 @@
 
 #user  nobody;
-worker_processes  1;
+worker_processes  auto;
 
 #error_log  logs/error.log;
 #error_log  logs/error.log  notice;
@@ -30,7 +30,7 @@ http {
     #keepalive_timeout  0;
     keepalive_timeout  65;
 
-    #gzip  on;
+    gzip  on;
 
     server {
         listen       80;
@@ -41,8 +41,8 @@ http {
         #access_log  logs/host.access.log  main;
 
         location / {
-            root   html;
-            index  index.html index.htm;
+            root   /usr/share/angie/html;
+	    index  index.html index.htm;
         }
 
         #error_page  404              /404.html;
@@ -51,7 +51,7 @@ http {
         #
         error_page   500 502 503 504  /50x.html;
         location = /50x.html {
-            root   html;
+             root   /usr/share/angie/html;
         }
 
         # proxy the PHP scripts to Apache listening on 127.0.0.1:80
@@ -87,7 +87,7 @@ http {
     #    server_name  somename  alias  another.alias;
 
     #    location / {
-    #        root   html;
+    #        root   /usr/share/angie/html;
     #        index  index.html index.htm;
     #    }
     #}
@@ -109,7 +109,7 @@ http {
     #    ssl_prefer_server_ciphers  on;
 
     #    location / {
-    #        root   html;
+    #        root   /usr/share/angie/html;
     #        index  index.html index.htm;
     #    }
     #}
-- 
2.51.0

