diff --git a/kcms/region_language/localegeneratorgeneratedglibc.cpp b/kcms/region_language/localegeneratorgeneratedglibc.cpp
index bd06daea5..6f2b178cc 100644
--- a/kcms/region_language/localegeneratorgeneratedglibc.cpp
+++ b/kcms/region_language/localegeneratorgeneratedglibc.cpp
@@ -9,5 +9,5 @@
 void LocaleGeneratorGeneratedGlibc::localesGenerate(const QStringList &list)
 {
     Q_UNUSED(list)
-    Q_EMIT needsFont();
+    Q_EMIT success();
 }
