diff --git a/kcms/region_language/localegeneratorgeneratedglibc.cpp b/kcms/region_language/localegeneratorgeneratedglibc.cpp
index a2f05a238f..32d8aff89e 100644
--- a/kcms/region_language/localegeneratorgeneratedglibc.cpp
+++ b/kcms/region_language/localegeneratorgeneratedglibc.cpp
@@ -9,7 +9,7 @@
 void LocaleGeneratorGeneratedGlibc::localesGenerate(const QStringList &list)
 {
     Q_UNUSED(list)
-    Q_EMIT needsFont();
+    Q_EMIT success();
 }
 
 #include "moc_localegeneratorgeneratedglibc.cpp"
