diff -Naur linux-5.4.29/drivers/video/fbdev/aty/atyfb_base.c linux-5.4.29.atyc1/drivers/video/fbdev/aty/atyfb_base.c
--- linux-5.4.29/drivers/video/fbdev/aty/atyfb_base.c	2020-04-01 04:02:18.000000000 -0500
+++ linux-5.4.29.atyc1/drivers/video/fbdev/aty/atyfb_base.c	2020-04-17 03:38:56.906110521 -0500
@@ -690,10 +690,16 @@
 {
 #ifdef CONFIG_FB_ATY_GENERIC_LCD
 	if (par->lcd_table != 0) {
+		unsigned long hs;
 		/* stop CRTC */
 		aty_st_le32(CRTC_GEN_CNTL, crtc->gen_cntl &
 			    ~(CRTC_EXT_DISP_EN | CRTC_EN), par);
 
+		/* VAIO PCG-C1VRX/K */
+		hs=aty_ld_le32(OVR_WID_LEFT_RIGHT,par);
+		hs &= ~0x003F003F;
+		aty_st_le32(OVR_WID_LEFT_RIGHT, hs, par);
+
 		/* update non-shadow registers first */
 		aty_st_lcd(CNFG_PANEL, crtc->lcd_config_panel, par);
 		aty_st_lcd(LCD_GEN_CNTL, crtc->lcd_gen_cntl &
