From a3aca6eaffbeb8586b0fc3776bf2bea86b1b34f3 Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Mon, 8 Jan 2024 21:52:58 -0800
Subject: [PATCH] update default config

---
 doc/example.conf.in | 129 ++++++++++++++++++++++++--------------------
 1 file changed, 70 insertions(+), 59 deletions(-)

diff --git a/doc/example.conf.in b/doc/example.conf.in
index 6bf1c668..0057bbc5 100644
--- a/doc/example.conf.in
+++ b/doc/example.conf.in
@@ -51,11 +51,18 @@ server:
 	# specify 0.0.0.0 and ::0 to bind to all available interfaces.
 	# specify every interface[@port] on a new 'interface:' labelled line.
 	# The listen interfaces are not changed on reload, only on restart.
+	# interface: 0.0.0.0
+	# interface: ::0
 	# interface: 192.0.2.153
 	# interface: 192.0.2.154
 	# interface: 192.0.2.154@5003
 	# interface: 2001:DB8::5
 	# interface: eth0@5003
+	# for DNS over TLS and raw DNS over port 80
+	# interface: 0.0.0.0@443
+	# interface: ::0@443
+	# interface: 0.0.0.0@80
+	# interface: ::0@80
 
 	# enable this feature to copy the source address of queries to reply.
 	# Socket options are not supported on all platforms. experimental.
@@ -121,7 +128,7 @@ server:
 
 	# use SO_REUSEPORT to distribute queries over threads.
 	# at extreme load it could be better to turn it off to distribute even.
-	# so-reuseport: yes
+	so-reuseport: yes
 
 	# use IP_TRANSPARENT so the interface: addresses can be non-local
 	# and you can config non-existing IPs that are going to work later on
@@ -281,7 +288,7 @@ server:
 	# tcp-idle-timeout: 30000
 
 	# Enable EDNS TCP keepalive option.
-	# edns-tcp-keepalive: no
+	edns-tcp-keepalive: yes
 
 	# Timeout for EDNS TCP keepalive, in msec.
 	# edns-tcp-keepalive-timeout: 120000
@@ -406,14 +413,14 @@ server:
 	# if given, user privileges are dropped (after binding port),
 	# and the given username is assumed. Default is user "unbound".
 	# If you give "" no privileges are dropped.
-	# username: "@UNBOUND_USERNAME@"
+	username: "@UNBOUND_USERNAME@"
 
 	# the working directory. The relative files in this config are
 	# relative to this directory. If you give "" the working directory
 	# is not changed.
 	# If you give a server: directory: dir before include: file statements
 	# then those includes can be relative to the working directory.
-	# directory: "@UNBOUND_RUN_DIR@"
+	directory: "@UNBOUND_RUN_DIR@"
 
 	# the log file, "" means log to stderr.
 	# Use of this option sets use-syslog to "no".
@@ -428,7 +435,7 @@ server:
 	# log-identity: ""
 
 	# print UTC timestamp in ascii to logfile, default is epoch in seconds.
-	# log-time-ascii: no
+	log-time-ascii: yes
 
 	# print one line with time, IP, name, type, class for every query.
 	# log-queries: no
@@ -497,22 +504,22 @@ server:
 	# harden-large-queries: no
 
 	# Harden against out of zone rrsets, to avoid spoofing attempts.
-	# harden-glue: yes
+	harden-glue: yes
 
 	# Harden against receiving dnssec-stripped data. If you turn it
 	# off, failing to validate dnskey data for a trustanchor will
 	# trigger insecure mode for that zone (like without a trustanchor).
 	# Default on, which insists on dnssec data for trust-anchored zones.
-	# harden-dnssec-stripped: yes
+	harden-dnssec-stripped: yes
 
 	# Harden against queries that fall under dnssec-signed nxdomain names.
-	# harden-below-nxdomain: yes
+	harden-below-nxdomain: yes
 
 	# Harden the referral path by performing additional queries for
 	# infrastructure data.  Validates the replies (if possible).
 	# Default off, because the lookups burden the server.  Experimental
 	# implementation of draft-wijngaards-dnsext-resolver-side-mitigation.
-	# harden-referral-path: no
+	harden-referral-path: yes
 
 	# Harden against algorithm downgrade when multiple algorithms are
 	# advertised in the DS record.  If no, allows the weakest algorithm
@@ -526,7 +533,7 @@ server:
 	# Sent minimum amount of information to upstream servers to enhance
 	# privacy. Only sent minimum required labels of the QNAME and set QTYPE
 	# to A when possible.
-	# qname-minimisation: yes
+	qname-minimisation: yes
 
 	# QNAME minimisation in strict mode. Do not fall-back to sending full
 	# QNAME to potentially broken nameservers. A lot of domains will not be
@@ -536,7 +543,7 @@ server:
 
 	# Aggressive NSEC uses the DNSSEC NSEC chain to synthesize NXDOMAIN
 	# and other denials, using information from previous NXDOMAINs answers.
-	# aggressive-nsec: yes
+	aggressive-nsec: yes
 
 	# Use 0x20-encoded random bits in the query to foil spoof attempts.
 	# This feature is an experimental implementation of draft dns-0x20.
@@ -569,7 +576,7 @@ server:
 	# threshold, a warning is printed and a defensive action is taken,
 	# the cache is cleared to flush potential poison out of it.
 	# A suggested value is 10000000, the default is 0 (turned off).
-	# unwanted-reply-threshold: 0
+	unwanted-reply-threshold: 0
 
 	# Do not query the following addresses. No DNS queries are sent there.
 	# List one address per entry. List classless netblocks with /size,
@@ -581,20 +588,20 @@ server:
 	# do-not-query-localhost: yes
 
 	# if yes, perform prefetching of almost expired message cache entries.
-	# prefetch: no
+	prefetch: yes
 
 	# if yes, perform key lookups adjacent to normal lookups.
-	# prefetch-key: no
+	prefetch-key: yes
 
 	# deny queries of type ANY with an empty response.
-	# deny-any: no
+	deny-any: yes
 
 	# if yes, Unbound rotates RRSet order in response.
-	# rrset-roundrobin: yes
+	rrset-roundrobin: yes
 
 	# if yes, Unbound doesn't insert authority/additional sections
 	# into response messages when those sections are not required.
-	# minimal-responses: yes
+	minimal-responses: yes
 
 	# true to disable DNSSEC lameness check in iterator.
 	# disable-dnssec-lame-check: no
@@ -604,7 +611,7 @@ server:
 	# most modules have to be listed at the beginning of the line,
 	# except cachedb(just before iterator), and python (at the beginning,
 	# or, just before the iterator).
-	# module-config: "validator iterator"
+	module-config: "ipsecmod validator iterator"
 
 	# File with trusted keys, kept uptodate using RFC5011 probes,
 	# initial file like trust-anchor-file, then it stores metadata.
@@ -615,13 +622,13 @@ server:
 	# And then enable the auto-trust-anchor-file config item.
 	# Please note usage of unbound-anchor root anchor is at your own risk
 	# and under the terms of our LICENSE (see that file in the source).
-	# auto-trust-anchor-file: "@UNBOUND_ROOTKEY_FILE@"
+	auto-trust-anchor-file: "@UNBOUND_ROOTKEY_FILE@"
 
 	# trust anchor signaling sends a RFC8145 key tag query after priming.
-	# trust-anchor-signaling: yes
+	trust-anchor-signaling: yes
 
 	# Root key trust anchor sentinel (draft-ietf-dnsop-kskroll-sentinel)
-	# root-key-sentinel: yes
+	root-key-sentinel: yes
 
 	# File with trusted keys for validation. Specify more than one file
 	# with several entries, one file per entry.
@@ -641,7 +648,7 @@ server:
 	# but has a different file format. Format is BIND-9 style format,
 	# the trusted-keys { name flag proto algo "key"; }; clauses are read.
 	# you need external update procedures to track changes in keys.
-	# trusted-keys-file: ""
+	trusted-keys-file: @UNBOUND_RUN_DIR@/keys.d/*.key
 
 	# Ignore chain of trust. Domain is treated as insecure.
 	# domain-insecure: "example.com"
@@ -669,14 +676,14 @@ server:
 	# unsecure data. Useful to shield the users of this validator from
 	# potential bogus data in the additional section. All unsigned data
 	# in the additional section is removed from secure messages.
-	# val-clean-additional: yes
+	val-clean-additional: yes
 
 	# Turn permissive mode on to permit bogus messages. Thus, messages
 	# for which security checks failed will be returned to clients,
 	# instead of SERVFAIL. It still performs the security checks, which
 	# result in interesting log files and possibly the AD bit in
 	# replies if the message is found secure. The default is off.
-	# val-permissive-mode: no
+	val-permissive-mode: no
 
 	# Ignore the CD flag in incoming queries and refuse them bogus data.
 	# Enable it if the only clients of Unbound are legacy servers (w2008)
@@ -690,11 +697,11 @@ server:
 
 	# Serve expired responses from cache, with serve-expired-reply-ttl in
 	# the response, and then attempt to fetch the data afresh.
-	# serve-expired: no
+	serve-expired: yes
 	#
 	# Limit serving of expired responses to configured seconds after
 	# expiration. 0 disables the limit.
-	# serve-expired-ttl: 0
+	serve-expired-ttl: 14400
 	#
 	# Set the TTL of expired records to the serve-expired-ttl value after a
 	# failed attempt to retrieve the record from upstream. This makes sure
@@ -865,6 +872,8 @@ server:
 	# you need to do the reverse notation yourself.
 	# local-data-ptr: "192.0.2.3 www.example.com"
 
+	include: @UNBOUND_RUN_DIR@/local.d/*.conf
+
 	# tag a localzone with a list of tag names (in "" with spaces between)
 	# local-zone-tag: "example.com" "tag2 tag3"
 
@@ -875,8 +884,8 @@ server:
 	# the TLS stream, and over HTTPS using HTTP/2 as specified in RFC8484.
 	# Give the certificate to use and private key.
 	# default is "" (disabled).  requires restart to take effect.
-	# tls-service-key: "path/to/privatekeyfile.key"
-	# tls-service-pem: "path/to/publiccertfile.pem"
+	# tls-service-key: "@UNBOUND_RUN_DIR@/unbound_server.key"
+	# tls-service-pem: "@UNBOUND_RUN_DIR@/unbound_server.pem"
 	# tls-port: 853
 	# https-port: 443
 
@@ -1005,12 +1014,12 @@ server:
 	# fast-server-num: 3
 
 	# Enable to attach Extended DNS Error codes (RFC8914) to responses.
-	# ede: no
+	ede: yes
 
 	# Enable to attach an Extended DNS Error (RFC8914) Code 3 - Stale
 	# Answer as EDNS0 option to expired responses.
 	# Note that the ede option above needs to be enabled for this to work.
-	# ede-serve-expired: no
+	ede-serve-expired: yes
 
 	# Specific options for ipsecmod. Unbound needs to be configured with
 	# --enable-ipsecmod for these to take effect.
@@ -1018,11 +1027,11 @@ server:
 	# Enable or disable ipsecmod (it still needs to be defined in
 	# module-config above). Can be used when ipsecmod needs to be
 	# enabled/disabled via remote-control(below).
-	# ipsecmod-enabled: yes
+	ipsecmod-enabled: no
 	#
 	# Path to executable external hook. It must be defined when ipsecmod is
 	# listed in module-config (above).
-	# ipsecmod-hook: "./my_executable"
+	ipsecmod-hook: "/usr/libexec/ipsec/_unbound-hook"
 	#
 	# When enabled Unbound will reply with SERVFAIL if the return value of
 	# the ipsecmod-hook is not 0.
@@ -1073,7 +1082,7 @@ dynlib:
 remote-control:
 	# Enable remote control with unbound-control(8) here.
 	# set up the keys and certificates with unbound-control-setup.
-	# control-enable: no
+	control-enable: yes
 
 	# what interfaces are listened to for remote control.
 	# give 0.0.0.0 and ::0 to listen to all interfaces.
@@ -1087,19 +1096,19 @@ remote-control:
 
 	# for localhost, you can disable use of TLS by setting this to "no"
 	# For local sockets this option is ignored, and TLS is not used.
-	# control-use-cert: "yes"
+	control-use-cert: "no"
 
 	# Unbound server key file.
-	# server-key-file: "@UNBOUND_RUN_DIR@/unbound_server.key"
+	server-key-file: "@UNBOUND_RUN_DIR@/unbound_server.key"
 
 	# Unbound server certificate file.
-	# server-cert-file: "@UNBOUND_RUN_DIR@/unbound_server.pem"
+	server-cert-file: "@UNBOUND_RUN_DIR@/unbound_server.pem"
 
 	# unbound-control key file.
-	# control-key-file: "@UNBOUND_RUN_DIR@/unbound_control.key"
+	control-key-file: "@UNBOUND_RUN_DIR@/unbound_control.key"
 
 	# unbound-control certificate file.
-	# control-cert-file: "@UNBOUND_RUN_DIR@/unbound_control.pem"
+	control-cert-file: "@UNBOUND_RUN_DIR@/unbound_control.pem"
 
 # Stub zones.
 # Create entries like below, to make all queries for 'example.com' and
@@ -1139,6 +1148,8 @@ remote-control:
 # 	name: "example.org"
 # 	forward-host: fwd.example.com
 
+include: @UNBOUND_RUN_DIR@/conf.d/*.conf
+
 # Authority zones
 # The data for these zones is kept locally, from a file or downloaded.
 # The data can be served to downstream clients, or used instead of the
@@ -1148,27 +1159,27 @@ remote-control:
 # download it), primary: fetches with AXFR and IXFR, or url to zonefile.
 # With allow-notify: you can give additional (apart from primaries and urls)
 # sources of notifies.
-# auth-zone:
-#	name: "."
-#	primary: 199.9.14.201         # b.root-servers.net
-#	primary: 192.33.4.12          # c.root-servers.net
-#	primary: 199.7.91.13          # d.root-servers.net
-#	primary: 192.5.5.241          # f.root-servers.net
-#	primary: 192.112.36.4         # g.root-servers.net
-#	primary: 193.0.14.129         # k.root-servers.net
-#	primary: 192.0.47.132         # xfr.cjr.dns.icann.org
-#	primary: 192.0.32.132         # xfr.lax.dns.icann.org
-#	primary: 2001:500:200::b      # b.root-servers.net
-#	primary: 2001:500:2::c        # c.root-servers.net
-#	primary: 2001:500:2d::d       # d.root-servers.net
-#	primary: 2001:500:2f::f       # f.root-servers.net
-#	primary: 2001:500:12::d0d     # g.root-servers.net
-#	primary: 2001:7fd::1          # k.root-servers.net
-#	primary: 2620:0:2830:202::132 # xfr.cjr.dns.icann.org
-#	primary: 2620:0:2d0:202::132  # xfr.lax.dns.icann.org
-#	fallback-enabled: yes
-#	for-downstream: no
-#	for-upstream: yes
+auth-zone:
+	name: "."
+	primary: 199.9.14.201         # b.root-servers.net
+	primary: 192.33.4.12          # c.root-servers.net
+	primary: 199.7.91.13          # d.root-servers.net
+	primary: 192.5.5.241          # f.root-servers.net
+	primary: 192.112.36.4         # g.root-servers.net
+	primary: 193.0.14.129         # k.root-servers.net
+	primary: 192.0.47.132         # xfr.cjr.dns.icann.org
+	primary: 192.0.32.132         # xfr.lax.dns.icann.org
+	primary: 2001:500:200::b      # b.root-servers.net
+	primary: 2001:500:2::c        # c.root-servers.net
+	primary: 2001:500:2d::d       # d.root-servers.net
+	primary: 2001:500:2f::f       # f.root-servers.net
+	primary: 2001:500:12::d0d     # g.root-servers.net
+	primary: 2001:7fd::1          # k.root-servers.net
+	primary: 2620:0:2830:202::132 # xfr.cjr.dns.icann.org
+	primary: 2620:0:2d0:202::132  # xfr.lax.dns.icann.org
+	fallback-enabled: yes
+	for-downstream: no
+	for-upstream: yes
 # auth-zone:
 #	name: "example.org"
 #	for-downstream: yes
-- 
2.39.1

