--- ash-0.4.0/jobs.c.orig	Tue Jul  3 19:10:28 2001
+++ ash-0.4.0/jobs.c	Tue Jul  3 19:12:11 2001
@@ -712,6 +712,7 @@
 		} else if (mode == FORK_BG) {
 			ignoresig(SIGINT);
 			ignoresig(SIGQUIT);
+			ignoresig(SIGHUP);
 			if ((jp == NULL || jp->nprocs == 0) &&
 			    ! fd0_redirected_p ()) {
 				close(0);
@@ -723,6 +724,7 @@
 		if (mode == FORK_BG) {
 			ignoresig(SIGINT);
 			ignoresig(SIGQUIT);
+			ignoresig(SIGHUP);
 			if ((jp == NULL || jp->nprocs == 0) &&
 			    ! fd0_redirected_p ()) {
 				close(0);
