From 622261b820e17e6ef376c3a12c31b63cc8562339 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Thu, 30 Nov 2023 11:26:18 +0800
Subject: [PATCH] fix: remove useless line to fix `ppc64el` arch build

---
 src/network.rs | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/network.rs b/src/network.rs
index 89bf93a..e8ca04b 100644
--- a/src/network.rs
+++ b/src/network.rs
@@ -131,7 +131,6 @@ pub fn get_variants() -> Result<Vec<VariantEntry>> {
 #[cfg(target_arch = "powerpc64")]
 #[inline]
 pub(crate) fn get_arch_name() -> Option<&'static str> {
-    use nix::libc;
     let mut endian: libc::c_int = -1;
     let result;
     unsafe {
