diff --git a/renpy/script.py b/renpy/script.py
index e061d6a4d..1343611ba 100644
--- a/renpy/script.py
+++ b/renpy/script.py
@@ -879,7 +879,7 @@ class Script(object):
                     bad_reduction_name=f"<{fn} transformed rpyc data>",
                 )
 
-                if not renpy.macapp:
+                if not renpy.macapp and os.access(os.path.dirname(rpycfn), os.W_OK):
                     try:
                         with open(rpycfn, "wb") as f:
                             self.write_rpyc_header(f)