From c7f0c4b5230f93d8de2f936745ff8670b8bbd90c Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Sat, 31 May 2025 20:26:04 +0800
Subject: [PATCH 3/3] AOSCOS: Bypass for read-only system directory

---
 renpy/script.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/renpy/script.py b/renpy/script.py
index 0b875fc5ae0f..411b300c7c5e 100644
--- a/renpy/script.py
+++ b/renpy/script.py
@@ -763,7 +763,7 @@ class Script(object):
 
                 pickle_data_after_static_transforms = dumps((data, stmts))
 
-                if not renpy.macapp:
+                if not renpy.macapp and not rpycfn.startswith("/usr/share/renpy"):
                     try:
                         with open(rpycfn, "wb") as f:
                             self.write_rpyc_header(f)
-- 
2.49.0

