diff --git a/src/Makefile.am b/src/Makefile.am
index ef7b092..c057721 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -19,8 +19,6 @@ if WITH_H265
 backend_c += h265.c
 endif
 
-backend_s = tiled_yuv.S
-
 backend_h = request.h object_heap.h config.h surface.h context.h buffer.h \
 	mpeg2.h picture.h subpicture.h image.h v4l2.h video.h media.h utils.h \
 	tiled_yuv.h h264.h h265.h
diff --git a/src/image.c b/src/image.c
index 476007b..352dfdb 100644
--- a/src/image.c
+++ b/src/image.c
@@ -175,11 +175,7 @@ VAStatus RequestDeriveImage(VADriverContextP context, VASurfaceID surface_id,
 
 	for (i = 0; i < surface_object->destination_planes_count; i++) {
 		if (!video_format_is_linear(driver_data->video_format))
-			tiled_to_planar(surface_object->destination_data[i],
-					buffer_object->data + image->offsets[i],
-					image->pitches[i], image->width,
-					i == 0 ? image->height :
-						 image->height / 2);
+			return VA_STATUS_ERROR_INVALID_BUFFER;
 		else {
 			memcpy(buffer_object->data + image->offsets[i],
 			       surface_object->destination_data[i],
diff --git a/src/video.c b/src/video.c
index b8a75a0..e23f957 100644
--- a/src/video.c
+++ b/src/video.c
@@ -45,16 +45,6 @@ static struct video_format formats[] = {
 		.planes_count		= 2,
 		.bpp			= 16,
 	},
-	{
-		.description		= "Sunxi tiled NV12 YUV",
-		.v4l2_format		= V4L2_PIX_FMT_SUNXI_TILED_NV12,
-		.v4l2_buffers_count	= 1,
-		.v4l2_mplane		= false,
-		.drm_format		= DRM_FORMAT_NV12,
-		.drm_modifier		= DRM_FORMAT_MOD_ALLWINNER_MB32_TILED,
-		.planes_count		= 2,
-		.bpp			= 16
-	},
 };
 
 static unsigned int formats_count = sizeof(formats) / sizeof(formats[0]);
