From 5de4ff8e8a5d564ce00e18064d300e7a545123cb Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Wed, 21 Aug 2024 19:41:55 -0700
Subject: [PATCH] setup.py: use cython by default

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 08fc003..ef26000 100644
--- a/setup.py
+++ b/setup.py
@@ -10,7 +10,7 @@ def pkgconfig(package, kw):
         kw.setdefault(flag_map.get(token[:2]), []).append(token[2:])
     return kw
 
-USE_CYTHON = False
+USE_CYTHON = True
 if "--cython" in sys.argv:
     sys.argv.remove("--cython")
     USE_CYTHON = True
-- 
2.46.0

