diff -Naur a/tools/compilers/roqvq/codec.cpp b/tools/compilers/roqvq/codec.cpp
--- a/tools/compilers/roqvq/codec.cpp	2018-05-27 19:25:56.498170400 +0000
+++ b/tools/compilers/roqvq/codec.cpp	2018-09-19 00:00:53.291587094 +0000
@@ -309,7 +309,7 @@
 int	codec::BestCodeword( unsigned char *tempvector, int dimension, VQDATA **codebook )
 {
 	VQDATA dist;
-	VQDATA bestDist = HUGE;
+	VQDATA bestDist = std::numeric_limits<double>::max();
 	VQDATA tempvq[64];
 	int bestIndex = -1;
 
@@ -1492,7 +1492,7 @@
 		//
 		// find the closest two and eliminate one
 		//
-		double bestDist = HUGE;
+		double bestDist = std::numeric_limits<double>::max();
 		double dist, simport;
 		int bestIndex = -1;
 		int bestOtherIndex = 0;
@@ -1558,7 +1558,7 @@
 		// until we have reduced it to 256 entries, find one to toss
 		//
 		do {
-			bestDist = HUGE;
+			bestDist = std::numeric_limits<double>::max();
 			bestIndex = -1;
 			bestOtherIndex = -1;
 			if (optimize) {
@@ -1573,7 +1573,7 @@
 				}
 			}
 			if (bestIndex == -1 || !optimize) {
-				bestDist = HUGE;
+				bestDist = std::numeric_limits<double>::max();
 				bestIndex = -1;
 				bestOtherIndex = 0;
 				aentries = 0;
