diff -Naur a/sys/scons/SConscript.darkmod b/sys/scons/SConscript.darkmod
--- a/sys/scons/SConscript.darkmod	2018-05-27 19:25:40.910170400 +0000
+++ b/sys/scons/SConscript.darkmod	2018-09-19 00:06:19.544890471 +0000
@@ -56,24 +56,16 @@
 		File('#/linux/ffmpeg/libswresample.a')
 	])
 if (TARGET_ARCH == 'x64'):
-	local_env.Append(LIBS = [
-		File('#/linux/libcurl/lib64/libcurl.a'),
-		File('#/linux/polarssl/lib64/libpolarssl.a'),
-		File('#/linux/ffmpeg/lib64/libavformat.a'),
-		File('#/linux/ffmpeg/lib64/libavcodec.a'),
-		File('#/linux/ffmpeg/lib64/libavutil.a'),
-		File('#/linux/ffmpeg/lib64/libswscale.a'),
-		File('#/linux/ffmpeg/lib64/libswresample.a')
-	])
+	pass
 
 # Add all standard libraries (taken from system)
 local_env.Append( LIBS = [ 'X11', 'Xext', 'Xxf86vm' ] ) # 'Xxf86dga', 
-#local_env.Append( LIBPATH = [ '/usr/X11R6/lib' ] )
-local_env.Append( LIBPATH = [ '/usr/lib/i386-linux-gnu' ] )
 local_env.Append( LIBS = [ 'openal' ] )
 local_env.Append( LIBS = [ 'dl' ] )
 local_env.Append( LIBS = [ 'pthread' ] )
 local_env.Append( LIBS = [ 'stdc++fs' ] )
+local_env.Append( LIBS = [ 'curl', 'boost_filesystem', 'libboost_system', 'boost_thread', 'png', 'z', 'rt' ] )
+local_env.Append( LIBS = [ 'avcodec', 'avdevice', 'avfilter', 'avformat', 'avutil', 'postproc', 'swresample', 'swscale', 'vorbis', 'vorbisfile' ])
 
 thedarkmod = local_env.Program( target = 'thedarkmod', source = source_list )
 Return( 'thedarkmod' )
diff -Naur a/sys/scons/SConscript.extlibs b/sys/scons/SConscript.extlibs
--- a/sys/scons/SConscript.extlibs	2018-05-27 19:25:40.911170400 +0000
+++ b/sys/scons/SConscript.extlibs	2018-09-19 00:02:15.961628475 +0000
@@ -33,16 +33,10 @@
 obj_list = []
 for f in extlibs_list:
 	obj_list += local_env.StaticObject( source = f )
-# Add ogg library
-obj_list += SConscript( '../../sys/scons/SConscript.ogg' )
-# Add curl library (built separately)
-obj_list += curl_lib
 # greebo: Use custom compiled devIL and libjpeg (since the engine requires 4 bytes per pixel format) in Linux
 obj_list += SConscript( '../../sys/scons/SConscript.devil' )
 obj_list += SConscript( '../../sys/scons/SConscript.libjpeg' )
-obj_list += SConscript( '../../sys/scons/SConscript.libpng' )
 # stgatilov: we need only zlib code, because internal minizip uses it
-obj_list += SConscript( '../../sys/scons/SConscript.zlib' )
 
 # Note: Unlike windows version, we built ExtLibs as static library,
 # because Linux forbids wrapping prebuilt static libraries into a shared library.
