--- tdm-orig/CMakeLists.txt	2020-06-27 13:39:29.788169300 -0600
+++ tdm/CMakeLists.txt	2021-02-05 15:57:23.828659080 -0700
@@ -206,17 +206,23 @@
 endif ()
 ucm_set_runtime(STATIC)
 
-find_package(ZLIB REQUIRED HINTS ${PACKAGE_CONFIG_PATH})
-find_package(CURL REQUIRED HINTS ${PACKAGE_CONFIG_PATH})
-find_package(OPENAL REQUIRED HINTS ${PACKAGE_CONFIG_PATH})
-find_package(OGG REQUIRED HINTS ${PACKAGE_CONFIG_PATH})
-find_package(VORBIS REQUIRED HINTS ${PACKAGE_CONFIG_PATH})
-find_package(DEVIL REQUIRED HINTS ${PACKAGE_CONFIG_PATH})
-find_package(JPEG REQUIRED HINTS ${PACKAGE_CONFIG_PATH})
-find_package(FFMPEG REQUIRED HINTS ${PACKAGE_CONFIG_PATH})
-find_package(PUGIXML REQUIRED HINTS ${PACKAGE_CONFIG_PATH})
-find_package(PNG REQUIRED HINTS ${PACKAGE_CONFIG_PATH})
-find_package(MBEDTLS REQUIRED HINTS ${PACKAGE_CONFIG_PATH})
+include(FindPkgConfig)
+if (NOT PKG_CONFIG_FOUND)
+    message(FATAL_ERROR "pkg-config not found!")
+endif (NOT PKG_CONFIG_FOUND)
+
+find_package(ZLIB REQUIRED)
+PKG_CHECK_MODULES(MINIZIP minizip)
+PKG_CHECK_MODULES(VORBIS vorbis vorbisfile)
+PKG_CHECK_MODULES(FFMPEG libavcodec libavformat libavutil libswresample libswscale)
+find_package(CURL REQUIRED)
+find_package(OpenAL REQUIRED)
+find_package(Ogg REQUIRED)
+find_package(DevIL REQUIRED)
+find_package(JPEG REQUIRED)
+find_package(pugixml REQUIRED)
+find_package(PNG REQUIRED)
+find_package(OpenSSL REQUIRED)
 
 include_directories(
 		"${PROJECT_SOURCE_DIR}"
@@ -224,31 +230,33 @@
 		"${PROJECT_SOURCE_DIR}/framework"
 		"${PROJECT_SOURCE_DIR}/ThirdParty/artefacts/doctest/include"
 		${ZLIB_INCLUDE_DIRS}
+                ${MINIZIP_INCLUDE_DIRS}
 		${CURL_INCLUDE_DIRS}
-		${OPENAL_INCLUDE_DIRS}
+                ${OPENAL_INCLUDE_DIR}
 		${OGG_INCLUDE_DIRS}
 		${VORBIS_INCLUDE_DIRS}
-		${DEVIL_INCLUDE_DIRS}
+		${IL_INCLUDE_DIR}
 		${JPEG_INCLUDE_DIRS}
 		${FFMPEG_INCLUDE_DIRS}
 		${PUGIXML_INCLUDE_DIRS}
 		${PNG_INCLUDE_DIRS}
-		${MBEDTLS_INCLUDE_DIRS}
+                ${OPENSSL_INCLUDE_DIR}
 )
 
 target_link_libraries(TheDarkMod
+                ${MINIZIP_LIBRARIES}
 		${ZLIB_LIBRARIES}
 		${CURL_LIBRARIES}
-		${OPENAL_LIBRARIES}
+                ${OPENAL_LIBRARY}
 		${VORBIS_LIBRARIES}
-		${OGG_LIBRARIES}
-		${DEVIL_LIBRARIES}
+                ${OGG_LIBRARIES}
+                ${IL_LIBRARIES}
 		${JPEG_LIBRARIES}
 		${FFMPEG_LIBRARIES}
 		${PNG_LIBRARIES}
-		${MBEDTLS_LIBRARIES}
+                ${OPENSSL_LIBRARIES}
+                pugixml::pugixml
 )
-target_link_libraries(TheDarkMod debug ${PUGIXML_LIBRARIES_D} optimized ${PUGIXML_LIBRARIES})
 
 if (WIN32)
 	target_link_libraries(TheDarkMod ws2_32 iphlpapi winmm bcrypt dinput8 dbghelp)
