--- a/sys/scons/SConscript.core	2017-03-11 06:24:20.000000000 +0800
+++ b/sys/scons/SConscript.core	2017-08-14 22:44:38.371513742 +0800
@@ -322,21 +322,13 @@
 
 # greebo: Add boost and image libraries
 local_env.Append(LIBS = [
-	# Uncomment these if you're on a x64 system - there is no libpng.a on Ubuntu x64 for instance
-	#File('/usr/lib32/libjpeg.so'),
-	#File('/usr/lib32/libpng.so'),
-	File('#/linux/polarssl/libpolarssl.a'),
-	File('#/linux/boost/lib/libboost_filesystem.a'),
-	File('#/linux/boost/lib/libboost_system.a'),
-	File('#/linux/boost/lib/libboost_thread.a'),
-	File('#/linux/boost/lib/libboost_regex.a')
+	'z', 'boost_filesystem', 'boost_system', 'boost_thread', 'boost_regex', 'curl'
 ])
 
 source_list = core_list
 source_list += idlib_objects
 source_list += [ '../../glimp/sys/scons/libglimp.a' ]
 source_list += libogg
-source_list += curl_lib
 source_list += sound_lib
 source_list += [ '../../openal/stubs.cpp' ]
 
--- a/sys/scons/SConscript.game	2017-03-11 06:24:20.000000000 +0800
+++ b/sys/scons/SConscript.game	2017-08-14 15:09:02.587411948 +0800
@@ -308,13 +308,7 @@
 local_env.Append( LINKFLAGS = [ '-lrt' ] ) #, '-lpng'
 
 local_env.Append(LIBS = [
-	# Uncomment this if you're on a x64 system - there is no libpng.a on Ubuntu x64 for instance
-	#File('/usr/lib32/libpng.so'),
-	File('#/linux/libcurl/libcurl.a'),
-	File('#/linux/polarssl/libpolarssl.a'),
-	File('#/linux/boost/lib/libboost_filesystem.a'),
-	File('#/linux/boost/lib/libboost_system.a'),
-	File('#/linux/boost/lib/libboost_thread.a')
+	'curl', 'boost_filesystem', 'libboost_system', 'boost_thread'
 ])
 
 if NO_GCH == '0':
--- a/sys/scons/SConscript.ziploader	2017-03-11 06:24:20.000000000 +0800
+++ b/sys/scons/SConscript.ziploader	2017-08-14 22:35:28.344094001 +0800
@@ -23,21 +23,6 @@
 Import( GLOBALS )
 
 ziploader_string = ' \
-	zlib/adler32.c \
-	zlib/compress.c \
-	zlib/crc32.c \
-	zlib/deflate.c \
-	zlib/gzclose.c \
-	zlib/gzlib.c \
-	zlib/gzread.c \
-	zlib/gzwrite.c \
-	zlib/infback.c \
-	zlib/inffast.c \
-	zlib/inflate.c \
-	zlib/inftrees.c \
-	zlib/trees.c \
-	zlib/uncompr.c \
-	zlib/zutil.c \
 	minizip/ioapi.c \
 	minizip/mztools.c \
 	minizip/unzip.c \
@@ -49,6 +34,7 @@
 	ziploader_list[ i ] = '../../lib/' + ziploader_list[ i ]
 
 local_env = g_env.Clone()
+local_env.Append( LINKFLAGS = [ '-lz' ] )
 
 ret_list = []
 
