--- a/sys/linux/main.cpp	2013-08-05 04:21:19.000000000 +0800
+++ b/sys/linux/main.cpp	2017-08-18 16:24:12.075087906 +0800
@@ -11,9 +11,9 @@
  
  Project: The Dark Mod (http://www.thedarkmod.com/)
  
- $Revision$ (Revision of last commit) 
- $Date$ (Date of last commit)
- $Author$ (Author of last commit)
+ $Revision: 5837 $ (Revision of last commit) 
+ $Date: 2013-08-04 20:21:19 +0000 (Sun, 04 Aug 2013) $ (Date of last commit)
+ $Author: taaaki $ (Author of last commit)
  
 ******************************************************************************/
 #include "../../idlib/precompiled.h"
@@ -25,6 +25,7 @@
 #include <errno.h>
 #include <unistd.h>
 #include <sys/stat.h>
+#include <sys/time.h>
 #include <sys/types.h>
 #include <fcntl.h>
 
@@ -285,22 +286,9 @@
 ===============
 */
 double Sys_GetClockTicks( void ) {
-#if defined( __i386__ )
-	unsigned long lo, hi;
-
-	__asm__ __volatile__ (
-						  "push %%ebx\n"			\
-						  "xor %%eax,%%eax\n"		\
-						  "cpuid\n"					\
-						  "rdtsc\n"					\
-						  "mov %%eax,%0\n"			\
-						  "mov %%edx,%1\n"			\
-						  "pop %%ebx\n"
-						  : "=r" (lo), "=r" (hi) );
-	return (double) lo + (double) 0xFFFFFFFF * hi;
-#else
-#error unsupported CPU
-#endif
+	struct timespec now;
+	clock_gettime( CLOCK_MONOTONIC, &now );
+	return now.tv_sec * 1000000000LL + now.tv_nsec;
 }
 
 /*
--- a/idlib/Lib.cpp	2017-03-10 15:24:25.000000000 -0700
+++ b/idlib/Lib.cpp	2017-09-06 10:20:48.071386879 -0600
@@ -576,9 +576,7 @@
 	idLib::sys->DebugPrintf( "\n\nASSERTION FAILED!\n%s(%d): '%s'\n", file, line, expression );
 #ifdef _WIN32
 	__asm int 0x03
-#elif defined( __linux__ )
-	__asm__ __volatile__ ("int $0x03");
-#elif defined( MACOS_X )
-	kill( getpid(), SIGINT );
+#elif defined( __linux__ ) || defined( MACOS_X )
+	abort();
 #endif
 }
