From 1156e07c6f654fc970a9ee2a7226536f426deb8d Mon Sep 17 00:00:00 2001
From: SkyBird233 <52884766+SkyBird233@users.noreply.github.com>
Date: Tue, 18 Aug 2026 14:41:12 +0800
Subject: [PATCH 2/4] Set default buildbot server for LoongArch

---
 config.def.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/config.def.h b/config.def.h
index a2f36dab48..4c43c42eab 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1859,6 +1859,8 @@
 #define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/linux/x86/latest/"
 #elif defined(__arm__) && __ARM_ARCH == 7 && defined(__ARM_PCS_VFP)
 #define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/linux/armhf/latest/"
+#elif defined(__loongarch__)
+#define DEFAULT_BUILDBOT_SERVER_URL "https://skybird233.github.io/libretro-loong/nightly/linux/loongarch64/latest/"
 #else
 #define DEFAULT_BUILDBOT_SERVER_URL ""
 #endif
-- 
2.55.0

