From 838de2292fe004d5972a303012196483aa8705fe Mon Sep 17 00:00:00 2001
From: SkyBird233 <52884766+SkyBird233@users.noreply.github.com>
Date: Tue, 18 Aug 2026 19:03:36 +0800
Subject: [PATCH 4/4] Add -DHAVE_DR_FLAC

---
 Makefile.common | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.common b/Makefile.common
index 57dabf5eac..f51adf52be 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -2020,7 +2020,7 @@ ifeq ($(HAVE_BUILTINFLAC),1)
    endif
    OBJ += $(FLACOBJ)
 else ifeq ($(HAVE_FLAC),1)
-   DEFINES += -DHAVE_FLAC
+   DEFINES += -DHAVE_FLAC -DHAVE_DR_FLAC
    LIBS    += $(FLAC_LIBS)
 endif
 
-- 
2.55.0

