diff -Naur tdebindings/korundum/rubylib/korundum/configure.in.in tdebindings.ruby23/korundum/rubylib/korundum/configure.in.in
--- tdebindings/korundum/rubylib/korundum/configure.in.in	2016-02-28 14:13:55.000000000 -0600
+++ tdebindings.ruby23/korundum/rubylib/korundum/configure.in.in	2016-11-14 19:08:32.604482286 -0600
@@ -39,6 +39,12 @@
       fi
     fi
     if test -z "$RUBY_VERSION"; then
+      RUBY_VERSION=`$PKGCONFIG ruby-2.3 --modversion 2>/dev/null`
+      if test -n "$RUBY_VERSION"; then
+        RUBY_CFLAGS=`$PKGCONFIG ruby-2.3 --cflags`
+      fi
+    fi
+    if test -z "$RUBY_VERSION"; then
       # pkgconfig file does not exist but might be 1.9.1: RUBY_VERSION defined in ruby/version.h
       if test -n "$RUBY -r rbconfig -e 'printf("%s",RbConfig::CONFIG@<:@"rubyhdrdir"@:>@)'"; then
         # rubyhdrdir parameter works only in >=1.9.1; this is 1.9.1
