diff --git a/data/pam/lightdm b/data/pam/lightdm
index fed8a9b4..4b1f9938 100644
--- a/data/pam/lightdm
+++ b/data/pam/lightdm
@@ -9,6 +9,9 @@ auth      required pam_env.so
 # Use /etc/passwd and /etc/shadow for passwords
 auth      required pam_unix.so
 
+-auth     optional pam_gnome_keyring.so
+-auth     optional pam_kwallet5.so
+
 # Check account is active, change password if required
 account   required pam_unix.so
 
@@ -18,3 +21,6 @@ password  required pam_unix.so
 # Setup session
 session   required pam_unix.so
 session   optional pam_systemd.so
+
+-session  optional pam_gnome_keyring.so auto_start
+-session  optional pam_kwallet5.so
