diff -Naur folks-0.13.2/meson.build folks-0.13.2.nomock/meson.build
--- folks-0.13.2/meson.build	2020-02-04 07:38:51.000000000 +0000
+++ folks-0.13.2.nomock/meson.build	2020-02-19 04:38:41.741278894 +0000
@@ -130,13 +130,6 @@
 
 if bluez_backend_enabled
   libebook_dep = dependency('libebook-1.2', version: '>=' + min_eds_version)
-  # Needed for the BlueZ tests
-  pymod = import('python')
-  py_installation = pymod.find_installation('python3',
-    modules: [
-      'dbusmock',
-    ],
-  )
 endif
 
 if import_tool_enabled
