From 25bca669267d39702c1a3135ff1f7cefa6854056 Mon Sep 17 00:00:00 2001
From: Willi Mann <willi@debian.org>
Date: Sun, 14 Sep 2025 16:13:09 +0200
Subject: [PATCH] DEBIAN: fix build with gcc-15

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 src/malloc.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/malloc.c b/src/malloc.c
index 1bca97a..f6ca4e8 100644
--- a/src/malloc.c
+++ b/src/malloc.c
@@ -165,8 +165,6 @@ my_strdup(char *src)
 #include <sys/types.h>
 #undef malloc
 
-void *malloc();
-
 /* Allocate an N-byte block of memory from the heap.
 If N is zero, allocate a 1-byte block.  */
 
-- 
2.52.0

