From e1cba1f97ec28c3538645d61dfcdc0c78529c200 Mon Sep 17 00:00:00 2001
From: lixianglai <lixianglai@loongson.cn>
Date: Wed, 13 Dec 2023 04:43:51 -0500
Subject: [PATCH 1002/1004] Add virtio-scsi controller if needed for LoongArch

Signed-off-by: lixianglai <lixianglai@loongson.cn>
---
 virtinst/guest.py | 27 ++++++++++++++++++++-------
 1 file changed, 20 insertions(+), 7 deletions(-)

diff --git a/virtinst/guest.py b/virtinst/guest.py
index e6636022..371b48a6 100644
--- a/virtinst/guest.py
+++ b/virtinst/guest.py
@@ -212,6 +212,7 @@ class Guest(XMLBuilder):
         self.skip_default_rng = False
         self.skip_default_tpm = False
         self.x86_cpu_default = self.cpu.SPECIAL_MODE_APP_DEFAULT
+        self.loongarch_cpu_default = self.cpu.SPECIAL_MODE_APP_DEFAULT
 
         # qemu 6.1, fairly new when we added this option, has an unfortunate
         # bug with >= 15 root ports, so we choose 14 instead of our original 16
@@ -352,7 +353,8 @@ class Guest(XMLBuilder):
         if (self.os.is_arm_machvirt() or
             self.os.is_riscv_virt() or
             self.os.is_s390x() or
-            self.os.is_pseries()):
+            self.os.is_pseries() or
+            self.os.is_loongarch64()):
             return True
 
         if not os_support:
@@ -541,7 +543,7 @@ class Guest(XMLBuilder):
             # and doesn't break QEMU internal snapshots
             prefer_efi = self.osinfo.requires_firmware_efi(self.os.arch)
         else:
-            prefer_efi = self.os.is_arm_machvirt() or self.conn.is_bhyve()
+            prefer_efi = self.os.is_arm_machvirt() or self.conn.is_bhyve() or self.os.is_loongarch64()
 
         log.debug("Prefer EFI => %s", prefer_efi)
         return prefer_efi
@@ -558,6 +560,8 @@ class Guest(XMLBuilder):
         """
         self.os.loader_ro = True
         self.os.loader_type = "pflash"
+        if (self.os.is_loongarch64()):
+            self.os.loader_type = "rom"
         self.os.loader = path
 
         # If the firmware name contains "secboot" it is probably build
@@ -902,7 +906,8 @@ class Guest(XMLBuilder):
             usb_tablet = True
         if (self.os.is_arm_machvirt() or
             self.os.is_riscv_virt() or
-            self.os.is_pseries()):
+            self.os.is_pseries() or
+            self.os.is_loongarch64()):
             usb_tablet = True
             usb_keyboard = True
 
@@ -984,7 +989,10 @@ class Guest(XMLBuilder):
             # For pseries, we always assume OS supports usb3
             if qemu_usb3:
                 usb3 = True
-
+        elif self.os.is_loongarch64():
+            # For loongarch64, we always assume OS supports usb3
+            if qemu_usb3:
+                usb3 = True
 
         if usb2:
             for dev in DeviceController.get_usb2_controllers(self.conn):
@@ -1016,7 +1024,8 @@ class Guest(XMLBuilder):
         if self.os.is_container() and not self.conn.is_vz():
             return
         if (not self.os.is_x86() and
-            not self.os.is_pseries()):
+            not self.os.is_pseries() and
+            not self.os.is_loongarch64()):
             return
         self.add_device(DeviceGraphics(self.conn))
 
@@ -1029,7 +1038,8 @@ class Guest(XMLBuilder):
                 self.os.is_arm_machvirt() or
                 self.os.is_riscv_virt() or
                 self.os.is_s390x() or
-                self.os.is_pseries()):
+                self.os.is_pseries() or
+                self.os.is_loongarch64()):
             return
 
         if (self.conn.is_qemu() and
@@ -1074,7 +1084,8 @@ class Guest(XMLBuilder):
                 self.os.is_arm_machvirt() or
                 self.os.is_riscv_virt() or
                 self.os.is_s390x() or
-                self.os.is_pseries()):
+                self.os.is_pseries() or
+                self.os.is_loongarch64()):
             return
 
         if self.osinfo.supports_virtioballoon(self._extra_drivers):
@@ -1101,6 +1112,8 @@ class Guest(XMLBuilder):
             return True
         if self.os.is_riscv_virt():
             return True
+        if self.os.is_loongarch64():
+            return True
         return False
 
     def _add_q35_pcie_controllers(self):
-- 
2.44.0

