From c18db0723a0a320863a321c91a2f32f11aaff571 Mon Sep 17 00:00:00 2001
From: Rick Liu <rickliu2000@foxmail.com>
Date: Wed, 3 Jun 2026 19:15:45 +0800
Subject: [PATCH] AOSCOS: fix build with boost 1.89

---
 resources/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/resources/CMakeLists.txt b/resources/CMakeLists.txt
index 299f404b0..415ea23c8 100644
--- a/resources/CMakeLists.txt
+++ b/resources/CMakeLists.txt
@@ -44,7 +44,7 @@ add_subdirectory( imap )
 if (Libkolabxml_FOUND)
     # The boost-provided CMake files are unreliable
     set(Boost_NO_BOOST_CMAKE ON)
-    find_package(Boost MODULE REQUIRED COMPONENTS atomic thread system)
+    find_package(Boost MODULE REQUIRED COMPONENTS atomic thread)
     unset(Boost_NO_BOOST_CMAKE)
     add_subdirectory( kolab )
 endif()
-- 
2.52.0

