From 05931305474901414b9ae29c37bc059604884d34 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Sat, 24 Aug 2024 15:50:29 -0700
Subject: [PATCH] utils/idna: fix utf7proc.h location

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 utils/idna.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utils/idna.c b/utils/idna.c
index 3ce3e1be1..d4e1bdff9 100644
--- a/utils/idna.c
+++ b/utils/idna.c
@@ -167,7 +167,7 @@ idna__ace_to_ucs4(const char *ace_label,
 
 #ifdef WITH_UTF8PROC
 
-#include <libutf8proc/utf8proc.h>
+#include <utf8proc.h>
 
 int32_t idna_contexto[] = {
 	/* CONTEXTO codepoints which have a rule defined */
-- 
2.46.0

