diff -Naur mlt-6.4.1/src/framework/mlt_property.h mlt-6.4.1.glibc226/src/framework/mlt_property.h
--- mlt-6.4.1/src/framework/mlt_property.h	2016-11-16 06:53:11.000000000 +0000
+++ mlt-6.4.1.glibc226/src/framework/mlt_property.h	2017-10-21 05:12:57.443538814 +0000
@@ -31,7 +31,7 @@
 #endif
 
 #if defined(__GLIBC__) || defined(__APPLE__) || (__FreeBSD_version >= 900506)
-#include <xlocale.h>
+#include <locale.h>
 #else
 typedef char* locale_t;
 #endif
