From 9ac31aaef99f6256a071cc14832567e5aff0b4fa Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Wed, 19 Mar 2025 23:23:41 -0700
Subject: [PATCH] fix: allocate null terminator for empty config_path

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 src/core/config.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/core/config.c b/src/core/config.c
index 663a8f3..fd2cd4d 100644
--- a/src/core/config.c
+++ b/src/core/config.c
@@ -241,7 +241,8 @@ static void init_wrapper_path(struct environment *env,
             *last_slash = '\0';
         }
     } else {
-        config_path = "";
+        config_path = malloc(1);
+        *config_path = '\0';
     }
 
     const char *const wrapper_paths =
-- 
2.49.0

