From 219972485cd51bc5d8f1681d553d58e8da577fb1 Mon Sep 17 00:00:00 2001
From: Mag Mell <sakiiily@aosc.io>
Date: Mon, 13 Apr 2026 16:38:34 +0800
Subject: [PATCH] fix build on librime 1.16

---
 src/fcitx-rime.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/fcitx-rime.c b/src/fcitx-rime.c
index 9f727ad..9a021cb 100644
--- a/src/fcitx-rime.c
+++ b/src/fcitx-rime.c
@@ -5,7 +5,7 @@
 #include <fcitx-config/xdg.h>
 #include <fcitx/module/freedesktop-notify/fcitx-freedesktop-notify.h>
 #include <libintl.h>
-#include <rime_api.h>
+#include <rime_api_deprecated.h>
 
 #define _(x) dgettext("fcitx-rime", (x))
 
@@ -603,7 +603,7 @@ void FcitxRimeToggleDeploy(void* arg)
 {
     FcitxRime* rime = (FcitxRime*) arg;
     if (rime->session_id) {
-        rime->api->sync_user_data(rime->session_id);
+        rime->api->sync_user_data();
         rime->session_id = 0;
     }
     rime->api->finalize();
-- 
2.52.0

