--- a/Makefile	2021-01-07 22:27:01.787460632 -0800
+++ b/Makefile	2021-01-07 22:26:10.466993436 -0800
@@ -1,8 +1,8 @@
 rtk_hciattach:hciattach.c hciattach_rtk.o  
-	cc -o rtk_hciattach hciattach.c hciattach_rtk.o  
+	cc $(CFLAGS) $(LDFLAGS) -o rtk_hciattach hciattach.c hciattach_rtk.o  
 
 hciattach_rtk.o:hciattach_rtk.c
-	cc -c hciattach_rtk.c
+	cc $(CFLAGS) -c hciattach_rtk.c
 
 clean:
 	rm -f *.o  rtk_hciattach
