--- a/ONScripter_event.cpp	2016-08-04 22:37:00.000000000 +0800
+++ b/ONScripter_event.cpp	2016-11-21 04:01:01.676712364 +0800
@@ -1377,19 +1377,14 @@
 #endif
 #if !defined(ANDROID) && !defined(IOS) && !defined(WINRT) && SDL_VERSION_ATLEAST(2,0,0)
           case SDL_MOUSEMOTION:
-            if (mouseMoveEvent( &event.motion )) return;
-            if (btndown_flag){
-                if (event.motion.state & SDL_BUTTON(SDL_BUTTON_LEFT))
+            if (mouseMoveEvent( &event.motion ));
+	    btndown_flag = true;
                     tmp_event.button.button = SDL_BUTTON_LEFT;
-                else if (event.motion.state & SDL_BUTTON(SDL_BUTTON_RIGHT))
-                    tmp_event.button.button = SDL_BUTTON_RIGHT;
-                else
-                    break;
 
                 tmp_event.button.type = SDL_MOUSEBUTTONDOWN;
                 ret = mousePressEvent( &tmp_event.button );
+		btndown_flag = false;
                 if (ret) return;
-            }
             break;
             
           case SDL_MOUSEBUTTONDOWN:
