Subject: YUV_420_USE_YV12 build option affects encoding of NTSC frames
From: Debian
Author: Ben Hutchings <ben@decadent.org.uk>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=466142
Forwarded: no
---
 libdv/encode.c |   15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

--- libdv.orig/libdv/encode.c
+++ libdv/libdv/encode.c
@@ -1712,15 +1712,15 @@ yuy2_to_ycb( uint8_t *data, int isPAL, s
 /* ---------------------------------------------------------------------------
  */
 static void
-yv12_to_ycb( uint8_t **in, int isPAL, short *img_y, short *img_cr, short *img_cb)
+yv12_to_ycb( uint8_t **in, short *img_y, short *img_cr, short *img_cb)
 {
-	register int total = DV_WIDTH * (isPAL ? DV_PAL_HEIGHT : DV_NTSC_HEIGHT);
+	register int total = DV_WIDTH * DV_PAL_HEIGHT;
 	register int i, j, k;
 	
 	
 	for (i = 0; i < total; i++) img_y[i]   = (((short) in[0][i]) - 128) << DCT_YUV_PRECISION;
 	
-	for (i = 0; i < (isPAL ? DV_PAL_HEIGHT : DV_NTSC_HEIGHT)/2; ++i) {
+	for (i = 0; i < (DV_PAL_HEIGHT / 2); ++i) {
 	  for (j = 0; j < DV_WIDTH/2 ; j++) {
 	    
 	    k = i * DV_WIDTH/2 + j;
@@ -1796,11 +1796,12 @@ int dv_encode_full_frame(dv_encoder_t *d
 			dv_enc->img_y, dv_enc->img_cr, dv_enc->img_cb);
 		break;
 	case e_dv_color_yuv:
-#ifndef YUV_420_USE_YV12
-		yuy2_to_ycb( in[0], dv_enc->isPAL, dv_enc->img_y, dv_enc->img_cr, dv_enc->img_cb);
-#else
-		yv12_to_ycb( in, dv_enc->isPAL, dv_enc->img_y, dv_enc->img_cr, dv_enc->img_cb);
+#ifdef YUV_420_USE_YV12
+		if (dv_enc->isPAL)
+			yv12_to_ycb( in, dv_enc->img_y, dv_enc->img_cr, dv_enc->img_cb);
+		else
 #endif
+			yuy2_to_ycb( in[0], dv_enc->isPAL, dv_enc->img_y, dv_enc->img_cr, dv_enc->img_cb);
 		break;
 	default:
 		fprintf(stderr, "Invalid value for color_space "
