From 66be1d90b0b09cf459664e5986009916d9ec20a3 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Thu, 14 May 2020 15:40:26 -0600
Subject: [PATCH] makefile: remove x86 specific flags

---
 source/Makefile.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/Makefile.mk b/source/Makefile.mk
index e27b2b9df..5db5c4022 100644
--- a/source/Makefile.mk
+++ b/source/Makefile.mk
@@ -102,7 +102,7 @@ endif
 # Set build and link flags
 
 BASE_FLAGS = -Wall -Wextra -pipe -DBUILDING_CARLA -DREAL_BUILD -MD -MP
-BASE_OPTS  = -O3 -ffast-math -mtune=generic -msse -msse2 -mfpmath=sse -fdata-sections -ffunction-sections
+BASE_OPTS  = -O3 -ffast-math -fdata-sections -ffunction-sections
 
 ifeq ($(MACOS),true)
 # MacOS linker flags
-- 
2.26.2

