From 1d9bada3dbf625c2db5f70611ad08dbe2074863c Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 7 Aug 2024 15:51:51 +0800
Subject: [PATCH] fix(pyproject.toml): drop release-management-only dep

---
 pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index ff499d3d..25a3c4f9 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -6,7 +6,6 @@ requires = [
     # Plugins
     "setuptools_scm[toml] >= 7; python_version >= '3.7'",
     "setuptools_scm[toml] < 7; python_version < '3.7'",
-    "setuptools_scm_git_archive >= 1.1; python_version < '3.7'",
 ]
 build-backend = "setuptools.build_meta"
 
-- 
2.46.0

