diff --git a/OpenRGB.pro b/OpenRGB.pro
index ef5a4f2..d6c8f8c 100644
--- a/OpenRGB.pro
+++ b/OpenRGB.pro
@@ -14,7 +14,7 @@ QT +=
 #-----------------------------------------------------------------------------------------------#
 # Set compiler to use C++17 to make std::filesystem available                                   #
 #-----------------------------------------------------------------------------------------------#
-CONFIG += c++17
+CONFIG += c++17 nostrip

 greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
