From ca02e9605d432039f7eefa3670d529a9a4b2c8a8 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sun, 12 Oct 2025 15:54:56 +0800
Subject: [PATCH 1/4] BLFS: Fixing includes

---
 growisofs.c   | 1 +
 transport.hxx | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/growisofs.c b/growisofs.c
index 6c47192..2602f1d 100644
--- a/growisofs.c
+++ b/growisofs.c
@@ -441,6 +441,7 @@
 #include <fcntl.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <assert.h>
 #include "mp.h"
 
diff --git a/transport.hxx b/transport.hxx
index 35a57a7..8ee1536 100644
--- a/transport.hxx
+++ b/transport.hxx
@@ -9,6 +9,7 @@
 #if defined(__unix) || defined(__unix__)
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 #include <unistd.h>
 #include <string.h>
 #include <sys/types.h>
@@ -40,6 +41,7 @@ inline long getmsecs()
 #include <stddef.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <limits.h>
 #define ssize_t		LONG_PTR
 #define off64_t		__int64
 
-- 
2.51.0

