From 371ae5f1cad0c0c3343295b2aaf926f2b353501d Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sun, 12 Oct 2025 15:59:01 +0800
Subject: [PATCH 2/4] fix dvddl

---
 growisofs_mmc.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/growisofs_mmc.cpp b/growisofs_mmc.cpp
index 4de8a14..cc25f0a 100644
--- a/growisofs_mmc.cpp
+++ b/growisofs_mmc.cpp
@@ -1640,9 +1640,7 @@ static void plus_r_dl_split (Scsi_Command &cmd,off64_t size)
     blocks += 15, blocks &= ~15;
 
     if (blocks <= split)
-	fprintf (stderr,":-( more than 50%% of space will be *wasted*!\n"
-			"    use single layer media for this recording\n"),
-	exit (FATAL_START(EMEDIUMTYPE));
+	fprintf (stderr,":-? more than 50%% of space will be *wasted*!\n");
 
     blocks /= 16;
     blocks += 1;
-- 
2.51.0

