From 6479e78c1327e91f703305c8c982b6b8bba470e9 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sun, 12 Oct 2025 15:59:37 +0800
Subject: [PATCH 3/4] fix wctomb

---
 transport.hxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/transport.hxx b/transport.hxx
index 8ee1536..679ce29 100644
--- a/transport.hxx
+++ b/transport.hxx
@@ -125,7 +125,7 @@ class autofree {
 extern "C" char *plusminus_locale()
 { static class __plusminus {
     private:
-	char str[4];
+	char str[MB_LEN_MAX];
     public:
 	__plusminus()	{   setlocale(LC_CTYPE,ENV_LOCALE);
 			    int l = wctomb(str,(wchar_t)(unsigned char)'±');
-- 
2.51.0

