diff --git a/core/kernel_interface.c b/core/kernel_interface.c
index 819a1ae..1dd005f 100644
--- a/core/kernel_interface.c
+++ b/core/kernel_interface.c
@@ -212,7 +212,7 @@ static  void  krnl_update_adapter_info(void* adp, adapter_info_t*  adapter_info,
     adapter_t *adapter = (adapter_t *)adp;
     if(adapter && adapter_info)
     {
-        adapter->hw_caps.snoop_only = (adapter_info->snoop_only)? 1 : 0;
+        adapter->hw_caps.snoop_only = adapter->hw_caps.snoop_only || ((adapter_info->snoop_only)? 1 : 0 ) ;
         adapter->low_top_address = adapter_info->low_top_addr;
         adapter->hw_caps.miu_channel_num = adapter_info->chan_num;
         adapter->hw_caps.chip_slice_mask = adapter_info->chip_slice_mask;
