diff -Naur compiz-0.9.13.1/metadata/core.xml.in compiz-0.9.13.1.steal/metadata/core.xml.in
--- compiz-0.9.13.1/metadata/core.xml.in	2016-11-11 16:20:40.000000000 +0000
+++ compiz-0.9.13.1.steal/metadata/core.xml.in	2018-09-30 07:37:25.892252305 +0000
@@ -119,7 +119,7 @@
 			<_long>Level of focus stealing prevention</_long>
 			<min>0</min>
 			<max>4</max>
-			<default>1</default>
+			<default>0</default>
 			<desc>
 				<value>0</value>
 				<_name>Off</_name>
