Index: compiz/plugins/opengl/include/opengl/opengl.h
===================================================================
--- compiz.orig/plugins/opengl/include/opengl/opengl.h	2014-03-03 15:29:29.898235681 +0100
+++ compiz/plugins/opengl/include/opengl/opengl.h	2014-03-03 15:29:29.894235669 +0100
@@ -590,6 +590,17 @@
 
 extern GLScreenPaintAttrib defaultScreenPaintAttrib;
 
+namespace compiz
+{
+namespace opengl
+{
+namespace internal
+{
+class DriverWorkaroundQuery;
+}
+}
+}
+
 class GLScreen;
 class GLFramebufferObject;
 class GLScreenInterface;
@@ -787,6 +798,13 @@
 
 	bool glInitContext (XVisualInfo *);
 
+	/**
+	 * This is only meant to be used internally, here to avoid unnecessarily
+	 * breaking the API and ABI. Fetch any driver workarounds in place
+	 */
+	const compiz::opengl::internal::DriverWorkaroundQuery &
+	fetchDriverWorkarounds ();
+
 	WRAPABLE_HND (0, GLScreenInterface, bool, glPaintOutput,
 		      const GLScreenPaintAttrib &, const GLMatrix &,
 		      const CompRegion &, CompOutput *, unsigned int);
Index: compiz/plugins/opengl/src/privates.h
===================================================================
--- compiz.orig/plugins/opengl/src/privates.h	2014-03-03 15:29:29.898235681 +0100
+++ compiz/plugins/opengl/src/privates.h	2014-03-03 15:29:29.894235669 +0100
@@ -47,6 +47,24 @@
 
 extern CompOutput *targetOutput;
 
+namespace compiz
+{
+namespace opengl
+{
+namespace internal
+{
+class DriverWorkaroundQuery
+{
+    public:
+
+	virtual ~DriverWorkaroundQuery () {};
+
+	virtual bool unsafeForExternalBinds () const = 0;
+};
+}
+}
+}
+
 class GLDoubleBuffer :
     public compiz::opengl::DoubleBuffer
 {
@@ -141,6 +159,7 @@
     public ScreenInterface,
     public CompositeScreenInterface,
     public compiz::composite::PaintHandler,
+    public compiz::opengl::internal::DriverWorkaroundQuery,
     public OpenglOptions
 {
     public:
@@ -232,6 +251,7 @@
 	std::vector<GLTexture::BindPixmapProc> bindPixmap;
 	bool hasCompositing;
 	bool commonFrontbuffer;
+	bool sharedMemoryTFP;
 	bool incorrectRefreshRate; // hack for NVIDIA specifying an incorrect
 				   // refresh rate, causing us to miss vblanks
 
@@ -266,6 +266,10 @@
 	std::vector<XToGLSync*>::size_type currentSyncNum;
 	XToGLSync *currentSync;
 	std::vector<XToGLSync*>::size_type warmupSyncs;
+
+    private:
+
+	bool unsafeForExternalBinds () const;
 };
 
 class PrivateGLWindow :
Index: compiz/plugins/opengl/src/screen.cpp
===================================================================
--- compiz.orig/plugins/opengl/src/screen.cpp	2014-03-03 15:29:29.898235681 +0100
+++ compiz/plugins/opengl/src/screen.cpp	2014-03-03 15:29:29.894235669 +0100
@@ -67,6 +67,7 @@
 
 
 using namespace compiz::opengl;
+namespace cglint = compiz::opengl::internal;
 
 namespace GL {
     #ifdef USE_GLES
@@ -537,6 +538,18 @@
 };
 
 bool
+PrivateGLScreen::unsafeForExternalBinds () const
+{
+    return sharedMemoryTFP;
+}
+
+const cglint::DriverWorkaroundQuery &
+GLScreen::fetchDriverWorkarounds ()
+{
+    return *priv;
+}
+
+bool
 GLScreen::glInitContext (XVisualInfo *visinfo)
 {
 #ifndef USE_GLES
@@ -855,6 +868,7 @@
 
     priv->commonFrontbuffer = true;
     priv->incorrectRefreshRate = false;
+    priv->sharedMemoryTFP = false;
     if (glRenderer != NULL && strstr (glRenderer, "on llvmpipe"))
     {
 	/*
@@ -875,6 +889,18 @@
 	priv->incorrectRefreshRate = true;
     }
 
+    if (glVendor != NULL && strstr (glVendor, "Humper"))
+    {
+	/*
+	 * VirtualBox uses XShm/XCopyArea in order to implement
+	 * texture_from_pixmap. Because of this, they require
+	 * a second X connection to track damage events and
+	 * changes to the pixmap, and this is incompatible with
+	 * taking a server grab on when querying and binding
+	 * externally managed pixmaps */
+	priv->sharedMemoryTFP = true;
+    }
+
     if (strstr (glExtensions, "GL_ARB_texture_non_power_of_two"))
 	GL::textureNonPowerOfTwo = true;
     GL::textureNonPowerOfTwoMipmap = GL::textureNonPowerOfTwo;
Index: compiz/plugins/opengl/src/texture.cpp
===================================================================
--- compiz.orig/plugins/opengl/src/texture.cpp	2014-03-03 15:29:29.898235681 +0100
+++ compiz/plugins/opengl/src/texture.cpp	2014-03-03 15:31:45.974666737 +0100
@@ -41,6 +41,7 @@
 #include "glx-tfp-bind.h"
 
 namespace cgl = compiz::opengl;
+namespace cglint = compiz::opengl::internal;
 
 #ifdef USE_GLES
 std::map<Damage, EglTexture*> boundPixmapTex;
@@ -646,6 +647,12 @@
 				 int               depth,
 				 cgl::PixmapSource source)
 {
+    GLScreen *gs = GLScreen::get (screen);
+    const cglint::DriverWorkaroundQuery &query (gs->fetchDriverWorkarounds ());
+
+    if (query.unsafeForExternalBinds () && source == cgl::ExternallyManaged)
+	return GLTexture::List ();
+
     GLTexture::List   rv (1);
     TfpTexture        *tex = NULL;
     unsigned int      target = 0;
