Index: compiz.fix_dist_upgrade/src/main.cpp
===================================================================
--- compiz.fix_dist_upgrade.orig/src/main.cpp	2013-02-22 01:06:33.342881314 +0800
+++ compiz.fix_dist_upgrade/src/main.cpp	2013-02-22 01:07:03.283029785 +0800
@@ -38,6 +38,11 @@
 #include "privatescreen.h"
 #include "privatestackdebugger.h"
 
+namespace
+{
+bool autoAddCCP = true;
+}
+
 void
 CompManager::usage ()
 {
@@ -52,6 +57,7 @@
             "  --debug               Enable debug mode\n"
             "  --version             Show the program version\n"
             "  --help                Show this summary\n"
+	    "  --no-auto-add-ccp     Do not automatically load the ccp plugin\n"
             , programName);
 }
 
@@ -116,6 +122,11 @@
 	    if (i + 1 < argc)
 		clientId = argv[++i];
 	}
+	else if (!strcmp (argv[i], "--no-auto-add-ccp"))
+	{
+	    compLogMessage ("core", CompLogLevelInfo, "running without the ccp plugin, no settings will be loaded");
+	    autoAddCCP = false;
+	}
 	else if (*argv[i] == '-')
 	{
 	    compLogMessage ("core", CompLogLevelWarn,
@@ -151,6 +162,9 @@
 
     modHandler = new ModifierHandler ();
 
+    if (initialPlugins.empty () && autoAddCCP)
+	initialPlugins.push_back ("ccp");
+
     if (!screen->init (displayName))
 	return false;
 
Index: compiz.fix_dist_upgrade/tests/xorg-gtest/src/compiz-xorg-gtest.cpp
===================================================================
--- compiz.fix_dist_upgrade.orig/tests/xorg-gtest/src/compiz-xorg-gtest.cpp	2013-02-22 01:06:33.342881314 +0800
+++ compiz.fix_dist_upgrade/tests/xorg-gtest/src/compiz-xorg-gtest.cpp	2013-02-22 01:06:33.342881314 +0800
@@ -462,6 +462,7 @@
     if (flags & ct::CompizProcess::ReplaceCurrentWM)
 	args.push_back ("--replace");
 
+    args.push_back ("--no-auto-add-ccp");
     args.push_back ("--send-startup-message");
 
     priv->mProcess.Start (compizBinaryPath, args);
