--- a/src/screen.cpp
+++ b/src/screen.cpp
@@ -3253,12 +3253,22 @@
 	     * This is so that we can detect taps on individual modifier
 	     * keys, and know to cancel the tap if <modifier>+k is pressed.
 	     */
-	    if (!(currentState & CompAction::StateIgnoreTap))
+            int minCode, maxCode;
+            XDisplayKeycodes (screen->dpy(), &minCode, &maxCode);
+
+            if ((currentState & CompAction::StateIgnoreTap))
+            {
+                KeySym sym_p = XStringToKeysym("p");
+                KeyCode code_p = XKeysymToKeycode(screen->dpy(), sym_p);
+
+                for (k = minCode; k <= maxCode; k++)
+                    if (k != code_p)
+                        grabUngrabOneKey (modifiers | ignore, k, grab);
+            }
+            else
             {
- 		int minCode, maxCode;
- 		XDisplayKeycodes (screen->dpy(), &minCode, &maxCode);
- 		for (k = minCode; k <= maxCode; k++)
- 		    grabUngrabOneKey (modifiers | modifierForKeycode | ignore, k, grab);
+                for (k = minCode; k <= maxCode; k++)
+                    grabUngrabOneKey (modifiers | modifierForKeycode | ignore, k, grab);
             }
 	}
 
