From 92a7669cf8b37c6b4def9977367f3f160cae9b1b Mon Sep 17 00:00:00 2001
From: Dugan Chen <thedoogster@gmail.com>
Date: Fri, 18 Nov 2016 23:47:36 -0800
Subject: [PATCH] Fix "too many args for cd command" with jc and FISH.

---
 bin/autojump.fish | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/autojump.fish b/bin/autojump.fish
index b9af80c..9134a53 100644
--- a/bin/autojump.fish
+++ b/bin/autojump.fish
@@ -74,7 +74,7 @@ function jc
         case '-*'
             j $argv
         case '*'
-            j (pwd) $argv
+            j (pwd)/$argv
     end
 end
 
-- 
2.27.0

