From 533cc7449b6b55f9a88bf6d53f3372ea83eca3d7 Mon Sep 17 00:00:00 2001
From: Dugan Chen <thedoogster@gmail.com>
Date: Thu, 3 Apr 2025 19:25:12 +0800
Subject: [PATCH 2/2] Fix "too many args for cd command" with jc and FISH.

---
 integrations/autojump.fish | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/integrations/autojump.fish b/integrations/autojump.fish
index e99550674470..a407a5ae2b9d 100644
--- a/integrations/autojump.fish
+++ b/integrations/autojump.fish
@@ -74,7 +74,7 @@ function jc
         case '-*'
             j $argv
         case '*'
-            j (pwd) $argv
+            j (pwd)/$argv
     end
 end
 
-- 
2.49.0

