--- a/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h
+++ b/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h
@@ -167,6 +167,15 @@
 #define WTF_CPU_BIG_ENDIAN 1
 #endif
 
+/* CPU(RISCV64) - RISC-V 64-bit */
+#if defined(__riscv)
+#if (__riscv_xlen == 64)
+#define WTF_CPU_RISCV64 1
+#elif (__riscv_xlen == 32)
+#define WTF_CPU_RISCV32 1
+#endif
+#endif
+
 /* CPU(SH4) - SuperH SH-4 */
 #if defined(__SH4__)
 #define WTF_CPU_SH4 1
@@ -919,7 +928,7 @@
 #endif
 
 #if !defined(WTF_USE_JSVALUE64) && !defined(WTF_USE_JSVALUE32) && !defined(WTF_USE_JSVALUE32_64)
-#if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS) || OS(SOLARIS) || OS(HPUX))) || (CPU(IA64) && !CPU(IA64_32)) || CPU(ALPHA) || CPU(AIX64) || CPU(SPARC64) || CPU(MIPS64) || CPU(AARCH64) || CPU(S390X)
+#if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS) || OS(SOLARIS) || OS(HPUX))) || (CPU(IA64) && !CPU(IA64_32)) || CPU(ALPHA) || CPU(AIX64) || CPU(SPARC64) || CPU(MIPS64) || CPU(AARCH64) || CPU(S390X) || CPU(RISCV64)
 #define WTF_USE_JSVALUE64 1
 #elif CPU(ARM) || CPU(PPC64)
 #define WTF_USE_JSVALUE32 1
--- a/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h
+++ b/src/3rdparty/webkit/Source/JavaScriptCore/wtf/Platform.h
@@ -187,6 +187,15 @@
 #define WTF_CPU_BIG_ENDIAN 1
 #endif
 
+/* CPU(RISCV64) - RISC-V 64-bit */
+#if defined(__riscv)
+#if (__riscv_xlen == 64)
+#define WTF_CPU_RISCV64 1
+#elif (__riscv_xlen == 32)
+#define WTF_CPU_RISCV32 1
+#endif
+#endif
+
 /* CPU(SH4) - SuperH SH-4 */
 #if defined(__SH4__)
 #define WTF_CPU_SH4 1
@@ -1000,6 +1009,7 @@
 #if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS))) \
     || (CPU(IA64) && !CPU(IA64_32)) \
     || CPU(ALPHA) \
+    || CPU(RISCV64) \
     || CPU(SPARC64) \
     || CPU(S390X) \
     || CPU(PPC64) \
--- a/src/corelib/arch/qatomic_arch.h
+++ b/src/corelib/arch/qatomic_arch.h
@@ -76,6 +76,8 @@
 #  include "QtCore/qatomic_parisc.h"
 #elif defined(QT_ARCH_POWERPC)
 #  include "QtCore/qatomic_powerpc.h"
+#elif defined(QT_ARCH_RISCV64)
+#  include "QtCore/qatomic_generic.h"
 #elif defined(QT_ARCH_S390)
 #  include "QtCore/qatomic_s390.h"
 #elif defined(QT_ARCH_SPARC)
--- /dev/null
+++ b/src/corelib/arch/riscv64/arch.pri
@@ -0,0 +1,8 @@
+#
+# RISC-V 64 architecture
+#
+
+unix:SOURCES += ../generic/qatomic_generic_unix.cpp
+win32:SOURCES += ../generic/qatomic_generic_windows.cpp
+
+QMAKE_CXXFLAGS += -fpermissive
--- a/src/dbus/qdbusintegrator.cpp
+++ b/src/dbus/qdbusintegrator.cpp
@@ -73,7 +73,7 @@
 static bool isDebugging;
 #define qDBusDebug              if (!::isDebugging); else qDebug
 
-Q_GLOBAL_STATIC_WITH_ARGS(const QString, orgFreedesktopDBusString, (QLatin1String(DBUS_SERVICE_DBUS)))
+Q_GLOBAL_STATIC_WITH_ARGS(QString, orgFreedesktopDBusString, (QLatin1String(DBUS_SERVICE_DBUS)))
 
 static inline QString dbusServiceString()
 { return *orgFreedesktopDBusString(); }
