From 38849c22aa34c5fad10091e066a520dd831462b3 Mon Sep 17 00:00:00 2001
From: Adam Williamson <awilliam@redhat.com>
Date: Fri, 24 Feb 2023 11:01:19 -0800
Subject: [PATCH] Include stdexcept when using logic_error

Signed-off-by: Adam Williamson <awilliam@redhat.com>
---
 src/cpp-utils/thread/LeftRight.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/cpp-utils/thread/LeftRight.h b/src/cpp-utils/thread/LeftRight.h
index 65ddba75..7b10458a 100644
--- a/src/cpp-utils/thread/LeftRight.h
+++ b/src/cpp-utils/thread/LeftRight.h
@@ -4,6 +4,7 @@
 #include <thread>
 #include <cpp-utils/macros.h>
 #include <array>
+#include <stdexcept>
 
 namespace cpputils {
 
-- 
2.40.0

