*** NetworkManager-1.8.0.orig/data/NetworkManager.service.in	2017-05-10 16:59:07.000000000 +0800
--- NetworkManager-1.8.0/data/NetworkManager.service.in	2017-06-03 00:06:56.554064887 +0800
***************
*** 1,6 ****
--- 1,7 ----
  [Unit]
  Description=Network Manager
  Documentation=man:NetworkManager(8)
+ ConditionCapability=CAP_NET_ADMIN
  Wants=network.target
  After=network-pre.target dbus.service
  Before=network.target @DISTRO_NETWORK_SERVICE@
