From 5d4796bdbc5c12eee09905fef0d085fe971b03d9 Mon Sep 17 00:00:00 2001
From: Laurent Montel <montel@kde.org>
Date: Wed, 3 Jan 2024 07:57:52 +0100
Subject: [PATCH 1/6] Fix compile with kconfiggroup + QStringLiteral

---
 floppy.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/floppy.cpp b/floppy.cpp
index ed33a6c..92cfe0d 100644
--- a/floppy.cpp
+++ b/floppy.cpp
@@ -618,7 +618,7 @@ void FloppyData::formatStatus(const QString &s, int p)
 
 void FloppyData::writeSettings()
 {
-    KConfigGroup config = KSharedConfig::openConfig()->group("GeneralData");
+    KConfigGroup config = KSharedConfig::openConfig()->group(QStringLiteral("GeneralData"));
 
     densityconfig = densityComboBox->currentText().trimmed();
     filesystemconfig = filesystemComboBox->currentText().trimmed();
@@ -645,7 +645,7 @@ void FloppyData::writeSettings()
 
 void FloppyData::readSettings()
 {
-    KConfigGroup config = KSharedConfig::openConfig()->group("GeneralData");
+    KConfigGroup config = KSharedConfig::openConfig()->group(QStringLiteral("GeneralData"));
 
     verifyconfig = config.readEntry("Verify", 1);
     labelconfig = config.readEntry("CreateLabel", 1);
-- 
2.51.0

