From 952280d556e96c6e12a6f538468d568dae2c2836 Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Fri, 8 Mar 2024 11:29:39 +0100
Subject: [PATCH 4/6] Port away from QRegExp

---
 CMakeLists.txt |  7 -------
 format.cpp     | 12 ++++++------
 2 files changed, 6 insertions(+), 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2f2b3bb..9d1b102 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,9 +39,6 @@ else()
 endif()
 
 find_package(Qt${QT_MAJOR_VERSION} ${REQUIRED_QT_VERSION} CONFIG REQUIRED Core DBus Widgets)
-if (QT_MAJOR_VERSION STREQUAL "6")
-    find_package(Qt6Core5Compat)
-endif()
 find_package(KF${KF_MAJOR_VERSION} ${KF_MIN_VERSION} REQUIRED COMPONENTS I18n CoreAddons Completion XmlGui)
 find_package(KF${KF_MAJOR_VERSION}DocTools ${KF_MIN_VERSION})
 set_package_properties(KF${KF_MAJOR_VERSION}DocTools PROPERTIES DESCRIPTION
@@ -66,10 +63,6 @@ ecm_qt_declare_logging_category(kfloppy HEADER kfloppy_debug.h IDENTIFIER KFLOPP
 target_link_libraries(kfloppy KF${KF_MAJOR_VERSION}::CoreAddons KF${KF_MAJOR_VERSION}::I18n KF${KF_MAJOR_VERSION}::Completion
     KF${KF_MAJOR_VERSION}::XmlGui Qt${QT_MAJOR_VERSION}::DBus KF${KF_MAJOR_VERSION}::Crash)
 
-if (TARGET Qt::Core5Compat)
-    target_link_libraries(kfloppy Qt::Core5Compat)
-endif()
-
 install(TARGETS kfloppy  ${KDE_INSTALL_TARGETS_DEFAULT_ARGS} )
 
 
diff --git a/format.cpp b/format.cpp
index 056c563..c093824 100644
--- a/format.cpp
+++ b/format.cpp
@@ -27,7 +27,7 @@
 #include <unistd.h>
 
 #include "qplatformdefs.h"
-#include <QRegExp>
+#include <QRegularExpression>
 #include <QStandardPaths>
 #include <QTimer>
 
@@ -446,10 +446,10 @@ void FDFormat::processStdOut(const QString &s)
     }
 #elif defined(ANY_LINUX)
     DEBUGS(s);
-    QRegExp regexp(QStringLiteral("([0-9]+)"));
+    const QRegularExpression regexp(QStringLiteral("([0-9]+)"));
     if (s.startsWith(QLatin1String("bad data at cyl")) || s.contains(QLatin1String("Problem reading cylinder"))) {
-        if (regexp.indexIn(s) > -1) {
-            const int track = regexp.cap(1).toInt();
+        if (const auto match = regexp.match(s); match.hasMatch()) {
+            const int track = match.capturedView(1).toInt();
             Q_EMIT status(i18n("Low-level formatting error at track %1.", track), -1);
         } else {
             // This error should not happen
@@ -473,9 +473,9 @@ void FDFormat::processStdOut(const QString &s)
         return;
     }
     // Check for numbers at last (as /dev/fd0u1440 has numbers too)
-    else if (regexp.indexIn(s) > -1) {
+    else if (const auto match = regexp.match(s); match.hasMatch()) {
         // Normal track number (formatting or verifying)
-        const int p = regexp.cap(1).toInt();
+        const int p = match.capturedView(1).toInt();
         if ((p >= 0) && (p < deviceInfo->tracks)) {
             Q_EMIT status(QString(), p * 100 / deviceInfo->tracks);
         }
-- 
2.51.0

