diff -Naur pip-18.0/src/pip/_internal/utils/outdated.py pip-18.0.noupdate/src/pip/_internal/utils/outdated.py
--- pip-18.0/src/pip/_internal/utils/outdated.py	2018-07-22 07:26:50.000000000 +0000
+++ pip-18.0.noupdate/src/pip/_internal/utils/outdated.py	2018-11-30 05:56:07.401080985 +0000
@@ -122,22 +122,6 @@
 
         remote_version = packaging_version.parse(pypi_version)
 
-        # Determine if our pypi_version is older
-        if (pip_version < remote_version and
-                pip_version.base_version != remote_version.base_version and
-                was_installed_by_pip('pip')):
-            # Advise "python -m pip" on Windows to avoid issues
-            # with overwriting pip.exe.
-            if WINDOWS:
-                pip_cmd = "python -m pip"
-            else:
-                pip_cmd = "pip"
-            logger.warning(
-                "You are using pip version %s, however version %s is "
-                "available.\nYou should consider upgrading via the "
-                "'%s install --upgrade pip' command.",
-                pip_version, pypi_version, pip_cmd
-            )
     except Exception:
         logger.debug(
             "There was an error checking the latest version of pip",
