From 3780f30ed4c2bd412a5794d501da286ba38bf0b8 Mon Sep 17 00:00:00 2001
From: Boyuan Yang <byang@debian.org>
Date: Mon, 2 Sep 2024 18:52:35 -0400
Subject: [PATCH] DEBIAN: Fix gcc14 FTBFS

Bug-Debian: https://bugs.debian.org/1077398
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 xcur2png.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xcur2png.c b/xcur2png.c
index 8723a10..ea235ff 100644
--- a/xcur2png.c
+++ b/xcur2png.c
@@ -671,7 +671,7 @@ int saveConfAndPNGs (const XcursorImages* xcIs, const char* xcurFilePart, int su
   int ret;
   int count = 0;
   char pngName[PATH_MAX] = {0};
-  extern dry_run;
+  extern int dry_run;
   
   //Write comment on config-file.
   fprintf (conffp,"#size\txhot\tyhot\tPath to PNG image\tdelay\n");
-- 
2.52.0

