diff --git a/external/llvm-project/clang/lib/Basic/Targets/LoongArch.h b/external/llvm-project/clang/lib/Basic/Targets/LoongArch.h
index 3ad5abca927b..2fcd133ea87b 100644
--- a/external/llvm-project/clang/lib/Basic/Targets/LoongArch.h
+++ b/external/llvm-project/clang/lib/Basic/Targets/LoongArch.h
@@ -60,6 +60,8 @@ public:
     SuitableAlign = 128;
     WCharType = SignedInt;
     WIntType = UnsignedInt;
+    HasFloat128 = true;
+    HasFloat16 = true;
   }

   bool setCPU(const std::string &Name) override {
diff --git a/external/llvm-project/llvm/lib/Analysis/ConstantFolding.cpp b/external/llvm-project/llvm/lib/Analysis/ConstantFolding.cpp
index b58f9b26a865..9a437ff6757e 100644
--- a/external/llvm-project/llvm/lib/Analysis/ConstantFolding.cpp
+++ b/external/llvm-project/llvm/lib/Analysis/ConstantFolding.cpp
@@ -1936,7 +1936,7 @@ Constant *GetConstantFoldFPValue(double V, Type *Ty) {
 }

 #if defined(HAS_IEE754_FLOAT128) && defined(HAS_LOGF128)
-Constant *GetConstantFoldFPValue128(float128 V, Type *Ty) {
+Constant *GetConstantFoldFPValue128(_Float128 V, Type *Ty) {
   if (Ty->isFP128Ty())
     return ConstantFP::get(Ty, V);
   llvm_unreachable("Can only constant fold fp128");
@@ -1982,7 +1982,7 @@ Constant *ConstantFoldFP(double (*NativeFP)(double), const APFloat &V,
 }

 #if defined(HAS_IEE754_FLOAT128) && defined(HAS_LOGF128)
-Constant *ConstantFoldFP128(float128 (*NativeFP)(float128), const APFloat &V,
+Constant *ConstantFoldFP128(_Float128 (*NativeFP)(_Float128), const APFloat &V,
                             Type *Ty) {
   llvm_fenv_clearexcept();
   float128 Result = NativeFP(V.convertToQuad());
diff --git a/external/llvm-project/llvm/lib/Target/LoongArch/LoongArchISelLowering.cpp b/external/llvm-project/llvm/lib/Target/LoongArch/LoongArchISelLowering.cpp
index 6946ed554a7e..5405460bbb18 100644
--- a/external/llvm-project/llvm/lib/Target/LoongArch/LoongArchISelLowering.cpp
+++ b/external/llvm-project/llvm/lib/Target/LoongArch/LoongArchISelLowering.cpp
@@ -159,6 +159,7 @@ LoongArchTargetLowering::LoongArchTargetLowering(const TargetMachine &TM,
     setOperationAction({ISD::SDIV, ISD::UDIV, ISD::SREM, ISD::UREM}, MVT::i32,
                        Custom);
     setOperationAction(ISD::LROUND, MVT::i32, Custom);
+    setOperationAction(ISD::SPLAT_VECTOR, MVT::f16, Custom);
   }

   // Set operations for LA32 only.
@@ -219,6 +220,9 @@ LoongArchTargetLowering::LoongArchTargetLowering(const TargetMachine &TM,
         setOperationAction(ISD::UINT_TO_FP, MVT::i64, Custom);
       }
     }
+
+    setLibcallName(RTLIB::FPEXT_F16_F32, "__extendhfsf2");
+    setLibcallName(RTLIB::FPROUND_F32_F16, "__truncsfhf2");
   }

   // Set operations for 'D' feature.
