diff -up sbcl-1.4.2/contrib/asdf-module.mk.optflags sbcl-1.4.2/contrib/asdf-module.mk
--- sbcl-1.4.2/contrib/asdf-module.mk.optflags	2017-11-30 08:21:48.000000000 -0600
+++ sbcl-1.4.2/contrib/asdf-module.mk	2017-12-01 10:40:50.979435420 -0600
@@ -24,6 +24,8 @@ ifeq (Linux,$(UNAME))
   EXTRA_CFLAGS+=-D_GNU_SOURCE
 endif
 
+#EXTRA_CFLAGS += $(RPM_OPT_FLAGS)
+
 export CC SBCL EXTRA_CFLAGS
 
 all: $(FASL) $(ASD)
diff -up sbcl-1.4.2/src/runtime/Config.arm-linux.optflags sbcl-1.4.2/src/runtime/Config.arm-linux
--- sbcl-1.4.2/src/runtime/Config.arm-linux.optflags	2017-11-30 08:21:48.000000000 -0600
+++ sbcl-1.4.2/src/runtime/Config.arm-linux	2017-12-01 10:40:50.979435420 -0600
@@ -9,7 +9,7 @@
 # provided with absolutely no warranty. See the COPYING and CREDITS
 # files for more information.
 
-CFLAGS += -marm -march=armv5
+#CFLAGS += -marm -march=armv5 
 NM = ./linux-nm
 
 ASSEM_SRC = arm-assem.S ldso-stubs.S
diff -up sbcl-1.4.2/src/runtime/GNUmakefile.optflags sbcl-1.4.2/src/runtime/GNUmakefile
--- sbcl-1.4.2/src/runtime/GNUmakefile.optflags	2017-12-01 10:40:50.979435420 -0600
+++ sbcl-1.4.2/src/runtime/GNUmakefile	2017-12-01 10:41:23.431758358 -0600
@@ -30,7 +30,7 @@ __LDFLAGS__ =
 
 include ../../output/prefix.def
 
-CFLAGS += -g -Wall -Wundef -Wsign-compare -Wpointer-arith -O3
+#CFLAGS += -g -Wall -Wundef -Wsign-compare -Wpointer-arith -O3
 ASFLAGS += $(CFLAGS)
 CPPFLAGS += -I. -DSBCL_PREFIX=\"$(SBCL_PREFIX)\"
 

