From 3ac975a41c198c23cf9c9e987c8057ebc0532710 Mon Sep 17 00:00:00 2001
From: Daniel Jacobowitz <dan@debian.org>
Date: Sat, 4 Apr 2026 01:40:43 +0800
Subject: [PATCH 01/25] DEBIAN: Build with tcpwrapper and SSL support

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 builddefs.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/builddefs.h b/builddefs.h
index e908352..683b46b 100644
--- a/builddefs.h
+++ b/builddefs.h
@@ -1,9 +1,9 @@
 #ifndef VSF_BUILDDEFS_H
 #define VSF_BUILDDEFS_H
 
-#undef VSF_BUILD_TCPWRAPPERS
+#define VSF_BUILD_TCPWRAPPERS 1
 #define VSF_BUILD_PAM
-#undef VSF_BUILD_SSL
+#define VSF_BUILD_SSL
 
 #endif /* VSF_BUILDDEFS_H */
 
-- 
2.52.0

