From 43365b2f8573445618e36cdc596460f55566921a Mon Sep 17 00:00:00 2001
From: Jiri Skala <jskala@redhat.com>
Date: Sat, 4 Apr 2026 01:43:34 +0800
Subject: [PATCH 05/25] DEBIAN: trim white spaces from option values (Closes:
 #419857, #536803)

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 parseconf.c |  2 +-
 str.c       | 12 ++++++++++++
 str.h       |  1 +
 sysutil.c   | 12 ++++++++++++
 sysutil.h   |  1 +
 5 files changed, 27 insertions(+), 1 deletion(-)

diff --git a/parseconf.c b/parseconf.c
index 19a1429..8141dcf 100644
--- a/parseconf.c
+++ b/parseconf.c
@@ -283,7 +283,7 @@ vsf_parseconf_load_setting(const char* p_setting, int errs_fatal)
         }
         else
         {
-          *p_curr_setting = str_strdup(&s_value_str);
+          *p_curr_setting = str_strdup_trimmed(&s_value_str);
         }
         return;
       }
diff --git a/str.c b/str.c
index 6596204..6abab18 100644
--- a/str.c
+++ b/str.c
@@ -104,6 +104,18 @@ str_strdup(const struct mystr* p_str)
   return vsf_sysutil_strdup(str_getbuf(p_str));
 }
 
+const char*
+str_strdup_trimmed(const struct mystr* p_str)
+{
+  const char* p_trimmed = str_getbuf(p_str);
+  int h, t, newlen;
+
+  for (h = 0; h < (int)str_getlen(p_str) && vsf_sysutil_isspace(p_trimmed[h]); h++) ;
+  for (t = str_getlen(p_str) - 1; t >= 0 && vsf_sysutil_isspace(p_trimmed[t]); t--) ;
+  newlen = t - h + 1;
+  return newlen ? vsf_sysutil_strndup(p_trimmed+h, (unsigned int)newlen) : 0L;
+}
+
 void
 str_alloc_alt_term(struct mystr* p_str, const char* p_src, char term)
 {
diff --git a/str.h b/str.h
index ab0a9a4..f945474 100644
--- a/str.h
+++ b/str.h
@@ -31,6 +31,7 @@ void str_alloc_ulong(struct mystr* p_str, unsigned long the_ulong);
 void str_alloc_filesize_t(struct mystr* p_str, filesize_t the_filesize);
 void str_copy(struct mystr* p_dest, const struct mystr* p_src);
 const char* str_strdup(const struct mystr* p_str);
+const char* str_strdup_trimmed(const struct mystr* p_str);
 void str_empty(struct mystr* p_str);
 void str_free(struct mystr* p_str);
 void str_trunc(struct mystr* p_str, unsigned int trunc_len);
diff --git a/sysutil.c b/sysutil.c
index 02a26ef..2125e98 100644
--- a/sysutil.c
+++ b/sysutil.c
@@ -1035,6 +1035,18 @@ vsf_sysutil_strdup(const char* p_str)
   return strdup(p_str);
 }
 
+char*
+vsf_sysutil_strndup(const char* p_str, unsigned int p_len)
+{
+  char *new = (char *)malloc(p_len+1);
+
+  if (new == NULL)
+    return NULL;
+
+  new[p_len]='\0';
+  return (char *)memcpy(new, p_str, p_len);
+}
+
 void
 vsf_sysutil_memclr(void* p_dest, unsigned int size)
 {
diff --git a/sysutil.h b/sysutil.h
index ae9b98c..40e7566 100644
--- a/sysutil.h
+++ b/sysutil.h
@@ -187,6 +187,7 @@ int vsf_sysutil_wait_get_exitcode(
 /* Various string functions */
 unsigned int vsf_sysutil_strlen(const char* p_text);
 char* vsf_sysutil_strdup(const char* p_str);
+char* vsf_sysutil_strndup(const char* p_str, unsigned int p_len);
 void vsf_sysutil_memclr(void* p_dest, unsigned int size);
 void vsf_sysutil_memcpy(void* p_dest, const void* p_src,
                         const unsigned int size);
-- 
2.52.0

