From 256ec89af70591a8b5c0eaafbae10af3df9f5e89 Mon Sep 17 00:00:00 2001
From: Martin Nagy <mnagy@redhat.com>
Date: Sat, 4 Apr 2026 01:44:16 +0800
Subject: [PATCH 06/25] DEBIAN: Fix file listing issue with wildcard (Bugzilla:
 #392181)

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 ls.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/ls.c b/ls.c
index 7e1376d..13665f7 100644
--- a/ls.c
+++ b/ls.c
@@ -289,6 +289,25 @@ vsf_filename_passes_filter(const struct mystr* p_filename_str,
       {
         goto out;
       }
+      if (!must_match_at_current_pos)
+      {
+        struct mystr scan_fwd = INIT_MYSTR;
+
+        str_mid_to_end(&name_remain_str, &scan_fwd,
+                        indexx + str_getlen(&s_match_needed_str));
+        /* We're allowed to be greedy, test if it match further along
+         * keep advancing indexx while we can still match.
+         */
+        while( (locate_result = str_locate_str(&scan_fwd, &s_match_needed_str)),
+            locate_result.found )
+        {
+          indexx += locate_result.index + str_getlen(&s_match_needed_str);
+          str_mid_to_end(&scan_fwd, &temp_str,
+                         locate_result.index + str_getlen(&s_match_needed_str));
+          str_copy(&scan_fwd, &temp_str);
+        }
+       str_free(&scan_fwd);
+      }
       /* Chop matched string out of remainder */
       str_mid_to_end(&name_remain_str, &temp_str,
                      indexx + str_getlen(&s_match_needed_str));
-- 
2.52.0

