From 98a7a127d4c4e0abbe89fee6eb017e8346bb0b99 Mon Sep 17 00:00:00 2001
From: Michael Cree <mcree@orcon.net.nz>
Date: Sat, 4 Apr 2026 01:46:59 +0800
Subject: [PATCH 10/25] DEBIAN: Fixing FTBFS on alpha (Closes: #656182)

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 sysdeputil.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sysdeputil.c b/sysdeputil.c
index 8c24305..ff7ee43 100644
--- a/sysdeputil.c
+++ b/sysdeputil.c
@@ -86,6 +86,9 @@
   #include <linux/unistd.h>
   #include <errno.h>
   #include <syscall.h>
+  #if defined(__alpha__)
+    #define __NR_getpid  __NR_getxpid
+  #endif
 #endif
 
 #if defined(__linux__) && !defined(__ia64__) && !defined(__s390__)
-- 
2.52.0

