From 784fd4d0bf5058997806a71cee6f57e37b764f66 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Frings-F=C3=BCrst?= <debian@jff-webhosting.net>
Date: Sat, 4 Apr 2026 02:10:18 +0800
Subject: [PATCH 18/25] DEBIAN: Change default of seccomp_sandbox

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=735357
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 tunables.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tunables.c b/tunables.c
index 3b55868..1f03c63 100644
--- a/tunables.c
+++ b/tunables.c
@@ -235,7 +235,7 @@ tunables_load_defaults()
   tunable_isolate_network = 1;
   tunable_ftp_enable = 1;
   tunable_http_enable = 0;
-  tunable_seccomp_sandbox = 1;
+  tunable_seccomp_sandbox = 0;
   tunable_allow_writeable_chroot = 0;
 
   tunable_accept_timeout = 60;
-- 
2.52.0

