From 71f150384a16130b8a156dbef4028affc5095d3a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Frings-F=C3=BCrst?= <debian@jff-webhosting.net>
Date: Sat, 4 Apr 2026 02:12:44 +0800
Subject: [PATCH 21/25] DEBIAN: Change the modus of the log files to the same
 as via logrotate

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 logging.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/logging.c b/logging.c
index ad531d6..648b4a5 100644
--- a/logging.c
+++ b/logging.c
@@ -44,7 +44,7 @@ vsf_log_init(struct vsf_session* p_sess)
     if (tunable_xferlog_file)
     {
       retval = vsf_sysutil_create_or_open_file_append(tunable_xferlog_file,
-                                                      0600);
+                                                      0640);
     }
     if (vsf_sysutil_retval_is_error(retval))
     {
@@ -60,7 +60,7 @@ vsf_log_init(struct vsf_session* p_sess)
       if (tunable_vsftpd_log_file)
       {
         retval = vsf_sysutil_create_or_open_file_append(tunable_vsftpd_log_file,
-                                                        0600);
+                                                        0640);
       }
       if (vsf_sysutil_retval_is_error(retval))
       {
-- 
2.52.0

