From 0dba26355ef3f416c9d647e28528b78ffeb75e19 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sat, 4 Apr 2026 02:13:32 +0800
Subject: [PATCH 22/25] DEBIAN: fix no-pam FTBFS

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 sysdeputil.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/sysdeputil.c b/sysdeputil.c
index 61e0fa9..162c8ac 100644
--- a/sysdeputil.c
+++ b/sysdeputil.c
@@ -293,15 +293,19 @@ vsf_sysdep_check_auth(struct mystr* p_user_str,
         return 0;
       }
       p_crypted = crypt(str_getbuf(p_pass_str), p_spwd->sp_pwdp);
-      if (!vsf_sysutil_strcmp(p_crypted, p_spwd->sp_pwdp))
+      if (NULL != p_crypted && !vsf_sysutil_strcmp(p_crypted, p_spwd->sp_pwdp))
       {
         return 1;
       }
+      else
+      {
+        return 0;
+      }
     }
   }
   #endif /* VSF_SYSDEP_HAVE_SHADOW */
   p_crypted = crypt(str_getbuf(p_pass_str), p_pwd->pw_passwd);
-  if (!vsf_sysutil_strcmp(p_crypted, p_pwd->pw_passwd))
+  if (NULL != p_crypted && !vsf_sysutil_strcmp(p_crypted, p_pwd->pw_passwd))
   {
     return 1;
   }
-- 
2.52.0

