From d862003ae974407508618c25e48253505609f7f8 Mon Sep 17 00:00:00 2001
From: Junde Yhi <lmy441900@aosc.xyz>
Date: Sat, 20 Jul 2019 18:14:50 +0800
Subject: [PATCH 1/2] Makefile: disable git update index

Signed-off-by: Junde Yhi <lmy441900@aosc.xyz>
---
 archive/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/archive/Makefile b/archive/Makefile
index 3c2fe1fed..bfc74efef 100644
--- a/archive/Makefile
+++ b/archive/Makefile
@@ -214,7 +214,6 @@ $(RESOURCES_DOT_GO):
 	APP="$$LANTERN_UI/app" && \
 	DIST="$$LANTERN_UI/dist" && \
 	echo 'var LANTERN_BUILD_REVISION = "$(GIT_REVISION_SHORTCODE)";' > $$APP/js/revision.js && \
-	git update-index --assume-unchanged $$APP/js/revision.js && \
 	DEST="$@" && \
 	cd $$LANTERN_UI && \
 	$(NPM) install && \
-- 
2.21.0

