diff --git a/build-aux/linux/inochi-creator.desktop b/build-aux/linux/inochi-creator.desktop
index 80a920e..0c71532 100644
--- a/build-aux/linux/inochi-creator.desktop
+++ b/build-aux/linux/inochi-creator.desktop
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=inochi-creator
 Exec=inochi-creator
-Icon=logo_256
+Icon=inochi-creator
 Type=Application
-Categories=Utility
\ No newline at end of file
+Categories=Utility
