diff --git a/source/creator/core/path.d b/source/creator/core/path.d
index 850ba23..c660e0a 100644
--- a/source/creator/core/path.d
+++ b/source/creator/core/path.d
@@ -152,12 +152,8 @@ string incGetAppLocalePath() {
 */
 string incGetAppLocalePathExtra() {
     
-    // AppImage locale dir is the root of the appimage
     version(linux) {
-        auto here = environment.get("HERE");
-        if (here) {
-            return here;
-        }    
+        return "/usr/share/inochi-creator";
     }
 
     return null;
