From ab6a5fd82e113d6814ed474c076e30b6c4ac4e2c Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Mon, 11 Aug 2025 21:08:14 +0800
Subject: [PATCH 3/3] fix: add missing header

---
 include/ntfs-3g/ntfstime.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/ntfs-3g/ntfstime.h b/include/ntfs-3g/ntfstime.h
index f3a89dd..f069371 100644
--- a/include/ntfs-3g/ntfstime.h
+++ b/include/ntfs-3g/ntfstime.h
@@ -29,6 +29,7 @@
 #endif
 #ifdef HAVE_SYS_STAT_H
 #include <sys/stat.h>
+#include <time.h>
 #endif
 #ifdef HAVE_GETTIMEOFDAY
 #include <sys/time.h>
-- 
2.50.1

