From 6ea8a21fa1f671210ad0f8175b24d4c2c2da5517 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Mon, 11 Aug 2025 21:08:14 +0800
Subject: [PATCH 3/3] AOSCOS: fix: add missing header

---
 include/ntfs-3g/ntfstime.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/ntfs-3g/ntfstime.h b/include/ntfs-3g/ntfstime.h
index f3a89dd8..f0693711 100644
--- a/include/ntfs-3g/ntfstime.h
+++ b/include/ntfs-3g/ntfstime.h
@@ -29,6 +29,7 @@
 #endif
 #ifdef HAVE_SYS_STAT_H
 #include <sys/stat.h>
+#include <time.h>
 #endif
 #ifdef HAVE_GETTIMEOFDAY
 #include <sys/time.h>
-- 
2.52.0

