From: =?utf-8?q?Hanno_B=C3=B6ck?= <hanno@hboeck.de>
Date: Wed, 1 Jul 2015 11:00:58 +0000
Subject: Don't free the 'tmp' variable when it's used later.

Origin: https://src.fedoraproject.org/rpms/mhash/c/baa57fdba46abadbac4d05762d8812db1cb1b93d?branch=master
Origin: https://sourceforge.net/p/mhash/mailman/message/34273046/
---
 src/hmac_test.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hmac_test.c b/src/hmac_test.c
index 1ecdef8..9c0e4d5 100644
--- a/src/hmac_test.c
+++ b/src/hmac_test.c
@@ -72,7 +72,7 @@ int main()
 		return(MUTILS_INVALID_RESULT);
 	}
 
-	mutils_free(tmp);
+	/* mutils_free(tmp); */
 
 	/* Test No 2 */	
 	
