From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Sun, 31 May 2020 21:40:34 +0100
Subject: _Bool to bool

---
 include/mutils/mutils.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/mutils/mutils.h b/include/mutils/mutils.h
index 2b3872e..8d5da59 100644
--- a/include/mutils/mutils.h
+++ b/include/mutils/mutils.h
@@ -97,7 +97,7 @@ typedef unsigned char mutils_word8;
  */
 
 #if defined(HAVE__BOOL)
-#define mutils_boolean _Bool
+#define mutils_boolean bool
 #else
 typedef char mutils_boolean;
 #endif
