--- a/Makefile	2018-09-12 22:58:58.393141656 +0800
+++ b/Makefile	2018-09-12 23:00:04.520604251 +0800
@@ -32,15 +32,15 @@
 	-update-rc.d earlyoom start 18 2 3 4 5 . stop 20 0 1 6 .
 
 earlyoom.%: earlyoom.%.in
-	sed "s|:TARGET:|$(PREFIX)$(BINDIR)|g;s|:SYSCONFDIR:|$(SYSCONFDIR)|g" $< > $@
+	sed "s|:TARGET:|$(BINDIR)|g;s|:SYSCONFDIR:|$(SYSCONFDIR)|g" $< > $@
 
 install-default: earlyoom.default install-man
 	install -d $(DESTDIR)$(SYSCONFDIR)/default/
 	install -m 644 $< $(DESTDIR)$(SYSCONFDIR)/default/earlyoom
 
 install-bin: earlyoom
-	install -d $(DESTDIR)$(PREFIX)$(BINDIR)/
-	install -m 755 $< $(DESTDIR)$(PREFIX)$(BINDIR)/
+	install -d $(DESTDIR)$(BINDIR)/
+	install -m 755 $< $(DESTDIR)$(BINDIR)/
 
 install-man: earlyoom.1.gz
 	install -d $(DESTDIR)$(PREFIX)/share/man/man1/
@@ -61,7 +61,7 @@
 	update-rc.d earlyoom remove
 
 uninstall-bin:
-	rm -f $(DESTDIR)$(PREFIX)$(BINDIR)/earlyoom
+	rm -f $(DESTDIR)$(BINDIR)/earlyoom
 
 format:
 	clang-format -i *.h *.c
