diff --git a/daemon/siginfo.c b/daemon/siginfo.c
index b1b8c79f..23e4ad52 100644
--- a/daemon/siginfo.c
+++ b/daemon/siginfo.c
@@ -58,10 +58,9 @@
 
 #define SIGINFO_MAX_BT_FRAMES 20
 
-#if defined(SA_SIGINFO) && !defined(__arm__) && !defined(__aarch64__) && !defined(__ia64__) && !defined(__alpha__) && !defined (__FreeBSD_kernel__) && !defined (__sh__) && !defined(__APPLE__)
+#if defined(SA_SIGINFO) && !defined(__arm__) && !defined(__aarch64__) && !defined(__ia64__) && !defined(__alpha__) && !defined (__FreeBSD_kernel__) && !defined (__sh__) && !defined(__APPLE__) && !defined(__loongarch__) && !defined(__riscv)
 # define USE_UCONTEXT
 # include <ucontext.h>
-#endif
 
 #if defined(__powerpc64__)
 # define SIGINFO_REGISTER(ucontext, index) ((ucontext)->uc_mcontext.gp_regs[index])
@@ -73,6 +72,8 @@
 # define SIGINFO_REGISTER(ucontext, index) ((ucontext)->uc_mcontext.gregs[index])
 #endif
 
+#endif
+
 #if defined(REG_RIP)
 # define SIGINFO_IP_REG REG_RIP
 # define SIGINFO_BP_REG REG_RBP
