From 3ec468c02e9832466ab7971eb776be3b99cbef9f Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Wed, 15 Dec 2021 21:32:35 -0600
Subject: [PATCH] quirk: use og cdparanoia for less bugs

---
 whipper/program/cdparanoia.py | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/whipper/program/cdparanoia.py b/whipper/program/cdparanoia.py
index 5c4b779..8745c5d 100644
--- a/whipper/program/cdparanoia.py
+++ b/whipper/program/cdparanoia.py
@@ -271,10 +271,10 @@ class ReadTrackTask(task.Task):
 
         bufsize = 1024
         if self._overread:
-            argv = ["cd-paranoia", "--stderr-progress",
+            argv = ["cdparanoia", "--stderr-progress",
                     "--sample-offset=%d" % self._offset, "--force-overread", ]
         else:
-            argv = ["cd-paranoia", "--stderr-progress",
+            argv = ["cdparanoia", "--stderr-progress",
                     "--sample-offset=%d" % self._offset, ]
         if self._device:
             argv.extend(["--force-cdrom-device", self._device, ])
@@ -285,7 +285,7 @@ class ReadTrackTask(task.Task):
         logger.debug('running %s', (" ".join(argv), ))
         if self._offset > 587:
             logger.warning(
-                "because of a cd-paranoia upstream bug whipper may fail to "
+                "because of a cdparanoia upstream bug whipper may fail to "
                 "work correctly when using offset values > 587 (current "
                 "value: %d) and print warnings like this: 'file size 0 did "
                 "not match expected size'. For more details please check the "
@@ -296,7 +296,7 @@ class ReadTrackTask(task.Task):
             )
         if stopTrack == 99:
             logger.warning(
-                "because of a cd-paranoia upstream bug whipper may fail to "
+                "because of a cdparanoia upstream bug whipper may fail to "
                 "rip the last track of a CD when it has got 99 of them. "
                 "For more details please check the following issue: "
                 "https://github.com/whipper-team/whipper/issues/302"
@@ -310,7 +310,7 @@ class ReadTrackTask(task.Task):
                                          close_fds=True)
         except OSError as e:
             if e.errno == errno.ENOENT:
-                raise common.MissingDependencyException('cd-paranoia')
+                raise common.MissingDependencyException('cdparanoia')
 
             raise
 
@@ -572,8 +572,8 @@ _VERSION_RE = re.compile(
 
 
 def getCdParanoiaVersion():
-    getter = common.VersionGetter('cd-paranoia',
-                                  ["cd-paranoia", "-V"],
+    getter = common.VersionGetter('cdparanoia',
+                                  ["cdparanoia", "-V"],
                                   _VERSION_RE,
                                   "%(version)s %(release)s")
 
@@ -599,12 +599,12 @@ class AnalyzeTask(ctask.PopenTask):
     def __init__(self, device=None):
         # cdparanoia -A *always* writes cdparanoia.log
         self.cwd = tempfile.mkdtemp(suffix='.whipper.cache')
-        self.command = ['cd-paranoia', '-A']
+        self.command = ['cdparanoia', '-A']
         if device:
             self.command += ['-d', device]
 
     def commandMissing(self):
-        raise common.MissingDependencyException('cd-paranoia')
+        raise common.MissingDependencyException('cdparanoia')
 
     def readbyteserr(self, bytes_stderr):
         self._output.append(bytes_stderr)
-- 
2.30.2

