From 553421b29fc962e6f0269a690ea908a741824a88 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu011@gmail.com>
Date: Wed, 10 Jun 2020 16:04:44 -0600
Subject: [PATCH] build: enable shell completions

---
 Cargo.toml |  2 +-
 build.rs   | 12 ++++++------
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index ea8580d..1b04c0d 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,7 +9,7 @@ description = "A fast static site generator with everything built-in"
 homepage = "https://www.getzola.org"
 repository = "https://github.com/getzola/zola"
 keywords = ["static", "site", "generator", "blog"]
-# build = "build.rs"
+build = "build.rs"
 
 [build-dependencies]
 clap = "2"
diff --git a/build.rs b/build.rs
index f6e8977..691f075 100644
--- a/build.rs
+++ b/build.rs
@@ -1,12 +1,12 @@
-// use clap::Shell;
+use clap::Shell;
 
 include!("src/cli.rs");
 
 fn main() {
     // disabled below as it fails in CI
-    //    let mut app = build_cli();
-    //    app.gen_completions("zola", Shell::Bash, "completions/");
-    //    app.gen_completions("zola", Shell::Fish, "completions/");
-    //    app.gen_completions("zola", Shell::Zsh, "completions/");
-    //    app.gen_completions("zola", Shell::PowerShell, "completions/");
+    let mut app = build_cli();
+    app.gen_completions("zola", Shell::Bash, "completions/");
+    app.gen_completions("zola", Shell::Fish, "completions/");
+    app.gen_completions("zola", Shell::Zsh, "completions/");
+    app.gen_completions("zola", Shell::PowerShell, "completions/");
 }
-- 
2.27.0

