diff --git a/Cargo.lock b/Cargo.lock
index 5225d59..11ca950 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -3258,6 +3258,7 @@ dependencies = [
  "open",
  "percent-encoding",
  "relative-path",
+ "search",
  "serde_json",
  "site",
  "termcolor",
diff --git a/Cargo.toml b/Cargo.toml
index 5445f5c..a429539 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -37,7 +37,7 @@ open = "1.2"
 globset = "0.4"
 relative-path = "1"
 serde_json = "1.0"
-
+search = { path = "components/search" }
 site = { path = "components/site" }
 errors = { path = "components/errors" }
 front_matter = { path = "components/front_matter" }
diff --git a/components/utils/src/de.rs b/components/utils/src/de.rs
index e294e92..568d71a 100644
--- a/components/utils/src/de.rs
+++ b/components/utils/src/de.rs
@@ -1,5 +1,4 @@
 use serde::{Deserialize, Deserializer};
-use serde_derive::Deserialize;
 use tera::{Map, Value};
 
 /// Used as an attribute when we want to convert from TOML to a string date
