Fix FTBFS with gcc4.4
--- trunk.orig/demo/cfa.cpp
+++ trunk/demo/cfa.cpp
@@ -19,7 +19,7 @@
  */
 
 
-
+#include <cstdio>
 #include <iostream>
 #include <libopenraw/libopenraw.h>
 #include <libopenraw/debug.h>
--- trunk.orig/demo/thumb.cpp
+++ trunk/demo/thumb.cpp
@@ -19,7 +19,7 @@
  */
 
 
-
+#include <cstdio>
 #include <iostream>
 #include <libopenraw/libopenraw.h>
 #include <libopenraw/debug.h>
