--- cde-2.3.2/config/cf/linux.cf	2020-01-12 17:10:57.000000000 -0800
+++ cde-2.3.2-patch/config/cf/linux.cf	2020-11-10 19:01:46.173992417 -0800
@@ -247,6 +247,9 @@
 # define ServerExtraDefines    -DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
 #endif /* AArch64Architecture */
 
+#ifdef Mips64Architecture
+# define LinuxMachineDefines    -D__mips64__
+#endif /* MipsArchitecture */
 
 #ifdef PpcArchitecture
 # ifndef OptimizedCDebugFlags
--- cde-2.3.2/config/cf/Imake.cf	2020-01-12 17:10:57.000000000 -0800
+++ cde-2.3.2-patch/config/cf/Imake.cf	2020-11-10 19:01:14.060714137 -0800
@@ -487,6 +487,26 @@
 #   define AArch64Architecture
 #   undef __aarch64__
 # endif
+# if defined(__mips__) || defined(mips)
+#   define MipsArchitecture
+#   ifdef __LP64__
+#     define Mips64Architecture
+#   endif
+#   ifdef arc
+#     define ArcArchitecture
+#     undef arc
+#   endif
+#   ifdef sgi
+#     define SGIArchitecture
+#     undef sgi
+#   endif
+#   ifdef pmax
+#     define PmaxArchitecture
+#     undef pmax
+#   endif
+#   undef mips
+#   undef __mips__
+# endif
 #endif /* linux */
 
 #if defined(sequent) || defined(_SEQUENT_)
