Description: Use stdbool.h to fix FTBFS with GCC 15.
Author: Bas Couwenberg <sebastic@debian.org>
Bug-Debian: https://bugs.debian.org/1097529
Forwarded: https://gitlab.com/osm-c-tools/osmctools/-/merge_requests/2

--- a/src/osmconvert.c
+++ b/src/osmconvert.c
@@ -461,6 +461,7 @@ const char* helptext=
 #define _FILE_OFFSET_BITS 64
 #include <zlib.h>
 #include <inttypes.h>
+#include <stdbool.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
@@ -471,7 +472,6 @@ const char* helptext=
 #include <fcntl.h>
 #include <signal.h>
 
-typedef enum {false= 0,true= 1} bool;
 typedef uint8_t byte;
 typedef unsigned int uint;
 #define isdig(x) isdigit((unsigned char)(x))
--- a/src/osmfilter.c
+++ b/src/osmfilter.c
@@ -374,6 +374,7 @@ const char* helptext=
 
 #define _FILE_OFFSET_BITS 64
 #include <inttypes.h>
+#include <stdbool.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
@@ -383,7 +384,6 @@ const char* helptext=
 #include <fcntl.h>
 #include <signal.h>
 
-typedef enum {false= 0,true= 1} bool;
 typedef uint8_t byte;
 typedef unsigned int uint;
 #define isdig(x) isdigit((unsigned char)(x))
--- a/src/osmupdate.c
+++ b/src/osmupdate.c
@@ -169,6 +169,7 @@ const char* helptext=
 
 #define _FILE_OFFSET_BITS 64
 #include <inttypes.h>
+#include <stdbool.h>
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
@@ -180,7 +181,6 @@ const char* helptext=
 #include <signal.h>
 #include <sys/stat.h>
 
-typedef enum {false= 0,true= 1} bool;
 typedef uint8_t byte;
 typedef unsigned int uint;
 #define isdig(x) isdigit((unsigned char)(x))
