From 22a1bd82461a65afb6a87b57bf286ebff221e7db Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Wed, 5 Feb 2020 08:35:39 +0000
Subject: [PATCH] install-packages: no fetch and update

---
 scripts/install-packages.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/install-packages.sh b/scripts/install-packages.sh
index 734f1a7..1f80ed4 100755
--- a/scripts/install-packages.sh
+++ b/scripts/install-packages.sh
@@ -33,7 +33,7 @@ install_package() {
     local recipe="$(resolve_recipe "$1")"
     local recipe_options=($(resolve_recipe_options "$1"))

-    fetch_or_update_package
+    #fetch_or_update_package

     if [[ -n "${recipe}" ]]; then
         require 'recipe'
--
2.23.0

