From 988453944641a4283f00777fa87f85ae726e6ea6 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 27 Jul 2020 21:39:09 +0800
Subject: [PATCH] Source AOSC schema bundle

---
 all-packages.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/all-packages.conf b/all-packages.conf
index 4adc89f..267cedd 100755
--- a/all-packages.conf
+++ b/all-packages.conf
@@ -4,3 +4,4 @@ package_list=()
 
 source "${root_dir:-.}"/preset-packages.conf
 source "${root_dir:-.}"/extra-packages.conf
+source "${root_dir:-.}"/aosc-schema-bundle.conf
-- 
2.27.0

