From 3e51867b3139b1d203250e2ed197d8fa6c3a3254 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Thu, 30 Jul 2020 23:29:08 +0800
Subject: [PATCH] Package name do not #rime-

---
 scripts/resolver.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/resolver.sh b/scripts/resolver.sh
index 3ec4132..4f0a6d4 100755
--- a/scripts/resolver.sh
+++ b/scripts/resolver.sh
@@ -19,7 +19,7 @@ resolve_user_name() {
 resolve_package_name() {
     local package="${1%%[@:]*}"
     local repo_name="${package##*/}"
-    local package_name="${repo_name#rime-}"
+    local package_name="${repo_name}"
     echo "${package_name}"
 }
 
-- 
2.27.0

