--- a/pyproject.toml	2026-07-19 01:11:16.098951878 +0800
+++ b/pyproject.toml	2026-07-19 01:11:03.558871474 +0800
@@ -1,8 +1,6 @@
 [build-system]
 build-backend = "mesonpy"
 requires = [
-    'flex-bin ; sys_platform == "linux" or sys_platform == "darwin"',
-    'bison-bin ; sys_platform == "linux" or sys_platform == "darwin"',
     'winflexbison-bin>=2.5.25.1 ; sys_platform == "win32"',
     "meson-python >= 0.14.0",
     "meson >= 1.2.1",
