From 0c73525023b821bb46afb6ae52567b2ddb352dce Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sun, 13 Apr 2025 18:46:51 +0800
Subject: [PATCH] Install the actual man pages instead of the sources

---
 Makefile | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index 87966a9..75d9a63 100644
--- a/Makefile
+++ b/Makefile
@@ -64,13 +64,13 @@ UTILS = $(INSTALLABLE) \
 LDLIBS=libgif.a -lm
 
 MANUAL_PAGES = \
-	doc/gif2rgb.xml \
-	doc/gifbuild.xml \
-	doc/gifclrmp.xml \
-	doc/giffix.xml \
-	doc/giflib.xml \
-	doc/giftext.xml \
-	doc/giftool.xml
+	doc/gif2rgb.1 \
+	doc/gifbuild.1 \
+	doc/gifclrmp.1 \
+	doc/giffix.1 \
+	doc/giflib.7 \
+	doc/giftext.1 \
+	doc/giftool.1
 
 SOEXTENSION	= so
 LIBGIFSO	= libgif.$(SOEXTENSION)
@@ -180,7 +180,7 @@ EXTRAS =     README \
 	     doc/gifstandard \
 
 DSOURCES = Makefile *.[ch]
-DOCS = doc/*.xml doc/*.1 doc/*.7 doc/*.html doc/index.html.in doc/00README doc/Makefile
+DOCS = doc/*.1 doc/*.1 doc/*.7 doc/*.html doc/index.html.in doc/00README doc/Makefile
 ALL =  $(DSOURCES) $(DOCS) tests pic $(EXTRAS)
 giflib-$(VERSION).tar.gz: $(ALL)
 	$(TAR) --transform='s:^:giflib-$(VERSION)/:' -czf giflib-$(VERSION).tar.gz $(ALL)
-- 
2.49.0

