From be069dedc05438bc001396abf112ae804687966c Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Thu, 19 Mar 2026 15:17:47 +0000
Subject: [PATCH] doc/Makefile: disable HTML documentation

---
 Makefile     | 2 +-
 doc/Makefile | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 75d9a63..7851b20 100644
--- a/Makefile
+++ b/Makefile
@@ -180,7 +180,7 @@ EXTRAS =     README \
 	     doc/gifstandard \
 
 DSOURCES = Makefile *.[ch]
-DOCS = doc/*.1 doc/*.1 doc/*.7 doc/*.html doc/index.html.in doc/00README doc/Makefile
+DOCS = doc/*.1 doc/*.1 doc/*.7 doc/00README doc/Makefile
 ALL =  $(DSOURCES) $(DOCS) tests pic $(EXTRAS)
 giflib-$(VERSION).tar.gz: $(ALL)
 	$(TAR) --transform='s:^:giflib-$(VERSION)/:' -czf giflib-$(VERSION).tar.gz $(ALL)
diff --git a/doc/Makefile b/doc/Makefile
index b34ece0..8e81c37 100644
--- a/doc/Makefile
+++ b/doc/Makefile
@@ -15,7 +15,7 @@
 .adoc.html:
 	asciidoc $<
 
-all: allhtml manpages
+all: manpages
 
 # The distinction between XMLMAN and XMLINTERNAL is because
 # some pages shouldn't be installed as part of a binary package;
-- 
2.52.0

