Description: don't reset terminal too early
Author: Serge Hallyn <serge.hallyn@ubuntu.com>
Forwarded: yes

Index: vde2-2.3.2/src/vdeterm.c
===================================================================
--- vde2-2.3.2.orig/src/vdeterm.c	2011-11-23 10:41:18.000000000 -0600
+++ vde2-2.3.2/src/vdeterm.c	2012-07-24 17:25:52.293126588 -0500
@@ -20,11 +20,13 @@
 
 char *prompt;
 static struct termios tiop;
+int termset = 0;
 
 static void cleanup(void)
 {
 	fprintf(stderr,"\n");
-	tcsetattr(STDIN_FILENO,TCSAFLUSH,&tiop);
+	if (termset)
+		tcsetattr(STDIN_FILENO,TCSAFLUSH,&tiop);
 }
 
 static void sig_handler(int sig)
@@ -135,6 +137,7 @@
 	newtiop.c_lflag &= ~ICANON;
 	newtiop.c_lflag &= ~ECHO;
 	tcsetattr(STDIN_FILENO,TCSAFLUSH,&newtiop);
+	termset = 1;
 	flags = fcntl(fd, F_GETFL);
 	flags |= O_NONBLOCK;
 	fcntl(fd, F_SETFL, flags);
