diff -Naur binutils-gdb/bfd/config.bfd binutils-gdb.n64/bfd/config.bfd
--- binutils-gdb/bfd/config.bfd	2018-01-31 23:55:49.747223789 +0000
+++ binutils-gdb.n64/bfd/config.bfd	2018-01-31 23:55:41.663608850 +0000
@@ -1188,11 +1188,21 @@
     targ_defvec=mips_elf32_be_vec
     targ_selvecs="mips_elf32_le_vec mips_elf64_be_vec mips_elf64_le_vec mips_ecoff_be_vec mips_ecoff_le_vec"
     ;;
-  mips64*el-*-linux*)
+  mips*64*el-*-linux*-gnuabi64)
+    targ_defvec=mips_elf64_trad_le_vec
+    targ_selvecs="mips_elf32_ntrad_be_vec mips_elf32_ntrad_le_vec mips_elf32_trad_be_vec mips_elf32_trad_le_vec mips_elf64_trad_be_vec"
+    want64=true
+    ;;
+  mips*64*-*-linux*-gnuabi64)
+    targ_defvec=mips_elf64_trad_be_vec
+    targ_selvecs="mips_elf32_ntrad_be_vec mips_elf32_ntrad_le_vec mips_elf32_trad_be_vec mips_elf32_trad_le_vec mips_elf64_trad_le_vec"
+    want64=true
+    ;;
+  mips*64*el-*-linux*)
     targ_defvec=mips_elf32_ntrad_le_vec
     targ_selvecs="mips_elf32_ntrad_be_vec mips_elf32_trad_le_vec mips_elf32_trad_be_vec mips_elf64_trad_le_vec mips_elf64_trad_be_vec"
     ;;
-  mips64*-*-linux*)
+  mips*64*-*-linux*)
     targ_defvec=mips_elf32_ntrad_be_vec
     targ_selvecs="mips_elf32_ntrad_le_vec mips_elf32_trad_be_vec mips_elf32_trad_le_vec mips_elf64_trad_be_vec mips_elf64_trad_le_vec"
     ;;
diff -Naur binutils-gdb/gas/configure binutils-gdb.n64/gas/configure
--- binutils-gdb/gas/configure	2018-01-31 23:55:49.750557238 +0000
+++ binutils-gdb.n64/gas/configure	2018-01-31 23:55:41.690276446 +0000
@@ -12380,7 +12380,10 @@
 	esac
 	# Decide which ABI to target by default.
 	case ${target} in
-	  mips64*-linux* | mips-sgi-irix6* | mips64*-freebsd* \
+          mips*64*-linux-gnuabi64)
+            mips_default_abi=N64_ABI
+            ;;
+          mips64*-linux* | mips-sgi-irix6* | mips64*-freebsd* \
 	  | mips64*-kfreebsd*-gnu | mips64*-ps2-elf*)
 	    mips_default_abi=N32_ABI
 	    ;;
diff -Naur binutils-gdb/gas/configure.ac binutils-gdb.n64/gas/configure.ac
--- binutils-gdb/gas/configure.ac	2018-01-31 23:55:49.750557238 +0000
+++ binutils-gdb.n64/gas/configure.ac	2018-01-31 23:55:41.690276446 +0000
@@ -345,6 +345,9 @@
 	esac
 	# Decide which ABI to target by default.
 	case ${target} in
+          mips*64*-linux-gnuabi64)
+            mips_default_abi=N64_ABI
+            ;;
 	  mips64*-linux* | mips-sgi-irix6* | mips64*-freebsd* \
 	  | mips64*-kfreebsd*-gnu | mips64*-ps2-elf*)
 	    mips_default_abi=N32_ABI
diff -Naur binutils-gdb/ld/configure.tgt binutils-gdb.n64/ld/configure.tgt
--- binutils-gdb/ld/configure.tgt	2018-01-31 23:55:49.763891036 +0000
+++ binutils-gdb.n64/ld/configure.tgt	2018-01-31 23:55:41.553605020 +0000
@@ -535,10 +535,16 @@
 mips*-*-vxworks*)	targ_emul=elf32ebmipvxworks
 			targ_extra_emuls="elf32elmipvxworks" ;;
 mips*-*-windiss)	targ_emul=elf32mipswindiss ;;
-mips64*el-*-linux-*)	targ_emul=elf32ltsmipn32
+mips*64*el-*-linux-gnuabi64)	targ_emul=elf64ltsmip
+			targ_extra_emuls="elf32btsmipn32 elf32ltsmip elf32btsmip elf32ltsmipn32 elf64btsmip"
+			targ_extra_libpath=$targ_extra_emuls ;;
+mips*64*-*-linux-gnuabi64)	targ_emul=elf64btsmip
+			targ_extra_emuls="elf32ltsmipn32 elf32btsmip elf32ltsmip elf32btsmipn32 elf64ltsmip"
+			targ_extra_libpath=$targ_extra_emuls ;;
+mips*64*el-*-linux-*)	targ_emul=elf32ltsmipn32
 			targ_extra_emuls="elf32btsmipn32 elf32ltsmip elf32btsmip elf64ltsmip elf64btsmip"
 			targ_extra_libpath=$targ_extra_emuls ;;
-mips64*-*-linux-*)	targ_emul=elf32btsmipn32
+mips*64*-*-linux-*)	targ_emul=elf32btsmipn32
 			targ_extra_emuls="elf32ltsmipn32 elf32btsmip elf32ltsmip elf64btsmip elf64ltsmip"
 			targ_extra_libpath=$targ_extra_emuls ;;
 mips*el-*-linux-*)	targ_emul=elf32ltsmip
