diff -Naur osm2pgrouting-2.3.2/CMakeLists.txt osm2pgrouting-2.3.2.gcc7/CMakeLists.txt
--- osm2pgrouting-2.3.2/CMakeLists.txt	2017-10-17 22:16:19.000000000 +0000
+++ osm2pgrouting-2.3.2.gcc7/CMakeLists.txt	2017-11-16 03:43:51.893163108 +0000
@@ -19,7 +19,7 @@
 
 FILE(GLOB osm2pgrouting_lib_SOURCES "${CMAKE_SOURCE_DIR}/src/*/*.cpp")
 
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_FILE_OFFSET_BITS=64 -std=c++0x")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_FILE_OFFSET_BITS=64 -std=c++14")
 set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -Wconversion -pedantic -Wextra  -frounding-math -Wno-deprecated -fmax-errors=10")
 
 

