diff -Naur simple-scan-42.1/data/simple-scan.desktop.in simple-scan-42.1.desktop/data/simple-scan.desktop.in
--- simple-scan-42.1/data/simple-scan.desktop.in	2022-04-18 22:55:01.487897400 -0700
+++ simple-scan-42.1.desktop/data/simple-scan.desktop.in	2023-07-03 07:10:04.636801441 -0700
@@ -9,4 +9,4 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Graphics;Scanning;
+Categories=Office;GNOME;GTK;
