diff -Naur libfreeaptx-0.1.1/Makefile libfreeaptx-0.1.1.aosc/Makefile
--- libfreeaptx-0.1.1/Makefile	2021-11-14 09:50:41.569107854 +0800
+++ libfreeaptx-0.1.1.aosc/Makefile	2021-11-14 11:17:04.111254801 +0800
@@ -8,9 +8,9 @@
 MKDIR = mkdir -p
 PRINTF = printf
 
-CFLAGS = -W -Wall -O3
-LDFLAGS = -s
-ARFLAGS = -rcs
+CFLAGS := -W -Wall -O3
+LDFLAGS := -s
+ARFLAGS := -rcs
 
 PREFIX = /usr/local
 BINDIR = bin
