diff --git a/sysctl.conf b/sysctl.conf
index e846a57d..5fed3529 100644
--- a/sysctl.conf
+++ b/sysctl.conf
@@ -56,7 +56,7 @@ net/ipv4/icmp_echo_ignore_broadcasts =1
 
 # This limits PID values to 4 digits, which allows tools like ps
 # to save screen space.
-kernel/pid_max=10000
+#kernel/pid_max=10000
 
 # Protects against creating or following links under certain conditions
 # See https://www.kernel.org/doc/Documentation/sysctl/fs.txt
