--- a/CMakeLists.txt	2024-06-03 16:58:51.556135094 +0800
+++ b/CMakeLists.txt	2024-06-03 16:58:53.850252932 +0800
@@ -6,7 +6,7 @@
 project(evolution-data-server
 	VERSION 3.44.4
 	LANGUAGES C CXX)
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 set(PROJECT_BUGREPORT "https://gitlab.gnome.org/GNOME/evolution-data-server/issues/")
 
 # Base Version: This is for API/version tracking for things like
