From 4a528a3af6abb6101b56befdde09bc0b85ed4cd7 Mon Sep 17 00:00:00 2001
From: Han Gao <rabenda.cn@gmail.com>
Date: Wed, 1 Oct 2025 21:48:27 +0800
Subject: [PATCH 2/5] Enable GLES on riscv64.

---
 toolkit/xre/glxtest/glxtest.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolkit/xre/glxtest/glxtest.cpp b/toolkit/xre/glxtest/glxtest.cpp
index 58f5c7e472..4bb3c777a7 100644
--- a/toolkit/xre/glxtest/glxtest.cpp
+++ b/toolkit/xre/glxtest/glxtest.cpp
@@ -437,7 +437,7 @@ static bool get_egl_gl_status(EGLDisplay dpy,
   PFNGLGETSTRING glGetString =
       cast<PFNGLGETSTRING>(eglGetProcAddress("glGetString"));
 
-#if defined(__arm__) || defined(__aarch64__)
+#if defined(__arm__) || defined(__aarch64__) || (defined(__riscv) && __riscv_xlen == 64)
   bool useGles = true;
 #else
   bool useGles = false;
-- 
2.51.0

