From f9bdcff6976b3372a6e2fcbe4a02b8e92d777564 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 5 Apr 2024 22:52:21 +0800
Subject: [PATCH 02/10] feat(netwerk): declare AOSC OS in user agent string

---
 netwerk/protocol/http/nsHttpHandler.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
index dcc5307984..00cc2b6c47 100644
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -748,6 +748,7 @@ void nsHttpHandler::BuildUserAgent() {
   mUserAgent.SetCapacity(mLegacyAppName.Length() + mLegacyAppVersion.Length() +
 #ifndef UA_SPARE_PLATFORM
                          mPlatform.Length() +
+                         mPlatform.Length() + 9 +
 #endif
                          mOscpu.Length() + mMisc.Length() + mProduct.Length() +
                          mProductSub.Length() + mAppName.Length() +
@@ -766,6 +767,7 @@ void nsHttpHandler::BuildUserAgent() {
   if (!mPlatform.IsEmpty()) {
     mUserAgent += mPlatform;
     mUserAgent.AppendLiteral("; ");
+    mUserAgent.AppendLiteral("AOSC OS; ");
   }
 #endif
   if (!mCompatDevice.IsEmpty()) {
-- 
2.44.0

