From 9b36d530d13752ac778f97c873bf92ada73b5407 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Fri, 5 Apr 2024 22:52:21 +0800
Subject: [PATCH 2/7] feat(netwerk): declare AOSC OS in user agent string

---
 netwerk/protocol/http/nsHttpHandler.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/netwerk/protocol/http/nsHttpHandler.cpp b/netwerk/protocol/http/nsHttpHandler.cpp
index 7d4a01bd9f..ed4bfeaa38 100644
--- a/netwerk/protocol/http/nsHttpHandler.cpp
+++ b/netwerk/protocol/http/nsHttpHandler.cpp
@@ -856,6 +856,7 @@ void nsHttpHandler::BuildUserAgent() {
   mUserAgent.SetCapacity(mLegacyAppName.Length() + mLegacyAppVersion.Length() +
 #ifndef UA_SPARE_PLATFORM
                          mPlatform.Length() +
+                         mPlatform.Length() + 9 +
 #endif
                          mOscpu.Length() + mMisc.Length() + mProduct.Length() +
                          mProductSub.Length() + mAppName.Length() +
@@ -874,6 +875,7 @@ void nsHttpHandler::BuildUserAgent() {
   if (!mPlatform.IsEmpty()) {
     mUserAgent += mPlatform;
     mUserAgent.AppendLiteral("; ");
+    mUserAgent.AppendLiteral("AOSC OS; ");
   }
 #endif
   if (!mCompatDevice.IsEmpty()) {
-- 
2.46.0

