From d8f34228e669b644fcea8f2c6de02f21c278ccc0 Mon Sep 17 00:00:00 2001
From: Jiangjin Wang <kaymw@aosc.io>
Date: Tue, 14 Nov 2023 18:16:46 -0800
Subject: [PATCH] Remove architectural limit on VA-API support

---
 toolkit/moz.configure | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 89ad639a2e..12b5d0d6d2 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -537,11 +537,8 @@ set_define("MOZ_WAYLAND", depends_if(wayland_headers)(lambda _: True))
 
 # Hardware-accelerated video decode with VAAPI and V4L2 on Linux
 # ==============================================================
-@depends(target, toolkit_gtk)
-def vaapi(target, toolkit_gtk):
-    # VAAPI is mostly used on x86(-64) but is sometimes used on ARM/ARM64 SOCs.
-    if target.cpu in ("arm", "aarch64", "x86", "x86_64") and toolkit_gtk:
-        return True
+set_config("MOZ_ENABLE_VAAPI", True, when=toolkit_gtk)
+set_define("MOZ_ENABLE_VAAPI", True, when=toolkit_gtk)
 
 
 @depends(target, toolkit_gtk)
@@ -552,9 +549,7 @@ def v4l2(target, toolkit_gtk):
         return True
 
 
-set_config("MOZ_ENABLE_VAAPI", True, when=vaapi)
 set_config("MOZ_ENABLE_V4L2", True, when=v4l2)
-set_define("MOZ_ENABLE_VAAPI", True, when=vaapi)
 set_define("MOZ_ENABLE_V4L2", True, when=v4l2)
 
 
-- 
2.39.1

