From 397d25f77597ecc38883743406821c33af2ee535 Mon Sep 17 00:00:00 2001
From: WANG Xuerui <xen0n@gentoo.org>
Date: Tue, 3 Sep 2024 15:39:18 +0800
Subject: [PATCH 06/11] enable webrtc for loongarch64 in toolkit/moz.configure

Signed-off-by: WANG Xuerui <xen0n@gentoo.org>
---
 toolkit/moz.configure | 1 +
 1 file changed, 1 insertion(+)

diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 011eee093dcf..0559658ab2ba 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -1480,6 +1480,7 @@ def webrtc_default(target):
         "aarch64",
         "x86",
         "ia64",
+        "loongarch64",
         "mips32",
         "mips64",
         "ppc",
-- 
2.47.0

