--- firefox-121.0/third_party/libwebrtc/common_audio/common_audio_c_gn/moz.build	2023-11-27 06:17:35.000000000 -0700
+++ firefox-121.0-new/third_party/libwebrtc/common_audio/common_audio_c_gn/moz.build	2023-11-27 21:27:12.597885957 -0700
@@ -255,6 +255,14 @@
         "/third_party/libwebrtc/common_audio/signal_processing/filter_ar_fast_q12.c"
     ]
 
+if CONFIG["CPU_ARCH"] == "loongarch64":
+
+    UNIFIED_SOURCES += [
+        "/third_party/libwebrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/third_party/libwebrtc/common_audio/signal_processing/complex_fft.c",
+        "/third_party/libwebrtc/common_audio/signal_processing/filter_ar_fast_q12.c"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86":
 
     DEFINES["WEBRTC_ENABLE_AVX2"] = True
--- firefox-121.0/third_party/libwebrtc/common_audio/third_party/spl_sqrt_floor/spl_sqrt_floor_gn/moz.build	2023-11-27 06:17:35.000000000 -0700
+++ firefox-121.0-new/third_party/libwebrtc/common_audio/third_party/spl_sqrt_floor/spl_sqrt_floor_gn/moz.build	2023-11-27 21:21:28.905664825 -0700
@@ -174,6 +174,12 @@
         "/third_party/libwebrtc/common_audio/third_party/spl_sqrt_floor/spl_sqrt_floor.c"
     ]
 
+if CONFIG["CPU_ARCH"] == "loongarch64":
+
+    UNIFIED_SOURCES += [
+        "/third_party/libwebrtc/common_audio/third_party/spl_sqrt_floor/spl_sqrt_floor.c"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86":
 
     DEFINES["WEBRTC_ENABLE_AVX2"] = True
--- firefox-121.0/third_party/libwebrtc/modules/audio_processing/aecm/aecm_core_gn/moz.build	2023-11-27 06:17:35.000000000 -0700
+++ firefox-121.0-new/third_party/libwebrtc/modules/audio_processing/aecm/aecm_core_gn/moz.build	2023-11-27 21:23:38.445132566 -0700
@@ -206,6 +206,12 @@
         "/third_party/libwebrtc/modules/audio_processing/aecm/aecm_core_c.cc"
     ]
 
+if CONFIG["CPU_ARCH"] == "loongarch64":
+
+    SOURCES += [
+        "/third_party/libwebrtc/modules/audio_processing/aecm/aecm_core_c.cc"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86":
 
     DEFINES["WEBRTC_ENABLE_AVX2"] = True
--- firefox-121.0/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn/moz.build	2023-11-27 06:17:35.000000000 -0700
+++ firefox-121.0-new/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn/moz.build	2023-11-27 21:49:10.133863026 -0700
@@ -390,6 +390,34 @@
         "/third_party/libwebrtc/modules/desktop_capture/linux/x11/x_window_property.cc"
     ]
 
+if CONFIG["CPU_ARCH"] == "loongarch64":
+
+    DEFINES["USE_X11"] = "1"
+    DEFINES["WEBRTC_USE_X11"] = True
+
+    OS_LIBS += [
+        "X11",
+        "Xcomposite",
+        "Xdamage",
+        "Xext",
+        "Xfixes",
+        "Xrandr",
+        "Xrender"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/third_party/libwebrtc/modules/desktop_capture/linux/x11/mouse_cursor_monitor_x11.cc",
+        "/third_party/libwebrtc/modules/desktop_capture/linux/x11/screen_capturer_x11.cc",
+        "/third_party/libwebrtc/modules/desktop_capture/linux/x11/shared_x_display.cc",
+        "/third_party/libwebrtc/modules/desktop_capture/linux/x11/window_capturer_x11.cc",
+        "/third_party/libwebrtc/modules/desktop_capture/linux/x11/window_finder_x11.cc",
+        "/third_party/libwebrtc/modules/desktop_capture/linux/x11/window_list_utils.cc",
+        "/third_party/libwebrtc/modules/desktop_capture/linux/x11/x_atom_cache.cc",
+        "/third_party/libwebrtc/modules/desktop_capture/linux/x11/x_error_trap.cc",
+        "/third_party/libwebrtc/modules/desktop_capture/linux/x11/x_server_pixel_buffer.cc",
+        "/third_party/libwebrtc/modules/desktop_capture/linux/x11/x_window_property.cc"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86":
 
     DEFINES["WEBRTC_ENABLE_AVX2"] = True
--- firefox-121.0/third_party/libwebrtc/modules/desktop_capture/primitives_gn/moz.build	2023-11-27 06:17:36.000000000 -0700
+++ firefox-121.0-new/third_party/libwebrtc/modules/desktop_capture/primitives_gn/moz.build	2023-11-27 20:09:21.612523160 -0700
@@ -148,6 +148,10 @@
 
     DEFINES["USE_X11"] = "1"
 
+if CONFIG["CPU_ARCH"] == "loongarch64":
+
+    DEFINES["USE_X11"] = "1"
+
 if CONFIG["CPU_ARCH"] == "x86":
 
     DEFINES["WEBRTC_ENABLE_AVX2"] = True
--- firefox-121.0/third_party/libwebrtc/moz.build	2023-11-27 06:17:34.000000000 -0700
+++ firefox-121.0-new/third_party/libwebrtc/moz.build	2023-11-27 20:08:46.305411224 -0700
@@ -692,3 +692,10 @@
         "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn",
         "/third_party/libwebrtc/modules/desktop_capture/primitives_gn"
     ]
+
+if CONFIG["CPU_ARCH"] == "loongarch64" and CONFIG["MOZ_X11"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    DIRS += [
+        "/third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn",
+        "/third_party/libwebrtc/modules/desktop_capture/primitives_gn"
+    ]
--- firefox-121.0/toolkit/moz.configure	2023-11-27 06:17:38.000000000 -0700
+++ firefox-121.0-new/toolkit/moz.configure	2023-11-27 20:08:30.603361285 -0700
@@ -1331,6 +1331,7 @@
         "ppc",
         "ppc64",
         "riscv64",
+        "loongarch64",
     )
 
     return os_match and cpu_match and target.endianness == "little"
