Origin: https://phabricator.services.mozilla.com/D204011
diff --git a/xpcom/reflect/xptcall/md/unix/xptcinvoke_loongarch64.cpp b/xpcom/reflect/xptcall/md/unix/xptcinvoke_loongarch64.cpp
--- a/xpcom/reflect/xptcall/md/unix/xptcinvoke_loongarch64.cpp
+++ b/xpcom/reflect/xptcall/md/unix/xptcinvoke_loongarch64.cpp
@@ -42,11 +42,13 @@
           break;
         case nsXPTType::T_U16:
           value = s->val.u16;
           break;
         case nsXPTType::T_U32:
-          value = s->val.u32;
+          // 32-bit values need to be sign-extended in 64-bit registers,
+          // so use the signed value here.
+          value = s->val.i32;
           break;
         case nsXPTType::T_U64:
           value = s->val.u64;
           break;
         case nsXPTType::T_BOOL:

