From 0c802396428178e63f001628597847b0ce80216f Mon Sep 17 00:00:00 2001
From: WANG Xuerui <xen0n@gentoo.org>
Date: Wed, 4 Sep 2024 16:45:05 +0800
Subject: [PATCH 08/11] Fix missing loongarch subdir in libpng moz.yaml

Signed-off-by: WANG Xuerui <xen0n@gentoo.org>
---
 media/libpng/moz.yaml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/media/libpng/moz.yaml b/media/libpng/moz.yaml
index 4d193089acd2..e3c63908cc35 100644
--- a/media/libpng/moz.yaml
+++ b/media/libpng/moz.yaml
@@ -37,6 +37,7 @@ vendoring:
     - arm
     - contrib/arm-neon/linux.c
     - intel
+    - loongarch
     - mips
     - powerpc
     - ANNOUNCE
-- 
2.47.0

