diff -Naur qownnotes-19.12.3/libraries/botan/botan.pri qownnotes-19.12.3.botan/libraries/botan/botan.pri
--- qownnotes-19.12.3/libraries/botan/botan.pri	2019-12-07 09:41:15.000000000 +0000
+++ qownnotes-19.12.3.botan/libraries/botan/botan.pri	2019-12-07 23:01:24.737565500 +0000
@@ -5,7 +5,7 @@
 equals(USE_SYSTEM_BOTAN, 1) {
     DEFINES += USE_SYSTEM_BOTAN
     CONFIG += link_pkgconfig
-    PKGCONFIG += botan-2.12
+    PKGCONFIG += botan-2
 } else {
 
 SOURCES += $$PWD/botan.cpp
