From c4b4dcadc6250a02cd003203902ea88beb613dd4 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Tue, 13 Feb 2024 21:01:03 -0800
Subject: [PATCH 1/6] Remove hardcoded CFLAGS

---
 build.gradle          | 10 +++++-----
 buildSrc/linux.gradle |  4 ++++
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/build.gradle b/build.gradle
index df82f630..81cf7be7 100644
--- a/build.gradle
+++ b/build.gradle
@@ -2607,12 +2607,12 @@ project(":web") {
                             cmakeArgs = "$cmakeArgs -DCMAKE_SYSTEM_PROCESSOR=i586"
                         }
                         // TODO: Use cflags and ldflags from all platforms
-                        def cFlags = webkitProperties.ccFlags?.join(' ') ?: ''
-                        def lFlags = webkitProperties.linkFlags?.join(' ') ?: ''
                         // -shared flag should be omitted while creating executable.
-                        def exeFlags = webkitProperties.linkFlags?.join(' ')?.replace('-shared', '') ?: ''
-                        cmakeArgs = "$cmakeArgs -DCMAKE_C_FLAGS='${cFlags}' -DCMAKE_CXX_FLAGS='${cFlags}' -DCMAKE_ASM_FLAGS='${cFlags}'"
-                        cmakeArgs = "$cmakeArgs -DCMAKE_SHARED_LINKER_FLAGS='${lFlags}' -DCMAKE_EXE_LINKER_FLAGS='${exeFlags}'"
+                        def cFlags = System.env.CFLAGS.trim()
+                        def cxxFlags = System.env.CXXFLAGS.trim()
+                        def lFlags = System.env.LDFLAGS.trim()
+                        cmakeArgs = "$cmakeArgs -DCMAKE_C_FLAGS='${cFlags}' -DCMAKE_CXX_FLAGS='${cxxFlags}' -DCMAKE_ASM_FLAGS='${cFlags}'"
+                        cmakeArgs = "$cmakeArgs -DCMAKE_SHARED_LINKER_FLAGS='${lFlags} -shared' -DCMAKE_EXE_LINKER_FLAGS='${lFlags}'"
                     } else if (t.name.startsWith("arm")) {
                         fail("ARM target is not supported as of now.")
                     }
diff --git a/buildSrc/linux.gradle b/buildSrc/linux.gradle
index c47b7b4c..8d58d97f 100644
--- a/buildSrc/linux.gradle
+++ b/buildSrc/linux.gradle
@@ -46,6 +46,10 @@ def commonFlags = [
         "-fstack-protector",
         "-W", "-Wall", "-Wno-unused", "-Wno-parentheses", "-Werror=implicit-function-declaration"] // warning flags
 
+if (System.env.CFLAGS) {
+    commonFlags = System.env.CFLAGS.tokenize(' ')
+}
+
 if (!IS_64) {
     commonFlags += "-m32"
 }
-- 
2.43.0

