From 7e6a75e52eb648f6d6e2ca496c777306c65323c0 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Tue, 13 Feb 2024 21:01:32 -0800
Subject: [PATCH 2/6] Remove arch restriction

---
 build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build.gradle b/build.gradle
index 81cf7be7..302cf112 100644
--- a/build.gradle
+++ b/build.gradle
@@ -713,7 +713,7 @@ if (IS_WINDOWS && OS_ARCH != "x86" && OS_ARCH != "amd64") {
     throw new Exception("Unknown and unsupported build architecture: $OS_ARCH")
 } else if (IS_MAC && OS_ARCH != "x86_64") {
     throw new Exception("Unknown and unsupported build architecture: $OS_ARCH")
-} else if (IS_LINUX && OS_ARCH != "i386" && OS_ARCH != "amd64") {
+} else if (IS_LINUX && false) {
     throw new Exception("Unknown and unsupported build architecture: $OS_ARCH")
 }
 
-- 
2.43.0

