From d3708cad0607ea260da6cac202e95647220ec76c Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Tue, 13 Feb 2024 21:02:15 -0800
Subject: [PATCH 4/6] Fix deprecation error

---
 build.gradle | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build.gradle b/build.gradle
index 302cf112..6921a5da 100644
--- a/build.gradle
+++ b/build.gradle
@@ -929,6 +929,7 @@ void addNative(Project project, String name) {
                 headers = headerDir
                 output(ccOutput)
                 params.addAll(variantProperties.ccFlags)
+                params.add("-Wno-deprecated-declarations")
                 compiler = variantProperties.compiler
                 source(variantProperties.nativeSource)
                 cleanTask.delete ccOutput
-- 
2.43.0

