--- libu2f-emu/meson.build	2023-08-18 21:56:36.130079089 +0800
+++ libu2f-emu.notest/meson.build	2023-08-18 21:56:54.233183406 +0800
@@ -37,15 +37,6 @@
 
 subdir('src')
 
-if not gtest.found()
-  warning('Skip test: gtest not found.')
-else
-  setup_dir = custom_target('setup_dir',
-    output: '.u2f.tmp',
-    command: ['@0@/scripts/setup.sh'.format(meson.source_root()), '@OUTPUT@'])
-  subdir('tests')
-endif
-
 subdir('examples')
 
 doxygen = find_program('doxygen', required : false)
