From 092af847a900afb844ff7cfa8738cb3e4b7849d2 Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sat, 20 Sep 2025 23:09:11 +0800
Subject: [PATCH 2/2] FEDORA Fix configure c99

---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index f6799b4..a77169c 100644
--- a/configure.in
+++ b/configure.in
@@ -85,7 +85,7 @@ AS_VAR_IF(utf8, "yes",[
     LIBS="$LIBICONV $LIBS"
     AC_DEFINE(KAKASI_SUPPORT_UTF8, 1, [KAKASI_SUPPORT_UTF8])
     AC_RUN_IFELSE([AC_LANG_PROGRAM([#include <iconv.h>],
-		    [if (iconv_open("EUC-JP", "UTF-8") == -1) exit(1);])],
+		    [if (iconv_open("EUC-JP", "UTF-8") == -1) return 1;])],
 	[],
 	[AC_MSG_ERROR([can not use EUC-JP or UTF-8 encoding on iconv])])
 ])
-- 
2.51.0

