diff -pruN kakasi-2.3.6.orig/lib/libkakasi.h kakasi-2.3.6/lib/libkakasi.h
--- kakasi-2.3.6.orig/lib/libkakasi.h	2001-08-27 11:11:56.000000000 +0900
+++ kakasi-2.3.6/lib/libkakasi.h	2025-06-03 14:15:40.434590691 +0900
@@ -38,12 +38,17 @@ int kakasi_free PARAMS((char *p));
 /*
  * NOTE: these are internal functions. Do not use.
  */
-void digest_start_copy();
+typedef struct character {
+    char type;
+    unsigned char c1, c2;
+} Character;
+
+void digest_start_copy(Character *, Character *);
 void put_separator();
-void putchars();
-void digest_out();
-int digest();
-void digest_shift();
+void putchars(Character *);
+void digest_out(Character *, int);
+int digest(Character *, int, Character *, int, int, int (*)(Character *, Character *));
+void digest_shift(Character *, int);
 
 void setcharbuffer PARAMS((unsigned char *s));
 char *getpbstr PARAMS((void));
diff -pruN kakasi-2.3.6.orig/src/jj2.c kakasi-2.3.6/src/jj2.c
--- kakasi-2.3.6.orig/src/jj2.c	2001-04-12 14:57:34.000000000 +0900
+++ kakasi-2.3.6/src/jj2.c	2025-06-03 14:15:25.564277469 +0900
@@ -201,7 +201,7 @@ static void
 J2convert(m, n, proc)
      Character *m;
      Character *n;
-     int (*proc)();
+     int (*proc)(Character *, Character *);
 {
     int mp=0, np=0;
     int ret;
diff -pruN kakasi-2.3.6.orig/src/kakasi.c kakasi-2.3.6/src/kakasi.c
--- kakasi-2.3.6.orig/src/kakasi.c	2013-02-06 15:05:02.000000000 +0900
+++ kakasi-2.3.6/src/kakasi.c	2025-06-03 14:15:25.564440801 +0900
@@ -97,7 +97,7 @@ Character right_paren[KAKASIBUF];
 
 #ifdef LIBRARY
 extern FILE *kanwadict;
-static int (*proc[8])()={NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL};
+static int (*proc[8])(Character *, Character *)={NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL};
 /* ASCII, JISROMAN, KATAKANA, GRAPHIC, ZENKAKU-KIGOU, ZENKAKU-KATAKANA, ZENKAKU-HIRAGANA, KANJI, */
 #endif
 
@@ -121,7 +121,7 @@ KAKASI_ATTR void digest_start_copy PARAM
 KAKASI_ATTR void put_separator PARAMS((void));
 KAKASI_ATTR void putchars PARAMS((Character *results));
 KAKASI_ATTR void digest_out PARAMS((Character *c, int ret));
-KAKASI_ATTR int digest PARAMS((Character *c, int clen, Character *r, int rlen, int type, int (*proc)(void)));
+KAKASI_ATTR int digest PARAMS((Character *c, int clen, Character *r, int rlen, int type, int (*proc)(Character *, Character *)));
 KAKASI_ATTR void digest_shift PARAMS((Character *c, int s));
 #ifdef LIBRARY
 static void free_jisyo PARAMS((void));
@@ -240,7 +240,7 @@ digest(c, clen, r, rlen, type, proc)
      Character *r;
      int rlen;
      int type;
-     int (*proc)();
+     int (*proc)(Character *, Character *);
 {
     int ret, i, j, k;
     Character new;
@@ -337,7 +337,7 @@ kakasi_getopt_argv(argc, argv)
 #ifndef LIBRARY
     Character c[KAKASIBUF], r[KAKASIBUF];
     int clen, ptype, pctype;
-    static int (*proc[8])()={NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL};
+    static int (*proc[8])(Character *, Character *)={NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL};
     /* ASCII, JISROMAN, KATAKANA, GRAPHIC, ZENKAKU-KIGOU, ZENKAKU-KATAKANA, ZENKAKU-HIRAGANA, KANJI, */
 #endif
 

