--- gnome-vfs-2.7.90/schemas/desktop_gnome_url_handlers.schemas.in.schema_about	2004-08-04 16:18:21.323425176 +0100
+++ gnome-vfs-2.7.90/schemas/desktop_gnome_url_handlers.schemas.in	2004-08-04 16:25:28.770443352 +0100
@@ -206,6 +206,40 @@
     </schema>
 
     <schema>
+      <key>/schemas/desktop/gnome/url-handlers/about/enabled</key>
+      <applyto>/desktop/gnome/url-handlers/about/enabled</applyto>
+      <owner>gnome</owner>
+      <type>bool</type>
+      <default>true</default>
+      <locale name="C">
+        <short>Handle about URLs</short>
+        <long>Set to true to have a program specified in "command" handle about URLs.</long>
+      </locale>
+    </schema>
+    <schema>
+      <key>/schemas/desktop/gnome/url-handlers/about/command</key>
+      <applyto>/desktop/gnome/url-handlers/about/command</applyto>
+      <owner>gnome</owner>
+      <type>string</type>
+      <default>firefox %s</default>
+      <locale name="C">
+        <short>about URL handler</short>
+        <long></long>
+      </locale>
+    </schema>
+    <schema>
+      <key>/schemas/desktop/gnome/url-handlers/about/needs_terminal</key>
+      <applyto>/desktop/gnome/url-handlers/about/needs_terminal</applyto>
+      <owner>gnome</owner>
+      <type>bool</type>
+      <default>false</default>
+      <locale name="C">
+        <short>Run program in terminal</short>
+        <long>True if the program to handle this URL should be run in a terminal.</long>
+      </locale>
+    </schema>
+
+    <schema>   
       <key>/schemas/desktop/gnome/url-handlers/mailto/enabled</key>
       <applyto>/desktop/gnome/url-handlers/mailto/enabled</applyto>
       <owner>gnome</owner>
