--- gnome-vfs-2.15.91/schemas/desktop_gnome_url_handlers.schemas.in.mailto-command	2006-08-10 16:23:54.000000000 -0400
+++ gnome-vfs-2.15.91/schemas/desktop_gnome_url_handlers.schemas.in	2006-08-10 16:24:14.000000000 -0400
@@ -255,7 +255,7 @@
       <applyto>/desktop/gnome/url-handlers/mailto/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>evolution %s</default>
+      <default>evolution --component=mail %s</default>
       <locale name="C">
         <short>The handler for "mailto" URLs</short>
         <long>The command used to handle "mailto" URLs, if enabled.</long>
