diff -Naur smartmontools-6.4/smartd.service.in smartmontools-6.4.condition/smartd.service.in
--- smartmontools-6.4/smartd.service.in	2014-03-03 11:24:00.000000000 -0600
+++ smartmontools-6.4.condition/smartd.service.in	2017-07-19 08:52:14.395500753 -0500
@@ -2,6 +2,7 @@
 Description=Self Monitoring and Reporting Technology (SMART) Daemon
 Documentation=man:smartd(8) man:smartd.conf(5)
 After=syslog.target
+ConditionVirtualization=no
 
 [Service]
 EnvironmentFile=-/usr/local/etc/sysconfig/smartmontools
