From 07d84267b8d741414601b1be529da5de3afd28af Mon Sep 17 00:00:00 2001
From: Arjen de Korte <build+github@de-korte.org>
Date: Sun, 28 Sep 2025 19:01:19 +0200
Subject: [PATCH 1/7] Boost.System has been header only since Boost 1.69.0

---
 cmake/dependencies.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cmake/dependencies.cmake b/cmake/dependencies.cmake
index 74c3761b..3387e7c0 100644
--- a/cmake/dependencies.cmake
+++ b/cmake/dependencies.cmake
@@ -7,7 +7,6 @@ find_package(Boost COMPONENTS
   filesystem
   iostreams
   regex
-  system
   thread
   REQUIRED
 )

From 8c9dca65bdf8f1bd3a2b0b791887d8b9b7da3fe6 Mon Sep 17 00:00:00 2001
From: Arjen de Korte <build+github@de-korte.org>
Date: Sun, 28 Sep 2025 19:08:36 +0200
Subject: [PATCH 2/7] Boost.System has been header only since Boost 1.69.0

---
 src/core/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 690f3826..657de549 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -57,7 +57,6 @@ target_link_libraries(lucene++
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
 )

From 32039e5f54e7dbc7a2c1e096bdd44c6297262522 Mon Sep 17 00:00:00 2001
From: Arjen de Korte <build+github@de-korte.org>
Date: Sun, 28 Sep 2025 19:13:16 +0200
Subject: [PATCH 3/7] Update CMakeLists.txt

---
 src/contrib/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/contrib/CMakeLists.txt b/src/contrib/CMakeLists.txt
index 7af68045..7252b73a 100644
--- a/src/contrib/CMakeLists.txt
+++ b/src/contrib/CMakeLists.txt
@@ -67,7 +67,6 @@ target_link_libraries(lucene++-contrib
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
     lucene++::lucene++)

From 7425cb10d3aa5415e20b3074743fc450195e243a Mon Sep 17 00:00:00 2001
From: Arjen de Korte <build+github@de-korte.org>
Date: Sun, 28 Sep 2025 19:16:05 +0200
Subject: [PATCH 4/7] Update CMakeLists.txt

---
 src/demo/deletefiles/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/demo/deletefiles/CMakeLists.txt b/src/demo/deletefiles/CMakeLists.txt
index b50d5bee..42131950 100644
--- a/src/demo/deletefiles/CMakeLists.txt
+++ b/src/demo/deletefiles/CMakeLists.txt
@@ -38,7 +38,6 @@ target_link_libraries(deletefiles
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
     lucene++::lucene++

From 2adbf0d482dec43ac9fe73be8f49c8980bc38cb3 Mon Sep 17 00:00:00 2001
From: Arjen de Korte <build+github@de-korte.org>
Date: Sun, 28 Sep 2025 19:16:41 +0200
Subject: [PATCH 5/7] Update CMakeLists.txt

---
 src/demo/indexfiles/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/demo/indexfiles/CMakeLists.txt b/src/demo/indexfiles/CMakeLists.txt
index a1dd582b..68ddced8 100644
--- a/src/demo/indexfiles/CMakeLists.txt
+++ b/src/demo/indexfiles/CMakeLists.txt
@@ -39,7 +39,6 @@ target_link_libraries(indexfiles
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
     lucene++::lucene++

From 6af2547d620a49df603026433a3eb484bada7e60 Mon Sep 17 00:00:00 2001
From: Arjen de Korte <build+github@de-korte.org>
Date: Sun, 28 Sep 2025 19:17:28 +0200
Subject: [PATCH 6/7] Update CMakeLists.txt

---
 src/demo/searchfiles/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/demo/searchfiles/CMakeLists.txt b/src/demo/searchfiles/CMakeLists.txt
index aaf559e4..dda95427 100644
--- a/src/demo/searchfiles/CMakeLists.txt
+++ b/src/demo/searchfiles/CMakeLists.txt
@@ -38,7 +38,6 @@ target_link_libraries(searchfiles
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
     lucene++::lucene++

From 39a3d7c4cd53b1daa3718083527d3c97419c4937 Mon Sep 17 00:00:00 2001
From: Arjen de Korte <build+github@de-korte.org>
Date: Sun, 28 Sep 2025 19:18:17 +0200
Subject: [PATCH 7/7] Update CMakeLists.txt

---
 src/test/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/test/CMakeLists.txt b/src/test/CMakeLists.txt
index f0b9b7e6..69393c00 100644
--- a/src/test/CMakeLists.txt
+++ b/src/test/CMakeLists.txt
@@ -59,7 +59,6 @@ target_link_libraries(lucene++-tester
     Boost::filesystem
     Boost::iostreams
     Boost::regex
-    Boost::system
     Boost::thread
     ZLIB::ZLIB
     gtest_main

