diff -Naur tmux/compat/utf8proc.c tmux.utf8proc/compat/utf8proc.c
--- tmux/compat/utf8proc.c	2022-12-15 20:26:33.613352112 -0800
+++ tmux.utf8proc/compat/utf8proc.c	2022-12-15 20:27:02.114710760 -0800
@@ -16,7 +16,7 @@
 
 #include <sys/types.h>
 
-#include <utf8proc.h>
+#include <libutf8proc/utf8proc.h>
 
 #include "compat.h"
 
diff -Naur tmux/compat.h tmux.utf8proc/compat.h
--- tmux/compat.h	2022-12-15 20:26:33.611352087 -0800
+++ tmux.utf8proc/compat.h	2022-12-15 20:26:56.610641521 -0800
@@ -45,7 +45,7 @@
 #endif
 
 #ifdef HAVE_UTF8PROC
-#include <utf8proc.h>
+#include <libutf8proc/utf8proc.h>
 #endif
 
 #ifndef __GNUC__
diff -Naur tmux/configure.ac tmux.utf8proc/configure.ac
--- tmux/configure.ac	2022-12-15 20:26:33.613352112 -0800
+++ tmux.utf8proc/configure.ac	2022-12-15 20:26:49.754555245 -0800
@@ -373,7 +373,7 @@
 	AS_HELP_STRING(--enable-utf8proc, use utf8proc if it is installed)
 )
 if test "x$enable_utf8proc" = xyes; then
-	AC_CHECK_HEADER(utf8proc.h, enable_utf8proc=yes, enable_utf8proc=no)
+	AC_CHECK_HEADER(libutf8proc/utf8proc.h, enable_utf8proc=yes, enable_utf8proc=no)
 	if test "x$enable_utf8proc" = xyes; then
 		AC_SEARCH_LIBS(
 			utf8proc_charwidth,
