From 679da92c54c75d7fca1fd825f1d42a750d115f2d Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Tue, 23 Apr 2019 10:25:20 -0700
Subject: [PATCH 3/4] tests: Test incomplete resizepart command

---
 tests/t3200-resize-partition.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/t3200-resize-partition.sh b/tests/t3200-resize-partition.sh
index 4565e99..06dbe68 100755
--- a/tests/t3200-resize-partition.sh
+++ b/tests/t3200-resize-partition.sh
@@ -63,6 +63,8 @@ parted -m -s $dev u s p > out 2>&1 || fail=1
 wait_for_dev_to_appear_ ${dev}1 || { warn_ "${dev}1 did not appear"  fail=1; }
 sleep 1
 
+# Running it without end should not core-dump or prompt
+parted -s $dev resizepart 1 > out 2> err || fail=1
 
 # extend the filesystem to end on sector 4096
 new_end=4096s
-- 
2.23.0

