From 74636ce7f34081e92c5680f34588217978306488 Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Thu, 19 Aug 2021 13:56:31 -0700
Subject: [PATCH 32/34] libparted: Check devpath before passing to strlen

---
 libparted/arch/linux.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libparted/arch/linux.c b/libparted/arch/linux.c
index 758d36a..430d02e 100644
--- a/libparted/arch/linux.c
+++ b/libparted/arch/linux.c
@@ -2453,6 +2453,9 @@ _device_get_part_path (PedDevice const *dev, int num)
 #else
         devpath = dev->path;
 #endif
+        if (!devpath)
+            return NULL;
+
         path_len = strlen (devpath);
         /* Check for devfs-style /disc => /partN transformation
            unconditionally; the system might be using udev with devfs rules,
-- 
2.31.1

