From e0a83f3bdf6cfb4b2b6d3631a5ab157291bfbdfa Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Sun, 29 Nov 2020 23:20:17 +0100
Subject: [PATCH 36/38] hurd: Support rumpdisk-based device names

Signed-off-by: Brian C. Lane <bcl@redhat.com>
---
 libparted/arch/gnu.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/libparted/arch/gnu.c b/libparted/arch/gnu.c
index 507a5ff..0797115 100644
--- a/libparted/arch/gnu.c
+++ b/libparted/arch/gnu.c
@@ -786,6 +786,15 @@ probe_standard_devices ()
 	_ped_device_probe ("/dev/hd6");
 	_ped_device_probe ("/dev/hd7");
 
+	_ped_device_probe ("/dev/wd0");
+	_ped_device_probe ("/dev/wd1");
+	_ped_device_probe ("/dev/wd2");
+	_ped_device_probe ("/dev/wd3");
+	_ped_device_probe ("/dev/wd4");
+	_ped_device_probe ("/dev/wd5");
+	_ped_device_probe ("/dev/wd6");
+	_ped_device_probe ("/dev/wd7");
+
 	return 1;
 }
 
-- 
2.31.1

