From a1f8bcde22bbd97ef7abae3c83ede77fc25a301c Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Thu, 30 Sep 2021 13:16:12 -0700
Subject: [PATCH 43/43] gpt: Revert to filesystem GUID when setting flag to off

---
 libparted/labels/gpt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libparted/labels/gpt.c b/libparted/labels/gpt.c
index 8b345d5..c5d7bb3 100644
--- a/libparted/labels/gpt.c
+++ b/libparted/labels/gpt.c
@@ -1565,7 +1565,7 @@ gpt_partition_set_flag (PedPartition *part, PedPartitionFlag flag, int state)
     if (state)
       gpt_part_data->type = p->type_uuid;
     else if (guid_cmp (gpt_part_data->type, p->type_uuid) == 0)
-      gpt_part_data->type = PARTITION_LINUX_DATA_GUID;
+      return gpt_partition_set_system (part, part->fs_type);
     return 1;
   }
 
-- 
2.31.1

