From d7bcb49f1313da7deae0ca41ab402980fb3094a4 Mon Sep 17 00:00:00 2001
From: Mike Fleetwood <mike.fleetwood@googlemail.com>
Date: Sat, 1 Oct 2016 16:40:18 +0100
Subject: [PATCH 47/53] tests: t3310-flags.sh: Add test for bsd table flags

Signed-off-by: Brian C. Lane <bcl@redhat.com>
---
 tests/t3310-flags.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/t3310-flags.sh b/tests/t3310-flags.sh
index 5cbf2da..3f80213 100644
--- a/tests/t3310-flags.sh
+++ b/tests/t3310-flags.sh
@@ -25,9 +25,11 @@ extract_flags()
   perl -nle '/^1:2048s:4095s:2048s::(?:PTNNAME)?:(.+);$/ and print $1' "$@"
 }
 
-for table_type in msdos gpt; do
+for table_type in bsd gpt msdos; do
 
   case $table_type in
+    bsd)   primary_or_name=''
+           ;;
     gpt)   primary_or_name='PTNNAME'
            ;;
     msdos) primary_or_name='primary'
-- 
2.7.4

