From c625f6769144472f3c099cdd512318cea18e2766 Mon Sep 17 00:00:00 2001
From: Sebastian Rasmussen <sebras@gmail.com>
Date: Thu, 19 Jan 2017 23:38:40 +0100
Subject: [PATCH 68/75] libparted: Fix typo in hfs error message

---
 libparted/fs/r/hfs/cache.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libparted/fs/r/hfs/cache.c b/libparted/fs/r/hfs/cache.c
index 68be366..03c2913 100644
--- a/libparted/fs/r/hfs/cache.c
+++ b/libparted/fs/r/hfs/cache.c
@@ -210,8 +210,8 @@ hfsc_cache_move_extent(HfsCPrivateCache* cache, uint32_t old_start,
 		ped_exception_throw (
 			PED_EXCEPTION_BUG,
 			PED_EXCEPTION_CANCEL,
-			_("Trying to move an extent from block Ox%X to block "
-			  "Ox%X, but another one already exists at this "
+			_("Trying to move an extent from block 0x%X to block "
+			  "0x%X, but another one already exists at this "
 			  "position.  This should not happen!"),
 			old_start, new_start);
 		return NULL;
-- 
2.9.3

