diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 248054301..00d7b6a57 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -329,13 +329,13 @@ target_link_libraries(migraphx PRIVATE nlohmann_json::nlohmann_json)
 find_package(SQLite3 REQUIRED)
 target_link_libraries(migraphx PRIVATE SQLite::SQLite3)

-find_package(msgpackc-cxx QUIET)
-if(NOT msgpackc-cxx_FOUND)
+find_package(msgpack-cxx QUIET)
+if(NOT msgpack-cxx_FOUND)
     find_package(msgpack REQUIRED)
 endif()
-target_link_libraries(migraphx PRIVATE msgpackc-cxx)
+target_link_libraries(migraphx PRIVATE msgpack-cxx)
 # Make this available to the tests
-target_link_libraries(migraphx INTERFACE $<BUILD_INTERFACE:msgpackc-cxx>)
+target_link_libraries(migraphx INTERFACE $<BUILD_INTERFACE:msgpack-cxx>)

 add_library(migraphx_all_targets INTERFACE)

