From eecb4690f18650b9139071301db63a24e6da0e39 Mon Sep 17 00:00:00 2001
From: Slávek Banko <slavek.banko@axis.cz>
Date: Sun, 14 May 2017 10:34:25 +0000
Subject: Fix crash on NULL pointer dereference

This resolves Bug 2773

Signed-off-by: Slávek Banko <slavek.banko@axis.cz>
---
diff --git a/korganizer/koeditorfreebusy.cpp b/korganizer/koeditorfreebusy.cpp
index 0ee9d4f..88c5414 100644
--- a/korganizer/koeditorfreebusy.cpp
+++ b/korganizer/koeditorfreebusy.cpp
@@ -850,10 +850,13 @@ void KOEditorFreeBusy::removeAttendee()
 void KOEditorFreeBusy::clearSelection() const
 {
   KDGanttViewItem *item = mGanttView->selectedItem();
-  if ( item )
+  if (item) {
     mGanttView->setSelected( item, false );
+  }
   mGanttView->repaint();
-  item->repaint();
+  if (item) {
+    item->repaint();
+  }
 }
 
 void KOEditorFreeBusy::setSelected( int index )
--
cgit v0.9.0.2-52-g1de3
