diff -Naur tdepim-trinity-14.0.7/libkholidays/scanholiday.c tdepim-trinity-14.0.7.gcc10/libkholidays/scanholiday.c
--- tdepim-trinity-14.0.7/libkholidays/scanholiday.c	2018-05-20 11:44:44.000000000 -0700
+++ tdepim-trinity-14.0.7.gcc10/libkholidays/scanholiday.c	2020-12-02 23:47:19.540129682 -0800
@@ -292,7 +292,7 @@
 #define yywrap() 1
 #define YY_SKIP_YYWRAP
 typedef unsigned char YY_CHAR;
-FILE *yyin = (FILE *) 0, *yyout = (FILE *) 0;
+FILE *yyout = (FILE *) NULL;
 typedef int yy_state_type;
 extern char *yytext;
 #define yytext_ptr yytext
