From 4767f333ca8a7997b4c4f9ec3f2f4fdb6e59d7ea Mon Sep 17 00:00:00 2001
From: François Andriot <francois.andriot@free.fr>
Date: Sun, 23 Jul 2017 17:52:10 +0000
Subject: Fix FTBFS with GCC7

Signed-off-by: François Andriot <francois.andriot@free.fr>
---
diff --git a/korganizer/koeditorattachments.cpp b/korganizer/koeditorattachments.cpp
index 4560d34..48b9acd 100644
--- a/korganizer/koeditorattachments.cpp
+++ b/korganizer/koeditorattachments.cpp
@@ -79,7 +79,7 @@ class AttachmentListItem : public TDEIconViewItem
       if ( att ) {
         mAttachment = new KCal::Attachment( *att );
       } else {
-        mAttachment = new KCal::Attachment( '\0' ); //use the non-uri constructor
+        mAttachment = new KCal::Attachment( TQChar('\0') ); //use the non-uri constructor
                                                     //as we want inline by default
       }
       readAttachment();
--
cgit v0.9.0.2-52-g1de3
