From b4abbc04ec3d3f91c317508c8e14afffc4aa0490 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 19 Aug 2024 15:21:20 +0800
Subject: [PATCH 4/4] Reset libtsm_deps require >= 4.0.0

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 650276e..ea5d7a7 100644
--- a/meson.build
+++ b/meson.build
@@ -45,7 +45,7 @@ systemddir = get_option('libdir') / 'systemd'
 fs = import('fs')
 
 xkbcommon_deps = dependency('xkbcommon', version: '>=0.5.0')
-libtsm_deps = dependency('libtsm', version: '>=4.1.0')
+libtsm_deps = dependency('libtsm', version: '>=4.0.0')
 libudev_deps = dependency('libudev', version: '>=172')
 libdrm_deps = dependency('libdrm')
 dl_deps = dependency('dl')
-- 
2.46.0

