From bf01b3a1b929306a6ab03adc8b733cbb81d480d7 Mon Sep 17 00:00:00 2001
From: Arseny Maslennikov <ar@cs.msu.ru>
Date: Sat, 4 Mar 2023 20:19:45 +0300
Subject: [PATCH 05/12] UPSTREAM: units: Order ourselves after plymouth

---
 docs/kmscon.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/docs/kmscon.service.in b/docs/kmscon.service.in
index 2ceb9e2..0fb352e 100644
--- a/docs/kmscon.service.in
+++ b/docs/kmscon.service.in
@@ -1,6 +1,7 @@
 [Unit]
 Description=KMS System Console
 Documentation=man:kmscon(1)
+After=plymouth-quit-wait.service
 After=systemd-user-sessions.service
 After=rc-local.service
 
-- 
2.47.0

