From f700cd6946ea5396e33c189f76d953e824bd7e0f Mon Sep 17 00:00:00 2001
From: abc1763613206 <abc1763613206@163.com>
Date: Thu, 14 May 2026 09:11:02 +0800
Subject: [PATCH 2/2] Add comments about compression in nginx.conf

---
 conf/nginx.conf | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/conf/nginx.conf b/conf/nginx.conf
index 2f8ed71be..5721bbced 100644
--- a/conf/nginx.conf
+++ b/conf/nginx.conf
@@ -30,7 +30,10 @@ http {
     #keepalive_timeout  0;
     keepalive_timeout  65;
 
+    # To enable gzip compression, uncomment the following line.
     #gzip  on;
+    # To enable brotli compression, uncomment the following line.
+    #brotli on;
 
     server {
         listen       80;
-- 
2.52.0

