diff --git a/conf/nginx.conf b/conf/nginx.conf
index 29bc085f2..9432b06a7 100644
--- a/conf/nginx.conf
+++ b/conf/nginx.conf
@@ -29,8 +29,10 @@ http {
 
     #keepalive_timeout  0;
     keepalive_timeout  65;
-
+    # To enable gzip compression, uncomment the following line.
     #gzip  on;
+    # To enable brotli compression, uncomment the following line.
+    #brotli on;
 
     server {
         listen       80;
