From ce7512c8515d4bd8458a92702fe495942161be60 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Sat, 6 Jan 2024 22:53:21 -0800
Subject: [PATCH 1/2] nvm.sh: drop hash -r

---
 nvm.sh | 2 --
 1 file changed, 2 deletions(-)

diff --git a/nvm.sh b/nvm.sh
index 1554b5b..9fe3f96 100755
--- a/nvm.sh
+++ b/nvm.sh
@@ -3953,7 +3953,6 @@ nvm() {
         fi
       else
         export PATH="${NEWPATH}"
-        \hash -r
         if [ "${NVM_SILENT:-0}" -ne 1 ]; then
           nvm_echo "${NVM_DIR}/*/bin removed from \${PATH}"
         fi
@@ -4096,7 +4095,6 @@ nvm() {
         export MANPATH
       fi
       export PATH
-      \hash -r
       export NVM_BIN="${NVM_VERSION_DIR}/bin"
       export NVM_INC="${NVM_VERSION_DIR}/include/node"
       if [ "${NVM_SYMLINK_CURRENT-}" = true ]; then
-- 
2.52.0

