From a7185ed2ae440717e29cde5d923b62d621be03f9 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Sat, 6 Jan 2024 22:53:21 -0800
Subject: [PATCH] nvm.sh: drop hash -r

---
 nvm.sh | 2 --
 1 file changed, 2 deletions(-)

diff --git a/nvm.sh b/nvm.sh
index 2e4378f..cb4dda5 100644
--- a/nvm.sh
+++ b/nvm.sh
@@ -3565,7 +3565,6 @@ nvm() {
         fi
       else
         export PATH="${NEWPATH}"
-        \hash -r
         if [ "${NVM_SILENT:-0}" -ne 1 ]; then
           nvm_echo "${NVM_DIR}/*/bin removed from \${PATH}"
         fi
@@ -3697,7 +3696,6 @@ nvm() {
         export MANPATH
       fi
       export PATH
-      \hash -r
       export NVM_BIN="${NVM_VERSION_DIR}/bin"
       export NVM_INC="${NVM_VERSION_DIR}/include/node"
       if [ "${NVM_SYMLINK_CURRENT-}" = true ]; then
-- 
2.39.1

