From 2bfd55e852ebfaed9a06caa62482d6d43e1f42a5 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Sat, 6 Jan 2024 22:53:21 -0800
Subject: [PATCH] nvm.sh: drop hash -r

---
 nvm.sh | 2 --
 1 file changed, 2 deletions(-)

diff --git a/nvm.sh b/nvm.sh
index d943896..89d69d9 100755
--- a/nvm.sh
+++ b/nvm.sh
@@ -3761,7 +3761,6 @@ nvm() {
         fi
       else
         export PATH="${NEWPATH}"
-        \hash -r
         if [ "${NVM_SILENT:-0}" -ne 1 ]; then
           nvm_echo "${NVM_DIR}/*/bin removed from \${PATH}"
         fi
@@ -3906,7 +3905,6 @@ nvm() {
         export MANPATH
       fi
       export PATH
-      \hash -r
       export NVM_BIN="${NVM_VERSION_DIR}/bin"
       export NVM_INC="${NVM_VERSION_DIR}/include/node"
       if [ "${NVM_SYMLINK_CURRENT-}" = true ]; then
-- 
2.49.0

