From 86c8c92a846e565502885739eac44399d41e1d1e Mon Sep 17 00:00:00 2001
From: RawDiamondMC <RawDiamondMC@aosc.io>
Date: Sun, 14 Jun 2026 16:42:48 +0800
Subject: [PATCH 2/2] nvm.sh: always use /root/.nvm when NVM_DIR is not set

---
 nvm.sh | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/nvm.sh b/nvm.sh
index 9fe3f96..bcdd114 100755
--- a/nvm.sh
+++ b/nvm.sh
@@ -461,13 +461,7 @@ fi
 
 # Auto detect the NVM_DIR when not set
 if [ -z "${NVM_DIR-}" ]; then
-  # shellcheck disable=SC2128
-  if [ -n "${BASH_SOURCE-}" ]; then
-    NVM_SCRIPT_SOURCE="${BASH_SOURCE}"
-  fi
-  # shellcheck disable=SC2086
-  NVM_DIR="$(nvm_cd ${NVM_CD_FLAGS} "$(dirname "${NVM_SCRIPT_SOURCE:-$0}")" >/dev/null && \pwd)"
-  export NVM_DIR
+  export NVM_DIR="${HOME}/.nvm"
 else
   # https://unix.stackexchange.com/a/198289
   case $NVM_DIR in
-- 
2.52.0

