From a74001dda8ae5b50ebe0c57a13cf053ddeefbc6d Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <origincode@aosc.io>
Date: Thu, 7 May 2026 11:31:58 +0800
Subject: [PATCH] Adapt to AOSC OS binutils binary naming scheme

Signed-off-by: Kaiyang Wu <origincode@aosc.io>
---
 Makefile  | 22 +++++++++++-----------
 configure | 16 ++++++++--------
 2 files changed, 19 insertions(+), 19 deletions(-)

diff --git a/Makefile b/Makefile
index eddeddc..5f1992c 100644
--- a/Makefile
+++ b/Makefile
@@ -18,11 +18,11 @@ ARCHS=amd64 x32 i386 \
 	loong64 \
 	arc \
 
-X86=x86_64-linux-gnu
-MIPS=mips-linux-gnu
-POWERPC=powerpc-linux-gnu
-ARM=arm-linux-gnueabihf
-SPARC=sparc64-linux-gnu
+X86=x86_64-aosc-linux-gnu
+MIPS=mips-aosc-linux-gnu
+POWERPC=powerpc-aosc-linux-gnu
+ARM=arm-aosc-linux-gnueabihf
+SPARC=sparc64-aosc-linux-gnu
 -include config
 all: $(ARCHS:%=arch-test-%)
 
@@ -116,8 +116,8 @@ arch-test-ppc64: ppc64.s
 	$(POWERPC)-ld -z noexecstack -melf64ppc -s ppc64.o -o $@
 
 arch-test-ppc64el: ppc64el.s
-	powerpc64le-linux-gnu-as -mpower8 $^ -o ppc64el.o
-	powerpc64le-linux-gnu-ld -z noexecstack -s ppc64el.o -o $@
+	powerpc64le-aosc-linux-gnu-as -mpower8 $^ -o ppc64el.o
+	powerpc64le-aosc-linux-gnu-ld -z noexecstack -s ppc64el.o -o $@
 
 arch-test-powerpcspe: powerpcspe.s
 	$(POWERPC)-as -a32 -me500 $^ -o powerpcspe.o
@@ -128,8 +128,8 @@ arch-test-s390x: s390x.s
 	s390x-linux-gnu-ld -z noexecstack -s s390x.o -o $@
 
 arch-test-arm64: arm64.s
-	aarch64-linux-gnu-as $^ -o arm64.o
-	aarch64-linux-gnu-ld -z noexecstack -s arm64.o -o $@
+	aarch64-aosc-linux-gnu-as $^ -o arm64.o
+	aarch64-aosc-linux-gnu-ld -z noexecstack -s arm64.o -o $@
 
 arch-test-arm64ilp32: arm64.s
 	aarch64-linux-gnu-as -mabi=ilp32 $^ -o arm64ilp32.o
@@ -180,8 +180,8 @@ arch-test-riscv64: riscv64.s
 	$(RISCV)-ld -z noexecstack -m elf64lriscv -s riscv64.o -o $@
 
 arch-test-loong64: loong64.s
-	loongarch64-linux-gnu-as $^ -o loong64.o
-	loongarch64-linux-gnu-ld -z noexecstack -s loong64.o -o $@
+	loongarch64-aosc-linux-gnu-as $^ -o loong64.o
+	loongarch64-aosc-linux-gnu-ld -z noexecstack -s loong64.o -o $@
 
 arch-test-arc: arc.s
 	arc-linux-gnu-as $^ -o arc.o
diff --git a/configure b/configure
index bc05051..17abf2e 100755
--- a/configure
+++ b/configure
@@ -30,14 +30,14 @@ sub alt
     }
 }
 
-alt qw(X86 x86_64-linux-gnu i386-linux-gnu i486-linux-gnu i586-linux-gnu
-    i686-linux-gnu x86_64-linux-gnux32);
-alt qw(MIPS mips-linux-gnu mipsel-linux-gnu mips64-linux-gnuabi64
-    mips64el-linux-gnuabi64);
-alt qw(POWERPC powerpc-linux-gnu powerpc64-linux-gnu);
-alt qw(ARM arm-linux-gnueabihf arm-linux-gnueabi);
-alt qw(SPARC sparc64-linux-gnu sparc-linux-gnu);
-alt qw(RISCV riscv64-linux-gnu riscv32-linux-gnu);
+alt qw(X86 x86_64-aosc-linux-gnu i386-aosc-linux-gnu i486-aosc-linux-gnu i586-aosc-linux-gnu
+    i686-aosc-linux-gnu x86_64-aosc-linux-gnux32);
+alt qw(MIPS mips-aosc-linux-gnu mipsel-aosc-linux-gnu mips64-aosc-linux-gnuabi64
+    mips64el-aosc-linux-gnuabi64);
+alt qw(POWERPC powerpc-aosc-linux-gnu powerpc64-aosc-linux-gnu);
+alt qw(ARM arm-aosc-linux-gnueabihf arm-aosc-linux-gnueabi);
+alt qw(SPARC sparc64-aosc-linux-gnu sparc-aosc-linux-gnu);
+alt qw(RISCV riscv64-aosc-linux-gnu riscv32-aosc-linux-gnu);
 
 while (/^arch-test-([a-z0-9-]+):.*\n\t([a-z0-9_.-]+-(?:gcc|as))/mg)
 {
-- 
2.52.0

