Description: avoid a FTBFS
Author: Paul Gevers <elbrus@debian.org>
Forwarded: not-yet

Index: lazarus/docs/html/build_lcl_docs.lpr
===================================================================
--- lazarus.orig/docs/html/build_lcl_docs.lpr
+++ lazarus/docs/html/build_lcl_docs.lpr
@@ -9,7 +9,7 @@ program update_lcl_docs;
 {$ENDIF}
 
 uses
-  Classes, Sysutils, GetOpts, LazFileUtils, FileUtil, UTF8Process, LazUtilities,
+  Classes, Sysutils, GetOpts, LazFileUtils, LazStringUtils, FileUtil, UTF8Process, LazUtilities,
   Process;
 
 var
