Index: lazarus/Makefile.fpc
===================================================================
--- lazarus/Makefile.fpc	(version 0.9.26)
+++ lazarus/Makefile.fpc	(working copy)
@@ -20,7 +20,7 @@
 
 [install]
 fpcpackage=n
-basedir=share/lazarus
+basedir=$(_LIB)/lazarus
 
 [dist]
 destdir=$(BASEDIR)/dist
@@ -28,7 +28,7 @@
 [prerules]
 RCPP?=$(strip $(firstword cpp$(SRCEXEEXT)))
 #
-LAZARUS_INSTALL_DIR=$(INSTALL_PREFIX)/share/lazarus
+LAZARUS_INSTALL_DIR=$(INSTALL_PREFIX)/$(_LIB)/lazarus
 INSTALL_MAN_DIR=$(INSTALL_PREFIX)/share/man
 ifneq ($(findstring $(OS_TARGET),win32 win64),)
 LAZARUS_INSTALL_DIR=C:\lazarus
@@ -256,8 +256,9 @@
 ifneq ($(findstring $(OS_TARGET),win32 win64),)
 
 else
+	$(MKDIR) $(INSTALL_PREFIX)/$(_LIB)
+	$(MKDIR) $(INSTALL_PREFIX)/$(_LIB)/lazarus
 	$(MKDIR) $(INSTALL_PREFIX)/share
-	$(MKDIR) $(INSTALL_PREFIX)/share/lazarus
 	$(MKDIR) $(INSTALL_PREFIX)/share/applications
 	$(MKDIR) $(INSTALL_PREFIX)/share/pixmaps
 	$(MKDIR) $(INSTALL_PREFIX)/share/mime/packages

