Description: Coupleof linking fixes for ppc64el port
Author: Mattias Klose <doko@debian.org>
Bug-Debian: https://bugs.debian.org/757134


--- /dev/null
+++ gtkmathview-0.8.0/acinclude.m4
@@ -0,0 +1 @@
+m4_pattern_allow(AM_BINRELOC)
--- gtkmathview-0.8.0.orig/src/Makefile.am
+++ gtkmathview-0.8.0/src/Makefile.am
@@ -32,5 +32,5 @@ libmathview_la_LIBADD = \
   $(MAYBE_BOXML_LIBADD) \
   $(top_builddir)/src/engine/common/libengine_common.la \
   $(top_builddir)/src/common/libcommon.la \
-  -lstdc++ \
+  -lstdc++ -lm\
   $(NULL)
