From d88574e42cda89caac6178f6701b3c10315a176a Mon Sep 17 00:00:00 2001
From: Jakub Schmidtke <sjakub@gmail.com>
Date: Tue, 6 Aug 2019 17:58:22 +0200
Subject: [PATCH] Fixed a GCC 9.1.0 build error

---
 src/dns.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/dns.h b/src/dns.h
index 78d6402..9f62ed3 100644
--- a/src/dns.h
+++ b/src/dns.h
@@ -161,9 +161,14 @@ DNS_PUBLIC int *dns_debug_p(void);
 #define DNS_PRAGMA_QUIET _Pragma("GCC diagnostic ignored \"-Woverride-init\"")
 #define DNS_PRAGMA_POP _Pragma("GCC diagnostic pop")
 
+#if __GNUC__ < 9
 /* GCC parses the _Pragma operator less elegantly than clang. */
 #define dns_quietinit(...) \
 	__extension__ ({ DNS_PRAGMA_PUSH DNS_PRAGMA_QUIET __VA_ARGS__; DNS_PRAGMA_POP })
+#else
+#define dns_quietinit(...) __VA_ARGS__
+#endif
+
 #else
 #define DNS_PRAGMA_PUSH
 #define DNS_PRAGMA_QUIET
