From 21d4d2c98bb24d06eb559c4c4aeaa9991969d025 Mon Sep 17 00:00:00 2001
From: Qing Yun <kf@aosc.io>
Date: Tue, 16 Jun 2026 21:34:02 +0800
Subject: [PATCH 8/9] AOSCOS: fix: adapt to drm_fb_helper_lastclose drop <=
 6.12

---
 linux/zx_pcie.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/linux/zx_pcie.c b/linux/zx_pcie.c
index 8e47a77..a63e5d6 100644
--- a/linux/zx_pcie.c
+++ b/linux/zx_pcie.c
@@ -512,6 +512,7 @@ static void zx_drm_postclose(struct drm_device *dev, struct drm_file *file)
 #endif
 }
 
+#if DRM_VERSION_CODE < KERNEL_VERSION(6, 12, 0)
 void  zx_drm_last_close(struct drm_device* dev)
 {
 #if DRM_VERSION_CODE < KERNEL_VERSION(4, 19, 0)
@@ -521,6 +522,7 @@ void  zx_drm_last_close(struct drm_device* dev)
     drm_fb_helper_lastclose(dev);
 #endif
 }
+#endif
 
 #if DRM_VERSION_CODE < KERNEL_VERSION(4,11,0)
 static int zx_drm_device_is_agp(struct drm_device * dev)
-- 
2.52.0

