From 9a22cfa065a11f663594de8815e9bb596370e360 Mon Sep 17 00:00:00 2001
From: Qing Yun <kf@aosc.io>
Date: Tue, 16 Jun 2026 21:36:11 +0800
Subject: [PATCH 9/9] AOSCOS: fix: adapt to -Werror=empty-body

---
 core/vidsch/vidsch_vm_update.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/vidsch/vidsch_vm_update.c b/core/vidsch/vidsch_vm_update.c
index d399858..988b348 100644
--- a/core/vidsch/vidsch_vm_update.c
+++ b/core/vidsch/vidsch_vm_update.c
@@ -200,9 +200,13 @@ int vidsch_wait_prepare_event(vidsch_mgr_t *sch_mgr, task_desc_t *task_desc, uns
 
             /* use wait_task_type to distinguish between vm_prepare_task and page_table_set_task */
             if (wait_task_type == task_type_vm_prepare)
+            {
                 zx_debug("sch_mgr->engine_index %d is waitting for vm_prepare_task more than %d x 100 ms\n", sch_mgr->engine_index, timeout_cnt);
+            }
             else if (wait_task_type == task_type_page_table_set)
+            {
                 zx_debug("sch_mgr->engine_index %d is waitting for page_table_set_task more than %d x 100 ms\n", sch_mgr->engine_index, timeout_cnt);
+            }
         }
 
     }while(force_wait && e_status != ZX_EVENT_BACK);
-- 
2.52.0

