diff -Naur mupen64plus-bundle-src-2.5.9/source/mupen64plus-core/src/asm_defines/asm_defines_gas.h mupen64plus-bundle-src-2.5.9-new/source/mupen64plus-core/src/asm_defines/asm_defines_gas.h
--- mupen64plus-bundle-src-2.5.9/source/mupen64plus-core/src/asm_defines/asm_defines_gas.h	1969-12-31 16:00:00.000000000 -0800
+++ mupen64plus-bundle-src-2.5.9-new/source/mupen64plus-core/src/asm_defines/asm_defines_gas.h	2021-02-01 03:12:07.166841398 -0800
@@ -0,0 +1,37 @@
+#define offsetof_struct_new_dynarec_hot_state_memory_map (0x000009e8)
+#define offsetof_struct_new_dynarec_hot_state_restore_candidate (0x000007e8)
+#define offsetof_struct_new_dynarec_hot_state_mini_ht (0x000005e8)
+#define offsetof_struct_new_dynarec_hot_state_rd (0x000005d8)
+#define offsetof_struct_new_dynarec_hot_state_rt (0x000005d0)
+#define offsetof_struct_new_dynarec_hot_state_rs (0x000005c8)
+#define offsetof_struct_new_dynarec_hot_state_fake_pc (0x000004f8)
+#define offsetof_struct_new_dynarec_hot_state_pc (0x000004f0)
+#define offsetof_struct_new_dynarec_hot_state_branch_target (0x000004e8)
+#define offsetof_struct_new_dynarec_hot_state_rounding_modes (0x000004d8)
+#define offsetof_struct_new_dynarec_hot_state_cp1_regs_double (0x000003d8)
+#define offsetof_struct_new_dynarec_hot_state_cp1_regs_simple (0x000002d8)
+#define offsetof_struct_new_dynarec_hot_state_cp0_regs (0x00000258)
+#define offsetof_struct_new_dynarec_hot_state_lo (0x00000250)
+#define offsetof_struct_new_dynarec_hot_state_hi (0x00000248)
+#define offsetof_struct_new_dynarec_hot_state_regs (0x00000148)
+#define offsetof_struct_new_dynarec_hot_state_fcr31 (0x00000140)
+#define offsetof_struct_new_dynarec_hot_state_fcr0 (0x0000013c)
+#define offsetof_struct_new_dynarec_hot_state_invc_ptr (0x00000118)
+#define offsetof_struct_new_dynarec_hot_state_stop (0x00000110)
+#define offsetof_struct_new_dynarec_hot_state_pcaddr (0x0000010c)
+#define offsetof_struct_new_dynarec_hot_state_pending_exception (0x00000108)
+#define offsetof_struct_new_dynarec_hot_state_cycle_count (0x00000104)
+#define offsetof_struct_new_dynarec_hot_state_next_interrupt (0x00000100)
+#define offsetof_struct_new_dynarec_hot_state_dynarec_local (0x00000000)
+#define offsetof_struct_r4300_core_extra_memory (0x00901000)
+#define offsetof_struct_r4300_core_new_dynarec_hot_state (0x02901000)
+#define offsetof_struct_cached_interp_invalid_code (0x00000000)
+#define offsetof_struct_r4300_core_cached_interp (0x000000f0)
+#define offsetof_struct_tlb_LUT_w (0x00400680)
+#define offsetof_struct_tlb_LUT_r (0x00000680)
+#define offsetof_struct_tlb_entries (0x00000000)
+#define offsetof_struct_cp0_tlb (0x0000026c)
+#define offsetof_struct_cp0_count_per_op (0x00000268)
+#define offsetof_struct_cp0_last_addr (0x00000264)
+#define offsetof_struct_r4300_core_cp0 (0x031019f0)
+#define offsetof_struct_device_r4300 (0x00000000)
diff -Naur mupen64plus-bundle-src-2.5.9/source/mupen64plus-core/src/asm_defines/asm_defines_nasm.h mupen64plus-bundle-src-2.5.9-new/source/mupen64plus-core/src/asm_defines/asm_defines_nasm.h
--- mupen64plus-bundle-src-2.5.9/source/mupen64plus-core/src/asm_defines/asm_defines_nasm.h	1969-12-31 16:00:00.000000000 -0800
+++ mupen64plus-bundle-src-2.5.9-new/source/mupen64plus-core/src/asm_defines/asm_defines_nasm.h	2021-02-01 03:12:07.166841398 -0800
@@ -0,0 +1,37 @@
+%define offsetof_struct_new_dynarec_hot_state_memory_map (0x000009e8)
+%define offsetof_struct_new_dynarec_hot_state_restore_candidate (0x000007e8)
+%define offsetof_struct_new_dynarec_hot_state_mini_ht (0x000005e8)
+%define offsetof_struct_new_dynarec_hot_state_rd (0x000005d8)
+%define offsetof_struct_new_dynarec_hot_state_rt (0x000005d0)
+%define offsetof_struct_new_dynarec_hot_state_rs (0x000005c8)
+%define offsetof_struct_new_dynarec_hot_state_fake_pc (0x000004f8)
+%define offsetof_struct_new_dynarec_hot_state_pc (0x000004f0)
+%define offsetof_struct_new_dynarec_hot_state_branch_target (0x000004e8)
+%define offsetof_struct_new_dynarec_hot_state_rounding_modes (0x000004d8)
+%define offsetof_struct_new_dynarec_hot_state_cp1_regs_double (0x000003d8)
+%define offsetof_struct_new_dynarec_hot_state_cp1_regs_simple (0x000002d8)
+%define offsetof_struct_new_dynarec_hot_state_cp0_regs (0x00000258)
+%define offsetof_struct_new_dynarec_hot_state_lo (0x00000250)
+%define offsetof_struct_new_dynarec_hot_state_hi (0x00000248)
+%define offsetof_struct_new_dynarec_hot_state_regs (0x00000148)
+%define offsetof_struct_new_dynarec_hot_state_fcr31 (0x00000140)
+%define offsetof_struct_new_dynarec_hot_state_fcr0 (0x0000013c)
+%define offsetof_struct_new_dynarec_hot_state_invc_ptr (0x00000118)
+%define offsetof_struct_new_dynarec_hot_state_stop (0x00000110)
+%define offsetof_struct_new_dynarec_hot_state_pcaddr (0x0000010c)
+%define offsetof_struct_new_dynarec_hot_state_pending_exception (0x00000108)
+%define offsetof_struct_new_dynarec_hot_state_cycle_count (0x00000104)
+%define offsetof_struct_new_dynarec_hot_state_next_interrupt (0x00000100)
+%define offsetof_struct_new_dynarec_hot_state_dynarec_local (0x00000000)
+%define offsetof_struct_r4300_core_extra_memory (0x00901000)
+%define offsetof_struct_r4300_core_new_dynarec_hot_state (0x02901000)
+%define offsetof_struct_cached_interp_invalid_code (0x00000000)
+%define offsetof_struct_r4300_core_cached_interp (0x000000f0)
+%define offsetof_struct_tlb_LUT_w (0x00400680)
+%define offsetof_struct_tlb_LUT_r (0x00000680)
+%define offsetof_struct_tlb_entries (0x00000000)
+%define offsetof_struct_cp0_tlb (0x0000026c)
+%define offsetof_struct_cp0_count_per_op (0x00000268)
+%define offsetof_struct_cp0_last_addr (0x00000264)
+%define offsetof_struct_r4300_core_cp0 (0x031019f0)
+%define offsetof_struct_device_r4300 (0x00000000)
diff -Naur mupen64plus-bundle-src-2.5.9/source/mupen64plus-core/src/device/r4300/idec.h mupen64plus-bundle-src-2.5.9-new/source/mupen64plus-core/src/device/r4300/idec.h
--- mupen64plus-bundle-src-2.5.9/source/mupen64plus-core/src/device/r4300/idec.h	2019-02-10 10:25:29.000000000 -0800
+++ mupen64plus-bundle-src-2.5.9-new/source/mupen64plus-core/src/device/r4300/idec.h	2021-02-01 03:06:30.780366353 -0800
@@ -82,6 +82,6 @@
 
 #define IDEC_U53(r4300, iw, u53, u5) (void*)(((char*)(r4300)) + idec_u53((iw), (u53), (u5)))
 
-const char* g_r4300_opcodes[R4300_OPCODES_COUNT];
+extern const char* g_r4300_opcodes[R4300_OPCODES_COUNT];
 
 #endif
diff -Naur mupen64plus-bundle-src-2.5.9/source/mupen64plus-core/src/main/workqueue.h mupen64plus-bundle-src-2.5.9-new/source/mupen64plus-core/src/main/workqueue.h
--- mupen64plus-bundle-src-2.5.9/source/mupen64plus-core/src/main/workqueue.h	2019-02-10 10:25:29.000000000 -0800
+++ mupen64plus-bundle-src-2.5.9-new/source/mupen64plus-core/src/main/workqueue.h	2021-02-01 03:08:39.088066504 -0800
@@ -27,7 +27,7 @@
 
 struct work_struct;
 
-struct work_struct *work;
+extern struct work_struct *work;
 typedef void (*work_func_t)(struct work_struct *work);
 struct work_struct {
     work_func_t func;
