From ba0a259537cf9bcebc562285570e55788d963e10 Mon Sep 17 00:00:00 2001
From: SkyBird233 <52884766+SkyBird233@users.noreply.github.com>
Date: Mon, 25 May 2026 10:36:32 +0800
Subject: [PATCH 2/3] Use system pandoc by default

---
 kernel/conf/export.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/conf/export.go b/kernel/conf/export.go
index c773a4222..767a66abc 100644
--- a/kernel/conf/export.go
+++ b/kernel/conf/export.go
@@ -58,7 +58,7 @@ func NewExport() *Export {
 		TagOpenMarker:           "#",
 		TagCloseMarker:          "#",
 		FileAnnotationRefMode:   0,
-		PandocBin:               "",
+		PandocBin:               "/usr/bin/pandoc",
 		RemoveAssetsID:          false,
 		MarkdownYFM:             false,
 		InlineMemo:              false,
-- 
2.52.0

