From d7bcddd89db4ce8ba202dd878e549ac18703d71f Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sat, 10 Feb 2024 23:29:32 +0800
Subject: [PATCH 3/4] Use Rust 2021 Edition

---
 Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 442f34d..abb92c9 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -2,7 +2,7 @@
 name = "inputplug"
 version = "0.4.0"
 authors = ["Andrej Shadura <andrew@shadura.me>"]
-edition = "2018"
+edition = "2021"
 description = "XInput monitor daemon"
 license = "MIT"
 repository = "https://github.com/andrewshadura/inputplug"
-- 
2.43.0

