From 1229ce634e4ca50bedf59e822d559b9e50b18c0e Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 28 Jan 2026 03:00:34 +0800
Subject: [PATCH 3/6] FEDORA: add missing errno.h include

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 tclhfs.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/tclhfs.c b/tclhfs.c
index c6182f3..636f985 100644
--- a/tclhfs.c
+++ b/tclhfs.c
@@ -43,8 +43,7 @@
 # include "charset.h"
 # include "suid.h"
 # include "version.h"
-
-extern int errno;
+# include <errno.h>
 
 # define ERROR(code, str)	(hfs_error = (str), errno = (code))
 
-- 
2.52.0

