From 56e3a5bf6280741bcbe5075699a43e43d3ad74aa Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 28 Jan 2026 03:00:59 +0800
Subject: [PATCH 4/6] DEBIAN: fix build with Tcl 8.6

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 config.h.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/config.h.in b/config.h.in
index 232e731..9e587bf 100644
--- a/config.h.in
+++ b/config.h.in
@@ -91,3 +91,5 @@ char *strstr(const char *, const char *);
 # ifndef HAVE_STRTOL
 long strtol(const char *, char **, int);
 # endif
+
+#define USE_INTERP_RESULT 1
-- 
2.52.0

