From 5c62516215d098be8b7d4b31fd5adca487135fe2 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 28 Jan 2026 03:01:16 +0800
Subject: [PATCH 5/6] DEBIAN: fix support for large files

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 libhfs/os/unix.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libhfs/os/unix.c b/libhfs/os/unix.c
index 736243d..474f807 100644
--- a/libhfs/os/unix.c
+++ b/libhfs/os/unix.c
@@ -19,6 +19,11 @@
  * $Id: unix.c,v 1.8 1998/11/02 22:09:13 rob Exp $
  */
 
+#ifdef __linux__
+#define _FILE_OFFSET_BITS 64
+#define _LARGE_FILES
+#endif
+
 # ifdef HAVE_CONFIG_H
 #  include "config.h"
 # endif
-- 
2.52.0

