From c3b874712301a84e49f1db18dc8c13d3bea7c8ad Mon Sep 17 00:00:00 2001
From: Paul Brossier <piem@piem.org>
Date: Wed, 27 Dec 2023 18:18:00 +0100
Subject: [PATCH 4/7] [source_avcodec] use a const pointer for AVCodec

---
 src/io/source_avcodec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/io/source_avcodec.c b/src/io/source_avcodec.c
index 011c3d78..bac74a5b 100644
--- a/src/io/source_avcodec.c
+++ b/src/io/source_avcodec.c
@@ -130,7 +130,7 @@ aubio_source_avcodec_t * new_aubio_source_avcodec(const char_t * path,
   AVCodecParameters *codecpar;
 #endif
 
-  AVCodec *codec;
+  const AVCodec *codec;
   uint_t i;
   int err;
   if (path == NULL) {
-- 
2.46.0

