From 665561fbd1230093cddc56b2b07a3073a71b5256 Mon Sep 17 00:00:00 2001
From: Paul Brossier <piem@piem.org>
Date: Wed, 27 Dec 2023 18:12:54 +0100
Subject: [PATCH] [waf] bump to 2.0.26

---
 scripts/get_waf.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/get_waf.sh b/scripts/get_waf.sh
index c7d921710514..0a40e09aa127 100755
--- a/scripts/get_waf.sh
+++ b/scripts/get_waf.sh
@@ -3,7 +3,7 @@
 set -e
 #set -x
 
-WAFVERSION=2.0.14
+WAFVERSION=2.0.26
 WAFTARBALL=waf-$WAFVERSION.tar.bz2
 WAFURL=https://waf.io/$WAFTARBALL
 WAFUPSTREAMKEY=https://gitlab.com/ita1024/waf/raw/master/utils/pubkey.asc

base-commit: 90bd27a23123fcc524c31787c9c8fc0ae4c79378
-- 
2.46.0

