From 119a6f23eaa607e8ae0cb4655e526a497fec0fe6 Mon Sep 17 00:00:00 2001
From: Paul Brossier <piem@piem.org>
Date: Wed, 27 Dec 2023 18:12:54 +0100
Subject: [PATCH 09/14] bump to 2.0.26

---
 scripts/get_waf.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/get_waf.sh b/scripts/get_waf.sh
index c7d92171..0a40e09a 100755
--- a/scripts/get_waf.sh
+++ b/scripts/get_waf.sh
@@ -3,7 +3,7 @@
 set -e
 #set -x
 
-WAFVERSION=2.0.14
+WAFVERSION=2.0.26
 WAFTARBALL=waf-$WAFVERSION.tar.bz2
 WAFURL=https://waf.io/$WAFTARBALL
 WAFUPSTREAMKEY=https://gitlab.com/ita1024/waf/raw/master/utils/pubkey.asc
-- 
2.53.0

