From 6f88c1e472786f9b9ccee08fd46bf4e542315cad Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Thu, 16 Nov 2023 15:55:24 +0800
Subject: [PATCH] Do not compile `sg` binary

---
 crates/cli/Cargo.toml | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/crates/cli/Cargo.toml b/crates/cli/Cargo.toml
index d04444b..99cd536 100644
--- a/crates/cli/Cargo.toml
+++ b/crates/cli/Cargo.toml
@@ -3,7 +3,7 @@ name = "ast-grep"
 description = "Search and Rewrite code at large scale using precise AST pattern"
 keywords = ["ast", "pattern", "codemod", "search", "rewrite"]
 categories = ["command-line-utilities", "development-tools", "parsing"]
-default-run = "sg"
+default-run = "ast-grep"
 # use relative path because maturin does not recognize
 readme = "../../README.md"
 license-file = "../../LICENSE"
@@ -17,10 +17,6 @@ homepage.workspace = true
 repository.workspace = true
 rust-version.workspace = true
 
-[[bin]]
-name = "sg"
-path = "src/main.rs"
-
 [[bin]]
 name = "ast-grep"
 path = "src/bin/ast-grep.rs"
-- 
2.39.1

