From a7929f19b7c94a63e40082a7779b644c80ff0664 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Thu, 16 Nov 2023 15:55:24 +0800
Subject: [PATCH] Do not compile `sg` binary

---
 crates/cli/Cargo.toml | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/crates/cli/Cargo.toml b/crates/cli/Cargo.toml
index 258c2e8d..cd290c49 100644
--- a/crates/cli/Cargo.toml
+++ b/crates/cli/Cargo.toml
@@ -21,10 +21,6 @@ rust-version.workspace = true
 name = "ast-grep"
 path = "src/main.rs"
 
-[[bin]]
-name = "sg"
-path = "src/bin/alias.rs"
-
 [dependencies]
 ast-grep-core.workspace = true
 ast-grep-config.workspace = true
-- 
2.49.0

