
diff -r bb0a5c371ff9 -r 090cb9c74b87 archivers/zip.c
--- a/archivers/zip.c	Fri Feb 17 20:41:56 2017 -0500
+++ b/archivers/zip.c	Fri Feb 17 20:48:02 2017 -0500
@@ -327,7 +327,8 @@
                 return(0);
 
             inflateEnd(&finfo->stream);
-            memcpy(&finfo->stream, &str, sizeof (z_stream));
+            inflateCopy(&finfo->stream, &str);
+            inflateEnd(&str);
             finfo->uncompressed_position = finfo->compressed_position = 0;
         } /* if */
 




