From 3c856a1da8ba1435ab8cc428f30396c4d3b6d50d Mon Sep 17 00:00:00 2001
From: CAB233 <yidaduizuoye@outlook.com>
Date: Sun, 21 Sep 2025 16:48:07 +0800
Subject: [PATCH 2/3] FEDORA: fix fluxbox after gcc11

---
 util/fluxbox-remote.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/fluxbox-remote.cc b/util/fluxbox-remote.cc
index 59852e6..504015b 100644
--- a/util/fluxbox-remote.cc
+++ b/util/fluxbox-remote.cc
@@ -73,7 +73,7 @@ int main(int argc, char **argv) {
     if (strcmp(cmd, "result") == 0) {
         XTextProperty text_prop;
         if (XGetTextProperty(disp, root, &text_prop, atom_result) != 0
-            && text_prop.value > 0
+            && text_prop.value != 0
             && text_prop.nitems > 0) {
 
             printf("%s", text_prop.value);
-- 
2.51.0

