From 4c2bbd5d7c5beb962117994c6c0d81660df5bf3b Mon Sep 17 00:00:00 2001
From: Neko205 <neko200553@gmail.com>
Date: Sun, 27 Apr 2025 19:06:43 +0800
Subject: [PATCH] feat: disable auto-update and version check

---
 runner/options.go | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/runner/options.go b/runner/options.go
index bbadb40..7864fd1 100644
--- a/runner/options.go
+++ b/runner/options.go
@@ -453,10 +453,12 @@ func ParseOptions() *Options {
 		flagSet.BoolVarP(&options.ListDSLVariable, "list-dsl-variables", "ldv", false, "list json output field keys name that support dsl matcher/filter"),
 	)
 
-	flagSet.CreateGroup("update", "Update",
-		flagSet.CallbackVarP(GetUpdateCallback(), "update", "up", "update httpx to latest version"),
-		flagSet.BoolVarP(&options.DisableUpdateCheck, "disable-update-check", "duc", false, "disable automatic httpx update check"),
-	)
+	//Disable auto-update and version detection
+	//flagSet.CreateGroup("update", "Update",
+	//	flagSet.CallbackVarP(GetUpdateCallback(), "update", "up", "update httpx to latest version"),
+	//	flagSet.BoolVarP(&options.DisableUpdateCheck, "disable-update-check", "duc", false, "disable automatic httpx update check"),
+	//)
+	options.DisableUpdateCheck = true
 
 	flagSet.CreateGroup("output", "Output",
 		flagSet.StringVarP(&options.Output, "output", "o", "", "file to write output results"),
-- 
2.49.0

