Author: Andreas Metzler
Description: Add support for make DESTDIR=...
Last-Update: 2023-03-25

--- a/Makefile.in
+++ b/Makefile.in
@@ -11,13 +11,14 @@ CFLAGS=@CFLAGS@
 LDFLAGS=@LDFLAGS@
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 
+prefix=@prefix@
 BINDIR=@prefix@/bin
-DATADIR=@prefix@/share/xfpt
-MANDIR=@prefix@/man
+DATADIR=@datadir@/xfpt
+MANDIR=@mandir@
 
 # BINDIR is the directory in which the commands are installed.
 # DATADIR is the directory in which the include styles are installed.
 # MANDIR is the directory in which the man pages are installed.
 
@@ -43,21 +44,21 @@ build:; @cd src; $(MAKE) all \
 clean:; cd src; $(MAKE) clean
 
 distclean:;     rm Makefile config.cache config.log config.status; \
                 cd src; $(MAKE) clean
 
-test:; cd testing; runtest
+test:; cd testing; ./runtest
 check: test
 
 install:        build
-		$(mkinstalldirs) $(BINDIR)
-		$(mkinstalldirs) $(DATADIR)
-		$(mkinstalldirs) $(MANDIR)
-		$(mkinstalldirs) $(MANDIR)/man1
-		$(INSTALL) src/xfpt $(BINDIR)/xfpt
-		$(INSTALL_DATA) share/* $(DATADIR)
-		$(INSTALL_DATA) doc/xfpt.1 $(MANDIR)/man1
+		$(mkinstalldirs) $(DESTDIR)$(BINDIR)
+		$(mkinstalldirs) $(DESTDIR)$(DATADIR)
+		$(mkinstalldirs) $(DESTDIR)$(MANDIR)
+		$(mkinstalldirs) $(DESTDIR)$(MANDIR)/man1
+		$(INSTALL) src/xfpt $(DESTDIR)$(BINDIR)/xfpt
+		$(INSTALL_DATA) share/* $(DESTDIR)$(DATADIR)
+		$(INSTALL_DATA) doc/xfpt.1 $(DESTDIR)$(MANDIR)/man1
 
 uninstall:;     rm -rf \
 		$(BINDIR)/xfpt \
 		$(DATADIR) \
 		$(MANDIR)/man1/xfpt.1
--- a/src/Makefile
+++ b/src/Makefile
@@ -43,8 +43,8 @@ read.o:        $(DEP) read.c
 tree.o:        $(DEP) tree.c
 xfpt.o:        $(DEP) xfpt.c
 
 # Clean up etc
 
-clean:;       /bin/rm -f *.o
+clean:;       /bin/rm -f *.o xfpt
 
 # End
