Description: Use CPPFLAGS
Author: Andreas Metzler <ametzler@debian.org>
Origin: vendor
Last-Update: 2023-03-25

--- a/Makefile.in
+++ b/Makefile.in
@@ -6,10 +6,11 @@ mkinstalldirs = $(SHELL) mkinstalldirs
 
 # These variables get set up by the configure script.
 
 CC=@CC@
 CFLAGS=@CFLAGS@
+CPPFLAGS=@CPPFLAGS@
 LDFLAGS=@LDFLAGS@
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 
@@ -36,10 +37,11 @@ build:; @cd src; $(MAKE) all \
                 BINDIR=$(BINDIR) \
                 DATADIR=$(DATADIR) \
                 MANDIR=$(MANDIR) \
                 CC="$(CC)" \
                 CFLAGS="$(CFLAGS)" \
+                CPPFLAGS="$(CPPFLAGS)" \
                 LDFLAGS="$(LDFLAGS)" \
                 FE="$(FE)"
 
 clean:; cd src; $(MAKE) clean
 
--- a/src/Makefile
+++ b/src/Makefile
@@ -13,11 +13,11 @@ all:  xfpt
 
 # Compile step for the modules
 
 .SUFFIXES:  .o .c
 .c.o:;  @echo "$(CC) $*.c"
-	       $(FE)$(CC) -c $(CFLAGS) -DDATADIR='"$(DATADIR)"' $*.c
+	       $(FE)$(CC) -c $(CPPFLAGS) $(CFLAGS) -DDATADIR='"$(DATADIR)"' $*.c
 
 # List of general dependencies
 
 DEP = ../Makefile Makefile xfpt.h structs.h functions.h globals.h mytypes.h
 
