diff -Naur sbcl-1.4.2/contrib/asdf-module.mk sbcl-1.4.2.optflags/contrib/asdf-module.mk
--- sbcl-1.4.2/contrib/asdf-module.mk	2017-11-30 14:21:48.000000000 +0000
+++ sbcl-1.4.2.optflags/contrib/asdf-module.mk	2017-12-02 16:16:32.060596661 +0000
@@ -24,6 +24,8 @@
   EXTRA_CFLAGS+=-D_GNU_SOURCE
 endif
 
+#EXTRA_CFLAGS += $(RPM_OPT_FLAGS)
+
 export CC SBCL EXTRA_CFLAGS
 
 all: $(FASL) $(ASD)
diff -Naur sbcl-1.4.2/src/runtime/Config.arm-linux sbcl-1.4.2.optflags/src/runtime/Config.arm-linux
--- sbcl-1.4.2/src/runtime/Config.arm-linux	2017-11-30 14:21:48.000000000 +0000
+++ sbcl-1.4.2.optflags/src/runtime/Config.arm-linux	2017-12-02 16:16:32.060596661 +0000
@@ -9,7 +9,7 @@
 # provided with absolutely no warranty. See the COPYING and CREDITS
 # files for more information.
 
-CFLAGS += -marm -march=armv5
+#CFLAGS += -marm -march=armv5 
 NM = ./linux-nm
 
 ASSEM_SRC = arm-assem.S ldso-stubs.S
diff -Naur sbcl-1.4.2/src/runtime/GNUmakefile sbcl-1.4.2.optflags/src/runtime/GNUmakefile
--- sbcl-1.4.2/src/runtime/GNUmakefile	2017-11-30 14:21:48.000000000 +0000
+++ sbcl-1.4.2.optflags/src/runtime/GNUmakefile	2017-12-02 16:17:11.232057468 +0000
@@ -1,4 +1,4 @@
-# -*- makefile -*- for the C-level run-time support for SBCL
+ -*- makefile -*- for the C-level run-time support for SBCL
 
 # This software is part of the SBCL system. See the README file for
 # more information.
@@ -30,7 +30,7 @@
 
 include ../../output/prefix.def
 
-CFLAGS += -g -Wall -Wundef -Wsign-compare -Wpointer-arith -O3
+#CFLAGS += -g -Wall -Wundef -Wsign-compare -Wpointer-arith -O3
 ASFLAGS += $(CFLAGS)
 CPPFLAGS += -I. -DSBCL_PREFIX=\"$(SBCL_PREFIX)\"
 

