diff -up sbcl-1.3.2/contrib/asdf-module.mk.optflags sbcl-1.3.2/contrib/asdf-module.mk
--- sbcl-1.3.2/contrib/asdf-module.mk.optflags	2016-01-31 10:14:20.000000000 -0600
+++ sbcl-1.3.2/contrib/asdf-module.mk	2016-02-01 08:54:24.147604689 -0600
@@ -24,6 +24,8 @@ ifeq (Linux,$(UNAME))
   EXTRA_CFLAGS=-D_GNU_SOURCE
 endif
 
+EXTRA_CFLAGS += $(RPM_OPT_FLAGS)
+
 export CC SBCL EXTRA_CFLAGS
 
 all: $(FASL) $(ASD)
diff -up sbcl-1.3.2/src/runtime/Config.arm-linux.optflags sbcl-1.3.2/src/runtime/Config.arm-linux
--- sbcl-1.3.2/src/runtime/Config.arm-linux.optflags	2016-01-31 10:14:21.000000000 -0600
+++ sbcl-1.3.2/src/runtime/Config.arm-linux	2016-02-01 08:54:24.147604689 -0600
@@ -9,7 +9,7 @@
 # provided with absolutely no warranty. See the COPYING and CREDITS
 # files for more information.
 
-CFLAGS += -marm -march=armv5
+#CFLAGS += $(RPM_OPT_FLAGS)
 NM = ./linux-nm
 
 ASSEM_SRC = arm-assem.S ldso-stubs.S
diff -up sbcl-1.3.2/src/runtime/GNUmakefile.optflags sbcl-1.3.2/src/runtime/GNUmakefile
--- sbcl-1.3.2/src/runtime/GNUmakefile.optflags	2016-01-31 10:14:21.000000000 -0600
+++ sbcl-1.3.2/src/runtime/GNUmakefile	2016-02-01 09:04:22.710896405 -0600
@@ -17,14 +17,14 @@ TARGET=sbcl
 # Defaults which might be overridden or modified by values in the
 # Config file. Most of them are same on most systems right now.
 # If you need to override one of these, do it in Config.
-LINKFLAGS = -g
+LINKFLAGS = -g
 NM = nm -gp
 DEPEND_FLAGS = -MM
 GREP = grep
 
 include ../../output/prefix.def
 
-CFLAGS = -g -Wall -Wsign-compare -O3
+#CFLAGS = -g -Wall -Wsign-compare -O3
 ASFLAGS = $(CFLAGS)
 CPPFLAGS = -I. -DSBCL_PREFIX=\"$(SBCL_PREFIX)\"
 

