diff --git a/drivers/media/platform/starfive/v4l2_driver/ov13850_mipi.c b/drivers/media/platform/starfive/v4l2_driver/ov13850_mipi.c
index c79d1ddd2..d91df01a1 100644
--- a/drivers/media/platform/starfive/v4l2_driver/ov13850_mipi.c
+++ b/drivers/media/platform/starfive/v4l2_driver/ov13850_mipi.c
@@ -1880,7 +1880,7 @@ static int ov13850_probe(struct i2c_client *client)
 	return ret;
 }
 
-static int ov13850_remove(struct i2c_client *client)
+static void ov13850_remove(struct i2c_client *client)
 {
 	struct v4l2_subdev *sd = i2c_get_clientdata(client);
 	struct ov13850_dev *sensor = to_ov13850_dev(sd);
@@ -1889,8 +1889,6 @@ static int ov13850_remove(struct i2c_client *client)
 	media_entity_cleanup(&sensor->sd.entity);
 	v4l2_ctrl_handler_free(&sensor->ctrls.handler);
 	mutex_destroy(&sensor->lock);
-
-	return 0;
 }
 
 static const struct i2c_device_id ov13850_id[] = {
@@ -1911,7 +1909,7 @@ static struct i2c_driver ov13850_i2c_driver = {
 		.of_match_table = ov13850_dt_ids,
 	},
 	.id_table = ov13850_id,
-	.probe_new = ov13850_probe,
+	.probe    = ov13850_probe,
 	.remove   = ov13850_remove,
 };
 
diff --git a/drivers/media/platform/starfive/v4l2_driver/ov5640.c b/drivers/media/platform/starfive/v4l2_driver/ov5640.c
index 3aba83c41..2b85ea52f 100644
--- a/drivers/media/platform/starfive/v4l2_driver/ov5640.c
+++ b/drivers/media/platform/starfive/v4l2_driver/ov5640.c
@@ -3186,7 +3186,7 @@ static int ov5640_probe(struct i2c_client *client)
 	return ret;
 }
 
-static int ov5640_remove(struct i2c_client *client)
+static void ov5640_remove(struct i2c_client *client)
 {
 	struct v4l2_subdev *sd = i2c_get_clientdata(client);
 	struct ov5640_dev *sensor = to_ov5640_dev(sd);
@@ -3195,8 +3195,6 @@ static int ov5640_remove(struct i2c_client *client)
 	media_entity_cleanup(&sensor->sd.entity);
 	v4l2_ctrl_handler_free(&sensor->ctrls.handler);
 	mutex_destroy(&sensor->lock);
-
-	return 0;
 }
 
 static const struct i2c_device_id ov5640_id[] = {
@@ -3217,7 +3215,7 @@ static struct i2c_driver ov5640_i2c_driver = {
 		.of_match_table	= ov5640_dt_ids,
 	},
 	.id_table = ov5640_id,
-	.probe_new = ov5640_probe,
+	.probe    = ov5640_probe,
 	.remove   = ov5640_remove,
 };
 
diff --git a/drivers/media/platform/starfive/v4l2_driver/sc2235.c b/drivers/media/platform/starfive/v4l2_driver/sc2235.c
index ffc548ec8..082d4690e 100644
--- a/drivers/media/platform/starfive/v4l2_driver/sc2235.c
+++ b/drivers/media/platform/starfive/v4l2_driver/sc2235.c
@@ -1904,7 +1904,7 @@ static struct i2c_driver sc2235_i2c_driver = {
 		.pm = &sc2235_pm_ops,
 	},
 	.id_table = sc2235_id,
-	.probe_new = sc2235_probe,
+	.probe    = sc2235_probe,
 	.remove   = sc2235_remove,
 };
 
diff --git a/drivers/media/platform/starfive/v4l2_driver/stfcamss.c b/drivers/media/platform/starfive/v4l2_driver/stfcamss.c
index 24bfe8da9..7e62569a1 100644
--- a/drivers/media/platform/starfive/v4l2_driver/stfcamss.c
+++ b/drivers/media/platform/starfive/v4l2_driver/stfcamss.c
@@ -1367,3 +1367,4 @@ late_initcall(stfcamss_init);
 module_exit(stfcamss_cleanup);
 
 MODULE_LICENSE("GPL");
+MODULE_IMPORT_NS(DMA_BUF);
\ No newline at end of file
-- 
2.46.0

