diff --git a/drivers/gpu/drm/img/img-rogue/services/server/devices/rogue/rgxdebug.c b/drivers/gpu/drm/img/img-rogue/services/server/devices/rogue/rgxdebug.c
index bccced1bf..b0ac7bfb2 100644
--- a/drivers/gpu/drm/img/img-rogue/services/server/devices/rogue/rgxdebug.c
+++ b/drivers/gpu/drm/img/img-rogue/services/server/devices/rogue/rgxdebug.c
@@ -2360,6 +2360,7 @@ static void _FillAppForFWFaults(PVRSRV_RGXDEV_INFO *psDevInfo,
 							RGXMEM_PROCESS_INFO *psProcInfo)
 {
 	IMG_UINT32 i, j;
+	const char *unknown_process_name = "(unknown)";
 
 	for (i = 0; i < DEVICEMEM_HISTORY_QUERY_INDEX_COUNT; i++)
 	{
@@ -2373,9 +2374,8 @@ static void _FillAppForFWFaults(PVRSRV_RGXDEV_INFO *psDevInfo,
 								psProcInfo);
 			if (!bFound)
 			{
-				OSStringLCopy(psProcInfo->szProcessName,
-								"(unknown)",
-								sizeof(psProcInfo->szProcessName));
+				strncpy(psProcInfo->szProcessName, unknown_process_name, sizeof(psProcInfo->szProcessName) - 1);
+				psProcInfo->szProcessName[sizeof(psProcInfo->szProcessName) - 1] = '\0';
 			}
 		}
 	}
-- 
2.46.0

