diff --git a/drivers/gpu/drm/i2c/tda998x_pin.c b/drivers/gpu/drm/i2c/tda998x_pin.c
index c730297e1..91964de4c 100644
--- a/drivers/gpu/drm/i2c/tda998x_pin.c
+++ b/drivers/gpu/drm/i2c/tda998x_pin.c
@@ -23,7 +23,7 @@ static const struct of_device_id tda998x_rgb_dt_ids[] = {
 	{ .compatible = "starfive,tda998x_rgb_pin", },
 	{ /* sentinel */ },
 };
-MODULE_DEVICE_TABLE(of, starfive_drm_dt_ids);
+MODULE_DEVICE_TABLE(of, tda998x_rgb_dt_ids);
 
 static struct platform_driver starfive_drm_platform_driver = {
 	.probe = starfive_drm_platform_probe,
-- 
2.46.0

