From f575d639645239b73691c93485bdc44912acb8e6 Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Thu, 22 Jan 2026 07:20:06 +0800
Subject: [PATCH] DEPS

---
 DEPS | 443 ++---------------------------------------------------------
 1 file changed, 11 insertions(+), 432 deletions(-)

diff --git a/DEPS b/DEPS
index 1ab6b2a1ce5..490c055d7a4 100644
--- a/DEPS
+++ b/DEPS
@@ -84,16 +84,16 @@ vars = {
   'ocmock_rev': 'c4ec0e3a7a9f56cfdbd0aa01f4f97bb4b75c5ef8', # v3.7.1
 
   # Download a prebuilt Dart SDK by default
-  'download_dart_sdk': True,
+  'download_dart_sdk': False,
 
   # Download a prebuilt esbuild by default
   'download_esbuild': True,
 
   # Checkout Android dependencies only on platforms where we build for Android targets.
-  'download_android_deps': 'host_os == "mac" or (host_os == "linux" and host_cpu == "x64")',
+  'download_android_deps': False,
 
   # Checkout Java dependencies only on platforms that do not have java installed on path.
-  'download_jdk': True,
+  'download_jdk': False,
 
   # Checkout Windows dependencies only if we are building on Windows.
   'download_windows_deps' : 'host_os == "win"',
@@ -106,7 +106,7 @@ vars = {
   # logic or condition may not work if this flag is False.
   # TODO(zijiehe): Make this condition more strict to only download fuchsia
   # dependencies when necessary: b/40935282
-  'download_fuchsia_deps': 'host_os == "linux"',
+  'download_fuchsia_deps': False,
   # Downloads the fuchsia SDK as listed in fuchsia_sdk_path var. This variable
   # is currently only used for the Fuchsia LSC process and is not intended for
   # local development.
@@ -372,68 +372,6 @@ deps = {
 
   # WARNING: end of dart dependencies list that is cleaned up automatically - see create_updated_flutter_deps.py.
 
-  # Prebuilt Dart SDK of the same revision as the Dart SDK source checkout
-  'engine/src/flutter/prebuilts/linux-x64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/linux-amd64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "linux" and download_dart_sdk'
-  },
-  'engine/src/flutter/prebuilts/linux-arm64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/linux-arm64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "linux" and download_dart_sdk'
-  },
-  'engine/src/flutter/prebuilts/macos-x64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/mac-amd64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "mac" and download_dart_sdk'
-  },
-  'engine/src/flutter/prebuilts/macos-arm64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/mac-arm64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "mac" and download_dart_sdk'
-  },
-  'engine/src/flutter/prebuilts/windows-x64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/windows-amd64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "win" and download_dart_sdk'
-  },
-  'engine/src/flutter/prebuilts/windows-arm64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/windows-arm64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "win" and download_dart_sdk'
-  },
-
   # esbuild download
   'engine/src/flutter/prebuilts/linux-x64/esbuild': {
     'packages': [
@@ -567,275 +505,20 @@ deps = {
   'engine/src/flutter/third_party/json':
   Var('flutter_git') + '/third_party/json.git' + '@' + '17d9eacd248f58b73f4d1be518ef649fe2295642',
 
-  'engine/src/flutter/third_party/gradle': {
-    'packages': [
-      {
-        # See tools/gradle/README.md for update instructions.
-        # Version here means the CIPD tag.
-        'version': 'version:8.11.1',
-        'package': 'flutter/gradle'
-      }
-    ],
-    'condition': 'download_android_deps',
-    'dep_type': 'cipd'
-  },
-
-  'engine/src/flutter/third_party/android_tools/trace_to_text': {
-    'packages': [
-      {
-        # 25.0 downloads for both mac-amd64 and mac-arm64
-        # 26.1 is not found with either platform
-        # 27.1 is the latest release of perfetto
-        'version': 'git_tag:v25.0',
-        'package': 'perfetto/trace_to_text/${{platform}}'
-      }
-    ],
-    'condition': 'download_android_deps',
-    'dep_type': 'cipd'
-  },
-
-   'engine/src/flutter/third_party/android_tools/google-java-format': {
-     'packages': [
-       {
-        'package': 'flutter/android/google-java-format',
-        'version': 'version:1.7-1'
-       }
-     ],
-     # We want to be able to format these as part of CI, and the CI step that
-     # checks formatting runs without downloading the rest of the Android build
-     # tooling. Therefore unlike all the other Android-related tools, we want to
-     # download this every time.
-     'dep_type': 'cipd',
-   },
-
-  'engine/src/flutter/third_party/android_tools': {
-     'packages': [
-       {
-        'package': 'flutter/android/sdk/all/${{platform}}',
-        'version': 'version:36v3'
-       }
-     ],
-     'condition': 'download_android_deps',
-     'dep_type': 'cipd',
-   },
-
-  'engine/src/flutter/third_party/android_embedding_dependencies': {
-     'packages': [
-       {
-        'package': 'flutter/android/embedding_bundle',
-        'version': 'last_updated:2024-09-10T16:32:16-0700'
-       }
-     ],
-     'condition': 'download_android_deps',
-     'dep_type': 'cipd',
-   },
-
-  'engine/src/flutter/third_party/java/openjdk': {
-     'packages': [
-       {
-        'package': 'flutter/java/openjdk/${{platform}}',
-        'version': 'version:21'
-       }
-     ],
-     # Always download the JDK since java is required for running the formatter.
-     'dep_type': 'cipd',
-   },
-
-  'engine/src/flutter/third_party/gn': {
-    'packages': [
-      {
-        'package': 'gn/gn/${{platform}}',
-        'version': 'git_revision:81b24e01531ecf0eff12ec9359a555ec3944ec4e'
-      },
-    ],
-    'dep_type': 'cipd',
-  },
-  'third_party/ninja': {
-    'packages': [
-      {
-        'package': 'infra/3pp/tools/ninja/${{platform}}',
-        'version': 'version:2@1.11.1.chromium.4',
-      }
-    ],
-    'dep_type': 'cipd',
-  },
-
-  'engine/src/flutter/prebuilts/emsdk': {
-   'url': Var('skia_git') + '/external/github.com/emscripten-core/emsdk.git' + '@' + '2514ec738de72cebbba7f4fdba0cf2fabcb779a5',
-   'condition': 'download_emsdk',
-  },
-
-  # Clang on mac and linux are expected to typically be the same revision.
-  # They are separated out so that the autoroller can more easily manage them.
-  'engine/src/flutter/buildtools/mac-x64/clang': {
-    'packages': [
-      {
-        'package': 'fuchsia/third_party/clang/mac-amd64',
-        'version': Var('clang_version'),
-      }
-    ],
-    'condition': 'host_os == "mac"', # On ARM64 Macs too because Goma doesn't support the host-arm64 toolchain.
-    'dep_type': 'cipd',
-  },
-
-  'engine/src/flutter/buildtools/mac-arm64/clang': {
-    'packages': [
-      {
-        'package': 'fuchsia/third_party/clang/mac-arm64',
-        'version': Var('clang_version'),
-      }
-    ],
-    'condition': 'host_os == "mac" and host_cpu == "arm64"',
-    'dep_type': 'cipd',
-  },
-
-  'engine/src/flutter/buildtools/linux-x64/clang': {
-    'packages': [
-      {
-        'package': 'fuchsia/third_party/clang/linux-amd64',
-        'version': Var('clang_version'),
-      }
-    ],
-    'condition': 'host_os == "linux" or host_os == "mac"',
-    'dep_type': 'cipd',
-  },
-
-  'engine/src/flutter/buildtools/linux-arm64/clang': {
-    'packages': [
-      {
-        'package': 'fuchsia/third_party/clang/linux-arm64',
-        'version': Var('clang_version'),
-      }
-    ],
-    'condition': 'host_os == "linux" and host_cpu == "arm64"',
-    'dep_type': 'cipd',
-  },
-
-  'engine/src/flutter/buildtools/windows-x64/clang': {
-    'packages': [
-      {
-        'package': 'fuchsia/third_party/clang/windows-amd64',
-        'version': Var('clang_version'),
-      }
-    ],
-    'condition': 'download_windows_deps',
-    'dep_type': 'cipd',
-  },
-
-  # RBE binaries and configs.
-  'engine/src/flutter/buildtools/linux-x64/reclient': {
+  'engine/src/flutter/third_party/android_tools/google-java-format': {
     'packages': [
       {
-        'package': 'infra/rbe/client/${{platform}}',
-        'version': Var('reclient_version'),
+      'package': 'flutter/android/google-java-format',
+      'version': 'version:1.7-1'
       }
     ],
-    'condition': 'use_rbe and host_os == "linux" and host_cpu == "x64"',
+    # We want to be able to format these as part of CI, and the CI step that
+    # checks formatting runs without downloading the rest of the Android build
+    # tooling. Therefore unlike all the other Android-related tools, we want to
+    # download this every time.
     'dep_type': 'cipd',
   },
 
-  'engine/src/flutter/buildtools/mac-arm64/reclient': {
-    'packages': [
-      {
-        'package': 'infra/rbe/client/${{platform}}',
-        'version': Var('reclient_version'),
-      }
-    ],
-    'condition': 'use_rbe and host_os == "mac" and host_cpu == "arm64"',
-    'dep_type': 'cipd',
-  },
-
-  'engine/src/flutter/buildtools/mac-x64/reclient': {
-    'packages': [
-      {
-        'package': 'infra/rbe/client/${{platform}}',
-        'version': Var('reclient_version'),
-      }
-    ],
-    'condition': 'use_rbe and host_os == "mac" and host_cpu == "x64"',
-    'dep_type': 'cipd',
-  },
-
-  'engine/src/flutter/buildtools/windows-x64/reclient': {
-    'packages': [
-      {
-        'package': 'infra/rbe/client/${{platform}}',
-        'version': Var('reclient_version'),
-      }
-    ],
-    'condition': 'use_rbe and download_windows_deps',
-    'dep_type': 'cipd',
-  },
-
-  'engine/src/flutter/build/rbe': {
-    'packages': [
-      {
-        'package': 'flutter_internal/rbe/reclient_cfgs',
-        'version': 'LNMZdvF2Y86Dq05IWthtVJ_PswIFSRiywIHrkfHhelUC',
-      }
-    ],
-    'condition': 'use_rbe',
-    'dep_type': 'cipd',
-  },
-
-  # gcloud
-  'engine/src/flutter/buildtools/linux-x64/gcloud': {
-    'packages': [
-      {
-        'package': 'infra/3pp/tools/gcloud/${{platform}}',
-        'version': Var('gcloud_version'),
-      }
-    ],
-    'condition': 'use_rbe and host_os == "linux" and host_cpu == "x64"',
-    'dep_type': 'cipd',
-  },
-
-  'engine/src/flutter/buildtools/mac-arm64/gcloud': {
-    'packages': [
-      {
-        'package': 'infra/3pp/tools/gcloud/${{platform}}',
-        'version': Var('gcloud_version'),
-      }
-    ],
-    'condition': 'use_rbe and host_os == "mac" and host_cpu == "arm64"',
-    'dep_type': 'cipd',
-  },
-
-  # Get the SDK from https://chrome-infra-packages.appspot.com/p/fuchsia/sdk/core at the 'latest' tag
-  # Get the toolchain from https://chrome-infra-packages.appspot.com/p/fuchsia/clang at the 'goma' tag
-  'engine/src/fuchsia/sdk/linux': {
-     'packages': [
-       {
-        'package': 'fuchsia/sdk/core/linux-amd64',
-        'version': 'xrIAL91ngrd-wNr9Sx_tu4Esw0_AoX814IP7RkmCJN4C'
-       }
-     ],
-     'condition': 'download_fuchsia_deps and not download_fuchsia_sdk',
-     'dep_type': 'cipd',
-   },
-
-  'engine/src/flutter/tools/fuchsia/test_scripts': {
-     'packages': [
-       {
-        'package': 'chromium/fuchsia/test-scripts',
-        'version': Var('fuchsia_test_scripts_version'),
-       }
-     ],
-     'condition': 'download_fuchsia_deps',
-     'dep_type': 'cipd',
-   },
-
-  'engine/src/flutter/tools/fuchsia/gn-sdk': {
-     'packages': [
-       {
-        'package': 'chromium/fuchsia/gn-sdk',
-        'version': Var('fuchsia_gn_sdk_version'),
-       }
-     ],
-     'condition': 'download_fuchsia_deps',
-     'dep_type': 'cipd',
-   },
-
   'engine/src/flutter/third_party/google_fonts_for_unit_tests': {
       'packages': [
         {
@@ -864,49 +547,6 @@ hooks = [
     'pattern': '.',
     'action': ['python3', 'engine/src/flutter/third_party/dart/tools/generate_sdk_version_file.py'],
   },
-  {
-    # Update the Windows toolchain if necessary.
-    'name': 'win_toolchain',
-    'condition': 'download_windows_deps',
-    'pattern': '.',
-    'action': ['python3', 'engine/src/build/vs_toolchain.py', 'update'],
-  },
-  {
-    'name': 'dia_dll',
-    'pattern': '.',
-    'condition': 'download_windows_deps',
-    'action': [
-      'python3',
-      'engine/src/flutter/tools/dia_dll.py',
-    ],
-  },
-  {
-    'name': 'linux_sysroot_x64',
-    'pattern': '.',
-    'condition': 'download_linux_deps',
-    'action': [
-      'python3',
-      'engine/src/build/linux/sysroot_scripts/install-sysroot.py',
-      '--arch=x64'],
-  },
-  {
-    'name': 'linux_sysroot_arm64',
-    'pattern': '.',
-    'condition': 'download_linux_deps',
-    'action': [
-      'python3',
-      'engine/src/build/linux/sysroot_scripts/install-sysroot.py',
-      '--arch=arm64'],
-  },
-  {
-    'name': 'linux_sysroot_riscv64',
-    'pattern': '.',
-    'condition': 'download_linux_deps',
-    'action': [
-      'python3',
-      'engine/src/build/linux/sysroot_scripts/install-sysroot.py',
-      '--arch=riscv64'],
-  },
   {
     'name': 'pub get --offline',
     'pattern': '.',
@@ -915,30 +555,6 @@ hooks = [
       'engine/src/flutter/tools/pub_get_offline.py',
     ]
   },
-  {
-    'name': 'Download Fuchsia SDK',
-    'pattern': '.',
-    'condition': 'download_fuchsia_deps and download_fuchsia_sdk',
-    'action': [
-      'python3',
-      'engine/src/flutter/tools/download_fuchsia_sdk.py',
-      '--fail-loudly',
-      '--verbose',
-      '--host-os',
-      Var('host_os'),
-      '--fuchsia-sdk-path',
-      Var('fuchsia_sdk_path'),
-    ]
-  },
-  {
-    'name': 'Activate Emscripten SDK',
-    'pattern': '.',
-    'condition': 'download_emsdk',
-    'action': [
-      'python3',
-      'engine/src/flutter/tools/activate_emsdk.py',
-    ]
-  },
   {
     'name': 'Setup githooks',
     'pattern': '.',
@@ -948,41 +564,4 @@ hooks = [
       'engine/src/flutter/tools/githooks/setup.py',
     ]
   },
-  {
-    'name': 'Download Fuchsia system images',
-    'pattern': '.',
-    'condition': 'run_fuchsia_emu',
-    'action': [
-      'env',
-      'DOWNLOAD_FUCHSIA_SDK={download_fuchsia_sdk}',
-      'FUCHSIA_SDK_PATH={fuchsia_sdk_path}',
-      'python3',
-      'engine/src/flutter/tools/fuchsia/with_envs.py',
-      'engine/src/flutter/tools/fuchsia/test_scripts/update_product_bundles.py',
-      'terminal.x64,terminal.qemu-arm64',
-    ]
-  },
-  # The following script checks if it is running in the LUCI environment,
-  # and does nothing if so. This is because Xcode is not yet installed in CI
-  # when this hook is run.
-  {
-    'name': 'Find the iOS and macOS SDKs',
-    'pattern': '.',
-    'condition': 'host_os == "mac"',
-    'action': [
-      'python3',
-      'engine/src/build/mac/darwin_sdk.py',
-      '--as-gclient-hook'
-    ]
-  },
-  {
-    'name': 'Generate Fuchsia GN build rules',
-    'pattern': '.',
-    'condition': 'download_fuchsia_deps',
-    'action': [
-      'python3',
-      'engine/src/flutter/tools/fuchsia/with_envs.py',
-      'engine/src/flutter/tools/fuchsia/test_scripts/gen_build_defs.py',
-    ],
-  },
 ]
-- 
2.52.0

