diff -Nur zssh_1.5c.debian.1-3/Makefile.in zssh_1.5c.debian.1-3.new/Makefile.in
--- zssh_1.5c.debian.1-3/Makefile.in	2007-09-25 01:24:58.000000000 -0700
+++ zssh_1.5c.debian.1-3.new/Makefile.in	2007-09-25 01:25:42.000000000 -0700
@@ -74,7 +74,7 @@
 	ln $(NAME) $@
 
 install :
-	strip $(NAME)
+	#strip $(NAME)
 	$(INSTALL) -m 0711 zssh $(DESTDIR)/@bindir@
 	#ln -f $(DESTDIR)/@bindir@/zssh  $(DESTDIR)/@bindir@/ztelnet
 	$(INSTALL) -m 0644 $(MANS) $(DESTDIR)/@mandir@/man1
