Author: Mihai Maties <mihai@xcyb.org>
Subject: Dead Peer Detection fix
Date: Tue Dec 28 11:09:26 CET 2010
Origin: http://lists.unix-ag.uni-kl.de/pipermail/vpnc-devel/2010-December/003492.html

--- a/vpnc.c
+++ b/vpnc.c
@@ -796,7 +796,9 @@
 		*/
 		s->ike.dpd_attempts = 6;
 		s->ike.dpd_sent = time(NULL);
+		s->ike.dpd_seqno = ntohl(s->ike.dpd_seqno);
 		s->ike.dpd_seqno++;
+		s->ike.dpd_seqno = htonl(s->ike.dpd_seqno);
 		send_dpd(s, 0, s->ike.dpd_seqno);
 	} else {
 		/* Our last dpd request has not yet been acked.  If it's been
@@ -1599,6 +1601,7 @@
 					if (s->ike.dpd_idle != 0) {
 						gcry_create_nonce(&s->ike.dpd_seqno, sizeof(s->ike.dpd_seqno));
 						s->ike.dpd_seqno &= 0x7FFFFFFF;
+						s->ike.dpd_seqno = htonl(s->ike.dpd_seqno);
 						s->ike.dpd_seqno_ack = s->ike.dpd_seqno;
 						s->ike.do_dpd = 1;
 						DEBUG(2, printf("peer is DPD capable (RFC3706)\n"));
