From 0afd3ec601b4dbb57b127896ba711fe9816e62ed Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Thu, 20 Nov 2025 16:34:26 +0800
Subject: [PATCH] Fix boost version less 1.84.0 multiple definition errors

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 454f24b..054c431 100644
--- a/configure.ac
+++ b/configure.ac
@@ -112,6 +112,7 @@ AC_CHECK_HEADERS(
     []
 )
 CPPFLAGS="${saved_CPPFLAGS}"
+BOOST_CFLAGS="-DBOOST_PHOENIX_STL_TUPLE_H_ ${BOOST_CFLAGS}"
 AC_SUBST([BOOST_CFLAGS])
 
 # ================
-- 
2.51.1

