Include <unistd.h> for the swab function.

diff --git a/src/cmsio1.c b/src/cmsio1.c
index a80a5ad435e77df3..f37ea7c38a8205c5 100755
--- a/src/cmsio1.c
+++ b/src/cmsio1.c
@@ -23,8 +23,13 @@
 // ICC profile serialization
 
 
+#define _DEFAULT_SOURCE
+#define _XOPEN_SOURCE
+
 #include "lcms.h"
 
+#include <unistd.h>
+
 // ----------------------------------------------------------------- Tag Serialization
 
 // Alignment of ICC file format uses 4 bytes DWORD
