From 8a310dcfed11b644b75614d74f8bb819e65283d2 Mon Sep 17 00:00:00 2001
From: Vicki Pfau <vi@endrift.com>
Date: Thu, 20 Jan 2022 23:49:56 -0800
Subject: [PATCH 2/4] FFmpeg: Fix GIF recording (fixes #2393)

diff --git a/src/feature/ffmpeg/ffmpeg-encoder.c b/src/feature/ffmpeg/ffmpeg-encoder.c
index 55c874692..ad76ca57b 100644
--- a/src/feature/ffmpeg/ffmpeg-encoder.c
+++ b/src/feature/ffmpeg/ffmpeg-encoder.c
@@ -785,7 +785,7 @@ void _ffmpegPostVideoFrame(struct mAVStream* stream, const color_t* pixels, size
 	sws_scale(encoder->scaleContext, (const uint8_t* const*) &pixels, (const int*) &stride, 0, encoder->iheight, encoder->videoFrame->data, encoder->videoFrame->linesize);
 
 	if (encoder->graph) {
-		if (av_buffersrc_add_frame(encoder->source, encoder->videoFrame) < 0) {
+		if (av_buffersrc_write_frame(encoder->source, encoder->videoFrame) < 0) {
 			return;
 		}
 		while (true) {
-- 
2.36.0

