From 31201d49033a0519e3272e3359ca75b97c183e6d Mon Sep 17 00:00:00 2001
From: Vicki Pfau <vi@endrift.com>
Date: Fri, 4 Feb 2022 23:58:57 -0800
Subject: [PATCH 4/4] FFmpeg: Fix build

---
 src/feature/ffmpeg/ffmpeg-encoder.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/feature/ffmpeg/ffmpeg-encoder.c b/src/feature/ffmpeg/ffmpeg-encoder.c
index be6bd3af5..2072afd29 100644
--- a/src/feature/ffmpeg/ffmpeg-encoder.c
+++ b/src/feature/ffmpeg/ffmpeg-encoder.c
@@ -12,7 +12,7 @@
 
 #include <libavcodec/version.h>
 #include <libavcodec/avcodec.h>
-#if LIBAVCODEC_VERSION_MAJOR >= 58
+#if LIBAVCODEC_VERSION_MAJOR >= 59
 #include <libavcodec/bsf.h>
 #endif
 
-- 
2.36.0

