diff -Naur mgba-0.5.1/src/platform/qt/AudioProcessorQt.cpp mgba-0.5.1.endian/src/platform/qt/AudioProcessorQt.cpp
--- mgba-0.5.1/src/platform/qt/AudioProcessorQt.cpp	2016-10-05 15:39:41.000000000 -0500
+++ mgba-0.5.1.endian/src/platform/qt/AudioProcessorQt.cpp	2016-11-06 16:49:01.988825308 -0600
@@ -50,7 +50,7 @@
 		format.setChannelCount(2);
 		format.setSampleSize(16);
 		format.setCodec("audio/pcm");
-		format.setByteOrder(QAudioFormat::LittleEndian);
+		format.setByteOrder(QAudioFormat::Endian(QSysInfo::ByteOrder));
 		format.setSampleType(QAudioFormat::SignedInt);
 
 		m_audioOutput = new QAudioOutput(format, this);
