diff -Naur codeblocks-20.03/src/plugins/contrib/BrowseTracker/BrowseMarks.cpp codeblocks-20.03-new/src/plugins/contrib/BrowseTracker/BrowseMarks.cpp
--- codeblocks-20.03/src/plugins/contrib/BrowseTracker/BrowseMarks.cpp	2020-03-28 06:35:46.000000000 -0600
+++ codeblocks-20.03-new/src/plugins/contrib/BrowseTracker/BrowseMarks.cpp	2021-02-06 21:33:14.209581941 -0700
@@ -52,7 +52,7 @@
 {
     //ctor
     #if defined(LOGGING)
-    if ( fullPath.IsEmpty() ) {asm("int3"); }
+    if ( fullPath.IsEmpty() ) {__builtin_trap(); }
     #endif
     wxFileName fname(fullPath);
     if ( fullPath.IsEmpty() )
@@ -78,7 +78,7 @@
     #endif
     EditorBase* eb = m_pEdMgr->GetEditor(m_filePath);
     #if defined(LOGGING)
-    if (not eb) asm("int3"); /*trap*/
+    if (not eb) __builtin_trap(); /*trap*/
     #endif
     if (not eb) return;
     cbEditor* cbed = Manager::Get()->GetEditorManager()->GetBuiltinEditor(eb);
@@ -267,7 +267,7 @@
     //-EditorBase* eb = m_pEditorBase;
     EditorBase* eb = m_pEdMgr->GetEditor(m_filePath);
     #if defined(LOGGING)
-    if (not eb) asm("int3"); /*trap*/
+    if (not eb) __builtin_trap(); /*trap*/
     #endif
     if (not eb) return;
     cbEditor* cbed = Manager::Get()->GetEditorManager()->GetBuiltinEditor(eb);
diff -Naur codeblocks-20.03/src/plugins/contrib/BrowseTracker/BrowseTracker.cpp codeblocks-20.03-new/src/plugins/contrib/BrowseTracker/BrowseTracker.cpp
--- codeblocks-20.03/src/plugins/contrib/BrowseTracker/BrowseTracker.cpp	2020-03-28 06:35:46.000000000 -0600
+++ codeblocks-20.03-new/src/plugins/contrib/BrowseTracker/BrowseTracker.cpp	2021-02-06 21:33:14.209581941 -0700
@@ -1309,7 +1309,7 @@
             #if defined(LOGGING)
             LOGIT( _T("BT RecordBrowseMarkByEb: pos[%d]line[%d]eb[%p][%s]"),
                 pos, m_CurrScrLine, eb, eb->GetShortName().c_str() );
-            if (not LineHasBrowseMarker(control, m_CurrScrLine)  ) {asm("int3"); /*trap*/}
+            if (not LineHasBrowseMarker(control, m_CurrScrLine)  ) {__builtin_trap(); /*trap*/}
             #endif
         }//if
     }while(0);//if do
diff -Naur codeblocks-20.03/src/plugins/contrib/BrowseTracker/ProjectData.cpp codeblocks-20.03-new/src/plugins/contrib/BrowseTracker/ProjectData.cpp
--- codeblocks-20.03/src/plugins/contrib/BrowseTracker/ProjectData.cpp	2020-03-28 06:35:46.000000000 -0600
+++ codeblocks-20.03-new/src/plugins/contrib/BrowseTracker/ProjectData.cpp	2021-02-06 21:33:14.209581941 -0700
@@ -55,7 +55,7 @@
 {
     //ctor
     #if defined(LOGGING)
-    if (not pcbProject) asm("int3"); /*trap*/;
+    if (not pcbProject) __builtin_trap(); /*trap*/;
     #endif
     if (not pcbProject) return;
     m_pCBProject = pcbProject;
@@ -162,7 +162,7 @@
 
     EditorBase* eb = m_pEdMgr->GetEditor(fullPath);
     #if defined(LOGGING)
-        if(not eb) asm("int3"); /*trap*/
+        if(not eb) __builtin_trap(); /*trap*/
     #endif
     if(not eb) return 0;
     wxString filePath = eb->GetFilename();
diff -Naur codeblocks-20.03/src/plugins/contrib/BrowseTracker/Version.h codeblocks-20.03-new/src/plugins/contrib/BrowseTracker/Version.h
--- codeblocks-20.03/src/plugins/contrib/BrowseTracker/Version.h	2020-03-28 06:35:46.000000000 -0600
+++ codeblocks-20.03-new/src/plugins/contrib/BrowseTracker/Version.h	2021-02-06 21:33:14.209581941 -0700
@@ -33,7 +33,7 @@
  #define LOGGING 1
  #undef LOGIT
  #define LOGIT wxLogMessage
- #define TRAP asm("int3")
+ #define TRAP __builtin_trap()
 #endif
 
 // ----------------------------------------------------------------------------
diff -Naur codeblocks-20.03/src/plugins/contrib/codesnippets/version.h codeblocks-20.03-new/src/plugins/contrib/codesnippets/version.h
--- codeblocks-20.03/src/plugins/contrib/codesnippets/version.h	2020-03-28 06:35:50.000000000 -0600
+++ codeblocks-20.03-new/src/plugins/contrib/codesnippets/version.h	2021-02-06 21:33:14.209581941 -0700
@@ -33,7 +33,7 @@
  #define LOGGING 1
  #undef LOGIT
  #define LOGIT wxLogMessage
- #define TRAP asm("int3")
+ #define TRAP __builtin_trap()
 #endif
 
 // ----------------------------------------------------------------------------
diff -Naur codeblocks-20.03/src/plugins/contrib/dragscroll/dragscrollcfg.h codeblocks-20.03-new/src/plugins/contrib/dragscroll/dragscrollcfg.h
--- codeblocks-20.03/src/plugins/contrib/dragscroll/dragscrollcfg.h	2020-03-28 06:35:51.000000000 -0600
+++ codeblocks-20.03-new/src/plugins/contrib/dragscroll/dragscrollcfg.h	2021-02-06 21:33:14.209581941 -0700
@@ -35,7 +35,7 @@
         wxString GetBitmapBaseName() const;
         void OnApply();
         void OnCancel(){}
-        virtual void InitDialog(){ asm("int3");} /*trap*/
+        virtual void InitDialog(){ __builtin_trap();} /*trap*/
 
         // pointer to owner of the configuration diaglog needed to
         // complete the OnApply/OnCancel EndModal() logic
diff -Naur codeblocks-20.03/src/plugins/contrib/keybinder/debugging.h codeblocks-20.03-new/src/plugins/contrib/keybinder/debugging.h
--- codeblocks-20.03/src/plugins/contrib/keybinder/debugging.h	2020-03-28 06:35:44.000000000 -0600
+++ codeblocks-20.03-new/src/plugins/contrib/keybinder/debugging.h	2021-02-06 21:33:14.209581941 -0700
@@ -14,7 +14,7 @@
  #define LOGGING 1
  #undef LOGIT
  #define LOGIT wxLogMessage
- #define TRAP asm("int3")
+ #define TRAP __builtin_trap()
 #endif
 
 // ----------------------------------------------------------------------------
diff -Naur codeblocks-20.03/src/plugins/contrib/MouseSap/MouseSap.h codeblocks-20.03-new/src/plugins/contrib/MouseSap/MouseSap.h
--- codeblocks-20.03/src/plugins/contrib/MouseSap/MouseSap.h	2020-03-28 06:35:48.000000000 -0600
+++ codeblocks-20.03-new/src/plugins/contrib/MouseSap/MouseSap.h	2021-02-06 21:33:14.209581941 -0700
@@ -44,7 +44,7 @@
  #define LOGGING 1
  #undef LOGIT
  #define LOGIT wxLogMessage
- #define TRAP asm("int3")
+ #define TRAP __builtin_trap()
 #endif
 
 // anchor to one and only MouseSap object
diff -Naur codeblocks-20.03/src/plugins/contrib/ThreadSearch/logging.h codeblocks-20.03-new/src/plugins/contrib/ThreadSearch/logging.h
--- codeblocks-20.03/src/plugins/contrib/ThreadSearch/logging.h	2020-03-28 06:35:49.000000000 -0600
+++ codeblocks-20.03-new/src/plugins/contrib/ThreadSearch/logging.h	2021-02-06 21:33:14.209581941 -0700
@@ -21,7 +21,7 @@
  #define LOGGING 1
  #undef LOGIT
  #define LOGIT wxLogMessage
- #define TRAP asm("int3")
+ #define TRAP __builtin_trap()
 #endif
 
 // ----------------------------------------------------------------------------
