--- host/CMakeLists.txt	2021-02-11 20:02:23.943489426 -0800
+++ host.atomic/CMakeLists.txt	2021-02-11 20:01:33.802951753 -0800
@@ -20,6 +20,8 @@
 #make sure our local CMake Modules path comes first
 list(INSERT CMAKE_MODULE_PATH 0 ${CMAKE_SOURCE_DIR}/cmake/Modules)
 
+set(CMAKE_EXE_LINKER_FLAGS "-latomic")
+
 ########################################################################
 # UHD Dependency Minimum Versions
 ########################################################################
