From: Jamie Strandboge <jamie@ubuntu.com>
Date: Mon, 29 Jan 2018 12:26:43 +0000
Subject: Allow access to socket directory used by recent ibus-daemon

Forwarded: https://gitlab.com/apparmor/apparmor/-/merge_requests/837
---
 profiles/apparmor.d/abstractions/ibus | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/profiles/apparmor.d/abstractions/ibus b/profiles/apparmor.d/abstractions/ibus
index 412bad6..a8b2bf1 100644
--- a/profiles/apparmor.d/abstractions/ibus
+++ b/profiles/apparmor.d/abstractions/ibus
@@ -16,5 +16,14 @@
   owner @{HOME}/.config/ibus/bus/ rw,
   owner @{HOME}/.config/ibus/bus/* rw,
 
+  # abstract path in ibus >= 1.5.22 uses $XDG_CACHE_HOME (ie, @{HOME}/.cache)
+  # This should use this, but due to LP: #1856738 we cannot
+  #unix (connect, receive, send)
+  #    type=stream
+  #    peer=(addr="@@{HOME}/.cache/ibus/dbus-*"),
+  unix (connect, receive, send)
+       type=stream
+       peer=(addr="@/home/*/.cache/ibus/dbus-*"),
+
   # Include additions to the abstraction
   include if exists <abstractions/ibus.d>
