diff --git a/setup.py b/setup.py
index b995328..cb20c25 100755
--- a/setup.py
+++ b/setup.py
@@ -10,10 +10,7 @@ except ImportError:
 
 import smmap
 
-if os.path.exists("README.md"):
-    long_description = open('README.md', encoding="utf-8").read().replace('\r\n', '\n')
-else:
-    long_description = "See https://github.com/gitpython-developers/smmap"
+long_description = "See https://github.com/gitpython-developers/smmap"
 
 setup(
     name="smmap",
