diff --git a/library/GnomeKeyring-1.0.metadata b/library/GnomeKeyring-1.0.metadata
index 04662fb2198818a7bf160ecffda951144b186bd3..d8451b93728d3e4e78ee15f1259c78795040d15f 100644
--- a/library/GnomeKeyring-1.0.metadata
+++ b/library/GnomeKeyring-1.0.metadata
@@ -54,7 +54,7 @@ set_info skip=false
 set_network_password skip=false
 store_password skip=false
 store_password_sync skip=false
-unlock skip=false
+@unlock skip=false
 
 find_items*.callback type_arguments="GnomeKeyring.Found"
 find_network_password*.callback type_arguments="GnomeKeyring.NetworkPasswordData"
