diff --git a/clang/include/clang/Basic/Builtins.td b/clang/include/clang/Basic/Builtins.td
index 29939242596b..16a8a40cdbfa 100644
--- a/clang/include/clang/Basic/Builtins.td
+++ b/clang/include/clang/Basic/Builtins.td
@@ -314,7 +314,7 @@ def NanF16 : Builtin {
 }

 def NanF128 : Builtin {
-  let Spellings = ["__builtin_nanf128"];
+  let Spellings = ["__builtin_nanf128", "__builtin_nansq"];
   let Attributes = [FunctionWithBuiltinPrefix, NoThrow, Pure, Constexpr];
   let Prototype = "__float128(char const*)";
 }
diff --git a/clang/lib/AST/ByteCode/InterpBuiltin.cpp b/clang/lib/AST/ByteCode/InterpBuiltin.cpp
index e657dbd2f9c7..a32a0718377d 100644
--- a/clang/lib/AST/ByteCode/InterpBuiltin.cpp
+++ b/clang/lib/AST/ByteCode/InterpBuiltin.cpp
@@ -2097,6 +2097,7 @@ bool InterpretBuiltin(InterpState &S, CodePtr OpPC, const Function *F,
     break;
   case Builtin::BI__builtin_nans:
   case Builtin::BI__builtin_nansf:
+  case Builtin::BI__builtin_nansq:
   case Builtin::BI__builtin_nansl:
   case Builtin::BI__builtin_nansf16:
   case Builtin::BI__builtin_nansf128:
diff --git a/clang/lib/AST/ExprConstant.cpp b/clang/lib/AST/ExprConstant.cpp
index 0e41e3dbc8a3..7bd707edf578 100644
--- a/clang/lib/AST/ExprConstant.cpp
+++ b/clang/lib/AST/ExprConstant.cpp
@@ -15541,6 +15541,7 @@ bool FloatExprEvaluator::VisitCallExpr(const CallExpr *E) {

   case Builtin::BI__builtin_nans:
   case Builtin::BI__builtin_nansf:
+  case Builtin::BI__builtin_nansq:
   case Builtin::BI__builtin_nansl:
   case Builtin::BI__builtin_nansf16:
   case Builtin::BI__builtin_nansf128:
