diff --git a/amd/comgr/cmake/opencl_pch.cmake b/amd/comgr/cmake/opencl_pch.cmake
index 1140cf8cf3f6..f4b5d9522d32 100644
--- a/amd/comgr/cmake/opencl_pch.cmake
+++ b/amd/comgr/cmake/opencl_pch.cmake
@@ -20,7 +20,7 @@ if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
   endif()
 else()
   get_target_property(clang_build_header_dir clang-resource-headers RUNTIME_OUTPUT_DIRECTORY)
-  set(OPENCL_C_H "${clang_build_header_dir}/opencl-c.h")
+  set(OPENCL_C_H "${CMAKE_SOURCE_DIR}/../clang/lib/Headers/opencl-c.h")
 endif()
 
 # Macro to create and install a custom target for generating PCH for given
diff --git a/clang/include/clang/Basic/Builtins.td b/clang/include/clang/Basic/Builtins.td
index 4133f6ff40cf..07c85801d51d 100644
--- a/clang/include/clang/Basic/Builtins.td
+++ b/clang/include/clang/Basic/Builtins.td
@@ -307,6 +307,13 @@ def NanF128 : Builtin {
   let Prototype = "__float128(char const*)";
 }
 
+def NansQ : Builtin {
+  let Spellings = ["__builtin_nansq"];
+  let Attributes = [FunctionWithBuiltinPrefix, NoThrow, Pure, Constexpr];
+  let Prototype = "__float128(char const*)";
+}
+
+
 def Nans : Builtin,
     Template<["float", "double", "long double", "_Float16", "__float128"],
              ["f",     "",       "l",           "f16",      "f128"]> {
diff --git a/clang/lib/AST/ExprConstant.cpp b/clang/lib/AST/ExprConstant.cpp
index e4895e574744..385923cb432b 100644
--- a/clang/lib/AST/ExprConstant.cpp
+++ b/clang/lib/AST/ExprConstant.cpp
@@ -14889,6 +14889,7 @@ bool FloatExprEvaluator::VisitCallExpr(const CallExpr *E) {
   case Builtin::BI__builtin_nansl:
   case Builtin::BI__builtin_nansf16:
   case Builtin::BI__builtin_nansf128:
+  case Builtin::BI__builtin_nansq:
     if (!TryEvaluateBuiltinNaN(Info.Ctx, E->getType(), E->getArg(0),
                                true, Result))
       return Error(E);
diff --git a/clang/lib/AST/Interp/InterpBuiltin.cpp b/clang/lib/AST/Interp/InterpBuiltin.cpp
index 98928b3c22d7..b7ac2b4810ec 100644
--- a/clang/lib/AST/Interp/InterpBuiltin.cpp
+++ b/clang/lib/AST/Interp/InterpBuiltin.cpp
@@ -1149,6 +1149,7 @@ bool InterpretBuiltin(InterpState &S, CodePtr OpPC, const Function *F,
   case Builtin::BI__builtin_nansl:
   case Builtin::BI__builtin_nansf16:
   case Builtin::BI__builtin_nansf128:
+  case Builtin::BI__builtin_nansq:
     if (!interp__builtin_nan(S, OpPC, Frame, F, /*Signaling=*/true))
       return false;
     break;
diff --git a/clang/lib/Basic/Targets/LoongArch.h b/clang/lib/Basic/Targets/LoongArch.h
index 5fc223483951..c8268dc91e5a 100644
--- a/clang/lib/Basic/Targets/LoongArch.h
+++ b/clang/lib/Basic/Targets/LoongArch.h
@@ -44,6 +44,8 @@ public:
     SuitableAlign = 128;
     WCharType = SignedInt;
     WIntType = UnsignedInt;
+    HasFloat128 = true;
+    HasFloat16 = true;
   }
 
   bool setCPU(const std::string &Name) override {
diff --git a/clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp b/clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp
index 2164bb330211..60f17295eeff 100644
--- a/clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp
+++ b/clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp
@@ -613,6 +613,7 @@ Expected<StringRef> linkDevice(ArrayRef<StringRef> InputFiles,
   case Triple::ppc64:
   case Triple::ppc64le:
   case Triple::systemz:
+  case Triple::loongarch64:
     return generic::clang(InputFiles, Args);
   default:
     return createStringError(Triple.getArchName() +
diff --git a/llvm/include/llvm/ADT/SmallVector.h b/llvm/include/llvm/ADT/SmallVector.h
index 09676d792dfe..4757855fdf29 100644
--- a/llvm/include/llvm/ADT/SmallVector.h
+++ b/llvm/include/llvm/ADT/SmallVector.h
@@ -29,6 +29,7 @@
 #include <new>
 #include <type_traits>
 #include <utility>
+#include <cstdint>
 
 namespace llvm {
 
diff --git a/llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUMCTargetDesc.h b/llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUMCTargetDesc.h
index 3ef00f75735b..40939eb503bf 100644
--- a/llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUMCTargetDesc.h
+++ b/llvm/lib/Target/AMDGPU/MCTargetDesc/AMDGPUMCTargetDesc.h
@@ -16,7 +16,7 @@
 #define LLVM_LIB_TARGET_AMDGPU_MCTARGETDESC_AMDGPUMCTARGETDESC_H
 
 #include <memory>
-
+#include <cstdint>
 namespace llvm {
 class Target;
 class MCAsmBackend;
diff --git a/llvm/lib/Target/LoongArch/LoongArchTargetMachine.cpp b/llvm/lib/Target/LoongArch/LoongArchTargetMachine.cpp
index e83fc08696ae..23497d5c6568 100644
--- a/llvm/lib/Target/LoongArch/LoongArchTargetMachine.cpp
+++ b/llvm/lib/Target/LoongArch/LoongArchTargetMachine.cpp
@@ -68,8 +68,7 @@ static CodeModel::Model
 getEffectiveLoongArchCodeModel(const Triple &TT,
                                std::optional<CodeModel::Model> CM) {
   if (!CM)
-    return CodeModel::Small;
-
+    return TT.isArch64Bit() ? CodeModel::Medium : CodeModel::Small;
   switch (*CM) {
   case CodeModel::Small:
     return *CM;
diff --git a/mlir/include/mlir/Dialect/Affine/IR/ValueBoundsOpInterfaceImpl.h b/mlir/include/mlir/Dialect/Affine/IR/ValueBoundsOpInterfaceImpl.h
index 451c466fa0c9..2e06a1b32e4b 100644
--- a/mlir/include/mlir/Dialect/Affine/IR/ValueBoundsOpInterfaceImpl.h
+++ b/mlir/include/mlir/Dialect/Affine/IR/ValueBoundsOpInterfaceImpl.h
@@ -10,7 +10,7 @@
 #define MLIR_DIALECT_AFFINE_IR_VALUEBOUNDSOPINTERFACEIMPL_H
 
 #include "mlir/Support/LLVM.h"
-
+#include <cstdint>
 namespace mlir {
 class DialectRegistry;
 class Value;
diff --git a/mlir/include/mlir/Target/SPIRV/Deserialization.h b/mlir/include/mlir/Target/SPIRV/Deserialization.h
index e39258beeaac..a346a7fd1e5f 100644
--- a/mlir/include/mlir/Target/SPIRV/Deserialization.h
+++ b/mlir/include/mlir/Target/SPIRV/Deserialization.h
@@ -15,6 +15,7 @@
 
 #include "mlir/IR/OwningOpRef.h"
 #include "mlir/Support/LLVM.h"
+#include <cstdint>
 
 namespace mlir {
 class MLIRContext;
diff --git a/openmp/libomptarget/libm/CMakeLists.txt b/openmp/libomptarget/libm/CMakeLists.txt
index e16ce67c527e..658afa26b894 100644
--- a/openmp/libomptarget/libm/CMakeLists.txt
+++ b/openmp/libomptarget/libm/CMakeLists.txt
@@ -51,8 +51,13 @@ foreach(mcpu ${amdgpu_mcpus})
   if(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "ppc64le")
     set(cpu_target powerpc64le-linux-gnu)
   endif()
+  if(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "loongarch64")
+    set(cpu_target loongarch64-linux-gnu)
+  endif()
   # Need omp.h from the build directory
+  separate_arguments(CXX_FLAGS UNIX_COMMAND ${CMAKE_CXX_FLAGS})
   set(omp_common_args
+     ${CXX_FLAGS}
     -c 
     -I ${CMAKE_BINARY_DIR}${_xdir}/runtime/src
     -emit-llvm
