diff --git a/clang/lib/Basic/Targets/LoongArch.h b/clang/lib/Basic/Targets/LoongArch.h
index 88dc433924d6..c829d7fad69c 100644
--- a/clang/lib/Basic/Targets/LoongArch.h
+++ b/clang/lib/Basic/Targets/LoongArch.h
@@ -61,6 +61,7 @@ public:
     WCharType = SignedInt;
     WIntType = UnsignedInt;
     BitIntMaxAlign = 128;
+    HasFloat128 = true;
   }
 
   bool setCPU(const std::string &Name) override {
