diff --git a/clang/lib/Basic/Targets/LoongArch.h b/clang/lib/Basic/Targets/LoongArch.h
index 5c34c84ff8d3..727b7ade7d36 100644
--- a/clang/lib/Basic/Targets/LoongArch.h
+++ b/clang/lib/Basic/Targets/LoongArch.h
@@ -56,6 +56,12 @@ public:
     SuitableAlign = 128;
     WCharType = SignedInt;
     WIntType = UnsignedInt;
+
+    HasFloat128 = true;
+    HasFloat16 = true;
+    HasBFloat16 = true;
+    BFloat16Format = &llvm::APFloat::BFloat();
+    BFloat16Width = BFloat16Align = 16;
   }

   bool setCPU(const std::string &Name) override {
