diff --git a/flang/include/flang/Optimizer/Builder/BoxValue.h b/flang/include/flang/Optimizer/Builder/BoxValue.h
index 43b70943aced..7a340f665b35 100644
--- a/flang/include/flang/Optimizer/Builder/BoxValue.h
+++ b/flang/include/flang/Optimizer/Builder/BoxValue.h
@@ -92,8 +92,9 @@ public:

   friend llvm::raw_ostream &operator<<(llvm::raw_ostream &,
                                        const CharBoxValue &);
+#ifndef INVISIBLE_LLVM_DUMP
   LLVM_DUMP_METHOD void dump() const { llvm::errs() << *this; }
-
+#endif
 protected:
   mlir::Value len;
 };
@@ -112,7 +113,9 @@ public:

   friend llvm::raw_ostream &operator<<(llvm::raw_ostream &,
                                        const PolymorphicValue &);
+#ifndef INVISIBLE_LLVM_DUMP
   LLVM_DUMP_METHOD void dump() const { llvm::errs() << *this; }
+#endif

 protected:
   mlir::Value sourceBox;
@@ -163,7 +166,9 @@ public:

   friend llvm::raw_ostream &operator<<(llvm::raw_ostream &,
                                        const ArrayBoxValue &);
+#ifndef INVISIBLE_LLVM_DUMP
   LLVM_DUMP_METHOD void dump() const { llvm::errs() << *this; }
+#endif
 };

 /// Expressions of type CHARACTER and with rank > 0.
@@ -184,7 +189,9 @@ public:

   friend llvm::raw_ostream &operator<<(llvm::raw_ostream &,
                                        const CharArrayBoxValue &);
+#ifndef INVISIBLE_LLVM_DUMP
   LLVM_DUMP_METHOD void dump() const { llvm::errs() << *this; }
+#endif
 };

 /// Expressions that are procedure POINTERs may need a set of references to
@@ -202,7 +209,9 @@ public:

   friend llvm::raw_ostream &operator<<(llvm::raw_ostream &,
                                        const ProcBoxValue &);
+#ifndef INVISIBLE_LLVM_DUMP
   LLVM_DUMP_METHOD void dump() const { llvm::errs() << *this; }
+#endif

 protected:
   mlir::Value hostContext;
@@ -307,7 +316,9 @@ public:
   }

   friend llvm::raw_ostream &operator<<(llvm::raw_ostream &, const BoxValue &);
+#ifndef INVISIBLE_LLVM_DUMP
   LLVM_DUMP_METHOD void dump() const { llvm::errs() << *this; }
+#endif

   llvm::ArrayRef<mlir::Value> getLBounds() const { return lbounds; }

@@ -392,7 +403,9 @@ public:
   llvm::ArrayRef<mlir::Value> nonDeferredLenParams() const { return lenParams; }
   friend llvm::raw_ostream &operator<<(llvm::raw_ostream &,
                                        const MutableBoxValue &);
+#ifndef INVISIBLE_LLVM_DUMP
   LLVM_DUMP_METHOD void dump() const { llvm::errs() << *this; }
+#endif

   /// Set of variable is used instead of a descriptor to hold the entity
   /// properties instead of a fir.ref<fir.box<>>.
@@ -537,7 +550,9 @@ public:
   }

   /// LLVM style debugging of extended values
+#ifndef INVISIBLE_LLVM_DUMP
   LLVM_DUMP_METHOD void dump() const { llvm::errs() << *this << '\n'; }
+#endif

   friend llvm::raw_ostream &operator<<(llvm::raw_ostream &,
                                        const ExtendedValue &);
diff --git a/flang/lib/Support/CMakeLists.txt b/flang/lib/Support/CMakeLists.txt
index 363f57ce97da..ac8e03fbe089 100644
--- a/flang/lib/Support/CMakeLists.txt
+++ b/flang/lib/Support/CMakeLists.txt
@@ -61,3 +61,5 @@ add_flang_library(FortranSupport
   MLIRIR
   MLIRSupport
 )
+
+target_compile_definitions(FortranSupport PRIVATE INVISIBLE_LLVM_DUMP)
