diff --git a/src/include/gdrwrap.h b/src/include/gdrwrap.h
index d3056908..b78f27f4 100644
--- a/src/include/gdrwrap.h
+++ b/src/include/gdrwrap.h
@@ -43,7 +43,7 @@ static inline void wc_store_fence(void) { asm volatile("sync") ; }
 #elif defined(__x86_64__)
 #include <immintrin.h>
 static inline void wc_store_fence(void) { _mm_sfence(); }
-#elif defined(__aarch64__) || defined(__riscv)
+#elif defined(__aarch64__) || defined(__riscv) || defined(__loongarch_lp64)
 #ifdef __cplusplus
 #include <atomic>
 static inline void wc_store_fence(void) { std::atomic_thread_fence(std::memory_order_release); }
