diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8281f23..1a7954bc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -318,11 +318,6 @@ if (HAVE_KERNARG_PRELOAD)
   message(STATUS "Kernarg preloading to SGPR enabled")
 endif()

-check_cxx_compiler_flag("-parallel-jobs=12" HAVE_PARALLEL_JOBS)
-if (HAVE_PARALLEL_JOBS)
-  message(STATUS "Parallel jobs enabled")
-endif()
-
 ## Disable building MSCCL++ if the build environment is invalid
 ## Currently MSCCL++ is supported only on gfx942 and gfx950, and only on Ubuntu and CentOS
 set(MSCCLPP_SUPPORTED_ARCHS "gfx942" "gfx942:xnack-" "gfx942:xnack+" "gfx950" "gfx950:xnack-" "gfx950:xnack+")
@@ -1125,11 +1120,6 @@ if(LL128_ENABLED)
   target_compile_definitions(rccl PRIVATE ENABLE_LL128)
 endif()

-## Set RCCL compile options
-if (HAVE_PARALLEL_JOBS)
-  target_compile_options(rccl PRIVATE -parallel-jobs=12)
-endif()
-
 if (ROCM_VERSION VERSION_GREATER_EQUAL "60200")
   target_compile_options(rccl PRIVATE --offload-compress)    # Compress GPU code at compile time.
   target_link_libraries(rccl PRIVATE --offload-compress)     # Compress GPU code at link time.
@@ -1280,15 +1270,6 @@ else()
     math(EXPR memory_in_gb "${memory_max_string} / 1024")
   endif()
 endif()
-## Reserve 16GB for each linker job. Limit max number of linker jobs to 16
-if (HAVE_PARALLEL_JOBS)
-  math(EXPR num_linker_jobs "(${memory_in_gb} + 15) / 16")
-  if (${num_linker_jobs} GREATER_EQUAL "16")
-    set(num_linker_jobs "16")
-  endif()
-  message(STATUS "Use ${num_linker_jobs} jobs for linking")
-  target_link_options(rccl PRIVATE -parallel-jobs=${num_linker_jobs})       # Use multiple threads to link
-endif()
 if(BUILD_ADDRESS_SANITIZER)
   target_link_options(rccl PRIVATE -fuse-ld=lld)
 endif()
