diff --git a/src/init.cc b/src/init.cc
index 689ea77764..055ae90537 100644
--- a/src/init.cc
+++ b/src/init.cc
@@ -77,7 +77,11 @@
 #include "latency_profiler/CollTraceFunc.h"
 #include "dda_all_reduce_ipc.h"
 #include "ipc_init.h"
+
+#if defined(__i386__) || defined(__x86_64__) || defined(_M_IX86) || defined(_M_X64)
+#define IS_X86_64
 #include  <cpuid.h>
+#endif
 
 #ifndef STR2
   #define STR2(v) #v
@@ -283,6 +287,7 @@ static ncclResult_t ncclInit() {
       if (verStr == NULL) break;
     }
     INFO(NCCL_INIT, "Kernel version: %s", verStr);
+#ifdef IS_X86_64
     if (strstr(verStr, "cray") == NULL) {
       unsigned int eax, ebx, ecx, edx;
       if (!__get_cpuid(1, &eax, &ebx, &ecx, &edx))
@@ -307,6 +312,7 @@ static ncclResult_t ncclInit() {
         WARN("Missing \"HSA_FORCE_FINE_GRAIN_PCIE=1\" from environment which can lead to low RCCL performance, system instablity or hang!");
 #endif
     }
+#endif
   std::call_once(initOnceFlag, initOnceFunc);
   return initResult;
 }
