diff --git a/src/init.cc b/src/init.cc
index a5fd77de92..c88eb0d9d4 100644
--- a/src/init.cc
+++ b/src/init.cc
@@ -72,7 +72,12 @@

 #include "latency_profiler/CollTrace.h"
 #include "latency_profiler/CollTraceFunc.h"
+#if defined(__i386__) || defined(__x86_64__) || defined(_M_IX86) || defined(_M_X64)
+#define IS_X86_64
+#endif
+#ifdef IS_X86_64
 #include  <cpuid.h>
+#endif

 #ifndef STR2
   #define STR2(v) #v
@@ -250,6 +255,7 @@ static ncclResult_t ncclInit() {
       if (verStr == NULL) break;
     }
     INFO(NCCL_INIT, "Kernel version: %s", verStr);
+#ifdef IS_X86_64
     if (strstr(verStr, "cray") == NULL) {
       unsigned int eax, ebx, ecx, edx;
       if (!__get_cpuid(1, &eax, &ebx, &ecx, &edx))
@@ -274,6 +280,7 @@ static ncclResult_t ncclInit() {
         WARN("Missing \"HSA_FORCE_FINE_GRAIN_PCIE=1\" from environment which can lead to low RCCL performance, system instablity or hang!");
 #endif
     }
+#endif
   std::call_once(initOnceFlag, initOnceFunc);
   return initResult;
 }
