From 92967853bceddf67f0773e222dccb19280d42a2d Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@suse.de>
Date: Sun, 16 Apr 2023 22:28:41 +0200
Subject: [PATCH] Default to rv64gc on riscv64

Fixes #4375
---
 driver/targetmachine.cpp | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/driver/targetmachine.cpp b/driver/targetmachine.cpp
index be47a1dc93..4668db1f74 100644
--- a/driver/targetmachine.cpp
+++ b/driver/targetmachine.cpp
@@ -120,7 +120,7 @@ const char *getABI(const llvm::Triple &triple) {
   case llvm::Triple::ppc64le:
     return "elfv2";
   case llvm::Triple::riscv64:
-    return "lp64";
+    return "lp64d";
   case llvm::Triple::riscv32:
     return "ilp32";
   default:
@@ -439,6 +439,15 @@ createTargetMachine(const std::string targetTriple, const std::string arch,
     features.push_back("+cx16");
   }
 
+  // Default to rv64gc if nothing has been selected
+  if (triple.getArch() == llvm::Triple::riscv64 && features.empty()) {
+    features.push_back("+m");
+    features.push_back("+a");
+    features.push_back("+f");
+    features.push_back("+d");
+    features.push_back("+c");
+  }
+
   // Handle cases where LLVM picks wrong default relocModel
 #if LDC_LLVM_VER >= 1600
   if (relocModel.has_value()) {}
