From e3447f501a86c91653fd9d7da45d4f4f433bc6de Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@suse.de>
Date: Sun, 16 Apr 2023 22:28:41 +0200
Subject: [PATCH 1/2] Default to rv64gc on riscv64

Fixes #4375
---
 driver/targetmachine.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/driver/targetmachine.cpp b/driver/targetmachine.cpp
index 73de806ac4..6aee1d199a 100644
--- a/driver/targetmachine.cpp
+++ b/driver/targetmachine.cpp
@@ -161,11 +161,7 @@ const char *getABI(const llvm::Triple &triple, const llvm::SmallVectorImpl<llvm:
   case llvm::Triple::ppc64le:
     return "elfv2";
   case llvm::Triple::riscv64:
-    if (isFeatureEnabled(features, "d"))
-      return "lp64d";
-    if (isFeatureEnabled(features, "f"))
-      return "lp64f";
-    return "lp64";
+    return "lp64d";
   case llvm::Triple::riscv32:
     return "ilp32";
 #if LDC_LLVM_VER >= 1600
-- 
2.44.0

