From 59d2d446d5526d93982a431c88279c2a46ee5491 Mon Sep 17 00:00:00 2001
From: WANG Rui <wangrui@loongson.cn>
Date: Tue, 14 May 2024 15:52:05 +0800
Subject: [PATCH 5/7] ABI: Update spec url for LoongArch

---
 gen/abi/loongarch64.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gen/abi/loongarch64.cpp b/gen/abi/loongarch64.cpp
index 413a4111b8..1c5d646c79 100644
--- a/gen/abi/loongarch64.cpp
+++ b/gen/abi/loongarch64.cpp
@@ -9,7 +9,7 @@
 //===----------------------------------------------------------------------===//
 //
 // ABI spec:
-// https://loongson.github.io/LoongArch-Documentation/LoongArch-ELF-ABI-EN.html
+// https://github.com/loongson/la-abi-specs/blob/release/lapcs.adoc
 //
 //===----------------------------------------------------------------------===//
 
-- 
2.45.1

