From fbf505daab151280c59d4049790904c7f29580f8 Mon Sep 17 00:00:00 2001
From: Johan Engelen <jbc.engelen@gmail.com>
Date: Sat, 25 Nov 2023 12:38:33 +0100
Subject: [PATCH 03/17] fixup

---
 gen/optimizer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gen/optimizer.cpp b/gen/optimizer.cpp
index 254436b8ec..843fbe2653 100644
--- a/gen/optimizer.cpp
+++ b/gen/optimizer.cpp
@@ -652,7 +652,7 @@ void runOptimizationPasses(llvm::Module *M) {
   StandardInstrumentations si(M->getContext(), debugLogging, /*VerifyEach=*/false, ppo);
 #endif
 
-#if LDC_LLVM_VER < 1600
+#if LDC_LLVM_VER < 1700
   si.registerCallbacks(pic, &fam);
 #else
   si.registerCallbacks(pic, &mam);
-- 
2.39.1

