From 95f9d7390d0bbb7bc5b24ebbbc41d7522cff055e Mon Sep 17 00:00:00 2001
From: Johan Engelen <jbc.engelen@gmail.com>
Date: Sat, 25 Nov 2023 14:15:34 +0100
Subject: [PATCH 06/17] fixup

---
 gen/optimizer.cpp | 44 ++++++++++++++++++++++----------------------
 1 file changed, 22 insertions(+), 22 deletions(-)

diff --git a/gen/optimizer.cpp b/gen/optimizer.cpp
index 843fbe2653..8eac6964a6 100644
--- a/gen/optimizer.cpp
+++ b/gen/optimizer.cpp
@@ -550,34 +550,34 @@ static void addGarbageCollect2StackPass(ModulePassManager &mpm,
 
 
 static llvm::Optional<PGOOptions> getPGOOptions() {
- //FIXME: Do we have these anywhere?
- bool debugInfoForProfiling=false;
- bool pseudoProbeForProfiling=false;
- if (opts::isInstrumentingForIRBasedPGO()) {
-    return PGOOptions(global.params.datafileInstrProf, "", "",
+  // FIXME: Do we have these anywhere?
+  bool debugInfoForProfiling = false;
+  bool pseudoProbeForProfiling = false;
+  if (opts::isInstrumentingForIRBasedPGO()) {
+    return PGOOptions(
+        global.params.datafileInstrProf, "", "",
 #if LDC_LLVM_VER >= 1700
-                      "" /*MemoryProfileUsePath*/, nullptr /*vfs::FileSystem*/,
+        "" /*MemoryProfileUsePath*/, llvm::vfs::getRealFileSystem(),
 #endif
-                      PGOOptions::PGOAction::IRInstr,
-                      PGOOptions::CSPGOAction::NoCSAction,
-                      debugInfoForProfiling, pseudoProbeForProfiling);
- } else if (opts::isUsingIRBasedPGOProfile()) {
-    return PGOOptions(global.params.datafileInstrProf, "", "",
+        PGOOptions::PGOAction::IRInstr, PGOOptions::CSPGOAction::NoCSAction,
+        debugInfoForProfiling, pseudoProbeForProfiling);
+  } else if (opts::isUsingIRBasedPGOProfile()) {
+    return PGOOptions(
+        global.params.datafileInstrProf, "", "",
 #if LDC_LLVM_VER >= 1700
-                      "" /*MemoryProfileUsePath*/, nullptr /*vfs::FileSystem*/,
+        "" /*MemoryProfileUsePath*/, llvm::vfs::getRealFileSystem(),
 #endif
-                      PGOOptions::PGOAction::IRUse,
-                      PGOOptions::CSPGOAction::NoCSAction,
-                      debugInfoForProfiling, pseudoProbeForProfiling);
- } else if (opts::isUsingSampleBasedPGOProfile()) {
-    return PGOOptions(global.params.datafileInstrProf, "", "",
+        PGOOptions::PGOAction::IRUse, PGOOptions::CSPGOAction::NoCSAction,
+        debugInfoForProfiling, pseudoProbeForProfiling);
+  } else if (opts::isUsingSampleBasedPGOProfile()) {
+    return PGOOptions(
+        global.params.datafileInstrProf, "", "",
 #if LDC_LLVM_VER >= 1700
-                      "" /*MemoryProfileUsePath*/, nullptr /*vfs::FileSystem*/,
+        "" /*MemoryProfileUsePath*/, llvm::vfs::getRealFileSystem(),
 #endif
-                      PGOOptions::PGOAction::SampleUse,
-                      PGOOptions::CSPGOAction::NoCSAction,
-                      debugInfoForProfiling, pseudoProbeForProfiling);
- }
+        PGOOptions::PGOAction::SampleUse, PGOOptions::CSPGOAction::NoCSAction,
+        debugInfoForProfiling, pseudoProbeForProfiling);
+  }
 #if LDC_LLVM_VER < 1600
   return None;
 #else
-- 
2.39.1

