From a3a8692675f2fd1abb57a948b9921926b7c18bfb Mon Sep 17 00:00:00 2001
From: Johan Engelen <jbc.engelen@gmail.com>
Date: Sat, 25 Nov 2023 14:21:10 +0100
Subject: [PATCH 07/17] fixup

---
 tests/plugins/addFuncEntryCall/addFuncEntryCallPass.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tests/plugins/addFuncEntryCall/addFuncEntryCallPass.cpp b/tests/plugins/addFuncEntryCall/addFuncEntryCallPass.cpp
index e138526d18..72899b0ca3 100644
--- a/tests/plugins/addFuncEntryCall/addFuncEntryCallPass.cpp
+++ b/tests/plugins/addFuncEntryCall/addFuncEntryCallPass.cpp
@@ -10,9 +10,12 @@
 #include "llvm/Pass.h"
 #include "llvm/IR/Function.h"
 #include "llvm/IR/IRBuilder.h"
-#include "llvm/IR/LegacyPassManager.h"
 #include "llvm/IR/Module.h"
+
+#if LLVM_VERSION < 1500 // legacy pass manager
+#include "llvm/IR/LegacyPassManager.h"
 #include "llvm/Transforms/IPO/PassManagerBuilder.h"
+#endif
 
 using namespace llvm;
 
-- 
2.39.1

