From b1eaeeaaf1def6c62a771dc85d2141b78e37a282 Mon Sep 17 00:00:00 2001
From: Johan Engelen <jbc.engelen@gmail.com>
Date: Sat, 25 Nov 2023 20:20:03 +0100
Subject: [PATCH 09/17] Replace deprecated function (LLVM17 opaque ptr) with
 equivalent call.

---
 gen/toir.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/gen/toir.cpp b/gen/toir.cpp
index 1fa7a7f5bd..c19d86c1b5 100644
--- a/gen/toir.cpp
+++ b/gen/toir.cpp
@@ -964,11 +964,13 @@ public:
     result = new DLValue(e->type, DtoBitCast(V, DtoPtrToType(e->type)));
   }
 
-  static llvm::PointerType * getWithSamePointeeType(llvm::PointerType *p, unsigned as) {
-#if LDC_LLVM_VER >= 1300
-    return llvm::PointerType::getWithSamePointeeType(p, as);
+  static llvm::PointerType * getWithSamePointeeType(llvm::PointerType *p, unsigned addressSpace) {
+#if LDC_LLVM_VER >= 1700
+    return llvm::PointerType::get(p->getContext(), addressSpace);
+#elif LDC_LLVM_VER >= 1300
+    return llvm::PointerType::getWithSamePointeeType(p, addressSpace);
 #else
-    return p->getPointerElementType()->getPointerTo(as);
+    return p->getPointerElementType()->getPointerTo(addressSpace);
 #endif
   }
 
-- 
2.39.1

