From 97e89bc494919244f75af1d87dc3745497ae7e88 Mon Sep 17 00:00:00 2001
From: Johan Engelen <jbc.engelen@gmail.com>
Date: Tue, 28 Nov 2023 23:25:14 +0100
Subject: [PATCH 11/17] Enable opaque pointers also for LLVM 16

---
 driver/cl_options.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/driver/cl_options.cpp b/driver/cl_options.cpp
index c2107e6dcf..3866c427c1 100644
--- a/driver/cl_options.cpp
+++ b/driver/cl_options.cpp
@@ -716,7 +716,7 @@ cl::opt<bool> dynamicCompileTlsWorkaround(
     cl::Hidden);
 #endif
 
-#if LDC_LLVM_VER >= 1700
+#if LDC_LLVM_VER >= 1600
 bool enableOpaqueIRPointers = true; // typed pointers are no longer supported from LLVM 17
 #elif LDC_LLVM_VER >= 1400
 bool enableOpaqueIRPointers = false;
-- 
2.39.1

