From 0aa40b91e0f4e8bd47d7194dbd63178a49f543de Mon Sep 17 00:00:00 2001
From: Johan Engelen <jbc.engelen@gmail.com>
Date: Fri, 1 Dec 2023 23:27:07 +0100
Subject: [PATCH 15/17] Silence the install_name_tool warning that we know is
 OK.

---
 CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 304ef21878..0874177d26 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -802,7 +802,12 @@ function(copy_compilerrt_lib llvm_lib_name ldc_lib_name fixup_dylib)
         set(ldc_lib_path ${PROJECT_BINARY_DIR}/lib${LIB_SUFFIX}/${ldc_lib_name})
         copy_and_rename_file(${llvm_lib_path} ${ldc_lib_path})
         if (fixup_dylib)
-            execute_process(COMMAND install_name_tool -id @rpath/${ldc_lib_name} ${ldc_lib_path})
+            execute_process(COMMAND install_name_tool -id @rpath/${ldc_lib_name} ${ldc_lib_path} ERROR_VARIABLE INSTALL_NAME_TOOL_STDERR)
+            if(${INSTALL_NAME_TOOL_STDERR} MATCHES "warning: changes being made to the file will invalidate the code signature")
+                # Eat the warning, it's ok.
+            else()
+                message(WARNING "install_name_tool stderr: ${INSTALL_NAME_TOOL_STDERR}")
+            endif()
             execute_process(COMMAND codesign --force -s - ${ldc_lib_path})
         endif()
         install(FILES ${ldc_lib_path} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
-- 
2.39.1

