From 37335aaad032fde5d9716d468fd83f168c65cf55 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Fri, 1 Dec 2023 23:28:48 -0800
Subject: [PATCH 17/17] gen/optimizer.cpp: drop unavailable
 opts::isUsingSampleBasedPGOProfile option

---
 gen/optimizer.cpp | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/gen/optimizer.cpp b/gen/optimizer.cpp
index 8eac6964a6..4e7af5921c 100644
--- a/gen/optimizer.cpp
+++ b/gen/optimizer.cpp
@@ -569,14 +569,6 @@ static llvm::Optional<PGOOptions> getPGOOptions() {
 #endif
         PGOOptions::PGOAction::IRUse, PGOOptions::CSPGOAction::NoCSAction,
         debugInfoForProfiling, pseudoProbeForProfiling);
-  } else if (opts::isUsingSampleBasedPGOProfile()) {
-    return PGOOptions(
-        global.params.datafileInstrProf, "", "",
-#if LDC_LLVM_VER >= 1700
-        "" /*MemoryProfileUsePath*/, llvm::vfs::getRealFileSystem(),
-#endif
-        PGOOptions::PGOAction::SampleUse, PGOOptions::CSPGOAction::NoCSAction,
-        debugInfoForProfiling, pseudoProbeForProfiling);
   }
 #if LDC_LLVM_VER < 1600
   return None;
-- 
2.39.1

