diff --git a/runtime/druntime/src/core/threadasm.S b/runtime/druntime/src/core/threadasm.S
index 596fb8cf31..796d6cc369 100644
--- a/runtime/druntime/src/core/threadasm.S
+++ b/runtime/druntime/src/core/threadasm.S
@@ -546,7 +546,7 @@ fiber_switchContext:
 .globl fiber_switchContext
 fiber_switchContext:
     .cfi_startproc
-    .cfi_undefined $ra
+    .cfi_undefined 1
     # reserve space on stack
     addi.d  $sp, $sp, -19 * 8
 
