diff --git a/nemo-preview/meson.build b/nemo-preview/meson.build
index df7d1c2..fc16740 100644
--- a/nemo-preview/meson.build
+++ b/nemo-preview/meson.build
@@ -22,7 +22,7 @@ gtk_dep = dependency('gtk+-3.0', version: '>=3.5.12')
 gtksourceview_dep = dependency('gtksourceview-4', version: '>=4.0.3')
 introspection_dep = dependency('gobject-introspection-1.0', version: '>=0.9.2')
 musicbrainz_dep = dependency('libmusicbrainz5')
-webkit_dep = dependency('webkit2gtk-4.1')
+webkit_dep = dependency('webkit2gtk-4.0')
 
 bindir = join_paths(get_option('prefix'), get_option('bindir'))
 datadir = join_paths(get_option('prefix'), get_option('datadir'))
diff --git a/nemo-preview/src/js/viewers/html.js b/nemo-preview/src/js/viewers/html.js
index 753fc98d..fc346b7b 100644
--- a/nemo-preview/src/js/viewers/html.js
+++ b/nemo-preview/src/js/viewers/html.js
@@ -26,7 +26,7 @@
  */
 
 imports.gi.versions.Gtk = '3.0';
-imports.gi.versions.WebKit2 = '4.1';
+imports.gi.versions.WebKit2 = '4.0';
 const Gtk = imports.gi.Gtk;
 const GLib = imports.gi.GLib;
 const GtkClutter = imports.gi.GtkClutter;
