From 28d5557db44b9e69129d7b1913511a0bf947bd83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=93=BA=E7=9B=96=E5=B4=BD?= <i@pugai.life>
Date: Sun, 28 Sep 2025 06:35:40 +0800
Subject: [PATCH] Use system glm

---
 meson.build | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 5e5bd4cdc7..4c1e1065a8 100644
--- a/meson.build
+++ b/meson.build
@@ -50,8 +50,7 @@ dep_x11 = dependency('x11')
 dep_wayland = dependency('wayland-client')
 vulkan_dep = dependency('vulkan')
 
-glm_proj = subproject('glm')
-glm_dep = glm_proj.get_variable('glm_dep')
+glm_dep = dependency('glm')
 stb_proj = subproject('stb')
 stb_dep = stb_proj.get_variable('stb_dep')
 
