From e8470296dd9ef579b3c3be4509024aba849c8e77 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Franti=C5=A1ek=20Zatloukal?= <fzatlouk@redhat.com>
Date: Sat, 10 Aug 2024 20:11:23 -0400
Subject: [PATCH] Allow to use system wlroots

---
 meson.build | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/meson.build b/meson.build
index b562ec3..c68f4c3 100644
--- a/meson.build
+++ b/meson.build
@@ -11,9 +11,6 @@ project(
 )
 
 fallbacks = get_option('force_fallback_for')
-if not (fallbacks.contains('wlroots') and fallbacks.contains('libliftoff') and fallbacks.contains('vkroots'))
-  error('!!!"force_fallback_for" is missing entries!!!\n\tPlease do not remove entries from force_fallback_for if you are packaging the project.\n\tWe pull in these projects at specific commits/forks/builds for a reason.\n\tIf you are not packaging, remove this line to continue.')
-endif
 
 add_project_arguments([
   '-DWLR_USE_UNSTABLE',
-- 
2.45.2
