diff --git a/XADMaster/lsar.m b/XADMaster/lsar.m
index 12fd742..0b29819 100644
--- a/XADMaster/lsar.m
+++ b/XADMaster/lsar.m
@@ -25,7 +25,7 @@
 #import "CSJSONPrinter.h"
 #import "CommandLineCommon.h"
 
-#define VERSION_STRING @"v1.10.7"
+#define VERSION_STRING @"v1.10.8"
 
 #define EntryDoesNotNeedTestingResult 0
 #define EntryIsNotSupportedResult 1
diff --git a/XADMaster/unar.m b/XADMaster/unar.m
index 18828c5..564978b 100644
--- a/XADMaster/unar.m
+++ b/XADMaster/unar.m
@@ -24,7 +24,7 @@
 #import "CommandLineCommon.h"
 #import "CSFileHandle.h"
 
-#define VERSION_STRING @"v1.10.7"
+#define VERSION_STRING @"v1.10.8"
 
 @interface Unarchiver:NSObject {}
 @end
