From d1ddeda7c9cdf93549d1a6d0da9d6ccc86ee4e88 Mon Sep 17 00:00:00 2001
From: Yao Zi <me@ziyao.cc>
Date: Thu, 23 Jul 2026 06:01:08 +0000
Subject: [PATCH 19/22] Select HAVE_SETMP

Signed-off-by: Yao Zi <me@ziyao.cc>
---
 arch/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/Kconfig b/arch/Kconfig
index 98e1e58c619..bf6520def20 100644
--- a/arch/Kconfig
+++ b/arch/Kconfig
@@ -112,6 +112,7 @@ config ARM
 
 config LOONGARCH
 	bool "LoongArch architecture"
+	select HAVE_SETJMP
 	select CREATE_ARCH_SYMLINK
 	select SUPPORT_OF_CONTROL
 	select OF_CONTROL
-- 
2.55.0

