From 7e30bce5b186ce098df42b03ebcf33f5c6639b3c Mon Sep 17 00:00:00 2001
From: Yao Zi <me@ziyao.cc>
Date: Thu, 23 Jul 2026 06:01:25 +0000
Subject: [PATCH 20/22] Clean up setjmp.h

Signed-off-by: Yao Zi <me@ziyao.cc>
---
 arch/loongarch/include/asm/setjmp.h | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/arch/loongarch/include/asm/setjmp.h b/arch/loongarch/include/asm/setjmp.h
index 295198a3896..4fec81fdc41 100644
--- a/arch/loongarch/include/asm/setjmp.h
+++ b/arch/loongarch/include/asm/setjmp.h
@@ -3,8 +3,8 @@
  * Copyright (C) 2024 Jiaxun Yang <jiaxun.yang@flygoat.com>
  */
 
-#ifndef _SETJMP_H_
-#define _SETJMP_H_
+#ifndef _ASM_SETJMP_H_
+#define _ASM_SETJMP_H_
 
 /*
  * This really should be opaque, but the EFI implementation wrongly
@@ -17,9 +17,4 @@ struct jmp_buf_data {
 	unsigned long ra;
 };
 
-typedef struct jmp_buf_data jmp_buf[1];
-
-int setjmp(jmp_buf jmp);
-void longjmp(jmp_buf jmp, int ret);
-
 #endif /* _SETJMP_H_ */
-- 
2.55.0

