From 53dcb5c750e8772f61661e1dd750a5c65b7058be Mon Sep 17 00:00:00 2001
From: elysia-best <el26yg@aosc.io>
Date: Sat, 21 Mar 2026 11:14:42 +0800
Subject: [PATCH] remove appimage target

Signed-off-by: elysia-best <el26yg@aosc.io>
---
 CMakeLists.txt | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2c34554..ccc3139 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -401,11 +401,6 @@ else()
             DESTINATION ${CMAKE_INSTALL_BINDIR}
             COMPONENT Desktop
         )
-        install(PROGRAMS misc/run-${PROJECT_NAME}
-            DESTINATION .
-            RENAME AppRun
-            COMPONENT AppImage
-        )
         install(FILES misc/${PROJECT_NAME_CAPITALIZED}.desktop
             DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/applications
             COMPONENT Desktop
-- 
2.52.0

