--- a/build.gradle	2019-11-03 12:30:22.784116983 -0700
+++ b/build.gradle	2019-11-03 12:30:49.827445418 -0700
@@ -929,6 +929,7 @@
                 headers = headerDir
                 output(ccOutput)
                 params.addAll(variantProperties.ccFlags)
+                params.add("-Wno-deprecated-declarations")
                 compiler = variantProperties.compiler
                 source(variantProperties.nativeSource)
                 cleanTask.delete ccOutput

