--- citra/CMakeLists.txt	2022-07-28 00:18:12.225302363 +0000
+++ citra.ffmpeg5/CMakeLists.txt	2022-07-28 00:18:37.442749836 +0000
@@ -235,9 +235,6 @@
     else()
         find_package(FFmpeg REQUIRED COMPONENTS avcodec)
     endif()
-    if ("${FFmpeg_avcodec_VERSION}" VERSION_LESS "57.48.101")
-        message(FATAL_ERROR "Found version for libavcodec is too low. The required version is at least 57.48.101 (included in FFmpeg 3.1 and later).")
-    endif()
 endif()
 
 if (ENABLE_FFMPEG_VIDEO_DUMPER)
