From 26701fe66aa7d2efc17042e2ec7004e4b0aee233 Mon Sep 17 00:00:00 2001
From: Jamie Nguyen <j@jamielinux.com>
Date: Fri, 5 Dec 2014 15:22:13 +0000
Subject: [PATCH 1/2] Improve solarized-dark colorscheme

---
 contrib/colorschemes/solarized-dark | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/contrib/colorschemes/solarized-dark b/contrib/colorschemes/solarized-dark
index d6cbb0d..4e9dc57 100644
--- a/contrib/colorschemes/solarized-dark
+++ b/contrib/colorschemes/solarized-dark
@@ -1,8 +1,14 @@
 # solarized color scheme for newsbeuter <http://www.newsbeuter.org/>
 # more on solarized: http://ethanschoonover.com/solarized
 
-color listnormal color244 color234
-color listfocus  color166 color235 
-color info       color136 color235 
-color background color244 color234
-color article    color244 color234
+color   article             default   default
+color   background          default   default
+color   info                default   black
+color   listfocus           black     yellow
+color   listfocus_unread    default   yellow
+color   listnormal          default   default
+color   listnormal_unread   white     default
+
+highlight article "^(Title):.*$"         blue    default
+highlight article "https?://[^ ]+"       red     default
+highlight article "\\[image\\ [0-9]+\\]" green   default
-- 
2.1.0

