--- a/src/java/org/apache/commons/httpclient/MultiThreadedHttpConnectionManager.java
+++ b/src/java/org/apache/commons/httpclient/MultiThreadedHttpConnectionManager.java
@@ -1025,7 +1025,7 @@
     private static class ConnectionSource {
         
         /** The connection pool that created the connection */
-        public ConnectionPool connectionPool;
+        public MultiThreadedHttpConnectionManager.ConnectionPool connectionPool;
 
         /** The connection's host configuration */
         public HostConfiguration hostConfiguration;
