diff -Naur systemsettings-5.25.0/categories/settings-personalization-regionalsettings.desktop systemsettings-5.25.0.cjk/categories/settings-personalization-regionalsettings.desktop
--- systemsettings-5.25.0/categories/settings-personalization-regionalsettings.desktop	2022-06-09 09:21:48.000000000 +0000
+++ systemsettings-5.25.0.cjk/categories/settings-personalization-regionalsettings.desktop	2022-06-16 13:33:54.686333823 +0000
@@ -31,9 +31,9 @@
 Name[ia]=Preferentias regional
 Name[id]=Pengaturan Regional
 Name[it]=Impostazioni regionali
-Name[ja]=地域の設定
+Name[ja]=地域の設定 (Regional Settings)
 Name[ka]=რეგიონის მორგება
-Name[ko]=지역 설정
+Name[ko]=지역 설정 (Regional Settings)
 Name[lt]=Regioninės nuostatos
 Name[ml]=പ്രാദേശിക സജ്ജീകരണങ്ങൾ
 Name[my]=ဒေသဆိုင်ရာဆက်တင်များ
@@ -60,5 +60,5 @@
 Name[uk]=Локалізація
 Name[vi]=Thiết lập vùng
 Name[x-test]=xxRegional Settingsxx
-Name[zh_CN]=区域设置
-Name[zh_TW]=區域設定
+Name[zh_CN]=区域设置 (Regional Settings)
+Name[zh_TW]=區域設定 (Regional Settings)
