diff -Naur systemsettings-5.9.0/categories/settings-workspace-desktopbehavior.desktop systemsettings-5.9.0.fix/categories/settings-workspace-desktopbehavior.desktop
--- systemsettings-5.9.0/categories/settings-workspace-desktopbehavior.desktop	2017-02-05 15:00:55.771615338 -0600
+++ systemsettings-5.9.0.fix/categories/settings-workspace-desktopbehavior.desktop	2017-02-05 15:00:11.290099262 -0600
@@ -48,5 +48,5 @@
 Name[tr]=Çalışma Alanı Davranışı
 Name[uk]=Поведінка стільниці
 Name[x-test]=xxDesktop Behaviorxx
-Name[zh_CN]=桌面行为
+Name[zh_CN]=桌面/n行为
 Name[zh_TW]=桌面行為
