diff -Naur a/categories/settings-personalization-regionalsettings.desktop b/categories/settings-personalization-regionalsettings.desktop
--- a/categories/settings-personalization-regionalsettings.desktop	2016-03-17 05:08:25.000000000 -0600
+++ b/categories/settings-personalization-regionalsettings.desktop	2016-03-27 00:07:28.925640781 -0600
@@ -25,8 +25,8 @@
 Name[hu]=Területi beállítások
 Name[id]=Pengaturan Regional
 Name[it]=Impostazioni regionali
-Name[ja]=地域の設定
-Name[ko]=지역 설정
+Name[ja]=地域の設定 (Localization)
+Name[ko]=지역 설정 (Localization)
 Name[lt]=Regioninės nuostatos
 Name[nb]=Regionale innstillinger
 Name[nds]=Regiooninstellen
@@ -48,5 +48,5 @@
 Name[tr]=Bölgesel Ayarlar
 Name[uk]=Локалізація
 Name[x-test]=xxRegional Settingsxx
-Name[zh_CN]=区域设置
-Name[zh_TW]=區域設定
+Name[zh_CN]=区域设置 (Localization)
+Name[zh_TW]=區域設定 (Localization)
