--- systemsettings-5.3.2/categories/settings-personalization-regionalsettings.desktop	2015-06-25 19:05:12.000000000 +0800
+++ systemsettings-5.3.2-changed/categories/settings-personalization-regionalsettings.desktop	2015-07-15 20:44:55.978686814 +0800
@@ -20,8 +20,8 @@
 Name[hu]=Területi beállítások
 Name[id]=Pengaturan Regional
 Name[it]=Impostazioni regionali
-Name[ja]=地域の設定
-Name[ko]=지역 설정
+Name[ja]=地域の設定 (Localization)
+Name[ko]=지역 설정 (Localization)
 Name[lt]=Regioniniai nustatymai
 Name[nb]=Regionale innstillinger
 Name[nds]=Regiooninstellen
@@ -41,5 +41,5 @@
 Name[tr]=Bölgesel Ayarlar
 Name[uk]=Локалізація
 Name[x-test]=xxRegional Settingsxx
-Name[zh_CN]=区域设置
-Name[zh_TW]=區域設定
+Name[zh_CN]=区域设置 (Localization)
+Name[zh_TW]=區域設定 (Localization)
