diff -Naur a/systemd/lxdm.service b/systemd/lxdm.service
--- a/systemd/lxdm.service	2012-10-03 02:27:02.000000000 -0500
+++ b/systemd/lxdm.service	2016-09-07 22:52:51.278300998 -0500
@@ -1,7 +1,7 @@
 [Unit]
 Description=LXDE Display Manager
 Conflicts=getty@tty1.service plymouth-quit.service
-After=systemd-user-sessions.service getty@tty1.service plymouth-quit.service
+After=systemd-user-sessions.service getty@tty1.service plymouth-quit.service multi-user.target
 
 [Service]
 ExecStart=/usr/sbin/lxdm
