From 7b0dd0eae675ea5e002953453283082f71ff0755 Mon Sep 17 00:00:00 2001
From: ShellWen <me@shellwen.com>
Date: Tue, 15 Jul 2025 02:23:30 +0800
Subject: [PATCH 1/2] relax dep version constraint to fix run error

---
 lolcat.gemspec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lolcat.gemspec b/lolcat.gemspec
index 633ba3f..929366b 100644
--- a/lolcat.gemspec
+++ b/lolcat.gemspec
@@ -13,7 +13,7 @@ Gem::Specification.new do |s|
 
   s.add_development_dependency "rake"
   s.add_dependency "paint", "~> 2.1"
-  s.add_dependency "optimist", "~> 3.0.0"
+  s.add_dependency "optimist", "~> 3.0"
   s.add_dependency "manpages", "~> 0.6.1"
 
   s.files         = `git ls-files`.split("\n")
-- 
2.50.0

