From 786336509d5699e888e0afd6ead3fea0d3c0769e Mon Sep 17 00:00:00 2001
From: ShellWen <me@shellwen.com>
Date: Tue, 15 Jul 2025 03:05:36 +0800
Subject: [PATCH 2/2] fix build error from tarball

---
 lolcat.gemspec | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lolcat.gemspec b/lolcat.gemspec
index 929366b..e44648d 100644
--- a/lolcat.gemspec
+++ b/lolcat.gemspec
@@ -16,8 +16,8 @@ Gem::Specification.new do |s|
   s.add_dependency "optimist", "~> 3.0"
   s.add_dependency "manpages", "~> 0.6.1"
 
-  s.files         = `git ls-files`.split("\n")
-  s.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
-  s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
+  s.files         = `find -type f`.split("\n")
+  s.test_files    = `echo`.split("\n")
+  s.executables   = `find bin -type f`.split("\n").map{ |f| File.basename(f) }
   s.require_paths = ["lib"]
 end
-- 
2.50.0

