Author: Daniel Baumann <mail@daniel-baumann.ch>
Description: Using ssl-cert for snakeoil certificate.

Index: trunk/vsftpd.conf
===================================================================
--- trunk.orig/vsftpd.conf
+++ trunk/vsftpd.conf
@@ -146,7 +146,10 @@ pam_service_name=vsftpd
 #
 # This option specifies the location of the RSA certificate to use for SSL
 # encrypted connections.
-rsa_cert_file=/etc/ssl/private/vsftpd.pem
+rsa_cert_file=/etc/ssl/certs/ssl-cert-snakeoil.pem
+rsa_private_key_file=/etc/ssl/private/ssl-cert-snakeoil.key
+ssl_enable=NO
+
 #
 # Uncomment this to indicate that vsftpd use a utf8 filesystem.
 #utf8_filesystem=YES
