diff --git a/injector.c b/injector.c
index 75848b5..bc6d0dd 100644
--- a/injector.c
+++ b/injector.c
@@ -168,8 +168,6 @@ search_mode_t mode=TUNNEL;
 void* packet_buffer;
 char* packet;
 
-static char stack[SIGSTKSZ];
-stack_t ss = { .ss_size = SIGSTKSZ, .ss_sp = stack, };
 
 struct {
 	uint64_t dummy_stack_hi[256];
@@ -1449,6 +1447,8 @@ int main(int argc, char** argv)
 	setvbuf(stderr, NULL, _IONBF, 0);
 	*/
 
+
+	stack_t ss = { .ss_size = SIGSTKSZ, .ss_sp = malloc(SIGSTKSZ), };
 	sigaltstack(&ss, 0);
 
 	initialize_ranges();
