From: jreiser
Date: 2018-08-25 19:53+0200
References: https://github.com/upx/upx/issues/207#issuecomment-415986002

---
 src/n2b_d.c |    2 +-
 src/n2e_d.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: ucl-1.03/src/n2b_d.c
===================================================================
--- ucl-1.03.orig/src/n2b_d.c
+++ ucl-1.03/src/n2b_d.c
@@ -101,7 +101,7 @@
             m_len += 2;
         }
         m_len += (m_off > 0xd00);
-        fail(olen + m_len > oend, UCL_E_OUTPUT_OVERRUN);
+        fail(olen + m_len + 1 > oend, UCL_E_OUTPUT_OVERRUN);
         fail(m_off > olen, UCL_E_LOOKBEHIND_OVERRUN);
 #ifdef TEST_OVERLAP
         olen += m_len + 1;
Index: ucl-1.03/src/n2e_d.c
===================================================================
--- ucl-1.03.orig/src/n2e_d.c
+++ ucl-1.03/src/n2e_d.c
@@ -109,7 +109,7 @@
             m_len += 3;
         }
         m_len += (m_off > 0x500);
-        fail(olen + m_len > oend, UCL_E_OUTPUT_OVERRUN);
+        fail(olen + m_len + 1 > oend, UCL_E_OUTPUT_OVERRUN);
         fail(m_off > olen, UCL_E_LOOKBEHIND_OVERRUN);
 #ifdef TEST_OVERLAP
         olen += m_len + 1;
