diff -up galculator-2.1.4/src/main.c.build galculator-2.1.4/src/main.c
--- galculator-2.1.4/src/main.c.build	2015-09-05 20:47:11.000000000 +0200
+++ galculator-2.1.4/src/main.c	2020-01-29 11:26:49.377343327 +0100
@@ -49,7 +49,6 @@
 
 #define MASK_NUMLOCK GDK_MOD2_MASK
 
-s_preferences		prefs;
 s_current_status 	current_status = {0, 0, 0, 0, FALSE, FALSE, TRUE};
 s_array				memory;
 s_constant 			*constant;
