From 61c0bb8c40823e07344c4fbdea7dbd2208d3124f Mon Sep 17 00:00:00 2001
From: MutsukiC <mutsuki@aosc.io>
Date: Wed, 3 Jun 2026 11:25:42 +0800
Subject: [PATCH 2/2] [UPSTREAM] Fix cookie header accounting against
 LimitRequestFields.

---
 modules/http2/h2_util.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules/http2/h2_util.c b/modules/http2/h2_util.c
index b377ff77f1..b3039451f3 100644
--- a/modules/http2/h2_util.c
+++ b/modules/http2/h2_util.c
@@ -1708,6 +1708,8 @@ static apr_status_t req_add_header(apr_table_t *headers, apr_pool_t *pool,
              && !ap_cstr_casecmpn("cookie", (const char *)nv->name, nv->namelen)) {
         existing = apr_table_get(headers, "cookie");
         if (existing) {
+            if (!nv->valuelen)
+                return APR_SUCCESS;
             /* Cookie header come separately in HTTP/2, but need
              * to be merged by "; " (instead of default ", ")
              */
@@ -1719,6 +1721,8 @@ static apr_status_t req_add_header(apr_table_t *headers, apr_pool_t *pool,
             apr_table_setn(headers, "Cookie",
                            apr_psprintf(pool, "%s; %.*s", existing,
                                         (int)nv->valuelen, nv->value));
+            /* Treat the merge as an "add" to not escape LimitRequestFields */
+            *pwas_added = 1;
             return APR_SUCCESS;
         }
     }
-- 
2.52.0

