diff -Naur ksystemlog-22.04.3/src/org.kde.ksystemlog.desktop ksystemlog-22.04.3.kdesu/src/org.kde.ksystemlog.desktop
--- ksystemlog-22.04.3/src/org.kde.ksystemlog.desktop	2022-07-02 03:23:41.000000000 +0000
+++ ksystemlog-22.04.3.kdesu/src/org.kde.ksystemlog.desktop	2022-09-17 05:37:00.629676449 +0000
@@ -5,7 +5,6 @@
 Terminal=false
 Categories=Qt;KDE;System;Monitor;
 X-DocPath=ksystemlog/index.html
-X-KDE-SubstituteUID=true
 Keywords=log;logfile;system messages;system log;kernel log;authentication log;X.org log;journald log;services logs;Event Viewer;eventvwr;eventlog;
 Keywords[ar]=سجل;رسائل النظام;رسائل;سجل النواة;سجل خادم العرض;عارض الأحداث;سجل الخدمات
 Keywords[az]=jurnal;jurnal faylı;sistem ismarıcı;sistem jurnalı;nüvə jurnalı;kimlik doğrulaması jurnalı;X.org jurnalı;journald jurnalı;xidmətlər jurnalı;Hadisələrə Baxış;eventvwr;hadisələr jurnalı;
