From dcd43c785f991932189809653c0006d711083ab2 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 16 Dec 2024 17:31:39 +0800
Subject: [PATCH 1/2] Fix pywatchman installation prefix and DESTDIR

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d0c7fc861..4776f8872 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -436,7 +436,7 @@ if(Python3_Development_FOUND)
       ${CMAKE_COMMAND} -E env
         CMAKE_CURRENT_SOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}
         ${Python3_EXECUTABLE} ${SETUP_PY} install
-        --root $ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}
+        --root $ENV{DESTDIR} --prefix ${CMAKE_INSTALL_PREFIX}
       WORKING_DIRECTORY ${PYWATCHMAN_BASE}
       RESULT_VARIABLE STATUS)
     if (NOT STATUS STREQUAL 0)
-- 
2.52.0

