From d6504f30fc8b9f97cb6569a08eaff6314dc01bd8 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 16 Dec 2024 17:31:39 +0800
Subject: [PATCH] Fix pywatchman installation prefix and DESTDIR

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9dfe4e3d9..b94837668 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -428,7 +428,7 @@ if(Python3_Development_FOUND)
       ${CMAKE_COMMAND} -E env
         CMAKE_CURRENT_SOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}
         ${Python3_EXECUTABLE} ${SETUP_PY} install
-        --root $ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}
+        --root $ENV{DESTDIR} --prefix ${CMAKE_INSTALL_PREFIX}
       WORKING_DIRECTORY ${PYWATCHMAN_BASE}
       RESULT_VARIABLE STATUS)
     if (NOT STATUS STREQUAL 0)
-- 
2.51.0

