From 642c599e48cc6dc0ff8656481942a168e7e915cf Mon Sep 17 00:00:00 2001
From: stydxm <stydxm@outlook.com>
Date: Tue, 2 Sep 2025 12:11:05 +0800
Subject: [PATCH 2/2] Update glog import logic for glog 0.7.1

---
 CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4776f8872..ab222ae6f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -342,7 +342,7 @@ endif()
 find_package(Gflags REQUIRED)
 include_directories(SYSTEM ${GFLAGS_INCLUDE_DIR})
 
-find_package(Glog REQUIRED)
+find_package(glog CONFIG QUIET)
 add_compile_definitions(GLOG_NO_ABBREVIATED_SEVERITIES)
 
 # We indirectly depend on boost.  This logic needs to match
@@ -488,7 +488,6 @@ target_link_libraries(third_party_deps INTERFACE
 )
 target_include_directories(third_party_deps INTERFACE
   ${FOLLY_INCLUDE_DIR}
-  ${GLOG_INCLUDE_DIR}
   ${GFLAGS_INCLUDE_DIR}
   ${Boost_INCLUDE_DIRS}
 )
-- 
2.52.0

