From ab58ed4c9faedd098c71f559958866f77af96e6a Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu@aosc.io>
Date: Thu, 9 Jul 2020 16:24:03 -0500
Subject: [PATCH] third_party: debundle sqlite3 soundtouch glew

---
 third_party/CMakeLists.txt | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index bf78dc11..48354231 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -41,12 +41,15 @@ add_library(nlohmann_json INTERFACE)
 target_include_directories(nlohmann_json INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/nlohmann_json)
 
 #sqlite
+find_package(SQLite3)
+if (NOT SQLite3_FOUND)
 add_library(sqlite3
     sqlite3/sqlite3.c
     sqlite3/sqlite3.h
     sqlite3/sqlite3ext.h
 )
 target_include_directories(sqlite3 PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/sqlite3)
+endif()
 
 #minimp3
 add_library(minimp3
@@ -57,6 +60,11 @@ add_library(minimp3
 target_include_directories(minimp3 PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/minimp3)
 
 #soundtouch
+find_package(PkgConfig)
+if (PKG_CONFIG_FOUND)
+  pkg_check_modules(SOUNDTOUCH IMPORTED_TARGET GLOBAL soundtouch)
+  add_library(soundtouch ALIAS PkgConfig::SOUNDTOUCH)
+else()
 add_library(soundtouch
     soundtouch/src/AAFilter.cpp
     soundtouch/src/AAFilter.h
@@ -90,8 +98,11 @@ add_library(soundtouch
 )
 target_include_directories(soundtouch PUBLIC soundtouch/include)
 target_include_directories(soundtouch PRIVATE soundtouch/src)
+endif()
 
 #GLEW
+find_package(GLEW)
+if (NOT GLEW_FOUND)
 add_library(GLEW
     glew/include/GL/glew.h
     glew/include/GL/glxew.h
@@ -100,4 +111,5 @@ add_library(GLEW
 )
 # GLEW is included statically and also doesn't need GLU(Which doesn't even exist on linux)
 target_compile_definitions(GLEW PUBLIC -DGLEW_NO_GLU -DGLEW_STATIC)
-target_include_directories(GLEW PUBLIC glew/include)
\ No newline at end of file
+target_include_directories(GLEW PUBLIC glew/include)
+endif()
-- 
2.27.0

