diff --git a/Audio/CMakeLists.txt b/Audio/CMakeLists.txt
index bc739b15..6b3498cb 100644
--- a/Audio/CMakeLists.txt
+++ b/Audio/CMakeLists.txt
@@ -38,7 +38,7 @@ target_include_directories(Audio PRIVATE
 )
 
 if(UNIX AND NOT APPLE)
-    target_compile_options(Audio PUBLIC -Wall -Wextra -Werror -Wno-unused-variable -Wno-unused-parameter -Wno-unused-function -Wno-unused-value) 
+    target_compile_options(Audio PUBLIC -Wall -Wextra -Wno-unused-variable -Wno-unused-parameter -Wno-unused-function -Wno-unused-value) 
     if(CMAKE_BUILD_TYPE STREQUAL "Release")
         target_compile_options(Audio PUBLIC -Wno-maybe-uninitialized) 
     endif(CMAKE_BUILD_TYPE STREQUAL "Release")
@@ -64,4 +64,4 @@ target_link_libraries(Audio cc-common)
 # Enable multiprocess compiling
 if(MSVC)
     target_compile_options(Audio PRIVATE /MP)
-endif(MSVC)
\ No newline at end of file
+endif(MSVC)
diff --git a/Graphics/CMakeLists.txt b/Graphics/CMakeLists.txt
index 15ffc968..301ebb7c 100644
--- a/Graphics/CMakeLists.txt
+++ b/Graphics/CMakeLists.txt
@@ -50,7 +50,7 @@ if(UNIX)
         find_library(OPEN_GL OpenGL)
         target_link_libraries(Graphics ${OPEN_GL})
     else(APPLE)
-        target_compile_options(Graphics PUBLIC -Wall -Wextra -Werror -Wno-unused-variable -Wno-unused-parameter -Wno-unused-function)
+        target_compile_options(Graphics PUBLIC -Wall -Wextra -Wno-unused-variable -Wno-unused-parameter -Wno-unused-function)
         target_link_libraries(Graphics GL)
         find_package(X11 REQUIRED)
         include_directories(${X11_INCLUDE_DIR})
@@ -62,4 +62,4 @@ target_link_libraries(Graphics cc-common)
 # Enable multiprocess compiling
 if(MSVC)
     target_compile_options(Graphics PRIVATE /MP)
-endif(MSVC)
\ No newline at end of file
+endif(MSVC)
diff --git a/Main/CMakeLists.txt b/Main/CMakeLists.txt
index a6fc7df9..28bfa853 100644
--- a/Main/CMakeLists.txt
+++ b/Main/CMakeLists.txt
@@ -79,7 +79,7 @@ if(MSVC)
 	set_target_properties(usc-game PROPERTIES LINK_FLAGS_DEBUG "/SUBSYSTEM:WINDOWS /DEBUG:FASTLINK")
 	set_target_properties(usc-game PROPERTIES LINK_FLAGS_RELEASE "/SUBSYSTEM:WINDOWS /DEBUG:FULL")
 elseif(NOT APPLE)
-    target_compile_options(usc-game PUBLIC -Wall -Wextra -Werror -Wno-unused-variable -Wno-unused-parameter -Wno-unused-function)
+    target_compile_options(usc-game PUBLIC -Wall -Wextra -Wno-unused-variable -Wno-unused-parameter -Wno-unused-function)
 endif(MSVC)
 
 # Dependencies
