diff --git a/libmalcontent-ui/meson.build b/libmalcontent-ui/meson.build
index 7d59df3..62e7d3d 100644
--- a/libmalcontent-ui/meson.build
+++ b/libmalcontent-ui/meson.build
@@ -104,6 +104,5 @@ if xmllint.found()
   )
 endif
 
-subdir('docs')
 # FIXME: Add tests
 #subdir('tests')
diff --git a/libmalcontent/meson.build b/libmalcontent/meson.build
index 7cf6375..18ae598 100644
--- a/libmalcontent/meson.build
+++ b/libmalcontent/meson.build
@@ -90,5 +90,4 @@ libmalcontent_gir = gnome.generate_gir(libmalcontent,
   dependencies: libmalcontent_dep,
 )
 
-subdir('docs')
-subdir('tests')
\ No newline at end of file
+subdir('tests')
diff --git a/meson.build b/meson.build
index e5b33ad..b529a68 100644
--- a/meson.build
+++ b/meson.build
@@ -146,11 +146,6 @@ add_test_setup('default',
   timeout_multiplier: 2,
 )
 
-dependency('gi-docgen',
-  version: '>= 2025.3',
-  native: true,
-)
-gidocgen = find_program('gi-docgen', native: true)
 api_docs_dir = datadir / 'doc'
 
 python_test_libraries_path = meson.project_source_root() / 'tests' / 'lib'
@@ -214,4 +209,3 @@ subdir('malcontent-webd-update')
 subdir('nss')
 subdir('pam')
 subdir('po')
-subdir('tests')
