From f4e7d2a6f7120b63e4be959910699ac841c60df6 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Fri, 24 May 2024 18:06:22 -0700
Subject: [PATCH] keep CEF binary file permissions

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 cmake/Modules/ObsHelpers.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/Modules/ObsHelpers.cmake b/cmake/Modules/ObsHelpers.cmake
index b4620ab56..1db7e9aa2 100644
--- a/cmake/Modules/ObsHelpers.cmake
+++ b/cmake/Modules/ObsHelpers.cmake
@@ -207,6 +207,7 @@ function(setup_target_browser target)
   install(
     DIRECTORY ${CEF_ROOT_DIR}/Release/
     DESTINATION ${OBS_PLUGIN_DESTINATION}
+    USE_SOURCE_PERMISSIONS
     COMPONENT ${target}_Runtime)
 
   install(
@@ -218,6 +219,7 @@ function(setup_target_browser target)
   install(
     DIRECTORY ${CEF_ROOT_DIR}/Release/
     DESTINATION ${OBS_OUTPUT_DIR}/$<CONFIG>/${OBS_PLUGIN_DESTINATION}
+    USE_SOURCE_PERMISSIONS
     COMPONENT obs_rundir
     EXCLUDE_FROM_ALL)
 endfunction()
-- 
2.45.1

