From 133b82225c37b41dbb079426a9cae612107081b3 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Wed, 18 Dec 2024 19:26:23 -0800
Subject: [PATCH] cmake: Add executable permission to CEF binary and libraries

Signed-off-by: Kaiyang Wu <self@origincode.me>
---
 cmake/linux/helpers.cmake | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/cmake/linux/helpers.cmake b/cmake/linux/helpers.cmake
index c0966e692..0b3918319 100644
--- a/cmake/linux/helpers.cmake
+++ b/cmake/linux/helpers.cmake
@@ -194,6 +194,7 @@ function(set_target_properties_obs target)
               "${cef_location}/libGLESv2.so"
               "${cef_location}/libvk_swiftshader.so"
               "${cef_location}/libvulkan.so.1"
+              "${cef_location}/snapshot_blob.bin"
               "${cef_location}/v8_context_snapshot.bin"
               "${cef_location}/vk_swiftshader_icd.json"
               "${cef_root_location}/Resources/chrome_100_percent.pak"
@@ -204,6 +205,17 @@ function(set_target_properties_obs target)
             COMPONENT Runtime
           )
 
+          install(
+            PROGRAMS
+              "${cef_location}/chrome-sandbox"
+              "${cef_location}/libEGL.so"
+              "${cef_location}/libGLESv2.so"
+              "${cef_location}/libvk_swiftshader.so"
+              "${cef_location}/libvulkan.so.1"
+            DESTINATION "${OBS_PLUGIN_DESTINATION}"
+            COMPONENT Runtime
+          )
+
           install(
             DIRECTORY "${cef_root_location}/Resources/locales"
             DESTINATION "${OBS_PLUGIN_DESTINATION}"
-- 
2.50.0

