--- a/Source/Pcap_DNSProxy/ReadParameter.cpp	2016-10-10 06:38:39.000000000 -0500
+++ b/Source/Pcap_DNSProxy/ReadParameter.cpp	2016-10-12 19:22:47.944690050 -0500
@@ -1587,7 +1587,7 @@
 			}
 		}
 	#endif
-		else if (Data.find("ListenProtocol=") == 0 && Data.length() > strlen("ListenProtocol="))
+		if (Data.find("ListenProtocol=") == 0 && Data.length() > strlen("ListenProtocol="))
 		{
 			CaseConvert(Data, true);
 
@@ -2519,7 +2519,7 @@
 				goto PrintDataFormatError;
 		}
 	#endif
-		else if ((Data.find("LocalMachineServerName=") == 0 && Data.length() > strlen("LocalMachineServerName=")) || 
+		if ((Data.find("LocalMachineServerName=") == 0 && Data.length() > strlen("LocalMachineServerName=")) || 
 			(Data.find("LocalhostServerName=") == 0 && Data.length() > strlen("LocalhostServerName="))) //Old version compatible support
 		{
 			size_t Offset = 0;
