--- a/Source/Pcap_DNSProxy/ReadParameter.cpp	2016-08-22 10:57:57.000000000 -0500
+++ b/Source/Pcap_DNSProxy/ReadParameter.cpp	2016-08-31 13:28:02.352121366 -0500
@@ -1826,7 +1826,7 @@
 			}
 		}
 	#endif
-		else if (Data.find("ListenProtocol=") == 0 && Data.length() > strlen("ListenProtocol="))
+		if (Data.find("ListenProtocol=") == 0 && Data.length() > strlen("ListenProtocol="))
 		{
 			CaseConvert(true, Data);
 
@@ -2762,7 +2762,7 @@
 				goto PrintDataFormatError;
 		}
 	#endif
-		else if (Data.find("LocalhostServerName=") == 0 && Data.length() > strlen("LocalhostServerName="))
+		if (Data.find("LocalhostServerName=") == 0 && Data.length() > strlen("LocalhostServerName="))
 		{
 			if (Data.length() > strlen("LocalhostServerName=") + DOMAIN_MINSIZE && Data.length() < strlen("LocalhostServerName=") + DOMAIN_DATA_MAXSIZE)
 			{
