From 4d6fd1244fa3685d2ad3fb08e0279460cdddf85e Mon Sep 17 00:00:00 2001
From: salieri <maliya355@outlook.com>
Date: Tue, 7 May 2024 17:35:02 +0800
Subject: [PATCH 4/4] let musicbox use python-3.8 native importlib metadata

---
 NEMbox/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/NEMbox/__init__.py b/NEMbox/__init__.py
index f038236..219b6fb 100644
--- a/NEMbox/__init__.py
+++ b/NEMbox/__init__.py
@@ -26,7 +26,7 @@ ________     __________________________  _____ _     _
 + ------------------------------------------ +
 
 """
-from importlib_metadata import version
+from importlib.metadata import version
 
 from .const import Constant
 from .utils import create_dir
-- 
2.34.1

