From 570180894dd56fdb7ba12827608a67f486c8317e Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 28 Feb 2024 21:52:54 +0800
Subject: [PATCH 01/22] [Fedora] QtBase: Fix build with libxkbcommon >= 1.6.0

---
 .../platformsupport/input/xkbcommon/qxkbcommon.cpp  | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/qtbase/src/platformsupport/input/xkbcommon/qxkbcommon.cpp b/qtbase/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
index f1e478274c..6d93768b10 100644
--- a/qtbase/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
+++ b/qtbase/src/platformsupport/input/xkbcommon/qxkbcommon.cpp
@@ -272,10 +272,15 @@ static constexpr const auto KeyTbl = qMakeArray(
         Xkb2Qt<XKB_KEY_dead_small_schwa,        Qt::Key_Dead_Small_Schwa>,
         Xkb2Qt<XKB_KEY_dead_capital_schwa,      Qt::Key_Dead_Capital_Schwa>,
         Xkb2Qt<XKB_KEY_dead_greek,              Qt::Key_Dead_Greek>,
-        Xkb2Qt<XKB_KEY_dead_lowline,            Qt::Key_Dead_Lowline>,
-        Xkb2Qt<XKB_KEY_dead_aboveverticalline,  Qt::Key_Dead_Aboveverticalline>,
-        Xkb2Qt<XKB_KEY_dead_belowverticalline,  Qt::Key_Dead_Belowverticalline>,
-        Xkb2Qt<XKB_KEY_dead_longsolidusoverlay, Qt::Key_Dead_Longsolidusoverlay>,
+        // These were removed in libxkbcommon 1.6.0
+        // XK_dead_lowline
+        Xkb2Qt<0xfe90,                          Qt::Key_Dead_Lowline>,
+        // XK_dead_aboveverticalline
+        Xkb2Qt<0xfe91,                          Qt::Key_Dead_Aboveverticalline>,
+        // XK_dead_belowverticalline
+        Xkb2Qt<0xfe92,                          Qt::Key_Dead_Belowverticalline>,
+        // XK_dead_longsolidusoverlay
+        Xkb2Qt<0xfe93,                          Qt::Key_Dead_Longsolidusoverlay>,
 
         // Special keys from X.org - This include multimedia keys,
         // wireless/bluetooth/uwb keys, special launcher keys, etc.
-- 
2.44.0

