From 1d784fef3189276ee8c68b470c67c1d88cea650f Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Tue, 13 Feb 2024 21:13:23 -0800
Subject: [PATCH 06/19] meta: add qtwebengine back into .gitmodules

---
 .gitmodules | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/.gitmodules b/.gitmodules
index eadc6bf481..28c57e4c12 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -286,3 +286,12 @@
 	url = ../qtquick3d.git
 	branch = 5.15
 	status = addon
+[submodule "qtwebengine"]
+	depends = qtdeclarative
+	recommends = qtquickcontrols qtquickcontrols2 qtlocation qtwebchannel qttools
+	path = qtwebengine
+	url = ../qtwebengine.git
+	branch = 5.15
+	status = addon
+	priority = 10
+
-- 
2.43.0

