diff --git a/qtlocation/src/3rdparty/mapbox-gl-native/platform/qt/src/bidi.cpp b/qtlocation/src/3rdparty/mapbox-gl-native/platform/qt/src/bidi.cpp
index b75d038..b367ee6 100644
--- a/qtlocation/src/3rdparty/mapbox-gl-native/platform/qt/src/bidi.cpp
+++ b/qtlocation/src/3rdparty/mapbox-gl-native/platform/qt/src/bidi.cpp
@@ -3,6 +3,7 @@
 #include <mbgl/text/bidi.hpp>
 
 #include <QString>
+#include <stdexcept>
 
 namespace mbgl {
 
diff -up a/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp.me a/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp
--- a/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp.me	2023-02-20 14:49:09.397384737 +0100
+++ b/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp	2023-02-20 14:50:09.654502801 +0100
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <cstdint>
+
 #include <mapbox/geometry/geometry.hpp>
 #include <mapbox/geometry/point_arithmetic.hpp>
 #include <mapbox/geometry/for_each_point.hpp>
diff -up a/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/string.hpp.me1 a/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/string.hpp
--- a/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/string.hpp.me1	2023-02-20 15:00:23.967862291 +0100
+++ b/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/string.hpp	2023-02-20 15:00:41.398058051 +0100
@@ -4,6 +4,7 @@
 #include <string>
 #include <cassert>
 #include <cstdlib>
+#include <cstdint>
 #include <exception>
 
 // Polyfill needed by Qt when building for Android with GCC
diff -up a/qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/gl/stencil_mode.hpp.me a/qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/gl/stencil_mode.hpp
--- a/qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/gl/stencil_mode.hpp.me	2023-02-20 15:12:32.276042434 +0100
+++ b/qtlocation/src/3rdparty/mapbox-gl-native/src/mbgl/gl/stencil_mode.hpp	2023-02-20 15:12:58.578501426 +0100
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdint>
 #include <mbgl/util/variant.hpp>
 
 namespace mbgl {
