diff --git a/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp b/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp
index a28c59a4..92d928a3 100644
--- a/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp
+++ b/qtlocation/src/3rdparty/mapbox-gl-native/include/mbgl/util/geometry.hpp
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdint>
 #include <mapbox/geometry/geometry.hpp>
 #include <mapbox/geometry/point_arithmetic.hpp>
 #include <mapbox/geometry/for_each_point.hpp>
