diff -Naur qt-5/qtwebkit/CMakeLists.txt qt-5.riscv64/qtwebkit/CMakeLists.txt
--- qt-5/qtwebkit/CMakeLists.txt	2024-01-01 02:13:25.316902468 -0800
+++ qt-5.riscv64/qtwebkit/CMakeLists.txt	2024-01-01 02:15:35.667691347 -0800
@@ -85,6 +85,8 @@
     set(WTF_CPU_S390 1)
 elseif (LOWERCASE_CMAKE_SYSTEM_PROCESSOR MATCHES "s390x")
     set(WTF_CPU_S390X 1)
+elseif (LOWERCASE_CMAKE_SYSTEM_PROCESSOR MATCHES "riscv64")
+    set(WTF_CPU_RISCV64 1)
 else ()
     message(FATAL_ERROR "Unknown CPU '${LOWERCASE_CMAKE_SYSTEM_PROCESSOR}'")
 endif ()
diff -Naur qt-5/qtwebkit/Source/JavaScriptCore/CMakeLists.txt qt-5.riscv64/qtwebkit/Source/JavaScriptCore/CMakeLists.txt
--- qt-5/qtwebkit/Source/JavaScriptCore/CMakeLists.txt	2024-01-01 02:13:25.318902473 -0800
+++ qt-5.riscv64/qtwebkit/Source/JavaScriptCore/CMakeLists.txt	2024-01-01 02:15:35.668691345 -0800
@@ -1288,6 +1288,7 @@
 elseif (WTF_CPU_MIPS)
 elseif (WTF_CPU_LOONGARCH64)
 elseif (WTF_CPU_SH4)
+elseif (WTF_CPU_RISCV64)
 elseif (WTF_CPU_X86)
 elseif (WTF_CPU_X86_64)
     if (MSVC AND ENABLE_JIT)
diff -Naur qt-5/qtwebkit/Source/WTF/wtf/Platform.h qt-5.riscv64/qtwebkit/Source/WTF/wtf/Platform.h
--- qt-5/qtwebkit/Source/WTF/wtf/Platform.h	2024-01-01 02:13:25.319902476 -0800
+++ qt-5.riscv64/qtwebkit/Source/WTF/wtf/Platform.h	2024-01-01 02:15:35.669691344 -0800
@@ -181,6 +181,11 @@
 #define WTF_CPU_LOONGARCH64 1
 #endif
 
+/* CPU(RISCV64) - RISCV64 */
+#if defined(__riscv) && defined(__riscv_xlen) && __riscv_xlen == 64
+#define WTF_CPU_RISCV64 1
+#endif
+
 /* CPU(ARM64) - Apple */
 #if (defined(__arm64__) && defined(__APPLE__)) || defined(__aarch64__)
 #define WTF_CPU_ARM64 1
@@ -713,7 +718,8 @@
     || CPU(LOONGARCH64) \
     || CPU(MIPS64) \
     || CPU(PPC64) \
-    || CPU(PPC64LE)
+    || CPU(PPC64LE) \
+    || CPU(RISCV64)
 #define USE_JSVALUE64 1
 #else
 #define USE_JSVALUE32_64 1
diff -Naur qt-5/qtwebkit/Source/WTF/wtf/dtoa/utils.h qt-5.riscv64/qtwebkit/Source/WTF/wtf/dtoa/utils.h
--- qt-5/qtwebkit/Source/WTF/wtf/dtoa/utils.h	2024-01-01 02:13:25.316902468 -0800
+++ qt-5.riscv64/qtwebkit/Source/WTF/wtf/dtoa/utils.h	2024-01-01 02:16:14.007626478 -0800
@@ -49,7 +49,7 @@
 defined(__ARMEL__) || \
 defined(_MIPS_ARCH_MIPS32R2)
 #define DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS 1
-#elif CPU(MIPS) || CPU(MIPS64) || CPU(PPC) || CPU(PPC64) || CPU(PPC64LE) || CPU(SH4) || CPU(S390) || CPU(S390X) || CPU(IA64) || CPU(ALPHA) || CPU(ARM64) || CPU(HPPA) || CPU(LOONGARCH64)
+#elif CPU(MIPS) || CPU(MIPS64) || CPU(PPC) || CPU(PPC64) || CPU(PPC64LE) || CPU(SH4) || CPU(S390) || CPU(S390X) || CPU(IA64) || CPU(ALPHA) || CPU(ARM64) || CPU(HPPA) || CPU(LOONGARCH64) || CPU(RISCV64)
 #define DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS 1
 #elif defined(_M_IX86) || defined(__i386__)
 #if defined(_WIN32)
