--- a/source/SConscript	2021-02-27 21:15:29.067551077 -0800
+++ b/source/SConscript	2021-02-27 21:15:48.876175804 -0800
@@ -50,7 +50,7 @@
 	else:
 		local_env.Append(LINKFLAGS = ['/SUBSYSTEM:WINDOWS', '/ENTRY:mainCRTStartup'], CPPDEFINES = ['XML_STATIC'])
 	objects.append(SConscript(['winres/SConscript'], exports='env'))
-elif local_env['BUILD_TARGET'] == 'linux2':
+elif local_env['BUILD_TARGET'] in ('linux', 'linux2'):
 	local_env.Append(LIBS=['rt', 'expat'])
 elif local_env['BUILD_TARGET'].startswith('gnu0'):
 	local_env.Append(LIBS=['rt', 'expat'])
