From 831132dbdd7657015b2344eb1a3b49bc8ee2aecb Mon Sep 17 00:00:00 2001
From: pch666777 <1005100717@qq.com>
Date: Sun, 26 Oct 2025 16:43:25 +0800
Subject: [PATCH] declare the weak function as noipa

---
 libXfont-1.5.4/src/stubs/stubs.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libXfont-1.5.4/src/stubs/stubs.h b/libXfont-1.5.4/src/stubs/stubs.h
index 9e71806..6fb9447 100644
--- a/libXfont-1.5.4/src/stubs/stubs.h
+++ b/libXfont-1.5.4/src/stubs/stubs.h
@@ -11,7 +11,9 @@
 
 /* this probably works for Mach-O too, but probably not for PE */
 #if defined(__ELF__) && defined(__GNUC__) && (__GNUC__ >= 3)
-#define weak __attribute__((weak))
+/* noipa needed to work around GCC PR95558 */
+/* The weak function should not be optimized */
+#define weak __attribute__((weak, noipa))
 #else
 #define weak
 #ifndef __SUNPRO_C /* Sun compilers use #pragma weak in .c files instead */
-- 
2.51.0

