--- DevIL/configure.ac	2021-02-20 00:06:16.886137290 -0600
+++ DevIL.unfucked/configure.ac	2021-02-20 00:05:44.224554873 -0600
@@ -156,35 +156,6 @@
 
 dnl AM_MAINTAINER_MODE
 
-dnl Now the phase of testing of extensions...
-TEST_EXT([Altivec],
-	 [-faltivec -maltivec],
-	 [vector int vec;vec=vec_splat_u32(0);],
-	 [ALTIVEC_GCC],
-	 [],
-	 [altivec])
-
-TEST_EXT([SSE],
-	 [-msse],
-	 [__m128 v;v=_mm_sub_ss(v,v);],
-	 [SSE],
-	 [#include <xmmintrin.h>],
-	 [sse])
-
-TEST_EXT([SSE2],
-	 [-msse2],
-	 [__m128i v;v=_mm_add_epi32(v,v);],
-	 [SSE2],
-	 [#include <emmintrin.h>],
-	 [sse2])
-
-TEST_EXT([SSE3],
-	 [-msse3],
-	 [__m128d v;v=_mm_addsub_pd(v,v);],
-	 [SSE3],
-	 [#include <pmmintrin.h>],
-	 [sse3])
-
 dnl Do we want to use ASM routines?
 AS_IF([test "x$enable_asm" = "xyes"],
       [TEST_ASM([i[[3456]]86*],
