From 07c56e1f31d9132911b71cccfc5d3fcae7917980 Mon Sep 17 00:00:00 2001
From: RawDiamondMC <RawDiamondMC@aosc.io>
Date: Sun, 21 Jun 2026 14:37:48 +0800
Subject: [PATCH 2/2] Change bug report link to aosc-os-abbs

Signed-off-by: RawDiamondMC <RawDiamondMC@aosc.io>
---
 desktop/src/main.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/desktop/src/main.rs b/desktop/src/main.rs
index 06fc6621d..933cbcff9 100644
--- a/desktop/src/main.rs
+++ b/desktop/src/main.rs
@@ -131,7 +131,7 @@ fn panic_hook(info: &PanicHookInfo) {
             params.push(("extra_info", extra_info.join("\n")));
         }
         if let Ok(url) = Url::parse_with_params(
-            "https://github.com/ruffle-rs/ruffle/issues/new?assignees=&labels=bug&template=crash_report.yml",
+            "https://github.com/AOSC-Dev/aosc-os-abbs/issues/new?&template=bug-report.yml",
             &params,
         ) {
             let _ = webbrowser::open(url.as_str());
-- 
2.55.0

