From a3dba89ec4083c5fadf4501a500d3b4799249000 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 5 Mar 2026 20:06:46 +0800
Subject: [PATCH 2/2] AOSCOS: disable AVX2 by default

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 build/arch.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/arch.mk b/build/arch.mk
index 6bea6007..e12ad2b7 100644
--- a/build/arch.mk
+++ b/build/arch.mk
@@ -1,5 +1,5 @@
 #for x86
-HAVE_AVX2 := Yes
+HAVE_AVX2 := No
 
 ifneq ($(filter %86 x86_64, $(ARCH)),)
 include $(SRC_PATH)build/x86-common.mk
-- 
2.52.0

