diff -Naur orca-42.3/orca-autostart.desktop.in orca-42.3.desktop/orca-autostart.desktop.in
--- orca-42.3/orca-autostart.desktop.in	2021-02-19 14:39:17.000000000 +0000
+++ orca-42.3.desktop/orca-autostart.desktop.in	2022-08-28 00:17:47.912484831 +0000
@@ -6,4 +6,3 @@
 AutostartCondition=GSettings org.gnome.desktop.a11y.applications screen-reader-enabled
 X-GNOME-AutoRestart=true
 #X-GNOME-Autostart-Phase=Initialization
-OnlyShowIn=GNOME;MATE;Unity;Cinnamon;
