diff -ur a/src/vdpau_decode.c b/src/vdpau_decode.c
--- a/src/vdpau_decode.c	2012-10-05 10:02:58.000000000 -0500
+++ b/src/vdpau_decode.c	2014-07-18 07:42:37.240166340 -0500
@@ -1289,7 +1289,7 @@
             driver_data,
             obj_context->vdp_decoder,
             obj_surface->vdp_surface,
-            (VdpPictureInfo)&obj_context->vdp_picture_info,
+            (VdpPictureInfo*)&obj_context->vdp_picture_info,
             obj_context->vdp_bitstream_buffers_count,
             obj_context->vdp_bitstream_buffers
         );

