--- lockdev-1.0.3.orig/debian/liblockdev1-dbg.docs
+++ lockdev-1.0.3/debian/liblockdev1-dbg.docs
@@ -0,0 +1 @@
+README.debug
--- lockdev-1.0.3.orig/debian/compat
+++ lockdev-1.0.3/debian/compat
@@ -0,0 +1 @@
+4
--- lockdev-1.0.3.orig/debian/changelog
+++ lockdev-1.0.3/debian/changelog
@@ -0,0 +1,341 @@
+lockdev (1.0.3-1.5) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Fix hurd-i386 FTBFS (Closes: #668951).
+
+ -- Samuel Thibault <sthibault@debian.org>  Fri, 08 Jun 2012 01:49:05 +0200
+
+lockdev (1.0.3-1.4) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Don't install copyright in both liblockdev1 and liblockdev1-dbg as
+    they share the same folde rin /usr/share/doc via symlink (Closes:
+    #563600)
+  * Use binary:Version instead of Source-Version
+
+ -- Christoph Egger <christoph@debian.org>  Fri, 19 Feb 2010 19:10:38 +0100
+
+lockdev (1.0.3-1.3) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Don't remove non-existant file (Closes: #549731)
+  * Add ${shlibs:Depends} to -perl and -db (Closes: #553240, #553242)
+  *  -dbg package is section debug
+  * add misc:Depends
+
+ -- Christoph Egger <christoph@debian.org>  Wed, 30 Dec 2009 20:50:40 +0100
+
+lockdev (1.0.3-1.2) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Brown paperbag release: Fix uninstallability of the -dbg-package.
+    Closes: #400676
+
+ -- Andreas Barth <aba@not.so.argh.org>  Tue, 28 Nov 2006 08:52:55 +0100
+
+lockdev (1.0.3-1.1) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Fix postinst. Closes: #366384
+    + don't use numerical compares for version history, but
+      dpkg --compare-versions
+    + do never wait for someone entering a key - that could be in
+      non-interactive mode ...
+  * Add upgrade information into NEWS.Debian.
+
+ -- Andreas Barth <aba@not.so.argh.org>  Mon, 27 Nov 2006 13:11:32 +0100
+
+lockdev (1.0.3-1) unstable; urgency=low
+
+  * New release
+  * Distinguish character and block devices when creating SVR4 type lockfiles
+    (closes: #331249)
+  * Fixed some typos in comments
+
+ -- Paweł Więcek <coven@debian.org>  Thu, 20 Apr 2006 21:11:48 +0200
+
+lockdev (1.0.2-1) unstable; urgency=low
+
+  * New release.
+  * debian/control:
+    + Added myself as an uploader.  Thanks to Pawel Wiecek for granting
+      me co-maintainership.
+    + Build-Depend upon debhelper 4.
+    + Update to Standards-Version 3.6.2.
+  * debian/copyright:
+    + Update FSF address.
+    + Add myself as a contributor.
+  * debian/compat: Use debhelper v4.
+  * Remove all postinst and preinst templates; debhelper v4 generates
+    ldconfig calls automatically.
+  * Devices within subdirectories of /dev are locked properly, using
+    ':' as a path delimiter in the device lock name.  For example,
+    /dev/pts/2 is locked as /var/lock/LCK..pts:2
+    (Closes: #115109, #135299).
+  * Device major and minor numbers are computed using GNU libc
+    builtins, rather than kernel headers (Closes: #331139).
+  * src/sample.c debugging is amended to use the proper debugging
+    functions (Closes: #331255).
+
+ -- Roger Leigh <rleigh@debian.org>  Mon,  3 Oct 2005 20:55:51 +0100
+
+lockdev (1.0.1-7) unstable; urgency=low
+
+  * Applied patch from Aurelien Jarno <aurel32@debian.org> to support
+    GNU/k*BSD (closes: #302457)
+  * Fixed patch from 1.0.1-5.1 NMU to clean after itself (!)
+
+ -- Pawel Wiecek <coven@debian.org>  Thu, 19 May 2005 21:25:43 +0200
+
+lockdev (1.0.1-6) unstable; urgency=low
+
+  * Applied patch from Kristof Koehler
+    <Kristof.Koehler@stud.uni-karlsruhe.de> to fix behavior when device is
+    locked by process with different UID (closes: #215807)
+
+ -- Pawel Wiecek <coven@debian.org>  Thu, 22 Jan 2004 14:34:28 +0100
+
+lockdev (1.0.1-5.1) unstable; urgency=low
+
+  * NMU
+  * Use -fPIC code in shared lib.  Closes: #190770
+
+ -- LaMont Jones <lamont@debian.org>  Sat,  7 Jun 2003 19:27:24 -0600
+
+lockdev (1.0.1-5) unstable; urgency=low
+
+  * Completely repackaged, so it now uses standard procedures instead of some
+    mysterious stuff (closes: #166600)
+  * Updated standards version
+  * Included perl 5.8 patch from -4.1 NMU (closes: #158941)
+  * Fixed a couple of lintian errors and warnings
+
+ -- Pawel Wiecek <coven@debian.org>  Fri, 25 Apr 2003 14:17:02 +0200
+
+lockdev (1.0.1-4.1) unstable; urgency=low
+
+  * NMU for perl 5.8.
+  * Bump perl build-dependency to >= 5.8.0-7.
+
+ -- Colin Watson <cjwatson@debian.org>  Sat, 31 Aug 2002 00:29:56 +0100
+
+lockdev (1.0.1-4) unstable; urgency=low
+
+  * New maintainer (closes: #112558)
+  * Removed umask calls (closes: #115108)
+  * Now calls ldconfig in postrm instead of prerm
+
+ -- Pawel Wiecek <coven@debian.org>  Tue,  9 Apr 2002 14:39:54 +0200
+
+lockdev (1.0.1-3) unstable; urgency=low
+
+  * debian/copyright: upstream location changed back to the Debian ftp
+    archives
+  * debian/control (Standards-Version): bumped to 3.5.6
+    (Maintainer): set to Debian QA Team
+  * debian/policy: standards-version bumped to 3.5.6
+
+ -- Gergely Nagy <algernon@debian.org>  Sun, 16 Sep 2001 22:34:51 +0200
+
+lockdev (1.0.1-2) unstable; urgency=low
+
+  * Added missing symlinks to some manual pages (Closes: #99549)
+  * Standards-Version bumped to 3.5.5
+  * Only call ldconfig upon configure and remove
+
+ -- Gergely Nagy <algernon@debian.org>  Sun, 10 Jun 2001 09:41:20 +0200
+
+lockdev (1.0.1-1) unstable; urgency=low
+
+  * New maintainer
+  * Acknowledge NMU: Closes: #41522
+  * Recompiled with new perl (Closes: #80662)
+  * Corrected missing symlink bug (Closes: #88888)
+  * Updated to latest perl policy (Closes: #95410)
+  * Added build-depends
+  * Bumped Standards-Version to 3.5.4
+
+ -- Gergely Nagy <algernon@debian.org>  Wed, 23 May 2001 17:24:23 +0200
+
+lockdev (1.0.0) unstable; urgency=low
+
+  * added management of pseudo device /dev/tty , thanx to Alan Cox.
+  * added unconditional malloc of supplied device argoment.
+  * changed name of the shared lib to end in .so
+  * changed Makefile.PL because the perl manpage disappeared.
+  * Changed symbols in the API as suggested by Daniel Quinlan and 
+    Erik Troan. Changed soname accordingly.
+  * Corrected manpages and added LSB proposal (draft 2).
+  * corrected cua device management: use linux/major.h to determine the
+    cua major number; conditioned to __linux__
+  * Changed LICENSE file to hold the LGPL 2.1
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Wed,  1 Dec 1999 12:59:10 +0200
+
+lockdev (0.11.1) unstable; urgency=low
+
+  * NMU for the perl upgrade. Closes: #41522
+  * Corrected the rules files to conform to perl policy 1.0.1.
+  * Compiled with perl-5.005.
+  * Corrected the location of the license.
+  * Upgraded standards-version to 2.5.1
+
+ -- Raphael Hertzog <rhertzog@hrnet.fr>  Mon, 19 Jul 1999 20:51:00 +0200
+
+lockdev (0.11) unstable; urgency=low
+
+  * Avoided repetute locking of the semaphore.
+  * Corrected special case of semaphore misbehaviour (found while
+    crosschecking between two instancies running under gdb).
+  * Added modification on-the-fly of major number of device from 4 to 5
+    in case it was referring to one of the 64 cua devices; it now locks
+    the major+minor numbers of the corresponding ttyS device. The name
+    remains unaffected, as it could be an arbitrary name.
+    (Thanx to Daniel Quinlan <quinlan@transmeta.com>)
+  * Tryed to use fcntl instead of flock, after suggestion of Daniel
+    Quinlan, to permit use on NFS mounted partitions.
+    I wasn't able to make it correctly work, so I conditioned the
+    changes and left the code there, for further (future)
+    investigations. Any help highly appreciated.
+  * changed third file (LCK..<pid>) to have three dots, and made it stay
+    there, as a quick reference. It doesn't hurt, and shouw the pid with
+    a simple ls -l
+  * Added full text of LGPL licence instead of GPL one added in the
+    previous version (oops).
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Wed,  2 Jun 1999 15:58:03 +0300
+
+lockdev (0.10) unstable; urgency=low
+
+  * Added ttylock.h as Compatibility API to AIX. Programs written for
+    that environment can be linked directly to lockdev library.
+    (On AIX the symbols are in libc, so special care should be used)
+  * changed location of manpages to /usr/share/man.
+  * changed the name of the source to lockdev_0.10.tar.gz
+  * changed address of FSF in Copyright file, added the full GPL text as
+    LICENSE (only in the source), corrected name of the mailing list.
+  * changed name of changelog, just to make lintian happier.
+  * corrected name of docs dir for perl package.
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Thu,  8 Apr 1999 21:04:53 +0300
+
+liblockdev (0.9a) frozen unstable; urgency=low
+
+  * Added more tests in postinst of liblockdev0g-dbg, to avoid strange
+    problems encountered by Adam P. Harris (closes: #22565)
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Mon, 18 May 1998 15:21:12 +0300
+
+liblockdev (0.9) frozen unstable; urgency=low
+
+  * cleaned symlinks in docs dirs.
+  * changed libs symlinks from /usr/lib to /lib
+  * stripped main shared library.
+  * Corrected wrong shlibs dependency (thanx to Hamish Moffatt, 
+    closes: #22282).
+  * Changed dependency of -dev from libc6 to libc6-dev, 
+    thanx to James Troup (closes: #16476).
+  * Added -I.. in Makefile.PL to allow compilation of the perl part even
+    when the -dev package isn't installed (thanx to Martin Schulze for
+    the patch) closes: #15834
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Tue, 12 May 1998 13:39:27 +0300
+
+liblockdev (0.8) unstable; urgency=low
+
+  * updated manpage (degugging section).
+  * added environment variable to set debug value, and signals 
+    to change those values at runtime (if compiled with -DDEBUG).
+  * avoided symlink to /usr/src changing .stab in asm source 
+    file (overwritten implicit rule to produce asm source).
+    Thanx to Martin Schulze <joey> for the help.
+  * changed dbg package to install shared debug lib in 
+    /usr/lib/debug and profile lib in /usr/lib/profile.
+  * splitted debian/rules in included debian/policy
+  * cleaned Makefile and changed to build debug shared lib.
+    Thanx to Yann Dirson.
+  * added change in umask value to allow full permissions to lock 
+    files.  (this is questionable: would it be better to add a 
+    suid program to check and remove dangling locks?)
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Tue, 23 Dec 1997 11:26:10 +0200
+
+liblockdev (0.7) unstable; urgency=low
+
+  * cleaned Makefile and debian/rules
+  * change in Makefile.PL to allow pristine compilation on new systems
+    (closes: #15834, thanx to Martin Schulze).
+  * added -dbg package with debug and profile libs.
+  * problem with lenght of dev_t in glibc for printf: cast needed :-)
+  * corrected ftp address in copyright file.
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Sun, 30 Nov 1997 01:00:50 +0200
+
+liblockdev (0.6) unstable; urgency=low
+
+  * added symlink to dirs in /usr/doc (one dir for all packages),
+     now /usr/doc/liblockdev is a symlink installed by -dev package,
+     as the liblockdev.a and liblockdev.so symlinks.
+     Instead, lblockdev-perl installs a copy of the docs because
+     it doesn't depend on the run-time binary package (as -dev does).
+  * cleaned spreading in the distribution: 
+     source and run-time to section 'libs', development into 'devel',
+     perl module into 'interpreters' (waiting for a perl section).
+  * created extension perl library LockDev:: with manpage LockDev(3pm)
+     and packaged as liblockdev0-perl.
+  * moved symlink so.0 -> so.0.6 to runtime package.
+  * added unlink of semaphore file (not really needed, just to clean up)
+  * corrected error in Synopsis of manpage (oops).
+  * cleaned source code: changed ANSI-C definition of functions to K&R,
+     as recommended in GNU coding style (but not for indentation).
+     Added ANSI-C prototypes for internal functions (exported one 
+     already have them in .h interface file.
+  * cleaned rules file, got rid of unneeded libc5 compatibility.
+  * Moved back include of stdio.h from lockdev.h to .c (closes:Bug#14154)
+     Thanx to Juan Cespedes <cespedes@debian.org>
+  * Corrected debian/rules to have binary-arch rule (closes:Bug#14062)
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Wed, 22 Oct 1997 13:45:34 +0300
+
+liblockdev (0.5) unstable; urgency=low
+
+  * moved to libc6 (glibc2); added a "g" after soname in package name.
+  * added rules to build also compatibility libc5 libs.
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Thu, 25 Sep 1997 11:51:22 +0300
+
+liblockdev0 (0.4) unstable; urgency=low
+
+  * Changed priority from extra to optional.
+  * cleaned Makefile and added separate static/debug library.
+  * Cleaned include file.
+  * Added global var liblockdev_debug for debugging level
+  * Added 2 functions to set and reset debugging level (only in static lib)
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Sat,  6 Sep 1997 02:20:01 +0300
+
+liblockdev0 (0.3) unstable; urgency=low
+
+  * avoided stripping of static lib (compiled with -DDEBUG)
+  * added semaphore lock to avoid race condition in
+    removing stale locks (thanx to Barak Pearlmutter)
+  * changed unlink of stale lock to a complex and
+    convoluted way to avoid conflicts between us and another
+    process that doesn't use this library. Still exists the
+    possibility of a conflict between two other processes that
+    don't use this library, but we are safe.
+  * cleaned code in lock_dev()
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Fri,  5 Sep 1997 11:05:21 +0300
+
+liblockdev0 (0.2) unstable; urgency=low
+
+  * fixed shared lib links.
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Thu,  4 Sep 1997 19:08:38 +0300
+
+liblockdev0 (0.1) unstable; urgency=low
+
+  * Initial Release.
+
+ -- Fabrizio Polacco <fpolacco@debian.org>  Thu,  4 Sep 1997 14:13:23 +0300
--- lockdev-1.0.3.orig/debian/rules
+++ lockdev-1.0.3/debian/rules
@@ -0,0 +1,145 @@
+#!/usr/bin/make -f
+# Sample debian/rules that uses debhelper.
+# GNU copyright 1997 to 1999 by Joey Hess.
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+
+ifndef PERL
+PERL = /usr/bin/perl
+endif
+
+archlib = `$(PERL) -MConfig -e 'print $$Config{installarchlib}'`
+config  = INSTALLDIRS=vendor
+config += INSTALLMAN1DIR=${d_man}/man1
+config += INSTALLMAN3DIR=${d_man}/man3
+config += INSTALLPRIVLIB=${d_ulib}/perl5
+config += INSTALLARCHLIB=${d_build}$(archlib)
+PERLVERSION=`$(PERL) -MConfig -e 'print $$Config{version}'`
+
+
+
+CFLAGS = -Wall -g
+
+ifneq (,$(findstring noopt,$(DEB_BUILD_OPTIONS)))
+	CFLAGS += -O0
+else
+	CFLAGS += -O2
+endif
+ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
+	INSTALL_PROGRAM += -s
+endif
+
+# shared library versions, option 1
+version=1.0.3
+major=1
+# option 2, assuming the library is created as src/.libs/libfoo.so.2.0.5 or so
+#version=`ls src/.libs/lib*.so.* | \
+# awk '{if (match($$0,/[0-9]+\.[0-9]+\.[0-9]+$$/)) print substr($$0,RSTART)}'`
+#major=`ls src/.libs/lib*.so.* | \
+# awk '{if (match($$0,/\.so\.[0-9]+$$/)) print substr($$0,RSTART+4)}'`
+
+configure: configure-stamp
+configure-stamp:
+	dh_testdir
+	# Add here commands to configure the package.
+
+	touch configure-stamp
+
+
+build: build-stamp
+build-stamp: configure-stamp 
+	dh_testdir
+
+	# Add here commands to compile the package.
+	$(MAKE)
+
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp configure-stamp
+
+	# Add here commands to clean up after the build process.
+	-$(MAKE) distclean
+
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_clean -k
+	dh_installdirs
+
+	# Add here commands to install the package into debian/
+	$(MAKE) install_run basedir=$(CURDIR)/debian/liblockdev1/usr
+	$(MAKE) install_debug basedir=$(CURDIR)/debian/liblockdev1-dbg/usr
+	$(MAKE) install_profile basedir=$(CURDIR)/debian/liblockdev1-dbg/usr
+	$(MAKE) install_dev basedir=$(CURDIR)/debian/liblockdev1-dev/usr
+	cd LockDev && $(MAKE) pure_install $(config) PREFIX=$(CURDIR)/debian/liblockdev1-perl/usr
+
+	mv $(CURDIR)/debian/liblockdev1-perl/usr/lib/perl5/LockDev.pm \
+	  $(CURDIR)/debian/liblockdev1-perl/usr/share/perl5
+
+	ln -s lockdev.3.gz \
+	  $(CURDIR)/debian/liblockdev1-dev/usr/share/man/man3/dev_lock.3.gz
+	ln -s lockdev.3.gz \
+	  $(CURDIR)/debian/liblockdev1-dev/usr/share/man/man3/dev_unlock.3.gz
+	ln -s lockdev.3.gz \
+	  $(CURDIR)/debian/liblockdev1-dev/usr/share/man/man3/dev_relock.3.gz
+	ln -s lockdev.3.gz \
+	  $(CURDIR)/debian/liblockdev1-dev/usr/share/man/man3/dev_testlock.3.gz
+	ln -s lockdev.3.gz \
+	  $(CURDIR)/debian/liblockdev1-dev/usr/share/man/man3/liblockdev.3.gz
+	ln -s liblockdev.$(version).so \
+	  $(CURDIR)/debian/liblockdev1-dev/usr/lib/liblockdev.so
+	ln -s liblockdev.$(version).so \
+	  $(CURDIR)/debian/liblockdev1/usr/lib/liblockdev.so.$(major)
+	
+	ln -s liblockdev1 \
+	  $(CURDIR)/debian/liblockdev1-dev/usr/share/doc/liblockdev1-dev
+	ln -s liblockdev1 \
+	  $(CURDIR)/debian/liblockdev1-dbg/usr/share/doc/liblockdev1-dbg
+
+
+# Build architecture-independent files here.
+binary-indep: build install
+# We have nothing to do by default.
+
+# Build architecture-dependent files here.
+binary-arch: build install
+	dh_testdir
+	dh_testroot
+	dh_installchangelogs ChangeLog
+	dh_installdocs
+	rm $(CURDIR)/debian/liblockdev1-dbg/usr/share/doc/liblockdev1/c*
+#	rm $(CURDIR)/debian/liblockdev1-dbg/usr/share/doc/liblockdev1/NEWS.Debia*
+	dh_installexamples
+#	dh_install
+#	dh_installmenu
+#	dh_installdebconf	
+#	dh_installlogrotate
+#	dh_installemacsen
+#	dh_installpam
+#	dh_installmime
+#	dh_installinit
+#	dh_installcron
+#	dh_installinfo
+	dh_installman
+	dh_link
+	dh_strip -Xprofile
+	dh_compress
+	dh_fixperms
+	dh_perl
+#	d_python
+	dh_makeshlibs -X usr/lib/debug -X usr/lib/profile
+	dh_installdeb
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install configure
--- lockdev-1.0.3.orig/debian/liblockdev1-dev.manpages
+++ lockdev-1.0.3/debian/liblockdev1-dev.manpages
@@ -0,0 +1 @@
+docs/lockdev.3
--- lockdev-1.0.3.orig/debian/liblockdev1-dbg.dirs
+++ lockdev-1.0.3/debian/liblockdev1-dbg.dirs
@@ -0,0 +1 @@
+usr/share/doc/liblockdev1
--- lockdev-1.0.3.orig/debian/control
+++ lockdev-1.0.3/debian/control
@@ -0,0 +1,55 @@
+Source: lockdev
+Section: libs
+Priority: optional
+Maintainer: Paweł Więcek <coven@debian.org>
+Uploaders: Roger Leigh <rleigh@debian.org>
+Standards-Version: 3.6.2
+Build-Depends: perl (>= 5.8.0-7), debhelper (>= 4.0.0)
+
+Package: liblockdev1
+Section: libs
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Description: Run-time shared library for locking devices
+ It provides a reliable way to put an exclusive lock to
+ devices using _both_ FSSTND and SVr4 methods.
+
+Package: liblockdev1-dev
+Section: libdevel
+Architecture: any
+Depends: liblockdev1 (= ${binary:Version}), libc6-dev, ${misc:Depends}
+Conflicts: liblockdev0-dev, liblockdev0g-dev
+Replaces: liblockdev0-dev, liblockdev0g-dev
+Description: Development library for locking devices
+ It provides a reliable way to put an exclusive lock to
+ devices using _both_ FSSTND and SVr4 methods.
+ .
+ Includes header files and static lib.
+
+Package: liblockdev1-dbg
+Section: debug
+Priority: extra
+Architecture: any
+Depends: liblockdev1 (= ${binary:Version}), liblockdev1-dev (= ${binary:Version}), ${shlibs:Depends}, ${misc:Depends}
+Conflicts: liblockdev0-dbg, liblockdev0g-dbg
+Replaces: liblockdev0-dbg, liblockdev0g-dbg
+Description: Debugging library for locking devices
+ It provides a reliable way to put an exclusive lock to
+ devices using _both_ FSSTND and SVr4 methods.
+ .
+ Includes static and shared debugging and profiling libs,
+ as well as the sources ready for use by gdb.
+
+Package: liblockdev1-perl
+Section: perl
+Architecture: any
+Depends: ${perl:Depends}, ${shlibs:Depends}, ${misc:Depends}
+Conflicts: liblockdev0-perl
+Replaces:  liblockdev0-perl
+Description: perl extension library for locking devices
+ It provides a reliable way to put an exclusive lock to
+ devices using _both_ FSSTND and SVr4 methods.
+ .
+ Includes a shared lib loadable by perl 5 (use LockDev),
+ the module LockDev.pm and its manpage LockDev(3pm) .
+
--- lockdev-1.0.3.orig/debian/liblockdev1-perl.dirs
+++ lockdev-1.0.3/debian/liblockdev1-perl.dirs
@@ -0,0 +1 @@
+usr/share/perl5
--- lockdev-1.0.3.orig/debian/liblockdev1-dev.dirs
+++ lockdev-1.0.3/debian/liblockdev1-dev.dirs
@@ -0,0 +1,2 @@
+usr/share/man/man3
+usr/share/doc
--- lockdev-1.0.3.orig/debian/NEWS
+++ lockdev-1.0.3/debian/NEWS
@@ -0,0 +1,8 @@
+lockdev (1.0.3-1.1) unstable; urgency=low
+
+  The format of device lock files has changed.
+
+  All programs that have locked device files using the old version of
+  liblockdev needs to be restarted.
+
+ -- Andreas Barth <aba@not.so.argh.org>  Mon, 27 Nov 2006 13:13:37 +0100
--- lockdev-1.0.3.orig/debian/liblockdev1.postinst
+++ lockdev-1.0.3/debian/liblockdev1.postinst
@@ -0,0 +1,22 @@
+#!/bin/sh -e
+
+if [ "$1" = "configure" ] && dpkg --compare-versions "$2" lt-nl 1.0.3-1;
+then
+  echo
+  echo "WARNING"
+  echo "Format of device lock files have changed, you will need to restart all"
+  echo "programs that have locked device files using the old version of liblockdev."
+  echo
+
+  if ls /var/lock/LCK.[0-9]* 2>/dev/null
+  then
+    echo "Right now the following programs use old lockfiles:"
+    echo
+    ps -p `cat /var/lock/LCK.[0-9]*`
+    echo
+  fi
+fi
+
+#DEBHELPER#
+
+exit 0
--- lockdev-1.0.3.orig/debian/copyright
+++ lockdev-1.0.3/debian/copyright
@@ -0,0 +1,49 @@
+liblockdev is (c) 1997, 1999 by Fabrizio Polacco <fab@prosa.it>
+
+   This program is free software; you can redistribute it and/or 
+   modify it under the terms of the GNU Lesser General Public 
+   License (LGPL) as published by the Free Software Foundation; 
+   version 2.1 dated February 1999.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General 
+   Public License (LGPL) along with this program;  if not, write 
+   to the Free Software Foundation, Inc.,
+   51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+
+On Debian GNU/Linux systems, the complete text of the GNU Library General
+Public License can be found in `/usr/share/common-licenses/LGPL'.
+
+It's primary source is the Debian archives:
+ ftp://ftp.debian.org/debian/pool/main/l/lockdev/
+
+================ CREDITS ================
+
+Barak Pearlmutter <bap@cs.unm.edu>
+	- code revision 
+	- suggestion for the use of the semaphore.
+Juan Cespedes <cespedes@debian.org>
+	- code revision 
+Daniel Quinlan <quinlan@transmeta.com>
+	- suggestion to support the cua devices.
+	- suggestions for naming of functions.
+Raphael Hertzog <rhertzog@hrnet.fr>
+	- NMU for new perl
+Adam P. Harris
+	- changes in debian postinst.
+Martin Schulze
+	- changes in perl makefile
+	- suggestions for changes in assembler stage for debug.
+Hamish Moffatt, James Troup
+	- dependencies check.
+Yann Dirson
+	- suggestions for a debugging shared library.
+Roger Leigh
+	- updated to ISO C89 C.
+	- work with modern glibc dev_t.
+	- work with devices in subdirectories, á la DevFS
+
--- lockdev-1.0.3.orig/src/lockdev.c
+++ lockdev-1.0.3/src/lockdev.c
@@ -125,14 +125,23 @@
 #include <sys/file.h>
 #if defined (__GNU_LIBRARY__)
 # include <sys/sysmacros.h>
+#ifdef __GNU__
+# define MAJOR(dev) major (dev)
+# define MINOR(dev) minor (dev)
+#else
 # define MAJOR(dev) gnu_dev_major (dev)
 # define MINOR(dev) gnu_dev_minor (dev)
+#endif
 #else
 #  error "put here a define for MAJOR and MINOR"
 #endif
 
 #include "lockdev.h"
 
+#ifndef MAXPATHLEN
+#define MAXPATHLEN 4096
+#endif
+
 /*
  *	PROTOTYPES for internal functions
  *	all internal functions names start with _dl_
