From 9bf3b2d5a8876669241e2cd8aa9eae3667beb27f Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Thu, 27 Nov 2025 01:09:58 +0800
Subject: [PATCH] Direct internally used script to its new location in lib

---
 src/providers/AppImageProvider.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/providers/AppImageProvider.py b/src/providers/AppImageProvider.py
index cd9821e..b9fcf9f 100644
--- a/src/providers/AppImageProvider.py
+++ b/src/providers/AppImageProvider.py
@@ -701,7 +701,7 @@ class AppImageProvider():
 
             if use_unsquashfs:
                 logging.debug('Testing with unsquashfs')
-                appimage_offset = terminal.sandbox_sh(['get_appimage_offset', file.get_path()])
+                appimage_offset = terminal.sandbox_sh(['/usr/lib/gearlever/get_appimage_offset', file.get_path()])
 
                 try:
                     terminal.sandbox_sh(['unsquashfs', '-o', appimage_offset, '-l', file.get_path()])
-- 
2.52.0

