From 1b3d2d8e4a75b074f3416235d2e104b47821ea83 Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Thu, 27 Nov 2025 20:41:49 +0800
Subject: [PATCH] Fix DesktopEntry type annotation in AppImageProvider

---
 src/providers/AppImageProvider.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/providers/AppImageProvider.py b/src/providers/AppImageProvider.py
index cd9821e..26f71ad 100644
--- a/src/providers/AppImageProvider.py
+++ b/src/providers/AppImageProvider.py
@@ -841,7 +841,7 @@ class AppImageProvider():
         folder = Settings.settings.get_string('appimages-default-folder')
         return re.sub(r'^~', GLib.get_home_dir(), folder)
 
-    def _get_app_version(self, extracted_appimage: Optional[ExtractedAppImage], desktop_entry: Optional[DesktopEntry] = None):
+    def _get_app_version(self, extracted_appimage: Optional[ExtractedAppImage], desktop_entry: Optional[DesktopEntry.DesktopEntry] = None):
         if not desktop_entry:
             desktop_entry = extracted_appimage.desktop_entry
 
-- 
2.52.0

