diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a394d1..d88e824 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,6 +45,11 @@ set_package_properties(BISON PROPERTIES
     PURPOSE "Required for the Predicate parser"
 )
 
+# flex-2.6.0 generates C90-incompatible code. KDEFrameworksCompilerSettings
+# forces C90 standard though, overriding our CFLAGS env variable, so we must
+# override the -std flag here
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
+
 if (CMAKE_SYSTEM_NAME MATCHES Linux)
     find_package( UDev )
 
