diff --git a/build-aux/linux/inochi-session.desktop b/build-aux/linux/inochi-session.desktop
index afa972e..037967e 100644
--- a/build-aux/linux/inochi-session.desktop
+++ b/build-aux/linux/inochi-session.desktop
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=inochi-session
 Exec=inochi-session
-Icon=icon_x256
+Icon=inochi-session
 Type=Application
-Categories=Utility
\ No newline at end of file
+Categories=Utility
