diff --git a/rdc_libs/rdc/src/RdcPerfTimer.cc b/rdc_libs/rdc/src/RdcPerfTimer.cc
index 47277bcc56..1f421a1e1f 100644
--- a/rdc_libs/rdc/src/RdcPerfTimer.cc
+++ b/rdc_libs/rdc/src/RdcPerfTimer.cc
@@ -22,7 +22,9 @@ THE SOFTWARE.
 
 #include "rdc_lib/RdcPerfTimer.h"
 
+#ifdef __x86_64__
 #include <x86intrin.h>
+#endif
 
 namespace amd {
 namespace rdc {
@@ -139,7 +141,13 @@ uint64_t RdcPerfTimer::CoarseTimestampUs() {
   clock_gettime(CLOCK_MONOTONIC_RAW, &ts);
   return uint64_t(ts.tv_sec) * 1000000 + ts.tv_nsec / 1000;
 }
-
+#ifdef __loongarch_lp64
+inline uint64_t __rdtscp(unsigned int* unused){
+    uint64_t val;
+    asm("rdtime.d %0, 0": "=r"(val));
+    return val;
+}
+#endif
 uint64_t RdcPerfTimer::MeasureTSCFreqHz() {
   // Make a coarse interval measurement of TSC ticks for 1 gigacycles.
   unsigned int unused;
