From: Dave Airlie <airlied@redhat.com>
Date: Thu, 4 Jun 2015 07:12:20 -0400
Subject: [PATCH] drm: i915: turn off wc mmaps

---
 drivers/gpu/drm/i915/i915_dma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_dma.c b/drivers/gpu/drm/i915/i915_dma.c
index d2df321ba634..775a5b11a366 100644
--- a/drivers/gpu/drm/i915/i915_dma.c
+++ b/drivers/gpu/drm/i915/i915_dma.c
@@ -151,7 +151,7 @@ static int i915_getparam(struct drm_device *dev, void *data,
 		value = 1;
 		break;
 	case I915_PARAM_MMAP_VERSION:
-		value = 1;
+		value = 0;
 		break;
 	case I915_PARAM_SUBSLICE_TOTAL:
 		value = INTEL_INFO(dev)->subslice_total;
