From fde7b5128ca9294f98b77b87be807ce1de599d5c Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 30 Apr 2024 00:42:36 -0700
Subject: [PATCH 1/8] [HACK] feat(data): add AOSC OS-specific armv4
 architecture

---
 data/cputable | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/cputable b/data/cputable
index 575c008e3..72d065af0 100644
--- a/data/cputable
+++ b/data/cputable
@@ -21,6 +21,7 @@
 alpha		alpha		alpha.*			64	little
 amd64		x86_64		(amd64|x86_64)		64	little
 arc		arc		arc			32	little
+armv4		arm		arm.*			32	little
 armeb		armeb		arm.*b			32	big
 arm		arm		arm.*			32	little
 arm64		aarch64		aarch64			64	little
-- 
2.45.0

