From d6b2de1272d735ca841c265ee1934b16fc6f8cf2 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 30 Apr 2024 00:45:44 -0700
Subject: [PATCH 3/8] [HACK] fix(data): rename i*86 as AOSC OS-specific i486

---
 data/cputable | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/cputable b/data/cputable
index 72d065af0..ca2d6b308 100644
--- a/data/cputable
+++ b/data/cputable
@@ -27,7 +27,7 @@ arm		arm		arm.*			32	little
 arm64		aarch64		aarch64			64	little
 hppa		hppa		hppa.*			32	big
 loong64		loongarch64	loongarch64		64	little
-i386		i686		(i[34567]86|pentium)	32	little
+i486		i486		i486			32	little
 ia64		ia64		ia64			64	little
 m68k		m68k		m68k			32	big
 mips		mips		mips(eb)?		32	big
-- 
2.45.0

