From 79bcfc68a9befaacc9ccb6cf8713f05f473f0e8e Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 30 Apr 2024 00:46:13 -0700
Subject: [PATCH 4/8] [HACK] fix(data): rename loong64 as AOSC OS-specific
 loongarch64

---
 data/cputable | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/cputable b/data/cputable
index ca2d6b308..9d79de9a7 100644
--- a/data/cputable
+++ b/data/cputable
@@ -26,7 +26,7 @@ armeb		armeb		arm.*b			32	big
 arm		arm		arm.*			32	little
 arm64		aarch64		aarch64			64	little
 hppa		hppa		hppa.*			32	big
-loong64		loongarch64	loongarch64		64	little
+loongarch64	loongarch64	loongarch64		64	little
 i486		i486		i486			32	little
 ia64		ia64		ia64			64	little
 m68k		m68k		m68k			32	big
-- 
2.45.0

