From 1b66947376bb74726eb46154997ed72a920c0b86 Mon Sep 17 00:00:00 2001
From: "Wang, Xiang" <KatyushaScarlet@Outlook.com>
Date: Sun, 23 Aug 2026 18:19:23 +0800
Subject: [PATCH 6/6] AOSCOS: data: add AOSC OS-specific armv5te architecture

---
 data/cputable   | 1 +
 data/tupletable | 1 +
 2 files changed, 2 insertions(+)

diff --git a/data/cputable b/data/cputable
index 4be844fd8..b6322a592 100644
--- a/data/cputable
+++ b/data/cputable
@@ -23,6 +23,7 @@ amd64		x86_64		(amd64|x86_64)		64	little
 arc		arc		arc			32	little
 armeb		armeb		arm.*b			32	big
 armv4l		armv4l		armv4.*l		32	little
+armv5l		armv5l		armv5.*l		32	little
 armv6lhf	armv6l		armv6.*l		32	little
 armv7lhf	armv7l		armv7.*l		32	little
 arm		arm		arm[^467]*		32	little
diff --git a/data/tupletable b/data/tupletable
index 7c120c0b5..ce75fe07c 100644
--- a/data/tupletable
+++ b/data/tupletable
@@ -29,6 +29,7 @@ eabihf-gnu-linux-armv6lhf	armv6hf
 eabihf-gnu-linux-armv7lhf	armv7hf
 eabihf-gnu-linux-arm		armhf
 eabi-gnu-linux-armv4l		armv4
+eabi-gnu-linux-armv5l		armv5te
 eabi-gnu-linux-arm		armel
 abin32-gnu-linux-mips64r6el	mipsn32r6el
 abin32-gnu-linux-mips64r6	mipsn32r6
-- 
2.55.0

