From 0bf684421c71071b04537cb2ca4f67776acabde1 Mon Sep 17 00:00:00 2001
From: Liu Zichen <tehtiam@aosc.io>
Date: Mon, 20 Jul 2026 23:24:59 +0800
Subject: [PATCH 1/1] Workaround build problem with Python support

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 99a00fd40..66ab16f8a 100644
--- a/Makefile
+++ b/Makefile
@@ -104,7 +104,7 @@ VPATH := $(YOSYS_SRC)
 UNITESTPATH := $(YOSYS_SRC)/tests/unit
 
 export CXXSTD ?= c++20
-CXXFLAGS := $(CXXFLAGS) -Wall -Wextra -Werror=unused -ggdb -I. -I"$(YOSYS_SRC)" -MD -MP -D_YOSYS_ -fPIC -I$(PREFIX)/include
+CXXFLAGS := $(CXXFLAGS) -Wall -Wextra -ggdb -I. -I"$(YOSYS_SRC)" -MD -MP -D_YOSYS_ -fPIC -I$(PREFIX)/include
 LIBS := $(LIBS) -lstdc++ -lm
 PLUGIN_LINKFLAGS :=
 PLUGIN_LIBS :=
-- 
2.52.0

