diff --git a/runtime/hsa-runtime/core/util/lnx/os_linux.cpp b/runtime/hsa-runtime/core/util/lnx/os_linux.cpp
index 064d4a5ab8..e575b59130 100644
--- a/runtime/hsa-runtime/core/util/lnx/os_linux.cpp
+++ b/runtime/hsa-runtime/core/util/lnx/os_linux.cpp
@@ -68,7 +68,7 @@
 #include <cpuid.h>
 #endif
 
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && !defined(__riscv) && !defined(__mips)
 #define ABS_ADDR(base, ptr) (ptr)
 #else
 #define ABS_ADDR(base, ptr) ((base) + (ptr))
diff --git a/runtime/hsa-runtime/core/util/locks.h b/runtime/hsa-runtime/core/util/locks.h
index 133dd06f0b..c7477d84ed 100644
--- a/runtime/hsa-runtime/core/util/locks.h
+++ b/runtime/hsa-runtime/core/util/locks.h
@@ -72,7 +72,11 @@ class HybridMutex {
     while (!lock_.compare_exchange_strong(old, 1)) {
       cnt--;
       if (cnt > maxSpinIterPause) {
+#if defined(_M_X64)
         _mm_pause();
+#else
+        os::YieldThread();
+#endif
       } else if (cnt-- > maxSpinIterYield) {
         os::YieldThread();
       } else {
diff --git a/runtime/hsa-runtime/core/util/utils.h b/runtime/hsa-runtime/core/util/utils.h
index 1839bacb68..47ec5184db 100644
--- a/runtime/hsa-runtime/core/util/utils.h
+++ b/runtime/hsa-runtime/core/util/utils.h
@@ -65,6 +65,30 @@
 #endif
 #if defined(_MSC_VER) && (defined(_M_IX86) || defined(_M_X64))
 #include "intrin.h"
+#elif defined(__aarch64__)
+#define _mm_clflush(ptr) asm volatile("DMB ish":::"memory")
+#define _mm_sfence(ptr) asm volatile("DMB ishst":::"memory")
+#define _mm_mfence(ptr) asm volatile("DMB ish":::"memory")
+#define _mm_pause() asm volatile("isb")
+#define _mm_lfence(ptr) asm volatile("DMB ish":::"memory")
+#elif defined(__loongarch_lp64)
+// See linux-kernel arch/loongarch/include/asm/barrier.h
+// full barrier-mb
+static inline void _mm_clflush(const void* ptr) { (void)ptr; asm volatile ("dbar 0" ::: "memory"); }
+// sfence->wmb->c_wsync->c_w_w
+// LoongArch barrier:  c_w_w completion previous(W) succeeding(W)
+static inline void _mm_sfence(void) { asm volatile ("dbar 0xa" ::: "memory"); }
+// mfence->mb->c_sync->crwrw
+// LoongArch barrier:  crwrw completion previous(R/W) succeeding(R/W)
+static inline void _mm_mfence(void) { asm volatile ("dbar 0" ::: "memory"); }
+// LoongArch does not have an alternative to mm_pause. It is set to execute empty here.
+static inline void _mm_pause(void) {}
+#elif defined(__riscv) && __riscv_xlen == 64
+#define _mm_clflush(ptr) asm volatile("fence rw,rw":::"memory")
+#define _mm_sfence(ptr) asm volatile("fence w,w":::"memory")
+#define _mm_mfence(ptr) asm volatile("fence rw,rw":::"memory")
+#define _mm_pause() asm volatile("fence rw,rw":::"memory")
+#define _mm_lfence(ptr) asm volatile("fence rw,rw":::"memory")
 #endif
 
 namespace rocr {
diff --git a/runtime/hsa-runtime/image/util.h b/runtime/hsa-runtime/image/util.h
index 18f4a11ba9..7e9aeb7876 100644
--- a/runtime/hsa-runtime/image/util.h
+++ b/runtime/hsa-runtime/image/util.h
@@ -74,8 +74,8 @@ namespace image {
 
 
 #if defined(__GNUC__)
-#include "mm_malloc.h"
 #if defined(__i386__) || defined(__x86_64__)
+#include "mm_malloc.h"
 #include <x86intrin.h>
 #elif defined(__loongarch64)
 #else
