diff --git a/runtime/hsa-runtime/core/util/utils.h b/runtime/hsa-runtime/core/util/utils.h
index 96e9e476e0..701a223b65 100644
--- a/runtime/hsa-runtime/core/util/utils.h
+++ b/runtime/hsa-runtime/core/util/utils.h
@@ -78,6 +78,12 @@ void log_printf(const char* file, int line, const char* format, ...);
 #if defined(__GNUC__)
 #if defined(__i386__) || defined(__x86_64__)
 #include <x86intrin.h>
+#elif defined(__loongarch_lp64)
+#define _mm_clflush(ptr) asm volatile ("dbar 0x700":::"memory")
+#define _mm_sfence(ptr) asm volatile ("dbar 0b10010":::"memory")
+#define _mm_mfence(ptr) asm volatile ("dbar 0b10000":::"memory")
+// LoongArch does not have an alternative to mm_pause. It is set to execute empty here.
+#define _mm_pause()
 #endif
 
 #define __forceinline __inline__ __attribute__((always_inline))
diff --git a/runtime/hsa-runtime/image/util.h b/runtime/hsa-runtime/image/util.h
index 18f4a11ba9..7e9aeb7876 100644
--- a/runtime/hsa-runtime/image/util.h
+++ b/runtime/hsa-runtime/image/util.h
@@ -74,8 +74,8 @@ namespace image {
 
 
 #if defined(__GNUC__)
-#include "mm_malloc.h"
 #if defined(__i386__) || defined(__x86_64__)
+#include "mm_malloc.h"
 #include <x86intrin.h>
 #elif defined(__loongarch64)
 #else
