From 1323071d4dbe83dc543b2358ccf1f5a232609093 Mon Sep 17 00:00:00 2001
From: Kaiyang Wu <self@origincode.me>
Date: Thu, 21 Nov 2024 17:02:13 -0800
Subject: [PATCH] AOSCOS: Install systemd unit to correct location

Signed-off-by: Kaiyang Wu <self@origincode.me>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 CMakeLists.txt | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4761b076..456007f8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -375,18 +375,8 @@ install(DIRECTORY man/
 set(WANT_SYSTEMD false CACHE BOOL "Install znc.service to systemd")
 if(WANT_SYSTEMD)
 	configure_file("znc.service.in" "znc.service")
-	set(SYSTEMD_DIR "" CACHE PATH "Path to systemd units")
-	if(SYSTEMD_DIR STREQUAL "" AND PKG_CONFIG_EXECUTABLE)
-		execute_process(COMMAND "${PKG_CONFIG_EXECUTABLE}"
-			--variable=systemdsystemunitdir systemd
-			OUTPUT_VARIABLE SYSTEMD_DIR)
-	endif()
-	if(SYSTEMD_DIR STREQUAL "")
-		message(FATAL_ERROR "Systemd is enabled, "
-			"but the unit dir can't be found.")
-	endif()
 	install(FILES "${PROJECT_BINARY_DIR}/znc.service"
-		DESTINATION "${SYSTEMD_DIR}")
+		DESTINATION "${CMAKE_INSTALL_LIBDIR}/systemd/system")
 endif()
 
 # On cygwin, if to link modules against znc.exe directly, modperl can't call
-- 
2.52.0

