diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ed5114..bf7005d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -266,7 +266,7 @@ if(NOT CMAKE_CROSSCOMPILING)
   enable_testing()
 endif()
 
-set(TEST_DATA_DIR ${CMAKE_CURRENT_SOURCE_DIR}/data)
+set(TEST_DATA_DIR ${CMAKE_CURRENT_SOURCE_DIR}/data CACHE PATH "Where presets are expected to be found")
 
 add_subdirectory(src)
 
diff --git a/src/config-kdsme.h.cmake b/src/config-kdsme.h.cmake
index 0100d36..9ace806 100644
--- a/src/config-kdsme.h.cmake
+++ b/src/config-kdsme.h.cmake
@@ -6,6 +6,8 @@
 /* Define to the full name of this package. */
 #define PACKAGE_NAME "${CMAKE_PROJECT_NAME}"
 
+#cmakedefine TEST_DATA_DIR
+
 #cmakedefine01 HAVE_GRAPHVIZ
 
 #cmakedefine01 WITH_INTERNAL_GRAPHVIZ
