--- a/src/helper.cpp	2018-05-22 09:50:09.656728368 +0200
+++ b/src/helper.cpp	2018-05-22 09:59:01.630672411 +0200
@@ -53,111 +53,19 @@
     QFile configFile(QStringLiteral(PLYMOUTH_CONFIG_PATH));
     configFile.setPermissions(QFileDevice::ReadOwner|QFileDevice::WriteOwner|QFileDevice::ReadGroup|QFileDevice::ReadOther);
 
-    //Special case: Ubuntu derivatives, which work different from everybody else
-    if (QFile::exists(QStringLiteral("/usr/sbin/update-alternatives"))) {
-        //find the .plymouth file in the theme
-        QDir dir(QStringLiteral(PLYMOUTH_THEMES_DIR) + theme);
-        QStringList themeFile = dir.entryList(QStringList() << QStringLiteral("*.plymouth"));
-        if (themeFile.count() != 1) {
-            reply = ActionReply::BackendError;
-            reply.setErrorDescription(i18n("Theme corrupted: .plymouth file not found inside theme."));
-            return reply;
-        }
-        int ret = 0;
-        QProcess checkProcess;
-        QByteArray data;
-        qDebug() << "Running update-alternatives --list default.plymouth now";
-        checkProcess.start(QStringLiteral("update-alternatives --list default.plymouth"), QStringList());
-        if (!checkProcess.waitForStarted()) {
-            reply = ActionReply::BackendError;
-            reply.setErrorDescription(i18n("Cannot start update-alternatives."));
-            return reply;
-        }
-        if (!checkProcess.waitForFinished()) {
-            reply = ActionReply::BackendError;
-            reply.setErrorDescription(i18n("update-alternatives failed to run."));
-            return reply;
-        }
-        else {
-            data = checkProcess.readAllStandardOutput();
-        }
-        ret = checkProcess.exitCode();
-        
-        if (ret != 0) {
-            reply = ActionReply(ActionReply::HelperErrorReply());
-            reply.setErrorCode(static_cast<ActionReply::Error>(ret));
-            reply.setErrorDescription(i18n("update-alternatives returned with error condition %1.", ret));
-            return reply;
-        }
-        QString installFile = dir.path() + QLatin1Char('/') + themeFile.first();
-        if (!data.contains(installFile.toUtf8())) {
-            qDebug() << "Plymouth file not found in update-alternatives. So install it";
-            QProcess installProcess;
-            installProcess.start(QStringLiteral("update-alternatives"), QStringList() << QStringLiteral("--install")
-                                 << QStringLiteral("/usr/share/plymouth/themes/default.plymouth")
-                                 << QStringLiteral("default.plymouth")
-                                 << installFile
-                                 << QStringLiteral("100"));
-            
-            if (!installProcess.waitForStarted()) {
-                reply = ActionReply::BackendError;
-                reply.setErrorDescription(i18n("Cannot start update-alternatives."));
-                return reply;
-            }
-            if (!installProcess.waitForFinished()) {
-                reply = ActionReply::BackendError;
-                reply.setErrorDescription(i18n("update-alternatives failed to run."));
-                return reply;
-            }
-            ret = installProcess.exitCode();
-            
-            if (ret != 0) {
-                reply = ActionReply(ActionReply::HelperErrorReply());
-                reply.setErrorCode(static_cast<ActionReply::Error>(ret));
-                reply.setErrorDescription(i18n("update-alternatives returned with error condition %1.", ret));
-                return reply;
-            }
-        }
-        else {
-            qDebug() << "Running update-alternatives --set  now";
-            QProcess process;
-            process.start(QStringLiteral("update-alternatives"), QStringList() << QStringLiteral("--set")
-                          << QStringLiteral("default.plymouth")
-                          << installFile);
-            if (!process.waitForStarted()) {
-                reply = ActionReply::BackendError;
-                reply.setErrorDescription(i18n("Cannot start update-alternatives."));
-                return reply;
-            }
-            if (!process.waitForFinished()) {
-                reply = ActionReply::BackendError;
-                reply.setErrorDescription(i18n("update-alternatives failed to run."));
-                return reply;
-            }
-            ret = process.exitCode();
-            
-            if (ret != 0) {
-                reply = ActionReply(ActionReply::HelperErrorReply());
-                reply.setErrorCode(static_cast<ActionReply::Error>(ret));
-                reply.setErrorDescription(i18n("update-alternatives returned with error condition %1.", ret));
-                return reply;
-            }
-        }
-    }
-
     int ret = 0;
 
     QProcess process;
-    qDebug() << "Running update-initramfs -u  now";
-    process.start(QStringLiteral("/usr/sbin/update-initramfs"), QStringList() << QStringLiteral("-u"));
+    qDebug() << "Running dracut --force  now";
+    process.start(QStringLiteral("/usr/bin/dracut"), QStringList() << QStringLiteral("--force"));
     if (!process.waitForStarted()) {
         reply = ActionReply::BackendError;
-        reply.setErrorDescription(i18n("Cannot start initramfs."));
+        reply.setErrorDescription(i18n("Cannot start dracut."));
         return reply;
     }
     if (!process.waitForFinished(60000)) {
         reply = ActionReply::BackendError;
-        reply.setErrorDescription(i18n("Initramfs failed to run."));
+        reply.setErrorDescription(i18n("Dracut failed to run."));
         return reply;
     }
     ret = process.exitCode();
@@ -167,7 +75,7 @@
     } else {
         reply = ActionReply(ActionReply::HelperErrorReply());
         reply.setErrorCode(static_cast<ActionReply::Error>(ret));
-        reply.setErrorDescription(i18n("Initramfs returned with error condition %1.", ret));
+        reply.setErrorDescription(i18n("Dracut returned with error condition %1.", ret));
         return reply;
     }
 }
@@ -238,43 +146,6 @@
 
     archive->close();
 
-    //Special case: Ubuntu derivatives, which work different from everybody else
-    if (QFile::exists(QStringLiteral("/usr/sbin/update-alternatives"))) {
-        //find the .plymouth file in the theme
-        QDir dir(themePath);
-        QStringList themeFile = dir.entryList(QStringList() << QStringLiteral("*.plymouth"));
-        if (themeFile.count() != 1) {
-            reply = ActionReply::BackendError;
-            reply.setErrorDescription(i18n("Theme corrupted: .plymouth file not found inside theme."));
-            return reply;
-        }
-        int ret = 0;
-        QProcess process;
-        process.start(QStringLiteral("update-alternatives"), QStringList() << QStringLiteral("--install")
-                      << QStringLiteral("/usr/share/plymouth/themes/default.plymouth")
-                      << QStringLiteral("default.plymouth")
-                      << themePath + QLatin1Char('/') + themeFile.first()
-                      << QStringLiteral("100"));
-        if (!process.waitForStarted()) {
-            reply = ActionReply::BackendError;
-            reply.setErrorDescription(i18n("Cannot start update-alternatives."));
-            return reply;
-        }
-        if (!process.waitForFinished()) {
-            reply = ActionReply::BackendError;
-            reply.setErrorDescription(i18n("update-alternatives failed to run."));
-            return reply;
-        }
-        ret = process.exitCode();
-
-        if (ret != 0) {
-            reply = ActionReply(ActionReply::HelperErrorReply());
-            reply.setErrorCode(static_cast<ActionReply::Error>(ret));
-            reply.setErrorDescription(i18n("update-alternatives returned with error condition %1.", ret));
-            return reply;
-        }
-    }
-
     QVariantMap map;
     map[QStringLiteral("plugin")] = themeName;
     map[QStringLiteral("path")] = themePath;
@@ -307,41 +178,6 @@
         return reply;
     }
 
-    //Special case: Ubuntu derivatives, which work different from everybody else
-    if (QFile::exists(QStringLiteral("/usr/sbin/update-alternatives"))) {
-        //find the .plymouth file in the theme
-        QStringList themeFile = dir.entryList(QStringList() << QStringLiteral("*.plymouth"));
-        if (themeFile.count() != 1) {
-            reply = ActionReply::BackendError;
-            reply.setErrorDescription(i18n("Theme corrupted: .plymouth file not found inside theme."));
-            return reply;
-        }
-        int ret = 0;
-        QProcess process;
-
-        process.start(QStringLiteral("update-alternatives"), QStringList() << QStringLiteral("--remove")
-                      << QStringLiteral("default.plymouth")
-                      << dir.path() + QLatin1Char('/') + themeFile.first());
-        if (!process.waitForStarted()) {
-            reply = ActionReply::BackendError;
-            reply.setErrorDescription(i18n("Cannot start update-alternatives."));
-            return reply;
-        }
-        if (!process.waitForFinished()) {
-            reply = ActionReply::BackendError;
-            reply.setErrorDescription(i18n("update-alternatives failed to run."));
-            return reply;
-        }
-        ret = process.exitCode();
-
-        if (ret != 0) {
-            reply = ActionReply(ActionReply::HelperErrorReply());
-            reply.setErrorCode(static_cast<ActionReply::Error>(ret));
-            reply.setErrorDescription(i18n("update-alternatives returned with error condition %1.", ret));
-            return reply;
-        }
-    }
-
     if (dir.removeRecursively()) {
         return ActionReply::SuccessReply();
     } else {
