diff --git a/deps/CMakeLists.txt b/deps/CMakeLists.txt
index a8bf132..f1c4f85 100644
--- a/deps/CMakeLists.txt
+++ b/deps/CMakeLists.txt
@@ -1,11 +1,7 @@
 cmake_minimum_required(VERSION 3.16)
-
+set(OpenGL_GL_PREFERENCE "GLVND")
 project(bindbc_imgui)
 
-include(cmake/PullSubmodules.cmake)
-
-PullSubmodules(${CMAKE_CURRENT_LIST_DIR}/.. submodulePrefixPath)
-
 message(STATUS "submodulePrefixPath: ${submodulePrefixPath}")
 
 ####################
@@ -44,7 +40,7 @@ else ()
     set(PlatformFolder "Unknown")
 endif()
 
-set(OutputDirectory ${CMAKE_CURRENT_LIST_DIR}/../libs/${PlatformArchitecture}/${PlatformFolder})
+set(OutputDirectory ${CMAKE_CURRENT_LIST_DIR}/../libs/${PlatformFolder})
 
 if (DEFINED STATIC_CIMGUI)
     set(OutputDirectory ${OutputDirectory}/Static)
@@ -99,7 +95,7 @@ endif()
 option(BUILD_SHARED_LIBS "Build shared instead of static libraries." OFF)
 set(IMGUI_SDL "yes" CACHE STRING "Build with SDL")
 set(IMGUI_OPENGL3 "yes" CACHE STRING "Build with OpenGL3")
-add_subdirectory(${cimgui_SUBMOD_DIR})
+add_subdirectory(cimgui)
 
 set_target_properties(cimgui PROPERTIES DEBUG_POSTFIX "")
 set_target_properties(cimgui PROPERTIES ARCHIVE_OUTPUT_DIRECTORY ${OutputDirectory})
diff --git a/dub.sdl b/dub.sdl
index de7d5e8..4345706 100644
--- a/dub.sdl
+++ b/dub.sdl
@@ -45,21 +45,13 @@ configuration "static_dynamicCRT" {
     copyFiles "libs/x86/win32/Static/DynamicCRT/SDL2.dll" platform="windows-x86"
     copyFiles "libs/x86/win32/Static/DynamicCRT/freetype.dll" platform="windows-x86"
     
-    // Linux x86_64
+    // Linux
     preBuildCommands \
-        "cmake -DSTATIC_CIMGUI= -S deps -B deps/build_linux_x64_cimguiStatic" \
-        "cmake --build deps/build_linux_x64_cimguiStatic --config Release" platform="linux-x86_64"
+        "cmake -GNinja -DSTATIC_CIMGUI= -S deps -B deps/build_linux_cimguiStatic" \
+        "cmake --build deps/build_linux_cimguiStatic --config Release" platform="linux"
 
-    sourceFiles "$PACKAGE_DIR/libs/x86_64/linux/Static/cimgui.a" platform="linux-x86_64"
-    libs "sdl2" "freetype" ":libstdc++.so" platform="linux-x86_64"
-
-    // Linux aarch64
-    preBuildCommands \
-        "cmake -DSTATIC_CIMGUI= -S deps -B deps/build_linux_aarch64_cimguiStatic" \
-        "cmake --build deps/build_linux_aarch64_cimguiStatic --config Release" platform="linux-aarch64"
-
-    sourceFiles "$PACKAGE_DIR/libs/aarch64/linux/Static/cimgui.a" platform="linux-aarch64"
-    libs "sdl2" "freetype" ":libstdc++.so" platform="linux-aarch64"
+    sourceFiles "$PACKAGE_DIR/libs/linux/Static/cimgui.a" platform="linux"
+    libs "sdl2" "freetype" ":libstdc++.so" platform="linux"
 }
 
 configuration "static_staticCRT" {
