From e9c49605e6542787b1cee2d3f62d9ab04cc2428e Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Thu, 22 Jan 2026 00:39:25 +0800
Subject: [PATCH] DEPS

---
 DEPS | 396 +----------------------------------------------------------
 1 file changed, 1 insertion(+), 395 deletions(-)

diff --git a/DEPS b/DEPS
index df7f2863e48..78519b143cb 100644
--- a/DEPS
+++ b/DEPS
@@ -81,7 +81,7 @@ vars = {
   "gn_version": "git_revision:81b24e01531ecf0eff12ec9359a555ec3944ec4e",
 
   "reclient_version": "re_client_version:28341fc74c68f05a5c8be35160ada940c4edb969",
-  "download_reclient": True,
+  "download_reclient": False,
 
   # Update from https://chrome-infra-packages.appspot.com/p/fuchsia/sdk/core
   "fuchsia_sdk_version": "version:29.20250925.5.1",
@@ -185,18 +185,6 @@ deps = {
     Var("chromium_git") + "/chromium/llvm-project/cfe/tools/clang-format.git" +
     "@" + Var("clang_format_scripts_rev"),
 
-  Var("dart_root") + "/benchmarks-internal": {
-    "url": Var("dart_internal_git") + "/benchmarks-internal.git" +
-           "@" + Var("benchmarks_internal_rev"),
-    "condition": "checkout_benchmarks_internal",
-  },
-  Var("dart_root") + "/tools/sdks/dart-sdk": {
-      "packages": [{
-          "package": "dart/dart-sdk/${{platform}}",
-          "version": Var("sdk_tag"),
-      }],
-      "dep_type": "cipd",
-  },
   Var("dart_root") + "/third_party/d8/linux/x64": {
       "packages": [{
           "package": "dart/third_party/d8/linux-amd64",
@@ -211,43 +199,6 @@ deps = {
       }],
       "dep_type": "cipd",
   },
-  Var("dart_root") + "/third_party/d8/macos/x64": {
-      "packages": [{
-          "package": "dart/third_party/d8/mac-amd64",
-          "version": Var("d8_tag"),
-      }],
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/third_party/d8/macos/arm64": {
-      "packages": [{
-          "package": "dart/third_party/d8/mac-arm64",
-          "version": Var("d8_tag"),
-      }],
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/third_party/d8/windows/x64": {
-      "packages": [{
-          "package": "dart/third_party/d8/windows-amd64",
-          "version": Var("d8_tag"),
-      }],
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/third_party/firefox_jsshell": {
-      "packages": [{
-          "package": "dart/third_party/jsshell/${{platform}}",
-          "version": Var("jsshell_tag"),
-      }],
-      "condition": "checkout_javascript_engines",
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/third_party/jsc": {
-      "packages": [{
-          "package": "dart/third_party/jsc/${{platform}}",
-          "version": Var("jsc_tag"),
-      }],
-      "condition": "checkout_javascript_engines and host_os == 'linux' and host_cpu == 'x64'",
-      "dep_type": "cipd",
-  },
   Var("dart_root") + "/third_party/devtools": {
       "packages": [{
           "package": "dart/third_party/flutter/devtools",
@@ -255,11 +206,6 @@ deps = {
       }],
       "dep_type": "cipd",
   },
-  Var("dart_root") + "/third_party/devtools_src": {
-      "url": Var("dart_git") + "external/github.com/flutter/devtools.git" +
-      "@" + Var("devtools_rev"),
-      "condition": "build_devtools_from_sources",
-  },
   Var("dart_root") + "/tests/co19/src": {
       "packages": [{
           "package": "dart/third_party/co19",
@@ -306,16 +252,6 @@ deps = {
       Var("dart_git") + "external/github.com/emscripten-core/emsdk.git" +
       "@" + Var("emsdk_rev"),
 
-  Var("dart_root") + "/third_party/flutter": {
-      "url": Var("dart_git") + "external/github.com/flutter/flutter.git" +
-      "@" + Var("flutter_rev"),
-      "condition": "build_devtools_from_sources",
-  },
-
-  Var("dart_root") + "/third_party/jinja2":
-      Var("chromium_git") + "/chromium/src/third_party/jinja2.git" +
-      "@" + Var("jinja2_rev"),
-
   Var("dart_root") + "/third_party/perfetto":
       Var("android_git") + "/platform/external/perfetto" +
       "@" + Var("perfetto_rev"),
@@ -353,10 +289,6 @@ deps = {
       Var("dart_git") + "dartdoc.git" + "@" + Var("dartdoc_rev"),
   Var("dart_root") + "/third_party/pkg/ecosystem":
       Var("dart_git") + "ecosystem.git" + "@" + Var("ecosystem_rev"),
-  Var("dart_root") + "/third_party/flute": {
-    "url": Var("dart_git") + "flute.git" + "@" + Var("flute_rev"),
-    "condition": "checkout_flute",
-  },
 
   Var("dart_root") + "/third_party/pkg/http":
       Var("dart_git") + "http.git" + "@" + Var("http_rev"),
@@ -364,12 +296,6 @@ deps = {
       Var("dart_git") + "i18n.git" + "@" + Var("i18n_rev"),
   Var("dart_root") + "/third_party/pkg/leak_tracker":
       Var("dart_git") + "leak_tracker.git" + "@" + Var("leak_tracker_rev"),
-  Var("dart_root") + "/third_party/pkg/material_color_utilities": {
-    "url": Var("dart_git") +
-           "external/github.com/material-foundation/material-color-utilities.git" +
-           "@" + Var("material_color_utilities_rev"),
-    "condition": "checkout_flute",
-  },
   Var("dart_root") + "/third_party/pkg/native":
       Var("dart_git") + "native.git" + "@" + Var("native_rev"),
   Var("dart_root") + "/third_party/pkg/protobuf":
@@ -400,216 +326,6 @@ deps = {
       "@" + Var("webkit_inspection_protocol_rev"),
   Var("dart_root") + "/third_party/pkg/web":
       Var("dart_git") + "web.git" + "@" + Var("web_rev"),
-
-  Var("dart_root") + "/buildtools/sysroot/linux": {
-      "packages": [
-          {
-              "package": "fuchsia/third_party/sysroot/linux",
-              "version": "git_revision:fa7a5a9710540f30ff98ae48b62f2cdf72ed2acd",
-          },
-      ],
-      "condition": "host_os == linux",
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/buildtools/sysroot/focal": {
-      "packages": [
-          {
-              "package": "fuchsia/third_party/sysroot/focal",
-              "version": "git_revision:fa7a5a9710540f30ff98ae48b62f2cdf72ed2acd",
-          },
-      ],
-      "condition": "host_os == linux",
-      "dep_type": "cipd",
-  },
-
-  # Keep consistent with pkg/test_runner/lib/src/options.dart.
-  Var("dart_root") + "/buildtools/linux-x64/clang": {
-      "packages": [
-          {
-              "package": "fuchsia/third_party/clang/linux-amd64",
-              "version": Var("clang_version"),
-          },
-      ],
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/buildtools/mac-x64/clang": {
-      "packages": [
-          {
-              "package": "fuchsia/third_party/clang/mac-amd64",
-              "version": Var("clang_version"),
-          },
-      ],
-      "condition": "host_os == mac", # On ARM64 Macs too because Goma doesn't support the host-arm64 toolchain.
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/buildtools/win-x64/clang": {
-      "packages": [
-          {
-              "package": "fuchsia/third_party/clang/windows-amd64",
-              "version": Var("clang_version"),
-          },
-      ],
-      "condition": "host_os == win", # On ARM64 Windows too because Fuchsia doesn't provide the host-arm64 toolchain.
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/buildtools/linux-arm64/clang": {
-      "packages": [
-          {
-              "package": "fuchsia/third_party/clang/linux-arm64",
-              "version": Var("clang_version"),
-          },
-      ],
-      "condition": "host_os == 'linux' and host_cpu == 'arm64'",
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/buildtools/mac-arm64/clang": {
-      "packages": [
-          {
-              "package": "fuchsia/third_party/clang/mac-arm64",
-              "version": Var("clang_version"),
-          },
-      ],
-      "condition": "host_os == 'mac' and host_cpu == 'arm64'",
-      "dep_type": "cipd",
-  },
-
-  Var("dart_root") + '/buildtools/reclient': {
-    'packages': [
-      {
-        'package': 'infra/rbe/client/${{platform}}',
-        'version': Var('reclient_version'),
-      }
-    ],
-    # Download reclient only on the platforms where it has packages available.
-    # Unfortunately windows-arm64 gclient uses x64 python which lies in
-    # host_cpu, so we have to use a variable to not download reclient there.
-    'condition': 'download_reclient and (((host_os == "linux" or host_os == "mac" or host_os == "win") and host_cpu == "x64") or (host_os == "mac" and host_cpu == "arm64"))',
-    'dep_type': 'cipd',
-  },
-
-  # windows-arm64 package is not available, use the windows-x64 package and
-  # rely on transparent emulation.
-  Var("dart_root") + '/buildtools/reclient-win': {
-    'packages': [
-      {
-        'package': 'infra/rbe/client/windows-amd64',
-        'version': Var('reclient_version'),
-      }
-    ],
-    'condition': 'host_os == "win"',
-    'dep_type': 'cipd',
-  },
-
-  # linux-arm64 package is not available, use the linux-x64 package and
-  # rely on transparent emulation.
-  Var("dart_root") + '/buildtools/reclient-linux': {
-    'packages': [
-      {
-        'package': 'infra/rbe/client/linux-amd64',
-        'version': Var('reclient_version'),
-      }
-    ],
-    'condition': 'host_os == "linux"',
-    'dep_type': 'cipd',
-  },
-
-  Var("dart_root") + "/third_party/webdriver/chrome": {
-    "packages": [
-      {
-        "package": "dart/third_party/chromedriver/${{platform}}",
-        "version": "version:" + Var("chrome_tag"),
-      }
-    ],
-    "condition": "download_chrome",
-    "dep_type": "cipd",
-  },
-
-  Var("dart_root") + "/buildtools": {
-      "packages": [
-          {
-              "package": "gn/gn/${{platform}}",
-              "version": Var("gn_version"),
-          },
-      ],
-      "condition": "host_os != 'win'",
-      "dep_type": "cipd",
-  },
-
-  # windows-arm64 package is not available, use the windows-x64 package and
-  # rely on transparent emulation.
-  Var("dart_root") + "/buildtools/win": {
-      "packages": [
-          {
-              "package": "gn/gn/windows-amd64",
-              "version": Var("gn_version"),
-          },
-      ],
-      "condition": "host_os == 'win'",
-      "dep_type": "cipd",
-  },
-
-  Var("dart_root") + "/buildtools/ninja": {
-      "packages": [{
-          "package": "infra/3pp/tools/ninja/${{platform}}",
-          "version": Var("ninja_tag"),
-      }],
-      "dep_type": "cipd",
-  },
-
-  Var("dart_root") + "/third_party/android_tools/ndk": {
-      "packages": [
-          {
-            "package": "flutter/android/ndk/${{os}}-amd64",
-            "version": "version:r27.0.10869015"
-          }
-      ],
-      "condition": "download_android_deps",
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/third_party/android_tools/sdk/platform-tools": {
-      "packages": [
-          {
-            "package": "flutter/android/sdk/platform-tools/linux-amd64",
-            "version": "1tZc4sOxZS6FQIvT5i0wwdycmM8AO7QZY32FC9_HfR4C"
-          }
-      ],
-      "condition": "download_android_deps",
-      "dep_type": "cipd",
-  },
-
-  Var("dart_root") + "/third_party/fuchsia/sdk/linux": {
-    "packages": [
-      {
-      "package": "fuchsia/sdk/core/${{platform}}",
-      "version": Var("fuchsia_sdk_version"),
-      }
-    ],
-    "condition": 'download_fuchsia_deps and host_os == "linux"',
-    "dep_type": "cipd",
-  },
-
-  Var("dart_root") + "/third_party/fuchsia/test_scripts": {
-    "packages": [
-      {
-      "package": "chromium/fuchsia/test-scripts",
-      "version": "JUeFbA8y0E-_pj-bgvR6glsLIO-gJIHlR9RtM5zxni8C",
-      }
-    ],
-    "condition": 'download_fuchsia_deps',
-    "dep_type": "cipd",
-  },
-
-  Var("dart_root") + "/third_party/fuchsia/gn-sdk": {
-    "packages": [
-      {
-      "package": "chromium/fuchsia/gn-sdk",
-      "version": "NAEC5tfgSl8g94nwpKsGtNMEdbiAlgwrNa9XQ7cIcbcC",
-      }
-    ],
-    "condition": 'download_fuchsia_deps',
-    "dep_type": "cipd",
-  },
-
   Var("dart_root") + "/pkg/front_end/test/types/benchmark_data": {
     "packages": [
       {
@@ -619,73 +335,6 @@ deps = {
     ],
     "dep_type": "cipd",
   },
-
-  # TODO(37531): Remove these cipd packages and build with sdk instead when
-  # benchmark runner gets support for that.
-  Var("dart_root") + "/benchmarks/FfiBoringssl/native/out/": {
-      "packages": [
-          {
-              "package": "dart/benchmarks/ffiboringssl",
-              "version": "commit:a86c69888b9a416f5249aacb4690a765be064969",
-          },
-      ],
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/benchmarks/FfiCall/native/out/": {
-      "packages": [
-          {
-              "package": "dart/benchmarks/fficall",
-              "version": "ebF5aRXKDananlaN4Y8b0bbCNHT1MnkGbWqfpCpiND4C",
-          },
-      ],
-          "dep_type": "cipd",
-  },
-  Var("dart_root") + "/benchmarks/NativeCall/native/out/": {
-      "packages": [
-          {
-              "package": "dart/benchmarks/nativecall",
-              "version": "w1JKzCIHSfDNIjqnioMUPq0moCXKwX67aUfhyrvw4E0C",
-          },
-      ],
-          "dep_type": "cipd",
-  },
-  Var("dart_root") + "/third_party/browsers/chrome": {
-      "packages": [
-          {
-              "package": "dart/browsers/chrome/${{platform}}",
-              "version": "version:" + Var("chrome_tag"),
-          },
-      ],
-      "condition": "download_chrome",
-      "dep_type": "cipd",
-  },
-  Var("dart_root") + "/third_party/browsers/firefox": {
-      "packages": [
-          {
-              "package": "dart/browsers/firefox/${{platform}}",
-              "version": "version:" + Var("firefox_tag"),
-          },
-      ],
-      "condition": "download_firefox",
-      "dep_type": "cipd",
-  },
-}
-
-deps_os = {
-  "win": {
-    Var("dart_root") + "/third_party/cygwin":
-        Var("chromium_git") + "/chromium/deps/cygwin.git" + "@" +
-        "c89e446b273697fadf3a10ff1007a97c0b7de6df",
-    Var("dart_root") + "/third_party/crashpad/crashpad":
-        Var("chromium_git") + "/crashpad/crashpad.git" + "@" +
-        Var("crashpad_rev"),
-    Var("dart_root") + "/third_party/mini_chromium/mini_chromium":
-        Var("chromium_git") + "/chromium/mini_chromium" + "@" +
-        Var("minichromium_rev"),
-    Var("dart_root") + "/third_party/googletest":
-        Var("fuchsia_git") + "/third_party/googletest" + "@" +
-        Var("googletest_rev"),
-  }
 }
 
 hooks = [
@@ -712,47 +361,4 @@ hooks = [
     'pattern': '.',
     'action': ['python3', Var("dart_root") + '/tools/buildtools/update.py'],
   },
-  {
-    # Update the Windows toolchain if necessary.
-    'name': 'win_toolchain',
-    'pattern': '.',
-    'action': ['python3', Var("dart_root") + '/build/vs_toolchain.py', 'update'],
-    'condition': 'checkout_win'
-  },
-  # Install and activate the empscripten SDK.
-  {
-    'name': 'install_emscripten',
-    'pattern': '.',
-    'action': ['python3', Var("dart_root") + '/third_party/emsdk/emsdk.py', 'install',
-        Var('emsdk_ver')],
-    'condition': 'download_emscripten'
-  },
-  {
-    'name': 'activate_emscripten',
-    'pattern': '.',
-    'action': ['python3', Var("dart_root") + '/third_party/emsdk/emsdk.py', 'activate',
-        Var('emsdk_ver')],
-    'condition': 'download_emscripten'
-  },
-  {
-    'name': 'Download Fuchsia system images',
-    'pattern': '.',
-    'action': [
-      'python3',
-      Var("dart_root") + '/build/fuchsia/with_envs.py',
-      Var("dart_root") + '/third_party/fuchsia/test_scripts/update_product_bundles.py',
-      'terminal.x64,terminal.qemu-arm64',
-    ],
-    'condition': 'download_fuchsia_deps'
-  },
-  {
-    'name': 'Generate Fuchsia GN build rules',
-    'pattern': '.',
-    'action': [
-      'python3',
-      Var("dart_root") + '/build/fuchsia/with_envs.py',
-      Var("dart_root") + '/third_party/fuchsia/test_scripts/gen_build_defs.py',
-    ],
-    'condition': 'download_fuchsia_deps'
-  },
 ]
-- 
2.52.0

