--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -97,7 +97,7 @@ config("compiler") {
     cflags_objcc += common_flags
 
     # Linker warnings.
-    if (current_cpu != "arm" && !is_mac && !is_ios) {
+    if (current_cpu != "arm64" && current_cpu != "arm" && !is_mac && !is_ios) {
       # TODO(jochen): Enable this on ChromeOS on arm. http://crbug.com/356580
       ldflags += [ "-Wl,--fatal-warnings" ]
     }

--- a/build/config/linux/BUILD.gn
+++ b/build/config/linux/BUILD.gn
@@ -11,16 +11,11 @@ config("sdk") {
   if (is_clang) {
     # Don't allow visible symbols from libc++ to be re-exported.
     ldflags += [ "-Wl,--exclude-libs=libc++.a" ]
+    ldflags += [ "-fuse-ld=lld" ]
   }
 
-  # Explicitly use static linking for libstdc++ and libgcc to minimize
-  # dependencies.
-  if (!use_flutter_cxx) {
-    ldflags += [
-      "-static-libgcc",
-      "-static-libstdc++",
-    ]
-  }
+  ldflags += string_split(getenv("LDFLAGS"), " ")
+  cflags += string_split(getenv("CFLAGS"), " ")
 
   if (sysroot != "") {
     cflags += [ "--sysroot=" + sysroot ]
@@ -35,13 +30,6 @@ config("sdk") {
                              ],
                              "value") ]
 
-    if (dart_sysroot == "alpine") {
-      # When using the alpine sysroot, prefers lld as linker.
-      if (is_clang) {
-        ldflags += [ "-fuse-ld=lld" ]
-      }
-    }
-
     # When using the pulled Debian sysroot with gcc, we have to specify these
     # explicitly.
     if (dart_sysroot == "debian" && !is_clang) {
