From 7ca1b50e34ac4ea493ecd0028d03721019fc0b24 Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Fri, 22 May 2026 18:48:12 +0800
Subject: [PATCH 3/3] shared-libcxx

---
 build/config/linux/BUILD.gn | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/build/config/linux/BUILD.gn b/build/config/linux/BUILD.gn
index 323e79b20e4..b2e73cbb761 100644
--- a/build/config/linux/BUILD.gn
+++ b/build/config/linux/BUILD.gn
@@ -13,14 +13,8 @@ config("sdk") {
     ldflags += [ "-Wl,--exclude-libs=libc++.a", "-fuse-ld=lld" ]
   }
 
-  # Explicitly use static linking for libstdc++ and libgcc to minimize
-  # dependencies.
-  if (!use_flutter_cxx) {
-    ldflags += [
-      "-static-libgcc",
-      "-static-libstdc++",
-    ]
-  }
+  ldflags += string_split(getenv("LDFLAGS"), " ")
+  cflags += string_split(getenv("CFLAGS"), " ")
 
   if (sysroot != "") {
     cflags += [ "--sysroot=" + sysroot ]
-- 
2.52.0

