From 278773d276660a4395b1062e4f0e94003560f4e8 Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Fri, 22 May 2026 20:04:14 +0800
Subject: [PATCH 4/4] skip toolchain version check

---
 build/config/compiler/BUILD.gn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 4af98d44ee9..9dde20f4be9 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -445,7 +445,7 @@ config("compiler") {
   # changes since artifacts from an older version of the toolchain may or may
   # not be compatible with newer ones. To achieve this, we insert a synthetic
   # define into the compile line.
-  if (is_clang && (is_linux || is_mac || is_ios) && dart_sysroot != "alpine") {
+  if (is_clang && (!is_linux || is_mac || is_ios) && dart_sysroot != "alpine") {
     if (is_linux && host_cpu == "arm64") {
       toolchain_stamp_file =
           "//buildtools/linux-arm64/clang/.versions/clang.cipd_version"
-- 
2.52.0

