From 7d4c1c6524069c67d7c3dcdac7d9bb64da838441 Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Wed, 21 Jan 2026 21:48:42 +0800
Subject: [PATCH] where we are heading prefixes are not needed

---
 build/toolchain/linux/BUILD.gn | 22 +++++++---------------
 1 file changed, 7 insertions(+), 15 deletions(-)

diff --git a/build/toolchain/linux/BUILD.gn b/build/toolchain/linux/BUILD.gn
index 5ff6db3d5d4..648fc486f70 100644
--- a/build/toolchain/linux/BUILD.gn
+++ b/build/toolchain/linux/BUILD.gn
@@ -44,16 +44,8 @@ if (use_rbe) {
   gcc_compiler_prefix = compiler_prefix
 }
 
-# Google's clang does not work for alpine, use alpine's system clang
-if (dart_sysroot == "alpine") {
-  rebased_clang_dir = "/usr/bin"
-} else if (host_cpu == "arm64") {
-  rebased_clang_dir =
-      rebase_path("//buildtools/linux-arm64/clang/bin", root_build_dir)
-} else {
-  rebased_clang_dir =
-      rebase_path("//buildtools/linux-x64/clang/bin", root_build_dir)
-}
+# use system clang
+rebased_clang_dir = "/usr/bin"
 
 template("gcc_toolchain_suite") {
   toolchain_suite(target_name) {
@@ -63,7 +55,7 @@ template("gcc_toolchain_suite") {
 }
 
 gcc_toolchain_suite("arm") {
-  prefix = "arm-linux-gnueabihf-"
+  prefix = ""
   if (arm_toolchain_prefix != "") {
     prefix = arm_toolchain_prefix
   }
@@ -106,7 +98,7 @@ gcc_toolchain_suite("clang_arm") {
 }
 
 gcc_toolchain_suite("arm64") {
-  prefix = "aarch64-linux-gnu-"
+  prefix = ""
   if (arm64_toolchain_prefix != "") {
     prefix = arm64_toolchain_prefix
   }
@@ -171,7 +163,7 @@ gcc_toolchain_suite("clang_x86") {
 }
 
 gcc_toolchain_suite("x86") {
-  prefix = "i686-linux-gnu-"
+  prefix = ""
   if (ia32_toolchain_prefix != "") {
     prefix = ia32_toolchain_prefix
   }
@@ -213,7 +205,7 @@ gcc_toolchain_suite("clang_x64") {
 }
 
 gcc_toolchain_suite("x64") {
-  prefix = "x86_64-linux-gnu-"
+  prefix = ""
   if (x64_toolchain_prefix != "") {
     prefix = x64_toolchain_prefix
   }
@@ -276,7 +268,7 @@ gcc_toolchain_suite("clang_riscv32") {
 }
 
 gcc_toolchain_suite("riscv64") {
-  prefix = "riscv64-linux-gnu-"
+  prefix = ""
   if (riscv64_toolchain_prefix != "") {
     prefix = riscv64_toolchain_prefix
   }
-- 
2.52.0
