From 5db7aaa6918badd85b3fcd8c20cd544b1c42cea9 Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Sun, 18 Jan 2026 23:21:44 +0800
Subject: [PATCH] fix gclient riscv64 venv

---
 depot_tools/.vpython3 | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/depot_tools/.vpython3 b/depot_tools/.vpython3
index ca4eafa71..c83fc0d00 100644
--- a/depot_tools/.vpython3
+++ b/depot_tools/.vpython3
@@ -131,15 +131,8 @@ wheel: <
   version: "version:0.4.4"
 >
 
-# Used by:
-#   tools/win/DebugVisualizers/PRESUBMIT.py
-wheel: <
-  name: "infra/python/wheels/lxml/${vpython_platform}"
-  version: "version:4.9.3"
->
-
 # To be used by developers running unit test coverage.
 wheel: <
   name: "infra/python/wheels/coverage-py3"
   version: "version:7.13.0"
->
\ No newline at end of file
+>
-- 
2.52.0
