From b736a5c1da505555b2d9bd21f05b90b4256c7b2f Mon Sep 17 00:00:00 2001
From: pugaizai <i@pugai.life>
Date: Wed, 21 Jan 2026 21:50:45 +0800
Subject: [PATCH] skip toolchain version check

---
 build/config/compiler/BUILD.gn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index bbbdfb9afb1..c72fa159cef 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -392,7 +392,7 @@ config("compiler") {
   # changes since artifacts from an older version of the toolchain may or may
   # not be compatible with newer ones. To achieve this, we insert a synthetic
   # define into the compile line.
-  if (is_clang && (is_linux || is_mac || is_ios) && dart_sysroot != "alpine") {
+  if (is_clang && (!is_linux || is_mac || is_ios) && dart_sysroot != "alpine") {
     if (is_linux && host_cpu == "arm64") {
       toolchain_stamp_file =
           "//buildtools/linux-arm64/clang/.versions/clang.cipd_version"
-- 
2.52.0

