From 82ddf7955970f6d692cb79aa1b57c3c07353b5d7 Mon Sep 17 00:00:00 2001
From: Ilikara <3435193369@qq.com>
Date: Sat, 7 Jun 2025 20:51:30 +0800
Subject: [PATCH] AOSCOS: remove gzip compress for man page

Signed-off-by: Ilikara <3435193369@qq.com>
---
 Makefile | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/Makefile b/Makefile
index 9028345..6da8321 100644
--- a/Makefile
+++ b/Makefile
@@ -252,9 +252,6 @@ endif
 ifeq (, $(findstring amiga, $(OS)))
 	$(INSTALL) -d $(DESTDIR)$(INSTALLDIRMAN)/man3
 	$(INSTALL) -m 644 man3/miniupnpc.3 $(DESTDIR)$(INSTALLDIRMAN)/man3/miniupnpc.3
-ifneq (, $(findstring linux, $(OS)))
-	gzip -f $(DESTDIR)$(INSTALLDIRMAN)/man3/miniupnpc.3
-endif
 endif
 
 install-static:	updateversion $(FILESTOINSTALL)
-- 
2.49.0

