From: <hesso@pool.math.tu-berlin.de>
Subject: Fix roff errors in the gdome-config(1) manpage.

--- gdome2-0.8.1+debian.orig/gdome-config.1.in	2008-04-30 21:58:27.000000000 +0200
+++ gdome2-0.8.1+debian/dome-config.1.in	2008-04-30 21:58:43.000000000 +0200
@@ -9,7 +9,6 @@
 linker flags that should be used to compile and link programs that use
 \fIgdome2\fP.
 .SH OPTIONS
-.l
 \fIgdome-config\fP accepts the following options:
 .TP 8
 .B  \-\-version
