From: Debian OpenSSL Team <pkg-openssl-devel@lists.alioth.debian.org>
Date: Thu, 13 Dec 2018 23:27:46 +0100
Subject: man-dir

===================================================================
---
 Makefile.org | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.org b/Makefile.org
index d24ff26c8cab..45afbe5cf65d 100644
--- a/Makefile.org
+++ b/Makefile.org
@@ -159,7 +159,7 @@ TESTS = alltests
 
 MAKEFILE= Makefile
 
-MANDIR=$(OPENSSLDIR)/man
+MANDIR=/usr/share/man
 MAN1=1
 MAN3=3
 MANSUFFIX=
