From: Debian OpenSSL Team <pkg-openssl-devel@lists.alioth.debian.org>
Date: Thu, 13 Dec 2018 23:27:46 +0100
Subject: valgrind

===================================================================
---
 crypto/rand/md_rand.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/crypto/rand/md_rand.c b/crypto/rand/md_rand.c
index 2983a3fda487..a16cc804cc56 100644
--- a/crypto/rand/md_rand.c
+++ b/crypto/rand/md_rand.c
@@ -488,6 +488,7 @@ int ssleay_rand_bytes(unsigned char *buf, int num, int pseudo, int lock)
             goto err;
 
 #ifndef PURIFY                  /* purify complains */
+#if 0
         /*
          * The following line uses the supplied buffer as a small source of
          * entropy: since this buffer is often uninitialised it may cause
@@ -497,6 +498,7 @@ int ssleay_rand_bytes(unsigned char *buf, int num, int pseudo, int lock)
          */
         if (!MD_Update(&m, buf, j))
             goto err;
+#endif
 #endif
 
         k = (st_idx + MD_DIGEST_LENGTH / 2) - st_num;
