From: Debian OpenSSL Team <pkg-openssl-devel@lists.alioth.debian.org>
Date: Thu, 13 Dec 2018 23:27:46 +0100
Subject: shared-lib-ext

===================================================================
---
 Configure | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Configure b/Configure
index b9e7c8b6aee7..b2918b52ba27 100755
--- a/Configure
+++ b/Configure
@@ -1852,7 +1852,8 @@ while (<IN>)
 	elsif ($shared_extension ne "" && $shared_extension =~ /^\.s([ol])\.[^\.]*\.[^\.]*$/)
 		{
 		my $sotmp = $1;
-		s/^SHARED_LIBS_LINK_EXTS=.*/SHARED_LIBS_LINK_EXTS=.s$sotmp.\$(SHLIB_MAJOR) .s$sotmp/;
+#		s/^SHARED_LIBS_LINK_EXTS=.*/SHARED_LIBS_LINK_EXTS=.s$sotmp.\$(SHLIB_MAJOR) .s$sotmp/;
+		s/^SHARED_LIBS_LINK_EXTS=.*/SHARED_LIBS_LINK_EXTS=.s$sotmp/;
 		}
 	elsif ($shared_extension ne "" && $shared_extension =~ /^\.[^\.]*\.[^\.]*\.dylib$/)
 		{
