diff -Naur src/emain.h src.glibc232/emain.h
--- src/emain.h	2009-10-12 00:44:52.000000000 -0700
+++ src.glibc232/emain.h	2021-06-16 21:14:09.803770162 -0700
@@ -562,10 +562,6 @@
 #include <unistd.h>
 #endif
 
-#ifdef _meDEF_SYS_ERRLIST
-extern const char *sys_errlist[];
-#endif
-
 /* Standard Types */
 typedef   signed char  meByte ;
 typedef unsigned char  meUByte ;
diff -Naur src/file.c src.glibc232/file.c
--- src/file.c	2009-10-12 00:44:52.000000000 -0700
+++ src.glibc232/file.c	2021-06-16 21:13:31.552297718 -0700
@@ -1325,7 +1325,7 @@
                    )
                 {
                     /* READ ONLY DIR */
-                    mlwrite(MWPAUSE,(meUByte *)"%s: %s", dirbuf, sys_errlist[errno]);
+                    mlwrite(MWPAUSE,(meUByte *)"%s: %s", dirbuf, strerror(errno));
                     /* Zap the filename - it is invalid.
                        We only want a buffer */
                     mlwrite (0,(meUByte *)"[New buffer %s]", getFileBaseName(fname));
@@ -1355,7 +1355,7 @@
             {
                 /* We are not allowed to read the file */
 #if ((defined _UNIX) || (defined _DOS))
-                mlwrite(MWABORT,(meUByte *)"[%s: %s]", fn, sys_errlist[errno]) ;
+                mlwrite(MWABORT,(meUByte *)"[%s: %s]", fn, strerror(errno)) ;
 #else
                 mlwrite(MWABORT,"[cannot read: %s]", fn) ;
 #endif
diff -Naur src/fileio.c src.glibc232/fileio.c
--- src/fileio.c	2009-10-12 00:44:52.000000000 -0700
+++ src.glibc232/fileio.c	2021-06-16 21:13:21.668175677 -0700
@@ -2331,7 +2331,7 @@
                                 if(meRename(filename,filename2) && (ffFileOp(filename,filename2,meRWFLAG_DELETE,-1) <= 0))
                                 {
                                     mlwrite(MWABORT|MWPAUSE,(meUByte *)"[Unable to backup file to %s (%d - %s)]", 
-                                            filename2,errno,sys_errlist[errno]) ;
+                                            filename2,errno,strerror(errno)) ;
                                     if(meUnlink(filename))
                                     {
                                         mlwrite(MWABORT|MWPAUSE,(meUByte *)"[Unable to remove backup file %s]", filename) ;
@@ -2346,7 +2346,7 @@
                         mlwrite(MWABORT|MWPAUSE,(meUByte *)"[Unable to remove backup file %s]", filename) ;
                     else if(meRename(filenameOld,filename) && (ffFileOp(filenameOld,filename,meRWFLAG_DELETE,-1) <= 0))
                         mlwrite(MWABORT|MWPAUSE,(meUByte *)"[Unable to backup file to %s (%d - %s)]", 
-                                filename,errno,sys_errlist[errno]) ;
+                                filename,errno,strerror(errno)) ;
                     else if(bp != NULL)
                     {
                         meUShort ss;
diff -Naur src/spawn.c src.glibc232/spawn.c
--- src/spawn.c	2009-10-12 00:44:53.000000000 -0700
+++ src.glibc232/spawn.c	2021-06-16 21:13:39.826399892 -0700
@@ -219,10 +219,10 @@
         case 0:
             /* we want the children to die on interrupt */
             execlp("xterm", "xterm", "-sl", "200", "-sb", NULL);
-            mlwrite(MWABORT,(meUByte *)"exec failed, %s", sys_errlist[errno]);
+            mlwrite(MWABORT,(meUByte *)"exec failed, %s", strerror(errno));
             meExit(127);
         case -1:
-            ss = mlwrite(MWABORT,(meUByte *)"exec failed, %s", sys_errlist[errno]);
+            ss = mlwrite(MWABORT,(meUByte *)"exec failed, %s", strerror(errno));
         default:
             ss = meTRUE ;
         }
