commit 60f2322055756e8cacfe96530a12c63e9694482c
Author: Paul Licameli <paul.licameli@gmail.com>
Date:   Fri Feb 26 14:45:09 2016 -0500

    Fix warnings for returning false instead of null

diff --git a/src/effects/vamp/LoadVamp.cpp b/src/effects/vamp/LoadVamp.cpp
index 2c73e1d..038a26f 100644
--- a/src/effects/vamp/LoadVamp.cpp
+++ b/src/effects/vamp/LoadVamp.cpp
@@ -266,7 +266,7 @@ Plugin *VampEffectsModule::FindPlugin(const wxString & path,
    Plugin *vp = PluginLoader::getInstance()->loadPlugin(key, 48000); // rate doesn't matter here
    if (!vp)
    {
-      return false;
+      return nullptr;
    }
 
    // We limit the listed plugin outputs to those whose results can
diff --git a/src/import/ImportFLAC.cpp b/src/import/ImportFLAC.cpp
index 2003456..84be299 100644
--- a/src/import/ImportFLAC.cpp
+++ b/src/import/ImportFLAC.cpp
@@ -299,7 +299,7 @@ ImportFileHandle *FLACImportPlugin::Open(const wxString &filename)
    int cnt;
    wxFile binaryFile;
    if (!binaryFile.Open(filename)) {
-      return false; // File not found
+      return nullptr; // File not found
    }
 
 #ifdef USE_LIBID3TAG
@@ -316,7 +316,7 @@ ImportFileHandle *FLACImportPlugin::Open(const wxString &filename)
 
    if (cnt == wxInvalidOffset || strncmp(buf, FLAC_HEADER, 4) != 0) {
       // File is not a FLAC file
-      return false;
+      return nullptr;
    }
 
    // Open the file for import
@@ -325,7 +325,7 @@ ImportFileHandle *FLACImportPlugin::Open(const wxString &filename)
    bool success = handle->Init();
    if (!success) {
       delete handle;
-      return NULL;
+      return nullptr;
    }
 
    return handle;
