diff -Nur orbit2-2.14.19/linc2/src/Makefile.am orbit2-2.14.19.build/linc2/src/Makefile.am
--- orbit2-2.14.19/linc2/src/Makefile.am	2010-02-09 13:05:35.000000000 +0100
+++ orbit2-2.14.19.build/linc2/src/Makefile.am	2013-04-02 16:51:33.434373357 +0200
@@ -12,8 +12,7 @@
 	-I$(top_builddir)/linc2/include  \
 	-I$(top_srcdir)/linc2/include    \
 	$(LINC_CFLAGS)             \
-	$(WARN_CFLAGS)             \
-	-DG_DISABLE_DEPRECATED
+	$(WARN_CFLAGS)             
 
 if OS_WIN32
 else
