Author: Steve Langasek <steve.langasek@ubuntu.com>
Description: Support pre-multiarch directory as a compatibility fallback
Bug-Debian: http://bugs.debian.org/641615
Bug-Ubuntu: https://bugs.launchpad.net/bugs/977966

diff -Nur -x '*.orig' -x '*~' trunk/src/orb/orb-core/orbit-typelib.c trunk.new/src/orb/orb-core/orbit-typelib.c
--- trunk/src/orb/orb-core/orbit-typelib.c	2012-04-18 20:39:57.000000000 +0000
+++ trunk.new/src/orb/orb-core/orbit-typelib.c	2012-04-20 23:27:57.969753251 +0000
@@ -218,6 +218,7 @@
 	paths = g_ptr_array_sized_new (8);
 
 	g_ptr_array_add (paths, g_strdup (ORBIT_TYPELIB_DIR));
+	g_ptr_array_add (paths, g_strdup ("/usr/lib/orbit-2.0"));
 
 	if ((path = g_getenv ("ORBIT_TYPELIB_PATH"))) {
 		char **strv;
