From 618b8d37b16124f39a2039cd80d00bac16f60a3b Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Mon, 29 Jun 2020 14:24:43 -0500
Subject: [PATCH] Fix config file install

---
 Makefile.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 3adbf8a..48227a6 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -781,10 +781,10 @@ install-exec-hook:
 @SUID_TRUE@	progname=$(DESTDIR)$(bindir)/`echo imwheel|sed 's/$(EXEEXT)$$//'|sed '$(transform)'|sed 's/$$/$(EXEEXT)/'` ; \
 @SUID_TRUE@	chown @suid@ $$progname ; \
 @SUID_TRUE@	chmod u+s $$progname
-	$(mkinstalldirs) $(ETCDIR)
-	-[ -f /etc/imwheelrc ] && mv /etc/imwheelrc $(ETCDIR)/imwheelrc || true
-	-[ -f /etc/X11/imwheelrc ] && mv /etc/X11/imwheelrc $(ETCDIR)/imwheelrc || true
-	-[ -f $(ETCDIR)/imwheelrc ] && install -m 644 imwheelrc $(ETCDIR)/imwheelrc.new || install -m 644 imwheelrc $(ETCDIR)/imwheelrc
+	$(mkinstalldirs) $(DESTDIR)$(ETCDIR)
+	-[ -f /etc/imwheelrc ] && mv /etc/imwheelrc $(DESTDIR)$(ETCDIR)/imwheelrc || true
+	-[ -f /etc/X11/imwheelrc ] && mv /etc/X11/imwheelrc $(DESTDIR)$(ETCDIR)/imwheelrc || true
+	-[ -f $(ETCDIR)/imwheelrc ] && install -m 644 imwheelrc $(DESTDIR)$(ETCDIR)/imwheelrc.new || install -m 644 imwheelrc $(DESTDIR)$(ETCDIR)/imwheelrc
 
 extras: $(EXTRA_PROGRAMS)
 
-- 
2.27.0

