From 66b67de3917f7eee48f1aa28fcdaa83c1723beb2 Mon Sep 17 00:00:00 2001
From: Essem <smswessem@gmail.com>
Date: Sat, 6 Dec 2025 16:49:18 -0600
Subject: [PATCH] Fix compilation on GCC 15

---
 lqr/lqr_energy_priv.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/lqr/lqr_energy_priv.h b/lqr/lqr_energy_priv.h
index cef9709..411aacd 100644
--- a/lqr/lqr_energy_priv.h
+++ b/lqr/lqr_energy_priv.h
@@ -56,11 +56,11 @@ gdouble lqr_carver_read_cached_rgba(LqrCarver *r, gint x, gint y, gint channel);
 gdouble lqr_carver_read_cached_custom(LqrCarver *r, gint x, gint y, gint channel);
 
 /* cache brightness (or luma or else) to speedup energy computation */
-gdouble *lqr_carver_generate_rcache_bright();
-gdouble *lqr_carver_generate_rcache_luma();
-gdouble *lqr_carver_generate_rcache_rgba();
-gdouble *lqr_carver_generate_rcache_custom();
-gdouble *lqr_carver_generate_rcache();
+gdouble *lqr_carver_generate_rcache_bright(LqrCarver *r);
+gdouble *lqr_carver_generate_rcache_luma(LqrCarver *r);
+gdouble *lqr_carver_generate_rcache_rgba(LqrCarver *r);
+gdouble *lqr_carver_generate_rcache_custom(LqrCarver *r);
+gdouble *lqr_carver_generate_rcache(LqrCarver *r);
 
 gfloat lqr_energy_builtin_grad_all(gint x, gint y, gint img_width, gint img_height, LqrReadingWindow *rwindow,
                                    LqrGradFunc gf);

