diff --git a/DOCS/Doxyfile_man b/DOCS/Doxyfile_man
index 1767cf5..a7bf53d 100644
--- a/DOCS/Doxyfile_man
+++ b/DOCS/Doxyfile_man
@@ -1850,7 +1850,7 @@ MAN_OUTPUT             = man
 # The default value is: .3.
 # This tag requires that the tag GENERATE_MAN is set to YES.
 
-MAN_EXTENSION          = .3
+MAN_EXTENSION          = .3l
 
 # The MAN_SUBDIR tag determines the name of the directory created within
 # MAN_OUTPUT in which the man pages are placed. If defaults to man followed by
