From a427602cec7dcf517dfb1c60648d6354ee38b7fa Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 6 Apr 2026 22:05:41 +0800
Subject: [PATCH 2/2] AOSCOS: rename man page category to 3l to avoid file
 conflict

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 DOCS/Doxyfile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/DOCS/Doxyfile b/DOCS/Doxyfile
index 5cdf3208c..2ccdd3622 100644
--- a/DOCS/Doxyfile
+++ b/DOCS/Doxyfile
@@ -1144,7 +1144,7 @@ IGNORE_PREFIX          =
 # If the GENERATE_HTML tag is set to YES, doxygen will generate HTML output
 # The default value is: YES.
 
-GENERATE_HTML          = YES
+GENERATE_HTML          = NO
 
 # The HTML_OUTPUT tag is used to specify where the HTML docs will be put. If a
 # relative path is entered the value of OUTPUT_DIRECTORY will be put in front of
@@ -1981,7 +1981,7 @@ RTF_SOURCE_CODE        = NO
 # classes and files.
 # The default value is: NO.
 
-GENERATE_MAN           = NO
+GENERATE_MAN           = YES
 
 # The MAN_OUTPUT tag is used to specify where the man pages will be put. If a
 # relative path is entered the value of OUTPUT_DIRECTORY will be put in front of
@@ -1999,7 +1999,7 @@ MAN_OUTPUT             = man
 # The default value is: .3.
 # This tag requires that the tag GENERATE_MAN is set to YES.
 
-MAN_EXTENSION          = .3
+MAN_EXTENSION          = .3l
 
 # The MAN_SUBDIR tag determines the name of the directory created within
 # MAN_OUTPUT in which the man pages are placed. If defaults to man followed by
-- 
2.52.0

