Description: don't build perf measurement tools in static builds
Bug: http://bugzilla.gnome.org/show_bug.cgi?id=346559

Index: gtk+2.0-2.22.0/Makefile.am
===================================================================
--- gtk+2.0-2.22.0.orig/Makefile.am	2010-09-23 05:00:34.000000000 +0200
+++ gtk+2.0-2.22.0/Makefile.am	2010-09-23 20:29:24.497578749 +0200
@@ -1,7 +1,12 @@
 ## Makefile.am for GTK+
 include $(top_srcdir)/Makefile.decl
 
-SRC_SUBDIRS = gdk gtk modules demos tests perf
+# don't build perf measurement tools in static builds
+if !ENABLE_STATIC
+MAYBE_PERF = perf
+endif
+
+SRC_SUBDIRS = gdk gtk modules demos tests $(MAYBE_PERF)
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
 
 # require automake 1.4
