=== modified file 'gtk/gtk.symbols'
Index: gtk+-2.24.12/gtk/gtk.symbols
===================================================================
--- gtk+-2.24.12.orig/gtk/gtk.symbols	2011-10-18 11:35:29.000000000 +1300
+++ gtk+-2.24.12/gtk/gtk.symbols	2012-09-04 16:42:40.382564335 +1200
@@ -5323,6 +5323,7 @@
 gtk_widget_get_realized
 gtk_widget_set_mapped
 gtk_widget_get_mapped
+ubuntu_gtk_widget_set_has_grab
 #endif
 #endif
 
Index: gtk+-2.24.12/gtk/gtkwidget.c
===================================================================
--- gtk+-2.24.12.orig/gtk/gtkwidget.c	2012-09-04 06:28:02.000000000 +1200
+++ gtk+-2.24.12/gtk/gtkwidget.c	2012-09-04 16:42:40.382564335 +1200
@@ -11449,5 +11449,15 @@
   return res;
 }
 
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  if (has_grab)
+    GTK_OBJECT_FLAGS (widget) |= GTK_HAS_GRAB;
+  else
+    GTK_OBJECT_FLAGS (widget) &= ~(GTK_HAS_GRAB);
+}
+
 #define __GTK_WIDGET_C__
 #include "gtkaliasdef.c"
Index: gtk+-2.24.12/gtk/gtkwidget.h
===================================================================
--- gtk+-2.24.12.orig/gtk/gtkwidget.h	2011-08-16 14:30:52.000000000 +1200
+++ gtk+-2.24.12/gtk/gtkwidget.h	2012-09-04 16:42:40.382564335 +1200
@@ -1346,6 +1346,9 @@
 						       GtkWidget *toplevel,
 						       gpointer   user_data);
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */
