Index: gtk+-2.24.19/gtk/gtksettings.c
===================================================================
--- gtk+-2.24.19.orig/gtk/gtksettings.c	2013-06-28 23:40:03.703367678 +0200
+++ gtk+-2.24.19/gtk/gtksettings.c	2013-06-28 23:40:03.695367587 +0200
@@ -331,7 +331,7 @@
                                              g_param_spec_string ("gtk-fallback-icon-theme",
 								  P_("Fallback Icon Theme Name"),
 								  P_("Name of a icon theme to fall back to"),
-								  NULL,
+								  "gnome",
 								  GTK_PARAM_READWRITE),
                                              NULL);
   g_assert (result == PROP_FALLBACK_ICON_THEME);
