Index: gtk+2.0-2.22.0/modules/input/imviqr.c
===================================================================
--- gtk+2.0-2.22.0.orig/modules/input/imviqr.c	2010-09-23 05:00:38.000000000 +0200
+++ gtk+2.0-2.22.0/modules/input/imviqr.c	2010-09-23 20:29:37.333633329 +0200
@@ -244,7 +244,7 @@
   N_("Vietnamese (VIQR)"), /* Human readable name */
   GETTEXT_PACKAGE,	   /* Translation domain */
    GTK_LOCALEDIR,	   /* Dir for bindtextdomain (not strictly needed for "gtk+") */
-  "vi"			   /* Languages for which this module is the default */
+  ""			   /* Languages for which this module is the default */
 };
 
 static const GtkIMContextInfo *info_list[] = {
