Index: gtk+-2.24.19/gtk/gtktreeview.c
===================================================================
--- gtk+-2.24.19.orig/gtk/gtktreeview.c	2013-06-28 23:40:08.435421781 +0200
+++ gtk+-2.24.19/gtk/gtktreeview.c	2013-06-28 23:40:08.427421690 +0200
@@ -9628,7 +9628,8 @@
 
       if (cursor_path)
 	{
-	  if (tree_view->priv->selection->type == GTK_SELECTION_MULTIPLE)
+	  if (tree_view->priv->selection->type == GTK_SELECTION_MULTIPLE ||
+	      tree_view->priv->selection->type == GTK_SELECTION_SINGLE)
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, FALSE, FALSE);
 	  else
 	    gtk_tree_view_real_set_cursor (tree_view, cursor_path, TRUE, FALSE);
