Index: gtk+-2.24.18/gtk/updateiconcache.c
===================================================================
--- gtk+-2.24.18.orig/gtk/updateiconcache.c	2013-05-17 11:56:55.032595934 +0200
+++ gtk+-2.24.18/gtk/updateiconcache.c	2013-05-17 11:56:55.032595934 +0200
@@ -678,7 +678,7 @@
 		      directories = g_list_append (directories, g_strdup (subdir));
 		    }
 		  else
-		    dir_index = 0xffff;
+		    continue;
 		}
 		
 	      image = g_new0 (Image, 1);
