--- a/modules/printbackends/lpr/gtkprintbackendlpr.c
+++ b/modules/printbackends/lpr/gtkprintbackendlpr.c
@@ -31,6 +31,8 @@
 #include <cairo.h>
 #include <cairo-ps.h>
 
+#include <cairo-pdf.h>
+
 #include <glib/gi18n-lib.h>
 
 #include <gtk/gtk.h>
@@ -209,7 +211,7 @@
 {
   cairo_surface_t *surface;
   
-  surface = cairo_ps_surface_create_for_stream (_cairo_write, cache_io, width, height);
+  surface = cairo_pdf_surface_create_for_stream (_cairo_write, cache_io, width, height);
 
   cairo_surface_set_fallback_resolution (surface,
                                          2.0 * gtk_print_settings_get_printer_lpi (settings),
