Index: gtk+-2.24.12/gtk/gtkrecentmanager.c
===================================================================
--- gtk+-2.24.12.orig/gtk/gtkrecentmanager.c	2012-09-04 06:28:02.000000000 +1200
+++ gtk+-2.24.12/gtk/gtkrecentmanager.c	2012-09-04 16:42:53.590563878 +1200
@@ -907,7 +907,7 @@
   GtkRecentManager *manager = user_data;
   GtkRecentData recent_data;
   GFileInfo *file_info;
-  gchar *uri;
+  gchar *uri, *basename;
 
   uri = g_file_get_uri (file);
 
@@ -931,7 +931,11 @@
       g_object_unref (file_info);
     }
   else
-    recent_data.mime_type = g_strdup (GTK_RECENT_DEFAULT_MIME); /* FIXME: maybe we should make up the MIME type from the filename's extension */
+    {
+      basename = g_file_get_basename (file);
+      recent_data.mime_type = g_content_type_guess (basename, NULL, 0, NULL);
+      g_free (basename);
+    }
 
   recent_data.app_name = g_strdup (g_get_application_name ());
   recent_data.app_exec = g_strjoin (" ", g_get_prgname (), "%u", NULL);
