Index: b/gdk/x11/gdksettings.c
===================================================================
--- a/gdk/x11/gdksettings.c
+++ b/gdk/x11/gdksettings.c
@@ -80,7 +80,8 @@
   "Net/EnableInputFeedbackSounds\0" "gtk-enable-input-feedback-sounds\0"
   "Net/EnableEventSounds\0"   "gtk-enable-event-sounds\0"
   "Gtk/CursorBlinkTimeout\0"  "gtk-cursor-blink-timeout\0"
-  "Gtk/AutoMnemonics\0"       "gtk-auto-mnemonics\0";
+  "Gtk/AutoMnemonics\0"       "gtk-auto-mnemonics\0"
+  "Gtk/ShellShowsMenubar\0"   "gtk-shell-shows-menubar\0";
 
 
 static const struct
@@ -133,5 +134,6 @@
   { 1573, 1603 },
   { 1636, 1658 },
   { 1682, 1705 },
-  { 1730, 1748 }
+  { 1730, 1748 },
+  { 1767, 1789 }
 };
Index: b/gtk/gtksettings.c
===================================================================
--- a/gtk/gtksettings.c
+++ b/gtk/gtksettings.c
@@ -139,7 +139,8 @@
   PROP_LABEL_SELECT_ON_FOCUS,
   PROP_COLOR_PALETTE,
   PROP_IM_PREEDIT_STYLE,
-  PROP_IM_STATUS_STYLE
+  PROP_IM_STATUS_STYLE,
+  PROP_SHELL_SHOWS_MENUBAR
 };
 
 /* --- prototypes --- */
@@ -1205,6 +1206,16 @@
                                                                 GTK_PARAM_READWRITE),
                                              gtk_rc_property_parse_enum);
   g_assert (result == PROP_IM_STATUS_STYLE);
+
+  result = settings_install_property_parser (class,
+                                             g_param_spec_boolean ("gtk-shell-shows-menubar",
+                                                                   P_("Desktop shell shows the menubar"),
+                                                                   P_("Set to TRUE if the desktop environment "
+                                                                      "is displaying the menubar, FALSE if "
+                                                                      "the app should display it itself."),
+                                                                   FALSE, GTK_PARAM_READWRITE),
+                                             NULL);
+  g_assert (result == PROP_SHELL_SHOWS_MENUBAR);
 }
 
 static void
