--- a/gtk/gui-thread-check/profiler/gui-thread-check.c	2019-07-17 20:49:24.379967437 +0200
+++ b/gtk/gui-thread-check/profiler/gui-thread-check.c	2019-07-17 20:49:47.272364112 +0200
@@ -93,6 +93,6 @@
 	
 	mono_profiler_install (NULL, NULL);
 	mono_profiler_install_enter_leave (simple_method_enter, NULL);
-	mono_profiler_set_events (MONO_PROFILE_ENTER_LEAVE);
+	mono_profiler_set_events (1 << 12);
 }
 
