From 4d4f8cbf272ce31e51a997bd73d18f529292ec4b Mon Sep 17 00:00:00 2001
From: cantabile <cantabile.desu@gmail.com>
Date: Sun, 26 Jan 2020 13:00:16 +0200
Subject: [PATCH] Fix compilation broken by typo two commits ago

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 0ce8e18f..123a3488 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -100,7 +100,7 @@ libvapoursynth_la_SOURCES += src/core/jitasm.h \
 							 src/core/kernel/x86/transpose_sse2.c
 
 libvapoursynth_la_LIBADD += libvapoursynth_avx2.la
-#endif # X86ASM
+endif # X86ASM
 
 if PYTHONMODULE
 pyexec_LTLIBRARIES = vapoursynth.la
-- 
2.26.2

