Index: mp3blaster-3.2.5/mpegsound/oggplayer.cc
===================================================================
--- mp3blaster-3.2.5.orig/mpegsound/oggplayer.cc	2009-03-06 17:18:00.000000000 +0100
+++ mp3blaster-3.2.5/mpegsound/oggplayer.cc	2009-03-06 17:18:30.000000000 +0100
@@ -18,7 +18,11 @@
 {
 	of = NULL;
 	wordsize = 2; //2 bytes
+#ifdef WORDS_BIGENDIAN
+	bigendian = 1;
+#else
 	bigendian = 0;
+#endif
 	signeddata = 1;
 	mono = 0;
 	downfreq = 0;
