Index: mp3blaster-3.2.5/mpegsound/oggplayer.cc
===================================================================
--- mp3blaster-3.2.5.orig/mpegsound/oggplayer.cc	2009-03-06 17:18:03.000000000 +0100
+++ mp3blaster-3.2.5/mpegsound/oggplayer.cc	2009-03-06 17:18:04.000000000 +0100
@@ -36,6 +36,7 @@
 	{
 		ov_clear(of);
 		delete of;
+		of = NULL;
 	}
 }
 
@@ -131,6 +132,8 @@
 {
 	if (of)
 		ov_clear(of);
+		delete of;
+		of = NULL;
 }
 
 void Oggplayer::setforcetomono(short flag)
