Index: mp3blaster-3.2.5/src/main.cc
===================================================================
--- mp3blaster-3.2.5.orig/src/main.cc	2009-03-06 17:18:05.000000000 +0100
+++ mp3blaster-3.2.5/src/main.cc	2009-03-06 17:18:07.000000000 +0100
@@ -1904,7 +1904,7 @@
 						stop_song(); //status => AC_NONE
 					} else {
 						//output buffer full, wait a bit.
-						USLEEP(10000);
+						// USLEEP(10000);
 					}
 				} else {
 					/* TODO: different thread for status updates. Now, this function
@@ -1913,7 +1913,7 @@
 					 * costly as well..
 					 */
 					update_play_display();
-					USLEEP(10000);
+					// USLEEP(10000);
 				}
 			}
 
