diff -Naur tesseract-4.1.1/src/ccutil/helpers.h tesseract-4.1.1.gcc13/src/ccutil/helpers.h
--- tesseract-4.1.1/src/ccutil/helpers.h	2019-12-26 14:21:51.000000000 +0000
+++ tesseract-4.1.1.gcc13/src/ccutil/helpers.h	2023-06-07 05:51:37.961264128 +0000
@@ -26,6 +26,7 @@
 #include <cstring>
 #include <functional>
 #include <string>
+#include <cstdint>
 
 // TODO(rays) Put the rest of the helpers in the namespace.
 namespace tesseract {
diff -Naur tesseract-4.1.1/src/dict/matchdefs.h tesseract-4.1.1.gcc13/src/dict/matchdefs.h
--- tesseract-4.1.1/src/dict/matchdefs.h	2019-12-26 14:21:51.000000000 +0000
+++ tesseract-4.1.1.gcc13/src/dict/matchdefs.h	2023-06-07 05:52:07.677264128 +0000
@@ -22,6 +22,7 @@
           Include Files and Type Defines
 ----------------------------------------------------------------------------**/
 #include <cstdio>
+#include <cstdint>
 #include "unichar.h"
 
 /* define the maximum number of classes defined for any matcher
