diff -Naur libatomic_ops-7.6.2-old/src/atomic_ops/sysdeps/gcc/riscv.h libatomic_ops-7.6.2/src/atomic_ops/sysdeps/gcc/riscv.h
--- libatomic_ops-7.6.2-old/src/atomic_ops/sysdeps/gcc/riscv.h	1970-01-01 08:00:00.000000000 +0800
+++ libatomic_ops-7.6.2/src/atomic_ops/sysdeps/gcc/riscv.h	2018-02-19 01:54:13.562102437 +0800
@@ -0,0 +1,24 @@
+/*
+ * THIS MATERIAL IS PROVIDED AS IS, WITH ABSOLUTELY NO WARRANTY EXPRESSED
+ * OR IMPLIED.  ANY USE IS AT YOUR OWN RISK.
+ *
+ * Permission is hereby granted to use or copy this program
+ * for any purpose,  provided the above notices are retained on all copies.
+ * Permission to modify the code and to distribute modified code is granted,
+ * provided the above notices are retained, and a notice that the code was
+ * modified is included with the above copyright notice.
+ *
+ */
+
+#define AO_PREFER_GENERALIZED
+#undef AO_PREFER_GENERALIZED
+
+#if (defined(__clang__) && !AO_CLANG_PREREQ(3, 8)) || defined(__APPLE_CC__)
+  /* __GCC_HAVE_SYNC_COMPARE_AND_SWAP_n macros are missing.     */
+# define AO_GCC_FORCE_HAVE_CAS
+#endif
+
+#include "generic.h"
+
+#undef AO_GCC_FORCE_HAVE_CAS
+#undef AO_GCC_HAVE_double_SYNC_CAS
diff -Naur libatomic_ops-7.6.2-old/src/atomic_ops.h libatomic_ops-7.6.2/src/atomic_ops.h
--- libatomic_ops-7.6.2-old/src/atomic_ops.h	2017-12-24 16:31:12.000000000 +0800
+++ libatomic_ops-7.6.2/src/atomic_ops.h	2018-02-19 01:54:13.565435808 +0800
@@ -352,6 +352,9 @@
 # if defined(__tile__)
 #   include "atomic_ops/sysdeps/gcc/tile.h"
 # endif
+# if defined(__riscv)
+#   include "atomic_ops/sysdeps/gcc/riscv.h"
+# endif
 #endif /* __GNUC__ && !AO_USE_PTHREAD_DEFS */
 
 #if (defined(__IBMC__) || defined(__IBMCPP__)) && !defined(__GNUC__) \
