diff --git a/pyproject.toml b/pyproject.toml
index 433de46..e50c713 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -18,7 +18,7 @@ classifiers = [
     "Topic :: Software Development :: Build Tools",
 ]
 dependencies = [
-    "flit-core~=3.5",
+    "flit-core>=3.5",
     "setuptools_scm>=6.4",
     "tomli; python_version < '3.11'",
 ]
@@ -33,7 +33,7 @@ repository = "https://gitlab.com/WillDaSilva/flit_scm"
 
 [build-system]
 requires = [
-    "flit-core~=3.5",
+    "flit-core>=3.5",
     "setuptools_scm>=6.4",
     "tomli; python_version < '3.11'",
 ]
