diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index afe58b9..a8c6773 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -953,6 +953,8 @@ if(HAVE_DBUS)
   list(APPEND HEADERS ${CMAKE_CURRENT_BINARY_DIR}/dbus/avahientrygroup.h)
   list(APPEND SOURCES ${CMAKE_CURRENT_BINARY_DIR}/dbus/avahientrygroup.cpp)
 
+  set_source_files_properties( dbus/org.freedesktop.UDisks.xml dbus/org.freedesktop.UDisks.Device.xml PROPERTIES NO_NAMESPACE TRUE)
+
   # DeviceKit DBUS interfaces
   if(HAVE_DEVICEKIT)
     set_source_files_properties(dbus/org.freedesktop.UDisks.xml
