diff -up clementine-1.2.3/gst/moodbar/CMakeLists.txt.moodbar_flags clementine-1.2.3/gst/moodbar/CMakeLists.txt
--- clementine-1.2.3/gst/moodbar/CMakeLists.txt.moodbar_flags	2014-09-22 04:44:59.000000000 -0500
+++ clementine-1.2.3/gst/moodbar/CMakeLists.txt	2015-05-04 14:02:29.008369066 -0500
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 2.6)
 
-set(CMAKE_C_FLAGS "-Wall")
-set(CMAKE_CXX_FLAGS "-Woverloaded-virtual -Wall")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Woverloaded-virtual -Wall")
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
 

