From 88ae103e999a7ffd10df16b82f5474b5ea801595 Mon Sep 17 00:00:00 2001
From: Matthieu Baerts <matttbe@gmail.com>
Date: Tue, 24 Mar 2015 22:56:11 +0100
Subject: [PATCH 3/3] lock-screen.sh used xdg-screensaver if available

---
 shared-files/scripts/lock-screen.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/shared-files/scripts/lock-screen.sh b/shared-files/scripts/lock-screen.sh
index 72ad7df..c1798f5 100755
--- a/shared-files/scripts/lock-screen.sh
+++ b/shared-files/scripts/lock-screen.sh
@@ -30,6 +30,8 @@ elif test -n "`echo $PS_OUTPUT | grep light-locker`"; then
 	light-locker-command --lock
 elif hash xlock 2> /dev/null; then
 	xlock
+elif hash xdg-screensaver 2> /dev/null; then
+	xdg-screensaver lock
 else # check is another "*-screensaver" daemon is running
 	# we need to relaunch ps, easier to parse compare to PS_OUTPUT which contains only one big line
 	SCREENSAVER=`ps -u $USER -wwo pid,cmd | grep "\-[s]creensaver" | awk '{print $2}'`
-- 
2.1.0

