From 2e3ede251589fb4d2bae0c3ad71c42c3fa1a3486 Mon Sep 17 00:00:00 2001
From: liushuyu <liushuyu001@gmail.com>
Date: Tue, 6 Feb 2024 07:55:22 -0800
Subject: [PATCH 87/87] libphobos/libdruntime: add autoconf

---
 libphobos/configure               | 13 +++++++++++++
 libphobos/libdruntime/Makefile.in |  2 ++
 2 files changed, 15 insertions(+)

diff --git a/libphobos/configure b/libphobos/configure
index 925c53c5f5e..bcc5af1fd32 100755
--- a/libphobos/configure
+++ b/libphobos/configure
@@ -690,6 +690,8 @@ DRUNTIME_CPU_S390_FALSE
 DRUNTIME_CPU_S390_TRUE
 DRUNTIME_CPU_SYSTEMZ_FALSE
 DRUNTIME_CPU_SYSTEMZ_TRUE
+DRUNTIME_CPU_LOONGARCH64_FALSE
+DRUNTIME_CPU_LOONGARCH64_TRUE
 DRUNTIME_CPU_X86_FALSE
 DRUNTIME_CPU_X86_TRUE
 DRUNTIME_CPU_POWERPC_FALSE
@@ -14173,6 +14175,9 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
       s390)
                druntime_target_cpu_parsed="s390"
                ;;
+      loongarch64*)
+               druntime_target_cpu_parsed="loongarch64"
+               ;;
   esac
    if test "$druntime_target_cpu_parsed" = "aarch64"; then
   DRUNTIME_CPU_AARCH64_TRUE=
@@ -14230,6 +14235,14 @@ else
   DRUNTIME_CPU_S390_FALSE=
 fi
 
+if test "$druntime_target_cpu_parsed" = "loongarch64"; then
+  DRUNTIME_CPU_LOONGARCH64_TRUE=
+  DRUNTIME_CPU_LOONGARCH64_FALSE='#'
+else
+  DRUNTIME_CPU_LOONGARCH64_TRUE='#'
+  DRUNTIME_CPU_LOONGARCH64_FALSE=
+fi
+
 
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for target OS" >&5
diff --git a/libphobos/libdruntime/Makefile.in b/libphobos/libdruntime/Makefile.in
index 797d6435a7c..665494cbebe 100644
--- a/libphobos/libdruntime/Makefile.in
+++ b/libphobos/libdruntime/Makefile.in
@@ -483,6 +483,7 @@ am__objects_23 = core/sys/solaris/dlfcn.lo core/sys/solaris/elf.lo \
 @DRUNTIME_CPU_POWERPC_TRUE@am__objects_28 = config/powerpc/libgdruntime_la-switchcontext.lo
 @DRUNTIME_CPU_X86_TRUE@@DRUNTIME_OS_MINGW_TRUE@am__objects_29 = config/mingw/libgdruntime_la-switchcontext.lo
 @DRUNTIME_CPU_X86_TRUE@@DRUNTIME_OS_MINGW_FALSE@am__objects_30 = config/x86/libgdruntime_la-switchcontext.lo
+@DRUNTIME_CPU_LOONGARCH64_TRUE@am__objects_128 = config/loongarch64/libgdruntime_la-switchcontext.lo
 @DRUNTIME_CPU_SYSTEMZ_TRUE@am__objects_31 = config/systemz/libgdruntime_la-get_tls_offset.lo
 @DRUNTIME_CPU_S390_TRUE@am__objects_32 = config/s390/libgdruntime_la-get_tls_offset.lo
 am__objects_33 = $(am__objects_6) $(am__objects_8) $(am__objects_10) \
@@ -506,6 +507,7 @@ am__objects_36 = core/stdc/libgdruntime_convenience_la-errno_.lo
 @DRUNTIME_CPU_POWERPC_TRUE@am__objects_41 = config/powerpc/libgdruntime_convenience_la-switchcontext.lo
 @DRUNTIME_CPU_X86_TRUE@@DRUNTIME_OS_MINGW_TRUE@am__objects_42 = config/mingw/libgdruntime_convenience_la-switchcontext.lo
 @DRUNTIME_CPU_X86_TRUE@@DRUNTIME_OS_MINGW_FALSE@am__objects_43 = config/x86/libgdruntime_convenience_la-switchcontext.lo
+@DRUNTIME_CPU_LOONGARCH64_TRUE@am__objects_129 = config/loongarch64/libgdruntime_convenience_la-switchcontext.lo
 @DRUNTIME_CPU_SYSTEMZ_TRUE@am__objects_44 = config/systemz/libgdruntime_convenience_la-get_tls_offset.lo
 @DRUNTIME_CPU_S390_TRUE@am__objects_45 = config/s390/libgdruntime_convenience_la-get_tls_offset.lo
 am__objects_46 = $(am__objects_6) $(am__objects_8) $(am__objects_10) \
-- 
2.43.0

