Description: tmp file is supposed to be writable
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: not-needed
Last-Update: 2012-04-20

Index: openjpeg/applications/jpip/libopenjpip/openjpip.c
===================================================================
--- openjpeg.orig/applications/jpip/libopenjpip/openjpip.c	2012-04-20 11:41:15.093920674 +0200
+++ openjpeg/applications/jpip/libopenjpip/openjpip.c	2012-04-20 11:41:27.837453425 +0200
@@ -127,7 +127,7 @@
 void send_responsedata( server_record_t *rec, QR_t *qr)
 {
   int fd;
-  char tmpfname[] = "tmpjpipstream.jpp";
+  const char tmpfname[] = "/tmp/tmpjpipstream.jpp";
   Byte_t *jpipstream;
   Byte8_t len_of_jpipstream;
 
