diff -Naur httpx-0.23.3/httpx/_config.py httpx-0.23.3.modded/httpx/_config.py
--- httpx-0.23.3/httpx/_config.py	2023-01-04 04:39:43.000000000 -0500
+++ httpx-0.23.3.modded/httpx/_config.py	2023-02-15 23:31:34.492573411 -0500
@@ -4,8 +4,6 @@
 import typing
 from pathlib import Path
 
-import certifi
-
 from ._compat import set_minimum_tls_version_1_2
 from ._models import Headers
 from ._types import CertTypes, HeaderTypes, TimeoutTypes, URLTypes, VerifyTypes
@@ -58,7 +56,7 @@
     SSL Configuration.
     """
 
-    DEFAULT_CA_BUNDLE_PATH = Path(certifi.where())
+    DEFAULT_CA_BUNDLE_PATH = Path("/etc/ssl/certs/ca-certificates.crt")
 
     def __init__(
         self,
diff -Naur httpx-0.23.3/pyproject.toml httpx-0.23.3.modded/pyproject.toml
--- httpx-0.23.3/pyproject.toml	2023-01-04 04:39:43.000000000 -0500
+++ httpx-0.23.3.modded/pyproject.toml	2023-02-15 23:32:15.577629788 -0500
@@ -28,7 +28,6 @@
     "Topic :: Internet :: WWW/HTTP",
 ]
 dependencies = [
-    "certifi",
     "httpcore>=0.15.0,<0.17.0",
     "rfc3986[idna2008]>=1.3,<2",
     "sniffio",
diff -Naur httpx-0.23.3/tests/test_config.py httpx-0.23.3.modded/tests/test_config.py
--- httpx-0.23.3/tests/test_config.py	2023-01-04 04:39:43.000000000 -0500
+++ httpx-0.23.3.modded/tests/test_config.py	2023-02-15 23:31:59.783608070 -0500
@@ -3,7 +3,6 @@
 import sys
 from pathlib import Path
 
-import certifi
 import pytest
 
 import httpx
@@ -21,7 +20,7 @@
 
 
 def test_load_ssl_config_verify_existing_file():
-    context = httpx.create_ssl_context(verify=certifi.where())
+    context = httpx.create_ssl_context(verify="/etc/ssl/certs/ca-certificates.crt")
     assert context.verify_mode == ssl.VerifyMode.CERT_REQUIRED
     assert context.check_hostname is True
 
@@ -44,7 +43,7 @@
 
 
 def test_load_ssl_config_verify_directory():
-    path = Path(certifi.where()).parent
+    path = Path("/etc/ssl/certs/ca-certificates.crt").parent
     context = httpx.create_ssl_context(verify=str(path))
     assert context.verify_mode == ssl.VerifyMode.CERT_REQUIRED
     assert context.check_hostname is True
