diff -Naur hdparm-9.65/Makefile hdparm-9.65.strip/Makefile
--- hdparm-9.65/Makefile	2020-06-26 03:55:43.000000000 -0700
+++ hdparm-9.65.strip/Makefile	2022-12-15 21:18:13.234122861 -0800
@@ -23,14 +23,13 @@
 OBJS = hdparm.o identify.o sgio.o sysfs.o geom.o fallocate.o fibmap.o fwdownload.o dvdspeed.o wdidle3.o apt.o
 
 all:
-	$(MAKE) -j4 hdparm
+	$(MAKE) hdparm
 
 static:
 	LDFLAGS=-static $(MAKE)
 
 hdparm: Makefile hdparm.h sgio.h $(OBJS)
 	$(CC) $(LDFLAGS) -o hdparm $(OBJS)
-	$(STRIP) hdparm
 
 hdparm.o:	hdparm.h sgio.h
 
