--- hdparm-9.58/Makefile	2017-05-01 12:08:42.000000000 -0700
+++ hdparm-9.58.Makefile/Makefile	2021-07-06 09:55:38.822809100 -0700
@@ -11,7 +11,6 @@
 oldmandir = $(manprefix)/man
 
 CC ?= gcc
-STRIP ?= strip
 
 CFLAGS := -O2 -W -Wall -Wbad-function-cast -Wcast-align -Wpointer-arith -Wcast-qual -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -fkeep-inline-functions -Wwrite-strings -Waggregate-return -Wnested-externs -Wtrigraphs $(CFLAGS)
 
@@ -25,11 +24,10 @@
 OBJS = hdparm.o identify.o sgio.o sysfs.o geom.o fallocate.o fibmap.o fwdownload.o dvdspeed.o wdidle3.o apt.o
 
 all:
-	$(MAKE) -j4 hdparm
+	$(MAKE) hdparm
 
 hdparm: hdparm.h sgio.h $(OBJS)
 	$(CC) $(LDFLAGS) -o hdparm $(OBJS)
-	$(STRIP) hdparm
 
 hdparm.o:	hdparm.h sgio.h
 
