diff -Naur hdparm-9.58.orig/Makefile hdparm-9.58/Makefile
--- hdparm-9.58.orig/Makefile	2021-09-05 23:58:33.222494242 +0800
+++ hdparm-9.58/Makefile	2021-09-05 23:58:43.810917769 +0800
@@ -15,7 +15,7 @@
 
 CFLAGS := -O2 -W -Wall -Wbad-function-cast -Wcast-align -Wpointer-arith -Wcast-qual -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -fkeep-inline-functions -Wwrite-strings -Waggregate-return -Wnested-externs -Wtrigraphs $(CFLAGS)
 
-LDFLAGS = -s
+LDFLAGS = 
 #LDFLAGS = -s -static
 INSTALL = install
 INSTALL_DATA = $(INSTALL) -m 644
@@ -29,7 +29,6 @@
 
 hdparm: hdparm.h sgio.h $(OBJS)
 	$(CC) $(LDFLAGS) -o hdparm $(OBJS)
-	$(STRIP) hdparm
 
 hdparm.o:	hdparm.h sgio.h
 
