Author: Fabian Greffrath <fabian@debian-unofficial.org>
Author: Michael Karcher <debian@mkarcher.dialup.fu-berlin.de>
Description: Not every arch has <io.h>, include <unistd.h> instead.

--- unace-nonfree-2.5.orig/makefile
+++ unace-nonfree-2.5/makefile
@@ -17,7 +17,7 @@ SRCS            = $(SRCSsl)/
 INCLSYSDIR 	= //usr/include/sys
 INCLDIR 	= //usr/include
 
-DEFINES		= -D__LINUX__ -D__unix__ -D__GCC__ -D__HAS_IO_H__
+DEFINES		= -D__LINUX__ -D__unix__ -D__GCC__ -U__HAS_IO_H__
 
 ifdef DEBUG
 LSWITCHES	=
--- unace-nonfree-2.5.orig/source/base/all/bufread/bufread.c
+++ unace-nonfree-2.5/source/base/all/bufread/bufread.c
@@ -3,6 +3,10 @@
 
 #ifdef __HAS_IO_H__
 	#include <io.h>
+#else
+#ifdef __unix__
+	#include <unistd.h>
+#endif
 #endif
 
 #include <string.h>
--- unace-nonfree-2.5.orig/source/base/all/lfn/lfn.h
+++ unace-nonfree-2.5/source/base/all/lfn/lfn.h
@@ -6,6 +6,10 @@
 
 #ifdef __HAS_IO_H__
 	#include <io.h>
+#else
+#ifdef __unix__
+	#include <unistd.h>
+#endif
 #endif
 
 #ifdef INCL_BASE_LFN_EXCLUSIVE
