Author: Michael Karcher <debian@mkarcher.dialup.fu-berlin.de>
Description: Fixes prototypes in unincore.h to match uninorm.c.

--- unace-nonfree-2.5.orig/source/base/all/uninorm/unincore.h
+++ unace-nonfree-2.5/source/base/all/uninorm/unincore.h
@@ -3,28 +3,30 @@
  *
  */
 
+typedef unsigned int uint;
+
 /* integer string functions                             */
-int    istrlen                  (int  * str             );
-int    istrcmp                  (int  * s1,  int  * s2  );
+uint   istrlen                  (uint * str             );
+uint   istrcmp                  (uint * s1,  uint * s2  );
 
 /* unicode data accessors                               */
-int    is_excluded              (int    c               );
-int    get_combining_class      (int    c               );
-int    get_recombined_codepoint (int    c1,  int    c2  );
-int  * get_decomposition        (int  * buf, int    c   );
+uint   is_excluded              (uint   c               );
+uint   get_combining_class      (uint   c               );
+int    get_recombined_codepoint (uint   c1,  uint   c2  );
+uint * get_decomposition        (uint * buf, uint   c   );
 
 /* utf en/decoding functions                            */
 void encode_utf8(char *buf, unsigned *str);
 void decode_utf8(unsigned *buf, char *str);
 
 /* (de)composition functions                            */
-void decompose_recursive      (int  * buf, int    c   );
-void canonical_order          (int  * str             );
-void canonical_composition    (int  * str             );
-void canonical_decomposition  (int  * buf, int  * str );
+void decompose_recursive      (uint * buf, uint   c   );
+void canonical_order          (uint * str             );
+void canonical_composition    (uint * str             );
+void canonical_decomposition  (uint * buf, uint * str );
 
 /* normalization functions                              */
-void normalize_nfd            (int  * buf, int  * str );
-void normalize_nfc            (int  * buf, int  * str );
+void normalize_nfd            (uint * buf, uint * str );
+void normalize_nfc            (uint * buf, uint * str );
 void normalize_nfd_utf8       (char * buf, char * str );
 void normalize_nfc_utf8       (char * buf, char * str );
--- unace-nonfree-2.5.orig/source/base/all/uninorm/uninorm.c
+++ unace-nonfree-2.5/source/base/all/uninorm/uninorm.c
@@ -101,4 +101,4 @@ INT BASE_UNINORM_CP850ToUTF8NFD(UCHAR *u
   }
 }
 
-*/
\ No newline at end of file
+*/
