Author: Fabian Greffrath <fabian+debian@greffrath.com>
Description: Fix format string vulnarability.

--- unace-nonfree-2.5.orig/source/apps/exe/output/output.c
+++ unace-nonfree-2.5/source/apps/exe/output/output.c
@@ -125,7 +125,7 @@ CHAR      DeleteStr[223];
     memset(&DeleteStr[74], ' ', 74);
     DeleteStr[222] = 0;
 
-    printf(DeleteStr);
+    printf("%s", DeleteStr);
 #ifndef __sun__
     fflush(stdout);
 #endif
