diff -Naur a/scripts/plymouth-update-initrd b/scripts/plymouth-update-initrd
--- a/scripts/plymouth-update-initrd	2012-09-27 00:50:53.000000000 +0800
+++ b/scripts/plymouth-update-initrd	2016-06-28 20:20:46.380815403 +0800
@@ -1,2 +1,5 @@
 #!/bin/bash
-mkinitrd -f /boot/initrd-$(uname -r).img $(uname -r)
+mkinitrd -f /boot/initramfs-$(uname -r).img $(uname -r)
+if [[ -x /usr/bin/grub-mkconfig ]]; then
+	grub-mkconfig -o /boot/grub/grub.cfg
+fi
