diff --git a-llvm/lib/Target/RISCV/MCTargetDesc/RISCVBaseInfo.cpp b-llvm/lib/Target/RISCV/MCTargetDesc/RISCVBaseInfo.cpp
index 60e86093d9f4..f209c51f302d 100644
--- a-llvm/lib/Target/RISCV/MCTargetDesc/RISCVBaseInfo.cpp
+++ b-llvm/lib/Target/RISCV/MCTargetDesc/RISCVBaseInfo.cpp
@@ -52,14 +52,19 @@ ABI computeTargetABI(const Triple &TT, FeatureBitset FeatureBits,
     return TargetABI;
 
   // For now, default to the ilp32/ilp32e/lp64 ABI if no explicit ABI is given
-  // or an invalid/unrecognised string is given. In the future, it might be
-  // worth changing this to default to ilp32f/lp64f and ilp32d/lp64d when
-  // hardware support for floating point is present.
+  // or an invalid/unrecognised string is given and targeting bare-metal,
+  // ilp32d/lp64d when targeting some operating system.
   if (IsRV32E)
     return ABI_ILP32E;
   if (IsRV64)
-    return ABI_LP64;
-  return ABI_ILP32;
+    if (TT.getOS() == Triple::UnknownOS)
+      return ABI_LP64;
+    else
+      return ABI_LP64D;
+  if (TT.getOS() == Triple::UnknownOS)
+    return ABI_ILP32;
+  else
+    return ABI_ILP32D;
 }
 
 ABI getTargetABI(StringRef ABIName) {
