From: Jonas Genannt <jonas.genannt@capi2name.de>
Subject: setserial Makefile patch


Index: setserial/Makefile.in
===================================================================
--- setserial.orig/Makefile.in	2012-06-14 16:18:55.000000000 +0200
+++ setserial/Makefile.in	2012-06-18 16:00:30.000000000 +0200
@@ -13,6 +13,9 @@
 CC = @CC@
 RM = rm -f
 CFLAGS = @CFLAGS@
+LDFLAGS = @LDFLAGS@
+CPPFLAGS = @CPPFLAGS@
+CXXFLAGS = @CXXFLAGS@
 DEFS = @DEFS@
 INCS = -I.
 TAR = tar
@@ -20,7 +21,7 @@
 all: setserial setserial.cat
 
 setserial: setserial.c
-	$(CC) $(CFLAGS) $(DEFS) $(INCS) setserial.c -o setserial
+	$(CC) $(CFLAGS) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) $(DEFS) $(INCS) setserial.c -o setserial
 
 setserial.cat: setserial.8
 	nroff -man setserial.8 > setserial.cat
@@ -28,7 +29,7 @@
 install: setserial setserial.8
 	$(INSTALL_PROGRAM) setserial $(DESTDIR)/bin
 	$(STRIP) $(DESTDIR)/bin/setserial
-	$(INSTALL_DATA) setserial.8 $(DESTDIR)/usr/man/man8
+	$(INSTALL_DATA) setserial.8 $(DESTDIR)/usr/share/man/man8
 
 clean:
 	$(RM) setserial setserial.o setserial.cat *~
