From: Jonas Genannt <jonas.genannt@capi2name.de>
Subject: setserial pcmcia README patch


diff -Naurp setserial-2.17.orig/pcmcia/README setserial-2.17/pcmcia/README
--- setserial-2.17.orig/pcmcia/README	1970-01-01 00:00:00.000000000 +0000
+++ setserial-2.17/pcmcia/README	2008-09-26 21:59:26.000000000 +0000
@@ -0,0 +1,5 @@
+I do not have pcmcia cards myself, but this file contains typical
+output from some pcmcia commands on systems which do. I use these
+for testing purposes.
+
+Gordon Russell <g.russell@dcs.napier.ac.uk>
