From: Jonas Genannt <jonas.genannt@capi2name.de>
Subject: setserial serial.conf patch


diff -Naurp setserial-2.17.orig/serial.conf setserial-2.17/serial.conf
--- setserial-2.17.orig/serial.conf	2008-09-26 21:59:11.000000000 +0000
+++ setserial-2.17/serial.conf	2008-09-26 21:59:26.000000000 +0000
@@ -1,3 +1,12 @@
+###AUTOSAVE-ONCE###
+###AUTOSAVE###
+#
+# If you want AUTOSAVE switched on, make the AUTOSAVE line above the
+# FIRST line of the file.
+#
+# AUTOSAVE-ONCE, if it is first, will save the port status and then move
+# itself to the second line of the file (thus effectively stopping further
+# AUTOSAVEs).
 #
 # This is a sample serial.conf file.  You should uncomment out and modify
 # the relevant lines as necessary.
