From: Jonas Genannt <jonas.genannt@capi2name.de>
Subject: setserial serial.confg.sparc patch


diff -Naurp setserial-2.17.orig/serial.conf.sparc setserial-2.17/serial.conf.sparc
--- setserial-2.17.orig/serial.conf.sparc	1970-01-01 00:00:00.000000000 +0000
+++ setserial-2.17/serial.conf.sparc	2008-09-26 21:59:26.000000000 +0000
@@ -0,0 +1,79 @@
+#
+# This is a sample serial.conf file.  You should uncomment out and modify
+# the relevant lines as necessary.
+#
+
+# These are the standard COM1 through COM4 devices
+#
+#/dev/ttyS0 uart 16450 port 0x3F8 irq 4
+#/dev/ttyS1 uart 16450 port 0x2F8 irq 3
+#/dev/ttyS2 uart 16450 port 0x3E8 irq 4
+#/dev/ttyS3 uart 16450 port 0x2E8 irq 3
+
+# These are the first set of AST Fourport ports
+#
+#/dev/ttyS4 uart 16450 port 0x1A0 irq 9 fourport
+#/dev/ttyS5 uart 16450 port 0x1A8 irq 9 fourport
+#/dev/ttyS6 uart 16450 port 0x1B0 irq 9 fourport
+#/dev/ttyS7 uart 16450 port 0x1B8 irq 9 fourport
+
+# This enables the new multiport masking feature, which is highly recommened
+# for AST FourPort boards.
+#
+#/dev/ttyS4 set_multiport port1 0x1BF mask1 0xf match1 0xf
+
+# These are the second set of AST Fourport ports
+#
+#/dev/ttyS8 uart 16450 port 0x2A0 irq 5 fourport
+#/dev/ttyS9 uart 16450 port 0x2A8 irq 5 fourport
+#/dev/ttyS10 uart 16450 port 0x2B0 irq 5 fourport
+#/dev/ttyS11 uart 16450 port 0x2B8 irq 5 fourport
+
+# This enables the new multiport masking feature, which is highly recommened
+# for AST FourPort boards.
+#
+#/dev/ttyS8 set_multiport port1 0x2BF mask1 0xf match1 0xf
+
+# These are the 3rd and 4th ports on the Accent Async board.
+#
+#/dev/ttyS12 uart 16450 port 0x330 irq 4
+#/dev/ttyS13 uart 16450 port 0x338 irq 4
+
+# These are two spare devices you can use to customize for 
+# some board which is not supported above....
+#
+#/dev/ttyS14 uart XXXXX port XXXX irq X
+#/dev/ttyS15 uart XXXXX port XXXX irq X
+
+# These are the ports used for either the Usenet Serial II
+# board, or the Boca Board 4, 8, or 16 port boards.
+#
+# Uncomment only the first 4 lines for the Usenet Serial II board,
+# and uncomment the first 4, 8, or all 16 lines for the
+# Boca Board BB-1004, BB-1008, and BB-2016 respectively.
+#
+#/dev/ttyS16 uart 16550A port 0x100 irq 12
+#/dev/ttyS17 uart 16550A port 0x108 irq 12
+#/dev/ttyS18 uart 16550A port 0x110 irq 12
+#/dev/ttyS19 uart 16550A port 0x118 irq 12
+#/dev/ttyS20 uart 16550A port 0x120 irq 12
+#/dev/ttyS21 uart 16550A port 0x128 irq 12
+#/dev/ttyS22 uart 16550A port 0x130 irq 12
+#/dev/ttyS23 uart 16550A port 0x138 irq 12
+#/dev/ttyS24 uart 16550A port 0x140 irq 12
+#/dev/ttyS25 uart 16550A port 0x148 irq 12
+#/dev/ttyS26 uart 16550A port 0x150 irq 12
+#/dev/ttyS27 uart 16550A port 0x158 irq 12
+#/dev/ttyS28 uart 16550A port 0x160 irq 12
+#/dev/ttyS29 uart 16550A port 0x168 irq 12
+#/dev/ttyS30 uart 16550A port 0x170 irq 12
+#/dev/ttyS31 uart 16550A port 0x178 irq 12
+
+# This enables the new multiport masking feature, which is highly recommened
+# for Bocaboard ports.  Uncomment only the first line if you have a 
+# BB-1004 or BB-1008.  Uncomment both lines if you have a BB-2016.
+# These numbers assume the Bocaboard is located at address 0x100.  If you
+# change this, remember to change the port1 and port2 addresses.
+#
+#/dev/ttyS16 set_multiport port1 0x107 mask1 0xff match1 0
+#/dev/ttyS16 set_multiport port2 0x147 mask2 0xff match2 0
