From: Jonas Genannt <jonas.genannt@capi2name.de>
Subject: setserial setserial.8.in patch


diff -Naurp setserial-2.17.orig/setserial.8.in setserial-2.17/setserial.8.in
--- setserial-2.17.orig/setserial.8.in	2008-09-26 21:59:11.000000000 +0000
+++ setserial-2.17/setserial.8.in	2008-09-26 21:59:26.000000000 +0000
@@ -6,6 +6,8 @@
 .\"
 .\" [tytso:19940519.2239EDT]  I did... - Ted Ts'o (tytso@mit.edu)
 .\"
+.\" [g.russell]  Updated for Debian - (gor@debian.org)
+.\"
 .TH SETSERIAL 8 "@RELEASE_MONTH@ @RELEASE_YEAR@" "Setserial @RELEASE_VERSION@
 .SH NAME
 setserial \- get/set Linux serial port information
@@ -29,18 +31,17 @@ on.
 During the normal bootup process, only COM ports 1-4 are initialized,
 using the default I/O ports and IRQ values, as listed below.  In order
 to initialize any additional serial ports, or to change the COM 1-4
-ports to a nonstadard configuration, the
+ports to a nonstandard configuration, the
 .B setserial 
 program should be used.  Typically it is called from an
-.I rc.serial
+.I setserial
 script, which is usually run out of 
-.IR /etc/rc.local .
+.IR /etc/init.d .
 
 The
 .I device
-argument or arguments specifies the serial device which should be configured or
-interrogated.  It will usually have the following form:
-.BR /dev/cua[0-3] .
+argument specifies which device to configure or to interrogate.
+Examples: /dev/ttyS0, /dev/ttyS1, /dev/ttyS2, /dev/ttyS3, etc.
 
 If no parameters are specified,
 .B setserial
@@ -108,7 +109,7 @@ to the automatic saving of serial flags 
 .SH PARAMETERS
 The following parameters can be assigned to a serial port.
 
-All argument values are assumed to be in decimal unless preceeded by "0x".
+All argument values are assumed to be in decimal unless preceded by "0x".
 
 .TP
 .BR port " port_number"
@@ -177,7 +178,7 @@ try to determine the IRQ.
 .B skip_test
 During autoconfiguration, skip the UART test.  Some internal modems do
 not have National Semiconductor compatible UART's, but have cheap
-imitations instead.  Some of these cheasy imitations UART's do not
+imitations instead.  Some of these cheesy imitations UART's do not
 fully support the loopback detection mode, which is used by the kernel
 to make sure there really is a UART at a particular address before
 attempting to configure it.  So for certain internal modems you will
@@ -224,7 +225,7 @@ Use 38.4kb when the application requests
 This parameter may be specified by a non-privileged user.
 .TP
 .BR divisor " divisor"
-This option sets the custom divison.  This divisor will be used then the
+This option sets the custom divisor.  This divisor will be used when the
 .B spd_cust
 option is selected and the serial port is set to 38.4kb by the
 application.
@@ -282,17 +283,17 @@ closed it.
 Do not lock out callout port accesses across different process groups.
 .TP
 .B hup_notify
-Notify a process blocked on opening a dial in line when a process has
+Notify a process blocked on opening a dialin line when a process has
 finished using a callout line (either by closing it or by the serial
 line being hung up) by returning EAGAIN to the open.  
 
 The application of this parameter is for getty's which are blocked on
-a serial port's dial in line.  This allows the getty to reset the
+a serial port's dialin line.  This allows the getty to reset the
 modem (which may have had its configuration modified by the
 application using the callout device) before blocking on the open again.
 .TP
 .B ^hup_notify
-Do not notify a process blocked on opening a dial in line when the
+Do not notify a process blocked on opening a dialin line when the
 callout device is hung up.
 .TP
 .B split_termios
@@ -315,7 +316,7 @@ the hangup is to occur.
 .B low_latency
 Minimize the receive latency of the serial device at the cost of
 greater CPU utilization.  (Normally there is an average of 5-10ms
-latency before characters are handed off to the line discpline to
+latency before characters are handed off to the line discipline to
 minimize overhead.)  This is off by default, but certain real-time
 applications may find this useful.
 .TP
@@ -338,10 +339,10 @@ The "standard MS-DOS" port associations 
 
 .nf
 .RS
-/dev/ttys0 (COM1), port 0x3f8, irq 4
-/dev/ttys1 (COM2), port 0x2f8, irq 3
-/dev/ttys2 (COM3), port 0x3e8, irq 4
-/dev/ttys3 (COM4), port 0x2e8, irq 3
+/dev/ttyS0 (COM1), port 0x3f8, irq 4
+/dev/ttyS1 (COM2), port 0x2f8, irq 3
+/dev/ttyS2 (COM3), port 0x3e8, irq 4
+/dev/ttyS3 (COM4), port 0x2e8, irq 3
 .RE
 .fi
 
@@ -404,7 +405,7 @@ IRQ  8      Real-time clock
 IRQ  9      Redirected to IRQ2
 IRQ 10      Reserved
 IRQ 11      Reserved
-IRQ 12      Reserved (Auxillary device in PS/2)
+IRQ 12      Reserved (Auxiliary device in PS/2)
 IRQ 13      Math coprocessor
 IRQ 14      Hard disk controller
 IRQ 15      Reserved
@@ -504,8 +505,8 @@ and a value too low will cause unnecessa
 CAUTION: Configuring a serial port to use an incorrect I/O port 
 can lock up your machine.
 .SH FILES
-.BR /etc/rc.local
-.BR /etc/rc.serial
+.BR /etc/serial.conf
+.BR /etc/init.d/setserial
 .SH "SEE ALSO"
 .BR tty (4),
 .BR ttys (4),
@@ -518,3 +519,6 @@ The original version of setserial was wr
 This version has since been rewritten from scratch by Theodore Ts'o
 (tytso@mit.edu) on 1/1/93.  Any bugs or problems are solely his
 responsibility.
+
+Debian related problems with this system should be sent to
+Gordon Russell (gor@debian.org).
