From: Jonas Genannt <jonas.genannt@capi2name.de>
Subject: setserial version.h patch


diff -Naurp setserial-2.17.orig/version.h setserial-2.17/version.h
--- setserial-2.17.orig/version.h       2008-09-26 21:59:11.000000000 +0000
+++ setserial-2.17/version.h    2008-09-26 21:59:26.000000000 +0000
@@ -2,7 +2,7 @@
  * version.h --- Defines the version number of setserial
  *
  * Copyright 1995, 1996, 1997, 1998 by Theodore Ts'o.  This file may be
- * redistributed under the GNU Public License.
+ * redistributed under The GNU General Public License.
  */

 #define SETSERIAL_VERSION "2.17"
