From 1d36d75403f9bd4c5009ecb502394536ce3a159f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Denis-Courmont?= <remi@remlab.net>
Date: Sat, 16 Jun 2018 21:31:45 +0300
Subject: [PATCH 1/3] configure: fix linking on RISC-V ISA

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 8fa2a87595..079a00df88 100644
--- a/configure.ac
+++ b/configure.ac
@@ -113,6 +113,7 @@ case "${host_os}" in
     ;;
   linux*)
     SYS=linux
+    test "${host_cpu}" = "riscv64" && CFLAGS="${CFLAGS} -pthread"
     ;;
   bsdi*)
     SYS=bsdi
-- 
2.46.0

