From 7e16ee2d21030868d685450b8de4cf3febda3708 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Fri, 31 Jul 2026 04:01:16 +0000
Subject: [PATCH] Makefile.in: do not run systemctl daemon-reload on `make
 install'

---
 Makefile.in | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 3bff462..c9e3340 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -152,7 +152,6 @@ ifneq ($(SYSTEMD_DIR), no)
 	if test ! -d $(DESTDIR)$(SYSTEMD_DIR); then $(INSTALL) -m 755 -d $(DESTDIR)$(SYSTEMD_DIR) ; fi
 	$(INSTALL) -m 644 script/fcron.init.systemd $(DESTDIR)$(SYSTEMD_DIR)/fcron.service
 	$(INSTALL) -m 644 script/fcron.init.systemd.reboot $(DESTDIR)$(SYSTEMD_DIR)/fcronreboot.service
-	systemctl daemon-reload
 endif
 
 install-staged: all
-- 
2.55.0

