From 61078461a66bd96f5d3f9975303520f627f959bb Mon Sep 17 00:00:00 2001
From: Cyan <cyan@cyano.uk>
Date: Thu, 20 Jun 2024 14:05:22 +0800
Subject: [PATCH 1/2] fix: set effective UID after setting effective GID to
 prevent EPERM

---
 fcrontab.c |  6 +++---
 subs.c     | 12 ++++++------
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/fcrontab.c b/fcrontab.c
index 10dc5b5..fe7ba02 100644
--- a/fcrontab.c
+++ b/fcrontab.c
@@ -175,10 +175,10 @@ xexit(int exit_val)
 
 #ifdef HAVE_LIBPAM
     /* we need those rights for pam to close properly */
-    if (geteuid() != fcrontab_uid && seteuid(fcrontab_uid) != 0)
-        die_e("could not change euid to %d", fcrontab_uid);
     if (getegid() != fcrontab_gid && setegid(fcrontab_gid) != 0)
         die_e("could not change egid to %d", fcrontab_gid);
+    if (geteuid() != fcrontab_uid && seteuid(fcrontab_uid) != 0)
+        die_e("could not change euid to %d", fcrontab_uid);
     pam_setcred(pamh, PAM_DELETE_CRED | PAM_SILENT);
     pam_end(pamh, pam_close_session(pamh, PAM_SILENT));
 #endif
@@ -1038,8 +1038,8 @@ main(int argc, char **argv)
 #ifdef USE_SETE_ID
     /* drop any privilege we may have: we will only get them back
      * temporarily every time we need it. */
-    seteuid_safe(useruid);
     setegid_safe(usergid);
+    seteuid_safe(useruid);
 #endif
 
 #ifdef HAVE_LIBPAM
diff --git a/subs.c b/subs.c
index b5a29be..d96d9ef 100644
--- a/subs.c
+++ b/subs.c
@@ -109,8 +109,8 @@ open_as_user(const char *pathname, uid_t openuid, gid_t opengid, int flags, ...)
         va_end(ap);
     }
 
-    seteuid_safe(openuid);
     setegid_safe(opengid);
+    seteuid_safe(openuid);
 
     if (flags & O_CREAT) {
         fd = open(pathname, flags, mode);
@@ -121,8 +121,8 @@ open_as_user(const char *pathname, uid_t openuid, gid_t opengid, int flags, ...)
     saved_errno = errno;
 
     /* change the effective uid/gid back to original values */
-    seteuid_safe(orig_euid);
     setegid_safe(orig_egid);
+    seteuid_safe(orig_euid);
 
     /* if open() didn't fail make sure we opened a 'normal' file */
     if (fd >= 0) {
@@ -271,15 +271,15 @@ remove_as_user(const char *pathname, uid_t removeuid, gid_t removegid)
     uid_t orig_euid = geteuid();
     gid_t orig_egid = getegid();
 
-    seteuid_safe(removeuid);
     setegid_safe(removegid);
+    seteuid_safe(removeuid);
 #endif                          /* def USE_SETE_ID */
 
     rval = remove(pathname);
 
 #ifdef USE_SETE_ID
-    seteuid_safe(orig_euid);
     setegid_safe(orig_egid);
+    seteuid_safe(orig_euid);
 #endif                          /* def USE_SETE_ID */
 
     return rval;
@@ -295,15 +295,15 @@ rename_as_user(const char *oldpath, const char *newpath, uid_t renameuid,
     uid_t orig_euid = geteuid();
     gid_t orig_egid = getegid();
 
-    seteuid_safe(renameuid);
     setegid_safe(renamegid);
+    seteuid_safe(renameuid);
 #endif                          /* def USE_SETE_ID */
 
     rval = rename(oldpath, newpath);
 
 #ifdef USE_SETE_ID
-    seteuid_safe(orig_euid);
     setegid_safe(orig_egid);
+    seteuid_safe(orig_euid);
 #endif                          /* def USE_SETE_ID */
 
     return rval;
-- 
2.45.2

