From 2aecb0c49b4bd5ea05cc71ab817236475f2263cb Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Thu, 20 Jun 2024 15:10:41 +0800
Subject: [PATCH 2/2] fix(Makefile.in): do not strip during install time

---
 Makefile.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 251445f..cbc942b 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -137,7 +137,7 @@ initscripts:
 documentation:
 	$(MAKE) -C doc doc-if-none
 
-install: install-staged strip perms 
+install: install-staged perms
 ifeq ($(BOOTINSTALL), 1)
 	$(SRCDIR)/script/boot-install "$(INSTALL) -o $(ROOTNAME)" $(DESTSBIN) $(DEBUG) $(FCRONTABS) $(ANSWERALL) $(SRCDIR)
 endif
@@ -174,7 +174,7 @@ endif
 	$(MAKE) -C doc install-staged
 
 
-perms: install-staged strip
+perms: install-staged
 # Note : we don't use "chown user:group file" because some systems use ":"
 #        and others "." as separator.
 	chown $(ROOTNAME) $(DESTDIR)$(DESTSBIN) 
-- 
2.45.2

