From 306e4df23a5f84cff133eb87e0a5f5171424a6a4 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 17 Mar 2025 15:35:14 +0800
Subject: [PATCH 1/2] Build whisker as a shared library

---
 thrift/compiler/CMakeLists.txt | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/thrift/compiler/CMakeLists.txt b/thrift/compiler/CMakeLists.txt
index ec0ee60543..d09582f105 100644
--- a/thrift/compiler/CMakeLists.txt
+++ b/thrift/compiler/CMakeLists.txt
@@ -166,7 +166,6 @@ target_link_libraries(
 
 add_library(
   whisker
-  STATIC
 
   whisker/ast.cc
   whisker/dsl.cc
@@ -183,9 +182,9 @@ add_library(
 )
 target_link_libraries(
   whisker
-  fmt::fmt
+  PUBLIC fmt::fmt
   Folly::folly
-  compiler_base
+  PRIVATE compiler_base
 )
 
 add_subdirectory(generate)
@@ -237,8 +236,6 @@ endif ()
 install(
   TARGETS thrift1
     compiler_ast
-    compiler_base
-    compiler_lib
     compiler
     whisker
     common
-- 
2.52.0

