From 2c022620a9dde49a0553f5e39a1b3c293f3ee273 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Mon, 17 Mar 2025 15:35:14 +0800
Subject: [PATCH] Build whisker as a shared library

---
 thrift/compiler/CMakeLists.txt | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/thrift/compiler/CMakeLists.txt b/thrift/compiler/CMakeLists.txt
index 0646111bf3..3180d5985a 100644
--- a/thrift/compiler/CMakeLists.txt
+++ b/thrift/compiler/CMakeLists.txt
@@ -156,7 +156,6 @@ target_link_libraries(
 
 add_library(
   whisker
-  STATIC
 
   whisker/detail/string.cc
   whisker/ast.cc
@@ -174,8 +173,8 @@ add_library(
 )
 target_link_libraries(
   whisker
-  fmt::fmt
-  compiler_base
+  PUBLIC fmt::fmt
+  PRIVATE compiler_base
 )
 
 add_subdirectory(generate)
@@ -228,8 +227,6 @@ endif ()
 install(
   TARGETS thrift1
     compiler_ast
-    compiler_base
-    compiler_lib
     compiler
     whisker
   EXPORT fbthrift-exports
-- 
2.49.0

