From f70618c985155318df3db3f388504902f314345a Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Tue, 24 Mar 2026 11:44:40 +0800
Subject: [PATCH 2/2] Satisfy inline atomic alignment requirements for clang

---
 thrift/lib/cpp2/server/ParallelConcurrencyController.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/thrift/lib/cpp2/server/ParallelConcurrencyController.h b/thrift/lib/cpp2/server/ParallelConcurrencyController.h
index 24c5516197..164652e4fb 100644
--- a/thrift/lib/cpp2/server/ParallelConcurrencyController.h
+++ b/thrift/lib/cpp2/server/ParallelConcurrencyController.h
@@ -90,7 +90,7 @@ class ParallelConcurrencyControllerBase : public ConcurrencyControllerBase,
  protected:
   const RequestExecutionMode requestExecutionMode_;
 
-  struct Counters {
+  struct __attribute__((aligned(8))) Counters {
     constexpr Counters() noexcept = default;
     // Number of requests that are being executed
     // by the executor
-- 
2.52.0

