From b912ff986b49197354fd0d791a787f1f17fb5043 Mon Sep 17 00:00:00 2001
From: Henry Chen <henry.chen@oss.cipunited.com>
Date: Tue, 2 Sep 2025 14:42:43 +0800
Subject: [PATCH 2/2] Add <algorithm> to DecoratorDataRuntime.h

---
 thrift/lib/cpp2/server/DecoratorDataRuntime.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/thrift/lib/cpp2/server/DecoratorDataRuntime.h b/thrift/lib/cpp2/server/DecoratorDataRuntime.h
index e988b60c0c..3021253138 100644
--- a/thrift/lib/cpp2/server/DecoratorDataRuntime.h
+++ b/thrift/lib/cpp2/server/DecoratorDataRuntime.h
@@ -16,6 +16,7 @@
 
 #pragma once
 
+#include <algorithm>
 #include <vector>
 #include <thrift/lib/cpp2/server/DecoratorData.h>
 #include <thrift/lib/cpp2/server/DecoratorDataKey.h>
-- 
2.51.0

