From 418c44e61fcda33e809076d1817acc2f2362d409 Mon Sep 17 00:00:00 2001
From: Qiyu Yan <yanqiyu@fedoraproject.org>
Date: Sun, 12 Mar 2023 12:48:39 +0800
Subject: [PATCH] remove -std=c++11 from gui/CMakeLists.txt

---
 gui/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gui/CMakeLists.txt b/gui/CMakeLists.txt
index 38b50d4..2327132 100644
--- a/gui/CMakeLists.txt
+++ b/gui/CMakeLists.txt
@@ -17,7 +17,7 @@ qt5_wrap_ui(KKC_CONFIG_SRCS dictwidget.ui adddictdialog.ui shortcutwidget.ui add
 add_library(fcitx5-kkc-config
   MODULE ${KKC_CONFIG_SRCS})
 
-add_definitions(-DQT_NO_KEYWORDS -std=c++11)
+add_definitions(-DQT_NO_KEYWORDS)
 
 target_include_directories(fcitx5-kkc-config PRIVATE ${PROJECT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR})
 
-- 
2.39.2
