From 9ae14bdc126d1408e3390c8c6eeae723177eac69 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sun, 28 Jul 2019 00:59:22 +0800
Subject: [PATCH] Move all executables to usr bin

---
 Makefile   | 2 +-
 fortune-zh | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index fb7d23b..7668030 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 DESTDIR=
 FORTUNES=/usr/share/games/fortunes
-GAMES=/usr/games
+GAMES=/usr/bin
 
 TEXTS = tang300 song100 chinese
 DATA  = tang300.dat song100.dat chinese.dat
diff --git a/fortune-zh b/fortune-zh
index 2383f65..725ad4f 100755
--- a/fortune-zh
+++ b/fortune-zh
@@ -2,7 +2,7 @@
 # fortune-zh
 set -e
 
-FORTUNE="/usr/games/fortune"
+FORTUNE="/usr/bin/fortune"
 [ -x $FORTUNE ] || ( echo "E: Please install package 'fortune-mod'."; false )
 
 # The old version (1.*) of fortune-zh contains only tang300 and song100.
-- 
2.22.0

