From a2bc8803b35d4df0731a4945804b0b1d645137d9 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sun, 28 Jul 2019 01:05:58 +0800
Subject: [PATCH] Move data file

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7668030..03cf95b 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 DESTDIR=
-FORTUNES=/usr/share/games/fortunes
+FORTUNES=/usr/share/fortune
 GAMES=/usr/bin
 
 TEXTS = tang300 song100 chinese
-- 
2.22.0

