diff -Naur fortune-zh/Makefile fortune-zh.compat/Makefile
--- fortune-zh/Makefile	2018-05-26 16:37:53.000000000 +0000
+++ fortune-zh.compat/Makefile	2018-07-13 00:17:48.284661360 +0000
@@ -15,7 +15,8 @@
 	strfile $< $<.dat
 chinese.dat: 
 	touch chinese
-	find chinese.d -type f,l -name '*.dat' -print0 | LC_ALL=C sort -z | xargs -0r cat | cat >> chinese
+	find chinese.d -type f -name '*.dat' -print0 | LC_ALL=C sort -z | xargs -0r cat | cat >> chinese
+	find chinese.d -type l -name '*.dat' -print0 | LC_ALL=C sort -z | xargs -0r cat | cat >> chinese
 	awk '{if ($$0 ~ /--/) {print "[33m" $$0 "[m"} else {print}}' chinese > chinese_colorize
 	sed -i -e 's/《/[32m《/g' -e 's/》/》[m/g' chinese_colorize
 	mv chinese_colorize chinese
