From 7516e0fb71154d4c383cdecfdf3f789dee678d09 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Sun, 28 Jul 2019 01:40:50 +0800
Subject: [PATCH] Makefile compat with findutils 4.6.0

---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 03cf95b..0d4579f 100644
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,8 @@ DATA  = tang300.dat song100.dat chinese.dat
 chinese.dat: 
 	python3 ./cookie2dat.py
 	touch chinese
-	find . -type f,l -name '*.dat' -print0 | LC_ALL=C sort -z | xargs -0r cat | cat >> chinese
+	find . -type f -name '*.dat' -print0 | LC_ALL=C sort -z | xargs -0r cat | cat >> chinese
+	find . -type l -name '*.dat' -print0 | LC_ALL=C sort -z | xargs -0r cat | cat >> chinese
 	awk '{if ($$0 ~ /--/) {print "[33m" $$0 "[m"} else {print}}' chinese > chinese_colorize
 	sed -i -e 's/《/[32m《/g' -e 's/》/》[m/g' chinese_colorize
 	mv chinese_colorize chinese
-- 
2.22.0

