diff --git a/configure.ac b/configure.ac
index a21d750..999eca3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -69,9 +69,6 @@ AM_MISSING_PROG([CASTXML], [castxml])
 LINGUAS=""
 AM_GNU_GETTEXT([external], [need-ngettext])
 AM_GNU_GETTEXT_VERSION([0.19.8])
-m4_ifdef([AM_GNU_GETTEXT_REQUIRE_VERSION],[
-AM_GNU_GETTEXT_REQUIRE_VERSION([0.19.8])
-])
 
 if test "$enable_static" = "yes"; then
 	AC_MSG_ERROR([p11-kit cannot be used as a static library])
