From b16b442019724b9cc27fedbce0916e2a65011eb1 Mon Sep 17 00:00:00 2001
From: Alexander Gorishnyak <kefir500@gmail.com>
Date: Sun, 5 Jan 2020 13:54:24 +0600
Subject: [PATCH] Fix unused parameter warnings

---
 libsecret.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/libsecret.cpp b/libsecret.cpp
index 70d1ddf..4fcf19f 100644
--- a/libsecret.cpp
+++ b/libsecret.cpp
@@ -244,6 +244,9 @@ bool LibSecretKeyring::findPassword(const QString &user, const QString &server,
                                NULL);
     return true;
 #else
+    Q_UNUSED(user)
+    Q_UNUSED(server)
+    Q_UNUSED(self)
     return false;
 #endif
 }
@@ -281,6 +284,12 @@ bool LibSecretKeyring::writePassword(const QString &display_name,
                               NULL);
     return true;
 #else
+    Q_UNUSED(display_name)
+    Q_UNUSED(user)
+    Q_UNUSED(server)
+    Q_UNUSED(mode)
+    Q_UNUSED(password)
+    Q_UNUSED(self)
     return false;
 #endif
 }
@@ -300,6 +309,9 @@ bool LibSecretKeyring::deletePassword(const QString &key, const QString &service
                               NULL);
     return true;
 #else
+    Q_UNUSED(key)
+    Q_UNUSED(service)
+    Q_UNUSED(self)
     return false;
 #endif
 }
-- 
2.27.0

