--- a/Eigen/src/Core/arch/AltiVec/PacketMath.h	2011-12-06 07:18:54.000000000 -0600
+++ b/Eigen/src/Core/arch/AltiVec/PacketMath.h	2021-02-13 02:49:47.061149121 -0600
@@ -191,7 +191,7 @@
 template<> inline v4f  ei_pset1(const float&  from)
 {
   // Taken from http://developer.apple.com/hardwaredrivers/ve/alignment.html
-  float __attribute__(aligned(16)) af[4];
+  float af[4] __attribute__((aligned(16)));
   af[0] = from;
   v4f vc = vec_ld(0, af);
   vc = vec_splat(vc, 0);
@@ -200,7 +200,7 @@
 
 template<> inline v4i    ei_pset1(const int&    from)
 {
-  int __attribute__(aligned(16)) ai[4];
+  int ai[4] __attribute__((aligned(16)));
   ai[0] = from;
   v4i vc = vec_ld(0, ai);
   vc = vec_splat(vc, 0);
@@ -248,14 +248,14 @@
 
 template<> inline float  ei_pfirst(const v4f&  a)
 {
-  float __attribute__(aligned(16)) af[4];
+  float af[4] __attribute__((aligned(16)));
   vec_st(a, 0, af);
   return af[0];
 }
 
 template<> inline int    ei_pfirst(const v4i&  a)
 {
-  int __attribute__(aligned(16)) ai[4];
+  int ai[4] __attribute__((aligned(16)));
   vec_st(a, 0, ai);
   return ai[0];
 }
