From 22ef5b66e8476aaa8a70e684063aba26843a5ac8 Mon Sep 17 00:00:00 2001
From: tsingkong <tsingkong@163.com>
Date: Wed, 5 Mar 2025 19:52:17 +0800
Subject: [PATCH] add support for loongarch

---
 backward.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/backward.hpp b/backward.hpp
index 2342b18..859d604 100644
--- a/backward.hpp
+++ b/backward.hpp
@@ -4280,6 +4280,8 @@ class SignalHandling {
     error_addr = reinterpret_cast<void *>(uctx->uc_mcontext->__ss.__rip);
 #elif defined(__APPLE__)
     error_addr = reinterpret_cast<void *>(uctx->uc_mcontext->__ss.__eip);
+#elif defined(__loongarch__)
+    error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.__pc);
 #else
 #warning ":/ sorry, ain't know no nothing none not of your architecture!"
 #endif

