From e054e61942048474ce380930a739ce4c9f54605d Mon Sep 17 00:00:00 2001
From: 2876225417 <2876225417@qq.com>
Date: Thu, 19 Jun 2025 15:59:11 +0800
Subject: [PATCH 2/3] UPSTREAM: Fix(tests): Add missing <cstdint> for modern
 compilers

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 tests/featuremap/featuremaptest.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/featuremap/featuremaptest.cpp b/tests/featuremap/featuremaptest.cpp
index dd0804d8..b46b8b49 100644
--- a/tests/featuremap/featuremaptest.cpp
+++ b/tests/featuremap/featuremaptest.cpp
@@ -29,6 +29,10 @@
 #include <string>
 #include <vector>
 
+#include <cstdint>
+
+
+
 #include <graphite2/Font.h>
 
 #include "inc/Endian.h"
-- 
2.52.0

