diff --git a/glib/poppler-private.h b/glib/poppler-private.h
index 3e6b4b17..9f232883 100644
--- a/glib/poppler-private.h
+++ b/glib/poppler-private.h
@@ -147,19 +147,5 @@ gboolean _poppler_convert_pdf_date_to_gtime (const GooString *date,
  * A convenience macro for boxed type implementations, which defines a
  * type_name_get_type() function registering the boxed type.
  */
-#define POPPLER_DEFINE_BOXED_TYPE(TypeName, type_name, copy_func, free_func)          \
-GType                                                                                 \
-type_name##_get_type (void)                                                           \
-{                                                                                     \
-        static volatile gsize g_define_type_id__volatile = 0;                         \
-	if (g_once_init_enter (&g_define_type_id__volatile)) {                        \
-	        GType g_define_type_id =                                              \
-		    g_boxed_type_register_static (g_intern_static_string (#TypeName), \
-		                                  (GBoxedCopyFunc) copy_func,         \
-						  (GBoxedFreeFunc) free_func);        \
-		g_once_init_leave (&g_define_type_id__volatile, g_define_type_id);    \
-	}                                                                             \
-	return g_define_type_id__volatile;                                            \
-}
-
+#define POPPLER_DEFINE_BOXED_TYPE G_DEFINE_BOXED_TYPE
 #endif
