Bug: https://bugs.freedesktop.org/show_bug.cgi?id=22363

Index: librarian/rarian-main.c
===================================================================
--- a/librarian/rarian-main.c	(revision 62)
+++ b/librarian/rarian-main.c	(working copy)
@@ -294,10 +294,12 @@
 	  char *suffix = NULL;
 	  
 	  suffix = strrchr (full_name, '.');
-	  if (!strcmp (suffix, ".document")) {
-	    process_file (full_name);
-	  } else if (!strcmp (suffix, ".section")) {
-	    process_section (full_name);
+	  if (suffix) {
+	    if (!strcmp (suffix, ".document")) {
+	      process_file (full_name);
+	    } else if (!strcmp (suffix, ".section")) {
+	      process_section (full_name);
+	    }
 	  }
 	} else if (S_ISDIR(buf.st_mode) && strcmp (dp->d_name, ".") &&
 		   strcmp (dp->d_name, "..") &&
