From ce759a6b4f4f4dc6f866b7e5ba89702532467704 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Tue, 18 May 2021 01:26:13 -0700
Subject: [PATCH] bitwarden fix webpack bundling

---
 package.json      | 2 ++
 webpack.config.js | 3 ++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/package.json b/package.json
index 8cc26c2..60c4448 100644
--- a/package.json
+++ b/package.json
@@ -76,6 +76,7 @@
     "dependencies": {
         "big-integer": "1.6.48",
         "browser-hrtime": "^1.1.8",
+        "browser-process-hrtime": "^1.0.0",
         "chalk": "2.4.1",
         "commander": "7.0.0",
         "form-data": "2.3.2",
@@ -89,6 +90,7 @@
         "open": "7.1.0",
         "papaparse": "5.2.0",
         "tldjs": "2.3.1",
+        "tslib": "^2.2.0",
         "zxcvbn": "4.4.2"
     }
 }
diff --git a/webpack.config.js b/webpack.config.js
index 19a3dfb..c73b1f3 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -37,6 +37,7 @@ const plugins = [
         raw: true
     }),
     new webpack.IgnorePlugin(/^encoding$/, /node-fetch/),
+    new webpack.IgnorePlugin(/canvas/),
 ];
 
 const config = {
@@ -69,7 +70,7 @@ const config = {
     },
     module: { rules: moduleRules },
     plugins: plugins,
-    externals: [nodeExternals()],
+    externals: [],
 };
 
 module.exports = config;
-- 
2.30.2

