From ce759a6b4f4f4dc6f866b7e5ba89702532467704 Mon Sep 17 00:00:00 2001
From: eatradish <sakiiily@aosc.io>
Date: Tue, 18 May 2021 01:26:13 -0700
Subject: [PATCH] bitwarden fix webpack bundling


diff --git a/webpack.config.js b/webpack.config.js
index 19a3dfb..c73b1f3 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -37,6 +37,7 @@ const plugins = [
         raw: true
     }),
     new webpack.IgnorePlugin(/^encoding$/, /node-fetch/),
+    new webpack.IgnorePlugin(/canvas/),
 ];
 
 const config = {
@@ -69,7 +70,7 @@ const config = {
     },
     module: { rules: moduleRules },
     plugins: plugins,
-    externals: [nodeExternals()],
+    externals: [],
 };
 
 module.exports = config;
--- a/package.json	2021-07-01 15:03:16.450583172 +0800
+++ b/package.json	2021-07-01 15:03:50.362005898 +0800
@@ -67,12 +67,14 @@
     "dependencies": {
         "@bitwarden/jslib-common": "file:jslib/common",
         "@bitwarden/jslib-node": "file:jslib/node",
+        "browser-process-hrtime": "^1.0.0",
         "commander": "7.2.0",
         "form-data": "4.0.0",
         "inquirer": "8.0.0",
         "jsdom": "^16.5.3",
         "node-forge": "0.10.0",
-        "open": "^8.0.8"
+        "open": "^8.0.8",
+        "tslib": "^2.3.0"
     },
     "engines": {
         "node": "~14",
