diff -Naur libdispatch-objc2-orig/libdispatch/src/CMakeLists.txt libdispatch-objc2/libdispatch/src/CMakeLists.txt
--- libdispatch/src/CMakeLists.txt	2020-12-13 00:22:54.636333615 -0800
+++ libdispatch/src/CMakeLists.txt	2020-12-13 00:20:43.671004777 -0800
@@ -110,7 +110,7 @@
     ${CMAKE_THREAD_LIBS_INIT}
     ${PTHREAD_WORKQUEUE_LIBRARIES}
     ${KQUEUE_LIBRARIES}
-    ${CBLOCKS_LIBRARIES}
+    -lobjc2
     ${CLOCK_GETTIME_LIBRARIES}
   )
 endforeach ()
diff -Naur libdispatch-objc2-orig/libdispatch/src/internal.h libdispatch-objc2/libdispatch/src/internal.h
--- libdispatch/src/internal.h	2020-12-13 00:22:54.637333633 -0800
+++ libdispatch/src/internal.h	2020-12-13 00:18:15.301312718 -0800
@@ -156,7 +156,7 @@
 #include <sys/event.h>
 #include <sys/mount.h>
 #include <sys/stat.h>
-#include <sys/sysctl.h>
+//#include <sys/sysctl.h>
 #include <sys/socket.h>
 #include <sys/time.h>
 #include <netinet/in.h>
