diff -Naur libva-vdpau-driver-0.7.4-orig/src/utils_glx.c libva-vdpau-driver-0.7.4/src/utils_glx.c
--- libva-vdpau-driver-0.7.4-orig/src/utils_glx.c	2012-10-05 23:02:58.000000000 +0800
+++ libva-vdpau-driver-0.7.4/src/utils_glx.c	2013-08-12 16:39:44.918145756 +0800
@@ -752,10 +752,10 @@
             get_proc_address("glActiveTextureARB");
         if (!gl_vtable->gl_active_texture)
             return NULL;
-        gl_vtable->gl_multi_tex_coord_2f = (PFNGLMULTITEXCOORD2FPROC)
+/*        gl_vtable->gl_multi_tex_coord_2f = (PFNGLMULTITEXCOORD2FPROC)
             get_proc_address("glMultiTexCoord2fARB");
         if (!gl_vtable->gl_multi_tex_coord_2f)
-            return NULL;
+            return NULL;*/
         gl_vtable->has_multitexture = 1;
     }
 
diff -Naur libva-vdpau-driver-0.7.4-orig/src/utils_glx.h libva-vdpau-driver-0.7.4/src/utils_glx.h
--- libva-vdpau-driver-0.7.4-orig/src/utils_glx.h	2012-10-05 23:02:58.000000000 +0800
+++ libva-vdpau-driver-0.7.4/src/utils_glx.h	2013-08-12 16:39:44.919145743 +0800
@@ -160,7 +160,7 @@
     PFNGLGETPROGRAMIVARBPROC              gl_get_program_iv;
     PFNGLPROGRAMLOCALPARAMETER4FVARBPROC  gl_program_local_parameter_4fv;
     PFNGLACTIVETEXTUREPROC                gl_active_texture;
-    PFNGLMULTITEXCOORD2FPROC              gl_multi_tex_coord_2f;
+//    PFNGLMULTITEXCOORD2FPROC              gl_multi_tex_coord_2f;
     PFNGLVDPAUINITNVPROC                  gl_vdpau_init;
     PFNGLVDPAUFININVPROC                  gl_vdpau_fini;
     PFNGLVDPAUREGISTERVIDEOSURFACENVPROC  gl_vdpau_register_video_surface;
diff -Naur libva-vdpau-driver-0.7.4-orig/src/vdpau_dump.c libva-vdpau-driver-0.7.4/src/vdpau_dump.c
--- libva-vdpau-driver-0.7.4-orig/src/vdpau_dump.c	2012-10-05 23:02:58.000000000 +0800
+++ libva-vdpau-driver-0.7.4/src/vdpau_dump.c	2013-08-12 16:39:44.918145756 +0800
@@ -59,8 +59,8 @@
         _(VAEncSequenceParameterBufferType);
         _(VAEncPictureParameterBufferType);
         _(VAEncSliceParameterBufferType);
-        _(VAEncH264VUIBufferType);
-        _(VAEncH264SEIBufferType);
+//        _(VAEncH264VUIBufferType);
+//        _(VAEncH264SEIBufferType);
 #endif
 #if VA_CHECK_VERSION(0,31,1)
         _(VAQMatrixBufferType);
