From d04eb0dd85f1d91bc3f1f9a19810389d092b26a1 Mon Sep 17 00:00:00 2001
From: Yuri Chornoivan <yurchor@ukr.net>
Date: Sat, 26 Oct 2019 22:53:08 +0300
Subject: [PATCH 12/14] Make action buttons translatable

---
 src/declarative/pkupdates.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/declarative/pkupdates.cpp b/src/declarative/pkupdates.cpp
index 9eac70f..401c887 100644
--- a/src/declarative/pkupdates.cpp
+++ b/src/declarative/pkupdates.cpp
@@ -497,11 +497,11 @@ void PkUpdates::onRequireRestart(PackageKit::Transaction::Restart type, const QS
         notification->setComponentName(s_componentName);
         notification->setIconName(s_pkUpdatesIconName);
         if (type == PackageKit::Transaction::RestartSystem) {
-            notification->setActions(QStringList{QLatin1String("Restart")});
+            notification->setActions(QStringList{i18nc("@action:button", "Restart")});
             notification->setTitle(i18n("Restart is required"));
             notification->setText(i18n("The computer will have to be restarted after the update for the changes to take effect."));
         } else {
-            notification->setActions(QStringList{QLatin1String("Logout")});
+            notification->setActions(QStringList{i18nc("@action:button", "Logout")});
             notification->setTitle(i18n("Session restart is required"));
             notification->setText(i18n("You will need to log out and back in after the update for the changes to take effect."));
         }
-- 
2.43.0

