From 37ed2595c6cca3b40617e68974166b818ec83b8f Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Tue, 30 Jan 2024 05:31:58 -0800
Subject: [PATCH 14/14] [Fedora] drop persistent notifications

---
 src/declarative/pkupdates.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/declarative/pkupdates.cpp b/src/declarative/pkupdates.cpp
index 1c958db..088783d 100644
--- a/src/declarative/pkupdates.cpp
+++ b/src/declarative/pkupdates.cpp
@@ -422,7 +422,7 @@ void PkUpdates::onFinished(PackageKit::Transaction::Exit status, uint runtime)
                 m_lastNotification = KNotification::event(s_eventIdUpdatesAvailable,
                                      QString(),
                                      i18np("You have 1 new update", "You have %1 new updates", upCount),
-                                     s_pkUpdatesIconName, nullptr, KNotification::Persistent,
+                                     s_pkUpdatesIconName, nullptr, KNotification::CloseOnTimeout,
                                      s_componentName);
                 connect(m_lastNotification, &KNotification::closed, this, [this] {
                     qCDebug(PLASMA_PK_UPDATES) << "Old notification closed";
-- 
2.43.0

